# Comparing `tmp/wgd-2.0.34.tar.gz` & `tmp/wgd-2.0.35.tar.gz`

## filetype from file(1)

```diff
@@ -1 +1 @@
-gzip compressed data, was "dist/wgd-2.0.34.tar", last modified: Sun Apr 28 17:05:05 2024, max compression
+gzip compressed data, was "dist/wgd-2.0.35.tar", last modified: Wed May  1 10:50:20 2024, max compression
```

## Comparing `wgd-2.0.34.tar` & `wgd-2.0.35.tar`

### file list

```diff
@@ -1,31 +1,31 @@
-drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-04-28 17:05:05.770809 wgd-2.0.34/
--rwxrwxrwx   0 heche     (1000) heche     (1000)    35071 2022-09-25 05:52:37.000000 wgd-2.0.34/LICENSE
--rwxrwxrwx   0 heche     (1000) heche     (1000)    97832 2024-04-28 17:05:05.770809 wgd-2.0.34/PKG-INFO
--rwxrwxrwx   0 heche     (1000) heche     (1000)    97594 2024-04-26 01:53:51.000000 wgd-2.0.34/README.md
--rwxrwxrwx   0 heche     (1000) heche     (1000)    62340 2024-04-24 15:16:17.000000 wgd-2.0.34/cli.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)      103 2024-04-28 17:05:05.775876 wgd-2.0.34/setup.cfg
--rwxrwxrwx   0 heche     (1000) heche     (1000)     1949 2024-04-28 16:47:24.000000 wgd-2.0.34/setup.py
-drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-04-28 17:05:04.212795 wgd-2.0.34/test/
--rwxrwxrwx   0 heche     (1000) heche     (1000)     4521 2023-06-06 22:05:06.000000 wgd-2.0.34/test/test_core.py
-drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-04-28 17:05:05.573748 wgd-2.0.34/wgd/
--rwxrwxrwx   0 heche     (1000) heche     (1000)      711 2022-09-25 05:52:37.000000 wgd-2.0.34/wgd/__init__.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    31126 2022-12-22 23:06:45.000000 wgd-2.0.34/wgd/beast.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)     1685 2022-09-30 07:30:21.000000 wgd-2.0.34/wgd/cluster.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    12579 2023-11-20 15:14:14.000000 wgd-2.0.34/wgd/codeml.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)   157655 2024-04-22 13:37:59.000000 wgd-2.0.34/wgd/core.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    17245 2024-04-22 09:47:26.000000 wgd-2.0.34/wgd/mcmctree.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    14140 2023-03-01 13:14:50.000000 wgd-2.0.34/wgd/mix.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)   119952 2024-04-28 16:43:47.000000 wgd-2.0.34/wgd/peak.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)     4722 2023-06-06 22:05:06.000000 wgd-2.0.34/wgd/postplot.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    54236 2024-04-16 08:22:14.000000 wgd-2.0.34/wgd/ratecorrect.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    18305 2024-03-04 04:24:50.000000 wgd-2.0.34/wgd/syn.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    21638 2023-10-09 15:32:16.000000 wgd-2.0.34/wgd/tree.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    31523 2023-06-23 20:25:14.000000 wgd-2.0.34/wgd/utils.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)   176561 2024-03-06 03:00:56.000000 wgd-2.0.34/wgd/viz.py
-drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-04-28 17:05:05.768982 wgd-2.0.34/wgd.egg-info/
--rwxrwxrwx   0 heche     (1000) heche     (1000)    97832 2024-04-28 17:05:00.000000 wgd-2.0.34/wgd.egg-info/PKG-INFO
--rwxrwxrwx   0 heche     (1000) heche     (1000)      416 2024-04-28 17:05:03.000000 wgd-2.0.34/wgd.egg-info/SOURCES.txt
--rwxrwxrwx   0 heche     (1000) heche     (1000)        1 2024-04-28 17:05:00.000000 wgd-2.0.34/wgd.egg-info/dependency_links.txt
--rwxrwxrwx   0 heche     (1000) heche     (1000)       32 2024-04-28 17:05:00.000000 wgd-2.0.34/wgd.egg-info/entry_points.txt
--rwxrwxrwx   0 heche     (1000) heche     (1000)      774 2024-04-28 17:05:00.000000 wgd-2.0.34/wgd.egg-info/requires.txt
--rwxrwxrwx   0 heche     (1000) heche     (1000)        8 2024-04-28 17:05:00.000000 wgd-2.0.34/wgd.egg-info/top_level.txt
+drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-05-01 10:50:20.623255 wgd-2.0.35/
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    35071 2022-09-25 05:52:37.000000 wgd-2.0.35/LICENSE
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    97683 2024-05-01 10:50:20.623255 wgd-2.0.35/PKG-INFO
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    97445 2024-05-01 07:38:13.000000 wgd-2.0.35/README.md
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    62375 2024-05-01 10:49:12.000000 wgd-2.0.35/cli.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)      103 2024-05-01 10:50:20.623255 wgd-2.0.35/setup.cfg
+-rwxrwxrwx   0 heche     (1000) heche     (1000)     1949 2024-05-01 10:49:22.000000 wgd-2.0.35/setup.py
+drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-05-01 10:50:20.448243 wgd-2.0.35/test/
+-rwxrwxrwx   0 heche     (1000) heche     (1000)     4521 2023-06-06 22:05:06.000000 wgd-2.0.35/test/test_core.py
+drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-05-01 10:50:20.607399 wgd-2.0.35/wgd/
+-rwxrwxrwx   0 heche     (1000) heche     (1000)      711 2022-09-25 05:52:37.000000 wgd-2.0.35/wgd/__init__.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    31126 2022-12-22 23:06:45.000000 wgd-2.0.35/wgd/beast.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)     1685 2022-09-30 07:30:21.000000 wgd-2.0.35/wgd/cluster.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    12579 2023-11-20 15:14:14.000000 wgd-2.0.35/wgd/codeml.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)   157655 2024-04-22 13:37:59.000000 wgd-2.0.35/wgd/core.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    17245 2024-04-22 09:47:26.000000 wgd-2.0.35/wgd/mcmctree.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    14140 2023-03-01 13:14:50.000000 wgd-2.0.35/wgd/mix.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)   120060 2024-05-01 09:43:52.000000 wgd-2.0.35/wgd/peak.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)     4722 2023-06-06 22:05:06.000000 wgd-2.0.35/wgd/postplot.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    54236 2024-04-16 08:22:14.000000 wgd-2.0.35/wgd/ratecorrect.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    18305 2024-03-04 04:24:50.000000 wgd-2.0.35/wgd/syn.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    21638 2023-10-09 15:32:16.000000 wgd-2.0.35/wgd/tree.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    31523 2023-06-23 20:25:14.000000 wgd-2.0.35/wgd/utils.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)   176151 2024-05-01 10:43:46.000000 wgd-2.0.35/wgd/viz.py
+drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-05-01 10:50:20.623255 wgd-2.0.35/wgd.egg-info/
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    97683 2024-05-01 10:50:20.000000 wgd-2.0.35/wgd.egg-info/PKG-INFO
+-rwxrwxrwx   0 heche     (1000) heche     (1000)      416 2024-05-01 10:50:20.000000 wgd-2.0.35/wgd.egg-info/SOURCES.txt
+-rwxrwxrwx   0 heche     (1000) heche     (1000)        1 2024-05-01 10:50:20.000000 wgd-2.0.35/wgd.egg-info/dependency_links.txt
+-rwxrwxrwx   0 heche     (1000) heche     (1000)       32 2024-05-01 10:50:20.000000 wgd-2.0.35/wgd.egg-info/entry_points.txt
+-rwxrwxrwx   0 heche     (1000) heche     (1000)      774 2024-05-01 10:50:20.000000 wgd-2.0.35/wgd.egg-info/requires.txt
+-rwxrwxrwx   0 heche     (1000) heche     (1000)        8 2024-05-01 10:50:20.000000 wgd-2.0.35/wgd.egg-info/top_level.txt
```

### Comparing `wgd-2.0.34/LICENSE` & `wgd-2.0.35/LICENSE`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/PKG-INFO` & `wgd-2.0.35/README.md`

 * *Files 2% similar despite different names*

```diff
@@ -1,6115 +1,6091 @@
-00000000: 4d65 7461 6461 7461 2d56 6572 7369 6f6e  Metadata-Version
-00000010: 3a20 322e 310a 4e61 6d65 3a20 7767 640a  : 2.1.Name: wgd.
-00000020: 5665 7273 696f 6e3a 2032 2e30 2e33 340a  Version: 2.0.34.
-00000030: 5375 6d6d 6172 793a 2077 6764 0a48 6f6d  Summary: wgd.Hom
-00000040: 652d 7061 6765 3a20 6874 7470 3a2f 2f67  e-page: http://g
-00000050: 6974 6875 622e 636f 6d2f 6865 6368 652d  ithub.com/heche-
-00000060: 7073 622f 7767 640a 4175 7468 6f72 3a20  psb/wgd.Author: 
-00000070: 4865 6e67 6368 6920 4368 656e 0a41 7574  Hengchi Chen.Aut
-00000080: 686f 722d 656d 6169 6c3a 2068 6563 6865  hor-email: heche
-00000090: 4070 7362 2e76 6962 2d75 6765 6e74 2e62  @psb.vib-ugent.b
-000000a0: 650a 4c69 6365 6e73 653a 2047 504c 0a44  e.License: GPL.D
-000000b0: 6573 6372 6970 7469 6f6e 2d43 6f6e 7465  escription-Conte
-000000c0: 6e74 2d54 7970 653a 2074 6578 742f 6d61  nt-Type: text/ma
-000000d0: 726b 646f 776e 0a4c 6963 656e 7365 2d46  rkdown.License-F
-000000e0: 696c 653a 204c 4943 454e 5345 0a0a 3c64  ile: LICENSE..<d
-000000f0: 6976 2061 6c69 676e 3d22 6365 6e74 6572  iv align="center
-00000100: 223e 0a0a 2320 6077 6764 2076 3260 203a  ">..# `wgd v2` :
-00000110: 2061 2073 7569 7465 2074 6f6f 6c20 6f66   a suite tool of
-00000120: 2057 4744 2069 6e66 6572 656e 6365 2061   WGD inference a
-00000130: 6e64 2074 696d 696e 670a 0a5b 215b 4275  nd timing..[![Bu
-00000140: 696c 6420 5374 6174 7573 5d28 6874 7470  ild Status](http
-00000150: 733a 2f2f 6170 702e 7472 6176 6973 2d63  s://app.travis-c
-00000160: 692e 636f 6d2f 6865 6368 652d 7073 622f  i.com/heche-psb/
-00000170: 7767 642e 7376 673f 6272 616e 6368 3d70  wgd.svg?branch=p
-00000180: 6879 6c6f 6461 7469 6e67 295d 2868 7474  hylodating)](htt
-00000190: 7073 3a2f 2f61 7070 2e74 7261 7669 732d  ps://app.travis-
-000001a0: 6369 2e63 6f6d 2f68 6563 6865 2d70 7362  ci.com/heche-psb
-000001b0: 2f77 6764 290a 5b21 5b44 6f63 756d 656e  /wgd).[![Documen
-000001c0: 7461 7469 6f6e 2053 7461 7475 735d 2868  tation Status](h
-000001d0: 7474 7073 3a2f 2f72 6561 6474 6865 646f  ttps://readthedo
-000001e0: 6373 2e6f 7267 2f70 726f 6a65 6374 732f  cs.org/projects/
-000001f0: 7767 6476 322f 6261 6467 652f 3f76 6572  wgdv2/badge/?ver
-00000200: 7369 6f6e 3d6c 6174 6573 7429 5d28 6874  sion=latest)](ht
-00000210: 7470 733a 2f2f 7767 6476 322e 7265 6164  tps://wgdv2.read
-00000220: 7468 6564 6f63 732e 696f 2f65 6e2f 6c61  thedocs.io/en/la
-00000230: 7465 7374 2f3f 6261 6467 653d 6c61 7465  test/?badge=late
-00000240: 7374 290a 5b21 5b6c 6963 656e 7365 5d28  st).[![license](
-00000250: 6874 7470 733a 2f2f 696d 672e 7368 6965  https://img.shie
-00000260: 6c64 732e 696f 2f70 7970 692f 6c2f 7767  lds.io/pypi/l/wg
-00000270: 642e 7376 6729 5d28 6874 7470 733a 2f2f  d.svg)](https://
-00000280: 7079 7069 2e70 7974 686f 6e2e 6f72 672f  pypi.python.org/
-00000290: 7079 7069 2f77 6764 290a 5b21 5b4c 6174  pypi/wgd).[![Lat
-000002a0: 6573 7420 5079 5049 2076 6572 7369 6f6e  est PyPI version
-000002b0: 5d28 6874 7470 733a 2f2f 696d 672e 7368  ](https://img.sh
-000002c0: 6965 6c64 732e 696f 2f70 7970 692f 762f  ields.io/pypi/v/
-000002d0: 7767 642e 7376 6729 5d28 6874 7470 733a  wgd.svg)](https:
-000002e0: 2f2f 7079 7069 2e70 7974 686f 6e2e 6f72  //pypi.python.or
-000002f0: 672f 7079 7069 2f77 6764 290a 5b21 5b69  g/pypi/wgd).[![i
-00000300: 6e73 7461 6c6c 2077 6974 6820 6269 6f63  nstall with bioc
-00000310: 6f6e 6461 5d28 6874 7470 733a 2f2f 696d  onda](https://im
-00000320: 672e 7368 6965 6c64 732e 696f 2f62 6164  g.shields.io/bad
-00000330: 6765 2f69 6e73 7461 6c6c 2532 3077 6974  ge/install%20wit
-00000340: 682d 6269 6f63 6f6e 6461 2d62 7269 6768  h-bioconda-brigh
-00000350: 7467 7265 656e 2e73 7667 3f73 7479 6c65  tgreen.svg?style
-00000360: 3d66 6c61 7429 5d28 6874 7470 3a2f 2f62  =flat)](http://b
-00000370: 696f 636f 6e64 612e 6769 7468 7562 2e69  ioconda.github.i
-00000380: 6f2f 7265 6369 7065 732f 7767 642f 5245  o/recipes/wgd/RE
-00000390: 4144 4d45 2e68 746d 6c29 0a5b 215b 446f  ADME.html).[![Do
-000003a0: 776e 6c6f 6164 735d 2868 7474 7073 3a2f  wnloads](https:/
-000003b0: 2f70 6570 792e 7465 6368 2f62 6164 6765  /pepy.tech/badge
-000003c0: 2f77 6764 295d 2868 7474 7073 3a2f 2f70  /wgd)](https://p
-000003d0: 6570 792e 7465 6368 2f70 726f 6a65 6374  epy.tech/project
-000003e0: 2f77 6764 290a 0a2a 2a48 656e 6763 6869  /wgd)..**Hengchi
-000003f0: 2043 6865 6e2c 2041 7274 6875 7220 5a77   Chen, Arthur Zw
-00000400: 6165 6e65 706f 656c 2c20 5976 6573 2056  aenepoel, Yves V
-00000410: 616e 2064 6520 5065 6572 2a2a 0a0a 5b2a  an de Peer**..[*
-00000420: 2a42 696f 696e 666f 726d 6174 6963 7320  *Bioinformatics 
-00000430: 2620 4576 6f6c 7574 696f 6e61 7279 2047  & Evolutionary G
-00000440: 656e 6f6d 6963 7320 4772 6f75 702a 2a5d  enomics Group**]
-00000450: 2868 7474 7073 3a2f 2f77 7777 2e76 616e  (https://www.van
-00000460: 6465 7065 6572 6c61 622e 6f72 672f 292a  depeerlab.org/)*
-00000470: 2a2c 2056 4942 2d55 4765 6e74 2043 656e  *, VIB-UGent Cen
-00000480: 7465 7220 666f 7220 506c 616e 7420 5379  ter for Plant Sy
-00000490: 7374 656d 7320 4269 6f6c 6f67 792a 2a0a  stems Biology**.
-000004a0: 0a5b 2a2a 496e 7472 6f64 7563 7469 6f6e  .[**Introduction
-000004b0: 2a2a 5d28 2369 6e74 726f 6475 6374 696f  **](#introductio
-000004c0: 6e29 207c 200a 5b2a 2a49 6e73 7461 6c6c  n) | .[**Install
-000004d0: 6174 696f 6e2a 2a5d 2823 696e 7374 616c  ation**](#instal
-000004e0: 6c61 7469 6f6e 2920 7c20 0a5b 2a2a 5061  lation) | .[**Pa
-000004f0: 7261 6d65 7465 7273 2a2a 5d28 2370 6172  rameters**](#par
-00000500: 616d 6574 6572 7329 207c 200a 5b2a 2a55  ameters) | .[**U
-00000510: 7361 6765 2a2a 5d28 2375 7361 6765 2920  sage**](#usage) 
-00000520: 7c20 0a5b 2a2a 496c 6c75 7374 7261 7469  | .[**Illustrati
-00000530: 6f6e 2a2a 5d28 2369 6c6c 7573 7472 6174  on**](#illustrat
-00000540: 696f 6e29 207c 0a5b 2a2a 446f 6375 6d65  ion) |.[**Docume
-00000550: 6e74 6174 696f 6e2a 2a5d 2868 7474 7073  ntation**](https
-00000560: 3a2f 2f77 6764 7632 2e72 6561 6474 6865  ://wgdv2.readthe
-00000570: 646f 6373 2e69 6f2f 656e 2f6c 6174 6573  docs.io/en/lates
-00000580: 742f 3f62 6164 6765 3d6c 6174 6573 7429  t/?badge=latest)
-00000590: 207c 0a5b 2a2a 4369 7461 7469 6f6e 2a2a   |.[**Citation**
-000005a0: 5d28 2363 6974 6174 696f 6e29 0a0a 3c2f  ](#citation)..</
-000005b0: 6469 763e 0a0a 6077 6764 2076 3260 2069  div>..`wgd v2` i
-000005c0: 7320 6120 7079 7468 6f6e 2070 6163 6b61  s a python packa
-000005d0: 6765 2075 7067 7261 6465 6420 6672 6f6d  ge upgraded from
-000005e0: 2074 6865 206f 7269 6769 6e61 6c20 6077   the original `w
-000005f0: 6764 6020 7061 636b 6167 6520 6169 6d69  gd` package aimi
-00000600: 6e67 2066 6f72 2074 6865 2069 6e66 6572  ng for the infer
-00000610: 656e 6365 2061 6e64 2074 696d 696e 6720  ence and timing 
-00000620: 6f66 2061 6e63 6965 6e74 2077 686f 6c65  of ancient whole
-00000630: 2d67 656e 6f6d 6520 6475 706c 6963 6174  -genome duplicat
-00000640: 696f 6e20 2857 4744 2920 6576 656e 7473  ion (WGD) events
-00000650: 2e20 466f 7220 7468 6520 7072 6f70 6f73  . For the propos
-00000660: 6520 6f66 2069 6c6c 7573 7472 6174 696e  e of illustratin
-00000670: 6720 7468 6520 7072 696e 6369 706c 6520  g the principle 
-00000680: 616e 6420 7573 6167 6520 6f66 2060 7767  and usage of `wg
-00000690: 6420 7632 602c 2077 6520 636f 6d70 696c  d v2`, we compil
-000006a0: 6564 2074 6869 7320 646f 6375 6d65 6e74  ed this document
-000006b0: 6174 696f 6e2e 2042 656c 6f77 2077 6520  ation. Below we 
-000006c0: 6669 7273 7420 6761 7665 2061 6e20 696e  first gave an in
-000006d0: 7472 6f64 7563 7469 6f6e 206f 7665 7220  troduction over 
-000006e0: 7468 6520 7363 6f70 6520 616e 6420 6d65  the scope and me
-000006f0: 6368 616e 6973 6d20 6f66 2060 7767 6420  chanism of `wgd 
-00000700: 7632 6020 616e 6420 7468 656e 2074 6865  v2` and then the
-00000710: 2070 7261 6374 6963 616c 2069 6e66 6f72   practical infor
-00000720: 6d61 7469 6f6e 2070 6572 7461 696e 696e  mation pertainin
-00000730: 6720 746f 2074 6865 2069 6e73 7461 6c6c  g to the install
-00000740: 6174 696f 6e20 616e 6420 7573 6167 652e  ation and usage.
-00000750: 2041 6e20 6578 616d 706c 6172 2077 6f72   An examplar wor
-00000760: 6b66 6c6f 7720 6973 2070 726f 7669 6465  kflow is provide
-00000770: 6420 696e 2074 6865 2074 7574 6f72 6961  d in the tutoria
-00000780: 6c20 7365 6374 696f 6e20 6f6e 2068 6f77  l section on how
-00000790: 2074 6f20 7365 656b 2065 7669 6465 6e63   to seek evidenc
-000007a0: 6520 666f 7220 6120 7075 7461 7469 7665  e for a putative
-000007b0: 2057 4744 2065 7665 6e74 2061 6e64 2070   WGD event and p
-000007c0: 6572 666f 726d 2070 726f 7065 7220 7469  erform proper ti
-000007d0: 6d69 6e67 2077 6974 6820 6120 6672 6573  ming with a fres
-000007e0: 686c 7920 6f62 7461 696e 6564 2067 656e  hly obtained gen
-000007f0: 6f6d 6520 6173 7365 6d62 6c79 2069 6e20  ome assembly in 
-00000800: 6861 6e64 2e20 466f 7220 7468 6f73 6520  hand. For those 
-00000810: 7768 6f20 6172 6520 696e 7465 7265 7374  who are interest
-00000820: 6564 2069 6e20 6d6f 7265 2074 6865 6f72  ed in more theor
-00000830: 6574 6963 616c 2064 6574 6169 6c73 2c20  etical details, 
-00000840: 7765 2072 6563 6f6d 6d65 6e64 2074 7572  we recommend tur
-00000850: 6e69 6e67 2074 6f20 6f75 7220 7061 7065  ning to our pape
-00000860: 7220 616e 6420 626f 6f6b 2063 6861 7074  r and book chapt
-00000870: 6572 2066 6f72 206d 6f72 6520 6465 7461  er for more deta
-00000880: 696c 6564 2064 6573 6372 6970 7469 6f6e  iled description
-00000890: 2061 6e64 2069 6e73 6967 6874 6675 6c20   and insightful 
-000008a0: 6469 7363 7573 7369 6f6e 732e 2054 6865  discussions. The
-000008b0: 206b 6579 2069 6d70 726f 7665 6420 6665   key improved fe
-000008c0: 6174 7572 6573 206f 6620 6077 6764 2076  atures of `wgd v
-000008d0: 3260 2061 7265 2064 656d 6f6e 7374 7261  2` are demonstra
-000008e0: 7465 6420 696e 206f 7572 206c 6174 6573  ted in our lates
-000008f0: 7420 6d61 6e75 7363 7269 7074 206f 6e20  t manuscript on 
-00000900: 5b42 696f 696e 666f 726d 6174 6963 735d  [Bioinformatics]
-00000910: 2868 7474 7073 3a2f 2f64 6f69 2e6f 7267  (https://doi.org
-00000920: 2f31 302e 3130 3933 2f62 696f 696e 666f  /10.1093/bioinfo
-00000930: 726d 6174 6963 732f 6274 6165 3237 3229  rmatics/btae272)
-00000940: 2e20 4966 2079 6f75 2075 7365 2060 7767  . If you use `wg
-00000950: 6420 7632 6020 696e 2079 6f75 7220 7265  d v2` in your re
-00000960: 7365 6172 6368 2c20 706c 6561 7365 2063  search, please c
-00000970: 6974 6520 7573 2061 7320 7375 6767 6573  ite us as sugges
-00000980: 7465 6420 696e 205b 4369 7461 7469 6f6e  ted in [Citation
-00000990: 5d28 2363 6974 6174 696f 6e29 2073 6563  ](#citation) sec
-000009a0: 7469 6f6e 2e0a 0a23 2320 496e 7472 6f64  tion...## Introd
-000009b0: 7563 7469 6f6e 0a0a 506f 6c79 706c 6f69  uction..Polyploi
-000009c0: 6469 7a61 7469 6f6e 732c 2074 6865 2065  dizations, the e
-000009d0: 766f 6c75 7469 6f6e 6172 7920 7072 6f63  volutionary proc
-000009e0: 6573 7320 7468 6174 2074 6865 2065 6e74  ess that the ent
-000009f0: 6972 6520 6765 6e6f 6d65 206f 6620 616e  ire genome of an
-00000a00: 206f 7267 616e 6973 6d20 6973 2064 7570   organism is dup
-00000a10: 6c69 6361 7465 642c 2061 6c73 6f20 6e61  licated, also na
-00000a20: 6d65 6420 6173 2077 686f 6c65 2d67 656e  med as whole-gen
-00000a30: 6f6d 6520 6475 706c 6963 6174 696f 6e73  ome duplications
-00000a40: 2028 5747 4473 292c 206f 6363 7572 2072   (WGDs), occur r
-00000a50: 6563 7572 7265 6e74 6c79 2061 6372 6f73  ecurrently acros
-00000a60: 7320 7468 6520 7472 6565 206f 6620 6c69  s the tree of li
-00000a70: 6665 2e20 5468 6572 6520 6172 6520 7477  fe. There are tw
-00000a80: 6f20 6d6f 6465 7320 6f66 2070 6f6c 7970  o modes of polyp
-00000a90: 6c6f 6964 697a 6174 696f 6e73 2c20 6175  loidizations, au
-00000aa0: 746f 706f 6c79 706c 6f69 6469 7a61 7469  topolyploidizati
-00000ab0: 6f6e 7320 616e 6420 616c 6c6f 706f 6c79  ons and allopoly
-00000ac0: 706c 6f69 6469 7a61 7469 6f6e 732e 2041  ploidizations. A
-00000ad0: 7574 6f70 6f6c 7970 6c6f 6964 697a 6174  utopolyploidizat
-00000ae0: 696f 6e73 2061 7265 2074 6865 2064 7570  ions are the dup
-00000af0: 6c69 6361 7469 6f6e 206f 6620 7468 6520  lication of the 
-00000b00: 7361 6d65 2067 656e 6f6d 652c 2072 6573  same genome, res
-00000b10: 756c 7469 6e67 2069 6e20 7477 6f20 6964  ulting in two id
-00000b20: 656e 7469 6361 6c20 7375 6267 656e 6f6d  entical subgenom
-00000b30: 6573 2061 7420 7468 6520 7469 6d65 2069  es at the time i
-00000b40: 7420 656d 6572 6765 642e 2057 6869 6c65  t emerged. While
-00000b50: 2074 6865 2061 6c6c 6f70 6f6c 7970 6c6f   the allopolyplo
-00000b60: 6964 697a 6174 696f 6e73 2061 7265 206e  idizations are n
-00000b70: 6f72 6d61 6c6c 7920 6163 6869 6576 6564  ormally achieved
-00000b80: 2069 6e20 7477 6f20 7374 6570 732c 2066   in two steps, f
-00000b90: 6972 7374 2074 6865 2068 7962 7269 6469  irst the hybridi
-00000ba0: 7a61 7469 6f6e 2062 6574 7765 656e 2074  zation between t
-00000bb0: 776f 2064 6966 6665 7265 6e74 2073 7065  wo different spe
-00000bc0: 6369 6573 2c20 7265 7375 6c74 696e 6720  cies, resulting 
-00000bd0: 696e 2074 6865 2061 7269 7369 6e67 206f  in the arising o
-00000be0: 6620 7472 616e 7369 656e 7420 686f 6d6f  f transient homo
-00000bf0: 706c 6f69 6479 2c73 6563 6f6e 6420 7468  ploidy,second th
-00000c00: 6520 6475 706c 6963 6174 696f 6e20 6f66  e duplication of
-00000c10: 2074 6865 2068 6f6d 6f70 6c6f 6964 792c   the homoploidy,
-00000c20: 2072 6573 756c 7469 6e67 2069 6e20 7468   resulting in th
-00000c30: 6520 656d 6572 6765 6e63 6520 6f66 2061  e emergence of a
-00000c40: 6c6c 6f70 6f6c 7970 6c6f 6964 792e 2044  llopolyploidy. D
-00000c50: 7565 2074 6f20 7468 6520 756e 7374 6162  ue to the unstab
-00000c60: 696c 6974 7920 616e 6420 756e 6261 6c61  ility and unbala
-00000c70: 6e63 6564 2074 6574 7261 736f 6d69 6320  nced tetrasomic 
-00000c80: 696e 6865 7269 7461 6e63 652c 2066 6f72  inheritance, for
-00000c90: 2069 6e73 7461 6e63 6520 7468 6520 6e75   instance the nu
-00000ca0: 636c 6561 722d 6379 746f 706c 6173 6d69  clear-cytoplasmi
-00000cb0: 6320 696e 636f 6d70 6174 6962 696c 6974  c incompatibilit
-00000cc0: 792c 2074 6865 2070 6f6c 7970 6c6f 6964  y, the polyploid
-00000cd0: 7920 6765 6e6f 6d65 2077 696c 6c20 7468  y genome will th
-00000ce0: 656e 2065 7870 6572 6965 6e63 6520 6120  en experience a 
-00000cf0: 7072 6f63 6573 7320 6361 6c6c 6564 2064  process called d
-00000d00: 6970 6c6f 6964 697a 6174 696f 6e2c 2061  iploidization, a
-00000d10: 6c73 6f20 6e61 6d65 6420 6173 2066 7261  lso named as fra
-00000d20: 6374 696f 6e61 7469 6f6e 2c20 6475 7269  ctionation, duri
-00000d30: 6e67 2077 6869 6368 2061 206c 6172 6765  ng which a large
-00000d40: 2070 6f72 7469 6f6e 206f 6620 6765 6e65   portion of gene
-00000d50: 2064 7570 6c69 6361 7465 7320 7769 6c6c   duplicates will
-00000d60: 2067 6574 206c 6f73 7420 616e 6420 6f6e   get lost and on
-00000d70: 6c79 2061 2066 7261 6374 696f 6e20 6361  ly a fraction ca
-00000d80: 6e20 6265 2072 6574 6169 6e65 642e 2054  n be retained. T
-00000d90: 6865 2074 7261 6365 7320 6f66 2070 6f6c  he traces of pol
-00000da0: 7970 6c6f 6964 697a 6174 696f 6e73 2063  yploidizations c
-00000db0: 616e 2062 6520 7468 7573 2075 6e65 6172  an be thus unear
-00000dc0: 7468 6564 2066 726f 6d20 7468 6573 6520  thed from these 
-00000dd0: 7265 7461 696e 6564 2067 656e 6520 6475  retained gene du
-00000de0: 706c 6963 6174 6573 2e20 5468 7265 6520  plicates. Three 
-00000df0: 6170 7072 6f61 6368 6573 2062 6173 6564  approaches based
-00000e00: 206f 6e20 6765 6e65 2064 7570 6c69 6361   on gene duplica
-00000e10: 7465 732c 206e 616d 656c 792c 202a 4b2a  tes, namely, *K*
-00000e20: 3c73 7562 3e53 3c2f 7375 623e 206d 6574  <sub>S</sub> met
-00000e30: 686f 642c 2067 656e 6520 7472 6565 202d  hod, gene tree -
-00000e40: 2073 7065 6369 6573 2074 7265 6520 7265   species tree re
-00000e50: 636f 6e63 696c 6961 7469 6f6e 206d 6574  conciliation met
-00000e60: 686f 6420 616e 6420 7379 6e74 656e 7920  hod and synteny 
-00000e70: 6d65 7468 6f64 2c20 6172 6520 636f 6d6d  method, are comm
-00000e80: 6f6e 6c79 2075 7365 6420 696e 2064 6574  only used in det
-00000e90: 6563 7469 6e67 2065 7669 6465 6e63 6520  ecting evidence 
-00000ea0: 666f 7220 5747 4473 2e20 5468 6520 6765  for WGDs. The ge
-00000eb0: 6e65 2074 7265 6520 2d20 7370 6563 6965  ne tree - specie
-00000ec0: 7320 7472 6565 2072 6563 6f6e 6369 6c69  s tree reconcili
-00000ed0: 6174 696f 6e20 6d65 7468 6f64 2069 7320  ation method is 
-00000ee0: 6e6f 7420 7769 7468 696e 2074 6865 2073  not within the s
-00000ef0: 636f 7065 206f 6620 6077 6764 2076 3260  cope of `wgd v2`
-00000f00: 2c20 6275 7420 7765 206b 696e 646c 7920  , but we kindly 
-00000f10: 7265 6665 7220 7265 6164 6572 7320 7768  refer readers wh
-00000f20: 6f20 6172 6520 696e 7465 7265 7374 6564  o are interested
-00000f30: 2074 6f20 7468 6520 7068 796c 6f67 656e   to the phylogen
-00000f40: 6f6d 6963 2070 726f 6772 616d 2064 6576  omic program dev
-00000f50: 656c 6f70 6564 2062 7920 4172 7468 7572  eloped by Arthur
-00000f60: 205a 7761 656e 6570 6f65 6c20 6e61 6d65   Zwaenepoel name
-00000f70: 6420 5b57 4841 4c45 5d28 6874 7470 733a  d [WHALE](https:
-00000f80: 2f2f 6769 7468 7562 2e63 6f6d 2f61 727a  //github.com/arz
-00000f90: 7761 2f57 6861 6c65 2e6a 6c29 2061 6e64  wa/Whale.jl) and
-00000fa0: 2074 6865 2061 7373 6f63 6961 7465 6420   the associated 
-00000fb0: 5b70 6170 6572 5d28 6874 7470 733a 2f2f  [paper](https://
-00000fc0: 646f 692e 6f72 672f 3130 2e31 3039 332f  doi.org/10.1093/
-00000fd0: 6d6f 6c62 6576 2f6d 737a 3038 3829 2066  molbev/msz088) f
-00000fe0: 6f72 206d 6f72 6520 7465 6368 6e69 6361  or more technica
-00000ff0: 6c20 616e 6420 7468 656f 7265 7469 6361  l and theoretica
-00001000: 6c20 6465 7461 696c 732e 0a0a 5468 6520  l details...The 
-00001010: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00001020: 6d65 7468 6f64 2069 7320 6573 7461 626c  method is establ
-00001030: 6973 6865 6420 6f6e 2061 206d 6f64 656c  ished on a model
-00001040: 206f 6620 6765 6e65 2066 616d 696c 7920   of gene family 
-00001050: 6576 6f6c 7574 696f 6e20 7468 6174 2065  evolution that e
-00001060: 6163 6820 6765 6e65 2066 616d 696c 7920  ach gene family 
-00001070: 6973 2061 6c6c 6f77 6564 2074 6f20 6576  is allowed to ev
-00001080: 6f6c 7665 2076 6961 2067 656e 6520 6475  olve via gene du
-00001090: 706c 6963 6174 696f 6e20 616e 6420 6c6f  plication and lo
-000010a0: 7373 2e20 4e6f 7465 2074 6861 7420 7468  ss. Note that th
-000010b0: 6520 6765 6e65 2066 616d 696c 7920 6865  e gene family he
-000010c0: 7265 2069 7320 6173 7375 6d65 6420 746f  re is assumed to
-000010d0: 2062 6520 7468 6520 636c 7573 7465 7220   be the cluster 
-000010e0: 6f66 2061 6c6c 2067 656e 6573 2064 6573  of all genes des
-000010f0: 6365 6e64 6564 2066 726f 6d20 616e 2061  cended from an a
-00001100: 6e63 6573 7472 616c 2067 656e 6520 696e  ncestral gene in
-00001110: 2061 2073 696e 676c 6520 6765 6e6f 6d65   a single genome
-00001120: 2e20 5265 636f 7665 7269 6e67 2074 6865  . Recovering the
-00001130: 2067 656e 6520 7472 6565 206f 6620 7375   gene tree of su
-00001140: 6368 2067 656e 6520 6661 6d69 6c79 2069  ch gene family i
-00001150: 6e66 6f72 6d73 2074 6865 2074 696d 696e  nforms the timin
-00001160: 672c 2073 6369 6c69 6365 7420 7468 6520  g, scilicet the 
-00001170: 6167 652c 206f 6620 6765 6e65 2064 7570  age, of gene dup
-00001180: 6c69 6361 7469 6f6e 2065 7665 6e74 732e  lication events.
-00001190: 2054 6865 2061 6765 2072 6566 6572 6564   The age refered
-000011a0: 2068 6572 652c 2069 7320 6e6f 7420 696e   here, is not in
-000011b0: 2072 6561 6c20 6765 6f6c 6f67 6963 616c   real geological
-000011c0: 2074 696d 652c 2062 7574 2069 6e20 7468   time, but in th
-000011d0: 6520 756e 6974 206f 6620 6576 6f6c 7574  e unit of evolut
-000011e0: 696f 6e61 7279 2064 6973 7461 6e63 652c  ionary distance,
-000011f0: 2069 2e65 2e2c 2074 6865 206e 756d 6265   i.e., the numbe
-00001200: 7220 6f66 2073 7562 7374 6974 7574 696f  r of substitutio
-00001210: 6e73 2070 6572 2073 6974 652e 2057 6865  ns per site. Whe
-00001220: 6e20 7468 6520 6576 6f6c 7574 696f 6e61  n the evolutiona
-00001230: 7279 2072 6174 6520 7265 6d61 696e 7320  ry rate remains 
-00001240: 6170 7072 6f78 696d 6174 656c 7920 636f  approximately co
-00001250: 6e73 7461 6e74 2c20 7468 6520 6576 6f6c  nstant, the evol
-00001260: 7574 696f 6e61 7279 2064 6973 7461 6e63  utionary distanc
-00001270: 6520 6973 2074 6865 6e20 7375 7070 6f73  e is then suppos
-00001280: 6564 2074 6f20 6265 2070 726f 706f 7274  ed to be proport
-00001290: 696f 6e61 6c20 746f 2074 6865 2072 6561  ional to the rea
-000012a0: 6c20 6576 6f6c 7574 696f 6e61 7279 2074  l evolutionary t
-000012b0: 696d 652e 2054 6865 2073 796e 6f6e 796d  ime. The synonym
-000012c0: 6f75 7320 6469 7374 616e 6365 202a 4b2a  ous distance *K*
-000012d0: 3c73 7562 3e53 3c2f 7375 623e 2c20 7468  <sub>S</sub>, th
-000012e0: 6520 6e75 6d62 6572 206f 6620 7379 6e6f  e number of syno
-000012f0: 6e79 6d6f 7573 2073 7562 7374 6974 7574  nymous substitut
-00001300: 696f 6e73 2070 6572 2073 796e 6f6e 796d  ions per synonym
-00001310: 6f75 7320 7369 7465 2c20 6973 2073 7563  ous site, is suc
-00001320: 6820 6361 6e64 6964 6174 6520 7468 6174  h candidate that
-00001330: 2073 796e 6f6e 796d 6f75 7320 7375 6273   synonymous subs
-00001340: 7469 7475 7469 6f6e 7320 776f 756c 6420  titutions would 
-00001350: 6e6f 7420 696e 6375 7220 7468 6520 6368  not incur the ch
-00001360: 616e 6765 206f 6620 616d 696e 6f20 6163  ange of amino ac
-00001370: 6964 2061 6e64 2061 7265 2074 6875 7320  id and are thus 
-00001380: 7265 6761 7264 6564 2061 7320 6e65 7574  regarded as neut
-00001390: 7261 6c2c 2077 6869 6368 2061 6363 6f72  ral, which accor
-000013a0: 6469 6e67 2074 6f20 7468 6520 6e65 7574  ding to the neut
-000013b0: 7261 6c20 7468 656f 7279 2073 686f 756c  ral theory shoul
-000013c0: 6420 6f63 6375 7220 696e 2063 6f6e 7374  d occur in const
-000013d0: 616e 7420 7261 7465 2e20 4769 7665 6e20  ant rate. Given 
-000013e0: 6120 6d6f 6465 6c20 6f66 2067 656e 6520  a model of gene 
-000013f0: 6661 6d69 6c79 2074 6861 7420 616c 6c6f  family that allo
-00001400: 7773 2074 6865 2067 656e 6520 746f 2064  ws the gene to d
-00001410: 7570 6c69 6361 7465 2061 6e64 2067 6574  uplicate and get
-00001420: 206c 6f73 7420 696e 2061 2066 6978 6564   lost in a fixed
-00001430: 2072 6174 652c 206f 6e65 2063 616e 2064   rate, one can d
-00001440: 6572 6976 6520 7468 6174 2074 6865 2070  erive that the p
-00001450: 726f 6261 6269 6c69 7479 2064 656e 7369  robability densi
-00001460: 7479 2066 756e 6374 696f 6e20 6f66 2074  ty function of t
-00001470: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
-00001480: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-00001490: 696f 6e20 6f66 2072 6574 6169 6e65 6420  ion of retained 
-000014a0: 6765 6e65 2064 7570 6c69 6361 7465 7320  gene duplicates 
-000014b0: 6973 2061 2071 7561 7369 2d65 7870 6f6e  is a quasi-expon
-000014c0: 656e 7469 616c 2066 756e 6374 696f 6e20  ential function 
-000014d0: 7468 6174 206d 6f73 7420 7265 7461 696e  that most retain
-000014e0: 6564 2067 656e 6520 6475 706c 6963 6174  ed gene duplicat
-000014f0: 6573 2061 7265 2072 6563 656e 746c 7920  es are recently 
-00001500: 626f 726e 6564 2077 6974 6820 7e30 2061  borned with ~0 a
-00001510: 6765 2077 6869 6c65 2061 7320 7468 6520  ge while as the 
-00001520: 6167 6520 676f 696e 6720 6f6c 6465 7220  age going older 
-00001530: 7468 6520 6173 736f 6369 6174 6564 206e  the associated n
-00001540: 756d 6265 7220 6f66 2072 6574 6169 6e65  umber of retaine
-00001550: 6420 6765 6e65 2064 7570 6c69 6361 7465  d gene duplicate
-00001560: 7320 6465 6361 7920 7175 6173 692d 6578  s decay quasi-ex
-00001570: 706f 6e65 6e74 6961 6c6c 792e 2054 6865  ponentially. The
-00001580: 7265 666f 7265 2c20 7468 6520 6f63 6375  refore, the occu
-00001590: 7261 6e63 6520 6f66 206c 6172 6765 2d73  rance of large-s
-000015a0: 6361 6c65 2067 656e 6520 6475 706c 6963  cale gene duplic
-000015b0: 6174 696f 6e20 6576 656e 7473 2c20 666f  ation events, fo
-000015c0: 7220 696e 7374 616e 6520 5747 4473 2c20  r instane WGDs, 
-000015d0: 7769 7468 2076 6172 6965 6420 7265 7465  with varied rete
-000015e0: 6e74 696f 6e20 7261 7465 2c20 7769 6c6c  ntion rate, will
-000015f0: 206c 6561 7665 2061 6e20 6167 6520 7065   leave an age pe
-00001600: 616b 2066 726f 6d20 7468 6520 6275 7273  ak from the burs
-00001610: 7420 6f66 2067 656e 6520 6475 706c 6963  t of gene duplic
-00001620: 6174 6573 2069 6e20 6120 7368 6f72 7420  ates in a short 
-00001630: 7469 6d65 2d66 7261 6d65 2075 706f 6e20  time-frame upon 
-00001640: 7468 6520 696e 6974 6961 6c20 6167 6520  the initial age 
-00001650: 6469 7374 7269 6275 7469 6f6e 2c20 616e  distribution, an
-00001660: 6420 6361 6e20 6265 2075 6e76 6569 6c65  d can be unveile
-00001670: 6420 6672 6f6d 206d 6978 7475 7265 206d  d from mixture m
-00001680: 6f64 656c 696e 6720 616e 616c 7973 6973  odeling analysis
-00001690: 2e20 486f 7765 7665 722c 2057 4744 7320  . However, WGDs 
-000016a0: 6964 656e 7469 6669 6564 2066 726f 6d20  identified from 
-000016b0: 7468 6520 7061 7261 6c6f 676f 7573 202a  the paralogous *
-000016c0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-000016d0: 6765 2064 6973 7472 6962 7574 696f 6e73  ge distributions
-000016e0: 2063 616e 206f 6e6c 7920 696e 666f 726d   can only inform
-000016f0: 2074 6865 2057 4744 2074 696d 696e 6720   the WGD timing 
-00001700: 696e 2074 6865 2074 696d 652d 7363 616c  in the time-scal
-00001710: 6520 6f66 2074 6861 7420 7370 6563 6966  e of that specif
-00001720: 6963 2073 7065 6369 6573 2c20 7768 6963  ic species, whic
-00001730: 6820 6973 206e 6f74 2063 6f6d 7061 7261  h is not compara
-00001740: 626c 6520 696e 2074 6865 2070 6879 6c6f  ble in the phylo
-00001750: 6765 6e65 7469 6320 636f 6e74 6578 742e  genetic context.
-00001760: 204f 6e6c 7920 7769 7468 2074 6865 206f   Only with the o
-00001770: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
-00001780: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-00001790: 6973 7472 6962 7574 696f 6e73 2c20 7768  istributions, wh
-000017a0: 6963 6820 636f 6e76 6572 7420 7468 6520  ich convert the 
-000017b0: 6573 7469 6d61 7465 6420 626f 6479 2066  estimated body f
-000017c0: 726f 6d20 7061 7261 6c6f 6775 6573 2074  rom paralogues t
-000017d0: 6f20 6f72 7468 6f6c 6f67 7565 7320 616e  o orthologues an
-000017e0: 6420 696e 666f 726d 2074 6865 2072 656c  d inform the rel
-000017f0: 6174 6976 6520 7469 6d69 6e67 206f 6620  ative timing of 
-00001800: 7370 6563 6961 7469 6f6e 2065 7665 6e74  speciation event
-00001810: 732c 2063 616e 2077 6520 6465 6369 7068  s, can we deciph
-00001820: 6572 2074 6865 2070 6879 6c6f 6765 6e65  er the phylogene
-00001830: 7469 6320 706c 6163 656d 656e 7420 6f66  tic placement of
-00001840: 2057 4744 7320 6166 7465 7220 7072 6f70   WGDs after prop
-00001850: 6572 2072 6174 6520 636f 7272 6563 7469  er rate correcti
-00001860: 6f6e 2e20 6077 6764 2076 3260 2069 7320  on. `wgd v2` is 
-00001870: 7375 6368 2070 726f 6772 616d 2074 6861  such program tha
-00001880: 7420 6865 6c70 7320 7573 6572 7320 636f  t helps users co
-00001890: 6e73 7472 7563 7420 7061 7261 6c6f 676f  nstruct paralogo
-000018a0: 7573 2061 6e64 206f 7274 686f 6c6f 676f  us and orthologo
-000018b0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-000018c0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-000018d0: 696f 6e73 2061 6e64 2072 6561 6c69 7a65  ions and realize
-000018e0: 2062 6f74 6820 7468 6520 6964 656e 7469   both the identi
-000018f0: 6669 6361 7469 6f6e 2061 6e64 2070 6c61  fication and pla
-00001900: 6365 6d65 6e74 206f 6620 5747 4473 2e0a  cement of WGDs..
-00001910: 0a49 6e20 7468 6520 7072 656d 6973 6520  .In the premise 
-00001920: 6f66 2070 6879 6c6f 6765 6e65 7469 6361  of phylogenetica
-00001930: 6c6c 7920 6c6f 6361 7465 6420 5747 4473  lly located WGDs
-00001940: 2c20 7468 6520 6162 736f 6c75 7465 2061  , the absolute a
-00001950: 6765 2028 696e 2067 656f 6c6f 6769 6361  ge (in geologica
-00001960: 6c20 7469 6d65 2920 6f66 2057 4744 7320  l time) of WGDs 
-00001970: 6361 6e20 616c 736f 2062 6520 696e 6665  can also be infe
-00001980: 7272 6564 2066 726f 6d20 7468 6f73 6520  rred from those 
-00001990: 5747 442d 7265 7461 696e 6564 2067 656e  WGD-retained gen
-000019a0: 6520 6475 706c 6963 6174 6573 2c20 616c  e duplicates, al
-000019b0: 7468 6f75 6768 2074 6865 7265 2068 6173  though there has
-000019c0: 2062 6565 6e20 6e6f 2065 6173 7920 6f72   been no easy or
-000019d0: 2073 7472 6169 6768 7466 6f72 7761 7264   straightforward
-000019e0: 2070 6970 656c 696e 6520 666f 7220 7468   pipeline for th
-000019f0: 6973 206a 6f62 2073 6f20 6661 722e 2049  is job so far. I
-00001a00: 6e20 6077 6764 2076 3260 2c20 7765 2064  n `wgd v2`, we d
-00001a10: 6576 656c 6f70 6564 2061 2066 6561 7369  eveloped a feasi
-00001a20: 626c 6520 696e 7465 6772 6174 6564 2070  ble integrated p
-00001a30: 6970 6c65 6c69 6e65 2066 6f72 2061 6273  ipleline for abs
-00001a40: 6f6c 7574 6520 6461 7469 6e67 206f 6620  olute dating of 
-00001a50: 5747 4473 2e20 5468 6520 7069 7065 6c69  WGDs. The pipeli
-00001a60: 6e65 2063 616e 2062 6520 726f 7567 686c  ne can be roughl
-00001a70: 7920 6469 7669 6465 6420 696e 746f 2074  y divided into t
-00001a80: 6872 6565 206d 6169 6e20 7374 6570 732e  hree main steps.
-00001a90: 2031 2920 5468 6520 636f 6e73 7472 7563   1) The construc
-00001aa0: 7469 6f6e 206f 6620 616e 6368 6f72 202a  tion of anchor *
-00001ab0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
-00001ac0: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
-00001ad0: 7468 6520 6465 6c69 6e65 6174 696f 6e20  the delineation 
-00001ae0: 6f66 2063 7265 6469 6162 6c65 202a 4b2a  of crediable *K*
-00001af0: 3c73 7562 3e53 3c2f 7375 623e 2072 616e  <sub>S</sub> ran
-00001b00: 6765 2061 646f 7074 6564 2066 6f72 2070  ge adopted for p
-00001b10: 6879 6c6f 6765 6e65 7469 6320 6461 7469  hylogenetic dati
-00001b20: 6e67 2c20 7573 696e 6720 6077 6764 2064  ng, using `wgd d
-00001b30: 6d64 602c 2060 7767 6420 6b73 6460 2c20  md`, `wgd ksd`, 
-00001b40: 6077 6764 2073 796e 6020 616e 6420 6077  `wgd syn` and `w
-00001b50: 6764 2070 6561 6b60 2e20 4e6f 7465 2074  gd peak`. Note t
-00001b60: 6861 7420 6865 7265 2077 6520 6f6e 6c79  hat here we only
-00001b70: 2063 6f6e 7369 6465 7220 6765 6e6f 6d65   consider genome
-00001b80: 2061 7373 656d 626c 7920 6265 6361 7573   assembly becaus
-00001b90: 6520 666f 7220 7472 616e 7363 7269 7074  e for transcript
-00001ba0: 6f6d 6520 6173 7365 6d62 6c79 2069 7427  ome assembly it'
-00001bb0: 7320 696d 706f 7373 6962 6c65 2074 6f20  s impossible to 
-00001bc0: 6469 7374 696e 6775 6973 6820 5747 442d  distinguish WGD-
-00001bd0: 6465 7269 7665 6420 6475 706c 6963 6174  derived duplicat
-00001be0: 6573 2066 726f 6d20 736d 616c 6c2d 7363  es from small-sc
-00001bf0: 616c 6520 6475 706c 6963 6174 696f 6e2d  ale duplication-
-00001c00: 6465 7269 7665 6420 6475 706c 6963 6174  derived duplicat
-00001c10: 6573 2c20 7768 6963 6820 6861 7070 656e  es, which happen
-00001c20: 6564 2069 6e20 6120 636f 6e74 696e 756f  ed in a continuo
-00001c30: 7573 2074 696d 652d 6672 616d 6520 696e  us time-frame in
-00001c40: 7374 6561 6420 6f66 206f 6e6c 7920 6120  stead of only a 
-00001c50: 7365 7061 7261 7465 2073 686f 7274 2074  separate short t
-00001c60: 696d 652d 6672 616d 6520 616e 6420 7468  ime-frame and th
-00001c70: 7573 2072 6566 6c65 6374 7320 7468 6520  us reflects the 
-00001c80: 6475 7261 7469 6f6e 206f 6620 7468 6174  duration of that
-00001c90: 2062 7261 6e63 6820 7261 7468 6572 2074   branch rather t
-00001ca0: 6861 6e20 7468 6520 7469 6d65 2061 7420  han the time at 
-00001cb0: 7768 6963 6820 5747 4420 6f63 6375 7272  which WGD occurr
-00001cc0: 6564 2e20 3229 2054 6865 2066 6f72 6d75  ed. 2) The formu
-00001cd0: 6c61 7469 6f6e 206f 6620 6120 7374 6172  lation of a star
-00001ce0: 7469 6e67 2074 7265 6520 7573 6564 2069  ting tree used i
-00001cf0: 6e20 7468 6520 7068 796c 6f67 656e 6574  n the phylogenet
-00001d00: 6963 2064 6174 696e 672c 2063 6f6d 706f  ic dating, compo
-00001d10: 7365 6420 6f66 2061 2066 6577 2073 7065  sed of a few spe
-00001d20: 6369 6573 2061 6e64 2061 6e6e 6f74 6174  cies and annotat
-00001d30: 6564 2077 6974 6820 666f 7373 696c 2063  ed with fossil c
-00001d40: 616c 6962 7261 7469 6f6e 2069 6e66 6f72  alibration infor
-00001d50: 6d61 7469 6f6e 2e20 5468 6973 2073 7465  mation. This ste
-00001d60: 7020 6973 2065 7373 656e 7469 616c 2066  p is essential f
-00001d70: 6f72 2074 6865 2072 6573 756c 7420 6f66  or the result of
-00001d80: 2061 6273 6f6c 7574 6520 5747 4420 6461   absolute WGD da
-00001d90: 7469 6e67 2074 6861 7420 7765 2073 7567  ting that we sug
-00001da0: 6765 7374 2075 7365 7273 206f 6620 7461  gest users of ta
-00001db0: 6b69 6e67 2067 7265 6174 2063 6175 7469  king great cauti
-00001dc0: 6f6e 2074 6f20 6173 7375 7265 2074 6865  on to assure the
-00001dd0: 2074 7265 6520 746f 706f 6c6f 6779 2061   tree topology a
-00001de0: 6e64 2070 726f 7065 7220 626f 756e 6473  nd proper bounds
-00001df0: 2066 6f72 2066 6f73 7369 6c20 6361 6c69   for fossil cali
-00001e00: 6272 6174 696f 6e73 2e20 3329 2054 6865  brations. 3) The
-00001e10: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
-00001e20: 206f 7274 686f 6772 6f75 7073 2063 6f6e   orthogroups con
-00001e30: 7369 7374 696e 6720 6f66 2063 6f6c 6c69  sisting of colli
-00001e40: 6e65 6172 2064 7570 6c69 6361 7465 7320  near duplicates 
-00001e50: 6f66 2074 6865 2066 6f63 616c 2073 7065  of the focal spe
-00001e60: 6369 6573 2061 6e64 2074 6865 6972 2072  cies and their r
-00001e70: 6563 6970 726f 6361 6c20 6265 7374 2068  eciprocal best h
-00001e80: 6974 7320 2852 4248 7329 2061 6761 696e  its (RBHs) again
-00001e90: 7374 206f 7468 6572 2073 7065 6369 6573  st other species
-00001ea0: 2069 6e20 7468 6520 7374 6172 7469 6e67   in the starting
-00001eb0: 2074 7265 6520 616e 6420 7468 6520 7068   tree and the ph
-00001ec0: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
-00001ed0: 6720 7573 696e 6720 6120 6d6f 6c65 6375  g using a molecu
-00001ee0: 6c61 7220 6461 7469 6e67 2070 726f 6772  lar dating progr
-00001ef0: 616d 2066 6f72 2069 6e73 7461 6e63 6520  am for instance 
-00001f00: 606d 636d 6374 7265 6560 2c20 7669 6120  `mcmctree`, via 
-00001f10: 7072 6f67 7261 6d20 6077 6764 2064 6d64  program `wgd dmd
-00001f20: 6020 616e 6420 6077 6764 2066 6f63 7573  ` and `wgd focus
-00001f30: 602e 2057 6520 7265 636f 6d6d 656e 6420  `. We recommend 
-00001f40: 7468 6520 7573 6167 6520 6f66 2042 6179  the usage of Bay
-00001f50: 6573 6961 6e20 6d6f 6c65 6375 6c61 7220  esian molecular 
-00001f60: 6461 7469 6e67 2070 726f 6772 616d 2060  dating program `
-00001f70: 6d63 6d63 7472 6565 602c 2077 6869 6368  mcmctree`, which
-00001f80: 2070 726f 7669 6465 7320 6120 7661 7269   provides a vari
-00001f90: 6574 7920 6f66 2064 6966 6665 7265 6e74  ety of different
-00001fa0: 2073 7562 7374 6974 7574 696f 6e20 616e   substitution an
-00001fb0: 6420 7261 7465 206d 6f64 656c 732e 204e  d rate models. N
-00001fc0: 6f6e 6574 6865 6c65 7373 2c20 7765 2075  onetheless, we u
-00001fd0: 7267 6520 7573 6572 7320 746f 2073 6574  rge users to set
-00001fe0: 2070 7269 6f72 2064 6973 7472 6962 7574   prior distribut
-00001ff0: 696f 6e20 6f66 2064 6966 6665 7265 6e74  ion of different
-00002000: 2070 6172 616d 6574 6572 7320 7769 7468   parameters with
-00002010: 2063 6175 7469 6f6e 2061 6e64 2061 7373   caution and ass
-00002020: 7572 6520 6164 6571 7561 7465 2073 616d  ure adequate sam
-00002030: 706c 696e 6720 6f66 2064 6966 6665 7265  pling of differe
-00002040: 6e74 2070 6172 616d 6574 6572 732e 0a0a  nt parameters...
-00002050: 2323 2049 6e73 7461 6c6c 6174 696f 6e0a  ## Installation.
-00002060: 0a54 6865 2065 6173 6965 7374 2077 6179  .The easiest way
-00002070: 2074 6f20 696e 7374 616c 6c20 6077 6764   to install `wgd
-00002080: 2076 3260 2069 7320 7573 696e 6720 6050   v2` is using `P
-00002090: 5950 4960 2e20 4e6f 7465 2074 6861 7420  YPI`. Note that 
-000020a0: 6966 2079 6f75 2077 616e 7420 746f 2067  if you want to g
-000020b0: 6574 2074 6865 206c 6174 6573 7420 7570  et the latest up
-000020c0: 6461 7465 2c20 7765 2073 7567 6765 7374  date, we suggest
-000020d0: 2069 6e73 7461 6c6c 696e 6720 6672 6f6d   installing from
-000020e0: 2074 6865 2073 6f75 7263 652c 2073 696e   the source, sin
-000020f0: 6365 2074 6865 2075 7064 6174 6520 6f6e  ce the update on
-00002100: 2060 5059 5049 6020 7769 6c6c 2062 6520   `PYPI` will be 
-00002110: 6465 6c61 7965 6420 636f 6d70 6172 6564  delayed compared
-00002120: 2074 6f20 6865 7265 206f 6620 736f 7572   to here of sour
-00002130: 6365 2e0a 0a60 6060 0a70 6970 2069 6e73  ce...```.pip ins
-00002140: 7461 6c6c 2077 6764 0a60 6060 0a0a 546f  tall wgd.```..To
-00002150: 2069 6e73 7461 6c6c 2060 7767 6420 7632   install `wgd v2
-00002160: 6020 696e 2061 2076 6972 7475 616c 2065  ` in a virtual e
-00002170: 6e76 6972 6f6e 6d65 6e74 2028 7768 6963  nvironment (whic
-00002180: 6820 7765 2073 7472 6f6e 676c 7920 7265  h we strongly re
-00002190: 636f 6d6d 656e 642c 206e 6f20 6d61 7474  commend, no matt
-000021a0: 6572 2079 6f75 2069 6e73 7461 6c6c 2066  er you install f
-000021b0: 726f 6d20 736f 7572 6365 2c20 6050 5950  rom source, `PYP
-000021c0: 4960 206f 7220 6062 696f 636f 6e64 6160  I` or `bioconda`
-000021d0: 292c 2074 6865 2066 6f6c 6c6f 7769 6e67  ), the following
-000021e0: 2063 6f6d 6d61 6e64 206c 696e 6573 2063   command lines c
-000021f0: 6f75 6c64 2062 6520 7573 6564 2e0a 0a60  ould be used...`
-00002200: 6060 0a67 6974 2063 6c6f 6e65 2068 7474  ``.git clone htt
-00002210: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-00002220: 6865 6368 652d 7073 622f 7767 640a 6364  heche-psb/wgd.cd
-00002230: 2077 6764 0a76 6972 7475 616c 656e 7620   wgd.virtualenv 
-00002240: 2d70 3d70 7974 686f 6e33 2045 4e56 2028  -p=python3 ENV (
-00002250: 6f72 2070 7974 686f 6e33 202d 6d20 7665  or python3 -m ve
-00002260: 6e76 2045 4e56 290a 736f 7572 6365 2045  nv ENV).source E
-00002270: 4e56 2f62 696e 2f61 6374 6976 6174 650a  NV/bin/activate.
-00002280: 7069 7020 696e 7374 616c 6c20 2d72 2072  pip install -r r
-00002290: 6571 7569 7265 6d65 6e74 732e 7478 740a  equirements.txt.
-000022a0: 7069 7020 696e 7374 616c 6c20 2e0a 6060  pip install ..``
-000022b0: 600a 0a57 6865 6e20 6d65 7420 7769 7468  `..When met with
-000022c0: 2070 6572 6d69 7373 696f 6e20 7072 6f62   permission prob
-000022d0: 6c65 6d20 696e 2069 6e73 7461 6c6c 6174  lem in installat
-000022e0: 696f 6e2c 2070 6c65 6173 6520 7472 7920  ion, please try 
-000022f0: 7468 6520 666f 6c6c 6f77 696e 6720 636f  the following co
-00002300: 6d6d 616e 6420 6c69 6e65 2e0a 0a60 6060  mmand line...```
-00002310: 0a70 6970 2069 6e73 7461 6c6c 202d 6520  .pip install -e 
-00002320: 2e0a 6060 600a 0a49 6620 6d75 6c74 6970  ..```..If multip
-00002330: 6c79 2076 6572 7369 6f6e 7320 6f66 2060  ly versions of `
-00002340: 7767 6460 2077 6572 6520 696e 7374 616c  wgd` were instal
-00002350: 6c65 6420 696e 2074 6865 2073 7973 7465  led in the syste
-00002360: 6d2c 2070 6c65 6173 6520 6164 6420 7468  m, please add th
-00002370: 6520 7269 6768 7420 7061 7468 206f 6620  e right path of 
-00002380: 696e 7465 7265 7374 6564 2076 6572 7369  interested versi
-00002390: 6f6e 2069 6e74 6f20 7468 6520 656e 7669  on into the envi
-000023a0: 726f 6e6d 656e 7420 7661 7269 6162 6c65  ronment variable
-000023b0: 732c 2066 6f72 2065 7861 6d70 6c65 0a0a  s, for example..
-000023c0: 6060 600a 6578 706f 7274 2050 4154 483d  ```.export PATH=
-000023d0: 2224 5041 5448 3a7e 2f2e 6c6f 6361 6c2f  "$PATH:~/.local/
-000023e0: 6269 6e2f 7767 6422 0a60 6060 0a0a 4e6f  bin/wgd".```..No
-000023f0: 7465 2074 6861 7420 7468 6520 7665 7273  te that the vers
-00002400: 696f 6e20 6f66 2060 6e75 6d70 7960 2069  ion of `numpy` i
-00002410: 7320 696d 706f 7274 616e 7420 2866 6f72  s important (for
-00002420: 206d 616e 7920 6f74 6865 7220 7061 636b   many other pack
-00002430: 6167 6573 2061 7265 2074 6865 2073 616d  ages are the sam
-00002440: 6520 6f66 2063 6f75 7273 6529 2c20 6573  e of course), es
-00002450: 7065 6369 616c 6c79 2066 6f72 2060 6661  pecially for `fa
-00002460: 7374 636c 7573 7465 7260 2070 6163 6b61  stcluster` packa
-00002470: 6765 2e20 496e 206f 7572 2074 6573 742c  ge. In our test,
-00002480: 2074 6865 2060 6e75 6d70 7960 2031 2e31   the `numpy` 1.1
-00002490: 392e 3020 776f 726b 7320 6669 6e65 206f  9.0 works fine o
-000024a0: 6e20 6070 7974 686f 6e33 2e36 2f38 602e  n `python3.6/8`.
-000024b0: 2049 6620 796f 7520 6d65 7420 736f 6d65   If you met some
-000024c0: 2065 7272 6f72 7320 6f72 2077 6172 6e69   errors or warni
-000024d0: 6e67 7320 6162 6f75 7420 606e 756d 7079  ngs about `numpy
-000024e0: 602c 206d 6179 6265 2063 6f6e 7369 6465  `, maybe conside
-000024f0: 7269 6e67 2070 7265 2d69 6e73 7461 6c6c  ring pre-install
-00002500: 2060 6e75 6d70 7960 2061 7320 312e 3139   `numpy` as 1.19
-00002510: 2e30 206f 7220 6f74 6865 7220 636c 6f73  .0 or other clos
-00002520: 652d 6279 2076 6572 7369 6f6e 7320 6265  e-by versions be
-00002530: 666f 7265 2079 6f75 2069 6e73 7461 6c6c  fore you install
-00002540: 2060 7767 6460 2e20 6077 6764 6020 7265   `wgd`. `wgd` re
-00002550: 6c69 6573 206f 6e20 6578 7465 726e 616c  lies on external
-00002560: 2073 6f66 7477 6172 6573 2069 6e63 6c75   softwares inclu
-00002570: 6469 6e67 2060 6469 616d 6f6e 6460 2061  ding `diamond` a
-00002580: 6e64 2060 6d63 6c60 2066 6f72 2060 7767  nd `mcl` for `wg
-00002590: 6420 646d 6460 2c20 6070 616d 6c20 7634  d dmd`, `paml v4
-000025a0: 2e39 6a60 2c20 606d 6166 6674 6020 2860  .9j`, `mafft` (`
-000025b0: 6d75 7363 6c65 6020 616e 6420 6070 7261  muscle` and `pra
-000025c0: 6e6b 6020 6966 2073 6574 292c 2060 6661  nk` if set), `fa
-000025d0: 7374 7472 6565 6020 286f 7220 6069 7174  sttree` (or `iqt
-000025e0: 7265 6560 2069 6620 7365 7429 2066 6f72  ree` if set) for
-000025f0: 2060 7767 6420 6b73 6460 2061 6e64 206f   `wgd ksd` and o
-00002600: 7074 696f 6e61 6c6c 7920 606d 7262 6179  ptionally `mrbay
-00002610: 6573 6020 666f 7220 7468 6520 7068 796c  es` for the phyl
-00002620: 6f67 656e 6574 6963 2069 6e66 6572 656e  ogenetic inferen
-00002630: 6365 2066 756e 6374 696f 6e20 696e 2060  ce function in `
-00002640: 7767 6420 646d 6460 2061 6e64 2060 7767  wgd dmd` and `wg
-00002650: 6420 666f 6375 7360 2028 606d 6166 6674  d focus` (`mafft
-00002660: 602c 2060 6d75 7363 6c65 6020 616e 6420  `, `muscle` and 
-00002670: 6070 7261 6e6b 6020 6173 2077 656c 6c20  `prank` as well 
-00002680: 7768 656e 2074 6865 2061 6e61 6c79 7369  when the analysi
-00002690: 7320 7265 7175 6972 6573 2073 6571 7565  s requires seque
-000026a0: 6e63 6520 616c 6967 6e6d 656e 7429 2e20  nce alignment). 
-000026b0: 536f 6d65 206f 7468 6572 206f 7074 696f  Some other optio
-000026c0: 6e61 6c20 736f 6674 7761 7265 7320 696e  nal softwares in
-000026d0: 636c 7564 696e 6720 6070 616d 6c20 7634  cluding `paml v4
-000026e0: 2e39 6a60 2c20 6072 3873 602c 2060 6265  .9j`, `r8s`, `be
-000026f0: 6173 7460 2c20 6065 6767 6e6f 6760 2c20  ast`, `eggnog`, 
-00002700: 6064 6961 6d6f 6e64 602c 2060 696e 7465  `diamond`, `inte
-00002710: 7270 726f 7363 616e 602c 2060 686d 6d65  rproscan`, `hmme
-00002720: 7220 7633 2e31 6232 6020 616e 6420 6061  r v3.1b2` and `a
-00002730: 7374 7261 6c2d 7072 6f60 2028 6068 6d6d  stral-pro` (`hmm
-00002740: 6572 2076 332e 3162 3260 2069 7320 616c  er v3.1b2` is al
-00002750: 736f 2066 6f72 2074 6865 205b 6f72 7468  so for the [orth
-00002760: 6f67 726f 7570 2061 7373 6967 6e6d 656e  ogroup assignmen
-00002770: 745d 2868 7474 7073 3a2f 2f77 6764 7632  t](https://wgdv2
-00002780: 2e72 6561 6474 6865 646f 6373 2e69 6f2f  .readthedocs.io/
-00002790: 656e 2f6c 6174 6573 742f 7573 6167 652e  en/latest/usage.
-000027a0: 6874 6d6c 236f 7274 686f 6772 6f75 702d  html#orthogroup-
-000027b0: 6173 7369 676e 6d65 6e74 2920 6675 6e63  assignment) func
-000027c0: 7469 6f6e 2069 6e20 6077 6764 2064 6d64  tion in `wgd dmd
-000027d0: 6020 616e 6420 6061 7374 7261 6c2d 7072  ` and `astral-pr
-000027e0: 6f60 2069 7320 616c 736f 2066 6f72 2074  o` is also for t
-000027f0: 6865 2066 756e 6374 696f 6e20 602d 2d63  he function `--c
-00002800: 6f6c 6c69 6e65 6172 636f 616c 6573 6365  ollinearcoalesce
-00002810: 6e63 6560 2069 6e20 6077 6764 2064 6d64  nce` in `wgd dmd
-00002820: 6029 2061 7265 2066 6f72 2074 6865 206d  `) are for the m
-00002830: 6f6c 6563 756c 6172 2064 6174 696e 672c  olecular dating,
-00002840: 2067 656e 6520 6661 6d69 6c79 2066 756e   gene family fun
-00002850: 6374 696f 6e20 616e 6e6f 7461 7469 6f6e  ction annotation
-00002860: 206f 7220 7068 796c 6f67 656e 6574 6963   or phylogenetic
-00002870: 2069 6e66 6572 656e 6365 2069 6e20 6077   inference in `w
-00002880: 6764 2066 6f63 7573 602e 0a0a 2323 2050  gd focus`...## P
-00002890: 6172 616d 6574 6572 730a 0a54 6865 7265  arameters..There
-000028a0: 2061 7265 2037 206d 6169 6e20 7072 6f67   are 7 main prog
-000028b0: 7261 6d73 2069 6e20 6077 6764 2076 3260  rams in `wgd v2`
-000028c0: 3a20 6064 6d64 602c 6066 6f63 7573 602c  : `dmd`,`focus`,
-000028d0: 606b 7364 602c 606d 6978 602c 6070 6561  `ksd`,`mix`,`pea
-000028e0: 6b60 2c60 7379 6e60 2c60 7669 7a60 2e20  k`,`syn`,`viz`. 
-000028f0: 4865 7265 6166 7465 7220 7765 2077 696c  Hereafter we wil
-00002900: 6c20 7072 6f76 6964 6520 6120 6465 7461  l provide a deta
-00002910: 696c 6564 2065 6c75 6369 6461 7469 6f6e  iled elucidation
-00002920: 206f 6e20 6561 6368 206f 6620 7468 6520   on each of the 
-00002930: 7072 6f67 7261 6d20 616e 6420 6974 7320  program and its 
-00002940: 6173 736f 6369 6174 6564 2070 6172 616d  associated param
-00002950: 6574 6572 732e 2050 6c65 6173 6520 7265  eters. Please re
-00002960: 6665 7220 746f 2074 6865 205b 5573 6167  fer to the [Usag
-00002970: 655d 2823 7573 6167 6529 2066 6f72 2074  e](#usage) for t
-00002980: 6865 2073 6365 6e61 7269 6f73 2074 6f20  he scenarios to 
-00002990: 7768 6963 6820 6561 6368 2070 6172 616d  which each param
-000029a0: 6574 6572 2061 7070 6c69 6573 2e0a 0a54  eter applies...T
-000029b0: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-000029c0: 646d 6460 2063 616e 2072 6561 6c69 7a65  dmd` can realize
-000029d0: 2074 6865 2064 656c 696e 6561 7469 6f6e   the delineation
-000029e0: 206f 6620 7768 6f6c 6520 7061 7261 6e6f   of whole parano
-000029f0: 6d65 2c20 5242 4873 2028 5265 6369 7072  me, RBHs (Recipr
-00002a00: 6f63 616c 2042 6573 7420 4869 7473 292c  ocal Best Hits),
-00002a10: 204d 5242 4873 2028 4d75 6c74 6970 6c65   MRBHs (Multiple
-00002a20: 2052 6563 6970 726f 6361 6c20 4265 7374   Reciprocal Best
-00002a30: 2048 6974 7329 2c20 6f72 7468 6f67 726f   Hits), orthogro
-00002a40: 7570 7320 616e 6420 736f 6d65 206f 7468  ups and some oth
-00002a50: 6572 206f 7274 686f 6772 6f75 702d 7265  er orthogroup-re
-00002a60: 6c61 7465 6420 6675 6e63 7469 6f6e 732c  lated functions,
-00002a70: 2069 6e63 6c75 6469 6e67 2063 6972 6375   including circu
-00002a80: 6d73 6372 6970 7469 6f6e 206f 6620 6e65  mscription of ne
-00002a90: 7374 6564 2073 696e 676c 652d 636f 7079  sted single-copy
-00002aa0: 206f 7274 686f 6772 6f75 7073 2028 4e53   orthogroups (NS
-00002ab0: 4f47 7329 2c20 756e 6269 6173 6564 2074  OGs), unbiased t
-00002ac0: 6573 7420 6f66 2073 696e 676c 652d 636f  est of single-co
-00002ad0: 7079 206f 7274 686f 6772 6f75 7073 2028  py orthogroups (
-00002ae0: 534f 4773 2920 6f76 6572 206d 6973 7369  SOGs) over missi
-00002af0: 6e67 2069 6e70 6172 616c 6f67 732c 2063  ng inparalogs, c
-00002b00: 6f6e 7374 7275 6374 696f 6e20 6f66 2042  onstruction of B
-00002b10: 5553 434f 2d67 7569 6465 6420 7369 6e67  USCO-guided sing
-00002b20: 6c65 2d63 6f70 7920 6f72 7468 6f67 726f  le-copy orthogro
-00002b30: 7570 7320 2853 4f47 7329 2c61 6e64 2074  ups (SOGs),and t
-00002b40: 6865 2063 6f6c 6c69 6e65 6172 2063 6f61  he collinear coa
-00002b50: 6c65 7363 656e 6365 2069 6e66 6572 656e  lescence inferen
-00002b60: 6365 206f 6620 7068 796c 6f67 656e 792e  ce of phylogeny.
-00002b70: 0a60 6060 0a77 6764 2064 6d64 2073 6571  .```.wgd dmd seq
-00002b80: 7565 6e63 6573 2028 6f70 7469 6f6e 290a  uences (option).
-00002b90: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002ba0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002bb0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002bc0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002bd0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002be0: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
-00002bf0: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
-00002c00: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
-00002c10: 5f64 6d64 0a2d 742c 202d 2d74 6d70 6469  _dmd.-t, --tmpdi
-00002c20: 722c 2074 6865 2074 656d 706f 7261 7279  r, the temporary
-00002c30: 2077 6f72 6b69 6e67 2064 6972 6563 746f   working directo
-00002c40: 7279 2c20 6465 6661 756c 7420 4e6f 6e65  ry, default None
-00002c50: 2c20 6966 204e 6f6e 6520 7761 7320 6769  , if None was gi
-00002c60: 7665 6e2c 2074 6865 2074 6d70 6469 7220  ven, the tmpdir 
-00002c70: 776f 756c 6420 6265 2061 7373 6967 6e65  would be assigne
-00002c80: 6420 7261 6e64 6f6d 206e 616d 6573 2069  d random names i
-00002c90: 6e20 6375 7272 656e 7420 6469 7265 6374  n current direct
-00002ca0: 6f72 7920 616e 6420 6175 746f 6d61 7465  ory and automate
-00002cb0: 6c79 2072 656d 6f76 6564 2061 7420 7468  ly removed at th
-00002cc0: 6520 636f 6d70 6c65 7469 6f6e 206f 6620  e completion of 
-00002cd0: 7072 6f67 7261 6d2c 2065 6c73 6520 7468  program, else th
-00002ce0: 6520 746d 7064 6972 2077 6f75 6c64 2062  e tmpdir would b
-00002cf0: 6520 6b65 7074 0a2d 702c 202d 2d70 726f  e kept.-p, --pro
-00002d00: 742c 2066 6c61 6720 6f70 7469 6f6e 2c20  t, flag option, 
-00002d10: 7768 6574 6865 7220 7573 696e 6720 7072  whether using pr
-00002d20: 6f74 6569 6e20 6f72 206e 7563 6c65 6f74  otein or nucleot
-00002d30: 6964 6520 7365 7175 656e 6365 730a 2d63  ide sequences.-c
-00002d40: 2c20 2d2d 6373 636f 7265 2c20 7468 6520  , --cscore, the 
-00002d50: 632d 7363 6f72 6520 746f 2072 6573 7472  c-score to restr
-00002d60: 6963 7420 7468 6520 686f 6d6f 6c6f 6720  ict the homolog 
-00002d70: 7369 6d69 6c61 7269 7479 206f 6620 4d52  similarity of MR
-00002d80: 4248 732c 2064 6566 6175 6c74 204e 6f6e  BHs, default Non
-00002d90: 652c 2069 6620 4e6f 6e65 2077 6173 2067  e, if None was g
-00002da0: 6976 656e 2c20 7468 6520 632d 7363 6f72  iven, the c-scor
-00002db0: 6520 6675 6e63 696f 6e20 776f 756c 646e  e funcion wouldn
-00002dc0: 2774 2062 6520 6163 7469 7661 7465 642c  't be activated,
-00002dd0: 2065 6c73 6520 6578 7065 6374 696e 6720   else expecting 
-00002de0: 6120 6465 6369 6d61 6c20 7769 7468 696e  a decimal within
-00002df0: 2074 6865 2072 616e 6765 206f 6620 3020   the range of 0 
-00002e00: 616e 6420 310a 2d49 2c20 2d2d 696e 666c  and 1.-I, --infl
-00002e10: 6174 696f 6e2c 2074 6865 2069 6e66 6c61  ation, the infla
-00002e20: 7469 6f6e 2066 6163 746f 7220 666f 7220  tion factor for 
-00002e30: 4d43 4c20 7072 6f67 7261 6d2c 2064 6566  MCL program, def
-00002e40: 6175 6c74 2032 2e30 2c20 7769 7468 2068  ault 2.0, with h
-00002e50: 6967 6865 7220 7661 6c75 6520 6c65 6164  igher value lead
-00002e60: 696e 6720 746f 206d 6f72 6520 6275 7420  ing to more but 
-00002e70: 736d 616c 6c65 7220 636c 7573 7465 7273  smaller clusters
-00002e80: 0a2d 652c 202d 2d65 7661 6c2c 2074 6865  .-e, --eval, the
-00002e90: 2065 2d76 616c 7565 2063 7574 2d6f 6666   e-value cut-off
-00002ea0: 2066 6f72 2073 696d 696c 6172 6974 7920   for similarity 
-00002eb0: 696e 2064 6961 6d6f 6e64 2061 6e64 2f6f  in diamond and/o
-00002ec0: 7220 686d 6d65 722c 2064 6566 6175 6c74  r hmmer, default
-00002ed0: 2031 652d 3130 0a2d 2d74 6f5f 7374 6f70   1e-10.--to_stop
-00002ee0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00002ef0: 6865 7468 6572 2074 6f20 7472 616e 736c  hether to transl
-00002f00: 6174 6520 7468 726f 7567 6820 5354 4f50  ate through STOP
-00002f10: 2063 6f64 6f6e 732c 2069 6620 7468 6520   codons, if the 
-00002f20: 666c 6167 2077 6173 2073 6574 2c20 7472  flag was set, tr
-00002f30: 616e 736c 6174 696f 6e20 7769 6c6c 2062  anslation will b
-00002f40: 6520 7465 726d 696e 6174 6564 2061 7420  e terminated at 
-00002f50: 7468 6520 6669 7273 7420 696e 2d66 7261  the first in-fra
-00002f60: 6d65 2073 746f 7020 636f 646f 6e2c 2065  me stop codon, e
-00002f70: 6c73 6520 6120 6675 6c6c 2074 7261 6e73  lse a full trans
-00002f80: 6c61 7469 6f6e 2063 6f6e 7469 6e75 696e  lation continuin
-00002f90: 6720 6f6e 2070 6173 7369 6e67 2061 6e79  g on passing any
-00002fa0: 2073 746f 7020 636f 646f 6e73 2077 6f75   stop codons wou
-00002fb0: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
-00002fc0: 2d2d 6364 732c 2066 6c61 6720 6f70 7469  --cds, flag opti
-00002fd0: 6f6e 2c20 7768 6574 6865 7220 746f 206f  on, whether to o
-00002fe0: 6e6c 7920 7472 616e 736c 6174 6520 7468  nly translate th
-00002ff0: 6520 636f 6d70 6c65 7465 2043 4453 2074  e complete CDS t
-00003000: 6861 7420 7374 6172 7473 2077 6974 6820  hat starts with 
-00003010: 6120 7661 6c69 6420 7374 6172 7420 636f  a valid start co
-00003020: 646f 6e20 616e 6420 6f6e 6c79 2063 6f6e  don and only con
-00003030: 7461 696e 7320 6120 7369 6e67 6c65 2069  tains a single i
-00003040: 6e2d 6672 616d 6520 7374 6f70 2063 6f64  n-frame stop cod
-00003050: 6f6e 2061 7420 7468 6520 656e 6420 616e  on at the end an
-00003060: 6420 6d75 7374 2062 6520 6469 7669 6461  d must be divida
-00003070: 626c 6520 6279 2074 6872 6565 2c20 6966  ble by three, if
-00003080: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00003090: 742c 206f 6e6c 7920 7468 6520 636f 6d70  t, only the comp
-000030a0: 6c65 7465 2043 4453 2077 6f75 6c64 2062  lete CDS would b
-000030b0: 6520 7472 616e 736c 6174 6564 0a2d 662c  e translated.-f,
-000030c0: 202d 2d66 6f63 7573 2c20 7468 6520 7370   --focus, the sp
-000030d0: 6563 6965 7320 746f 2062 6520 6d65 7267  ecies to be merg
-000030e0: 6564 206f 6e20 6c6f 6361 6c20 4d52 4248  ed on local MRBH
-000030f0: 732c 2064 6566 6175 6c74 204e 6f6e 652c  s, default None,
-00003100: 2069 6620 4e6f 6e65 2077 6173 2067 6976   if None was giv
-00003110: 656e 2c20 7468 6520 6c6f 6361 6c20 4d52  en, the local MR
-00003120: 4248 7320 776f 756c 646e 2774 2062 6520  BHs wouldn't be 
-00003130: 696e 6665 7272 6564 0a2d 6170 2c20 2d2d  inferred.-ap, --
-00003140: 616e 6368 6f72 706f 696e 7473 2c20 7468  anchorpoints, th
-00003150: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
-00003160: 6461 7461 2066 696c 6520 6672 6f6d 2069  data file from i
-00003170: 2d61 6468 6f72 6520 666f 7220 636f 6e73  -adhore for cons
-00003180: 7472 7563 7469 6e67 2074 6865 206f 7274  tructing the ort
-00003190: 686f 6772 6f75 7073 2077 6974 6820 616e  hogroups with an
-000031a0: 6368 6f72 2070 6169 7273 2c20 6465 6661  chor pairs, defa
-000031b0: 756c 7420 4e6f 6e65 0a2d 736d 2c20 2d2d  ult None.-sm, --
-000031c0: 7365 676d 656e 7473 2c20 7468 6520 7365  segments, the se
-000031d0: 676d 656e 7473 2064 6174 6166 696c 6520  gments datafile 
-000031e0: 7573 6564 2069 6e20 636f 6c6c 696e 6561  used in collinea
-000031f0: 7220 636f 616c 6573 6365 6e63 6520 616e  r coalescence an
-00003200: 616c 7973 6973 2069 6620 696e 6974 6961  alysis if initia
-00003210: 7465 642c 2064 6566 6175 6c74 204e 6f6e  ted, default Non
-00003220: 650a 2d6c 652c 202d 2d6c 6973 7465 6c65  e.-le, --listele
-00003230: 6d65 6e74 732c 2074 6865 206c 6973 7473  ments, the lists
-00003240: 6567 6d65 6e74 7320 6461 7461 2066 696c  egments data fil
-00003250: 6520 7573 6564 2069 6e20 636f 6c6c 696e  e used in collin
-00003260: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
-00003270: 616e 616c 7973 6973 2069 6620 696e 6974  analysis if init
-00003280: 6961 7465 642c 2064 6566 6175 6c74 204e  iated, default N
-00003290: 6f6e 650a 2d67 742c 202d 2d67 656e 6574  one.-gt, --genet
-000032a0: 6162 6c65 2c20 7468 6520 6765 6e65 2074  able, the gene t
-000032b0: 6162 6c65 2064 6174 6166 696c 6520 7573  able datafile us
-000032c0: 6564 2069 6e20 636f 6c6c 696e 6561 7220  ed in collinear 
-000032d0: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
-000032e0: 7973 6973 2069 6620 696e 6974 6961 7465  ysis if initiate
-000032f0: 642c 2064 6566 6175 6c74 204e 6f6e 650a  d, default None.
-00003300: 2d63 6f63 2c20 2d2d 636f 6c6c 696e 6561  -coc, --collinea
-00003310: 7263 6f61 6c65 7363 656e 6365 2c20 666c  rcoalescence, fl
-00003320: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-00003330: 6572 2074 6f20 696e 6974 6961 7465 2074  er to initiate t
-00003340: 6865 2063 6f6c 6c69 6e65 6172 2063 6f61  he collinear coa
-00003350: 6c65 7363 656e 6365 2061 6e61 6c79 7369  lescence analysi
-00003360: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
-00003370: 6173 2073 6574 2c20 7468 6520 616e 616c  as set, the anal
-00003380: 7973 6973 2077 6f75 6c64 2062 6520 696e  ysis would be in
-00003390: 6974 6961 7465 640a 2d6b 662c 202d 2d6b  itiated.-kf, --k
-000033a0: 6565 7066 6173 7461 2c20 666c 6167 206f  eepfasta, flag o
-000033b0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000033c0: 6f20 6f75 7470 7574 2074 6865 2073 6571  o output the seq
-000033d0: 7565 6e63 6520 696e 666f 726d 6174 696f  uence informatio
-000033e0: 6e20 6f66 204d 5242 4873 2c20 6966 2074  n of MRBHs, if t
-000033f0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00003400: 2074 6865 2073 6571 7565 6e63 6573 206f   the sequences o
-00003410: 6620 4d52 4248 7320 776f 756c 6420 6265  f MRBHs would be
-00003420: 2069 6e20 6f75 7470 7574 0a2d 6b64 2c20   in output.-kd, 
-00003430: 2d2d 6b65 6570 6475 706c 6963 6174 6573  --keepduplicates
-00003440: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00003450: 6865 7468 6572 2074 6f20 616c 6c6f 7720  hether to allow 
-00003460: 7468 6520 7361 6d65 2067 656e 6520 746f  the same gene to
-00003470: 206f 6363 7572 2069 6e20 6469 6666 6572   occur in differ
-00003480: 656e 7420 4d52 4248 7320 286f 6e6c 7920  ent MRBHs (only 
-00003490: 6d65 616e 696e 6766 756c 2077 6865 6e20  meaningful when 
-000034a0: 7468 6520 6373 636f 7265 2077 6173 2075  the cscore was u
-000034b0: 7365 6429 2c20 6966 2074 6865 2066 6c61  sed), if the fla
-000034c0: 6720 7761 7320 7365 742c 2074 6865 2073  g was set, the s
-000034d0: 616d 6520 6765 6e65 2063 6f75 6c64 2062  ame gene could b
-000034e0: 6520 6173 7369 676e 6564 2074 6f20 6469  e assigned to di
-000034f0: 6666 6572 656e 7420 4d52 4248 730a 2d67  fferent MRBHs.-g
-00003500: 6d2c 202d 2d67 6c6f 6261 6c6d 7262 682c  m, --globalmrbh,
-00003510: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00003520: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00003530: 6520 676c 6f62 616c 204d 5242 4873 2063  e global MRBHs c
-00003540: 6f6e 7374 7275 6374 696f 6e2c 2069 6620  onstruction, if 
-00003550: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00003560: 2c20 7468 6520 2d2d 666f 6375 7320 6f70  , the --focus op
-00003570: 7469 6f6e 2077 6f75 6c64 2062 6520 6967  tion would be ig
-00003580: 6e6f 7265 6420 616e 6420 6f6e 6c79 2067  nored and only g
-00003590: 6c6f 6261 6c20 4d52 4248 7320 776f 756c  lobal MRBHs woul
-000035a0: 6420 6265 2062 7569 6c74 0a2d 6e2c 202d  d be built.-n, -
-000035b0: 2d6e 7468 7265 6164 732c 2074 6865 206e  -nthreads, the n
-000035c0: 756d 6265 7220 6f66 2074 6872 6561 6473  umber of threads
-000035d0: 2074 6f20 7573 652c 2064 6566 6175 6c74   to use, default
-000035e0: 2034 0a2d 6f69 2c20 2d2d 6f72 7468 6f69   4.-oi, --orthoi
-000035f0: 6e66 6572 2c20 666c 6167 206f 7074 696f  nfer, flag optio
-00003600: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
-00003610: 6974 6961 7465 206f 7274 686f 6772 6f75  itiate orthogrou
-00003620: 7020 696e 6665 726e 6563 652c 2069 6620  p infernece, if 
-00003630: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00003640: 2c20 7468 6520 6f72 7468 6f67 726f 7570  , the orthogroup
-00003650: 2069 6e66 6572 6e65 6365 2070 726f 6772   infernece progr
-00003660: 616d 2077 6f75 6c64 2062 6520 696e 6974  am would be init
-00003670: 6961 7465 640a 2d6f 6f2c 202d 2d6f 6e6c  iated.-oo, --onl
-00003680: 796f 7274 686f 2c20 666c 6167 206f 7074  yortho, flag opt
-00003690: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-000036a0: 6f6e 6c79 2063 6f6e 6475 6374 206f 7274  only conduct ort
-000036b0: 686f 6772 6f75 7020 696e 6665 726e 6563  hogroup infernec
-000036c0: 652c 2069 6620 7468 6520 666c 6167 2077  e, if the flag w
-000036d0: 6173 2073 6574 2c20 6f6e 6c79 2074 6865  as set, only the
-000036e0: 206f 7274 686f 6772 6f75 7020 696e 6665   orthogroup infe
-000036f0: 726e 6563 6520 7069 7065 6c69 6e65 2077  rnece pipeline w
-00003700: 6f75 6c64 2062 6520 7065 7266 6f72 6d65  ould be performe
-00003710: 6420 7768 696c 6520 7468 6520 6f74 6865  d while the othe
-00003720: 7220 616e 616c 7973 6973 2077 6f75 6c64  r analysis would
-00003730: 6e27 7420 6265 2069 6e69 7469 6174 6564  n't be initiated
-00003740: 0a2d 676e 2c20 2d2d 6765 746e 736f 672c  .-gn, --getnsog,
-00003750: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00003760: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00003770: 6520 7468 6520 7365 6172 6368 696e 6720  e the searching 
-00003780: 666f 7220 6e65 7374 6564 2073 696e 676c  for nested singl
-00003790: 652d 636f 7079 2067 656e 6520 6661 6d69  e-copy gene fami
-000037a0: 6c69 6573 2028 4e53 4f47 7329 2028 6f6e  lies (NSOGs) (on
-000037b0: 6c79 206d 6561 6e69 6e67 6675 6c20 7768  ly meaningful wh
-000037c0: 656e 2074 6865 206f 7274 686f 6772 6f75  en the orthogrou
-000037d0: 7020 696e 6665 726e 6563 6520 7069 7065  p infernece pipe
-000037e0: 6c69 6e65 2077 6173 2061 6374 6976 6174  line was activat
-000037f0: 6564 292c 2069 6620 7468 6520 666c 6167  ed), if the flag
-00003800: 2077 6173 2073 6574 2c20 6164 6469 7469   was set, additi
-00003810: 6f6e 616c 204e 534f 4773 2061 6e61 6c79  onal NSOGs analy
-00003820: 7369 7320 776f 756c 6420 6265 2070 6572  sis would be per
-00003830: 666f 726d 6564 2062 6573 6964 6573 2074  formed besides t
-00003840: 6865 2062 6173 6963 206f 7274 686f 6772  he basic orthogr
-00003850: 6f75 7020 696e 6665 726e 6563 650a 2d74  oup infernece.-t
-00003860: 7265 652c 202d 2d74 7265 655f 6d65 7468  ree, --tree_meth
-00003870: 6f64 2c20 7768 6963 6820 6765 6e65 2074  od, which gene t
-00003880: 7265 6520 696e 6665 7265 6e63 6520 7072  ree inference pr
-00003890: 6f67 7261 6d20 746f 2069 6e76 6f6b 6520  ogram to invoke 
-000038a0: 286f 6e6c 7920 6d65 616e 696e 6766 756c  (only meaningful
-000038b0: 2077 6865 6e20 7468 6520 636f 6c6c 696e   when the collin
-000038c0: 6561 7220 636f 616c 6573 6365 6e63 652c  ear coalescence,
-000038d0: 2067 656e 652d 746f 2d66 616d 696c 7920   gene-to-family 
-000038e0: 6173 7369 676e 6d65 6e74 206f 7220 4e53  assignment or NS
-000038f0: 4f47 7320 616e 616c 7973 6973 2077 6572  OGs analysis wer
-00003900: 6520 6163 7469 7661 7465 6429 2c20 6465  e activated), de
-00003910: 6661 756c 7420 6661 7374 7472 6565 0a2d  fault fasttree.-
-00003920: 7473 2c20 2d2d 7472 6565 7365 742c 2074  ts, --treeset, t
-00003930: 6865 2070 6172 616d 6574 6572 7320 7365  he parameters se
-00003940: 7474 696e 6720 666f 7220 6765 6e65 2074  tting for gene t
-00003950: 7265 6520 696e 6665 7265 6e63 652c 2064  ree inference, d
-00003960: 6566 6175 6c74 204e 6f6e 652c 2074 6869  efault None, thi
-00003970: 7320 6f70 7469 6f6e 2063 616e 2062 6520  s option can be 
-00003980: 7072 6f76 6964 6564 206d 756c 7469 706c  provided multipl
-00003990: 6520 7469 6d65 730a 2d6d 632c 202d 2d6d  e times.-mc, --m
-000039a0: 736f 6763 7574 2c20 7468 6520 7261 7469  sogcut, the rati
-000039b0: 6f20 6375 746f 6666 2066 6f72 206d 6f73  o cutoff for mos
-000039c0: 746c 7920 7369 6e67 6c65 2d63 6f70 7920  tly single-copy 
-000039d0: 6661 6d69 6c79 2028 6d65 616e 696e 6766  family (meaningf
-000039e0: 756c 2077 6865 6e20 6163 7469 7661 7469  ul when activati
-000039f0: 6e67 2074 6865 206f 7274 686f 6772 6f75  ng the orthogrou
-00003a00: 7020 696e 6665 726e 6563 6520 7069 7065  p infernece pipe
-00003a10: 6c69 6e65 2920 616e 6420 7370 6563 6965  line) and specie
-00003a20: 7320 7265 7072 6573 656e 7461 7469 6f6e  s representation
-00003a30: 2069 6e20 636f 6c6c 696e 6561 7220 636f   in collinear co
-00003a40: 616c 6573 6365 6e63 6520 616e 616c 7973  alescence analys
-00003a50: 6973 2c20 6465 6661 756c 7420 302e 382e  is, default 0.8.
-00003a60: 0a2d 6761 2c20 2d2d 6765 6e65 6173 7369  .-ga, --geneassi
-00003a70: 676e 2c20 666c 6167 206f 7074 696f 6e2c  gn, flag option,
-00003a80: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
-00003a90: 6961 7465 2074 6865 2067 656e 652d 746f  iate the gene-to
-00003aa0: 2d66 616d 696c 7920 6173 7369 676e 6d65  -family assignme
-00003ab0: 6e74 2061 6e61 6c79 7369 732c 2069 6620  nt analysis, if 
-00003ac0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00003ad0: 2c20 7468 6520 616e 616c 7973 6973 2077  , the analysis w
-00003ae0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
-00003af0: 640a 2d73 612c 202d 2d73 6571 3261 7373  d.-sa, --seq2ass
-00003b00: 6967 6e2c 2074 6865 2071 7565 7269 6564  ign, the queried
-00003b10: 2073 6571 7565 6e63 6573 2064 6174 6120   sequences data 
-00003b20: 6669 6c65 2069 6e20 6765 6e65 2d74 6f2d  file in gene-to-
-00003b30: 6661 6d69 6c79 2061 7373 6967 6e6d 656e  family assignmen
-00003b40: 7420 616e 616c 7973 6973 2c20 6465 6661  t analysis, defa
-00003b50: 756c 7420 4e6f 6e65 2c20 7468 6973 206f  ult None, this o
-00003b60: 7074 696f 6e20 6361 6e20 6265 2070 726f  ption can be pro
-00003b70: 7669 6465 6420 6d75 6c74 6970 6c65 2074  vided multiple t
-00003b80: 696d 6573 0a2d 6661 2c20 2d2d 6661 6d32  imes.-fa, --fam2
-00003b90: 6173 7369 676e 2c20 7468 6520 7175 6572  assign, the quer
-00003ba0: 6965 6420 6661 6d69 6c69 7920 6461 7461  ied familiy data
-00003bb0: 2066 696c 6520 696e 2067 656e 652d 746f   file in gene-to
-00003bc0: 2d66 616d 696c 7920 6173 7369 676e 6d65  -family assignme
-00003bd0: 6e74 2061 6e61 6c79 7369 732c 2064 6566  nt analysis, def
-00003be0: 6175 6c74 204e 6f6e 650a 2d63 632c 202d  ault None.-cc, -
-00003bf0: 2d63 6f6e 6361 742c 2066 6c61 6720 6f70  -concat, flag op
-00003c00: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00003c10: 2069 6e69 7469 6174 6520 7468 6520 636f   initiate the co
-00003c20: 6e63 6174 656e 6174 696f 6e20 7069 7065  ncatenation pipe
-00003c30: 6c69 6e65 2066 6f72 206f 7274 686f 6772  line for orthogr
-00003c40: 6f75 7020 696e 6665 726e 6563 652c 2069  oup infernece, i
-00003c50: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00003c60: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
-00003c70: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00003c80: 7465 640a 2d74 652c 202d 2d74 6573 7473  ted.-te, --tests
-00003c90: 6f67 2c20 666c 6167 206f 7074 696f 6e2c  og, flag option,
-00003ca0: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
-00003cb0: 6961 7465 2074 6865 2075 6e62 6961 7365  iate the unbiase
-00003cc0: 6420 7465 7374 206f 6620 7369 6e67 6c65  d test of single
-00003cd0: 2d63 6f70 7920 6765 6e65 2066 616d 696c  -copy gene famil
-00003ce0: 6965 732c 2069 6620 7468 6520 666c 6167  ies, if the flag
-00003cf0: 2077 6173 2073 6574 2c20 7468 6520 616e   was set, the an
-00003d00: 616c 7973 6973 2077 6f75 6c64 2062 6520  alysis would be 
-00003d10: 696e 6974 6961 7465 640a 2d62 732c 202d  initiated.-bs, -
-00003d20: 2d62 696e 732c 2074 6865 206e 756d 6265  -bins, the numbe
-00003d30: 7220 6f66 2062 696e 7320 6469 7669 6465  r of bins divide
-00003d40: 6420 696e 2074 6865 2067 656e 6520 6c65  d in the gene le
-00003d50: 6e67 7468 206e 6f72 6d61 6c69 7a61 7469  ngth normalizati
-00003d60: 6f6e 2c20 6465 6661 756c 7420 3130 300a  on, default 100.
-00003d70: 2d6e 702c 202d 2d6e 6f72 6d61 6c69 7a65  -np, --normalize
-00003d80: 6470 6572 6365 6e74 2c20 7468 6520 7065  dpercent, the pe
-00003d90: 7263 656e 7461 6765 206f 6620 7570 7065  rcentage of uppe
-00003da0: 7220 6869 7473 2075 7365 6420 666f 7220  r hits used for 
-00003db0: 6765 6e65 206c 656e 6774 6820 6e6f 726d  gene length norm
-00003dc0: 616c 697a 6174 696f 6e2c 2064 6566 6175  alization, defau
-00003dd0: 6c74 2035 0a2d 6e6e 2c20 2d2d 6e6f 6e6f  lt 5.-nn, --nono
-00003de0: 726d 616c 697a 6174 696f 6e2c 2066 6c61  rmalization, fla
-00003df0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00003e00: 7220 746f 2063 616c 6c20 6f66 6620 7468  r to call off th
-00003e10: 6520 6e6f 726d 616c 697a 6174 696f 6e2c  e normalization,
-00003e20: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00003e30: 2073 6574 2c20 6e6f 206e 6f72 6d61 6c69   set, no normali
-00003e40: 7a61 7469 6f6e 2077 6f75 6c64 2062 6520  zation would be 
-00003e50: 636f 6e64 7563 7465 640a 2d62 736f 672c  conducted.-bsog,
-00003e60: 202d 2d62 7573 636f 736f 672c 2066 6c61   --buscosog, fla
-00003e70: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00003e80: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
-00003e90: 6520 6275 7363 6f2d 6775 6964 6564 2073  e busco-guided s
-00003ea0: 696e 676c 652d 636f 7079 2067 656e 6520  ingle-copy gene 
-00003eb0: 6661 6d69 6c79 2061 6e61 6c79 7369 732c  family analysis,
-00003ec0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00003ed0: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
-00003ee0: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
-00003ef0: 6961 7465 640a 2d62 686d 6d2c 202d 2d62  iated.-bhmm, --b
-00003f00: 7573 636f 686d 6d2c 2074 6865 2048 4d4d  uscohmm, the HMM
-00003f10: 2070 726f 6669 6c65 2064 6174 6166 696c   profile datafil
-00003f20: 6520 696e 2074 6865 2062 7573 636f 2d67  e in the busco-g
-00003f30: 7569 6465 6420 7369 6e67 6c65 2d63 6f70  uided single-cop
-00003f40: 7920 6765 6e65 2066 616d 696c 7920 616e  y gene family an
-00003f50: 616c 7973 6973 2c20 6465 6661 756c 7420  alysis, default 
-00003f60: 4e6f 6e65 0a2d 6263 7466 2c20 2d2d 6275  None.-bctf, --bu
-00003f70: 7363 6f63 7574 6f66 662c 2074 6865 2048  scocutoff, the H
-00003f80: 4d4d 2073 636f 7265 2063 7574 6f66 6620  MM score cutoff 
-00003f90: 6461 7461 6669 6c65 2069 6e20 7468 6520  datafile in the 
-00003fa0: 6275 7363 6f2d 6775 6964 6564 2073 696e  busco-guided sin
-00003fb0: 676c 652d 636f 7079 2067 656e 6520 6661  gle-copy gene fa
-00003fc0: 6d69 6c79 2061 6e61 6c79 7369 732c 2064  mily analysis, d
-00003fd0: 6566 6175 6c74 204e 6f6e 650a 2d6f 6620  efault None.-of 
-00003fe0: 2c2d 2d6f 6766 6f72 6d61 742c 2066 6c61  ,--ogformat, fla
-00003ff0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00004000: 7220 746f 2061 6464 2069 6e64 6578 2074  r to add index t
-00004010: 6f20 7468 6520 5242 4820 6661 6d69 6c69  o the RBH famili
-00004020: 6573 0a60 6060 0a0a 5468 6520 7072 6f67  es.```..The prog
-00004030: 7261 6d20 6077 6764 2066 6f63 7573 6020  ram `wgd focus` 
-00004040: 6361 6e20 7265 616c 697a 6520 7468 6520  can realize the 
-00004050: 636f 6e63 6174 656e 6174 696f 6e2d 6261  concatenation-ba
-00004060: 7365 6420 616e 6420 636f 616c 6573 6365  sed and coalesce
-00004070: 6e63 652d 6261 7365 6420 7068 796c 6f67  nce-based phylog
-00004080: 656e 6574 6963 2069 6e66 6572 656e 6365  enetic inference
-00004090: 2061 6e64 2070 6879 6c6f 6765 6e65 7469   and phylogeneti
-000040a0: 6320 6461 7469 6e67 206f 6620 5747 4473  c dating of WGDs
-000040b0: 2065 7463 2e0a 6060 600a 7767 6420 666f   etc..```.wgd fo
-000040c0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
-000040d0: 7565 6e63 6573 2028 6f70 7469 6f6e 290a  uences (option).
-000040e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000040f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004100: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004110: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004120: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004130: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
-00004140: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
-00004150: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
-00004160: 5f66 6f63 7573 0a2d 742c 202d 2d74 6d70  _focus.-t, --tmp
-00004170: 6469 722c 2074 6865 2074 656d 706f 7261  dir, the tempora
-00004180: 7279 2077 6f72 6b69 6e67 2064 6972 6563  ry working direc
-00004190: 746f 7279 2c20 6465 6661 756c 7420 4e6f  tory, default No
-000041a0: 6e65 2c20 6966 204e 6f6e 6520 7761 7320  ne, if None was 
-000041b0: 6769 7665 6e2c 2074 6865 2074 6d70 6469  given, the tmpdi
-000041c0: 7220 7769 6c6c 2062 6520 6173 7369 676e  r will be assign
-000041d0: 6564 2072 616e 646f 6d20 6e61 6d65 7320  ed random names 
-000041e0: 696e 2063 7572 7265 6e74 2064 6972 6563  in current direc
-000041f0: 746f 7279 2061 6e64 2061 7574 6f6d 6174  tory and automat
-00004200: 656c 7920 7265 6d6f 7665 6420 6174 2074  ely removed at t
-00004210: 6865 2063 6f6d 706c 6574 696f 6e20 6f66  he completion of
-00004220: 2070 726f 6772 616d 2c20 656c 7365 2074   program, else t
-00004230: 6865 2074 6d70 6469 7220 776f 756c 6420  he tmpdir would 
-00004240: 6265 206b 6570 740a 2d6e 2c20 2d2d 6e74  be kept.-n, --nt
-00004250: 6872 6561 6473 2c20 7468 6520 6e75 6d62  hreads, the numb
-00004260: 6572 206f 6620 7468 7265 6164 7320 746f  er of threads to
-00004270: 2075 7365 2c20 6465 6661 756c 7420 340a   use, default 4.
-00004280: 2d2d 746f 5f73 746f 702c 2066 6c61 6720  --to_stop, flag 
-00004290: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-000042a0: 746f 2074 7261 6e73 6c61 7465 2074 6872  to translate thr
-000042b0: 6f75 6768 2053 544f 5020 636f 646f 6e73  ough STOP codons
-000042c0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-000042d0: 7320 7365 742c 2074 7261 6e73 6c61 7469  s set, translati
-000042e0: 6f6e 2077 696c 6c20 6265 2074 6572 6d69  on will be termi
-000042f0: 6e61 7465 6420 6174 2074 6865 2066 6972  nated at the fir
-00004300: 7374 2069 6e2d 6672 616d 6520 7374 6f70  st in-frame stop
-00004310: 2063 6f64 6f6e 2c20 656c 7365 2061 2066   codon, else a f
-00004320: 756c 6c20 7472 616e 736c 6174 696f 6e20  ull translation 
-00004330: 636f 6e74 696e 7569 6e67 206f 6e20 7061  continuing on pa
-00004340: 7374 2061 6e79 2073 746f 7020 636f 646f  st any stop codo
-00004350: 6e73 2077 6f75 6c64 2062 6520 696e 6974  ns would be init
-00004360: 6961 7465 640a 2d2d 6364 732c 2066 6c61  iated.--cds, fla
-00004370: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00004380: 7220 746f 206f 6e6c 7920 7472 616e 736c  r to only transl
-00004390: 6174 6520 7468 6520 636f 6d70 6c65 7465  ate the complete
-000043a0: 2043 4453 2074 6861 7420 7374 6172 7473   CDS that starts
-000043b0: 2077 6974 6820 6120 7661 6c69 6420 7374   with a valid st
-000043c0: 6172 7420 636f 646f 6e20 616e 6420 6f6e  art codon and on
-000043d0: 6c79 2063 6f6e 7461 696e 7320 6120 7369  ly contains a si
-000043e0: 6e67 6c65 2069 6e2d 6672 616d 6520 7374  ngle in-frame st
-000043f0: 6f70 2063 6f64 6f6e 2061 7420 7468 6520  op codon at the 
-00004400: 656e 6420 616e 6420 6d75 7374 2062 6520  end and must be 
-00004410: 6469 7669 6461 626c 6520 6279 2074 6872  dividable by thr
-00004420: 6565 2c20 6966 2074 6865 2066 6c61 6720  ee, if the flag 
-00004430: 7761 7320 7365 742c 206f 6e6c 7920 7468  was set, only th
-00004440: 6520 636f 6d70 6c65 7465 2043 4453 2077  e complete CDS w
-00004450: 6f75 6c64 2062 6520 7472 616e 736c 6174  ould be translat
-00004460: 6564 0a2d 2d73 7472 6970 5f67 6170 732c  ed.--strip_gaps,
-00004470: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004480: 6574 6865 7220 746f 2064 726f 7020 616c  ether to drop al
-00004490: 6c20 6761 7073 2069 6e20 6d75 6c74 6970  l gaps in multip
-000044a0: 6c65 2073 6571 7565 6e63 6520 616c 6967  le sequence alig
-000044b0: 6e6d 656e 742c 2069 6620 7468 6520 666c  nment, if the fl
-000044c0: 6167 2077 6173 2073 6574 2c20 616c 6c20  ag was set, all 
-000044d0: 6761 7073 2077 6f75 6c64 2062 6520 6472  gaps would be dr
-000044e0: 6f70 7065 640a 2d61 2c20 2d2d 616c 6967  opped.-a, --alig
-000044f0: 6e65 722c 2077 6869 6368 2061 6c69 676e  ner, which align
-00004500: 6d65 6e74 2070 726f 6772 616d 2074 6f20  ment program to 
-00004510: 7573 652c 2064 6566 6175 6c74 206d 6166  use, default maf
-00004520: 6674 0a2d 7472 6565 2c20 2d2d 7472 6565  ft.-tree, --tree
-00004530: 5f6d 6574 686f 642c 2077 6869 6368 2067  _method, which g
-00004540: 656e 6520 7472 6565 2069 6e66 6572 656e  ene tree inferen
-00004550: 6365 2070 726f 6772 616d 2074 6f20 696e  ce program to in
-00004560: 766f 6b65 2c20 6465 6661 756c 7420 6661  voke, default fa
-00004570: 7374 7472 6565 0a2d 7473 2c20 2d2d 7472  sttree.-ts, --tr
-00004580: 6565 7365 742c 2074 6865 2070 6172 616d  eeset, the param
-00004590: 6574 6572 7320 7365 7474 696e 6720 666f  eters setting fo
-000045a0: 7220 6765 6e65 2074 7265 6520 696e 6665  r gene tree infe
-000045b0: 7265 6e63 652c 2064 6566 6175 6c74 204e  rence, default N
-000045c0: 6f6e 652c 2074 6869 7320 6f70 7469 6f6e  one, this option
-000045d0: 2063 616e 2062 6520 7072 6f76 6964 6564   can be provided
-000045e0: 206d 756c 7469 706c 6520 7469 6d65 730a   multiple times.
-000045f0: 2d2d 636f 6e63 6174 656e 6174 696f 6e2c  --concatenation,
-00004600: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004610: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00004620: 6520 7468 6520 636f 6e63 6174 656e 6174  e the concatenat
-00004630: 696f 6e2d 6261 7365 6420 7370 6563 6965  ion-based specie
-00004640: 7320 7472 6565 2069 6e66 6572 656e 6365  s tree inference
-00004650: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00004660: 7320 7365 742c 2063 6f6e 6361 7465 6e61  s set, concatena
-00004670: 7469 6f6e 2d62 6173 6564 2073 7065 6369  tion-based speci
-00004680: 6573 2074 7265 6520 776f 756c 6420 6265  es tree would be
-00004690: 2069 6e66 6572 6564 0a2d 2d63 6f61 6c65   infered.--coale
-000046a0: 7363 656e 6365 2c20 666c 6167 206f 7074  scence, flag opt
-000046b0: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-000046c0: 696e 6974 6961 7465 2074 6865 2063 6f61  initiate the coa
-000046d0: 6c65 7363 656e 6365 2d62 6173 6564 2073  lescence-based s
-000046e0: 7065 6369 6573 2074 7265 6520 696e 6665  pecies tree infe
-000046f0: 7265 6e63 652c 2069 6620 7468 6520 666c  rence, if the fl
-00004700: 6167 2077 6173 2073 6574 2c20 636f 616c  ag was set, coal
-00004710: 6573 6365 6e63 652d 6261 7365 6420 7370  escence-based sp
-00004720: 6563 6965 7320 7472 6565 2077 6f75 6c64  ecies tree would
-00004730: 2062 6520 696e 6665 7265 640a 2d73 702c   be infered.-sp,
-00004740: 202d 2d73 7065 6369 6573 7472 6565 2c20   --speciestree, 
-00004750: 7370 6563 6965 7320 7472 6565 2064 6174  species tree dat
-00004760: 6166 696c 6520 666f 7220 6461 7469 6e67  afile for dating
-00004770: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-00004780: 642c 202d 2d64 6174 696e 672c 2077 6869  d, --dating, whi
-00004790: 6368 206d 6f6c 6563 756c 6172 2064 6174  ch molecular dat
-000047a0: 696e 6720 7072 6f67 7261 6d20 746f 2075  ing program to u
-000047b0: 7365 2c20 6465 6661 756c 7420 6e6f 6e65  se, default none
-000047c0: 0a2d 6473 2c20 2d2d 6461 7469 6e67 7365  .-ds, --datingse
-000047d0: 742c 2074 6865 2070 6172 616d 6574 6572  t, the parameter
-000047e0: 7320 7365 7474 696e 6720 666f 7220 6461  s setting for da
-000047f0: 7469 6e67 2070 726f 6772 616d 2c20 6465  ting program, de
-00004800: 6661 756c 7420 4e6f 6e65 2c20 7468 6973  fault None, this
-00004810: 206f 7074 696f 6e20 6361 6e20 6265 2070   option can be p
-00004820: 726f 7669 6465 6420 6d75 6c74 6970 6c65  rovided multiple
-00004830: 2074 696d 6573 0a2d 6e73 2c20 2d2d 6e73   times.-ns, --ns
-00004840: 6974 6573 2c20 7468 6520 6e73 6974 6573  ites, the nsites
-00004850: 2069 6e66 6f72 6d61 7469 6f6e 2066 6f72   information for
-00004860: 2072 3873 2064 6174 696e 672c 2064 6566   r8s dating, def
-00004870: 6175 6c74 204e 6f6e 650a 2d6f 742c 202d  ault None.-ot, -
-00004880: 2d6f 7574 6772 6f75 702c 2074 6865 206f  -outgroup, the o
-00004890: 7574 6772 6f75 7020 696e 666f 726d 6174  utgroup informat
-000048a0: 696f 6e20 666f 7220 7238 7320 6461 7469  ion for r8s dati
-000048b0: 6e67 2c20 6465 6661 756c 7420 4e6f 6e65  ng, default None
-000048c0: 0a2d 7074 2c20 2d2d 7061 7274 6974 696f  .-pt, --partitio
-000048d0: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
-000048e0: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
-000048f0: 6174 6520 7061 7274 6974 696f 6e20 6461  ate partition da
-00004900: 7469 6e67 2061 6e61 6c79 7369 7320 666f  ting analysis fo
-00004910: 7220 636f 646f 6e2c 2069 6620 7468 6520  r codon, if the 
-00004920: 666c 6167 2077 6173 2073 6574 2c20 616e  flag was set, an
-00004930: 2061 6464 6974 696f 6e61 6c20 7061 7274   additional part
-00004940: 6974 696f 6e20 6461 7469 6e67 2061 6e61  ition dating ana
-00004950: 6c79 7369 7320 776f 756c 6420 6265 2069  lysis would be i
-00004960: 6e69 7469 6174 6564 0a2d 616d 2c20 2d2d  nitiated.-am, --
-00004970: 6161 6d6f 6465 6c2c 2077 6869 6368 2070  aamodel, which p
-00004980: 726f 7465 696e 206d 6f64 656c 2074 6f20  rotein model to 
-00004990: 6265 2075 7365 6420 696e 206d 636d 6374  be used in mcmct
-000049a0: 7265 652c 2064 6566 6175 6c74 2070 6f69  ree, default poi
-000049b0: 7373 6f6e 0a2d 6b73 2c20 666c 6167 206f  sson.-ks, flag o
-000049c0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000049d0: 6f20 696e 6974 6961 7465 204b 7320 6361  o initiate Ks ca
-000049e0: 6c63 756c 6174 696f 6e20 666f 7220 686f  lculation for ho
-000049f0: 6d6f 6c6f 6775 6573 2069 6e20 7468 6520  mologues in the 
-00004a00: 7072 6f76 6964 6564 206f 7274 686f 6c6f  provided ortholo
-00004a10: 676f 7573 2067 656e 6520 6661 6d69 6c79  gous gene family
-00004a20: 0a2d 2d61 6e6e 6f74 6174 696f 6e2c 2077  .--annotation, w
-00004a30: 6869 6368 2061 6e6e 6f74 6174 696f 6e20  hich annotation 
-00004a40: 7072 6f67 7261 6d20 746f 2075 7365 2c20  program to use, 
-00004a50: 6465 6661 756c 7420 4e6f 6e65 0a2d 2d70  default None.--p
-00004a60: 6169 7277 6973 652c 2066 6c61 6720 6f70  airwise, flag op
-00004a70: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00004a80: 2069 6e69 7469 6174 6520 7061 6972 7769   initiate pairwi
-00004a90: 7365 204b 7320 6573 7469 6d61 7469 6f6e  se Ks estimation
-00004aa0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00004ab0: 7320 7365 742c 2070 6169 7277 6973 6520  s set, pairwise 
-00004ac0: 4b73 2076 616c 7565 7320 776f 756c 6420  Ks values would 
-00004ad0: 6265 2065 7374 696d 6174 6564 0a2d 6564  be estimated.-ed
-00004ae0: 2c20 2d2d 6567 676e 6f67 6461 7461 2c20  , --eggnogdata, 
-00004af0: 7468 6520 6567 676e 6f67 2061 6e6e 6f74  the eggnog annot
-00004b00: 6174 696f 6e20 6461 7461 6669 6c65 2c20  ation datafile, 
-00004b10: 6465 6661 756c 7420 4e6f 6e65 0a2d 2d70  default None.--p
-00004b20: 6661 6d2c 2077 6869 6368 206f 7074 696f  fam, which optio
-00004b30: 6e20 746f 2075 7365 2066 6f72 2070 6661  n to use for pfa
-00004b40: 6d20 616e 6e6f 7461 7469 6f6e 2c20 6465  m annotation, de
-00004b50: 6661 756c 7420 4e6f 6e65 0a2d 2d64 6d6e  fault None.--dmn
-00004b60: 622c 2074 6865 2064 6961 6d6f 6e64 2064  b, the diamond d
-00004b70: 6174 6162 6173 6520 666f 7220 616e 6e6f  atabase for anno
-00004b80: 7461 7469 6f6e 2c20 6465 6661 756c 7420  tation, default 
-00004b90: 4e6f 6e65 0a2d 2d68 6d6d 2c20 7468 6520  None.--hmm, the 
-00004ba0: 484d 4d20 7072 6f66 696c 6520 666f 7220  HMM profile for 
-00004bb0: 616e 6e6f 7461 7469 6f6e 2c20 6465 6661  annotation, defa
-00004bc0: 756c 7420 4e6f 6e65 0a2d 2d65 7661 6c75  ult None.--evalu
-00004bd0: 652c 2074 6865 2065 2d76 616c 7565 2063  e, the e-value c
-00004be0: 7574 2d6f 6666 2066 6f72 2061 6e6e 6f74  ut-off for annot
-00004bf0: 6174 696f 6e2c 2064 6566 6175 6c74 2031  ation, default 1
-00004c00: 652d 3130 0a2d 2d65 7865 7061 7468 2c20  e-10.--exepath, 
-00004c10: 7468 6520 7061 7468 2074 6f20 7468 6520  the path to the 
-00004c20: 696e 7465 7270 726f 7363 616e 2065 7865  interproscan exe
-00004c30: 6375 7461 626c 652c 2064 6566 6175 6c74  cutable, default
-00004c40: 204e 6f6e 650a 2d66 2c20 2d2d 666f 7373   None.-f, --foss
-00004c50: 696c 2c20 7468 6520 666f 7373 696c 2063  il, the fossil c
-00004c60: 616c 6962 7261 7469 6f6e 2069 6e66 6f72  alibration infor
-00004c70: 6d61 7469 6f6e 2069 6e20 4265 6173 742c  mation in Beast,
-00004c80: 2064 6566 6175 6c74 2028 2763 6c61 6465   default ('clade
-00004c90: 313b 636c 6164 6532 272c 2027 7461 7861  1;clade2', 'taxa
-00004ca0: 312c 7461 7861 323b 7461 7861 332c 7461  1,taxa2;taxa3,ta
-00004cb0: 7861 3427 2c20 2734 3b35 272c 2027 302e  xa4', '4;5', '0.
-00004cc0: 353b 302e 3627 2c20 2734 3030 3b35 3030  5;0.6', '400;500
-00004cd0: 2729 0a2d 7268 2c20 2d2d 726f 6f74 6865  ').-rh, --roothe
-00004ce0: 6967 6874 2c20 7468 6520 726f 6f74 2068  ight, the root h
-00004cf0: 6569 6768 7420 6361 6c69 6272 6174 696f  eight calibratio
-00004d00: 6e20 696e 666f 2069 6e20 4265 6173 742c  n info in Beast,
-00004d10: 2064 6566 6175 6c74 2028 342c 302e 352c   default (4,0.5,
-00004d20: 3430 3029 0a2d 6373 2c20 2d2d 6368 6169  400).-cs, --chai
-00004d30: 6e73 6574 2c20 7468 6520 7061 7261 6d65  nset, the parame
-00004d40: 7465 7273 206f 6620 4d43 4d43 2063 6861  ters of MCMC cha
-00004d50: 696e 2069 6e20 4265 6173 742c 2064 6566  in in Beast, def
-00004d60: 6175 6c74 2028 3130 3030 302c 3130 3029  ault (10000,100)
-00004d70: 0a2d 2d62 6561 7374 6c67 6a61 722c 2074  .--beastlgjar, t
-00004d80: 6865 2070 6174 6820 746f 2062 6561 7374  he path to beast
-00004d90: 4c47 2e6a 6172 2c20 6465 6661 756c 7420  LG.jar, default 
-00004da0: 4e6f 6e65 0a2d 2d62 6561 676c 652c 2066  None.--beagle, f
-00004db0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00004dc0: 6865 7220 746f 2075 7365 2062 6561 676c  her to use beagl
-00004dd0: 6520 696e 2042 6561 7374 2c20 6966 2074  e in Beast, if t
-00004de0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00004df0: 2062 6561 676c 6520 776f 756c 6420 6265   beagle would be
-00004e00: 2075 7365 640a 2d2d 7072 6f74 636f 6364   used.--protcocd
-00004e10: 6174 696e 672c 2066 6c61 6720 6f70 7469  ating, flag opti
-00004e20: 6f6e 2c20 7768 6574 6865 7220 746f 206f  on, whether to o
-00004e30: 6e6c 7920 696e 6974 6961 7465 2074 6865  nly initiate the
-00004e40: 2070 726f 7465 696e 2d63 6f6e 6361 7465   protein-concate
-00004e50: 6e61 7469 6f6e 2d62 6173 6564 2064 6174  nation-based dat
-00004e60: 696e 6720 616e 616c 7973 6973 2c20 6966  ing analysis, if
-00004e70: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00004e80: 742c 2074 6865 2061 6e61 6c79 7369 7320  t, the analysis 
-00004e90: 776f 756c 6420 6265 2069 6e69 7469 6174  would be initiat
-00004ea0: 6564 0a2d 2d70 726f 7464 6174 696e 672c  ed.--protdating,
-00004eb0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004ec0: 6574 6865 7220 746f 206f 6e6c 7920 696e  ether to only in
-00004ed0: 6974 6961 7465 2074 6865 2070 726f 7465  itiate the prote
-00004ee0: 696e 2d62 6173 6564 2064 6174 696e 6720  in-based dating 
-00004ef0: 616e 616c 7973 6973 2c20 6966 2074 6865  analysis, if the
-00004f00: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00004f10: 6865 2061 6e61 6c79 7369 7320 776f 756c  he analysis woul
-00004f20: 6420 6265 2069 6e69 7469 6174 6564 0a60  d be initiated.`
-00004f30: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
-00004f40: 6077 6764 206b 7364 6020 6361 6e20 7265  `wgd ksd` can re
-00004f50: 616c 697a 6520 7468 6520 636f 6e73 7472  alize the constr
-00004f60: 7563 7469 6f6e 206f 6620 2a4b 2a3c 7375  uction of *K*<su
-00004f70: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-00004f80: 7374 7269 6275 7469 6f6e 2061 6e64 2072  stribution and r
-00004f90: 6174 6520 636f 7272 6563 7469 6f6e 2e0a  ate correction..
-00004fa0: 6060 600a 7767 6420 6b73 6420 6661 6d69  ```.wgd ksd fami
-00004fb0: 6c69 6573 2073 6571 7565 6e63 6573 2028  lies sequences (
-00004fc0: 6f70 7469 6f6e 290a 2d2d 2d2d 2d2d 2d2d  option).--------
-00004fd0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004fe0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004ff0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005000: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005010: 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c 202d 2d6f  --------.-o, --o
-00005020: 7574 6469 722c 2074 6865 206f 7574 7075  utdir, the outpu
-00005030: 7420 6469 7265 6374 6f72 792c 2064 6566  t directory, def
-00005040: 6175 6c74 2077 6764 5f6b 7364 0a2d 742c  ault wgd_ksd.-t,
-00005050: 202d 2d74 6d70 6469 722c 2074 6865 2074   --tmpdir, the t
-00005060: 656d 706f 7261 7279 2077 6f72 6b69 6e67  emporary working
-00005070: 2064 6972 6563 746f 7279 2c20 6465 6661   directory, defa
-00005080: 756c 7420 4e6f 6e65 2c20 6966 204e 6f6e  ult None, if Non
-00005090: 6520 7761 7320 6769 7665 6e2c 2074 6865  e was given, the
-000050a0: 2074 6d70 6469 7220 7769 6c6c 2062 6520   tmpdir will be 
-000050b0: 6173 7369 676e 6564 2072 616e 646f 6d20  assigned random 
-000050c0: 6e61 6d65 7320 696e 2063 7572 7265 6e74  names in current
-000050d0: 2064 6972 6563 746f 7279 2061 6e64 2061   directory and a
-000050e0: 7574 6f6d 6174 656c 7920 7265 6d6f 7665  utomately remove
-000050f0: 6420 6174 2074 6865 2063 6f6d 706c 6574  d at the complet
-00005100: 696f 6e20 6f66 2070 726f 6772 616d 2c20  ion of program, 
-00005110: 656c 7365 2074 6865 2074 6d70 6469 7220  else the tmpdir 
-00005120: 776f 756c 6420 6265 206b 6570 740a 2d6e  would be kept.-n
-00005130: 2c20 2d2d 6e74 6872 6561 6473 2c20 7468  , --nthreads, th
-00005140: 6520 6e75 6d62 6572 206f 6620 7468 7265  e number of thre
-00005150: 6164 7320 746f 2075 7365 2c20 6465 6661  ads to use, defa
-00005160: 756c 7420 340a 2d2d 746f 5f73 746f 702c  ult 4.--to_stop,
-00005170: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00005180: 6574 6865 7220 746f 2074 7261 6e73 6c61  ether to transla
-00005190: 7465 2074 6872 6f75 6768 2053 544f 5020  te through STOP 
-000051a0: 636f 646f 6e73 2c20 6966 2074 6865 2066  codons, if the f
-000051b0: 6c61 6720 7761 7320 7365 742c 2074 7261  lag was set, tra
-000051c0: 6e73 6c61 7469 6f6e 2077 696c 6c20 6265  nslation will be
-000051d0: 2074 6572 6d69 6e61 7465 6420 6174 2074   terminated at t
-000051e0: 6865 2066 6972 7374 2069 6e2d 6672 616d  he first in-fram
-000051f0: 6520 7374 6f70 2063 6f64 6f6e 2c20 656c  e stop codon, el
-00005200: 7365 2061 2066 756c 6c20 7472 616e 736c  se a full transl
-00005210: 6174 696f 6e20 636f 6e74 696e 7569 6e67  ation continuing
-00005220: 206f 6e20 7061 7374 2061 6e79 2073 746f   on past any sto
-00005230: 7020 636f 646f 6e73 2077 6f75 6c64 2062  p codons would b
-00005240: 6520 696e 6974 6961 7465 640a 2d2d 6364  e initiated.--cd
-00005250: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
-00005260: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
-00005270: 7472 616e 736c 6174 6520 7468 6520 636f  translate the co
-00005280: 6d70 6c65 7465 2043 4453 2074 6861 7420  mplete CDS that 
-00005290: 7374 6172 7473 2077 6974 6820 6120 7661  starts with a va
-000052a0: 6c69 6420 7374 6172 7420 636f 646f 6e20  lid start codon 
-000052b0: 616e 6420 6f6e 6c79 2063 6f6e 7461 696e  and only contain
-000052c0: 7320 6120 7369 6e67 6c65 2069 6e2d 6672  s a single in-fr
-000052d0: 616d 6520 7374 6f70 2063 6f64 6f6e 2061  ame stop codon a
-000052e0: 7420 7468 6520 656e 6420 616e 6420 6d75  t the end and mu
-000052f0: 7374 2062 6520 6469 7669 6461 626c 6520  st be dividable 
-00005300: 6279 2074 6872 6565 2c20 6966 2074 6865  by three, if the
-00005310: 2066 6c61 6720 7761 7320 7365 742c 206f   flag was set, o
-00005320: 6e6c 7920 7468 6520 636f 6d70 6c65 7465  nly the complete
-00005330: 2043 4453 2077 6f75 6c64 2062 6520 7472   CDS would be tr
-00005340: 616e 736c 6174 6564 0a2d 2d70 6169 7277  anslated.--pairw
-00005350: 6973 652c 2066 6c61 6720 6f70 7469 6f6e  ise, flag option
-00005360: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
-00005370: 7469 6174 6520 7061 6972 7769 7365 204b  tiate pairwise K
-00005380: 7320 6573 7469 6d61 7469 6f6e 2c20 6966  s estimation, if
-00005390: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-000053a0: 742c 2070 6169 7277 6973 6520 4b73 2076  t, pairwise Ks v
-000053b0: 616c 7565 7320 776f 756c 6420 6265 2065  alues would be e
-000053c0: 7374 696d 6174 6564 0a2d 2d73 7472 6970  stimated.--strip
-000053d0: 5f67 6170 732c 2066 6c61 6720 6f70 7469  _gaps, flag opti
-000053e0: 6f6e 2c20 7768 6574 6865 7220 746f 2064  on, whether to d
-000053f0: 726f 7020 616c 6c20 6761 7073 2069 6e20  rop all gaps in 
-00005400: 6d75 6c74 6970 6c65 2073 6571 7565 6e63  multiple sequenc
-00005410: 6520 616c 6967 6e6d 656e 742c 2069 6620  e alignment, if 
-00005420: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00005430: 2c20 616c 6c20 6761 7073 2077 6f75 6c64  , all gaps would
-00005440: 2062 6520 6472 6f70 7065 640a 2d61 2c20   be dropped.-a, 
-00005450: 2d2d 616c 6967 6e65 722c 2077 6869 6368  --aligner, which
-00005460: 2061 6c69 676e 6d65 6e74 2070 726f 6772   alignment progr
-00005470: 616d 2074 6f20 7573 652c 2064 6566 6175  am to use, defau
-00005480: 6c74 206d 6166 6674 200a 2d74 7265 652c  lt mafft .-tree,
-00005490: 202d 2d74 7265 655f 6d65 7468 6f64 2c20   --tree_method, 
-000054a0: 7768 6963 6820 6765 6e65 2074 7265 6520  which gene tree 
-000054b0: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
-000054c0: 6d20 746f 2069 6e76 6f6b 652c 2064 6566  m to invoke, def
-000054d0: 6175 6c74 2066 6173 7474 7265 650a 2d2d  ault fasttree.--
-000054e0: 7472 6565 5f6f 7074 696f 6e73 2c20 6f70  tree_options, op
-000054f0: 7469 6f6e 7320 696e 2074 7265 6520 696e  tions in tree in
-00005500: 6665 7265 6e63 6520 6173 2061 2063 6f6d  ference as a com
-00005510: 6d61 2073 6570 6172 6174 6564 2073 7472  ma separated str
-00005520: 696e 672c 2064 6566 6175 6c74 204e 6f6e  ing, default Non
-00005530: 650a 2d2d 6e6f 6465 5f61 7665 7261 6765  e.--node_average
-00005540: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00005550: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
-00005560: 7465 206e 6f64 652d 6176 6572 6167 6520  te node-average 
-00005570: 7761 7920 6f66 2064 652d 7265 6475 6e64  way of de-redund
-00005580: 616e 6379 2069 6e73 7465 6164 206f 6620  ancy instead of 
-00005590: 6e6f 6465 2d77 6569 6768 7465 642c 2069  node-weighted, i
-000055a0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-000055b0: 6574 2c20 7468 6520 6e6f 6465 2d61 7665  et, the node-ave
-000055c0: 7261 6769 6e67 2064 652d 7265 6475 6e64  raging de-redund
-000055d0: 616e 6379 2077 6f75 6c64 2062 6520 696e  ancy would be in
-000055e0: 6974 6961 7465 640a 2d73 722c 202d 2d73  itiated.-sr, --s
-000055f0: 7061 6972 2c20 7468 6520 7370 6563 6965  pair, the specie
-00005600: 7320 7061 6972 2074 6f20 6265 2070 6c6f  s pair to be plo
-00005610: 7474 6564 2c20 6465 6661 756c 7420 4e6f  tted, default No
-00005620: 6e65 2c20 7468 6973 206f 7074 696f 6e20  ne, this option 
-00005630: 6361 6e20 6265 2070 726f 7669 6465 6420  can be provided 
-00005640: 6d75 6c74 6970 6c65 2074 696d 6573 0a2d  multiple times.-
-00005650: 7370 2c20 2d2d 7370 6563 6965 7374 7265  sp, --speciestre
-00005660: 652c 2074 6865 2073 7065 6369 6573 2074  e, the species t
-00005670: 7265 6520 746f 2070 6572 666f 726d 2072  ree to perform r
-00005680: 6174 6520 636f 7272 6563 7469 6f6e 2c20  ate correction, 
-00005690: 6465 6661 756c 7420 4e6f 6e65 2c20 6966  default None, if
-000056a0: 204e 6f6e 6520 7761 7320 6769 7665 6e2c   None was given,
-000056b0: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
-000056c0: 7469 6f6e 2061 6e61 6c79 7369 7320 776f  tion analysis wo
-000056d0: 756c 6420 6265 2063 616c 6c65 6420 6f66  uld be called of
-000056e0: 660a 2d72 772c 202d 2d72 6577 6569 6768  f.-rw, --reweigh
-000056f0: 742c 2066 6c61 6720 6f70 7469 6f6e 2c20  t, flag option, 
-00005700: 7768 6574 6865 7220 746f 2072 6563 616c  whether to recal
-00005710: 6375 6c61 7465 2074 6865 2077 6569 6768  culate the weigh
-00005720: 7420 7065 7220 7370 6563 6965 7320 7061  t per species pa
-00005730: 6972 2c20 6966 2074 6865 2066 6c61 6720  ir, if the flag 
-00005740: 7761 7320 7365 742c 2074 6865 2077 6569  was set, the wei
-00005750: 6768 7420 776f 756c 6420 6265 2072 6563  ght would be rec
-00005760: 616c 6375 6c61 7465 640a 2d6f 722c 202d  alculated.-or, -
-00005770: 2d6f 6e6c 7972 6f6f 746f 7574 2c20 666c  -onlyrootout, fl
-00005780: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-00005790: 6572 2074 6f20 6f6e 6c79 2063 6f6e 6475  er to only condu
-000057a0: 6374 2072 6174 6520 636f 7272 6563 7469  ct rate correcti
-000057b0: 6f6e 2075 7369 6e67 2074 6865 206f 7574  on using the out
-000057c0: 6772 6f75 7020 6174 2072 6f6f 7420 6173  group at root as
-000057d0: 206f 7574 6772 6f75 702c 2069 6620 7468   outgroup, if th
-000057e0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-000057f0: 6f6e 6c79 2074 6865 206f 7574 6772 6f75  only the outgrou
-00005800: 7020 6174 2072 6f6f 7420 776f 756c 6420  p at root would 
-00005810: 6265 2075 7365 6420 6173 206f 7574 6772  be used as outgr
-00005820: 6f75 700a 2d65 706b 2c20 2d2d 6578 7472  oup.-epk, --extr
-00005830: 6170 6172 616e 6f6d 656b 732c 2065 7874  aparanomeks, ext
-00005840: 7261 2070 6172 616e 6f6d 6520 4b73 2064  ra paranome Ks d
-00005850: 6174 6120 746f 2070 6c6f 7420 696e 2074  ata to plot in t
-00005860: 6865 206d 6978 6564 204b 7320 6469 7374  he mixed Ks dist
-00005870: 7269 6275 7469 6f6e 2c20 6465 6661 756c  ribution, defaul
-00005880: 7420 4e6f 6e65 0a2d 6170 2c20 2d2d 616e  t None.-ap, --an
-00005890: 6368 6f72 706f 696e 7473 2c20 616e 6368  chorpoints, anch
-000058a0: 6f72 706f 696e 7473 2e74 7874 2066 696c  orpoints.txt fil
-000058b0: 6520 746f 2070 6c6f 7420 616e 6368 6f72  e to plot anchor
-000058c0: 204b 7320 696e 2074 6865 206d 6978 6564   Ks in the mixed
-000058d0: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-000058e0: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-000058f0: 706b 2c20 2d2d 706c 6f74 6b64 652c 2066  pk, --plotkde, f
-00005900: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00005910: 6865 7220 746f 2070 6c6f 7420 6b64 6520  her to plot kde 
-00005920: 6375 7276 6520 6f66 206f 7274 686f 6c6f  curve of ortholo
-00005930: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
-00005940: 7469 6f6e 206f 7665 7220 6869 7374 6f67  tion over histog
-00005950: 7261 6d20 696e 2074 6865 206d 6978 6564  ram in the mixed
-00005960: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00005970: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00005980: 7320 7365 742c 2074 6865 206b 6465 2063  s set, the kde c
-00005990: 7572 7665 2077 6f75 6c64 2062 6520 706c  urve would be pl
-000059a0: 6f74 7465 640a 2d70 6167 2c20 2d2d 706c  otted.-pag, --pl
-000059b0: 6f74 6170 676d 6d2c 2066 6c61 6720 6f70  otapgmm, flag op
-000059c0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-000059d0: 2070 6572 666f 726d 2061 6e64 2070 6c6f   perform and plo
-000059e0: 7420 6d69 7874 7572 6520 6d6f 6465 6c69  t mixture modeli
-000059f0: 6e67 206f 6620 616e 6368 6f72 204b 7320  ng of anchor Ks 
-00005a00: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
-00005a10: 6469 7374 7269 6275 7469 6f6e 2c20 6966  distribution, if
-00005a20: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00005a30: 742c 2074 6865 206d 6978 7475 7265 206d  t, the mixture m
-00005a40: 6f64 656c 696e 6720 6f66 2061 6e63 686f  odeling of ancho
-00005a50: 7220 4b73 2077 6f75 6c64 2062 6520 706c  r Ks would be pl
-00005a60: 6f74 7465 640a 2d70 656d 2c20 2d2d 706c  otted.-pem, --pl
-00005a70: 6f74 656c 6d6d 2c20 666c 6167 206f 7074  otelmm, flag opt
-00005a80: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-00005a90: 7065 7266 6f72 6d20 616e 6420 706c 6f74  perform and plot
-00005aa0: 2065 6c6d 6d20 6d69 7874 7572 6520 6d6f   elmm mixture mo
-00005ab0: 6465 6c69 6e67 206f 6620 7061 7261 6e6f  deling of parano
-00005ac0: 6d65 204b 7320 696e 2074 6865 206d 6978  me Ks in the mix
-00005ad0: 6564 204b 7320 6469 7374 7269 6275 7469  ed Ks distributi
-00005ae0: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
-00005af0: 7761 7320 7365 742c 2074 6865 2065 6c6d  was set, the elm
-00005b00: 6d20 6d69 7874 7572 6520 6d6f 6465 6c69  m mixture modeli
-00005b10: 6e67 206f 6620 7061 7261 6e6f 6d65 204b  ng of paranome K
-00005b20: 7320 776f 756c 6420 6265 2070 6c6f 7474  s would be plott
-00005b30: 6564 0a2d 632c 202d 2d63 6f6d 706f 6e65  ed.-c, --compone
-00005b40: 6e74 732c 2074 6865 2072 616e 6765 206f  nts, the range o
-00005b50: 6620 7468 6520 6e75 6d62 6572 206f 6620  f the number of 
-00005b60: 636f 6d70 6f6e 656e 7473 2074 6f20 6669  components to fi
-00005b70: 7420 696e 2061 6e63 686f 7220 4b73 206d  t in anchor Ks m
-00005b80: 6978 7475 7265 206d 6f64 656c 696e 672c  ixture modeling,
-00005b90: 2064 6566 6175 6c74 2028 312c 3429 0a2d   default (1,4).-
-00005ba0: 786c 2c20 2d2d 786c 696d 2c20 7468 6520  xl, --xlim, the 
-00005bb0: 7820 6178 6973 206c 696d 6974 206f 6620  x axis limit of 
-00005bc0: 4b73 2064 6973 7472 6962 7574 696f 6e0a  Ks distribution.
-00005bd0: 2d79 6c2c 202d 2d79 6c69 6d2c 2074 6865  -yl, --ylim, the
-00005be0: 2079 2061 7869 7320 6c69 6d69 7420 6f66   y axis limit of
-00005bf0: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00005c00: 0a2d 6164 6f2c 202d 2d61 646a 7573 746f  .-ado, --adjusto
-00005c10: 7274 686f 2c20 666c 6167 206f 7074 696f  rtho, flag optio
-00005c20: 6e2c 2077 6865 7468 6572 2074 6f20 6164  n, whether to ad
-00005c30: 6a75 7374 2074 6865 2068 6973 746f 6772  just the histogr
-00005c40: 616d 2068 6569 6768 7420 6f66 206f 7274  am height of ort
-00005c50: 686f 6c6f 676f 7573 204b 7320 6173 2074  hologous Ks as t
-00005c60: 6f20 6d61 7463 6820 7468 6520 6865 6967  o match the heig
-00005c70: 6874 206f 6620 7061 7261 6c6f 676f 7573  ht of paralogous
-00005c80: 204b 732c 2069 6620 7468 6520 666c 6167   Ks, if the flag
-00005c90: 2077 6173 2073 6574 2c20 7468 6520 6164   was set, the ad
-00005ca0: 6a75 7374 6d65 6e74 2077 6f75 6c64 2062  justment would b
-00005cb0: 6520 636f 6e64 7563 7465 640a 2d61 6466  e conducted.-adf
-00005cc0: 2c20 2d2d 6164 6a75 7374 6661 6374 6f72  , --adjustfactor
-00005cd0: 2c20 7468 6520 6164 6a75 7374 6d65 6e74  , the adjustment
-00005ce0: 2066 6163 746f 7220 6f66 206f 7274 686f   factor of ortho
-00005cf0: 6c6f 676f 7573 204b 732c 2064 6566 6175  logous Ks, defau
-00005d00: 6c74 2030 2e35 0a2d 6f61 2c20 2d2d 6f6b  lt 0.5.-oa, --ok
-00005d10: 616c 7068 612c 2074 6865 206f 7061 6369  alpha, the opaci
-00005d20: 7479 206f 6620 6f72 7468 6f6c 6f67 6f75  ty of orthologou
-00005d30: 7320 4b73 2064 6973 7472 6962 7574 696f  s Ks distributio
-00005d40: 6e20 696e 206d 6978 6564 2070 6c6f 742c  n in mixed plot,
-00005d50: 2064 6566 6175 6c74 2030 2e35 0a2d 6661   default 0.5.-fa
-00005d60: 2c20 2d2d 666f 6375 7332 616c 6c2c 2073  , --focus2all, s
-00005d70: 6574 2066 6f63 616c 2073 7065 6369 6573  et focal species
-00005d80: 2061 6e64 206c 6574 2073 7065 6369 6573   and let species
-00005d90: 2070 6169 7220 746f 2062 6520 6265 7477   pair to be betw
-00005da0: 6565 6e20 666f 6361 6c20 616e 6420 616c  een focal and al
-00005db0: 6c20 7468 6520 7265 6d61 696e 696e 6720  l the remaining 
-00005dc0: 7370 6563 6965 732c 2064 6566 6175 6c74  species, default
-00005dd0: 204e 6f6e 650a 2d6b 732c 202d 2d6b 7374   None.-ks, --kst
-00005de0: 7265 652c 2066 6c61 6720 6f70 7469 6f6e  ree, flag option
-00005df0: 2c20 7768 6574 6865 7220 746f 2069 6e66  , whether to inf
-00005e00: 6572 204b 7320 7472 6565 2c20 6966 2074  er Ks tree, if t
-00005e10: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00005e20: 2074 6865 204b 7320 7472 6565 2069 6e66   the Ks tree inf
-00005e30: 6572 656e 6365 2061 6e61 6c79 7369 7320  erence analysis 
-00005e40: 776f 756c 6420 6265 2069 6e69 7469 6174  would be initiat
-00005e50: 6564 0a2d 6f63 6b2c 202d 2d6f 6e6c 7963  ed.-ock, --onlyc
-00005e60: 6f6e 6361 746b 7374 7265 652c 2066 6c61  oncatkstree, fla
-00005e70: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00005e80: 7220 746f 206f 6e6c 7920 696e 6665 7220  r to only infer 
-00005e90: 4b73 2074 7265 6520 756e 6465 7220 636f  Ks tree under co
-00005ea0: 6e63 6174 656e 6174 6564 2061 6c69 676e  ncatenated align
-00005eb0: 6d65 6e74 2c20 6966 2074 6865 2066 6c61  ment, if the fla
-00005ec0: 6720 7761 7320 7365 742c 206f 6e6c 7920  g was set, only 
-00005ed0: 7468 6520 4b73 2074 7265 6520 756e 6465  the Ks tree unde
-00005ee0: 7220 636f 6e63 6174 656e 6174 6564 2061  r concatenated a
-00005ef0: 6c69 676e 6d65 6e74 2077 6f75 6c64 2062  lignment would b
-00005f00: 6520 6361 6c63 756c 6174 6564 0a2d 6373  e calculated.-cs
-00005f10: 2c20 2d2d 636c 6173 7369 632c 2066 6c61  , --classic, fla
-00005f20: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00005f30: 7220 746f 2064 7261 7720 6d69 7865 6420  r to draw mixed 
-00005f40: 4b73 2070 6c6f 7420 696e 2061 2063 6c61  Ks plot in a cla
-00005f50: 7373 6963 206d 616e 6e65 7220 7768 6572  ssic manner wher
-00005f60: 6520 7468 6520 6675 6c6c 206f 7274 686f  e the full ortho
-00005f70: 6c6f 676f 7573 204b 7320 6469 7374 7269  logous Ks distri
-00005f80: 6275 7469 6f6e 2069 7320 6472 6177 6564  bution is drawed
-00005f90: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00005fa0: 7320 7365 742c 2074 6865 2063 6c61 7373  s set, the class
-00005fb0: 6963 206d 6978 6564 204b 7320 706c 6f74  ic mixed Ks plot
-00005fc0: 2077 6f75 6c64 2062 6520 6472 6177 6e0a   would be drawn.
-00005fd0: 2d74 612c 202d 2d74 6f70 6172 726f 772c  -ta, --toparrow,
-00005fe0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00005ff0: 6574 6865 7220 746f 2061 646a 7573 7420  ether to adjust 
-00006000: 7468 6520 6172 726f 7720 746f 2062 6520  the arrow to be 
-00006010: 6174 2074 6865 2074 6f70 206f 6620 7468  at the top of th
-00006020: 6520 706c 6f74 2c20 696e 7374 6561 6420  e plot, instead 
-00006030: 6f66 2062 6569 6e67 2063 6f6f 7264 696e  of being coordin
-00006040: 6174 6564 2061 7320 7468 6520 4b44 4520  ated as the KDE 
-00006050: 6f66 2074 6865 206f 7274 686f 6c6f 676f  of the orthologo
-00006060: 7573 204b 7320 6469 7374 7269 6275 7469  us Ks distributi
-00006070: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
-00006080: 7761 7320 7365 742c 2074 6865 2061 7272  was set, the arr
-00006090: 6f77 2077 6f75 6c64 2062 6520 7365 7420  ow would be set 
-000060a0: 6174 2074 6865 2074 6f70 0a2d 6273 2c20  at the top.-bs, 
-000060b0: 2d2d 626f 6f74 7374 7261 702c 2074 6865  --bootstrap, the
-000060c0: 206e 756d 6265 7220 6f66 2062 6f6f 7473   number of boots
-000060d0: 7472 6170 2072 6570 6c69 6361 7465 7320  trap replicates 
-000060e0: 6f66 206f 7274 686f 6c6f 6720 4b73 2064  of ortholog Ks d
-000060f0: 6973 7472 6962 7574 696f 6e20 696e 206d  istribution in m
-00006100: 6978 6564 2070 6c6f 740a 6060 600a 0a54  ixed plot.```..T
-00006110: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-00006120: 6d69 7860 2063 616e 2072 6561 6c69 7a65  mix` can realize
-00006130: 2074 6865 206d 6978 7475 7265 206d 6f64   the mixture mod
-00006140: 656c 2063 6c75 7374 6572 696e 6720 616e  el clustering an
-00006150: 616c 7973 6973 206f 6620 2a4b 2a3c 7375  alysis of *K*<su
-00006160: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-00006170: 7374 7269 6275 7469 6f6e 2e0a 6060 600a  stribution..```.
-00006180: 7767 6420 6d69 7820 6b73 5f64 6174 6166  wgd mix ks_dataf
-00006190: 696c 6520 286f 7074 696f 6e29 0a2d 2d2d  ile (option).---
-000061a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000061b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000061c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000061d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000061e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d0a 2d66  -------------.-f
-000061f0: 2c20 2d2d 6669 6c74 6572 732c 2074 6865  , --filters, the
-00006200: 2063 7574 6f66 6620 616c 6967 6e6d 656e   cutoff alignmen
-00006210: 7420 6c65 6e67 7468 2c20 6465 6661 756c  t length, defaul
-00006220: 7420 3330 300a 2d72 2c20 2d2d 6b73 5f72  t 300.-r, --ks_r
-00006230: 616e 6765 2c20 7468 6520 4b73 2072 616e  ange, the Ks ran
-00006240: 6765 2074 6f20 6265 2063 6f6e 7369 6465  ge to be conside
-00006250: 7265 642c 2064 6566 6175 6c74 2028 302c  red, default (0,
-00006260: 2035 290a 2d62 2c20 2d2d 6269 6e73 2c20   5).-b, --bins, 
-00006270: 7468 6520 6e75 6d62 6572 206f 6620 6269  the number of bi
-00006280: 6e73 2069 6e20 4b73 2064 6973 7472 6962  ns in Ks distrib
-00006290: 7574 696f 6e2c 2064 6566 6175 6c74 2035  ution, default 5
-000062a0: 300a 2d6f 2c20 2d2d 6f75 7464 6972 2c20  0.-o, --outdir, 
-000062b0: 7468 6520 6f75 7470 7574 2064 6972 6563  the output direc
-000062c0: 746f 7279 2c20 6465 6661 756c 7420 7767  tory, default wg
-000062d0: 645f 6d69 780a 2d2d 6d65 7468 6f64 2c20  d_mix.--method, 
-000062e0: 7768 6963 6820 6d69 7874 7572 6520 6d6f  which mixture mo
-000062f0: 6465 6c20 746f 2075 7365 2c20 6465 6661  del to use, defa
-00006300: 756c 7420 676d 6d0a 2d6e 2c20 2d2d 636f  ult gmm.-n, --co
-00006310: 6d70 6f6e 656e 7473 2c20 7468 6520 7261  mponents, the ra
-00006320: 6e67 6520 6f66 2074 6865 206e 756d 6265  nge of the numbe
-00006330: 7220 6f66 2063 6f6d 706f 6e65 6e74 7320  r of components 
-00006340: 746f 2066 6974 2c20 6465 6661 756c 7420  to fit, default 
-00006350: 2831 2c20 3429 0a2d 672c 202d 2d67 616d  (1, 4).-g, --gam
-00006360: 6d61 2c20 7468 6520 6761 6d6d 6120 7061  ma, the gamma pa
-00006370: 7261 6d65 7465 7220 666f 7220 6267 6d6d  rameter for bgmm
-00006380: 206d 6f64 656c 732c 2064 6566 6175 6c74   models, default
-00006390: 2030 2e30 3031 0a2d 6e69 2c20 2d2d 6e5f   0.001.-ni, --n_
-000063a0: 696e 6974 2c20 7468 6520 6e75 6d62 6572  init, the number
-000063b0: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
-000063c0: 6961 6c69 7a61 7469 6f6e 732c 2064 6566  ializations, def
-000063d0: 6175 6c74 2032 3030 0a2d 6d69 2c20 2d2d  ault 200.-mi, --
-000063e0: 6d61 785f 6974 6572 2c20 7468 6520 6d61  max_iter, the ma
-000063f0: 7869 6d75 6d20 6e75 6d62 6572 206f 6620  ximum number of 
-00006400: 6974 6572 6174 696f 6e73 2c20 6465 6661  iterations, defa
-00006410: 756c 7420 3230 300a 6060 600a 0a54 6865  ult 200.```..The
-00006420: 2070 726f 6772 616d 2060 7767 6420 7065   program `wgd pe
-00006430: 616b 6020 6361 6e20 7265 616c 697a 6520  ak` can realize 
-00006440: 7468 6520 7365 6172 6368 206f 6620 6372  the search of cr
-00006450: 6564 6961 626c 6520 2a4b 2a3c 7375 623e  ediable *K*<sub>
-00006460: 533c 2f73 7562 3e20 7261 6e67 6520 7573  S</sub> range us
-00006470: 6564 2069 6e20 5747 4420 6461 7469 6e67  ed in WGD dating
-00006480: 2e0a 6060 600a 7767 6420 7065 616b 206b  ..```.wgd peak k
-00006490: 735f 6461 7461 6669 6c65 2028 6f70 7469  s_datafile (opti
-000064a0: 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  on).------------
-000064b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000064c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000064d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000064e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000064f0: 2d2d 2d2d 0a2d 6170 2c20 2d2d 616e 6368  ----.-ap, --anch
-00006500: 6f72 706f 696e 7473 2c20 7468 6520 616e  orpoints, the an
-00006510: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
-00006520: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
-00006530: 6e65 0a2d 736d 2c20 2d2d 7365 676d 656e  ne.-sm, --segmen
-00006540: 7473 2c20 7468 6520 7365 676d 656e 7473  ts, the segments
-00006550: 2064 6174 6166 696c 652c 2064 6566 6175   datafile, defau
-00006560: 6c74 204e 6f6e 650a 2d6c 652c 202d 2d6c  lt None.-le, --l
-00006570: 6973 7465 6c65 6d65 6e74 732c 2074 6865  istelements, the
-00006580: 206c 6973 7473 6567 6d65 6e74 7320 6461   listsegments da
-00006590: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
-000065a0: 4e6f 6e65 200a 2d6d 702c 202d 2d6d 756c  None .-mp, --mul
-000065b0: 7469 706c 6963 6f6e 7061 6972 732c 2074  tipliconpairs, t
-000065c0: 6865 206d 756c 7469 706c 6963 6f6e 7061  he multipliconpa
-000065d0: 6972 7320 6461 7461 6669 6c65 2c20 6465  irs datafile, de
-000065e0: 6661 756c 7420 4e6f 6e65 0a2d 6f2c 202d  fault None.-o, -
-000065f0: 2d6f 7574 6469 722c 2074 6865 206f 7574  -outdir, the out
-00006600: 7075 7420 6469 7265 6374 6f72 792c 2064  put directory, d
-00006610: 6566 6175 6c74 2077 6764 5f70 6561 6b0a  efault wgd_peak.
-00006620: 2d61 662c 202d 2d61 6c69 676e 6669 6c74  -af, --alignfilt
-00006630: 6572 2c20 6375 746f 6666 2066 6f72 2061  er, cutoff for a
-00006640: 6c69 676e 6d65 6e74 2069 6465 6e74 6974  lignment identit
-00006650: 792c 206c 656e 6774 6820 616e 6420 636f  y, length and co
-00006660: 7665 7261 6765 2c20 6465 6661 756c 7420  verage, default 
-00006670: 302e 302c 2030 2c20 302e 300a 2d72 2c20  0.0, 0, 0.0.-r, 
-00006680: 2d2d 6b73 7261 6e67 652c 2072 616e 6765  --ksrange, range
-00006690: 206f 6620 4b73 2074 6f20 6265 2061 6e61   of Ks to be ana
-000066a0: 6c79 7a65 642c 2064 6566 6175 6c74 2028  lyzed, default (
-000066b0: 302c 2035 290a 2d62 772c 202d 2d62 696e  0, 5).-bw, --bin
-000066c0: 5f77 6964 7468 2c20 6261 6e64 7769 6474  _width, bandwidt
-000066d0: 6820 6f66 204b 7320 6469 7374 7269 6275  h of Ks distribu
-000066e0: 7469 6f6e 2c20 6465 6661 756c 7420 302e  tion, default 0.
-000066f0: 310a 2d69 632c 202d 2d77 6569 6768 7473  1.-ic, --weights
-00006700: 5f6f 7574 6c69 6572 735f 696e 636c 7564  _outliers_includ
-00006710: 6564 2c20 666c 6167 206f 7074 696f 6e2c  ed, flag option,
-00006720: 2077 6865 7468 6572 2074 6f20 696e 636c   whether to incl
-00006730: 7564 6520 4b73 206f 7574 6c69 6572 732c  ude Ks outliers,
-00006740: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00006750: 2073 6574 2c20 4b73 206f 7574 6c69 6572   set, Ks outlier
-00006760: 7320 776f 756c 6420 6265 2069 6e63 6c75  s would be inclu
-00006770: 6465 6420 696e 2074 6865 2061 6e61 6c79  ded in the analy
-00006780: 7369 730a 2d6d 2c20 2d2d 6d65 7468 6f64  sis.-m, --method
-00006790: 2c20 7768 6963 6820 6d69 7874 7572 6520  , which mixture 
-000067a0: 6d6f 6465 6c20 746f 2075 7365 2c20 6465  model to use, de
-000067b0: 6661 756c 7420 676d 6d0a 2d2d 7365 6564  fault gmm.--seed
-000067c0: 2c20 7261 6e64 6f6d 2073 6565 6420 6769  , random seed gi
-000067d0: 7665 6e20 746f 2069 6e69 7469 616c 697a  ven to initializ
-000067e0: 6174 696f 6e2c 2064 6566 6175 6c74 2032  ation, default 2
-000067f0: 3335 3238 3930 0a2d 6569 2c20 2d2d 656d  352890.-ei, --em
-00006800: 5f69 7465 722c 2074 6865 206e 756d 6265  _iter, the numbe
-00006810: 7220 6f66 2045 4d20 6974 6572 6174 696f  r of EM iteratio
-00006820: 6e73 2074 6f20 7065 7266 6f72 6d2c 2064  ns to perform, d
-00006830: 6566 6175 6c74 2032 3030 0a2d 6e69 2c20  efault 200.-ni, 
-00006840: 2d2d 6e5f 696e 6974 2c20 7468 6520 6e75  --n_init, the nu
-00006850: 6d62 6572 206f 6620 6b2d 6d65 616e 7320  mber of k-means 
-00006860: 696e 6974 6961 6c69 7a61 7469 6f6e 732c  initializations,
-00006870: 2064 6566 6175 6c74 2032 3030 0a2d 6e2c   default 200.-n,
-00006880: 202d 2d63 6f6d 706f 6e65 6e74 732c 2074   --components, t
-00006890: 6865 2072 616e 6765 206f 6620 7468 6520  he range of the 
-000068a0: 6e75 6d62 6572 206f 6620 636f 6d70 6f6e  number of compon
-000068b0: 656e 7473 2074 6f20 6669 742c 2064 6566  ents to fit, def
-000068c0: 6175 6c74 2028 312c 2034 290a 2d67 2c20  ault (1, 4).-g, 
-000068d0: 2d2d 6761 6d6d 612c 2074 6865 2067 616d  --gamma, the gam
-000068e0: 6d61 2070 6172 616d 6574 6572 2066 6f72  ma parameter for
-000068f0: 2062 676d 6d20 6d6f 6465 6c73 2c20 6465   bgmm models, de
-00006900: 6661 756c 7420 3165 2d33 0a2d 2d62 6f6f  fault 1e-3.--boo
-00006910: 7473 2c20 7468 6520 6e75 6d62 6572 206f  ts, the number o
-00006920: 6620 626f 6f74 7374 7261 7020 7265 706c  f bootstrap repl
-00006930: 6963 6174 6573 206f 6620 6b64 652c 2064  icates of kde, d
-00006940: 6566 6175 6c74 2032 3030 0a2d 2d77 6569  efault 200.--wei
-00006950: 6768 7465 642c 2066 6c61 6720 6f70 7469  ghted, flag opti
-00006960: 6f6e 2c20 7768 6574 6865 7220 746f 2075  on, whether to u
-00006970: 7365 206e 6f64 652d 7765 6967 6874 6564  se node-weighted
-00006980: 206d 6574 686f 6420 6f66 2064 652d 7265   method of de-re
-00006990: 6475 6e64 616e 6379 2c20 6966 2074 6865  dundancy, if the
-000069a0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-000069b0: 6865 206e 6f64 652d 7765 6967 6874 6564  he node-weighted
-000069c0: 206d 6574 686f 6420 776f 756c 6420 6265   method would be
-000069d0: 2075 7365 640a 2d70 2c20 2d2d 706c 6f74   used.-p, --plot
-000069e0: 2c20 7468 6520 706c 6f74 7469 6e67 206d  , the plotting m
-000069f0: 6574 686f 6420 746f 2062 6520 7573 6564  ethod to be used
-00006a00: 2c20 6465 6661 756c 7420 6964 656e 7469  , default identi
-00006a10: 6361 6c0a 2d62 6d2c 202d 2d62 775f 6d65  cal.-bm, --bw_me
-00006a20: 7468 6f64 2c20 7468 6520 6261 6e64 7769  thod, the bandwi
-00006a30: 6474 6820 6d65 7468 6f64 2074 6f20 6265  dth method to be
-00006a40: 2075 7365 6420 696e 2061 6e61 6c79 7a69   used in analyzi
-00006a50: 6e67 2074 6865 2070 6561 6b20 6f66 2057  ng the peak of W
-00006a60: 4744 2064 6174 6573 2c20 6465 6661 756c  GD dates, defaul
-00006a70: 7420 7369 6c76 6572 6d61 6e0a 2d2d 6e5f  t silverman.--n_
-00006a80: 6d65 646f 6964 732c 2074 6865 206e 756d  medoids, the num
-00006a90: 6265 7220 6f66 206d 6564 6f69 6473 2074  ber of medoids t
-00006aa0: 6f20 6669 742c 2064 6566 6175 6c74 2032  o fit, default 2
-00006ab0: 0a2d 6b6d 2c20 2d2d 6b64 656d 6574 686f  .-km, --kdemetho
-00006ac0: 642c 2074 6865 206b 6465 206d 6574 686f  d, the kde metho
-00006ad0: 6420 746f 2062 6520 7573 6564 2069 6e20  d to be used in 
-00006ae0: 616e 616c 797a 696e 6720 7468 6520 7065  analyzing the pe
-00006af0: 616b 206f 6620 5747 4420 6461 7465 732c  ak of WGD dates,
-00006b00: 206b 6d65 646f 6964 7320 616e 616c 7973   kmedoids analys
-00006b10: 6973 206f 7220 7468 6520 6261 7369 6320  is or the basic 
-00006b20: 4b73 2070 6c6f 7474 696e 672c 2064 6566  Ks plotting, def
-00006b30: 6175 6c74 2073 6369 7079 0a2d 2d6e 5f63  ault scipy.--n_c
-00006b40: 6c75 7374 6572 732c 2074 6865 206e 756d  lusters, the num
-00006b50: 6265 7220 6f66 2063 6c75 7374 6572 7320  ber of clusters 
-00006b60: 746f 2070 6c6f 7420 456c 626f 7720 6c6f  to plot Elbow lo
-00006b70: 7373 2066 756e 6374 696f 6e2c 2064 6566  ss function, def
-00006b80: 6175 6c74 2035 0a2d 6764 2c20 2d2d 6775  ault 5.-gd, --gu
-00006b90: 6964 652c 2074 6865 2072 6567 696d 6520  ide, the regime 
-00006ba0: 7265 7369 6469 6e67 2061 6e63 686f 7273  residing anchors
-00006bb0: 2c20 6465 6661 756c 7420 5365 676d 656e  , default Segmen
-00006bc0: 740a 2d70 7263 742c 202d 2d70 726f 6d69  t.-prct, --promi
-00006bd0: 6e65 6e63 655f 6375 746f 6666 2c20 7468  nence_cutoff, th
-00006be0: 6520 7072 6f6d 696e 656e 6365 2063 7574  e prominence cut
-00006bf0: 6f66 6620 6f66 2061 6363 6570 7461 626c  off of acceptabl
-00006c00: 6520 7065 616b 7320 696e 2070 6561 6b20  e peaks in peak 
-00006c10: 6669 6e64 696e 6720 7374 6570 732c 2064  finding steps, d
-00006c20: 6566 6175 6c74 2030 2e31 0a2d 7268 2c20  efault 0.1.-rh, 
-00006c30: 2d2d 7265 6c5f 6865 6967 6874 2c20 7468  --rel_height, th
-00006c40: 6520 7265 6c61 7469 7665 2068 6569 6768  e relative heigh
-00006c50: 7420 6174 2077 6869 6368 2074 6865 2070  t at which the p
-00006c60: 6561 6b20 7769 6474 6820 6973 206d 6561  eak width is mea
-00006c70: 7375 7265 642c 2064 6566 6175 6c74 2030  sured, default 0
-00006c80: 2e34 0a2d 6b64 2c20 2d2d 6b73 746f 6461  .4.-kd, --kstoda
-00006c90: 7465 2c20 7468 6520 7261 6e67 6520 6f66  te, the range of
-00006ca0: 204b 7320 746f 2062 6520 6461 7465 6420   Ks to be dated 
-00006cb0: 696e 2068 6575 7269 7374 6963 2073 6561  in heuristic sea
-00006cc0: 7263 682c 2064 6566 6175 6c74 2028 302e  rch, default (0.
-00006cd0: 352c 2031 2e35 290a 2d78 6c2c 202d 2d78  5, 1.5).-xl, --x
-00006ce0: 6c69 6d2c 2074 6865 2078 2061 7869 7320  lim, the x axis 
-00006cf0: 6c69 6d69 7420 6f66 2047 4d4d 204b 7320  limit of GMM Ks 
-00006d00: 6469 7374 7269 6275 7469 6f6e 0a2d 796c  distribution.-yl
-00006d10: 2c20 2d2d 796c 696d 2c20 7468 6520 7920  , --ylim, the y 
-00006d20: 6178 6973 206c 696d 6974 206f 6620 474d  axis limit of GM
-00006d30: 4d20 4b73 2064 6973 7472 6962 7574 696f  M Ks distributio
-00006d40: 6e0a 2d2d 6d61 6e75 616c 7365 742c 2066  n.--manualset, f
-00006d50: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00006d60: 6865 7220 746f 206f 7574 7075 7420 616e  her to output an
-00006d70: 6368 6f72 2070 6169 7273 2077 6974 6820  chor pairs with 
-00006d80: 6d61 6e75 616c 6c79 2073 6574 204b 7320  manually set Ks 
-00006d90: 7261 6e67 652c 2069 6620 7468 6520 666c  range, if the fl
-00006da0: 6167 2077 6173 2073 6574 2c20 6d61 6e75  ag was set, manu
-00006db0: 616c 6c79 2073 6574 204b 7320 7261 6e67  ally set Ks rang
-00006dc0: 6520 776f 756c 6420 6265 206f 7574 7075  e would be outpu
-00006dd0: 7474 6564 0a2d 2d63 692c 2074 6865 2063  tted.--ci, the c
-00006de0: 6f6e 6669 6465 6e63 6520 6c65 7665 6c20  onfidence level 
-00006df0: 6f66 206c 6f67 2d6e 6f72 6d61 6c20 6469  of log-normal di
-00006e00: 7374 7269 6275 7469 6f6e 2074 6f20 6461  stribution to da
-00006e10: 7465 2c20 6465 6661 756c 7420 3935 0a2d  te, default 95.-
-00006e20: 2d68 6472 2c20 7468 6520 6869 6768 6573  -hdr, the highes
-00006e30: 7420 6465 6e73 6974 7920 7265 6769 6f6e  t density region
-00006e40: 2028 4844 5229 2061 7070 6c69 6564 2069   (HDR) applied i
-00006e50: 6e20 7468 6520 7365 676d 656e 742d 6775  n the segment-gu
-00006e60: 6964 6564 2061 6e63 686f 7220 7061 6972  ided anchor pair
-00006e70: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00006e80: 2c20 6465 6661 756c 7420 3935 0a2d 2d68  , default 95.--h
-00006e90: 6575 7269 7374 6963 2c20 666c 6167 206f  euristic, flag o
-00006ea0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-00006eb0: 6f20 696e 6974 6961 7465 2068 6575 7269  o initiate heuri
-00006ec0: 7374 6963 206d 6574 686f 6420 6f66 2064  stic method of d
-00006ed0: 6566 696e 696e 6720 4349 2066 6f72 2064  efining CI for d
-00006ee0: 6174 696e 672c 2069 6620 7468 6520 666c  ating, if the fl
-00006ef0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
-00006f00: 6865 7572 6973 7469 6320 6d65 7468 6f64  heuristic method
-00006f10: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00006f20: 7465 640a 2d6b 632c 202d 2d6b 7363 7574  ted.-kc, --kscut
-00006f30: 6f66 662c 2074 6865 204b 7320 7361 7475  off, the Ks satu
-00006f40: 7261 7469 6f6e 2063 7574 6f66 6620 696e  ration cutoff in
-00006f50: 2064 6174 696e 672c 2064 6566 6175 6c74   dating, default
-00006f60: 2035 0a2d 2d6b 6565 7074 6d70 6669 672c   5.--keeptmpfig,
-00006f70: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00006f80: 6574 6865 7220 746f 206b 6565 7020 7465  ether to keep te
-00006f90: 6d70 6f72 6172 7920 6669 6775 7265 7320  mporary figures 
-00006fa0: 696e 2070 6561 6b20 6669 6e64 696e 6720  in peak finding 
-00006fb0: 7072 6f63 6573 732c 2069 6620 7468 6520  process, if the 
-00006fc0: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
-00006fd0: 6f73 6520 6669 6775 7265 7320 776f 756c  ose figures woul
-00006fe0: 6420 6265 206b 6570 740a 6060 600a 0a54  d be kept.```..T
-00006ff0: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-00007000: 7379 6e60 2063 616e 2072 6561 6c69 7a65  syn` can realize
-00007010: 2074 6865 2069 6e74 7261 2d20 616e 6420   the intra- and 
-00007020: 696e 7465 722d 7370 6563 6966 6963 2073  inter-specific s
-00007030: 796e 7465 6e79 2069 6e66 6572 656e 6365  ynteny inference
-00007040: 2e0a 6060 600a 7767 6420 7379 6e20 6661  ..```.wgd syn fa
-00007050: 6d69 6c69 6573 2067 6666 7320 286f 7074  milies gffs (opt
-00007060: 696f 6e29 0a2d 2d2d 2d2d 2d2d 2d2d 2d2d  ion).-----------
-00007070: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007080: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007090: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000070a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000070b0: 2d2d 2d2d 2d0a 2d6b 732c 202d 2d6b 735f  -----.-ks, --ks_
-000070c0: 6469 7374 7269 6275 7469 6f6e 2c20 6b73  distribution, ks
-000070d0: 2064 6973 7472 6962 7574 696f 6e20 6461   distribution da
-000070e0: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
-000070f0: 4e6f 6e65 0a2d 6f2c 202d 2d6f 7574 6469  None.-o, --outdi
-00007100: 722c 2074 6865 206f 7574 7075 7420 6469  r, the output di
-00007110: 7265 6374 6f72 792c 2064 6566 6175 6c74  rectory, default
-00007120: 2077 6764 5f73 796e 0a2d 662c 202d 2d66   wgd_syn.-f, --f
-00007130: 6561 7475 7265 2c20 7468 6520 6665 6174  eature, the feat
-00007140: 7572 6520 666f 7220 7061 7273 696e 6720  ure for parsing 
-00007150: 6765 6e65 2049 4473 2066 726f 6d20 4746  gene IDs from GF
-00007160: 4620 6669 6c65 732c 2064 6566 6175 6c74  F files, default
-00007170: 2067 656e 650a 2d61 2c20 2d2d 6174 7472   gene.-a, --attr
-00007180: 6962 7574 652c 2074 6865 2061 7474 7269  ibute, the attri
-00007190: 6275 7465 2066 6f72 2070 6172 7369 6e67  bute for parsing
-000071a0: 2074 6865 2067 656e 6520 4944 7320 6672   the gene IDs fr
-000071b0: 6f6d 2074 6865 2047 4646 2066 696c 6573  om the GFF files
-000071c0: 2c20 6465 6661 756c 7420 4944 0a2d 6174  , default ID.-at
-000071d0: 672c 202d 2d61 6464 6974 696f 6e61 6c67  g, --additionalg
-000071e0: 6666 696e 666f 2c20 7468 6520 6665 6174  ffinfo, the feat
-000071f0: 7572 6520 616e 6420 6174 7472 6962 7574  ure and attribut
-00007200: 6520 696e 666f 726d 6174 696f 6e20 6f66  e information of
-00007210: 2061 6464 6974 696f 6e61 6c20 6766 6633   additional gff3
-00007220: 2066 696c 6573 2069 6620 6469 6666 6572   files if differ
-00007230: 656e 7420 696e 2074 6865 2066 6f72 6d61  ent in the forma
-00007240: 7420 6f66 2028 6665 6174 7572 653b 6174  t of (feature;at
-00007250: 7472 6962 7574 6529 272c 2064 6566 6175  tribute)', defau
-00007260: 6c74 204e 6f6e 650a 2d6d 6c2c 202d 2d6d  lt None.-ml, --m
-00007270: 696e 6c65 6e2c 2074 6865 206d 696e 696d  inlen, the minim
-00007280: 756d 206c 656e 6774 6820 6f66 2061 2073  um length of a s
-00007290: 6361 6666 6f6c 6420 746f 2062 6520 696e  caffold to be in
-000072a0: 636c 7564 6564 2069 6e20 646f 7470 6c6f  cluded in dotplo
-000072b0: 742c 2064 6566 6175 6c74 202d 312c 2069  t, default -1, i
-000072c0: 6620 2d31 2077 6173 2073 6574 2c20 7468  f -1 was set, th
-000072d0: 6520 3130 2520 6f66 2074 6865 206c 6f6e  e 10% of the lon
-000072e0: 6765 7374 2073 6361 6666 6f6c 6420 776f  gest scaffold wo
-000072f0: 756c 6420 6265 2073 6574 0a2d 6d73 2c20  uld be set.-ms, 
-00007300: 2d2d 6d61 7873 697a 652c 2074 6865 206d  --maxsize, the m
-00007310: 6178 696d 756d 2066 616d 696c 7920 7369  aximum family si
-00007320: 7a65 2074 6f20 6265 2069 6e63 6c75 6465  ze to be include
-00007330: 642c 2064 6566 6175 6c74 2032 3030 0a2d  d, default 200.-
-00007340: 722c 202d 2d6b 735f 7261 6e67 652c 2074  r, --ks_range, t
-00007350: 6865 204b 7320 7261 6e67 6520 696e 2063  he Ks range in c
-00007360: 6f6c 6f72 6564 2064 6f74 706c 6f74 2c20  olored dotplot, 
-00007370: 6465 6661 756c 7420 2830 2c20 3529 0a2d  default (0, 5).-
-00007380: 2d69 6164 686f 7265 5f6f 7074 696f 6e73  -iadhore_options
-00007390: 2c20 7468 6520 7061 7261 6d65 7465 7220  , the parameter 
-000073a0: 7365 7474 696e 6720 696e 2069 6164 686f  setting in iadho
-000073b0: 7265 2c20 6465 6661 756c 7420 6173 2061  re, default as a
-000073c0: 2073 7472 696e 6720 6f66 206c 656e 6774   string of lengt
-000073d0: 6820 7a65 726f 0a2d 6d67 2c20 2d2d 6d69  h zero.-mg, --mi
-000073e0: 6e73 6567 6c65 6e2c 2074 6865 206d 696e  nseglen, the min
-000073f0: 696d 756d 206c 656e 6774 6820 6f66 2073  imum length of s
-00007400: 6567 6d65 6e74 7320 746f 2069 6e63 6c75  egments to inclu
-00007410: 6465 2069 6e20 7261 7469 6f20 6966 203c  de in ratio if <
-00007420: 3d20 312c 2064 6566 6175 6c74 2031 3030  = 1, default 100
-00007430: 3030 0a2d 6b72 2c20 2d2d 6b65 6570 7265  00.-kr, --keepre
-00007440: 6475 6e2c 2066 6c61 6720 6f70 7469 6f6e  dun, flag option
-00007450: 2c20 7768 6574 6865 7220 746f 206b 6565  , whether to kee
-00007460: 7020 7265 6475 6e64 616e 7420 6d75 6c74  p redundant mult
-00007470: 6970 6c69 636f 6e73 2c20 6966 2074 6865  iplicons, if the
-00007480: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00007490: 6865 2072 6564 756e 6461 6e74 206d 756c  he redundant mul
-000074a0: 7469 706c 6963 6f6e 7320 776f 756c 6420  tiplicons would 
-000074b0: 6265 206b 6570 740a 2d6d 676e 2c20 2d2d  be kept.-mgn, --
-000074c0: 6d69 6e67 656e 656e 756d 2c20 7468 6520  mingenenum, the 
-000074d0: 6d69 6e69 6d75 6d20 6e75 6d62 6572 206f  minimum number o
-000074e0: 6620 6765 6e65 7320 666f 7220 6120 7365  f genes for a se
-000074f0: 676d 656e 7420 746f 2062 6520 636f 6e73  gment to be cons
-00007500: 6964 6572 6564 2c20 6465 6661 756c 7420  idered, default 
-00007510: 3330 0a2d 6473 2c20 2d2d 646f 7473 697a  30.-ds, --dotsiz
-00007520: 652c 2074 6865 2064 6f74 2073 697a 6520  e, the dot size 
-00007530: 696e 2064 6f74 2070 6c6f 742c 2064 6566  in dot plot, def
-00007540: 6175 6c74 2030 2e33 0a2d 6161 2c20 2d2d  ault 0.3.-aa, --
-00007550: 6170 616c 7068 612c 2074 6865 206f 7061  apalpha, the opa
-00007560: 6369 7479 206f 6620 616e 6368 6f72 2064  city of anchor d
-00007570: 6f74 732c 2064 6566 6175 6c74 2031 0a2d  ots, default 1.-
-00007580: 6861 2c20 2d2d 686f 616c 7068 612c 2074  ha, --hoalpha, t
-00007590: 6865 206f 7061 6369 7479 206f 6620 686f  he opacity of ho
-000075a0: 6d6f 6c6f 6720 646f 7473 2c20 6465 6661  molog dots, defa
-000075b0: 756c 7420 300a 2d73 7274 2c20 2d2d 7368  ult 0.-srt, --sh
-000075c0: 6f77 7265 616c 7469 636b 2c20 666c 6167  owrealtick, flag
-000075d0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-000075e0: 2074 6f20 7368 6f77 2074 6865 2072 6561   to show the rea
-000075f0: 6c20 7469 636b 2069 6e20 6765 6e65 7320  l tick in genes 
-00007600: 6f72 2062 6173 6573 2c20 6966 2074 6865  or bases, if the
-00007610: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00007620: 6865 2072 6561 6c20 7469 636b 2077 6f75  he real tick wou
-00007630: 6c64 2062 6520 7368 6f77 6564 0a2d 746c  ld be showed.-tl
-00007640: 732c 202d 2d74 6963 6b6c 6162 656c 7369  s, --ticklabelsi
-00007650: 7a65 2c20 7468 6520 6c61 6265 6c20 7369  ze, the label si
-00007660: 7a65 206f 6620 7469 636b 2c20 6465 6661  ze of tick, defa
-00007670: 756c 7420 350a 2d67 722c 202d 2d67 6973  ult 5.-gr, --gis
-00007680: 7472 622c 2066 6c61 6720 6f70 7469 6f6e  trb, flag option
-00007690: 2c20 7768 6574 6865 7220 746f 2075 7365  , whether to use
-000076a0: 2067 6973 745f 7261 696e 626f 7720 6173   gist_rainbow as
-000076b0: 2063 6f6c 6f72 206d 6170 206f 6620 646f   color map of do
-000076c0: 7470 6c6f 740a 2d6e 2c20 2d2d 6e74 6872  tplot.-n, --nthr
-000076d0: 6561 6473 2c20 7468 6520 6e75 6d62 6572  eads, the number
-000076e0: 206f 6620 7468 7265 6164 7320 746f 2075   of threads to u
-000076f0: 7365 2069 6e20 7379 6e74 656e 7920 696e  se in synteny in
-00007700: 6665 7265 6e63 652c 2064 6566 6175 6c74  ference, default
-00007710: 2034 0a60 6060 0a0a 5468 6520 7072 6f67   4.```..The prog
-00007720: 7261 6d20 6077 6764 2076 697a 6020 6361  ram `wgd viz` ca
-00007730: 6e20 7265 616c 697a 6520 7468 6520 7669  n realize the vi
-00007740: 7375 616c 697a 6174 696f 6e20 6f66 202a  sualization of *
-00007750: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-00007760: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
-00007770: 616e 6420 7379 6e74 656e 792e 0a60 6060  and synteny..```
-00007780: 0a77 6764 2076 697a 2028 6f70 7469 6f6e  .wgd viz (option
-00007790: 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ).--------------
-000077a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000077b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000077c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000077d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000077e0: 2d2d 0a2d 642c 202d 2d64 6174 6166 696c  --.-d, --datafil
-000077f0: 652c 2074 6865 204b 7320 6461 7461 6669  e, the Ks datafi
-00007800: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
-00007810: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
-00007820: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
-00007830: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
-00007840: 5f76 697a 0a2d 7372 2c20 2d2d 7370 6169  _viz.-sr, --spai
-00007850: 722c 2074 6865 2073 7065 6369 6573 2070  r, the species p
-00007860: 6169 7220 746f 2062 6520 706c 6f74 7465  air to be plotte
-00007870: 642c 2064 6566 6175 6c74 204e 6f6e 652c  d, default None,
-00007880: 2074 6869 7320 6f70 7469 6f6e 2063 616e   this option can
-00007890: 2062 6520 7072 6f76 6964 6564 206d 756c   be provided mul
-000078a0: 7469 706c 6520 7469 6d65 730a 2d66 612c  tiple times.-fa,
-000078b0: 202d 2d66 6f63 7573 3261 6c6c 2c20 7365   --focus2all, se
-000078c0: 7420 666f 6361 6c20 7370 6563 6965 7320  t focal species 
-000078d0: 616e 6420 6c65 7420 7370 6563 6965 7320  and let species 
-000078e0: 7061 6972 2074 6f20 6265 2062 6574 7765  pair to be betwe
-000078f0: 656e 2066 6f63 616c 2061 6e64 2061 6c6c  en focal and all
-00007900: 2074 6865 2072 656d 6169 6e69 6e67 2073   the remaining s
-00007910: 7065 6369 6573 2c20 6465 6661 756c 7420  pecies, default 
-00007920: 4e6f 6e65 0a2d 6773 2c20 2d2d 6773 6d61  None.-gs, --gsma
-00007930: 702c 2074 6865 2067 656e 6520 6e61 6d65  p, the gene name
-00007940: 2d73 7065 6369 6573 206e 616d 6520 6d61  -species name ma
-00007950: 702c 2064 6566 6175 6c74 204e 6f6e 650a  p, default None.
-00007960: 2d73 702c 202d 2d73 7065 6369 6573 7472  -sp, --speciestr
-00007970: 6565 2c20 7468 6520 7370 6563 6965 7320  ee, the species 
-00007980: 7472 6565 2074 6f20 7065 7266 6f72 6d20  tree to perform 
-00007990: 7261 7465 2063 6f72 7265 6374 696f 6e2c  rate correction,
-000079a0: 2064 6566 6175 6c74 204e 6f6e 652c 2069   default None, i
-000079b0: 6620 4e6f 6e65 2077 6173 2067 6976 656e  f None was given
-000079c0: 2c20 7468 6520 7261 7465 2063 6f72 7265  , the rate corre
-000079d0: 6374 696f 6e20 616e 616c 7973 6973 2077  ction analysis w
-000079e0: 6f75 6c64 2062 6520 6361 6c6c 6564 206f  ould be called o
-000079f0: 6666 0a2d 706b 2c20 2d2d 706c 6f74 6b64  ff.-pk, --plotkd
-00007a00: 652c 2066 6c61 6720 6f70 7469 6f6e 2c20  e, flag option, 
-00007a10: 7768 6574 6865 7220 746f 2070 6c6f 7420  whether to plot 
-00007a20: 6b64 6520 6375 7276 6520 7570 6f6e 2068  kde curve upon h
-00007a30: 6973 746f 6772 616d 2c20 6966 2074 6865  istogram, if the
-00007a40: 2066 6c61 6720 7761 7320 7365 742c 206b   flag was set, k
-00007a50: 6465 2063 7572 7665 2077 6f75 6c64 2062  de curve would b
-00007a60: 6520 6164 6465 640a 2d72 772c 202d 2d72  e added.-rw, --r
-00007a70: 6577 6569 6768 742c 2066 6c61 6720 6f70  eweight, flag op
-00007a80: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00007a90: 2072 6563 616c 6375 6c61 7465 2074 6865   recalculate the
-00007aa0: 2077 6569 6768 7420 7065 7220 7370 6563   weight per spec
-00007ab0: 6965 7320 7061 6972 2c20 6966 2074 6865  ies pair, if the
-00007ac0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00007ad0: 6865 2077 6569 6768 7420 776f 756c 6420  he weight would 
-00007ae0: 6265 2072 6563 616c 6375 6c61 7465 640a  be recalculated.
-00007af0: 2d6f 722c 202d 2d6f 6e6c 7972 6f6f 746f  -or, --onlyrooto
-00007b00: 7574 2c20 666c 6167 206f 7074 696f 6e2c  ut, flag option,
-00007b10: 2077 6865 7468 6572 2074 6f20 6f6e 6c79   whether to only
-00007b20: 2063 6f6e 6475 6374 2072 6174 6520 636f   conduct rate co
-00007b30: 7272 6563 7469 6f6e 2075 7369 6e67 2074  rrection using t
-00007b40: 6865 206f 7574 6772 6f75 7020 6174 2072  he outgroup at r
-00007b50: 6f6f 7420 6173 206f 7574 6772 6f75 702c  oot as outgroup,
-00007b60: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00007b70: 2073 6574 2c20 6f6e 6c79 2074 6865 206f   set, only the o
-00007b80: 7574 6772 6f75 7020 6174 2072 6f6f 7420  utgroup at root 
-00007b90: 776f 756c 6420 6265 2075 7365 6420 6173  would be used as
-00007ba0: 206f 7574 6772 6f75 700a 2d69 7465 722c   outgroup.-iter,
-00007bb0: 202d 2d65 6d5f 6974 6572 6174 696f 6e73   --em_iterations
-00007bc0: 2c20 7468 6520 6d61 7869 6d75 6d20 454d  , the maximum EM
-00007bd0: 2069 7465 7261 7469 6f6e 732c 2064 6566   iterations, def
-00007be0: 6175 6c74 2032 3030 0a2d 696e 6974 2c20  ault 200.-init, 
-00007bf0: 2d2d 656d 5f69 6e69 7469 616c 697a 6174  --em_initializat
-00007c00: 696f 6e73 2c20 7468 6520 6d61 7869 6d75  ions, the maximu
-00007c10: 6d20 454d 2069 6e69 7469 616c 697a 6174  m EM initializat
-00007c20: 696f 6e73 2c20 6465 6661 756c 7420 3230  ions, default 20
-00007c30: 300a 2d70 7263 742c 202d 2d70 726f 6d69  0.-prct, --promi
-00007c40: 6e65 6e63 655f 6375 746f 6666 2c20 7468  nence_cutoff, th
-00007c50: 6520 7072 6f6d 696e 656e 6365 2063 7574  e prominence cut
-00007c60: 6f66 6620 6f66 2061 6363 6570 7461 626c  off of acceptabl
-00007c70: 6520 7065 616b 732c 2064 6566 6175 6c74  e peaks, default
-00007c80: 2030 2e31 0a2d 7268 2c20 2d2d 7265 6c5f   0.1.-rh, --rel_
-00007c90: 6865 6967 6874 2c20 7468 6520 7265 6c61  height, the rela
-00007ca0: 7469 7665 2068 6569 6768 7420 6174 2077  tive height at w
-00007cb0: 6869 6368 2074 6865 2070 6561 6b20 7769  hich the peak wi
-00007cc0: 6474 6820 6973 206d 6561 7375 7265 642c  dth is measured,
-00007cd0: 2064 6566 6175 6c74 2030 2e34 0a2d 736d   default 0.4.-sm
-00007ce0: 2c20 2d2d 7365 676d 656e 7473 2c20 7468  , --segments, th
-00007cf0: 6520 7365 676d 656e 7473 2064 6174 6166  e segments dataf
-00007d00: 696c 652c 2064 6566 6175 6c74 204e 6f6e  ile, default Non
-00007d10: 650a 2d6d 6c2c 202d 2d6d 696e 6c65 6e2c  e.-ml, --minlen,
-00007d20: 2074 6865 206d 696e 696d 756d 206c 656e   the minimum len
-00007d30: 6774 6820 6f66 2061 2073 6361 6666 6f6c  gth of a scaffol
-00007d40: 6420 746f 2062 6520 696e 636c 7564 6564  d to be included
-00007d50: 2069 6e20 646f 7470 6c6f 742c 2064 6566   in dotplot, def
-00007d60: 6175 6c74 202d 312c 2069 6620 2d31 2077  ault -1, if -1 w
-00007d70: 6173 2073 6574 2c20 7468 6520 3130 2520  as set, the 10% 
-00007d80: 6f66 2074 6865 206c 6f6e 6765 7374 2073  of the longest s
-00007d90: 6361 6666 6f6c 6473 2077 696c 6c20 6265  caffolds will be
-00007da0: 2073 6574 0a2d 6d73 2c20 2d2d 6d61 7873   set.-ms, --maxs
-00007db0: 697a 652c 2074 6865 206d 6178 696d 756d  ize, the maximum
-00007dc0: 2066 616d 696c 7920 7369 7a65 2074 6f20   family size to 
-00007dd0: 6265 2069 6e63 6c75 6465 642c 2064 6566  be included, def
-00007de0: 6175 6c74 2032 3030 0a2d 6170 2c20 2d2d  ault 200.-ap, --
-00007df0: 616e 6368 6f72 706f 696e 7473 2c20 7468  anchorpoints, th
-00007e00: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
-00007e10: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
-00007e20: 7420 4e6f 6e65 0a2d 6d74 2c20 2d2d 6d75  t None.-mt, --mu
-00007e30: 6c74 6970 6c69 636f 6e2c 2074 6865 206d  ltiplicon, the m
-00007e40: 756c 7469 706c 6963 6f6e 7320 6461 7461  ultiplicons data
-00007e50: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
-00007e60: 6e65 0a2d 6774 2c20 2d2d 6765 6e65 7461  ne.-gt, --geneta
-00007e70: 626c 652c 2074 6865 2067 656e 6520 7461  ble, the gene ta
-00007e80: 626c 6520 6461 7461 6669 6c65 2c20 6465  ble datafile, de
-00007e90: 6661 756c 7420 4e6f 6e65 0a2d 6d67 2c20  fault None.-mg, 
-00007ea0: 2d2d 6d69 6e73 6567 6c65 6e2c 2074 6865  --minseglen, the
-00007eb0: 206d 696e 696d 756d 206c 656e 6774 6820   minimum length 
-00007ec0: 6f66 2073 6567 6d65 6e74 7320 746f 2069  of segments to i
-00007ed0: 6e63 6c75 6465 2c20 696e 2072 6174 696f  nclude, in ratio
-00007ee0: 2069 6620 3c3d 2031 2c20 6465 6661 756c   if <= 1, defaul
-00007ef0: 7420 3130 3030 300a 2d6d 676e 2c20 2d2d  t 10000.-mgn, --
-00007f00: 6d69 6e67 656e 656e 756d 2c20 7468 6520  mingenenum, the 
-00007f10: 6d69 6e69 6d75 6d20 6e75 6d62 6572 206f  minimum number o
-00007f20: 6620 6765 6e65 7320 666f 7220 6120 7365  f genes for a se
-00007f30: 676d 656e 7420 746f 2062 6520 636f 6e73  gment to be cons
-00007f40: 6964 6572 6564 2c20 6465 6661 756c 7420  idered, default 
-00007f50: 3330 0a2d 6b72 2c20 2d2d 6b65 6570 7265  30.-kr, --keepre
-00007f60: 6475 6e2c 2066 6c61 6720 6f70 7469 6f6e  dun, flag option
-00007f70: 2c20 7768 6574 6865 7220 746f 206b 6565  , whether to kee
-00007f80: 7020 7265 6475 6e64 616e 7420 6d75 6c74  p redundant mult
-00007f90: 6970 6c69 636f 6e73 2c20 6966 2074 6865  iplicons, if the
-00007fa0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00007fb0: 6865 2072 6564 756e 6461 6e74 206d 756c  he redundant mul
-00007fc0: 7469 706c 6963 6f6e 7320 776f 756c 6420  tiplicons would 
-00007fd0: 6265 206b 6570 740a 2d65 706b 2c20 2d2d  be kept.-epk, --
-00007fe0: 6578 7472 6170 6172 616e 6f6d 656b 732c  extraparanomeks,
-00007ff0: 2065 7874 7261 2070 6172 616e 6f6d 6520   extra paranome 
-00008000: 4b73 2064 6174 6120 746f 2070 6c6f 7420  Ks data to plot 
-00008010: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
-00008020: 6469 7374 7269 6275 7469 6f6e 2c20 6465  distribution, de
-00008030: 6661 756c 7420 4e6f 6e65 0a2d 7061 672c  fault None.-pag,
-00008040: 202d 2d70 6c6f 7461 7067 6d6d 2c20 666c   --plotapgmm, fl
-00008050: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-00008060: 6572 2074 6f20 636f 6e64 7563 7420 616e  er to conduct an
-00008070: 6420 706c 6f74 206d 6978 7475 7265 206d  d plot mixture m
-00008080: 6f64 656c 696e 6720 6f66 2061 6e63 686f  odeling of ancho
-00008090: 7220 4b73 2069 6e20 7468 6520 6d69 7865  r Ks in the mixe
-000080a0: 6420 4b73 2064 6973 7472 6962 7574 696f  d Ks distributio
-000080b0: 6e2c 2069 6620 7468 6520 666c 6167 2077  n, if the flag w
-000080c0: 6173 2073 6574 2c20 7468 6520 6d69 7874  as set, the mixt
-000080d0: 7572 6520 6d6f 6465 6c69 6e67 206f 6620  ure modeling of 
-000080e0: 616e 6368 6f72 204b 7320 776f 756c 6420  anchor Ks would 
-000080f0: 6265 2063 6f6e 6475 6374 6564 2061 6e64  be conducted and
-00008100: 2070 6c6f 7474 6564 0a2d 7065 6d2c 202d   plotted.-pem, -
-00008110: 2d70 6c6f 7465 6c6d 6d2c 2066 6c61 6720  -plotelmm, flag 
-00008120: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00008130: 746f 2063 6f6e 6475 6374 2061 6e64 2070  to conduct and p
-00008140: 6c6f 7420 656c 6d6d 206d 6978 7475 7265  lot elmm mixture
-00008150: 206d 6f64 656c 696e 6720 6f66 2070 6172   modeling of par
-00008160: 616e 6f6d 6520 4b73 2069 6e20 7468 6520  anome Ks in the 
-00008170: 6d69 7865 6420 4b73 2064 6973 7472 6962  mixed Ks distrib
-00008180: 7574 696f 6e2c 2069 6620 7468 6520 666c  ution, if the fl
-00008190: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
-000081a0: 656c 6d6d 206d 6978 7475 7265 206d 6f64  elmm mixture mod
-000081b0: 656c 696e 6720 6f66 2070 6172 616e 6f6d  eling of paranom
-000081c0: 6520 4b73 2077 6f75 6c64 2062 6520 636f  e Ks would be co
-000081d0: 6e64 7563 7465 6420 616e 6420 706c 6f74  nducted and plot
-000081e0: 7465 640a 2d63 2c20 2d2d 636f 6d70 6f6e  ted.-c, --compon
-000081f0: 656e 7473 2c20 7468 6520 7261 6e67 6520  ents, the range 
-00008200: 6f66 2074 6865 206e 756d 6265 7220 6f66  of the number of
-00008210: 2063 6f6d 706f 6e65 6e74 7320 746f 2066   components to f
-00008220: 6974 2069 6e20 616e 6368 6f72 204b 7320  it in anchor Ks 
-00008230: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
-00008240: 2c20 6465 6661 756c 7420 2831 2c34 290a  , default (1,4).
-00008250: 2d70 7379 2c20 2d2d 706c 6f74 7379 6e2c  -psy, --plotsyn,
-00008260: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00008270: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00008280: 6520 7468 6520 7379 6e74 656e 7920 706c  e the synteny pl
-00008290: 6f74 2c20 6f6e 6c79 2077 6865 6e20 7468  ot, only when th
-000082a0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-000082b0: 7468 6520 7379 6e74 656e 7920 706c 6f74  the synteny plot
-000082c0: 2077 6f75 6c64 2062 6520 7072 6f64 7563   would be produc
-000082d0: 6564 0a2d 6473 2c20 2d2d 646f 7473 697a  ed.-ds, --dotsiz
-000082e0: 652c 2074 6865 2064 6f74 2073 697a 6520  e, the dot size 
-000082f0: 696e 2064 6f74 2070 6c6f 742c 2064 6566  in dot plot, def
-00008300: 6175 6c74 2030 2e33 0a2d 6161 2c20 2d2d  ault 0.3.-aa, --
-00008310: 6170 616c 7068 612c 2074 6865 206f 7061  apalpha, the opa
-00008320: 6369 7479 206f 6620 616e 6368 6f72 2064  city of anchor d
-00008330: 6f74 732c 2064 6566 6175 6c74 2031 0a2d  ots, default 1.-
-00008340: 6861 2c20 2d2d 686f 616c 7068 612c 2074  ha, --hoalpha, t
-00008350: 6865 206f 7061 6369 7479 206f 6620 686f  he opacity of ho
-00008360: 6d6f 6c6f 6720 646f 7473 2c20 6465 6661  molog dots, defa
-00008370: 756c 7420 300a 2d73 7274 2c20 2d2d 7368  ult 0.-srt, --sh
-00008380: 6f77 7265 616c 7469 636b 2c20 666c 6167  owrealtick, flag
-00008390: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-000083a0: 2074 6f20 7368 6f77 2074 6865 2072 6561   to show the rea
-000083b0: 6c20 7469 636b 2069 6e20 6765 6e65 7320  l tick in genes 
-000083c0: 6f72 2062 6173 6573 2c20 6966 2074 6865  or bases, if the
-000083d0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-000083e0: 6865 2072 6561 6c20 7469 636b 2077 6f75  he real tick wou
-000083f0: 6c64 2062 6520 7368 6f77 6564 0a2d 746c  ld be showed.-tl
-00008400: 732c 202d 2d74 6963 6b6c 6162 656c 7369  s, --ticklabelsi
-00008410: 7a65 2c20 7468 6520 6c61 6265 6c20 7369  ze, the label si
-00008420: 7a65 206f 6620 7469 636b 2c20 6465 6661  ze of tick, defa
-00008430: 756c 7420 350a 2d78 6c2c 202d 2d78 6c69  ult 5.-xl, --xli
-00008440: 6d2c 2074 6865 2078 2061 7869 7320 6c69  m, the x axis li
-00008450: 6d69 7420 6f66 204b 7320 6469 7374 7269  mit of Ks distri
-00008460: 6275 7469 6f6e 0a2d 796c 2c20 2d2d 796c  bution.-yl, --yl
-00008470: 696d 2c20 7468 6520 7920 6178 6973 206c  im, the y axis l
-00008480: 696d 6974 206f 6620 4b73 2064 6973 7472  imit of Ks distr
-00008490: 6962 7574 696f 6e0a 2d61 646f 2c20 2d2d  ibution.-ado, --
-000084a0: 6164 6a75 7374 6f72 7468 6f2c 2066 6c61  adjustortho, fla
-000084b0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-000084c0: 7220 746f 2061 646a 7573 7420 7468 6520  r to adjust the 
-000084d0: 6869 7374 6f67 7261 6d20 6865 6967 6874  histogram height
-000084e0: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
-000084f0: 4b73 2061 7320 746f 206d 6174 6368 2074  Ks as to match t
-00008500: 6865 2068 6569 6768 7420 6f66 2070 6172  he height of par
-00008510: 616c 6f67 6f75 7320 4b73 2c20 6966 2074  alogous Ks, if t
-00008520: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00008530: 2074 6865 2061 646a 7573 746d 656e 7420   the adjustment 
-00008540: 776f 756c 6420 6265 2063 6f6e 6475 6374  would be conduct
-00008550: 6564 0a2d 6164 662c 202d 2d61 646a 7573  ed.-adf, --adjus
-00008560: 7466 6163 746f 722c 2074 6865 2061 646a  tfactor, the adj
-00008570: 7573 746d 656e 7420 6661 6374 6f72 206f  ustment factor o
-00008580: 6620 6f72 7468 6f6c 6f67 6f75 7320 4b73  f orthologous Ks
-00008590: 2c20 6465 6661 756c 7420 302e 350a 2d6f  , default 0.5.-o
-000085a0: 612c 202d 2d6f 6b61 6c70 6861 2c20 7468  a, --okalpha, th
-000085b0: 6520 6f70 6163 6974 7920 6f66 206f 7274  e opacity of ort
-000085c0: 686f 6c6f 676f 7573 204b 7320 6469 7374  hologous Ks dist
-000085d0: 7269 6275 7469 6f6e 2069 6e20 6d69 7865  ribution in mixe
-000085e0: 6420 706c 6f74 2c20 6465 6661 756c 7420  d plot, default 
-000085f0: 302e 350a 2d63 732c 202d 2d63 6c61 7373  0.5.-cs, --class
-00008600: 6963 2c20 666c 6167 206f 7074 696f 6e2c  ic, flag option,
-00008610: 2077 6865 7468 6572 2074 6f20 6472 6177   whether to draw
-00008620: 206d 6978 6564 204b 7320 706c 6f74 2069   mixed Ks plot i
-00008630: 6e20 6120 636c 6173 7369 6320 6d61 6e6e  n a classic mann
-00008640: 6572 2077 6865 7265 2074 6865 2066 756c  er where the ful
-00008650: 6c20 6f72 7468 6f6c 6f67 6f75 7320 4b73  l orthologous Ks
-00008660: 2064 6973 7472 6962 7574 696f 6e20 6973   distribution is
-00008670: 2064 7261 7765 642c 2069 6620 7468 6520   drawed, if the 
-00008680: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
-00008690: 6520 636c 6173 7369 6320 6d69 7865 6420  e classic mixed 
-000086a0: 4b73 2070 6c6f 7420 776f 756c 6420 6265  Ks plot would be
-000086b0: 2064 7261 776e 0a2d 7461 2c20 2d2d 746f   drawn.-ta, --to
-000086c0: 7061 7272 6f77 2c20 666c 6167 206f 7074  parrow, flag opt
-000086d0: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-000086e0: 6164 6a75 7374 2074 6865 2061 7272 6f77  adjust the arrow
-000086f0: 2074 6f20 6265 2061 7420 7468 6520 746f   to be at the to
-00008700: 7020 6f66 2074 6865 2070 6c6f 742c 2069  p of the plot, i
-00008710: 6e73 7465 6164 206f 6620 6265 696e 6720  nstead of being 
-00008720: 636f 6f72 6469 6e61 7465 6420 6173 2074  coordinated as t
-00008730: 6865 204b 4445 206f 6620 7468 6520 6f72  he KDE of the or
-00008740: 7468 6f6c 6f67 6f75 7320 4b73 2064 6973  thologous Ks dis
-00008750: 7472 6962 7574 696f 6e2c 2069 6620 7468  tribution, if th
-00008760: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00008770: 7468 6520 6172 726f 7720 776f 756c 6420  the arrow would 
-00008780: 6265 2073 6574 2061 7420 7468 6520 746f  be set at the to
-00008790: 700a 2d6e 612c 202d 2d6e 6f64 6561 7665  p.-na, --nodeave
-000087a0: 7261 6765 642c 2066 6c61 6720 6f70 7469  raged, flag opti
-000087b0: 6f6e 2c20 7768 6574 6865 7220 746f 2075  on, whether to u
-000087c0: 7365 206e 6f64 652d 6176 6572 6167 6564  se node-averaged
-000087d0: 206d 6574 686f 6420 666f 7220 6465 2d72   method for de-r
-000087e0: 6564 756e 6461 6e63 792c 2069 6620 7468  edundancy, if th
-000087f0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00008800: 7468 6520 6e6f 6465 2d61 7665 7261 6765  the node-average
-00008810: 6420 6d65 7468 6f64 2077 6f75 6c64 2062  d method would b
-00008820: 6520 696e 6974 6961 7465 640a 2d62 732c  e initiated.-bs,
-00008830: 202d 2d62 6f6f 7473 7472 6170 2c20 7468   --bootstrap, th
-00008840: 6520 6e75 6d62 6572 206f 6620 626f 6f74  e number of boot
-00008850: 7374 7261 7020 7265 706c 6963 6174 6573  strap replicates
-00008860: 206f 6620 6f72 7468 6f6c 6f67 204b 7320   of ortholog Ks 
-00008870: 6469 7374 7269 6275 7469 6f6e 2069 6e20  distribution in 
-00008880: 6d69 7865 6420 706c 6f74 0a2d 6772 2c20  mixed plot.-gr, 
-00008890: 2d2d 6769 7374 7262 2c20 666c 6167 206f  --gistrb, flag o
-000088a0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000088b0: 6f20 7573 6520 6769 7374 5f72 6169 6e62  o use gist_rainb
-000088c0: 6f77 2061 7320 636f 6c6f 7220 6d61 7020  ow as color map 
-000088d0: 6f66 2064 6f74 706c 6f74 0a2d 6e2c 202d  of dotplot.-n, -
-000088e0: 2d6e 7468 7265 6164 732c 2074 6865 206e  -nthreads, the n
-000088f0: 756d 6265 7220 6f66 2074 6872 6561 6473  umber of threads
-00008900: 2074 6f20 7573 6520 696e 2062 6f6f 7473   to use in boots
-00008910: 7472 6170 2073 616d 706c 696e 672c 2064  trap sampling, d
-00008920: 6566 6175 6c74 2031 0a60 6060 0a0a 2323  efault 1.```..##
-00008930: 2055 7361 6765 0a0a 4865 7265 2077 6520   Usage..Here we 
-00008940: 7072 6f76 6964 6564 2074 6865 2062 6173  provided the bas
-00008950: 6963 2075 7361 6765 2066 6f72 2065 6163  ic usage for eac
-00008960: 6820 7072 6f67 7261 6d20 616e 6420 7468  h program and th
-00008970: 6520 7265 6c65 7661 6e74 2070 6172 616d  e relevant param
-00008980: 6574 6572 7320 616e 6420 7375 6767 6573  eters and sugges
-00008990: 7469 6f6e 7320 6f6e 2070 6172 616d 6574  tions on paramet
-000089a0: 6572 697a 6174 696f 6e2e 2041 2072 656d  erization. A rem
-000089b0: 696e 6465 7220 7468 6174 2074 6865 2067  inder that the g
-000089c0: 6976 656e 2063 6f72 6573 2061 6e64 2074  iven cores and t
-000089d0: 6872 6561 6473 2063 616e 2073 6967 6e69  hreads can signi
-000089e0: 6669 6361 6e74 6c79 2069 6d70 6163 7420  ficantly impact 
-000089f0: 7468 6520 7275 6e20 7469 6d65 2061 6e64  the run time and
-00008a00: 2074 6875 7320 7765 2061 6464 6564 2073   thus we added s
-00008a10: 6f6d 6520 7265 706f 7274 2069 6e66 6f72  ome report infor
-00008a20: 6d61 7469 6f6e 2070 6572 7461 696e 696e  mation pertainin
-00008a30: 6720 746f 2074 6865 2073 7973 7465 6d20  g to the system 
-00008a40: 6f66 2075 7365 7273 2074 6f20 6661 6369  of users to faci
-00008a50: 6c69 7461 7465 2074 6865 2065 6666 6963  litate the effic
-00008a60: 6965 6e74 2073 6574 7469 6e67 206f 6620  ient setting of 
-00008a70: 7468 7265 6164 7320 616e 6420 6d65 6d6f  threads and memo
-00008a80: 7279 2e20 5468 6520 6c6f 6769 6361 6c20  ry. The logical 
-00008a90: 4350 5573 2072 6570 6f72 7465 6420 7265  CPUs reported re
-00008aa0: 7072 6573 656e 7473 2074 6865 206e 756d  presents the num
-00008ab0: 6265 7220 6f66 2070 6879 7369 6361 6c20  ber of physical 
-00008ac0: 636f 7265 7320 6d75 6c74 6970 6c69 6564  cores multiplied
-00008ad0: 2062 7920 7468 6520 6e75 6d62 6572 206f   by the number o
-00008ae0: 6620 7468 7265 6164 7320 7468 6174 2063  f threads that c
-00008af0: 616e 2072 756e 206f 6e20 6561 6368 2063  an run on each c
-00008b00: 6f72 652c 2061 6c73 6f20 6b6e 6f77 6e20  ore, also known 
-00008b10: 6173 2048 7970 6572 2054 6872 6561 6469  as Hyper Threadi
-00008b20: 6e67 2e20 5468 6520 6e75 6d62 6572 206f  ng. The number o
-00008b30: 6620 6c6f 6769 6361 6c20 4350 5573 206d  f logical CPUs m
-00008b40: 6179 206e 6f74 206e 6563 6573 7361 7269  ay not necessari
-00008b50: 6c79 2062 6520 6571 7569 7661 6c65 6e74  ly be equivalent
-00008b60: 2074 6f20 7468 6520 6163 7475 616c 206e   to the actual n
-00008b70: 756d 6265 7220 6f66 2043 5055 7320 7468  umber of CPUs th
-00008b80: 6520 6375 7272 656e 7420 7072 6f63 6573  e current proces
-00008b90: 7320 6361 6e20 7573 652e 2054 6865 2061  s can use. The a
-00008ba0: 7661 696c 6162 6c65 206d 656d 6f72 7920  vailable memory 
-00008bb0: 7265 6665 7273 2074 6f20 7468 6520 6d65  refers to the me
-00008bc0: 6d6f 7279 2074 6861 7420 6361 6e20 6265  mory that can be
-00008bd0: 2067 6976 656e 2069 6e73 7461 6e74 6c79   given instantly
-00008be0: 2074 6f20 7072 6f63 6573 7365 7320 7769   to processes wi
-00008bf0: 7468 6f75 7420 7468 6520 7379 7374 656d  thout the system
-00008c00: 2067 6f69 6e67 2069 6e74 6f20 7377 6170   going into swap
-00008c10: 2061 6e64 2072 6566 6c65 6374 7320 7468   and reflects th
-00008c20: 6520 6163 7475 616c 206d 656d 6f72 7920  e actual memory 
-00008c30: 6176 6169 6c61 626c 652e 2054 6865 2066  available. The f
-00008c40: 7265 6520 6d65 6d6f 7279 2072 6566 6572  ree memory refer
-00008c50: 7320 746f 2074 6865 206d 656d 6f72 7920  s to the memory 
-00008c60: 6e6f 7420 6265 696e 6720 7573 6564 2061  not being used a
-00008c70: 7420 616c 6c20 287a 6572 6f65 6429 2074  t all (zeroed) t
-00008c80: 6861 7420 6973 2072 6561 6469 6c79 2061  hat is readily a
-00008c90: 7661 696c 6162 6c65 2e20 5468 6520 6465  vailable. The de
-00008ca0: 7363 7269 7074 696f 6e20 6162 6f76 6520  scription above 
-00008cb0: 7265 6665 7273 2074 6f20 7468 6520 646f  refers to the do
-00008cc0: 6375 6d65 6e74 6174 696f 6e20 6f66 2060  cumentation of `
-00008cd0: 7073 7574 696c 602e 0a0a 2323 2320 7767  psutil`...### wg
-00008ce0: 6420 646d 640a 0a2a 2a54 6865 2064 656c  d dmd..**The del
-00008cf0: 696e 6561 7469 6f6e 206f 6620 7768 6f6c  ineation of whol
-00008d00: 6520 7061 7261 6e6f 6d65 2a2a 0a60 6060  e paranome**.```
-00008d10: 0a77 6764 2064 6d64 2041 7175 696c 6567  .wgd dmd Aquileg
-00008d20: 6961 5f63 6f65 7275 6c65 6120 2d49 2032  ia_coerulea -I 2
-00008d30: 202d 6520 3165 2d31 3020 2d62 7320 3130   -e 1e-10 -bs 10
-00008d40: 3020 2d6e 7020 3520 282d 6e6e 2920 282d  0 -np 5 (-nn) (-
-00008d50: 2d74 6f5f 7374 6f70 2920 282d 2d63 6473  -to_stop) (--cds
-00008d60: 2920 282d 6e20 3429 2028 2d6f 2077 6764  ) (-n 4) (-o wgd
-00008d70: 5f64 6d64 2920 282d 7420 776f 726b 696e  _dmd) (-t workin
-00008d80: 675f 746d 7029 0a60 6060 200a 0a4e 6f74  g_tmp).``` ..Not
-00008d90: 6520 7468 6174 2077 6520 646f 6e27 7420  e that we don't 
-00008da0: 7072 6f76 6964 6520 7468 6520 6461 7461  provide the data
-00008db0: 206f 6620 7468 6973 2063 6f64 696e 6720   of this coding 
-00008dc0: 7365 7175 656e 6365 2028 6364 7329 2066  sequence (cds) f
-00008dd0: 696c 6520 6041 7175 696c 6567 6961 5f63  ile `Aquilegia_c
-00008de0: 6f65 7275 6c65 6160 2062 7574 2069 7420  oerulea` but it 
-00008df0: 6361 6e20 6265 2065 6173 696c 7920 646f  can be easily do
-00008e00: 776e 6c6f 6164 6564 2061 7420 5b50 6879  wnloaded at [Phy
-00008e10: 746f 7a6f 6d65 5d28 6874 7470 733a 2f2f  tozome](https://
-00008e20: 7068 7974 6f7a 6f6d 652d 6e65 7874 2e6a  phytozome-next.j
-00008e30: 6769 2e64 6f65 2e67 6f76 2f69 6e66 6f2f  gi.doe.gov/info/
-00008e40: 4163 6f65 7275 6c65 615f 7633 5f31 2920  Acoerulea_v3_1) 
-00008e50: 2873 616d 6520 666f 7220 6f74 6865 7220  (same for other 
-00008e60: 5573 6167 6520 646f 6329 2e20 4120 7265  Usage doc). A re
-00008e70: 6d69 6e64 6572 2074 6861 7420 696e 2074  minder that in t
-00008e80: 6865 2069 7373 7565 7320 736f 6d65 2075  he issues some u
-00008e90: 7365 7273 2064 6964 6e27 7420 646f 776e  sers didn't down
-00008ea0: 6c6f 6164 2074 6865 2060 4163 6f65 7275  load the `Acoeru
-00008eb0: 6c65 615f 3332 325f 7633 2e31 2e63 6473  lea_322_v3.1.cds
-00008ec0: 5f70 7269 6d61 7279 5472 616e 7363 7269  _primaryTranscri
-00008ed0: 7074 4f6e 6c79 2e66 612e 677a 6020 6275  ptOnly.fa.gz` bu
-00008ee0: 7420 696e 7374 6561 6420 7468 6520 6041  t instead the `A
-00008ef0: 636f 6572 756c 6561 5f33 3232 5f76 332e  coerulea_322_v3.
-00008f00: 312e 6364 732e 6661 2e67 7a60 2066 696c  1.cds.fa.gz` fil
-00008f10: 652e 2046 6f72 2074 6865 2063 6f6e 7374  e. For the const
-00008f20: 7275 6374 696f 6e20 6f66 2077 686f 6c65  ruction of whole
-00008f30: 2070 6172 616e 6f6d 6520 2a4b 2a3c 7375   paranome *K*<su
-00008f40: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
-00008f50: 6275 7469 6f6e 2c20 2a2a 6f6e 6c79 206f  bution, **only o
-00008f60: 6e65 2074 7261 6e73 6372 6970 7420 2874  ne transcript (t
-00008f70: 6865 2070 7269 6d61 7279 206f 6e65 2920  he primary one) 
-00008f80: 7065 7220 6765 6e65 2073 686f 756c 6420  per gene should 
-00008f90: 6265 2069 6e63 6c75 6465 642a 2a20 7375  be included** su
-00008fa0: 6368 2074 6861 7420 7468 6520 2a4b 2a3c  ch that the *K*<
-00008fb0: 7375 623e 533c 2f73 7562 3e20 6973 2072  sub>S</sub> is r
-00008fc0: 6561 6c6c 7920 696e 6469 6361 7469 6e67  eally indicating
-00008fd0: 2074 6865 2061 6765 206f 6620 6765 6e65   the age of gene
-00008fe0: 2064 7570 6c69 6361 7469 6f6e 2065 7665   duplication eve
-00008ff0: 6e74 2c20 696e 7374 6561 6420 6f66 2061  nt, instead of a
-00009000: 6c74 6572 6e61 7469 7665 2073 706c 6963  lternative splic
-00009010: 696e 672e 2054 7261 6e73 6372 6970 746f  ing. Transcripto
-00009020: 6d65 2064 6174 6120 7368 6f75 6c64 2062  me data should b
-00009030: 6520 6361 7265 6675 6c6c 7920 7472 6561  e carefully trea
-00009040: 7465 6420 7769 7468 2064 652d 7265 6475  ted with de-redu
-00009050: 6e64 616e 6379 2073 6f20 6173 2074 6f20  ndancy so as to 
-00009060: 7265 6475 6365 2074 6865 2066 616c 7365  reduce the false
-00009070: 2070 6f73 6974 6976 6520 6475 706c 6963   positive duplic
-00009080: 6174 696f 6e20 6275 6d70 2063 6175 7365  ation bump cause
-00009090: 6420 6279 2070 6572 7661 7369 7665 2061  d by pervasive a
-000090a0: 6c74 6572 6e61 7469 7665 2073 706c 6963  lternative splic
-000090b0: 696e 672e 2048 6572 6520 7468 6520 696e  ing. Here the in
-000090c0: 666c 6174 696f 6e20 6661 6374 6f72 2070  flation factor p
-000090d0: 6172 616d 6574 6572 2c20 6769 7665 6e20  arameter, given 
-000090e0: 6279 2060 2d49 6020 6f72 2060 2d2d 696e  by `-I` or `--in
-000090f0: 666c 6174 696f 6e60 2c20 6166 6665 6374  flation`, affect
-00009100: 7320 7468 6520 6772 616e 756c 6172 6974  s the granularit
-00009110: 7920 6f72 2072 6573 6f6c 7574 696f 6e20  y or resolution 
-00009120: 6f66 2074 6865 2063 6c75 7374 6572 696e  of the clusterin
-00009130: 6720 6f75 7463 6f6d 6520 616e 6420 696d  g outcome and im
-00009140: 706c 6963 6974 6c79 2063 6f6e 7472 6f6c  plicitly control
-00009150: 6c73 2074 6865 206e 756d 6265 7220 6f66  ls the number of
-00009160: 2063 6c75 7374 6572 732c 2077 6974 6820   clusters, with 
-00009170: 6c6f 7720 7661 6c75 6573 2073 7563 6820  low values such 
-00009180: 6173 2031 2e33 206f 7220 312e 3420 6c65  as 1.3 or 1.4 le
-00009190: 6164 696e 6720 746f 2066 6577 6572 2062  ading to fewer b
-000091a0: 7574 206c 6172 6765 7220 636c 7573 7465  ut larger cluste
-000091b0: 7273 2061 6e64 2068 6967 6820 7661 6c75  rs and high valu
-000091c0: 6573 2073 7563 6820 6173 2035 206f 7220  es such as 5 or 
-000091d0: 3620 6c65 6164 696e 6720 746f 206d 6f72  6 leading to mor
-000091e0: 6520 6275 7420 736d 616c 6c65 7220 636c  e but smaller cl
-000091f0: 7573 7465 7273 2e20 5765 2073 6574 2074  usters. We set t
-00009200: 6865 2064 6566 6175 6c74 2076 616c 7565  he default value
-00009210: 2061 7320 3220 6173 2073 7567 6765 7374   as 2 as suggest
-00009220: 6564 2062 7920 5b4d 434c 5d28 6874 7470  ed by [MCL](http
-00009230: 733a 2f2f 6d69 6361 6e73 2e6f 7267 2f6d  s://micans.org/m
-00009240: 636c 2f29 2e20 5468 6520 652d 7661 6c75  cl/). The e-valu
-00009250: 6520 6375 742d 6f66 6620 666f 7220 7365  e cut-off for se
-00009260: 7175 656e 6365 2073 696d 696c 6172 6974  quence similarit
-00009270: 792c 2067 6976 656e 2062 7920 602d 6560  y, given by `-e`
-00009280: 206f 7220 602d 2d65 7661 6c60 2c20 7768   or `--eval`, wh
-00009290: 6963 6820 6465 6e6f 7465 7320 7468 6520  ich denotes the 
-000092a0: 6578 7065 6374 6564 2076 616c 7565 206f  expected value o
-000092b0: 6620 7468 6520 6869 7420 7175 616e 7469  f the hit quanti
-000092c0: 6669 6573 2074 6865 206e 756d 6265 7220  fies the number 
-000092d0: 6f66 2061 6c69 676e 6d65 6e74 7320 6f66  of alignments of
-000092e0: 2073 696d 696c 6172 206f 7220 6265 7474   similar or bett
-000092f0: 6572 2071 7561 6c69 7479 2074 6861 7420  er quality that 
-00009300: 796f 7520 6578 7065 6374 2074 6f20 6669  you expect to fi
-00009310: 6e64 2073 6561 7263 6869 6e67 2074 6869  nd searching thi
-00009320: 7320 7175 6572 7920 6167 6169 6e73 7420  s query against 
-00009330: 6120 6461 7461 6261 7365 206f 6620 7261  a database of ra
-00009340: 6e64 6f6d 2073 6571 7565 6e63 6573 2074  ndom sequences t
-00009350: 6865 2073 616d 6520 7369 7a65 2061 7320  he same size as 
-00009360: 7468 6520 6163 7475 616c 2074 6172 6765  the actual targe
-00009370: 7420 6461 7461 6261 7365 2c20 6973 2074  t database, is t
-00009380: 6865 206b 6579 2070 6172 616d 6574 6572  he key parameter
-00009390: 206d 6561 7375 7269 6e67 2074 6865 2073   measuring the s
-000093a0: 6967 6e69 6669 6361 6e63 6520 6f66 2061  ignificance of a
-000093b0: 2068 6974 2c20 7768 6963 6820 6973 2073   hit, which is s
-000093c0: 6574 2068 6572 6520 6173 2064 6566 6175  et here as defau
-000093d0: 6c74 2031 652d 3130 2e20 4e6f 7465 2074  lt 1e-10. Note t
-000093e0: 6861 7420 5b44 4941 4d4f 4e44 5d28 6874  hat [DIAMOND](ht
-000093f0: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-00009400: 2f62 6275 6368 6669 6e6b 2f64 6961 6d6f  /bbuchfink/diamo
-00009410: 6e64 2f77 696b 692f 312e 2d54 7574 6f72  nd/wiki/1.-Tutor
-00009420: 6961 6c29 2069 7473 656c 6620 6279 2064  ial) itself by d
-00009430: 6566 6175 6c74 206f 6e6c 7920 7265 706f  efault only repo
-00009440: 7274 7320 616c 6c20 616c 6967 6e6d 656e  rts all alignmen
-00009450: 7473 2077 6974 6820 652d 7661 6c75 6520  ts with e-value 
-00009460: 3c20 302e 3030 312e 2054 6865 2070 6572  < 0.001. The per
-00009470: 6365 6e74 6167 6520 6f66 2075 7070 6572  centage of upper
-00009480: 2068 6974 7320 7573 6564 2066 6f72 2067   hits used for g
-00009490: 656e 6520 6c65 6e67 7468 206e 6f72 6d61  ene length norma
-000094a0: 6c69 7a61 7469 6f6e 2c20 6769 7665 6e20  lization, given 
-000094b0: 6279 2060 2d6e 7060 206f 7220 602d 2d6e  by `-np` or `--n
-000094c0: 6f72 6d61 6c69 7a65 6470 6572 6365 6e74  ormalizedpercent
-000094d0: 602c 2077 6869 6368 2064 6574 6572 6d69  `, which determi
-000094e0: 6e65 7320 7468 6520 7570 7065 7220 7065  nes the upper pe
-000094f0: 7263 656e 7469 6c65 206f 6620 6869 7473  rcentile of hits
-00009500: 2070 6572 2062 696e 2028 6361 7465 676f   per bin (catego
-00009510: 7269 7a65 6420 6279 2067 656e 6520 6c65  rized by gene le
-00009520: 6e67 7468 2920 7573 6564 2069 6e20 7468  ngth) used in th
-00009530: 6520 6669 7420 6f66 206c 696e 6561 7220  e fit of linear 
-00009540: 7265 6772 6573 7369 6f6e 2c20 636f 6e73  regression, cons
-00009550: 6964 6572 696e 6720 7468 6174 206e 6f74  idering that not
-00009560: 2061 6c6c 2068 6974 7320 7065 7220 6269   all hits per bi
-00009570: 6e20 7368 6f77 2061 7070 6172 656e 7420  n show apparent 
-00009580: 6c69 6e65 6172 2072 656c 6174 696f 6e73  linear relations
-00009590: 6869 702c 2069 7320 7365 7420 6173 2064  hip, is set as d
-000095a0: 6566 6175 6c74 2035 2c20 696e 6469 6361  efault 5, indica
-000095b0: 7469 6e67 2074 6865 2075 7361 6765 206f  ting the usage o
-000095c0: 6620 746f 7020 3525 2068 6974 7320 7065  f top 5% hits pe
-000095d0: 7220 6269 6e2e 2054 6865 206e 756d 6265  r bin. The numbe
-000095e0: 7220 6f66 2062 696e 7320 6469 7669 6465  r of bins divide
-000095f0: 6420 696e 2067 656e 6520 6c65 6e67 7468  d in gene length
-00009600: 206e 6f72 6d61 6c69 7a61 7469 6f6e 2c20   normalization, 
-00009610: 6769 7665 6e20 6279 2060 2d62 7360 206f  given by `-bs` o
-00009620: 7220 602d 2d62 696e 7360 2c20 6465 7465  r `--bins`, dete
-00009630: 726d 696e 6573 2074 6865 206e 756d 6265  rmines the numbe
-00009640: 7220 6f66 2062 696e 7320 746f 2063 6174  r of bins to cat
-00009650: 6567 6f72 697a 6520 7468 6520 6765 6e65  egorize the gene
-00009660: 206c 656e 6774 682c 2069 7320 7365 7420   length, is set 
-00009670: 6173 2064 6566 6175 6c74 2031 3030 2e20  as default 100. 
-00009680: 5468 6520 7061 7261 6d65 7465 7220 602d  The parameter `-
-00009690: 6e6e 6020 6f72 2060 2d2d 6e6f 6e6f 726d  nn` or `--nonorm
-000096a0: 616c 697a 6174 696f 6e60 2063 616e 2062  alization` can b
-000096b0: 6520 7365 7420 746f 2063 616c 6c20 6f66  e set to call of
-000096c0: 6620 7468 6520 6e6f 726d 616c 697a 6174  f the normalizat
-000096d0: 696f 6e20 7072 6f63 6573 732c 2061 6c74  ion process, alt
-000096e0: 686f 7567 6820 6974 2773 2073 7567 6765  hough it's sugge
-000096f0: 7374 6564 2074 6f20 636f 6e64 7563 7420  sted to conduct 
-00009700: 7468 6520 6e6f 726d 616c 697a 6174 696f  the normalizatio
-00009710: 6e20 746f 2061 6371 7569 7265 206d 6f72  n to acquire mor
-00009720: 6520 6163 6375 7261 7465 2067 656e 6520  e accurate gene 
-00009730: 6661 6d69 6c79 2063 6c75 7374 6572 696e  family clusterin
-00009740: 6720 7265 7375 6c74 2e20 5468 6520 7061  g result. The pa
-00009750: 7261 6d65 7465 7273 2060 2d2d 746f 5f73  rameters `--to_s
-00009760: 746f 7060 2061 6e64 2060 2d2d 6364 7360  top` and `--cds`
-00009770: 2063 6f6e 7472 6f6c 2074 6865 2062 6568   control the beh
-00009780: 6176 696f 7572 206f 6620 7472 616e 736c  aviour of transl
-00009790: 6174 696e 6720 636f 6469 6e67 2073 6571  ating coding seq
-000097a0: 7565 6e63 6520 696e 746f 2061 6d69 6e6f  uence into amino
-000097b0: 2061 6369 6420 7365 7175 656e 6365 2e20   acid sequence. 
-000097c0: 4966 2074 6865 2060 2d2d 746f 5f73 746f  If the `--to_sto
-000097d0: 7060 2077 6173 2073 6574 2c20 7468 6520  p` was set, the 
-000097e0: 7472 616e 736c 6174 696f 6e20 776f 756c  translation woul
-000097f0: 6420 6265 2074 6572 6d69 6e61 7465 6420  d be terminated 
-00009800: 6174 2074 6865 2066 6972 7374 2069 6e2d  at the first in-
-00009810: 6672 616d 6520 7374 6f70 2063 6f64 6f6e  frame stop codon
-00009820: 2c20 6f74 6865 7277 6973 6520 7468 6520  , otherwise the 
-00009830: 7472 616e 736c 6174 696f 6e20 776f 756c  translation woul
-00009840: 6420 7369 6d70 6c79 2073 6b69 7020 616e  d simply skip an
-00009850: 7920 7374 6f70 2063 6f64 6f6e 732e 2049  y stop codons. I
-00009860: 6620 7468 6520 602d 2d63 6473 6020 7761  f the `--cds` wa
-00009870: 7320 7365 742c 2073 6571 7565 6e63 6573  s set, sequences
-00009880: 2074 6861 7420 646f 6573 6e27 7420 7374   that doesn't st
-00009890: 6172 7420 7769 7468 2061 2076 616c 6964  art with a valid
-000098a0: 2073 7461 7274 2063 6f64 6f6e 2c20 6f72   start codon, or
-000098b0: 2063 6f6e 7461 696e 7320 6d6f 7265 2074   contains more t
-000098c0: 6861 6e20 6f6e 6520 696e 2d66 7261 6d65  han one in-frame
-000098d0: 2073 746f 7020 636f 646f 6e2c 206f 7220   stop codon, or 
-000098e0: 6973 206e 6f74 2064 6976 6964 6162 6c65  is not dividable
-000098f0: 2062 7920 332c 2077 6f75 6c64 2062 6520   by 3, would be 
-00009900: 7369 6d70 6c79 2064 726f 7070 6564 2c20  simply dropped, 
-00009910: 7375 6368 2074 6861 7420 6f6e 6c79 2073  such that only s
-00009920: 7472 6963 7420 636f 6d70 6c65 7465 2063  trict complete c
-00009930: 6f64 696e 6720 7365 7175 656e 6365 7320  oding sequences 
-00009940: 776f 756c 6420 6265 2069 6e63 6c75 6465  would be include
-00009950: 6420 696e 2074 6865 2073 7562 7365 7175  d in the subsequ
-00009960: 656e 7420 616e 616c 7973 6973 2e20 5468  ent analysis. Th
-00009970: 6520 6578 6163 7420 6265 6861 7669 6f75  e exact behaviou
-00009980: 7220 6f66 2060 2d2d 746f 5f73 746f 7060  r of `--to_stop`
-00009990: 2061 6e64 2060 2d2d 6364 7360 2069 7320   and `--cds` is 
-000099a0: 6465 6669 6e65 6420 616e 6420 6465 7363  defined and desc
-000099b0: 7269 6265 6420 696e 2074 6865 205b 6269  ribed in the [bi
-000099c0: 6f70 7974 686f 6e5d 2868 7474 7073 3a2f  opython](https:/
-000099d0: 2f62 696f 7079 7468 6f6e 2e6f 7267 2f64  /biopython.org/d
-000099e0: 6f63 732f 312e 3735 2f61 7069 2f42 696f  ocs/1.75/api/Bio
-000099f0: 2e53 6571 2e68 746d 6c23 4269 6f2e 5365  .Seq.html#Bio.Se
-00009a00: 712e 5365 712e 7472 616e 736c 6174 6529  q.Seq.translate)
-00009a10: 206c 6962 7261 7279 2e20 5468 6520 6e75   library. The nu
-00009a20: 6d62 6572 206f 6620 7061 7261 6c6c 656c  mber of parallel
-00009a30: 2074 6872 6561 6473 2062 7920 7468 6520   threads by the 
-00009a40: 6f70 7469 6f6e 2060 2d6e 6020 6f72 2060  option `-n` or `
-00009a50: 2d2d 6e74 6872 6561 6473 6020 6361 6e20  --nthreads` can 
-00009a60: 6265 2073 6574 2074 6f20 6163 6365 6c65  be set to accele
-00009a70: 7261 7465 2074 6865 2063 616c 6375 6c61  rate the calcula
-00009a80: 7469 6f6e 2077 6974 6869 6e20 6064 6961  tion within `dia
-00009a90: 6d6f 6e64 602e 2054 6865 2064 6972 6563  mond`. The direc
-00009aa0: 746f 7279 206f 6620 6f75 7470 7574 206f  tory of output o
-00009ab0: 7220 696e 7465 726d 6564 6961 7465 2066  r intermediate f
-00009ac0: 696c 6573 2069 7320 6465 7465 726d 696e  iles is determin
-00009ad0: 6564 2062 7920 7468 6520 7061 7261 6d65  ed by the parame
-00009ae0: 7465 7220 602d 6f60 206f 7220 602d 2d6f  ter `-o` or `--o
-00009af0: 7574 6469 7260 2c20 616e 6420 602d 7460  utdir`, and `-t`
-00009b00: 206f 7220 602d 2d74 6d70 6469 7260 2c20   or `--tmpdir`, 
-00009b10: 7768 6963 6820 7769 6c6c 2062 6520 6372  which will be cr
-00009b20: 6561 7465 6420 6279 2074 6865 2070 726f  eated by the pro
-00009b30: 6772 616d 2069 7473 656c 6620 616e 6420  gram itself and 
-00009b40: 6265 206f 7665 7277 7269 7474 656e 2069  be overwritten i
-00009b50: 6620 7468 6520 666f 6c64 6572 2068 6173  f the folder has
-00009b60: 2061 6c72 6561 6479 2062 6565 6e20 6372   already been cr
-00009b70: 6561 7465 642e 204e 6f74 6520 7468 6174  eated. Note that
-00009b80: 2074 6865 2073 6f66 7477 6172 6520 6064   the software `d
-00009b90: 6961 6d6f 6e64 6020 7368 6f75 6c64 2062  iamond` should b
-00009ba0: 6520 7072 652d 696e 7374 616c 6c65 6420  e pre-installed 
-00009bb0: 616e 6420 7365 7420 696e 2074 6865 2065  and set in the e
-00009bc0: 6e76 6972 6f6e 6d65 6e74 2070 6174 6820  nvironment path 
-00009bd0: 696e 2061 6c6c 2074 6865 2061 6e61 6c79  in all the analy
-00009be0: 7369 7320 7065 7266 6f72 6d65 6420 6279  sis performed by
-00009bf0: 2060 7767 6420 646d 6460 2065 7863 6570   `wgd dmd` excep
-00009c00: 7420 666f 7220 7468 6520 636f 6c6c 696e  t for the collin
-00009c10: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
-00009c20: 616e 616c 7973 6973 2e0a 0a2a 2a57 6520  analysis...**We 
-00009c30: 7375 6767 6573 7420 7468 6174 2074 6865  suggest that the
-00009c40: 2064 6566 6175 6c74 2073 6574 7469 6e67   default setting
-00009c50: 2069 6e20 7768 6963 6820 7468 6520 696e   in which the in
-00009c60: 666c 6174 696f 6e20 6661 6374 6f72 2069  flation factor i
-00009c70: 7320 7365 7420 6173 2032 2c20 652d 7661  s set as 2, e-va
-00009c80: 6c75 6520 6375 742d 6f66 6620 6173 2031  lue cut-off as 1
-00009c90: 652d 3130 2061 6e64 206f 7468 6572 2070  e-10 and other p
-00009ca0: 6172 616d 6574 6572 7320 696e 2064 6566  arameters in def
-00009cb0: 6175 6c74 2069 7320 6120 676f 6f64 2073  ault is a good s
-00009cc0: 7461 7274 696e 6720 706f 696e 742c 2075  tarting point, u
-00009cd0: 6e6c 6573 7320 796f 7520 7370 6563 6966  nless you specif
-00009ce0: 6963 616c 6c79 2077 616e 7420 746f 2065  ically want to e
-00009cf0: 7870 6c6f 7265 2074 6865 2065 6666 6563  xplore the effec
-00009d00: 7473 206f 6620 6469 6666 6572 656e 7420  ts of different 
-00009d10: 7061 7261 6d65 7465 7273 2e2a 2a20 5375  parameters.** Su
-00009d20: 6368 2074 6861 7420 7468 6520 636f 6d6d  ch that the comm
-00009d30: 616e 6420 666f 7220 7468 6520 6465 6c69  and for the deli
-00009d40: 6e65 6174 696f 6e20 6f66 2077 686f 6c65  neation of whole
-00009d50: 2070 6172 616e 6f6d 6520 6973 2073 696d   paranome is sim
-00009d60: 706c 7920 6173 2062 656c 6f77 2e0a 0a60  ply as below...`
-00009d70: 6060 0a77 6764 2064 6d64 2041 7175 696c  ``.wgd dmd Aquil
-00009d80: 6567 6961 5f63 6f65 7275 6c65 610a 6060  egia_coerulea.``
-00009d90: 600a 0a2a 2a54 6865 2064 656c 696e 6561  `..**The delinea
-00009da0: 7469 6f6e 206f 6620 5242 4873 2a2a 0a60  tion of RBHs**.`
-00009db0: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
-00009dc0: 6e63 6531 2073 6571 7565 6e63 6532 202d  nce1 sequence2 -
-00009dd0: 6520 3165 2d31 3020 2d62 7320 3130 3020  e 1e-10 -bs 100 
-00009de0: 2d6e 7020 3520 282d 6e6e 2920 282d 6e20  -np 5 (-nn) (-n 
-00009df0: 3429 2028 2d63 2030 2e39 2920 282d 2d6f  4) (-c 0.9) (--o
-00009e00: 6766 6f72 6d61 7429 2028 2d2d 746f 5f73  gformat) (--to_s
-00009e10: 746f 7029 2028 2d2d 6364 7329 2028 2d6f  top) (--cds) (-o
-00009e20: 2077 6764 5f64 6d64 2920 282d 7420 776f   wgd_dmd) (-t wo
-00009e30: 726b 696e 675f 746d 7029 0a60 6060 0a0a  rking_tmp).```..
-00009e40: 546f 2064 656c 696e 6561 7465 2052 4248  To delineate RBH
-00009e50: 7320 6265 7477 6565 6e20 7477 6f20 6364  s between two cd
-00009e60: 7320 7365 7175 656e 6365 2066 696c 6573  s sequence files
-00009e70: 2c20 7468 6520 7265 6c65 7661 6e74 2070  , the relevant p
-00009e80: 6172 616d 6574 6572 2069 7320 6d6f 7374  arameter is most
-00009e90: 6c79 2074 6865 2073 616d 6520 6173 2077  ly the same as w
-00009ea0: 686f 6c65 2070 6172 616e 6f6d 6520 696e  hole paranome in
-00009eb0: 6665 7265 6e63 652c 2065 7863 6570 7420  ference, except 
-00009ec0: 666f 7220 7468 6520 7061 7261 6d65 7465  for the paramete
-00009ed0: 7220 602d 6360 206f 7220 602d 2d63 7363  r `-c` or `--csc
-00009ee0: 6f72 6560 2c20 7768 6963 6820 7261 6e67  ore`, which rang
-00009ef0: 6573 2062 6574 7765 656e 2030 2061 6e64  es between 0 and
-00009f00: 2031 2061 6e64 2069 7320 7573 6564 2074   1 and is used t
-00009f10: 6f20 7265 6c61 7820 7468 6520 7369 6d69  o relax the simi
-00009f20: 6c61 7269 7479 2063 7574 6f66 6620 6672  larity cutoff fr
-00009f30: 6f6d 2074 6865 2065 7863 6c75 7369 7665  om the exclusive
-00009f40: 2072 6563 6970 726f 6361 6c20 6265 7374   reciprocal best
-00009f50: 2068 6974 7320 746f 2061 2063 6572 7461   hits to a certa
-00009f60: 696e 2072 6174 696f 2061 7320 746f 2074  in ratio as to t
-00009f70: 6865 2062 6573 7420 6869 7473 2e20 466f  he best hits. Fo
-00009f80: 7220 696e 7374 616e 6365 2c20 6966 2074  r instance, if t
-00009f90: 6865 2067 656e 6520 6231 2066 726f 6d20  he gene b1 from 
-00009fa0: 6765 6e6f 6d65 2042 2068 6173 2074 6865  genome B has the
-00009fb0: 2062 6573 7420 6869 7420 6765 6e65 2061   best hit gene a
-00009fc0: 3120 6672 6f6d 2067 656e 6f6d 6520 4120  1 from genome A 
-00009fd0: 7769 7468 2074 6865 2062 6974 2073 636f  with the bit sco
-00009fe0: 7265 2061 7320 3130 302c 2077 6869 6368  re as 100, which
-00009ff0: 2069 7320 6120 7363 6f72 696e 6720 6d61   is a scoring ma
-0000a000: 7472 6978 2069 6e64 6570 656e 6465 6e74  trix independent
-0000a010: 206d 6561 7375 7265 206f 6620 7468 6520   measure of the 
-0000a020: 286c 6f63 616c 2920 7369 6d69 6c61 7269  (local) similari
-0000a030: 7479 206f 6620 7468 6520 7477 6f20 616c  ty of the two al
-0000a040: 6967 6e65 6420 7365 7175 656e 6365 732c  igned sequences,
-0000a050: 2077 6974 6820 6c61 7267 6572 2076 616c   with larger val
-0000a060: 7565 7320 696e 6469 6361 7469 6e67 2068  ues indicating h
-0000a070: 6967 6865 7220 7369 6d69 6c61 7269 7469  igher similariti
-0000a080: 6573 2c20 6769 7665 6e20 7468 6520 602d  es, given the `-
-0000a090: 6320 302e 3960 2c20 6765 6e65 7320 6672  c 0.9`, genes fr
-0000a0a0: 6f6d 2067 656e 6f6d 6520 4120 7768 6963  om genome A whic
-0000a0b0: 6820 6861 7320 7468 6520 6269 7420 7363  h has the bit sc
-0000a0c0: 6f72 6520 7769 7468 2067 656e 6520 6231  ore with gene b1
-0000a0d0: 2068 6967 6865 7220 7468 616e 2030 2e39   higher than 0.9
-0000a0e0: 7831 3030 2077 696c 6c20 616c 736f 2062  x100 will also b
-0000a0f0: 6520 7772 6974 7465 6e20 696e 2074 6865  e written in the
-0000a100: 2072 6573 756c 7420 6669 6c65 2c20 7768   result file, wh
-0000a110: 6963 6820 696e 2061 2073 656e 7365 2061  ich in a sense a
-0000a120: 7265 206e 6f74 2052 4248 7320 616e 796d  re not RBHs anym
-0000a130: 6f72 6520 6f66 2063 6f75 7273 652c 2062  ore of course, b
-0000a140: 7574 2074 6865 2068 6967 686c 7920 7369  ut the highly si
-0000a150: 6d69 6c61 7220 686f 6d6f 6c6f 6775 6520  milar homologue 
-0000a160: 7061 6972 732e 2049 6620 6d6f 7265 2074  pairs. If more t
-0000a170: 6861 6e20 3220 7365 7175 656e 6365 2066  han 2 sequence f
-0000a180: 696c 6573 2077 6572 6520 7072 6f76 6964  iles were provid
-0000a190: 6564 2c20 6576 6572 7920 7061 6972 2d77  ed, every pair-w
-0000a1a0: 6973 6520 5242 4873 2077 6f75 6c64 2062  ise RBHs would b
-0000a1b0: 6520 6361 6c63 756c 6174 6564 2065 7863  e calculated exc
-0000a1c0: 6570 7420 666f 7220 7175 6572 7969 6e67  ept for querying
-0000a1d0: 2074 6865 2073 616d 6520 7365 7175 656e   the same sequen
-0000a1e0: 6365 2069 7473 656c 662e 2054 6865 206e  ce itself. The n
-0000a1f0: 756d 6265 7220 6f66 2070 6172 616c 6c65  umber of paralle
-0000a200: 6c20 7468 7265 6164 7320 746f 2062 6f6f  l threads to boo
-0000a210: 7374 6572 2074 6865 2072 756e 6e69 6e67  ster the running
-0000a220: 2073 7065 6564 2063 616e 2062 6520 7365   speed can be se
-0000a230: 7420 6279 2074 6865 206f 7074 696f 6e20  t by the option 
-0000a240: 602d 6e60 206f 7220 602d 2d6e 7468 7265  `-n` or `--nthre
-0000a250: 6164 7360 2077 6869 6368 2069 7320 7375  ads` which is su
-0000a260: 6767 6573 7465 6420 746f 2062 6520 7365  ggested to be se
-0000a270: 7420 6173 2028 4e2d 3129 4e2f 3220 7768  t as (N-1)N/2 wh
-0000a280: 6572 6520 4e20 6973 2074 6865 206e 756d  ere N is the num
-0000a290: 6265 7220 6f66 2063 6473 2066 696c 6573  ber of cds files
-0000a2a0: 2074 6f20 6163 6869 6576 6520 7468 6520   to achieve the 
-0000a2b0: 6869 6768 6573 7420 6566 6669 6369 656e  highest efficien
-0000a2c0: 6379 2e20 5468 6520 6f70 7469 6f6e 2060  cy. The option `
-0000a2d0: 2d2d 6f67 666f 726d 6174 6020 6361 6e20  --ogformat` can 
-0000a2e0: 6265 2073 6574 2074 6f20 6164 6420 696e  be set to add in
-0000a2f0: 6465 7820 2866 6f72 2069 6e73 7461 6e63  dex (for instanc
-0000a300: 6520 4746 3030 3030 3030 3031 2920 746f  e GF00000001) to
-0000a310: 2074 6865 206f 7574 7075 7420 5242 4820   the output RBH 
-0000a320: 6765 6e65 2066 616d 696c 6965 7320 7768  gene families wh
-0000a330: 6963 6820 6361 6e20 6265 2066 7572 7468  ich can be furth
-0000a340: 6572 2075 7365 6420 696e 2074 6865 202a  er used in the *
-0000a350: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2063  K*<sub>S</sub> c
-0000a360: 616c 6375 6c61 7469 6f6e 2062 7920 6077  alculation by `w
-0000a370: 6764 206b 7364 602e 0a0a 2a2a 5468 6520  gd ksd`...**The 
-0000a380: 7375 6767 6573 7465 6420 636f 6d6d 616e  suggested comman
-0000a390: 6420 746f 2073 7461 7274 2077 6974 6820  d to start with 
-0000a3a0: 6973 2061 6c73 6f20 756e 6465 7220 7468  is also under th
-0000a3b0: 6520 6465 6661 756c 7420 7365 7474 696e  e default settin
-0000a3c0: 6720 7769 7468 2074 6865 2063 6f6d 6d61  g with the comma
-0000a3d0: 6e64 2073 686f 776e 2062 656c 6f77 2a2a  nd shown below**
-0000a3e0: 0a0a 6060 600a 7767 6420 646d 6420 7365  ..```.wgd dmd se
-0000a3f0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
-0000a400: 320a 6060 600a 0a2a 2a54 6865 2064 656c  2.```..**The del
-0000a410: 696e 6561 7469 6f6e 206f 6620 6c6f 6361  ineation of loca
-0000a420: 6c20 4d52 4248 732a 2a0a 6060 600a 7767  l MRBHs**.```.wg
-0000a430: 6420 646d 6420 7365 7175 656e 6365 3120  d dmd sequence1 
-0000a440: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
-0000a450: 6365 3320 2d66 2073 6571 7565 6e63 6531  ce3 -f sequence1
-0000a460: 202d 6520 3165 2d31 3020 2d62 7320 3130   -e 1e-10 -bs 10
-0000a470: 3020 2d6e 7020 3520 282d 6e6e 2920 282d  0 -np 5 (-nn) (-
-0000a480: 6e20 3429 2028 2d6b 6629 2028 2d6b 6429  n 4) (-kf) (-kd)
-0000a490: 2028 2d63 2030 2e39 2920 282d 2d74 6f5f   (-c 0.9) (--to_
-0000a4a0: 7374 6f70 2920 282d 2d63 6473 2920 282d  stop) (--cds) (-
-0000a4b0: 6f20 7767 645f 646d 6429 2028 2d74 2077  o wgd_dmd) (-t w
-0000a4c0: 6f72 6b69 6e67 5f74 6d70 290a 6060 600a  orking_tmp).```.
-0000a4d0: 0a2a 2a54 6865 2064 6973 7469 6e63 7469  .**The distincti
-0000a4e0: 6f6e 2062 6574 7765 656e 206c 6f63 616c  on between local
-0000a4f0: 2061 6e64 2067 6c6f 6261 6c20 2868 6572   and global (her
-0000a500: 6575 6e64 6572 2920 4d52 4248 7320 6973  eunder) MRBHs is
-0000a510: 2074 6861 7420 6c6f 6361 6c20 4d52 4248   that local MRBH
-0000a520: 7320 6172 6520 7468 6520 7265 7375 6c74  s are the result
-0000a530: 7320 6f66 206d 6572 6765 6420 5242 4873  s of merged RBHs
-0000a540: 206f 6e20 6120 6a6f 696e 7420 666f 6361   on a joint foca
-0000a550: 6c20 7370 6563 6965 732c 2066 6f72 2069  l species, for i
-0000a560: 6e73 7461 6e63 6520 696e 2061 2074 6872  nstance in a thr
-0000a570: 6565 2073 7065 6369 6573 2073 7973 7465  ee species syste
-0000a580: 6d20 2841 2c28 422c 4329 292c 2074 6865  m (A,(B,C)), the
-0000a590: 206c 6f63 616c 204d 5242 4873 206f 6620   local MRBHs of 
-0000a5a0: 4120 6f6e 6c79 2072 6571 7569 7265 2074  A only require t
-0000a5b0: 6865 2063 616c 6375 6c61 7469 6f6e 206f  he calculation o
-0000a5c0: 6620 5242 4873 2062 6574 7765 656e 2041  f RBHs between A
-0000a5d0: 2061 6e64 2043 2028 6465 6e6f 7465 6420   and C (denoted 
-0000a5e0: 6173 2041 4329 2061 6e64 2041 4220 616e  as AC) and AB an
-0000a5f0: 6420 7468 656e 2074 6865 206d 6572 6769  d then the mergi
-0000a600: 6e67 206f 6620 4142 2061 6e64 2041 4320  ng of AB and AC 
-0000a610: 6174 2074 6865 2061 7869 7320 6f66 2041  at the axis of A
-0000a620: 2c20 7768 696c 6520 7468 6520 676c 6f61  , while the gloa
-0000a630: 626c 204d 5242 4873 2061 7265 2069 6e64  bl MRBHs are ind
-0000a640: 6570 656e 6465 6e74 206f 6620 666f 6361  ependent of foca
-0000a650: 6c20 7370 6563 6965 7320 696e 2074 6861  l species in tha
-0000a660: 7420 6974 206a 7573 7420 6361 6c6c 7320  t it just calls 
-0000a670: 7468 6520 6361 6c63 756c 6174 696f 6e73  the calculations
-0000a680: 206f 6620 616c 6c20 706f 7373 6962 6c65   of all possible
-0000a690: 2073 7065 6369 6573 2070 6169 7220 286e   species pair (n
-0000a6a0: 6f74 2073 656c 6620 746f 2073 656c 6620  ot self to self 
-0000a6b0: 7370 6563 6965 7320 7061 6972 292c 2073  species pair), s
-0000a6c0: 7563 6820 7468 6174 2041 422c 2041 432c  uch that AB, AC,
-0000a6d0: 2061 6e64 2042 4320 6172 6520 616c 6c20   and BC are all 
-0000a6e0: 746f 2062 6520 6361 6c63 756c 6174 6564  to be calculated
-0000a6f0: 2061 6e64 206d 6572 6765 642e 2a2a 0a0a   and merged.**..
-0000a700: 5477 6f20 7479 7065 7320 6f66 204d 5242  Two types of MRB
-0000a710: 4873 2061 7320 696e 7465 7072 6574 6174  Hs as intepretat
-0000a720: 6564 2061 626f 7665 2063 616e 2062 6520  ed above can be 
-0000a730: 6465 6c69 6e65 6174 6564 2062 7920 6077  delineated by `w
-0000a740: 6764 2064 6d64 602c 2074 6865 206c 6f63  gd dmd`, the loc
-0000a750: 616c 204d 5242 4873 2061 6e64 2074 6865  al MRBHs and the
-0000a760: 2067 6c6f 6261 6c20 4d52 4248 732e 2054   global MRBHs. T
-0000a770: 6865 206c 6f63 616c 204d 5242 4873 2061  he local MRBHs a
-0000a780: 7265 2063 6f6e 7374 7275 6374 6564 2062  re constructed b
-0000a790: 7920 6d65 7267 696e 6720 616c 6c20 7468  y merging all th
-0000a7a0: 6520 7265 6c65 7661 6e74 2052 4248 7320  e relevant RBHs 
-0000a7b0: 6f6e 6c79 2077 6974 6820 7468 6520 666f  only with the fo
-0000a7c0: 6361 6c20 7370 6563 6965 732c 2077 6869  cal species, whi
-0000a7d0: 6368 2069 7320 7365 7420 6279 2060 2d66  ch is set by `-f
-0000a7e0: 6020 6f72 2060 2d2d 666f 6375 7360 2e20  ` or `--focus`. 
-0000a7f0: 5468 6520 7061 7261 6d65 7465 7220 602d  The parameter `-
-0000a800: 6b66 6020 6f72 2060 2d2d 6b65 6570 6661  kf` or `--keepfa
-0000a810: 7374 6160 2063 616e 2062 6520 7365 7420  sta` can be set 
-0000a820: 746f 2072 6574 6169 6e20 7468 6520 7365  to retain the se
-0000a830: 7175 656e 6365 2069 6e66 6f72 6d61 7469  quence informati
-0000a840: 6f6e 206f 6620 6561 6368 204d 5242 482e  on of each MRBH.
-0000a850: 2054 6865 2070 6172 616d 6574 6572 2060   The parameter `
-0000a860: 2d6b 6460 206f 7220 602d 2d6b 6565 7064  -kd` or `--keepd
-0000a870: 7570 6c69 6361 7465 7360 2064 6574 6572  uplicates` deter
-0000a880: 6d69 6e65 7320 7768 6574 6865 7220 7468  mines whether th
-0000a890: 6520 7361 6d65 2067 656e 6573 2063 616e  e same genes can
-0000a8a0: 2061 7070 6561 7220 696e 2064 6966 6665   appear in diffe
-0000a8b0: 7265 6e74 206c 6f63 616c 204d 5242 4873  rent local MRBHs
-0000a8c0: 2e20 4e6f 726d 616c 6c79 2074 6865 7265  . Normally there
-0000a8d0: 2077 696c 6c20 6265 206e 6f20 6475 706c   will be no dupl
-0000a8e0: 6963 6174 6573 2069 6e20 7468 6520 6c6f  icates in the lo
-0000a8f0: 6361 6c20 4d52 4248 7320 6275 7420 6966  cal MRBHs but if
-0000a900: 2075 7365 7273 2073 6574 2074 6865 2060   users set the `
-0000a910: 2d63 6020 6173 2030 2e39 2028 6f72 2061  -c` as 0.9 (or a
-0000a920: 6e79 2076 616c 7565 2073 6d61 6c6c 6572  ny value smaller
-0000a930: 2074 6861 6e20 3129 2c20 6974 2773 206c   than 1), it's l
-0000a940: 696b 656c 7920 7468 6174 2074 6865 2073  ikely that the s
-0000a950: 616d 6520 6765 6e65 2077 696c 6c20 6861  ame gene will ha
-0000a960: 7665 2063 6861 6e63 6520 746f 2061 7070  ve chance to app
-0000a970: 6561 7220 6d75 6c74 6970 6c65 2074 696d  ear multiple tim
-0000a980: 6573 2069 6e20 6469 6666 6572 656e 7420  es in different 
-0000a990: 6c6f 6361 6c20 4d52 4248 732e 2054 6861  local MRBHs. Tha
-0000a9a0: 7420 6973 2074 6f20 7361 792c 2074 6865  t is to say, the
-0000a9b0: 2070 6172 616d 6574 6572 2060 2d6b 6460   parameter `-kd`
-0000a9c0: 2069 7320 6d65 616e 696e 6766 756c 206f   is meaningful o
-0000a9d0: 6e6c 7920 7768 656e 2069 7427 7320 7365  nly when it's se
-0000a9e0: 7420 746f 6765 7468 6572 2077 6974 6820  t together with 
-0000a9f0: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
-0000aa00: 6360 2e20 5468 6520 6e75 6d62 6572 206f  c`. The number o
-0000aa10: 6620 7061 7261 6c6c 656c 2074 6872 6561  f parallel threa
-0000aa20: 6473 2069 7320 7375 6767 6573 7465 6420  ds is suggested 
-0000aa30: 746f 2062 6520 7365 7420 6173 2074 6865  to be set as the
-0000aa40: 206e 756d 6265 7220 6f66 2063 6473 2066   number of cds f
-0000aa50: 696c 6573 206d 696e 7573 2031 2e0a 0a2a  iles minus 1...*
-0000aa60: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
-0000aa70: 7274 696e 6720 7275 6e20 6973 2075 6e64  rting run is und
-0000aa80: 6572 2074 6865 2064 6566 6175 6c74 2070  er the default p
-0000aa90: 6172 616d 6574 6572 2077 6974 6820 7468  arameter with th
-0000aaa0: 6520 636f 6d6d 616e 6420 7368 6f77 6e20  e command shown 
-0000aab0: 6173 2062 656c 6f77 2e2a 2a0a 0a60 6060  as below.**..```
-0000aac0: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
-0000aad0: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000aae0: 7565 6e63 6533 202d 6620 7365 7175 656e  uence3 -f sequen
-0000aaf0: 6365 310a 6060 600a 0a2a 2a54 6865 2064  ce1.```..**The d
-0000ab00: 656c 696e 6561 7469 6f6e 206f 6620 676c  elineation of gl
-0000ab10: 6f62 616c 204d 5242 4873 2a2a 0a60 6060  obal MRBHs**.```
-0000ab20: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
-0000ab30: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000ab40: 7565 6e63 6533 202d 676d 202d 6520 3165  uence3 -gm -e 1e
-0000ab50: 2d31 3020 2d62 7320 3130 3020 2d6e 7020  -10 -bs 100 -np 
-0000ab60: 3520 282d 6e6e 2920 282d 6e20 3429 2028  5 (-nn) (-n 4) (
-0000ab70: 2d6b 6629 2028 2d6b 6429 2028 2d63 2030  -kf) (-kd) (-c 0
-0000ab80: 2e39 2920 282d 2d74 6f5f 7374 6f70 2920  .9) (--to_stop) 
-0000ab90: 282d 2d63 6473 2920 282d 6f20 7767 645f  (--cds) (-o wgd_
-0000aba0: 646d 6429 2028 2d74 2077 6f72 6b69 6e67  dmd) (-t working
-0000abb0: 5f74 6d70 290a 6060 600a 0a54 6865 2067  _tmp).```..The g
-0000abc0: 6c6f 6261 6c20 4d52 4248 7320 6973 2063  lobal MRBHs is c
-0000abd0: 6f6e 7374 7275 6374 6564 2062 7920 6578  onstructed by ex
-0000abe0: 6861 7573 7469 7665 6c79 206d 6572 6769  haustively mergi
-0000abf0: 6e67 2061 6c6c 2074 6865 2070 6f73 7369  ng all the possi
-0000ac00: 626c 6520 7061 6972 2d77 6973 6520 5242  ble pair-wise RB
-0000ac10: 4873 2065 7863 6570 7420 666f 7220 7175  Hs except for qu
-0000ac20: 6572 7969 6e67 2074 6865 2073 6571 7565  erying the seque
-0000ac30: 6e63 6520 6974 7365 6c66 2c20 7768 6963  nce itself, whic
-0000ac40: 6820 6361 6e20 6265 2069 6e69 7469 6174  h can be initiat
-0000ac50: 6564 2062 7920 6164 6420 7468 6520 666c  ed by add the fl
-0000ac60: 6167 2060 2d67 6d60 206f 7220 602d 2d67  ag `-gm` or `--g
-0000ac70: 6c6f 6261 6c6d 7262 6860 2e20 5468 6520  lobalmrbh`. The 
-0000ac80: 7265 7374 206f 6620 7265 6c65 7661 6e74  rest of relevant
-0000ac90: 2070 6172 616d 6574 6572 7320 7374 6179   parameters stay
-0000aca0: 7320 7468 6520 7361 6d65 2061 7320 7468  s the same as th
-0000acb0: 6520 6c6f 6361 6c20 4d52 4248 732e 2054  e local MRBHs. T
-0000acc0: 6865 206e 756d 6265 7220 6f66 2070 6172  he number of par
-0000acd0: 616c 6c65 6c20 7468 7265 6164 7320 6973  allel threads is
-0000ace0: 2073 7567 6765 7374 6564 2074 6f20 6265   suggested to be
-0000acf0: 2073 6574 2061 7320 284e 2d31 294e 2f32   set as (N-1)N/2
-0000ad00: 2074 6f6f 2077 6865 7265 204e 2069 7320   too where N is 
-0000ad10: 7468 6520 6e75 6d62 6572 206f 6620 6364  the number of cd
-0000ad20: 7320 6669 6c65 7320 746f 2061 6368 6965  s files to achie
-0000ad30: 7665 2074 6865 2068 6967 6865 7374 2065  ve the highest e
-0000ad40: 6666 6963 6965 6e63 792e 0a0a 2a2a 4120  fficiency...**A 
-0000ad50: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
-0000ad60: 6e67 2072 756e 2069 7320 756e 6465 7220  ng run is under 
-0000ad70: 7468 6520 6465 6661 756c 7420 7061 7261  the default para
-0000ad80: 6d65 7465 7220 7769 7468 2074 6865 2063  meter with the c
-0000ad90: 6f6d 6d61 6e64 2073 686f 776e 2061 7320  ommand shown as 
-0000ada0: 6265 6c6f 772e 2a2a 0a0a 6060 600a 7767  below.**..```.wg
-0000adb0: 6420 646d 6420 7365 7175 656e 6365 3120  d dmd sequence1 
-0000adc0: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
-0000add0: 6365 3320 2d67 6d0a 6060 600a 0a2a 2a54  ce3 -gm.```..**T
-0000ade0: 6865 2064 656c 696e 6561 7469 6f6e 206f  he delineation o
-0000adf0: 6620 6f72 7468 6f67 726f 7570 732a 2a0a  f orthogroups**.
-0000ae00: 6060 600a 7767 6420 646d 6420 7365 7175  ```.wgd dmd sequ
-0000ae10: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000ae20: 7365 7175 656e 6365 3320 2d6f 6920 2d6f  sequence3 -oi -o
-0000ae30: 6f20 2d65 2031 652d 3130 202d 6273 2031  o -e 1e-10 -bs 1
-0000ae40: 3030 202d 6e70 2035 2028 2d6e 6e29 2028  00 -np 5 (-nn) (
-0000ae50: 2d63 6329 2028 2d74 6529 2028 2d6d 6320  -cc) (-te) (-mc 
-0000ae60: 302e 3829 2028 2d67 6e29 2028 2d74 7265  0.8) (-gn) (-tre
-0000ae70: 6520 2766 6173 7474 7265 6527 2920 282d  e 'fasttree') (-
-0000ae80: 7473 2027 2d66 6173 7465 7374 2729 2028  ts '-fastest') (
-0000ae90: 2d6e 2034 2920 282d 2d74 6f5f 7374 6f70  -n 4) (--to_stop
-0000aea0: 2920 282d 2d63 6473 2920 282d 6f20 7767  ) (--cds) (-o wg
-0000aeb0: 645f 646d 6429 2028 2d74 2077 6f72 6b69  d_dmd) (-t worki
-0000aec0: 6e67 5f74 6d70 290a 6060 600a 0a49 6e20  ng_tmp).```..In 
-0000aed0: 6077 6764 2076 3260 2c20 7765 2061 6c73  `wgd v2`, we als
-0000aee0: 6f20 696d 706c 656d 656e 7465 6420 616e  o implemented an
-0000aef0: 2061 6c67 6f72 6974 686d 206f 6620 6465   algorithm of de
-0000af00: 6c69 6e65 6174 696e 6720 6f72 7468 6f67  lineating orthog
-0000af10: 726f 7570 732c 2077 6869 6368 2063 616e  roups, which can
-0000af20: 2062 6520 696e 6974 6961 7465 6420 7769   be initiated wi
-0000af30: 7468 2074 6865 2070 6172 616d 6574 6572  th the parameter
-0000af40: 2060 2d6f 6960 206f 7220 602d 2d6f 7274   `-oi` or `--ort
-0000af50: 686f 696e 6665 7260 2e20 5477 6f20 7761  hoinfer`. Two wa
-0000af60: 7973 206f 6620 6465 6c69 6e65 6174 696f  ys of delineatio
-0000af70: 6e20 6361 6e20 6265 2063 686f 7365 6e2c  n can be chosen,
-0000af80: 2074 6865 2063 6f6e 6361 7465 6e61 7469   the concatenati
-0000af90: 6f6e 2077 6179 2028 7365 7420 6279 2074  on way (set by t
-0000afa0: 6865 2070 6172 616d 6574 6572 2060 2d63  he parameter `-c
-0000afb0: 6360 206f 7220 602d 2d63 6f6e 6361 7460  c` or `--concat`
-0000afc0: 2920 6f72 2074 6865 206e 6f6e 2d63 6f6e  ) or the non-con
-0000afd0: 6361 7465 6e61 7469 6f6e 2028 6465 6661  catenation (defa
-0000afe0: 756c 7429 2077 6179 2e20 496e 2062 7269  ult) way. In bri
-0000aff0: 6566 2c20 7468 6520 636f 6e63 6174 656e  ef, the concaten
-0000b000: 6174 696f 6e20 7761 7920 6f66 2064 656c  ation way of del
-0000b010: 696e 6561 7469 6e67 206f 7274 686f 6772  ineating orthogr
-0000b020: 6f75 7073 2073 7461 7274 7320 7769 7468  oups starts with
-0000b030: 2063 6f6e 6361 7465 6e61 7469 6e67 2061   concatenating a
-0000b040: 6c6c 2074 6865 2073 6571 7565 6e63 6573  ll the sequences
-0000b050: 2069 6e74 6f20 6120 7369 6e67 6c65 2073   into a single s
-0000b060: 6571 7565 6e63 6520 6669 6c65 2061 6e64  equence file and
-0000b070: 2074 6865 6e20 696e 6665 7272 696e 6720   then inferring 
-0000b080: 7468 6520 7768 6f6c 6520 7061 7261 6e6f  the whole parano
-0000b090: 6d65 206f 6620 7468 6973 2073 696e 676c  me of this singl
-0000b0a0: 6520 7365 7175 656e 6365 2066 696c 6520  e sequence file 
-0000b0b0: 7769 7468 2074 6865 2063 6c75 7374 6572  with the cluster
-0000b0c0: 696e 6720 7265 7375 6c74 7320 6d61 7070  ing results mapp
-0000b0d0: 6564 2062 6163 6b20 746f 2074 6865 2062  ed back to the b
-0000b0e0: 656c 6f6e 6769 6e67 2073 7065 6369 6573  elonging species
-0000b0f0: 2e20 5768 696c 6520 7468 6520 6e6f 6e2d  . While the non-
-0000b100: 636f 6e63 6174 656e 6174 696f 6e20 7761  concatenation wa
-0000b110: 7920 7374 6172 7473 2077 6974 6820 7265  y starts with re
-0000b120: 7370 6563 7469 7665 2070 6169 722d 7769  spective pair-wi
-0000b130: 7365 2064 6961 6d6f 6e64 2073 6561 7263  se diamond searc
-0000b140: 6820 2869 6e63 6c75 6469 6e67 2071 7565  h (including que
-0000b150: 7279 696e 6720 7468 6520 7361 6d65 2073  rying the same s
-0000b160: 6571 7565 6e63 6520 6974 7365 6c66 2920  equence itself) 
-0000b170: 616e 6420 7468 656e 2061 6c6c 2074 6865  and then all the
-0000b180: 2073 6571 7565 6e63 6520 7369 6d69 6c61   sequence simila
-0000b190: 7269 7479 2074 6162 6c65 7320 7769 6c6c  rity tables will
-0000b1a0: 2062 6520 636f 6e63 6174 656e 6174 6564   be concatenated
-0000b1b0: 2061 6e64 2063 6c75 7374 6572 6564 2069   and clustered i
-0000b1c0: 6e74 6f20 6f72 7468 6f67 726f 7570 732e  nto orthogroups.
-0000b1d0: 2053 6f6d 6520 6f74 6865 7220 706f 7373   Some other poss
-0000b1e0: 6962 6c79 2075 7365 6675 6c20 706f 7374  ibly useful post
-0000b1f0: 2d63 6c75 7374 6572 696e 6720 6675 6e63  -clustering func
-0000b200: 7469 6f6e 7320 6361 6e20 6265 2069 6e69  tions can be ini
-0000b210: 7469 6174 6564 2c20 696e 636c 7564 696e  tiated, includin
-0000b220: 6720 7468 6520 7061 7261 6d65 7465 7220  g the parameter 
-0000b230: 602d 7465 6020 6f72 2060 2d2d 7465 7374  `-te` or `--test
-0000b240: 736f 6760 2c20 7768 6963 6820 6361 6e20  sog`, which can 
-0000b250: 6265 2073 6574 2074 6f20 7374 6172 7420  be set to start 
-0000b260: 7468 6520 756e 6269 6173 6564 2074 6573  the unbiased tes
-0000b270: 7420 6f66 2073 696e 676c 652d 636f 7079  t of single-copy
-0000b280: 2067 656e 6520 6661 6d69 6c69 6573 2028   gene families (
-0000b290: 6e6f 7465 2074 6861 7420 7468 6973 2066  note that this f
-0000b2a0: 756e 6374 696f 6e20 6e65 6564 7320 6068  unction needs `h
-0000b2b0: 6d6d 6572 6020 2876 332e 3162 3229 2074  mmer` (v3.1b2) t
-0000b2c0: 6f20 6265 2069 6e73 7461 6c6c 6564 2069  o be installed i
-0000b2d0: 6e20 7468 6520 656e 7669 726f 6e6d 656e  n the environmen
-0000b2e0: 7420 7061 7468 292c 2074 6865 2070 6172  t path), the par
-0000b2f0: 616d 6574 6572 2060 2d6d 6360 206f 7220  ameter `-mc` or 
-0000b300: 602d 2d6d 736f 6763 7574 602c 2072 616e  `--msogcut`, ran
-0000b310: 6769 6e67 2062 6574 7765 656e 2030 2074  ging between 0 t
-0000b320: 6f20 312c 2077 6869 6368 2063 616e 2062  o 1, which can b
-0000b330: 6520 7365 7420 746f 2073 6561 7263 6820  e set to search 
-0000b340: 7468 6520 736f 2d63 616c 6c65 6420 6d6f  the so-called mo
-0000b350: 7374 6c79 2073 696e 676c 652d 636f 7079  stly single-copy
-0000b360: 2066 616d 696c 7920 7768 6963 6820 6861   family which ha
-0000b370: 7320 6869 6768 6572 2074 6861 6e20 6365  s higher than ce
-0000b380: 7274 6169 6e20 6375 742d 6f66 6620 7065  rtain cut-off pe
-0000b390: 7263 656e 7461 6765 206f 6620 7370 6563  rcentage of spec
-0000b3a0: 6965 7320 636f 7665 7261 6765 2c20 7468  ies coverage, th
-0000b3b0: 6520 7061 7261 6d65 7465 7220 602d 676e  e parameter `-gn
-0000b3c0: 6020 6f72 2060 2d2d 6765 746e 736f 6760  ` or `--getnsog`
-0000b3d0: 2c20 7768 6963 6820 6361 6e20 6265 2073  , which can be s
-0000b3e0: 6574 2074 6f20 7365 6172 6368 2066 6f72  et to search for
-0000b3f0: 206e 6573 7465 6420 7369 6e67 6c65 2d63   nested single-c
-0000b400: 6f70 7920 6765 6e65 2066 616d 696c 6965  opy gene familie
-0000b410: 7320 284e 534f 4773 2920 7768 6963 6820  s (NSOGs) which 
-0000b420: 6973 206f 7269 6769 6e61 6c6c 7920 6d75  is originally mu
-0000b430: 6c74 6979 2d63 6f70 7920 6275 7420 6861  ltiy-copy but ha
-0000b440: 7320 6120 286d 6f73 746c 7929 2073 696e  s a (mostly) sin
-0000b450: 676c 652d 636f 7079 2062 7261 6e63 6820  gle-copy branch 
-0000b460: 2877 6869 6368 2072 6571 7569 7265 7320  (which requires 
-0000b470: 7468 6520 6368 6f73 656e 2074 7265 652d  the chosen tree-
-0000b480: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
-0000b490: 6d20 7365 7420 6279 2060 2d74 7265 6560  m set by `-tree`
-0000b4a0: 206f 7220 602d 2d74 7265 655f 6d65 7468   or `--tree_meth
-0000b4b0: 6f64 6020 746f 2062 6520 7072 652d 696e  od` to be pre-in
-0000b4c0: 7374 616c 6c65 6420 696e 2074 6865 2065  stalled in the e
-0000b4d0: 6e76 6972 6f6e 6d65 6e74 2070 6174 6820  nvironment path 
-0000b4e0: 7769 7468 2074 6865 2070 6172 616d 6574  with the paramet
-0000b4f0: 6572 7320 7365 7474 696e 6720 666f 7220  ers setting for 
-0000b500: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
-0000b510: 6e63 6520 636f 6e74 726f 6c6c 6564 2062  nce controlled b
-0000b520: 7920 602d 7473 6020 6f72 2060 2d2d 7472  y `-ts` or `--tr
-0000b530: 6565 7365 7460 292e 2054 6865 2070 726f  eeset`). The pro
-0000b540: 6772 616d 2060 7767 6420 646d 6460 2077  gram `wgd dmd` w
-0000b550: 6f75 6c64 2073 7469 6c6c 2063 6f6e 6475  ould still condu
-0000b560: 6374 2074 6865 2052 4248 7320 6361 6c63  ct the RBHs calc
-0000b570: 756c 6174 696f 6e20 756e 6c65 7373 2074  ulation unless t
-0000b580: 6865 2070 6172 616d 6574 6572 2060 2d6f  he parameter `-o
-0000b590: 6f60 206f 7220 602d 2d6f 6e6c 796f 7274  o` or `--onlyort
-0000b5a0: 686f 6020 7761 7320 7365 742e 2049 6620  ho` was set. If 
-0000b5b0: 6f6e 6520 6f6e 6c79 2077 616e 7473 2074  one only wants t
-0000b5c0: 6f20 696e 6665 7220 7468 6520 6f72 7468  o infer the orth
-0000b5d0: 6f67 726f 7570 732c 2069 7427 7320 7375  ogroups, it's su
-0000b5e0: 6767 6573 7465 6420 746f 2061 6464 2074  ggested to add t
-0000b5f0: 6865 2066 6c61 6720 602d 6f6f 6020 746f  he flag `-oo` to
-0000b600: 206a 7573 7420 696d 706c 656d 656e 7420   just implement 
-0000b610: 7468 6520 6f72 7468 6f67 726f 7570 7320  the orthogroups 
-0000b620: 6465 6c69 6e65 6174 696f 6e20 616e 616c  delineation anal
-0000b630: 7973 6973 2e20 5468 6520 6e75 6d62 6572  ysis. The number
-0000b640: 206f 6620 7061 7261 6c6c 656c 2074 6872   of parallel thr
-0000b650: 6561 6473 2069 7320 7375 6767 6573 7465  eads is suggeste
-0000b660: 6420 746f 2062 6520 7365 7420 6173 2028  d to be set as (
-0000b670: 4e2b 3129 4e2f 3220 7768 6572 6520 4e20  N+1)N/2 where N 
-0000b680: 6973 2074 6865 206e 756d 6265 7220 6f66  is the number of
-0000b690: 2063 6473 2066 696c 6573 2074 6f20 6163   cds files to ac
-0000b6a0: 6869 6576 6520 7468 6520 6869 6768 6573  hieve the highes
-0000b6b0: 7420 6566 6669 6369 656e 6379 2073 696e  t efficiency sin
-0000b6c0: 6365 2074 6865 2073 656c 662d 636f 6d70  ce the self-comp
-0000b6d0: 6172 6973 6f6e 2069 7320 616c 736f 2069  arison is also i
-0000b6e0: 6e63 6c75 6465 642e 0a0a 2a2a 5468 6520  ncluded...**The 
-0000b6f0: 6465 6661 756c 7420 7365 7474 696e 6720  default setting 
-0000b700: 6f66 2070 6172 616d 6574 6572 7320 6973  of parameters is
-0000b710: 2061 2072 6561 736f 6e61 626c 6520 7374   a reasonable st
-0000b720: 6172 7469 6e67 2070 6f69 6e74 2077 6974  arting point wit
-0000b730: 6820 7468 6520 636f 6d6d 616e 6420 6173  h the command as
-0000b740: 2062 656c 6f77 2e2a 2a0a 0a60 6060 0a77   below.**..```.w
-0000b750: 6764 2064 6d64 2073 6571 7565 6e63 6531  gd dmd sequence1
-0000b760: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
-0000b770: 6e63 6533 202d 6f69 202d 6f6f 0a60 6060  nce3 -oi -oo.```
-0000b780: 0a0a 2a2a 5468 6520 636f 6c6c 696e 6561  ..**The collinea
-0000b790: 7220 636f 616c 6573 6365 6e63 6520 696e  r coalescence in
-0000b7a0: 6665 7265 6e63 6520 6f66 2070 6879 6c6f  ference of phylo
-0000b7b0: 6765 6e79 2a2a 0a60 6060 0a77 6764 2064  geny**.```.wgd d
-0000b7c0: 6d64 2073 6571 7565 6e63 6531 2073 6571  md sequence1 seq
-0000b7d0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000b7e0: 202d 6170 2061 7064 6174 6120 2d73 6d20   -ap apdata -sm 
-0000b7f0: 736d 6461 7461 202d 6c65 206c 6564 6174  smdata -le ledat
-0000b800: 6120 2d67 7420 6774 6461 7461 202d 636f  a -gt gtdata -co
-0000b810: 6320 282d 7472 6565 2027 6661 7374 7472  c (-tree 'fasttr
-0000b820: 6565 2729 2028 2d74 7320 272d 6661 7374  ee') (-ts '-fast
-0000b830: 6573 7427 2920 282d 6e20 3429 2028 2d2d  est') (-n 4) (--
-0000b840: 746f 5f73 746f 7029 2028 2d2d 6364 7329  to_stop) (--cds)
-0000b850: 2028 2d6f 2077 6764 5f64 6d64 2920 282d   (-o wgd_dmd) (-
-0000b860: 7420 776f 726b 696e 675f 746d 7029 0a60  t working_tmp).`
-0000b870: 6060 0a0a 4120 6e6f 7665 6c20 7068 796c  ``..A novel phyl
-0000b880: 6f67 656e 6574 6963 2069 6e66 6572 656e  ogenetic inferen
-0000b890: 6365 206d 6574 686f 6420 6e61 6d65 6420  ce method named 
-0000b8a0: 2263 6f6c 6c69 6e65 6172 2063 6f61 6c65  "collinear coale
-0000b8b0: 7363 656e 6365 2069 6e66 6572 656e 6365  scence inference
-0000b8c0: 2220 6973 2061 6c73 6f20 696d 706c 656d  " is also implem
-0000b8d0: 656e 7465 6420 696e 2060 7767 6420 7632  ented in `wgd v2
-0000b8e0: 602e 2046 6f72 2074 6869 7320 616e 616c  `. For this anal
-0000b8f0: 7973 6973 2c20 7573 6572 7320 6e65 6564  ysis, users need
-0000b900: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
-0000b910: 616e 6368 6f72 2070 6f69 6e74 7320 6669  anchor points fi
-0000b920: 6c65 2062 7920 602d 6170 6020 6f72 2060  le by `-ap` or `
-0000b930: 2d2d 616e 6368 6f72 706f 696e 7473 602c  --anchorpoints`,
-0000b940: 2074 6865 2063 6f6c 6c69 6e65 6172 2073   the collinear s
-0000b950: 6567 6d65 6e74 7320 6669 6c65 2062 7920  egments file by 
-0000b960: 602d 736d 6020 6f72 2060 2d2d 7365 676d  `-sm` or `--segm
-0000b970: 656e 7473 602c 2074 6865 206c 6973 7473  ents`, the lists
-0000b980: 6567 6d65 6e74 7320 6669 6c65 2062 7920  egments file by 
-0000b990: 602d 6c65 6020 6f72 2060 2d2d 6c69 7374  `-le` or `--list
-0000b9a0: 656c 656d 656e 7473 602c 2061 6e64 2074  elements`, and t
-0000b9b0: 6865 2067 656e 6520 7461 626c 6520 6669  he gene table fi
-0000b9c0: 6c65 2062 7920 602d 6774 6020 6f72 2060  le by `-gt` or `
-0000b9d0: 2d2d 6765 6e65 7461 626c 6560 2c20 616c  --genetable`, al
-0000b9e0: 6c20 6f66 2077 6869 6368 2063 616e 2062  l of which can b
-0000b9f0: 6520 7072 6f64 7563 6564 2069 6e20 7468  e produced in th
-0000ba00: 6520 7072 6f67 7261 6d20 6077 6764 2073  e program `wgd s
-0000ba10: 796e 602e 2054 6865 2070 6172 616d 6574  yn`. The paramet
-0000ba20: 6572 2060 2d63 6f63 6020 6f72 2060 2d2d  er `-coc` or `--
-0000ba30: 636f 6c6c 696e 6561 7263 6f61 6c65 7363  collinearcoalesc
-0000ba40: 656e 6365 6020 6e65 6564 7320 746f 2062  ence` needs to b
-0000ba50: 6520 7365 7420 746f 2073 7461 7274 2074  e set to start t
-0000ba60: 6869 7320 616e 616c 7973 6973 2e20 5468  his analysis. Th
-0000ba70: 6520 7472 6565 2d69 6e66 6572 656e 6365  e tree-inference
-0000ba80: 2070 726f 6772 616d 2061 6e64 2074 6865   program and the
-0000ba90: 2061 7373 6f63 6961 7465 6420 7061 7261   associated para
-0000baa0: 6d65 7465 7273 2063 616e 2062 6520 7365  meters can be se
-0000bab0: 7420 6a75 7374 2061 7320 6162 6f76 6520  t just as above 
-0000bac0: 6279 2060 2d74 7265 6560 206f 7220 602d  by `-tree` or `-
-0000bad0: 2d74 7265 655f 6d65 7468 6f64 6020 616e  -tree_method` an
-0000bae0: 6420 602d 7473 6020 6f72 2060 2d2d 7472  d `-ts` or `--tr
-0000baf0: 6565 7365 7460 2e20 506c 6561 7365 2061  eeset`. Please a
-0000bb00: 6c73 6f20 6d61 6b65 2073 7572 6520 7468  lso make sure th
-0000bb10: 6520 6368 6f73 656e 2074 7265 652d 696e  e chosen tree-in
-0000bb20: 6665 7265 6e63 6520 7072 6f67 7261 6d20  ference program 
-0000bb30: 6973 2069 6e73 7461 6c6c 6564 2069 6e20  is installed in 
-0000bb40: 7468 6520 656e 7669 726f 6e6d 656e 7420  the environment 
-0000bb50: 7061 7468 2e20 5468 6520 7072 6f67 7261  path. The progra
-0000bb60: 6d20 6061 7374 7261 6c2d 7072 6f60 2069  m `astral-pro` i
-0000bb70: 7320 7265 7175 6972 6564 2074 6f20 6265  s required to be
-0000bb80: 2069 6e73 7461 6c6c 6564 2069 6e20 7468   installed in th
-0000bb90: 6520 656e 7669 726f 6e6d 656e 7420 7061  e environment pa
-0000bba0: 7468 2074 6f6f 2e20 4e6f 7465 2074 6861  th too. Note tha
-0000bbb0: 7420 7468 6572 6520 7368 6f75 6c64 2062  t there should b
-0000bbc0: 6520 6e6f 2064 7570 6c69 6361 7465 6420  e no duplicated 
-0000bbd0: 6765 6e65 2049 4473 2069 6e20 7468 6520  gene IDs in the 
-0000bbe0: 7365 7175 656e 6365 2066 696c 652e 2054  sequence file. T
-0000bbf0: 6865 2070 6172 616c 6c65 6c20 7468 7265  he parallel thre
-0000bc00: 6164 7320 6865 7265 2061 7265 2074 6f20  ads here are to 
-0000bc10: 6163 6365 6c65 7261 7465 2074 6865 2073  accelerate the s
-0000bc20: 6571 7565 6e63 6520 616c 6967 6e6d 656e  equence alignmen
-0000bc30: 7420 616e 6420 6765 6e65 2074 7265 6520  t and gene tree 
-0000bc40: 696e 6665 7265 6e63 6520 666f 7220 6561  inference for ea
-0000bc50: 6368 2067 656e 6520 6661 6d69 6c79 2061  ch gene family a
-0000bc60: 6e64 2074 6875 7320 7375 6767 6573 7465  nd thus suggeste
-0000bc70: 6420 746f 2062 6520 7365 7420 6173 206d  d to be set as m
-0000bc80: 7563 6820 6173 2074 6865 206e 756d 6265  uch as the numbe
-0000bc90: 7220 6f66 2067 656e 6520 6661 6d69 6c69  r of gene famili
-0000bca0: 6573 2e0a 0a2a 2a41 2073 7567 6765 7374  es...**A suggest
-0000bcb0: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
-0000bcc0: 6f66 2074 6869 7320 616e 616c 7973 6973  of this analysis
-0000bcd0: 2069 7320 7769 7468 2074 6865 2073 696d   is with the sim
-0000bce0: 706c 6520 636f 6d6d 616e 6420 6265 6c6f  ple command belo
-0000bcf0: 772e 2a2a 0a0a 6060 600a 7767 6420 646d  w.**..```.wgd dm
-0000bd00: 6420 7365 7175 656e 6365 3120 7365 7175  d sequence1 sequ
-0000bd10: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
-0000bd20: 2d61 7020 6170 6461 7461 202d 736d 2073  -ap apdata -sm s
-0000bd30: 6d64 6174 6120 2d6c 6520 6c65 6461 7461  mdata -le ledata
-0000bd40: 202d 6774 2067 7464 6174 6120 2d63 6f63   -gt gtdata -coc
-0000bd50: 0a60 6060 0a0a 2323 2320 7767 6420 666f  .```..### wgd fo
-0000bd60: 6375 730a 0a2a 2a54 6865 2063 6f6e 6361  cus..**The conca
-0000bd70: 7465 6e61 7469 6f6e 2d62 6173 6564 2f63  tenation-based/c
-0000bd80: 6f61 6c65 7363 656e 6365 2d62 6173 6564  oalescence-based
-0000bd90: 2070 6879 6c6f 6765 6e65 7469 6320 696e   phylogenetic in
-0000bda0: 6665 7265 6e63 652a 2a0a 6060 600a 7767  ference**.```.wg
-0000bdb0: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
-0000bdc0: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000bdd0: 6e63 6532 2073 6571 7565 6e63 6533 2028  nce2 sequence3 (
-0000bde0: 2d2d 636f 6e63 6174 656e 6174 696f 6e29  --concatenation)
-0000bdf0: 2028 2d2d 636f 616c 6573 6365 6e63 6529   (--coalescence)
-0000be00: 2028 2d74 7265 6520 2766 6173 7474 7265   (-tree 'fasttre
-0000be10: 6527 2920 282d 7473 2027 2d66 6173 7465  e') (-ts '-faste
-0000be20: 7374 2729 2028 2d6e 2034 2920 282d 2d74  st') (-n 4) (--t
-0000be30: 6f5f 7374 6f70 2920 282d 2d63 6473 2920  o_stop) (--cds) 
-0000be40: 282d 6f20 7767 645f 666f 6375 7329 2028  (-o wgd_focus) (
-0000be50: 2d74 2077 6f72 6b69 6e67 5f74 6d70 290a  -t working_tmp).
-0000be60: 6060 600a 0a54 6865 2070 726f 6772 616d  ```..The program
-0000be70: 2060 7767 6420 666f 6375 7360 2069 6d70   `wgd focus` imp
-0000be80: 6c65 6d65 6e74 6564 2074 776f 2062 6173  lemented two bas
-0000be90: 6963 2070 6879 6c6f 6765 6e65 7469 6320  ic phylogenetic 
-0000bea0: 696e 6665 7265 6e63 6520 6d65 7468 6f64  inference method
-0000beb0: 732c 2069 2e65 2e2c 2063 6f6e 6361 7465  s, i.e., concate
-0000bec0: 6e61 7469 6f6e 2d62 6173 6564 2061 6e64  nation-based and
-0000bed0: 2063 6f61 6c65 7363 656e 6365 2d62 6173   coalescence-bas
-0000bee0: 6564 206d 6574 686f 6473 2e20 546f 2069  ed methods. To i
-0000bef0: 6e69 7469 6174 6520 7468 6573 6520 616e  nitiate these an
-0000bf00: 616c 7973 6973 2c20 7573 6572 7320 6e65  alysis, users ne
-0000bf10: 6564 2074 6f20 7365 7420 7468 6520 666c  ed to set the fl
-0000bf20: 6167 206f 7074 696f 6e20 602d 2d63 6f6e  ag option `--con
-0000bf30: 6361 7465 6e61 7469 6f6e 6020 6f72 2060  catenation` or `
-0000bf40: 2d2d 636f 616c 6573 6365 6e63 6560 2e20  --coalescence`. 
-0000bf50: 5468 6520 636f 6e63 6174 656e 6174 696f  The concatenatio
-0000bf60: 6e2d 6261 7365 6420 6d65 7468 6f64 2069  n-based method i
-0000bf70: 6e63 6c75 6465 7320 6120 6665 7720 6d61  ncludes a few ma
-0000bf80: 6a6f 7220 7374 6570 732c 2069 2e65 2e2c  jor steps, i.e.,
-0000bf90: 2074 6865 206d 756c 7469 706c 6520 7365   the multiple se
-0000bfa0: 7175 656e 6365 2061 6c69 676e 6d65 6e74  quence alignment
-0000bfb0: 2028 4d53 4129 206f 6620 6561 6368 2067   (MSA) of each g
-0000bfc0: 656e 6520 6661 6d69 6c79 2c20 7468 6520  ene family, the 
-0000bfd0: 636f 6e63 6174 656e 6174 696f 6e20 6f66  concatenation of
-0000bfe0: 2061 6c6c 2067 656e 6520 6661 6d69 6c69   all gene famili
-0000bff0: 6573 2061 6e64 2074 6865 6e20 7468 6520  es and then the 
-0000c000: 6765 6e65 2074 7265 6520 2861 6c73 6f20  gene tree (also 
-0000c010: 7370 6563 6965 7320 7472 6565 2069 6e20  species tree in 
-0000c020: 7468 6973 2063 6173 6529 2069 6e66 6572  this case) infer
-0000c030: 656e 6365 2e20 5468 6520 636f 616c 6573  ence. The coales
-0000c040: 6365 6e63 652d 6261 7365 6420 6d65 7468  cence-based meth
-0000c050: 6f64 2077 696c 6c20 696e 7374 6561 6420  od will instead 
-0000c060: 7065 7266 6f72 6d20 7468 6520 4d53 4120  perform the MSA 
-0000c070: 6f66 2065 6163 6820 6765 6e65 2066 616d  of each gene fam
-0000c080: 696c 7920 616e 6420 7468 6520 6765 6e65  ily and the gene
-0000c090: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
-0000c0a0: 6261 7365 6420 6f6e 2065 6163 6820 4d53  based on each MS
-0000c0b0: 412c 2061 6e64 2074 6865 6e20 696e 6665  A, and then infe
-0000c0c0: 7220 7468 6520 7370 6563 6965 7320 7472  r the species tr
-0000c0d0: 6565 2062 6173 6564 206f 6e20 7468 6573  ee based on thes
-0000c0e0: 6520 696e 6469 7669 6475 616c 2067 656e  e individual gen
-0000c0f0: 6520 7472 6565 732e 2054 6865 2074 7265  e trees. The tre
-0000c100: 652d 696e 6665 7265 6e63 6520 7072 6f67  e-inference prog
-0000c110: 7261 6d20 616e 6420 7468 6520 6173 736f  ram and the asso
-0000c120: 6369 6174 6564 2070 6172 616d 6574 6572  ciated parameter
-0000c130: 7320 6361 6e20 6265 2073 6574 206a 7573  s can be set jus
-0000c140: 7420 6173 2061 626f 7665 2062 7920 602d  t as above by `-
-0000c150: 7472 6565 6020 6f72 2060 2d2d 7472 6565  tree` or `--tree
-0000c160: 5f6d 6574 686f 6460 2061 6e64 2060 2d74  _method` and `-t
-0000c170: 7360 206f 7220 602d 2d74 7265 6573 6574  s` or `--treeset
-0000c180: 602e 2050 6c65 6173 6520 616c 736f 206d  `. Please also m
-0000c190: 616b 6520 7375 7265 2074 6865 2063 686f  ake sure the cho
-0000c1a0: 7365 6e20 7472 6565 2d69 6e66 6572 656e  sen tree-inferen
-0000c1b0: 6365 2070 726f 6772 616d 2069 7320 696e  ce program is in
-0000c1c0: 7374 616c 6c65 6420 696e 2074 6865 2065  stalled in the e
-0000c1d0: 6e76 6972 6f6e 6d65 6e74 2070 6174 682e  nvironment path.
-0000c1e0: 2054 6865 2070 726f 6772 616d 2060 6173   The program `as
-0000c1f0: 7472 616c 2d70 726f 6020 6973 2072 6571  tral-pro` is req
-0000c200: 7569 7265 6420 746f 2062 6520 696e 7374  uired to be inst
-0000c210: 616c 6c65 6420 696e 2074 6865 2065 6e76  alled in the env
-0000c220: 6972 6f6e 6d65 6e74 2070 6174 6820 6966  ironment path if
-0000c230: 2074 6865 2063 6f61 6c65 7363 656e 6365   the coalescence
-0000c240: 2d62 6173 6564 206d 6574 686f 6420 6973  -based method is
-0000c250: 2063 686f 7365 6e2e 2054 6865 2070 6172   chosen. The par
-0000c260: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
-0000c270: 7265 2061 7265 2074 6f20 6163 6365 6c65  re are to accele
-0000c280: 7261 7465 2074 6865 2073 6571 7565 6e63  rate the sequenc
-0000c290: 6520 616c 6967 6e6d 656e 7420 616e 6420  e alignment and 
-0000c2a0: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
-0000c2b0: 6e63 6520 666f 7220 6561 6368 2067 656e  nce for each gen
-0000c2c0: 6520 6661 6d69 6c79 2074 6f6f 2061 6e64  e family too and
-0000c2d0: 2074 6875 7320 7375 6767 6573 7465 6420   thus suggested 
-0000c2e0: 746f 2062 6520 7365 7420 6173 206d 7563  to be set as muc
-0000c2f0: 6820 6173 2074 6865 206e 756d 6265 7220  h as the number 
-0000c300: 6f66 2067 656e 6520 6661 6d69 6c69 6573  of gene families
-0000c310: 2e0a 0a2a 2a41 2073 7567 6765 7374 6564  ...**A suggested
-0000c320: 2073 7461 7274 696e 6720 7275 6e20 6f66   starting run of
-0000c330: 2074 6869 7320 616e 616c 7973 6973 2069   this analysis i
-0000c340: 7320 7769 7468 2074 6865 2073 696d 706c  s with the simpl
-0000c350: 6520 636f 6d6d 616e 6420 6265 6c6f 772e  e command below.
-0000c360: 2a2a 0a0a 6060 600a 7767 6420 666f 6375  **..```.wgd focu
-0000c370: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
-0000c380: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
-0000c390: 6571 7565 6e63 6533 202d 2d63 6f6e 6361  equence3 --conca
-0000c3a0: 7465 6e61 7469 6f6e 0a77 6764 2066 6f63  tenation.wgd foc
-0000c3b0: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
-0000c3c0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000c3d0: 7365 7175 656e 6365 3320 2d2d 636f 616c  sequence3 --coal
-0000c3e0: 6573 6365 6e63 650a 6060 600a 0a2a 2a54  escence.```..**T
-0000c3f0: 6865 2066 756e 6374 696f 6e61 6c20 616e  he functional an
-0000c400: 6e6f 7461 7469 6f6e 206f 6620 6765 6e65  notation of gene
-0000c410: 2066 616d 696c 6965 732a 2a0a 6060 600a   families**.```.
-0000c420: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
-0000c430: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
-0000c440: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000c450: 202d 2d61 6e6e 6f74 6174 696f 6e20 6567   --annotation eg
-0000c460: 676e 6f67 202d 6564 2065 6464 6174 6120  gnog -ed eddata 
-0000c470: 2d2d 646d 6e62 2064 6264 6174 610a 6060  --dmnb dbdata.``
-0000c480: 600a 0a54 6865 2070 726f 6772 616d 2060  `..The program `
-0000c490: 7767 6420 666f 6375 7360 2061 6c73 6f20  wgd focus` also 
-0000c4a0: 6164 6465 6420 736f 6d65 2077 7261 7070  added some wrapp
-0000c4b0: 696e 6720 6675 6e63 7469 6f6e 7320 666f  ing functions fo
-0000c4c0: 7220 6675 6e63 7469 6f6e 616c 2061 6e6e  r functional ann
-0000c4d0: 6f74 6174 696f 6e20 6f66 2067 656e 6520  otation of gene 
-0000c4e0: 6661 6d69 6c69 6573 206f 6e20 7468 6520  families on the 
-0000c4f0: 686f 6f64 206f 6620 6461 7461 6261 7365  hood of database
-0000c500: 7320 616e 6420 736f 6674 7761 7265 7320  s and softwares 
-0000c510: 666f 7220 696e 7374 616e 6365 2060 4567  for instance `Eg
-0000c520: 674e 4f47 6020 616e 6420 6045 6767 4e4f  gNOG` and `EggNO
-0000c530: 472d 6d61 7070 6572 602e 2046 6f72 2074  G-mapper`. For t
-0000c540: 6865 2061 6e6e 6f74 6174 696f 6e20 7573  he annotation us
-0000c550: 696e 6720 6045 6767 4e4f 472d 6d61 7070  ing `EggNOG-mapp
-0000c560: 6572 602c 2075 7365 7273 206e 6565 6420  er`, users need 
-0000c570: 746f 2070 726f 7669 6465 2074 6865 2070  to provide the p
-0000c580: 6174 6820 746f 2074 6865 2065 6767 4e4f  ath to the eggNO
-0000c590: 4720 616e 6e6f 7461 7469 6f6e 2064 6174  G annotation dat
-0000c5a0: 6162 6173 6520 7669 6120 7468 6520 6f70  abase via the op
-0000c5b0: 7469 6f6e 2060 2d65 6460 2c20 7468 6520  tion `-ed`, the 
-0000c5c0: 7061 7468 2074 6f20 7468 6520 6469 616d  path to the diam
-0000c5d0: 6f6e 642d 636f 6d70 6174 6962 6c65 2064  ond-compatible d
-0000c5e0: 6174 6162 6173 6520 7669 6120 7468 6520  atabase via the 
-0000c5f0: 6f70 7469 6f6e 2060 2d2d 646d 6e64 5f64  option `--dmnd_d
-0000c600: 6260 2061 6e64 2074 6865 206f 7074 696f  b` and the optio
-0000c610: 6e20 602d 2d61 6e6e 6f74 6174 696f 6e60  n `--annotation`
-0000c620: 2073 6574 2061 7320 2265 6767 6e6f 6722   set as "eggnog"
-0000c630: 2e20 5468 6520 6d61 6e6e 6572 2068 6f77  . The manner how
-0000c640: 2050 4641 4d20 616e 6e6f 7461 7469 6f6e   PFAM annotation
-0000c650: 2077 696c 6c20 6265 2070 6572 666f 726d   will be perform
-0000c660: 6564 2063 616e 2062 6520 636f 6e74 726f  ed can be contro
-0000c670: 6c6c 6564 2076 6961 2074 6865 206f 7074  lled via the opt
-0000c680: 696f 6e20 6070 6661 6d60 2c20 6569 7468  ion `pfam`, eith
-0000c690: 6572 2022 6e6f 6e65 222c 2022 7265 616c  er "none", "real
-0000c6a0: 6967 6e22 206f 7220 2264 656e 6f76 6f22  ign" or "denovo"
-0000c6b0: 2c20 7468 6520 6465 7461 696c 6564 2065  , the detailed e
-0000c6c0: 7870 6c61 6e61 7469 6f6e 2063 616e 2062  xplanation can b
-0000c6d0: 6520 666f 756e 6420 6174 2074 6865 2077  e found at the w
-0000c6e0: 696b 6920 6f66 205b 4567 674e 4f47 2d6d  iki of [EggNOG-m
-0000c6f0: 6170 7065 725d 2868 7474 7073 3a2f 2f67  apper](https://g
-0000c700: 6974 6875 622e 636f 6d2f 6567 676e 6f67  ithub.com/eggnog
-0000c710: 6462 2f65 6767 6e6f 672d 6d61 7070 6572  db/eggnog-mapper
-0000c720: 2f77 696b 6929 2e20 506c 6561 7365 2070  /wiki). Please p
-0000c730: 7265 2d69 6e73 7461 6c6c 2074 6865 2060  re-install the `
-0000c740: 4567 674e 4f47 2d6d 6170 7065 7260 2070  EggNOG-mapper` p
-0000c750: 7974 686f 6e20 7061 636b 6167 6520 6966  ython package if
-0000c760: 2075 7369 6e67 2074 6869 7320 6675 6e63   using this func
-0000c770: 7469 6f6e 2e20 466f 7220 7468 6520 616e  tion. For the an
-0000c780: 6e6f 7461 7469 6f6e 2075 7369 6e67 2060  notation using `
-0000c790: 686d 6d73 6361 6e60 2c20 7768 6174 2069  hmmscan`, what i
-0000c7a0: 7320 696d 706c 656d 656e 7465 6420 696e  s implemented in
-0000c7b0: 2060 7767 6420 7632 6020 6973 2061 2073   `wgd v2` is a s
-0000c7c0: 696d 706c 6520 6275 6e64 6c65 2066 756e  imple bundle fun
-0000c7d0: 6374 696f 6e20 746f 2070 6572 666f 726d  ction to perform
-0000c7e0: 2060 686d 6d73 6361 6e60 2061 6e61 6c79   `hmmscan` analy
-0000c7f0: 7369 7320 666f 7220 6120 6769 7665 6e20  sis for a given 
-0000c800: 686d 6d20 7072 6f66 696c 6520 616e 6420  hmm profile and 
-0000c810: 7365 7420 6f66 2067 656e 6520 6661 6d69  set of gene fami
-0000c820: 6c69 6573 2073 7563 6820 7468 6174 2075  lies such that u
-0000c830: 7365 7273 206e 6565 6420 746f 2073 6574  sers need to set
-0000c840: 2074 6865 206f 7074 696f 6e20 602d 2d61   the option `--a
-0000c850: 6e6e 6f74 6174 696f 6e60 2061 7320 2268  nnotation` as "h
-0000c860: 6d6d 7066 616d 2220 616e 6420 7072 6f76  mmpfam" and prov
-0000c870: 6964 6520 7468 6520 686d 6d70 726f 6669  ide the hmmprofi
-0000c880: 6c65 2076 6961 2074 6865 206f 7074 696f  le via the optio
-0000c890: 6e20 602d 2d68 6d6d 602e 2046 6f72 2074  n `--hmm`. For t
-0000c8a0: 6865 2061 6e6e 6f74 6174 696f 6e20 7573  he annotation us
-0000c8b0: 696e 6720 6069 6e74 6572 7072 6f73 6361  ing `interprosca
-0000c8c0: 6e60 2c20 7573 6572 7320 6e65 6564 2074  n`, users need t
-0000c8d0: 6f20 7072 6f76 6964 6520 7468 6520 7061  o provide the pa
-0000c8e0: 7468 2074 6f20 7468 6520 696e 7465 7270  th to the interp
-0000c8f0: 726f 7363 616e 2069 6e73 7461 6c6c 6174  roscan installat
-0000c900: 696f 6e20 666f 6c64 6572 2077 6865 7265  ion folder where
-0000c910: 2074 6865 7265 2069 7320 6120 6069 6e74   there is a `int
-0000c920: 6572 7072 6f73 6361 6e2e 7368 6020 6669  erproscan.sh` fi
-0000c930: 6c65 2076 6961 2074 6865 206f 7074 696f  le via the optio
-0000c940: 6e20 602d 2d65 7865 7061 7468 6020 616e  n `--exepath` an
-0000c950: 6420 7365 7420 7468 6520 7468 6520 6f70  d set the the op
-0000c960: 7469 6f6e 2060 2d2d 616e 6e6f 7461 7469  tion `--annotati
-0000c970: 6f6e 6020 6173 2022 696e 7465 7270 726f  on` as "interpro
-0000c980: 7363 616e 222e 2054 6865 2070 6172 616c  scan". The paral
-0000c990: 6c65 6c20 7468 7265 6164 7320 6865 7265  lel threads here
-0000c9a0: 2061 7265 2074 6f20 7061 7261 6c6c 656c   are to parallel
-0000c9b0: 697a 6520 7468 6520 616e 616c 7973 6973  ize the analysis
-0000c9c0: 2066 6f72 2065 6163 6820 6765 6e65 2066   for each gene f
-0000c9d0: 616d 696c 7920 616e 6420 7468 7573 2073  amily and thus s
-0000c9e0: 7567 6765 7374 6564 2074 6f20 6265 2073  uggested to be s
-0000c9f0: 6574 2061 7320 6d75 6368 2061 7320 7468  et as much as th
-0000ca00: 6520 6e75 6d62 6572 206f 6620 6765 6e65  e number of gene
-0000ca10: 2066 616d 696c 6965 732e 0a0a 2a2a 4120   families...**A 
-0000ca20: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
-0000ca30: 6e67 2072 756e 206f 6620 7468 6973 2061  ng run of this a
-0000ca40: 6e61 6c79 7369 7320 6361 6e20 6265 2077  nalysis can be w
-0000ca50: 6974 6820 7468 6520 636f 6d6d 616e 6420  ith the command 
-0000ca60: 6265 6c6f 772e 2a2a 0a0a 6060 600a 7767  below.**..```.wg
-0000ca70: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
-0000ca80: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000ca90: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
-0000caa0: 2d61 6e6e 6f74 6174 696f 6e20 6567 676e  -annotation eggn
-0000cab0: 6f67 202d 6564 2065 6464 6174 6120 2d2d  og -ed eddata --
-0000cac0: 646d 6e62 2064 6264 6174 610a 7767 6420  dmnb dbdata.wgd 
-0000cad0: 666f 6375 7320 6661 6d69 6c69 6573 2073  focus families s
-0000cae0: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
-0000caf0: 6532 2073 6571 7565 6e63 6533 202d 2d61  e2 sequence3 --a
-0000cb00: 6e6e 6f74 6174 696f 6e20 686d 6d70 6661  nnotation hmmpfa
-0000cb10: 6d20 2d2d 686d 6d20 686d 6d64 6174 610a  m --hmm hmmdata.
-0000cb20: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
-0000cb30: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
-0000cb40: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000cb50: 202d 2d61 6e6e 6f74 6174 696f 6e20 696e   --annotation in
-0000cb60: 7465 7270 726f 7363 616e 202d 2d65 7865  terproscan --exe
-0000cb70: 7061 7468 2024 5041 5448 0a60 6060 0a0a  path $PATH.```..
-0000cb80: 2a2a 5468 6520 7068 796c 6f67 656e 6574  **The phylogenet
-0000cb90: 6963 2064 6174 696e 6720 6f66 2057 4744  ic dating of WGD
-0000cba0: 732a 2a0a 6060 600a 7767 6420 666f 6375  s**.```.wgd focu
-0000cbb0: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
-0000cbc0: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
-0000cbd0: 6571 7565 6e63 6533 202d 6420 6d63 6d63  equence3 -d mcmc
-0000cbe0: 7472 6565 202d 7370 2073 7064 6174 6120  tree -sp spdata 
-0000cbf0: 282d 2d70 726f 7463 6f63 6461 7469 6e67  (--protcocdating
-0000cc00: 2920 282d 2d70 6172 7469 7469 6f6e 2920  ) (--partition) 
-0000cc10: 282d 2d61 616d 6f64 656c 206c 6729 2028  (--aamodel lg) (
-0000cc20: 2d64 7320 2762 7572 6e69 6e20 3d20 3230  -ds 'burnin = 20
-0000cc30: 3030 2729 2028 2d64 7320 2773 616d 7066  00') (-ds 'sampf
-0000cc40: 7265 7120 3d20 3130 3030 2729 2028 2d64  req = 1000') (-d
-0000cc50: 7320 276e 7361 6d70 6c65 203d 2032 3030  s 'nsample = 200
-0000cc60: 3030 2729 2020 282d 6e20 3429 2028 2d2d  00')  (-n 4) (--
-0000cc70: 746f 5f73 746f 7029 2028 2d2d 6364 7329  to_stop) (--cds)
-0000cc80: 2028 2d6f 2077 6764 5f66 6f63 7573 2920   (-o wgd_focus) 
-0000cc90: 282d 7420 776f 726b 696e 675f 746d 7029  (-t working_tmp)
-0000cca0: 0a60 6060 0a0a 5468 6520 6162 736f 6c75  .```..The absolu
-0000ccb0: 7465 2064 6174 696e 6720 6f66 2057 4744  te dating of WGD
-0000ccc0: 7320 6973 2061 2073 7065 6369 6669 6320  s is a specific 
-0000ccd0: 7069 7065 6c69 6e65 2069 6d70 6c65 6d65  pipeline impleme
-0000cce0: 6e74 6564 2069 6e20 6077 6764 2076 3260  nted in `wgd v2`
-0000ccf0: 2075 7369 6e67 2074 6865 206d 6574 686f   using the metho
-0000cd00: 6420 6f66 2070 6879 6c6f 6765 6e65 7469  d of phylogeneti
-0000cd10: 6320 6461 7469 6e67 2e20 5468 6520 6661  c dating. The fa
-0000cd20: 6d69 6c69 6573 2075 7365 6420 696e 2074  milies used in t
-0000cd30: 6869 7320 7374 6570 2063 616e 2062 6520  his step can be 
-0000cd40: 7072 6f64 7563 6564 2066 726f 6d20 6077  produced from `w
-0000cd50: 6764 2064 6d64 6020 616e 6420 6077 6764  gd dmd` and `wgd
-0000cd60: 2070 6561 6b60 2e20 4e6f 7465 2074 6861   peak`. Note tha
-0000cd70: 7420 6865 7265 2077 6520 6f6e 6c79 2064  t here we only d
-0000cd80: 6973 6375 7373 2068 6f77 2074 6f20 6461  iscuss how to da
-0000cd90: 7465 2057 4744 7320 7769 7468 2067 656e  te WGDs with gen
-0000cda0: 6f6d 6520 6173 7365 6d62 6c79 2c20 696e  ome assembly, in
-0000cdb0: 7374 6561 6420 6f66 2074 7261 6e73 6372  stead of transcr
-0000cdc0: 6970 746f 6d65 2061 7373 656d 626c 7920  iptome assembly 
-0000cdd0: 2877 6869 6368 2077 696c 6c20 6265 2064  (which will be d
-0000cde0: 6973 6375 7373 6564 2069 6e20 6120 7365  iscussed in a se
-0000cdf0: 7061 7261 7465 2073 6563 7469 6f6e 2068  parate section h
-0000ce00: 6572 6575 6e64 6572 292e 2054 6865 2061  ereunder). The a
-0000ce10: 7373 756d 7074 696f 6e20 7765 206d 6164  ssumption we mad
-0000ce20: 6520 6865 7265 2069 7320 7468 6174 206e  e here is that n
-0000ce30: 6f74 2061 6c6c 2061 6e63 686f 7220 7061  ot all anchor pa
-0000ce40: 6972 7320 2863 6f6c 6c69 6e65 6172 2064  irs (collinear d
-0000ce50: 7570 6c69 6361 7465 7329 2061 7265 2073  uplicates) are s
-0000ce60: 7569 7461 626c 6520 666f 7220 7068 796c  uitable for phyl
-0000ce70: 6f67 656e 6574 6963 2064 6174 696e 672c  ogenetic dating,
-0000ce80: 2066 6f72 2069 6e73 7461 6e63 6520 7468   for instance th
-0000ce90: 6f73 6520 6661 7374 6c79 206f 7220 736c  ose fastly or sl
-0000cea0: 6f77 6c79 2065 766f 6c76 696e 6720 6765  owly evolving ge
-0000ceb0: 6e65 2064 7570 6c69 6361 7465 732c 2062  ne duplicates, b
-0000cec0: 6563 6175 7365 2074 6865 7927 7265 2070  ecause they're p
-0000ced0: 726f 6e65 2074 6f20 6769 7665 2062 6961  rone to give bia
-0000cee0: 7365 6420 6461 7469 6e67 2065 7374 696d  sed dating estim
-0000cef0: 6174 696f 6e2e 2053 6f20 6173 2074 6f20  ation. So as to 
-0000cf00: 7265 7472 6965 7665 2074 6865 2022 7265  retrieve the "re
-0000cf10: 6c69 6162 6c65 2220 616e 6368 6f72 2070  liable" anchor p
-0000cf20: 6169 7273 2c20 7765 2069 6d70 6c65 6d65  airs, we impleme
-0000cf30: 6e74 6564 2073 6f6d 6520 6d65 7468 6f64  nted some method
-0000cf40: 7320 6f66 2069 6465 6e74 6966 7969 6e67  s of identifying
-0000cf50: 2063 7265 6469 6162 6c65 2061 6e63 686f   crediable ancho
-0000cf60: 7220 7061 6972 7320 6261 7365 6420 6f6e  r pairs based on
-0000cf70: 2074 6865 6972 204b 7320 7661 6c75 6573   their Ks values
-0000cf80: 2061 6e64 2f6f 7220 7265 7369 6469 6e67   and/or residing
-0000cf90: 2063 6f6c 6c69 6e65 6172 2073 6567 6d65   collinear segme
-0000cfa0: 6e74 732e 2046 6f72 2061 2067 656e 6f6d  nts. For a genom
-0000cfb0: 6520 7769 7468 2063 6c65 6172 2073 6967  e with clear sig
-0000cfc0: 6e61 6c73 206f 6620 7075 7461 7469 7665  nals of putative
-0000cfd0: 2057 4744 732c 2073 7563 6820 6173 2074   WGDs, such as t
-0000cfe0: 6865 202a 4171 7569 6c65 6769 6120 636f  he *Aquilegia co
-0000cff0: 6572 756c 6561 2a20 696e 2074 6865 2065  erulea* in the e
-0000d000: 7861 6d70 6c65 2c20 6120 6865 7572 6973  xample, a heuris
-0000d010: 7469 6320 6d65 7468 6f64 2074 6861 7420  tic method that 
-0000d020: 6170 706c 6965 7320 7468 6520 7072 696e  applies the prin
-0000d030: 6369 706c 6520 6f66 2068 6f77 205b 6b73  ciple of how [ks
-0000d040: 7261 7465 735d 2868 7474 7073 3a2f 2f67  rates](https://g
-0000d050: 6974 6875 622e 636f 6d2f 5649 422d 5053  ithub.com/VIB-PS
-0000d060: 422f 6b73 7261 7465 7329 2066 696e 6420  B/ksrates) find 
-0000d070: 7468 6520 696e 6974 6961 6c20 7065 616b  the initial peak
-0000d080: 7320 616e 6420 7468 6569 7220 7061 7261  s and their para
-0000d090: 6d65 7465 7273 2077 6173 2069 6d70 6c65  meters was imple
-0000d0a0: 6d65 6e74 6564 2074 6f20 6669 6e64 2074  mented to find t
-0000d0b0: 6865 2039 3525 2063 6f6e 6669 6465 6e63  he 95% confidenc
-0000d0c0: 6520 6c65 7665 6c20 6f66 2074 6865 2061  e level of the a
-0000d0d0: 7373 756d 6564 206c 6f67 6e6f 726d 616c  ssumed lognormal
-0000d0e0: 2064 6973 7472 6962 7574 696f 6e20 6f66   distribution of
-0000d0f0: 2074 6865 2061 6e63 686f 7220 7061 6972   the anchor pair
-0000d100: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000d110: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-0000d120: 6e20 746f 2066 696c 7465 7220 616e 6368  n to filter anch
-0000d130: 6f72 2070 6169 7273 2077 6974 6820 746f  or pairs with to
-0000d140: 6f20 6869 6768 206f 7220 6c6f 7720 2a4b  o high or low *K
-0000d150: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
-0000d160: 6573 2c20 7468 6520 6578 616d 706c 6172  es, the examplar
-0000d170: 2063 6f6d 6d61 6e64 206f 6620 7768 6963   command of whic
-0000d180: 6820 6973 2073 686f 7765 6420 696e 2074  h is showed in t
-0000d190: 6865 2060 7767 6420 7065 616b 6020 7365  he `wgd peak` se
-0000d1a0: 6374 696f 6e20 6f66 205b 496c 6c75 7374  ction of [Illust
-0000d1b0: 7261 7469 6f6e 5d28 6874 7470 733a 2f2f  ration](https://
-0000d1c0: 6769 7468 7562 2e63 6f6d 2f68 6563 6865  github.com/heche
-0000d1d0: 2d70 7362 2f77 6764 3f74 6162 3d72 6561  -psb/wgd?tab=rea
-0000d1e0: 646d 652d 6f76 2d66 696c 6523 696c 6c75  dme-ov-file#illu
-0000d1f0: 7374 7261 7469 6f6e 292e 2049 6620 7468  stration). If th
-0000d200: 6973 2068 6575 7269 7374 6963 206d 6574  is heuristic met
-0000d210: 686f 6420 6661 696c 6564 2074 6f20 6769  hod failed to gi
-0000d220: 7665 2072 6561 736f 6e61 626c 6520 7265  ve reasonable re
-0000d230: 7375 6c74 732c 2077 6869 6368 2069 7320  sults, which is 
-0000d240: 7573 7561 6c6c 7920 6475 6520 746f 2074  usually due to t
-0000d250: 6865 2065 6666 6563 7420 6f66 206d 756c  he effect of mul
-0000d260: 7469 706c 6520 6164 6a61 6365 6e74 2057  tiple adjacent W
-0000d270: 4744 7320 7468 6174 2062 6c75 7273 2074  GDs that blurs t
-0000d280: 6865 2070 6561 6b2d 6669 6e64 696e 6720  he peak-finding 
-0000d290: 7072 6f63 6573 732c 2075 7365 7273 2063  process, users c
-0000d2a0: 616e 2074 7572 6e20 746f 2074 6865 2063  an turn to the c
-0000d2b0: 6f6c 6c69 6e65 6172 2073 6567 6d65 6e74  ollinear segment
-0000d2c0: 732d 6775 6964 6564 2061 6e63 686f 7220  s-guided anchor 
-0000d2d0: 7061 6972 2063 6c75 7374 6572 696e 6720  pair clustering 
-0000d2e0: 696d 706c 656d 656e 7465 6420 696e 2060  implemented in `
-0000d2f0: 7767 6420 7632 602c 2069 6e20 7768 6963  wgd v2`, in whic
-0000d300: 6820 6120 636f 6c6c 696e 6561 7220 7365  h a collinear se
-0000d310: 676d 656e 742d 7769 7365 2047 4d4d 2063  gment-wise GMM c
-0000d320: 6c75 7374 6572 696e 6720 7769 6c6c 2062  lustering will b
-0000d330: 6520 6669 7273 7420 636f 6e64 7563 7465  e first conducte
-0000d340: 6420 6261 7365 6420 6f6e 2074 6865 2022  d based on the "
-0000d350: 736f 2d63 616c 6c65 6422 2073 6567 6d65  so-called" segme
-0000d360: 6e74 202a 4b2a 3c73 7562 3e53 3c2f 7375  nt *K*<sub>S</su
-0000d370: 623e 2061 6765 2072 6570 7265 7365 6e74  b> age represent
-0000d380: 6564 2062 7920 7468 6520 6d65 6469 616e  ed by the median
-0000d390: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000d3a0: 2061 6765 206f 6620 616c 6c20 7468 6520   age of all the 
-0000d3b0: 7265 7369 6469 6e67 2073 796e 7465 6c6f  residing syntelo
-0000d3c0: 6773 2028 6e6f 7465 2074 6861 7420 7468  gs (note that th
-0000d3d0: 6520 6765 6e65 2064 7570 6c69 6361 7465  e gene duplicate
-0000d3e0: 2070 6169 7273 2061 646f 7074 6564 2069   pairs adopted i
-0000d3f0: 6e20 7468 6973 2073 7465 7020 6973 2066  n this step is f
-0000d400: 726f 6d20 7468 6520 6669 6c65 2060 6d75  rom the file `mu
-0000d410: 6c74 6970 6c69 636f 6e5f 7061 6972 732e  ltiplicon_pairs.
-0000d420: 7478 7460 2077 6869 6368 2063 6f6e 7461  txt` which conta
-0000d430: 696e 7320 7468 6520 6675 6c6c 2073 6574  ins the full set
-0000d440: 206f 6620 7379 6e74 656c 6f67 7329 2c20   of syntelogs), 
-0000d450: 696e 7374 6561 6420 6f66 2074 6865 2073  instead of the s
-0000d460: 6d61 6c6c 6572 2067 656e 6520 7365 7420  maller gene set 
-0000d470: 6f66 2061 6e63 686f 7220 7061 6972 732e  of anchor pairs.
-0000d480: 2054 6865 2064 6973 7469 6e63 7469 6f6e   The distinction
-0000d490: 2062 6574 7765 656e 2061 6e63 686f 7220   between anchor 
-0000d4a0: 7061 6972 7320 616e 6420 7379 6e74 656c  pairs and syntel
-0000d4b0: 6f67 7320 6973 2074 6861 7420 7468 6520  ogs is that the 
-0000d4c0: 6c61 7474 6572 2072 6566 6572 7320 746f  latter refers to
-0000d4d0: 206d 756c 7469 706c 6520 7365 7473 206f   multiple sets o
-0000d4e0: 6620 6765 6e65 7320 6465 7269 7665 6420  f genes derived 
-0000d4f0: 6672 6f6d 2074 6865 2073 616d 6520 616e  from the same an
-0000d500: 6365 7374 7261 6c20 6765 6e6f 6d69 6320  cestral genomic 
-0000d510: 7265 6769 6f6e 2077 6869 6c65 2074 6865  region while the
-0000d520: 2066 6f72 6d65 7220 696d 706c 6965 7320   former implies 
-0000d530: 7468 6520 6c61 7474 6572 2062 7574 2072  the latter but r
-0000d540: 6571 7569 7265 7320 6164 6469 7469 6f6e  equires addition
-0000d550: 616c 6c79 2074 6865 2063 6f6e 7365 7276  ally the conserv
-0000d560: 6564 2067 656e 6520 6f72 6465 722c 2062  ed gene order, b
-0000d570: 6f74 6820 6f66 2077 6869 6368 2061 7265  oth of which are
-0000d580: 2c20 7769 7468 696e 2061 2067 656e 6f6d  , within a genom
-0000d590: 652c 2061 7373 756d 6564 2074 6f20 6265  e, assumed to be
-0000d5a0: 206f 7269 6769 6e61 7465 6420 6672 6f6d   originated from
-0000d5b0: 2074 6865 2064 7570 6c69 6361 7469 6f6e   the duplication
-0000d5c0: 206f 6620 6120 636f 6d6d 6f6e 2061 6e63   of a common anc
-0000d5d0: 6573 7472 616c 2067 656e 6f6d 6963 2072  estral genomic r
-0000d5e0: 6567 696f 6e20 616e 6420 6173 2073 7563  egion and as suc
-0000d5f0: 6820 6465 656d 6564 2065 7669 6465 6e63  h deemed evidenc
-0000d600: 6520 666f 7220 5747 442e 2054 6865 6e20  e for WGD. Then 
-0000d610: 7468 6520 7379 6e74 656c 6f67 7320 7769  the syntelogs wi
-0000d620: 6c6c 2062 6520 6d61 7070 6564 2062 6163  ll be mapped bac
-0000d630: 6b20 6163 636f 7264 696e 6720 746f 2074  k according to t
-0000d640: 6865 2063 6c75 7374 6572 696e 6720 7265  he clustering re
-0000d650: 7375 6c74 7320 6f66 2074 6865 6972 2061  sults of their a
-0000d660: 6666 696c 6961 7465 6420 7365 676d 656e  ffiliated segmen
-0000d670: 7473 2e20 5369 6e63 6520 7468 6520 4761  ts. Since the Ga
-0000d680: 7573 7369 616e 2073 6861 7065 206f 6620  ussian shape of 
-0000d690: 7468 6520 7365 676d 656e 7420 636c 7573  the segment clus
-0000d6a0: 7465 7220 646f 6573 6e27 7420 6e65 6365  ter doesn't nece
-0000d6b0: 7373 6172 696c 7920 696d 6974 6174 6520  ssarily imitate 
-0000d6c0: 7468 6520 7368 6170 6520 6f66 2074 6865  the shape of the
-0000d6d0: 2072 6573 6964 696e 6720 7379 6e74 656c   residing syntel
-0000d6e0: 6f67 732c 2073 6f20 6173 2074 6f20 7265  ogs, so as to re
-0000d6f0: 7472 6965 7665 2074 6865 2022 7265 6c69  trieve the "reli
-0000d700: 6162 6c65 2220 6765 6e65 2073 6574 7320  able" gene sets 
-0000d710: 666f 7220 7068 796c 6f67 656e 6574 6963  for phylogenetic
-0000d720: 2064 6174 696e 672c 2077 6520 6164 706f   dating, we adpo
-0000d730: 7465 6420 7468 6520 2839 3525 2920 6869  ted the (95%) hi
-0000d740: 6768 6573 7420 6465 6e73 6974 7920 7265  ghest density re
-0000d750: 6769 6f6e 2028 4844 5229 206f 6620 7468  gion (HDR) of th
-0000d760: 6520 7379 6e74 656c 6f67 202a 4b2a 3c73  e syntelog *K*<s
-0000d770: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-0000d780: 6973 7472 6962 7574 696f 6e20 666f 7220  istribution for 
-0000d790: 7468 6520 7068 796c 6f67 656e 6574 6963  the phylogenetic
-0000d7a0: 2064 6174 696e 672c 2074 6865 2063 616c   dating, the cal
-0000d7b0: 6375 6c61 7469 6f6e 206f 6620 7468 6520  culation of the 
-0000d7c0: 2839 3525 2920 4844 5220 696e 2074 6865  (95%) HDR in the
-0000d7d0: 2066 756e 6374 696f 6e20 5b63 616c 6375   function [calcu
-0000d7e0: 6c61 7465 4850 445d 2868 7474 7073 3a2f  lateHPD](https:/
-0000d7f0: 2f67 6974 6875 622e 636f 6d2f 6865 6368  /github.com/hech
-0000d800: 652d 7073 622f 7767 642f 626c 6f62 2f70  e-psb/wgd/blob/p
-0000d810: 6879 6c6f 6461 7469 6e67 2f77 6764 2f70  hylodating/wgd/p
-0000d820: 6561 6b2e 7079 234c 3137 3537 2920 7365  eak.py#L1757) se
-0000d830: 656b 7320 7468 6520 7368 6f72 7465 7374  eks the shortest
-0000d840: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000d850: 2072 616e 6765 2028 612c 6229 2077 6869   range (a,b) whi
-0000d860: 6368 2073 6174 6973 6669 6573 2074 6865  ch satisfies the
-0000d870: 2072 6571 7569 7265 6d65 6e74 206f 6620   requirement of 
-0000d880: 7370 616e 6e69 6e67 206d 6f72 6520 7468  spanning more th
-0000d890: 616e 2028 3935 2529 206f 6620 616c 6c20  an (95%) of all 
-0000d8a0: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
-0000d8b0: 7562 3e20 7661 6c75 6573 2e20 4f6e 2074  ub> values. On t
-0000d8c0: 6865 2070 7265 6d69 7365 206f 6620 6964  he premise of id
-0000d8d0: 656e 7469 6669 6564 2061 6e63 686f 7220  entified anchor 
-0000d8e0: 7061 6972 7320 286e 6f74 6520 7468 6174  pairs (note that
-0000d8f0: 2074 6865 2073 796e 7465 6c6f 6773 2061   the syntelogs a
-0000d900: 7265 2061 6c73 6f20 7265 6665 7272 6564  re also referred
-0000d910: 2061 7320 616e 6368 6f72 2070 6169 7273   as anchor pairs
-0000d920: 2068 6572 6561 6674 6572 292c 2077 6520   hereafter), we 
-0000d930: 696d 706c 656d 656e 7465 6420 696e 2074  implemented in t
-0000d940: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-0000d950: 646d 6460 2074 6865 2073 6f2d 6361 6c6c  dmd` the so-call
-0000d960: 6564 2061 6e63 686f 722d 6177 6172 6520  ed anchor-aware 
-0000d970: 6c6f 6361 6c20 4d52 4248 7320 6f72 206f  local MRBHs or o
-0000d980: 7274 686f 6772 6f75 7073 2c20 696e 2077  rthogroups, in w
-0000d990: 6869 6368 2074 6865 206f 7269 6769 6e61  hich the origina
-0000d9a0: 6c20 6c6f 6361 6c20 4d52 4248 7320 6172  l local MRBHs ar
-0000d9b0: 6520 6675 7274 6865 7220 6d65 7267 6564  e further merged
-0000d9c0: 2077 6974 6820 616e 6368 6f72 2070 6169   with anchor pai
-0000d9d0: 7273 2073 7563 6820 7468 6174 2065 6163  rs such that eac
-0000d9e0: 6820 6f72 7468 6f67 726f 7570 2063 6f6e  h orthogroup con
-0000d9f0: 7461 696e 7320 7468 6520 616e 6368 6f72  tains the anchor
-0000da00: 2070 6169 7220 616e 6420 7468 6520 6f72   pair and the or
-0000da10: 7468 6f6c 6f67 7565 732e 2057 6974 6820  thologues. With 
-0000da20: 7468 6973 206f 7274 686f 6772 6f75 702c  this orthogroup,
-0000da30: 2075 7365 7273 2074 6865 6e20 6e65 6564   users then need
-0000da40: 206f 6e65 2073 7461 7274 696e 6720 7472   one starting tr
-0000da50: 6565 2066 696c 6520 2861 7320 7368 6f77  ee file (as show
-0000da60: 6e20 696e 2074 6865 205b 496c 6c75 7374  n in the [Illust
-0000da70: 7261 7469 6f6e 5d28 6874 7470 733a 2f2f  ration](https://
-0000da80: 6769 7468 7562 2e63 6f6d 2f68 6563 6865  github.com/heche
-0000da90: 2d70 7362 2f77 6764 2f74 7265 652f 7068  -psb/wgd/tree/ph
-0000daa0: 796c 6f64 6174 696e 673f 7461 623d 7265  ylodating?tab=re
-0000dab0: 6164 6d65 2d6f 762d 6669 6c65 2369 6c6c  adme-ov-file#ill
-0000dac0: 7573 7472 6174 696f 6e29 2070 6172 7429  ustration) part)
-0000dad0: 2069 6e64 6963 6174 696e 6720 7468 6520   indicating the 
-0000dae0: 7472 6565 2074 6f70 6f6c 6f67 7920 616e  tree topology an
-0000daf0: 6420 666f 7373 696c 2063 616c 6962 7261  d fossil calibra
-0000db00: 7469 6f6e 2069 6e66 6f72 6d61 7469 6f6e  tion information
-0000db10: 2066 6f72 2074 6865 2066 696e 616c 2070   for the final p
-0000db20: 6879 6c6f 6765 6e65 7469 6320 6461 7469  hylogenetic dati
-0000db30: 6e67 2075 7369 6e67 2074 6865 2070 726f  ng using the pro
-0000db40: 6772 616d 2060 7767 6420 666f 6375 7360  gram `wgd focus`
-0000db50: 2e20 5573 6572 7320 6361 6e20 7365 7420  . Users can set 
-0000db60: 7468 6520 666c 6167 206f 7074 696f 6e20  the flag option 
-0000db70: 602d 2d70 726f 7463 6f63 6461 7469 6e67  `--protcocdating
-0000db80: 6020 746f 206f 6e6c 7920 636f 6e64 7563  ` to only conduc
-0000db90: 7420 7468 6520 6461 7469 6e67 206f 6620  t the dating of 
-0000dba0: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
-0000dbb0: 6e20 7072 6f74 6569 6e20 4d53 4120 6f72  n protein MSA or
-0000dbc0: 2074 6865 2066 6c61 6720 6f70 7469 6f6e   the flag option
-0000dbd0: 2060 2d2d 7072 6f74 6461 7469 6e67 6020   `--protdating` 
-0000dbe0: 746f 206f 6e6c 7920 636f 6e64 7563 7420  to only conduct 
-0000dbf0: 7468 6520 6461 7469 6e67 206f 6620 7468  the dating of th
-0000dc00: 6520 7072 6f74 6569 6e20 4d53 4173 2c20  e protein MSAs, 
-0000dc10: 6e6f 7465 6420 7468 6174 2074 6865 7365  noted that these
-0000dc20: 2074 776f 206f 7074 696f 6e73 206f 6e6c   two options onl
-0000dc30: 7920 776f 726b 2066 6f72 2074 6865 2060  y work for the `
-0000dc40: 6d63 6d63 7472 6565 6020 6f70 7469 6f6e  mcmctree` option
-0000dc50: 2073 6f20 6661 722e 2054 6865 2066 6c61   so far. The fla
-0000dc60: 6720 6f70 7469 6f6e 2060 2d2d 7061 7274  g option `--part
-0000dc70: 6974 696f 6e60 2063 616e 2062 6520 7365  ition` can be se
-0000dc80: 7420 746f 2070 6572 666f 726d 2060 6d63  t to perform `mc
-0000dc90: 6d63 7472 6565 6020 616e 616c 7973 6973  mctree` analysis
-0000dca0: 2075 7369 6e67 2074 6865 2070 6172 7469   using the parti
-0000dcb0: 7469 6f6e 6564 2064 6174 6120 2869 2e65  tioned data (i.e
-0000dcc0: 2e2c 2031 7374 2c20 326e 6420 616e 6420  ., 1st, 2nd and 
-0000dcd0: 3372 6420 706f 7369 7469 6f6e 206f 6620  3rd position of 
-0000dce0: 636f 646f 6e29 2069 6e73 7465 6164 206f  codon) instead o
-0000dcf0: 6620 7573 696e 6720 7468 6520 636f 646f  f using the codo
-0000dd00: 6e20 6173 2061 2077 686f 6c65 2e20 5468  n as a whole. Th
-0000dd10: 6520 6f70 7469 6f6e 2060 2d2d 6161 6d6f  e option `--aamo
-0000dd20: 6465 6c60 2063 616e 2062 6520 7365 7420  del` can be set 
-0000dd30: 746f 2064 6574 6572 6d69 6e65 2074 6865  to determine the
-0000dd40: 2061 6d69 6e6f 2061 6369 6420 6d6f 6465   amino acid mode
-0000dd50: 6c20 6170 706c 6965 6420 696e 2060 6d63  l applied in `mc
-0000dd60: 6d63 7472 6565 6020 616e 616c 7973 6973  mctree` analysis
-0000dd70: 2e20 5468 6520 6f70 7469 6f6e 2060 2d64  . The option `-d
-0000dd80: 7360 2063 616e 2062 6520 7573 6564 2074  s` can be used t
-0000dd90: 6f20 7365 7420 7061 7261 6d65 7465 7273  o set parameters
-0000dda0: 2066 6f72 2074 6865 206d 6f6c 6563 756c   for the molecul
-0000ddb0: 6172 2064 6174 696e 6720 7072 6f67 7261  ar dating progra
-0000ddc0: 6d2e 2054 6865 2070 6172 616c 6c65 6c20  m. The parallel 
-0000ddd0: 7468 7265 6164 7320 6865 7265 2061 7265  threads here are
-0000dde0: 2074 6f20 7061 7261 6c6c 656c 697a 6520   to parallelize 
-0000ddf0: 7468 6520 616e 616c 7973 6973 2066 6f72  the analysis for
-0000de00: 2065 6163 6820 6765 6e65 2066 616d 696c   each gene famil
-0000de10: 7920 616e 6420 7468 7573 2073 7567 6765  y and thus sugge
-0000de20: 7374 6564 2074 6f20 6265 2073 6574 2061  sted to be set a
-0000de30: 7320 6d75 6368 2061 7320 7468 6520 6e75  s much as the nu
-0000de40: 6d62 6572 206f 6620 6765 6e65 2066 616d  mber of gene fam
-0000de50: 696c 6965 732e 0a0a 2a2a 4120 7375 6767  ilies...**A sugg
-0000de60: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-0000de70: 756e 2063 616e 2062 6520 7769 7468 2074  un can be with t
-0000de80: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
-0000de90: 2e2a 2a0a 0a60 6060 0a77 6764 2066 6f63  .**..```.wgd foc
-0000dea0: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
-0000deb0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000dec0: 7365 7175 656e 6365 3320 2d64 206d 636d  sequence3 -d mcm
-0000ded0: 6374 7265 6520 2d73 7020 7370 6461 7461  ctree -sp spdata
-0000dee0: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
-0000def0: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
-0000df00: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
-0000df10: 3320 2d64 2072 3873 202d 7370 2073 7064  3 -d r8s -sp spd
-0000df20: 6174 6120 2d2d 6e73 6974 6573 206e 7369  ata --nsites nsi
-0000df30: 7465 696e 666f 0a77 6764 2066 6f63 7573  teinfo.wgd focus
-0000df40: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
-0000df50: 6365 3120 7365 7175 656e 6365 3220 7365  ce1 sequence2 se
-0000df60: 7175 656e 6365 3320 2d64 2062 6561 7374  quence3 -d beast
-0000df70: 202d 2d66 6f73 7369 6c20 666f 7373 696c   --fossil fossil
-0000df80: 696e 666f 202d 2d72 6f6f 7468 6569 6768  info --rootheigh
-0000df90: 7420 726f 6f74 6865 6967 6874 696e 666f  t rootheightinfo
-0000dfa0: 202d 2d63 6861 696e 7365 7420 6368 6169   --chainset chai
-0000dfb0: 6e73 6574 696e 666f 202d 2d62 6561 7374  nsetinfo --beast
-0000dfc0: 6c67 6a61 7220 2450 4154 480a 6060 600a  lgjar $PATH.```.
-0000dfd0: 0a23 2323 2077 6764 206b 7364 0a0a 2a2a  .### wgd ksd..**
-0000dfe0: 5468 6520 636f 6e73 7472 7563 7469 6f6e  The construction
-0000dff0: 206f 6620 7768 6f6c 6520 7061 7261 6e6f   of whole parano
-0000e000: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
-0000e010: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-0000e020: 696f 6e2a 2a0a 6060 600a 7767 6420 6b73  ion**.```.wgd ks
-0000e030: 6420 6661 6d69 6c69 6573 2073 6571 7565  d families seque
-0000e040: 6e63 6520 282d 6f20 7767 645f 6b73 6420  nce (-o wgd_ksd 
-0000e050: 2d74 2077 6764 5f6b 7364 5f74 6d70 202d  -t wgd_ksd_tmp -
-0000e060: 2d6e 7468 7265 6164 7320 3420 2d2d 746f  -nthreads 4 --to
-0000e070: 5f73 746f 7020 2d2d 6364 7320 2d2d 7061  _stop --cds --pa
-0000e080: 6972 7769 7365 202d 2d73 7472 6970 5f67  irwise --strip_g
-0000e090: 6170 7320 2d2d 616c 6967 6e65 7220 6d61  aps --aligner ma
-0000e0a0: 6666 7420 2d2d 7472 6565 5f6d 6574 686f  fft --tree_metho
-0000e0b0: 6420 6661 7374 7472 6565 202d 2d6e 6f64  d fasttree --nod
-0000e0c0: 655f 6176 6572 6167 6529 0a60 6060 0a0a  e_average).```..
-0000e0d0: 5468 6520 7072 6f67 7261 6d20 6077 6764  The program `wgd
-0000e0e0: 206b 7364 602c 2061 7320 696d 7069 6564   ksd`, as impied
-0000e0f0: 2062 7920 6974 7320 6e61 6d65 2c20 6973   by its name, is
-0000e100: 2066 6f72 2074 6865 2063 6f6e 7374 7275   for the constru
-0000e110: 6374 696f 6e20 6f66 202a 4b2a 3c73 7562  ction of *K*<sub
-0000e120: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-0000e130: 7472 6962 7574 696f 6e2e 2045 7863 6570  tribution. Excep
-0000e140: 7420 666f 7220 7468 6520 6166 6f72 656d  t for the aforem
-0000e150: 656e 7469 6f6e 6564 2070 6172 616d 6574  entioned paramet
-0000e160: 6572 7320 7375 6368 2061 7320 602d 2d74  ers such as `--t
-0000e170: 6f5f 7374 6f70 6020 616e 6420 602d 2d63  o_stop` and `--c
-0000e180: 6473 602c 2074 6865 7265 2061 7265 2073  ds`, there are s
-0000e190: 6f6d 6520 696d 706f 7274 616e 7420 7061  ome important pa
-0000e1a0: 7261 6d65 7465 7273 2074 6861 7420 6861  rameters that ha
-0000e1b0: 7665 2063 7275 6369 616c 2069 6d70 6163  ve crucial impac
-0000e1c0: 7420 6f6e 2074 6865 202a 4b2a 3c73 7562  t on the *K*<sub
-0000e1d0: 3e53 3c2f 7375 623e 2065 7374 696d 6174  >S</sub> estimat
-0000e1e0: 696f 6e2e 2054 6865 206f 7074 696f 6e20  ion. The option 
-0000e1f0: 602d 2d70 6169 7277 6973 6560 2069 7320  `--pairwise` is 
-0000e200: 6120 7665 7279 2069 6d70 6f72 7461 6e74  a very important
-0000e210: 2070 6172 616d 6574 6572 2066 6f72 2074   parameter for t
-0000e220: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
-0000e230: 623e 2065 7374 696d 6174 696f 6e2c 2077  b> estimation, w
-0000e240: 6974 6820 7768 6963 6820 7468 6520 6043  ith which the `C
-0000e250: 4f44 454d 4c60 2077 696c 6c20 6361 6c63  ODEML` will calc
-0000e260: 756c 6174 6520 7468 6520 2a4b 2a3c 7375  ulate the *K*<su
-0000e270: 623e 533c 2f73 7562 3e20 666f 7220 6561  b>S</sub> for ea
-0000e280: 6368 2067 656e 6520 7061 6972 2073 6570  ch gene pair sep
-0000e290: 6172 6174 656c 7920 6261 7365 6420 6f6e  arately based on
-0000e2a0: 2074 6865 2061 6c69 676e 6d65 6e74 206f   the alignment o
-0000e2b0: 6620 6f6e 6c79 2074 6865 7365 2074 776f  f only these two
-0000e2c0: 2067 656e 6573 2069 6e73 7465 6164 206f   genes instead o
-0000e2d0: 6620 7468 6520 7768 6f6c 6520 616c 6967  f the whole alig
-0000e2e0: 6e6d 656e 7420 6f66 2074 6865 2066 616d  nment of the fam
-0000e2f0: 696c 792c 2073 7563 6820 7468 6174 206c  ily, such that l
-0000e300: 6573 7320 6761 7073 2061 7265 2065 7870  ess gaps are exp
-0000e310: 6563 7465 6420 616e 6420 7468 7573 2074  ected and thus t
-0000e320: 6865 2061 6c69 676e 6d65 6e74 2069 6e20  he alignment in 
-0000e330: 7468 6520 636f 6e73 6964 6572 6174 696f  the consideratio
-0000e340: 6e20 6f66 2060 434f 4445 4d4c 6020 7769  n of `CODEML` wi
-0000e350: 6c6c 2062 6520 6c6f 6e67 6572 2028 6265  ll be longer (be
-0000e360: 6361 7573 6520 6043 4f44 454d 4c60 2077  cause `CODEML` w
-0000e370: 696c 6c20 6175 746f 6d61 7465 6c79 2073  ill automately s
-0000e380: 6b69 7020 6576 6572 7920 636f 6c75 6d6e  kip every column
-0000e390: 2077 6974 6820 6761 702c 2072 6567 6172   with gap, regar
-0000e3a0: 646c 6573 7320 6f66 2077 6865 7468 6572  dless of whether
-0000e3b0: 2069 7427 7320 616e 206f 7665 7261 6c6c   it's an overall
-0000e3c0: 206f 7220 7061 7274 6961 6c20 6761 7029   or partial gap)
-0000e3d0: 2c20 7769 7468 6f75 7420 7768 6963 6820  , without which 
-0000e3e0: 7468 6520 6043 4f44 454d 4c60 2077 696c  the `CODEML` wil
-0000e3f0: 6c20 6361 6c63 756c 6174 6520 7468 6520  l calculate the 
-0000e400: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000e410: 6261 7365 6420 6f6e 2074 6865 2077 686f  based on the who
-0000e420: 6c65 2061 6c69 676e 6d65 6e74 206f 6620  le alignment of 
-0000e430: 7468 6520 6661 6d69 6c79 2c20 7768 6963  the family, whic
-0000e440: 6820 6d69 6768 7420 6861 7665 206e 6f20  h might have no 
-0000e450: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000e460: 7265 7375 6c74 2061 7420 616c 6c20 6966  result at all if
-0000e470: 2074 6865 2073 7472 6970 7065 6420 616c   the stripped al
-0000e480: 6967 6e6d 656e 7420 6c65 6e67 7468 2028  ignment length (
-0000e490: 7265 6d6f 7669 6e67 2061 6c6c 2067 6170  removing all gap
-0000e4a0: 2d63 6f6e 7461 696e 696e 6720 636f 6c75  -containing colu
-0000e4b0: 6d6e 7329 2077 6173 207a 6572 6f2c 2061  mns) was zero, a
-0000e4c0: 2063 6175 7365 206f 6620 6469 6666 6572   cause of differ
-0000e4d0: 656e 7420 6e75 6d62 6572 206f 6620 2a4b  ent number of *K
-0000e4e0: 2a3c 7375 623e 533c 2f73 7562 3e20 6573  *<sub>S</sub> es
-0000e4f0: 7469 6d61 7465 7320 6265 7477 6565 6e20  timates between 
-0000e500: 2270 6169 7277 6973 6520 6d6f 6465 2220  "pairwise mode" 
-0000e510: 616e 6420 226e 6f6e 2d70 6169 7277 6973  and "non-pairwis
-0000e520: 6520 6d6f 6465 222e 2049 7427 7320 6469  e mode". It's di
-0000e530: 6666 6963 756c 7420 746f 2073 6179 2077  fficult to say w
-0000e540: 6869 6368 206d 6f64 6520 6973 206d 6f72  hich mode is mor
-0000e550: 6520 6964 6561 6c2c 2061 6c74 686f 7567  e ideal, althoug
-0000e560: 6820 7468 6520 226e 6f6e 2d70 6169 7277  h the "non-pairw
-0000e570: 6973 6520 6d6f 6465 2220 2864 6566 6175  ise mode" (defau
-0000e580: 6c74 2073 6574 7469 6e67 2920 7768 6963  lt setting) whic
-0000e590: 6820 7275 6e73 206f 6e20 7468 6520 7768  h runs on the wh
-0000e5a0: 6f6c 6520 616c 6967 6e6d 656e 7420 696e  ole alignment in
-0000e5b0: 7374 6561 6420 6f66 2061 206c 6f63 616c  stead of a local
-0000e5c0: 2061 6c69 676e 6d65 6e74 2c20 6d69 6768   alignment, migh
-0000e5d0: 7420 6265 206d 6f72 6520 6269 6f6c 6f67  t be more biolog
-0000e5e0: 6963 616c 6c79 2063 6f6e 7365 7276 6564  ically conserved
-0000e5f0: 2069 6e20 7468 6174 2069 7420 6173 7375   in that it assu
-0000e600: 7265 7320 7468 6520 6576 6f6c 7574 696f  res the evolutio
-0000e610: 6e20 6f66 2065 6163 6820 636f 6c75 6d6e  n of each column
-0000e620: 2074 6f20 6265 2073 7461 7274 6564 2066   to be started f
-0000e630: 726f 6d20 7468 6520 726f 6f74 206f 6620  rom the root of 
-0000e640: 7468 6520 6661 6d69 6c79 2061 6e64 2061  the family and a
-0000e650: 6c6c 2074 6865 2067 656e 6520 6475 706c  ll the gene dupl
-0000e660: 6963 6174 6573 2061 7265 2074 616b 656e  icates are taken
-0000e670: 2069 6e74 6f20 6163 636f 756e 7420 696e   into account in
-0000e680: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-0000e690: 7375 623e 2065 7374 696d 6174 696f 6e20  sub> estimation 
-0000e6a0: 7072 6f63 6573 732e 2054 6865 206f 7074  process. The opt
-0000e6b0: 696f 6e20 602d 2d73 7472 6970 5f67 6170  ion `--strip_gap
-0000e6c0: 7360 2063 616e 2072 656d 6f76 6520 616c  s` can remove al
-0000e6d0: 6c20 7468 6520 6761 702d 636f 6e74 6169  l the gap-contai
-0000e6e0: 6e69 6e67 2063 6f6c 756d 6e73 2c20 7769  ning columns, wi
-0000e6f0: 7468 206f 7220 7769 7468 6f75 7420 7768  th or without wh
-0000e700: 6963 6820 7468 6520 7265 7375 6c74 206f  ich the result o
-0000e710: 6620 226e 6f6e 2d70 6169 7277 6973 6520  f "non-pairwise 
-0000e720: 6d6f 6465 2220 776f 6e27 7420 6265 2061  mode" won't be a
-0000e730: 6666 6563 7465 642c 2077 6869 6c65 2077  ffected, while w
-0000e740: 6974 6820 7768 6963 6820 7468 6520 7265  ith which the re
-0000e750: 7375 6c74 206f 6620 2270 6169 7277 6973  sult of "pairwis
-0000e760: 6520 6d6f 6465 2220 7769 6c6c 2062 6520  e mode" will be 
-0000e770: 616c 7465 7265 642e 2054 6865 206f 7074  altered. The opt
-0000e780: 696f 6e20 602d 2d61 6c69 676e 6572 6020  ion `--aligner` 
-0000e790: 616e 6420 602d 2d61 6c6e 5f6f 7074 696f  and `--aln_optio
-0000e7a0: 6e73 6020 7768 6963 6820 6465 6369 6465  ns` which decide
-0000e7b0: 2077 6869 6368 2061 6c69 676e 6d65 6e74   which alignment
-0000e7c0: 2070 726f 6772 616d 2074 6f20 6265 2075   program to be u
-0000e7d0: 7365 6420 616e 6420 7768 6963 6820 7061  sed and which pa
-0000e7e0: 7261 6d65 7465 7220 746f 2062 6520 7365  rameter to be se
-0000e7f0: 742c 2077 696c 6c20 6861 7665 2069 6d70  t, will have imp
-0000e800: 6163 7420 6f6e 2074 6865 202a 4b2a 3c73  act on the *K*<s
-0000e810: 7562 3e53 3c2f 7375 623e 2072 6573 756c  ub>S</sub> resul
-0000e820: 7473 2c20 6e6f 7465 6420 7468 6174 2074  ts, noted that t
-0000e830: 6865 2064 6566 6175 6c74 2070 726f 6772  he default progr
-0000e840: 616d 2069 7320 606d 6166 6674 6020 616e  am is `mafft` an
-0000e850: 6420 7468 6520 7061 7261 6d65 7465 7220  d the parameter 
-0000e860: 6973 2060 2d2d 6175 746f 602e 2054 6865  is `--auto`. The
-0000e870: 206f 7074 696f 6e20 602d 2d74 7265 655f   option `--tree_
-0000e880: 6d65 7468 6f64 6020 616e 6420 602d 2d74  method` and `--t
-0000e890: 7265 655f 6f70 7469 6f6e 7360 2064 6563  ree_options` dec
-0000e8a0: 6964 6520 7768 6963 6820 6765 6e65 2074  ide which gene t
-0000e8b0: 7265 6520 696e 6665 7265 6e63 6520 7072  ree inference pr
-0000e8c0: 6f67 7261 6d20 746f 2062 6520 7573 6564  ogram to be used
-0000e8d0: 2061 6e64 2077 6869 6368 2070 6172 616d   and which param
-0000e8e0: 6574 6572 2074 6f20 6265 2073 6574 2c20  eter to be set, 
-0000e8f0: 776f 6e27 7420 6166 6665 6374 2074 6865  won't affect the
-0000e900: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000e910: 2065 7374 696d 6174 696f 6e20 6974 7365   estimation itse
-0000e920: 6c66 2062 7574 2074 6865 2072 6573 756c  lf but the resul
-0000e930: 7420 6f66 2064 652d 7265 6475 6e64 616e  t of de-redundan
-0000e940: 6379 2c20 6e6f 7465 6420 7468 6174 2077  cy, noted that w
-0000e950: 6520 696d 706c 656d 656e 7465 6420 6120  e implemented a 
-0000e960: 6275 696c 742d 696e 2067 656e 6520 7472  built-in gene tr
-0000e970: 6565 2069 6e66 6572 656e 6365 206d 6574  ee inference met
-0000e980: 686f 6420 6261 7365 6420 6f6e 2074 6865  hod based on the
-0000e990: 2041 7665 7261 6765 204c 696e 6b61 6765   Average Linkage
-0000e9a0: 2043 6c75 7374 6572 696e 6720 2841 4c43   Clustering (ALC
-0000e9b0: 2920 2874 6875 7320 6120 6469 7374 616e  ) (thus a distan
-0000e9c0: 6365 2d62 6173 6564 2074 7265 6529 2077  ce-based tree) w
-0000e9d0: 6974 6820 7468 6520 602d 2d74 7265 655f  ith the `--tree_
-0000e9e0: 6d65 7468 6f64 6020 7365 7420 6173 2022  method` set as "
-0000e9f0: 636c 7573 7465 7222 2e20 5477 6f20 6d65  cluster". Two me
-0000ea00: 7468 6f64 7320 6f66 2064 652d 7265 6475  thods of de-redu
-0000ea10: 6e64 616e 6379 2061 7265 2069 6d70 6c65  ndancy are imple
-0000ea20: 6d65 6e74 6564 2069 6e20 6077 6764 2076  mented in `wgd v
-0000ea30: 3260 2c20 6e61 6d65 6c79 206e 6f64 652d  2`, namely node-
-0000ea40: 7765 6967 6874 6564 2061 6e64 206e 6f64  weighted and nod
-0000ea50: 652d 6176 6572 6167 6564 206d 6574 686f  e-averaged metho
-0000ea60: 6473 2e20 5468 6520 6e6f 6465 2d77 6569  ds. The node-wei
-0000ea70: 6768 7465 6420 6d65 7468 6f64 2061 6368  ghted method ach
-0000ea80: 6965 7665 7320 7468 6520 6465 2d72 6564  ieves the de-red
-0000ea90: 756e 6461 6e63 7920 7669 6120 7765 6967  undancy via weig
-0000eaa0: 6869 6e67 2074 6865 202a 4b2a 3c73 7562  hing the *K*<sub
-0000eab0: 3e53 3c2f 7375 623e 2076 616c 7565 2061  >S</sub> value a
-0000eac0: 7373 6f63 6961 7465 6420 7769 7468 2065  ssociated with e
-0000ead0: 6163 6820 6765 6e65 2070 6169 7220 7375  ach gene pair su
-0000eae0: 6368 2074 6861 7420 7468 6520 7765 6967  ch that the weig
-0000eaf0: 6874 206f 6620 6120 7369 6e67 6c65 2064  ht of a single d
-0000eb00: 7570 6c69 6361 7469 6f6e 2065 7665 6e74  uplication event
-0000eb10: 2073 756d 7320 7570 2074 6f20 3120 286e   sums up to 1 (n
-0000eb20: 6f74 6564 2074 6861 7420 7468 6520 6e75  oted that the nu
-0000eb30: 6d62 6572 206f 6620 2a4b 2a3c 7375 623e  mber of *K*<sub>
-0000eb40: 533c 2f73 7562 3e20 6573 7469 6d61 7469  S</sub> estimati
-0000eb50: 6f6e 2072 656d 6169 6e73 2074 6865 2073  on remains the s
-0000eb60: 616d 6529 2077 6869 6c65 2074 6865 206e  ame) while the n
-0000eb70: 6f64 652d 6176 6572 6167 6564 206d 6574  ode-averaged met
-0000eb80: 686f 6420 7265 616c 697a 6573 2074 6865  hod realizes the
-0000eb90: 2064 652d 7265 6475 6e64 616e 6379 2076   de-redundancy v
-0000eba0: 6961 2063 616c 6375 6c61 7469 6e67 2070  ia calculating p
-0000ebb0: 6572 2067 656e 6520 7472 6565 206e 6f64  er gene tree nod
-0000ebc0: 6520 6f6e 6520 6176 6572 6167 6564 202a  e one averaged *
-0000ebd0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2076  K*<sub>S</sub> v
-0000ebe0: 616c 7565 2074 6f20 7265 7072 6573 656e  alue to represen
-0000ebf0: 7420 7468 6520 6167 6520 6f66 2065 6163  t the age of eac
-0000ec00: 6820 6475 706c 6963 6174 696f 6e20 6576  h duplication ev
-0000ec10: 656e 742e 2054 6865 206f 7074 696f 6e20  ent. The option 
-0000ec20: 602d 2d6e 6f64 655f 6176 6572 6167 6560  `--node_average`
-0000ec30: 2063 616e 2062 6520 7365 7420 746f 2063   can be set to c
-0000ec40: 686f 6f73 6520 7468 6520 6e6f 6465 2d61  hoose the node-a
-0000ec50: 7665 7261 6765 6420 7761 7920 6f66 2064  veraged way of d
-0000ec60: 652d 7265 6475 6e64 616e 6379 2e20 4469  e-redundancy. Di
-0000ec70: 6666 6572 656e 7420 6d65 7468 6f64 7320  fferent methods 
-0000ec80: 6f66 2064 652d 7265 6475 6e64 616e 6379  of de-redundancy
-0000ec90: 2077 696c 6c20 6861 7665 2069 6d70 6163   will have impac
-0000eca0: 7420 6f6e 2074 6865 2064 6574 6563 7469  t on the detecti
-0000ecb0: 6f6e 206f 6620 5747 4420 7369 676e 616c  on of WGD signal
-0000ecc0: 732c 2077 6869 6368 2068 6173 2062 6565  s, which has bee
-0000ecd0: 6e20 696e 7665 7374 6967 6174 6564 2069  n investigated i
-0000ece0: 6e20 7468 6973 205b 6c69 7465 7261 7475  n this [literatu
-0000ecf0: 7265 5d28 6874 7470 733a 2f2f 646f 692e  re](https://doi.
-0000ed00: 6f72 672f 3130 2e31 3039 332f 6762 652f  org/10.1093/gbe/
-0000ed10: 6576 7932 3030 292e 2054 6865 2070 6172  evy200). The par
-0000ed20: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
-0000ed30: 7265 2061 7265 2074 6f20 7061 7261 6c6c  re are to parall
-0000ed40: 656c 697a 6520 7468 6520 616e 616c 7973  elize the analys
-0000ed50: 6973 2066 6f72 2065 6163 6820 6765 6e65  is for each gene
-0000ed60: 2066 616d 696c 7920 616e 6420 7468 7573   family and thus
-0000ed70: 2073 7567 6765 7374 6564 2074 6f20 6265   suggested to be
-0000ed80: 2073 6574 2061 7320 6d75 6368 2061 7320   set as much as 
-0000ed90: 7468 6520 6e75 6d62 6572 206f 6620 6765  the number of ge
-0000eda0: 6e65 2066 616d 696c 6965 732e 0a0a 2a2a  ne families...**
-0000edb0: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-0000edc0: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
-0000edd0: 2063 6f6d 6d61 6e64 2061 7320 6265 6c6f   command as belo
-0000ede0: 772a 2a0a 0a60 6060 0a77 6764 206b 7364  w**..```.wgd ksd
-0000edf0: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
-0000ee00: 6365 0a60 6060 0a0a 2a2a 5468 6520 636f  ce.```..**The co
-0000ee10: 6e73 7472 7563 7469 6f6e 206f 6620 6f72  nstruction of or
-0000ee20: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
-0000ee30: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-0000ee40: 7374 7269 6275 7469 6f6e 2a2a 0a60 6060  stribution**.```
-0000ee50: 0a77 6764 206b 7364 2066 616d 696c 6965  .wgd ksd familie
-0000ee60: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
-0000ee70: 656e 6365 3220 282d 2d72 6577 6569 6768  ence2 (--reweigh
-0000ee80: 7429 0a60 6060 0a0a 4672 6f6d 2070 6172  t).```..From par
-0000ee90: 616c 6f67 6f75 7320 746f 206f 7274 686f  alogous to ortho
-0000eea0: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-0000eeb0: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-0000eec0: 6962 7574 696f 6e2c 2075 7365 7273 206f  ibution, users o
-0000eed0: 6e6c 7920 6e65 6564 2074 6f20 7072 6f76  nly need to prov
-0000eee0: 6964 6520 6d6f 7265 2063 6473 2066 696c  ide more cds fil
-0000eef0: 6573 2e20 4e6f 7465 2074 6861 7420 7769  es. Note that wi
-0000ef00: 7468 206f 7274 686f 6c6f 676f 7573 2067  th orthologous g
-0000ef10: 656e 6520 6661 6d69 6c69 6573 2074 6865  ene families the
-0000ef20: 2077 6569 6768 7469 6e67 206d 6574 686f   weighting metho
-0000ef30: 6420 6361 6e20 6265 2073 6574 2074 6f20  d can be set to 
-0000ef40: 6265 2063 616c 6375 6c61 7465 6420 7065  be calculated pe
-0000ef50: 7220 7370 6563 6965 7320 7061 6972 2069  r species pair i
-0000ef60: 6e73 7465 6164 206f 6620 636f 6e73 6964  nstead of consid
-0000ef70: 6572 696e 6720 7468 6520 7768 6f6c 6520  ering the whole 
-0000ef80: 6661 6d69 6c79 2062 6563 6175 7365 2077  family because w
-0000ef90: 6865 6e20 706c 6f74 7469 6e67 206f 7274  hen plotting ort
-0000efa0: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
-0000efb0: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-0000efc0: 7472 6962 7574 696f 6e20 6265 7477 6565  tribution betwee
-0000efd0: 6e20 7477 6f20 7370 6563 6965 7320 7468  n two species th
-0000efe0: 6520 7765 6967 6874 2063 616c 6375 6c61  e weight calcula
-0000eff0: 7465 6420 6672 6f6d 2074 6869 7320 7370  ted from this sp
-0000f000: 6563 6966 6963 2073 7065 6369 6573 2070  ecific species p
-0000f010: 6169 7220 7368 6f75 6c64 2062 6520 636f  air should be co
-0000f020: 6e73 6572 7665 6420 7768 696c 6520 7468  nserved while th
-0000f030: 6520 6f6e 6520 6361 6c63 756c 6174 6564  e one calculated
-0000f040: 2066 726f 6d20 7468 6520 7768 6f6c 6520   from the whole 
-0000f050: 6661 6d69 6c79 2077 696c 6c20 7661 7279  family will vary
-0000f060: 2077 6974 6820 7468 6520 6e75 6d62 6572   with the number
-0000f070: 206f 6620 7370 6563 6965 732e 2054 6f20   of species. To 
-0000f080: 696e 6974 6961 7465 2074 6865 2077 6569  initiate the wei
-0000f090: 6768 7469 6e67 2070 6572 2073 7065 6369  ghting per speci
-0000f0a0: 6573 2070 6169 722c 2074 6865 206f 7074  es pair, the opt
-0000f0b0: 696f 6e20 602d 2d72 6577 6569 6768 7460  ion `--reweight`
-0000f0c0: 2063 616e 2062 6520 7365 742e 0a0a 2a2a   can be set...**
-0000f0d0: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-0000f0e0: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
-0000f0f0: 2063 6f6d 6d61 6e64 2061 7320 6265 6c6f   command as belo
-0000f100: 772a 2a0a 0a60 6060 0a77 6764 206b 7364  w**..```.wgd ksd
-0000f110: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
-0000f120: 6365 3120 7365 7175 656e 6365 320a 6060  ce1 sequence2.``
-0000f130: 600a 0a2a 2a54 6865 2063 6f6e 7374 7275  `..**The constru
-0000f140: 6374 696f 6e20 6f66 202a 4b2a 3c73 7562  ction of *K*<sub
-0000f150: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-0000f160: 7472 6962 7574 696f 6e20 7769 7468 2072  tribution with r
-0000f170: 6174 6520 636f 7272 6563 7469 6f6e 2a2a  ate correction**
-0000f180: 0a60 6060 0a77 6764 206b 7364 2066 616d  .```.wgd ksd fam
-0000f190: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
-0000f1a0: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
-0000f1b0: 6365 3320 2d2d 666f 6375 7332 616c 6c20  ce3 --focus2all 
-0000f1c0: 7365 7175 656e 6365 3120 2d73 7020 7370  sequence1 -sp sp
-0000f1d0: 6461 7461 0a60 6060 0a0a 496e 7370 6972  data.```..Inspir
-0000f1e0: 6564 2062 7920 7468 6520 7261 7465 2063  ed by the rate c
-0000f1f0: 6f72 7265 6374 696f 6e20 616c 676f 7269  orrection algori
-0000f200: 7468 6d20 696e 205b 6b73 7261 7465 735d  thm in [ksrates]
-0000f210: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
-0000f220: 636f 6d2f 5649 422d 5053 422f 6b73 7261  com/VIB-PSB/ksra
-0000f230: 7465 7329 2c20 7765 2069 6d70 6c65 6d65  tes), we impleme
-0000f240: 6e74 6564 2074 6865 2072 6174 6520 636f  nted the rate co
-0000f250: 7272 6563 7469 6f6e 2061 6e61 6c79 7369  rrection analysi
-0000f260: 7320 616c 736f 2069 6e20 6077 6764 2076  s also in `wgd v
-0000f270: 3260 2c20 7768 6963 6820 6973 206d 6f73  2`, which is mos
-0000f280: 746c 7920 7468 6520 7361 6d65 2061 7320  tly the same as 
-0000f290: 5b6b 7372 6174 6573 5d28 6874 7470 733a  [ksrates](https:
-0000f2a0: 2f2f 6769 7468 7562 2e63 6f6d 2f56 4942  //github.com/VIB
-0000f2b0: 2d50 5342 2f6b 7372 6174 6573 2920 6275  -PSB/ksrates) bu
-0000f2c0: 7420 6469 6666 6572 7320 696e 2074 6865  t differs in the
-0000f2d0: 2063 616c 6375 6c61 7469 6f6e 206f 6620   calculation of 
-0000f2e0: 7468 6520 7374 616e 6461 7264 2064 6576  the standard dev
-0000f2f0: 6961 7469 6f6e 206f 6620 7265 7363 616c  iation of rescal
-0000f300: 6564 202a 4b2a 3c73 7562 3e53 3c2f 7375  ed *K*<sub>S</su
-0000f310: 623e 2061 6765 732e 2054 6f20 7065 7266  b> ages. To perf
-0000f320: 6f72 6d20 7468 6520 7261 7465 2063 6f72  orm the rate cor
-0000f330: 7265 6374 696f 6e20 616e 616c 7973 6973  rection analysis
-0000f340: 2c20 7573 6572 7320 6361 6e20 7573 6520  , users can use 
-0000f350: 626f 7468 2074 6865 2060 7767 6420 6b73  both the `wgd ks
-0000f360: 6460 2061 6e64 2060 7767 6420 7669 7a60  d` and `wgd viz`
-0000f370: 2070 726f 6772 616d 2e20 466f 7220 7468   program. For th
-0000f380: 6520 6077 6764 206b 7364 6020 7072 6f67  e `wgd ksd` prog
-0000f390: 7261 6d2c 2075 7365 7273 206e 6565 6420  ram, users need 
-0000f3a0: 746f 2070 726f 7669 6465 2061 2073 7065  to provide a spe
-0000f3b0: 6369 6573 2074 7265 6520 7669 6120 7468  cies tree via th
-0000f3c0: 6520 6f70 7469 6f6e 2060 2d2d 7370 6563  e option `--spec
-0000f3d0: 6965 7374 7265 6560 206f 6e20 7768 6963  iestree` on whic
-0000f3e0: 6820 7261 7465 2063 6f72 7265 6374 696f  h rate correctio
-0000f3f0: 6e20 6361 6e20 6265 2063 6f6e 6475 6374  n can be conduct
-0000f400: 6564 2e20 4e6f 7465 2074 6861 7420 756e  ed. Note that un
-0000f410: 6e65 6365 7373 6172 7920 6272 6163 6b65  necessary bracke
-0000f420: 7473 206d 6967 6874 206c 6561 6420 746f  ts might lead to
-0000f430: 2075 6e65 7870 6563 7465 6420 6572 726f   unexpected erro
-0000f440: 7273 2c20 666f 7220 696e 7374 616e 6365  rs, for instance
-0000f450: 2061 2074 7265 6520 6028 412c 2842 2c43   a tree `(A,(B,C
-0000f460: 2929 3b60 2073 686f 756c 6420 6e6f 7420  ));` should not 
-0000f470: 6265 2072 6570 7265 7365 6e74 6564 2061  be represented a
-0000f480: 7320 6028 412c 2828 422c 4329 2929 3b60  s `(A,((B,C)));`
-0000f490: 2e20 5468 6520 7365 7420 6f66 2073 7065  . The set of spe
-0000f4a0: 6369 6573 2070 6169 7273 2074 6f20 6265  cies pairs to be
-0000f4b0: 2073 686f 776e 2069 7320 666c 6578 6962   shown is flexib
-0000f4c0: 6c65 2074 6861 7420 7468 6520 6d6f 7374  le that the most
-0000f4d0: 2063 6f6e 7665 6e69 656e 7420 6f70 7469   convenient opti
-0000f4e0: 6f6e 2069 7320 602d 2d66 6f63 7573 3261  on is `--focus2a
-0000f4f0: 6c6c 6020 7768 6963 6820 7369 6d70 6c79  ll` which simply
-0000f500: 2073 686f 7773 2061 6c6c 2074 6865 2070   shows all the p
-0000f510: 6f73 7369 626c 6520 666f 6361 6c2d 7369  ossible focal-si
-0000f520: 7374 6572 2073 7065 6369 6573 2070 6169  ster species pai
-0000f530: 7273 2c20 6f72 2075 7365 7273 2063 616e  rs, or users can
-0000f540: 206d 616e 7561 6c6c 7920 7365 7420 7468   manually set th
-0000f550: 6520 7370 6563 6965 7320 7061 6972 7320  e species pairs 
-0000f560: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
-0000f570: 2d2d 7370 6169 7260 2e20 4e6f 7465 2074  --spair`. Note t
-0000f580: 6861 7420 6966 2074 6865 2073 7065 6369  hat if the speci
-0000f590: 6573 2070 6169 7273 2077 6572 6520 6d61  es pairs were ma
-0000f5a0: 6e75 616c 6c79 2073 6574 2c20 6974 2077  nually set, it w
-0000f5b0: 6f75 6c64 2062 6520 6e65 6564 6564 2074  ould be needed t
-0000f5c0: 6f20 7365 7420 7468 6520 6f70 7469 6f6e  o set the option
-0000f5d0: 2060 2d2d 636c 6173 7369 6360 2e20 5468   `--classic`. Th
-0000f5e0: 6520 6f70 7469 6f6e 2060 6f6e 6c79 726f  e option `onlyro
-0000f5f0: 6f74 6f75 7460 2063 616e 2062 6520 7365  otout` can be se
-0000f600: 7420 746f 206f 6e6c 7920 636f 6e73 6964  t to only consid
-0000f610: 6572 206f 7574 6772 6f75 7020 6174 2074  er outgroup at t
-0000f620: 6865 2072 6f6f 742c 2069 6e73 7465 6164  he root, instead
-0000f630: 206f 6620 616c 6c20 7468 6520 706f 7373   of all the poss
-0000f640: 6962 6c65 206f 7574 6772 6f75 7073 2070  ible outgroups p
-0000f650: 6572 2066 6f63 616c 2d73 6973 7465 7220  er focal-sister 
-0000f660: 7370 6563 6965 7320 7061 6972 2c20 7768  species pair, wh
-0000f670: 6963 6820 6861 7320 696d 7061 6374 206f  ich has impact o
-0000f680: 6e20 7468 6520 6669 6e61 6c20 636f 7272  n the final corr
-0000f690: 6563 7465 6420 2a4b 2a3c 7375 623e 533c  ected *K*<sub>S<
-0000f6a0: 2f73 7562 3e20 6167 6573 2e20 536f 6d65  /sub> ages. Some
-0000f6b0: 206f 7468 6572 206f 7074 696f 6e73 2077   other options w
-0000f6c0: 6869 6368 2068 6176 6520 6e6f 2069 6d70  hich have no imp
-0000f6d0: 6163 7420 6f6e 2074 6865 2072 6174 6520  act on the rate 
-0000f6e0: 636f 7272 6563 7469 6f6e 2062 7574 2061  correction but a
-0000f6f0: 6464 206d 6f72 6520 6c61 7965 7273 206f  dd more layers o
-0000f700: 7220 6368 616e 6765 2074 6865 2061 7070  r change the app
-0000f710: 6561 7261 6e63 6520 6f6e 2074 6865 206d  earance on the m
-0000f720: 6978 6564 202a 4b2a 3c73 7562 3e53 3c2f  ixed *K*<sub>S</
-0000f730: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-0000f740: 7574 696f 6e2c 2069 6e63 6c75 6465 2060  ution, include `
-0000f750: 706c 6f74 6170 676d 6d60 2061 6e64 2060  plotapgmm` and `
-0000f760: 706c 6f74 656c 6d6d 6020 6574 632e 2054  plotelmm` etc. T
-0000f770: 6865 206f 7074 696f 6e20 6070 6c6f 7461  he option `plota
-0000f780: 7067 6d6d 6020 6361 6e20 6265 2073 6574  pgmm` can be set
-0000f790: 2074 6f20 6361 6c6c 2074 6865 2047 4d4d   to call the GMM
-0000f7a0: 2061 6e61 6c79 7369 7320 7570 6f6e 2074   analysis upon t
-0000f7b0: 6865 2061 6e63 686f 7220 7061 6972 202a  he anchor pair *
-0000f7c0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-0000f7d0: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
-0000f7e0: 616e 6420 706c 6f74 2074 6865 2063 6c75  and plot the clu
-0000f7f0: 7374 6572 696e 6720 7265 7375 6c74 2075  stering result u
-0000f800: 706f 6e20 7468 6520 6d69 7865 6420 2a4b  pon the mixed *K
-0000f810: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
-0000f820: 6520 6469 7374 7269 6275 7469 6f6e 2c20  e distribution, 
-0000f830: 7768 6963 6820 6861 7320 746f 2062 6520  which has to be 
-0000f840: 636f 2d73 6574 2077 6974 6820 7468 6520  co-set with the 
-0000f850: 6f70 7469 6f6e 2060 2d2d 616e 6368 6f72  option `--anchor
-0000f860: 706f 696e 7473 6020 7072 6f76 6964 696e  points` providin
-0000f870: 6720 7468 6520 616e 6368 6f72 2070 6169  g the anchor pai
-0000f880: 7273 2069 6e66 6f72 6d61 7469 6f6e 2e20  rs information. 
-0000f890: 5468 6520 6f70 7469 6f6e 2060 2d2d 706c  The option `--pl
-0000f8a0: 6f74 656c 6d6d 6020 6361 6e20 6265 2073  otelmm` can be s
-0000f8b0: 6574 2074 6f20 6361 6c6c 2074 6865 2045  et to call the E
-0000f8c0: 4c4d 4d20 616e 616c 7973 6973 2075 706f  LMM analysis upo
-0000f8d0: 6e20 7468 6520 7061 7261 6c6f 676f 7573  n the paralogous
-0000f8e0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000f8f0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-0000f900: 6e2e 0a0a 5468 6572 6520 6172 6520 3231  n...There are 21
-0000f910: 2063 6f6c 756d 6e73 2069 6e20 7468 6520   columns in the 
-0000f920: 7265 7375 6c74 2060 2a6b 732e 7473 7660  result `*ks.tsv`
-0000f930: 2066 696c 6520 6265 7369 6465 7320 7468   file besides th
-0000f940: 6520 696e 6465 7820 636f 6c75 6d6e 7320  e index columns 
-0000f950: 6070 6169 7260 2061 7320 7468 6520 756e  `pair` as the un
-0000f960: 6971 7565 2069 6465 6e74 6966 6965 7220  ique identifier 
-0000f970: 666f 7220 6561 6368 2067 656e 6520 7061  for each gene pa
-0000f980: 6972 2e20 5468 6520 604e 602c 2060 5360  ir. The `N`, `S`
-0000f990: 2c20 6064 4e60 2c20 6064 4e2f 6453 602c  , `dN`, `dN/dS`,
-0000f9a0: 2060 6453 602c 2060 6c60 2061 6e64 2060   `dS`, `l` and `
-0000f9b0: 7460 2061 7265 2066 726f 6d20 7468 6520  t` are from the 
-0000f9c0: 636f 6465 6d6c 2072 6573 756c 7473 2c20  codeml results, 
-0000f9d0: 7265 7072 6573 656e 7469 6e67 2074 6865  representing the
-0000f9e0: 204e 2065 7374 696d 6174 652c 2074 6865   N estimate, the
-0000f9f0: 2053 2065 7374 696d 6174 652c 2074 6865   S estimate, the
-0000fa00: 2064 4e20 6573 7469 6d61 7465 2c20 7468   dN estimate, th
-0000fa10: 6520 644e 2f64 5320 286f 6d65 6761 2920  e dN/dS (omega) 
-0000fa20: 6573 7469 6d61 7465 2c20 7468 6520 6453  estimate, the dS
-0000fa30: 2065 7374 696d 6174 652c 2074 6865 206c   estimate, the l
-0000fa40: 6f67 2d6c 696b 656c 6968 6f6f 6420 616e  og-likelihood an
-0000fa50: 6420 7468 6520 7420 6573 7469 6d61 7465  d the t estimate
-0000fa60: 2c20 7265 7370 6563 7469 7665 6c79 2e20  , respectively. 
-0000fa70: 5468 6520 6061 6c69 676e 6d65 6e74 636f  The `alignmentco
-0000fa80: 7665 7261 6765 602c 2060 616c 6967 6e6d  verage`, `alignm
-0000fa90: 656e 7469 6465 6e74 6974 7960 2061 6e64  entidentity` and
-0000faa0: 2060 616c 6967 6e6d 656e 746c 656e 6774   `alignmentlengt
-0000fab0: 6860 2061 7265 2074 6865 2069 6e66 6f72  h` are the infor
-0000fac0: 6d61 7469 6f6e 2070 6572 7461 696e 696e  mation pertainin
-0000fad0: 6720 746f 2074 6865 2061 6c69 676e 6d65  g to the alignme
-0000fae0: 6e74 2066 6f72 2065 6163 6820 6661 6d69  nt for each fami
-0000faf0: 6c79 2c20 7265 7072 6573 656e 7469 6e67  ly, representing
-0000fb00: 2074 6865 2072 6174 696f 206f 6620 7468   the ratio of th
-0000fb10: 6520 7374 7269 7070 6564 2061 6c69 676e  e stripped align
-0000fb20: 6d65 6e74 206c 656e 6774 6820 636f 6d70  ment length comp
-0000fb30: 6172 6564 2074 6f20 7468 6520 6675 6c6c  ared to the full
-0000fb40: 2061 6c69 676e 6d65 6e74 206c 656e 6774   alignment lengt
-0000fb50: 682c 2074 6865 2072 6174 696f 206f 6620  h, the ratio of 
-0000fb60: 636f 6c75 6d6e 7320 7769 7468 2069 6465  columns with ide
-0000fb70: 6e74 6963 616c 206e 7563 6c65 6f74 6964  ntical nucleotid
-0000fb80: 6573 2063 6f6d 7061 7265 6420 746f 2074  es compared to t
-0000fb90: 6865 206f 7665 7261 6c6c 2063 6f6c 756d  he overall colum
-0000fba0: 6e73 206f 6620 7468 6520 7374 7269 7070  ns of the stripp
-0000fbb0: 6564 2061 6c69 676e 6d65 6e74 2c20 616e  ed alignment, an
-0000fbc0: 6420 7468 6520 6c65 6e67 7468 206f 6620  d the length of 
-0000fbd0: 7468 6520 6675 6c6c 2061 6c69 676e 6d65  the full alignme
-0000fbe0: 6e74 2c20 7265 7370 6563 7469 7665 6c79  nt, respectively
-0000fbf0: 2e0a 2323 2320 7767 6420 6d69 780a 0a2a  ..### wgd mix..*
-0000fc00: 2a54 6865 206d 6978 7475 7265 206d 6f64  *The mixture mod
-0000fc10: 656c 2063 6c75 7374 6572 696e 6720 616e  el clustering an
-0000fc20: 616c 7973 6973 206f 6620 2a4b 2a3c 7375  alysis of *K*<su
-0000fc30: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-0000fc40: 7374 7269 6275 7469 6f6e 2a2a 0a60 6060  stribution**.```
-0000fc50: 0a77 6764 206d 6978 206b 7364 6174 6120  .wgd mix ksdata 
-0000fc60: 282d 2d6e 5f69 6e69 7420 3230 3020 2d2d  (--n_init 200 --
-0000fc70: 6d61 785f 6974 6572 2032 3030 202d 2d6b  max_iter 200 --k
-0000fc80: 735f 7261 6e67 6520 3020 3520 2d2d 6669  s_range 0 5 --fi
-0000fc90: 6c74 6572 7320 3330 3020 2d2d 6269 6e73  lters 300 --bins
-0000fca0: 2035 3020 2d2d 636f 6d70 6f6e 656e 7473   50 --components
-0000fcb0: 2031 2034 202d 2d67 616d 6d61 2030 2e30   1 4 --gamma 0.0
-0000fcc0: 3031 290a 6060 600a 0a54 6869 7320 7061  01).```..This pa
-0000fcd0: 7274 206f 6620 4761 7573 7369 616e 206d  rt of Gaussian m
-0000fce0: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
-0000fcf0: 2847 4d4d 2920 616e 616c 7973 6973 2069  (GMM) analysis i
-0000fd00: 7320 696e 6865 7269 7465 6420 6672 6f6d  s inherited from
-0000fd10: 2074 6865 206f 7269 6769 6e61 6c20 6077   the original `w
-0000fd20: 6764 6020 7072 6f67 7261 6d2c 2062 7574  gd` program, but
-0000fd30: 2077 7269 7465 7320 6164 6469 7469 6f6e   writes addition
-0000fd40: 616c 6c79 2074 6865 2070 726f 6261 6269  ally the probabi
-0000fd50: 6c69 7479 206f 6620 6561 6368 202a 4b2a  lity of each *K*
-0000fd60: 3c73 7562 3e53 3c2f 7375 623e 2076 616c  <sub>S</sub> val
-0000fd70: 7565 2069 6e74 6f20 7468 6520 6669 6e61  ue into the fina
-0000fd80: 6c20 6461 7461 6672 616d 652e 2042 6173  l dataframe. Bas
-0000fd90: 6963 616c 6c79 2c20 7573 6572 7320 6e65  ically, users ne
-0000fda0: 6564 2074 6f20 7072 6f76 6964 6520 7769  ed to provide wi
-0000fdb0: 7468 2061 2028 6e6f 726d 616c 6c79 2066  th a (normally f
-0000fdc0: 726f 6d20 7768 6f6c 652d 7061 7261 6e6f  rom whole-parano
-0000fdd0: 6d65 206f 7220 616e 6368 6f72 2d70 6169  me or anchor-pai
-0000fde0: 7273 2920 2a4b 2a3c 7375 623e 533c 2f73  rs) *K*<sub>S</s
-0000fdf0: 7562 3e20 6461 7461 6669 6c65 2061 6e64  ub> datafile and
-0000fe00: 2074 6865 2047 4d4d 2061 6e61 6c79 7369   the GMM analysi
-0000fe10: 7320 7769 6c6c 2062 6520 636f 6e64 7563  s will be conduc
-0000fe20: 7465 6420 7570 6f6e 2074 6865 2064 6174  ted upon the dat
-0000fe30: 6166 696c 652e 2053 6f6d 6520 7061 7261  afile. Some para
-0000fe40: 6d65 7465 7273 2063 616e 2061 6666 6563  meters can affec
-0000fe50: 7420 7468 6520 7265 7375 6c74 732c 2069  t the results, i
-0000fe60: 6e63 6c75 6469 6e67 2060 2d2d 6e5f 696e  ncluding `--n_in
-0000fe70: 6974 602c 2077 6869 6368 2073 6574 7320  it`, which sets 
-0000fe80: 7468 6520 6e75 6d62 6572 206f 6620 6b2d  the number of k-
-0000fe90: 6d65 616e 7320 696e 6974 6961 6c69 7a61  means initializa
-0000fea0: 7469 6f6e 7320 2864 6566 6175 6c74 2032  tions (default 2
-0000feb0: 3030 292c 2060 2d2d 6d61 785f 6974 6572  00), `--max_iter
-0000fec0: 602c 2077 6869 6368 2073 6574 7320 7468  `, which sets th
-0000fed0: 6520 6d61 7869 6d75 6d20 6e75 6d62 6572  e maximum number
-0000fee0: 206f 6620 6974 6572 6174 696f 6e73 2028   of iterations (
-0000fef0: 6465 6661 756c 7420 3230 3029 2c20 602d  default 200), `-
-0000ff00: 2d6d 6574 686f 6460 2c20 7768 6963 6820  -method`, which 
-0000ff10: 6465 7465 726d 696e 6573 2077 6869 6368  determines which
-0000ff20: 2063 6c75 7374 6572 696e 6720 6d65 7468   clustering meth
-0000ff30: 6f64 2074 6f20 7573 6520 2864 6566 6175  od to use (defau
-0000ff40: 6c74 2067 6d6d 292c 2060 2d2d 6761 6d6d  lt gmm), `--gamm
-0000ff50: 6160 2c20 7768 6963 6820 7365 7473 2074  a`, which sets t
-0000ff60: 6865 2067 616d 6d61 2070 6172 616d 6574  he gamma paramet
-0000ff70: 6572 2066 6f72 2074 6865 2062 676d 6d20  er for the bgmm 
-0000ff80: 6d6f 6465 6c20 2864 6566 6175 6c74 2030  model (default 0
-0000ff90: 2e30 3031 292c 2060 2d2d 636f 6d70 6f6e  .001), `--compon
-0000ffa0: 656e 7473 602c 2077 6869 6368 2073 6574  ents`, which set
-0000ffb0: 7320 7468 6520 7261 6e67 6520 6f66 2074  s the range of t
-0000ffc0: 6865 206e 756d 6265 7220 6f66 2063 6f6d  he number of com
-0000ffd0: 706f 6e65 6e74 7320 746f 2066 6974 2028  ponents to fit (
-0000ffe0: 6465 6661 756c 7420 3120 3429 2c20 616e  default 1 4), an
-0000fff0: 6420 7468 6520 6461 7461 2066 696c 7465  d the data filte
-00010000: 7269 6e67 2070 6172 616d 6574 6572 7320  ring parameters 
-00010010: 602d 2d66 696c 7465 7273 6020 7768 6963  `--filters` whic
-00010020: 6820 6669 6c74 6572 7320 6461 7461 2062  h filters data b
-00010030: 6173 6564 206f 6e20 616c 6967 6e6d 656e  ased on alignmen
-00010040: 7420 6c65 6e67 7468 2c20 602d 2d6b 735f  t length, `--ks_
-00010050: 7261 6e67 6560 2077 6869 6368 2066 696c  range` which fil
-00010060: 7465 7273 2064 6174 6120 6261 7365 6420  ters data based 
-00010070: 6f6e 202a 4b2a 3c73 7562 3e53 3c2f 7375  on *K*<sub>S</su
-00010080: 623e 2076 616c 7565 7320 616e 6420 7468  b> values and th
-00010090: 6520 7061 7261 6d65 7465 7220 602d 2d62  e parameter `--b
-000100a0: 696e 7360 2077 6869 6368 2073 6574 7320  ins` which sets 
-000100b0: 7468 6520 6e75 6d62 6572 206f 6620 6269  the number of bi
-000100c0: 6e73 2069 6e20 2a4b 2a3c 7375 623e 533c  ns in *K*<sub>S<
-000100d0: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-000100e0: 6f6e 2028 6465 6661 756c 7420 3530 292e  on (default 50).
-000100f0: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
-00010100: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
-00010110: 2075 7365 2063 6f6d 6d61 6e64 2073 696d   use command sim
-00010120: 706c 7920 6173 2062 656c 6f77 2a2a 0a0a  ply as below**..
-00010130: 6060 600a 7767 6420 6d69 7820 6b73 6461  ```.wgd mix ksda
-00010140: 7461 0a60 6060 0a0a 2323 2320 7767 6420  ta.```..### wgd 
-00010150: 7065 616b 0a0a 2a2a 5468 6520 7365 6172  peak..**The sear
-00010160: 6368 206f 6620 6372 6564 6961 626c 6520  ch of crediable 
-00010170: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00010180: 7261 6e67 6520 7573 6564 2069 6e20 5747  range used in WG
-00010190: 4420 6461 7469 6e67 2a2a 0a60 6060 0a77  D dating**.```.w
-000101a0: 6764 2070 6561 6b20 6b73 6461 7461 202d  gd peak ksdata -
-000101b0: 6170 2061 7064 6174 6120 2d73 6d20 736d  ap apdata -sm sm
-000101c0: 6461 7461 202d 6c65 206c 6564 6174 6120  data -le ledata 
-000101d0: 2d6d 7020 6d70 6461 7461 202d 2d68 6575  -mp mpdata --heu
-000101e0: 7269 7374 6963 2028 2d2d 616c 6967 6e66  ristic (--alignf
-000101f0: 696c 7465 7220 302e 3020 3020 302e 3020  ilter 0.0 0 0.0 
-00010200: 2d2d 6b73 7261 6e67 6520 3020 3520 2d2d  --ksrange 0 5 --
-00010210: 6269 6e5f 7769 6474 6820 302e 3120 2d2d  bin_width 0.1 --
-00010220: 6775 6964 6520 7365 676d 656e 7420 2d2d  guide segment --
-00010230: 7072 6f6d 696e 656e 6365 5f63 7574 6f66  prominence_cutof
-00010240: 6620 302e 3120 2d2d 7265 6c5f 6865 6967  f 0.1 --rel_heig
-00010250: 6874 2030 2e34 202d 2d63 6920 3935 202d  ht 0.4 --ci 95 -
-00010260: 2d68 6472 2039 3520 2d2d 6b73 6375 746f  -hdr 95 --kscuto
-00010270: 6666 2035 290a 6060 600a 0a41 7320 6d65  ff 5).```..As me
-00010280: 6e74 696f 6e65 6420 7072 6576 696f 7573  ntioned previous
-00010290: 6c79 2c20 6120 6865 7572 6973 7469 6320  ly, a heuristic 
-000102a0: 6d65 7468 6f64 2061 6e64 2061 2063 6f6c  method and a col
-000102b0: 6c69 6e65 6172 2073 6567 6d65 6e74 732d  linear segments-
-000102c0: 6775 6964 6564 2061 6e63 686f 7220 7061  guided anchor pa
-000102d0: 6972 2063 6c75 7374 6572 696e 6720 666f  ir clustering fo
-000102e0: 7220 7468 6520 7365 6172 6368 206f 6620  r the search of 
-000102f0: 6372 6564 6961 626c 6520 2a4b 2a3c 7375  crediable *K*<su
-00010300: 623e 533c 2f73 7562 3e20 7261 6e67 6520  b>S</sub> range 
-00010310: 7573 6564 2069 6e20 5747 4420 6461 7469  used in WGD dati
-00010320: 6e67 2061 7265 2069 6d70 6c65 6d65 6e74  ng are implement
-00010330: 6564 2069 6e20 6077 6764 2076 3260 2e20  ed in `wgd v2`. 
-00010340: 5573 6572 7320 6e65 6564 2074 6f20 7072  Users need to pr
-00010350: 6f76 6964 6520 7468 6520 616e 6368 6f72  ovide the anchor
-00010360: 2070 6f69 6e74 732c 2073 6567 6d65 6e74   points, segment
-00010370: 732c 206c 6973 7473 6567 6d65 6e74 732c  s, listsegments,
-00010380: 206d 756c 7469 706c 6963 6f6e 7061 6972   multipliconpair
-00010390: 7320 6461 7461 6669 6c65 2066 726f 6d20  s datafile from 
-000103a0: 6069 2d61 6468 6f72 6560 2074 6f20 6163  `i-adhore` to ac
-000103b0: 6869 6576 6520 7468 6520 636c 7573 7465  hieve the cluste
-000103c0: 7269 6e67 2066 756e 6374 696f 6e2e 2053  ring function. S
-000103d0: 6f6d 6520 7061 7261 6d65 7465 7273 2074  ome parameters t
-000103e0: 6861 7420 6361 6e20 696d 7061 6374 2074  hat can impact t
-000103f0: 6865 2072 6573 756c 7473 2069 6e63 6c75  he results inclu
-00010400: 6465 2060 2d2d 616c 6967 6e66 696c 7465  de `--alignfilte
-00010410: 7260 2c20 7768 6963 6820 6669 6c74 6572  r`, which filter
-00010420: 7320 7468 6520 6461 7461 2062 6173 6564  s the data based
-00010430: 206f 6e20 616c 6967 6e6d 656e 7420 6964   on alignment id
-00010440: 656e 7469 7479 2c20 6c65 6e67 7468 2061  entity, length a
-00010450: 6e64 2063 6f76 6572 6167 652c 2060 2d2d  nd coverage, `--
-00010460: 6b73 7261 6e67 6560 2c20 7768 6963 6820  ksrange`, which 
-00010470: 7365 7473 2074 6865 2072 616e 6765 206f  sets the range o
-00010480: 6620 4b73 2074 6f20 6265 2061 6e61 6c79  f Ks to be analy
-00010490: 7a65 642c 2060 2d2d 6269 6e5f 7769 6474  zed, `--bin_widt
-000104a0: 6860 2c20 7768 6963 6820 7365 7473 2074  h`, which sets t
-000104b0: 6865 2062 616e 6477 6964 7468 206f 6620  he bandwidth of 
-000104c0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-000104d0: 6469 7374 7269 6275 7469 6f6e 2c20 602d  distribution, `-
-000104e0: 2d77 6569 6768 7473 5f6f 7574 6c69 6572  -weights_outlier
-000104f0: 735f 696e 636c 7564 6564 6020 7768 6963  s_included` whic
-00010500: 6820 6465 7465 726d 696e 6573 2077 6865  h determines whe
-00010510: 7468 6572 2074 6f20 696e 636c 7564 6520  ther to include 
-00010520: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00010530: 6f75 746c 6965 7273 2028 7768 6f73 6520  outliers (whose 
-00010540: 7661 6c75 6520 6973 206f 7665 7220 3529  value is over 5)
-00010550: 2069 6e20 616e 616c 7973 6973 2c20 602d   in analysis, `-
-00010560: 2d6d 6574 686f 6460 2077 6869 6368 2064  -method` which d
-00010570: 6574 6572 6d69 6e65 7320 7768 6963 6820  etermines which 
-00010580: 636c 7573 7465 7269 6e67 206d 6574 686f  clustering metho
-00010590: 6420 746f 2075 7365 2028 6465 6661 756c  d to use (defaul
-000105a0: 7420 676d 6d29 2c20 602d 2d73 6565 6460  t gmm), `--seed`
-000105b0: 2077 6869 6368 2073 6574 7320 7468 6520   which sets the 
-000105c0: 7261 6e64 6f6d 2073 6565 6420 6769 7665  random seed give
-000105d0: 6e20 746f 2069 6e69 7469 616c 697a 6174  n to initializat
-000105e0: 696f 6e20 2864 6566 6175 6c74 2032 3335  ion (default 235
-000105f0: 3238 3930 292c 2060 2d2d 6e5f 696e 6974  2890), `--n_init
-00010600: 602c 2077 6869 6368 2073 6574 7320 7468  `, which sets th
-00010610: 6520 6e75 6d62 6572 206f 6620 6b2d 6d65  e number of k-me
-00010620: 616e 7320 696e 6974 6961 6c69 7a61 7469  ans initializati
-00010630: 6f6e 7320 2864 6566 6175 6c74 2032 3030  ons (default 200
-00010640: 292c 2060 2d2d 656d 5f69 7465 7260 2c20  ), `--em_iter`, 
-00010650: 7768 6963 6820 7365 7473 2074 6865 206d  which sets the m
-00010660: 6178 696d 756d 206e 756d 6265 7220 6f66  aximum number of
-00010670: 2069 7465 7261 7469 6f6e 7320 2864 6566   iterations (def
-00010680: 6175 6c74 2032 3030 292c 2060 2d2d 6761  ault 200), `--ga
-00010690: 6d6d 6160 2c20 7768 6963 6820 7365 7473  mma`, which sets
-000106a0: 2074 6865 2067 616d 6d61 2070 6172 616d   the gamma param
-000106b0: 6574 6572 2066 6f72 2074 6865 2062 676d  eter for the bgm
-000106c0: 6d20 6d6f 6465 6c20 2864 6566 6175 6c74  m model (default
-000106d0: 2030 2e30 3031 292c 2060 2d2d 636f 6d70   0.001), `--comp
-000106e0: 6f6e 656e 7473 602c 2077 6869 6368 2073  onents`, which s
-000106f0: 6574 7320 7468 6520 7261 6e67 6520 6f66  ets the range of
-00010700: 2074 6865 206e 756d 6265 7220 6f66 2063   the number of c
-00010710: 6f6d 706f 6e65 6e74 7320 746f 2066 6974  omponents to fit
-00010720: 2028 6465 6661 756c 7420 3120 3429 2c20   (default 1 4), 
-00010730: 602d 2d77 6569 6768 7465 6460 2077 6869  `--weighted` whi
-00010740: 6368 2064 6574 6572 6d69 6e65 7320 7768  ch determines wh
-00010750: 6574 6865 7220 746f 2075 7365 206e 6f64  ether to use nod
-00010760: 652d 7765 6967 6874 6564 206d 6574 686f  e-weighted metho
-00010770: 6420 666f 7220 6465 2d72 6564 756e 6461  d for de-redunda
-00010780: 6e63 792c 2060 2d2d 6775 6964 6560 2077  ncy, `--guide` w
-00010790: 6869 6368 2064 6574 6572 6d69 6e65 7320  hich determines 
-000107a0: 7768 6963 6820 7265 6769 6d65 2072 6573  which regime res
-000107b0: 6964 696e 6720 616e 6368 6f72 7320 746f  iding anchors to
-000107c0: 2062 6520 7573 6564 2028 6465 6661 756c   be used (defaul
-000107d0: 7420 5365 676d 656e 7429 2c20 602d 2d70  t Segment), `--p
-000107e0: 726f 6d69 6e65 6e63 655f 6375 746f 6666  rominence_cutoff
-000107f0: 6020 7768 6963 6820 7365 7473 2074 6865  ` which sets the
-00010800: 2070 726f 6d69 6e65 6e63 6520 6375 746f   prominence cuto
-00010810: 6666 206f 6620 6163 6365 7074 6162 6c65  ff of acceptable
-00010820: 2070 6561 6b73 2069 6e20 7065 616b 2066   peaks in peak f
-00010830: 696e 6469 6e67 2070 726f 6365 7373 2c20  inding process, 
-00010840: 602d 2d72 656c 5f68 6569 6768 7460 2077  `--rel_height` w
-00010850: 6869 6368 2073 6574 7320 7468 6520 7265  hich sets the re
-00010860: 6c61 7469 7665 2068 6569 6768 7420 6174  lative height at
-00010870: 2077 6869 6368 2074 6865 2070 6561 6b20   which the peak 
-00010880: 7769 6474 6820 6973 206d 6561 7375 7265  width is measure
-00010890: 642c 2060 2d2d 6b73 746f 6461 7465 6020  d, `--kstodate` 
-000108a0: 7768 6963 6820 6d61 6e75 616c 6c79 2073  which manually s
-000108b0: 6574 7320 7468 6520 7261 6e67 6520 6f66  ets the range of
-000108c0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000108d0: 2074 6f20 6265 2064 6174 6564 2069 6e20   to be dated in 
-000108e0: 6865 7572 6973 7469 6320 7365 6172 6368  heuristic search
-000108f0: 2061 6e64 206e 6565 6473 2074 6f20 6265   and needs to be
-00010900: 2063 6f2d 7365 7420 7769 7468 206f 7074   co-set with opt
-00010910: 696f 6e20 602d 2d6d 616e 7561 6c73 6574  ion `--manualset
-00010920: 602c 2060 2d2d 786c 696d 6020 616e 6420  `, `--xlim` and 
-00010930: 602d 2d79 6c69 6d60 2064 6574 6572 6d69  `--ylim` determi
-00010940: 6e69 6e67 2074 6865 2078 2061 6e64 2079  ning the x and y
-00010950: 2061 7869 7320 6c69 6d69 7420 6f66 2047   axis limit of G
-00010960: 4d4d 202a 4b2a 3c73 7562 3e53 3c2f 7375  MM *K*<sub>S</su
-00010970: 623e 2064 6973 7472 6962 7574 696f 6e2c  b> distribution,
-00010980: 2060 2d2d 6369 6020 7365 7474 696e 6720   `--ci` setting 
-00010990: 7468 6520 636f 6e66 6964 656e 6365 206c  the confidence l
-000109a0: 6576 656c 206f 6620 6c6f 672d 6e6f 726d  evel of log-norm
-000109b0: 616c 2064 6973 7472 6962 7574 696f 6e20  al distribution 
-000109c0: 746f 2064 6174 6520 2864 6566 6175 6c74  to date (default
-000109d0: 2039 3529 2c20 602d 2d68 6472 6020 7365   95), `--hdr` se
-000109e0: 7474 696e 6720 7468 6520 6869 6768 6573  tting the highes
-000109f0: 7420 6465 6e73 6974 7920 7265 6769 6f6e  t density region
-00010a00: 2028 4844 5229 2061 7070 6c69 6564 2069   (HDR) applied i
-00010a10: 6e20 7468 6520 7365 676d 656e 742d 6775  n the segment-gu
-00010a20: 6964 6564 2061 6e63 686f 7220 7061 6972  ided anchor pair
-00010a30: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00010a40: 2064 6973 7472 6962 7574 696f 6e20 2864   distribution (d
-00010a50: 6566 6175 6c74 2039 3529 2c20 602d 2d68  efault 95), `--h
-00010a60: 6575 7269 7374 6963 6020 6465 7465 726d  euristic` determ
-00010a70: 696e 696e 6720 7768 6574 6865 7220 746f  ining whether to
-00010a80: 2069 6e69 7469 6174 6520 6865 7572 6973   initiate heuris
-00010a90: 7469 6320 6d65 7468 6f64 206f 6620 6465  tic method of de
-00010aa0: 6669 6e69 6e67 2043 4920 666f 7220 6461  fining CI for da
-00010ab0: 7469 6e67 2c20 602d 2d6b 7363 7574 6f66  ting, `--kscutof
-00010ac0: 6660 2073 6574 7469 6e67 2074 6865 202a  f` setting the *
-00010ad0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2073  K*<sub>S</sub> s
-00010ae0: 6174 7572 6174 696f 6e20 6375 746f 6666  aturation cutoff
-00010af0: 2069 6e20 6461 7469 6e67 2028 6465 6661   in dating (defa
-00010b00: 756c 7420 3529 2e0a 0a46 6f75 7220 7265  ult 5)...Four re
-00010b10: 7375 6c74 2073 7562 666c 6f64 6572 7320  sult subfloders 
-00010b20: 7769 6c6c 2062 6520 7072 6f64 7563 6564  will be produced
-00010b30: 2c20 6e61 6d65 6c79 2060 416e 6368 6f72  , namely `Anchor
-00010b40: 4b73 5f46 696e 6450 6561 6b60 2c20 6041  Ks_FindPeak`, `A
-00010b50: 6e63 686f 724b 735f 474d 4d60 2c20 6053  nchorKs_GMM`, `S
-00010b60: 6567 6d65 6e74 4775 6964 654b 735f 474d  egmentGuideKs_GM
-00010b70: 4d60 2061 6e64 2060 5365 676d 656e 744b  M` and `SegmentK
-00010b80: 735f 4669 6e64 5065 616b 602e 2054 6865  s_FindPeak`. The
-00010b90: 2060 416e 6368 6f72 4b73 5f46 696e 6450   `AnchorKs_FindP
-00010ba0: 6561 6b60 2073 7562 666c 6f64 6572 2063  eak` subfloder c
-00010bb0: 6f6e 7461 696e 7320 7265 7375 6c74 7320  ontains results 
-00010bc0: 6f66 2074 6865 2064 6574 6563 7465 6420  of the detected 
-00010bd0: 7065 616b 7320 6279 2074 6865 2060 7369  peaks by the `si
-00010be0: 676e 616c 6020 6d6f 6475 6c65 206f 6620  gnal` module of 
-00010bf0: 6053 6369 5079 6020 6c69 6272 6172 7920  `SciPy` library 
-00010c00: 616e 6420 7468 6520 6173 7375 6d65 6420  and the assumed 
-00010c10: 6869 6768 6573 7420 6d61 7373 2070 6172  highest mass par
-00010c20: 7420 2872 6566 6572 7265 6420 746f 2061  t (referred to a
-00010c30: 7320 4869 6768 4d61 7373 2068 6572 6561  s HighMass herea
-00010c40: 6674 6572 2920 6f66 2065 6163 6820 7065  fter) of each pe
-00010c50: 616b 2c20 7768 6963 6820 6361 6e20 6265  ak, which can be
-00010c60: 2075 7365 6420 666f 7220 6675 7274 6865   used for furthe
-00010c70: 7220 5747 4420 6461 7469 6e67 2e20 5468  r WGD dating. Th
-00010c80: 6520 6041 6e63 686f 724b 735f 474d 4d60  e `AnchorKs_GMM`
-00010c90: 2073 686f 7773 2074 6865 2047 4d4d 2072   shows the GMM r
-00010ca0: 6573 756c 7473 2075 706f 6e20 7468 6520  esults upon the 
-00010cb0: 6f72 6967 696e 616c 2061 6e63 686f 7220  original anchor 
-00010cc0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00010cd0: 6469 7374 7269 6275 7469 6f6e 2062 7920  distribution by 
-00010ce0: 7468 6520 606d 6978 7475 7265 6020 6d6f  the `mixture` mo
-00010cf0: 6475 6c65 206f 6620 6073 6369 6b69 742d  dule of `scikit-
-00010d00: 6c65 6172 6e60 206c 6962 7261 7279 2061  learn` library a
-00010d10: 6e64 2074 776f 2073 7562 666c 6f64 6572  nd two subfloder
-00010d20: 732c 2060 4c6f 6747 4d4d 5f43 4960 2063  s, `LogGMM_CI` c
-00010d30: 6f6e 7461 696e 696e 6720 7468 6520 7265  ontaining the re
-00010d40: 7375 6c74 7320 6f66 2039 3525 2043 4920  sults of 95% CI 
-00010d50: 6f66 2065 6163 6820 636f 6d70 6f6e 656e  of each componen
-00010d60: 742c 2060 4869 6768 4d61 7373 5f43 4960  t, `HighMass_CI`
-00010d70: 2063 6f6e 7461 696e 696e 6720 7468 6520   containing the 
-00010d80: 4869 6768 4d61 7373 206f 6620 6561 6368  HighMass of each
-00010d90: 2063 6f6d 706f 6e65 6e74 2c20 7768 6963   component, whic
-00010da0: 6820 6361 6e20 6265 2075 7365 6420 666f  h can be used fo
-00010db0: 7220 6675 7274 6865 7220 5747 4420 6461  r further WGD da
-00010dc0: 7469 6e67 2e20 5468 6520 6053 6567 6d65  ting. The `Segme
-00010dd0: 6e74 4775 6964 654b 735f 474d 4d60 2073  ntGuideKs_GMM` s
-00010de0: 7562 666f 6c64 6572 2070 7265 7365 6e74  ubfolder present
-00010df0: 7320 7265 7375 6c74 7320 6f66 2073 6567  s results of seg
-00010e00: 6d65 6e74 202a 4b2a 3c73 7562 3e53 3c2f  ment *K*<sub>S</
-00010e10: 7375 623e 2047 4d4d 2077 6869 6368 2061  sub> GMM which a
-00010e20: 7265 206d 6170 7065 6420 6261 636b 2074  re mapped back t
-00010e30: 6f20 7468 6520 7265 7369 6469 6e67 2061  o the residing a
-00010e40: 6e63 686f 7220 7061 6972 7320 616e 6420  nchor pairs and 
-00010e50: 7468 6520 6173 736f 6369 6174 6564 2039  the associated 9
-00010e60: 3525 2048 4452 2061 6e64 2048 6967 684d  5% HDR and HighM
-00010e70: 6173 7320 6f66 2065 6163 6820 7365 676d  ass of each segm
-00010e80: 656e 7420 636c 7573 7465 7220 696e 2073  ent cluster in s
-00010e90: 7562 666c 6f64 6572 7320 6f66 2060 4844  ubfloders of `HD
-00010ea0: 525f 4349 6020 616e 6420 6048 6967 684d  R_CI` and `HighM
-00010eb0: 6173 735f 4349 602e 2054 6865 2060 5365  ass_CI`. The `Se
-00010ec0: 676d 656e 744b 735f 4669 6e64 5065 616b  gmentKs_FindPeak
-00010ed0: 6020 7375 6266 6f6c 6465 7220 6973 2073  ` subfolder is s
-00010ee0: 696d 696c 6172 2074 6f20 6041 6e63 686f  imilar to `Ancho
-00010ef0: 724b 735f 4669 6e64 5065 616b 6020 6275  rKs_FindPeak` bu
-00010f00: 7420 7769 7468 2073 6567 6d65 6e74 202a  t with segment *
-00010f10: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2069  K*<sub>S</sub> i
-00010f20: 6e73 7465 6164 2e20 5468 6520 2a4b 2a3c  nstead. The *K*<
-00010f30: 7375 623e 533c 2f73 7562 3e20 696e 2060  sub>S</sub> in `
-00010f40: 4d75 6c74 6970 6c69 636f 6e60 2063 616e  Multiplicon` can
-00010f50: 2061 6c73 6f20 6265 2063 616c 6375 6c61   also be calcula
-00010f60: 7465 6420 696e 2070 6c61 6365 206f 6620  ted in place of 
-00010f70: 6053 6567 6d65 6e74 6020 7573 696e 6720  `Segment` using 
-00010f80: 7468 6520 6f70 7469 6f6e 2060 2d2d 6775  the option `--gu
-00010f90: 6964 6560 2061 7320 7375 6368 2074 6865  ide` as such the
-00010fa0: 2072 6573 756c 7420 7469 746c 652c 206c   result title, l
-00010fb0: 6162 656c 2c20 6669 6c65 2061 6e64 2066  abel, file and f
-00010fc0: 6f6c 6465 7220 6e61 6d65 7320 7769 6c6c  older names will
-00010fd0: 2062 6520 6368 616e 6765 6420 6163 636f   be changed acco
-00010fe0: 7264 696e 676c 792e 0a0a 2a2a 4120 7375  rdingly...**A su
-00010ff0: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
-00011000: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
-00011010: 6d61 6e64 2073 696d 706c 7920 6173 2062  mand simply as b
-00011020: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
-00011030: 7065 616b 206b 7364 6174 6120 2d61 7020  peak ksdata -ap 
-00011040: 6170 6461 7461 202d 736d 2073 6d64 6174  apdata -sm smdat
-00011050: 6120 2d6c 6520 6c65 6461 7461 202d 6d70  a -le ledata -mp
-00011060: 206d 7064 6174 6120 2d2d 6865 7572 6973   mpdata --heuris
-00011070: 7469 630a 6060 600a 0a23 2323 2077 6764  tic.```..### wgd
-00011080: 2073 796e 0a0a 2a2a 5468 6520 696e 7472   syn..**The intr
-00011090: 612d 7370 6563 6966 6963 2073 796e 7465  a-specific synte
-000110a0: 6e79 2069 6e66 6572 656e 6365 2a2a 0a60  ny inference**.`
-000110b0: 6060 0a77 6764 2073 796e 2066 616d 696c  ``.wgd syn famil
-000110c0: 6965 7320 6766 6620 282d 2d6b 735f 6469  ies gff (--ks_di
-000110d0: 7374 7269 6275 7469 6f6e 206b 7364 6174  stribution ksdat
-000110e0: 6120 2d66 2067 656e 6520 2d61 2049 4420  a -f gene -a ID 
-000110f0: 2d2d 6d69 6e6c 656e 202d 3120 2d2d 6d69  --minlen -1 --mi
-00011100: 6e73 6567 6c65 6e20 3130 3030 3020 2d2d  nseglen 10000 --
-00011110: 6d69 6e67 656e 656e 756d 2033 3029 0a60  mingenenum 30).`
-00011120: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
-00011130: 6077 6764 2073 796e 6020 6973 206d 6169  `wgd syn` is mai
-00011140: 6e6c 7920 6465 616c 696e 6720 7769 7468  nly dealing with
-00011150: 2063 6f6c 6c69 6e65 6172 6974 7920 6f72   collinearity or
-00011160: 2073 796e 7465 6e79 2028 626f 7468 2072   synteny (both r
-00011170: 6566 6572 7265 6420 746f 2061 7320 7379  eferred to as sy
-00011180: 6e74 656e 7920 6865 7265 6166 7465 7229  nteny hereafter)
-00011190: 2061 6e61 6c79 7369 732e 2054 776f 2069   analysis. Two i
-000111a0: 6e70 7574 2066 696c 6573 2061 7265 2065  nput files are e
-000111b0: 7373 656e 7469 616c 2c20 7468 6520 6765  ssential, the ge
-000111c0: 6e65 2066 616d 696c 7920 6669 6c65 2061  ne family file a
-000111d0: 6e64 2074 6865 2067 6666 3320 6669 6c65  nd the gff3 file
-000111e0: 2e20 5468 6520 6765 6e65 2066 616d 696c  . The gene famil
-000111f0: 7920 6669 6c65 2069 7320 696e 2074 6865  y file is in the
-00011200: 2066 6f72 6d61 7420 6173 2060 4f72 7468   format as `Orth
-00011210: 6f46 696e 6465 7260 2e20 5468 6520 736f  oFinder`. The so
-00011220: 6674 7761 7265 2060 692d 6164 686f 7265  ftware `i-adhore
-00011230: 6020 6973 2061 2070 7265 7265 7175 6973  ` is a prerequis
-00011240: 6974 652e 2057 6974 6820 6465 6661 756c  ite. With defaul
-00011250: 7420 7061 7261 6d65 7465 7273 2c20 7468  t parameters, th
-00011260: 6520 7072 6f67 7261 6d20 6261 7369 6361  e program basica
-00011270: 6c6c 7920 636f 6e64 7563 7473 2031 2920  lly conducts 1) 
-00011280: 6669 6c74 6572 696e 6720 6765 6e65 2066  filtering gene f
-00011290: 616d 696c 6965 7320 6261 7365 6420 6f6e  amilies based on
-000112a0: 206d 6178 696d 756d 2066 616d 696c 7920   maximum family 
-000112b0: 7369 7a65 2032 2920 7265 7472 6965 7669  size 2) retrievi
-000112c0: 6e67 2067 656e 6520 706f 7369 7469 6f6e  ng gene position
-000112d0: 2061 6e64 2073 6361 6666 6f6c 6420 696e   and scaffold in
-000112e0: 666f 726d 6174 696f 6e20 6672 6f6d 2067  formation from g
-000112f0: 6666 3320 6669 6c65 2033 2920 7072 6f64  ff3 file 3) prod
-00011300: 7563 696e 6720 7468 6520 636f 6e66 6967  ucing the config
-00011310: 7572 6174 696f 6e20 6669 6c65 2061 6e64  uration file and
-00011320: 2061 7373 6f63 6961 7465 6420 6461 7461   associated data
-00011330: 6669 6c65 7320 666f 7220 6069 2d61 6468  files for `i-adh
-00011340: 6f72 6560 2034 2920 6361 6c6c 696e 6720  ore` 4) calling 
-00011350: 6069 2d61 6468 6f72 6560 2067 6976 656e  `i-adhore` given
-00011360: 2074 6865 2070 6172 616d 6574 6572 7320   the parameters 
-00011370: 7365 7420 746f 2069 6e66 6572 2073 796e  set to infer syn
-00011380: 7465 6e79 2035 2920 7669 7375 616c 697a  teny 5) visualiz
-00011390: 696e 6720 7468 6520 7379 6e74 656e 7920  ing the synteny 
-000113a0: 696e 2022 646f 7470 6c6f 7422 2069 6e20  in "dotplot" in 
-000113b0: 7468 6520 756e 6974 206f 6620 6765 6e65  the unit of gene
-000113c0: 7320 616e 6420 6261 7365 732c 2069 6e20  s and bases, in 
-000113d0: 2253 796e 6465 7074 6822 2070 6c6f 7420  "Syndepth" plot 
-000113e0: 7368 6f77 696e 6720 7468 6520 6469 7374  showing the dist
-000113f0: 7269 6275 7469 6f6e 206f 6620 6469 6666  ribution of diff
-00011400: 6572 656e 7420 6361 7465 676f 7269 6573  erent categories
-00011410: 206f 6620 636f 6c6c 696e 6561 7269 7479   of collinearity
-00011420: 2072 6174 696f 7320 7769 7468 696e 2061   ratios within a
-00011430: 6e64 2062 6574 7765 656e 2073 7065 6369  nd between speci
-00011440: 6573 2c20 696e 2022 6475 7053 7461 636b  es, in "dupStack
-00011450: 2220 706c 6f74 2073 686f 7769 6e67 206d  " plot showing m
-00011460: 756c 7469 706c 6963 6f6e 7320 7769 7468  ultiplicons with
-00011470: 2064 6966 6665 7265 6e74 206d 756c 7469   different multi
-00011480: 706c 6963 6174 696f 6e20 6c65 7665 6c73  plication levels
-00011490: 2e20 3629 2069 6620 7769 7468 202a 4b2a  . 6) if with *K*
-000114a0: 3c73 7562 3e53 3c2f 7375 623e 2064 6174  <sub>S</sub> dat
-000114b0: 612c 2061 2022 2a4b 2a3c 7375 623e 533c  a, a "*K*<sub>S<
-000114c0: 2f73 7562 3e20 646f 7470 6c6f 7422 2077  /sub> dotplot" w
-000114d0: 6974 6820 646f 7473 2061 6e6e 6f74 6174  ith dots annotat
-000114e0: 6564 2069 6e20 2a4b 2a3c 7375 623e 533c  ed in *K*<sub>S<
-000114f0: 2f73 7562 3e20 7661 6c75 6573 2061 6e64  /sub> values and
-00011500: 2061 202a 4b2a 3c73 7562 3e53 3c2f 7375   a *K*<sub>S</su
-00011510: 623e 2064 6973 7472 6962 7574 696f 6e20  b> distribution 
-00011520: 7769 7468 2061 6e63 686f 7220 7061 6972  with anchor pair
-00011530: 7320 6465 6e6f 7465 6420 7769 6c6c 2062  s denoted will b
-00011540: 6520 7072 6f64 7563 6564 2e20 5468 6520  e produced. The 
-00011550: 6765 6e65 2069 6e66 6f72 6d61 7469 6f6e  gene information
-00011560: 2069 6e20 7468 6520 6765 6e65 2066 616d   in the gene fam
-00011570: 696c 7920 6669 6c65 2061 6e64 2067 6666  ily file and gff
-00011580: 3320 6669 6c65 2073 686f 756c 6420 6265  3 file should be
-00011590: 206d 6174 6368 6564 2077 6869 6368 2072   matched which r
-000115a0: 6571 7569 7265 7320 7573 6572 7320 746f  equires users to
-000115b0: 2073 6574 2070 726f 7065 7220 602d 2d66   set proper `--f
-000115c0: 6561 7475 7265 6020 616e 6420 602d 2d61  eature` and `--a
-000115d0: 7474 7269 6275 7465 602e 2054 6865 206d  ttribute`. The m
-000115e0: 6178 696d 756d 2066 616d 696c 7920 7369  aximum family si
-000115f0: 7a65 2074 6f20 6265 2069 6e63 6c75 6465  ze to be include
-00011600: 6420 6361 6e20 6265 2073 6574 2076 6961  d can be set via
-00011610: 2074 6865 206f 7074 696f 6e20 602d 2d6d   the option `--m
-00011620: 6178 7369 7a65 602c 206e 6f74 6564 2074  axsize`, noted t
-00011630: 6861 7420 7468 6973 2066 696c 7465 7269  hat this filteri
-00011640: 6e67 2069 7320 6d61 696e 6c79 2074 6f20  ng is mainly to 
-00011650: 6472 6f70 2074 686f 7365 2068 7567 6520  drop those huge 
-00011660: 7461 6e64 656d 2064 7570 6c69 6361 7465  tandem duplicate
-00011670: 7320 6661 6d69 6c79 2061 6e64 2074 7261  s family and tra
-00011680: 6e73 706f 7361 626c 6520 656c 656d 656e  nsposable elemen
-00011690: 7473 2028 5445 7329 2066 616d 696c 792c  ts (TEs) family,
-000116a0: 2061 6e64 206e 6f74 206d 616e 6461 746f   and not mandato
-000116b0: 7279 2e20 5573 6572 7320 6361 6e20 6669  ry. Users can fi
-000116c0: 6c74 6572 2074 686f 7365 2066 7261 676d  lter those fragm
-000116d0: 656e 7461 7279 2073 6361 6666 6f6c 6473  entary scaffolds
-000116e0: 2076 6961 2074 6865 206f 7074 696f 6e20   via the option 
-000116f0: 602d 2d6d 696e 6c65 6e60 2e20 5468 6520  `--minlen`. The 
-00011700: 6d69 6e69 6d75 6d20 6c65 6e67 7468 2061  minimum length a
-00011710: 6e64 206e 756d 6265 7220 6f66 2067 656e  nd number of gen
-00011720: 6573 2066 6f72 2061 2073 6567 6d65 6e74  es for a segment
-00011730: 2074 6f20 6265 2063 6f6e 7369 6465 7265   to be considere
-00011740: 6420 6361 6e20 6265 2073 6574 2076 6961  d can be set via
-00011750: 2074 6865 206f 7074 696f 6e20 602d 2d6d   the option `--m
-00011760: 696e 7365 676c 656e 6020 616e 6420 602d  inseglen` and `-
-00011770: 2d6d 696e 6765 6e65 6e75 6d60 2e20 5265  -mingenenum`. Re
-00011780: 6475 6e64 616e 7420 6d75 6c74 6970 6c69  dundant multipli
-00011790: 636f 6e73 2063 616e 2062 6520 6b65 7074  cons can be kept
-000117a0: 2062 7920 7365 7420 7468 6520 666c 6167   by set the flag
-000117b0: 206f 7074 696f 6e20 602d 2d6b 6565 7072   option `--keepr
-000117c0: 6564 756e 602e 0a0a 2a2a 4120 7375 6767  edun`...**A sugg
-000117d0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-000117e0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-000117f0: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
-00011800: 6f77 2a2a 0a0a 6060 600a 7767 6420 7379  ow**..```.wgd sy
-00011810: 6e20 6661 6d69 6c69 6573 2067 6666 0a60  n families gff.`
-00011820: 6060 0a0a 2a2a 5468 6520 696e 7465 722d  ``..**The inter-
-00011830: 7370 6563 6966 6963 2073 796e 7465 6e79  specific synteny
-00011840: 2069 6e66 6572 656e 6365 2a2a 0a60 6060   inference**.```
-00011850: 0a77 6764 2073 796e 2066 616d 696c 6965  .wgd syn familie
-00011860: 7320 6766 6631 2067 6666 3220 282d 2d61  s gff1 gff2 (--a
-00011870: 6464 6974 696f 6e61 6c67 6666 696e 666f  dditionalgffinfo
-00011880: 2022 6d52 4e41 3b4e 616d 6522 202d 2d61   "mRNA;Name" --a
-00011890: 6464 6974 696f 6e61 6c67 6666 696e 666f  dditionalgffinfo
-000118a0: 2022 6765 6e65 3b49 4422 290a 6060 600a   "gene;ID").```.
-000118b0: 0a46 6f72 206d 756c 7469 2d73 7065 6369  .For multi-speci
-000118c0: 6573 2073 796e 7465 6e79 2069 6e66 6572  es synteny infer
-000118d0: 656e 6365 2c20 6966 2075 7365 7273 2068  ence, if users h
-000118e0: 6176 6520 6766 6633 2066 696c 6573 2077  ave gff3 files w
-000118f0: 6869 6368 2068 6176 6520 6469 6666 6572  hich have differ
-00011900: 656e 7420 6665 6174 7572 6573 206f 7220  ent features or 
-00011910: 6174 7472 6962 7574 6573 2066 6f72 2067  attributes for g
-00011920: 656e 6520 706f 7369 7469 6f6e 2069 6e66  ene position inf
-00011930: 6f72 6d61 7469 6f6e 2072 6574 7269 6576  ormation retriev
-00011940: 616c 2c20 7468 6520 6f70 7469 6f6e 2060  al, the option `
-00011950: 2d2d 6164 6469 7469 6f6e 616c 6766 6669  --additionalgffi
-00011960: 6e66 6f60 2063 616e 2062 6520 7365 7420  nfo` can be set 
-00011970: 746f 2070 726f 7669 6465 2074 6865 2061  to provide the a
-00011980: 6464 6974 696f 6e61 6c20 696e 666f 726d  dditional inform
-00011990: 6174 696f 6e2e 2054 6865 2072 656d 6169  ation. The remai
-000119a0: 6e69 6e67 2070 6172 616d 6574 6572 2073  ning parameter s
-000119b0: 6574 7469 6e67 2069 7320 7468 6520 7361  etting is the sa
-000119c0: 6d65 2061 7320 7468 6520 696e 7472 612d  me as the intra-
-000119d0: 7370 6563 6966 6963 2073 796e 7465 6e79  specific synteny
-000119e0: 2069 6e66 6572 656e 6365 2e0a 0a2a 2a41   inference...**A
-000119f0: 2073 7567 6765 7374 6564 2073 7461 7274   suggested start
-00011a00: 696e 6720 7275 6e20 6361 6e20 7573 6520  ing run can use 
-00011a10: 636f 6d6d 616e 6420 7369 6d70 6c79 2061  command simply a
-00011a20: 7320 6265 6c6f 772a 2a0a 0a60 6060 0a77  s below**..```.w
-00011a30: 6764 2073 796e 2066 616d 696c 6965 7320  gd syn families 
-00011a40: 6766 6631 2067 6666 320a 6060 600a 0a23  gff1 gff2.```..#
-00011a50: 2323 2077 6764 2076 697a 0a0a 2a2a 5468  ## wgd viz..**Th
-00011a60: 6520 7669 7375 616c 697a 6174 696f 6e20  e visualization 
-00011a70: 6f66 202a 4b2a 3c73 7562 3e53 3c2f 7375  of *K*<sub>S</su
-00011a80: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-00011a90: 696f 6e20 616e 6420 454c 4d4d 2061 6e61  ion and ELMM ana
-00011aa0: 6c79 7369 732a 2a0a 6060 600a 7767 6420  lysis**.```.wgd 
-00011ab0: 7669 7a20 2d64 206b 7364 6174 610a 6060  viz -d ksdata.``
-00011ac0: 600a 0a54 6865 2070 726f 6772 616d 2060  `..The program `
-00011ad0: 7767 6420 7669 7a60 2069 7320 6d61 696e  wgd viz` is main
-00011ae0: 6c79 2066 6f72 2074 6865 2070 7572 706f  ly for the purpo
-00011af0: 7365 206f 6620 2a4b 2a3c 7375 623e 533c  se of *K*<sub>S<
-00011b00: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-00011b10: 6f6e 2061 6e64 2073 796e 7465 6e79 2076  on and synteny v
-00011b20: 6973 7561 6c69 7a61 7469 6f6e 2c20 7769  isualization, wi
-00011b30: 7468 2073 6f6d 6520 6f70 7469 6f6e 616c  th some optional
-00011b40: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
-00011b50: 6720 616e 616c 7973 6973 2e20 5468 6520  g analysis. The 
-00011b60: 6261 7369 6320 6675 6e63 7469 6f6e 2069  basic function i
-00011b70: 7320 6a75 7374 2074 6f20 706c 6f74 2074  s just to plot t
-00011b80: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
-00011b90: 623e 2064 6973 7472 6962 7574 696f 6e20  b> distribution 
-00011ba0: 616e 6420 636f 6e64 7563 7420 616e 2045  and conduct an E
-00011bb0: 4c4d 4d20 616e 616c 7973 6973 2069 6e20  LMM analysis in 
-00011bc0: 7365 6172 6368 206f 6620 706f 7465 6e74  search of potent
-00011bd0: 6961 6c20 5747 4420 636f 6d70 6f6e 656e  ial WGD componen
-00011be0: 7473 2e20 536f 6d65 206b 6579 2070 6172  ts. Some key par
-00011bf0: 616d 6574 6572 7320 6166 6665 6374 696e  ameters affectin
-00011c00: 6720 7468 6520 454c 4d4d 2072 6573 756c  g the ELMM resul
-00011c10: 7420 696e 636c 7564 6520 602d 2d70 726f  t include `--pro
-00011c20: 6d69 6e65 6e63 655f 6375 746f 6666 602c  minence_cutoff`,
-00011c30: 2060 2d2d 7265 6c5f 6865 6967 6874 602c   `--rel_height`,
-00011c40: 2060 2d2d 656d 5f69 7465 7261 7469 6f6e   `--em_iteration
-00011c50: 7360 2061 6e64 2060 2d2d 656d 5f69 6e69  s` and `--em_ini
-00011c60: 7469 616c 697a 6174 696f 6e73 602c 2061  tializations`, a
-00011c70: 6c6c 206f 6620 7768 6963 6820 6861 7665  ll of which have
-00011c80: 2062 6565 6e20 6578 706c 6169 6e65 6420   been explained 
-00011c90: 6168 6561 642e 0a0a 2a2a 4120 7375 6767  ahead...**A sugg
-00011ca0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-00011cb0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-00011cc0: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
-00011cd0: 6f77 2a2a 0a0a 6060 600a 7767 6420 7669  ow**..```.wgd vi
-00011ce0: 7a20 2d64 206b 7364 6174 610a 6060 600a  z -d ksdata.```.
-00011cf0: 0a2a 2a54 6865 2076 6973 7561 6c69 7a61  .**The visualiza
-00011d00: 7469 6f6e 206f 6620 2a4b 2a3c 7375 623e  tion of *K*<sub>
-00011d10: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
-00011d20: 7269 6275 7469 6f6e 2077 6974 6820 7261  ribution with ra
-00011d30: 7465 2063 6f72 7265 6374 696f 6e2a 2a0a  te correction**.
-00011d40: 6060 600a 7767 6420 7669 7a20 2d64 206b  ```.wgd viz -d k
-00011d50: 7364 6174 6120 2d73 7020 7370 6461 7461  sdata -sp spdata
-00011d60: 202d 2d66 6f63 7573 3261 6c6c 2066 6f63   --focus2all foc
-00011d70: 616c 5f73 7065 6369 6573 202d 2d65 7874  al_species --ext
-00011d80: 7261 7061 7261 6e6f 6d65 6b73 206b 7364  raparanomeks ksd
-00011d90: 6174 6120 282d 2d61 6e63 686f 7270 6f69  ata (--anchorpoi
-00011da0: 6e74 7320 6170 6461 7461 202d 2d70 6c6f  nts apdata --plo
-00011db0: 7461 7067 6d6d 202d 2d70 6c6f 7465 6c6d  tapgmm --plotelm
-00011dc0: 6d29 0a60 6060 0a0a 4265 7369 6465 7320  m).```..Besides 
-00011dd0: 7468 6520 6261 7369 6320 2a4b 2a3c 7375  the basic *K*<su
-00011de0: 623e 533c 2f73 7562 3e20 706c 6f74 2c20  b>S</sub> plot, 
-00011df0: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
-00011e00: 6520 636f 7272 6563 7469 6f6e 2063 616e  e correction can
-00011e10: 2061 6c73 6f20 6265 2061 6368 6965 7665   also be achieve
-00011e20: 6420 6769 7665 6e20 6174 206c 6561 7374  d given at least
-00011e30: 2061 2073 7065 6369 6573 2074 7265 6520   a species tree 
-00011e40: 2876 6961 2074 6865 206f 7074 696f 6e20  (via the option 
-00011e50: 602d 2d73 7065 6369 6573 7472 6565 6029  `--speciestree`)
-00011e60: 2061 6e64 2061 2066 6f63 616c 2073 7065   and a focal spe
-00011e70: 6369 6573 2028 6569 7468 6572 2076 6961  cies (either via
-00011e80: 2074 6865 206f 7074 696f 6e20 602d 2d66   the option `--f
-00011e90: 6f63 7573 3261 6c6c 6020 6f72 2076 6961  ocus2all` or via
-00011ea0: 2074 6865 206f 7074 696f 6e20 602d 2d73   the option `--s
-00011eb0: 7061 6972 6020 696e 2074 6865 2066 6f72  pair` in the for
-00011ec0: 6d20 6f66 2022 2466 6f63 616c 5f73 7065  m of "$focal_spe
-00011ed0: 6369 6573 3b24 666f 6361 6c5f 7370 6563  cies;$focal_spec
-00011ee0: 6965 7322 292e 2049 7427 7320 7375 6767  ies"). It's sugg
-00011ef0: 6573 7465 6420 7468 6174 2074 6865 206f  ested that the o
-00011f00: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
-00011f10: 7562 3e53 3c2f 7375 623e 2064 6174 6120  ub>S</sub> data 
-00011f20: 6973 2070 726f 7669 6465 6420 6279 2074  is provided by t
-00011f30: 6865 2060 2d2d 6461 7461 6669 6c65 6020  he `--datafile` 
-00011f40: 6f70 7469 6f6e 2061 6e64 2074 6865 2070  option and the p
-00011f50: 6172 616c 6f67 6f75 7320 2a4b 2a3c 7375  aralogous *K*<su
-00011f60: 623e 533c 2f73 7562 3e20 6461 7461 2069  b>S</sub> data i
-00011f70: 7320 7072 6f76 6964 6564 2062 7920 7468  s provided by th
-00011f80: 6520 602d 2d65 7874 7261 7061 7261 6e6f  e `--extraparano
-00011f90: 6d65 6b73 6020 6f70 7469 6f6e 2c20 616c  meks` option, al
-00011fa0: 7468 6f75 6768 2069 7427 7320 616c 6c6f  though it's allo
-00011fb0: 7765 6420 746f 206f 6e6c 7920 7072 6f76  wed to only prov
-00011fc0: 6964 6520 2a4b 2a3c 7375 623e 533c 2f73  ide *K*<sub>S</s
-00011fd0: 7562 3e20 6461 7461 2076 6961 2074 6865  ub> data via the
-00011fe0: 2060 2d2d 6461 7461 6669 6c65 6020 6f70   `--datafile` op
-00011ff0: 7469 6f6e 2061 6e64 2064 6570 6f73 6974  tion and deposit
-00012000: 2062 6f74 6820 6f72 7468 6f6c 6f67 6f75   both orthologou
-00012010: 7326 7061 7261 6c6f 676f 7573 202a 4b2a  s&paralogous *K*
-00012020: 3c73 7562 3e53 3c2f 7375 623e 2064 6174  <sub>S</sub> dat
-00012030: 6120 7468 6572 656f 6e2e 2054 6865 7265  a thereon. There
-00012040: 2061 7265 2074 776f 2074 7970 6573 206f   are two types o
-00012050: 6620 6d69 7865 6420 706c 6f74 7320 2874  f mixed plots (t
-00012060: 6865 2022 6d69 7865 6422 2068 6572 6520  he "mixed" here 
-00012070: 7265 6665 7273 2074 6f20 6d69 7865 6420  refers to mixed 
-00012080: 6f72 7468 6f6c 6f67 6f75 7320 616e 6420  orthologous and 
-00012090: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
-000120a0: 7562 3e53 3c2f 7375 623e 2064 6973 7472  ub>S</sub> distr
-000120b0: 6962 7574 696f 6e73 292c 206f 6e65 206f  ibutions), one o
-000120c0: 6620 7768 6963 6820 6973 2073 696d 696c  f which is simil
-000120d0: 6172 2074 6f20 7768 6174 2060 6b73 7261  ar to what `ksra
-000120e0: 7465 7360 2070 6c6f 7473 2c20 7768 696c  tes` plots, whil
-000120f0: 6520 7468 6520 6f74 6865 7220 6f66 2077  e the other of w
-00012100: 6869 6368 2069 7320 6c69 6b65 2074 6865  hich is like the
-00012110: 2063 6f6e 7665 6e74 696f 6e61 6c20 2a4b   conventional *K
-00012120: 2a3c 7375 623e 533c 2f73 7562 3e20 706c  *<sub>S</sub> pl
-00012130: 6f74 2074 6861 7420 626f 7468 2074 6865  ot that both the
-00012140: 206f 7269 6769 6e61 6c20 6f72 7468 6f6c   original orthol
-00012150: 6f67 6f75 7320 616e 6420 7061 7261 6c6f  ogous and paralo
-00012160: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
-00012170: 7375 623e 2064 6973 7472 6962 7574 696f  sub> distributio
-00012180: 6e73 2061 7265 2074 7275 7468 6675 6c6c  ns are truthfull
-00012190: 7920 706c 6f74 7465 6420 696e 7374 6561  y plotted instea
-000121a0: 6420 6f66 206a 7573 7420 6265 696e 6720  d of just being 
-000121b0: 7265 7072 6573 656e 7465 6420 6279 2073  represented by s
-000121c0: 6f6d 6520 7665 7274 6963 616c 206c 696e  ome vertical lin
-000121d0: 6573 2e20 5765 2073 7567 6765 7374 2075  es. We suggest u
-000121e0: 7365 7273 2074 6f20 6164 6f70 7420 7468  sers to adopt th
-000121f0: 6520 606b 7372 6174 6573 602d 6c69 6b65  e `ksrates`-like
-00012200: 2070 6c6f 7473 2c20 7768 6963 6820 6973   plots, which is
-00012210: 2074 6865 2064 6566 6175 6c74 2e20 4f74   the default. Ot
-00012220: 6865 7277 6973 6520 7468 6520 6f70 7469  herwise the opti
-00012230: 6f6e 2060 2d2d 636c 6173 7369 6360 2077  on `--classic` w
-00012240: 696c 6c20 6265 206e 6565 6465 6420 746f  ill be needed to
-00012250: 2073 6574 2e20 4578 7472 6120 6d69 7874   set. Extra mixt
-00012260: 7572 6520 6d6f 6465 6c69 6e67 2061 6e61  ure modeling ana
-00012270: 6c79 7369 7320 6361 6e20 6265 2069 6e69  lysis can be ini
-00012280: 7469 6174 6564 2076 6961 2074 6865 206f  tiated via the o
-00012290: 7074 696f 6e20 602d 2d70 6c6f 7465 6c6d  ption `--plotelm
-000122a0: 6d60 2061 6e64 2060 2d2d 706c 6f74 6170  m` and `--plotap
-000122b0: 676d 6d60 2077 6974 6820 7468 6520 616e  gmm` with the an
-000122c0: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
-000122d0: 6669 6c65 2070 726f 7669 6465 6420 6279  file provided by
-000122e0: 2074 6865 2060 2d2d 616e 6368 6f72 706f   the `--anchorpo
-000122f0: 696e 7473 6020 6f70 7469 6f6e 2e0a 0a2a  ints` option...*
-00012300: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
-00012310: 7274 696e 6720 7275 6e20 6361 6e20 7573  rting run can us
-00012320: 6520 636f 6d6d 616e 6420 7369 6d70 6c79  e command simply
-00012330: 2061 7320 6265 6c6f 772a 2a0a 0a60 6060   as below**..```
-00012340: 0a77 6764 2076 697a 202d 6420 6b73 6461  .wgd viz -d ksda
-00012350: 7461 202d 7370 2073 7064 6174 6120 2d2d  ta -sp spdata --
-00012360: 666f 6375 7332 616c 6c20 666f 6361 6c5f  focus2all focal_
-00012370: 7370 6563 6965 7320 2d2d 6578 7472 6170  species --extrap
-00012380: 6172 616e 6f6d 656b 7320 6b73 6461 7461  aranomeks ksdata
-00012390: 0a60 6060 0a0a 2a2a 5468 6520 7669 7375  .```..**The visu
-000123a0: 616c 697a 6174 696f 6e20 6f66 2073 796e  alization of syn
-000123b0: 7465 6e79 2a2a 0a60 6060 0a77 6764 2076  teny**.```.wgd v
-000123c0: 697a 202d 6170 2061 7064 6174 6120 2d73  iz -ap apdata -s
-000123d0: 6d20 736d 6461 7461 202d 6d74 206d 7464  m smdata -mt mtd
-000123e0: 6174 6120 2d67 7420 6774 6461 7461 202d  ata -gt gtdata -
-000123f0: 2d70 6c6f 7473 796e 2028 2d2d 6d69 6e6c  -plotsyn (--minl
-00012400: 656e 202d 3120 2d2d 6d69 6e73 6567 6c65  en -1 --minsegle
-00012410: 6e20 3130 3030 3020 2d2d 6d69 6e67 656e  n 10000 --mingen
-00012420: 656e 756d 2033 3029 0a60 6060 0a0a 436f  enum 30).```..Co
-00012430: 6d70 6172 6564 2074 6f20 7468 6520 6f72  mpared to the or
-00012440: 6967 696e 616c 2060 7767 6460 2c20 7468  iginal `wgd`, th
-00012450: 6520 6077 6764 2076 697a 6020 7072 6f67  e `wgd viz` prog
-00012460: 7261 6d20 6164 6465 6420 7379 6e74 656e  ram added synten
-00012470: 7920 7669 7375 616c 697a 6174 696f 6e20  y visualization 
-00012480: 7069 7065 6c69 6e65 2e20 5573 6572 7320  pipeline. Users 
-00012490: 6e65 6564 2074 6f20 7072 6f76 6964 6520  need to provide 
-000124a0: 7468 6520 666c 6167 206f 7074 696f 6e20  the flag option 
-000124b0: 602d 2d70 6c6f 7473 796e 6020 746f 2069  `--plotsyn` to i
-000124c0: 6e69 7469 6174 6520 7468 6973 2070 6172  nitiate this par
-000124d0: 7420 6f66 2070 6970 656c 696e 652e 2054  t of pipeline. T
-000124e0: 6869 7320 7374 6570 2061 7373 756d 6573  his step assumes
-000124f0: 2074 6861 7420 7573 6572 7320 6861 7665   that users have
-00012500: 206f 6274 6169 6e65 6420 616c 7265 6164   obtained alread
-00012510: 7920 7468 6520 7379 6e74 656e 6963 2072  y the syntenic r
-00012520: 6573 756c 7473 2066 726f 6d20 6069 2d61  esults from `i-a
-00012530: 6468 6f72 6560 2061 6e64 2075 7365 7320  dhore` and uses 
-00012540: 7468 6f73 6520 7265 7375 6c74 2066 696c  those result fil
-00012550: 6573 2074 6f20 7265 616c 697a 6520 7468  es to realize th
-00012560: 6520 7379 6e74 656e 7920 7669 7375 616c  e synteny visual
-00012570: 697a 6174 696f 6e2e 2053 696d 6c69 6172  ization. Simliar
-00012580: 2074 6f20 6077 6764 2073 796e 602c 2074   to `wgd syn`, t
-00012590: 6865 2065 7874 7261 202a 4b2a 3c73 7562  he extra *K*<sub
-000125a0: 3e53 3c2f 7375 623e 2064 6174 6120 6669  >S</sub> data fi
-000125b0: 6c65 2063 616e 2062 6520 7472 616e 736d  le can be transm
-000125c0: 6974 7465 6420 7669 6120 7468 6520 6f70  itted via the op
-000125d0: 7469 6f6e 2060 2d2d 6461 7461 6669 6c65  tion `--datafile
-000125e0: 6020 2869 6e73 7465 6164 206f 6620 602d  ` (instead of `-
-000125f0: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
-00012600: 6020 6f70 7469 6f6e 292e 2042 6173 6963  ` option). Basic
-00012610: 616c 6c79 2c20 7468 6520 7379 6e74 656e  ally, the synten
-00012620: 6963 2072 6573 756c 7420 6669 6c65 7320  ic result files 
-00012630: 7265 7175 6972 6564 2061 7265 2074 6865  required are the
-00012640: 2061 6e63 686f 7220 706f 696e 7473 2064   anchor points d
-00012650: 6174 6166 696c 652c 206d 756c 7469 706c  atafile, multipl
-00012660: 6963 6f6e 7320 6461 7461 6669 6c65 2c20  icons datafile, 
-00012670: 6765 6e65 2074 6162 6c65 2064 6174 6166  gene table dataf
-00012680: 696c 6520 2861 7574 6f6d 6174 656c 7920  ile (automately 
-00012690: 7072 6f64 7563 6564 2062 7920 6077 6764  produced by `wgd
-000126a0: 2073 796e 6029 2c20 616e 6420 7365 676d   syn`), and segm
-000126b0: 656e 7473 2064 6174 6166 696c 652e 0a0a  ents datafile...
-000126c0: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
-000126d0: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
-000126e0: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
-000126f0: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
-00012700: 600a 7767 6420 7669 7a20 2d61 7020 6170  `.wgd viz -ap ap
-00012710: 6461 7461 202d 736d 2073 6d64 6174 6120  data -sm smdata 
-00012720: 2d6d 7420 6d74 6461 7461 202d 6774 2067  -mt mtdata -gt g
-00012730: 7464 6174 6120 2d2d 706c 6f74 7379 6e0a  tdata --plotsyn.
-00012740: 6060 600a 0a23 2320 496c 6c75 7374 7261  ```..## Illustra
-00012750: 7469 6f6e 0a0a 5765 2069 6c6c 7573 7472  tion..We illustr
-00012760: 6174 6520 6f75 7220 7072 6f67 7261 6d20  ate our program 
-00012770: 6f6e 2061 6e20 6578 656d 706c 6172 7920  on an exemplary 
-00012780: 5747 4420 696e 6665 7265 6e63 6520 616e  WGD inference an
-00012790: 6420 6461 7469 6e67 2075 706f 6e20 7370  d dating upon sp
-000127a0: 6563 6965 7320 2a41 7175 696c 6567 6961  ecies *Aquilegia
-000127b0: 2063 6f65 7275 6c65 612a 2e0a 0a54 6865   coerulea*...The
-000127c0: 202a 4171 7569 6c65 6769 6120 636f 6572   *Aquilegia coer
-000127d0: 756c 6561 2a20 7761 7320 7265 706f 7274  ulea* was report
-000127e0: 6564 2074 6f20 6578 7065 7269 656e 6365  ed to experience
-000127f0: 2061 6e20 7061 6c65 6f2d 706f 6c79 706c   an paleo-polypl
-00012800: 6f69 6469 7a61 7469 6f6e 2065 7665 6e74  oidization event
-00012810: 2061 6674 6572 2074 6865 2064 6976 6572   after the diver
-00012820: 6765 6e63 6520 6f66 2063 6f72 6520 6575  gence of core eu
-00012830: 6469 636f 7473 2c20 7768 6963 6820 6973  dicots, which is
-00012840: 206c 696b 656c 7920 7368 6172 6564 2062   likely shared b
-00012850: 7920 616c 6c20 5261 6e75 6e63 756c 616c  y all Ranunculal
-00012860: 6573 2e0a 0a46 6972 7374 2061 626f 7665  es...First above
-00012870: 2061 6c6c 2c20 6c65 7427 7320 6465 6c69   all, let's deli
-00012880: 6e65 6174 6520 7468 6520 7768 6f6c 6520  neate the whole 
-00012890: 7061 7261 6e6f 6d65 202a 4b2a 3c73 7562  paranome *K*<sub
-000128a0: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-000128b0: 7472 6962 7574 696f 6e20 616e 6420 6861  tribution and ha
-000128c0: 7665 2061 2062 6173 6963 206f 6273 6572  ve a basic obser
-000128d0: 7661 7469 6f6e 2066 6f72 2070 6f74 656e  vation for poten
-000128e0: 7469 616c 6c79 2063 6f6e 6365 6976 6162  tially conceivab
-000128f0: 6c65 2057 4744 732c 2075 7369 6e67 2074  le WGDs, using t
-00012900: 6865 2063 6f6d 6d61 6e64 206c 696e 6520  he command line 
-00012910: 6265 6c6f 772e 0a0a 6060 600a 7767 6420  below...```.wgd 
-00012920: 646d 6420 4171 7569 6c65 6769 615f 636f  dmd Aquilegia_co
-00012930: 6572 756c 6561 0a77 6764 206b 7364 2077  erulea.wgd ksd w
-00012940: 6764 5f64 6d64 2f41 7175 696c 6567 6961  gd_dmd/Aquilegia
-00012950: 5f63 6f65 7275 6c65 612e 7473 7620 4171  _coerulea.tsv Aq
-00012960: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00012970: 0a60 6060 0a0a 5468 6520 636f 6e73 7472  .```..The constr
-00012980: 7563 7465 6420 7768 6f6c 6520 7061 7261  ucted whole para
-00012990: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
-000129a0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-000129b0: 7574 696f 6e20 6f66 202a 4171 7569 6c65  ution of *Aquile
-000129c0: 6769 6120 636f 6572 756c 6561 2a20 6973  gia coerulea* is
-000129d0: 2061 7320 6265 6c6f 772c 2077 6520 6361   as below, we ca
-000129e0: 6e20 7365 6520 7468 6174 2074 6865 7265  n see that there
-000129f0: 2073 6565 6d73 2074 6f20 6265 2061 2068   seems to be a h
-00012a00: 756d 7020 6174 202a 4b2a 3c73 7562 3e53  ump at *K*<sub>S
-00012a10: 3c2f 7375 623e 2031 2062 7574 206e 6f74  </sub> 1 but not
-00012a20: 2063 6c65 6172 2e0a 0a21 5b5d 2864 6174   clear...![](dat
-00012a30: 612f 4171 7569 6c65 6769 615f 636f 6572  a/Aquilegia_coer
-00012a40: 756c 6561 2e74 7376 2e6b 7364 5f77 702e  ulea.tsv.ksd_wp.
-00012a50: 7376 6729 0a0a 5765 2074 6865 6e20 636f  svg)..We then co
-00012a60: 6e73 7472 7563 7420 7468 6520 616e 6368  nstruct the anch
-00012a70: 6f72 202a 4b2a 3c73 7562 3e53 3c2f 7375  or *K*<sub>S</su
-00012a80: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-00012a90: 696f 6e20 7573 696e 6720 7468 6520 636f  ion using the co
-00012aa0: 6d6d 616e 6420 6c69 6e65 2062 656c 6f77  mmand line below
-00012ab0: 2e0a 0a60 6060 0a77 6764 2073 796e 202d  ...```.wgd syn -
-00012ac0: 6620 6d52 4e41 202d 6120 4e61 6d65 2077  f mRNA -a Name w
-00012ad0: 6764 5f64 6d64 2f41 7175 696c 6567 6961  gd_dmd/Aquilegia
-00012ae0: 5f63 6f65 7275 6c65 612e 7473 7620 4171  _coerulea.tsv Aq
-00012af0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00012b00: 2e67 6666 3320 2d6b 7320 7767 645f 6b73  .gff3 -ks wgd_ks
-00012b10: 642f 4171 7569 6c65 6769 615f 636f 6572  d/Aquilegia_coer
-00012b20: 756c 6561 2e74 7376 2e6b 732e 7473 760a  ulea.tsv.ks.tsv.
-00012b30: 6060 600a 0a41 7320 7368 6f77 6e20 6265  ```..As shown be
-00012b40: 6c6f 772c 2074 6865 7265 2061 7265 2073  low, there are s
-00012b50: 6f6d 6520 7265 7461 696e 6564 2061 6e63  ome retained anc
-00012b60: 686f 7220 7061 6972 7320 7769 7468 202a  hor pairs with *
-00012b70: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2062  K*<sub>S</sub> b
-00012b80: 6574 7765 656e 2031 2061 6e64 2032 2c20  etween 1 and 2, 
-00012b90: 7768 6963 6820 7365 656d 7320 746f 2073  which seems to s
-00012ba0: 7567 6765 7374 2061 2057 4744 2065 7665  uggest a WGD eve
-00012bb0: 6e74 2e0a 0a21 5b5d 2864 6174 612f 4171  nt...![](data/Aq
-00012bc0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00012bd0: 2e74 7376 2e6b 7364 5f77 705f 6170 2e73  .tsv.ksd_wp_ap.s
-00012be0: 7667 290a 0a54 6865 2061 7373 6f63 6961  vg)..The associa
-00012bf0: 7465 6420 6064 7570 5374 6163 6b60 2070  ted `dupStack` p
-00012c00: 6c6f 7420 7368 6f77 7320 7468 6174 2074  lot shows that t
-00012c10: 6865 7265 2061 7265 206e 756d 6572 6f75  here are numerou
-00012c20: 7320 6475 706c 6963 6174 6564 2073 6567  s duplicated seg
-00012c30: 6d65 6e74 7320 6163 726f 7373 206d 6f73  ments across mos
-00012c40: 7420 6f66 2074 6865 2063 6872 6f6d 6f73  t of the chromos
-00012c50: 6f6d 6573 2e0a 0a21 5b5d 2864 6174 612f  omes...![](data/
-00012c60: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00012c70: 6561 5f41 7175 696c 6567 6961 5f63 6f65  ea_Aquilegia_coe
-00012c80: 7275 6c65 615f 6d75 6c74 6970 6c69 636f  rulea_multiplico
-00012c90: 6e73 5f6c 6576 656c 2e73 7667 290a 0a57  ns_level.svg)..W
-00012ca0: 6520 696d 706c 656d 656e 7465 6420 7477  e implemented tw
-00012cb0: 6f20 7479 7065 7320 6f66 2064 6f74 2070  o types of dot p
-00012cc0: 6c6f 7473 2069 6e20 6f78 666f 7264 2067  lots in oxford g
-00012cd0: 7269 643a 206f 6e65 2069 6e20 7468 6520  rid: one in the 
-00012ce0: 756e 6974 206f 6620 6261 7365 7320 616e  unit of bases an
-00012cf0: 6420 7468 6520 6f74 6865 7220 696e 2074  d the other in t
-00012d00: 6865 2075 6e69 7420 6f66 2067 656e 6573  he unit of genes
-00012d10: 2c20 7768 6963 6820 6361 6e20 6265 2063  , which can be c
-00012d20: 6f6c 6f72 6564 2062 7920 2a4b 2a3c 7375  olored by *K*<su
-00012d30: 623e 533c 2f73 7562 3e20 7661 6c75 6573  b>S</sub> values
-00012d40: 2067 6976 656e 202a 4b2a 3c73 7562 3e53   given *K*<sub>S
-00012d50: 3c2f 7375 623e 2064 6174 612e 0a0a 215b  </sub> data...![
-00012d60: 5d28 6461 7461 2f41 7175 696c 6567 6961  ](data/Aquilegia
-00012d70: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
-00012d80: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
-00012d90: 4b73 2e64 6f74 5f75 6e69 745f 6765 6e65  Ks.dot_unit_gene
-00012da0: 2e70 6e67 290a 0a41 7320 7368 6f77 6e20  .png)..As shown 
-00012db0: 6162 6f76 652c 2074 6865 2064 6f74 2070  above, the dot p
-00012dc0: 6c6f 7420 696e 2074 6865 2075 6e69 7420  lot in the unit 
-00012dd0: 6f66 2067 656e 6573 2070 7265 7365 6e74  of genes present
-00012de0: 7320 6e75 6d65 726f 7573 2064 656e 7365  s numerous dense
-00012df0: 6c79 2061 6767 7265 6761 7465 6420 286c  ly aggregated (l
-00012e00: 696e 652d 6c69 6b65 2920 616e 6368 6f72  ine-like) anchor
-00012e10: 2070 6f69 6e74 7320 6174 206d 6f73 7420   points at most 
-00012e20: 6f66 2074 6865 2063 6872 6f6d 6f73 6f6d  of the chromosom
-00012e30: 6573 2077 6974 6820 636f 6e73 6973 7465  es with consiste
-00012e40: 6e74 202a 4b2a 3c73 7562 3e53 3c2f 7375  nt *K*<sub>S</su
-00012e50: 623e 2061 6765 2062 6574 7765 656e 2031  b> age between 1
-00012e60: 2061 6e64 2032 2e20 5468 6520 646f 7420   and 2. The dot 
-00012e70: 706c 6f74 2069 6e20 7468 6520 756e 6974  plot in the unit
-00012e80: 206f 6620 6261 7365 7320 7368 6f77 7320   of bases shows 
-00012e90: 7468 6520 7361 6d65 2070 6174 7465 726e  the same pattern
-00012ea0: 2c20 6173 206d 616e 6966 6573 7465 6420  , as manifested 
-00012eb0: 6265 6c6f 772e 0a0a 215b 5d28 6461 7461  below...![](data
-00012ec0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
-00012ed0: 6c65 612d 7673 2d41 7175 696c 6567 6961  lea-vs-Aquilegia
-00012ee0: 5f63 6f65 7275 6c65 615f 4b73 2e64 6f74  _coerulea_Ks.dot
-00012ef0: 2e70 6e67 290a 0a54 6865 2064 6f74 2070  .png)..The dot p
-00012f00: 6c6f 7473 2077 6974 686f 7574 202a 4b2a  lots without *K*
-00012f10: 3c73 7562 3e53 3c2f 7375 623e 2061 6e6e  <sub>S</sub> ann
-00012f20: 6f74 6174 696f 6e20 7769 6c6c 2061 6c73  otation will als
-00012f30: 6f20 6265 2061 7574 6f6d 6174 656c 7920  o be automately 
-00012f40: 7072 6f64 7563 6564 2c20 6173 2073 686f  produced, as sho
-00012f50: 776e 2062 656c 6f77 2e0a 0a21 5b5d 2864  wn below...![](d
-00012f60: 6174 612f 4171 7569 6c65 6769 615f 636f  ata/Aquilegia_co
-00012f70: 6572 756c 6561 2d76 732d 4171 7569 6c65  erulea-vs-Aquile
-00012f80: 6769 615f 636f 6572 756c 6561 2e64 6f74  gia_coerulea.dot
-00012f90: 5f75 6e69 745f 6765 6e65 2e70 6e67 290a  _unit_gene.png).
-00012fa0: 0a21 5b5d 2864 6174 612f 4171 7569 6c65  .![](data/Aquile
-00012fb0: 6769 615f 636f 6572 756c 6561 2d76 732d  gia_coerulea-vs-
-00012fc0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00012fd0: 6561 2e64 6f74 2e70 6e67 290a 0a4e 6f74  ea.dot.png)..Not
-00012fe0: 6520 7468 6174 2074 6865 206f 7061 6369  e that the opaci
-00012ff0: 7479 206f 6620 616e 6368 6f72 2064 6f74  ty of anchor dot
-00013000: 7320 616e 6420 616c 6c20 686f 6d6f 6c6f  s and all homolo
-00013010: 6720 646f 7473 2063 616e 2062 6520 7365  g dots can be se
-00013020: 7420 6279 2074 6865 206f 7074 696f 6e20  t by the option 
-00013030: 6061 7061 6c70 6861 6020 616e 6420 6068  `apalpha` and `h
-00013040: 6f61 6c70 6861 6020 7365 7061 7261 7465  oalpha` separate
-00013050: 6c79 2e20 4966 206f 6e65 206a 7573 7420  ly. If one just 
-00013060: 7761 6e74 7320 746f 2073 6565 2074 6865  wants to see the
-00013070: 2061 6e63 686f 7220 646f 7473 2c20 7365   anchor dots, se
-00013080: 7474 696e 6720 7468 6520 6068 6f61 6c70  tting the `hoalp
-00013090: 6861 6020 6173 2030 2028 6f72 206f 7468  ha` as 0 (or oth
-000130a0: 6572 206d 696e 7573 6375 6c65 2076 616c  er minuscule val
-000130b0: 7565 7329 2077 696c 6c20 646f 2e20 4966  ues) will do. If
-000130c0: 206f 6e65 2077 616e 7473 2074 6f20 7365   one wants to se
-000130d0: 6520 7468 6520 6469 7374 7269 6275 7469  e the distributi
-000130e0: 6f6e 206f 6620 7768 6f6c 6520 646f 7473  on of whole dots
-000130f0: 2062 6574 7465 722c 2073 6574 7469 6e67   better, setting
-00013100: 2074 6865 2060 686f 616c 7068 6160 2068   the `hoalpha` h
-00013110: 6967 6865 7220 2861 6e64 2060 6170 616c  igher (and `apal
-00013120: 7068 6160 206c 6f77 6572 2920 7769 6c6c  pha` lower) will
-00013130: 2064 6f2e 2054 6865 2060 646f 7473 697a   do. The `dotsiz
-00013140: 6560 206f 7074 696f 6e20 6361 6e20 6265  e` option can be
-00013150: 2063 616c 6c65 6420 746f 2061 646a 7573   called to adjus
-00013160: 7420 7468 6520 7369 7a65 206f 6620 646f  t the size of do
-00013170: 7473 2e0a 0a41 2066 7572 7468 6572 2061  ts...A further a
-00013180: 7373 6f63 6961 7465 6420 5379 6e64 6570  ssociated Syndep
-00013190: 7468 2070 6c6f 7420 7368 6f77 7320 7468  th plot shows th
-000131a0: 6174 2074 6865 7265 2061 7265 206d 6f72  at there are mor
-000131b0: 6520 7468 616e 2038 3020 6475 706c 6963  e than 80 duplic
-000131c0: 6174 6564 2073 6567 6d65 6e74 732c 2077  ated segments, w
-000131d0: 6869 6368 2064 6f6d 696e 6174 6573 2074  hich dominates t
-000131e0: 6865 2077 686f 6c65 2063 6f6c 6c69 6e65  he whole colline
-000131f0: 6172 2072 6174 696f 2063 6174 6567 6f72  ar ratio categor
-00013200: 792e 0a0a 215b 5d28 6461 7461 2f53 796e  y...![](data/Syn
-00013210: 6465 7074 682e 7376 6729 0a0a 5765 2063  depth.svg)..We c
-00013220: 616e 2066 6974 2061 6e20 454c 4d4d 206d  an fit an ELMM m
-00013230: 6978 7475 7265 206d 6f64 656c 2075 706f  ixture model upo
-00013240: 6e20 7468 6520 7768 6f6c 6520 7061 7261  n the whole para
-00013250: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
-00013260: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-00013270: 7574 696f 6e20 746f 2073 6565 206d 6f72  ution to see mor
-00013280: 6520 6163 6375 7261 7465 6c79 2074 6865  e accurately the
-00013290: 2073 6967 6e69 6669 6361 6e63 6520 616e   significance an
-000132a0: 6420 6c6f 6361 7469 6f6e 206f 6620 706f  d location of po
-000132b0: 7465 6e74 6961 6c20 5747 4473 2c20 7573  tential WGDs, us
-000132c0: 696e 6720 7468 6520 636f 6d6d 616e 6420  ing the command 
-000132d0: 6c69 6e65 2062 656c 6f77 2e0a 0a60 6060  line below...```
-000132e0: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
-000132f0: 6b73 642f 4171 7569 6c65 6769 615f 636f  ksd/Aquilegia_co
-00013300: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
-00013310: 760a 6060 600a 0a54 6865 2072 6573 756c  v.```..The resul
-00013320: 7420 6f66 2045 4c4d 4d20 6d69 7874 7572  t of ELMM mixtur
-00013330: 6520 6d6f 6465 6c20 636c 7573 7465 7269  e model clusteri
-00013340: 6e67 2073 686f 7773 2074 6861 7420 7468  ng shows that th
-00013350: 6572 6520 6973 2061 206c 696b 656c 7920  ere is a likely 
-00013360: 5747 4420 636f 6d70 6f6e 656e 7420 6174  WGD component at
-00013370: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00013380: 2031 2e31 392e 0a0a 215b 5d28 6461 7461   1.19...![](data
-00013390: 2f65 6c6d 6d5f 4171 7569 6c65 6769 615f  /elmm_Aquilegia_
-000133a0: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
-000133b0: 7473 765f 6265 7374 5f6d 6f64 656c 735f  tsv_best_models_
-000133c0: 7765 6967 6874 6564 2e73 7667 290a 0a4c  weighted.svg)..L
-000133d0: 6574 2773 2064 6f20 6120 6d69 7874 7572  et's do a mixtur
-000133e0: 6520 6d6f 6465 6c20 636c 7573 7465 7269  e model clusteri
-000133f0: 6e67 2066 6f72 2061 6e63 686f 7220 2a4b  ng for anchor *K
-00013400: 2a3c 7375 623e 533c 2f73 7562 3e20 746f  *<sub>S</sub> to
-00013410: 6f2c 2075 7369 6e67 2074 6865 2063 6f6d  o, using the com
-00013420: 6d61 6e64 206c 696e 6520 6265 6c6f 772e  mand line below.
-00013430: 0a0a 6060 600a 7767 6420 7065 616b 2077  ..```.wgd peak w
-00013440: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-00013450: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-00013460: 2e74 7376 202d 2d61 6e63 686f 7270 6f69  .tsv --anchorpoi
-00013470: 6e74 7320 7767 645f 7379 6e2f 6961 6468  nts wgd_syn/iadh
-00013480: 6f72 652d 6f75 742f 616e 6368 6f72 706f  ore-out/anchorpo
-00013490: 696e 7473 2e74 7874 202d 2d73 6567 6d65  ints.txt --segme
-000134a0: 6e74 7320 7767 645f 7379 6e2f 6961 6468  nts wgd_syn/iadh
-000134b0: 6f72 652d 6f75 742f 7365 676d 656e 7473  ore-out/segments
-000134c0: 2e74 7874 202d 2d6c 6973 7465 6c65 6d65  .txt --listeleme
-000134d0: 6e74 7320 7767 645f 7379 6e2f 6961 6468  nts wgd_syn/iadh
-000134e0: 6f72 652d 6f75 742f 6c69 7374 5f65 6c65  ore-out/list_ele
-000134f0: 6d65 6e74 732e 7478 7420 2d2d 6d75 6c74  ments.txt --mult
-00013500: 6970 6c69 636f 6e70 6169 7273 2077 6764  ipliconpairs wgd
-00013510: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
-00013520: 2f6d 756c 7469 706c 6963 6f6e 5f70 6169  /multiplicon_pai
-00013530: 7273 2e74 7874 202d 2d77 6569 6768 7465  rs.txt --weighte
-00013540: 640a 6060 600a 0a54 6865 2061 6e63 686f  d.```..The ancho
-00013550: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
-00013560: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
-00013570: 6f6e 2061 6c73 6f20 6861 7320 6120 6c69  on also has a li
-00013580: 6b65 6c79 2057 4744 2063 6f6d 706f 6e65  kely WGD compone
-00013590: 6e74 2077 6974 6820 6d6f 6465 2031 2e32  nt with mode 1.2
-000135a0: 382e 0a0a 215b 5d28 6461 7461 2f4f 7269  8...![](data/Ori
-000135b0: 6769 6e61 6c5f 416e 6368 6f72 4b73 5f47  ginal_AnchorKs_G
-000135c0: 4d4d 5f43 6f6d 706f 6e65 6e74 335f 6e6f  MM_Component3_no
-000135d0: 6465 5f77 6569 6768 7465 645f 4c6f 676e  de_weighted_Logn
-000135e0: 6f72 6d61 6c2e 7376 6729 0a0a 4e6f 7720  ormal.svg)..Now 
-000135f0: 7468 6174 2077 6520 6861 7665 2073 6565  that we have see
-00013600: 6e20 7468 6520 6576 6964 656e 6365 206f  n the evidence o
-00013610: 6620 6e75 6d65 726f 7573 2064 7570 6c69  f numerous dupli
-00013620: 6361 7465 6420 7365 676d 656e 7473 2061  cated segments a
-00013630: 6e64 2074 6865 2061 6767 7265 6761 7469  nd the aggregati
-00013640: 6f6e 206f 6620 6475 706c 6963 6174 6573  on of duplicates
-00013650: 2061 6765 2069 6e20 2a4b 2a3c 7375 623e   age in *K*<sub>
-00013660: 533c 2f73 7562 3e20 6172 6f75 6e64 2031  S</sub> around 1
-00013670: 2e32 2066 6f72 2061 6e63 686f 7220 7061  .2 for anchor pa
-00013680: 6972 7320 616e 6420 6e6f 6e2d 616e 6368  irs and non-anch
-00013690: 6f72 2070 6169 7273 2074 6872 6f75 6768  or pairs through
-000136a0: 6f75 7420 7468 6520 7768 6f6c 6520 6765  out the whole ge
-000136b0: 6e6f 6d65 2e20 5765 2063 616e 2063 6c61  nome. We can cla
-000136c0: 696d 2077 6974 6820 736f 6d65 2063 6f6e  im with some con
-000136d0: 6669 6465 6e63 6520 7468 6174 202a 4171  fidence that *Aq
-000136e0: 7569 6c65 6769 6120 636f 6572 756c 6561  uilegia coerulea
-000136f0: 2a20 6d69 6768 7420 6861 7665 2065 7870  * might have exp
-00013700: 6572 6965 6e63 6564 2061 2070 616c 656f  erienced a paleo
-00013710: 2d70 6f6c 7970 6c6f 6964 697a 6174 696f  -polyploidizatio
-00013720: 6e20 6576 656e 742e 204e 6578 742c 204c  n event. Next, L
-00013730: 6574 2773 2068 6176 6520 6120 6675 7274  et's have a furt
-00013740: 6865 7220 6c6f 6f6b 2061 626f 7574 2069  her look about i
-00013750: 7473 2070 6879 6c6f 6765 6e65 7469 6320  ts phylogenetic 
-00013760: 6c6f 6361 7469 6f6e 2e20 5765 206b 6e6f  location. We kno
-00013770: 7720 7468 6174 2074 6865 7265 2061 7265  w that there are
-00013780: 2075 6e63 6572 7461 696e 7469 6573 2061   uncertainties a
-00013790: 626f 7574 2077 6865 7468 6572 2074 6869  bout whether thi
-000137a0: 7320 7075 7461 7469 7665 2070 616c 656f  s putative paleo
-000137b0: 2d70 6f6c 7970 6c6f 6964 697a 6174 696f  -polyploidizatio
-000137c0: 6e20 6576 656e 7420 6973 2073 6861 7265  n event is share
-000137d0: 6420 7769 7468 2061 6c6c 2065 7564 6963  d with all eudic
-000137e0: 6f74 7320 6f72 206e 6f74 2e20 5765 2063  ots or not. We c
-000137f0: 616e 2063 686f 6f73 6520 736f 6d65 206f  an choose some o
-00013800: 7468 6572 2065 7564 6963 6f74 2067 656e  ther eudicot gen
-00013810: 6f6d 6573 2074 6f20 7365 6520 7468 6520  omes to see the 
-00013820: 6f72 6465 7269 6e67 206f 6620 7370 6563  ordering of spec
-00013830: 6961 7469 6f6e 2061 6e64 2070 6f6c 7970  iation and polyp
-00013840: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
-00013850: 7473 2e20 4865 7265 2077 6520 6368 6f6f  ts. Here we choo
-00013860: 7365 202a 5669 7469 7320 7669 6e69 6665  se *Vitis vinife
-00013870: 7261 2a2c 202a 5072 6f74 6561 2063 796e  ra*, *Protea cyn
-00013880: 6172 6f69 6465 732a 2061 6e64 202a 4163  aroides* and *Ac
-00013890: 6f72 7573 2061 6d65 7269 6361 6e75 732a  orus americanus*
-000138a0: 2069 6e20 7468 6520 666f 6c6c 6f77 696e   in the followin
-000138b0: 6720 2a4b 2a3c 7375 623e 533c 2f73 7562  g *K*<sub>S</sub
-000138c0: 3e20 616e 616c 7973 6973 2e20 4669 7273  > analysis. Firs
-000138d0: 742c 2077 6520 6275 696c 7420 6120 676c  t, we built a gl
-000138e0: 6f62 616c 204d 5242 4820 6661 6d69 6c79  obal MRBH family
-000138f0: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-00013900: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
-00013910: 6764 2064 6d64 202d 2d67 6c6f 6261 6c6d  gd dmd --globalm
-00013920: 7262 6820 4171 7569 6c65 6769 615f 636f  rbh Aquilegia_co
-00013930: 6572 756c 6561 2050 726f 7465 615f 6379  erulea Protea_cy
-00013940: 6e61 726f 6964 6573 2041 636f 7275 735f  naroides Acorus_
-00013950: 616d 6572 6963 616e 7573 2056 6974 6973  americanus Vitis
-00013960: 5f76 696e 6966 6572 6120 2d6f 2077 6764  _vinifera -o wgd
-00013970: 5f67 6c6f 6261 6c6d 7262 680a 6060 600a  _globalmrbh.```.
-00013980: 0a49 6e20 7468 6520 676c 6f62 616c 204d  .In the global M
-00013990: 5242 4820 6661 6d69 6c79 2c20 6576 6572  RBH family, ever
-000139a0: 7920 7061 6972 206f 6620 6f72 7468 6f6c  y pair of orthol
-000139b0: 6f67 6f75 7320 6765 6e65 7320 6973 2074  ogous genes is t
-000139c0: 6865 2072 6563 6970 726f 6361 6c20 6265  he reciprocal be
-000139d0: 7374 2068 6974 2c20 7375 6767 6573 7469  st hit, suggesti
-000139e0: 6e67 2074 7275 6520 6f72 7468 6f6c 6f67  ng true ortholog
-000139f0: 6f75 7320 7265 6c61 7469 6f6e 7368 6970  ous relationship
-00013a00: 732e 2057 6520 776f 756c 6420 7573 6520  s. We would use 
-00013a10: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
-00013a20: 7562 3e20 7661 6c75 6573 2061 7373 6f63  ub> values assoc
-00013a30: 6961 7465 6420 7769 7468 2074 6865 7365  iated with these
-00013a40: 206f 7274 686f 6c6f 676f 7573 2070 6169   orthologous pai
-00013a50: 7273 2074 6f20 6465 6c69 6d69 7420 7468  rs to delimit th
-00013a60: 6520 6469 7665 7267 656e 6365 202a 4b2a  e divergence *K*
-00013a70: 3c73 7562 3e53 3c2f 7375 623e 2070 6561  <sub>S</sub> pea
-00013a80: 6b2e 2054 6f67 6574 6865 7220 7769 7468  k. Together with
-00013a90: 2074 6865 2077 686f 6c65 2070 6172 616e   the whole paran
-00013aa0: 6f6d 6520 2a4b 2a3c 7375 623e 533c 2f73  ome *K*<sub>S</s
-00013ab0: 7562 3e20 6469 7374 7269 6275 7469 6f6e  ub> distribution
-00013ac0: 2c20 7765 2063 6f6e 6475 6374 2074 6865  , we conduct the
-00013ad0: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-00013ae0: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-00013af0: 6e64 2062 656c 6f77 2e0a 0a21 2153 696e  nd below...!!Sin
-00013b00: 6365 2060 7767 6460 2076 6572 7369 6f6e  ce `wgd` version
-00013b10: 2032 2e30 2e32 342c 2077 6520 7265 7772   2.0.24, we rewr
-00013b20: 6f74 6520 6120 636c 6561 6e65 7220 616e  ote a cleaner an
-00013b30: 6420 7175 6963 6b65 7220 7761 7920 6f66  d quicker way of
-00013b40: 2064 6f69 6e67 2073 7562 7374 6974 7574   doing substitut
-00013b50: 696f 6e20 7261 7465 2063 6f72 7265 6374  ion rate correct
-00013b60: 696f 6e2e 2049 7427 7320 6e6f 7420 7265  ion. It's not re
-00013b70: 7175 6972 6564 2074 6f20 7479 7065 2069  quired to type i
-00013b80: 6e20 616e 7920 7370 6569 6365 7320 7061  n any speices pa
-00013b90: 6972 2061 6e64 2061 2073 6572 6965 7320  ir and a series 
-00013ba0: 6f66 202a 4b2a 3c73 7562 3e53 3c2f 7375  of *K*<sub>S</su
-00013bb0: 623e 2070 6c6f 7473 2077 696c 6c20 6265  b> plots will be
-00013bc0: 2070 726f 6475 6365 642e 2054 6865 2072   produced. The r
-00013bd0: 6571 7569 7265 6420 6669 6c65 7320 6172  equired files ar
-00013be0: 6520 6f72 7468 6f6c 6f67 6f75 7320 2a4b  e orthologous *K
-00013bf0: 2a3c 7375 623e 533c 2f73 7562 3e20 7461  *<sub>S</sub> ta
-00013c00: 626c 652c 2070 6172 616c 6f67 6f75 7320  ble, paralogous 
-00013c10: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013c20: 7461 626c 652c 2061 2073 7065 6369 6573  table, a species
-00013c30: 2074 7265 6520 616e 6420 6120 666f 6375   tree and a focu
-00013c40: 7365 6420 7370 6563 6965 7320 2874 6865  sed species (the
-00013c50: 206f 6e65 2069 6e70 7574 7465 6420 7769   one inputted wi
-00013c60: 7468 2070 6172 616c 6f67 6f75 7320 2a4b  th paralogous *K
-00013c70: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
-00013c80: 7461 292e 2055 7365 7273 2063 616e 2063  ta). Users can c
-00013c90: 686f 6f73 6520 746f 2061 6464 206f 6e65  hoose to add one
-00013ca0: 206d 6f72 6520 6c61 7965 7220 6f66 2065   more layer of e
-00013cb0: 6c6d 6d20 6d6f 6465 6c69 6e67 206f 6e20  lmm modeling on 
-00013cc0: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
-00013cd0: 7562 3e53 3c2f 7375 623e 2076 616c 7565  ub>S</sub> value
-00013ce0: 7320 616e 642f 6f72 2067 6d6d 206d 6f64  s and/or gmm mod
-00013cf0: 656c 696e 6720 6f6e 2061 6e63 686f 7220  eling on anchor 
-00013d00: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013d10: 7661 6c75 6573 2e20 5468 6520 6f72 7468  values. The orth
-00013d20: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
-00013d30: 533c 2f73 7562 3e20 7661 6c75 6573 2063  S</sub> values c
-00013d40: 616e 2062 6520 6361 6c63 756c 6174 6564  an be calculated
-00013d50: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-00013d60: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
-00013d70: 6764 206b 7364 2077 6764 5f67 6c6f 6261  gd ksd wgd_globa
-00013d80: 6c6d 7262 682f 676c 6f62 616c 5f4d 5242  lmrbh/global_MRB
-00013d90: 482e 7473 7620 7365 7173 2a20 2d6f 2077  H.tsv seqs* -o w
-00013da0: 6764 5f67 6c6f 6261 6c6d 7262 685f 6b73  gd_globalmrbh_ks
-00013db0: 0a60 6060 0a0a 5769 7468 2074 6865 2063  .```..With the c
-00013dc0: 616c 6375 6c61 7465 6420 6f72 7468 6f6c  alculated orthol
-00013dd0: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
-00013de0: 2f73 7562 3e20 7461 626c 652c 2077 6520  /sub> table, we 
-00013df0: 6361 6e20 7573 6520 7468 6520 636f 6d6d  can use the comm
-00013e00: 616e 6420 6265 6c6f 7720 746f 2063 6f6e  and below to con
-00013e10: 6475 6374 2074 6865 2072 6174 6520 636f  duct the rate co
-00013e20: 7272 6563 7469 6f6e 2061 6e64 2f6f 7220  rrection and/or 
-00013e30: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
-00013e40: 2061 6e61 6c79 7369 732e 0a0a 6060 600a   analysis...```.
-00013e50: 7767 6420 7669 7a20 2d64 2077 6764 5f67  wgd viz -d wgd_g
-00013e60: 6c6f 6261 6c6d 7262 685f 6b73 2f67 6c6f  lobalmrbh_ks/glo
-00013e70: 6261 6c5f 4d52 4248 2e74 7376 2e6b 732e  bal_MRBH.tsv.ks.
-00013e80: 7473 7620 2d66 6120 4171 7569 6c65 6769  tsv -fa Aquilegi
-00013e90: 615f 636f 6572 756c 6561 202d 6570 6b20  a_coerulea -epk 
-00013ea0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00013eb0: 6561 2e6b 732e 7473 7620 2d61 7020 616e  ea.ks.tsv -ap an
-00013ec0: 6368 6f72 706f 696e 7473 2e74 7874 202d  chorpoints.txt -
-00013ed0: 7370 2073 7065 6369 6573 7472 6565 2e6e  sp speciestree.n
-00013ee0: 7720 2d6f 2077 6764 5f76 697a 5f6d 6978  w -o wgd_viz_mix
-00013ef0: 6564 5f4b 7320 2d2d 706c 6f74 656c 6d6d  ed_Ks --plotelmm
-00013f00: 202d 2d70 6c6f 7461 7067 6d6d 0a60 6060   --plotapgmm.```
-00013f10: 0a0a 6060 600a 7767 6420 6b73 6420 7767  ..```.wgd ksd wg
-00013f20: 645f 676c 6f62 616c 6d72 6268 2f67 6c6f  d_globalmrbh/glo
-00013f30: 6261 6c5f 4d52 4248 2e74 7376 2073 6571  bal_MRBH.tsv seq
-00013f40: 732a 202d 2d65 7874 7261 7061 7261 6e6f  s* --extraparano
-00013f50: 6d65 6b73 2077 6764 5f6b 7364 2f41 7175  meks wgd_ksd/Aqu
-00013f60: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
-00013f70: 7473 762e 6b73 2e74 7376 202d 7370 2073  tsv.ks.tsv -sp s
-00013f80: 7065 6369 6573 7472 6565 2e6e 7720 2d2d  peciestree.nw --
-00013f90: 7265 7765 6967 6874 202d 6f20 7767 645f  reweight -o wgd_
-00013fa0: 676c 6f62 616c 6d72 6268 5f6b 7320 2d2d  globalmrbh_ks --
-00013fb0: 7370 6169 7220 2241 7175 696c 6567 6961  spair "Aquilegia
-00013fc0: 5f63 6f65 7275 6c65 613b 5072 6f74 6561  _coerulea;Protea
-00013fd0: 5f63 796e 6172 6f69 6465 7322 202d 2d73  _cynaroides" --s
-00013fe0: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
-00013ff0: 636f 6572 756c 6561 3b56 6974 6973 5f76  coerulea;Vitis_v
-00014000: 696e 6966 6572 6122 202d 2d73 7061 6972  inifera" --spair
-00014010: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
-00014020: 756c 6561 3b41 636f 7275 735f 616d 6572  ulea;Acorus_amer
-00014030: 6963 616e 7573 2220 2d2d 7370 6169 7220  icanus" --spair 
-00014040: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-00014050: 6c65 613b 4171 7569 6c65 6769 615f 636f  lea;Aquilegia_co
-00014060: 6572 756c 6561 2220 2d2d 706c 6f74 6b64  erulea" --plotkd
-00014070: 6520 282d 6170 2077 6764 5f73 796e 2f69  e (-ap wgd_syn/i
-00014080: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
-00014090: 7270 6f69 6e74 732e 7478 7429 0a60 6060  rpoints.txt).```
-000140a0: 0a0a 5468 6520 6669 6c65 2060 7370 6563  ..The file `spec
-000140b0: 6965 7374 7265 652e 6e77 6020 6973 2074  iestree.nw` is t
-000140c0: 6865 2074 6578 7420 6669 6c65 206f 6620  he text file of 
-000140d0: 7370 6563 6965 7320 7472 6565 2069 6e20  species tree in 
-000140e0: 6e65 7769 636b 2074 6861 7420 7261 7465  newick that rate
-000140f0: 2063 6f72 7265 6374 696f 6e20 776f 756c   correction woul
-00014100: 6420 6265 2063 6f6e 6475 6374 6564 206f  d be conducted o
-00014110: 6e2e 2049 7473 2063 6f6e 7465 6e74 2069  n. Its content i
-00014120: 7320 6173 2062 656c 6f77 2e20 5573 6572  s as below. User
-00014130: 7320 6e65 6564 2074 6f20 7072 6f76 6964  s need to provid
-00014140: 6520 7468 6520 7370 6563 6965 7320 7061  e the species pa
-00014150: 6972 7320 746f 2062 6520 706c 6f74 7465  irs to be plotte
-00014160: 642e 2057 6520 7375 6767 6573 7420 6164  d. We suggest ad
-00014170: 6469 6e67 2074 6865 206f 7074 696f 6e20  ding the option 
-00014180: 602d 2d72 6577 6569 6768 7460 2074 6f20  `--reweight` to 
-00014190: 7265 6361 6c63 756c 6174 6520 7468 6520  recalculate the 
-000141a0: 7765 6967 6874 2070 6572 2073 7065 6369  weight per speci
-000141b0: 6573 2070 6169 7220 7375 6368 2074 6861  es pair such tha
-000141c0: 7420 7468 6520 7765 6967 6874 206f 6620  t the weight of 
-000141d0: 6f72 7468 6f6c 6f67 6f75 7320 6765 6e65  orthologous gene
-000141e0: 2070 6169 7273 2077 696c 6c20 6265 636f   pairs will beco
-000141f0: 6d65 2031 2061 7320 7468 6520 7061 7261  me 1 as the para
-00014200: 6c6f 676f 7573 2067 656e 6520 7061 6972  logous gene pair
-00014210: 732e 2054 6865 2066 6c61 6720 602d 2d70  s. The flag `--p
-00014220: 6c6f 746b 6465 6020 6361 6e20 6265 2061  lotkde` can be a
-00014230: 6464 6564 2077 6865 6e20 7468 6520 6b64  dded when the kd
-00014240: 6520 6375 7276 6520 6f66 206f 7274 686f  e curve of ortho
-00014250: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-00014260: 3c2f 7375 623e 2069 7320 6465 7369 7265  </sub> is desire
-00014270: 642e 2045 7874 7261 2063 6f6c 6c69 6e65  d. Extra colline
-00014280: 6172 2064 6174 6120 6361 6e20 6265 2061  ar data can be a
-00014290: 6464 6564 2062 7920 7468 6520 6f70 7469  dded by the opti
-000142a0: 6f6e 2060 2d61 7060 2e0a 0a60 6060 0a28  on `-ap`...```.(
-000142b0: 2828 5669 7469 735f 7669 6e69 6665 7261  ((Vitis_vinifera
-000142c0: 2c50 726f 7465 615f 6379 6e61 726f 6964  ,Protea_cynaroid
-000142d0: 6573 292c 4171 7569 6c65 6769 615f 636f  es),Aquilegia_co
-000142e0: 6572 756c 6561 292c 4163 6f72 7573 5f61  erulea),Acorus_a
-000142f0: 6d65 7269 6361 6e75 7329 3b0a 6060 600a  mericanus);.```.
-00014300: 0a21 5b5d 2864 6174 612f 4171 7569 6c65  .![](data/Aquile
-00014310: 6769 615f 636f 6572 756c 6561 5f47 6c6f  gia_coerulea_Glo
-00014320: 6261 6c6d 7262 684b 735f 436f 7272 6563  balmrbhKs_Correc
-00014330: 7465 642e 6b73 642e 7376 6729 0a0a 4173  ted.ksd.svg)..As
-00014340: 2073 686f 776e 2061 626f 7665 2c20 6265   shown above, be
-00014350: 6361 7573 6520 6f66 2074 6865 2068 6967  cause of the hig
-00014360: 6865 7220 7375 6273 7469 7475 7469 6f6e  her substitution
-00014370: 2072 6174 6520 6f66 202a 4171 7569 6c65   rate of *Aquile
-00014380: 6769 6120 636f 6572 756c 6561 2a2c 2074  gia coerulea*, t
-00014390: 6865 206f 7269 6769 6e61 6c20 6f72 7468  he original orth
-000143a0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
-000143b0: 533c 2f73 7562 3e20 7661 6c75 6573 2077  S</sub> values w
-000143c0: 6572 6520 6163 7475 616c 6c79 2075 6e64  ere actually und
-000143d0: 6572 6573 7469 6d61 7465 6420 696e 2074  erestimated in t
-000143e0: 6865 2074 696d 652d 6672 616d 6520 6f66  he time-frame of
-000143f0: 202a 4171 7569 6c65 6769 6120 636f 6572   *Aquilegia coer
-00014400: 756c 6561 2a2e 2057 6865 6e20 7765 2072  ulea*. When we r
-00014410: 6563 6f76 6572 6564 2074 6865 2064 6976  ecovered the div
-00014420: 6572 6765 6e63 6520 7375 6273 7469 7475  ergence substitu
-00014430: 7469 6f6e 2064 6973 7461 6e63 6520 696e  tion distance in
-00014440: 2074 6572 6d73 206f 6620 7477 6f20 7469   terms of two ti
-00014450: 6d65 7320 6f66 2074 6865 2062 7261 6e63  mes of the branc
-00014460: 682d 7370 6563 6966 6963 2063 6f6e 7472  h-specific contr
-00014470: 6962 7574 696f 6e20 6f66 202a 4171 7569  ibution of *Aqui
-00014480: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
-00014490: 7369 6e63 6520 6974 7320 6469 7665 7267  since its diverg
-000144a0: 656e 6365 2077 6974 6820 7468 6520 7369  ence with the si
-000144b0: 7374 6572 2073 7065 6369 6573 2070 6c75  ster species plu
-000144c0: 7320 7468 6520 7368 6172 6564 2073 7562  s the shared sub
-000144d0: 7374 6974 7574 696f 6e20 6469 7374 616e  stitution distan
-000144e0: 6365 2062 6566 6f72 6520 6469 7665 7267  ce before diverg
-000144f0: 656e 6365 2028 696e 2072 656c 6174 6976  ence (in relativ
-00014500: 6520 746f 2074 6865 206f 7574 6772 6f75  e to the outgrou
-00014510: 7029 2c20 7468 6520 636f 7272 6563 7465  p), the correcte
-00014520: 6420 2a4b 2a3c 7375 623e 533c 2f73 7562  d *K*<sub>S</sub
-00014530: 3e20 6d6f 6465 2062 6563 616d 6520 6c61  > mode became la
-00014540: 7267 6572 2e0a 0a49 6620 6f6e 6520 6861  rger...If one ha
-00014550: 6420 7468 6520 6f72 7468 6f6c 6f67 6f75  d the orthologou
-00014560: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
-00014570: 3e20 6461 7461 2061 6c72 6561 6479 2c20  > data already, 
-00014580: 6f6e 6520 636f 756c 6420 616c 736f 2061  one could also a
-00014590: 7070 6c79 2074 6865 2070 726f 6772 616d  pply the program
-000145a0: 2060 7767 6420 7669 7a60 2074 6f20 636f   `wgd viz` to co
-000145b0: 6e64 7563 7420 7468 6520 7261 7465 2063  nduct the rate c
-000145c0: 6f72 7265 6374 696f 6e20 616e 616c 7973  orrection analys
-000145d0: 6973 2075 7369 6e67 2074 6865 2063 6f6d  is using the com
-000145e0: 6d61 6e64 2062 656c 6f77 2e20 4e6f 7465  mand below. Note
-000145f0: 2074 6861 7420 7468 6520 6f72 6465 7220   that the order 
-00014600: 6f66 2067 6976 656e 2060 7370 6169 7260  of given `spair`
-00014610: 206f 7074 696f 6e73 2064 6563 6964 6573   options decides
-00014620: 2074 6865 2063 6f6c 6f72 206f 6620 7468   the color of th
-00014630: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-00014640: 3e20 6469 7374 7269 6275 7469 6f6e 206f  > distribution o
-00014650: 6620 6561 6368 2073 7065 6369 6573 2070  f each species p
-00014660: 6169 722e 2054 6865 2061 6464 6974 696f  air. The additio
-00014670: 6e61 6c20 6f70 7469 6f6e 2060 666f 6375  nal option `focu
-00014680: 7332 616c 6c60 2069 6e20 6077 6764 206b  s2all` in `wgd k
-00014690: 7364 6020 616e 6420 6077 6764 2076 697a  sd` and `wgd viz
-000146a0: 6020 6361 6e20 6265 2075 7365 6420 746f  ` can be used to
-000146b0: 2074 656c 6c20 7468 6520 7072 6f67 7261   tell the progra
-000146c0: 6d20 7468 6174 2074 6865 2073 7065 6369  m that the speci
-000146d0: 6573 2070 6169 7273 2061 7265 2062 6574  es pairs are bet
-000146e0: 7765 656e 2074 6865 2066 6f63 7573 2073  ween the focus s
-000146f0: 7065 6369 6573 2061 6e64 2061 6c6c 2074  pecies and all t
-00014700: 6865 206f 7468 6572 2073 7065 6369 6573  he other species
-00014710: 2073 7563 6820 7468 6174 2075 7365 7273   such that users
-00014720: 2064 6f6e 2774 206e 6565 6420 746f 2074   don't need to t
-00014730: 7970 6520 696e 2065 6163 6820 7061 6972  ype in each pair
-00014740: 2069 6e64 6976 6964 7561 6c6c 792e 0a0a   individually...
-00014750: 6060 600a 7767 6420 7669 7a20 2d64 2077  ```.wgd viz -d w
-00014760: 6764 5f67 6c6f 6261 6c6d 7262 685f 6b73  gd_globalmrbh_ks
-00014770: 2f67 6c6f 6261 6c5f 4d52 4248 2e74 7376  /global_MRBH.tsv
-00014780: 2e6b 732e 7473 7620 2d2d 6578 7472 6170  .ks.tsv --extrap
-00014790: 6172 616e 6f6d 656b 7320 7767 645f 6b73  aranomeks wgd_ks
-000147a0: 642f 4171 7569 6c65 6769 615f 636f 6572  d/Aquilegia_coer
-000147b0: 756c 6561 2e74 7376 2e6b 732e 7473 7620  ulea.tsv.ks.tsv 
-000147c0: 2d73 7020 7370 6563 6965 7374 7265 652e  -sp speciestree.
-000147d0: 6e77 202d 2d72 6577 6569 6768 7420 2d61  nw --reweight -a
-000147e0: 7020 7767 645f 7379 6e2f 6961 6468 6f72  p wgd_syn/iadhor
-000147f0: 652d 6f75 742f 616e 6368 6f72 706f 696e  e-out/anchorpoin
-00014800: 7473 2e74 7874 202d 6f20 7767 645f 7669  ts.txt -o wgd_vi
-00014810: 7a5f 6d69 7865 645f 4b73 202d 2d73 7061  z_mixed_Ks --spa
-00014820: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-00014830: 6572 756c 6561 3b50 726f 7465 615f 6379  erulea;Protea_cy
-00014840: 6e61 726f 6964 6573 2220 2d2d 7370 6169  naroides" --spai
-00014850: 7220 2241 7175 696c 6567 6961 5f63 6f65  r "Aquilegia_coe
-00014860: 7275 6c65 613b 5669 7469 735f 7669 6e69  rulea;Vitis_vini
-00014870: 6665 7261 2220 2d2d 7370 6169 7220 2241  fera" --spair "A
-00014880: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00014890: 613b 4163 6f72 7573 5f61 6d65 7269 6361  a;Acorus_america
-000148a0: 6e75 7322 202d 2d73 7061 6972 2022 4171  nus" --spair "Aq
-000148b0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000148c0: 3b41 7175 696c 6567 6961 5f63 6f65 7275  ;Aquilegia_coeru
-000148d0: 6c65 6122 202d 2d70 6c6f 746b 6465 2028  lea" --plotkde (
-000148e0: 2d2d 6773 6d61 7020 6765 6e65 5f73 7065  --gsmap gene_spe
-000148f0: 6369 6573 2e6d 6170 290a 6060 600a 0a57  cies.map).```..W
-00014900: 6974 6820 7468 6520 6f70 7469 6f6e 2060  ith the option `
-00014910: 666f 6375 7332 616c 6c60 2c20 7468 6520  focus2all`, the 
-00014920: 636f 6d6d 616e 6420 6361 6e20 6265 2061  command can be a
-00014930: 6c73 6f20 6c69 6b65 2074 6869 732c 2073  lso like this, s
-00014940: 616d 6520 7769 7468 2060 7767 6420 6b73  ame with `wgd ks
-00014950: 6460 2e0a 0a60 6060 0a77 6764 2076 697a  d`...```.wgd viz
-00014960: 202d 6420 7767 645f 676c 6f62 616c 6d72   -d wgd_globalmr
-00014970: 6268 5f6b 732f 676c 6f62 616c 5f4d 5242  bh_ks/global_MRB
-00014980: 482e 7473 762e 6b73 2e74 7376 202d 2d65  H.tsv.ks.tsv --e
-00014990: 7874 7261 7061 7261 6e6f 6d65 6b73 2077  xtraparanomeks w
-000149a0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-000149b0: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-000149c0: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
-000149d0: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
-000149e0: 6874 202d 6170 2077 6764 5f73 796e 2f69  ht -ap wgd_syn/i
-000149f0: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
-00014a00: 7270 6f69 6e74 732e 7478 7420 2d6f 2077  rpoints.txt -o w
-00014a10: 6764 5f76 697a 5f6d 6978 6564 5f4b 7320  gd_viz_mixed_Ks 
-00014a20: 2d2d 666f 6375 7332 616c 6c20 4171 7569  --focus2all Aqui
-00014a30: 6c65 6769 615f 636f 6572 756c 6561 202d  legia_coerulea -
-00014a40: 2d70 6c6f 746b 6465 0a60 6060 0a0a 4e6f  -plotkde.```..No
-00014a50: 7465 2074 6861 7420 7765 2063 616e 2065  te that we can e
-00014a60: 6173 696c 7920 7368 6f77 2074 6861 7420  asily show that 
-00014a70: 2a41 7175 696c 6567 6961 2063 6f65 7275  *Aquilegia coeru
-00014a80: 6c65 612a 2068 6173 2068 6967 6865 7220  lea* has higher 
-00014a90: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
-00014aa0: 6520 7468 616e 202a 5072 6f74 6561 2063  e than *Protea c
-00014ab0: 796e 6172 6f69 6465 732a 2061 6e64 202a  ynaroides* and *
-00014ac0: 5669 7469 7320 7669 6e69 6665 7261 2a20  Vitis vinifera* 
-00014ad0: 6279 2063 6f6d 7061 7269 6e67 2074 6865  by comparing the
-00014ae0: 6972 2073 7562 7374 6974 7574 696f 6e20  ir substitution 
-00014af0: 6469 7374 616e 6365 2069 6e20 7265 6761  distance in rega
-00014b00: 7264 2074 6f20 7468 6520 7361 6d65 2064  rd to the same d
-00014b10: 6976 6572 6765 6e63 6520 6576 656e 7420  ivergence event 
-00014b20: 7769 7468 206f 7574 6772 6f75 7020 7370  with outgroup sp
-00014b30: 6563 6965 7320 2a41 636f 7275 735f 616d  ecies *Acorus_am
-00014b40: 6572 6963 616e 7573 2a2c 2075 7369 6e67  ericanus*, using
-00014b50: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e0a   command below..
-00014b60: 0a60 6060 0a77 6764 2076 697a 202d 6420  .```.wgd viz -d 
-00014b70: 7767 645f 676c 6f62 616c 6d72 6268 5f6b  wgd_globalmrbh_k
-00014b80: 732f 676c 6f62 616c 5f4d 5242 482e 7473  s/global_MRBH.ts
-00014b90: 762e 6b73 2e74 7376 202d 7370 2073 7065  v.ks.tsv -sp spe
-00014ba0: 6369 6573 7472 6565 2e6e 7720 2d2d 7265  ciestree.nw --re
-00014bb0: 7765 6967 6874 202d 6f20 7767 645f 7669  weight -o wgd_vi
-00014bc0: 7a5f 436f 6d70 6172 655f 7261 7465 202d  z_Compare_rate -
-00014bd0: 2d73 7061 6972 2022 4163 6f72 7573 5f61  -spair "Acorus_a
-00014be0: 6d65 7269 6361 6e75 733b 5072 6f74 6561  mericanus;Protea
-00014bf0: 5f63 796e 6172 6f69 6465 7322 202d 2d73  _cynaroides" --s
-00014c00: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
-00014c10: 636f 6572 756c 6561 3b41 636f 7275 735f  coerulea;Acorus_
-00014c20: 616d 6572 6963 616e 7573 2220 2d2d 7370  americanus" --sp
-00014c30: 6169 7220 2256 6974 6973 5f76 696e 6966  air "Vitis_vinif
-00014c40: 6572 613b 4163 6f72 7573 5f61 6d65 7269  era;Acorus_ameri
-00014c50: 6361 6e75 7322 202d 2d67 736d 6170 2067  canus" --gsmap g
-00014c60: 656e 655f 7370 6563 6965 732e 6d61 7020  ene_species.map 
-00014c70: 2d2d 706c 6f74 6b64 650a 6060 600a 0a21  --plotkde.```..!
-00014c80: 5b5d 2864 6174 612f 5261 775f 4f72 7468  [](data/Raw_Orth
-00014c90: 6f6c 6f67 7565 735f 436f 6d70 6172 655f  ologues_Compare_
-00014ca0: 7261 7465 2e6b 7364 2e73 7667 290a 0a41  rate.ksd.svg)..A
-00014cb0: 7320 6469 7370 6c61 7965 6420 6162 6f76  s displayed abov
-00014cc0: 652c 2074 6865 206f 7274 686f 6c6f 676f  e, the orthologo
-00014cd0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-00014ce0: 623e 2076 616c 7565 7320 6265 7774 6565  b> values bewtee
-00014cf0: 6e20 2a41 7175 696c 6567 6961 2063 6f65  n *Aquilegia coe
-00014d00: 7275 6c65 612a 2061 6e64 202a 4163 6f72  rulea* and *Acor
-00014d10: 7573 2061 6d65 7269 6361 6e75 732a 2068  us americanus* h
-00014d20: 6173 2074 6865 2068 6967 6865 7374 206d  as the highest m
-00014d30: 6f64 652c 2069 6e64 6963 6174 696e 6774  ode, indicatingt
-00014d40: 6865 2066 6173 7465 7220 7375 6273 7469  he faster substi
-00014d50: 7475 7469 6f6e 2072 6174 6520 6f66 202a  tution rate of *
-00014d60: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
-00014d70: 6561 2a20 636f 6d70 6172 6564 2074 6f20  ea* compared to 
-00014d80: 2a50 726f 7465 6120 6379 6e61 726f 6964  *Protea cynaroid
-00014d90: 6573 2a20 616e 6420 2a56 6974 6973 2076  es* and *Vitis v
-00014da0: 696e 6966 6572 612a 2e0a 0a42 6566 6f72  inifera*...Befor
-00014db0: 6520 7632 2e30 2e32 312c 2074 6865 2067  e v2.0.21, the g
-00014dc0: 656e 652d 7370 6563 6965 7320 6d61 7020  ene-species map 
-00014dd0: 6669 6c65 2069 7320 6e65 6363 6573 7361  file is neccessa
-00014de0: 7269 6c79 206e 6565 6465 6420 666f 7220  rily needed for 
-00014df0: 6974 7320 696d 706c 656d 656e 7461 7469  its implementati
-00014e00: 6f6e 2069 6e20 6077 6764 2076 697a 602c  on in `wgd viz`,
-00014e10: 2077 6869 6368 2073 686f 756c 6420 6265   which should be
-00014e20: 2061 7574 6f6d 6174 656c 7920 7072 6f64   automately prod
-00014e30: 7563 6564 2062 7920 7468 6520 6c61 7374  uced by the last
-00014e40: 2060 7767 6420 6b73 6460 2073 7465 7020   `wgd ksd` step 
-00014e50: 6769 7665 6e20 7468 6520 6073 7061 6972  given the `spair
-00014e60: 6020 616e 6420 6073 7065 6369 6573 7472  ` and `speciestr
-00014e70: 6565 6020 7061 7261 6d65 7465 7273 2e20  ee` parameters. 
-00014e80: 5468 6520 6067 656e 655f 7370 6563 6965  The `gene_specie
-00014e90: 732e 6d61 7060 2068 6173 2063 6f6e 7465  s.map` has conte
-00014ea0: 6e74 7320 6173 2062 656c 6f77 2069 6e20  nts as below in 
-00014eb0: 7768 6963 6820 6561 6368 206c 696e 6520  which each line 
-00014ec0: 6973 2074 6865 206a 6f69 6e65 6420 7374  is the joined st
-00014ed0: 7269 6e67 206f 6620 6765 6e65 206e 616d  ring of gene nam
-00014ee0: 6520 616e 6420 7370 6563 6965 7320 6e61  e and species na
-00014ef0: 6d65 2062 7920 7370 6163 652e 2041 6674  me by space. Aft
-00014f00: 6572 2076 322e 302e 3231 2028 696e 636c  er v2.0.21 (incl
-00014f10: 7564 6564 292c 2074 6865 2067 656e 652d  uded), the gene-
-00014f20: 7370 6563 6965 7320 6d61 7020 6669 6c65  species map file
-00014f30: 2069 7320 6e6f 7420 6e65 6363 6573 7361   is not neccessa
-00014f40: 7269 6c79 206e 6565 6465 6420 616e 796d  rily needed anym
-00014f50: 6f72 652e 0a0a 6060 600a 4171 636f 6536  ore...```.Aqcoe6
-00014f60: 4730 3537 3830 302e 3120 4171 7569 6c65  G057800.1 Aquile
-00014f70: 6769 615f 636f 6572 756c 6561 0a56 7669  gia_coerulea.Vvi
-00014f80: 5f56 4954 5f32 3031 7330 3031 3167 3031  _VIT_201s0011g01
-00014f90: 3533 302e 3120 5669 7469 735f 7669 6e69  530.1 Vitis_vini
-00014fa0: 6665 7261 0a50 6379 5f50 726f 6379 3031  fera.Pcy_Procy01
-00014fb0: 6730 3835 3130 2050 726f 7465 615f 6379  g08510 Protea_cy
-00014fc0: 6e61 726f 6964 6573 0a41 616d 5f41 636f  naroides.Aam_Aco
-00014fd0: 7261 2e30 3447 3134 3239 3030 2e31 2041  ra.04G142900.1 A
-00014fe0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
-00014ff0: 0a60 6060 0a0a 4120 6d6f 7265 2063 6f6d  .```..A more com
-00015000: 706c 6578 2070 6c6f 7420 6361 6e20 6265  plex plot can be
-00015010: 206d 6164 6520 6279 2061 6464 2074 6865   made by add the
-00015020: 2066 6c61 6720 602d 2d70 6c6f 7465 6c6d   flag `--plotelm
-00015030: 6d60 2073 7563 6820 7468 6174 2074 6865  m` such that the
-00015040: 2045 4c4d 4d20 6d69 7874 7572 6520 6d6f   ELMM mixture mo
-00015050: 6465 6c69 6e67 206f 6620 7072 6f76 6964  deling of provid
-00015060: 6564 2070 6172 616e 6f6d 6520 2a4b 2a3c  ed paranome *K*<
-00015070: 7375 623e 533c 2f73 7562 3e20 6361 6e20  sub>S</sub> can 
-00015080: 6265 2073 7570 6572 696d 706f 7365 642c  be superimposed,
-00015090: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-000150a0: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
-000150b0: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
-000150c0: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
-000150d0: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
-000150e0: 7376 202d 2d65 7874 7261 7061 7261 6e6f  sv --extraparano
-000150f0: 6d65 6b73 2077 6764 5f6b 7364 2f41 7175  meks wgd_ksd/Aqu
-00015100: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
-00015110: 7473 762e 6b73 2e74 7376 202d 7370 2073  tsv.ks.tsv -sp s
-00015120: 7065 6369 6573 7472 6565 2e6e 7720 2d2d  peciestree.nw --
-00015130: 7265 7765 6967 6874 202d 6170 2077 6764  reweight -ap wgd
-00015140: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
-00015150: 2f61 6e63 686f 7270 6f69 6e74 732e 7478  /anchorpoints.tx
-00015160: 7420 2d6f 2077 6764 5f76 697a 5f6d 6978  t -o wgd_viz_mix
-00015170: 6564 5f4b 735f 656c 6d6d 202d 2d73 7061  ed_Ks_elmm --spa
-00015180: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-00015190: 6572 756c 6561 3b50 726f 7465 615f 6379  erulea;Protea_cy
-000151a0: 6e61 726f 6964 6573 2220 2d2d 7370 6169  naroides" --spai
-000151b0: 7220 2241 7175 696c 6567 6961 5f63 6f65  r "Aquilegia_coe
-000151c0: 7275 6c65 613b 5669 7469 735f 7669 6e69  rulea;Vitis_vini
-000151d0: 6665 7261 2220 2d2d 7370 6169 7220 2241  fera" --spair "A
-000151e0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-000151f0: 613b 4163 6f72 7573 5f61 6d65 7269 6361  a;Acorus_america
-00015200: 6e75 7322 202d 2d73 7061 6972 2022 4171  nus" --spair "Aq
-00015210: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00015220: 3b41 7175 696c 6567 6961 5f63 6f65 7275  ;Aquilegia_coeru
-00015230: 6c65 6122 202d 2d67 736d 6170 2067 656e  lea" --gsmap gen
-00015240: 655f 7370 6563 6965 732e 6d61 7020 2d2d  e_species.map --
-00015250: 706c 6f74 6b64 6520 2d2d 706c 6f74 656c  plotkde --plotel
-00015260: 6d6d 0a60 6060 0a0a 215b 5d28 6461 7461  mm.```..![](data
-00015270: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
-00015280: 6c65 615f 476c 6f62 616c 6d72 6268 4b73  lea_GlobalmrbhKs
-00015290: 5f45 6c6d 6d5f 436f 7272 6563 7465 642e  _Elmm_Corrected.
-000152a0: 6b73 642e 7376 6729 0a0a 4672 6f6d 2074  ksd.svg)..From t
-000152b0: 6865 206d 6978 6564 202a 4b2a 3c73 7562  he mixed *K*<sub
-000152c0: 3e53 3c2f 7375 623e 2070 6c6f 7420 6162  >S</sub> plot ab
-000152d0: 6f76 652c 2077 6520 6361 6e20 7365 6520  ove, we can see 
-000152e0: 7468 6174 2074 6865 206f 7074 696d 697a  that the optimiz
-000152f0: 6564 206c 6f67 6e6f 726d 616c 2063 6f6d  ed lognormal com
-00015300: 706f 6e65 6e74 2062 2077 6974 6820 6d6f  ponent b with mo
-00015310: 6465 2031 2e32 2069 7320 796f 756e 6765  de 1.2 is younge
-00015320: 7220 7468 616e 2074 6865 2063 6f72 7265  r than the corre
-00015330: 6374 6564 206f 7274 686f 6c6f 676f 7573  cted orthologous
-00015340: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00015350: 206d 6f64 6520 7769 7468 202a 5072 6f74   mode with *Prot
-00015360: 6561 2063 796e 6172 6f69 6465 732a 2061  ea cynaroides* a
-00015370: 6e64 202a 5669 7469 7320 7669 6e69 6665  nd *Vitis vinife
-00015380: 7261 2a20 2831 2e33 3920 616e 6420 312e  ra* (1.39 and 1.
-00015390: 3437 2c20 7265 7370 6563 7469 7665 6c79  47, respectively
-000153a0: 292e 0a0a 4265 7369 6465 732c 2077 6520  )...Besides, we 
-000153b0: 6361 6e20 616c 736f 2061 6464 2074 6865  can also add the
-000153c0: 2047 4d4d 206d 6978 7475 7265 206d 6f64   GMM mixture mod
-000153d0: 656c 696e 6720 6f66 2061 6e63 686f 7220  eling of anchor 
-000153e0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-000153f0: 7661 6c75 6573 2077 6974 6820 7468 6520  values with the 
-00015400: 666c 6167 2060 2d2d 706c 6f74 6170 676d  flag `--plotapgm
-00015410: 6d60 2c20 7573 696e 6720 7468 6520 636f  m`, using the co
-00015420: 6d6d 616e 6420 6265 6c6f 772e 0a0a 6060  mmand below...``
-00015430: 600a 7767 6420 7669 7a20 2d64 2077 6764  `.wgd viz -d wgd
-00015440: 5f67 6c6f 6261 6c6d 7262 685f 6b73 2f67  _globalmrbh_ks/g
-00015450: 6c6f 6261 6c5f 4d52 4248 2e74 7376 2e6b  lobal_MRBH.tsv.k
-00015460: 732e 7473 7620 2d2d 6578 7472 6170 6172  s.tsv --extrapar
-00015470: 616e 6f6d 656b 7320 7767 645f 6b73 642f  anomeks wgd_ksd/
-00015480: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00015490: 6561 2e74 7376 2e6b 732e 7473 7620 2d73  ea.tsv.ks.tsv -s
-000154a0: 7020 7370 6563 6965 7374 7265 652e 6e77  p speciestree.nw
-000154b0: 202d 2d72 6577 6569 6768 7420 2d61 7020   --reweight -ap 
-000154c0: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
-000154d0: 6f75 742f 616e 6368 6f72 706f 696e 7473  out/anchorpoints
-000154e0: 2e74 7874 202d 6f20 7767 645f 7669 7a5f  .txt -o wgd_viz_
-000154f0: 6d69 7865 645f 4b73 5f65 6c6d 6d20 2d2d  mixed_Ks_elmm --
-00015500: 7370 6169 7220 2241 7175 696c 6567 6961  spair "Aquilegia
-00015510: 5f63 6f65 7275 6c65 613b 5072 6f74 6561  _coerulea;Protea
-00015520: 5f63 796e 6172 6f69 6465 7322 202d 2d73  _cynaroides" --s
-00015530: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
-00015540: 636f 6572 756c 6561 3b56 6974 6973 5f76  coerulea;Vitis_v
-00015550: 696e 6966 6572 6122 202d 2d73 7061 6972  inifera" --spair
-00015560: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
-00015570: 756c 6561 3b41 636f 7275 735f 616d 6572  ulea;Acorus_amer
-00015580: 6963 616e 7573 2220 2d2d 7370 6169 7220  icanus" --spair 
-00015590: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-000155a0: 6c65 613b 4171 7569 6c65 6769 615f 636f  lea;Aquilegia_co
-000155b0: 6572 756c 6561 2220 2d2d 6773 6d61 7020  erulea" --gsmap 
-000155c0: 6765 6e65 5f73 7065 6369 6573 2e6d 6170  gene_species.map
-000155d0: 202d 2d70 6c6f 746b 6465 202d 2d70 6c6f   --plotkde --plo
-000155e0: 7461 7067 6d6d 6d0a 6060 600a 0a21 5b5d  tapgmmm.```..![]
-000155f0: 2864 6174 612f 4171 7569 6c65 6769 615f  (data/Aquilegia_
-00015600: 636f 6572 756c 6561 5f47 6c6f 6261 6c6d  coerulea_Globalm
-00015610: 7262 684b 735f 4170 676d 6d5f 436f 7272  rbhKs_Apgmm_Corr
-00015620: 6563 7465 642e 6b73 642e 7376 6729 0a0a  ected.ksd.svg)..
-00015630: 4173 206d 616e 6966 6573 7465 6420 6162  As manifested ab
-00015640: 6f76 652c 2074 6865 2061 6e63 686f 7220  ove, the anchor 
-00015650: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00015660: 636f 6d70 6f6e 656e 7420 3220 7769 7468  component 2 with
-00015670: 206d 6f64 6520 312e 3238 2069 7320 616c   mode 1.28 is al
-00015680: 736f 2079 6f75 6e67 6572 2074 6861 6e20  so younger than 
-00015690: 7468 6520 636f 7272 6563 7465 6420 6f72  the corrected or
-000156a0: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
-000156b0: 623e 533c 2f73 7562 3e20 6d6f 6465 2077  b>S</sub> mode w
-000156c0: 6974 6820 2a50 726f 7465 6120 6379 6e61  ith *Protea cyna
-000156d0: 726f 6964 6573 2a20 616e 6420 2a56 6974  roides* and *Vit
-000156e0: 6973 2076 696e 6966 6572 612a 2e20 4275  is vinifera*. Bu
-000156f0: 7420 7765 206e 6565 6420 746f 2062 6520  t we need to be 
-00015700: 6f66 2063 6f75 7273 6520 6361 7574 696f  of course cautio
-00015710: 7573 2074 6861 7420 7375 6368 2064 6973  us that such dis
-00015720: 7469 6e63 7469 6f6e 2063 6f6d 6573 2077  tinction comes w
-00015730: 6974 6820 7468 6520 756e 6365 7274 6169  ith the uncertai
-00015740: 6e74 6965 7320 696e 7472 6f64 7563 6564  nties introduced
-00015750: 2066 726f 6d20 7468 6520 6170 706c 6965   from the applie
-00015760: 6420 6d69 7874 7572 6520 6d6f 6465 6c69  d mixture modeli
-00015770: 6e67 206d 6574 686f 646f 6c6f 6779 2069  ng methodology i
-00015780: 6e20 7465 726d 7320 6f66 2066 6f72 2069  n terms of for i
-00015790: 6e73 7461 6e63 6520 6469 6666 6572 656e  nstance differen
-000157a0: 7420 696e 6974 6961 6c69 7a61 7469 6f6e  t initialization
-000157b0: 2070 6f69 6e74 7320 616e 6420 7468 6520   points and the 
-000157c0: 6973 7375 6520 6f66 206f 7665 7266 6974  issue of overfit
-000157d0: 7469 6e67 2061 6e64 2074 6865 2073 6973  ting and the sis
-000157e0: 7465 7220 7370 6563 6965 7373 2061 646f  ter speciess ado
-000157f0: 7074 6564 2069 6e20 7468 6174 2074 6865  pted in that the
-00015800: 7265 206d 6967 6874 2062 6520 7370 6563  re might be spec
-00015810: 6965 7320 7769 7468 206d 6f72 6520 6469  ies with more di
-00015820: 7370 6172 6174 6520 7375 6273 7469 7475  sparate substitu
-00015830: 7469 6f6e 2072 6174 6573 2074 6861 6e20  tion rates than 
-00015840: 7468 6520 6f6e 6520 7765 2063 686f 7365  the one we chose
-00015850: 2e0a 0a41 6464 696e 6720 626f 7468 2045  ...Adding both E
-00015860: 4c4d 4d20 7265 7375 6c74 2066 6f72 2070  LMM result for p
-00015870: 6172 616e 6f6d 6520 616e 6420 474d 4d20  aranome and GMM 
-00015880: 7265 7375 6c74 2066 6f72 2061 6e63 686f  result for ancho
-00015890: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
-000158a0: 3e20 6361 6e20 6265 2061 6368 6965 7665  > can be achieve
-000158b0: 6420 6a75 7374 2062 7920 6164 6420 7468  d just by add th
-000158c0: 6520 7477 6f20 666c 6167 7320 6d65 6e74  e two flags ment
-000158d0: 696f 6e65 6420 6162 6f76 652c 2075 7369  ioned above, usi
-000158e0: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
-000158f0: 656c 6f77 2e0a 0a60 6060 0a77 6764 2076  elow...```.wgd v
-00015900: 697a 202d 6420 7767 645f 676c 6f62 616c  iz -d wgd_global
-00015910: 6d72 6268 5f6b 732f 676c 6f62 616c 5f4d  mrbh_ks/global_M
-00015920: 5242 482e 7473 762e 6b73 2e74 7376 202d  RBH.tsv.ks.tsv -
-00015930: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
-00015940: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
-00015950: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-00015960: 6b73 2e74 7376 202d 7370 2073 7065 6369  ks.tsv -sp speci
-00015970: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
-00015980: 6967 6874 202d 6170 2077 6764 5f73 796e  ight -ap wgd_syn
-00015990: 2f69 6164 686f 7265 2d6f 7574 2f61 6e63  /iadhore-out/anc
-000159a0: 686f 7270 6f69 6e74 732e 7478 7420 2d6f  horpoints.txt -o
-000159b0: 2077 6764 5f76 697a 5f6d 6978 6564 5f4b   wgd_viz_mixed_K
-000159c0: 735f 656c 6d6d 202d 2d73 7061 6972 2022  s_elmm --spair "
-000159d0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000159e0: 6561 3b50 726f 7465 615f 6379 6e61 726f  ea;Protea_cynaro
-000159f0: 6964 6573 2220 2d2d 7370 6169 7220 2241  ides" --spair "A
-00015a00: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00015a10: 613b 5669 7469 735f 7669 6e69 6665 7261  a;Vitis_vinifera
-00015a20: 2220 2d2d 7370 6169 7220 2241 7175 696c  " --spair "Aquil
-00015a30: 6567 6961 5f63 6f65 7275 6c65 613b 4163  egia_coerulea;Ac
-00015a40: 6f72 7573 5f61 6d65 7269 6361 6e75 7322  orus_americanus"
-00015a50: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
-00015a60: 6769 615f 636f 6572 756c 6561 3b41 7175  gia_coerulea;Aqu
-00015a70: 696c 6567 6961 5f63 6f65 7275 6c65 6122  ilegia_coerulea"
-00015a80: 202d 2d67 736d 6170 2067 656e 655f 7370   --gsmap gene_sp
-00015a90: 6563 6965 732e 6d61 7020 2d2d 706c 6f74  ecies.map --plot
-00015aa0: 6b64 6520 2d2d 706c 6f74 656c 6d6d 202d  kde --plotelmm -
-00015ab0: 2d70 6c6f 7461 7067 6d6d 6d0a 6060 600a  -plotapgmmm.```.
-00015ac0: 0a21 5b5d 2864 6174 612f 4171 7569 6c65  .![](data/Aquile
-00015ad0: 6769 615f 636f 6572 756c 6561 5f47 6c6f  gia_coerulea_Glo
-00015ae0: 6261 6c6d 7262 684b 735f 456c 6d6d 5f41  balmrbhKs_Elmm_A
-00015af0: 7067 6d6d 5f43 6f72 7265 6374 6564 2e6b  pgmm_Corrected.k
-00015b00: 7364 2e73 7667 290a 0a41 6e20 616c 7465  sd.svg)..An alte
-00015b10: 726e 6174 6976 6520 7761 7920 746f 2063  rnative way to c
-00015b20: 616c 6375 6c61 7465 2074 6865 206f 7274  alculate the ort
-00015b30: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
-00015b40: 3e53 3c2f 7375 623e 2069 7320 746f 2064  >S</sub> is to d
-00015b50: 6972 6563 746c 7920 7573 6520 7468 6520  irectly use the 
-00015b60: 6f72 7468 6f67 726f 7570 7320 696e 7374  orthogroups inst
-00015b70: 6561 6420 6f66 2067 6c6f 6261 6c20 4d52  ead of global MR
-00015b80: 4248 2066 616d 696c 792e 2054 6861 7420  BH family. That 
-00015b90: 7761 7920 7765 2064 6f6e 2774 2075 7365  way we don't use
-00015ba0: 2074 6865 2070 7265 2d69 6e66 6572 7265   the pre-inferre
-00015bb0: 6420 7061 7261 6e6f 6d65 202a 4b2a 3c73  d paranome *K*<s
-00015bc0: 7562 3e53 3c2f 7375 623e 2062 7574 2074  ub>S</sub> but t
-00015bd0: 6865 2070 6172 616c 6f67 6f75 7320 6765  he paralogous ge
-00015be0: 6e65 2070 6169 7273 2069 6e73 6964 6520  ne pairs inside 
-00015bf0: 6561 6368 206f 7274 686f 6772 6f75 7020  each orthogroup 
-00015c00: 696e 7374 6561 642e 2054 6f20 6163 6869  instead. To achi
-00015c10: 6576 6520 7468 6174 2c20 7765 2066 6972  eve that, we fir
-00015c20: 7374 206e 6565 6420 746f 2069 6e66 6572  st need to infer
-00015c30: 206f 7274 686f 6772 6f75 7073 2075 7369   orthogroups usi
-00015c40: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
-00015c50: 656c 6f77 2e0a 0a60 6060 0a77 6764 2064  elow...```.wgd d
-00015c60: 6d64 2041 7175 696c 6567 6961 5f63 6f65  md Aquilegia_coe
-00015c70: 7275 6c65 6120 5072 6f74 6561 5f63 796e  rulea Protea_cyn
-00015c80: 6172 6f69 6465 7320 4163 6f72 7573 5f61  aroides Acorus_a
-00015c90: 6d65 7269 6361 6e75 7320 5669 7469 735f  mericanus Vitis_
-00015ca0: 7669 6e69 6665 7261 202d 2d6f 7274 686f  vinifera --ortho
-00015cb0: 696e 6665 7220 2d6f 2077 6764 5f6f 7274  infer -o wgd_ort
-00015cc0: 686f 2028 2d2d 6f6e 6c79 6f72 7468 6f29  ho (--onlyortho)
-00015cd0: 200a 6060 600a 0a55 7365 7273 2063 616e   .```..Users can
-00015ce0: 2064 6563 6964 6520 746f 206f 6e6c 7920   decide to only 
-00015cf0: 636f 6e64 7563 7420 7468 6520 6f72 7468  conduct the orth
-00015d00: 6f67 726f 7570 2061 6e61 6c79 7369 7320  ogroup analysis 
-00015d10: 7768 696c 6520 736b 6970 7069 6e67 206f  while skipping o
-00015d20: 7468 6572 2061 6e61 6c79 7369 7320 6279  ther analysis by
-00015d30: 2061 6464 696e 6720 7468 6520 666c 6167   adding the flag
-00015d40: 2060 2d2d 6f6e 6c79 6f72 7468 6f60 2e20   `--onlyortho`. 
-00015d50: 4e65 7874 2073 7465 7020 6973 2074 6865  Next step is the
-00015d60: 2073 616d 6520 7769 7468 2067 6c6f 6261   same with globa
-00015d70: 6c20 4d52 4248 2066 616d 696c 7920 6578  l MRBH family ex
-00015d80: 6365 7074 2074 6861 7420 7765 2064 6f6e  cept that we don
-00015d90: 2774 2075 7365 2074 6865 2065 7874 7261  't use the extra
-00015da0: 2070 7265 2d69 6e66 6572 7265 6420 7061   pre-inferred pa
-00015db0: 7261 6e6f 6d65 202a 4b2a 3c73 7562 3e53  ranome *K*<sub>S
-00015dc0: 3c2f 7375 623e 2061 6e79 6d6f 7265 2e20  </sub> anymore. 
-00015dd0: 5765 2069 6e66 6572 2074 6865 202a 4b2a  We infer the *K*
-00015de0: 3c73 7562 3e53 3c2f 7375 623e 2075 7369  <sub>S</sub> usi
-00015df0: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
-00015e00: 656c 6f77 2e20 4e6f 7465 2074 6861 7420  elow. Note that 
-00015e10: 7468 6520 7072 6f67 7261 6d20 6077 6764  the program `wgd
-00015e20: 2076 697a 6020 6361 6e20 706c 6f74 2074   viz` can plot t
-00015e30: 6865 2073 616d 6520 6a75 7374 2061 7320  he same just as 
-00015e40: 7368 6f77 6e20 6162 6f76 652e 0a0a 6060  shown above...``
-00015e50: 600a 7767 6420 6b73 6420 7767 645f 6f72  `.wgd ksd wgd_or
-00015e60: 7468 6f2f 4f72 7468 6f67 726f 7570 732e  tho/Orthogroups.
-00015e70: 7370 2e74 7376 202d 7370 2073 7065 6369  sp.tsv -sp speci
-00015e80: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
-00015e90: 6967 6874 202d 6f20 7767 645f 6f72 7468  ight -o wgd_orth
-00015ea0: 6f5f 6b73 202d 2d73 7061 6972 2022 4171  o_ks --spair "Aq
-00015eb0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00015ec0: 3b50 726f 7465 615f 6379 6e61 726f 6964  ;Protea_cynaroid
-00015ed0: 6573 2220 2d2d 7370 6169 7220 2241 7175  es" --spair "Aqu
-00015ee0: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-00015ef0: 5669 7469 735f 7669 6e69 6665 7261 2220  Vitis_vinifera" 
-00015f00: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
-00015f10: 6961 5f63 6f65 7275 6c65 613b 4163 6f72  ia_coerulea;Acor
-00015f20: 7573 5f61 6d65 7269 6361 6e75 7322 202d  us_americanus" -
-00015f30: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
-00015f40: 615f 636f 6572 756c 6561 3b41 7175 696c  a_coerulea;Aquil
-00015f50: 6567 6961 5f63 6f65 7275 6c65 6122 202d  egia_coerulea" -
-00015f60: 2d70 6c6f 746b 6465 2028 2d61 7020 7767  -plotkde (-ap wg
-00015f70: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
-00015f80: 742f 616e 6368 6f72 706f 696e 7473 2e74  t/anchorpoints.t
-00015f90: 7874 290a 6060 600a 0a21 5b5d 2864 6174  xt).```..![](dat
-00015fa0: 612f 4171 7569 6c65 6769 615f 636f 6572  a/Aquilegia_coer
-00015fb0: 756c 6561 5f4f 7274 686f 4b73 5f43 6f72  ulea_OrthoKs_Cor
-00015fc0: 7265 6374 6564 2e6b 7364 2e73 7667 290a  rected.ksd.svg).
-00015fd0: 0a41 7320 7368 6f77 6e20 6162 6f76 652c  .As shown above,
-00015fe0: 2074 6865 206e 756d 6265 7220 6f66 2062   the number of b
-00015ff0: 6f74 6820 7061 7261 6c6f 676f 7573 2067  oth paralogous g
-00016000: 656e 6520 7061 6972 7320 616e 6420 6f72  ene pairs and or
-00016010: 7468 6f6c 6f67 6f75 7320 6765 6e65 2070  thologous gene p
-00016020: 6169 7273 2069 7320 6469 6666 6572 656e  airs is differen
-00016030: 7420 7468 616e 2074 6865 206f 6e65 2066  t than the one f
-00016040: 726f 6d20 676c 6f62 616c 204d 5242 4820  rom global MRBH 
-00016050: 6661 6d69 6c79 2069 6e20 7468 6174 2068  family in that h
-00016060: 6572 6520 7765 2070 6c6f 7474 6564 2061  ere we plotted a
-00016070: 6c6c 206f 7274 686f 6c6f 676f 7573 2067  ll orthologous g
-00016080: 656e 6520 7061 6972 7320 696e 7374 6561  ene pairs instea
-00016090: 6420 6f66 206f 6e6c 7920 676c 6f62 616c  d of only global
-000160a0: 204d 5242 4820 616e 6420 706f 7465 6e74   MRBH and potent
-000160b0: 6961 6c6c 7920 6e65 7720 7061 7261 6c6f  ially new paralo
-000160c0: 676f 7573 2067 656e 6520 7061 6972 7320  gous gene pairs 
-000160d0: 6d69 6768 7420 6265 2070 726f 6475 6365  might be produce
-000160e0: 6420 696e 2074 6865 206f 7274 686f 6772  d in the orthogr
-000160f0: 6f75 7020 696e 6665 7265 6e63 6520 7374  oup inference st
-00016100: 6570 2c20 746f 6765 7468 6572 2077 6974  ep, together wit
-00016110: 6820 6469 6666 6572 656e 7420 7265 6361  h different reca
-00016120: 6c63 756c 6174 6564 2077 6569 6768 7473  lculated weights
-00016130: 2e0a 0a49 6620 6f6e 6520 6f6e 6c79 2077  ...If one only w
-00016140: 616e 7465 6420 746f 2070 6c6f 7420 7468  anted to plot th
-00016150: 6520 6f72 7468 6f6c 6f67 6f75 7320 2a4b  e orthologous *K
-00016160: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-00016170: 7374 7269 6275 7469 6f6e 2c20 6173 2077  stribution, as w
-00016180: 6520 6861 7665 2061 6c72 6561 6479 2073  e have already s
-00016190: 686f 776e 2061 626f 7665 2c20 6974 2063  hown above, it c
-000161a0: 616e 2062 6520 6561 7369 6c79 2061 6368  an be easily ach
-000161b0: 6965 7665 6420 6279 2072 656d 6f76 696e  ieved by removin
-000161c0: 6720 7468 6520 7061 7261 6c6f 676f 7573  g the paralogous
-000161d0: 2073 7065 6369 6573 2070 6169 7220 6041   species pair `A
-000161e0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-000161f0: 613b 4171 7569 6c65 6769 615f 636f 6572  a;Aquilegia_coer
-00016200: 756c 6561 602c 2075 7369 6e67 2074 6865  ulea`, using the
-00016210: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e0a   command below..
-00016220: 0a60 6060 0a77 6764 2076 697a 202d 6420  .```.wgd viz -d 
-00016230: 7767 645f 676c 6f62 616c 6d72 6268 5f6b  wgd_globalmrbh_k
-00016240: 732f 676c 6f62 616c 5f4d 5242 482e 7473  s/global_MRBH.ts
-00016250: 762e 6b73 2e74 7376 202d 7370 2073 7065  v.ks.tsv -sp spe
-00016260: 6369 6573 7472 6565 2e6e 7720 2d2d 7265  ciestree.nw --re
-00016270: 7765 6967 6874 202d 6f20 7767 645f 676c  weight -o wgd_gl
-00016280: 6f62 616c 6d72 6268 5f6f 6e6c 796f 7274  obalmrbh_onlyort
-00016290: 686f 5f6b 7320 2d2d 7370 6169 7220 2241  ho_ks --spair "A
-000162a0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-000162b0: 613b 5072 6f74 6561 5f63 796e 6172 6f69  a;Protea_cynaroi
-000162c0: 6465 7322 202d 2d73 7061 6972 2022 4171  des" --spair "Aq
-000162d0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000162e0: 3b56 6974 6973 5f76 696e 6966 6572 6122  ;Vitis_vinifera"
-000162f0: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
-00016300: 6769 615f 636f 6572 756c 6561 3b41 636f  gia_coerulea;Aco
-00016310: 7275 735f 616d 6572 6963 616e 7573 2220  rus_americanus" 
-00016320: 2d2d 6773 6d61 7020 6765 6e65 5f73 7065  --gsmap gene_spe
-00016330: 6369 6573 2e6d 6170 0a60 6060 0a0a 5765  cies.map.```..We
-00016340: 2063 616e 2063 6c65 6172 6c79 2073 6565   can clearly see
-00016350: 2074 6861 7420 2a56 6974 6973 2076 696e   that *Vitis vin
-00016360: 6966 6572 612a 2068 6173 2068 6967 6865  ifera* has highe
-00016370: 7220 7375 6273 7469 7475 7469 6f6e 2072  r substitution r
-00016380: 6174 6520 7468 616e 202a 5072 6f74 6561  ate than *Protea
-00016390: 2063 796e 6172 6f69 6465 732a 2069 6e20   cynaroides* in 
-000163a0: 7468 6174 2074 6865 6972 206f 7274 686f  that their ortho
-000163b0: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-000163c0: 3c2f 7375 623e 2070 6561 6b73 2077 6974  </sub> peaks wit
-000163d0: 6820 2a41 7175 696c 6567 6961 2063 6f65  h *Aquilegia coe
-000163e0: 7275 6c65 612a 2c20 616c 7468 6f75 6768  rulea*, although
-000163f0: 2072 6570 7265 7365 6e74 696e 6720 7468   representing th
-00016400: 6520 7361 6d65 2064 6976 6572 6765 6e63  e same divergenc
-00016410: 6520 6576 656e 742c 2064 6966 6665 7265  e event, differe
-00016420: 6420 696e 2073 7562 7374 6974 7574 696f  d in substitutio
-00016430: 6e20 6469 7374 616e 6365 2e0a 215b 5d28  n distance..![](
-00016440: 6461 7461 2f52 6177 5f4f 7274 686f 6c6f  data/Raw_Ortholo
-00016450: 6775 6573 2e6b 7364 2e73 7667 290a 0a41  gues.ksd.svg)..A
-00016460: 6674 6572 2074 6865 2070 6879 6c6f 6765  fter the phyloge
-00016470: 6e65 7469 6320 7469 6d69 6e67 206f 6620  netic timing of 
-00016480: 7468 6520 5261 6e75 6e63 756c 616c 6573  the Ranunculales
-00016490: 2057 4744 2c20 7765 2063 616e 2066 7572   WGD, we can fur
-000164a0: 7468 6572 2069 6e66 6572 2069 7473 2061  ther infer its a
-000164b0: 6273 6f6c 7574 6520 6167 652e 2046 6972  bsolute age. Fir
-000164c0: 7374 2077 6520 696e 6665 7220 7468 6520  st we infer the 
-000164d0: 6372 6564 6962 6c65 2072 616e 6765 206f  credible range o
-000164e0: 6620 616e 6368 6f72 2070 6169 7273 2062  f anchor pairs b
-000164f0: 7920 2a4b 2a3c 7375 623e 533c 2f73 7562  y *K*<sub>S</sub
-00016500: 3e20 6865 7572 6973 7469 6361 6c6c 7920  > heuristically 
-00016510: 7573 696e 6720 7468 6520 7072 6f67 7261  using the progra
-00016520: 6d20 6077 6764 2070 6561 6b60 2e0a 0a60  m `wgd peak`...`
-00016530: 6060 0a77 6764 2070 6561 6b20 2d2d 6865  ``.wgd peak --he
-00016540: 7572 6973 7469 6320 7767 645f 6b73 642f  uristic wgd_ksd/
-00016550: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00016560: 6561 2e74 7376 2e6b 732e 7473 7620 2d61  ea.tsv.ks.tsv -a
-00016570: 7020 7767 645f 7379 6e2f 6961 6468 6f72  p wgd_syn/iadhor
-00016580: 652d 6f75 742f 616e 6368 6f72 706f 696e  e-out/anchorpoin
-00016590: 7473 2e74 7874 202d 736d 2077 6764 5f73  ts.txt -sm wgd_s
-000165a0: 796e 2f69 6164 686f 7265 2d6f 7574 2f73  yn/iadhore-out/s
-000165b0: 6567 6d65 6e74 732e 7478 7420 2d6c 6520  egments.txt -le 
-000165c0: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
-000165d0: 6f75 742f 6c69 7374 5f65 6c65 6d65 6e74  out/list_element
-000165e0: 732e 7478 7420 2d6d 7020 7767 645f 7379  s.txt -mp wgd_sy
-000165f0: 6e2f 6961 6468 6f72 652d 6f75 742f 6d75  n/iadhore-out/mu
-00016600: 6c74 6970 6c69 636f 6e5f 7061 6972 732e  ltiplicon_pairs.
-00016610: 7478 7420 2d6f 2077 6764 5f70 6561 6b0a  txt -o wgd_peak.
-00016620: 6060 600a 0a21 5b5d 2864 6174 612f 416e  ```..![](data/An
-00016630: 6368 6f72 4b73 5f50 6561 6b43 495f 4171  chorKs_PeakCI_Aq
-00016640: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00016650: 2e74 7376 2e6b 732e 7473 765f 6e6f 6465  .tsv.ks.tsv_node
-00016660: 5f77 6569 6768 7465 642e 7376 6729 0a0a  _weighted.svg)..
-00016670: 4173 2073 686f 776e 2061 626f 7665 2c20  As shown above, 
-00016680: 7765 2061 7373 756d 6564 2061 206c 6f67  we assumed a log
-00016690: 6e6f 726d 616c 2064 6973 7472 6962 7574  normal distribut
-000166a0: 696f 6e20 6174 2074 6865 2070 6561 6b20  ion at the peak 
-000166b0: 6c6f 6361 7469 6f6e 2064 6574 6563 7465  location detecte
-000166c0: 6420 6279 2074 6865 2060 7369 676e 616c  d by the `signal
-000166d0: 6020 6d6f 6475 6c65 206f 6620 6073 6369  ` module of `sci
-000166e0: 7079 6020 6c69 6272 6172 792e 2054 6865  py` library. The
-000166f0: 2039 3525 2063 6f6e 6669 6465 6e63 6520   95% confidence 
-00016700: 6c65 7665 6c20 6f66 2074 6865 206c 6f67  level of the log
-00016710: 6e6f 726d 616c 2064 6973 7472 6962 7574  normal distribut
-00016720: 696f 6e20 7761 7320 6170 706c 6965 642c  ion was applied,
-00016730: 2069 2e65 2e2c 2030 2e36 382d 322e 3734   i.e., 0.68-2.74
-00016740: 2c20 696e 2066 7572 7468 6572 206d 6f6c  , in further mol
-00016750: 6563 756c 6172 2064 6174 696e 672e 2054  ecular dating. T
-00016760: 6865 2066 696c 6520 6041 7175 696c 6567  he file `Aquileg
-00016770: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-00016780: 6b73 2e74 7376 5f39 3525 4349 5f41 505f  ks.tsv_95%CI_AP_
-00016790: 666f 725f 6461 7469 6e67 5f77 6569 6768  for_dating_weigh
-000167a0: 7465 645f 666f 726d 6174 2e74 7376 6020  ted_format.tsv` 
-000167b0: 6973 2077 6861 7420 7765 206e 6565 6420  is what we need 
-000167c0: 666f 7220 6e65 7874 2073 7465 702e 2054  for next step. T
-000167d0: 6f20 6275 696c 6420 7468 6520 6f72 7468  o build the orth
-000167e0: 6f67 726f 7570 7320 7573 6564 2069 6e20  ogroups used in 
-000167f0: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
-00016800: 696e 672c 2077 6520 6e65 6564 2074 6f20  ing, we need to 
-00016810: 7365 6c65 6374 2073 6f6d 6520 7370 6563  select some spec
-00016820: 6965 7320 616e 6420 666f 726d 2061 2073  ies and form a s
-00016830: 7461 7274 696e 6720 7472 6565 2077 6974  tarting tree wit
-00016840: 6820 7072 6f70 6572 2066 6f73 7369 6c20  h proper fossil 
-00016850: 6361 6c69 6272 6174 696f 6e73 2e20 5765  calibrations. We
-00016860: 2070 726f 7669 6465 206f 6e65 2069 6e20   provide one in 
-00016870: 6d63 6d63 7472 6565 2066 6f72 6d61 7420  mcmctree format 
-00016880: 6173 2062 656c 6f77 2e0a 0a60 6060 0a31  as below...```.1
-00016890: 3720 310a 2828 2828 506f 7461 6d6f 6765  7 1.((((Potamoge
-000168a0: 746f 6e5f 6163 7574 6966 6f6c 6975 732c  ton_acutifolius,
-000168b0: 2853 7069 726f 6465 6c61 5f69 6e74 6572  (Spirodela_inter
-000168c0: 6d65 6469 612c 416d 6f72 7068 6f70 6861  media,Amorphopha
-000168d0: 6c6c 7573 5f6b 6f6e 6a61 6329 292c 2841  llus_konjac)),(A
-000168e0: 6361 6e74 686f 6368 6c61 6d79 735f 6272  canthochlamys_br
-000168f0: 6163 7465 6174 612c 2844 696f 7363 6f72  acteata,(Dioscor
-00016900: 6561 5f61 6c61 7461 2c44 696f 7363 6f72  ea_alata,Dioscor
-00016910: 6561 5f72 6f74 756e 6461 7461 2929 273e  ea_rotundata))'>
-00016920: 302e 3536 3030 3c31 2e32 3836 3327 2927  0.5600<1.2863')'
-00016930: 3e30 2e38 3336 303c 312e 3238 3633 272c  >0.8360<1.2863',
-00016940: 2841 636f 7275 735f 616d 6572 6963 616e  (Acorus_american
-00016950: 7573 2c41 636f 7275 735f 7461 7461 7269  us,Acorus_tatari
-00016960: 6e6f 7769 6929 2927 3e30 2e38 3336 303c  nowii))'>0.8360<
-00016970: 312e 3238 3633 272c 2828 2828 5465 7472  1.2863',((((Tetr
-00016980: 6163 656e 7472 6f6e 5f73 696e 656e 7365  acentron_sinense
-00016990: 2c54 726f 6368 6f64 656e 6472 6f6e 5f61  ,Trochodendron_a
-000169a0: 7261 6c69 6f69 6465 7329 2c28 4275 7875  ralioides),(Buxu
-000169b0: 735f 6175 7374 726f 7975 6e6e 616e 656e  s_austroyunnanen
-000169c0: 7369 732c 4275 7875 735f 7369 6e69 6361  sis,Buxus_sinica
-000169d0: 2929 273e 312e 3130 3830 3c31 2e32 3836  ))'>1.1080<1.286
-000169e0: 3327 2c28 4e65 6c75 6d62 6f5f 6e75 6369  3',(Nelumbo_nuci
-000169f0: 6665 7261 2c28 5465 6c6f 7065 615f 7370  fera,(Telopea_sp
-00016a00: 6563 696f 7369 7373 696d 612c 5072 6f74  eciosissima,Prot
-00016a10: 6561 5f63 796e 6172 6f69 6465 7329 2929  ea_cynaroides)))
-00016a20: 273e 312e 3130 3830 3c31 2e32 3836 3327  '>1.1080<1.2863'
-00016a30: 2c28 4171 7569 6c65 6769 615f 636f 6572  ,(Aquilegia_coer
-00016a40: 756c 6561 5f61 7031 2c41 7175 696c 6567  ulea_ap1,Aquileg
-00016a50: 6961 5f63 6f65 7275 6c65 615f 6170 3229  ia_coerulea_ap2)
-00016a60: 2927 3e31 2e31 3038 303c 312e 3238 3633  )'>1.1080<1.2863
-00016a70: 2729 273e 312e 3237 3230 3c32 2e34 3732  ')'>1.2720<2.472
-00016a80: 3027 3b0a 6060 600a 0a41 7320 7072 6573  0';.```..As pres
-00016a90: 656e 7465 6420 6162 6f76 652c 2074 6865  ented above, the
-00016aa0: 2066 6f63 7573 2073 7065 6369 6573 2074   focus species t
-00016ab0: 6861 7420 6973 2061 626f 7574 2074 6f20  hat is about to 
-00016ac0: 6265 2064 6174 6564 206e 6565 6473 2074  be dated needs t
-00016ad0: 6f20 6265 2072 6570 6c61 6365 6420 7769  o be replaced wi
-00016ae0: 7468 2060 2841 7175 696c 6567 6961 5f63  th `(Aquilegia_c
-00016af0: 6f65 7275 6c65 615f 6170 312c 4171 7569  oerulea_ap1,Aqui
-00016b00: 6c65 6769 615f 636f 6572 756c 6561 5f61  legia_coerulea_a
-00016b10: 7032 2960 2e20 5769 7468 2074 6869 7320  p2)`. With this 
-00016b20: 7374 6172 7469 6e67 2074 7265 6520 616e  starting tree an
-00016b30: 6420 7072 6564 6f77 6e6c 6f61 6465 6420  d predownloaded 
-00016b40: 6364 7320 6669 6c65 7320 6f66 2061 6c6c  cds files of all
-00016b50: 2074 6865 2073 7065 6369 6573 2c20 7765   the species, we
-00016b60: 2063 616e 2062 7569 6c64 2074 6865 206f   can build the o
-00016b70: 7274 686f 6772 6f75 7020 7573 6564 2069  rthogroup used i
-00016b80: 6e20 7468 6520 6669 6e61 6c20 6d6f 6c65  n the final mole
-00016b90: 6375 6c61 7220 6461 7469 6e67 2075 7369  cular dating usi
-00016ba0: 6e67 2074 6865 2063 6f6d 6d61 6e64 2061  ng the command a
-00016bb0: 7320 6265 6c6f 772e 0a0a 6060 600a 7767  s below...```.wg
-00016bc0: 6420 646d 6420 2d66 2041 7175 696c 6567  d dmd -f Aquileg
-00016bd0: 6961 5f63 6f65 7275 6c65 6120 2d61 7020  ia_coerulea -ap 
-00016be0: 7767 645f 7065 616b 2f41 7175 696c 6567  wgd_peak/Aquileg
-00016bf0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-00016c00: 6b73 2e74 7376 5f39 3525 4349 5f41 505f  ks.tsv_95%CI_AP_
-00016c10: 666f 725f 6461 7469 6e67 5f77 6569 6768  for_dating_weigh
-00016c20: 7465 645f 666f 726d 6174 2e74 7376 202d  ted_format.tsv -
-00016c30: 6f20 7767 645f 646d 645f 6f72 7468 6f20  o wgd_dmd_ortho 
-00016c40: 506f 7461 6d6f 6765 746f 6e5f 6163 7574  Potamogeton_acut
-00016c50: 6966 6f6c 6975 7320 5370 6972 6f64 656c  ifolius Spirodel
-00016c60: 615f 696e 7465 726d 6564 6961 2041 6d6f  a_intermedia Amo
-00016c70: 7270 686f 7068 616c 6c75 735f 6b6f 6e6a  rphophallus_konj
-00016c80: 6163 2041 6361 6e74 686f 6368 6c61 6d79  ac Acanthochlamy
-00016c90: 735f 6272 6163 7465 6174 6120 4469 6f73  s_bracteata Dios
-00016ca0: 636f 7265 615f 616c 6174 6120 4469 6f73  corea_alata Dios
-00016cb0: 636f 7265 615f 726f 7475 6e64 6174 6120  corea_rotundata 
-00016cc0: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
-00016cd0: 7320 4163 6f72 7573 5f74 6174 6172 696e  s Acorus_tatarin
-00016ce0: 6f77 6969 2054 6574 7261 6365 6e74 726f  owii Tetracentro
-00016cf0: 6e5f 7369 6e65 6e73 6520 5472 6f63 686f  n_sinense Trocho
-00016d00: 6465 6e64 726f 6e5f 6172 616c 696f 6964  dendron_aralioid
-00016d10: 6573 2042 7578 7573 5f61 7573 7472 6f79  es Buxus_austroy
-00016d20: 756e 6e61 6e65 6e73 6973 2042 7578 7573  unnanensis Buxus
-00016d30: 5f73 696e 6963 6120 4e65 6c75 6d62 6f5f  _sinica Nelumbo_
-00016d40: 6e75 6369 6665 7261 2054 656c 6f70 6561  nucifera Telopea
-00016d50: 5f73 7065 6369 6f73 6973 7369 6d61 2050  _speciosissima P
-00016d60: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
-00016d70: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
-00016d80: 6c65 610a 6060 600a 0a54 6865 2072 6573  lea.```..The res
-00016d90: 756c 7420 6669 6c65 2060 6d65 7267 655f  ult file `merge_
-00016da0: 666f 6375 735f 6170 2e74 7376 6020 6973  focus_ap.tsv` is
-00016db0: 2077 6861 7420 7765 206e 6565 6420 666f   what we need fo
-00016dc0: 7220 7468 6520 6669 6e61 6c20 7374 6570  r the final step
-00016dd0: 206f 6620 6d6f 6c65 6375 6c61 7220 6461   of molecular da
-00016de0: 7469 6e67 2069 6e20 7072 6f67 7261 6d20  ting in program 
-00016df0: 6077 6764 2066 6f63 7573 602e 0a0a 6060  `wgd focus`...``
-00016e00: 600a 7767 6420 666f 6375 7320 2d2d 7072  `.wgd focus --pr
-00016e10: 6f74 6461 7469 6e67 202d 2d61 616d 6f64  otdating --aamod
-00016e20: 656c 206c 6720 7767 645f 646d 645f 6f72  el lg wgd_dmd_or
-00016e30: 7468 6f2f 6d65 7267 655f 666f 6375 735f  tho/merge_focus_
-00016e40: 6170 2e74 7376 202d 7370 2064 6174 696e  ap.tsv -sp datin
-00016e50: 675f 7472 6565 2e6e 7720 2d6f 2077 6764  g_tree.nw -o wgd
-00016e60: 5f64 6174 696e 6720 2d64 206d 636d 6374  _dating -d mcmct
-00016e70: 7265 6520 2d64 7320 2762 7572 6e69 6e20  ree -ds 'burnin 
-00016e80: 3d20 3230 3030 2720 2d64 7320 2773 616d  = 2000' -ds 'sam
-00016e90: 7066 7265 7120 3d20 3130 3030 2720 2d64  pfreq = 1000' -d
-00016ea0: 7320 276e 7361 6d70 6c65 203d 2032 3030  s 'nsample = 200
-00016eb0: 3030 2720 506f 7461 6d6f 6765 746f 6e5f  00' Potamogeton_
-00016ec0: 6163 7574 6966 6f6c 6975 7320 5370 6972  acutifolius Spir
-00016ed0: 6f64 656c 615f 696e 7465 726d 6564 6961  odela_intermedia
-00016ee0: 2041 6d6f 7270 686f 7068 616c 6c75 735f   Amorphophallus_
-00016ef0: 6b6f 6e6a 6163 2041 6361 6e74 686f 6368  konjac Acanthoch
-00016f00: 6c61 6d79 735f 6272 6163 7465 6174 6120  lamys_bracteata 
-00016f10: 4469 6f73 636f 7265 615f 616c 6174 6120  Dioscorea_alata 
-00016f20: 4469 6f73 636f 7265 615f 726f 7475 6e64  Dioscorea_rotund
-00016f30: 6174 6120 4163 6f72 7573 5f61 6d65 7269  ata Acorus_ameri
-00016f40: 6361 6e75 7320 4163 6f72 7573 5f74 6174  canus Acorus_tat
-00016f50: 6172 696e 6f77 6969 2054 6574 7261 6365  arinowii Tetrace
-00016f60: 6e74 726f 6e5f 7369 6e65 6e73 6520 5472  ntron_sinense Tr
-00016f70: 6f63 686f 6465 6e64 726f 6e5f 6172 616c  ochodendron_aral
-00016f80: 696f 6964 6573 2042 7578 7573 5f61 7573  ioides Buxus_aus
-00016f90: 7472 6f79 756e 6e61 6e65 6e73 6973 2042  troyunnanensis B
-00016fa0: 7578 7573 5f73 696e 6963 6120 4e65 6c75  uxus_sinica Nelu
-00016fb0: 6d62 6f5f 6e75 6369 6665 7261 2054 656c  mbo_nucifera Tel
-00016fc0: 6f70 6561 5f73 7065 6369 6f73 6973 7369  opea_speciosissi
-00016fd0: 6d61 2050 726f 7465 615f 6379 6e61 726f  ma Protea_cynaro
-00016fe0: 6964 6573 2041 7175 696c 6567 6961 5f63  ides Aquilegia_c
-00016ff0: 6f65 7275 6c65 610a 6060 600a 0a48 6572  oerulea.```..Her
-00017000: 6520 7765 206f 6e6c 7920 696d 706c 656d  e we only implem
-00017010: 656e 7465 6420 7468 6520 636f 6e63 6174  ented the concat
-00017020: 656e 6174 696f 6e20 616e 616c 7973 6973  enation analysis
-00017030: 2075 7369 6e67 2070 726f 7465 696e 2073   using protein s
-00017040: 6571 7565 6e63 6520 6279 2061 6464 696e  equence by addin
-00017050: 6720 7468 6520 666c 6167 2060 2d2d 7072  g the flag `--pr
-00017060: 6f74 6461 7469 6e67 6020 616e 6420 7765  otdating` and we
-00017070: 2073 6574 2074 6865 2070 6172 616d 6574   set the paramet
-00017080: 6572 2066 6f72 2060 6d63 6d63 7472 6565  er for `mcmctree
-00017090: 6020 7669 6120 7468 6520 6f70 7469 6f6e  ` via the option
-000170a0: 2060 2d64 7360 2e20 4e6f 7465 2074 6861   `-ds`. Note tha
-000170b0: 7420 6f74 6865 7220 6461 7469 6e67 2070  t other dating p
-000170c0: 726f 6772 616d 2073 7563 6820 6173 2060  rogram such as `
-000170d0: 7238 7360 2061 6e64 2060 6265 6173 7460  r8s` and `beast`
-000170e0: 2061 7265 2061 6c73 6f20 6176 6169 6c61   are also availa
-000170f0: 626c 6520 6769 7665 6e20 736f 6d65 206d  ble given some m
-00017100: 616e 6461 746f 7279 2070 6172 616d 6574  andatory paramet
-00017110: 6572 732e 2054 6865 2066 696e 616c 206c  ers. The final l
-00017120: 6f67 206f 6620 7468 6520 7375 6363 6573  og of the succes
-00017130: 7366 756c 2072 756e 2069 7320 6173 2062  sful run is as b
-00017140: 656c 6f77 2e0a 0a60 6060 0a31 363a 3034  elow...```.16:04
-00017150: 3a32 3520 494e 464f 2020 2020 2052 756e  :25 INFO     Run
-00017160: 6e69 6e67 206d 636d 6374 7265 6520 7573  ning mcmctree us
-00017170: 696e 6720 4865 7373 6961 6e20 6d61 7472  ing Hessian matr
-00017180: 6978 206f 6620 4c47 2b47 616d 6d61 2020  ix of LG+Gamma  
-00017190: 636f 7265 2e70 793a 3936 370a 2020 2020  core.py:967.    
-000171a0: 2020 2020 2020 2020 2020 2020 2020 666f                fo
-000171b0: 7220 7072 6f74 6569 6e20 6d6f 6465 6c0a  r protein model.
-000171c0: 3233 3a34 393a 3337 2049 4e46 4f20 2020  23:49:37 INFO   
-000171d0: 2020 506f 7374 6572 696f 7220 6d65 616e    Posterior mean
-000171e0: 2066 6f72 2074 6865 2061 6765 7320 6f66   for the ages of
-000171f0: 2077 6764 2069 7320 3131 322e 3839 3435   wgd is 112.8945
-00017200: 206d 636d 6374 7265 652e 7079 3a32 3936   mcmctree.py:296
-00017210: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00017220: 2020 206d 696c 6c69 6f6e 2079 6561 7273     million years
-00017230: 2066 726f 6d20 436f 6e63 6174 656e 6174   from Concatenat
-00017240: 6564 2070 6570 7469 6465 0a20 2020 2020  ed peptide.     
-00017250: 2020 2020 2020 2020 2020 2020 2061 6c69               ali
-00017260: 676e 6d65 6e74 2061 6e64 2039 3525 2063  gnment and 95% c
-00017270: 7265 6469 6269 6c69 7479 2069 6e74 6572  redibility inter
-00017280: 7661 6c73 2028 4349 290a 2020 2020 2020  vals (CI).      
-00017290: 2020 2020 2020 2020 2020 2020 6973 2031              is 1
-000172a0: 3031 2e32 3234 2d31 3233 2e31 3231 206d  01.224-123.121 m
-000172b0: 696c 6c69 6f6e 2079 6561 7273 0a20 2020  illion years.   
-000172c0: 2020 2020 2020 494e 464f 2020 2020 2054        INFO     T
-000172d0: 6f74 616c 2072 756e 2074 696d 653a 2032  otal run time: 2
-000172e0: 3931 3735 7320 2020 2020 2020 2020 2020  9175s           
-000172f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00017300: 2020 2063 6c69 2e70 793a 3234 310a 2020     cli.py:241.  
-00017310: 2020 2020 2020 2049 4e46 4f20 2020 2020         INFO     
-00017320: 446f 6e65 2020 2020 2020 2020 2020 2020  Done            
-00017330: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00017340: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00017350: 2020 2020 636c 692e 7079 3a32 3432 0a60      cli.py:242.`
-00017360: 6060 0a0a 546f 2076 6973 7561 6c69 7a65  ``..To visualize
-00017370: 2074 6865 2064 6174 652c 2077 6520 616c   the date, we al
-00017380: 736f 2070 726f 7669 6465 6420 6120 7079  so provided a py
-00017390: 7468 6f6e 2073 6372 6970 7420 746f 2070  thon script to p
-000173a0: 6c6f 7420 7468 6520 5747 4420 6461 7465  lot the WGD date
-000173b0: 7320 696e 2074 6865 2060 7767 6460 2066  s in the `wgd` f
-000173c0: 6f6c 6465 722e 2055 7365 7273 206e 6565  older. Users nee
-000173d0: 6420 746f 2065 7874 7261 6374 2074 6865  d to extract the
-000173e0: 2072 6177 2064 6174 6573 2066 726f 6d20   raw dates from 
-000173f0: 7468 6520 606d 636d 632e 7478 7460 2066  the `mcmc.txt` f
-00017400: 6f72 2074 6865 2057 4744 206e 6f64 6520  or the WGD node 
-00017410: 6669 7273 7420 616e 6420 7361 7665 2069  first and save i
-00017420: 7420 6173 2066 696c 6520 6064 6174 6573  t as file `dates
-00017430: 2e74 7874 6020 286f 7220 7768 6174 6576  .txt` (or whatev
-00017440: 6572 2070 7265 6665 7272 6564 206e 616d  er preferred nam
-00017450: 6529 2e20 416e 2065 7861 6d70 6c65 2063  e). An example c
-00017460: 6f6d 6d61 6e64 2069 7320 6173 2062 656c  ommand is as bel
-00017470: 6f77 2e0a 0a60 6060 0a70 7974 686f 6e20  ow...```.python 
-00017480: 2450 4154 482f 706f 7374 706c 6f74 2e70  $PATH/postplot.p
-00017490: 7920 706f 7374 6469 7320 6461 7465 732e  y postdis dates.
-000174a0: 7478 7420 2d2d 7065 7263 656e 7469 6c65  txt --percentile
-000174b0: 2039 3020 2d2d 7469 746c 6520 2257 4744   90 --title "WGD
-000174c0: 2064 6174 6522 202d 2d68 7064 202d 6f20   date" --hpd -o 
-000174d0: 2252 616e 756e 6375 6c61 6c65 735f 5747  "Ranunculales_WG
-000174e0: 445f 6461 7465 2e73 7667 220a 6060 600a  D_date.svg".```.
-000174f0: 0a21 5b5d 2864 6174 612f 5261 6e75 6e63  .![](data/Ranunc
-00017500: 756c 616c 6573 5f57 4744 5f64 6174 652e  ulales_WGD_date.
-00017510: 7376 6729 0a0a 5468 6520 706f 7374 6572  svg)..The poster
-00017520: 696f 7220 6d65 616e 2c20 6d65 6469 616e  ior mean, median
-00017530: 2061 6e64 206d 6f64 6520 6f66 2074 6865   and mode of the
-00017540: 2052 616e 756e 6375 6c61 6c65 7320 5747   Ranunculales WG
-00017550: 4420 6167 6520 6973 2031 3132 2e39 322c  D age is 112.92,
-00017560: 2031 3133 2e34 3420 616e 6420 3131 322e   113.44 and 112.
-00017570: 3534 206d 7961 2c20 7769 7468 2039 3025  54 mya, with 90%
-00017580: 2048 5044 2031 3035 2e30 3720 2d20 3132   HPD 105.07 - 12
-00017590: 322e 3332 206d 7961 2061 7320 6d61 6e69  2.32 mya as mani
-000175a0: 6665 7374 6564 2061 626f 7665 2e0a 0a23  fested above...#
-000175b0: 2323 204b 7374 7265 650a 0a49 6e20 6164  ## Kstree..In ad
-000175c0: 6469 7469 6f6e 2074 6f20 7061 6972 7769  dition to pairwi
-000175d0: 7365 202a 4b2a 3c73 7562 3e53 3c2f 7375  se *K*<sub>S</su
-000175e0: 623e 2065 7374 696d 6174 696f 6e2c 2061  b> estimation, a
-000175f0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00017600: 2074 7265 6520 7769 7468 2062 7261 6e63   tree with branc
-00017610: 6820 6c65 6e67 7468 2069 6e20 2a4b 2a3c  h length in *K*<
-00017620: 7375 623e 533c 2f73 7562 3e20 756e 6974  sub>S</sub> unit
-00017630: 2063 616e 2061 6c73 6f20 6265 2064 6572   can also be der
-00017640: 6976 6564 2066 726f 6d20 7468 6520 7072  ived from the pr
-00017650: 6f67 7261 6d20 6077 6764 206b 7364 6020  ogram `wgd ksd` 
-00017660: 6769 7665 6e20 7468 6520 6f70 7469 6f6e  given the option
-00017670: 2060 2d2d 6b73 7472 6565 6020 616e 6420   `--kstree` and 
-00017680: 602d 2d73 7065 6369 6573 7472 6565 602e  `--speciestree`.
-00017690: 204e 6f74 6520 7468 6174 2074 6865 2061   Note that the a
-000176a0: 6464 6974 696f 6e61 6c20 6f70 7469 6f6e  dditional option
-000176b0: 2060 2d2d 6f6e 6c79 636f 6e63 6174 6b73   `--onlyconcatks
-000176c0: 7472 6565 6020 7769 6c6c 206f 6e6c 7920  tree` will only 
-000176d0: 6361 6c6c 2074 6865 202a 4b2a 3c73 7562  call the *K*<sub
-000176e0: 3e53 3c2f 7375 623e 2065 7374 696d 6174  >S</sub> estimat
-000176f0: 696f 6e20 666f 7220 7468 6520 636f 6e63  ion for the conc
-00017700: 6174 656e 6174 6564 2061 6c69 676e 6d65  atenated alignme
-00017710: 6e74 2072 6174 6865 7220 7468 616e 2061  nt rather than a
-00017720: 6c6c 2074 6865 2061 6c69 676e 6d65 6e74  ll the alignment
-00017730: 732e 2055 7365 7273 206e 6565 6420 746f  s. Users need to
-00017740: 2070 726f 7669 6465 2061 2070 7265 7365   provide a prese
-00017750: 7420 7370 6563 6965 7320 7472 6565 2066  t species tree f
-00017760: 6f72 2074 6865 202a 4b2a 3c73 7562 3e53  or the *K*<sub>S
-00017770: 3c2f 7375 623e 2074 7265 6520 696e 6665  </sub> tree infe
-00017780: 7265 6e63 6520 6f66 2074 6865 2063 6f6e  rence of the con
-00017790: 6361 7465 6e61 7465 6420 616c 6967 6e6d  catenated alignm
-000177a0: 656e 7420 7768 696c 6520 7468 6520 7265  ent while the re
-000177b0: 6d61 696e 696e 6720 616c 6967 6e6d 656e  maining alignmen
-000177c0: 7473 2077 696c 6c20 6265 2061 6761 696e  ts will be again
-000177d0: 7374 2061 6e20 6175 746f 6d61 7465 6c79  st an automately
-000177e0: 2069 6e66 6572 7265 6420 7472 6565 2066   inferred tree f
-000177f0: 726f 6d20 6066 6173 7474 7265 6560 206f  rom `fasttree` o
-00017800: 7220 6069 7174 7265 6560 2e20 496e 2074  r `iqtree`. In t
-00017810: 6865 2065 6e64 2c20 7573 6572 7320 7769  he end, users wi
-00017820: 6c6c 2067 6574 2061 202a 4b2a 3c73 7562  ll get a *K*<sub
-00017830: 3e53 3c2f 7375 623e 2074 7265 652c 2061  >S</sub> tree, a
-00017840: 202a 4b2a 3c73 7562 3e41 3c2f 7375 623e   *K*<sub>A</sub>
-00017850: 2074 7265 6520 616e 6420 6120 cf89 2074   tree and a .. t
-00017860: 7265 6520 7065 7220 6661 6d20 616e 6420  ree per fam and 
-00017870: 666f 7220 7468 6520 636f 6e63 6174 656e  for the concaten
-00017880: 6174 6564 2061 6c69 676e 6d65 6e74 2e0a  ated alignment..
-00017890: 0a60 6060 0a77 6764 206b 7364 2064 6174  .```.wgd ksd dat
-000178a0: 612f 6b73 7472 6565 5f64 6174 612f 6661  a/kstree_data/fa
-000178b0: 6d2e 7473 7620 6461 7461 2f6b 7374 7265  m.tsv data/kstre
-000178c0: 655f 6461 7461 2f41 636f 7275 735f 7461  e_data/Acorus_ta
-000178d0: 7461 7269 6e6f 7769 6920 6461 7461 2f6b  tarinowii data/k
-000178e0: 7374 7265 655f 6461 7461 2f41 6d62 6f72  stree_data/Ambor
-000178f0: 656c 6c61 5f74 7269 6368 6f70 6f64 6120  ella_trichopoda 
-00017900: 6461 7461 2f6b 7374 7265 655f 6461 7461  data/kstree_data
-00017910: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
-00017920: 6c65 6120 6461 7461 2f6b 7374 7265 655f  lea data/kstree_
-00017930: 6461 7461 2f41 7269 7374 6f6c 6f63 6869  data/Aristolochi
-00017940: 615f 6669 6d62 7269 6174 6120 6461 7461  a_fimbriata data
-00017950: 2f6b 7374 7265 655f 6461 7461 2f43 7963  /kstree_data/Cyc
-00017960: 6173 5f70 616e 7a68 6968 7561 656e 7369  as_panzhihuaensi
-00017970: 202d 2d6b 7374 7265 6520 2d2d 7370 6563   --kstree --spec
-00017980: 6965 7374 7265 6520 6461 7461 2f6b 7374  iestree data/kst
-00017990: 7265 655f 6461 7461 2f73 7065 6369 6573  ree_data/species
-000179a0: 5f74 7265 6531 2e6e 7720 2d2d 6f6e 6c79  _tree1.nw --only
-000179b0: 636f 6e63 6174 6b73 7472 6565 202d 6f20  concatkstree -o 
-000179c0: 7767 645f 6b73 7472 6565 5f74 6f70 6f6c  wgd_kstree_topol
-000179d0: 6f67 7931 0a60 6060 0a0a 215b 5d28 6461  ogy1.```..![](da
-000179e0: 7461 2f6b 7374 7265 655f 7265 7375 6c74  ta/kstree_result
-000179f0: 732f 6b73 7472 6565 2e73 7667 290a 0a41  s/kstree.svg)..A
-00017a00: 626f 7665 2077 6520 7573 6564 2074 6872  bove we used thr
-00017a10: 6565 2061 6c74 6572 6e61 7469 7665 2074  ee alternative t
-00017a20: 6f70 6f6c 6f67 6965 7320 746f 2069 6e66  opologies to inf
-00017a30: 6572 2074 6865 202a 4b2a 3c73 7562 3e53  er the *K*<sub>S
-00017a40: 3c2f 7375 623e 2074 7265 6520 7768 6963  </sub> tree whic
-00017a50: 6820 6c65 6420 746f 2064 6966 6665 7265  h led to differe
-00017a60: 6e74 2062 7261 6e63 6820 6c65 6e67 7468  nt branch length
-00017a70: 2065 7374 696d 6174 696f 6e2e 204e 6f74   estimation. Not
-00017a80: 6520 7468 6174 2074 6865 2066 616d 696c  e that the famil
-00017a90: 6965 7320 7765 2075 7365 6420 7765 7265  ies we used were
-00017aa0: 206f 6e6c 7920 7477 6f20 676c 6f62 616c   only two global
-00017ab0: 204d 5242 4820 6661 6d69 6c69 6573 2066   MRBH families f
-00017ac0: 6f72 2074 6865 2070 7572 706f 7365 206f  or the purpose o
-00017ad0: 6620 696c 6c75 7374 7261 7469 6f6e 2e20  f illustration. 
-00017ae0: 546f 2061 6371 7569 7265 2061 6e20 6163  To acquire an ac
-00017af0: 6375 7261 7465 2070 726f 6669 6c65 206f  curate profile o
-00017b00: 6620 7468 6520 7375 6273 7469 7475 7469  f the substituti
-00017b10: 6f6e 2072 6174 6520 7661 7269 6174 696f  on rate variatio
-00017b20: 6e2c 206f 7274 686f 6c6f 6775 6573 2061  n, orthologues a
-00017b30: 7420 7468 6520 7768 6f6c 6520 6765 6e6f  t the whole geno
-00017b40: 6d65 2073 6361 6c65 2073 686f 756c 6420  me scale should 
-00017b50: 6265 2075 7365 642e 0a0a 2323 2043 6974  be used...## Cit
-00017b60: 6174 696f 6e0a 200a 506c 6561 7365 2063  ation. .Please c
-00017b70: 6974 6520 7573 2061 7420 6874 7470 733a  ite us at https:
-00017b80: 2f2f 646f 692e 6f72 672f 3130 2e31 3030  //doi.org/10.100
-00017b90: 372f 3937 382d 312d 3037 3136 2d32 3536  7/978-1-0716-256
-00017ba0: 312d 335f 3120 616e 6420 6874 7470 733a  1-3_1 and https:
-00017bb0: 2f2f 646f 692e 6f72 672f 3130 2e31 3039  //doi.org/10.109
-00017bc0: 332f 6269 6f69 6e66 6f72 6d61 7469 6373  3/bioinformatics
-00017bd0: 2f62 7461 6532 3732 2e0a 0a60 6060 0a48  /btae272...```.H
-00017be0: 656e 6763 6869 2043 6865 6e2c 2041 7274  engchi Chen, Art
-00017bf0: 6875 7220 5a77 6165 6e65 706f 656c 2028  hur Zwaenepoel (
-00017c00: 3230 3233 292e 2049 6e66 6572 656e 6365  2023). Inference
-00017c10: 206f 6620 416e 6369 656e 7420 506f 6c79   of Ancient Poly
-00017c20: 706c 6f69 6479 2066 726f 6d20 4765 6e6f  ploidy from Geno
-00017c30: 6d69 6320 4461 7461 2e20 496e 3a20 5661  mic Data. In: Va
-00017c40: 6e20 6465 2050 6565 722c 2059 2e20 2865  n de Peer, Y. (e
-00017c50: 6473 2920 506f 6c79 706c 6f69 6479 2e20  ds) Polyploidy. 
-00017c60: 4d65 7468 6f64 7320 696e 204d 6f6c 6563  Methods in Molec
-00017c70: 756c 6172 2042 696f 6c6f 6779 2c20 766f  ular Biology, vo
-00017c80: 6c20 3235 3435 2e20 4875 6d61 6e61 2c20  l 2545. Humana, 
-00017c90: 4e65 7720 596f 726b 2c20 4e59 2e20 6874  New York, NY. ht
-00017ca0: 7470 733a 2f2f 646f 692e 6f72 672f 3130  tps://doi.org/10
-00017cb0: 2e31 3030 372f 3937 382d 312d 3037 3136  .1007/978-1-0716
-00017cc0: 2d32 3536 312d 335f 310a 4865 6e67 6368  -2561-3_1.Hengch
-00017cd0: 6920 4368 656e 2c20 4172 7468 7572 205a  i Chen, Arthur Z
-00017ce0: 7761 656e 6570 6f65 6c2c 2059 7665 7320  waenepoel, Yves 
-00017cf0: 5661 6e20 6465 2050 6565 7220 2832 3032  Van de Peer (202
-00017d00: 3429 2e20 7767 6420 7632 3a20 6120 7375  4). wgd v2: a su
-00017d10: 6974 6520 6f66 2074 6f6f 6c73 2074 6f20  ite of tools to 
-00017d20: 756e 636f 7665 7220 616e 6420 6461 7465  uncover and date
-00017d30: 2061 6e63 6965 6e74 2070 6f6c 7970 6c6f   ancient polyplo
-00017d40: 6964 7920 616e 6420 7768 6f6c 652d 6765  idy and whole-ge
-00017d50: 6e6f 6d65 2064 7570 6c69 6361 7469 6f6e  nome duplication
-00017d60: 2e20 4269 6f69 6e66 6f72 6d61 7469 6373  . Bioinformatics
-00017d70: 2c20 6874 7470 733a 2f2f 646f 692e 6f72  , https://doi.or
-00017d80: 672f 3130 2e31 3039 332f 6269 6f69 6e66  g/10.1093/bioinf
-00017d90: 6f72 6d61 7469 6373 2f62 7461 6532 3732  ormatics/btae272
-00017da0: 0a60 6060 0a0a 466f 7220 6369 7461 7469  .```..For citati
-00017db0: 6f6e 206f 6620 7468 6520 746f 6f6c 7320  on of the tools 
-00017dc0: 7573 6564 2069 6e20 7767 642c 2070 6c65  used in wgd, ple
-00017dd0: 6173 6520 636f 6e73 756c 7420 7468 6520  ase consult the 
-00017de0: 646f 6375 6d65 6e74 6174 696f 6e20 6174  documentation at
-00017df0: 0a68 7474 7073 3a2f 2f77 6764 7632 2e72  .https://wgdv2.r
-00017e00: 6561 6474 6865 646f 6373 2e69 6f2f 656e  eadthedocs.io/en
-00017e10: 2f6c 6174 6573 742f 6369 7461 7469 6f6e  /latest/citation
-00017e20: 2e68 746d 6c2e 0a0a                      .html...
+00000000: 3c64 6976 2061 6c69 676e 3d22 6365 6e74  <div align="cent
+00000010: 6572 223e 0a0a 2320 6077 6764 2076 3260  er">..# `wgd v2`
+00000020: 203a 2061 2073 7569 7465 2074 6f6f 6c20   : a suite tool 
+00000030: 6f66 2057 4744 2069 6e66 6572 656e 6365  of WGD inference
+00000040: 2061 6e64 2074 696d 696e 670a 0a5b 215b   and timing..[![
+00000050: 4275 696c 6420 5374 6174 7573 5d28 6874  Build Status](ht
+00000060: 7470 733a 2f2f 6170 702e 7472 6176 6973  tps://app.travis
+00000070: 2d63 692e 636f 6d2f 6865 6368 652d 7073  -ci.com/heche-ps
+00000080: 622f 7767 642e 7376 673f 6272 616e 6368  b/wgd.svg?branch
+00000090: 3d70 6879 6c6f 6461 7469 6e67 295d 2868  =phylodating)](h
+000000a0: 7474 7073 3a2f 2f61 7070 2e74 7261 7669  ttps://app.travi
+000000b0: 732d 6369 2e63 6f6d 2f68 6563 6865 2d70  s-ci.com/heche-p
+000000c0: 7362 2f77 6764 290a 5b21 5b44 6f63 756d  sb/wgd).[![Docum
+000000d0: 656e 7461 7469 6f6e 2053 7461 7475 735d  entation Status]
+000000e0: 2868 7474 7073 3a2f 2f72 6561 6474 6865  (https://readthe
+000000f0: 646f 6373 2e6f 7267 2f70 726f 6a65 6374  docs.org/project
+00000100: 732f 7767 6476 322f 6261 6467 652f 3f76  s/wgdv2/badge/?v
+00000110: 6572 7369 6f6e 3d6c 6174 6573 7429 5d28  ersion=latest)](
+00000120: 6874 7470 733a 2f2f 7767 6476 322e 7265  https://wgdv2.re
+00000130: 6164 7468 6564 6f63 732e 696f 2f65 6e2f  adthedocs.io/en/
+00000140: 6c61 7465 7374 2f3f 6261 6467 653d 6c61  latest/?badge=la
+00000150: 7465 7374 290a 5b21 5b6c 6963 656e 7365  test).[![license
+00000160: 5d28 6874 7470 733a 2f2f 696d 672e 7368  ](https://img.sh
+00000170: 6965 6c64 732e 696f 2f70 7970 692f 6c2f  ields.io/pypi/l/
+00000180: 7767 642e 7376 6729 5d28 6874 7470 733a  wgd.svg)](https:
+00000190: 2f2f 7079 7069 2e70 7974 686f 6e2e 6f72  //pypi.python.or
+000001a0: 672f 7079 7069 2f77 6764 290a 5b21 5b4c  g/pypi/wgd).[![L
+000001b0: 6174 6573 7420 5079 5049 2076 6572 7369  atest PyPI versi
+000001c0: 6f6e 5d28 6874 7470 733a 2f2f 696d 672e  on](https://img.
+000001d0: 7368 6965 6c64 732e 696f 2f70 7970 692f  shields.io/pypi/
+000001e0: 762f 7767 642e 7376 6729 5d28 6874 7470  v/wgd.svg)](http
+000001f0: 733a 2f2f 7079 7069 2e70 7974 686f 6e2e  s://pypi.python.
+00000200: 6f72 672f 7079 7069 2f77 6764 290a 5b21  org/pypi/wgd).[!
+00000210: 5b69 6e73 7461 6c6c 2077 6974 6820 6269  [install with bi
+00000220: 6f63 6f6e 6461 5d28 6874 7470 733a 2f2f  oconda](https://
+00000230: 696d 672e 7368 6965 6c64 732e 696f 2f62  img.shields.io/b
+00000240: 6164 6765 2f69 6e73 7461 6c6c 2532 3077  adge/install%20w
+00000250: 6974 682d 6269 6f63 6f6e 6461 2d62 7269  ith-bioconda-bri
+00000260: 6768 7467 7265 656e 2e73 7667 3f73 7479  ghtgreen.svg?sty
+00000270: 6c65 3d66 6c61 7429 5d28 6874 7470 3a2f  le=flat)](http:/
+00000280: 2f62 696f 636f 6e64 612e 6769 7468 7562  /bioconda.github
+00000290: 2e69 6f2f 7265 6369 7065 732f 7767 642f  .io/recipes/wgd/
+000002a0: 5245 4144 4d45 2e68 746d 6c29 0a5b 215b  README.html).[![
+000002b0: 446f 776e 6c6f 6164 735d 2868 7474 7073  Downloads](https
+000002c0: 3a2f 2f70 6570 792e 7465 6368 2f62 6164  ://pepy.tech/bad
+000002d0: 6765 2f77 6764 295d 2868 7474 7073 3a2f  ge/wgd)](https:/
+000002e0: 2f70 6570 792e 7465 6368 2f70 726f 6a65  /pepy.tech/proje
+000002f0: 6374 2f77 6764 290a 0a2a 2a48 656e 6763  ct/wgd)..**Hengc
+00000300: 6869 2043 6865 6e2c 2041 7274 6875 7220  hi Chen, Arthur 
+00000310: 5a77 6165 6e65 706f 656c 2c20 5976 6573  Zwaenepoel, Yves
+00000320: 2056 616e 2064 6520 5065 6572 2a2a 0a0a   Van de Peer**..
+00000330: 5b2a 2a42 696f 696e 666f 726d 6174 6963  [**Bioinformatic
+00000340: 7320 2620 4576 6f6c 7574 696f 6e61 7279  s & Evolutionary
+00000350: 2047 656e 6f6d 6963 7320 4772 6f75 702a   Genomics Group*
+00000360: 2a5d 2868 7474 7073 3a2f 2f77 7777 2e76  *](https://www.v
+00000370: 616e 6465 7065 6572 6c61 622e 6f72 672f  andepeerlab.org/
+00000380: 292a 2a2c 2056 4942 2d55 4765 6e74 2043  )**, VIB-UGent C
+00000390: 656e 7465 7220 666f 7220 506c 616e 7420  enter for Plant 
+000003a0: 5379 7374 656d 7320 4269 6f6c 6f67 792a  Systems Biology*
+000003b0: 2a0a 0a5b 2a2a 496e 7472 6f64 7563 7469  *..[**Introducti
+000003c0: 6f6e 2a2a 5d28 2369 6e74 726f 6475 6374  on**](#introduct
+000003d0: 696f 6e29 207c 200a 5b2a 2a49 6e73 7461  ion) | .[**Insta
+000003e0: 6c6c 6174 696f 6e2a 2a5d 2823 696e 7374  llation**](#inst
+000003f0: 616c 6c61 7469 6f6e 2920 7c20 0a5b 2a2a  allation) | .[**
+00000400: 5061 7261 6d65 7465 7273 2a2a 5d28 2370  Parameters**](#p
+00000410: 6172 616d 6574 6572 7329 207c 200a 5b2a  arameters) | .[*
+00000420: 2a55 7361 6765 2a2a 5d28 2375 7361 6765  *Usage**](#usage
+00000430: 2920 7c20 0a5b 2a2a 496c 6c75 7374 7261  ) | .[**Illustra
+00000440: 7469 6f6e 2a2a 5d28 2369 6c6c 7573 7472  tion**](#illustr
+00000450: 6174 696f 6e29 207c 0a5b 2a2a 446f 6375  ation) |.[**Docu
+00000460: 6d65 6e74 6174 696f 6e2a 2a5d 2868 7474  mentation**](htt
+00000470: 7073 3a2f 2f77 6764 7632 2e72 6561 6474  ps://wgdv2.readt
+00000480: 6865 646f 6373 2e69 6f2f 656e 2f6c 6174  hedocs.io/en/lat
+00000490: 6573 742f 3f62 6164 6765 3d6c 6174 6573  est/?badge=lates
+000004a0: 7429 207c 0a5b 2a2a 4369 7461 7469 6f6e  t) |.[**Citation
+000004b0: 2a2a 5d28 2363 6974 6174 696f 6e29 0a0a  **](#citation)..
+000004c0: 3c2f 6469 763e 0a0a 6077 6764 2076 3260  </div>..`wgd v2`
+000004d0: 2069 7320 6120 7079 7468 6f6e 2070 6163   is a python pac
+000004e0: 6b61 6765 2075 7067 7261 6465 6420 6672  kage upgraded fr
+000004f0: 6f6d 2074 6865 206f 7269 6769 6e61 6c20  om the original 
+00000500: 6077 6764 6020 7061 636b 6167 6520 6169  `wgd` package ai
+00000510: 6d69 6e67 2066 6f72 2074 6865 2069 6e66  ming for the inf
+00000520: 6572 656e 6365 2061 6e64 2074 696d 696e  erence and timin
+00000530: 6720 6f66 2061 6e63 6965 6e74 2077 686f  g of ancient who
+00000540: 6c65 2d67 656e 6f6d 6520 6475 706c 6963  le-genome duplic
+00000550: 6174 696f 6e20 2857 4744 2920 6576 656e  ation (WGD) even
+00000560: 7473 2e20 466f 7220 7468 6520 7072 6f70  ts. For the prop
+00000570: 6f73 6520 6f66 2069 6c6c 7573 7472 6174  ose of illustrat
+00000580: 696e 6720 7468 6520 7072 696e 6369 706c  ing the principl
+00000590: 6520 616e 6420 7573 6167 6520 6f66 2060  e and usage of `
+000005a0: 7767 6420 7632 602c 2077 6520 636f 6d70  wgd v2`, we comp
+000005b0: 696c 6564 2074 6869 7320 646f 6375 6d65  iled this docume
+000005c0: 6e74 6174 696f 6e2e 2042 656c 6f77 2077  ntation. Below w
+000005d0: 6520 6669 7273 7420 6761 7665 2061 6e20  e first gave an 
+000005e0: 696e 7472 6f64 7563 7469 6f6e 206f 7665  introduction ove
+000005f0: 7220 7468 6520 7363 6f70 6520 616e 6420  r the scope and 
+00000600: 6d65 6368 616e 6973 6d20 6f66 2060 7767  mechanism of `wg
+00000610: 6420 7632 6020 616e 6420 7468 656e 2074  d v2` and then t
+00000620: 6865 2070 7261 6374 6963 616c 2069 6e66  he practical inf
+00000630: 6f72 6d61 7469 6f6e 2070 6572 7461 696e  ormation pertain
+00000640: 696e 6720 746f 2074 6865 2069 6e73 7461  ing to the insta
+00000650: 6c6c 6174 696f 6e20 616e 6420 7573 6167  llation and usag
+00000660: 652e 2041 6e20 6578 616d 706c 6172 2077  e. An examplar w
+00000670: 6f72 6b66 6c6f 7720 6973 2070 726f 7669  orkflow is provi
+00000680: 6465 6420 696e 2074 6865 2074 7574 6f72  ded in the tutor
+00000690: 6961 6c20 7365 6374 696f 6e20 6f6e 2068  ial section on h
+000006a0: 6f77 2074 6f20 7365 656b 2065 7669 6465  ow to seek evide
+000006b0: 6e63 6520 666f 7220 6120 7075 7461 7469  nce for a putati
+000006c0: 7665 2057 4744 2065 7665 6e74 2061 6e64  ve WGD event and
+000006d0: 2070 6572 666f 726d 2070 726f 7065 7220   perform proper 
+000006e0: 7469 6d69 6e67 2077 6974 6820 6120 6672  timing with a fr
+000006f0: 6573 686c 7920 6f62 7461 696e 6564 2067  eshly obtained g
+00000700: 656e 6f6d 6520 6173 7365 6d62 6c79 2069  enome assembly i
+00000710: 6e20 6861 6e64 2e20 466f 7220 7468 6f73  n hand. For thos
+00000720: 6520 7768 6f20 6172 6520 696e 7465 7265  e who are intere
+00000730: 7374 6564 2069 6e20 6d6f 7265 2074 6865  sted in more the
+00000740: 6f72 6574 6963 616c 2064 6574 6169 6c73  oretical details
+00000750: 2c20 7765 2072 6563 6f6d 6d65 6e64 2074  , we recommend t
+00000760: 7572 6e69 6e67 2074 6f20 6f75 7220 7061  urning to our pa
+00000770: 7065 7220 616e 6420 626f 6f6b 2063 6861  per and book cha
+00000780: 7074 6572 2066 6f72 206d 6f72 6520 6465  pter for more de
+00000790: 7461 696c 6564 2064 6573 6372 6970 7469  tailed descripti
+000007a0: 6f6e 2061 6e64 2069 6e73 6967 6874 6675  on and insightfu
+000007b0: 6c20 6469 7363 7573 7369 6f6e 732e 2054  l discussions. T
+000007c0: 6865 206b 6579 2069 6d70 726f 7665 6420  he key improved 
+000007d0: 6665 6174 7572 6573 206f 6620 6077 6764  features of `wgd
+000007e0: 2076 3260 2061 7265 2064 656d 6f6e 7374   v2` are demonst
+000007f0: 7261 7465 6420 696e 206f 7572 206c 6174  rated in our lat
+00000800: 6573 7420 6d61 6e75 7363 7269 7074 206f  est manuscript o
+00000810: 6e20 5b42 696f 696e 666f 726d 6174 6963  n [Bioinformatic
+00000820: 735d 2868 7474 7073 3a2f 2f64 6f69 2e6f  s](https://doi.o
+00000830: 7267 2f31 302e 3130 3933 2f62 696f 696e  rg/10.1093/bioin
+00000840: 666f 726d 6174 6963 732f 6274 6165 3237  formatics/btae27
+00000850: 3229 2e20 4966 2079 6f75 2075 7365 2060  2). If you use `
+00000860: 7767 6420 7632 6020 696e 2079 6f75 7220  wgd v2` in your 
+00000870: 7265 7365 6172 6368 2c20 706c 6561 7365  research, please
+00000880: 2063 6974 6520 7573 2061 7320 7375 6767   cite us as sugg
+00000890: 6573 7465 6420 696e 205b 4369 7461 7469  ested in [Citati
+000008a0: 6f6e 5d28 2363 6974 6174 696f 6e29 2073  on](#citation) s
+000008b0: 6563 7469 6f6e 2e0a 0a23 2320 496e 7472  ection...## Intr
+000008c0: 6f64 7563 7469 6f6e 0a0a 506f 6c79 706c  oduction..Polypl
+000008d0: 6f69 6469 7a61 7469 6f6e 732c 2074 6865  oidizations, the
+000008e0: 2065 766f 6c75 7469 6f6e 6172 7920 7072   evolutionary pr
+000008f0: 6f63 6573 7320 7468 6174 2074 6865 2065  ocess that the e
+00000900: 6e74 6972 6520 6765 6e6f 6d65 206f 6620  ntire genome of 
+00000910: 616e 206f 7267 616e 6973 6d20 6973 2064  an organism is d
+00000920: 7570 6c69 6361 7465 642c 2061 6c73 6f20  uplicated, also 
+00000930: 6e61 6d65 6420 6173 2077 686f 6c65 2d67  named as whole-g
+00000940: 656e 6f6d 6520 6475 706c 6963 6174 696f  enome duplicatio
+00000950: 6e73 2028 5747 4473 292c 206f 6363 7572  ns (WGDs), occur
+00000960: 2072 6563 7572 7265 6e74 6c79 2061 6372   recurrently acr
+00000970: 6f73 7320 7468 6520 7472 6565 206f 6620  oss the tree of 
+00000980: 6c69 6665 2e20 5468 6572 6520 6172 6520  life. There are 
+00000990: 7477 6f20 6d6f 6465 7320 6f66 2070 6f6c  two modes of pol
+000009a0: 7970 6c6f 6964 697a 6174 696f 6e73 2c20  yploidizations, 
+000009b0: 6175 746f 706f 6c79 706c 6f69 6469 7a61  autopolyploidiza
+000009c0: 7469 6f6e 7320 616e 6420 616c 6c6f 706f  tions and allopo
+000009d0: 6c79 706c 6f69 6469 7a61 7469 6f6e 732e  lyploidizations.
+000009e0: 2041 7574 6f70 6f6c 7970 6c6f 6964 697a   Autopolyploidiz
+000009f0: 6174 696f 6e73 2061 7265 2074 6865 2064  ations are the d
+00000a00: 7570 6c69 6361 7469 6f6e 206f 6620 7468  uplication of th
+00000a10: 6520 7361 6d65 2067 656e 6f6d 652c 2072  e same genome, r
+00000a20: 6573 756c 7469 6e67 2069 6e20 7477 6f20  esulting in two 
+00000a30: 6964 656e 7469 6361 6c20 7375 6267 656e  identical subgen
+00000a40: 6f6d 6573 2061 7420 7468 6520 7469 6d65  omes at the time
+00000a50: 2069 7420 656d 6572 6765 642e 2057 6869   it emerged. Whi
+00000a60: 6c65 2074 6865 2061 6c6c 6f70 6f6c 7970  le the allopolyp
+00000a70: 6c6f 6964 697a 6174 696f 6e73 2061 7265  loidizations are
+00000a80: 206e 6f72 6d61 6c6c 7920 6163 6869 6576   normally achiev
+00000a90: 6564 2069 6e20 7477 6f20 7374 6570 732c  ed in two steps,
+00000aa0: 2066 6972 7374 2074 6865 2068 7962 7269   first the hybri
+00000ab0: 6469 7a61 7469 6f6e 2062 6574 7765 656e  dization between
+00000ac0: 2074 776f 2064 6966 6665 7265 6e74 2073   two different s
+00000ad0: 7065 6369 6573 2c20 7265 7375 6c74 696e  pecies, resultin
+00000ae0: 6720 696e 2074 6865 2061 7269 7369 6e67  g in the arising
+00000af0: 206f 6620 7472 616e 7369 656e 7420 686f   of transient ho
+00000b00: 6d6f 706c 6f69 6479 2c73 6563 6f6e 6420  moploidy,second 
+00000b10: 7468 6520 6475 706c 6963 6174 696f 6e20  the duplication 
+00000b20: 6f66 2074 6865 2068 6f6d 6f70 6c6f 6964  of the homoploid
+00000b30: 792c 2072 6573 756c 7469 6e67 2069 6e20  y, resulting in 
+00000b40: 7468 6520 656d 6572 6765 6e63 6520 6f66  the emergence of
+00000b50: 2061 6c6c 6f70 6f6c 7970 6c6f 6964 792e   allopolyploidy.
+00000b60: 2044 7565 2074 6f20 7468 6520 756e 7374   Due to the unst
+00000b70: 6162 696c 6974 7920 616e 6420 756e 6261  ability and unba
+00000b80: 6c61 6e63 6564 2074 6574 7261 736f 6d69  lanced tetrasomi
+00000b90: 6320 696e 6865 7269 7461 6e63 652c 2066  c inheritance, f
+00000ba0: 6f72 2069 6e73 7461 6e63 6520 7468 6520  or instance the 
+00000bb0: 6e75 636c 6561 722d 6379 746f 706c 6173  nuclear-cytoplas
+00000bc0: 6d69 6320 696e 636f 6d70 6174 6962 696c  mic incompatibil
+00000bd0: 6974 792c 2074 6865 2070 6f6c 7970 6c6f  ity, the polyplo
+00000be0: 6964 7920 6765 6e6f 6d65 2077 696c 6c20  idy genome will 
+00000bf0: 7468 656e 2065 7870 6572 6965 6e63 6520  then experience 
+00000c00: 6120 7072 6f63 6573 7320 6361 6c6c 6564  a process called
+00000c10: 2064 6970 6c6f 6964 697a 6174 696f 6e2c   diploidization,
+00000c20: 2061 6c73 6f20 6e61 6d65 6420 6173 2066   also named as f
+00000c30: 7261 6374 696f 6e61 7469 6f6e 2c20 6475  ractionation, du
+00000c40: 7269 6e67 2077 6869 6368 2061 206c 6172  ring which a lar
+00000c50: 6765 2070 6f72 7469 6f6e 206f 6620 6765  ge portion of ge
+00000c60: 6e65 2064 7570 6c69 6361 7465 7320 7769  ne duplicates wi
+00000c70: 6c6c 2067 6574 206c 6f73 7420 616e 6420  ll get lost and 
+00000c80: 6f6e 6c79 2061 2066 7261 6374 696f 6e20  only a fraction 
+00000c90: 6361 6e20 6265 2072 6574 6169 6e65 642e  can be retained.
+00000ca0: 2054 6865 2074 7261 6365 7320 6f66 2070   The traces of p
+00000cb0: 6f6c 7970 6c6f 6964 697a 6174 696f 6e73  olyploidizations
+00000cc0: 2063 616e 2062 6520 7468 7573 2075 6e65   can be thus une
+00000cd0: 6172 7468 6564 2066 726f 6d20 7468 6573  arthed from thes
+00000ce0: 6520 7265 7461 696e 6564 2067 656e 6520  e retained gene 
+00000cf0: 6475 706c 6963 6174 6573 2e20 5468 7265  duplicates. Thre
+00000d00: 6520 6170 7072 6f61 6368 6573 2062 6173  e approaches bas
+00000d10: 6564 206f 6e20 6765 6e65 2064 7570 6c69  ed on gene dupli
+00000d20: 6361 7465 732c 206e 616d 656c 792c 202a  cates, namely, *
+00000d30: 4b2a 3c73 7562 3e53 3c2f 7375 623e 206d  K*<sub>S</sub> m
+00000d40: 6574 686f 642c 2067 656e 6520 7472 6565  ethod, gene tree
+00000d50: 202d 2073 7065 6369 6573 2074 7265 6520   - species tree 
+00000d60: 7265 636f 6e63 696c 6961 7469 6f6e 206d  reconciliation m
+00000d70: 6574 686f 6420 616e 6420 7379 6e74 656e  ethod and synten
+00000d80: 7920 6d65 7468 6f64 2c20 6172 6520 636f  y method, are co
+00000d90: 6d6d 6f6e 6c79 2075 7365 6420 696e 2064  mmonly used in d
+00000da0: 6574 6563 7469 6e67 2065 7669 6465 6e63  etecting evidenc
+00000db0: 6520 666f 7220 5747 4473 2e20 5468 6520  e for WGDs. The 
+00000dc0: 6765 6e65 2074 7265 6520 2d20 7370 6563  gene tree - spec
+00000dd0: 6965 7320 7472 6565 2072 6563 6f6e 6369  ies tree reconci
+00000de0: 6c69 6174 696f 6e20 6d65 7468 6f64 2069  liation method i
+00000df0: 7320 6e6f 7420 7769 7468 696e 2074 6865  s not within the
+00000e00: 2073 636f 7065 206f 6620 6077 6764 2076   scope of `wgd v
+00000e10: 3260 2c20 6275 7420 7765 206b 696e 646c  2`, but we kindl
+00000e20: 7920 7265 6665 7220 7265 6164 6572 7320  y refer readers 
+00000e30: 7768 6f20 6172 6520 696e 7465 7265 7374  who are interest
+00000e40: 6564 2074 6f20 7468 6520 7068 796c 6f67  ed to the phylog
+00000e50: 656e 6f6d 6963 2070 726f 6772 616d 2064  enomic program d
+00000e60: 6576 656c 6f70 6564 2062 7920 4172 7468  eveloped by Arth
+00000e70: 7572 205a 7761 656e 6570 6f65 6c20 6e61  ur Zwaenepoel na
+00000e80: 6d65 6420 5b57 4841 4c45 5d28 6874 7470  med [WHALE](http
+00000e90: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f61  s://github.com/a
+00000ea0: 727a 7761 2f57 6861 6c65 2e6a 6c29 2061  rzwa/Whale.jl) a
+00000eb0: 6e64 2074 6865 2061 7373 6f63 6961 7465  nd the associate
+00000ec0: 6420 5b70 6170 6572 5d28 6874 7470 733a  d [paper](https:
+00000ed0: 2f2f 646f 692e 6f72 672f 3130 2e31 3039  //doi.org/10.109
+00000ee0: 332f 6d6f 6c62 6576 2f6d 737a 3038 3829  3/molbev/msz088)
+00000ef0: 2066 6f72 206d 6f72 6520 7465 6368 6e69   for more techni
+00000f00: 6361 6c20 616e 6420 7468 656f 7265 7469  cal and theoreti
+00000f10: 6361 6c20 6465 7461 696c 732e 0a0a 5468  cal details...Th
+00000f20: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+00000f30: 3e20 6d65 7468 6f64 2069 7320 6573 7461  > method is esta
+00000f40: 626c 6973 6865 6420 6f6e 2061 206d 6f64  blished on a mod
+00000f50: 656c 206f 6620 6765 6e65 2066 616d 696c  el of gene famil
+00000f60: 7920 6576 6f6c 7574 696f 6e20 7468 6174  y evolution that
+00000f70: 2065 6163 6820 6765 6e65 2066 616d 696c   each gene famil
+00000f80: 7920 6973 2061 6c6c 6f77 6564 2074 6f20  y is allowed to 
+00000f90: 6576 6f6c 7665 2076 6961 2067 656e 6520  evolve via gene 
+00000fa0: 6475 706c 6963 6174 696f 6e20 616e 6420  duplication and 
+00000fb0: 6c6f 7373 2e20 4e6f 7465 2074 6861 7420  loss. Note that 
+00000fc0: 7468 6520 6765 6e65 2066 616d 696c 7920  the gene family 
+00000fd0: 6865 7265 2069 7320 6173 7375 6d65 6420  here is assumed 
+00000fe0: 746f 2062 6520 7468 6520 636c 7573 7465  to be the cluste
+00000ff0: 7220 6f66 2061 6c6c 2067 656e 6573 2064  r of all genes d
+00001000: 6573 6365 6e64 6564 2066 726f 6d20 616e  escended from an
+00001010: 2061 6e63 6573 7472 616c 2067 656e 6520   ancestral gene 
+00001020: 696e 2061 2073 696e 676c 6520 6765 6e6f  in a single geno
+00001030: 6d65 2e20 5265 636f 7665 7269 6e67 2074  me. Recovering t
+00001040: 6865 2067 656e 6520 7472 6565 206f 6620  he gene tree of 
+00001050: 7375 6368 2067 656e 6520 6661 6d69 6c79  such gene family
+00001060: 2069 6e66 6f72 6d73 2074 6865 2074 696d   informs the tim
+00001070: 696e 672c 2073 6369 6c69 6365 7420 7468  ing, scilicet th
+00001080: 6520 6167 652c 206f 6620 6765 6e65 2064  e age, of gene d
+00001090: 7570 6c69 6361 7469 6f6e 2065 7665 6e74  uplication event
+000010a0: 732e 2054 6865 2061 6765 2072 6566 6572  s. The age refer
+000010b0: 6564 2068 6572 652c 2069 7320 6e6f 7420  ed here, is not 
+000010c0: 696e 2072 6561 6c20 6765 6f6c 6f67 6963  in real geologic
+000010d0: 616c 2074 696d 652c 2062 7574 2069 6e20  al time, but in 
+000010e0: 7468 6520 756e 6974 206f 6620 6576 6f6c  the unit of evol
+000010f0: 7574 696f 6e61 7279 2064 6973 7461 6e63  utionary distanc
+00001100: 652c 2069 2e65 2e2c 2074 6865 206e 756d  e, i.e., the num
+00001110: 6265 7220 6f66 2073 7562 7374 6974 7574  ber of substitut
+00001120: 696f 6e73 2070 6572 2073 6974 652e 2057  ions per site. W
+00001130: 6865 6e20 7468 6520 6576 6f6c 7574 696f  hen the evolutio
+00001140: 6e61 7279 2072 6174 6520 7265 6d61 696e  nary rate remain
+00001150: 7320 6170 7072 6f78 696d 6174 656c 7920  s approximately 
+00001160: 636f 6e73 7461 6e74 2c20 7468 6520 6576  constant, the ev
+00001170: 6f6c 7574 696f 6e61 7279 2064 6973 7461  olutionary dista
+00001180: 6e63 6520 6973 2074 6865 6e20 7375 7070  nce is then supp
+00001190: 6f73 6564 2074 6f20 6265 2070 726f 706f  osed to be propo
+000011a0: 7274 696f 6e61 6c20 746f 2074 6865 2072  rtional to the r
+000011b0: 6561 6c20 6576 6f6c 7574 696f 6e61 7279  eal evolutionary
+000011c0: 2074 696d 652e 2054 6865 2073 796e 6f6e   time. The synon
+000011d0: 796d 6f75 7320 6469 7374 616e 6365 202a  ymous distance *
+000011e0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2c20  K*<sub>S</sub>, 
+000011f0: 7468 6520 6e75 6d62 6572 206f 6620 7379  the number of sy
+00001200: 6e6f 6e79 6d6f 7573 2073 7562 7374 6974  nonymous substit
+00001210: 7574 696f 6e73 2070 6572 2073 796e 6f6e  utions per synon
+00001220: 796d 6f75 7320 7369 7465 2c20 6973 2073  ymous site, is s
+00001230: 7563 6820 6361 6e64 6964 6174 6520 7468  uch candidate th
+00001240: 6174 2073 796e 6f6e 796d 6f75 7320 7375  at synonymous su
+00001250: 6273 7469 7475 7469 6f6e 7320 776f 756c  bstitutions woul
+00001260: 6420 6e6f 7420 696e 6375 7220 7468 6520  d not incur the 
+00001270: 6368 616e 6765 206f 6620 616d 696e 6f20  change of amino 
+00001280: 6163 6964 2061 6e64 2061 7265 2074 6875  acid and are thu
+00001290: 7320 7265 6761 7264 6564 2061 7320 6e65  s regarded as ne
+000012a0: 7574 7261 6c2c 2077 6869 6368 2061 6363  utral, which acc
+000012b0: 6f72 6469 6e67 2074 6f20 7468 6520 6e65  ording to the ne
+000012c0: 7574 7261 6c20 7468 656f 7279 2073 686f  utral theory sho
+000012d0: 756c 6420 6f63 6375 7220 696e 2063 6f6e  uld occur in con
+000012e0: 7374 616e 7420 7261 7465 2e20 4769 7665  stant rate. Give
+000012f0: 6e20 6120 6d6f 6465 6c20 6f66 2067 656e  n a model of gen
+00001300: 6520 6661 6d69 6c79 2074 6861 7420 616c  e family that al
+00001310: 6c6f 7773 2074 6865 2067 656e 6520 746f  lows the gene to
+00001320: 2064 7570 6c69 6361 7465 2061 6e64 2067   duplicate and g
+00001330: 6574 206c 6f73 7420 696e 2061 2066 6978  et lost in a fix
+00001340: 6564 2072 6174 652c 206f 6e65 2063 616e  ed rate, one can
+00001350: 2064 6572 6976 6520 7468 6174 2074 6865   derive that the
+00001360: 2070 726f 6261 6269 6c69 7479 2064 656e   probability den
+00001370: 7369 7479 2066 756e 6374 696f 6e20 6f66  sity function of
+00001380: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+00001390: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+000013a0: 7574 696f 6e20 6f66 2072 6574 6169 6e65  ution of retaine
+000013b0: 6420 6765 6e65 2064 7570 6c69 6361 7465  d gene duplicate
+000013c0: 7320 6973 2061 2071 7561 7369 2d65 7870  s is a quasi-exp
+000013d0: 6f6e 656e 7469 616c 2066 756e 6374 696f  onential functio
+000013e0: 6e20 7468 6174 206d 6f73 7420 7265 7461  n that most reta
+000013f0: 696e 6564 2067 656e 6520 6475 706c 6963  ined gene duplic
+00001400: 6174 6573 2061 7265 2072 6563 656e 746c  ates are recentl
+00001410: 7920 626f 726e 6564 2077 6974 6820 7e30  y borned with ~0
+00001420: 2061 6765 2077 6869 6c65 2061 7320 7468   age while as th
+00001430: 6520 6167 6520 676f 696e 6720 6f6c 6465  e age going olde
+00001440: 7220 7468 6520 6173 736f 6369 6174 6564  r the associated
+00001450: 206e 756d 6265 7220 6f66 2072 6574 6169   number of retai
+00001460: 6e65 6420 6765 6e65 2064 7570 6c69 6361  ned gene duplica
+00001470: 7465 7320 6465 6361 7920 7175 6173 692d  tes decay quasi-
+00001480: 6578 706f 6e65 6e74 6961 6c6c 792e 2054  exponentially. T
+00001490: 6865 7265 666f 7265 2c20 7468 6520 6f63  herefore, the oc
+000014a0: 6375 7261 6e63 6520 6f66 206c 6172 6765  curance of large
+000014b0: 2d73 6361 6c65 2067 656e 6520 6475 706c  -scale gene dupl
+000014c0: 6963 6174 696f 6e20 6576 656e 7473 2c20  ication events, 
+000014d0: 666f 7220 696e 7374 616e 6520 5747 4473  for instane WGDs
+000014e0: 2c20 7769 7468 2076 6172 6965 6420 7265  , with varied re
+000014f0: 7465 6e74 696f 6e20 7261 7465 2c20 7769  tention rate, wi
+00001500: 6c6c 206c 6561 7665 2061 6e20 6167 6520  ll leave an age 
+00001510: 7065 616b 2066 726f 6d20 7468 6520 6275  peak from the bu
+00001520: 7273 7420 6f66 2067 656e 6520 6475 706c  rst of gene dupl
+00001530: 6963 6174 6573 2069 6e20 6120 7368 6f72  icates in a shor
+00001540: 7420 7469 6d65 2d66 7261 6d65 2075 706f  t time-frame upo
+00001550: 6e20 7468 6520 696e 6974 6961 6c20 6167  n the initial ag
+00001560: 6520 6469 7374 7269 6275 7469 6f6e 2c20  e distribution, 
+00001570: 616e 6420 6361 6e20 6265 2075 6e76 6569  and can be unvei
+00001580: 6c65 6420 6672 6f6d 206d 6978 7475 7265  led from mixture
+00001590: 206d 6f64 656c 696e 6720 616e 616c 7973   modeling analys
+000015a0: 6973 2e20 486f 7765 7665 722c 2057 4744  is. However, WGD
+000015b0: 7320 6964 656e 7469 6669 6564 2066 726f  s identified fro
+000015c0: 6d20 7468 6520 7061 7261 6c6f 676f 7573  m the paralogous
+000015d0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000015e0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
+000015f0: 6e73 2063 616e 206f 6e6c 7920 696e 666f  ns can only info
+00001600: 726d 2074 6865 2057 4744 2074 696d 696e  rm the WGD timin
+00001610: 6720 696e 2074 6865 2074 696d 652d 7363  g in the time-sc
+00001620: 616c 6520 6f66 2074 6861 7420 7370 6563  ale of that spec
+00001630: 6966 6963 2073 7065 6369 6573 2c20 7768  ific species, wh
+00001640: 6963 6820 6973 206e 6f74 2063 6f6d 7061  ich is not compa
+00001650: 7261 626c 6520 696e 2074 6865 2070 6879  rable in the phy
+00001660: 6c6f 6765 6e65 7469 6320 636f 6e74 6578  logenetic contex
+00001670: 742e 204f 6e6c 7920 7769 7468 2074 6865  t. Only with the
+00001680: 206f 7274 686f 6c6f 676f 7573 202a 4b2a   orthologous *K*
+00001690: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+000016a0: 2064 6973 7472 6962 7574 696f 6e73 2c20   distributions, 
+000016b0: 7768 6963 6820 636f 6e76 6572 7420 7468  which convert th
+000016c0: 6520 6573 7469 6d61 7465 6420 626f 6479  e estimated body
+000016d0: 2066 726f 6d20 7061 7261 6c6f 6775 6573   from paralogues
+000016e0: 2074 6f20 6f72 7468 6f6c 6f67 7565 7320   to orthologues 
+000016f0: 616e 6420 696e 666f 726d 2074 6865 2072  and inform the r
+00001700: 656c 6174 6976 6520 7469 6d69 6e67 206f  elative timing o
+00001710: 6620 7370 6563 6961 7469 6f6e 2065 7665  f speciation eve
+00001720: 6e74 732c 2063 616e 2077 6520 6465 6369  nts, can we deci
+00001730: 7068 6572 2074 6865 2070 6879 6c6f 6765  pher the phyloge
+00001740: 6e65 7469 6320 706c 6163 656d 656e 7420  netic placement 
+00001750: 6f66 2057 4744 7320 6166 7465 7220 7072  of WGDs after pr
+00001760: 6f70 6572 2072 6174 6520 636f 7272 6563  oper rate correc
+00001770: 7469 6f6e 2e20 6077 6764 2076 3260 2069  tion. `wgd v2` i
+00001780: 7320 7375 6368 2070 726f 6772 616d 2074  s such program t
+00001790: 6861 7420 6865 6c70 7320 7573 6572 7320  hat helps users 
+000017a0: 636f 6e73 7472 7563 7420 7061 7261 6c6f  construct paralo
+000017b0: 676f 7573 2061 6e64 206f 7274 686f 6c6f  gous and ortholo
+000017c0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+000017d0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+000017e0: 7574 696f 6e73 2061 6e64 2072 6561 6c69  utions and reali
+000017f0: 7a65 2062 6f74 6820 7468 6520 6964 656e  ze both the iden
+00001800: 7469 6669 6361 7469 6f6e 2061 6e64 2070  tification and p
+00001810: 6c61 6365 6d65 6e74 206f 6620 5747 4473  lacement of WGDs
+00001820: 2e0a 0a49 6e20 7468 6520 7072 656d 6973  ...In the premis
+00001830: 6520 6f66 2070 6879 6c6f 6765 6e65 7469  e of phylogeneti
+00001840: 6361 6c6c 7920 6c6f 6361 7465 6420 5747  cally located WG
+00001850: 4473 2c20 7468 6520 6162 736f 6c75 7465  Ds, the absolute
+00001860: 2061 6765 2028 696e 2067 656f 6c6f 6769   age (in geologi
+00001870: 6361 6c20 7469 6d65 2920 6f66 2057 4744  cal time) of WGD
+00001880: 7320 6361 6e20 616c 736f 2062 6520 696e  s can also be in
+00001890: 6665 7272 6564 2066 726f 6d20 7468 6f73  ferred from thos
+000018a0: 6520 5747 442d 7265 7461 696e 6564 2067  e WGD-retained g
+000018b0: 656e 6520 6475 706c 6963 6174 6573 2c20  ene duplicates, 
+000018c0: 616c 7468 6f75 6768 2074 6865 7265 2068  although there h
+000018d0: 6173 2062 6565 6e20 6e6f 2065 6173 7920  as been no easy 
+000018e0: 6f72 2073 7472 6169 6768 7466 6f72 7761  or straightforwa
+000018f0: 7264 2070 6970 656c 696e 6520 666f 7220  rd pipeline for 
+00001900: 7468 6973 206a 6f62 2073 6f20 6661 722e  this job so far.
+00001910: 2049 6e20 6077 6764 2076 3260 2c20 7765   In `wgd v2`, we
+00001920: 2064 6576 656c 6f70 6564 2061 2066 6561   developed a fea
+00001930: 7369 626c 6520 696e 7465 6772 6174 6564  sible integrated
+00001940: 2070 6970 6c65 6c69 6e65 2066 6f72 2061   pipleline for a
+00001950: 6273 6f6c 7574 6520 6461 7469 6e67 206f  bsolute dating o
+00001960: 6620 5747 4473 2e20 5468 6520 7069 7065  f WGDs. The pipe
+00001970: 6c69 6e65 2063 616e 2062 6520 726f 7567  line can be roug
+00001980: 686c 7920 6469 7669 6465 6420 696e 746f  hly divided into
+00001990: 2074 6872 6565 206d 6169 6e20 7374 6570   three main step
+000019a0: 732e 2031 2920 5468 6520 636f 6e73 7472  s. 1) The constr
+000019b0: 7563 7469 6f6e 206f 6620 616e 6368 6f72  uction of anchor
+000019c0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000019d0: 2064 6973 7472 6962 7574 696f 6e20 616e   distribution an
+000019e0: 6420 7468 6520 6465 6c69 6e65 6174 696f  d the delineatio
+000019f0: 6e20 6f66 2063 7265 6469 6162 6c65 202a  n of crediable *
+00001a00: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2072  K*<sub>S</sub> r
+00001a10: 616e 6765 2061 646f 7074 6564 2066 6f72  ange adopted for
+00001a20: 2070 6879 6c6f 6765 6e65 7469 6320 6461   phylogenetic da
+00001a30: 7469 6e67 2c20 7573 696e 6720 6077 6764  ting, using `wgd
+00001a40: 2064 6d64 602c 2060 7767 6420 6b73 6460   dmd`, `wgd ksd`
+00001a50: 2c20 6077 6764 2073 796e 6020 616e 6420  , `wgd syn` and 
+00001a60: 6077 6764 2070 6561 6b60 2e20 4e6f 7465  `wgd peak`. Note
+00001a70: 2074 6861 7420 6865 7265 2077 6520 6f6e   that here we on
+00001a80: 6c79 2063 6f6e 7369 6465 7220 6765 6e6f  ly consider geno
+00001a90: 6d65 2061 7373 656d 626c 7920 6265 6361  me assembly beca
+00001aa0: 7573 6520 666f 7220 7472 616e 7363 7269  use for transcri
+00001ab0: 7074 6f6d 6520 6173 7365 6d62 6c79 2069  ptome assembly i
+00001ac0: 7427 7320 696d 706f 7373 6962 6c65 2074  t's impossible t
+00001ad0: 6f20 6469 7374 696e 6775 6973 6820 5747  o distinguish WG
+00001ae0: 442d 6465 7269 7665 6420 6475 706c 6963  D-derived duplic
+00001af0: 6174 6573 2066 726f 6d20 736d 616c 6c2d  ates from small-
+00001b00: 7363 616c 6520 6475 706c 6963 6174 696f  scale duplicatio
+00001b10: 6e2d 6465 7269 7665 6420 6475 706c 6963  n-derived duplic
+00001b20: 6174 6573 2c20 7768 6963 6820 6861 7070  ates, which happ
+00001b30: 656e 6564 2069 6e20 6120 636f 6e74 696e  ened in a contin
+00001b40: 756f 7573 2074 696d 652d 6672 616d 6520  uous time-frame 
+00001b50: 696e 7374 6561 6420 6f66 206f 6e6c 7920  instead of only 
+00001b60: 6120 7365 7061 7261 7465 2073 686f 7274  a separate short
+00001b70: 2074 696d 652d 6672 616d 6520 616e 6420   time-frame and 
+00001b80: 7468 7573 2072 6566 6c65 6374 7320 7468  thus reflects th
+00001b90: 6520 6475 7261 7469 6f6e 206f 6620 7468  e duration of th
+00001ba0: 6174 2062 7261 6e63 6820 7261 7468 6572  at branch rather
+00001bb0: 2074 6861 6e20 7468 6520 7469 6d65 2061   than the time a
+00001bc0: 7420 7768 6963 6820 5747 4420 6f63 6375  t which WGD occu
+00001bd0: 7272 6564 2e20 3229 2054 6865 2066 6f72  rred. 2) The for
+00001be0: 6d75 6c61 7469 6f6e 206f 6620 6120 7374  mulation of a st
+00001bf0: 6172 7469 6e67 2074 7265 6520 7573 6564  arting tree used
+00001c00: 2069 6e20 7468 6520 7068 796c 6f67 656e   in the phylogen
+00001c10: 6574 6963 2064 6174 696e 672c 2063 6f6d  etic dating, com
+00001c20: 706f 7365 6420 6f66 2061 2066 6577 2073  posed of a few s
+00001c30: 7065 6369 6573 2061 6e64 2061 6e6e 6f74  pecies and annot
+00001c40: 6174 6564 2077 6974 6820 666f 7373 696c  ated with fossil
+00001c50: 2063 616c 6962 7261 7469 6f6e 2069 6e66   calibration inf
+00001c60: 6f72 6d61 7469 6f6e 2e20 5468 6973 2073  ormation. This s
+00001c70: 7465 7020 6973 2065 7373 656e 7469 616c  tep is essential
+00001c80: 2066 6f72 2074 6865 2072 6573 756c 7420   for the result 
+00001c90: 6f66 2061 6273 6f6c 7574 6520 5747 4420  of absolute WGD 
+00001ca0: 6461 7469 6e67 2074 6861 7420 7765 2073  dating that we s
+00001cb0: 7567 6765 7374 2075 7365 7273 206f 6620  uggest users of 
+00001cc0: 7461 6b69 6e67 2067 7265 6174 2063 6175  taking great cau
+00001cd0: 7469 6f6e 2074 6f20 6173 7375 7265 2074  tion to assure t
+00001ce0: 6865 2074 7265 6520 746f 706f 6c6f 6779  he tree topology
+00001cf0: 2061 6e64 2070 726f 7065 7220 626f 756e   and proper boun
+00001d00: 6473 2066 6f72 2066 6f73 7369 6c20 6361  ds for fossil ca
+00001d10: 6c69 6272 6174 696f 6e73 2e20 3329 2054  librations. 3) T
+00001d20: 6865 2063 6f6e 7374 7275 6374 696f 6e20  he construction 
+00001d30: 6f66 206f 7274 686f 6772 6f75 7073 2063  of orthogroups c
+00001d40: 6f6e 7369 7374 696e 6720 6f66 2063 6f6c  onsisting of col
+00001d50: 6c69 6e65 6172 2064 7570 6c69 6361 7465  linear duplicate
+00001d60: 7320 6f66 2074 6865 2066 6f63 616c 2073  s of the focal s
+00001d70: 7065 6369 6573 2061 6e64 2074 6865 6972  pecies and their
+00001d80: 2072 6563 6970 726f 6361 6c20 6265 7374   reciprocal best
+00001d90: 2068 6974 7320 2852 4248 7329 2061 6761   hits (RBHs) aga
+00001da0: 696e 7374 206f 7468 6572 2073 7065 6369  inst other speci
+00001db0: 6573 2069 6e20 7468 6520 7374 6172 7469  es in the starti
+00001dc0: 6e67 2074 7265 6520 616e 6420 7468 6520  ng tree and the 
+00001dd0: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
+00001de0: 696e 6720 7573 696e 6720 6120 6d6f 6c65  ing using a mole
+00001df0: 6375 6c61 7220 6461 7469 6e67 2070 726f  cular dating pro
+00001e00: 6772 616d 2066 6f72 2069 6e73 7461 6e63  gram for instanc
+00001e10: 6520 606d 636d 6374 7265 6560 2c20 7669  e `mcmctree`, vi
+00001e20: 6120 7072 6f67 7261 6d20 6077 6764 2064  a program `wgd d
+00001e30: 6d64 6020 616e 6420 6077 6764 2066 6f63  md` and `wgd foc
+00001e40: 7573 602e 2057 6520 7265 636f 6d6d 656e  us`. We recommen
+00001e50: 6420 7468 6520 7573 6167 6520 6f66 2042  d the usage of B
+00001e60: 6179 6573 6961 6e20 6d6f 6c65 6375 6c61  ayesian molecula
+00001e70: 7220 6461 7469 6e67 2070 726f 6772 616d  r dating program
+00001e80: 2060 6d63 6d63 7472 6565 602c 2077 6869   `mcmctree`, whi
+00001e90: 6368 2070 726f 7669 6465 7320 6120 7661  ch provides a va
+00001ea0: 7269 6574 7920 6f66 2064 6966 6665 7265  riety of differe
+00001eb0: 6e74 2073 7562 7374 6974 7574 696f 6e20  nt substitution 
+00001ec0: 616e 6420 7261 7465 206d 6f64 656c 732e  and rate models.
+00001ed0: 204e 6f6e 6574 6865 6c65 7373 2c20 7765   Nonetheless, we
+00001ee0: 2075 7267 6520 7573 6572 7320 746f 2073   urge users to s
+00001ef0: 6574 2070 7269 6f72 2064 6973 7472 6962  et prior distrib
+00001f00: 7574 696f 6e20 6f66 2064 6966 6665 7265  ution of differe
+00001f10: 6e74 2070 6172 616d 6574 6572 7320 7769  nt parameters wi
+00001f20: 7468 2063 6175 7469 6f6e 2061 6e64 2061  th caution and a
+00001f30: 7373 7572 6520 6164 6571 7561 7465 2073  ssure adequate s
+00001f40: 616d 706c 696e 6720 6f66 2064 6966 6665  ampling of diffe
+00001f50: 7265 6e74 2070 6172 616d 6574 6572 732e  rent parameters.
+00001f60: 0a0a 2323 2049 6e73 7461 6c6c 6174 696f  ..## Installatio
+00001f70: 6e0a 0a54 6865 2065 6173 6965 7374 2077  n..The easiest w
+00001f80: 6179 2074 6f20 696e 7374 616c 6c20 6077  ay to install `w
+00001f90: 6764 2076 3260 2069 7320 7573 696e 6720  gd v2` is using 
+00001fa0: 6050 5950 4960 2e20 4e6f 7465 2074 6861  `PYPI`. Note tha
+00001fb0: 7420 6966 2079 6f75 2077 616e 7420 746f  t if you want to
+00001fc0: 2067 6574 2074 6865 206c 6174 6573 7420   get the latest 
+00001fd0: 7570 6461 7465 2c20 7765 2073 7567 6765  update, we sugge
+00001fe0: 7374 2069 6e73 7461 6c6c 696e 6720 6672  st installing fr
+00001ff0: 6f6d 2074 6865 2073 6f75 7263 652c 2073  om the source, s
+00002000: 696e 6365 2074 6865 2075 7064 6174 6520  ince the update 
+00002010: 6f6e 2060 5059 5049 6020 7769 6c6c 2062  on `PYPI` will b
+00002020: 6520 6465 6c61 7965 6420 636f 6d70 6172  e delayed compar
+00002030: 6564 2074 6f20 6865 7265 206f 6620 736f  ed to here of so
+00002040: 7572 6365 2e0a 0a60 6060 0a70 6970 2069  urce...```.pip i
+00002050: 6e73 7461 6c6c 2077 6764 0a60 6060 0a0a  nstall wgd.```..
+00002060: 546f 2069 6e73 7461 6c6c 2060 7767 6420  To install `wgd 
+00002070: 7632 6020 696e 2061 2076 6972 7475 616c  v2` in a virtual
+00002080: 2065 6e76 6972 6f6e 6d65 6e74 2028 7768   environment (wh
+00002090: 6963 6820 7765 2073 7472 6f6e 676c 7920  ich we strongly 
+000020a0: 7265 636f 6d6d 656e 642c 206e 6f20 6d61  recommend, no ma
+000020b0: 7474 6572 2079 6f75 2069 6e73 7461 6c6c  tter you install
+000020c0: 2066 726f 6d20 736f 7572 6365 2c20 6050   from source, `P
+000020d0: 5950 4960 206f 7220 6062 696f 636f 6e64  YPI` or `biocond
+000020e0: 6160 292c 2074 6865 2066 6f6c 6c6f 7769  a`), the followi
+000020f0: 6e67 2063 6f6d 6d61 6e64 206c 696e 6573  ng command lines
+00002100: 2063 6f75 6c64 2062 6520 7573 6564 2e0a   could be used..
+00002110: 0a60 6060 0a67 6974 2063 6c6f 6e65 2068  .```.git clone h
+00002120: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
+00002130: 6d2f 6865 6368 652d 7073 622f 7767 640a  m/heche-psb/wgd.
+00002140: 6364 2077 6764 0a76 6972 7475 616c 656e  cd wgd.virtualen
+00002150: 7620 2d70 3d70 7974 686f 6e33 2045 4e56  v -p=python3 ENV
+00002160: 2028 6f72 2070 7974 686f 6e33 202d 6d20   (or python3 -m 
+00002170: 7665 6e76 2045 4e56 290a 736f 7572 6365  venv ENV).source
+00002180: 2045 4e56 2f62 696e 2f61 6374 6976 6174   ENV/bin/activat
+00002190: 650a 7069 7020 696e 7374 616c 6c20 2d72  e.pip install -r
+000021a0: 2072 6571 7569 7265 6d65 6e74 732e 7478   requirements.tx
+000021b0: 740a 7069 7020 696e 7374 616c 6c20 2e0a  t.pip install ..
+000021c0: 6060 600a 0a57 6865 6e20 6d65 7420 7769  ```..When met wi
+000021d0: 7468 2070 6572 6d69 7373 696f 6e20 7072  th permission pr
+000021e0: 6f62 6c65 6d20 696e 2069 6e73 7461 6c6c  oblem in install
+000021f0: 6174 696f 6e2c 2070 6c65 6173 6520 7472  ation, please tr
+00002200: 7920 7468 6520 666f 6c6c 6f77 696e 6720  y the following 
+00002210: 636f 6d6d 616e 6420 6c69 6e65 2e0a 0a60  command line...`
+00002220: 6060 0a70 6970 2069 6e73 7461 6c6c 202d  ``.pip install -
+00002230: 6520 2e0a 6060 600a 0a49 6620 6d75 6c74  e ..```..If mult
+00002240: 6970 6c79 2076 6572 7369 6f6e 7320 6f66  iply versions of
+00002250: 2060 7767 6460 2077 6572 6520 696e 7374   `wgd` were inst
+00002260: 616c 6c65 6420 696e 2074 6865 2073 7973  alled in the sys
+00002270: 7465 6d2c 2070 6c65 6173 6520 6164 6420  tem, please add 
+00002280: 7468 6520 7269 6768 7420 7061 7468 206f  the right path o
+00002290: 6620 696e 7465 7265 7374 6564 2076 6572  f interested ver
+000022a0: 7369 6f6e 2069 6e74 6f20 7468 6520 656e  sion into the en
+000022b0: 7669 726f 6e6d 656e 7420 7661 7269 6162  vironment variab
+000022c0: 6c65 732c 2066 6f72 2065 7861 6d70 6c65  les, for example
+000022d0: 0a0a 6060 600a 6578 706f 7274 2050 4154  ..```.export PAT
+000022e0: 483d 2224 5041 5448 3a7e 2f2e 6c6f 6361  H="$PATH:~/.loca
+000022f0: 6c2f 6269 6e2f 7767 6422 0a60 6060 0a0a  l/bin/wgd".```..
+00002300: 4e6f 7465 2074 6861 7420 7468 6520 7665  Note that the ve
+00002310: 7273 696f 6e20 6f66 2060 6e75 6d70 7960  rsion of `numpy`
+00002320: 2069 7320 696d 706f 7274 616e 7420 2866   is important (f
+00002330: 6f72 206d 616e 7920 6f74 6865 7220 7061  or many other pa
+00002340: 636b 6167 6573 2061 7265 2074 6865 2073  ckages are the s
+00002350: 616d 6520 6f66 2063 6f75 7273 6529 2c20  ame of course), 
+00002360: 6573 7065 6369 616c 6c79 2066 6f72 2060  especially for `
+00002370: 6661 7374 636c 7573 7465 7260 2070 6163  fastcluster` pac
+00002380: 6b61 6765 2e20 496e 206f 7572 2074 6573  kage. In our tes
+00002390: 742c 2074 6865 2060 6e75 6d70 7960 2031  t, the `numpy` 1
+000023a0: 2e31 392e 3020 776f 726b 7320 6669 6e65  .19.0 works fine
+000023b0: 206f 6e20 6070 7974 686f 6e33 2e36 2f38   on `python3.6/8
+000023c0: 602e 2049 6620 796f 7520 6d65 7420 736f  `. If you met so
+000023d0: 6d65 2065 7272 6f72 7320 6f72 2077 6172  me errors or war
+000023e0: 6e69 6e67 7320 6162 6f75 7420 606e 756d  nings about `num
+000023f0: 7079 602c 206d 6179 6265 2063 6f6e 7369  py`, maybe consi
+00002400: 6465 7269 6e67 2070 7265 2d69 6e73 7461  dering pre-insta
+00002410: 6c6c 2060 6e75 6d70 7960 2061 7320 312e  ll `numpy` as 1.
+00002420: 3139 2e30 206f 7220 6f74 6865 7220 636c  19.0 or other cl
+00002430: 6f73 652d 6279 2076 6572 7369 6f6e 7320  ose-by versions 
+00002440: 6265 666f 7265 2079 6f75 2069 6e73 7461  before you insta
+00002450: 6c6c 2060 7767 6460 2e20 6077 6764 6020  ll `wgd`. `wgd` 
+00002460: 7265 6c69 6573 206f 6e20 6578 7465 726e  relies on extern
+00002470: 616c 2073 6f66 7477 6172 6573 2069 6e63  al softwares inc
+00002480: 6c75 6469 6e67 2060 6469 616d 6f6e 6460  luding `diamond`
+00002490: 2061 6e64 2060 6d63 6c60 2066 6f72 2060   and `mcl` for `
+000024a0: 7767 6420 646d 6460 2c20 6070 616d 6c20  wgd dmd`, `paml 
+000024b0: 7634 2e39 6a60 2c20 606d 6166 6674 6020  v4.9j`, `mafft` 
+000024c0: 2860 6d75 7363 6c65 6020 616e 6420 6070  (`muscle` and `p
+000024d0: 7261 6e6b 6020 6966 2073 6574 292c 2060  rank` if set), `
+000024e0: 6661 7374 7472 6565 6020 286f 7220 6069  fasttree` (or `i
+000024f0: 7174 7265 6560 2069 6620 7365 7429 2066  qtree` if set) f
+00002500: 6f72 2060 7767 6420 6b73 6460 2061 6e64  or `wgd ksd` and
+00002510: 206f 7074 696f 6e61 6c6c 7920 606d 7262   optionally `mrb
+00002520: 6179 6573 6020 666f 7220 7468 6520 7068  ayes` for the ph
+00002530: 796c 6f67 656e 6574 6963 2069 6e66 6572  ylogenetic infer
+00002540: 656e 6365 2066 756e 6374 696f 6e20 696e  ence function in
+00002550: 2060 7767 6420 646d 6460 2061 6e64 2060   `wgd dmd` and `
+00002560: 7767 6420 666f 6375 7360 2028 606d 6166  wgd focus` (`maf
+00002570: 6674 602c 2060 6d75 7363 6c65 6020 616e  ft`, `muscle` an
+00002580: 6420 6070 7261 6e6b 6020 6173 2077 656c  d `prank` as wel
+00002590: 6c20 7768 656e 2074 6865 2061 6e61 6c79  l when the analy
+000025a0: 7369 7320 7265 7175 6972 6573 2073 6571  sis requires seq
+000025b0: 7565 6e63 6520 616c 6967 6e6d 656e 7429  uence alignment)
+000025c0: 2e20 536f 6d65 206f 7468 6572 206f 7074  . Some other opt
+000025d0: 696f 6e61 6c20 736f 6674 7761 7265 7320  ional softwares 
+000025e0: 696e 636c 7564 696e 6720 6070 616d 6c20  including `paml 
+000025f0: 7634 2e39 6a60 2c20 6072 3873 602c 2060  v4.9j`, `r8s`, `
+00002600: 6265 6173 7460 2c20 6065 6767 6e6f 6760  beast`, `eggnog`
+00002610: 2c20 6064 6961 6d6f 6e64 602c 2060 696e  , `diamond`, `in
+00002620: 7465 7270 726f 7363 616e 602c 2060 686d  terproscan`, `hm
+00002630: 6d65 7220 7633 2e31 6232 6020 616e 6420  mer v3.1b2` and 
+00002640: 6061 7374 7261 6c2d 7072 6f60 2028 6068  `astral-pro` (`h
+00002650: 6d6d 6572 2076 332e 3162 3260 2069 7320  mmer v3.1b2` is 
+00002660: 616c 736f 2066 6f72 2074 6865 205b 6f72  also for the [or
+00002670: 7468 6f67 726f 7570 2061 7373 6967 6e6d  thogroup assignm
+00002680: 656e 745d 2868 7474 7073 3a2f 2f77 6764  ent](https://wgd
+00002690: 7632 2e72 6561 6474 6865 646f 6373 2e69  v2.readthedocs.i
+000026a0: 6f2f 656e 2f6c 6174 6573 742f 7573 6167  o/en/latest/usag
+000026b0: 652e 6874 6d6c 236f 7274 686f 6772 6f75  e.html#orthogrou
+000026c0: 702d 6173 7369 676e 6d65 6e74 2920 6675  p-assignment) fu
+000026d0: 6e63 7469 6f6e 2069 6e20 6077 6764 2064  nction in `wgd d
+000026e0: 6d64 6020 616e 6420 6061 7374 7261 6c2d  md` and `astral-
+000026f0: 7072 6f60 2069 7320 616c 736f 2066 6f72  pro` is also for
+00002700: 2074 6865 2066 756e 6374 696f 6e20 602d   the function `-
+00002710: 2d63 6f6c 6c69 6e65 6172 636f 616c 6573  -collinearcoales
+00002720: 6365 6e63 6560 2069 6e20 6077 6764 2064  cence` in `wgd d
+00002730: 6d64 6029 2061 7265 2066 6f72 2074 6865  md`) are for the
+00002740: 206d 6f6c 6563 756c 6172 2064 6174 696e   molecular datin
+00002750: 672c 2067 656e 6520 6661 6d69 6c79 2066  g, gene family f
+00002760: 756e 6374 696f 6e20 616e 6e6f 7461 7469  unction annotati
+00002770: 6f6e 206f 7220 7068 796c 6f67 656e 6574  on or phylogenet
+00002780: 6963 2069 6e66 6572 656e 6365 2069 6e20  ic inference in 
+00002790: 6077 6764 2066 6f63 7573 602e 0a0a 2323  `wgd focus`...##
+000027a0: 2050 6172 616d 6574 6572 730a 0a54 6865   Parameters..The
+000027b0: 7265 2061 7265 2037 206d 6169 6e20 7072  re are 7 main pr
+000027c0: 6f67 7261 6d73 2069 6e20 6077 6764 2076  ograms in `wgd v
+000027d0: 3260 3a20 6064 6d64 602c 6066 6f63 7573  2`: `dmd`,`focus
+000027e0: 602c 606b 7364 602c 606d 6978 602c 6070  `,`ksd`,`mix`,`p
+000027f0: 6561 6b60 2c60 7379 6e60 2c60 7669 7a60  eak`,`syn`,`viz`
+00002800: 2e20 4865 7265 6166 7465 7220 7765 2077  . Hereafter we w
+00002810: 696c 6c20 7072 6f76 6964 6520 6120 6465  ill provide a de
+00002820: 7461 696c 6564 2065 6c75 6369 6461 7469  tailed elucidati
+00002830: 6f6e 206f 6e20 6561 6368 206f 6620 7468  on on each of th
+00002840: 6520 7072 6f67 7261 6d20 616e 6420 6974  e program and it
+00002850: 7320 6173 736f 6369 6174 6564 2070 6172  s associated par
+00002860: 616d 6574 6572 732e 2050 6c65 6173 6520  ameters. Please 
+00002870: 7265 6665 7220 746f 2074 6865 205b 5573  refer to the [Us
+00002880: 6167 655d 2823 7573 6167 6529 2066 6f72  age](#usage) for
+00002890: 2074 6865 2073 6365 6e61 7269 6f73 2074   the scenarios t
+000028a0: 6f20 7768 6963 6820 6561 6368 2070 6172  o which each par
+000028b0: 616d 6574 6572 2061 7070 6c69 6573 2e0a  ameter applies..
+000028c0: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
+000028d0: 6420 646d 6460 2063 616e 2072 6561 6c69  d dmd` can reali
+000028e0: 7a65 2074 6865 2064 656c 696e 6561 7469  ze the delineati
+000028f0: 6f6e 206f 6620 7768 6f6c 6520 7061 7261  on of whole para
+00002900: 6e6f 6d65 2c20 5242 4873 2028 5265 6369  nome, RBHs (Reci
+00002910: 7072 6f63 616c 2042 6573 7420 4869 7473  procal Best Hits
+00002920: 292c 204d 5242 4873 2028 4d75 6c74 6970  ), MRBHs (Multip
+00002930: 6c65 2052 6563 6970 726f 6361 6c20 4265  le Reciprocal Be
+00002940: 7374 2048 6974 7329 2c20 6f72 7468 6f67  st Hits), orthog
+00002950: 726f 7570 7320 616e 6420 736f 6d65 206f  roups and some o
+00002960: 7468 6572 206f 7274 686f 6772 6f75 702d  ther orthogroup-
+00002970: 7265 6c61 7465 6420 6675 6e63 7469 6f6e  related function
+00002980: 732c 2069 6e63 6c75 6469 6e67 2063 6972  s, including cir
+00002990: 6375 6d73 6372 6970 7469 6f6e 206f 6620  cumscription of 
+000029a0: 6e65 7374 6564 2073 696e 676c 652d 636f  nested single-co
+000029b0: 7079 206f 7274 686f 6772 6f75 7073 2028  py orthogroups (
+000029c0: 4e53 4f47 7329 2c20 756e 6269 6173 6564  NSOGs), unbiased
+000029d0: 2074 6573 7420 6f66 2073 696e 676c 652d   test of single-
+000029e0: 636f 7079 206f 7274 686f 6772 6f75 7073  copy orthogroups
+000029f0: 2028 534f 4773 2920 6f76 6572 206d 6973   (SOGs) over mis
+00002a00: 7369 6e67 2069 6e70 6172 616c 6f67 732c  sing inparalogs,
+00002a10: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
+00002a20: 2042 5553 434f 2d67 7569 6465 6420 7369   BUSCO-guided si
+00002a30: 6e67 6c65 2d63 6f70 7920 6f72 7468 6f67  ngle-copy orthog
+00002a40: 726f 7570 7320 2853 4f47 7329 2c61 6e64  roups (SOGs),and
+00002a50: 2074 6865 2063 6f6c 6c69 6e65 6172 2063   the collinear c
+00002a60: 6f61 6c65 7363 656e 6365 2069 6e66 6572  oalescence infer
+00002a70: 656e 6365 206f 6620 7068 796c 6f67 656e  ence of phylogen
+00002a80: 792e 0a60 6060 0a77 6764 2064 6d64 2073  y..```.wgd dmd s
+00002a90: 6571 7565 6e63 6573 2028 6f70 7469 6f6e  equences (option
+00002aa0: 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ).--------------
+00002ab0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002ac0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002ad0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002ae0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002af0: 2d2d 0a2d 6f2c 202d 2d6f 7574 6469 722c  --.-o, --outdir,
+00002b00: 2074 6865 206f 7574 7075 7420 6469 7265   the output dire
+00002b10: 6374 6f72 792c 2064 6566 6175 6c74 2077  ctory, default w
+00002b20: 6764 5f64 6d64 0a2d 742c 202d 2d74 6d70  gd_dmd.-t, --tmp
+00002b30: 6469 722c 2074 6865 2074 656d 706f 7261  dir, the tempora
+00002b40: 7279 2077 6f72 6b69 6e67 2064 6972 6563  ry working direc
+00002b50: 746f 7279 2c20 6465 6661 756c 7420 4e6f  tory, default No
+00002b60: 6e65 2c20 6966 204e 6f6e 6520 7761 7320  ne, if None was 
+00002b70: 6769 7665 6e2c 2074 6865 2074 6d70 6469  given, the tmpdi
+00002b80: 7220 776f 756c 6420 6265 2061 7373 6967  r would be assig
+00002b90: 6e65 6420 7261 6e64 6f6d 206e 616d 6573  ned random names
+00002ba0: 2069 6e20 6375 7272 656e 7420 6469 7265   in current dire
+00002bb0: 6374 6f72 7920 616e 6420 6175 746f 6d61  ctory and automa
+00002bc0: 7465 6c79 2072 656d 6f76 6564 2061 7420  tely removed at 
+00002bd0: 7468 6520 636f 6d70 6c65 7469 6f6e 206f  the completion o
+00002be0: 6620 7072 6f67 7261 6d2c 2065 6c73 6520  f program, else 
+00002bf0: 7468 6520 746d 7064 6972 2077 6f75 6c64  the tmpdir would
+00002c00: 2062 6520 6b65 7074 0a2d 702c 202d 2d70   be kept.-p, --p
+00002c10: 726f 742c 2066 6c61 6720 6f70 7469 6f6e  rot, flag option
+00002c20: 2c20 7768 6574 6865 7220 7573 696e 6720  , whether using 
+00002c30: 7072 6f74 6569 6e20 6f72 206e 7563 6c65  protein or nucle
+00002c40: 6f74 6964 6520 7365 7175 656e 6365 730a  otide sequences.
+00002c50: 2d63 2c20 2d2d 6373 636f 7265 2c20 7468  -c, --cscore, th
+00002c60: 6520 632d 7363 6f72 6520 746f 2072 6573  e c-score to res
+00002c70: 7472 6963 7420 7468 6520 686f 6d6f 6c6f  trict the homolo
+00002c80: 6720 7369 6d69 6c61 7269 7479 206f 6620  g similarity of 
+00002c90: 4d52 4248 732c 2064 6566 6175 6c74 204e  MRBHs, default N
+00002ca0: 6f6e 652c 2069 6620 4e6f 6e65 2077 6173  one, if None was
+00002cb0: 2067 6976 656e 2c20 7468 6520 632d 7363   given, the c-sc
+00002cc0: 6f72 6520 6675 6e63 696f 6e20 776f 756c  ore funcion woul
+00002cd0: 646e 2774 2062 6520 6163 7469 7661 7465  dn't be activate
+00002ce0: 642c 2065 6c73 6520 6578 7065 6374 696e  d, else expectin
+00002cf0: 6720 6120 6465 6369 6d61 6c20 7769 7468  g a decimal with
+00002d00: 696e 2074 6865 2072 616e 6765 206f 6620  in the range of 
+00002d10: 3020 616e 6420 310a 2d49 2c20 2d2d 696e  0 and 1.-I, --in
+00002d20: 666c 6174 696f 6e2c 2074 6865 2069 6e66  flation, the inf
+00002d30: 6c61 7469 6f6e 2066 6163 746f 7220 666f  lation factor fo
+00002d40: 7220 4d43 4c20 7072 6f67 7261 6d2c 2064  r MCL program, d
+00002d50: 6566 6175 6c74 2032 2e30 2c20 7769 7468  efault 2.0, with
+00002d60: 2068 6967 6865 7220 7661 6c75 6520 6c65   higher value le
+00002d70: 6164 696e 6720 746f 206d 6f72 6520 6275  ading to more bu
+00002d80: 7420 736d 616c 6c65 7220 636c 7573 7465  t smaller cluste
+00002d90: 7273 0a2d 652c 202d 2d65 7661 6c2c 2074  rs.-e, --eval, t
+00002da0: 6865 2065 2d76 616c 7565 2063 7574 2d6f  he e-value cut-o
+00002db0: 6666 2066 6f72 2073 696d 696c 6172 6974  ff for similarit
+00002dc0: 7920 696e 2064 6961 6d6f 6e64 2061 6e64  y in diamond and
+00002dd0: 2f6f 7220 686d 6d65 722c 2064 6566 6175  /or hmmer, defau
+00002de0: 6c74 2031 652d 3130 0a2d 2d74 6f5f 7374  lt 1e-10.--to_st
+00002df0: 6f70 2c20 666c 6167 206f 7074 696f 6e2c  op, flag option,
+00002e00: 2077 6865 7468 6572 2074 6f20 7472 616e   whether to tran
+00002e10: 736c 6174 6520 7468 726f 7567 6820 5354  slate through ST
+00002e20: 4f50 2063 6f64 6f6e 732c 2069 6620 7468  OP codons, if th
+00002e30: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00002e40: 7472 616e 736c 6174 696f 6e20 7769 6c6c  translation will
+00002e50: 2062 6520 7465 726d 696e 6174 6564 2061   be terminated a
+00002e60: 7420 7468 6520 6669 7273 7420 696e 2d66  t the first in-f
+00002e70: 7261 6d65 2073 746f 7020 636f 646f 6e2c  rame stop codon,
+00002e80: 2065 6c73 6520 6120 6675 6c6c 2074 7261   else a full tra
+00002e90: 6e73 6c61 7469 6f6e 2063 6f6e 7469 6e75  nslation continu
+00002ea0: 696e 6720 6f6e 2070 6173 7369 6e67 2061  ing on passing a
+00002eb0: 6e79 2073 746f 7020 636f 646f 6e73 2077  ny stop codons w
+00002ec0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+00002ed0: 640a 2d2d 6364 732c 2066 6c61 6720 6f70  d.--cds, flag op
+00002ee0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00002ef0: 206f 6e6c 7920 7472 616e 736c 6174 6520   only translate 
+00002f00: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
+00002f10: 2074 6861 7420 7374 6172 7473 2077 6974   that starts wit
+00002f20: 6820 6120 7661 6c69 6420 7374 6172 7420  h a valid start 
+00002f30: 636f 646f 6e20 616e 6420 6f6e 6c79 2063  codon and only c
+00002f40: 6f6e 7461 696e 7320 6120 7369 6e67 6c65  ontains a single
+00002f50: 2069 6e2d 6672 616d 6520 7374 6f70 2063   in-frame stop c
+00002f60: 6f64 6f6e 2061 7420 7468 6520 656e 6420  odon at the end 
+00002f70: 616e 6420 6d75 7374 2062 6520 6469 7669  and must be divi
+00002f80: 6461 626c 6520 6279 2074 6872 6565 2c20  dable by three, 
+00002f90: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00002fa0: 7365 742c 206f 6e6c 7920 7468 6520 636f  set, only the co
+00002fb0: 6d70 6c65 7465 2043 4453 2077 6f75 6c64  mplete CDS would
+00002fc0: 2062 6520 7472 616e 736c 6174 6564 0a2d   be translated.-
+00002fd0: 662c 202d 2d66 6f63 7573 2c20 7468 6520  f, --focus, the 
+00002fe0: 7370 6563 6965 7320 746f 2062 6520 6d65  species to be me
+00002ff0: 7267 6564 206f 6e20 6c6f 6361 6c20 4d52  rged on local MR
+00003000: 4248 732c 2064 6566 6175 6c74 204e 6f6e  BHs, default Non
+00003010: 652c 2069 6620 4e6f 6e65 2077 6173 2067  e, if None was g
+00003020: 6976 656e 2c20 7468 6520 6c6f 6361 6c20  iven, the local 
+00003030: 4d52 4248 7320 776f 756c 646e 2774 2062  MRBHs wouldn't b
+00003040: 6520 696e 6665 7272 6564 0a2d 6170 2c20  e inferred.-ap, 
+00003050: 2d2d 616e 6368 6f72 706f 696e 7473 2c20  --anchorpoints, 
+00003060: 7468 6520 616e 6368 6f72 2070 6f69 6e74  the anchor point
+00003070: 7320 6461 7461 2066 696c 6520 6672 6f6d  s data file from
+00003080: 2069 2d61 6468 6f72 6520 666f 7220 636f   i-adhore for co
+00003090: 6e73 7472 7563 7469 6e67 2074 6865 206f  nstructing the o
+000030a0: 7274 686f 6772 6f75 7073 2077 6974 6820  rthogroups with 
+000030b0: 616e 6368 6f72 2070 6169 7273 2c20 6465  anchor pairs, de
+000030c0: 6661 756c 7420 4e6f 6e65 0a2d 736d 2c20  fault None.-sm, 
+000030d0: 2d2d 7365 676d 656e 7473 2c20 7468 6520  --segments, the 
+000030e0: 7365 676d 656e 7473 2064 6174 6166 696c  segments datafil
+000030f0: 6520 7573 6564 2069 6e20 636f 6c6c 696e  e used in collin
+00003100: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
+00003110: 616e 616c 7973 6973 2069 6620 696e 6974  analysis if init
+00003120: 6961 7465 642c 2064 6566 6175 6c74 204e  iated, default N
+00003130: 6f6e 650a 2d6c 652c 202d 2d6c 6973 7465  one.-le, --liste
+00003140: 6c65 6d65 6e74 732c 2074 6865 206c 6973  lements, the lis
+00003150: 7473 6567 6d65 6e74 7320 6461 7461 2066  tsegments data f
+00003160: 696c 6520 7573 6564 2069 6e20 636f 6c6c  ile used in coll
+00003170: 696e 6561 7220 636f 616c 6573 6365 6e63  inear coalescenc
+00003180: 6520 616e 616c 7973 6973 2069 6620 696e  e analysis if in
+00003190: 6974 6961 7465 642c 2064 6566 6175 6c74  itiated, default
+000031a0: 204e 6f6e 650a 2d67 742c 202d 2d67 656e   None.-gt, --gen
+000031b0: 6574 6162 6c65 2c20 7468 6520 6765 6e65  etable, the gene
+000031c0: 2074 6162 6c65 2064 6174 6166 696c 6520   table datafile 
+000031d0: 7573 6564 2069 6e20 636f 6c6c 696e 6561  used in collinea
+000031e0: 7220 636f 616c 6573 6365 6e63 6520 616e  r coalescence an
+000031f0: 616c 7973 6973 2069 6620 696e 6974 6961  alysis if initia
+00003200: 7465 642c 2064 6566 6175 6c74 204e 6f6e  ted, default Non
+00003210: 650a 2d63 6f63 2c20 2d2d 636f 6c6c 696e  e.-coc, --collin
+00003220: 6561 7263 6f61 6c65 7363 656e 6365 2c20  earcoalescence, 
+00003230: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00003240: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
+00003250: 2074 6865 2063 6f6c 6c69 6e65 6172 2063   the collinear c
+00003260: 6f61 6c65 7363 656e 6365 2061 6e61 6c79  oalescence analy
+00003270: 7369 732c 2069 6620 7468 6520 666c 6167  sis, if the flag
+00003280: 2077 6173 2073 6574 2c20 7468 6520 616e   was set, the an
+00003290: 616c 7973 6973 2077 6f75 6c64 2062 6520  alysis would be 
+000032a0: 696e 6974 6961 7465 640a 2d6b 662c 202d  initiated.-kf, -
+000032b0: 2d6b 6565 7066 6173 7461 2c20 666c 6167  -keepfasta, flag
+000032c0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
+000032d0: 2074 6f20 6f75 7470 7574 2074 6865 2073   to output the s
+000032e0: 6571 7565 6e63 6520 696e 666f 726d 6174  equence informat
+000032f0: 696f 6e20 6f66 204d 5242 4873 2c20 6966  ion of MRBHs, if
+00003300: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00003310: 742c 2074 6865 2073 6571 7565 6e63 6573  t, the sequences
+00003320: 206f 6620 4d52 4248 7320 776f 756c 6420   of MRBHs would 
+00003330: 6265 2069 6e20 6f75 7470 7574 0a2d 6b64  be in output.-kd
+00003340: 2c20 2d2d 6b65 6570 6475 706c 6963 6174  , --keepduplicat
+00003350: 6573 2c20 666c 6167 206f 7074 696f 6e2c  es, flag option,
+00003360: 2077 6865 7468 6572 2074 6f20 616c 6c6f   whether to allo
+00003370: 7720 7468 6520 7361 6d65 2067 656e 6520  w the same gene 
+00003380: 746f 206f 6363 7572 2069 6e20 6469 6666  to occur in diff
+00003390: 6572 656e 7420 4d52 4248 7320 286f 6e6c  erent MRBHs (onl
+000033a0: 7920 6d65 616e 696e 6766 756c 2077 6865  y meaningful whe
+000033b0: 6e20 7468 6520 6373 636f 7265 2077 6173  n the cscore was
+000033c0: 2075 7365 6429 2c20 6966 2074 6865 2066   used), if the f
+000033d0: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+000033e0: 2073 616d 6520 6765 6e65 2063 6f75 6c64   same gene could
+000033f0: 2062 6520 6173 7369 676e 6564 2074 6f20   be assigned to 
+00003400: 6469 6666 6572 656e 7420 4d52 4248 730a  different MRBHs.
+00003410: 2d67 6d2c 202d 2d67 6c6f 6261 6c6d 7262  -gm, --globalmrb
+00003420: 682c 2066 6c61 6720 6f70 7469 6f6e 2c20  h, flag option, 
+00003430: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
+00003440: 6174 6520 676c 6f62 616c 204d 5242 4873  ate global MRBHs
+00003450: 2063 6f6e 7374 7275 6374 696f 6e2c 2069   construction, i
+00003460: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00003470: 6574 2c20 7468 6520 2d2d 666f 6375 7320  et, the --focus 
+00003480: 6f70 7469 6f6e 2077 6f75 6c64 2062 6520  option would be 
+00003490: 6967 6e6f 7265 6420 616e 6420 6f6e 6c79  ignored and only
+000034a0: 2067 6c6f 6261 6c20 4d52 4248 7320 776f   global MRBHs wo
+000034b0: 756c 6420 6265 2062 7569 6c74 0a2d 6e2c  uld be built.-n,
+000034c0: 202d 2d6e 7468 7265 6164 732c 2074 6865   --nthreads, the
+000034d0: 206e 756d 6265 7220 6f66 2074 6872 6561   number of threa
+000034e0: 6473 2074 6f20 7573 652c 2064 6566 6175  ds to use, defau
+000034f0: 6c74 2034 0a2d 6f69 2c20 2d2d 6f72 7468  lt 4.-oi, --orth
+00003500: 6f69 6e66 6572 2c20 666c 6167 206f 7074  oinfer, flag opt
+00003510: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+00003520: 696e 6974 6961 7465 206f 7274 686f 6772  initiate orthogr
+00003530: 6f75 7020 696e 6665 726e 6563 652c 2069  oup infernece, i
+00003540: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00003550: 6574 2c20 7468 6520 6f72 7468 6f67 726f  et, the orthogro
+00003560: 7570 2069 6e66 6572 6e65 6365 2070 726f  up infernece pro
+00003570: 6772 616d 2077 6f75 6c64 2062 6520 696e  gram would be in
+00003580: 6974 6961 7465 640a 2d6f 6f2c 202d 2d6f  itiated.-oo, --o
+00003590: 6e6c 796f 7274 686f 2c20 666c 6167 206f  nlyortho, flag o
+000035a0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+000035b0: 6f20 6f6e 6c79 2063 6f6e 6475 6374 206f  o only conduct o
+000035c0: 7274 686f 6772 6f75 7020 696e 6665 726e  rthogroup infern
+000035d0: 6563 652c 2069 6620 7468 6520 666c 6167  ece, if the flag
+000035e0: 2077 6173 2073 6574 2c20 6f6e 6c79 2074   was set, only t
+000035f0: 6865 206f 7274 686f 6772 6f75 7020 696e  he orthogroup in
+00003600: 6665 726e 6563 6520 7069 7065 6c69 6e65  fernece pipeline
+00003610: 2077 6f75 6c64 2062 6520 7065 7266 6f72   would be perfor
+00003620: 6d65 6420 7768 696c 6520 7468 6520 6f74  med while the ot
+00003630: 6865 7220 616e 616c 7973 6973 2077 6f75  her analysis wou
+00003640: 6c64 6e27 7420 6265 2069 6e69 7469 6174  ldn't be initiat
+00003650: 6564 0a2d 676e 2c20 2d2d 6765 746e 736f  ed.-gn, --getnso
+00003660: 672c 2066 6c61 6720 6f70 7469 6f6e 2c20  g, flag option, 
+00003670: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
+00003680: 6174 6520 7468 6520 7365 6172 6368 696e  ate the searchin
+00003690: 6720 666f 7220 6e65 7374 6564 2073 696e  g for nested sin
+000036a0: 676c 652d 636f 7079 2067 656e 6520 6661  gle-copy gene fa
+000036b0: 6d69 6c69 6573 2028 4e53 4f47 7329 2028  milies (NSOGs) (
+000036c0: 6f6e 6c79 206d 6561 6e69 6e67 6675 6c20  only meaningful 
+000036d0: 7768 656e 2074 6865 206f 7274 686f 6772  when the orthogr
+000036e0: 6f75 7020 696e 6665 726e 6563 6520 7069  oup infernece pi
+000036f0: 7065 6c69 6e65 2077 6173 2061 6374 6976  peline was activ
+00003700: 6174 6564 292c 2069 6620 7468 6520 666c  ated), if the fl
+00003710: 6167 2077 6173 2073 6574 2c20 6164 6469  ag was set, addi
+00003720: 7469 6f6e 616c 204e 534f 4773 2061 6e61  tional NSOGs ana
+00003730: 6c79 7369 7320 776f 756c 6420 6265 2070  lysis would be p
+00003740: 6572 666f 726d 6564 2062 6573 6964 6573  erformed besides
+00003750: 2074 6865 2062 6173 6963 206f 7274 686f   the basic ortho
+00003760: 6772 6f75 7020 696e 6665 726e 6563 650a  group infernece.
+00003770: 2d74 7265 652c 202d 2d74 7265 655f 6d65  -tree, --tree_me
+00003780: 7468 6f64 2c20 7768 6963 6820 6765 6e65  thod, which gene
+00003790: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
+000037a0: 7072 6f67 7261 6d20 746f 2069 6e76 6f6b  program to invok
+000037b0: 6520 286f 6e6c 7920 6d65 616e 696e 6766  e (only meaningf
+000037c0: 756c 2077 6865 6e20 7468 6520 636f 6c6c  ul when the coll
+000037d0: 696e 6561 7220 636f 616c 6573 6365 6e63  inear coalescenc
+000037e0: 652c 2067 656e 652d 746f 2d66 616d 696c  e, gene-to-famil
+000037f0: 7920 6173 7369 676e 6d65 6e74 206f 7220  y assignment or 
+00003800: 4e53 4f47 7320 616e 616c 7973 6973 2077  NSOGs analysis w
+00003810: 6572 6520 6163 7469 7661 7465 6429 2c20  ere activated), 
+00003820: 6465 6661 756c 7420 6661 7374 7472 6565  default fasttree
+00003830: 0a2d 7473 2c20 2d2d 7472 6565 7365 742c  .-ts, --treeset,
+00003840: 2074 6865 2070 6172 616d 6574 6572 7320   the parameters 
+00003850: 7365 7474 696e 6720 666f 7220 6765 6e65  setting for gene
+00003860: 2074 7265 6520 696e 6665 7265 6e63 652c   tree inference,
+00003870: 2064 6566 6175 6c74 204e 6f6e 652c 2074   default None, t
+00003880: 6869 7320 6f70 7469 6f6e 2063 616e 2062  his option can b
+00003890: 6520 7072 6f76 6964 6564 206d 756c 7469  e provided multi
+000038a0: 706c 6520 7469 6d65 730a 2d6d 632c 202d  ple times.-mc, -
+000038b0: 2d6d 736f 6763 7574 2c20 7468 6520 7261  -msogcut, the ra
+000038c0: 7469 6f20 6375 746f 6666 2066 6f72 206d  tio cutoff for m
+000038d0: 6f73 746c 7920 7369 6e67 6c65 2d63 6f70  ostly single-cop
+000038e0: 7920 6661 6d69 6c79 2028 6d65 616e 696e  y family (meanin
+000038f0: 6766 756c 2077 6865 6e20 6163 7469 7661  gful when activa
+00003900: 7469 6e67 2074 6865 206f 7274 686f 6772  ting the orthogr
+00003910: 6f75 7020 696e 6665 726e 6563 6520 7069  oup infernece pi
+00003920: 7065 6c69 6e65 2920 616e 6420 7370 6563  peline) and spec
+00003930: 6965 7320 7265 7072 6573 656e 7461 7469  ies representati
+00003940: 6f6e 2069 6e20 636f 6c6c 696e 6561 7220  on in collinear 
+00003950: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
+00003960: 7973 6973 2c20 6465 6661 756c 7420 302e  ysis, default 0.
+00003970: 382e 0a2d 6761 2c20 2d2d 6765 6e65 6173  8..-ga, --geneas
+00003980: 7369 676e 2c20 666c 6167 206f 7074 696f  sign, flag optio
+00003990: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
+000039a0: 6974 6961 7465 2074 6865 2067 656e 652d  itiate the gene-
+000039b0: 746f 2d66 616d 696c 7920 6173 7369 676e  to-family assign
+000039c0: 6d65 6e74 2061 6e61 6c79 7369 732c 2069  ment analysis, i
+000039d0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+000039e0: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
+000039f0: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+00003a00: 7465 640a 2d73 612c 202d 2d73 6571 3261  ted.-sa, --seq2a
+00003a10: 7373 6967 6e2c 2074 6865 2071 7565 7269  ssign, the queri
+00003a20: 6564 2073 6571 7565 6e63 6573 2064 6174  ed sequences dat
+00003a30: 6120 6669 6c65 2069 6e20 6765 6e65 2d74  a file in gene-t
+00003a40: 6f2d 6661 6d69 6c79 2061 7373 6967 6e6d  o-family assignm
+00003a50: 656e 7420 616e 616c 7973 6973 2c20 6465  ent analysis, de
+00003a60: 6661 756c 7420 4e6f 6e65 2c20 7468 6973  fault None, this
+00003a70: 206f 7074 696f 6e20 6361 6e20 6265 2070   option can be p
+00003a80: 726f 7669 6465 6420 6d75 6c74 6970 6c65  rovided multiple
+00003a90: 2074 696d 6573 0a2d 6661 2c20 2d2d 6661   times.-fa, --fa
+00003aa0: 6d32 6173 7369 676e 2c20 7468 6520 7175  m2assign, the qu
+00003ab0: 6572 6965 6420 6661 6d69 6c69 7920 6461  eried familiy da
+00003ac0: 7461 2066 696c 6520 696e 2067 656e 652d  ta file in gene-
+00003ad0: 746f 2d66 616d 696c 7920 6173 7369 676e  to-family assign
+00003ae0: 6d65 6e74 2061 6e61 6c79 7369 732c 2064  ment analysis, d
+00003af0: 6566 6175 6c74 204e 6f6e 650a 2d63 632c  efault None.-cc,
+00003b00: 202d 2d63 6f6e 6361 742c 2066 6c61 6720   --concat, flag 
+00003b10: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00003b20: 746f 2069 6e69 7469 6174 6520 7468 6520  to initiate the 
+00003b30: 636f 6e63 6174 656e 6174 696f 6e20 7069  concatenation pi
+00003b40: 7065 6c69 6e65 2066 6f72 206f 7274 686f  peline for ortho
+00003b50: 6772 6f75 7020 696e 6665 726e 6563 652c  group infernece,
+00003b60: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00003b70: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
+00003b80: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
+00003b90: 6961 7465 640a 2d74 652c 202d 2d74 6573  iated.-te, --tes
+00003ba0: 7473 6f67 2c20 666c 6167 206f 7074 696f  tsog, flag optio
+00003bb0: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
+00003bc0: 6974 6961 7465 2074 6865 2075 6e62 6961  itiate the unbia
+00003bd0: 7365 6420 7465 7374 206f 6620 7369 6e67  sed test of sing
+00003be0: 6c65 2d63 6f70 7920 6765 6e65 2066 616d  le-copy gene fam
+00003bf0: 696c 6965 732c 2069 6620 7468 6520 666c  ilies, if the fl
+00003c00: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+00003c10: 616e 616c 7973 6973 2077 6f75 6c64 2062  analysis would b
+00003c20: 6520 696e 6974 6961 7465 640a 2d62 732c  e initiated.-bs,
+00003c30: 202d 2d62 696e 732c 2074 6865 206e 756d   --bins, the num
+00003c40: 6265 7220 6f66 2062 696e 7320 6469 7669  ber of bins divi
+00003c50: 6465 6420 696e 2074 6865 2067 656e 6520  ded in the gene 
+00003c60: 6c65 6e67 7468 206e 6f72 6d61 6c69 7a61  length normaliza
+00003c70: 7469 6f6e 2c20 6465 6661 756c 7420 3130  tion, default 10
+00003c80: 300a 2d6e 702c 202d 2d6e 6f72 6d61 6c69  0.-np, --normali
+00003c90: 7a65 6470 6572 6365 6e74 2c20 7468 6520  zedpercent, the 
+00003ca0: 7065 7263 656e 7461 6765 206f 6620 7570  percentage of up
+00003cb0: 7065 7220 6869 7473 2075 7365 6420 666f  per hits used fo
+00003cc0: 7220 6765 6e65 206c 656e 6774 6820 6e6f  r gene length no
+00003cd0: 726d 616c 697a 6174 696f 6e2c 2064 6566  rmalization, def
+00003ce0: 6175 6c74 2035 0a2d 6e6e 2c20 2d2d 6e6f  ault 5.-nn, --no
+00003cf0: 6e6f 726d 616c 697a 6174 696f 6e2c 2066  normalization, f
+00003d00: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00003d10: 6865 7220 746f 2063 616c 6c20 6f66 6620  her to call off 
+00003d20: 7468 6520 6e6f 726d 616c 697a 6174 696f  the normalizatio
+00003d30: 6e2c 2069 6620 7468 6520 666c 6167 2077  n, if the flag w
+00003d40: 6173 2073 6574 2c20 6e6f 206e 6f72 6d61  as set, no norma
+00003d50: 6c69 7a61 7469 6f6e 2077 6f75 6c64 2062  lization would b
+00003d60: 6520 636f 6e64 7563 7465 640a 2d62 736f  e conducted.-bso
+00003d70: 672c 202d 2d62 7573 636f 736f 672c 2066  g, --buscosog, f
+00003d80: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00003d90: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+00003da0: 7468 6520 6275 7363 6f2d 6775 6964 6564  the busco-guided
+00003db0: 2073 696e 676c 652d 636f 7079 2067 656e   single-copy gen
+00003dc0: 6520 6661 6d69 6c79 2061 6e61 6c79 7369  e family analysi
+00003dd0: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
+00003de0: 6173 2073 6574 2c20 7468 6520 616e 616c  as set, the anal
+00003df0: 7973 6973 2077 6f75 6c64 2062 6520 696e  ysis would be in
+00003e00: 6974 6961 7465 640a 2d62 686d 6d2c 202d  itiated.-bhmm, -
+00003e10: 2d62 7573 636f 686d 6d2c 2074 6865 2048  -buscohmm, the H
+00003e20: 4d4d 2070 726f 6669 6c65 2064 6174 6166  MM profile dataf
+00003e30: 696c 6520 696e 2074 6865 2062 7573 636f  ile in the busco
+00003e40: 2d67 7569 6465 6420 7369 6e67 6c65 2d63  -guided single-c
+00003e50: 6f70 7920 6765 6e65 2066 616d 696c 7920  opy gene family 
+00003e60: 616e 616c 7973 6973 2c20 6465 6661 756c  analysis, defaul
+00003e70: 7420 4e6f 6e65 0a2d 6263 7466 2c20 2d2d  t None.-bctf, --
+00003e80: 6275 7363 6f63 7574 6f66 662c 2074 6865  buscocutoff, the
+00003e90: 2048 4d4d 2073 636f 7265 2063 7574 6f66   HMM score cutof
+00003ea0: 6620 6461 7461 6669 6c65 2069 6e20 7468  f datafile in th
+00003eb0: 6520 6275 7363 6f2d 6775 6964 6564 2073  e busco-guided s
+00003ec0: 696e 676c 652d 636f 7079 2067 656e 6520  ingle-copy gene 
+00003ed0: 6661 6d69 6c79 2061 6e61 6c79 7369 732c  family analysis,
+00003ee0: 2064 6566 6175 6c74 204e 6f6e 650a 2d6f   default None.-o
+00003ef0: 6620 2c2d 2d6f 6766 6f72 6d61 742c 2066  f ,--ogformat, f
+00003f00: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00003f10: 6865 7220 746f 2061 6464 2069 6e64 6578  her to add index
+00003f20: 2074 6f20 7468 6520 5242 4820 6661 6d69   to the RBH fami
+00003f30: 6c69 6573 0a60 6060 0a0a 5468 6520 7072  lies.```..The pr
+00003f40: 6f67 7261 6d20 6077 6764 2066 6f63 7573  ogram `wgd focus
+00003f50: 6020 6361 6e20 7265 616c 697a 6520 7468  ` can realize th
+00003f60: 6520 636f 6e63 6174 656e 6174 696f 6e2d  e concatenation-
+00003f70: 6261 7365 6420 616e 6420 636f 616c 6573  based and coales
+00003f80: 6365 6e63 652d 6261 7365 6420 7068 796c  cence-based phyl
+00003f90: 6f67 656e 6574 6963 2069 6e66 6572 656e  ogenetic inferen
+00003fa0: 6365 2061 6e64 2070 6879 6c6f 6765 6e65  ce and phylogene
+00003fb0: 7469 6320 6461 7469 6e67 206f 6620 5747  tic dating of WG
+00003fc0: 4473 2065 7463 2e0a 6060 600a 7767 6420  Ds etc..```.wgd 
+00003fd0: 666f 6375 7320 6661 6d69 6c69 6573 2073  focus families s
+00003fe0: 6571 7565 6e63 6573 2028 6f70 7469 6f6e  equences (option
+00003ff0: 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ).--------------
+00004000: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004010: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004020: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004030: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004040: 2d2d 0a2d 6f2c 202d 2d6f 7574 6469 722c  --.-o, --outdir,
+00004050: 2074 6865 206f 7574 7075 7420 6469 7265   the output dire
+00004060: 6374 6f72 792c 2064 6566 6175 6c74 2077  ctory, default w
+00004070: 6764 5f66 6f63 7573 0a2d 742c 202d 2d74  gd_focus.-t, --t
+00004080: 6d70 6469 722c 2074 6865 2074 656d 706f  mpdir, the tempo
+00004090: 7261 7279 2077 6f72 6b69 6e67 2064 6972  rary working dir
+000040a0: 6563 746f 7279 2c20 6465 6661 756c 7420  ectory, default 
+000040b0: 4e6f 6e65 2c20 6966 204e 6f6e 6520 7761  None, if None wa
+000040c0: 7320 6769 7665 6e2c 2074 6865 2074 6d70  s given, the tmp
+000040d0: 6469 7220 7769 6c6c 2062 6520 6173 7369  dir will be assi
+000040e0: 676e 6564 2072 616e 646f 6d20 6e61 6d65  gned random name
+000040f0: 7320 696e 2063 7572 7265 6e74 2064 6972  s in current dir
+00004100: 6563 746f 7279 2061 6e64 2061 7574 6f6d  ectory and autom
+00004110: 6174 656c 7920 7265 6d6f 7665 6420 6174  ately removed at
+00004120: 2074 6865 2063 6f6d 706c 6574 696f 6e20   the completion 
+00004130: 6f66 2070 726f 6772 616d 2c20 656c 7365  of program, else
+00004140: 2074 6865 2074 6d70 6469 7220 776f 756c   the tmpdir woul
+00004150: 6420 6265 206b 6570 740a 2d6e 2c20 2d2d  d be kept.-n, --
+00004160: 6e74 6872 6561 6473 2c20 7468 6520 6e75  nthreads, the nu
+00004170: 6d62 6572 206f 6620 7468 7265 6164 7320  mber of threads 
+00004180: 746f 2075 7365 2c20 6465 6661 756c 7420  to use, default 
+00004190: 340a 2d2d 746f 5f73 746f 702c 2066 6c61  4.--to_stop, fla
+000041a0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+000041b0: 7220 746f 2074 7261 6e73 6c61 7465 2074  r to translate t
+000041c0: 6872 6f75 6768 2053 544f 5020 636f 646f  hrough STOP codo
+000041d0: 6e73 2c20 6966 2074 6865 2066 6c61 6720  ns, if the flag 
+000041e0: 7761 7320 7365 742c 2074 7261 6e73 6c61  was set, transla
+000041f0: 7469 6f6e 2077 696c 6c20 6265 2074 6572  tion will be ter
+00004200: 6d69 6e61 7465 6420 6174 2074 6865 2066  minated at the f
+00004210: 6972 7374 2069 6e2d 6672 616d 6520 7374  irst in-frame st
+00004220: 6f70 2063 6f64 6f6e 2c20 656c 7365 2061  op codon, else a
+00004230: 2066 756c 6c20 7472 616e 736c 6174 696f   full translatio
+00004240: 6e20 636f 6e74 696e 7569 6e67 206f 6e20  n continuing on 
+00004250: 7061 7374 2061 6e79 2073 746f 7020 636f  past any stop co
+00004260: 646f 6e73 2077 6f75 6c64 2062 6520 696e  dons would be in
+00004270: 6974 6961 7465 640a 2d2d 6364 732c 2066  itiated.--cds, f
+00004280: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00004290: 6865 7220 746f 206f 6e6c 7920 7472 616e  her to only tran
+000042a0: 736c 6174 6520 7468 6520 636f 6d70 6c65  slate the comple
+000042b0: 7465 2043 4453 2074 6861 7420 7374 6172  te CDS that star
+000042c0: 7473 2077 6974 6820 6120 7661 6c69 6420  ts with a valid 
+000042d0: 7374 6172 7420 636f 646f 6e20 616e 6420  start codon and 
+000042e0: 6f6e 6c79 2063 6f6e 7461 696e 7320 6120  only contains a 
+000042f0: 7369 6e67 6c65 2069 6e2d 6672 616d 6520  single in-frame 
+00004300: 7374 6f70 2063 6f64 6f6e 2061 7420 7468  stop codon at th
+00004310: 6520 656e 6420 616e 6420 6d75 7374 2062  e end and must b
+00004320: 6520 6469 7669 6461 626c 6520 6279 2074  e dividable by t
+00004330: 6872 6565 2c20 6966 2074 6865 2066 6c61  hree, if the fla
+00004340: 6720 7761 7320 7365 742c 206f 6e6c 7920  g was set, only 
+00004350: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
+00004360: 2077 6f75 6c64 2062 6520 7472 616e 736c   would be transl
+00004370: 6174 6564 0a2d 2d73 7472 6970 5f67 6170  ated.--strip_gap
+00004380: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
+00004390: 7768 6574 6865 7220 746f 2064 726f 7020  whether to drop 
+000043a0: 616c 6c20 6761 7073 2069 6e20 6d75 6c74  all gaps in mult
+000043b0: 6970 6c65 2073 6571 7565 6e63 6520 616c  iple sequence al
+000043c0: 6967 6e6d 656e 742c 2069 6620 7468 6520  ignment, if the 
+000043d0: 666c 6167 2077 6173 2073 6574 2c20 616c  flag was set, al
+000043e0: 6c20 6761 7073 2077 6f75 6c64 2062 6520  l gaps would be 
+000043f0: 6472 6f70 7065 640a 2d61 2c20 2d2d 616c  dropped.-a, --al
+00004400: 6967 6e65 722c 2077 6869 6368 2061 6c69  igner, which ali
+00004410: 676e 6d65 6e74 2070 726f 6772 616d 2074  gnment program t
+00004420: 6f20 7573 652c 2064 6566 6175 6c74 206d  o use, default m
+00004430: 6166 6674 0a2d 7472 6565 2c20 2d2d 7472  afft.-tree, --tr
+00004440: 6565 5f6d 6574 686f 642c 2077 6869 6368  ee_method, which
+00004450: 2067 656e 6520 7472 6565 2069 6e66 6572   gene tree infer
+00004460: 656e 6365 2070 726f 6772 616d 2074 6f20  ence program to 
+00004470: 696e 766f 6b65 2c20 6465 6661 756c 7420  invoke, default 
+00004480: 6661 7374 7472 6565 0a2d 7473 2c20 2d2d  fasttree.-ts, --
+00004490: 7472 6565 7365 742c 2074 6865 2070 6172  treeset, the par
+000044a0: 616d 6574 6572 7320 7365 7474 696e 6720  ameters setting 
+000044b0: 666f 7220 6765 6e65 2074 7265 6520 696e  for gene tree in
+000044c0: 6665 7265 6e63 652c 2064 6566 6175 6c74  ference, default
+000044d0: 204e 6f6e 652c 2074 6869 7320 6f70 7469   None, this opti
+000044e0: 6f6e 2063 616e 2062 6520 7072 6f76 6964  on can be provid
+000044f0: 6564 206d 756c 7469 706c 6520 7469 6d65  ed multiple time
+00004500: 730a 2d2d 636f 6e63 6174 656e 6174 696f  s.--concatenatio
+00004510: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
+00004520: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
+00004530: 6174 6520 7468 6520 636f 6e63 6174 656e  ate the concaten
+00004540: 6174 696f 6e2d 6261 7365 6420 7370 6563  ation-based spec
+00004550: 6965 7320 7472 6565 2069 6e66 6572 656e  ies tree inferen
+00004560: 6365 2c20 6966 2074 6865 2066 6c61 6720  ce, if the flag 
+00004570: 7761 7320 7365 742c 2063 6f6e 6361 7465  was set, concate
+00004580: 6e61 7469 6f6e 2d62 6173 6564 2073 7065  nation-based spe
+00004590: 6369 6573 2074 7265 6520 776f 756c 6420  cies tree would 
+000045a0: 6265 2069 6e66 6572 6564 0a2d 2d63 6f61  be infered.--coa
+000045b0: 6c65 7363 656e 6365 2c20 666c 6167 206f  lescence, flag o
+000045c0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+000045d0: 6f20 696e 6974 6961 7465 2074 6865 2063  o initiate the c
+000045e0: 6f61 6c65 7363 656e 6365 2d62 6173 6564  oalescence-based
+000045f0: 2073 7065 6369 6573 2074 7265 6520 696e   species tree in
+00004600: 6665 7265 6e63 652c 2069 6620 7468 6520  ference, if the 
+00004610: 666c 6167 2077 6173 2073 6574 2c20 636f  flag was set, co
+00004620: 616c 6573 6365 6e63 652d 6261 7365 6420  alescence-based 
+00004630: 7370 6563 6965 7320 7472 6565 2077 6f75  species tree wou
+00004640: 6c64 2062 6520 696e 6665 7265 640a 2d73  ld be infered.-s
+00004650: 702c 202d 2d73 7065 6369 6573 7472 6565  p, --speciestree
+00004660: 2c20 7370 6563 6965 7320 7472 6565 2064  , species tree d
+00004670: 6174 6166 696c 6520 666f 7220 6461 7469  atafile for dati
+00004680: 6e67 2c20 6465 6661 756c 7420 4e6f 6e65  ng, default None
+00004690: 0a2d 642c 202d 2d64 6174 696e 672c 2077  .-d, --dating, w
+000046a0: 6869 6368 206d 6f6c 6563 756c 6172 2064  hich molecular d
+000046b0: 6174 696e 6720 7072 6f67 7261 6d20 746f  ating program to
+000046c0: 2075 7365 2c20 6465 6661 756c 7420 6e6f   use, default no
+000046d0: 6e65 0a2d 6473 2c20 2d2d 6461 7469 6e67  ne.-ds, --dating
+000046e0: 7365 742c 2074 6865 2070 6172 616d 6574  set, the paramet
+000046f0: 6572 7320 7365 7474 696e 6720 666f 7220  ers setting for 
+00004700: 6461 7469 6e67 2070 726f 6772 616d 2c20  dating program, 
+00004710: 6465 6661 756c 7420 4e6f 6e65 2c20 7468  default None, th
+00004720: 6973 206f 7074 696f 6e20 6361 6e20 6265  is option can be
+00004730: 2070 726f 7669 6465 6420 6d75 6c74 6970   provided multip
+00004740: 6c65 2074 696d 6573 0a2d 6e73 2c20 2d2d  le times.-ns, --
+00004750: 6e73 6974 6573 2c20 7468 6520 6e73 6974  nsites, the nsit
+00004760: 6573 2069 6e66 6f72 6d61 7469 6f6e 2066  es information f
+00004770: 6f72 2072 3873 2064 6174 696e 672c 2064  or r8s dating, d
+00004780: 6566 6175 6c74 204e 6f6e 650a 2d6f 742c  efault None.-ot,
+00004790: 202d 2d6f 7574 6772 6f75 702c 2074 6865   --outgroup, the
+000047a0: 206f 7574 6772 6f75 7020 696e 666f 726d   outgroup inform
+000047b0: 6174 696f 6e20 666f 7220 7238 7320 6461  ation for r8s da
+000047c0: 7469 6e67 2c20 6465 6661 756c 7420 4e6f  ting, default No
+000047d0: 6e65 0a2d 7074 2c20 2d2d 7061 7274 6974  ne.-pt, --partit
+000047e0: 696f 6e2c 2066 6c61 6720 6f70 7469 6f6e  ion, flag option
+000047f0: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
+00004800: 7469 6174 6520 7061 7274 6974 696f 6e20  tiate partition 
+00004810: 6461 7469 6e67 2061 6e61 6c79 7369 7320  dating analysis 
+00004820: 666f 7220 636f 646f 6e2c 2069 6620 7468  for codon, if th
+00004830: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00004840: 616e 2061 6464 6974 696f 6e61 6c20 7061  an additional pa
+00004850: 7274 6974 696f 6e20 6461 7469 6e67 2061  rtition dating a
+00004860: 6e61 6c79 7369 7320 776f 756c 6420 6265  nalysis would be
+00004870: 2069 6e69 7469 6174 6564 0a2d 616d 2c20   initiated.-am, 
+00004880: 2d2d 6161 6d6f 6465 6c2c 2077 6869 6368  --aamodel, which
+00004890: 2070 726f 7465 696e 206d 6f64 656c 2074   protein model t
+000048a0: 6f20 6265 2075 7365 6420 696e 206d 636d  o be used in mcm
+000048b0: 6374 7265 652c 2064 6566 6175 6c74 2070  ctree, default p
+000048c0: 6f69 7373 6f6e 0a2d 6b73 2c20 666c 6167  oisson.-ks, flag
+000048d0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
+000048e0: 2074 6f20 696e 6974 6961 7465 204b 7320   to initiate Ks 
+000048f0: 6361 6c63 756c 6174 696f 6e20 666f 7220  calculation for 
+00004900: 686f 6d6f 6c6f 6775 6573 2069 6e20 7468  homologues in th
+00004910: 6520 7072 6f76 6964 6564 206f 7274 686f  e provided ortho
+00004920: 6c6f 676f 7573 2067 656e 6520 6661 6d69  logous gene fami
+00004930: 6c79 0a2d 2d61 6e6e 6f74 6174 696f 6e2c  ly.--annotation,
+00004940: 2077 6869 6368 2061 6e6e 6f74 6174 696f   which annotatio
+00004950: 6e20 7072 6f67 7261 6d20 746f 2075 7365  n program to use
+00004960: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+00004970: 2d70 6169 7277 6973 652c 2066 6c61 6720  -pairwise, flag 
+00004980: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00004990: 746f 2069 6e69 7469 6174 6520 7061 6972  to initiate pair
+000049a0: 7769 7365 204b 7320 6573 7469 6d61 7469  wise Ks estimati
+000049b0: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
+000049c0: 7761 7320 7365 742c 2070 6169 7277 6973  was set, pairwis
+000049d0: 6520 4b73 2076 616c 7565 7320 776f 756c  e Ks values woul
+000049e0: 6420 6265 2065 7374 696d 6174 6564 0a2d  d be estimated.-
+000049f0: 6564 2c20 2d2d 6567 676e 6f67 6461 7461  ed, --eggnogdata
+00004a00: 2c20 7468 6520 6567 676e 6f67 2061 6e6e  , the eggnog ann
+00004a10: 6f74 6174 696f 6e20 6461 7461 6669 6c65  otation datafile
+00004a20: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+00004a30: 2d70 6661 6d2c 2077 6869 6368 206f 7074  -pfam, which opt
+00004a40: 696f 6e20 746f 2075 7365 2066 6f72 2070  ion to use for p
+00004a50: 6661 6d20 616e 6e6f 7461 7469 6f6e 2c20  fam annotation, 
+00004a60: 6465 6661 756c 7420 4e6f 6e65 0a2d 2d64  default None.--d
+00004a70: 6d6e 622c 2074 6865 2064 6961 6d6f 6e64  mnb, the diamond
+00004a80: 2064 6174 6162 6173 6520 666f 7220 616e   database for an
+00004a90: 6e6f 7461 7469 6f6e 2c20 6465 6661 756c  notation, defaul
+00004aa0: 7420 4e6f 6e65 0a2d 2d68 6d6d 2c20 7468  t None.--hmm, th
+00004ab0: 6520 484d 4d20 7072 6f66 696c 6520 666f  e HMM profile fo
+00004ac0: 7220 616e 6e6f 7461 7469 6f6e 2c20 6465  r annotation, de
+00004ad0: 6661 756c 7420 4e6f 6e65 0a2d 2d65 7661  fault None.--eva
+00004ae0: 6c75 652c 2074 6865 2065 2d76 616c 7565  lue, the e-value
+00004af0: 2063 7574 2d6f 6666 2066 6f72 2061 6e6e   cut-off for ann
+00004b00: 6f74 6174 696f 6e2c 2064 6566 6175 6c74  otation, default
+00004b10: 2031 652d 3130 0a2d 2d65 7865 7061 7468   1e-10.--exepath
+00004b20: 2c20 7468 6520 7061 7468 2074 6f20 7468  , the path to th
+00004b30: 6520 696e 7465 7270 726f 7363 616e 2065  e interproscan e
+00004b40: 7865 6375 7461 626c 652c 2064 6566 6175  xecutable, defau
+00004b50: 6c74 204e 6f6e 650a 2d66 2c20 2d2d 666f  lt None.-f, --fo
+00004b60: 7373 696c 2c20 7468 6520 666f 7373 696c  ssil, the fossil
+00004b70: 2063 616c 6962 7261 7469 6f6e 2069 6e66   calibration inf
+00004b80: 6f72 6d61 7469 6f6e 2069 6e20 4265 6173  ormation in Beas
+00004b90: 742c 2064 6566 6175 6c74 2028 2763 6c61  t, default ('cla
+00004ba0: 6465 313b 636c 6164 6532 272c 2027 7461  de1;clade2', 'ta
+00004bb0: 7861 312c 7461 7861 323b 7461 7861 332c  xa1,taxa2;taxa3,
+00004bc0: 7461 7861 3427 2c20 2734 3b35 272c 2027  taxa4', '4;5', '
+00004bd0: 302e 353b 302e 3627 2c20 2734 3030 3b35  0.5;0.6', '400;5
+00004be0: 3030 2729 0a2d 7268 2c20 2d2d 726f 6f74  00').-rh, --root
+00004bf0: 6865 6967 6874 2c20 7468 6520 726f 6f74  height, the root
+00004c00: 2068 6569 6768 7420 6361 6c69 6272 6174   height calibrat
+00004c10: 696f 6e20 696e 666f 2069 6e20 4265 6173  ion info in Beas
+00004c20: 742c 2064 6566 6175 6c74 2028 342c 302e  t, default (4,0.
+00004c30: 352c 3430 3029 0a2d 6373 2c20 2d2d 6368  5,400).-cs, --ch
+00004c40: 6169 6e73 6574 2c20 7468 6520 7061 7261  ainset, the para
+00004c50: 6d65 7465 7273 206f 6620 4d43 4d43 2063  meters of MCMC c
+00004c60: 6861 696e 2069 6e20 4265 6173 742c 2064  hain in Beast, d
+00004c70: 6566 6175 6c74 2028 3130 3030 302c 3130  efault (10000,10
+00004c80: 3029 0a2d 2d62 6561 7374 6c67 6a61 722c  0).--beastlgjar,
+00004c90: 2074 6865 2070 6174 6820 746f 2062 6561   the path to bea
+00004ca0: 7374 4c47 2e6a 6172 2c20 6465 6661 756c  stLG.jar, defaul
+00004cb0: 7420 4e6f 6e65 0a2d 2d62 6561 676c 652c  t None.--beagle,
+00004cc0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00004cd0: 6574 6865 7220 746f 2075 7365 2062 6561  ether to use bea
+00004ce0: 676c 6520 696e 2042 6561 7374 2c20 6966  gle in Beast, if
+00004cf0: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00004d00: 742c 2062 6561 676c 6520 776f 756c 6420  t, beagle would 
+00004d10: 6265 2075 7365 640a 2d2d 7072 6f74 636f  be used.--protco
+00004d20: 6364 6174 696e 672c 2066 6c61 6720 6f70  cdating, flag op
+00004d30: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00004d40: 206f 6e6c 7920 696e 6974 6961 7465 2074   only initiate t
+00004d50: 6865 2070 726f 7465 696e 2d63 6f6e 6361  he protein-conca
+00004d60: 7465 6e61 7469 6f6e 2d62 6173 6564 2064  tenation-based d
+00004d70: 6174 696e 6720 616e 616c 7973 6973 2c20  ating analysis, 
+00004d80: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00004d90: 7365 742c 2074 6865 2061 6e61 6c79 7369  set, the analysi
+00004da0: 7320 776f 756c 6420 6265 2069 6e69 7469  s would be initi
+00004db0: 6174 6564 0a2d 2d70 726f 7464 6174 696e  ated.--protdatin
+00004dc0: 672c 2066 6c61 6720 6f70 7469 6f6e 2c20  g, flag option, 
+00004dd0: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
+00004de0: 696e 6974 6961 7465 2074 6865 2070 726f  initiate the pro
+00004df0: 7465 696e 2d62 6173 6564 2064 6174 696e  tein-based datin
+00004e00: 6720 616e 616c 7973 6973 2c20 6966 2074  g analysis, if t
+00004e10: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00004e20: 2074 6865 2061 6e61 6c79 7369 7320 776f   the analysis wo
+00004e30: 756c 6420 6265 2069 6e69 7469 6174 6564  uld be initiated
+00004e40: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
+00004e50: 6d20 6077 6764 206b 7364 6020 6361 6e20  m `wgd ksd` can 
+00004e60: 7265 616c 697a 6520 7468 6520 636f 6e73  realize the cons
+00004e70: 7472 7563 7469 6f6e 206f 6620 2a4b 2a3c  truction of *K*<
+00004e80: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+00004e90: 6469 7374 7269 6275 7469 6f6e 2061 6e64  distribution and
+00004ea0: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
+00004eb0: 2e0a 6060 600a 7767 6420 6b73 6420 6661  ..```.wgd ksd fa
+00004ec0: 6d69 6c69 6573 2073 6571 7565 6e63 6573  milies sequences
+00004ed0: 2028 6f70 7469 6f6e 290a 2d2d 2d2d 2d2d   (option).------
+00004ee0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004ef0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004f00: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004f10: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004f20: 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c 202d  ----------.-o, -
+00004f30: 2d6f 7574 6469 722c 2074 6865 206f 7574  -outdir, the out
+00004f40: 7075 7420 6469 7265 6374 6f72 792c 2064  put directory, d
+00004f50: 6566 6175 6c74 2077 6764 5f6b 7364 0a2d  efault wgd_ksd.-
+00004f60: 742c 202d 2d74 6d70 6469 722c 2074 6865  t, --tmpdir, the
+00004f70: 2074 656d 706f 7261 7279 2077 6f72 6b69   temporary worki
+00004f80: 6e67 2064 6972 6563 746f 7279 2c20 6465  ng directory, de
+00004f90: 6661 756c 7420 4e6f 6e65 2c20 6966 204e  fault None, if N
+00004fa0: 6f6e 6520 7761 7320 6769 7665 6e2c 2074  one was given, t
+00004fb0: 6865 2074 6d70 6469 7220 7769 6c6c 2062  he tmpdir will b
+00004fc0: 6520 6173 7369 676e 6564 2072 616e 646f  e assigned rando
+00004fd0: 6d20 6e61 6d65 7320 696e 2063 7572 7265  m names in curre
+00004fe0: 6e74 2064 6972 6563 746f 7279 2061 6e64  nt directory and
+00004ff0: 2061 7574 6f6d 6174 656c 7920 7265 6d6f   automately remo
+00005000: 7665 6420 6174 2074 6865 2063 6f6d 706c  ved at the compl
+00005010: 6574 696f 6e20 6f66 2070 726f 6772 616d  etion of program
+00005020: 2c20 656c 7365 2074 6865 2074 6d70 6469  , else the tmpdi
+00005030: 7220 776f 756c 6420 6265 206b 6570 740a  r would be kept.
+00005040: 2d6e 2c20 2d2d 6e74 6872 6561 6473 2c20  -n, --nthreads, 
+00005050: 7468 6520 6e75 6d62 6572 206f 6620 7468  the number of th
+00005060: 7265 6164 7320 746f 2075 7365 2c20 6465  reads to use, de
+00005070: 6661 756c 7420 340a 2d2d 746f 5f73 746f  fault 4.--to_sto
+00005080: 702c 2066 6c61 6720 6f70 7469 6f6e 2c20  p, flag option, 
+00005090: 7768 6574 6865 7220 746f 2074 7261 6e73  whether to trans
+000050a0: 6c61 7465 2074 6872 6f75 6768 2053 544f  late through STO
+000050b0: 5020 636f 646f 6e73 2c20 6966 2074 6865  P codons, if the
+000050c0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+000050d0: 7261 6e73 6c61 7469 6f6e 2077 696c 6c20  ranslation will 
+000050e0: 6265 2074 6572 6d69 6e61 7465 6420 6174  be terminated at
+000050f0: 2074 6865 2066 6972 7374 2069 6e2d 6672   the first in-fr
+00005100: 616d 6520 7374 6f70 2063 6f64 6f6e 2c20  ame stop codon, 
+00005110: 656c 7365 2061 2066 756c 6c20 7472 616e  else a full tran
+00005120: 736c 6174 696f 6e20 636f 6e74 696e 7569  slation continui
+00005130: 6e67 206f 6e20 7061 7374 2061 6e79 2073  ng on past any s
+00005140: 746f 7020 636f 646f 6e73 2077 6f75 6c64  top codons would
+00005150: 2062 6520 696e 6974 6961 7465 640a 2d2d   be initiated.--
+00005160: 6364 732c 2066 6c61 6720 6f70 7469 6f6e  cds, flag option
+00005170: 2c20 7768 6574 6865 7220 746f 206f 6e6c  , whether to onl
+00005180: 7920 7472 616e 736c 6174 6520 7468 6520  y translate the 
+00005190: 636f 6d70 6c65 7465 2043 4453 2074 6861  complete CDS tha
+000051a0: 7420 7374 6172 7473 2077 6974 6820 6120  t starts with a 
+000051b0: 7661 6c69 6420 7374 6172 7420 636f 646f  valid start codo
+000051c0: 6e20 616e 6420 6f6e 6c79 2063 6f6e 7461  n and only conta
+000051d0: 696e 7320 6120 7369 6e67 6c65 2069 6e2d  ins a single in-
+000051e0: 6672 616d 6520 7374 6f70 2063 6f64 6f6e  frame stop codon
+000051f0: 2061 7420 7468 6520 656e 6420 616e 6420   at the end and 
+00005200: 6d75 7374 2062 6520 6469 7669 6461 626c  must be dividabl
+00005210: 6520 6279 2074 6872 6565 2c20 6966 2074  e by three, if t
+00005220: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00005230: 206f 6e6c 7920 7468 6520 636f 6d70 6c65   only the comple
+00005240: 7465 2043 4453 2077 6f75 6c64 2062 6520  te CDS would be 
+00005250: 7472 616e 736c 6174 6564 0a2d 2d70 6169  translated.--pai
+00005260: 7277 6973 652c 2066 6c61 6720 6f70 7469  rwise, flag opti
+00005270: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
+00005280: 6e69 7469 6174 6520 7061 6972 7769 7365  nitiate pairwise
+00005290: 204b 7320 6573 7469 6d61 7469 6f6e 2c20   Ks estimation, 
+000052a0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+000052b0: 7365 742c 2070 6169 7277 6973 6520 4b73  set, pairwise Ks
+000052c0: 2076 616c 7565 7320 776f 756c 6420 6265   values would be
+000052d0: 2065 7374 696d 6174 6564 0a2d 2d73 7472   estimated.--str
+000052e0: 6970 5f67 6170 732c 2066 6c61 6720 6f70  ip_gaps, flag op
+000052f0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00005300: 2064 726f 7020 616c 6c20 6761 7073 2069   drop all gaps i
+00005310: 6e20 6d75 6c74 6970 6c65 2073 6571 7565  n multiple seque
+00005320: 6e63 6520 616c 6967 6e6d 656e 742c 2069  nce alignment, i
+00005330: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00005340: 6574 2c20 616c 6c20 6761 7073 2077 6f75  et, all gaps wou
+00005350: 6c64 2062 6520 6472 6f70 7065 640a 2d61  ld be dropped.-a
+00005360: 2c20 2d2d 616c 6967 6e65 722c 2077 6869  , --aligner, whi
+00005370: 6368 2061 6c69 676e 6d65 6e74 2070 726f  ch alignment pro
+00005380: 6772 616d 2074 6f20 7573 652c 2064 6566  gram to use, def
+00005390: 6175 6c74 206d 6166 6674 200a 2d74 7265  ault mafft .-tre
+000053a0: 652c 202d 2d74 7265 655f 6d65 7468 6f64  e, --tree_method
+000053b0: 2c20 7768 6963 6820 6765 6e65 2074 7265  , which gene tre
+000053c0: 6520 696e 6665 7265 6e63 6520 7072 6f67  e inference prog
+000053d0: 7261 6d20 746f 2069 6e76 6f6b 652c 2064  ram to invoke, d
+000053e0: 6566 6175 6c74 2066 6173 7474 7265 650a  efault fasttree.
+000053f0: 2d2d 7472 6565 5f6f 7074 696f 6e73 2c20  --tree_options, 
+00005400: 6f70 7469 6f6e 7320 696e 2074 7265 6520  options in tree 
+00005410: 696e 6665 7265 6e63 6520 6173 2061 2063  inference as a c
+00005420: 6f6d 6d61 2073 6570 6172 6174 6564 2073  omma separated s
+00005430: 7472 696e 672c 2064 6566 6175 6c74 204e  tring, default N
+00005440: 6f6e 650a 2d2d 6e6f 6465 5f61 7665 7261  one.--node_avera
+00005450: 6765 2c20 666c 6167 206f 7074 696f 6e2c  ge, flag option,
+00005460: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
+00005470: 6961 7465 206e 6f64 652d 6176 6572 6167  iate node-averag
+00005480: 6520 7761 7920 6f66 2064 652d 7265 6475  e way of de-redu
+00005490: 6e64 616e 6379 2069 6e73 7465 6164 206f  ndancy instead o
+000054a0: 6620 6e6f 6465 2d77 6569 6768 7465 642c  f node-weighted,
+000054b0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+000054c0: 2073 6574 2c20 7468 6520 6e6f 6465 2d61   set, the node-a
+000054d0: 7665 7261 6769 6e67 2064 652d 7265 6475  veraging de-redu
+000054e0: 6e64 616e 6379 2077 6f75 6c64 2062 6520  ndancy would be 
+000054f0: 696e 6974 6961 7465 640a 2d73 722c 202d  initiated.-sr, -
+00005500: 2d73 7061 6972 2c20 7468 6520 7370 6563  -spair, the spec
+00005510: 6965 7320 7061 6972 2074 6f20 6265 2070  ies pair to be p
+00005520: 6c6f 7474 6564 2c20 6465 6661 756c 7420  lotted, default 
+00005530: 4e6f 6e65 2c20 7468 6973 206f 7074 696f  None, this optio
+00005540: 6e20 6361 6e20 6265 2070 726f 7669 6465  n can be provide
+00005550: 6420 6d75 6c74 6970 6c65 2074 696d 6573  d multiple times
+00005560: 0a2d 7370 2c20 2d2d 7370 6563 6965 7374  .-sp, --speciest
+00005570: 7265 652c 2074 6865 2073 7065 6369 6573  ree, the species
+00005580: 2074 7265 6520 746f 2070 6572 666f 726d   tree to perform
+00005590: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
+000055a0: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
+000055b0: 6966 204e 6f6e 6520 7761 7320 6769 7665  if None was give
+000055c0: 6e2c 2074 6865 2072 6174 6520 636f 7272  n, the rate corr
+000055d0: 6563 7469 6f6e 2061 6e61 6c79 7369 7320  ection analysis 
+000055e0: 776f 756c 6420 6265 2063 616c 6c65 6420  would be called 
+000055f0: 6f66 660a 2d72 772c 202d 2d72 6577 6569  off.-rw, --rewei
+00005600: 6768 742c 2066 6c61 6720 6f70 7469 6f6e  ght, flag option
+00005610: 2c20 7768 6574 6865 7220 746f 2072 6563  , whether to rec
+00005620: 616c 6375 6c61 7465 2074 6865 2077 6569  alculate the wei
+00005630: 6768 7420 7065 7220 7370 6563 6965 7320  ght per species 
+00005640: 7061 6972 2c20 6966 2074 6865 2066 6c61  pair, if the fla
+00005650: 6720 7761 7320 7365 742c 2074 6865 2077  g was set, the w
+00005660: 6569 6768 7420 776f 756c 6420 6265 2072  eight would be r
+00005670: 6563 616c 6375 6c61 7465 640a 2d6f 722c  ecalculated.-or,
+00005680: 202d 2d6f 6e6c 7972 6f6f 746f 7574 2c20   --onlyrootout, 
+00005690: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+000056a0: 7468 6572 2074 6f20 6f6e 6c79 2063 6f6e  ther to only con
+000056b0: 6475 6374 2072 6174 6520 636f 7272 6563  duct rate correc
+000056c0: 7469 6f6e 2075 7369 6e67 2074 6865 206f  tion using the o
+000056d0: 7574 6772 6f75 7020 6174 2072 6f6f 7420  utgroup at root 
+000056e0: 6173 206f 7574 6772 6f75 702c 2069 6620  as outgroup, if 
+000056f0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00005700: 2c20 6f6e 6c79 2074 6865 206f 7574 6772  , only the outgr
+00005710: 6f75 7020 6174 2072 6f6f 7420 776f 756c  oup at root woul
+00005720: 6420 6265 2075 7365 6420 6173 206f 7574  d be used as out
+00005730: 6772 6f75 700a 2d65 706b 2c20 2d2d 6578  group.-epk, --ex
+00005740: 7472 6170 6172 616e 6f6d 656b 732c 2065  traparanomeks, e
+00005750: 7874 7261 2070 6172 616e 6f6d 6520 4b73  xtra paranome Ks
+00005760: 2064 6174 6120 746f 2070 6c6f 7420 696e   data to plot in
+00005770: 2074 6865 206d 6978 6564 204b 7320 6469   the mixed Ks di
+00005780: 7374 7269 6275 7469 6f6e 2c20 6465 6661  stribution, defa
+00005790: 756c 7420 4e6f 6e65 0a2d 6170 2c20 2d2d  ult None.-ap, --
+000057a0: 616e 6368 6f72 706f 696e 7473 2c20 616e  anchorpoints, an
+000057b0: 6368 6f72 706f 696e 7473 2e74 7874 2066  chorpoints.txt f
+000057c0: 696c 6520 746f 2070 6c6f 7420 616e 6368  ile to plot anch
+000057d0: 6f72 204b 7320 696e 2074 6865 206d 6978  or Ks in the mix
+000057e0: 6564 204b 7320 6469 7374 7269 6275 7469  ed Ks distributi
+000057f0: 6f6e 2c20 6465 6661 756c 7420 4e6f 6e65  on, default None
+00005800: 0a2d 706b 2c20 2d2d 706c 6f74 6b64 652c  .-pk, --plotkde,
+00005810: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00005820: 6574 6865 7220 746f 2070 6c6f 7420 6b64  ether to plot kd
+00005830: 6520 6375 7276 6520 6f66 206f 7274 686f  e curve of ortho
+00005840: 6c6f 676f 7573 204b 7320 6469 7374 7269  logous Ks distri
+00005850: 6275 7469 6f6e 206f 7665 7220 6869 7374  bution over hist
+00005860: 6f67 7261 6d20 696e 2074 6865 206d 6978  ogram in the mix
+00005870: 6564 204b 7320 6469 7374 7269 6275 7469  ed Ks distributi
+00005880: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
+00005890: 7761 7320 7365 742c 2074 6865 206b 6465  was set, the kde
+000058a0: 2063 7572 7665 2077 6f75 6c64 2062 6520   curve would be 
+000058b0: 706c 6f74 7465 640a 2d70 6167 2c20 2d2d  plotted.-pag, --
+000058c0: 706c 6f74 6170 676d 6d2c 2066 6c61 6720  plotapgmm, flag 
+000058d0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+000058e0: 746f 2070 6572 666f 726d 2061 6e64 2070  to perform and p
+000058f0: 6c6f 7420 6d69 7874 7572 6520 6d6f 6465  lot mixture mode
+00005900: 6c69 6e67 206f 6620 616e 6368 6f72 204b  ling of anchor K
+00005910: 7320 696e 2074 6865 206d 6978 6564 204b  s in the mixed K
+00005920: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+00005930: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00005940: 7365 742c 2074 6865 206d 6978 7475 7265  set, the mixture
+00005950: 206d 6f64 656c 696e 6720 6f66 2061 6e63   modeling of anc
+00005960: 686f 7220 4b73 2077 6f75 6c64 2062 6520  hor Ks would be 
+00005970: 706c 6f74 7465 640a 2d70 656d 2c20 2d2d  plotted.-pem, --
+00005980: 706c 6f74 656c 6d6d 2c20 666c 6167 206f  plotelmm, flag o
+00005990: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+000059a0: 6f20 7065 7266 6f72 6d20 616e 6420 706c  o perform and pl
+000059b0: 6f74 2065 6c6d 6d20 6d69 7874 7572 6520  ot elmm mixture 
+000059c0: 6d6f 6465 6c69 6e67 206f 6620 7061 7261  modeling of para
+000059d0: 6e6f 6d65 204b 7320 696e 2074 6865 206d  nome Ks in the m
+000059e0: 6978 6564 204b 7320 6469 7374 7269 6275  ixed Ks distribu
+000059f0: 7469 6f6e 2c20 6966 2074 6865 2066 6c61  tion, if the fla
+00005a00: 6720 7761 7320 7365 742c 2074 6865 2065  g was set, the e
+00005a10: 6c6d 6d20 6d69 7874 7572 6520 6d6f 6465  lmm mixture mode
+00005a20: 6c69 6e67 206f 6620 7061 7261 6e6f 6d65  ling of paranome
+00005a30: 204b 7320 776f 756c 6420 6265 2070 6c6f   Ks would be plo
+00005a40: 7474 6564 0a2d 632c 202d 2d63 6f6d 706f  tted.-c, --compo
+00005a50: 6e65 6e74 732c 2074 6865 2072 616e 6765  nents, the range
+00005a60: 206f 6620 7468 6520 6e75 6d62 6572 206f   of the number o
+00005a70: 6620 636f 6d70 6f6e 656e 7473 2074 6f20  f components to 
+00005a80: 6669 7420 696e 2061 6e63 686f 7220 4b73  fit in anchor Ks
+00005a90: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
+00005aa0: 672c 2064 6566 6175 6c74 2028 312c 3429  g, default (1,4)
+00005ab0: 0a2d 786c 2c20 2d2d 786c 696d 2c20 7468  .-xl, --xlim, th
+00005ac0: 6520 7820 6178 6973 206c 696d 6974 206f  e x axis limit o
+00005ad0: 6620 4b73 2064 6973 7472 6962 7574 696f  f Ks distributio
+00005ae0: 6e0a 2d79 6c2c 202d 2d79 6c69 6d2c 2074  n.-yl, --ylim, t
+00005af0: 6865 2079 2061 7869 7320 6c69 6d69 7420  he y axis limit 
+00005b00: 6f66 204b 7320 6469 7374 7269 6275 7469  of Ks distributi
+00005b10: 6f6e 0a2d 6164 6f2c 202d 2d61 646a 7573  on.-ado, --adjus
+00005b20: 746f 7274 686f 2c20 666c 6167 206f 7074  tortho, flag opt
+00005b30: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+00005b40: 6164 6a75 7374 2074 6865 2068 6973 746f  adjust the histo
+00005b50: 6772 616d 2068 6569 6768 7420 6f66 206f  gram height of o
+00005b60: 7274 686f 6c6f 676f 7573 204b 7320 6173  rthologous Ks as
+00005b70: 2074 6f20 6d61 7463 6820 7468 6520 6865   to match the he
+00005b80: 6967 6874 206f 6620 7061 7261 6c6f 676f  ight of paralogo
+00005b90: 7573 204b 732c 2069 6620 7468 6520 666c  us Ks, if the fl
+00005ba0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+00005bb0: 6164 6a75 7374 6d65 6e74 2077 6f75 6c64  adjustment would
+00005bc0: 2062 6520 636f 6e64 7563 7465 640a 2d61   be conducted.-a
+00005bd0: 6466 2c20 2d2d 6164 6a75 7374 6661 6374  df, --adjustfact
+00005be0: 6f72 2c20 7468 6520 6164 6a75 7374 6d65  or, the adjustme
+00005bf0: 6e74 2066 6163 746f 7220 6f66 206f 7274  nt factor of ort
+00005c00: 686f 6c6f 676f 7573 204b 732c 2064 6566  hologous Ks, def
+00005c10: 6175 6c74 2030 2e35 0a2d 6f61 2c20 2d2d  ault 0.5.-oa, --
+00005c20: 6f6b 616c 7068 612c 2074 6865 206f 7061  okalpha, the opa
+00005c30: 6369 7479 206f 6620 6f72 7468 6f6c 6f67  city of ortholog
+00005c40: 6f75 7320 4b73 2064 6973 7472 6962 7574  ous Ks distribut
+00005c50: 696f 6e20 696e 206d 6978 6564 2070 6c6f  ion in mixed plo
+00005c60: 742c 2064 6566 6175 6c74 2030 2e35 0a2d  t, default 0.5.-
+00005c70: 6661 2c20 2d2d 666f 6375 7332 616c 6c2c  fa, --focus2all,
+00005c80: 2073 6574 2066 6f63 616c 2073 7065 6369   set focal speci
+00005c90: 6573 2061 6e64 206c 6574 2073 7065 6369  es and let speci
+00005ca0: 6573 2070 6169 7220 746f 2062 6520 6265  es pair to be be
+00005cb0: 7477 6565 6e20 666f 6361 6c20 616e 6420  tween focal and 
+00005cc0: 616c 6c20 7468 6520 7265 6d61 696e 696e  all the remainin
+00005cd0: 6720 7370 6563 6965 732c 2064 6566 6175  g species, defau
+00005ce0: 6c74 204e 6f6e 650a 2d6b 732c 202d 2d6b  lt None.-ks, --k
+00005cf0: 7374 7265 652c 2066 6c61 6720 6f70 7469  stree, flag opti
+00005d00: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
+00005d10: 6e66 6572 204b 7320 7472 6565 2c20 6966  nfer Ks tree, if
+00005d20: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00005d30: 742c 2074 6865 204b 7320 7472 6565 2069  t, the Ks tree i
+00005d40: 6e66 6572 656e 6365 2061 6e61 6c79 7369  nference analysi
+00005d50: 7320 776f 756c 6420 6265 2069 6e69 7469  s would be initi
+00005d60: 6174 6564 0a2d 6f63 6b2c 202d 2d6f 6e6c  ated.-ock, --onl
+00005d70: 7963 6f6e 6361 746b 7374 7265 652c 2066  yconcatkstree, f
+00005d80: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00005d90: 6865 7220 746f 206f 6e6c 7920 696e 6665  her to only infe
+00005da0: 7220 4b73 2074 7265 6520 756e 6465 7220  r Ks tree under 
+00005db0: 636f 6e63 6174 656e 6174 6564 2061 6c69  concatenated ali
+00005dc0: 676e 6d65 6e74 2c20 6966 2074 6865 2066  gnment, if the f
+00005dd0: 6c61 6720 7761 7320 7365 742c 206f 6e6c  lag was set, onl
+00005de0: 7920 7468 6520 4b73 2074 7265 6520 756e  y the Ks tree un
+00005df0: 6465 7220 636f 6e63 6174 656e 6174 6564  der concatenated
+00005e00: 2061 6c69 676e 6d65 6e74 2077 6f75 6c64   alignment would
+00005e10: 2062 6520 6361 6c63 756c 6174 6564 0a2d   be calculated.-
+00005e20: 6373 2c20 2d2d 636c 6173 7369 632c 2066  cs, --classic, f
+00005e30: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00005e40: 6865 7220 746f 2064 7261 7720 6d69 7865  her to draw mixe
+00005e50: 6420 4b73 2070 6c6f 7420 696e 2061 2063  d Ks plot in a c
+00005e60: 6c61 7373 6963 206d 616e 6e65 7220 7768  lassic manner wh
+00005e70: 6572 6520 7468 6520 6675 6c6c 206f 7274  ere the full ort
+00005e80: 686f 6c6f 676f 7573 204b 7320 6469 7374  hologous Ks dist
+00005e90: 7269 6275 7469 6f6e 2069 7320 6472 6177  ribution is draw
+00005ea0: 6564 2c20 6966 2074 6865 2066 6c61 6720  ed, if the flag 
+00005eb0: 7761 7320 7365 742c 2074 6865 2063 6c61  was set, the cla
+00005ec0: 7373 6963 206d 6978 6564 204b 7320 706c  ssic mixed Ks pl
+00005ed0: 6f74 2077 6f75 6c64 2062 6520 6472 6177  ot would be draw
+00005ee0: 6e0a 2d74 612c 202d 2d74 6f70 6172 726f  n.-ta, --toparro
+00005ef0: 772c 2066 6c61 6720 6f70 7469 6f6e 2c20  w, flag option, 
+00005f00: 7768 6574 6865 7220 746f 2061 646a 7573  whether to adjus
+00005f10: 7420 7468 6520 6172 726f 7720 746f 2062  t the arrow to b
+00005f20: 6520 6174 2074 6865 2074 6f70 206f 6620  e at the top of 
+00005f30: 7468 6520 706c 6f74 2c20 696e 7374 6561  the plot, instea
+00005f40: 6420 6f66 2062 6569 6e67 2063 6f6f 7264  d of being coord
+00005f50: 696e 6174 6564 2061 7320 7468 6520 4b44  inated as the KD
+00005f60: 4520 6f66 2074 6865 206f 7274 686f 6c6f  E of the ortholo
+00005f70: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
+00005f80: 7469 6f6e 2c20 6966 2074 6865 2066 6c61  tion, if the fla
+00005f90: 6720 7761 7320 7365 742c 2074 6865 2061  g was set, the a
+00005fa0: 7272 6f77 2077 6f75 6c64 2062 6520 7365  rrow would be se
+00005fb0: 7420 6174 2074 6865 2074 6f70 0a2d 6273  t at the top.-bs
+00005fc0: 2c20 2d2d 626f 6f74 7374 7261 702c 2074  , --bootstrap, t
+00005fd0: 6865 206e 756d 6265 7220 6f66 2062 6f6f  he number of boo
+00005fe0: 7473 7472 6170 2072 6570 6c69 6361 7465  tstrap replicate
+00005ff0: 7320 6f66 206f 7274 686f 6c6f 6720 4b73  s of ortholog Ks
+00006000: 2064 6973 7472 6962 7574 696f 6e20 696e   distribution in
+00006010: 206d 6978 6564 2070 6c6f 740a 6060 600a   mixed plot.```.
+00006020: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
+00006030: 6420 6d69 7860 2063 616e 2072 6561 6c69  d mix` can reali
+00006040: 7a65 2074 6865 206d 6978 7475 7265 206d  ze the mixture m
+00006050: 6f64 656c 2063 6c75 7374 6572 696e 6720  odel clustering 
+00006060: 616e 616c 7973 6973 206f 6620 2a4b 2a3c  analysis of *K*<
+00006070: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+00006080: 6469 7374 7269 6275 7469 6f6e 2e0a 6060  distribution..``
+00006090: 600a 7767 6420 6d69 7820 6b73 5f64 6174  `.wgd mix ks_dat
+000060a0: 6166 696c 6520 286f 7074 696f 6e29 0a2d  afile (option).-
+000060b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000060c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000060d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000060e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000060f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d0a  ---------------.
+00006100: 2d66 2c20 2d2d 6669 6c74 6572 732c 2074  -f, --filters, t
+00006110: 6865 2063 7574 6f66 6620 616c 6967 6e6d  he cutoff alignm
+00006120: 656e 7420 6c65 6e67 7468 2c20 6465 6661  ent length, defa
+00006130: 756c 7420 3330 300a 2d72 2c20 2d2d 6b73  ult 300.-r, --ks
+00006140: 5f72 616e 6765 2c20 7468 6520 4b73 2072  _range, the Ks r
+00006150: 616e 6765 2074 6f20 6265 2063 6f6e 7369  ange to be consi
+00006160: 6465 7265 642c 2064 6566 6175 6c74 2028  dered, default (
+00006170: 302c 2035 290a 2d62 2c20 2d2d 6269 6e73  0, 5).-b, --bins
+00006180: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
+00006190: 6269 6e73 2069 6e20 4b73 2064 6973 7472  bins in Ks distr
+000061a0: 6962 7574 696f 6e2c 2064 6566 6175 6c74  ibution, default
+000061b0: 2035 300a 2d6f 2c20 2d2d 6f75 7464 6972   50.-o, --outdir
+000061c0: 2c20 7468 6520 6f75 7470 7574 2064 6972  , the output dir
+000061d0: 6563 746f 7279 2c20 6465 6661 756c 7420  ectory, default 
+000061e0: 7767 645f 6d69 780a 2d2d 6d65 7468 6f64  wgd_mix.--method
+000061f0: 2c20 7768 6963 6820 6d69 7874 7572 6520  , which mixture 
+00006200: 6d6f 6465 6c20 746f 2075 7365 2c20 6465  model to use, de
+00006210: 6661 756c 7420 676d 6d0a 2d6e 2c20 2d2d  fault gmm.-n, --
+00006220: 636f 6d70 6f6e 656e 7473 2c20 7468 6520  components, the 
+00006230: 7261 6e67 6520 6f66 2074 6865 206e 756d  range of the num
+00006240: 6265 7220 6f66 2063 6f6d 706f 6e65 6e74  ber of component
+00006250: 7320 746f 2066 6974 2c20 6465 6661 756c  s to fit, defaul
+00006260: 7420 2831 2c20 3429 0a2d 672c 202d 2d67  t (1, 4).-g, --g
+00006270: 616d 6d61 2c20 7468 6520 6761 6d6d 6120  amma, the gamma 
+00006280: 7061 7261 6d65 7465 7220 666f 7220 6267  parameter for bg
+00006290: 6d6d 206d 6f64 656c 732c 2064 6566 6175  mm models, defau
+000062a0: 6c74 2030 2e30 3031 0a2d 6e69 2c20 2d2d  lt 0.001.-ni, --
+000062b0: 6e5f 696e 6974 2c20 7468 6520 6e75 6d62  n_init, the numb
+000062c0: 6572 206f 6620 6b2d 6d65 616e 7320 696e  er of k-means in
+000062d0: 6974 6961 6c69 7a61 7469 6f6e 732c 2064  itializations, d
+000062e0: 6566 6175 6c74 2032 3030 0a2d 6d69 2c20  efault 200.-mi, 
+000062f0: 2d2d 6d61 785f 6974 6572 2c20 7468 6520  --max_iter, the 
+00006300: 6d61 7869 6d75 6d20 6e75 6d62 6572 206f  maximum number o
+00006310: 6620 6974 6572 6174 696f 6e73 2c20 6465  f iterations, de
+00006320: 6661 756c 7420 3230 300a 6060 600a 0a54  fault 200.```..T
+00006330: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
+00006340: 7065 616b 6020 6361 6e20 7265 616c 697a  peak` can realiz
+00006350: 6520 7468 6520 7365 6172 6368 206f 6620  e the search of 
+00006360: 6372 6564 6961 626c 6520 2a4b 2a3c 7375  crediable *K*<su
+00006370: 623e 533c 2f73 7562 3e20 7261 6e67 6520  b>S</sub> range 
+00006380: 7573 6564 2069 6e20 5747 4420 6461 7469  used in WGD dati
+00006390: 6e67 2e0a 6060 600a 7767 6420 7065 616b  ng..```.wgd peak
+000063a0: 206b 735f 6461 7461 6669 6c65 2028 6f70   ks_datafile (op
+000063b0: 7469 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d  tion).----------
+000063c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000063d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000063e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000063f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006400: 2d2d 2d2d 2d2d 0a2d 6170 2c20 2d2d 616e  ------.-ap, --an
+00006410: 6368 6f72 706f 696e 7473 2c20 7468 6520  chorpoints, the 
+00006420: 616e 6368 6f72 2070 6f69 6e74 7320 6461  anchor points da
+00006430: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
+00006440: 4e6f 6e65 0a2d 736d 2c20 2d2d 7365 676d  None.-sm, --segm
+00006450: 656e 7473 2c20 7468 6520 7365 676d 656e  ents, the segmen
+00006460: 7473 2064 6174 6166 696c 652c 2064 6566  ts datafile, def
+00006470: 6175 6c74 204e 6f6e 650a 2d6c 652c 202d  ault None.-le, -
+00006480: 2d6c 6973 7465 6c65 6d65 6e74 732c 2074  -listelements, t
+00006490: 6865 206c 6973 7473 6567 6d65 6e74 7320  he listsegments 
+000064a0: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
+000064b0: 7420 4e6f 6e65 200a 2d6d 702c 202d 2d6d  t None .-mp, --m
+000064c0: 756c 7469 706c 6963 6f6e 7061 6972 732c  ultipliconpairs,
+000064d0: 2074 6865 206d 756c 7469 706c 6963 6f6e   the multiplicon
+000064e0: 7061 6972 7320 6461 7461 6669 6c65 2c20  pairs datafile, 
+000064f0: 6465 6661 756c 7420 4e6f 6e65 0a2d 6f2c  default None.-o,
+00006500: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
+00006510: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
+00006520: 2064 6566 6175 6c74 2077 6764 5f70 6561   default wgd_pea
+00006530: 6b0a 2d61 662c 202d 2d61 6c69 676e 6669  k.-af, --alignfi
+00006540: 6c74 6572 2c20 6375 746f 6666 2066 6f72  lter, cutoff for
+00006550: 2061 6c69 676e 6d65 6e74 2069 6465 6e74   alignment ident
+00006560: 6974 792c 206c 656e 6774 6820 616e 6420  ity, length and 
+00006570: 636f 7665 7261 6765 2c20 6465 6661 756c  coverage, defaul
+00006580: 7420 302e 302c 2030 2c20 302e 300a 2d72  t 0.0, 0, 0.0.-r
+00006590: 2c20 2d2d 6b73 7261 6e67 652c 2072 616e  , --ksrange, ran
+000065a0: 6765 206f 6620 4b73 2074 6f20 6265 2061  ge of Ks to be a
+000065b0: 6e61 6c79 7a65 642c 2064 6566 6175 6c74  nalyzed, default
+000065c0: 2028 302c 2035 290a 2d62 772c 202d 2d62   (0, 5).-bw, --b
+000065d0: 696e 5f77 6964 7468 2c20 6261 6e64 7769  in_width, bandwi
+000065e0: 6474 6820 6f66 204b 7320 6469 7374 7269  dth of Ks distri
+000065f0: 6275 7469 6f6e 2c20 6465 6661 756c 7420  bution, default 
+00006600: 302e 310a 2d69 632c 202d 2d77 6569 6768  0.1.-ic, --weigh
+00006610: 7473 5f6f 7574 6c69 6572 735f 696e 636c  ts_outliers_incl
+00006620: 7564 6564 2c20 666c 6167 206f 7074 696f  uded, flag optio
+00006630: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
+00006640: 636c 7564 6520 4b73 206f 7574 6c69 6572  clude Ks outlier
+00006650: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
+00006660: 6173 2073 6574 2c20 4b73 206f 7574 6c69  as set, Ks outli
+00006670: 6572 7320 776f 756c 6420 6265 2069 6e63  ers would be inc
+00006680: 6c75 6465 6420 696e 2074 6865 2061 6e61  luded in the ana
+00006690: 6c79 7369 730a 2d6d 2c20 2d2d 6d65 7468  lysis.-m, --meth
+000066a0: 6f64 2c20 7768 6963 6820 6d69 7874 7572  od, which mixtur
+000066b0: 6520 6d6f 6465 6c20 746f 2075 7365 2c20  e model to use, 
+000066c0: 6465 6661 756c 7420 676d 6d0a 2d2d 7365  default gmm.--se
+000066d0: 6564 2c20 7261 6e64 6f6d 2073 6565 6420  ed, random seed 
+000066e0: 6769 7665 6e20 746f 2069 6e69 7469 616c  given to initial
+000066f0: 697a 6174 696f 6e2c 2064 6566 6175 6c74  ization, default
+00006700: 2032 3335 3238 3930 0a2d 6569 2c20 2d2d   2352890.-ei, --
+00006710: 656d 5f69 7465 722c 2074 6865 206e 756d  em_iter, the num
+00006720: 6265 7220 6f66 2045 4d20 6974 6572 6174  ber of EM iterat
+00006730: 696f 6e73 2074 6f20 7065 7266 6f72 6d2c  ions to perform,
+00006740: 2064 6566 6175 6c74 2032 3030 0a2d 6e69   default 200.-ni
+00006750: 2c20 2d2d 6e5f 696e 6974 2c20 7468 6520  , --n_init, the 
+00006760: 6e75 6d62 6572 206f 6620 6b2d 6d65 616e  number of k-mean
+00006770: 7320 696e 6974 6961 6c69 7a61 7469 6f6e  s initialization
+00006780: 732c 2064 6566 6175 6c74 2032 3030 0a2d  s, default 200.-
+00006790: 6e2c 202d 2d63 6f6d 706f 6e65 6e74 732c  n, --components,
+000067a0: 2074 6865 2072 616e 6765 206f 6620 7468   the range of th
+000067b0: 6520 6e75 6d62 6572 206f 6620 636f 6d70  e number of comp
+000067c0: 6f6e 656e 7473 2074 6f20 6669 742c 2064  onents to fit, d
+000067d0: 6566 6175 6c74 2028 312c 2034 290a 2d67  efault (1, 4).-g
+000067e0: 2c20 2d2d 6761 6d6d 612c 2074 6865 2067  , --gamma, the g
+000067f0: 616d 6d61 2070 6172 616d 6574 6572 2066  amma parameter f
+00006800: 6f72 2062 676d 6d20 6d6f 6465 6c73 2c20  or bgmm models, 
+00006810: 6465 6661 756c 7420 3165 2d33 0a2d 2d62  default 1e-3.--b
+00006820: 6f6f 7473 2c20 7468 6520 6e75 6d62 6572  oots, the number
+00006830: 206f 6620 626f 6f74 7374 7261 7020 7265   of bootstrap re
+00006840: 706c 6963 6174 6573 206f 6620 6b64 652c  plicates of kde,
+00006850: 2064 6566 6175 6c74 2032 3030 0a2d 2d77   default 200.--w
+00006860: 6569 6768 7465 642c 2066 6c61 6720 6f70  eighted, flag op
+00006870: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00006880: 2075 7365 206e 6f64 652d 7765 6967 6874   use node-weight
+00006890: 6564 206d 6574 686f 6420 6f66 2064 652d  ed method of de-
+000068a0: 7265 6475 6e64 616e 6379 2c20 6966 2074  redundancy, if t
+000068b0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+000068c0: 2074 6865 206e 6f64 652d 7765 6967 6874   the node-weight
+000068d0: 6564 206d 6574 686f 6420 776f 756c 6420  ed method would 
+000068e0: 6265 2075 7365 640a 2d70 2c20 2d2d 706c  be used.-p, --pl
+000068f0: 6f74 2c20 7468 6520 706c 6f74 7469 6e67  ot, the plotting
+00006900: 206d 6574 686f 6420 746f 2062 6520 7573   method to be us
+00006910: 6564 2c20 6465 6661 756c 7420 6964 656e  ed, default iden
+00006920: 7469 6361 6c0a 2d62 6d2c 202d 2d62 775f  tical.-bm, --bw_
+00006930: 6d65 7468 6f64 2c20 7468 6520 6261 6e64  method, the band
+00006940: 7769 6474 6820 6d65 7468 6f64 2074 6f20  width method to 
+00006950: 6265 2075 7365 6420 696e 2061 6e61 6c79  be used in analy
+00006960: 7a69 6e67 2074 6865 2070 6561 6b20 6f66  zing the peak of
+00006970: 2057 4744 2064 6174 6573 2c20 6465 6661   WGD dates, defa
+00006980: 756c 7420 7369 6c76 6572 6d61 6e0a 2d2d  ult silverman.--
+00006990: 6e5f 6d65 646f 6964 732c 2074 6865 206e  n_medoids, the n
+000069a0: 756d 6265 7220 6f66 206d 6564 6f69 6473  umber of medoids
+000069b0: 2074 6f20 6669 742c 2064 6566 6175 6c74   to fit, default
+000069c0: 2032 0a2d 6b6d 2c20 2d2d 6b64 656d 6574   2.-km, --kdemet
+000069d0: 686f 642c 2074 6865 206b 6465 206d 6574  hod, the kde met
+000069e0: 686f 6420 746f 2062 6520 7573 6564 2069  hod to be used i
+000069f0: 6e20 616e 616c 797a 696e 6720 7468 6520  n analyzing the 
+00006a00: 7065 616b 206f 6620 5747 4420 6461 7465  peak of WGD date
+00006a10: 732c 206b 6d65 646f 6964 7320 616e 616c  s, kmedoids anal
+00006a20: 7973 6973 206f 7220 7468 6520 6261 7369  ysis or the basi
+00006a30: 6320 4b73 2070 6c6f 7474 696e 672c 2064  c Ks plotting, d
+00006a40: 6566 6175 6c74 2073 6369 7079 0a2d 2d6e  efault scipy.--n
+00006a50: 5f63 6c75 7374 6572 732c 2074 6865 206e  _clusters, the n
+00006a60: 756d 6265 7220 6f66 2063 6c75 7374 6572  umber of cluster
+00006a70: 7320 746f 2070 6c6f 7420 456c 626f 7720  s to plot Elbow 
+00006a80: 6c6f 7373 2066 756e 6374 696f 6e2c 2064  loss function, d
+00006a90: 6566 6175 6c74 2035 0a2d 6764 2c20 2d2d  efault 5.-gd, --
+00006aa0: 6775 6964 652c 2074 6865 2072 6567 696d  guide, the regim
+00006ab0: 6520 7265 7369 6469 6e67 2061 6e63 686f  e residing ancho
+00006ac0: 7273 2c20 6465 6661 756c 7420 5365 676d  rs, default Segm
+00006ad0: 656e 740a 2d70 7263 742c 202d 2d70 726f  ent.-prct, --pro
+00006ae0: 6d69 6e65 6e63 655f 6375 746f 6666 2c20  minence_cutoff, 
+00006af0: 7468 6520 7072 6f6d 696e 656e 6365 2063  the prominence c
+00006b00: 7574 6f66 6620 6f66 2061 6363 6570 7461  utoff of accepta
+00006b10: 626c 6520 7065 616b 7320 696e 2070 6561  ble peaks in pea
+00006b20: 6b20 6669 6e64 696e 6720 7374 6570 732c  k finding steps,
+00006b30: 2064 6566 6175 6c74 2030 2e31 0a2d 7268   default 0.1.-rh
+00006b40: 2c20 2d2d 7265 6c5f 6865 6967 6874 2c20  , --rel_height, 
+00006b50: 7468 6520 7265 6c61 7469 7665 2068 6569  the relative hei
+00006b60: 6768 7420 6174 2077 6869 6368 2074 6865  ght at which the
+00006b70: 2070 6561 6b20 7769 6474 6820 6973 206d   peak width is m
+00006b80: 6561 7375 7265 642c 2064 6566 6175 6c74  easured, default
+00006b90: 2030 2e34 0a2d 6b64 2c20 2d2d 6b73 746f   0.4.-kd, --ksto
+00006ba0: 6461 7465 2c20 7468 6520 7261 6e67 6520  date, the range 
+00006bb0: 6f66 204b 7320 746f 2062 6520 6461 7465  of Ks to be date
+00006bc0: 6420 696e 2068 6575 7269 7374 6963 2073  d in heuristic s
+00006bd0: 6561 7263 682c 2064 6566 6175 6c74 2028  earch, default (
+00006be0: 302e 352c 2031 2e35 290a 2d78 6c2c 202d  0.5, 1.5).-xl, -
+00006bf0: 2d78 6c69 6d2c 2074 6865 2078 2061 7869  -xlim, the x axi
+00006c00: 7320 6c69 6d69 7420 6f66 2047 4d4d 204b  s limit of GMM K
+00006c10: 7320 6469 7374 7269 6275 7469 6f6e 0a2d  s distribution.-
+00006c20: 796c 2c20 2d2d 796c 696d 2c20 7468 6520  yl, --ylim, the 
+00006c30: 7920 6178 6973 206c 696d 6974 206f 6620  y axis limit of 
+00006c40: 474d 4d20 4b73 2064 6973 7472 6962 7574  GMM Ks distribut
+00006c50: 696f 6e0a 2d2d 6d61 6e75 616c 7365 742c  ion.--manualset,
+00006c60: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00006c70: 6574 6865 7220 746f 206f 7574 7075 7420  ether to output 
+00006c80: 616e 6368 6f72 2070 6169 7273 2077 6974  anchor pairs wit
+00006c90: 6820 6d61 6e75 616c 6c79 2073 6574 204b  h manually set K
+00006ca0: 7320 7261 6e67 652c 2069 6620 7468 6520  s range, if the 
+00006cb0: 666c 6167 2077 6173 2073 6574 2c20 6d61  flag was set, ma
+00006cc0: 6e75 616c 6c79 2073 6574 204b 7320 7261  nually set Ks ra
+00006cd0: 6e67 6520 776f 756c 6420 6265 206f 7574  nge would be out
+00006ce0: 7075 7474 6564 0a2d 2d63 692c 2074 6865  putted.--ci, the
+00006cf0: 2063 6f6e 6669 6465 6e63 6520 6c65 7665   confidence leve
+00006d00: 6c20 6f66 206c 6f67 2d6e 6f72 6d61 6c20  l of log-normal 
+00006d10: 6469 7374 7269 6275 7469 6f6e 2074 6f20  distribution to 
+00006d20: 6461 7465 2c20 6465 6661 756c 7420 3935  date, default 95
+00006d30: 0a2d 2d68 6472 2c20 7468 6520 6869 6768  .--hdr, the high
+00006d40: 6573 7420 6465 6e73 6974 7920 7265 6769  est density regi
+00006d50: 6f6e 2028 4844 5229 2061 7070 6c69 6564  on (HDR) applied
+00006d60: 2069 6e20 7468 6520 7365 676d 656e 742d   in the segment-
+00006d70: 6775 6964 6564 2061 6e63 686f 7220 7061  guided anchor pa
+00006d80: 6972 204b 7320 6469 7374 7269 6275 7469  ir Ks distributi
+00006d90: 6f6e 2c20 6465 6661 756c 7420 3935 0a2d  on, default 95.-
+00006da0: 2d68 6575 7269 7374 6963 2c20 666c 6167  -heuristic, flag
+00006db0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
+00006dc0: 2074 6f20 696e 6974 6961 7465 2068 6575   to initiate heu
+00006dd0: 7269 7374 6963 206d 6574 686f 6420 6f66  ristic method of
+00006de0: 2064 6566 696e 696e 6720 4349 2066 6f72   defining CI for
+00006df0: 2064 6174 696e 672c 2069 6620 7468 6520   dating, if the 
+00006e00: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+00006e10: 6520 6865 7572 6973 7469 6320 6d65 7468  e heuristic meth
+00006e20: 6f64 2077 6f75 6c64 2062 6520 696e 6974  od would be init
+00006e30: 6961 7465 640a 2d6b 632c 202d 2d6b 7363  iated.-kc, --ksc
+00006e40: 7574 6f66 662c 2074 6865 204b 7320 7361  utoff, the Ks sa
+00006e50: 7475 7261 7469 6f6e 2063 7574 6f66 6620  turation cutoff 
+00006e60: 696e 2064 6174 696e 672c 2064 6566 6175  in dating, defau
+00006e70: 6c74 2035 0a2d 2d6b 6565 7074 6d70 6669  lt 5.--keeptmpfi
+00006e80: 672c 2066 6c61 6720 6f70 7469 6f6e 2c20  g, flag option, 
+00006e90: 7768 6574 6865 7220 746f 206b 6565 7020  whether to keep 
+00006ea0: 7465 6d70 6f72 6172 7920 6669 6775 7265  temporary figure
+00006eb0: 7320 696e 2070 6561 6b20 6669 6e64 696e  s in peak findin
+00006ec0: 6720 7072 6f63 6573 732c 2069 6620 7468  g process, if th
+00006ed0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00006ee0: 7468 6f73 6520 6669 6775 7265 7320 776f  those figures wo
+00006ef0: 756c 6420 6265 206b 6570 740a 6060 600a  uld be kept.```.
+00006f00: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
+00006f10: 6420 7379 6e60 2063 616e 2072 6561 6c69  d syn` can reali
+00006f20: 7a65 2074 6865 2069 6e74 7261 2d20 616e  ze the intra- an
+00006f30: 6420 696e 7465 722d 7370 6563 6966 6963  d inter-specific
+00006f40: 2073 796e 7465 6e79 2069 6e66 6572 656e   synteny inferen
+00006f50: 6365 2e0a 6060 600a 7767 6420 7379 6e20  ce..```.wgd syn 
+00006f60: 6661 6d69 6c69 6573 2067 6666 7320 286f  families gffs (o
+00006f70: 7074 696f 6e29 0a2d 2d2d 2d2d 2d2d 2d2d  ption).---------
+00006f80: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006f90: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006fa0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006fb0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006fc0: 2d2d 2d2d 2d2d 2d0a 2d6b 732c 202d 2d6b  -------.-ks, --k
+00006fd0: 735f 6469 7374 7269 6275 7469 6f6e 2c20  s_distribution, 
+00006fe0: 6b73 2064 6973 7472 6962 7574 696f 6e20  ks distribution 
+00006ff0: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
+00007000: 7420 4e6f 6e65 0a2d 6f2c 202d 2d6f 7574  t None.-o, --out
+00007010: 6469 722c 2074 6865 206f 7574 7075 7420  dir, the output 
+00007020: 6469 7265 6374 6f72 792c 2064 6566 6175  directory, defau
+00007030: 6c74 2077 6764 5f73 796e 0a2d 662c 202d  lt wgd_syn.-f, -
+00007040: 2d66 6561 7475 7265 2c20 7468 6520 6665  -feature, the fe
+00007050: 6174 7572 6520 666f 7220 7061 7273 696e  ature for parsin
+00007060: 6720 6765 6e65 2049 4473 2066 726f 6d20  g gene IDs from 
+00007070: 4746 4620 6669 6c65 732c 2064 6566 6175  GFF files, defau
+00007080: 6c74 2067 656e 650a 2d61 2c20 2d2d 6174  lt gene.-a, --at
+00007090: 7472 6962 7574 652c 2074 6865 2061 7474  tribute, the att
+000070a0: 7269 6275 7465 2066 6f72 2070 6172 7369  ribute for parsi
+000070b0: 6e67 2074 6865 2067 656e 6520 4944 7320  ng the gene IDs 
+000070c0: 6672 6f6d 2074 6865 2047 4646 2066 696c  from the GFF fil
+000070d0: 6573 2c20 6465 6661 756c 7420 4944 0a2d  es, default ID.-
+000070e0: 6174 672c 202d 2d61 6464 6974 696f 6e61  atg, --additiona
+000070f0: 6c67 6666 696e 666f 2c20 7468 6520 6665  lgffinfo, the fe
+00007100: 6174 7572 6520 616e 6420 6174 7472 6962  ature and attrib
+00007110: 7574 6520 696e 666f 726d 6174 696f 6e20  ute information 
+00007120: 6f66 2061 6464 6974 696f 6e61 6c20 6766  of additional gf
+00007130: 6633 2066 696c 6573 2069 6620 6469 6666  f3 files if diff
+00007140: 6572 656e 7420 696e 2074 6865 2066 6f72  erent in the for
+00007150: 6d61 7420 6f66 2028 6665 6174 7572 653b  mat of (feature;
+00007160: 6174 7472 6962 7574 6529 272c 2064 6566  attribute)', def
+00007170: 6175 6c74 204e 6f6e 650a 2d6d 6c2c 202d  ault None.-ml, -
+00007180: 2d6d 696e 6c65 6e2c 2074 6865 206d 696e  -minlen, the min
+00007190: 696d 756d 206c 656e 6774 6820 6f66 2061  imum length of a
+000071a0: 2073 6361 6666 6f6c 6420 746f 2062 6520   scaffold to be 
+000071b0: 696e 636c 7564 6564 2069 6e20 646f 7470  included in dotp
+000071c0: 6c6f 742c 2064 6566 6175 6c74 202d 312c  lot, default -1,
+000071d0: 2069 6620 2d31 2077 6173 2073 6574 2c20   if -1 was set, 
+000071e0: 7468 6520 3130 2520 6f66 2074 6865 206c  the 10% of the l
+000071f0: 6f6e 6765 7374 2073 6361 6666 6f6c 6420  ongest scaffold 
+00007200: 776f 756c 6420 6265 2073 6574 0a2d 6d73  would be set.-ms
+00007210: 2c20 2d2d 6d61 7873 697a 652c 2074 6865  , --maxsize, the
+00007220: 206d 6178 696d 756d 2066 616d 696c 7920   maximum family 
+00007230: 7369 7a65 2074 6f20 6265 2069 6e63 6c75  size to be inclu
+00007240: 6465 642c 2064 6566 6175 6c74 2032 3030  ded, default 200
+00007250: 0a2d 722c 202d 2d6b 735f 7261 6e67 652c  .-r, --ks_range,
+00007260: 2074 6865 204b 7320 7261 6e67 6520 696e   the Ks range in
+00007270: 2063 6f6c 6f72 6564 2064 6f74 706c 6f74   colored dotplot
+00007280: 2c20 6465 6661 756c 7420 2830 2c20 3529  , default (0, 5)
+00007290: 0a2d 2d69 6164 686f 7265 5f6f 7074 696f  .--iadhore_optio
+000072a0: 6e73 2c20 7468 6520 7061 7261 6d65 7465  ns, the paramete
+000072b0: 7220 7365 7474 696e 6720 696e 2069 6164  r setting in iad
+000072c0: 686f 7265 2c20 6465 6661 756c 7420 6173  hore, default as
+000072d0: 2061 2073 7472 696e 6720 6f66 206c 656e   a string of len
+000072e0: 6774 6820 7a65 726f 0a2d 6d67 2c20 2d2d  gth zero.-mg, --
+000072f0: 6d69 6e73 6567 6c65 6e2c 2074 6865 206d  minseglen, the m
+00007300: 696e 696d 756d 206c 656e 6774 6820 6f66  inimum length of
+00007310: 2073 6567 6d65 6e74 7320 746f 2069 6e63   segments to inc
+00007320: 6c75 6465 2069 6e20 7261 7469 6f20 6966  lude in ratio if
+00007330: 203c 3d20 312c 2064 6566 6175 6c74 2031   <= 1, default 1
+00007340: 3030 3030 0a2d 6b72 2c20 2d2d 6b65 6570  0000.-kr, --keep
+00007350: 7265 6475 6e2c 2066 6c61 6720 6f70 7469  redun, flag opti
+00007360: 6f6e 2c20 7768 6574 6865 7220 746f 206b  on, whether to k
+00007370: 6565 7020 7265 6475 6e64 616e 7420 6d75  eep redundant mu
+00007380: 6c74 6970 6c69 636f 6e73 2c20 6966 2074  ltiplicons, if t
+00007390: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+000073a0: 2074 6865 2072 6564 756e 6461 6e74 206d   the redundant m
+000073b0: 756c 7469 706c 6963 6f6e 7320 776f 756c  ultiplicons woul
+000073c0: 6420 6265 206b 6570 740a 2d6d 676e 2c20  d be kept.-mgn, 
+000073d0: 2d2d 6d69 6e67 656e 656e 756d 2c20 7468  --mingenenum, th
+000073e0: 6520 6d69 6e69 6d75 6d20 6e75 6d62 6572  e minimum number
+000073f0: 206f 6620 6765 6e65 7320 666f 7220 6120   of genes for a 
+00007400: 7365 676d 656e 7420 746f 2062 6520 636f  segment to be co
+00007410: 6e73 6964 6572 6564 2c20 6465 6661 756c  nsidered, defaul
+00007420: 7420 3330 0a2d 6473 2c20 2d2d 646f 7473  t 30.-ds, --dots
+00007430: 697a 652c 2074 6865 2064 6f74 2073 697a  ize, the dot siz
+00007440: 6520 696e 2064 6f74 2070 6c6f 742c 2064  e in dot plot, d
+00007450: 6566 6175 6c74 2030 2e33 0a2d 6161 2c20  efault 0.3.-aa, 
+00007460: 2d2d 6170 616c 7068 612c 2074 6865 206f  --apalpha, the o
+00007470: 7061 6369 7479 206f 6620 616e 6368 6f72  pacity of anchor
+00007480: 2064 6f74 732c 2064 6566 6175 6c74 2031   dots, default 1
+00007490: 0a2d 6861 2c20 2d2d 686f 616c 7068 612c  .-ha, --hoalpha,
+000074a0: 2074 6865 206f 7061 6369 7479 206f 6620   the opacity of 
+000074b0: 686f 6d6f 6c6f 6720 646f 7473 2c20 6465  homolog dots, de
+000074c0: 6661 756c 7420 300a 2d73 7274 2c20 2d2d  fault 0.-srt, --
+000074d0: 7368 6f77 7265 616c 7469 636b 2c20 666c  showrealtick, fl
+000074e0: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+000074f0: 6572 2074 6f20 7368 6f77 2074 6865 2072  er to show the r
+00007500: 6561 6c20 7469 636b 2069 6e20 6765 6e65  eal tick in gene
+00007510: 7320 6f72 2062 6173 6573 2c20 6966 2074  s or bases, if t
+00007520: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00007530: 2074 6865 2072 6561 6c20 7469 636b 2077   the real tick w
+00007540: 6f75 6c64 2062 6520 7368 6f77 6564 0a2d  ould be showed.-
+00007550: 746c 732c 202d 2d74 6963 6b6c 6162 656c  tls, --ticklabel
+00007560: 7369 7a65 2c20 7468 6520 6c61 6265 6c20  size, the label 
+00007570: 7369 7a65 206f 6620 7469 636b 2c20 6465  size of tick, de
+00007580: 6661 756c 7420 350a 2d67 722c 202d 2d67  fault 5.-gr, --g
+00007590: 6973 7472 622c 2066 6c61 6720 6f70 7469  istrb, flag opti
+000075a0: 6f6e 2c20 7768 6574 6865 7220 746f 2075  on, whether to u
+000075b0: 7365 2067 6973 745f 7261 696e 626f 7720  se gist_rainbow 
+000075c0: 6173 2063 6f6c 6f72 206d 6170 206f 6620  as color map of 
+000075d0: 646f 7470 6c6f 740a 2d6e 2c20 2d2d 6e74  dotplot.-n, --nt
+000075e0: 6872 6561 6473 2c20 7468 6520 6e75 6d62  hreads, the numb
+000075f0: 6572 206f 6620 7468 7265 6164 7320 746f  er of threads to
+00007600: 2075 7365 2069 6e20 7379 6e74 656e 7920   use in synteny 
+00007610: 696e 6665 7265 6e63 652c 2064 6566 6175  inference, defau
+00007620: 6c74 2034 0a60 6060 0a0a 5468 6520 7072  lt 4.```..The pr
+00007630: 6f67 7261 6d20 6077 6764 2076 697a 6020  ogram `wgd viz` 
+00007640: 6361 6e20 7265 616c 697a 6520 7468 6520  can realize the 
+00007650: 7669 7375 616c 697a 6174 696f 6e20 6f66  visualization of
+00007660: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00007670: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
+00007680: 6e20 616e 6420 7379 6e74 656e 792e 0a60  n and synteny..`
+00007690: 6060 0a77 6764 2076 697a 2028 6f70 7469  ``.wgd viz (opti
+000076a0: 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  on).------------
+000076b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000076c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000076d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000076e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000076f0: 2d2d 2d2d 0a2d 642c 202d 2d64 6174 6166  ----.-d, --dataf
+00007700: 696c 652c 2074 6865 204b 7320 6461 7461  ile, the Ks data
+00007710: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
+00007720: 6e65 0a2d 6f2c 202d 2d6f 7574 6469 722c  ne.-o, --outdir,
+00007730: 2074 6865 206f 7574 7075 7420 6469 7265   the output dire
+00007740: 6374 6f72 792c 2064 6566 6175 6c74 2077  ctory, default w
+00007750: 6764 5f76 697a 0a2d 7372 2c20 2d2d 7370  gd_viz.-sr, --sp
+00007760: 6169 722c 2074 6865 2073 7065 6369 6573  air, the species
+00007770: 2070 6169 7220 746f 2062 6520 706c 6f74   pair to be plot
+00007780: 7465 642c 2064 6566 6175 6c74 204e 6f6e  ted, default Non
+00007790: 652c 2074 6869 7320 6f70 7469 6f6e 2063  e, this option c
+000077a0: 616e 2062 6520 7072 6f76 6964 6564 206d  an be provided m
+000077b0: 756c 7469 706c 6520 7469 6d65 730a 2d66  ultiple times.-f
+000077c0: 612c 202d 2d66 6f63 7573 3261 6c6c 2c20  a, --focus2all, 
+000077d0: 7365 7420 666f 6361 6c20 7370 6563 6965  set focal specie
+000077e0: 7320 616e 6420 6c65 7420 7370 6563 6965  s and let specie
+000077f0: 7320 7061 6972 2074 6f20 6265 2062 6574  s pair to be bet
+00007800: 7765 656e 2066 6f63 616c 2061 6e64 2061  ween focal and a
+00007810: 6c6c 2074 6865 2072 656d 6169 6e69 6e67  ll the remaining
+00007820: 2073 7065 6369 6573 2c20 6465 6661 756c   species, defaul
+00007830: 7420 4e6f 6e65 0a2d 6773 2c20 2d2d 6773  t None.-gs, --gs
+00007840: 6d61 702c 2074 6865 2067 656e 6520 6e61  map, the gene na
+00007850: 6d65 2d73 7065 6369 6573 206e 616d 6520  me-species name 
+00007860: 6d61 702c 2064 6566 6175 6c74 204e 6f6e  map, default Non
+00007870: 650a 2d73 702c 202d 2d73 7065 6369 6573  e.-sp, --species
+00007880: 7472 6565 2c20 7468 6520 7370 6563 6965  tree, the specie
+00007890: 7320 7472 6565 2074 6f20 7065 7266 6f72  s tree to perfor
+000078a0: 6d20 7261 7465 2063 6f72 7265 6374 696f  m rate correctio
+000078b0: 6e2c 2064 6566 6175 6c74 204e 6f6e 652c  n, default None,
+000078c0: 2069 6620 4e6f 6e65 2077 6173 2067 6976   if None was giv
+000078d0: 656e 2c20 7468 6520 7261 7465 2063 6f72  en, the rate cor
+000078e0: 7265 6374 696f 6e20 616e 616c 7973 6973  rection analysis
+000078f0: 2077 6f75 6c64 2062 6520 6361 6c6c 6564   would be called
+00007900: 206f 6666 0a2d 706b 2c20 2d2d 706c 6f74   off.-pk, --plot
+00007910: 6b64 652c 2066 6c61 6720 6f70 7469 6f6e  kde, flag option
+00007920: 2c20 7768 6574 6865 7220 746f 2070 6c6f  , whether to plo
+00007930: 7420 6b64 6520 6375 7276 6520 7570 6f6e  t kde curve upon
+00007940: 2068 6973 746f 6772 616d 2c20 6966 2074   histogram, if t
+00007950: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00007960: 206b 6465 2063 7572 7665 2077 6f75 6c64   kde curve would
+00007970: 2062 6520 6164 6465 640a 2d72 772c 202d   be added.-rw, -
+00007980: 2d72 6577 6569 6768 742c 2066 6c61 6720  -reweight, flag 
+00007990: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+000079a0: 746f 2072 6563 616c 6375 6c61 7465 2074  to recalculate t
+000079b0: 6865 2077 6569 6768 7420 7065 7220 7370  he weight per sp
+000079c0: 6563 6965 7320 7061 6972 2c20 6966 2074  ecies pair, if t
+000079d0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+000079e0: 2074 6865 2077 6569 6768 7420 776f 756c   the weight woul
+000079f0: 6420 6265 2072 6563 616c 6375 6c61 7465  d be recalculate
+00007a00: 640a 2d6f 722c 202d 2d6f 6e6c 7972 6f6f  d.-or, --onlyroo
+00007a10: 746f 7574 2c20 666c 6167 206f 7074 696f  tout, flag optio
+00007a20: 6e2c 2077 6865 7468 6572 2074 6f20 6f6e  n, whether to on
+00007a30: 6c79 2063 6f6e 6475 6374 2072 6174 6520  ly conduct rate 
+00007a40: 636f 7272 6563 7469 6f6e 2075 7369 6e67  correction using
+00007a50: 2074 6865 206f 7574 6772 6f75 7020 6174   the outgroup at
+00007a60: 2072 6f6f 7420 6173 206f 7574 6772 6f75   root as outgrou
+00007a70: 702c 2069 6620 7468 6520 666c 6167 2077  p, if the flag w
+00007a80: 6173 2073 6574 2c20 6f6e 6c79 2074 6865  as set, only the
+00007a90: 206f 7574 6772 6f75 7020 6174 2072 6f6f   outgroup at roo
+00007aa0: 7420 776f 756c 6420 6265 2075 7365 6420  t would be used 
+00007ab0: 6173 206f 7574 6772 6f75 700a 2d69 7465  as outgroup.-ite
+00007ac0: 722c 202d 2d65 6d5f 6974 6572 6174 696f  r, --em_iteratio
+00007ad0: 6e73 2c20 7468 6520 6d61 7869 6d75 6d20  ns, the maximum 
+00007ae0: 454d 2069 7465 7261 7469 6f6e 732c 2064  EM iterations, d
+00007af0: 6566 6175 6c74 2032 3030 0a2d 696e 6974  efault 200.-init
+00007b00: 2c20 2d2d 656d 5f69 6e69 7469 616c 697a  , --em_initializ
+00007b10: 6174 696f 6e73 2c20 7468 6520 6d61 7869  ations, the maxi
+00007b20: 6d75 6d20 454d 2069 6e69 7469 616c 697a  mum EM initializ
+00007b30: 6174 696f 6e73 2c20 6465 6661 756c 7420  ations, default 
+00007b40: 3230 300a 2d70 7263 742c 202d 2d70 726f  200.-prct, --pro
+00007b50: 6d69 6e65 6e63 655f 6375 746f 6666 2c20  minence_cutoff, 
+00007b60: 7468 6520 7072 6f6d 696e 656e 6365 2063  the prominence c
+00007b70: 7574 6f66 6620 6f66 2061 6363 6570 7461  utoff of accepta
+00007b80: 626c 6520 7065 616b 732c 2064 6566 6175  ble peaks, defau
+00007b90: 6c74 2030 2e31 0a2d 7268 2c20 2d2d 7265  lt 0.1.-rh, --re
+00007ba0: 6c5f 6865 6967 6874 2c20 7468 6520 7265  l_height, the re
+00007bb0: 6c61 7469 7665 2068 6569 6768 7420 6174  lative height at
+00007bc0: 2077 6869 6368 2074 6865 2070 6561 6b20   which the peak 
+00007bd0: 7769 6474 6820 6973 206d 6561 7375 7265  width is measure
+00007be0: 642c 2064 6566 6175 6c74 2030 2e34 0a2d  d, default 0.4.-
+00007bf0: 736d 2c20 2d2d 7365 676d 656e 7473 2c20  sm, --segments, 
+00007c00: 7468 6520 7365 676d 656e 7473 2064 6174  the segments dat
+00007c10: 6166 696c 652c 2064 6566 6175 6c74 204e  afile, default N
+00007c20: 6f6e 650a 2d6d 6c2c 202d 2d6d 696e 6c65  one.-ml, --minle
+00007c30: 6e2c 2074 6865 206d 696e 696d 756d 206c  n, the minimum l
+00007c40: 656e 6774 6820 6f66 2061 2073 6361 6666  ength of a scaff
+00007c50: 6f6c 6420 746f 2062 6520 696e 636c 7564  old to be includ
+00007c60: 6564 2069 6e20 646f 7470 6c6f 742c 2064  ed in dotplot, d
+00007c70: 6566 6175 6c74 202d 312c 2069 6620 2d31  efault -1, if -1
+00007c80: 2077 6173 2073 6574 2c20 7468 6520 3130   was set, the 10
+00007c90: 2520 6f66 2074 6865 206c 6f6e 6765 7374  % of the longest
+00007ca0: 2073 6361 6666 6f6c 6473 2077 696c 6c20   scaffolds will 
+00007cb0: 6265 2073 6574 0a2d 6d73 2c20 2d2d 6d61  be set.-ms, --ma
+00007cc0: 7873 697a 652c 2074 6865 206d 6178 696d  xsize, the maxim
+00007cd0: 756d 2066 616d 696c 7920 7369 7a65 2074  um family size t
+00007ce0: 6f20 6265 2069 6e63 6c75 6465 642c 2064  o be included, d
+00007cf0: 6566 6175 6c74 2032 3030 0a2d 6170 2c20  efault 200.-ap, 
+00007d00: 2d2d 616e 6368 6f72 706f 696e 7473 2c20  --anchorpoints, 
+00007d10: 7468 6520 616e 6368 6f72 2070 6f69 6e74  the anchor point
+00007d20: 7320 6461 7461 6669 6c65 2c20 6465 6661  s datafile, defa
+00007d30: 756c 7420 4e6f 6e65 0a2d 6d74 2c20 2d2d  ult None.-mt, --
+00007d40: 6d75 6c74 6970 6c69 636f 6e2c 2074 6865  multiplicon, the
+00007d50: 206d 756c 7469 706c 6963 6f6e 7320 6461   multiplicons da
+00007d60: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
+00007d70: 4e6f 6e65 0a2d 6774 2c20 2d2d 6765 6e65  None.-gt, --gene
+00007d80: 7461 626c 652c 2074 6865 2067 656e 6520  table, the gene 
+00007d90: 7461 626c 6520 6461 7461 6669 6c65 2c20  table datafile, 
+00007da0: 6465 6661 756c 7420 4e6f 6e65 0a2d 6d67  default None.-mg
+00007db0: 2c20 2d2d 6d69 6e73 6567 6c65 6e2c 2074  , --minseglen, t
+00007dc0: 6865 206d 696e 696d 756d 206c 656e 6774  he minimum lengt
+00007dd0: 6820 6f66 2073 6567 6d65 6e74 7320 746f  h of segments to
+00007de0: 2069 6e63 6c75 6465 2c20 696e 2072 6174   include, in rat
+00007df0: 696f 2069 6620 3c3d 2031 2c20 6465 6661  io if <= 1, defa
+00007e00: 756c 7420 3130 3030 300a 2d6d 676e 2c20  ult 10000.-mgn, 
+00007e10: 2d2d 6d69 6e67 656e 656e 756d 2c20 7468  --mingenenum, th
+00007e20: 6520 6d69 6e69 6d75 6d20 6e75 6d62 6572  e minimum number
+00007e30: 206f 6620 6765 6e65 7320 666f 7220 6120   of genes for a 
+00007e40: 7365 676d 656e 7420 746f 2062 6520 636f  segment to be co
+00007e50: 6e73 6964 6572 6564 2c20 6465 6661 756c  nsidered, defaul
+00007e60: 7420 3330 0a2d 6b72 2c20 2d2d 6b65 6570  t 30.-kr, --keep
+00007e70: 7265 6475 6e2c 2066 6c61 6720 6f70 7469  redun, flag opti
+00007e80: 6f6e 2c20 7768 6574 6865 7220 746f 206b  on, whether to k
+00007e90: 6565 7020 7265 6475 6e64 616e 7420 6d75  eep redundant mu
+00007ea0: 6c74 6970 6c69 636f 6e73 2c20 6966 2074  ltiplicons, if t
+00007eb0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00007ec0: 2074 6865 2072 6564 756e 6461 6e74 206d   the redundant m
+00007ed0: 756c 7469 706c 6963 6f6e 7320 776f 756c  ultiplicons woul
+00007ee0: 6420 6265 206b 6570 740a 2d65 706b 2c20  d be kept.-epk, 
+00007ef0: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
+00007f00: 732c 2065 7874 7261 2070 6172 616e 6f6d  s, extra paranom
+00007f10: 6520 4b73 2064 6174 6120 746f 2070 6c6f  e Ks data to plo
+00007f20: 7420 696e 2074 6865 206d 6978 6564 204b  t in the mixed K
+00007f30: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+00007f40: 6465 6661 756c 7420 4e6f 6e65 0a2d 7061  default None.-pa
+00007f50: 672c 202d 2d70 6c6f 7461 7067 6d6d 2c20  g, --plotapgmm, 
+00007f60: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00007f70: 7468 6572 2074 6f20 636f 6e64 7563 7420  ther to conduct 
+00007f80: 616e 6420 706c 6f74 206d 6978 7475 7265  and plot mixture
+00007f90: 206d 6f64 656c 696e 6720 6f66 2061 6e63   modeling of anc
+00007fa0: 686f 7220 4b73 2069 6e20 7468 6520 6d69  hor Ks in the mi
+00007fb0: 7865 6420 4b73 2064 6973 7472 6962 7574  xed Ks distribut
+00007fc0: 696f 6e2c 2069 6620 7468 6520 666c 6167  ion, if the flag
+00007fd0: 2077 6173 2073 6574 2c20 7468 6520 6d69   was set, the mi
+00007fe0: 7874 7572 6520 6d6f 6465 6c69 6e67 206f  xture modeling o
+00007ff0: 6620 616e 6368 6f72 204b 7320 776f 756c  f anchor Ks woul
+00008000: 6420 6265 2063 6f6e 6475 6374 6564 2061  d be conducted a
+00008010: 6e64 2070 6c6f 7474 6564 0a2d 7065 6d2c  nd plotted.-pem,
+00008020: 202d 2d70 6c6f 7465 6c6d 6d2c 2066 6c61   --plotelmm, fla
+00008030: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00008040: 7220 746f 2063 6f6e 6475 6374 2061 6e64  r to conduct and
+00008050: 2070 6c6f 7420 656c 6d6d 206d 6978 7475   plot elmm mixtu
+00008060: 7265 206d 6f64 656c 696e 6720 6f66 2070  re modeling of p
+00008070: 6172 616e 6f6d 6520 4b73 2069 6e20 7468  aranome Ks in th
+00008080: 6520 6d69 7865 6420 4b73 2064 6973 7472  e mixed Ks distr
+00008090: 6962 7574 696f 6e2c 2069 6620 7468 6520  ibution, if the 
+000080a0: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+000080b0: 6520 656c 6d6d 206d 6978 7475 7265 206d  e elmm mixture m
+000080c0: 6f64 656c 696e 6720 6f66 2070 6172 616e  odeling of paran
+000080d0: 6f6d 6520 4b73 2077 6f75 6c64 2062 6520  ome Ks would be 
+000080e0: 636f 6e64 7563 7465 6420 616e 6420 706c  conducted and pl
+000080f0: 6f74 7465 640a 2d63 2c20 2d2d 636f 6d70  otted.-c, --comp
+00008100: 6f6e 656e 7473 2c20 7468 6520 7261 6e67  onents, the rang
+00008110: 6520 6f66 2074 6865 206e 756d 6265 7220  e of the number 
+00008120: 6f66 2063 6f6d 706f 6e65 6e74 7320 746f  of components to
+00008130: 2066 6974 2069 6e20 616e 6368 6f72 204b   fit in anchor K
+00008140: 7320 6d69 7874 7572 6520 6d6f 6465 6c69  s mixture modeli
+00008150: 6e67 2c20 6465 6661 756c 7420 2831 2c34  ng, default (1,4
+00008160: 290a 2d70 7379 2c20 2d2d 706c 6f74 7379  ).-psy, --plotsy
+00008170: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
+00008180: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
+00008190: 6174 6520 7468 6520 7379 6e74 656e 7920  ate the synteny 
+000081a0: 706c 6f74 2c20 6f6e 6c79 2077 6865 6e20  plot, only when 
+000081b0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+000081c0: 2c20 7468 6520 7379 6e74 656e 7920 706c  , the synteny pl
+000081d0: 6f74 2077 6f75 6c64 2062 6520 7072 6f64  ot would be prod
+000081e0: 7563 6564 0a2d 6473 2c20 2d2d 646f 7473  uced.-ds, --dots
+000081f0: 697a 652c 2074 6865 2064 6f74 2073 697a  ize, the dot siz
+00008200: 6520 696e 2064 6f74 2070 6c6f 742c 2064  e in dot plot, d
+00008210: 6566 6175 6c74 2030 2e33 0a2d 6161 2c20  efault 0.3.-aa, 
+00008220: 2d2d 6170 616c 7068 612c 2074 6865 206f  --apalpha, the o
+00008230: 7061 6369 7479 206f 6620 616e 6368 6f72  pacity of anchor
+00008240: 2064 6f74 732c 2064 6566 6175 6c74 2031   dots, default 1
+00008250: 0a2d 6861 2c20 2d2d 686f 616c 7068 612c  .-ha, --hoalpha,
+00008260: 2074 6865 206f 7061 6369 7479 206f 6620   the opacity of 
+00008270: 686f 6d6f 6c6f 6720 646f 7473 2c20 6465  homolog dots, de
+00008280: 6661 756c 7420 300a 2d73 7274 2c20 2d2d  fault 0.-srt, --
+00008290: 7368 6f77 7265 616c 7469 636b 2c20 666c  showrealtick, fl
+000082a0: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+000082b0: 6572 2074 6f20 7368 6f77 2074 6865 2072  er to show the r
+000082c0: 6561 6c20 7469 636b 2069 6e20 6765 6e65  eal tick in gene
+000082d0: 7320 6f72 2062 6173 6573 2c20 6966 2074  s or bases, if t
+000082e0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+000082f0: 2074 6865 2072 6561 6c20 7469 636b 2077   the real tick w
+00008300: 6f75 6c64 2062 6520 7368 6f77 6564 0a2d  ould be showed.-
+00008310: 746c 732c 202d 2d74 6963 6b6c 6162 656c  tls, --ticklabel
+00008320: 7369 7a65 2c20 7468 6520 6c61 6265 6c20  size, the label 
+00008330: 7369 7a65 206f 6620 7469 636b 2c20 6465  size of tick, de
+00008340: 6661 756c 7420 350a 2d78 6c2c 202d 2d78  fault 5.-xl, --x
+00008350: 6c69 6d2c 2074 6865 2078 2061 7869 7320  lim, the x axis 
+00008360: 6c69 6d69 7420 6f66 204b 7320 6469 7374  limit of Ks dist
+00008370: 7269 6275 7469 6f6e 0a2d 796c 2c20 2d2d  ribution.-yl, --
+00008380: 796c 696d 2c20 7468 6520 7920 6178 6973  ylim, the y axis
+00008390: 206c 696d 6974 206f 6620 4b73 2064 6973   limit of Ks dis
+000083a0: 7472 6962 7574 696f 6e0a 2d61 646f 2c20  tribution.-ado, 
+000083b0: 2d2d 6164 6a75 7374 6f72 7468 6f2c 2066  --adjustortho, f
+000083c0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+000083d0: 6865 7220 746f 2061 646a 7573 7420 7468  her to adjust th
+000083e0: 6520 6869 7374 6f67 7261 6d20 6865 6967  e histogram heig
+000083f0: 6874 206f 6620 6f72 7468 6f6c 6f67 6f75  ht of orthologou
+00008400: 7320 4b73 2061 7320 746f 206d 6174 6368  s Ks as to match
+00008410: 2074 6865 2068 6569 6768 7420 6f66 2070   the height of p
+00008420: 6172 616c 6f67 6f75 7320 4b73 2c20 6966  aralogous Ks, if
+00008430: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00008440: 742c 2074 6865 2061 646a 7573 746d 656e  t, the adjustmen
+00008450: 7420 776f 756c 6420 6265 2063 6f6e 6475  t would be condu
+00008460: 6374 6564 0a2d 6164 662c 202d 2d61 646a  cted.-adf, --adj
+00008470: 7573 7466 6163 746f 722c 2074 6865 2061  ustfactor, the a
+00008480: 646a 7573 746d 656e 7420 6661 6374 6f72  djustment factor
+00008490: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
+000084a0: 4b73 2c20 6465 6661 756c 7420 302e 350a  Ks, default 0.5.
+000084b0: 2d6f 612c 202d 2d6f 6b61 6c70 6861 2c20  -oa, --okalpha, 
+000084c0: 7468 6520 6f70 6163 6974 7920 6f66 206f  the opacity of o
+000084d0: 7274 686f 6c6f 676f 7573 204b 7320 6469  rthologous Ks di
+000084e0: 7374 7269 6275 7469 6f6e 2069 6e20 6d69  stribution in mi
+000084f0: 7865 6420 706c 6f74 2c20 6465 6661 756c  xed plot, defaul
+00008500: 7420 302e 350a 2d63 732c 202d 2d63 6c61  t 0.5.-cs, --cla
+00008510: 7373 6963 2c20 666c 6167 206f 7074 696f  ssic, flag optio
+00008520: 6e2c 2077 6865 7468 6572 2074 6f20 6472  n, whether to dr
+00008530: 6177 206d 6978 6564 204b 7320 706c 6f74  aw mixed Ks plot
+00008540: 2069 6e20 6120 636c 6173 7369 6320 6d61   in a classic ma
+00008550: 6e6e 6572 2077 6865 7265 2074 6865 2066  nner where the f
+00008560: 756c 6c20 6f72 7468 6f6c 6f67 6f75 7320  ull orthologous 
+00008570: 4b73 2064 6973 7472 6962 7574 696f 6e20  Ks distribution 
+00008580: 6973 2064 7261 7765 642c 2069 6620 7468  is drawed, if th
+00008590: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+000085a0: 7468 6520 636c 6173 7369 6320 6d69 7865  the classic mixe
+000085b0: 6420 4b73 2070 6c6f 7420 776f 756c 6420  d Ks plot would 
+000085c0: 6265 2064 7261 776e 0a2d 7461 2c20 2d2d  be drawn.-ta, --
+000085d0: 746f 7061 7272 6f77 2c20 666c 6167 206f  toparrow, flag o
+000085e0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+000085f0: 6f20 6164 6a75 7374 2074 6865 2061 7272  o adjust the arr
+00008600: 6f77 2074 6f20 6265 2061 7420 7468 6520  ow to be at the 
+00008610: 746f 7020 6f66 2074 6865 2070 6c6f 742c  top of the plot,
+00008620: 2069 6e73 7465 6164 206f 6620 6265 696e   instead of bein
+00008630: 6720 636f 6f72 6469 6e61 7465 6420 6173  g coordinated as
+00008640: 2074 6865 204b 4445 206f 6620 7468 6520   the KDE of the 
+00008650: 6f72 7468 6f6c 6f67 6f75 7320 4b73 2064  orthologous Ks d
+00008660: 6973 7472 6962 7574 696f 6e2c 2069 6620  istribution, if 
+00008670: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00008680: 2c20 7468 6520 6172 726f 7720 776f 756c  , the arrow woul
+00008690: 6420 6265 2073 6574 2061 7420 7468 6520  d be set at the 
+000086a0: 746f 700a 2d6e 612c 202d 2d6e 6f64 6561  top.-na, --nodea
+000086b0: 7665 7261 6765 642c 2066 6c61 6720 6f70  veraged, flag op
+000086c0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+000086d0: 2075 7365 206e 6f64 652d 6176 6572 6167   use node-averag
+000086e0: 6564 206d 6574 686f 6420 666f 7220 6465  ed method for de
+000086f0: 2d72 6564 756e 6461 6e63 792c 2069 6620  -redundancy, if 
+00008700: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00008710: 2c20 7468 6520 6e6f 6465 2d61 7665 7261  , the node-avera
+00008720: 6765 6420 6d65 7468 6f64 2077 6f75 6c64  ged method would
+00008730: 2062 6520 696e 6974 6961 7465 640a 2d62   be initiated.-b
+00008740: 732c 202d 2d62 6f6f 7473 7472 6170 2c20  s, --bootstrap, 
+00008750: 7468 6520 6e75 6d62 6572 206f 6620 626f  the number of bo
+00008760: 6f74 7374 7261 7020 7265 706c 6963 6174  otstrap replicat
+00008770: 6573 206f 6620 6f72 7468 6f6c 6f67 204b  es of ortholog K
+00008780: 7320 6469 7374 7269 6275 7469 6f6e 2069  s distribution i
+00008790: 6e20 6d69 7865 6420 706c 6f74 0a2d 6772  n mixed plot.-gr
+000087a0: 2c20 2d2d 6769 7374 7262 2c20 666c 6167  , --gistrb, flag
+000087b0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
+000087c0: 2074 6f20 7573 6520 6769 7374 5f72 6169   to use gist_rai
+000087d0: 6e62 6f77 2061 7320 636f 6c6f 7220 6d61  nbow as color ma
+000087e0: 7020 6f66 2064 6f74 706c 6f74 0a2d 6e2c  p of dotplot.-n,
+000087f0: 202d 2d6e 7468 7265 6164 732c 2074 6865   --nthreads, the
+00008800: 206e 756d 6265 7220 6f66 2074 6872 6561   number of threa
+00008810: 6473 2074 6f20 7573 6520 696e 2062 6f6f  ds to use in boo
+00008820: 7473 7472 6170 2073 616d 706c 696e 672c  tstrap sampling,
+00008830: 2064 6566 6175 6c74 2031 0a60 6060 0a0a   default 1.```..
+00008840: 2323 2055 7361 6765 0a0a 4865 7265 2077  ## Usage..Here w
+00008850: 6520 7072 6f76 6964 6564 2074 6865 2062  e provided the b
+00008860: 6173 6963 2075 7361 6765 2066 6f72 2065  asic usage for e
+00008870: 6163 6820 7072 6f67 7261 6d20 616e 6420  ach program and 
+00008880: 7468 6520 7265 6c65 7661 6e74 2070 6172  the relevant par
+00008890: 616d 6574 6572 7320 616e 6420 7375 6767  ameters and sugg
+000088a0: 6573 7469 6f6e 7320 6f6e 2070 6172 616d  estions on param
+000088b0: 6574 6572 697a 6174 696f 6e2e 2041 2072  eterization. A r
+000088c0: 656d 696e 6465 7220 7468 6174 2074 6865  eminder that the
+000088d0: 2067 6976 656e 2063 6f72 6573 2061 6e64   given cores and
+000088e0: 2074 6872 6561 6473 2063 616e 2073 6967   threads can sig
+000088f0: 6e69 6669 6361 6e74 6c79 2069 6d70 6163  nificantly impac
+00008900: 7420 7468 6520 7275 6e20 7469 6d65 2061  t the run time a
+00008910: 6e64 2074 6875 7320 7765 2061 6464 6564  nd thus we added
+00008920: 2073 6f6d 6520 7265 706f 7274 2069 6e66   some report inf
+00008930: 6f72 6d61 7469 6f6e 2070 6572 7461 696e  ormation pertain
+00008940: 696e 6720 746f 2074 6865 2073 7973 7465  ing to the syste
+00008950: 6d20 6f66 2075 7365 7273 2074 6f20 6661  m of users to fa
+00008960: 6369 6c69 7461 7465 2074 6865 2065 6666  cilitate the eff
+00008970: 6963 6965 6e74 2073 6574 7469 6e67 206f  icient setting o
+00008980: 6620 7468 7265 6164 7320 616e 6420 6d65  f threads and me
+00008990: 6d6f 7279 2e20 5468 6520 6c6f 6769 6361  mory. The logica
+000089a0: 6c20 4350 5573 2072 6570 6f72 7465 6420  l CPUs reported 
+000089b0: 7265 7072 6573 656e 7473 2074 6865 206e  represents the n
+000089c0: 756d 6265 7220 6f66 2070 6879 7369 6361  umber of physica
+000089d0: 6c20 636f 7265 7320 6d75 6c74 6970 6c69  l cores multipli
+000089e0: 6564 2062 7920 7468 6520 6e75 6d62 6572  ed by the number
+000089f0: 206f 6620 7468 7265 6164 7320 7468 6174   of threads that
+00008a00: 2063 616e 2072 756e 206f 6e20 6561 6368   can run on each
+00008a10: 2063 6f72 652c 2061 6c73 6f20 6b6e 6f77   core, also know
+00008a20: 6e20 6173 2048 7970 6572 2054 6872 6561  n as Hyper Threa
+00008a30: 6469 6e67 2e20 5468 6520 6e75 6d62 6572  ding. The number
+00008a40: 206f 6620 6c6f 6769 6361 6c20 4350 5573   of logical CPUs
+00008a50: 206d 6179 206e 6f74 206e 6563 6573 7361   may not necessa
+00008a60: 7269 6c79 2062 6520 6571 7569 7661 6c65  rily be equivale
+00008a70: 6e74 2074 6f20 7468 6520 6163 7475 616c  nt to the actual
+00008a80: 206e 756d 6265 7220 6f66 2043 5055 7320   number of CPUs 
+00008a90: 7468 6520 6375 7272 656e 7420 7072 6f63  the current proc
+00008aa0: 6573 7320 6361 6e20 7573 652e 2054 6865  ess can use. The
+00008ab0: 2061 7661 696c 6162 6c65 206d 656d 6f72   available memor
+00008ac0: 7920 7265 6665 7273 2074 6f20 7468 6520  y refers to the 
+00008ad0: 6d65 6d6f 7279 2074 6861 7420 6361 6e20  memory that can 
+00008ae0: 6265 2067 6976 656e 2069 6e73 7461 6e74  be given instant
+00008af0: 6c79 2074 6f20 7072 6f63 6573 7365 7320  ly to processes 
+00008b00: 7769 7468 6f75 7420 7468 6520 7379 7374  without the syst
+00008b10: 656d 2067 6f69 6e67 2069 6e74 6f20 7377  em going into sw
+00008b20: 6170 2061 6e64 2072 6566 6c65 6374 7320  ap and reflects 
+00008b30: 7468 6520 6163 7475 616c 206d 656d 6f72  the actual memor
+00008b40: 7920 6176 6169 6c61 626c 652e 2054 6865  y available. The
+00008b50: 2066 7265 6520 6d65 6d6f 7279 2072 6566   free memory ref
+00008b60: 6572 7320 746f 2074 6865 206d 656d 6f72  ers to the memor
+00008b70: 7920 6e6f 7420 6265 696e 6720 7573 6564  y not being used
+00008b80: 2061 7420 616c 6c20 287a 6572 6f65 6429   at all (zeroed)
+00008b90: 2074 6861 7420 6973 2072 6561 6469 6c79   that is readily
+00008ba0: 2061 7661 696c 6162 6c65 2e20 5468 6520   available. The 
+00008bb0: 6465 7363 7269 7074 696f 6e20 6162 6f76  description abov
+00008bc0: 6520 7265 6665 7273 2074 6f20 7468 6520  e refers to the 
+00008bd0: 646f 6375 6d65 6e74 6174 696f 6e20 6f66  documentation of
+00008be0: 2060 7073 7574 696c 602e 0a0a 2323 2320   `psutil`...### 
+00008bf0: 7767 6420 646d 640a 0a2a 2a54 6865 2064  wgd dmd..**The d
+00008c00: 656c 696e 6561 7469 6f6e 206f 6620 7768  elineation of wh
+00008c10: 6f6c 6520 7061 7261 6e6f 6d65 2a2a 0a60  ole paranome**.`
+00008c20: 6060 0a77 6764 2064 6d64 2041 7175 696c  ``.wgd dmd Aquil
+00008c30: 6567 6961 5f63 6f65 7275 6c65 6120 2d49  egia_coerulea -I
+00008c40: 2032 202d 6520 3165 2d31 3020 2d62 7320   2 -e 1e-10 -bs 
+00008c50: 3130 3020 2d6e 7020 3520 282d 6e6e 2920  100 -np 5 (-nn) 
+00008c60: 282d 2d74 6f5f 7374 6f70 2920 282d 2d63  (--to_stop) (--c
+00008c70: 6473 2920 282d 6e20 3429 2028 2d6f 2077  ds) (-n 4) (-o w
+00008c80: 6764 5f64 6d64 2920 282d 7420 776f 726b  gd_dmd) (-t work
+00008c90: 696e 675f 746d 7029 0a60 6060 200a 0a4e  ing_tmp).``` ..N
+00008ca0: 6f74 6520 7468 6174 2077 6520 646f 6e27  ote that we don'
+00008cb0: 7420 7072 6f76 6964 6520 7468 6520 6461  t provide the da
+00008cc0: 7461 206f 6620 7468 6973 2063 6f64 696e  ta of this codin
+00008cd0: 6720 7365 7175 656e 6365 2028 6364 7329  g sequence (cds)
+00008ce0: 2066 696c 6520 6041 7175 696c 6567 6961   file `Aquilegia
+00008cf0: 5f63 6f65 7275 6c65 6160 2062 7574 2069  _coerulea` but i
+00008d00: 7420 6361 6e20 6265 2065 6173 696c 7920  t can be easily 
+00008d10: 646f 776e 6c6f 6164 6564 2061 7420 5b50  downloaded at [P
+00008d20: 6879 746f 7a6f 6d65 5d28 6874 7470 733a  hytozome](https:
+00008d30: 2f2f 7068 7974 6f7a 6f6d 652d 6e65 7874  //phytozome-next
+00008d40: 2e6a 6769 2e64 6f65 2e67 6f76 2f69 6e66  .jgi.doe.gov/inf
+00008d50: 6f2f 4163 6f65 7275 6c65 615f 7633 5f31  o/Acoerulea_v3_1
+00008d60: 2920 2873 616d 6520 666f 7220 6f74 6865  ) (same for othe
+00008d70: 7220 5573 6167 6520 646f 6329 2e20 4120  r Usage doc). A 
+00008d80: 7265 6d69 6e64 6572 2074 6861 7420 696e  reminder that in
+00008d90: 2074 6865 2069 7373 7565 7320 736f 6d65   the issues some
+00008da0: 2075 7365 7273 2064 6964 6e27 7420 646f   users didn't do
+00008db0: 776e 6c6f 6164 2074 6865 2060 4163 6f65  wnload the `Acoe
+00008dc0: 7275 6c65 615f 3332 325f 7633 2e31 2e63  rulea_322_v3.1.c
+00008dd0: 6473 5f70 7269 6d61 7279 5472 616e 7363  ds_primaryTransc
+00008de0: 7269 7074 4f6e 6c79 2e66 612e 677a 6020  riptOnly.fa.gz` 
+00008df0: 6275 7420 696e 7374 6561 6420 7468 6520  but instead the 
+00008e00: 6041 636f 6572 756c 6561 5f33 3232 5f76  `Acoerulea_322_v
+00008e10: 332e 312e 6364 732e 6661 2e67 7a60 2066  3.1.cds.fa.gz` f
+00008e20: 696c 652e 2046 6f72 2074 6865 2063 6f6e  ile. For the con
+00008e30: 7374 7275 6374 696f 6e20 6f66 2077 686f  struction of who
+00008e40: 6c65 2070 6172 616e 6f6d 6520 2a4b 2a3c  le paranome *K*<
+00008e50: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00008e60: 7269 6275 7469 6f6e 2c20 2a2a 6f6e 6c79  ribution, **only
+00008e70: 206f 6e65 2074 7261 6e73 6372 6970 7420   one transcript 
+00008e80: 2874 6865 2070 7269 6d61 7279 206f 6e65  (the primary one
+00008e90: 2920 7065 7220 6765 6e65 2073 686f 756c  ) per gene shoul
+00008ea0: 6420 6265 2069 6e63 6c75 6465 642a 2a20  d be included** 
+00008eb0: 7375 6368 2074 6861 7420 7468 6520 2a4b  such that the *K
+00008ec0: 2a3c 7375 623e 533c 2f73 7562 3e20 6973  *<sub>S</sub> is
+00008ed0: 2072 6561 6c6c 7920 696e 6469 6361 7469   really indicati
+00008ee0: 6e67 2074 6865 2061 6765 206f 6620 6765  ng the age of ge
+00008ef0: 6e65 2064 7570 6c69 6361 7469 6f6e 2065  ne duplication e
+00008f00: 7665 6e74 2c20 696e 7374 6561 6420 6f66  vent, instead of
+00008f10: 2061 6c74 6572 6e61 7469 7665 2073 706c   alternative spl
+00008f20: 6963 696e 672e 2054 7261 6e73 6372 6970  icing. Transcrip
+00008f30: 746f 6d65 2064 6174 6120 7368 6f75 6c64  tome data should
+00008f40: 2062 6520 6361 7265 6675 6c6c 7920 7472   be carefully tr
+00008f50: 6561 7465 6420 7769 7468 2064 652d 7265  eated with de-re
+00008f60: 6475 6e64 616e 6379 2073 6f20 6173 2074  dundancy so as t
+00008f70: 6f20 7265 6475 6365 2074 6865 2066 616c  o reduce the fal
+00008f80: 7365 2070 6f73 6974 6976 6520 6475 706c  se positive dupl
+00008f90: 6963 6174 696f 6e20 6275 6d70 2063 6175  ication bump cau
+00008fa0: 7365 6420 6279 2070 6572 7661 7369 7665  sed by pervasive
+00008fb0: 2061 6c74 6572 6e61 7469 7665 2073 706c   alternative spl
+00008fc0: 6963 696e 672e 2048 6572 6520 7468 6520  icing. Here the 
+00008fd0: 696e 666c 6174 696f 6e20 6661 6374 6f72  inflation factor
+00008fe0: 2070 6172 616d 6574 6572 2c20 6769 7665   parameter, give
+00008ff0: 6e20 6279 2060 2d49 6020 6f72 2060 2d2d  n by `-I` or `--
+00009000: 696e 666c 6174 696f 6e60 2c20 6166 6665  inflation`, affe
+00009010: 6374 7320 7468 6520 6772 616e 756c 6172  cts the granular
+00009020: 6974 7920 6f72 2072 6573 6f6c 7574 696f  ity or resolutio
+00009030: 6e20 6f66 2074 6865 2063 6c75 7374 6572  n of the cluster
+00009040: 696e 6720 6f75 7463 6f6d 6520 616e 6420  ing outcome and 
+00009050: 696d 706c 6963 6974 6c79 2063 6f6e 7472  implicitly contr
+00009060: 6f6c 6c73 2074 6865 206e 756d 6265 7220  olls the number 
+00009070: 6f66 2063 6c75 7374 6572 732c 2077 6974  of clusters, wit
+00009080: 6820 6c6f 7720 7661 6c75 6573 2073 7563  h low values suc
+00009090: 6820 6173 2031 2e33 206f 7220 312e 3420  h as 1.3 or 1.4 
+000090a0: 6c65 6164 696e 6720 746f 2066 6577 6572  leading to fewer
+000090b0: 2062 7574 206c 6172 6765 7220 636c 7573   but larger clus
+000090c0: 7465 7273 2061 6e64 2068 6967 6820 7661  ters and high va
+000090d0: 6c75 6573 2073 7563 6820 6173 2035 206f  lues such as 5 o
+000090e0: 7220 3620 6c65 6164 696e 6720 746f 206d  r 6 leading to m
+000090f0: 6f72 6520 6275 7420 736d 616c 6c65 7220  ore but smaller 
+00009100: 636c 7573 7465 7273 2e20 5765 2073 6574  clusters. We set
+00009110: 2074 6865 2064 6566 6175 6c74 2076 616c   the default val
+00009120: 7565 2061 7320 3220 6173 2073 7567 6765  ue as 2 as sugge
+00009130: 7374 6564 2062 7920 5b4d 434c 5d28 6874  sted by [MCL](ht
+00009140: 7470 733a 2f2f 6d69 6361 6e73 2e6f 7267  tps://micans.org
+00009150: 2f6d 636c 2f29 2e20 5468 6520 652d 7661  /mcl/). The e-va
+00009160: 6c75 6520 6375 742d 6f66 6620 666f 7220  lue cut-off for 
+00009170: 7365 7175 656e 6365 2073 696d 696c 6172  sequence similar
+00009180: 6974 792c 2067 6976 656e 2062 7920 602d  ity, given by `-
+00009190: 6560 206f 7220 602d 2d65 7661 6c60 2c20  e` or `--eval`, 
+000091a0: 7768 6963 6820 6465 6e6f 7465 7320 7468  which denotes th
+000091b0: 6520 6578 7065 6374 6564 2076 616c 7565  e expected value
+000091c0: 206f 6620 7468 6520 6869 7420 7175 616e   of the hit quan
+000091d0: 7469 6669 6573 2074 6865 206e 756d 6265  tifies the numbe
+000091e0: 7220 6f66 2061 6c69 676e 6d65 6e74 7320  r of alignments 
+000091f0: 6f66 2073 696d 696c 6172 206f 7220 6265  of similar or be
+00009200: 7474 6572 2071 7561 6c69 7479 2074 6861  tter quality tha
+00009210: 7420 796f 7520 6578 7065 6374 2074 6f20  t you expect to 
+00009220: 6669 6e64 2073 6561 7263 6869 6e67 2074  find searching t
+00009230: 6869 7320 7175 6572 7920 6167 6169 6e73  his query agains
+00009240: 7420 6120 6461 7461 6261 7365 206f 6620  t a database of 
+00009250: 7261 6e64 6f6d 2073 6571 7565 6e63 6573  random sequences
+00009260: 2074 6865 2073 616d 6520 7369 7a65 2061   the same size a
+00009270: 7320 7468 6520 6163 7475 616c 2074 6172  s the actual tar
+00009280: 6765 7420 6461 7461 6261 7365 2c20 6973  get database, is
+00009290: 2074 6865 206b 6579 2070 6172 616d 6574   the key paramet
+000092a0: 6572 206d 6561 7375 7269 6e67 2074 6865  er measuring the
+000092b0: 2073 6967 6e69 6669 6361 6e63 6520 6f66   significance of
+000092c0: 2061 2068 6974 2c20 7768 6963 6820 6973   a hit, which is
+000092d0: 2073 6574 2068 6572 6520 6173 2064 6566   set here as def
+000092e0: 6175 6c74 2031 652d 3130 2e20 4e6f 7465  ault 1e-10. Note
+000092f0: 2074 6861 7420 5b44 4941 4d4f 4e44 5d28   that [DIAMOND](
+00009300: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
+00009310: 6f6d 2f62 6275 6368 6669 6e6b 2f64 6961  om/bbuchfink/dia
+00009320: 6d6f 6e64 2f77 696b 692f 312e 2d54 7574  mond/wiki/1.-Tut
+00009330: 6f72 6961 6c29 2069 7473 656c 6620 6279  orial) itself by
+00009340: 2064 6566 6175 6c74 206f 6e6c 7920 7265   default only re
+00009350: 706f 7274 7320 616c 6c20 616c 6967 6e6d  ports all alignm
+00009360: 656e 7473 2077 6974 6820 652d 7661 6c75  ents with e-valu
+00009370: 6520 3c20 302e 3030 312e 2054 6865 2070  e < 0.001. The p
+00009380: 6572 6365 6e74 6167 6520 6f66 2075 7070  ercentage of upp
+00009390: 6572 2068 6974 7320 7573 6564 2066 6f72  er hits used for
+000093a0: 2067 656e 6520 6c65 6e67 7468 206e 6f72   gene length nor
+000093b0: 6d61 6c69 7a61 7469 6f6e 2c20 6769 7665  malization, give
+000093c0: 6e20 6279 2060 2d6e 7060 206f 7220 602d  n by `-np` or `-
+000093d0: 2d6e 6f72 6d61 6c69 7a65 6470 6572 6365  -normalizedperce
+000093e0: 6e74 602c 2077 6869 6368 2064 6574 6572  nt`, which deter
+000093f0: 6d69 6e65 7320 7468 6520 7570 7065 7220  mines the upper 
+00009400: 7065 7263 656e 7469 6c65 206f 6620 6869  percentile of hi
+00009410: 7473 2070 6572 2062 696e 2028 6361 7465  ts per bin (cate
+00009420: 676f 7269 7a65 6420 6279 2067 656e 6520  gorized by gene 
+00009430: 6c65 6e67 7468 2920 7573 6564 2069 6e20  length) used in 
+00009440: 7468 6520 6669 7420 6f66 206c 696e 6561  the fit of linea
+00009450: 7220 7265 6772 6573 7369 6f6e 2c20 636f  r regression, co
+00009460: 6e73 6964 6572 696e 6720 7468 6174 206e  nsidering that n
+00009470: 6f74 2061 6c6c 2068 6974 7320 7065 7220  ot all hits per 
+00009480: 6269 6e20 7368 6f77 2061 7070 6172 656e  bin show apparen
+00009490: 7420 6c69 6e65 6172 2072 656c 6174 696f  t linear relatio
+000094a0: 6e73 6869 702c 2069 7320 7365 7420 6173  nship, is set as
+000094b0: 2064 6566 6175 6c74 2035 2c20 696e 6469   default 5, indi
+000094c0: 6361 7469 6e67 2074 6865 2075 7361 6765  cating the usage
+000094d0: 206f 6620 746f 7020 3525 2068 6974 7320   of top 5% hits 
+000094e0: 7065 7220 6269 6e2e 2054 6865 206e 756d  per bin. The num
+000094f0: 6265 7220 6f66 2062 696e 7320 6469 7669  ber of bins divi
+00009500: 6465 6420 696e 2067 656e 6520 6c65 6e67  ded in gene leng
+00009510: 7468 206e 6f72 6d61 6c69 7a61 7469 6f6e  th normalization
+00009520: 2c20 6769 7665 6e20 6279 2060 2d62 7360  , given by `-bs`
+00009530: 206f 7220 602d 2d62 696e 7360 2c20 6465   or `--bins`, de
+00009540: 7465 726d 696e 6573 2074 6865 206e 756d  termines the num
+00009550: 6265 7220 6f66 2062 696e 7320 746f 2063  ber of bins to c
+00009560: 6174 6567 6f72 697a 6520 7468 6520 6765  ategorize the ge
+00009570: 6e65 206c 656e 6774 682c 2069 7320 7365  ne length, is se
+00009580: 7420 6173 2064 6566 6175 6c74 2031 3030  t as default 100
+00009590: 2e20 5468 6520 7061 7261 6d65 7465 7220  . The parameter 
+000095a0: 602d 6e6e 6020 6f72 2060 2d2d 6e6f 6e6f  `-nn` or `--nono
+000095b0: 726d 616c 697a 6174 696f 6e60 2063 616e  rmalization` can
+000095c0: 2062 6520 7365 7420 746f 2063 616c 6c20   be set to call 
+000095d0: 6f66 6620 7468 6520 6e6f 726d 616c 697a  off the normaliz
+000095e0: 6174 696f 6e20 7072 6f63 6573 732c 2061  ation process, a
+000095f0: 6c74 686f 7567 6820 6974 2773 2073 7567  lthough it's sug
+00009600: 6765 7374 6564 2074 6f20 636f 6e64 7563  gested to conduc
+00009610: 7420 7468 6520 6e6f 726d 616c 697a 6174  t the normalizat
+00009620: 696f 6e20 746f 2061 6371 7569 7265 206d  ion to acquire m
+00009630: 6f72 6520 6163 6375 7261 7465 2067 656e  ore accurate gen
+00009640: 6520 6661 6d69 6c79 2063 6c75 7374 6572  e family cluster
+00009650: 696e 6720 7265 7375 6c74 2e20 5468 6520  ing result. The 
+00009660: 7061 7261 6d65 7465 7273 2060 2d2d 746f  parameters `--to
+00009670: 5f73 746f 7060 2061 6e64 2060 2d2d 6364  _stop` and `--cd
+00009680: 7360 2063 6f6e 7472 6f6c 2074 6865 2062  s` control the b
+00009690: 6568 6176 696f 7572 206f 6620 7472 616e  ehaviour of tran
+000096a0: 736c 6174 696e 6720 636f 6469 6e67 2073  slating coding s
+000096b0: 6571 7565 6e63 6520 696e 746f 2061 6d69  equence into ami
+000096c0: 6e6f 2061 6369 6420 7365 7175 656e 6365  no acid sequence
+000096d0: 2e20 4966 2074 6865 2060 2d2d 746f 5f73  . If the `--to_s
+000096e0: 746f 7060 2077 6173 2073 6574 2c20 7468  top` was set, th
+000096f0: 6520 7472 616e 736c 6174 696f 6e20 776f  e translation wo
+00009700: 756c 6420 6265 2074 6572 6d69 6e61 7465  uld be terminate
+00009710: 6420 6174 2074 6865 2066 6972 7374 2069  d at the first i
+00009720: 6e2d 6672 616d 6520 7374 6f70 2063 6f64  n-frame stop cod
+00009730: 6f6e 2c20 6f74 6865 7277 6973 6520 7468  on, otherwise th
+00009740: 6520 7472 616e 736c 6174 696f 6e20 776f  e translation wo
+00009750: 756c 6420 7369 6d70 6c79 2073 6b69 7020  uld simply skip 
+00009760: 616e 7920 7374 6f70 2063 6f64 6f6e 732e  any stop codons.
+00009770: 2049 6620 7468 6520 602d 2d63 6473 6020   If the `--cds` 
+00009780: 7761 7320 7365 742c 2073 6571 7565 6e63  was set, sequenc
+00009790: 6573 2074 6861 7420 646f 6573 6e27 7420  es that doesn't 
+000097a0: 7374 6172 7420 7769 7468 2061 2076 616c  start with a val
+000097b0: 6964 2073 7461 7274 2063 6f64 6f6e 2c20  id start codon, 
+000097c0: 6f72 2063 6f6e 7461 696e 7320 6d6f 7265  or contains more
+000097d0: 2074 6861 6e20 6f6e 6520 696e 2d66 7261   than one in-fra
+000097e0: 6d65 2073 746f 7020 636f 646f 6e2c 206f  me stop codon, o
+000097f0: 7220 6973 206e 6f74 2064 6976 6964 6162  r is not dividab
+00009800: 6c65 2062 7920 332c 2077 6f75 6c64 2062  le by 3, would b
+00009810: 6520 7369 6d70 6c79 2064 726f 7070 6564  e simply dropped
+00009820: 2c20 7375 6368 2074 6861 7420 6f6e 6c79  , such that only
+00009830: 2073 7472 6963 7420 636f 6d70 6c65 7465   strict complete
+00009840: 2063 6f64 696e 6720 7365 7175 656e 6365   coding sequence
+00009850: 7320 776f 756c 6420 6265 2069 6e63 6c75  s would be inclu
+00009860: 6465 6420 696e 2074 6865 2073 7562 7365  ded in the subse
+00009870: 7175 656e 7420 616e 616c 7973 6973 2e20  quent analysis. 
+00009880: 5468 6520 6578 6163 7420 6265 6861 7669  The exact behavi
+00009890: 6f75 7220 6f66 2060 2d2d 746f 5f73 746f  our of `--to_sto
+000098a0: 7060 2061 6e64 2060 2d2d 6364 7360 2069  p` and `--cds` i
+000098b0: 7320 6465 6669 6e65 6420 616e 6420 6465  s defined and de
+000098c0: 7363 7269 6265 6420 696e 2074 6865 205b  scribed in the [
+000098d0: 6269 6f70 7974 686f 6e5d 2868 7474 7073  biopython](https
+000098e0: 3a2f 2f62 696f 7079 7468 6f6e 2e6f 7267  ://biopython.org
+000098f0: 2f64 6f63 732f 312e 3735 2f61 7069 2f42  /docs/1.75/api/B
+00009900: 696f 2e53 6571 2e68 746d 6c23 4269 6f2e  io.Seq.html#Bio.
+00009910: 5365 712e 5365 712e 7472 616e 736c 6174  Seq.Seq.translat
+00009920: 6529 206c 6962 7261 7279 2e20 5468 6520  e) library. The 
+00009930: 6e75 6d62 6572 206f 6620 7061 7261 6c6c  number of parall
+00009940: 656c 2074 6872 6561 6473 2062 7920 7468  el threads by th
+00009950: 6520 6f70 7469 6f6e 2060 2d6e 6020 6f72  e option `-n` or
+00009960: 2060 2d2d 6e74 6872 6561 6473 6020 6361   `--nthreads` ca
+00009970: 6e20 6265 2073 6574 2074 6f20 6163 6365  n be set to acce
+00009980: 6c65 7261 7465 2074 6865 2063 616c 6375  lerate the calcu
+00009990: 6c61 7469 6f6e 2077 6974 6869 6e20 6064  lation within `d
+000099a0: 6961 6d6f 6e64 602e 2054 6865 2064 6972  iamond`. The dir
+000099b0: 6563 746f 7279 206f 6620 6f75 7470 7574  ectory of output
+000099c0: 206f 7220 696e 7465 726d 6564 6961 7465   or intermediate
+000099d0: 2066 696c 6573 2069 7320 6465 7465 726d   files is determ
+000099e0: 696e 6564 2062 7920 7468 6520 7061 7261  ined by the para
+000099f0: 6d65 7465 7220 602d 6f60 206f 7220 602d  meter `-o` or `-
+00009a00: 2d6f 7574 6469 7260 2c20 616e 6420 602d  -outdir`, and `-
+00009a10: 7460 206f 7220 602d 2d74 6d70 6469 7260  t` or `--tmpdir`
+00009a20: 2c20 7768 6963 6820 7769 6c6c 2062 6520  , which will be 
+00009a30: 6372 6561 7465 6420 6279 2074 6865 2070  created by the p
+00009a40: 726f 6772 616d 2069 7473 656c 6620 616e  rogram itself an
+00009a50: 6420 6265 206f 7665 7277 7269 7474 656e  d be overwritten
+00009a60: 2069 6620 7468 6520 666f 6c64 6572 2068   if the folder h
+00009a70: 6173 2061 6c72 6561 6479 2062 6565 6e20  as already been 
+00009a80: 6372 6561 7465 642e 204e 6f74 6520 7468  created. Note th
+00009a90: 6174 2074 6865 2073 6f66 7477 6172 6520  at the software 
+00009aa0: 6064 6961 6d6f 6e64 6020 7368 6f75 6c64  `diamond` should
+00009ab0: 2062 6520 7072 652d 696e 7374 616c 6c65   be pre-installe
+00009ac0: 6420 616e 6420 7365 7420 696e 2074 6865  d and set in the
+00009ad0: 2065 6e76 6972 6f6e 6d65 6e74 2070 6174   environment pat
+00009ae0: 6820 696e 2061 6c6c 2074 6865 2061 6e61  h in all the ana
+00009af0: 6c79 7369 7320 7065 7266 6f72 6d65 6420  lysis performed 
+00009b00: 6279 2060 7767 6420 646d 6460 2065 7863  by `wgd dmd` exc
+00009b10: 6570 7420 666f 7220 7468 6520 636f 6c6c  ept for the coll
+00009b20: 696e 6561 7220 636f 616c 6573 6365 6e63  inear coalescenc
+00009b30: 6520 616e 616c 7973 6973 2e0a 0a2a 2a57  e analysis...**W
+00009b40: 6520 7375 6767 6573 7420 7468 6174 2074  e suggest that t
+00009b50: 6865 2064 6566 6175 6c74 2073 6574 7469  he default setti
+00009b60: 6e67 2069 6e20 7768 6963 6820 7468 6520  ng in which the 
+00009b70: 696e 666c 6174 696f 6e20 6661 6374 6f72  inflation factor
+00009b80: 2069 7320 7365 7420 6173 2032 2c20 652d   is set as 2, e-
+00009b90: 7661 6c75 6520 6375 742d 6f66 6620 6173  value cut-off as
+00009ba0: 2031 652d 3130 2061 6e64 206f 7468 6572   1e-10 and other
+00009bb0: 2070 6172 616d 6574 6572 7320 696e 2064   parameters in d
+00009bc0: 6566 6175 6c74 2069 7320 6120 676f 6f64  efault is a good
+00009bd0: 2073 7461 7274 696e 6720 706f 696e 742c   starting point,
+00009be0: 2075 6e6c 6573 7320 796f 7520 7370 6563   unless you spec
+00009bf0: 6966 6963 616c 6c79 2077 616e 7420 746f  ifically want to
+00009c00: 2065 7870 6c6f 7265 2074 6865 2065 6666   explore the eff
+00009c10: 6563 7473 206f 6620 6469 6666 6572 656e  ects of differen
+00009c20: 7420 7061 7261 6d65 7465 7273 2e2a 2a20  t parameters.** 
+00009c30: 5375 6368 2074 6861 7420 7468 6520 636f  Such that the co
+00009c40: 6d6d 616e 6420 666f 7220 7468 6520 6465  mmand for the de
+00009c50: 6c69 6e65 6174 696f 6e20 6f66 2077 686f  lineation of who
+00009c60: 6c65 2070 6172 616e 6f6d 6520 6973 2073  le paranome is s
+00009c70: 696d 706c 7920 6173 2062 656c 6f77 2e0a  imply as below..
+00009c80: 0a60 6060 0a77 6764 2064 6d64 2041 7175  .```.wgd dmd Aqu
+00009c90: 696c 6567 6961 5f63 6f65 7275 6c65 610a  ilegia_coerulea.
+00009ca0: 6060 600a 0a2a 2a54 6865 2064 656c 696e  ```..**The delin
+00009cb0: 6561 7469 6f6e 206f 6620 5242 4873 2a2a  eation of RBHs**
+00009cc0: 0a60 6060 0a77 6764 2064 6d64 2073 6571  .```.wgd dmd seq
+00009cd0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+00009ce0: 202d 6520 3165 2d31 3020 2d62 7320 3130   -e 1e-10 -bs 10
+00009cf0: 3020 2d6e 7020 3520 282d 6e6e 2920 282d  0 -np 5 (-nn) (-
+00009d00: 6e20 3429 2028 2d63 2030 2e39 2920 282d  n 4) (-c 0.9) (-
+00009d10: 2d6f 6766 6f72 6d61 7429 2028 2d2d 746f  -ogformat) (--to
+00009d20: 5f73 746f 7029 2028 2d2d 6364 7329 2028  _stop) (--cds) (
+00009d30: 2d6f 2077 6764 5f64 6d64 2920 282d 7420  -o wgd_dmd) (-t 
+00009d40: 776f 726b 696e 675f 746d 7029 0a60 6060  working_tmp).```
+00009d50: 0a0a 546f 2064 656c 696e 6561 7465 2052  ..To delineate R
+00009d60: 4248 7320 6265 7477 6565 6e20 7477 6f20  BHs between two 
+00009d70: 6364 7320 7365 7175 656e 6365 2066 696c  cds sequence fil
+00009d80: 6573 2c20 7468 6520 7265 6c65 7661 6e74  es, the relevant
+00009d90: 2070 6172 616d 6574 6572 2069 7320 6d6f   parameter is mo
+00009da0: 7374 6c79 2074 6865 2073 616d 6520 6173  stly the same as
+00009db0: 2077 686f 6c65 2070 6172 616e 6f6d 6520   whole paranome 
+00009dc0: 696e 6665 7265 6e63 652c 2065 7863 6570  inference, excep
+00009dd0: 7420 666f 7220 7468 6520 7061 7261 6d65  t for the parame
+00009de0: 7465 7220 602d 6360 206f 7220 602d 2d63  ter `-c` or `--c
+00009df0: 7363 6f72 6560 2c20 7768 6963 6820 7261  score`, which ra
+00009e00: 6e67 6573 2062 6574 7765 656e 2030 2061  nges between 0 a
+00009e10: 6e64 2031 2061 6e64 2069 7320 7573 6564  nd 1 and is used
+00009e20: 2074 6f20 7265 6c61 7820 7468 6520 7369   to relax the si
+00009e30: 6d69 6c61 7269 7479 2063 7574 6f66 6620  milarity cutoff 
+00009e40: 6672 6f6d 2074 6865 2065 7863 6c75 7369  from the exclusi
+00009e50: 7665 2072 6563 6970 726f 6361 6c20 6265  ve reciprocal be
+00009e60: 7374 2068 6974 7320 746f 2061 2063 6572  st hits to a cer
+00009e70: 7461 696e 2072 6174 696f 2061 7320 746f  tain ratio as to
+00009e80: 2074 6865 2062 6573 7420 6869 7473 2e20   the best hits. 
+00009e90: 466f 7220 696e 7374 616e 6365 2c20 6966  For instance, if
+00009ea0: 2074 6865 2067 656e 6520 6231 2066 726f   the gene b1 fro
+00009eb0: 6d20 6765 6e6f 6d65 2042 2068 6173 2074  m genome B has t
+00009ec0: 6865 2062 6573 7420 6869 7420 6765 6e65  he best hit gene
+00009ed0: 2061 3120 6672 6f6d 2067 656e 6f6d 6520   a1 from genome 
+00009ee0: 4120 7769 7468 2074 6865 2062 6974 2073  A with the bit s
+00009ef0: 636f 7265 2061 7320 3130 302c 2077 6869  core as 100, whi
+00009f00: 6368 2069 7320 6120 7363 6f72 696e 6720  ch is a scoring 
+00009f10: 6d61 7472 6978 2069 6e64 6570 656e 6465  matrix independe
+00009f20: 6e74 206d 6561 7375 7265 206f 6620 7468  nt measure of th
+00009f30: 6520 286c 6f63 616c 2920 7369 6d69 6c61  e (local) simila
+00009f40: 7269 7479 206f 6620 7468 6520 7477 6f20  rity of the two 
+00009f50: 616c 6967 6e65 6420 7365 7175 656e 6365  aligned sequence
+00009f60: 732c 2077 6974 6820 6c61 7267 6572 2076  s, with larger v
+00009f70: 616c 7565 7320 696e 6469 6361 7469 6e67  alues indicating
+00009f80: 2068 6967 6865 7220 7369 6d69 6c61 7269   higher similari
+00009f90: 7469 6573 2c20 6769 7665 6e20 7468 6520  ties, given the 
+00009fa0: 602d 6320 302e 3960 2c20 6765 6e65 7320  `-c 0.9`, genes 
+00009fb0: 6672 6f6d 2067 656e 6f6d 6520 4120 7768  from genome A wh
+00009fc0: 6963 6820 6861 7320 7468 6520 6269 7420  ich has the bit 
+00009fd0: 7363 6f72 6520 7769 7468 2067 656e 6520  score with gene 
+00009fe0: 6231 2068 6967 6865 7220 7468 616e 2030  b1 higher than 0
+00009ff0: 2e39 7831 3030 2077 696c 6c20 616c 736f  .9x100 will also
+0000a000: 2062 6520 7772 6974 7465 6e20 696e 2074   be written in t
+0000a010: 6865 2072 6573 756c 7420 6669 6c65 2c20  he result file, 
+0000a020: 7768 6963 6820 696e 2061 2073 656e 7365  which in a sense
+0000a030: 2061 7265 206e 6f74 2052 4248 7320 616e   are not RBHs an
+0000a040: 796d 6f72 6520 6f66 2063 6f75 7273 652c  ymore of course,
+0000a050: 2062 7574 2074 6865 2068 6967 686c 7920   but the highly 
+0000a060: 7369 6d69 6c61 7220 686f 6d6f 6c6f 6775  similar homologu
+0000a070: 6520 7061 6972 732e 2049 6620 6d6f 7265  e pairs. If more
+0000a080: 2074 6861 6e20 3220 7365 7175 656e 6365   than 2 sequence
+0000a090: 2066 696c 6573 2077 6572 6520 7072 6f76   files were prov
+0000a0a0: 6964 6564 2c20 6576 6572 7920 7061 6972  ided, every pair
+0000a0b0: 2d77 6973 6520 5242 4873 2077 6f75 6c64  -wise RBHs would
+0000a0c0: 2062 6520 6361 6c63 756c 6174 6564 2065   be calculated e
+0000a0d0: 7863 6570 7420 666f 7220 7175 6572 7969  xcept for queryi
+0000a0e0: 6e67 2074 6865 2073 616d 6520 7365 7175  ng the same sequ
+0000a0f0: 656e 6365 2069 7473 656c 662e 2054 6865  ence itself. The
+0000a100: 206e 756d 6265 7220 6f66 2070 6172 616c   number of paral
+0000a110: 6c65 6c20 7468 7265 6164 7320 746f 2062  lel threads to b
+0000a120: 6f6f 7374 6572 2074 6865 2072 756e 6e69  ooster the runni
+0000a130: 6e67 2073 7065 6564 2063 616e 2062 6520  ng speed can be 
+0000a140: 7365 7420 6279 2074 6865 206f 7074 696f  set by the optio
+0000a150: 6e20 602d 6e60 206f 7220 602d 2d6e 7468  n `-n` or `--nth
+0000a160: 7265 6164 7360 2077 6869 6368 2069 7320  reads` which is 
+0000a170: 7375 6767 6573 7465 6420 746f 2062 6520  suggested to be 
+0000a180: 7365 7420 6173 2028 4e2d 3129 4e2f 3220  set as (N-1)N/2 
+0000a190: 7768 6572 6520 4e20 6973 2074 6865 206e  where N is the n
+0000a1a0: 756d 6265 7220 6f66 2063 6473 2066 696c  umber of cds fil
+0000a1b0: 6573 2074 6f20 6163 6869 6576 6520 7468  es to achieve th
+0000a1c0: 6520 6869 6768 6573 7420 6566 6669 6369  e highest effici
+0000a1d0: 656e 6379 2e20 5468 6520 6f70 7469 6f6e  ency. The option
+0000a1e0: 2060 2d2d 6f67 666f 726d 6174 6020 6361   `--ogformat` ca
+0000a1f0: 6e20 6265 2073 6574 2074 6f20 6164 6420  n be set to add 
+0000a200: 696e 6465 7820 2866 6f72 2069 6e73 7461  index (for insta
+0000a210: 6e63 6520 4746 3030 3030 3030 3031 2920  nce GF00000001) 
+0000a220: 746f 2074 6865 206f 7574 7075 7420 5242  to the output RB
+0000a230: 4820 6765 6e65 2066 616d 696c 6965 7320  H gene families 
+0000a240: 7768 6963 6820 6361 6e20 6265 2066 7572  which can be fur
+0000a250: 7468 6572 2075 7365 6420 696e 2074 6865  ther used in the
+0000a260: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000a270: 2063 616c 6375 6c61 7469 6f6e 2062 7920   calculation by 
+0000a280: 6077 6764 206b 7364 602e 0a0a 2a2a 5468  `wgd ksd`...**Th
+0000a290: 6520 7375 6767 6573 7465 6420 636f 6d6d  e suggested comm
+0000a2a0: 616e 6420 746f 2073 7461 7274 2077 6974  and to start wit
+0000a2b0: 6820 6973 2061 6c73 6f20 756e 6465 7220  h is also under 
+0000a2c0: 7468 6520 6465 6661 756c 7420 7365 7474  the default sett
+0000a2d0: 696e 6720 7769 7468 2074 6865 2063 6f6d  ing with the com
+0000a2e0: 6d61 6e64 2073 686f 776e 2062 656c 6f77  mand shown below
+0000a2f0: 2a2a 0a0a 6060 600a 7767 6420 646d 6420  **..```.wgd dmd 
+0000a300: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
+0000a310: 6365 320a 6060 600a 0a2a 2a54 6865 2064  ce2.```..**The d
+0000a320: 656c 696e 6561 7469 6f6e 206f 6620 6c6f  elineation of lo
+0000a330: 6361 6c20 4d52 4248 732a 2a0a 6060 600a  cal MRBHs**.```.
+0000a340: 7767 6420 646d 6420 7365 7175 656e 6365  wgd dmd sequence
+0000a350: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
+0000a360: 656e 6365 3320 2d66 2073 6571 7565 6e63  ence3 -f sequenc
+0000a370: 6531 202d 6520 3165 2d31 3020 2d62 7320  e1 -e 1e-10 -bs 
+0000a380: 3130 3020 2d6e 7020 3520 282d 6e6e 2920  100 -np 5 (-nn) 
+0000a390: 282d 6e20 3429 2028 2d6b 6629 2028 2d6b  (-n 4) (-kf) (-k
+0000a3a0: 6429 2028 2d63 2030 2e39 2920 282d 2d74  d) (-c 0.9) (--t
+0000a3b0: 6f5f 7374 6f70 2920 282d 2d63 6473 2920  o_stop) (--cds) 
+0000a3c0: 282d 6f20 7767 645f 646d 6429 2028 2d74  (-o wgd_dmd) (-t
+0000a3d0: 2077 6f72 6b69 6e67 5f74 6d70 290a 6060   working_tmp).``
+0000a3e0: 600a 0a2a 2a54 6865 2064 6973 7469 6e63  `..**The distinc
+0000a3f0: 7469 6f6e 2062 6574 7765 656e 206c 6f63  tion between loc
+0000a400: 616c 2061 6e64 2067 6c6f 6261 6c20 2868  al and global (h
+0000a410: 6572 6575 6e64 6572 2920 4d52 4248 7320  ereunder) MRBHs 
+0000a420: 6973 2074 6861 7420 6c6f 6361 6c20 4d52  is that local MR
+0000a430: 4248 7320 6172 6520 7468 6520 7265 7375  BHs are the resu
+0000a440: 6c74 7320 6f66 206d 6572 6765 6420 5242  lts of merged RB
+0000a450: 4873 206f 6e20 6120 6a6f 696e 7420 666f  Hs on a joint fo
+0000a460: 6361 6c20 7370 6563 6965 732c 2066 6f72  cal species, for
+0000a470: 2069 6e73 7461 6e63 6520 696e 2061 2074   instance in a t
+0000a480: 6872 6565 2073 7065 6369 6573 2073 7973  hree species sys
+0000a490: 7465 6d20 2841 2c28 422c 4329 292c 2074  tem (A,(B,C)), t
+0000a4a0: 6865 206c 6f63 616c 204d 5242 4873 206f  he local MRBHs o
+0000a4b0: 6620 4120 6f6e 6c79 2072 6571 7569 7265  f A only require
+0000a4c0: 2074 6865 2063 616c 6375 6c61 7469 6f6e   the calculation
+0000a4d0: 206f 6620 5242 4873 2062 6574 7765 656e   of RBHs between
+0000a4e0: 2041 2061 6e64 2043 2028 6465 6e6f 7465   A and C (denote
+0000a4f0: 6420 6173 2041 4329 2061 6e64 2041 4220  d as AC) and AB 
+0000a500: 616e 6420 7468 656e 2074 6865 206d 6572  and then the mer
+0000a510: 6769 6e67 206f 6620 4142 2061 6e64 2041  ging of AB and A
+0000a520: 4320 6174 2074 6865 2061 7869 7320 6f66  C at the axis of
+0000a530: 2041 2c20 7768 696c 6520 7468 6520 676c   A, while the gl
+0000a540: 6f61 626c 204d 5242 4873 2061 7265 2069  oabl MRBHs are i
+0000a550: 6e64 6570 656e 6465 6e74 206f 6620 666f  ndependent of fo
+0000a560: 6361 6c20 7370 6563 6965 7320 696e 2074  cal species in t
+0000a570: 6861 7420 6974 206a 7573 7420 6361 6c6c  hat it just call
+0000a580: 7320 7468 6520 6361 6c63 756c 6174 696f  s the calculatio
+0000a590: 6e73 206f 6620 616c 6c20 706f 7373 6962  ns of all possib
+0000a5a0: 6c65 2073 7065 6369 6573 2070 6169 7220  le species pair 
+0000a5b0: 286e 6f74 2073 656c 6620 746f 2073 656c  (not self to sel
+0000a5c0: 6620 7370 6563 6965 7320 7061 6972 292c  f species pair),
+0000a5d0: 2073 7563 6820 7468 6174 2041 422c 2041   such that AB, A
+0000a5e0: 432c 2061 6e64 2042 4320 6172 6520 616c  C, and BC are al
+0000a5f0: 6c20 746f 2062 6520 6361 6c63 756c 6174  l to be calculat
+0000a600: 6564 2061 6e64 206d 6572 6765 642e 2a2a  ed and merged.**
+0000a610: 0a0a 5477 6f20 7479 7065 7320 6f66 204d  ..Two types of M
+0000a620: 5242 4873 2061 7320 696e 7465 7072 6574  RBHs as intepret
+0000a630: 6174 6564 2061 626f 7665 2063 616e 2062  ated above can b
+0000a640: 6520 6465 6c69 6e65 6174 6564 2062 7920  e delineated by 
+0000a650: 6077 6764 2064 6d64 602c 2074 6865 206c  `wgd dmd`, the l
+0000a660: 6f63 616c 204d 5242 4873 2061 6e64 2074  ocal MRBHs and t
+0000a670: 6865 2067 6c6f 6261 6c20 4d52 4248 732e  he global MRBHs.
+0000a680: 2054 6865 206c 6f63 616c 204d 5242 4873   The local MRBHs
+0000a690: 2061 7265 2063 6f6e 7374 7275 6374 6564   are constructed
+0000a6a0: 2062 7920 6d65 7267 696e 6720 616c 6c20   by merging all 
+0000a6b0: 7468 6520 7265 6c65 7661 6e74 2052 4248  the relevant RBH
+0000a6c0: 7320 6f6e 6c79 2077 6974 6820 7468 6520  s only with the 
+0000a6d0: 666f 6361 6c20 7370 6563 6965 732c 2077  focal species, w
+0000a6e0: 6869 6368 2069 7320 7365 7420 6279 2060  hich is set by `
+0000a6f0: 2d66 6020 6f72 2060 2d2d 666f 6375 7360  -f` or `--focus`
+0000a700: 2e20 5468 6520 7061 7261 6d65 7465 7220  . The parameter 
+0000a710: 602d 6b66 6020 6f72 2060 2d2d 6b65 6570  `-kf` or `--keep
+0000a720: 6661 7374 6160 2063 616e 2062 6520 7365  fasta` can be se
+0000a730: 7420 746f 2072 6574 6169 6e20 7468 6520  t to retain the 
+0000a740: 7365 7175 656e 6365 2069 6e66 6f72 6d61  sequence informa
+0000a750: 7469 6f6e 206f 6620 6561 6368 204d 5242  tion of each MRB
+0000a760: 482e 2054 6865 2070 6172 616d 6574 6572  H. The parameter
+0000a770: 2060 2d6b 6460 206f 7220 602d 2d6b 6565   `-kd` or `--kee
+0000a780: 7064 7570 6c69 6361 7465 7360 2064 6574  pduplicates` det
+0000a790: 6572 6d69 6e65 7320 7768 6574 6865 7220  ermines whether 
+0000a7a0: 7468 6520 7361 6d65 2067 656e 6573 2063  the same genes c
+0000a7b0: 616e 2061 7070 6561 7220 696e 2064 6966  an appear in dif
+0000a7c0: 6665 7265 6e74 206c 6f63 616c 204d 5242  ferent local MRB
+0000a7d0: 4873 2e20 4e6f 726d 616c 6c79 2074 6865  Hs. Normally the
+0000a7e0: 7265 2077 696c 6c20 6265 206e 6f20 6475  re will be no du
+0000a7f0: 706c 6963 6174 6573 2069 6e20 7468 6520  plicates in the 
+0000a800: 6c6f 6361 6c20 4d52 4248 7320 6275 7420  local MRBHs but 
+0000a810: 6966 2075 7365 7273 2073 6574 2074 6865  if users set the
+0000a820: 2060 2d63 6020 6173 2030 2e39 2028 6f72   `-c` as 0.9 (or
+0000a830: 2061 6e79 2076 616c 7565 2073 6d61 6c6c   any value small
+0000a840: 6572 2074 6861 6e20 3129 2c20 6974 2773  er than 1), it's
+0000a850: 206c 696b 656c 7920 7468 6174 2074 6865   likely that the
+0000a860: 2073 616d 6520 6765 6e65 2077 696c 6c20   same gene will 
+0000a870: 6861 7665 2063 6861 6e63 6520 746f 2061  have chance to a
+0000a880: 7070 6561 7220 6d75 6c74 6970 6c65 2074  ppear multiple t
+0000a890: 696d 6573 2069 6e20 6469 6666 6572 656e  imes in differen
+0000a8a0: 7420 6c6f 6361 6c20 4d52 4248 732e 2054  t local MRBHs. T
+0000a8b0: 6861 7420 6973 2074 6f20 7361 792c 2074  hat is to say, t
+0000a8c0: 6865 2070 6172 616d 6574 6572 2060 2d6b  he parameter `-k
+0000a8d0: 6460 2069 7320 6d65 616e 696e 6766 756c  d` is meaningful
+0000a8e0: 206f 6e6c 7920 7768 656e 2069 7427 7320   only when it's 
+0000a8f0: 7365 7420 746f 6765 7468 6572 2077 6974  set together wit
+0000a900: 6820 7468 6520 7061 7261 6d65 7465 7220  h the parameter 
+0000a910: 602d 6360 2e20 5468 6520 6e75 6d62 6572  `-c`. The number
+0000a920: 206f 6620 7061 7261 6c6c 656c 2074 6872   of parallel thr
+0000a930: 6561 6473 2069 7320 7375 6767 6573 7465  eads is suggeste
+0000a940: 6420 746f 2062 6520 7365 7420 6173 2074  d to be set as t
+0000a950: 6865 206e 756d 6265 7220 6f66 2063 6473  he number of cds
+0000a960: 2066 696c 6573 206d 696e 7573 2031 2e0a   files minus 1..
+0000a970: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
+0000a980: 7461 7274 696e 6720 7275 6e20 6973 2075  tarting run is u
+0000a990: 6e64 6572 2074 6865 2064 6566 6175 6c74  nder the default
+0000a9a0: 2070 6172 616d 6574 6572 2077 6974 6820   parameter with 
+0000a9b0: 7468 6520 636f 6d6d 616e 6420 7368 6f77  the command show
+0000a9c0: 6e20 6173 2062 656c 6f77 2e2a 2a0a 0a60  n as below.**..`
+0000a9d0: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
+0000a9e0: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
+0000a9f0: 6571 7565 6e63 6533 202d 6620 7365 7175  equence3 -f sequ
+0000aa00: 656e 6365 310a 6060 600a 0a2a 2a54 6865  ence1.```..**The
+0000aa10: 2064 656c 696e 6561 7469 6f6e 206f 6620   delineation of 
+0000aa20: 676c 6f62 616c 204d 5242 4873 2a2a 0a60  global MRBHs**.`
+0000aa30: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
+0000aa40: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
+0000aa50: 6571 7565 6e63 6533 202d 676d 202d 6520  equence3 -gm -e 
+0000aa60: 3165 2d31 3020 2d62 7320 3130 3020 2d6e  1e-10 -bs 100 -n
+0000aa70: 7020 3520 282d 6e6e 2920 282d 6e20 3429  p 5 (-nn) (-n 4)
+0000aa80: 2028 2d6b 6629 2028 2d6b 6429 2028 2d63   (-kf) (-kd) (-c
+0000aa90: 2030 2e39 2920 282d 2d74 6f5f 7374 6f70   0.9) (--to_stop
+0000aaa0: 2920 282d 2d63 6473 2920 282d 6f20 7767  ) (--cds) (-o wg
+0000aab0: 645f 646d 6429 2028 2d74 2077 6f72 6b69  d_dmd) (-t worki
+0000aac0: 6e67 5f74 6d70 290a 6060 600a 0a54 6865  ng_tmp).```..The
+0000aad0: 2067 6c6f 6261 6c20 4d52 4248 7320 6973   global MRBHs is
+0000aae0: 2063 6f6e 7374 7275 6374 6564 2062 7920   constructed by 
+0000aaf0: 6578 6861 7573 7469 7665 6c79 206d 6572  exhaustively mer
+0000ab00: 6769 6e67 2061 6c6c 2074 6865 2070 6f73  ging all the pos
+0000ab10: 7369 626c 6520 7061 6972 2d77 6973 6520  sible pair-wise 
+0000ab20: 5242 4873 2065 7863 6570 7420 666f 7220  RBHs except for 
+0000ab30: 7175 6572 7969 6e67 2074 6865 2073 6571  querying the seq
+0000ab40: 7565 6e63 6520 6974 7365 6c66 2c20 7768  uence itself, wh
+0000ab50: 6963 6820 6361 6e20 6265 2069 6e69 7469  ich can be initi
+0000ab60: 6174 6564 2062 7920 6164 6420 7468 6520  ated by add the 
+0000ab70: 666c 6167 2060 2d67 6d60 206f 7220 602d  flag `-gm` or `-
+0000ab80: 2d67 6c6f 6261 6c6d 7262 6860 2e20 5468  -globalmrbh`. Th
+0000ab90: 6520 7265 7374 206f 6620 7265 6c65 7661  e rest of releva
+0000aba0: 6e74 2070 6172 616d 6574 6572 7320 7374  nt parameters st
+0000abb0: 6179 7320 7468 6520 7361 6d65 2061 7320  ays the same as 
+0000abc0: 7468 6520 6c6f 6361 6c20 4d52 4248 732e  the local MRBHs.
+0000abd0: 2054 6865 206e 756d 6265 7220 6f66 2070   The number of p
+0000abe0: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
+0000abf0: 6973 2073 7567 6765 7374 6564 2074 6f20  is suggested to 
+0000ac00: 6265 2073 6574 2061 7320 284e 2d31 294e  be set as (N-1)N
+0000ac10: 2f32 2074 6f6f 2077 6865 7265 204e 2069  /2 too where N i
+0000ac20: 7320 7468 6520 6e75 6d62 6572 206f 6620  s the number of 
+0000ac30: 6364 7320 6669 6c65 7320 746f 2061 6368  cds files to ach
+0000ac40: 6965 7665 2074 6865 2068 6967 6865 7374  ieve the highest
+0000ac50: 2065 6666 6963 6965 6e63 792e 0a0a 2a2a   efficiency...**
+0000ac60: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
+0000ac70: 7469 6e67 2072 756e 2069 7320 756e 6465  ting run is unde
+0000ac80: 7220 7468 6520 6465 6661 756c 7420 7061  r the default pa
+0000ac90: 7261 6d65 7465 7220 7769 7468 2074 6865  rameter with the
+0000aca0: 2063 6f6d 6d61 6e64 2073 686f 776e 2061   command shown a
+0000acb0: 7320 6265 6c6f 772e 2a2a 0a0a 6060 600a  s below.**..```.
+0000acc0: 7767 6420 646d 6420 7365 7175 656e 6365  wgd dmd sequence
+0000acd0: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
+0000ace0: 656e 6365 3320 2d67 6d0a 6060 600a 0a2a  ence3 -gm.```..*
+0000acf0: 2a54 6865 2064 656c 696e 6561 7469 6f6e  *The delineation
+0000ad00: 206f 6620 6f72 7468 6f67 726f 7570 732a   of orthogroups*
+0000ad10: 2a0a 6060 600a 7767 6420 646d 6420 7365  *.```.wgd dmd se
+0000ad20: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
+0000ad30: 3220 7365 7175 656e 6365 3320 2d6f 6920  2 sequence3 -oi 
+0000ad40: 2d6f 6f20 2d65 2031 652d 3130 202d 6273  -oo -e 1e-10 -bs
+0000ad50: 2031 3030 202d 6e70 2035 2028 2d6e 6e29   100 -np 5 (-nn)
+0000ad60: 2028 2d63 6329 2028 2d74 6529 2028 2d6d   (-cc) (-te) (-m
+0000ad70: 6320 302e 3829 2028 2d67 6e29 2028 2d74  c 0.8) (-gn) (-t
+0000ad80: 7265 6520 2766 6173 7474 7265 6527 2920  ree 'fasttree') 
+0000ad90: 282d 7473 2027 2d66 6173 7465 7374 2729  (-ts '-fastest')
+0000ada0: 2028 2d6e 2034 2920 282d 2d74 6f5f 7374   (-n 4) (--to_st
+0000adb0: 6f70 2920 282d 2d63 6473 2920 282d 6f20  op) (--cds) (-o 
+0000adc0: 7767 645f 646d 6429 2028 2d74 2077 6f72  wgd_dmd) (-t wor
+0000add0: 6b69 6e67 5f74 6d70 290a 6060 600a 0a49  king_tmp).```..I
+0000ade0: 6e20 6077 6764 2076 3260 2c20 7765 2061  n `wgd v2`, we a
+0000adf0: 6c73 6f20 696d 706c 656d 656e 7465 6420  lso implemented 
+0000ae00: 616e 2061 6c67 6f72 6974 686d 206f 6620  an algorithm of 
+0000ae10: 6465 6c69 6e65 6174 696e 6720 6f72 7468  delineating orth
+0000ae20: 6f67 726f 7570 732c 2077 6869 6368 2063  ogroups, which c
+0000ae30: 616e 2062 6520 696e 6974 6961 7465 6420  an be initiated 
+0000ae40: 7769 7468 2074 6865 2070 6172 616d 6574  with the paramet
+0000ae50: 6572 2060 2d6f 6960 206f 7220 602d 2d6f  er `-oi` or `--o
+0000ae60: 7274 686f 696e 6665 7260 2e20 5477 6f20  rthoinfer`. Two 
+0000ae70: 7761 7973 206f 6620 6465 6c69 6e65 6174  ways of delineat
+0000ae80: 696f 6e20 6361 6e20 6265 2063 686f 7365  ion can be chose
+0000ae90: 6e2c 2074 6865 2063 6f6e 6361 7465 6e61  n, the concatena
+0000aea0: 7469 6f6e 2077 6179 2028 7365 7420 6279  tion way (set by
+0000aeb0: 2074 6865 2070 6172 616d 6574 6572 2060   the parameter `
+0000aec0: 2d63 6360 206f 7220 602d 2d63 6f6e 6361  -cc` or `--conca
+0000aed0: 7460 2920 6f72 2074 6865 206e 6f6e 2d63  t`) or the non-c
+0000aee0: 6f6e 6361 7465 6e61 7469 6f6e 2028 6465  oncatenation (de
+0000aef0: 6661 756c 7429 2077 6179 2e20 496e 2062  fault) way. In b
+0000af00: 7269 6566 2c20 7468 6520 636f 6e63 6174  rief, the concat
+0000af10: 656e 6174 696f 6e20 7761 7920 6f66 2064  enation way of d
+0000af20: 656c 696e 6561 7469 6e67 206f 7274 686f  elineating ortho
+0000af30: 6772 6f75 7073 2073 7461 7274 7320 7769  groups starts wi
+0000af40: 7468 2063 6f6e 6361 7465 6e61 7469 6e67  th concatenating
+0000af50: 2061 6c6c 2074 6865 2073 6571 7565 6e63   all the sequenc
+0000af60: 6573 2069 6e74 6f20 6120 7369 6e67 6c65  es into a single
+0000af70: 2073 6571 7565 6e63 6520 6669 6c65 2061   sequence file a
+0000af80: 6e64 2074 6865 6e20 696e 6665 7272 696e  nd then inferrin
+0000af90: 6720 7468 6520 7768 6f6c 6520 7061 7261  g the whole para
+0000afa0: 6e6f 6d65 206f 6620 7468 6973 2073 696e  nome of this sin
+0000afb0: 676c 6520 7365 7175 656e 6365 2066 696c  gle sequence fil
+0000afc0: 6520 7769 7468 2074 6865 2063 6c75 7374  e with the clust
+0000afd0: 6572 696e 6720 7265 7375 6c74 7320 6d61  ering results ma
+0000afe0: 7070 6564 2062 6163 6b20 746f 2074 6865  pped back to the
+0000aff0: 2062 656c 6f6e 6769 6e67 2073 7065 6369   belonging speci
+0000b000: 6573 2e20 5768 696c 6520 7468 6520 6e6f  es. While the no
+0000b010: 6e2d 636f 6e63 6174 656e 6174 696f 6e20  n-concatenation 
+0000b020: 7761 7920 7374 6172 7473 2077 6974 6820  way starts with 
+0000b030: 7265 7370 6563 7469 7665 2070 6169 722d  respective pair-
+0000b040: 7769 7365 2064 6961 6d6f 6e64 2073 6561  wise diamond sea
+0000b050: 7263 6820 2869 6e63 6c75 6469 6e67 2071  rch (including q
+0000b060: 7565 7279 696e 6720 7468 6520 7361 6d65  uerying the same
+0000b070: 2073 6571 7565 6e63 6520 6974 7365 6c66   sequence itself
+0000b080: 2920 616e 6420 7468 656e 2061 6c6c 2074  ) and then all t
+0000b090: 6865 2073 6571 7565 6e63 6520 7369 6d69  he sequence simi
+0000b0a0: 6c61 7269 7479 2074 6162 6c65 7320 7769  larity tables wi
+0000b0b0: 6c6c 2062 6520 636f 6e63 6174 656e 6174  ll be concatenat
+0000b0c0: 6564 2061 6e64 2063 6c75 7374 6572 6564  ed and clustered
+0000b0d0: 2069 6e74 6f20 6f72 7468 6f67 726f 7570   into orthogroup
+0000b0e0: 732e 2053 6f6d 6520 6f74 6865 7220 706f  s. Some other po
+0000b0f0: 7373 6962 6c79 2075 7365 6675 6c20 706f  ssibly useful po
+0000b100: 7374 2d63 6c75 7374 6572 696e 6720 6675  st-clustering fu
+0000b110: 6e63 7469 6f6e 7320 6361 6e20 6265 2069  nctions can be i
+0000b120: 6e69 7469 6174 6564 2c20 696e 636c 7564  nitiated, includ
+0000b130: 696e 6720 7468 6520 7061 7261 6d65 7465  ing the paramete
+0000b140: 7220 602d 7465 6020 6f72 2060 2d2d 7465  r `-te` or `--te
+0000b150: 7374 736f 6760 2c20 7768 6963 6820 6361  stsog`, which ca
+0000b160: 6e20 6265 2073 6574 2074 6f20 7374 6172  n be set to star
+0000b170: 7420 7468 6520 756e 6269 6173 6564 2074  t the unbiased t
+0000b180: 6573 7420 6f66 2073 696e 676c 652d 636f  est of single-co
+0000b190: 7079 2067 656e 6520 6661 6d69 6c69 6573  py gene families
+0000b1a0: 2028 6e6f 7465 2074 6861 7420 7468 6973   (note that this
+0000b1b0: 2066 756e 6374 696f 6e20 6e65 6564 7320   function needs 
+0000b1c0: 6068 6d6d 6572 6020 2876 332e 3162 3229  `hmmer` (v3.1b2)
+0000b1d0: 2074 6f20 6265 2069 6e73 7461 6c6c 6564   to be installed
+0000b1e0: 2069 6e20 7468 6520 656e 7669 726f 6e6d   in the environm
+0000b1f0: 656e 7420 7061 7468 292c 2074 6865 2070  ent path), the p
+0000b200: 6172 616d 6574 6572 2060 2d6d 6360 206f  arameter `-mc` o
+0000b210: 7220 602d 2d6d 736f 6763 7574 602c 2072  r `--msogcut`, r
+0000b220: 616e 6769 6e67 2062 6574 7765 656e 2030  anging between 0
+0000b230: 2074 6f20 312c 2077 6869 6368 2063 616e   to 1, which can
+0000b240: 2062 6520 7365 7420 746f 2073 6561 7263   be set to searc
+0000b250: 6820 7468 6520 736f 2d63 616c 6c65 6420  h the so-called 
+0000b260: 6d6f 7374 6c79 2073 696e 676c 652d 636f  mostly single-co
+0000b270: 7079 2066 616d 696c 7920 7768 6963 6820  py family which 
+0000b280: 6861 7320 6869 6768 6572 2074 6861 6e20  has higher than 
+0000b290: 6365 7274 6169 6e20 6375 742d 6f66 6620  certain cut-off 
+0000b2a0: 7065 7263 656e 7461 6765 206f 6620 7370  percentage of sp
+0000b2b0: 6563 6965 7320 636f 7665 7261 6765 2c20  ecies coverage, 
+0000b2c0: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
+0000b2d0: 676e 6020 6f72 2060 2d2d 6765 746e 736f  gn` or `--getnso
+0000b2e0: 6760 2c20 7768 6963 6820 6361 6e20 6265  g`, which can be
+0000b2f0: 2073 6574 2074 6f20 7365 6172 6368 2066   set to search f
+0000b300: 6f72 206e 6573 7465 6420 7369 6e67 6c65  or nested single
+0000b310: 2d63 6f70 7920 6765 6e65 2066 616d 696c  -copy gene famil
+0000b320: 6965 7320 284e 534f 4773 2920 7768 6963  ies (NSOGs) whic
+0000b330: 6820 6973 206f 7269 6769 6e61 6c6c 7920  h is originally 
+0000b340: 6d75 6c74 6979 2d63 6f70 7920 6275 7420  multiy-copy but 
+0000b350: 6861 7320 6120 286d 6f73 746c 7929 2073  has a (mostly) s
+0000b360: 696e 676c 652d 636f 7079 2062 7261 6e63  ingle-copy branc
+0000b370: 6820 2877 6869 6368 2072 6571 7569 7265  h (which require
+0000b380: 7320 7468 6520 6368 6f73 656e 2074 7265  s the chosen tre
+0000b390: 652d 696e 6665 7265 6e63 6520 7072 6f67  e-inference prog
+0000b3a0: 7261 6d20 7365 7420 6279 2060 2d74 7265  ram set by `-tre
+0000b3b0: 6560 206f 7220 602d 2d74 7265 655f 6d65  e` or `--tree_me
+0000b3c0: 7468 6f64 6020 746f 2062 6520 7072 652d  thod` to be pre-
+0000b3d0: 696e 7374 616c 6c65 6420 696e 2074 6865  installed in the
+0000b3e0: 2065 6e76 6972 6f6e 6d65 6e74 2070 6174   environment pat
+0000b3f0: 6820 7769 7468 2074 6865 2070 6172 616d  h with the param
+0000b400: 6574 6572 7320 7365 7474 696e 6720 666f  eters setting fo
+0000b410: 7220 6765 6e65 2074 7265 6520 696e 6665  r gene tree infe
+0000b420: 7265 6e63 6520 636f 6e74 726f 6c6c 6564  rence controlled
+0000b430: 2062 7920 602d 7473 6020 6f72 2060 2d2d   by `-ts` or `--
+0000b440: 7472 6565 7365 7460 292e 2054 6865 2070  treeset`). The p
+0000b450: 726f 6772 616d 2060 7767 6420 646d 6460  rogram `wgd dmd`
+0000b460: 2077 6f75 6c64 2073 7469 6c6c 2063 6f6e   would still con
+0000b470: 6475 6374 2074 6865 2052 4248 7320 6361  duct the RBHs ca
+0000b480: 6c63 756c 6174 696f 6e20 756e 6c65 7373  lculation unless
+0000b490: 2074 6865 2070 6172 616d 6574 6572 2060   the parameter `
+0000b4a0: 2d6f 6f60 206f 7220 602d 2d6f 6e6c 796f  -oo` or `--onlyo
+0000b4b0: 7274 686f 6020 7761 7320 7365 742e 2049  rtho` was set. I
+0000b4c0: 6620 6f6e 6520 6f6e 6c79 2077 616e 7473  f one only wants
+0000b4d0: 2074 6f20 696e 6665 7220 7468 6520 6f72   to infer the or
+0000b4e0: 7468 6f67 726f 7570 732c 2069 7427 7320  thogroups, it's 
+0000b4f0: 7375 6767 6573 7465 6420 746f 2061 6464  suggested to add
+0000b500: 2074 6865 2066 6c61 6720 602d 6f6f 6020   the flag `-oo` 
+0000b510: 746f 206a 7573 7420 696d 706c 656d 656e  to just implemen
+0000b520: 7420 7468 6520 6f72 7468 6f67 726f 7570  t the orthogroup
+0000b530: 7320 6465 6c69 6e65 6174 696f 6e20 616e  s delineation an
+0000b540: 616c 7973 6973 2e20 5468 6520 6e75 6d62  alysis. The numb
+0000b550: 6572 206f 6620 7061 7261 6c6c 656c 2074  er of parallel t
+0000b560: 6872 6561 6473 2069 7320 7375 6767 6573  hreads is sugges
+0000b570: 7465 6420 746f 2062 6520 7365 7420 6173  ted to be set as
+0000b580: 2028 4e2b 3129 4e2f 3220 7768 6572 6520   (N+1)N/2 where 
+0000b590: 4e20 6973 2074 6865 206e 756d 6265 7220  N is the number 
+0000b5a0: 6f66 2063 6473 2066 696c 6573 2074 6f20  of cds files to 
+0000b5b0: 6163 6869 6576 6520 7468 6520 6869 6768  achieve the high
+0000b5c0: 6573 7420 6566 6669 6369 656e 6379 2073  est efficiency s
+0000b5d0: 696e 6365 2074 6865 2073 656c 662d 636f  ince the self-co
+0000b5e0: 6d70 6172 6973 6f6e 2069 7320 616c 736f  mparison is also
+0000b5f0: 2069 6e63 6c75 6465 642e 0a0a 2a2a 5468   included...**Th
+0000b600: 6520 6465 6661 756c 7420 7365 7474 696e  e default settin
+0000b610: 6720 6f66 2070 6172 616d 6574 6572 7320  g of parameters 
+0000b620: 6973 2061 2072 6561 736f 6e61 626c 6520  is a reasonable 
+0000b630: 7374 6172 7469 6e67 2070 6f69 6e74 2077  starting point w
+0000b640: 6974 6820 7468 6520 636f 6d6d 616e 6420  ith the command 
+0000b650: 6173 2062 656c 6f77 2e2a 2a0a 0a60 6060  as below.**..```
+0000b660: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
+0000b670: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
+0000b680: 7565 6e63 6533 202d 6f69 202d 6f6f 0a60  uence3 -oi -oo.`
+0000b690: 6060 0a0a 2a2a 5468 6520 636f 6c6c 696e  ``..**The collin
+0000b6a0: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
+0000b6b0: 696e 6665 7265 6e63 6520 6f66 2070 6879  inference of phy
+0000b6c0: 6c6f 6765 6e79 2a2a 0a60 6060 0a77 6764  logeny**.```.wgd
+0000b6d0: 2064 6d64 2073 6571 7565 6e63 6531 2073   dmd sequence1 s
+0000b6e0: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000b6f0: 6533 202d 6170 2061 7064 6174 6120 2d73  e3 -ap apdata -s
+0000b700: 6d20 736d 6461 7461 202d 6c65 206c 6564  m smdata -le led
+0000b710: 6174 6120 2d67 7420 6774 6461 7461 202d  ata -gt gtdata -
+0000b720: 636f 6320 282d 7472 6565 2027 6661 7374  coc (-tree 'fast
+0000b730: 7472 6565 2729 2028 2d74 7320 272d 6661  tree') (-ts '-fa
+0000b740: 7374 6573 7427 2920 282d 6e20 3429 2028  stest') (-n 4) (
+0000b750: 2d2d 746f 5f73 746f 7029 2028 2d2d 6364  --to_stop) (--cd
+0000b760: 7329 2028 2d6f 2077 6764 5f64 6d64 2920  s) (-o wgd_dmd) 
+0000b770: 282d 7420 776f 726b 696e 675f 746d 7029  (-t working_tmp)
+0000b780: 0a60 6060 0a0a 4120 6e6f 7665 6c20 7068  .```..A novel ph
+0000b790: 796c 6f67 656e 6574 6963 2069 6e66 6572  ylogenetic infer
+0000b7a0: 656e 6365 206d 6574 686f 6420 6e61 6d65  ence method name
+0000b7b0: 6420 2263 6f6c 6c69 6e65 6172 2063 6f61  d "collinear coa
+0000b7c0: 6c65 7363 656e 6365 2069 6e66 6572 656e  lescence inferen
+0000b7d0: 6365 2220 6973 2061 6c73 6f20 696d 706c  ce" is also impl
+0000b7e0: 656d 656e 7465 6420 696e 2060 7767 6420  emented in `wgd 
+0000b7f0: 7632 602e 2046 6f72 2074 6869 7320 616e  v2`. For this an
+0000b800: 616c 7973 6973 2c20 7573 6572 7320 6e65  alysis, users ne
+0000b810: 6564 2074 6f20 7072 6f76 6964 6520 7468  ed to provide th
+0000b820: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
+0000b830: 6669 6c65 2062 7920 602d 6170 6020 6f72  file by `-ap` or
+0000b840: 2060 2d2d 616e 6368 6f72 706f 696e 7473   `--anchorpoints
+0000b850: 602c 2074 6865 2063 6f6c 6c69 6e65 6172  `, the collinear
+0000b860: 2073 6567 6d65 6e74 7320 6669 6c65 2062   segments file b
+0000b870: 7920 602d 736d 6020 6f72 2060 2d2d 7365  y `-sm` or `--se
+0000b880: 676d 656e 7473 602c 2074 6865 206c 6973  gments`, the lis
+0000b890: 7473 6567 6d65 6e74 7320 6669 6c65 2062  tsegments file b
+0000b8a0: 7920 602d 6c65 6020 6f72 2060 2d2d 6c69  y `-le` or `--li
+0000b8b0: 7374 656c 656d 656e 7473 602c 2061 6e64  stelements`, and
+0000b8c0: 2074 6865 2067 656e 6520 7461 626c 6520   the gene table 
+0000b8d0: 6669 6c65 2062 7920 602d 6774 6020 6f72  file by `-gt` or
+0000b8e0: 2060 2d2d 6765 6e65 7461 626c 6560 2c20   `--genetable`, 
+0000b8f0: 616c 6c20 6f66 2077 6869 6368 2063 616e  all of which can
+0000b900: 2062 6520 7072 6f64 7563 6564 2069 6e20   be produced in 
+0000b910: 7468 6520 7072 6f67 7261 6d20 6077 6764  the program `wgd
+0000b920: 2073 796e 602e 2054 6865 2070 6172 616d   syn`. The param
+0000b930: 6574 6572 2060 2d63 6f63 6020 6f72 2060  eter `-coc` or `
+0000b940: 2d2d 636f 6c6c 696e 6561 7263 6f61 6c65  --collinearcoale
+0000b950: 7363 656e 6365 6020 6e65 6564 7320 746f  scence` needs to
+0000b960: 2062 6520 7365 7420 746f 2073 7461 7274   be set to start
+0000b970: 2074 6869 7320 616e 616c 7973 6973 2e20   this analysis. 
+0000b980: 5468 6520 7472 6565 2d69 6e66 6572 656e  The tree-inferen
+0000b990: 6365 2070 726f 6772 616d 2061 6e64 2074  ce program and t
+0000b9a0: 6865 2061 7373 6f63 6961 7465 6420 7061  he associated pa
+0000b9b0: 7261 6d65 7465 7273 2063 616e 2062 6520  rameters can be 
+0000b9c0: 7365 7420 6a75 7374 2061 7320 6162 6f76  set just as abov
+0000b9d0: 6520 6279 2060 2d74 7265 6560 206f 7220  e by `-tree` or 
+0000b9e0: 602d 2d74 7265 655f 6d65 7468 6f64 6020  `--tree_method` 
+0000b9f0: 616e 6420 602d 7473 6020 6f72 2060 2d2d  and `-ts` or `--
+0000ba00: 7472 6565 7365 7460 2e20 506c 6561 7365  treeset`. Please
+0000ba10: 2061 6c73 6f20 6d61 6b65 2073 7572 6520   also make sure 
+0000ba20: 7468 6520 6368 6f73 656e 2074 7265 652d  the chosen tree-
+0000ba30: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
+0000ba40: 6d20 6973 2069 6e73 7461 6c6c 6564 2069  m is installed i
+0000ba50: 6e20 7468 6520 656e 7669 726f 6e6d 656e  n the environmen
+0000ba60: 7420 7061 7468 2e20 5468 6520 7072 6f67  t path. The prog
+0000ba70: 7261 6d20 6061 7374 7261 6c2d 7072 6f60  ram `astral-pro`
+0000ba80: 2069 7320 7265 7175 6972 6564 2074 6f20   is required to 
+0000ba90: 6265 2069 6e73 7461 6c6c 6564 2069 6e20  be installed in 
+0000baa0: 7468 6520 656e 7669 726f 6e6d 656e 7420  the environment 
+0000bab0: 7061 7468 2074 6f6f 2e20 4e6f 7465 2074  path too. Note t
+0000bac0: 6861 7420 7468 6572 6520 7368 6f75 6c64  hat there should
+0000bad0: 2062 6520 6e6f 2064 7570 6c69 6361 7465   be no duplicate
+0000bae0: 6420 6765 6e65 2049 4473 2069 6e20 7468  d gene IDs in th
+0000baf0: 6520 7365 7175 656e 6365 2066 696c 652e  e sequence file.
+0000bb00: 2054 6865 2070 6172 616c 6c65 6c20 7468   The parallel th
+0000bb10: 7265 6164 7320 6865 7265 2061 7265 2074  reads here are t
+0000bb20: 6f20 6163 6365 6c65 7261 7465 2074 6865  o accelerate the
+0000bb30: 2073 6571 7565 6e63 6520 616c 6967 6e6d   sequence alignm
+0000bb40: 656e 7420 616e 6420 6765 6e65 2074 7265  ent and gene tre
+0000bb50: 6520 696e 6665 7265 6e63 6520 666f 7220  e inference for 
+0000bb60: 6561 6368 2067 656e 6520 6661 6d69 6c79  each gene family
+0000bb70: 2061 6e64 2074 6875 7320 7375 6767 6573   and thus sugges
+0000bb80: 7465 6420 746f 2062 6520 7365 7420 6173  ted to be set as
+0000bb90: 206d 7563 6820 6173 2074 6865 206e 756d   much as the num
+0000bba0: 6265 7220 6f66 2067 656e 6520 6661 6d69  ber of gene fami
+0000bbb0: 6c69 6573 2e0a 0a2a 2a41 2073 7567 6765  lies...**A sugge
+0000bbc0: 7374 6564 2073 7461 7274 696e 6720 7275  sted starting ru
+0000bbd0: 6e20 6f66 2074 6869 7320 616e 616c 7973  n of this analys
+0000bbe0: 6973 2069 7320 7769 7468 2074 6865 2073  is is with the s
+0000bbf0: 696d 706c 6520 636f 6d6d 616e 6420 6265  imple command be
+0000bc00: 6c6f 772e 2a2a 0a0a 6060 600a 7767 6420  low.**..```.wgd 
+0000bc10: 646d 6420 7365 7175 656e 6365 3120 7365  dmd sequence1 se
+0000bc20: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+0000bc30: 3320 2d61 7020 6170 6461 7461 202d 736d  3 -ap apdata -sm
+0000bc40: 2073 6d64 6174 6120 2d6c 6520 6c65 6461   smdata -le leda
+0000bc50: 7461 202d 6774 2067 7464 6174 6120 2d63  ta -gt gtdata -c
+0000bc60: 6f63 0a60 6060 0a0a 2323 2320 7767 6420  oc.```..### wgd 
+0000bc70: 666f 6375 730a 0a2a 2a54 6865 2063 6f6e  focus..**The con
+0000bc80: 6361 7465 6e61 7469 6f6e 2d62 6173 6564  catenation-based
+0000bc90: 2f63 6f61 6c65 7363 656e 6365 2d62 6173  /coalescence-bas
+0000bca0: 6564 2070 6879 6c6f 6765 6e65 7469 6320  ed phylogenetic 
+0000bcb0: 696e 6665 7265 6e63 652a 2a0a 6060 600a  inference**.```.
+0000bcc0: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
+0000bcd0: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
+0000bce0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+0000bcf0: 2028 2d2d 636f 6e63 6174 656e 6174 696f   (--concatenatio
+0000bd00: 6e29 2028 2d2d 636f 616c 6573 6365 6e63  n) (--coalescenc
+0000bd10: 6529 2028 2d74 7265 6520 2766 6173 7474  e) (-tree 'fastt
+0000bd20: 7265 6527 2920 282d 7473 2027 2d66 6173  ree') (-ts '-fas
+0000bd30: 7465 7374 2729 2028 2d6e 2034 2920 282d  test') (-n 4) (-
+0000bd40: 2d74 6f5f 7374 6f70 2920 282d 2d63 6473  -to_stop) (--cds
+0000bd50: 2920 282d 6f20 7767 645f 666f 6375 7329  ) (-o wgd_focus)
+0000bd60: 2028 2d74 2077 6f72 6b69 6e67 5f74 6d70   (-t working_tmp
+0000bd70: 290a 6060 600a 0a54 6865 2070 726f 6772  ).```..The progr
+0000bd80: 616d 2060 7767 6420 666f 6375 7360 2069  am `wgd focus` i
+0000bd90: 6d70 6c65 6d65 6e74 6564 2074 776f 2062  mplemented two b
+0000bda0: 6173 6963 2070 6879 6c6f 6765 6e65 7469  asic phylogeneti
+0000bdb0: 6320 696e 6665 7265 6e63 6520 6d65 7468  c inference meth
+0000bdc0: 6f64 732c 2069 2e65 2e2c 2063 6f6e 6361  ods, i.e., conca
+0000bdd0: 7465 6e61 7469 6f6e 2d62 6173 6564 2061  tenation-based a
+0000bde0: 6e64 2063 6f61 6c65 7363 656e 6365 2d62  nd coalescence-b
+0000bdf0: 6173 6564 206d 6574 686f 6473 2e20 546f  ased methods. To
+0000be00: 2069 6e69 7469 6174 6520 7468 6573 6520   initiate these 
+0000be10: 616e 616c 7973 6973 2c20 7573 6572 7320  analysis, users 
+0000be20: 6e65 6564 2074 6f20 7365 7420 7468 6520  need to set the 
+0000be30: 666c 6167 206f 7074 696f 6e20 602d 2d63  flag option `--c
+0000be40: 6f6e 6361 7465 6e61 7469 6f6e 6020 6f72  oncatenation` or
+0000be50: 2060 2d2d 636f 616c 6573 6365 6e63 6560   `--coalescence`
+0000be60: 2e20 5468 6520 636f 6e63 6174 656e 6174  . The concatenat
+0000be70: 696f 6e2d 6261 7365 6420 6d65 7468 6f64  ion-based method
+0000be80: 2069 6e63 6c75 6465 7320 6120 6665 7720   includes a few 
+0000be90: 6d61 6a6f 7220 7374 6570 732c 2069 2e65  major steps, i.e
+0000bea0: 2e2c 2074 6865 206d 756c 7469 706c 6520  ., the multiple 
+0000beb0: 7365 7175 656e 6365 2061 6c69 676e 6d65  sequence alignme
+0000bec0: 6e74 2028 4d53 4129 206f 6620 6561 6368  nt (MSA) of each
+0000bed0: 2067 656e 6520 6661 6d69 6c79 2c20 7468   gene family, th
+0000bee0: 6520 636f 6e63 6174 656e 6174 696f 6e20  e concatenation 
+0000bef0: 6f66 2061 6c6c 2067 656e 6520 6661 6d69  of all gene fami
+0000bf00: 6c69 6573 2061 6e64 2074 6865 6e20 7468  lies and then th
+0000bf10: 6520 6765 6e65 2074 7265 6520 2861 6c73  e gene tree (als
+0000bf20: 6f20 7370 6563 6965 7320 7472 6565 2069  o species tree i
+0000bf30: 6e20 7468 6973 2063 6173 6529 2069 6e66  n this case) inf
+0000bf40: 6572 656e 6365 2e20 5468 6520 636f 616c  erence. The coal
+0000bf50: 6573 6365 6e63 652d 6261 7365 6420 6d65  escence-based me
+0000bf60: 7468 6f64 2077 696c 6c20 696e 7374 6561  thod will instea
+0000bf70: 6420 7065 7266 6f72 6d20 7468 6520 4d53  d perform the MS
+0000bf80: 4120 6f66 2065 6163 6820 6765 6e65 2066  A of each gene f
+0000bf90: 616d 696c 7920 616e 6420 7468 6520 6765  amily and the ge
+0000bfa0: 6e65 2074 7265 6520 696e 6665 7265 6e63  ne tree inferenc
+0000bfb0: 6520 6261 7365 6420 6f6e 2065 6163 6820  e based on each 
+0000bfc0: 4d53 412c 2061 6e64 2074 6865 6e20 696e  MSA, and then in
+0000bfd0: 6665 7220 7468 6520 7370 6563 6965 7320  fer the species 
+0000bfe0: 7472 6565 2062 6173 6564 206f 6e20 7468  tree based on th
+0000bff0: 6573 6520 696e 6469 7669 6475 616c 2067  ese individual g
+0000c000: 656e 6520 7472 6565 732e 2054 6865 2074  ene trees. The t
+0000c010: 7265 652d 696e 6665 7265 6e63 6520 7072  ree-inference pr
+0000c020: 6f67 7261 6d20 616e 6420 7468 6520 6173  ogram and the as
+0000c030: 736f 6369 6174 6564 2070 6172 616d 6574  sociated paramet
+0000c040: 6572 7320 6361 6e20 6265 2073 6574 206a  ers can be set j
+0000c050: 7573 7420 6173 2061 626f 7665 2062 7920  ust as above by 
+0000c060: 602d 7472 6565 6020 6f72 2060 2d2d 7472  `-tree` or `--tr
+0000c070: 6565 5f6d 6574 686f 6460 2061 6e64 2060  ee_method` and `
+0000c080: 2d74 7360 206f 7220 602d 2d74 7265 6573  -ts` or `--trees
+0000c090: 6574 602e 2050 6c65 6173 6520 616c 736f  et`. Please also
+0000c0a0: 206d 616b 6520 7375 7265 2074 6865 2063   make sure the c
+0000c0b0: 686f 7365 6e20 7472 6565 2d69 6e66 6572  hosen tree-infer
+0000c0c0: 656e 6365 2070 726f 6772 616d 2069 7320  ence program is 
+0000c0d0: 696e 7374 616c 6c65 6420 696e 2074 6865  installed in the
+0000c0e0: 2065 6e76 6972 6f6e 6d65 6e74 2070 6174   environment pat
+0000c0f0: 682e 2054 6865 2070 726f 6772 616d 2060  h. The program `
+0000c100: 6173 7472 616c 2d70 726f 6020 6973 2072  astral-pro` is r
+0000c110: 6571 7569 7265 6420 746f 2062 6520 696e  equired to be in
+0000c120: 7374 616c 6c65 6420 696e 2074 6865 2065  stalled in the e
+0000c130: 6e76 6972 6f6e 6d65 6e74 2070 6174 6820  nvironment path 
+0000c140: 6966 2074 6865 2063 6f61 6c65 7363 656e  if the coalescen
+0000c150: 6365 2d62 6173 6564 206d 6574 686f 6420  ce-based method 
+0000c160: 6973 2063 686f 7365 6e2e 2054 6865 2070  is chosen. The p
+0000c170: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
+0000c180: 6865 7265 2061 7265 2074 6f20 6163 6365  here are to acce
+0000c190: 6c65 7261 7465 2074 6865 2073 6571 7565  lerate the seque
+0000c1a0: 6e63 6520 616c 6967 6e6d 656e 7420 616e  nce alignment an
+0000c1b0: 6420 6765 6e65 2074 7265 6520 696e 6665  d gene tree infe
+0000c1c0: 7265 6e63 6520 666f 7220 6561 6368 2067  rence for each g
+0000c1d0: 656e 6520 6661 6d69 6c79 2074 6f6f 2061  ene family too a
+0000c1e0: 6e64 2074 6875 7320 7375 6767 6573 7465  nd thus suggeste
+0000c1f0: 6420 746f 2062 6520 7365 7420 6173 206d  d to be set as m
+0000c200: 7563 6820 6173 2074 6865 206e 756d 6265  uch as the numbe
+0000c210: 7220 6f66 2067 656e 6520 6661 6d69 6c69  r of gene famili
+0000c220: 6573 2e0a 0a2a 2a41 2073 7567 6765 7374  es...**A suggest
+0000c230: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
+0000c240: 6f66 2074 6869 7320 616e 616c 7973 6973  of this analysis
+0000c250: 2069 7320 7769 7468 2074 6865 2073 696d   is with the sim
+0000c260: 706c 6520 636f 6d6d 616e 6420 6265 6c6f  ple command belo
+0000c270: 772e 2a2a 0a0a 6060 600a 7767 6420 666f  w.**..```.wgd fo
+0000c280: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+0000c290: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+0000c2a0: 2073 6571 7565 6e63 6533 202d 2d63 6f6e   sequence3 --con
+0000c2b0: 6361 7465 6e61 7469 6f6e 0a77 6764 2066  catenation.wgd f
+0000c2c0: 6f63 7573 2066 616d 696c 6965 7320 7365  ocus families se
+0000c2d0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
+0000c2e0: 3220 7365 7175 656e 6365 3320 2d2d 636f  2 sequence3 --co
+0000c2f0: 616c 6573 6365 6e63 650a 6060 600a 0a2a  alescence.```..*
+0000c300: 2a54 6865 2066 756e 6374 696f 6e61 6c20  *The functional 
+0000c310: 616e 6e6f 7461 7469 6f6e 206f 6620 6765  annotation of ge
+0000c320: 6e65 2066 616d 696c 6965 732a 2a0a 6060  ne families**.``
+0000c330: 600a 7767 6420 666f 6375 7320 6661 6d69  `.wgd focus fami
+0000c340: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
+0000c350: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000c360: 6533 202d 2d61 6e6e 6f74 6174 696f 6e20  e3 --annotation 
+0000c370: 6567 676e 6f67 202d 6564 2065 6464 6174  eggnog -ed eddat
+0000c380: 6120 2d2d 646d 6e62 2064 6264 6174 610a  a --dmnb dbdata.
+0000c390: 6060 600a 0a54 6865 2070 726f 6772 616d  ```..The program
+0000c3a0: 2060 7767 6420 666f 6375 7360 2061 6c73   `wgd focus` als
+0000c3b0: 6f20 6164 6465 6420 736f 6d65 2077 7261  o added some wra
+0000c3c0: 7070 696e 6720 6675 6e63 7469 6f6e 7320  pping functions 
+0000c3d0: 666f 7220 6675 6e63 7469 6f6e 616c 2061  for functional a
+0000c3e0: 6e6e 6f74 6174 696f 6e20 6f66 2067 656e  nnotation of gen
+0000c3f0: 6520 6661 6d69 6c69 6573 206f 6e20 7468  e families on th
+0000c400: 6520 686f 6f64 206f 6620 6461 7461 6261  e hood of databa
+0000c410: 7365 7320 616e 6420 736f 6674 7761 7265  ses and software
+0000c420: 7320 666f 7220 696e 7374 616e 6365 2060  s for instance `
+0000c430: 4567 674e 4f47 6020 616e 6420 6045 6767  EggNOG` and `Egg
+0000c440: 4e4f 472d 6d61 7070 6572 602e 2046 6f72  NOG-mapper`. For
+0000c450: 2074 6865 2061 6e6e 6f74 6174 696f 6e20   the annotation 
+0000c460: 7573 696e 6720 6045 6767 4e4f 472d 6d61  using `EggNOG-ma
+0000c470: 7070 6572 602c 2075 7365 7273 206e 6565  pper`, users nee
+0000c480: 6420 746f 2070 726f 7669 6465 2074 6865  d to provide the
+0000c490: 2070 6174 6820 746f 2074 6865 2065 6767   path to the egg
+0000c4a0: 4e4f 4720 616e 6e6f 7461 7469 6f6e 2064  NOG annotation d
+0000c4b0: 6174 6162 6173 6520 7669 6120 7468 6520  atabase via the 
+0000c4c0: 6f70 7469 6f6e 2060 2d65 6460 2c20 7468  option `-ed`, th
+0000c4d0: 6520 7061 7468 2074 6f20 7468 6520 6469  e path to the di
+0000c4e0: 616d 6f6e 642d 636f 6d70 6174 6962 6c65  amond-compatible
+0000c4f0: 2064 6174 6162 6173 6520 7669 6120 7468   database via th
+0000c500: 6520 6f70 7469 6f6e 2060 2d2d 646d 6e64  e option `--dmnd
+0000c510: 5f64 6260 2061 6e64 2074 6865 206f 7074  _db` and the opt
+0000c520: 696f 6e20 602d 2d61 6e6e 6f74 6174 696f  ion `--annotatio
+0000c530: 6e60 2073 6574 2061 7320 2265 6767 6e6f  n` set as "eggno
+0000c540: 6722 2e20 5468 6520 6d61 6e6e 6572 2068  g". The manner h
+0000c550: 6f77 2050 4641 4d20 616e 6e6f 7461 7469  ow PFAM annotati
+0000c560: 6f6e 2077 696c 6c20 6265 2070 6572 666f  on will be perfo
+0000c570: 726d 6564 2063 616e 2062 6520 636f 6e74  rmed can be cont
+0000c580: 726f 6c6c 6564 2076 6961 2074 6865 206f  rolled via the o
+0000c590: 7074 696f 6e20 6070 6661 6d60 2c20 6569  ption `pfam`, ei
+0000c5a0: 7468 6572 2022 6e6f 6e65 222c 2022 7265  ther "none", "re
+0000c5b0: 616c 6967 6e22 206f 7220 2264 656e 6f76  align" or "denov
+0000c5c0: 6f22 2c20 7468 6520 6465 7461 696c 6564  o", the detailed
+0000c5d0: 2065 7870 6c61 6e61 7469 6f6e 2063 616e   explanation can
+0000c5e0: 2062 6520 666f 756e 6420 6174 2074 6865   be found at the
+0000c5f0: 2077 696b 6920 6f66 205b 4567 674e 4f47   wiki of [EggNOG
+0000c600: 2d6d 6170 7065 725d 2868 7474 7073 3a2f  -mapper](https:/
+0000c610: 2f67 6974 6875 622e 636f 6d2f 6567 676e  /github.com/eggn
+0000c620: 6f67 6462 2f65 6767 6e6f 672d 6d61 7070  ogdb/eggnog-mapp
+0000c630: 6572 2f77 696b 6929 2e20 506c 6561 7365  er/wiki). Please
+0000c640: 2070 7265 2d69 6e73 7461 6c6c 2074 6865   pre-install the
+0000c650: 2060 4567 674e 4f47 2d6d 6170 7065 7260   `EggNOG-mapper`
+0000c660: 2070 7974 686f 6e20 7061 636b 6167 6520   python package 
+0000c670: 6966 2075 7369 6e67 2074 6869 7320 6675  if using this fu
+0000c680: 6e63 7469 6f6e 2e20 466f 7220 7468 6520  nction. For the 
+0000c690: 616e 6e6f 7461 7469 6f6e 2075 7369 6e67  annotation using
+0000c6a0: 2060 686d 6d73 6361 6e60 2c20 7768 6174   `hmmscan`, what
+0000c6b0: 2069 7320 696d 706c 656d 656e 7465 6420   is implemented 
+0000c6c0: 696e 2060 7767 6420 7632 6020 6973 2061  in `wgd v2` is a
+0000c6d0: 2073 696d 706c 6520 6275 6e64 6c65 2066   simple bundle f
+0000c6e0: 756e 6374 696f 6e20 746f 2070 6572 666f  unction to perfo
+0000c6f0: 726d 2060 686d 6d73 6361 6e60 2061 6e61  rm `hmmscan` ana
+0000c700: 6c79 7369 7320 666f 7220 6120 6769 7665  lysis for a give
+0000c710: 6e20 686d 6d20 7072 6f66 696c 6520 616e  n hmm profile an
+0000c720: 6420 7365 7420 6f66 2067 656e 6520 6661  d set of gene fa
+0000c730: 6d69 6c69 6573 2073 7563 6820 7468 6174  milies such that
+0000c740: 2075 7365 7273 206e 6565 6420 746f 2073   users need to s
+0000c750: 6574 2074 6865 206f 7074 696f 6e20 602d  et the option `-
+0000c760: 2d61 6e6e 6f74 6174 696f 6e60 2061 7320  -annotation` as 
+0000c770: 2268 6d6d 7066 616d 2220 616e 6420 7072  "hmmpfam" and pr
+0000c780: 6f76 6964 6520 7468 6520 686d 6d70 726f  ovide the hmmpro
+0000c790: 6669 6c65 2076 6961 2074 6865 206f 7074  file via the opt
+0000c7a0: 696f 6e20 602d 2d68 6d6d 602e 2046 6f72  ion `--hmm`. For
+0000c7b0: 2074 6865 2061 6e6e 6f74 6174 696f 6e20   the annotation 
+0000c7c0: 7573 696e 6720 6069 6e74 6572 7072 6f73  using `interpros
+0000c7d0: 6361 6e60 2c20 7573 6572 7320 6e65 6564  can`, users need
+0000c7e0: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
+0000c7f0: 7061 7468 2074 6f20 7468 6520 696e 7465  path to the inte
+0000c800: 7270 726f 7363 616e 2069 6e73 7461 6c6c  rproscan install
+0000c810: 6174 696f 6e20 666f 6c64 6572 2077 6865  ation folder whe
+0000c820: 7265 2074 6865 7265 2069 7320 6120 6069  re there is a `i
+0000c830: 6e74 6572 7072 6f73 6361 6e2e 7368 6020  nterproscan.sh` 
+0000c840: 6669 6c65 2076 6961 2074 6865 206f 7074  file via the opt
+0000c850: 696f 6e20 602d 2d65 7865 7061 7468 6020  ion `--exepath` 
+0000c860: 616e 6420 7365 7420 7468 6520 7468 6520  and set the the 
+0000c870: 6f70 7469 6f6e 2060 2d2d 616e 6e6f 7461  option `--annota
+0000c880: 7469 6f6e 6020 6173 2022 696e 7465 7270  tion` as "interp
+0000c890: 726f 7363 616e 222e 2054 6865 2070 6172  roscan". The par
+0000c8a0: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
+0000c8b0: 7265 2061 7265 2074 6f20 7061 7261 6c6c  re are to parall
+0000c8c0: 656c 697a 6520 7468 6520 616e 616c 7973  elize the analys
+0000c8d0: 6973 2066 6f72 2065 6163 6820 6765 6e65  is for each gene
+0000c8e0: 2066 616d 696c 7920 616e 6420 7468 7573   family and thus
+0000c8f0: 2073 7567 6765 7374 6564 2074 6f20 6265   suggested to be
+0000c900: 2073 6574 2061 7320 6d75 6368 2061 7320   set as much as 
+0000c910: 7468 6520 6e75 6d62 6572 206f 6620 6765  the number of ge
+0000c920: 6e65 2066 616d 696c 6965 732e 0a0a 2a2a  ne families...**
+0000c930: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
+0000c940: 7469 6e67 2072 756e 206f 6620 7468 6973  ting run of this
+0000c950: 2061 6e61 6c79 7369 7320 6361 6e20 6265   analysis can be
+0000c960: 2077 6974 6820 7468 6520 636f 6d6d 616e   with the comman
+0000c970: 6420 6265 6c6f 772e 2a2a 0a0a 6060 600a  d below.**..```.
+0000c980: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
+0000c990: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
+0000c9a0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+0000c9b0: 202d 2d61 6e6e 6f74 6174 696f 6e20 6567   --annotation eg
+0000c9c0: 676e 6f67 202d 6564 2065 6464 6174 6120  gnog -ed eddata 
+0000c9d0: 2d2d 646d 6e62 2064 6264 6174 610a 7767  --dmnb dbdata.wg
+0000c9e0: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
+0000c9f0: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
+0000ca00: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
+0000ca10: 2d61 6e6e 6f74 6174 696f 6e20 686d 6d70  -annotation hmmp
+0000ca20: 6661 6d20 2d2d 686d 6d20 686d 6d64 6174  fam --hmm hmmdat
+0000ca30: 610a 7767 6420 666f 6375 7320 6661 6d69  a.wgd focus fami
+0000ca40: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
+0000ca50: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000ca60: 6533 202d 2d61 6e6e 6f74 6174 696f 6e20  e3 --annotation 
+0000ca70: 696e 7465 7270 726f 7363 616e 202d 2d65  interproscan --e
+0000ca80: 7865 7061 7468 2024 5041 5448 0a60 6060  xepath $PATH.```
+0000ca90: 0a0a 2a2a 5468 6520 7068 796c 6f67 656e  ..**The phylogen
+0000caa0: 6574 6963 2064 6174 696e 6720 6f66 2057  etic dating of W
+0000cab0: 4744 732a 2a0a 6060 600a 7767 6420 666f  GDs**.```.wgd fo
+0000cac0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+0000cad0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+0000cae0: 2073 6571 7565 6e63 6533 202d 6420 6d63   sequence3 -d mc
+0000caf0: 6d63 7472 6565 202d 7370 2073 7064 6174  mctree -sp spdat
+0000cb00: 6120 282d 2d70 726f 7463 6f63 6461 7469  a (--protcocdati
+0000cb10: 6e67 2920 282d 2d70 6172 7469 7469 6f6e  ng) (--partition
+0000cb20: 2920 282d 2d61 616d 6f64 656c 206c 6729  ) (--aamodel lg)
+0000cb30: 2028 2d64 7320 2762 7572 6e69 6e20 3d20   (-ds 'burnin = 
+0000cb40: 3230 3030 2729 2028 2d64 7320 2773 616d  2000') (-ds 'sam
+0000cb50: 7066 7265 7120 3d20 3130 3030 2729 2028  pfreq = 1000') (
+0000cb60: 2d64 7320 276e 7361 6d70 6c65 203d 2032  -ds 'nsample = 2
+0000cb70: 3030 3030 2729 2020 282d 6e20 3429 2028  0000')  (-n 4) (
+0000cb80: 2d2d 746f 5f73 746f 7029 2028 2d2d 6364  --to_stop) (--cd
+0000cb90: 7329 2028 2d6f 2077 6764 5f66 6f63 7573  s) (-o wgd_focus
+0000cba0: 2920 282d 7420 776f 726b 696e 675f 746d  ) (-t working_tm
+0000cbb0: 7029 0a60 6060 0a0a 5468 6520 6162 736f  p).```..The abso
+0000cbc0: 6c75 7465 2064 6174 696e 6720 6f66 2057  lute dating of W
+0000cbd0: 4744 7320 6973 2061 2073 7065 6369 6669  GDs is a specifi
+0000cbe0: 6320 7069 7065 6c69 6e65 2069 6d70 6c65  c pipeline imple
+0000cbf0: 6d65 6e74 6564 2069 6e20 6077 6764 2076  mented in `wgd v
+0000cc00: 3260 2075 7369 6e67 2074 6865 206d 6574  2` using the met
+0000cc10: 686f 6420 6f66 2070 6879 6c6f 6765 6e65  hod of phylogene
+0000cc20: 7469 6320 6461 7469 6e67 2e20 5468 6520  tic dating. The 
+0000cc30: 6661 6d69 6c69 6573 2075 7365 6420 696e  families used in
+0000cc40: 2074 6869 7320 7374 6570 2063 616e 2062   this step can b
+0000cc50: 6520 7072 6f64 7563 6564 2066 726f 6d20  e produced from 
+0000cc60: 6077 6764 2064 6d64 6020 616e 6420 6077  `wgd dmd` and `w
+0000cc70: 6764 2070 6561 6b60 2e20 4e6f 7465 2074  gd peak`. Note t
+0000cc80: 6861 7420 6865 7265 2077 6520 6f6e 6c79  hat here we only
+0000cc90: 2064 6973 6375 7373 2068 6f77 2074 6f20   discuss how to 
+0000cca0: 6461 7465 2057 4744 7320 7769 7468 2067  date WGDs with g
+0000ccb0: 656e 6f6d 6520 6173 7365 6d62 6c79 2c20  enome assembly, 
+0000ccc0: 696e 7374 6561 6420 6f66 2074 7261 6e73  instead of trans
+0000ccd0: 6372 6970 746f 6d65 2061 7373 656d 626c  criptome assembl
+0000cce0: 7920 2877 6869 6368 2077 696c 6c20 6265  y (which will be
+0000ccf0: 2064 6973 6375 7373 6564 2069 6e20 6120   discussed in a 
+0000cd00: 7365 7061 7261 7465 2073 6563 7469 6f6e  separate section
+0000cd10: 2068 6572 6575 6e64 6572 292e 2054 6865   hereunder). The
+0000cd20: 2061 7373 756d 7074 696f 6e20 7765 206d   assumption we m
+0000cd30: 6164 6520 6865 7265 2069 7320 7468 6174  ade here is that
+0000cd40: 206e 6f74 2061 6c6c 2061 6e63 686f 7220   not all anchor 
+0000cd50: 7061 6972 7320 2863 6f6c 6c69 6e65 6172  pairs (collinear
+0000cd60: 2064 7570 6c69 6361 7465 7329 2061 7265   duplicates) are
+0000cd70: 2073 7569 7461 626c 6520 666f 7220 7068   suitable for ph
+0000cd80: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
+0000cd90: 672c 2066 6f72 2069 6e73 7461 6e63 6520  g, for instance 
+0000cda0: 7468 6f73 6520 6661 7374 6c79 206f 7220  those fastly or 
+0000cdb0: 736c 6f77 6c79 2065 766f 6c76 696e 6720  slowly evolving 
+0000cdc0: 6765 6e65 2064 7570 6c69 6361 7465 732c  gene duplicates,
+0000cdd0: 2062 6563 6175 7365 2074 6865 7927 7265   because they're
+0000cde0: 2070 726f 6e65 2074 6f20 6769 7665 2062   prone to give b
+0000cdf0: 6961 7365 6420 6461 7469 6e67 2065 7374  iased dating est
+0000ce00: 696d 6174 696f 6e2e 2053 6f20 6173 2074  imation. So as t
+0000ce10: 6f20 7265 7472 6965 7665 2074 6865 2022  o retrieve the "
+0000ce20: 7265 6c69 6162 6c65 2220 616e 6368 6f72  reliable" anchor
+0000ce30: 2070 6169 7273 2c20 7765 2069 6d70 6c65   pairs, we imple
+0000ce40: 6d65 6e74 6564 2073 6f6d 6520 6d65 7468  mented some meth
+0000ce50: 6f64 7320 6f66 2069 6465 6e74 6966 7969  ods of identifyi
+0000ce60: 6e67 2063 7265 6469 6162 6c65 2061 6e63  ng crediable anc
+0000ce70: 686f 7220 7061 6972 7320 6261 7365 6420  hor pairs based 
+0000ce80: 6f6e 2074 6865 6972 204b 7320 7661 6c75  on their Ks valu
+0000ce90: 6573 2061 6e64 2f6f 7220 7265 7369 6469  es and/or residi
+0000cea0: 6e67 2063 6f6c 6c69 6e65 6172 2073 6567  ng collinear seg
+0000ceb0: 6d65 6e74 732e 2046 6f72 2061 2067 656e  ments. For a gen
+0000cec0: 6f6d 6520 7769 7468 2063 6c65 6172 2073  ome with clear s
+0000ced0: 6967 6e61 6c73 206f 6620 7075 7461 7469  ignals of putati
+0000cee0: 7665 2057 4744 732c 2073 7563 6820 6173  ve WGDs, such as
+0000cef0: 2074 6865 202a 4171 7569 6c65 6769 6120   the *Aquilegia 
+0000cf00: 636f 6572 756c 6561 2a20 696e 2074 6865  coerulea* in the
+0000cf10: 2065 7861 6d70 6c65 2c20 6120 6865 7572   example, a heur
+0000cf20: 6973 7469 6320 6d65 7468 6f64 2074 6861  istic method tha
+0000cf30: 7420 6170 706c 6965 7320 7468 6520 7072  t applies the pr
+0000cf40: 696e 6369 706c 6520 6f66 2068 6f77 205b  inciple of how [
+0000cf50: 6b73 7261 7465 735d 2868 7474 7073 3a2f  ksrates](https:/
+0000cf60: 2f67 6974 6875 622e 636f 6d2f 5649 422d  /github.com/VIB-
+0000cf70: 5053 422f 6b73 7261 7465 7329 2066 696e  PSB/ksrates) fin
+0000cf80: 6420 7468 6520 696e 6974 6961 6c20 7065  d the initial pe
+0000cf90: 616b 7320 616e 6420 7468 6569 7220 7061  aks and their pa
+0000cfa0: 7261 6d65 7465 7273 2077 6173 2069 6d70  rameters was imp
+0000cfb0: 6c65 6d65 6e74 6564 2074 6f20 6669 6e64  lemented to find
+0000cfc0: 2074 6865 2039 3525 2063 6f6e 6669 6465   the 95% confide
+0000cfd0: 6e63 6520 6c65 7665 6c20 6f66 2074 6865  nce level of the
+0000cfe0: 2061 7373 756d 6564 206c 6f67 6e6f 726d   assumed lognorm
+0000cff0: 616c 2064 6973 7472 6962 7574 696f 6e20  al distribution 
+0000d000: 6f66 2074 6865 2061 6e63 686f 7220 7061  of the anchor pa
+0000d010: 6972 202a 4b2a 3c73 7562 3e53 3c2f 7375  ir *K*<sub>S</su
+0000d020: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+0000d030: 696f 6e20 746f 2066 696c 7465 7220 616e  ion to filter an
+0000d040: 6368 6f72 2070 6169 7273 2077 6974 6820  chor pairs with 
+0000d050: 746f 6f20 6869 6768 206f 7220 6c6f 7720  too high or low 
+0000d060: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000d070: 6167 6573 2c20 7468 6520 6578 616d 706c  ages, the exampl
+0000d080: 6172 2063 6f6d 6d61 6e64 206f 6620 7768  ar command of wh
+0000d090: 6963 6820 6973 2073 686f 7765 6420 696e  ich is showed in
+0000d0a0: 2074 6865 2060 7767 6420 7065 616b 6020   the `wgd peak` 
+0000d0b0: 7365 6374 696f 6e20 6f66 205b 496c 6c75  section of [Illu
+0000d0c0: 7374 7261 7469 6f6e 5d28 6874 7470 733a  stration](https:
+0000d0d0: 2f2f 6769 7468 7562 2e63 6f6d 2f68 6563  //github.com/hec
+0000d0e0: 6865 2d70 7362 2f77 6764 3f74 6162 3d72  he-psb/wgd?tab=r
+0000d0f0: 6561 646d 652d 6f76 2d66 696c 6523 696c  eadme-ov-file#il
+0000d100: 6c75 7374 7261 7469 6f6e 292e 2049 6620  lustration). If 
+0000d110: 7468 6973 2068 6575 7269 7374 6963 206d  this heuristic m
+0000d120: 6574 686f 6420 6661 696c 6564 2074 6f20  ethod failed to 
+0000d130: 6769 7665 2072 6561 736f 6e61 626c 6520  give reasonable 
+0000d140: 7265 7375 6c74 732c 2077 6869 6368 2069  results, which i
+0000d150: 7320 7573 7561 6c6c 7920 6475 6520 746f  s usually due to
+0000d160: 2074 6865 2065 6666 6563 7420 6f66 206d   the effect of m
+0000d170: 756c 7469 706c 6520 6164 6a61 6365 6e74  ultiple adjacent
+0000d180: 2057 4744 7320 7468 6174 2062 6c75 7273   WGDs that blurs
+0000d190: 2074 6865 2070 6561 6b2d 6669 6e64 696e   the peak-findin
+0000d1a0: 6720 7072 6f63 6573 732c 2075 7365 7273  g process, users
+0000d1b0: 2063 616e 2074 7572 6e20 746f 2074 6865   can turn to the
+0000d1c0: 2063 6f6c 6c69 6e65 6172 2073 6567 6d65   collinear segme
+0000d1d0: 6e74 732d 6775 6964 6564 2061 6e63 686f  nts-guided ancho
+0000d1e0: 7220 7061 6972 2063 6c75 7374 6572 696e  r pair clusterin
+0000d1f0: 6720 696d 706c 656d 656e 7465 6420 696e  g implemented in
+0000d200: 2060 7767 6420 7632 602c 2069 6e20 7768   `wgd v2`, in wh
+0000d210: 6963 6820 6120 636f 6c6c 696e 6561 7220  ich a collinear 
+0000d220: 7365 676d 656e 742d 7769 7365 2047 4d4d  segment-wise GMM
+0000d230: 2063 6c75 7374 6572 696e 6720 7769 6c6c   clustering will
+0000d240: 2062 6520 6669 7273 7420 636f 6e64 7563   be first conduc
+0000d250: 7465 6420 6261 7365 6420 6f6e 2074 6865  ted based on the
+0000d260: 2022 736f 2d63 616c 6c65 6422 2073 6567   "so-called" seg
+0000d270: 6d65 6e74 202a 4b2a 3c73 7562 3e53 3c2f  ment *K*<sub>S</
+0000d280: 7375 623e 2061 6765 2072 6570 7265 7365  sub> age represe
+0000d290: 6e74 6564 2062 7920 7468 6520 6d65 6469  nted by the medi
+0000d2a0: 616e 202a 4b2a 3c73 7562 3e53 3c2f 7375  an *K*<sub>S</su
+0000d2b0: 623e 2061 6765 206f 6620 616c 6c20 7468  b> age of all th
+0000d2c0: 6520 7265 7369 6469 6e67 2073 796e 7465  e residing synte
+0000d2d0: 6c6f 6773 2028 6e6f 7465 2074 6861 7420  logs (note that 
+0000d2e0: 7468 6520 6765 6e65 2064 7570 6c69 6361  the gene duplica
+0000d2f0: 7465 2070 6169 7273 2061 646f 7074 6564  te pairs adopted
+0000d300: 2069 6e20 7468 6973 2073 7465 7020 6973   in this step is
+0000d310: 2066 726f 6d20 7468 6520 6669 6c65 2060   from the file `
+0000d320: 6d75 6c74 6970 6c69 636f 6e5f 7061 6972  multiplicon_pair
+0000d330: 732e 7478 7460 2077 6869 6368 2063 6f6e  s.txt` which con
+0000d340: 7461 696e 7320 7468 6520 6675 6c6c 2073  tains the full s
+0000d350: 6574 206f 6620 7379 6e74 656c 6f67 7329  et of syntelogs)
+0000d360: 2c20 696e 7374 6561 6420 6f66 2074 6865  , instead of the
+0000d370: 2073 6d61 6c6c 6572 2067 656e 6520 7365   smaller gene se
+0000d380: 7420 6f66 2061 6e63 686f 7220 7061 6972  t of anchor pair
+0000d390: 732e 2054 6865 2064 6973 7469 6e63 7469  s. The distincti
+0000d3a0: 6f6e 2062 6574 7765 656e 2061 6e63 686f  on between ancho
+0000d3b0: 7220 7061 6972 7320 616e 6420 7379 6e74  r pairs and synt
+0000d3c0: 656c 6f67 7320 6973 2074 6861 7420 7468  elogs is that th
+0000d3d0: 6520 6c61 7474 6572 2072 6566 6572 7320  e latter refers 
+0000d3e0: 746f 206d 756c 7469 706c 6520 7365 7473  to multiple sets
+0000d3f0: 206f 6620 6765 6e65 7320 6465 7269 7665   of genes derive
+0000d400: 6420 6672 6f6d 2074 6865 2073 616d 6520  d from the same 
+0000d410: 616e 6365 7374 7261 6c20 6765 6e6f 6d69  ancestral genomi
+0000d420: 6320 7265 6769 6f6e 2077 6869 6c65 2074  c region while t
+0000d430: 6865 2066 6f72 6d65 7220 696d 706c 6965  he former implie
+0000d440: 7320 7468 6520 6c61 7474 6572 2062 7574  s the latter but
+0000d450: 2072 6571 7569 7265 7320 6164 6469 7469   requires additi
+0000d460: 6f6e 616c 6c79 2074 6865 2063 6f6e 7365  onally the conse
+0000d470: 7276 6564 2067 656e 6520 6f72 6465 722c  rved gene order,
+0000d480: 2062 6f74 6820 6f66 2077 6869 6368 2061   both of which a
+0000d490: 7265 2c20 7769 7468 696e 2061 2067 656e  re, within a gen
+0000d4a0: 6f6d 652c 2061 7373 756d 6564 2074 6f20  ome, assumed to 
+0000d4b0: 6265 206f 7269 6769 6e61 7465 6420 6672  be originated fr
+0000d4c0: 6f6d 2074 6865 2064 7570 6c69 6361 7469  om the duplicati
+0000d4d0: 6f6e 206f 6620 6120 636f 6d6d 6f6e 2061  on of a common a
+0000d4e0: 6e63 6573 7472 616c 2067 656e 6f6d 6963  ncestral genomic
+0000d4f0: 2072 6567 696f 6e20 616e 6420 6173 2073   region and as s
+0000d500: 7563 6820 6465 656d 6564 2065 7669 6465  uch deemed evide
+0000d510: 6e63 6520 666f 7220 5747 442e 2054 6865  nce for WGD. The
+0000d520: 6e20 7468 6520 7379 6e74 656c 6f67 7320  n the syntelogs 
+0000d530: 7769 6c6c 2062 6520 6d61 7070 6564 2062  will be mapped b
+0000d540: 6163 6b20 6163 636f 7264 696e 6720 746f  ack according to
+0000d550: 2074 6865 2063 6c75 7374 6572 696e 6720   the clustering 
+0000d560: 7265 7375 6c74 7320 6f66 2074 6865 6972  results of their
+0000d570: 2061 6666 696c 6961 7465 6420 7365 676d   affiliated segm
+0000d580: 656e 7473 2e20 5369 6e63 6520 7468 6520  ents. Since the 
+0000d590: 4761 7573 7369 616e 2073 6861 7065 206f  Gaussian shape o
+0000d5a0: 6620 7468 6520 7365 676d 656e 7420 636c  f the segment cl
+0000d5b0: 7573 7465 7220 646f 6573 6e27 7420 6e65  uster doesn't ne
+0000d5c0: 6365 7373 6172 696c 7920 696d 6974 6174  cessarily imitat
+0000d5d0: 6520 7468 6520 7368 6170 6520 6f66 2074  e the shape of t
+0000d5e0: 6865 2072 6573 6964 696e 6720 7379 6e74  he residing synt
+0000d5f0: 656c 6f67 732c 2073 6f20 6173 2074 6f20  elogs, so as to 
+0000d600: 7265 7472 6965 7665 2074 6865 2022 7265  retrieve the "re
+0000d610: 6c69 6162 6c65 2220 6765 6e65 2073 6574  liable" gene set
+0000d620: 7320 666f 7220 7068 796c 6f67 656e 6574  s for phylogenet
+0000d630: 6963 2064 6174 696e 672c 2077 6520 6164  ic dating, we ad
+0000d640: 706f 7465 6420 7468 6520 2839 3525 2920  poted the (95%) 
+0000d650: 6869 6768 6573 7420 6465 6e73 6974 7920  highest density 
+0000d660: 7265 6769 6f6e 2028 4844 5229 206f 6620  region (HDR) of 
+0000d670: 7468 6520 7379 6e74 656c 6f67 202a 4b2a  the syntelog *K*
+0000d680: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000d690: 2064 6973 7472 6962 7574 696f 6e20 666f   distribution fo
+0000d6a0: 7220 7468 6520 7068 796c 6f67 656e 6574  r the phylogenet
+0000d6b0: 6963 2064 6174 696e 672c 2074 6865 2063  ic dating, the c
+0000d6c0: 616c 6375 6c61 7469 6f6e 206f 6620 7468  alculation of th
+0000d6d0: 6520 2839 3525 2920 4844 5220 696e 2074  e (95%) HDR in t
+0000d6e0: 6865 2066 756e 6374 696f 6e20 5b63 616c  he function [cal
+0000d6f0: 6375 6c61 7465 4850 445d 2868 7474 7073  culateHPD](https
+0000d700: 3a2f 2f67 6974 6875 622e 636f 6d2f 6865  ://github.com/he
+0000d710: 6368 652d 7073 622f 7767 642f 626c 6f62  che-psb/wgd/blob
+0000d720: 2f70 6879 6c6f 6461 7469 6e67 2f77 6764  /phylodating/wgd
+0000d730: 2f70 6561 6b2e 7079 234c 3137 3537 2920  /peak.py#L1757) 
+0000d740: 7365 656b 7320 7468 6520 7368 6f72 7465  seeks the shorte
+0000d750: 7374 202a 4b2a 3c73 7562 3e53 3c2f 7375  st *K*<sub>S</su
+0000d760: 623e 2072 616e 6765 2028 612c 6229 2077  b> range (a,b) w
+0000d770: 6869 6368 2073 6174 6973 6669 6573 2074  hich satisfies t
+0000d780: 6865 2072 6571 7569 7265 6d65 6e74 206f  he requirement o
+0000d790: 6620 7370 616e 6e69 6e67 206d 6f72 6520  f spanning more 
+0000d7a0: 7468 616e 2028 3935 2529 206f 6620 616c  than (95%) of al
+0000d7b0: 6c20 7468 6520 2a4b 2a3c 7375 623e 533c  l the *K*<sub>S<
+0000d7c0: 2f73 7562 3e20 7661 6c75 6573 2e20 4f6e  /sub> values. On
+0000d7d0: 2074 6865 2070 7265 6d69 7365 206f 6620   the premise of 
+0000d7e0: 6964 656e 7469 6669 6564 2061 6e63 686f  identified ancho
+0000d7f0: 7220 7061 6972 7320 286e 6f74 6520 7468  r pairs (note th
+0000d800: 6174 2074 6865 2073 796e 7465 6c6f 6773  at the syntelogs
+0000d810: 2061 7265 2061 6c73 6f20 7265 6665 7272   are also referr
+0000d820: 6564 2061 7320 616e 6368 6f72 2070 6169  ed as anchor pai
+0000d830: 7273 2068 6572 6561 6674 6572 292c 2077  rs hereafter), w
+0000d840: 6520 696d 706c 656d 656e 7465 6420 696e  e implemented in
+0000d850: 2074 6865 2070 726f 6772 616d 2060 7767   the program `wg
+0000d860: 6420 646d 6460 2074 6865 2073 6f2d 6361  d dmd` the so-ca
+0000d870: 6c6c 6564 2061 6e63 686f 722d 6177 6172  lled anchor-awar
+0000d880: 6520 6c6f 6361 6c20 4d52 4248 7320 6f72  e local MRBHs or
+0000d890: 206f 7274 686f 6772 6f75 7073 2c20 696e   orthogroups, in
+0000d8a0: 2077 6869 6368 2074 6865 206f 7269 6769   which the origi
+0000d8b0: 6e61 6c20 6c6f 6361 6c20 4d52 4248 7320  nal local MRBHs 
+0000d8c0: 6172 6520 6675 7274 6865 7220 6d65 7267  are further merg
+0000d8d0: 6564 2077 6974 6820 616e 6368 6f72 2070  ed with anchor p
+0000d8e0: 6169 7273 2073 7563 6820 7468 6174 2065  airs such that e
+0000d8f0: 6163 6820 6f72 7468 6f67 726f 7570 2063  ach orthogroup c
+0000d900: 6f6e 7461 696e 7320 7468 6520 616e 6368  ontains the anch
+0000d910: 6f72 2070 6169 7220 616e 6420 7468 6520  or pair and the 
+0000d920: 6f72 7468 6f6c 6f67 7565 732e 2057 6974  orthologues. Wit
+0000d930: 6820 7468 6973 206f 7274 686f 6772 6f75  h this orthogrou
+0000d940: 702c 2075 7365 7273 2074 6865 6e20 6e65  p, users then ne
+0000d950: 6564 206f 6e65 2073 7461 7274 696e 6720  ed one starting 
+0000d960: 7472 6565 2066 696c 6520 2861 7320 7368  tree file (as sh
+0000d970: 6f77 6e20 696e 2074 6865 205b 496c 6c75  own in the [Illu
+0000d980: 7374 7261 7469 6f6e 5d28 6874 7470 733a  stration](https:
+0000d990: 2f2f 6769 7468 7562 2e63 6f6d 2f68 6563  //github.com/hec
+0000d9a0: 6865 2d70 7362 2f77 6764 2f74 7265 652f  he-psb/wgd/tree/
+0000d9b0: 7068 796c 6f64 6174 696e 673f 7461 623d  phylodating?tab=
+0000d9c0: 7265 6164 6d65 2d6f 762d 6669 6c65 2369  readme-ov-file#i
+0000d9d0: 6c6c 7573 7472 6174 696f 6e29 2070 6172  llustration) par
+0000d9e0: 7429 2069 6e64 6963 6174 696e 6720 7468  t) indicating th
+0000d9f0: 6520 7472 6565 2074 6f70 6f6c 6f67 7920  e tree topology 
+0000da00: 616e 6420 666f 7373 696c 2063 616c 6962  and fossil calib
+0000da10: 7261 7469 6f6e 2069 6e66 6f72 6d61 7469  ration informati
+0000da20: 6f6e 2066 6f72 2074 6865 2066 696e 616c  on for the final
+0000da30: 2070 6879 6c6f 6765 6e65 7469 6320 6461   phylogenetic da
+0000da40: 7469 6e67 2075 7369 6e67 2074 6865 2070  ting using the p
+0000da50: 726f 6772 616d 2060 7767 6420 666f 6375  rogram `wgd focu
+0000da60: 7360 2e20 5573 6572 7320 6361 6e20 7365  s`. Users can se
+0000da70: 7420 7468 6520 666c 6167 206f 7074 696f  t the flag optio
+0000da80: 6e20 602d 2d70 726f 7463 6f63 6461 7469  n `--protcocdati
+0000da90: 6e67 6020 746f 206f 6e6c 7920 636f 6e64  ng` to only cond
+0000daa0: 7563 7420 7468 6520 6461 7469 6e67 206f  uct the dating o
+0000dab0: 6620 7468 6520 636f 6e63 6174 656e 6174  f the concatenat
+0000dac0: 696f 6e20 7072 6f74 6569 6e20 4d53 4120  ion protein MSA 
+0000dad0: 6f72 2074 6865 2066 6c61 6720 6f70 7469  or the flag opti
+0000dae0: 6f6e 2060 2d2d 7072 6f74 6461 7469 6e67  on `--protdating
+0000daf0: 6020 746f 206f 6e6c 7920 636f 6e64 7563  ` to only conduc
+0000db00: 7420 7468 6520 6461 7469 6e67 206f 6620  t the dating of 
+0000db10: 7468 6520 7072 6f74 6569 6e20 4d53 4173  the protein MSAs
+0000db20: 2c20 6e6f 7465 6420 7468 6174 2074 6865  , noted that the
+0000db30: 7365 2074 776f 206f 7074 696f 6e73 206f  se two options o
+0000db40: 6e6c 7920 776f 726b 2066 6f72 2074 6865  nly work for the
+0000db50: 2060 6d63 6d63 7472 6565 6020 6f70 7469   `mcmctree` opti
+0000db60: 6f6e 2073 6f20 6661 722e 2054 6865 2066  on so far. The f
+0000db70: 6c61 6720 6f70 7469 6f6e 2060 2d2d 7061  lag option `--pa
+0000db80: 7274 6974 696f 6e60 2063 616e 2062 6520  rtition` can be 
+0000db90: 7365 7420 746f 2070 6572 666f 726d 2060  set to perform `
+0000dba0: 6d63 6d63 7472 6565 6020 616e 616c 7973  mcmctree` analys
+0000dbb0: 6973 2075 7369 6e67 2074 6865 2070 6172  is using the par
+0000dbc0: 7469 7469 6f6e 6564 2064 6174 6120 2869  titioned data (i
+0000dbd0: 2e65 2e2c 2031 7374 2c20 326e 6420 616e  .e., 1st, 2nd an
+0000dbe0: 6420 3372 6420 706f 7369 7469 6f6e 206f  d 3rd position o
+0000dbf0: 6620 636f 646f 6e29 2069 6e73 7465 6164  f codon) instead
+0000dc00: 206f 6620 7573 696e 6720 7468 6520 636f   of using the co
+0000dc10: 646f 6e20 6173 2061 2077 686f 6c65 2e20  don as a whole. 
+0000dc20: 5468 6520 6f70 7469 6f6e 2060 2d2d 6161  The option `--aa
+0000dc30: 6d6f 6465 6c60 2063 616e 2062 6520 7365  model` can be se
+0000dc40: 7420 746f 2064 6574 6572 6d69 6e65 2074  t to determine t
+0000dc50: 6865 2061 6d69 6e6f 2061 6369 6420 6d6f  he amino acid mo
+0000dc60: 6465 6c20 6170 706c 6965 6420 696e 2060  del applied in `
+0000dc70: 6d63 6d63 7472 6565 6020 616e 616c 7973  mcmctree` analys
+0000dc80: 6973 2e20 5468 6520 6f70 7469 6f6e 2060  is. The option `
+0000dc90: 2d64 7360 2063 616e 2062 6520 7573 6564  -ds` can be used
+0000dca0: 2074 6f20 7365 7420 7061 7261 6d65 7465   to set paramete
+0000dcb0: 7273 2066 6f72 2074 6865 206d 6f6c 6563  rs for the molec
+0000dcc0: 756c 6172 2064 6174 696e 6720 7072 6f67  ular dating prog
+0000dcd0: 7261 6d2e 2054 6865 2070 6172 616c 6c65  ram. The paralle
+0000dce0: 6c20 7468 7265 6164 7320 6865 7265 2061  l threads here a
+0000dcf0: 7265 2074 6f20 7061 7261 6c6c 656c 697a  re to paralleliz
+0000dd00: 6520 7468 6520 616e 616c 7973 6973 2066  e the analysis f
+0000dd10: 6f72 2065 6163 6820 6765 6e65 2066 616d  or each gene fam
+0000dd20: 696c 7920 616e 6420 7468 7573 2073 7567  ily and thus sug
+0000dd30: 6765 7374 6564 2074 6f20 6265 2073 6574  gested to be set
+0000dd40: 2061 7320 6d75 6368 2061 7320 7468 6520   as much as the 
+0000dd50: 6e75 6d62 6572 206f 6620 6765 6e65 2066  number of gene f
+0000dd60: 616d 696c 6965 732e 0a0a 2a2a 4120 7375  amilies...**A su
+0000dd70: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
+0000dd80: 2072 756e 2063 616e 2062 6520 7769 7468   run can be with
+0000dd90: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+0000dda0: 6f77 2e2a 2a0a 0a60 6060 0a77 6764 2066  ow.**..```.wgd f
+0000ddb0: 6f63 7573 2066 616d 696c 6965 7320 7365  ocus families se
+0000ddc0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
+0000ddd0: 3220 7365 7175 656e 6365 3320 2d64 206d  2 sequence3 -d m
+0000dde0: 636d 6374 7265 6520 2d73 7020 7370 6461  cmctree -sp spda
+0000ddf0: 7461 0a77 6764 2066 6f63 7573 2066 616d  ta.wgd focus fam
+0000de00: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
+0000de10: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
+0000de20: 6365 3320 2d64 2072 3873 202d 7370 2073  ce3 -d r8s -sp s
+0000de30: 7064 6174 6120 2d2d 6e73 6974 6573 206e  pdata --nsites n
+0000de40: 7369 7465 696e 666f 0a77 6764 2066 6f63  siteinfo.wgd foc
+0000de50: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
+0000de60: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
+0000de70: 7365 7175 656e 6365 3320 2d64 2062 6561  sequence3 -d bea
+0000de80: 7374 202d 2d66 6f73 7369 6c20 666f 7373  st --fossil foss
+0000de90: 696c 696e 666f 202d 2d72 6f6f 7468 6569  ilinfo --roothei
+0000dea0: 6768 7420 726f 6f74 6865 6967 6874 696e  ght rootheightin
+0000deb0: 666f 202d 2d63 6861 696e 7365 7420 6368  fo --chainset ch
+0000dec0: 6169 6e73 6574 696e 666f 202d 2d62 6561  ainsetinfo --bea
+0000ded0: 7374 6c67 6a61 7220 2450 4154 480a 6060  stlgjar $PATH.``
+0000dee0: 600a 0a23 2323 2077 6764 206b 7364 0a0a  `..### wgd ksd..
+0000def0: 2a2a 5468 6520 636f 6e73 7472 7563 7469  **The constructi
+0000df00: 6f6e 206f 6620 7768 6f6c 6520 7061 7261  on of whole para
+0000df10: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
+0000df20: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+0000df30: 7574 696f 6e2a 2a0a 6060 600a 7767 6420  ution**.```.wgd 
+0000df40: 6b73 6420 6661 6d69 6c69 6573 2073 6571  ksd families seq
+0000df50: 7565 6e63 6520 282d 6f20 7767 645f 6b73  uence (-o wgd_ks
+0000df60: 6420 2d74 2077 6764 5f6b 7364 5f74 6d70  d -t wgd_ksd_tmp
+0000df70: 202d 2d6e 7468 7265 6164 7320 3420 2d2d   --nthreads 4 --
+0000df80: 746f 5f73 746f 7020 2d2d 6364 7320 2d2d  to_stop --cds --
+0000df90: 7061 6972 7769 7365 202d 2d73 7472 6970  pairwise --strip
+0000dfa0: 5f67 6170 7320 2d2d 616c 6967 6e65 7220  _gaps --aligner 
+0000dfb0: 6d61 6666 7420 2d2d 7472 6565 5f6d 6574  mafft --tree_met
+0000dfc0: 686f 6420 6661 7374 7472 6565 202d 2d6e  hod fasttree --n
+0000dfd0: 6f64 655f 6176 6572 6167 6529 0a60 6060  ode_average).```
+0000dfe0: 0a0a 5468 6520 7072 6f67 7261 6d20 6077  ..The program `w
+0000dff0: 6764 206b 7364 602c 2061 7320 696d 7069  gd ksd`, as impi
+0000e000: 6564 2062 7920 6974 7320 6e61 6d65 2c20  ed by its name, 
+0000e010: 6973 2066 6f72 2074 6865 2063 6f6e 7374  is for the const
+0000e020: 7275 6374 696f 6e20 6f66 202a 4b2a 3c73  ruction of *K*<s
+0000e030: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+0000e040: 6973 7472 6962 7574 696f 6e2e 2045 7863  istribution. Exc
+0000e050: 6570 7420 666f 7220 7468 6520 6166 6f72  ept for the afor
+0000e060: 656d 656e 7469 6f6e 6564 2070 6172 616d  ementioned param
+0000e070: 6574 6572 7320 7375 6368 2061 7320 602d  eters such as `-
+0000e080: 2d74 6f5f 7374 6f70 6020 616e 6420 602d  -to_stop` and `-
+0000e090: 2d63 6473 602c 2074 6865 7265 2061 7265  -cds`, there are
+0000e0a0: 2073 6f6d 6520 696d 706f 7274 616e 7420   some important 
+0000e0b0: 7061 7261 6d65 7465 7273 2074 6861 7420  parameters that 
+0000e0c0: 6861 7665 2063 7275 6369 616c 2069 6d70  have crucial imp
+0000e0d0: 6163 7420 6f6e 2074 6865 202a 4b2a 3c73  act on the *K*<s
+0000e0e0: 7562 3e53 3c2f 7375 623e 2065 7374 696d  ub>S</sub> estim
+0000e0f0: 6174 696f 6e2e 2054 6865 206f 7074 696f  ation. The optio
+0000e100: 6e20 602d 2d70 6169 7277 6973 6560 2069  n `--pairwise` i
+0000e110: 7320 6120 7665 7279 2069 6d70 6f72 7461  s a very importa
+0000e120: 6e74 2070 6172 616d 6574 6572 2066 6f72  nt parameter for
+0000e130: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+0000e140: 7375 623e 2065 7374 696d 6174 696f 6e2c  sub> estimation,
+0000e150: 2077 6974 6820 7768 6963 6820 7468 6520   with which the 
+0000e160: 6043 4f44 454d 4c60 2077 696c 6c20 6361  `CODEML` will ca
+0000e170: 6c63 756c 6174 6520 7468 6520 2a4b 2a3c  lculate the *K*<
+0000e180: 7375 623e 533c 2f73 7562 3e20 666f 7220  sub>S</sub> for 
+0000e190: 6561 6368 2067 656e 6520 7061 6972 2073  each gene pair s
+0000e1a0: 6570 6172 6174 656c 7920 6261 7365 6420  eparately based 
+0000e1b0: 6f6e 2074 6865 2061 6c69 676e 6d65 6e74  on the alignment
+0000e1c0: 206f 6620 6f6e 6c79 2074 6865 7365 2074   of only these t
+0000e1d0: 776f 2067 656e 6573 2069 6e73 7465 6164  wo genes instead
+0000e1e0: 206f 6620 7468 6520 7768 6f6c 6520 616c   of the whole al
+0000e1f0: 6967 6e6d 656e 7420 6f66 2074 6865 2066  ignment of the f
+0000e200: 616d 696c 792c 2073 7563 6820 7468 6174  amily, such that
+0000e210: 206c 6573 7320 6761 7073 2061 7265 2065   less gaps are e
+0000e220: 7870 6563 7465 6420 616e 6420 7468 7573  xpected and thus
+0000e230: 2074 6865 2061 6c69 676e 6d65 6e74 2069   the alignment i
+0000e240: 6e20 7468 6520 636f 6e73 6964 6572 6174  n the considerat
+0000e250: 696f 6e20 6f66 2060 434f 4445 4d4c 6020  ion of `CODEML` 
+0000e260: 7769 6c6c 2062 6520 6c6f 6e67 6572 2028  will be longer (
+0000e270: 6265 6361 7573 6520 6043 4f44 454d 4c60  because `CODEML`
+0000e280: 2077 696c 6c20 6175 746f 6d61 7465 6c79   will automately
+0000e290: 2073 6b69 7020 6576 6572 7920 636f 6c75   skip every colu
+0000e2a0: 6d6e 2077 6974 6820 6761 702c 2072 6567  mn with gap, reg
+0000e2b0: 6172 646c 6573 7320 6f66 2077 6865 7468  ardless of wheth
+0000e2c0: 6572 2069 7427 7320 616e 206f 7665 7261  er it's an overa
+0000e2d0: 6c6c 206f 7220 7061 7274 6961 6c20 6761  ll or partial ga
+0000e2e0: 7029 2c20 7769 7468 6f75 7420 7768 6963  p), without whic
+0000e2f0: 6820 7468 6520 6043 4f44 454d 4c60 2077  h the `CODEML` w
+0000e300: 696c 6c20 6361 6c63 756c 6174 6520 7468  ill calculate th
+0000e310: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+0000e320: 3e20 6261 7365 6420 6f6e 2074 6865 2077  > based on the w
+0000e330: 686f 6c65 2061 6c69 676e 6d65 6e74 206f  hole alignment o
+0000e340: 6620 7468 6520 6661 6d69 6c79 2c20 7768  f the family, wh
+0000e350: 6963 6820 6d69 6768 7420 6861 7665 206e  ich might have n
+0000e360: 6f20 2a4b 2a3c 7375 623e 533c 2f73 7562  o *K*<sub>S</sub
+0000e370: 3e20 7265 7375 6c74 2061 7420 616c 6c20  > result at all 
+0000e380: 6966 2074 6865 2073 7472 6970 7065 6420  if the stripped 
+0000e390: 616c 6967 6e6d 656e 7420 6c65 6e67 7468  alignment length
+0000e3a0: 2028 7265 6d6f 7669 6e67 2061 6c6c 2067   (removing all g
+0000e3b0: 6170 2d63 6f6e 7461 696e 696e 6720 636f  ap-containing co
+0000e3c0: 6c75 6d6e 7329 2077 6173 207a 6572 6f2c  lumns) was zero,
+0000e3d0: 2061 2063 6175 7365 206f 6620 6469 6666   a cause of diff
+0000e3e0: 6572 656e 7420 6e75 6d62 6572 206f 6620  erent number of 
+0000e3f0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000e400: 6573 7469 6d61 7465 7320 6265 7477 6565  estimates betwee
+0000e410: 6e20 2270 6169 7277 6973 6520 6d6f 6465  n "pairwise mode
+0000e420: 2220 616e 6420 226e 6f6e 2d70 6169 7277  " and "non-pairw
+0000e430: 6973 6520 6d6f 6465 222e 2049 7427 7320  ise mode". It's 
+0000e440: 6469 6666 6963 756c 7420 746f 2073 6179  difficult to say
+0000e450: 2077 6869 6368 206d 6f64 6520 6973 206d   which mode is m
+0000e460: 6f72 6520 6964 6561 6c2c 2061 6c74 686f  ore ideal, altho
+0000e470: 7567 6820 7468 6520 226e 6f6e 2d70 6169  ugh the "non-pai
+0000e480: 7277 6973 6520 6d6f 6465 2220 2864 6566  rwise mode" (def
+0000e490: 6175 6c74 2073 6574 7469 6e67 2920 7768  ault setting) wh
+0000e4a0: 6963 6820 7275 6e73 206f 6e20 7468 6520  ich runs on the 
+0000e4b0: 7768 6f6c 6520 616c 6967 6e6d 656e 7420  whole alignment 
+0000e4c0: 696e 7374 6561 6420 6f66 2061 206c 6f63  instead of a loc
+0000e4d0: 616c 2061 6c69 676e 6d65 6e74 2c20 6d69  al alignment, mi
+0000e4e0: 6768 7420 6265 206d 6f72 6520 6269 6f6c  ght be more biol
+0000e4f0: 6f67 6963 616c 6c79 2063 6f6e 7365 7276  ogically conserv
+0000e500: 6564 2069 6e20 7468 6174 2069 7420 6173  ed in that it as
+0000e510: 7375 7265 7320 7468 6520 6576 6f6c 7574  sures the evolut
+0000e520: 696f 6e20 6f66 2065 6163 6820 636f 6c75  ion of each colu
+0000e530: 6d6e 2074 6f20 6265 2073 7461 7274 6564  mn to be started
+0000e540: 2066 726f 6d20 7468 6520 726f 6f74 206f   from the root o
+0000e550: 6620 7468 6520 6661 6d69 6c79 2061 6e64  f the family and
+0000e560: 2061 6c6c 2074 6865 2067 656e 6520 6475   all the gene du
+0000e570: 706c 6963 6174 6573 2061 7265 2074 616b  plicates are tak
+0000e580: 656e 2069 6e74 6f20 6163 636f 756e 7420  en into account 
+0000e590: 696e 2074 6865 202a 4b2a 3c73 7562 3e53  in the *K*<sub>S
+0000e5a0: 3c2f 7375 623e 2065 7374 696d 6174 696f  </sub> estimatio
+0000e5b0: 6e20 7072 6f63 6573 732e 2054 6865 206f  n process. The o
+0000e5c0: 7074 696f 6e20 602d 2d73 7472 6970 5f67  ption `--strip_g
+0000e5d0: 6170 7360 2063 616e 2072 656d 6f76 6520  aps` can remove 
+0000e5e0: 616c 6c20 7468 6520 6761 702d 636f 6e74  all the gap-cont
+0000e5f0: 6169 6e69 6e67 2063 6f6c 756d 6e73 2c20  aining columns, 
+0000e600: 7769 7468 206f 7220 7769 7468 6f75 7420  with or without 
+0000e610: 7768 6963 6820 7468 6520 7265 7375 6c74  which the result
+0000e620: 206f 6620 226e 6f6e 2d70 6169 7277 6973   of "non-pairwis
+0000e630: 6520 6d6f 6465 2220 776f 6e27 7420 6265  e mode" won't be
+0000e640: 2061 6666 6563 7465 642c 2077 6869 6c65   affected, while
+0000e650: 2077 6974 6820 7768 6963 6820 7468 6520   with which the 
+0000e660: 7265 7375 6c74 206f 6620 2270 6169 7277  result of "pairw
+0000e670: 6973 6520 6d6f 6465 2220 7769 6c6c 2062  ise mode" will b
+0000e680: 6520 616c 7465 7265 642e 2054 6865 206f  e altered. The o
+0000e690: 7074 696f 6e20 602d 2d61 6c69 676e 6572  ption `--aligner
+0000e6a0: 6020 616e 6420 602d 2d61 6c6e 5f6f 7074  ` and `--aln_opt
+0000e6b0: 696f 6e73 6020 7768 6963 6820 6465 6369  ions` which deci
+0000e6c0: 6465 2077 6869 6368 2061 6c69 676e 6d65  de which alignme
+0000e6d0: 6e74 2070 726f 6772 616d 2074 6f20 6265  nt program to be
+0000e6e0: 2075 7365 6420 616e 6420 7768 6963 6820   used and which 
+0000e6f0: 7061 7261 6d65 7465 7220 746f 2062 6520  parameter to be 
+0000e700: 7365 742c 2077 696c 6c20 6861 7665 2069  set, will have i
+0000e710: 6d70 6163 7420 6f6e 2074 6865 202a 4b2a  mpact on the *K*
+0000e720: 3c73 7562 3e53 3c2f 7375 623e 2072 6573  <sub>S</sub> res
+0000e730: 756c 7473 2c20 6e6f 7465 6420 7468 6174  ults, noted that
+0000e740: 2074 6865 2064 6566 6175 6c74 2070 726f   the default pro
+0000e750: 6772 616d 2069 7320 606d 6166 6674 6020  gram is `mafft` 
+0000e760: 616e 6420 7468 6520 7061 7261 6d65 7465  and the paramete
+0000e770: 7220 6973 2060 2d2d 6175 746f 602e 2054  r is `--auto`. T
+0000e780: 6865 206f 7074 696f 6e20 602d 2d74 7265  he option `--tre
+0000e790: 655f 6d65 7468 6f64 6020 616e 6420 602d  e_method` and `-
+0000e7a0: 2d74 7265 655f 6f70 7469 6f6e 7360 2064  -tree_options` d
+0000e7b0: 6563 6964 6520 7768 6963 6820 6765 6e65  ecide which gene
+0000e7c0: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
+0000e7d0: 7072 6f67 7261 6d20 746f 2062 6520 7573  program to be us
+0000e7e0: 6564 2061 6e64 2077 6869 6368 2070 6172  ed and which par
+0000e7f0: 616d 6574 6572 2074 6f20 6265 2073 6574  ameter to be set
+0000e800: 2c20 776f 6e27 7420 6166 6665 6374 2074  , won't affect t
+0000e810: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+0000e820: 623e 2065 7374 696d 6174 696f 6e20 6974  b> estimation it
+0000e830: 7365 6c66 2062 7574 2074 6865 2072 6573  self but the res
+0000e840: 756c 7420 6f66 2064 652d 7265 6475 6e64  ult of de-redund
+0000e850: 616e 6379 2c20 6e6f 7465 6420 7468 6174  ancy, noted that
+0000e860: 2077 6520 696d 706c 656d 656e 7465 6420   we implemented 
+0000e870: 6120 6275 696c 742d 696e 2067 656e 6520  a built-in gene 
+0000e880: 7472 6565 2069 6e66 6572 656e 6365 206d  tree inference m
+0000e890: 6574 686f 6420 6261 7365 6420 6f6e 2074  ethod based on t
+0000e8a0: 6865 2041 7665 7261 6765 204c 696e 6b61  he Average Linka
+0000e8b0: 6765 2043 6c75 7374 6572 696e 6720 2841  ge Clustering (A
+0000e8c0: 4c43 2920 2874 6875 7320 6120 6469 7374  LC) (thus a dist
+0000e8d0: 616e 6365 2d62 6173 6564 2074 7265 6529  ance-based tree)
+0000e8e0: 2077 6974 6820 7468 6520 602d 2d74 7265   with the `--tre
+0000e8f0: 655f 6d65 7468 6f64 6020 7365 7420 6173  e_method` set as
+0000e900: 2022 636c 7573 7465 7222 2e20 5477 6f20   "cluster". Two 
+0000e910: 6d65 7468 6f64 7320 6f66 2064 652d 7265  methods of de-re
+0000e920: 6475 6e64 616e 6379 2061 7265 2069 6d70  dundancy are imp
+0000e930: 6c65 6d65 6e74 6564 2069 6e20 6077 6764  lemented in `wgd
+0000e940: 2076 3260 2c20 6e61 6d65 6c79 206e 6f64   v2`, namely nod
+0000e950: 652d 7765 6967 6874 6564 2061 6e64 206e  e-weighted and n
+0000e960: 6f64 652d 6176 6572 6167 6564 206d 6574  ode-averaged met
+0000e970: 686f 6473 2e20 5468 6520 6e6f 6465 2d77  hods. The node-w
+0000e980: 6569 6768 7465 6420 6d65 7468 6f64 2061  eighted method a
+0000e990: 6368 6965 7665 7320 7468 6520 6465 2d72  chieves the de-r
+0000e9a0: 6564 756e 6461 6e63 7920 7669 6120 7765  edundancy via we
+0000e9b0: 6967 6869 6e67 2074 6865 202a 4b2a 3c73  ighing the *K*<s
+0000e9c0: 7562 3e53 3c2f 7375 623e 2076 616c 7565  ub>S</sub> value
+0000e9d0: 2061 7373 6f63 6961 7465 6420 7769 7468   associated with
+0000e9e0: 2065 6163 6820 6765 6e65 2070 6169 7220   each gene pair 
+0000e9f0: 7375 6368 2074 6861 7420 7468 6520 7765  such that the we
+0000ea00: 6967 6874 206f 6620 6120 7369 6e67 6c65  ight of a single
+0000ea10: 2064 7570 6c69 6361 7469 6f6e 2065 7665   duplication eve
+0000ea20: 6e74 2073 756d 7320 7570 2074 6f20 3120  nt sums up to 1 
+0000ea30: 286e 6f74 6564 2074 6861 7420 7468 6520  (noted that the 
+0000ea40: 6e75 6d62 6572 206f 6620 2a4b 2a3c 7375  number of *K*<su
+0000ea50: 623e 533c 2f73 7562 3e20 6573 7469 6d61  b>S</sub> estima
+0000ea60: 7469 6f6e 2072 656d 6169 6e73 2074 6865  tion remains the
+0000ea70: 2073 616d 6529 2077 6869 6c65 2074 6865   same) while the
+0000ea80: 206e 6f64 652d 6176 6572 6167 6564 206d   node-averaged m
+0000ea90: 6574 686f 6420 7265 616c 697a 6573 2074  ethod realizes t
+0000eaa0: 6865 2064 652d 7265 6475 6e64 616e 6379  he de-redundancy
+0000eab0: 2076 6961 2063 616c 6375 6c61 7469 6e67   via calculating
+0000eac0: 2070 6572 2067 656e 6520 7472 6565 206e   per gene tree n
+0000ead0: 6f64 6520 6f6e 6520 6176 6572 6167 6564  ode one averaged
+0000eae0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000eaf0: 2076 616c 7565 2074 6f20 7265 7072 6573   value to repres
+0000eb00: 656e 7420 7468 6520 6167 6520 6f66 2065  ent the age of e
+0000eb10: 6163 6820 6475 706c 6963 6174 696f 6e20  ach duplication 
+0000eb20: 6576 656e 742e 2054 6865 206f 7074 696f  event. The optio
+0000eb30: 6e20 602d 2d6e 6f64 655f 6176 6572 6167  n `--node_averag
+0000eb40: 6560 2063 616e 2062 6520 7365 7420 746f  e` can be set to
+0000eb50: 2063 686f 6f73 6520 7468 6520 6e6f 6465   choose the node
+0000eb60: 2d61 7665 7261 6765 6420 7761 7920 6f66  -averaged way of
+0000eb70: 2064 652d 7265 6475 6e64 616e 6379 2e20   de-redundancy. 
+0000eb80: 4469 6666 6572 656e 7420 6d65 7468 6f64  Different method
+0000eb90: 7320 6f66 2064 652d 7265 6475 6e64 616e  s of de-redundan
+0000eba0: 6379 2077 696c 6c20 6861 7665 2069 6d70  cy will have imp
+0000ebb0: 6163 7420 6f6e 2074 6865 2064 6574 6563  act on the detec
+0000ebc0: 7469 6f6e 206f 6620 5747 4420 7369 676e  tion of WGD sign
+0000ebd0: 616c 732c 2077 6869 6368 2068 6173 2062  als, which has b
+0000ebe0: 6565 6e20 696e 7665 7374 6967 6174 6564  een investigated
+0000ebf0: 2069 6e20 7468 6973 205b 6c69 7465 7261   in this [litera
+0000ec00: 7475 7265 5d28 6874 7470 733a 2f2f 646f  ture](https://do
+0000ec10: 692e 6f72 672f 3130 2e31 3039 332f 6762  i.org/10.1093/gb
+0000ec20: 652f 6576 7932 3030 292e 2054 6865 2070  e/evy200). The p
+0000ec30: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
+0000ec40: 6865 7265 2061 7265 2074 6f20 7061 7261  here are to para
+0000ec50: 6c6c 656c 697a 6520 7468 6520 616e 616c  llelize the anal
+0000ec60: 7973 6973 2066 6f72 2065 6163 6820 6765  ysis for each ge
+0000ec70: 6e65 2066 616d 696c 7920 616e 6420 7468  ne family and th
+0000ec80: 7573 2073 7567 6765 7374 6564 2074 6f20  us suggested to 
+0000ec90: 6265 2073 6574 2061 7320 6d75 6368 2061  be set as much a
+0000eca0: 7320 7468 6520 6e75 6d62 6572 206f 6620  s the number of 
+0000ecb0: 6765 6e65 2066 616d 696c 6965 732e 0a0a  gene families...
+0000ecc0: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
+0000ecd0: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
+0000ece0: 7365 2063 6f6d 6d61 6e64 2061 7320 6265  se command as be
+0000ecf0: 6c6f 772a 2a0a 0a60 6060 0a77 6764 206b  low**..```.wgd k
+0000ed00: 7364 2066 616d 696c 6965 7320 7365 7175  sd families sequ
+0000ed10: 656e 6365 0a60 6060 0a0a 2a2a 5468 6520  ence.```..**The 
+0000ed20: 636f 6e73 7472 7563 7469 6f6e 206f 6620  construction of 
+0000ed30: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
+0000ed40: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+0000ed50: 6469 7374 7269 6275 7469 6f6e 2a2a 0a60  distribution**.`
+0000ed60: 6060 0a77 6764 206b 7364 2066 616d 696c  ``.wgd ksd famil
+0000ed70: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+0000ed80: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+0000ed90: 3320 282d 2d72 6577 6569 6768 7429 0a60  3 (--reweight).`
+0000eda0: 6060 0a0a 4672 6f6d 2070 6172 616c 6f67  ``..From paralog
+0000edb0: 6f75 7320 746f 206f 7274 686f 6c6f 676f  ous to orthologo
+0000edc0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+0000edd0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+0000ede0: 696f 6e2c 2075 7365 7273 206f 6e6c 7920  ion, users only 
+0000edf0: 6e65 6564 2074 6f20 7072 6f76 6964 6520  need to provide 
+0000ee00: 6d6f 7265 2063 6473 2066 696c 6573 2e20  more cds files. 
+0000ee10: 4e6f 7465 2074 6861 7420 7769 7468 206f  Note that with o
+0000ee20: 7274 686f 6c6f 676f 7573 2067 656e 6520  rthologous gene 
+0000ee30: 6661 6d69 6c69 6573 2074 6865 2077 6569  families the wei
+0000ee40: 6768 7469 6e67 206d 6574 686f 6420 6361  ghting method ca
+0000ee50: 6e20 6265 2073 6574 2074 6f20 6265 2063  n be set to be c
+0000ee60: 616c 6375 6c61 7465 6420 7065 7220 7370  alculated per sp
+0000ee70: 6563 6965 7320 7061 6972 2069 6e73 7465  ecies pair inste
+0000ee80: 6164 206f 6620 636f 6e73 6964 6572 696e  ad of considerin
+0000ee90: 6720 7468 6520 7768 6f6c 6520 6661 6d69  g the whole fami
+0000eea0: 6c79 2062 6563 6175 7365 2077 6865 6e20  ly because when 
+0000eeb0: 706c 6f74 7469 6e67 206f 7274 686f 6c6f  plotting ortholo
+0000eec0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+0000eed0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+0000eee0: 7574 696f 6e20 6265 7477 6565 6e20 7477  ution between tw
+0000eef0: 6f20 7370 6563 6965 7320 7468 6520 7765  o species the we
+0000ef00: 6967 6874 2063 616c 6375 6c61 7465 6420  ight calculated 
+0000ef10: 6672 6f6d 2074 6869 7320 7370 6563 6966  from this specif
+0000ef20: 6963 2073 7065 6369 6573 2070 6169 7220  ic species pair 
+0000ef30: 7368 6f75 6c64 2062 6520 636f 6e73 6572  should be conser
+0000ef40: 7665 6420 7768 696c 6520 7468 6520 6f6e  ved while the on
+0000ef50: 6520 6361 6c63 756c 6174 6564 2066 726f  e calculated fro
+0000ef60: 6d20 7468 6520 7768 6f6c 6520 6661 6d69  m the whole fami
+0000ef70: 6c79 2077 696c 6c20 7661 7279 2077 6974  ly will vary wit
+0000ef80: 6820 7468 6520 6e75 6d62 6572 206f 6620  h the number of 
+0000ef90: 7370 6563 6965 732e 2054 6f20 696e 6974  species. To init
+0000efa0: 6961 7465 2074 6865 2077 6569 6768 7469  iate the weighti
+0000efb0: 6e67 2070 6572 2073 7065 6369 6573 2070  ng per species p
+0000efc0: 6169 722c 2074 6865 206f 7074 696f 6e20  air, the option 
+0000efd0: 602d 2d72 6577 6569 6768 7460 2063 616e  `--reweight` can
+0000efe0: 2062 6520 7365 742e 0a0a 2a2a 4120 7375   be set...**A su
+0000eff0: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
+0000f000: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
+0000f010: 6d61 6e64 2061 7320 6265 6c6f 772a 2a0a  mand as below**.
+0000f020: 0a60 6060 0a77 6764 206b 7364 2066 616d  .```.wgd ksd fam
+0000f030: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
+0000f040: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
+0000f050: 6365 330a 6060 600a 0a2a 2a54 6865 2063  ce3.```..**The c
+0000f060: 6f6e 7374 7275 6374 696f 6e20 6f66 202a  onstruction of *
+0000f070: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+0000f080: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
+0000f090: 7769 7468 2072 6174 6520 636f 7272 6563  with rate correc
+0000f0a0: 7469 6f6e 2a2a 0a60 6060 0a77 6764 206b  tion**.```.wgd k
+0000f0b0: 7364 2066 616d 696c 6965 7320 7365 7175  sd families sequ
+0000f0c0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
+0000f0d0: 7365 7175 656e 6365 3320 2d2d 666f 6375  sequence3 --focu
+0000f0e0: 7332 616c 6c20 7365 7175 656e 6365 3120  s2all sequence1 
+0000f0f0: 2d73 7020 7370 6461 7461 202d 2d65 7874  -sp spdata --ext
+0000f100: 7261 7061 7261 6e6f 6d65 6b73 2070 6172  raparanomeks par
+0000f110: 616e 6f6d 654b 7364 6174 6120 282d 2d70  anomeKsdata (--p
+0000f120: 6c6f 7465 6c6d 6d20 2d2d 706c 6f74 6170  lotelmm --plotap
+0000f130: 676d 6d20 2d2d 616e 6368 6f72 706f 696e  gmm --anchorpoin
+0000f140: 7473 2061 7064 6174 6120 2d2d 7265 7765  ts apdata --rewe
+0000f150: 6967 6874 202d 2d6f 6e6c 7972 6f6f 746f  ight --onlyrooto
+0000f160: 7574 290a 6060 600a 0a49 6e73 7069 7265  ut).```..Inspire
+0000f170: 6420 6279 2074 6865 2072 6174 6520 636f  d by the rate co
+0000f180: 7272 6563 7469 6f6e 2061 6c67 6f72 6974  rrection algorit
+0000f190: 686d 2069 6e20 5b6b 7372 6174 6573 5d28  hm in [ksrates](
+0000f1a0: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
+0000f1b0: 6f6d 2f56 4942 2d50 5342 2f6b 7372 6174  om/VIB-PSB/ksrat
+0000f1c0: 6573 292c 2077 6520 696d 706c 656d 656e  es), we implemen
+0000f1d0: 7465 6420 7468 6520 7261 7465 2063 6f72  ted the rate cor
+0000f1e0: 7265 6374 696f 6e20 616e 616c 7973 6973  rection analysis
+0000f1f0: 2061 6c73 6f20 696e 2060 7767 6420 7632   also in `wgd v2
+0000f200: 602c 2077 6869 6368 2069 7320 6d6f 7374  `, which is most
+0000f210: 6c79 2074 6865 2073 616d 6520 6173 205b  ly the same as [
+0000f220: 6b73 7261 7465 735d 2868 7474 7073 3a2f  ksrates](https:/
+0000f230: 2f67 6974 6875 622e 636f 6d2f 5649 422d  /github.com/VIB-
+0000f240: 5053 422f 6b73 7261 7465 7329 2062 7574  PSB/ksrates) but
+0000f250: 2064 6966 6665 7273 2069 6e20 7468 6520   differs in the 
+0000f260: 6361 6c63 756c 6174 696f 6e20 6f66 2074  calculation of t
+0000f270: 6865 2073 7461 6e64 6172 6420 6465 7669  he standard devi
+0000f280: 6174 696f 6e20 6f66 2072 6573 6361 6c65  ation of rescale
+0000f290: 6420 2a4b 2a3c 7375 623e 533c 2f73 7562  d *K*<sub>S</sub
+0000f2a0: 3e20 6167 6573 2e20 546f 2070 6572 666f  > ages. To perfo
+0000f2b0: 726d 2074 6865 2072 6174 6520 636f 7272  rm the rate corr
+0000f2c0: 6563 7469 6f6e 2061 6e61 6c79 7369 732c  ection analysis,
+0000f2d0: 2075 7365 7273 2063 616e 2075 7365 2062   users can use b
+0000f2e0: 6f74 6820 7468 6520 6077 6764 206b 7364  oth the `wgd ksd
+0000f2f0: 6020 616e 6420 6077 6764 2076 697a 6020  ` and `wgd viz` 
+0000f300: 7072 6f67 7261 6d2e 2046 6f72 2074 6865  program. For the
+0000f310: 2060 7767 6420 6b73 6460 2070 726f 6772   `wgd ksd` progr
+0000f320: 616d 2c20 7573 6572 7320 6e65 6564 2074  am, users need t
+0000f330: 6f20 7072 6f76 6964 6520 6120 7370 6563  o provide a spec
+0000f340: 6965 7320 7472 6565 2076 6961 2074 6865  ies tree via the
+0000f350: 206f 7074 696f 6e20 602d 2d73 7065 6369   option `--speci
+0000f360: 6573 7472 6565 6020 6f6e 2077 6869 6368  estree` on which
+0000f370: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
+0000f380: 2063 616e 2062 6520 636f 6e64 7563 7465   can be conducte
+0000f390: 642e 204e 6f74 6520 7468 6174 2075 6e6e  d. Note that unn
+0000f3a0: 6563 6573 7361 7279 2062 7261 636b 6574  ecessary bracket
+0000f3b0: 7320 6d69 6768 7420 6c65 6164 2074 6f20  s might lead to 
+0000f3c0: 756e 6578 7065 6374 6564 2065 7272 6f72  unexpected error
+0000f3d0: 732c 2066 6f72 2069 6e73 7461 6e63 6520  s, for instance 
+0000f3e0: 6120 7472 6565 2060 2841 2c28 422c 4329  a tree `(A,(B,C)
+0000f3f0: 293b 6020 7368 6f75 6c64 206e 6f74 2062  );` should not b
+0000f400: 6520 7265 7072 6573 656e 7465 6420 6173  e represented as
+0000f410: 2060 2841 2c28 2842 2c43 2929 293b 602e   `(A,((B,C)));`.
+0000f420: 2054 6865 2073 6574 206f 6620 7370 6563   The set of spec
+0000f430: 6965 7320 7061 6972 7320 746f 2062 6520  ies pairs to be 
+0000f440: 7368 6f77 6e20 6973 2066 6c65 7869 626c  shown is flexibl
+0000f450: 6520 7468 6174 2074 6865 206d 6f73 7420  e that the most 
+0000f460: 636f 6e76 656e 6965 6e74 206f 7074 696f  convenient optio
+0000f470: 6e20 6973 2060 2d2d 666f 6375 7332 616c  n is `--focus2al
+0000f480: 6c60 2077 6869 6368 2073 696d 706c 7920  l` which simply 
+0000f490: 7368 6f77 7320 616c 6c20 7468 6520 706f  shows all the po
+0000f4a0: 7373 6962 6c65 2066 6f63 616c 2d73 6973  ssible focal-sis
+0000f4b0: 7465 7220 7370 6563 6965 7320 7061 6972  ter species pair
+0000f4c0: 732c 206f 7220 7573 6572 7320 6361 6e20  s, or users can 
+0000f4d0: 6d61 6e75 616c 6c79 2073 6574 2074 6865  manually set the
+0000f4e0: 2073 7065 6369 6573 2070 6169 7273 2076   species pairs v
+0000f4f0: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
+0000f500: 2d73 7061 6972 602e 204e 6f74 6520 7468  -spair`. Note th
+0000f510: 6174 2069 6620 7468 6520 7370 6563 6965  at if the specie
+0000f520: 7320 7061 6972 7320 7765 7265 206d 616e  s pairs were man
+0000f530: 7561 6c6c 7920 7365 742c 2069 7420 776f  ually set, it wo
+0000f540: 756c 6420 6265 206e 6565 6465 6420 746f  uld be needed to
+0000f550: 2063 6f2d 7365 7420 7468 6520 6f70 7469   co-set the opti
+0000f560: 6f6e 2060 2d2d 636c 6173 7369 6360 2e20  on `--classic`. 
+0000f570: 5468 6520 6f70 7469 6f6e 2060 2d2d 6f6e  The option `--on
+0000f580: 6c79 726f 6f74 6f75 7460 2063 616e 2062  lyrootout` can b
+0000f590: 6520 7365 7420 746f 206f 6e6c 7920 636f  e set to only co
+0000f5a0: 6e73 6964 6572 206f 7574 6772 6f75 7020  nsider outgroup 
+0000f5b0: 6174 2074 6865 2072 6f6f 742c 2069 6e73  at the root, ins
+0000f5c0: 7465 6164 206f 6620 616c 6c20 7468 6520  tead of all the 
+0000f5d0: 706f 7373 6962 6c65 206f 7574 6772 6f75  possible outgrou
+0000f5e0: 7073 2070 6572 2066 6f63 616c 2d73 6973  ps per focal-sis
+0000f5f0: 7465 7220 7370 6563 6965 7320 7061 6972  ter species pair
+0000f600: 2c20 7768 6963 6820 6861 7320 696d 7061  , which has impa
+0000f610: 6374 206f 6e20 7468 6520 6669 6e61 6c20  ct on the final 
+0000f620: 636f 7272 6563 7465 6420 2a4b 2a3c 7375  corrected *K*<su
+0000f630: 623e 533c 2f73 7562 3e20 6167 6573 2e20  b>S</sub> ages. 
+0000f640: 5765 2073 7567 6765 7374 206f 6620 7573  We suggest of us
+0000f650: 696e 6720 616c 6c20 7468 6520 706f 7373  ing all the poss
+0000f660: 6962 6c65 206f 7574 6772 6f75 7073 2070  ible outgroups p
+0000f670: 6572 2066 6f63 616c 2d73 6973 7465 7220  er focal-sister 
+0000f680: 7370 6563 6965 7320 7061 6972 2061 7320  species pair as 
+0000f690: 666f 7220 6120 6c65 7373 2062 6961 7365  for a less biase
+0000f6a0: 6420 7265 7375 6c74 2e20 5468 6520 7061  d result. The pa
+0000f6b0: 7261 6e6f 6d65 202a 4b2a 3c73 7562 3e53  ranome *K*<sub>S
+0000f6c0: 3c2f 7375 623e 2064 6174 6120 7368 6f75  </sub> data shou
+0000f6d0: 6c64 2062 6520 7072 6f76 6964 6564 2076  ld be provided v
+0000f6e0: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
+0000f6f0: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
+0000f700: 602e 200a 0a53 6f6d 6520 6f74 6865 7220  `. ..Some other 
+0000f710: 6f70 7469 6f6e 7320 7768 6963 6820 6861  options which ha
+0000f720: 7665 206e 6f20 696d 7061 6374 206f 6e20  ve no impact on 
+0000f730: 7468 6520 7261 7465 2063 6f72 7265 6374  the rate correct
+0000f740: 696f 6e20 6275 7420 6164 6420 6d6f 7265  ion but add more
+0000f750: 206c 6179 6572 7320 6f72 2063 6861 6e67   layers or chang
+0000f760: 6520 7468 6520 6170 7065 6172 616e 6365  e the appearance
+0000f770: 206f 6e20 7468 6520 6d69 7865 6420 2a4b   on the mixed *K
+0000f780: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
+0000f790: 6520 6469 7374 7269 6275 7469 6f6e 2c20  e distribution, 
+0000f7a0: 696e 636c 7564 6520 602d 2d70 6c6f 7461  include `--plota
+0000f7b0: 7067 6d6d 6020 616e 6420 602d 2d70 6c6f  pgmm` and `--plo
+0000f7c0: 7465 6c6d 6d60 2065 7463 2e20 5468 6520  telmm` etc. The 
+0000f7d0: 6f70 7469 6f6e 2060 2d2d 706c 6f74 6170  option `--plotap
+0000f7e0: 676d 6d60 2063 616e 2062 6520 7365 7420  gmm` can be set 
+0000f7f0: 746f 2063 616c 6c20 7468 6520 474d 4d20  to call the GMM 
+0000f800: 616e 616c 7973 6973 2075 706f 6e20 7468  analysis upon th
+0000f810: 6520 616e 6368 6f72 2070 6169 7220 2a4b  e anchor pair *K
+0000f820: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
+0000f830: 6520 6469 7374 7269 6275 7469 6f6e 2061  e distribution a
+0000f840: 6e64 2070 6c6f 7420 7468 6520 636c 7573  nd plot the clus
+0000f850: 7465 7269 6e67 2072 6573 756c 7420 7570  tering result up
+0000f860: 6f6e 2074 6865 206d 6978 6564 202a 4b2a  on the mixed *K*
+0000f870: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000f880: 2064 6973 7472 6962 7574 696f 6e2c 2077   distribution, w
+0000f890: 6869 6368 2068 6173 2074 6f20 6265 2063  hich has to be c
+0000f8a0: 6f2d 7365 7420 7769 7468 2074 6865 206f  o-set with the o
+0000f8b0: 7074 696f 6e20 602d 2d61 6e63 686f 7270  ption `--anchorp
+0000f8c0: 6f69 6e74 7360 2070 726f 7669 6469 6e67  oints` providing
+0000f8d0: 2074 6865 2061 6e63 686f 7220 7061 6972   the anchor pair
+0000f8e0: 7320 696e 666f 726d 6174 696f 6e2e 2054  s information. T
+0000f8f0: 6865 206f 7074 696f 6e20 602d 2d70 6c6f  he option `--plo
+0000f900: 7465 6c6d 6d60 2063 616e 2062 6520 7365  telmm` can be se
+0000f910: 7420 746f 2063 616c 6c20 7468 6520 454c  t to call the EL
+0000f920: 4d4d 2061 6e61 6c79 7369 7320 7570 6f6e  MM analysis upon
+0000f930: 2074 6865 2077 686f 6c65 2070 6172 616e   the whole paran
+0000f940: 6f6d 6520 2a4b 2a3c 7375 623e 533c 2f73  ome *K*<sub>S</s
+0000f950: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+0000f960: 7469 6f6e 2e20 2a2a 4e6f 7465 2074 6861  tion. **Note tha
+0000f970: 7420 7468 6520 7370 6563 6965 7320 6e61  t the species na
+0000f980: 6d65 7320 7072 6573 656e 7420 696e 2074  mes present in t
+0000f990: 6865 2073 7065 6369 6573 2074 7265 6520  he species tree 
+0000f9a0: 6669 6c65 2073 686f 756c 6420 6d61 7463  file should matc
+0000f9b0: 6820 7468 6520 6e61 6d65 7320 6f66 2074  h the names of t
+0000f9c0: 6865 2063 6f72 7265 7370 6f6e 6469 6e67  he corresponding
+0000f9d0: 2073 6571 7565 6e63 6520 6669 6c65 732a   sequence files*
+0000f9e0: 2a20 466f 7220 696e 7374 616e 6365 2c20  * For instance, 
+0000f9f0: 6769 7665 6e20 7468 6520 6364 7320 6669  given the cds fi
+0000fa00: 6c65 206e 616d 6573 2027 412e 6364 7327  le names 'A.cds'
+0000fa10: 2c27 422e 6364 7327 2c27 432e 6364 7327  ,'B.cds','C.cds'
+0000fa20: 2c20 7468 6520 7370 6563 6965 7320 7472  , the species tr
+0000fa30: 6565 2063 6f75 6c64 2062 6520 2728 412e  ee could be '(A.
+0000fa40: 6364 732c 2842 2e63 6473 2c43 2e63 6473  cds,(B.cds,C.cds
+0000fa50: 2929 3b27 2072 6174 6865 7220 7468 616e  ));' rather than
+0000fa60: 2027 2841 2c28 422c 4329 293b 272e 2054   '(A,(B,C));'. T
+0000fa70: 6865 7265 2069 7320 6e6f 2072 6571 7569  here is no requi
+0000fa80: 7265 6d65 6e74 2066 6f72 2074 6865 206e  rement for the n
+0000fa90: 616d 6520 6f66 2074 6865 2070 6172 616e  ame of the paran
+0000faa0: 6f6d 6520 2a4b 2a3c 7375 623e 533c 2f73  ome *K*<sub>S</s
+0000fab0: 7562 3e20 6461 7461 6669 6c65 2077 6869  ub> datafile whi
+0000fac0: 6368 2063 616e 2062 6520 6e61 6d65 6420  ch can be named 
+0000fad0: 696e 2077 6861 7465 7665 7220 6d61 6e6e  in whatever mann
+0000fae0: 6572 2075 7365 7273 2070 7265 6665 722e  er users prefer.
+0000faf0: 2054 6865 2027 474d 4d27 2069 7320 7468   The 'GMM' is th
+0000fb00: 6520 6162 6272 6576 6961 7469 6f6e 206f  e abbreviation o
+0000fb10: 6620 4761 7573 7369 616e 204d 6978 7475  f Gaussian Mixtu
+0000fb20: 7265 204d 6f64 656c 696e 6720 7768 696c  re Modeling whil
+0000fb30: 6520 7468 6520 2745 4c4d 4d27 2072 6566  e the 'ELMM' ref
+0000fb40: 6572 7320 746f 2045 7870 6f6e 656e 7469  ers to Exponenti
+0000fb50: 616c 2d4c 6f67 6e6f 726d 616c 204d 6978  al-Lognormal Mix
+0000fb60: 7475 7265 204d 6f64 656c 696e 6720 6173  ture Modeling as
+0000fb70: 205b 6b73 7261 7465 735d 2868 7474 7073   [ksrates](https
+0000fb80: 3a2f 2f67 6974 6875 622e 636f 6d2f 5649  ://github.com/VI
+0000fb90: 422d 5053 422f 6b73 7261 7465 7329 2069  B-PSB/ksrates) i
+0000fba0: 6e74 6572 7072 6574 732e 0a0a 5468 6572  nterprets...Ther
+0000fbb0: 6520 6172 6520 3231 2063 6f6c 756d 6e73  e are 21 columns
+0000fbc0: 2069 6e20 7468 6520 7265 7375 6c74 2060   in the result `
+0000fbd0: 2e6b 732e 7473 7660 2066 696c 6520 6265  .ks.tsv` file be
+0000fbe0: 7369 6465 7320 7468 6520 696e 6465 7820  sides the index 
+0000fbf0: 636f 6c75 6d6e 7320 6070 6169 7260 2061  columns `pair` a
+0000fc00: 7320 7468 6520 756e 6971 7565 2069 6465  s the unique ide
+0000fc10: 6e74 6966 6965 7220 666f 7220 6561 6368  ntifier for each
+0000fc20: 2067 656e 6520 7061 6972 2e20 5468 6520   gene pair. The 
+0000fc30: 604e 602c 2060 5360 2c20 6064 4e60 2c20  `N`, `S`, `dN`, 
+0000fc40: 6064 4e2f 6453 602c 2060 6453 602c 2060  `dN/dS`, `dS`, `
+0000fc50: 6c60 2061 6e64 2060 7460 2061 7265 2066  l` and `t` are f
+0000fc60: 726f 6d20 7468 6520 636f 6465 6d6c 2072  rom the codeml r
+0000fc70: 6573 756c 7473 2c20 7265 7072 6573 656e  esults, represen
+0000fc80: 7469 6e67 2074 6865 204e 2065 7374 696d  ting the N estim
+0000fc90: 6174 652c 2074 6865 2053 2065 7374 696d  ate, the S estim
+0000fca0: 6174 652c 2074 6865 2064 4e20 6573 7469  ate, the dN esti
+0000fcb0: 6d61 7465 2c20 7468 6520 644e 2f64 5320  mate, the dN/dS 
+0000fcc0: 286f 6d65 6761 2920 6573 7469 6d61 7465  (omega) estimate
+0000fcd0: 2c20 7468 6520 6453 2065 7374 696d 6174  , the dS estimat
+0000fce0: 652c 2074 6865 206c 6f67 2d6c 696b 656c  e, the log-likel
+0000fcf0: 6968 6f6f 6420 616e 6420 7468 6520 7420  ihood and the t 
+0000fd00: 6573 7469 6d61 7465 2c20 7265 7370 6563  estimate, respec
+0000fd10: 7469 7665 6c79 2e20 5468 6520 6061 6c69  tively. The `ali
+0000fd20: 676e 6d65 6e74 636f 7665 7261 6765 602c  gnmentcoverage`,
+0000fd30: 2060 616c 6967 6e6d 656e 7469 6465 6e74   `alignmentident
+0000fd40: 6974 7960 2061 6e64 2060 616c 6967 6e6d  ity` and `alignm
+0000fd50: 656e 746c 656e 6774 6860 2061 7265 2074  entlength` are t
+0000fd60: 6865 2069 6e66 6f72 6d61 7469 6f6e 2070  he information p
+0000fd70: 6572 7461 696e 696e 6720 746f 2074 6865  ertaining to the
+0000fd80: 2061 6c69 676e 6d65 6e74 2066 6f72 2065   alignment for e
+0000fd90: 6163 6820 6661 6d69 6c79 2c20 7265 7072  ach family, repr
+0000fda0: 6573 656e 7469 6e67 2074 6865 2072 6174  esenting the rat
+0000fdb0: 696f 206f 6620 7468 6520 7374 7269 7070  io of the stripp
+0000fdc0: 6564 2061 6c69 676e 6d65 6e74 206c 656e  ed alignment len
+0000fdd0: 6774 6820 636f 6d70 6172 6564 2074 6f20  gth compared to 
+0000fde0: 7468 6520 6675 6c6c 2061 6c69 676e 6d65  the full alignme
+0000fdf0: 6e74 206c 656e 6774 682c 2074 6865 2072  nt length, the r
+0000fe00: 6174 696f 206f 6620 636f 6c75 6d6e 7320  atio of columns 
+0000fe10: 7769 7468 2069 6465 6e74 6963 616c 206e  with identical n
+0000fe20: 7563 6c65 6f74 6964 6573 2063 6f6d 7061  ucleotides compa
+0000fe30: 7265 6420 746f 2074 6865 206f 7665 7261  red to the overa
+0000fe40: 6c6c 2063 6f6c 756d 6e73 206f 6620 7468  ll columns of th
+0000fe50: 6520 7374 7269 7070 6564 2061 6c69 676e  e stripped align
+0000fe60: 6d65 6e74 2c20 616e 6420 7468 6520 6c65  ment, and the le
+0000fe70: 6e67 7468 206f 6620 7468 6520 6675 6c6c  ngth of the full
+0000fe80: 2061 6c69 676e 6d65 6e74 2c20 7265 7370   alignment, resp
+0000fe90: 6563 7469 7665 6c79 2e0a 2323 2320 7767  ectively..### wg
+0000fea0: 6420 6d69 780a 0a2a 2a54 6865 206d 6978  d mix..**The mix
+0000feb0: 7475 7265 206d 6f64 656c 2063 6c75 7374  ture model clust
+0000fec0: 6572 696e 6720 616e 616c 7973 6973 206f  ering analysis o
+0000fed0: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
+0000fee0: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+0000fef0: 6f6e 2a2a 0a60 6060 0a77 6764 206d 6978  on**.```.wgd mix
+0000ff00: 206b 7364 6174 6120 282d 2d6e 5f69 6e69   ksdata (--n_ini
+0000ff10: 7420 3230 3020 2d2d 6d61 785f 6974 6572  t 200 --max_iter
+0000ff20: 2032 3030 202d 2d6b 735f 7261 6e67 6520   200 --ks_range 
+0000ff30: 3020 3520 2d2d 6669 6c74 6572 7320 3330  0 5 --filters 30
+0000ff40: 3020 2d2d 6269 6e73 2035 3020 2d2d 636f  0 --bins 50 --co
+0000ff50: 6d70 6f6e 656e 7473 2031 2034 202d 2d67  mponents 1 4 --g
+0000ff60: 616d 6d61 2030 2e30 3031 290a 6060 600a  amma 0.001).```.
+0000ff70: 0a54 6869 7320 7061 7274 206f 6620 4761  .This part of Ga
+0000ff80: 7573 7369 616e 206d 6978 7475 7265 206d  ussian mixture m
+0000ff90: 6f64 656c 696e 6720 2847 4d4d 2920 616e  odeling (GMM) an
+0000ffa0: 616c 7973 6973 2069 7320 696e 6865 7269  alysis is inheri
+0000ffb0: 7465 6420 6672 6f6d 2074 6865 206f 7269  ted from the ori
+0000ffc0: 6769 6e61 6c20 6077 6764 6020 7072 6f67  ginal `wgd` prog
+0000ffd0: 7261 6d2c 2062 7574 2077 7269 7465 7320  ram, but writes 
+0000ffe0: 6164 6469 7469 6f6e 616c 6c79 2074 6865  additionally the
+0000fff0: 2070 726f 6261 6269 6c69 7479 206f 6620   probability of 
+00010000: 6561 6368 202a 4b2a 3c73 7562 3e53 3c2f  each *K*<sub>S</
+00010010: 7375 623e 2076 616c 7565 2069 6e74 6f20  sub> value into 
+00010020: 7468 6520 6669 6e61 6c20 6461 7461 6672  the final datafr
+00010030: 616d 652e 2042 6173 6963 616c 6c79 2c20  ame. Basically, 
+00010040: 7573 6572 7320 6e65 6564 2074 6f20 7072  users need to pr
+00010050: 6f76 6964 6520 7769 7468 2061 2028 6e6f  ovide with a (no
+00010060: 726d 616c 6c79 2066 726f 6d20 7768 6f6c  rmally from whol
+00010070: 652d 7061 7261 6e6f 6d65 206f 7220 616e  e-paranome or an
+00010080: 6368 6f72 2d70 6169 7273 2920 2a4b 2a3c  chor-pairs) *K*<
+00010090: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
+000100a0: 6669 6c65 2061 6e64 2074 6865 2047 4d4d  file and the GMM
+000100b0: 2061 6e61 6c79 7369 7320 7769 6c6c 2062   analysis will b
+000100c0: 6520 636f 6e64 7563 7465 6420 7570 6f6e  e conducted upon
+000100d0: 2074 6865 2064 6174 6166 696c 652e 2053   the datafile. S
+000100e0: 6f6d 6520 7061 7261 6d65 7465 7273 2063  ome parameters c
+000100f0: 616e 2061 6666 6563 7420 7468 6520 7265  an affect the re
+00010100: 7375 6c74 732c 2069 6e63 6c75 6469 6e67  sults, including
+00010110: 2060 2d2d 6e5f 696e 6974 602c 2077 6869   `--n_init`, whi
+00010120: 6368 2073 6574 7320 7468 6520 6e75 6d62  ch sets the numb
+00010130: 6572 206f 6620 6b2d 6d65 616e 7320 696e  er of k-means in
+00010140: 6974 6961 6c69 7a61 7469 6f6e 7320 2864  itializations (d
+00010150: 6566 6175 6c74 2032 3030 292c 2060 2d2d  efault 200), `--
+00010160: 6d61 785f 6974 6572 602c 2077 6869 6368  max_iter`, which
+00010170: 2073 6574 7320 7468 6520 6d61 7869 6d75   sets the maximu
+00010180: 6d20 6e75 6d62 6572 206f 6620 6974 6572  m number of iter
+00010190: 6174 696f 6e73 2028 6465 6661 756c 7420  ations (default 
+000101a0: 3230 3029 2c20 602d 2d6d 6574 686f 6460  200), `--method`
+000101b0: 2c20 7768 6963 6820 6465 7465 726d 696e  , which determin
+000101c0: 6573 2077 6869 6368 2063 6c75 7374 6572  es which cluster
+000101d0: 696e 6720 6d65 7468 6f64 2074 6f20 7573  ing method to us
+000101e0: 6520 2864 6566 6175 6c74 2067 6d6d 292c  e (default gmm),
+000101f0: 2060 2d2d 6761 6d6d 6160 2c20 7768 6963   `--gamma`, whic
+00010200: 6820 7365 7473 2074 6865 2067 616d 6d61  h sets the gamma
+00010210: 2070 6172 616d 6574 6572 2066 6f72 2074   parameter for t
+00010220: 6865 2062 676d 6d20 6d6f 6465 6c20 2864  he bgmm model (d
+00010230: 6566 6175 6c74 2030 2e30 3031 292c 2060  efault 0.001), `
+00010240: 2d2d 636f 6d70 6f6e 656e 7473 602c 2077  --components`, w
+00010250: 6869 6368 2073 6574 7320 7468 6520 7261  hich sets the ra
+00010260: 6e67 6520 6f66 2074 6865 206e 756d 6265  nge of the numbe
+00010270: 7220 6f66 2063 6f6d 706f 6e65 6e74 7320  r of components 
+00010280: 746f 2066 6974 2028 6465 6661 756c 7420  to fit (default 
+00010290: 3120 3429 2c20 616e 6420 7468 6520 6461  1 4), and the da
+000102a0: 7461 2066 696c 7465 7269 6e67 2070 6172  ta filtering par
+000102b0: 616d 6574 6572 7320 602d 2d66 696c 7465  ameters `--filte
+000102c0: 7273 6020 7768 6963 6820 6669 6c74 6572  rs` which filter
+000102d0: 7320 6461 7461 2062 6173 6564 206f 6e20  s data based on 
+000102e0: 616c 6967 6e6d 656e 7420 6c65 6e67 7468  alignment length
+000102f0: 2c20 602d 2d6b 735f 7261 6e67 6560 2077  , `--ks_range` w
+00010300: 6869 6368 2066 696c 7465 7273 2064 6174  hich filters dat
+00010310: 6120 6261 7365 6420 6f6e 202a 4b2a 3c73  a based on *K*<s
+00010320: 7562 3e53 3c2f 7375 623e 2076 616c 7565  ub>S</sub> value
+00010330: 7320 616e 6420 7468 6520 7061 7261 6d65  s and the parame
+00010340: 7465 7220 602d 2d62 696e 7360 2077 6869  ter `--bins` whi
+00010350: 6368 2073 6574 7320 7468 6520 6e75 6d62  ch sets the numb
+00010360: 6572 206f 6620 6269 6e73 2069 6e20 2a4b  er of bins in *K
+00010370: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+00010380: 7374 7269 6275 7469 6f6e 2028 6465 6661  stribution (defa
+00010390: 756c 7420 3530 292e 0a0a 2a2a 4120 7375  ult 50)...**A su
+000103a0: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
+000103b0: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
+000103c0: 6d61 6e64 2073 696d 706c 7920 6173 2062  mand simply as b
+000103d0: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
+000103e0: 6d69 7820 6b73 6461 7461 0a60 6060 0a0a  mix ksdata.```..
+000103f0: 2323 2320 7767 6420 7065 616b 0a0a 2a2a  ### wgd peak..**
+00010400: 5468 6520 7365 6172 6368 206f 6620 6372  The search of cr
+00010410: 6564 6961 626c 6520 2a4b 2a3c 7375 623e  ediable *K*<sub>
+00010420: 533c 2f73 7562 3e20 7261 6e67 6520 7573  S</sub> range us
+00010430: 6564 2069 6e20 5747 4420 6461 7469 6e67  ed in WGD dating
+00010440: 2a2a 0a60 6060 0a77 6764 2070 6561 6b20  **.```.wgd peak 
+00010450: 6b73 6461 7461 202d 6170 2061 7064 6174  ksdata -ap apdat
+00010460: 6120 2d73 6d20 736d 6461 7461 202d 6c65  a -sm smdata -le
+00010470: 206c 6564 6174 6120 2d6d 7020 6d70 6461   ledata -mp mpda
+00010480: 7461 202d 2d68 6575 7269 7374 6963 2028  ta --heuristic (
+00010490: 2d2d 616c 6967 6e66 696c 7465 7220 302e  --alignfilter 0.
+000104a0: 3020 3020 302e 3020 2d2d 6b73 7261 6e67  0 0 0.0 --ksrang
+000104b0: 6520 3020 3520 2d2d 6269 6e5f 7769 6474  e 0 5 --bin_widt
+000104c0: 6820 302e 3120 2d2d 6775 6964 6520 7365  h 0.1 --guide se
+000104d0: 676d 656e 7420 2d2d 7072 6f6d 696e 656e  gment --prominen
+000104e0: 6365 5f63 7574 6f66 6620 302e 3120 2d2d  ce_cutoff 0.1 --
+000104f0: 7265 6c5f 6865 6967 6874 2030 2e34 202d  rel_height 0.4 -
+00010500: 2d63 6920 3935 202d 2d68 6472 2039 3520  -ci 95 --hdr 95 
+00010510: 2d2d 6b73 6375 746f 6666 2035 290a 6060  --kscutoff 5).``
+00010520: 600a 0a41 7320 6d65 6e74 696f 6e65 6420  `..As mentioned 
+00010530: 7072 6576 696f 7573 6c79 2c20 6120 6865  previously, a he
+00010540: 7572 6973 7469 6320 6d65 7468 6f64 2061  uristic method a
+00010550: 6e64 2061 2063 6f6c 6c69 6e65 6172 2073  nd a collinear s
+00010560: 6567 6d65 6e74 732d 6775 6964 6564 2061  egments-guided a
+00010570: 6e63 686f 7220 7061 6972 2063 6c75 7374  nchor pair clust
+00010580: 6572 696e 6720 666f 7220 7468 6520 7365  ering for the se
+00010590: 6172 6368 206f 6620 6372 6564 6961 626c  arch of crediabl
+000105a0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+000105b0: 3e20 7261 6e67 6520 7573 6564 2069 6e20  > range used in 
+000105c0: 5747 4420 6461 7469 6e67 2061 7265 2069  WGD dating are i
+000105d0: 6d70 6c65 6d65 6e74 6564 2069 6e20 6077  mplemented in `w
+000105e0: 6764 2076 3260 2e20 5573 6572 7320 6e65  gd v2`. Users ne
+000105f0: 6564 2074 6f20 7072 6f76 6964 6520 7468  ed to provide th
+00010600: 6520 616e 6368 6f72 2070 6f69 6e74 732c  e anchor points,
+00010610: 2073 6567 6d65 6e74 732c 206c 6973 7473   segments, lists
+00010620: 6567 6d65 6e74 732c 206d 756c 7469 706c  egments, multipl
+00010630: 6963 6f6e 7061 6972 7320 6461 7461 6669  iconpairs datafi
+00010640: 6c65 2066 726f 6d20 6069 2d61 6468 6f72  le from `i-adhor
+00010650: 6560 2074 6f20 6163 6869 6576 6520 7468  e` to achieve th
+00010660: 6520 636c 7573 7465 7269 6e67 2066 756e  e clustering fun
+00010670: 6374 696f 6e2e 2053 6f6d 6520 7061 7261  ction. Some para
+00010680: 6d65 7465 7273 2074 6861 7420 6361 6e20  meters that can 
+00010690: 696d 7061 6374 2074 6865 2072 6573 756c  impact the resul
+000106a0: 7473 2069 6e63 6c75 6465 2060 2d2d 616c  ts include `--al
+000106b0: 6967 6e66 696c 7465 7260 2c20 7768 6963  ignfilter`, whic
+000106c0: 6820 6669 6c74 6572 7320 7468 6520 6461  h filters the da
+000106d0: 7461 2062 6173 6564 206f 6e20 616c 6967  ta based on alig
+000106e0: 6e6d 656e 7420 6964 656e 7469 7479 2c20  nment identity, 
+000106f0: 6c65 6e67 7468 2061 6e64 2063 6f76 6572  length and cover
+00010700: 6167 652c 2060 2d2d 6b73 7261 6e67 6560  age, `--ksrange`
+00010710: 2c20 7768 6963 6820 7365 7473 2074 6865  , which sets the
+00010720: 2072 616e 6765 206f 6620 4b73 2074 6f20   range of Ks to 
+00010730: 6265 2061 6e61 6c79 7a65 642c 2060 2d2d  be analyzed, `--
+00010740: 6269 6e5f 7769 6474 6860 2c20 7768 6963  bin_width`, whic
+00010750: 6820 7365 7473 2074 6865 2062 616e 6477  h sets the bandw
+00010760: 6964 7468 206f 6620 2a4b 2a3c 7375 623e  idth of *K*<sub>
+00010770: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+00010780: 7469 6f6e 2c20 602d 2d77 6569 6768 7473  tion, `--weights
+00010790: 5f6f 7574 6c69 6572 735f 696e 636c 7564  _outliers_includ
+000107a0: 6564 6020 7768 6963 6820 6465 7465 726d  ed` which determ
+000107b0: 696e 6573 2077 6865 7468 6572 2074 6f20  ines whether to 
+000107c0: 696e 636c 7564 6520 2a4b 2a3c 7375 623e  include *K*<sub>
+000107d0: 533c 2f73 7562 3e20 6f75 746c 6965 7273  S</sub> outliers
+000107e0: 2028 7768 6f73 6520 7661 6c75 6520 6973   (whose value is
+000107f0: 206f 7665 7220 3529 2069 6e20 616e 616c   over 5) in anal
+00010800: 7973 6973 2c20 602d 2d6d 6574 686f 6460  ysis, `--method`
+00010810: 2077 6869 6368 2064 6574 6572 6d69 6e65   which determine
+00010820: 7320 7768 6963 6820 636c 7573 7465 7269  s which clusteri
+00010830: 6e67 206d 6574 686f 6420 746f 2075 7365  ng method to use
+00010840: 2028 6465 6661 756c 7420 676d 6d29 2c20   (default gmm), 
+00010850: 602d 2d73 6565 6460 2077 6869 6368 2073  `--seed` which s
+00010860: 6574 7320 7468 6520 7261 6e64 6f6d 2073  ets the random s
+00010870: 6565 6420 6769 7665 6e20 746f 2069 6e69  eed given to ini
+00010880: 7469 616c 697a 6174 696f 6e20 2864 6566  tialization (def
+00010890: 6175 6c74 2032 3335 3238 3930 292c 2060  ault 2352890), `
+000108a0: 2d2d 6e5f 696e 6974 602c 2077 6869 6368  --n_init`, which
+000108b0: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
+000108c0: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
+000108d0: 6961 6c69 7a61 7469 6f6e 7320 2864 6566  ializations (def
+000108e0: 6175 6c74 2032 3030 292c 2060 2d2d 656d  ault 200), `--em
+000108f0: 5f69 7465 7260 2c20 7768 6963 6820 7365  _iter`, which se
+00010900: 7473 2074 6865 206d 6178 696d 756d 206e  ts the maximum n
+00010910: 756d 6265 7220 6f66 2069 7465 7261 7469  umber of iterati
+00010920: 6f6e 7320 2864 6566 6175 6c74 2032 3030  ons (default 200
+00010930: 292c 2060 2d2d 6761 6d6d 6160 2c20 7768  ), `--gamma`, wh
+00010940: 6963 6820 7365 7473 2074 6865 2067 616d  ich sets the gam
+00010950: 6d61 2070 6172 616d 6574 6572 2066 6f72  ma parameter for
+00010960: 2074 6865 2062 676d 6d20 6d6f 6465 6c20   the bgmm model 
+00010970: 2864 6566 6175 6c74 2030 2e30 3031 292c  (default 0.001),
+00010980: 2060 2d2d 636f 6d70 6f6e 656e 7473 602c   `--components`,
+00010990: 2077 6869 6368 2073 6574 7320 7468 6520   which sets the 
+000109a0: 7261 6e67 6520 6f66 2074 6865 206e 756d  range of the num
+000109b0: 6265 7220 6f66 2063 6f6d 706f 6e65 6e74  ber of component
+000109c0: 7320 746f 2066 6974 2028 6465 6661 756c  s to fit (defaul
+000109d0: 7420 3120 3429 2c20 602d 2d77 6569 6768  t 1 4), `--weigh
+000109e0: 7465 6460 2077 6869 6368 2064 6574 6572  ted` which deter
+000109f0: 6d69 6e65 7320 7768 6574 6865 7220 746f  mines whether to
+00010a00: 2075 7365 206e 6f64 652d 7765 6967 6874   use node-weight
+00010a10: 6564 206d 6574 686f 6420 666f 7220 6465  ed method for de
+00010a20: 2d72 6564 756e 6461 6e63 792c 2060 2d2d  -redundancy, `--
+00010a30: 6775 6964 6560 2077 6869 6368 2064 6574  guide` which det
+00010a40: 6572 6d69 6e65 7320 7768 6963 6820 7265  ermines which re
+00010a50: 6769 6d65 2072 6573 6964 696e 6720 616e  gime residing an
+00010a60: 6368 6f72 7320 746f 2062 6520 7573 6564  chors to be used
+00010a70: 2028 6465 6661 756c 7420 5365 676d 656e   (default Segmen
+00010a80: 7429 2c20 602d 2d70 726f 6d69 6e65 6e63  t), `--prominenc
+00010a90: 655f 6375 746f 6666 6020 7768 6963 6820  e_cutoff` which 
+00010aa0: 7365 7473 2074 6865 2070 726f 6d69 6e65  sets the promine
+00010ab0: 6e63 6520 6375 746f 6666 206f 6620 6163  nce cutoff of ac
+00010ac0: 6365 7074 6162 6c65 2070 6561 6b73 2069  ceptable peaks i
+00010ad0: 6e20 7065 616b 2066 696e 6469 6e67 2070  n peak finding p
+00010ae0: 726f 6365 7373 2c20 602d 2d72 656c 5f68  rocess, `--rel_h
+00010af0: 6569 6768 7460 2077 6869 6368 2073 6574  eight` which set
+00010b00: 7320 7468 6520 7265 6c61 7469 7665 2068  s the relative h
+00010b10: 6569 6768 7420 6174 2077 6869 6368 2074  eight at which t
+00010b20: 6865 2070 6561 6b20 7769 6474 6820 6973  he peak width is
+00010b30: 206d 6561 7375 7265 642c 2060 2d2d 6b73   measured, `--ks
+00010b40: 746f 6461 7465 6020 7768 6963 6820 6d61  todate` which ma
+00010b50: 6e75 616c 6c79 2073 6574 7320 7468 6520  nually sets the 
+00010b60: 7261 6e67 6520 6f66 202a 4b2a 3c73 7562  range of *K*<sub
+00010b70: 3e53 3c2f 7375 623e 2074 6f20 6265 2064  >S</sub> to be d
+00010b80: 6174 6564 2069 6e20 6865 7572 6973 7469  ated in heuristi
+00010b90: 6320 7365 6172 6368 2061 6e64 206e 6565  c search and nee
+00010ba0: 6473 2074 6f20 6265 2063 6f2d 7365 7420  ds to be co-set 
+00010bb0: 7769 7468 206f 7074 696f 6e20 602d 2d6d  with option `--m
+00010bc0: 616e 7561 6c73 6574 602c 2060 2d2d 786c  anualset`, `--xl
+00010bd0: 696d 6020 616e 6420 602d 2d79 6c69 6d60  im` and `--ylim`
+00010be0: 2064 6574 6572 6d69 6e69 6e67 2074 6865   determining the
+00010bf0: 2078 2061 6e64 2079 2061 7869 7320 6c69   x and y axis li
+00010c00: 6d69 7420 6f66 2047 4d4d 202a 4b2a 3c73  mit of GMM *K*<s
+00010c10: 7562 3e53 3c2f 7375 623e 2064 6973 7472  ub>S</sub> distr
+00010c20: 6962 7574 696f 6e2c 2060 2d2d 6369 6020  ibution, `--ci` 
+00010c30: 7365 7474 696e 6720 7468 6520 636f 6e66  setting the conf
+00010c40: 6964 656e 6365 206c 6576 656c 206f 6620  idence level of 
+00010c50: 6c6f 672d 6e6f 726d 616c 2064 6973 7472  log-normal distr
+00010c60: 6962 7574 696f 6e20 746f 2064 6174 6520  ibution to date 
+00010c70: 2864 6566 6175 6c74 2039 3529 2c20 602d  (default 95), `-
+00010c80: 2d68 6472 6020 7365 7474 696e 6720 7468  -hdr` setting th
+00010c90: 6520 6869 6768 6573 7420 6465 6e73 6974  e highest densit
+00010ca0: 7920 7265 6769 6f6e 2028 4844 5229 2061  y region (HDR) a
+00010cb0: 7070 6c69 6564 2069 6e20 7468 6520 7365  pplied in the se
+00010cc0: 676d 656e 742d 6775 6964 6564 2061 6e63  gment-guided anc
+00010cd0: 686f 7220 7061 6972 202a 4b2a 3c73 7562  hor pair *K*<sub
+00010ce0: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
+00010cf0: 7574 696f 6e20 2864 6566 6175 6c74 2039  ution (default 9
+00010d00: 3529 2c20 602d 2d68 6575 7269 7374 6963  5), `--heuristic
+00010d10: 6020 6465 7465 726d 696e 696e 6720 7768  ` determining wh
+00010d20: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
+00010d30: 6520 6865 7572 6973 7469 6320 6d65 7468  e heuristic meth
+00010d40: 6f64 206f 6620 6465 6669 6e69 6e67 2043  od of defining C
+00010d50: 4920 666f 7220 6461 7469 6e67 2c20 602d  I for dating, `-
+00010d60: 2d6b 7363 7574 6f66 6660 2073 6574 7469  -kscutoff` setti
+00010d70: 6e67 2074 6865 202a 4b2a 3c73 7562 3e53  ng the *K*<sub>S
+00010d80: 3c2f 7375 623e 2073 6174 7572 6174 696f  </sub> saturatio
+00010d90: 6e20 6375 746f 6666 2069 6e20 6461 7469  n cutoff in dati
+00010da0: 6e67 2028 6465 6661 756c 7420 3529 2e0a  ng (default 5)..
+00010db0: 0a46 6f75 7220 7265 7375 6c74 2073 7562  .Four result sub
+00010dc0: 666c 6f64 6572 7320 7769 6c6c 2062 6520  floders will be 
+00010dd0: 7072 6f64 7563 6564 2c20 6e61 6d65 6c79  produced, namely
+00010de0: 2060 416e 6368 6f72 4b73 5f46 696e 6450   `AnchorKs_FindP
+00010df0: 6561 6b60 2c20 6041 6e63 686f 724b 735f  eak`, `AnchorKs_
+00010e00: 474d 4d60 2c20 6053 6567 6d65 6e74 4775  GMM`, `SegmentGu
+00010e10: 6964 654b 735f 474d 4d60 2061 6e64 2060  ideKs_GMM` and `
+00010e20: 5365 676d 656e 744b 735f 4669 6e64 5065  SegmentKs_FindPe
+00010e30: 616b 602e 2054 6865 2060 416e 6368 6f72  ak`. The `Anchor
+00010e40: 4b73 5f46 696e 6450 6561 6b60 2073 7562  Ks_FindPeak` sub
+00010e50: 666c 6f64 6572 2063 6f6e 7461 696e 7320  floder contains 
+00010e60: 7265 7375 6c74 7320 6f66 2074 6865 2064  results of the d
+00010e70: 6574 6563 7465 6420 7065 616b 7320 6279  etected peaks by
+00010e80: 2074 6865 2060 7369 676e 616c 6020 6d6f   the `signal` mo
+00010e90: 6475 6c65 206f 6620 6053 6369 5079 6020  dule of `SciPy` 
+00010ea0: 6c69 6272 6172 7920 616e 6420 7468 6520  library and the 
+00010eb0: 6173 7375 6d65 6420 6869 6768 6573 7420  assumed highest 
+00010ec0: 6d61 7373 2070 6172 7420 2872 6566 6572  mass part (refer
+00010ed0: 7265 6420 746f 2061 7320 4869 6768 4d61  red to as HighMa
+00010ee0: 7373 2068 6572 6561 6674 6572 2920 6f66  ss hereafter) of
+00010ef0: 2065 6163 6820 7065 616b 2c20 7768 6963   each peak, whic
+00010f00: 6820 6361 6e20 6265 2075 7365 6420 666f  h can be used fo
+00010f10: 7220 6675 7274 6865 7220 5747 4420 6461  r further WGD da
+00010f20: 7469 6e67 2e20 5468 6520 6041 6e63 686f  ting. The `Ancho
+00010f30: 724b 735f 474d 4d60 2073 686f 7773 2074  rKs_GMM` shows t
+00010f40: 6865 2047 4d4d 2072 6573 756c 7473 2075  he GMM results u
+00010f50: 706f 6e20 7468 6520 6f72 6967 696e 616c  pon the original
+00010f60: 2061 6e63 686f 7220 2a4b 2a3c 7375 623e   anchor *K*<sub>
+00010f70: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+00010f80: 7469 6f6e 2062 7920 7468 6520 606d 6978  tion by the `mix
+00010f90: 7475 7265 6020 6d6f 6475 6c65 206f 6620  ture` module of 
+00010fa0: 6073 6369 6b69 742d 6c65 6172 6e60 206c  `scikit-learn` l
+00010fb0: 6962 7261 7279 2061 6e64 2074 776f 2073  ibrary and two s
+00010fc0: 7562 666c 6f64 6572 732c 2060 4c6f 6747  ubfloders, `LogG
+00010fd0: 4d4d 5f43 4960 2063 6f6e 7461 696e 696e  MM_CI` containin
+00010fe0: 6720 7468 6520 7265 7375 6c74 7320 6f66  g the results of
+00010ff0: 2039 3525 2043 4920 6f66 2065 6163 6820   95% CI of each 
+00011000: 636f 6d70 6f6e 656e 742c 2060 4869 6768  component, `High
+00011010: 4d61 7373 5f43 4960 2063 6f6e 7461 696e  Mass_CI` contain
+00011020: 696e 6720 7468 6520 4869 6768 4d61 7373  ing the HighMass
+00011030: 206f 6620 6561 6368 2063 6f6d 706f 6e65   of each compone
+00011040: 6e74 2c20 7768 6963 6820 6361 6e20 6265  nt, which can be
+00011050: 2075 7365 6420 666f 7220 6675 7274 6865   used for furthe
+00011060: 7220 5747 4420 6461 7469 6e67 2e20 5468  r WGD dating. Th
+00011070: 6520 6053 6567 6d65 6e74 4775 6964 654b  e `SegmentGuideK
+00011080: 735f 474d 4d60 2073 7562 666f 6c64 6572  s_GMM` subfolder
+00011090: 2070 7265 7365 6e74 7320 7265 7375 6c74   presents result
+000110a0: 7320 6f66 2073 6567 6d65 6e74 202a 4b2a  s of segment *K*
+000110b0: 3c73 7562 3e53 3c2f 7375 623e 2047 4d4d  <sub>S</sub> GMM
+000110c0: 2077 6869 6368 2061 7265 206d 6170 7065   which are mappe
+000110d0: 6420 6261 636b 2074 6f20 7468 6520 7265  d back to the re
+000110e0: 7369 6469 6e67 2061 6e63 686f 7220 7061  siding anchor pa
+000110f0: 6972 7320 616e 6420 7468 6520 6173 736f  irs and the asso
+00011100: 6369 6174 6564 2039 3525 2048 4452 2061  ciated 95% HDR a
+00011110: 6e64 2048 6967 684d 6173 7320 6f66 2065  nd HighMass of e
+00011120: 6163 6820 7365 676d 656e 7420 636c 7573  ach segment clus
+00011130: 7465 7220 696e 2073 7562 666c 6f64 6572  ter in subfloder
+00011140: 7320 6f66 2060 4844 525f 4349 6020 616e  s of `HDR_CI` an
+00011150: 6420 6048 6967 684d 6173 735f 4349 602e  d `HighMass_CI`.
+00011160: 2054 6865 2060 5365 676d 656e 744b 735f   The `SegmentKs_
+00011170: 4669 6e64 5065 616b 6020 7375 6266 6f6c  FindPeak` subfol
+00011180: 6465 7220 6973 2073 696d 696c 6172 2074  der is similar t
+00011190: 6f20 6041 6e63 686f 724b 735f 4669 6e64  o `AnchorKs_Find
+000111a0: 5065 616b 6020 6275 7420 7769 7468 2073  Peak` but with s
+000111b0: 6567 6d65 6e74 202a 4b2a 3c73 7562 3e53  egment *K*<sub>S
+000111c0: 3c2f 7375 623e 2069 6e73 7465 6164 2e20  </sub> instead. 
+000111d0: 5468 6520 2a4b 2a3c 7375 623e 533c 2f73  The *K*<sub>S</s
+000111e0: 7562 3e20 696e 2060 4d75 6c74 6970 6c69  ub> in `Multipli
+000111f0: 636f 6e60 2063 616e 2061 6c73 6f20 6265  con` can also be
+00011200: 2063 616c 6375 6c61 7465 6420 696e 2070   calculated in p
+00011210: 6c61 6365 206f 6620 6053 6567 6d65 6e74  lace of `Segment
+00011220: 6020 7573 696e 6720 7468 6520 6f70 7469  ` using the opti
+00011230: 6f6e 2060 2d2d 6775 6964 6560 2061 7320  on `--guide` as 
+00011240: 7375 6368 2074 6865 2072 6573 756c 7420  such the result 
+00011250: 7469 746c 652c 206c 6162 656c 2c20 6669  title, label, fi
+00011260: 6c65 2061 6e64 2066 6f6c 6465 7220 6e61  le and folder na
+00011270: 6d65 7320 7769 6c6c 2062 6520 6368 616e  mes will be chan
+00011280: 6765 6420 6163 636f 7264 696e 676c 792e  ged accordingly.
+00011290: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
+000112a0: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
+000112b0: 2075 7365 2063 6f6d 6d61 6e64 2073 696d   use command sim
+000112c0: 706c 7920 6173 2062 656c 6f77 2a2a 0a0a  ply as below**..
+000112d0: 6060 600a 7767 6420 7065 616b 206b 7364  ```.wgd peak ksd
+000112e0: 6174 6120 2d61 7020 6170 6461 7461 202d  ata -ap apdata -
+000112f0: 736d 2073 6d64 6174 6120 2d6c 6520 6c65  sm smdata -le le
+00011300: 6461 7461 202d 6d70 206d 7064 6174 6120  data -mp mpdata 
+00011310: 2d2d 6865 7572 6973 7469 630a 6060 600a  --heuristic.```.
+00011320: 0a23 2323 2077 6764 2073 796e 0a0a 2a2a  .### wgd syn..**
+00011330: 5468 6520 696e 7472 612d 7370 6563 6966  The intra-specif
+00011340: 6963 2073 796e 7465 6e79 2069 6e66 6572  ic synteny infer
+00011350: 656e 6365 2a2a 0a60 6060 0a77 6764 2073  ence**.```.wgd s
+00011360: 796e 2066 616d 696c 6965 7320 6766 6620  yn families gff 
+00011370: 282d 2d6b 735f 6469 7374 7269 6275 7469  (--ks_distributi
+00011380: 6f6e 206b 7364 6174 6120 2d66 2067 656e  on ksdata -f gen
+00011390: 6520 2d61 2049 4420 2d2d 6d69 6e6c 656e  e -a ID --minlen
+000113a0: 202d 3120 2d2d 6d69 6e73 6567 6c65 6e20   -1 --minseglen 
+000113b0: 3130 3030 3020 2d2d 6d69 6e67 656e 656e  10000 --mingenen
+000113c0: 756d 2033 3029 0a60 6060 0a0a 5468 6520  um 30).```..The 
+000113d0: 7072 6f67 7261 6d20 6077 6764 2073 796e  program `wgd syn
+000113e0: 6020 6973 206d 6169 6e6c 7920 6465 616c  ` is mainly deal
+000113f0: 696e 6720 7769 7468 2063 6f6c 6c69 6e65  ing with colline
+00011400: 6172 6974 7920 6f72 2073 796e 7465 6e79  arity or synteny
+00011410: 2028 626f 7468 2072 6566 6572 7265 6420   (both referred 
+00011420: 746f 2061 7320 7379 6e74 656e 7920 6865  to as synteny he
+00011430: 7265 6166 7465 7229 2061 6e61 6c79 7369  reafter) analysi
+00011440: 732e 2054 776f 2069 6e70 7574 2066 696c  s. Two input fil
+00011450: 6573 2061 7265 2065 7373 656e 7469 616c  es are essential
+00011460: 2c20 7468 6520 6765 6e65 2066 616d 696c  , the gene famil
+00011470: 7920 6669 6c65 2061 6e64 2074 6865 2067  y file and the g
+00011480: 6666 3320 6669 6c65 2e20 5468 6520 6765  ff3 file. The ge
+00011490: 6e65 2066 616d 696c 7920 6669 6c65 2069  ne family file i
+000114a0: 7320 696e 2074 6865 2066 6f72 6d61 7420  s in the format 
+000114b0: 6173 2060 4f72 7468 6f46 696e 6465 7260  as `OrthoFinder`
+000114c0: 2e20 5468 6520 736f 6674 7761 7265 2060  . The software `
+000114d0: 692d 6164 686f 7265 6020 6973 2061 2070  i-adhore` is a p
+000114e0: 7265 7265 7175 6973 6974 652e 2057 6974  rerequisite. Wit
+000114f0: 6820 6465 6661 756c 7420 7061 7261 6d65  h default parame
+00011500: 7465 7273 2c20 7468 6520 7072 6f67 7261  ters, the progra
+00011510: 6d20 6261 7369 6361 6c6c 7920 636f 6e64  m basically cond
+00011520: 7563 7473 2031 2920 6669 6c74 6572 696e  ucts 1) filterin
+00011530: 6720 6765 6e65 2066 616d 696c 6965 7320  g gene families 
+00011540: 6261 7365 6420 6f6e 206d 6178 696d 756d  based on maximum
+00011550: 2066 616d 696c 7920 7369 7a65 2032 2920   family size 2) 
+00011560: 7265 7472 6965 7669 6e67 2067 656e 6520  retrieving gene 
+00011570: 706f 7369 7469 6f6e 2061 6e64 2073 6361  position and sca
+00011580: 6666 6f6c 6420 696e 666f 726d 6174 696f  ffold informatio
+00011590: 6e20 6672 6f6d 2067 6666 3320 6669 6c65  n from gff3 file
+000115a0: 2033 2920 7072 6f64 7563 696e 6720 7468   3) producing th
+000115b0: 6520 636f 6e66 6967 7572 6174 696f 6e20  e configuration 
+000115c0: 6669 6c65 2061 6e64 2061 7373 6f63 6961  file and associa
+000115d0: 7465 6420 6461 7461 6669 6c65 7320 666f  ted datafiles fo
+000115e0: 7220 6069 2d61 6468 6f72 6560 2034 2920  r `i-adhore` 4) 
+000115f0: 6361 6c6c 696e 6720 6069 2d61 6468 6f72  calling `i-adhor
+00011600: 6560 2067 6976 656e 2074 6865 2070 6172  e` given the par
+00011610: 616d 6574 6572 7320 7365 7420 746f 2069  ameters set to i
+00011620: 6e66 6572 2073 796e 7465 6e79 2035 2920  nfer synteny 5) 
+00011630: 7669 7375 616c 697a 696e 6720 7468 6520  visualizing the 
+00011640: 7379 6e74 656e 7920 696e 2022 646f 7470  synteny in "dotp
+00011650: 6c6f 7422 2069 6e20 7468 6520 756e 6974  lot" in the unit
+00011660: 206f 6620 6765 6e65 7320 616e 6420 6261   of genes and ba
+00011670: 7365 732c 2069 6e20 2253 796e 6465 7074  ses, in "Syndept
+00011680: 6822 2070 6c6f 7420 7368 6f77 696e 6720  h" plot showing 
+00011690: 7468 6520 6469 7374 7269 6275 7469 6f6e  the distribution
+000116a0: 206f 6620 6469 6666 6572 656e 7420 6361   of different ca
+000116b0: 7465 676f 7269 6573 206f 6620 636f 6c6c  tegories of coll
+000116c0: 696e 6561 7269 7479 2072 6174 696f 7320  inearity ratios 
+000116d0: 7769 7468 696e 2061 6e64 2062 6574 7765  within and betwe
+000116e0: 656e 2073 7065 6369 6573 2c20 696e 2022  en species, in "
+000116f0: 6475 7053 7461 636b 2220 706c 6f74 2073  dupStack" plot s
+00011700: 686f 7769 6e67 206d 756c 7469 706c 6963  howing multiplic
+00011710: 6f6e 7320 7769 7468 2064 6966 6665 7265  ons with differe
+00011720: 6e74 206d 756c 7469 706c 6963 6174 696f  nt multiplicatio
+00011730: 6e20 6c65 7665 6c73 2e20 3629 2069 6620  n levels. 6) if 
+00011740: 7769 7468 202a 4b2a 3c73 7562 3e53 3c2f  with *K*<sub>S</
+00011750: 7375 623e 2064 6174 612c 2061 2022 2a4b  sub> data, a "*K
+00011760: 2a3c 7375 623e 533c 2f73 7562 3e20 646f  *<sub>S</sub> do
+00011770: 7470 6c6f 7422 2077 6974 6820 646f 7473  tplot" with dots
+00011780: 2061 6e6e 6f74 6174 6564 2069 6e20 2a4b   annotated in *K
+00011790: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
+000117a0: 6c75 6573 2061 6e64 2061 202a 4b2a 3c73  lues and a *K*<s
+000117b0: 7562 3e53 3c2f 7375 623e 2064 6973 7472  ub>S</sub> distr
+000117c0: 6962 7574 696f 6e20 7769 7468 2061 6e63  ibution with anc
+000117d0: 686f 7220 7061 6972 7320 6465 6e6f 7465  hor pairs denote
+000117e0: 6420 7769 6c6c 2062 6520 7072 6f64 7563  d will be produc
+000117f0: 6564 2e20 5468 6520 6765 6e65 2069 6e66  ed. The gene inf
+00011800: 6f72 6d61 7469 6f6e 2069 6e20 7468 6520  ormation in the 
+00011810: 6765 6e65 2066 616d 696c 7920 6669 6c65  gene family file
+00011820: 2061 6e64 2067 6666 3320 6669 6c65 2073   and gff3 file s
+00011830: 686f 756c 6420 6265 206d 6174 6368 6564  hould be matched
+00011840: 2077 6869 6368 2072 6571 7569 7265 7320   which requires 
+00011850: 7573 6572 7320 746f 2073 6574 2070 726f  users to set pro
+00011860: 7065 7220 602d 2d66 6561 7475 7265 6020  per `--feature` 
+00011870: 616e 6420 602d 2d61 7474 7269 6275 7465  and `--attribute
+00011880: 602e 2054 6865 206d 6178 696d 756d 2066  `. The maximum f
+00011890: 616d 696c 7920 7369 7a65 2074 6f20 6265  amily size to be
+000118a0: 2069 6e63 6c75 6465 6420 6361 6e20 6265   included can be
+000118b0: 2073 6574 2076 6961 2074 6865 206f 7074   set via the opt
+000118c0: 696f 6e20 602d 2d6d 6178 7369 7a65 602c  ion `--maxsize`,
+000118d0: 206e 6f74 6564 2074 6861 7420 7468 6973   noted that this
+000118e0: 2066 696c 7465 7269 6e67 2069 7320 6d61   filtering is ma
+000118f0: 696e 6c79 2074 6f20 6472 6f70 2074 686f  inly to drop tho
+00011900: 7365 2068 7567 6520 7461 6e64 656d 2064  se huge tandem d
+00011910: 7570 6c69 6361 7465 7320 6661 6d69 6c79  uplicates family
+00011920: 2061 6e64 2074 7261 6e73 706f 7361 626c   and transposabl
+00011930: 6520 656c 656d 656e 7473 2028 5445 7329  e elements (TEs)
+00011940: 2066 616d 696c 792c 2061 6e64 206e 6f74   family, and not
+00011950: 206d 616e 6461 746f 7279 2e20 5573 6572   mandatory. User
+00011960: 7320 6361 6e20 6669 6c74 6572 2074 686f  s can filter tho
+00011970: 7365 2066 7261 676d 656e 7461 7279 2073  se fragmentary s
+00011980: 6361 6666 6f6c 6473 2076 6961 2074 6865  caffolds via the
+00011990: 206f 7074 696f 6e20 602d 2d6d 696e 6c65   option `--minle
+000119a0: 6e60 2e20 5468 6520 6d69 6e69 6d75 6d20  n`. The minimum 
+000119b0: 6c65 6e67 7468 2061 6e64 206e 756d 6265  length and numbe
+000119c0: 7220 6f66 2067 656e 6573 2066 6f72 2061  r of genes for a
+000119d0: 2073 6567 6d65 6e74 2074 6f20 6265 2063   segment to be c
+000119e0: 6f6e 7369 6465 7265 6420 6361 6e20 6265  onsidered can be
+000119f0: 2073 6574 2076 6961 2074 6865 206f 7074   set via the opt
+00011a00: 696f 6e20 602d 2d6d 696e 7365 676c 656e  ion `--minseglen
+00011a10: 6020 616e 6420 602d 2d6d 696e 6765 6e65  ` and `--mingene
+00011a20: 6e75 6d60 2e20 5265 6475 6e64 616e 7420  num`. Redundant 
+00011a30: 6d75 6c74 6970 6c69 636f 6e73 2063 616e  multiplicons can
+00011a40: 2062 6520 6b65 7074 2062 7920 7365 7420   be kept by set 
+00011a50: 7468 6520 666c 6167 206f 7074 696f 6e20  the flag option 
+00011a60: 602d 2d6b 6565 7072 6564 756e 602e 0a0a  `--keepredun`...
+00011a70: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
+00011a80: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
+00011a90: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
+00011aa0: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
+00011ab0: 600a 7767 6420 7379 6e20 6661 6d69 6c69  `.wgd syn famili
+00011ac0: 6573 2067 6666 0a60 6060 0a0a 2a2a 5468  es gff.```..**Th
+00011ad0: 6520 696e 7465 722d 7370 6563 6966 6963  e inter-specific
+00011ae0: 2073 796e 7465 6e79 2069 6e66 6572 656e   synteny inferen
+00011af0: 6365 2a2a 0a60 6060 0a77 6764 2073 796e  ce**.```.wgd syn
+00011b00: 2066 616d 696c 6965 7320 6766 6631 2067   families gff1 g
+00011b10: 6666 3220 282d 2d61 6464 6974 696f 6e61  ff2 (--additiona
+00011b20: 6c67 6666 696e 666f 2022 6d52 4e41 3b4e  lgffinfo "mRNA;N
+00011b30: 616d 6522 202d 2d61 6464 6974 696f 6e61  ame" --additiona
+00011b40: 6c67 6666 696e 666f 2022 6765 6e65 3b49  lgffinfo "gene;I
+00011b50: 4422 290a 6060 600a 0a46 6f72 206d 756c  D").```..For mul
+00011b60: 7469 2d73 7065 6369 6573 2073 796e 7465  ti-species synte
+00011b70: 6e79 2069 6e66 6572 656e 6365 2c20 6966  ny inference, if
+00011b80: 2075 7365 7273 2068 6176 6520 6766 6633   users have gff3
+00011b90: 2066 696c 6573 2077 6869 6368 2068 6176   files which hav
+00011ba0: 6520 6469 6666 6572 656e 7420 6665 6174  e different feat
+00011bb0: 7572 6573 206f 7220 6174 7472 6962 7574  ures or attribut
+00011bc0: 6573 2066 6f72 2067 656e 6520 706f 7369  es for gene posi
+00011bd0: 7469 6f6e 2069 6e66 6f72 6d61 7469 6f6e  tion information
+00011be0: 2072 6574 7269 6576 616c 2c20 7468 6520   retrieval, the 
+00011bf0: 6f70 7469 6f6e 2060 2d2d 6164 6469 7469  option `--additi
+00011c00: 6f6e 616c 6766 6669 6e66 6f60 2063 616e  onalgffinfo` can
+00011c10: 2062 6520 7365 7420 746f 2070 726f 7669   be set to provi
+00011c20: 6465 2074 6865 2061 6464 6974 696f 6e61  de the additiona
+00011c30: 6c20 696e 666f 726d 6174 696f 6e2e 2054  l information. T
+00011c40: 6865 2072 656d 6169 6e69 6e67 2070 6172  he remaining par
+00011c50: 616d 6574 6572 2073 6574 7469 6e67 2069  ameter setting i
+00011c60: 7320 7468 6520 7361 6d65 2061 7320 7468  s the same as th
+00011c70: 6520 696e 7472 612d 7370 6563 6966 6963  e intra-specific
+00011c80: 2073 796e 7465 6e79 2069 6e66 6572 656e   synteny inferen
+00011c90: 6365 2e0a 0a2a 2a41 2073 7567 6765 7374  ce...**A suggest
+00011ca0: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
+00011cb0: 6361 6e20 7573 6520 636f 6d6d 616e 6420  can use command 
+00011cc0: 7369 6d70 6c79 2061 7320 6265 6c6f 772a  simply as below*
+00011cd0: 2a0a 0a60 6060 0a77 6764 2073 796e 2066  *..```.wgd syn f
+00011ce0: 616d 696c 6965 7320 6766 6631 2067 6666  amilies gff1 gff
+00011cf0: 320a 6060 600a 0a23 2323 2077 6764 2076  2.```..### wgd v
+00011d00: 697a 0a0a 2a2a 5468 6520 7669 7375 616c  iz..**The visual
+00011d10: 697a 6174 696f 6e20 6f66 202a 4b2a 3c73  ization of *K*<s
+00011d20: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+00011d30: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
+00011d40: 454c 4d4d 2061 6e61 6c79 7369 732a 2a0a  ELMM analysis**.
+00011d50: 6060 600a 7767 6420 7669 7a20 2d64 206b  ```.wgd viz -d k
+00011d60: 7364 6174 610a 6060 600a 0a54 6865 2070  sdata.```..The p
+00011d70: 726f 6772 616d 2060 7767 6420 7669 7a60  rogram `wgd viz`
+00011d80: 2069 7320 6d61 696e 6c79 2066 6f72 2074   is mainly for t
+00011d90: 6865 2070 7572 706f 7365 206f 6620 2a4b  he purpose of *K
+00011da0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+00011db0: 7374 7269 6275 7469 6f6e 2061 6e64 2073  stribution and s
+00011dc0: 796e 7465 6e79 2076 6973 7561 6c69 7a61  ynteny visualiza
+00011dd0: 7469 6f6e 2c20 7769 7468 2073 6f6d 6520  tion, with some 
+00011de0: 6f70 7469 6f6e 616c 206d 6978 7475 7265  optional mixture
+00011df0: 206d 6f64 656c 696e 6720 616e 616c 7973   modeling analys
+00011e00: 6973 2e20 5468 6520 6261 7369 6320 6675  is. The basic fu
+00011e10: 6e63 7469 6f6e 2069 7320 6a75 7374 2074  nction is just t
+00011e20: 6f20 706c 6f74 2074 6865 202a 4b2a 3c73  o plot the *K*<s
+00011e30: 7562 3e53 3c2f 7375 623e 2064 6973 7472  ub>S</sub> distr
+00011e40: 6962 7574 696f 6e20 616e 6420 636f 6e64  ibution and cond
+00011e50: 7563 7420 616e 2045 4c4d 4d20 616e 616c  uct an ELMM anal
+00011e60: 7973 6973 2069 6e20 7365 6172 6368 206f  ysis in search o
+00011e70: 6620 706f 7465 6e74 6961 6c20 5747 4420  f potential WGD 
+00011e80: 636f 6d70 6f6e 656e 7473 2e20 536f 6d65  components. Some
+00011e90: 206b 6579 2070 6172 616d 6574 6572 7320   key parameters 
+00011ea0: 6166 6665 6374 696e 6720 7468 6520 454c  affecting the EL
+00011eb0: 4d4d 2072 6573 756c 7420 696e 636c 7564  MM result includ
+00011ec0: 6520 602d 2d70 726f 6d69 6e65 6e63 655f  e `--prominence_
+00011ed0: 6375 746f 6666 6020 616e 6420 602d 2d72  cutoff` and `--r
+00011ee0: 656c 5f68 6569 6768 7460 2c20 7768 6963  el_height`, whic
+00011ef0: 6820 6861 7665 2062 6565 6e20 6578 706c  h have been expl
+00011f00: 6169 6e65 6420 6168 6561 642c 2060 2d2d  ained ahead, `--
+00011f10: 656d 5f69 7465 7261 7469 6f6e 7360 2061  em_iterations` a
+00011f20: 6e64 2060 2d2d 656d 5f69 6e69 7469 616c  nd `--em_initial
+00011f30: 697a 6174 696f 6e73 6020 6465 7465 726d  izations` determ
+00011f40: 696e 696e 6720 7468 6520 6d61 7869 6d75  ining the maximu
+00011f50: 6d20 6974 6572 6174 696f 6e73 2061 6e64  m iterations and
+00011f60: 2069 6e69 7469 616c 697a 6174 696f 6e73   initializations
+00011f70: 2069 6e20 7468 6520 454d 2061 6c67 6f72   in the EM algor
+00011f80: 6974 686d 2e0a 0a2a 2a41 2073 7567 6765  ithm...**A sugge
+00011f90: 7374 6564 2073 7461 7274 696e 6720 7275  sted starting ru
+00011fa0: 6e20 6361 6e20 7573 6520 636f 6d6d 616e  n can use comman
+00011fb0: 6420 7369 6d70 6c79 2061 7320 6265 6c6f  d simply as belo
+00011fc0: 772a 2a0a 0a60 6060 0a77 6764 2076 697a  w**..```.wgd viz
+00011fd0: 202d 6420 6b73 6461 7461 0a60 6060 0a0a   -d ksdata.```..
+00011fe0: 2a2a 5468 6520 7669 7375 616c 697a 6174  **The visualizat
+00011ff0: 696f 6e20 6f66 202a 4b2a 3c73 7562 3e53  ion of *K*<sub>S
+00012000: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
+00012010: 6962 7574 696f 6e20 7769 7468 2072 6174  ibution with rat
+00012020: 6520 636f 7272 6563 7469 6f6e 2a2a 0a60  e correction**.`
+00012030: 6060 0a77 6764 2076 697a 202d 6420 6b73  ``.wgd viz -d ks
+00012040: 6461 7461 202d 7370 2073 7064 6174 6120  data -sp spdata 
+00012050: 2d2d 666f 6375 7332 616c 6c20 666f 6361  --focus2all foca
+00012060: 6c5f 7370 6563 6965 7320 2d2d 6578 7472  l_species --extr
+00012070: 6170 6172 616e 6f6d 656b 7320 6b73 6461  aparanomeks ksda
+00012080: 7461 2028 2d2d 616e 6368 6f72 706f 696e  ta (--anchorpoin
+00012090: 7473 2061 7064 6174 6120 2d2d 706c 6f74  ts apdata --plot
+000120a0: 6170 676d 6d20 2d2d 706c 6f74 656c 6d6d  apgmm --plotelmm
+000120b0: 290a 6060 600a 0a42 6573 6964 6573 2074  ).```..Besides t
+000120c0: 6865 2062 6173 6963 202a 4b2a 3c73 7562  he basic *K*<sub
+000120d0: 3e53 3c2f 7375 623e 2070 6c6f 742c 2073  >S</sub> plot, s
+000120e0: 7562 7374 6974 7574 696f 6e20 7261 7465  ubstitution rate
+000120f0: 2063 6f72 7265 6374 696f 6e20 6361 6e20   correction can 
+00012100: 616c 736f 2062 6520 6163 6869 6576 6564  also be achieved
+00012110: 2067 6976 656e 2061 7420 6c65 6173 7420   given at least 
+00012120: 6120 7370 6563 6965 7320 7472 6565 2028  a species tree (
+00012130: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
+00012140: 2d2d 7370 6563 6965 7374 7265 6560 2920  --speciestree`) 
+00012150: 616e 6420 6120 666f 6361 6c20 7370 6563  and a focal spec
+00012160: 6965 7320 2865 6974 6865 7220 7669 6120  ies (either via 
+00012170: 7468 6520 6f70 7469 6f6e 2060 2d2d 666f  the option `--fo
+00012180: 6375 7332 616c 6c60 206f 7220 7669 6120  cus2all` or via 
+00012190: 7468 6520 6f70 7469 6f6e 2060 2d2d 7370  the option `--sp
+000121a0: 6169 7260 2069 6e20 7468 6520 666f 726d  air` in the form
+000121b0: 206f 6620 2224 666f 6361 6c5f 7370 6563   of "$focal_spec
+000121c0: 6965 733b 2466 6f63 616c 5f73 7065 6369  ies;$focal_speci
+000121d0: 6573 2229 2e20 4974 2773 2073 7567 6765  es"). It's sugge
+000121e0: 7374 6564 2074 6861 7420 7468 6520 6f72  sted that the or
+000121f0: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
+00012200: 623e 533c 2f73 7562 3e20 6461 7461 2069  b>S</sub> data i
+00012210: 7320 7072 6f76 6964 6564 2062 7920 7468  s provided by th
+00012220: 6520 602d 2d64 6174 6166 696c 6560 206f  e `--datafile` o
+00012230: 7074 696f 6e20 616e 6420 7468 6520 7061  ption and the pa
+00012240: 7261 6c6f 676f 7573 202a 4b2a 3c73 7562  ralogous *K*<sub
+00012250: 3e53 3c2f 7375 623e 2064 6174 6120 6973  >S</sub> data is
+00012260: 2070 726f 7669 6465 6420 6279 2074 6865   provided by the
+00012270: 2060 2d2d 6578 7472 6170 6172 616e 6f6d   `--extraparanom
+00012280: 656b 7360 206f 7074 696f 6e2c 2061 6c74  eks` option, alt
+00012290: 686f 7567 6820 6974 2773 2061 6c6c 6f77  hough it's allow
+000122a0: 6564 2074 6f20 6f6e 6c79 2070 726f 7669  ed to only provi
+000122b0: 6465 202a 4b2a 3c73 7562 3e53 3c2f 7375  de *K*<sub>S</su
+000122c0: 623e 2064 6174 6120 7669 6120 7468 6520  b> data via the 
+000122d0: 602d 2d64 6174 6166 696c 6560 206f 7074  `--datafile` opt
+000122e0: 696f 6e20 616e 6420 6465 706f 7369 7420  ion and deposit 
+000122f0: 626f 7468 206f 7274 686f 6c6f 676f 7573  both orthologous
+00012300: 2061 6e64 2070 6172 616c 6f67 6f75 7320   and paralogous 
+00012310: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00012320: 6461 7461 2074 6865 7265 6f6e 2e20 5468  data thereon. Th
+00012330: 6572 6520 6172 6520 7477 6f20 7479 7065  ere are two type
+00012340: 7320 6f66 206d 6978 6564 2070 6c6f 7473  s of mixed plots
+00012350: 2028 7468 6520 226d 6978 6564 2220 6865   (the "mixed" he
+00012360: 7265 2072 6566 6572 7320 746f 206d 6978  re refers to mix
+00012370: 6564 206f 7274 686f 6c6f 676f 7573 2061  ed orthologous a
+00012380: 6e64 2070 6172 616c 6f67 6f75 7320 2a4b  nd paralogous *K
+00012390: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+000123a0: 7374 7269 6275 7469 6f6e 7329 2c20 6f6e  stributions), on
+000123b0: 6520 6f66 2077 6869 6368 2069 7320 7369  e of which is si
+000123c0: 6d69 6c61 7220 746f 2077 6861 7420 606b  milar to what `k
+000123d0: 7372 6174 6573 6020 706c 6f74 732c 2077  srates` plots, w
+000123e0: 6869 6c65 2074 6865 206f 7468 6572 206f  hile the other o
+000123f0: 6620 7768 6963 6820 6973 206c 696b 6520  f which is like 
+00012400: 7468 6520 636f 6e76 656e 7469 6f6e 616c  the conventional
+00012410: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00012420: 2070 6c6f 7420 7468 6174 2062 6f74 6820   plot that both 
+00012430: 7468 6520 6f72 6967 696e 616c 206f 7274  the original ort
+00012440: 686f 6c6f 676f 7573 2061 6e64 2070 6172  hologous and par
+00012450: 616c 6f67 6f75 7320 2a4b 2a3c 7375 623e  alogous *K*<sub>
+00012460: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+00012470: 7469 6f6e 7320 6172 6520 7472 7574 6866  tions are truthf
+00012480: 756c 6c79 2070 6c6f 7474 6564 2069 6e73  ully plotted ins
+00012490: 7465 6164 206f 6620 6a75 7374 2062 6569  tead of just bei
+000124a0: 6e67 2072 6570 7265 7365 6e74 6564 2062  ng represented b
+000124b0: 7920 736f 6d65 2076 6572 7469 6361 6c20  y some vertical 
+000124c0: 6c69 6e65 732e 2057 6520 7375 6767 6573  lines. We sugges
+000124d0: 7420 7573 6572 7320 746f 2061 646f 7074  t users to adopt
+000124e0: 2074 6865 2060 6b73 7261 7465 7360 2d6c   the `ksrates`-l
+000124f0: 696b 6520 706c 6f74 732c 2077 6869 6368  ike plots, which
+00012500: 2069 7320 7468 6520 6465 6661 756c 742e   is the default.
+00012510: 204f 7468 6572 7769 7365 2074 6865 206f   Otherwise the o
+00012520: 7074 696f 6e20 602d 2d63 6c61 7373 6963  ption `--classic
+00012530: 6020 7769 6c6c 2062 6520 6e65 6564 6564  ` will be needed
+00012540: 2074 6f20 7365 742e 2045 7874 7261 206d   to set. Extra m
+00012550: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
+00012560: 616e 616c 7973 6973 2063 616e 2062 6520  analysis can be 
+00012570: 696e 6974 6961 7465 6420 7669 6120 7468  initiated via th
+00012580: 6520 6f70 7469 6f6e 2060 2d2d 706c 6f74  e option `--plot
+00012590: 656c 6d6d 6020 616e 6420 602d 2d70 6c6f  elmm` and `--plo
+000125a0: 7461 7067 6d6d 6020 7769 7468 2074 6865  tapgmm` with the
+000125b0: 2061 6e63 686f 7220 706f 696e 7473 2064   anchor points d
+000125c0: 6174 6166 696c 6520 7072 6f76 6964 6564  atafile provided
+000125d0: 2062 7920 7468 6520 602d 2d61 6e63 686f   by the `--ancho
+000125e0: 7270 6f69 6e74 7360 206f 7074 696f 6e2e  rpoints` option.
+000125f0: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
+00012600: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
+00012610: 2075 7365 2063 6f6d 6d61 6e64 2073 696d   use command sim
+00012620: 706c 7920 6173 2062 656c 6f77 2a2a 0a0a  ply as below**..
+00012630: 6060 600a 7767 6420 7669 7a20 2d64 206b  ```.wgd viz -d k
+00012640: 7364 6174 6120 2d73 7020 7370 6461 7461  sdata -sp spdata
+00012650: 202d 2d66 6f63 7573 3261 6c6c 2066 6f63   --focus2all foc
+00012660: 616c 5f73 7065 6369 6573 202d 2d65 7874  al_species --ext
+00012670: 7261 7061 7261 6e6f 6d65 6b73 206b 7364  raparanomeks ksd
+00012680: 6174 610a 6060 600a 0a2a 2a54 6865 2076  ata.```..**The v
+00012690: 6973 7561 6c69 7a61 7469 6f6e 206f 6620  isualization of 
+000126a0: 7379 6e74 656e 792a 2a0a 6060 600a 7767  synteny**.```.wg
+000126b0: 6420 7669 7a20 2d61 7020 6170 6461 7461  d viz -ap apdata
+000126c0: 202d 736d 2073 6d64 6174 6120 2d6d 7420   -sm smdata -mt 
+000126d0: 6d74 6461 7461 202d 6774 2067 7464 6174  mtdata -gt gtdat
+000126e0: 6120 2d2d 706c 6f74 7379 6e20 282d 2d6d  a --plotsyn (--m
+000126f0: 696e 6c65 6e20 2d31 202d 2d6d 696e 7365  inlen -1 --minse
+00012700: 676c 656e 2031 3030 3030 202d 2d6d 696e  glen 10000 --min
+00012710: 6765 6e65 6e75 6d20 3330 290a 6060 600a  genenum 30).```.
+00012720: 0a43 6f6d 7061 7265 6420 746f 2074 6865  .Compared to the
+00012730: 206f 7269 6769 6e61 6c20 6077 6764 602c   original `wgd`,
+00012740: 2074 6865 2060 7767 6420 7669 7a60 2070   the `wgd viz` p
+00012750: 726f 6772 616d 2061 6464 6564 2073 796e  rogram added syn
+00012760: 7465 6e79 2076 6973 7561 6c69 7a61 7469  teny visualizati
+00012770: 6f6e 2070 6970 656c 696e 652e 2055 7365  on pipeline. Use
+00012780: 7273 206e 6565 6420 746f 2070 726f 7669  rs need to provi
+00012790: 6465 2074 6865 2066 6c61 6720 6f70 7469  de the flag opti
+000127a0: 6f6e 2060 2d2d 706c 6f74 7379 6e60 2074  on `--plotsyn` t
+000127b0: 6f20 696e 6974 6961 7465 2074 6869 7320  o initiate this 
+000127c0: 7061 7274 206f 6620 7069 7065 6c69 6e65  part of pipeline
+000127d0: 2e20 5468 6973 2073 7465 7020 6173 7375  . This step assu
+000127e0: 6d65 7320 7468 6174 2075 7365 7273 2068  mes that users h
+000127f0: 6176 6520 6f62 7461 696e 6564 2061 6c72  ave obtained alr
+00012800: 6561 6479 2074 6865 2073 796e 7465 6e69  eady the synteni
+00012810: 6320 7265 7375 6c74 7320 6672 6f6d 2060  c results from `
+00012820: 692d 6164 686f 7265 6020 616e 6420 7573  i-adhore` and us
+00012830: 6573 2074 686f 7365 2072 6573 756c 7420  es those result 
+00012840: 6669 6c65 7320 746f 2072 6561 6c69 7a65  files to realize
+00012850: 2074 6865 2073 796e 7465 6e79 2076 6973   the synteny vis
+00012860: 7561 6c69 7a61 7469 6f6e 2e20 5369 6d6c  ualization. Siml
+00012870: 6961 7220 746f 2060 7767 6420 7379 6e60  iar to `wgd syn`
+00012880: 2c20 7468 6520 6578 7472 6120 2a4b 2a3c  , the extra *K*<
+00012890: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
+000128a0: 2066 696c 6520 6361 6e20 6265 2074 7261   file can be tra
+000128b0: 6e73 6d69 7474 6564 2076 6961 2074 6865  nsmitted via the
+000128c0: 206f 7074 696f 6e20 602d 2d64 6174 6166   option `--dataf
+000128d0: 696c 6560 2028 696e 7374 6561 6420 6f66  ile` (instead of
+000128e0: 2060 2d2d 6578 7472 6170 6172 616e 6f6d   `--extraparanom
+000128f0: 656b 7360 206f 7074 696f 6e29 2e20 4261  eks` option). Ba
+00012900: 7369 6361 6c6c 792c 2074 6865 2073 796e  sically, the syn
+00012910: 7465 6e69 6320 7265 7375 6c74 2066 696c  tenic result fil
+00012920: 6573 2072 6571 7569 7265 6420 6172 6520  es required are 
+00012930: 7468 6520 616e 6368 6f72 2070 6f69 6e74  the anchor point
+00012940: 7320 6461 7461 6669 6c65 2c20 6d75 6c74  s datafile, mult
+00012950: 6970 6c69 636f 6e73 2064 6174 6166 696c  iplicons datafil
+00012960: 652c 2067 656e 6520 7461 626c 6520 6461  e, gene table da
+00012970: 7461 6669 6c65 2028 6175 746f 6d61 7465  tafile (automate
+00012980: 6c79 2070 726f 6475 6365 6420 6279 2060  ly produced by `
+00012990: 7767 6420 7379 6e60 292c 2061 6e64 2073  wgd syn`), and s
+000129a0: 6567 6d65 6e74 7320 6461 7461 6669 6c65  egments datafile
+000129b0: 2e0a 0a2a 2a41 2073 7567 6765 7374 6564  ...**A suggested
+000129c0: 2073 7461 7274 696e 6720 7275 6e20 6361   starting run ca
+000129d0: 6e20 7573 6520 636f 6d6d 616e 6420 7369  n use command si
+000129e0: 6d70 6c79 2061 7320 6265 6c6f 772a 2a0a  mply as below**.
+000129f0: 0a60 6060 0a77 6764 2076 697a 202d 6170  .```.wgd viz -ap
+00012a00: 2061 7064 6174 6120 2d73 6d20 736d 6461   apdata -sm smda
+00012a10: 7461 202d 6d74 206d 7464 6174 6120 2d67  ta -mt mtdata -g
+00012a20: 7420 6774 6461 7461 202d 2d70 6c6f 7473  t gtdata --plots
+00012a30: 796e 0a60 6060 0a0a 2323 2049 6c6c 7573  yn.```..## Illus
+00012a40: 7472 6174 696f 6e0a 0a57 6520 696c 6c75  tration..We illu
+00012a50: 7374 7261 7465 206f 7572 2070 726f 6772  strate our progr
+00012a60: 616d 206f 6e20 616e 2065 7865 6d70 6c61  am on an exempla
+00012a70: 7279 2057 4744 2069 6e66 6572 656e 6365  ry WGD inference
+00012a80: 2061 6e64 2064 6174 696e 6720 7570 6f6e   and dating upon
+00012a90: 2073 7065 6369 6573 202a 4171 7569 6c65   species *Aquile
+00012aa0: 6769 6120 636f 6572 756c 6561 2a2e 0a0a  gia coerulea*...
+00012ab0: 5468 6520 2a41 7175 696c 6567 6961 2063  The *Aquilegia c
+00012ac0: 6f65 7275 6c65 612a 2077 6173 2072 6570  oerulea* was rep
+00012ad0: 6f72 7465 6420 746f 2065 7870 6572 6965  orted to experie
+00012ae0: 6e63 6520 616e 2070 616c 656f 2d70 6f6c  nce an paleo-pol
+00012af0: 7970 6c6f 6964 697a 6174 696f 6e20 6576  yploidization ev
+00012b00: 656e 7420 6166 7465 7220 7468 6520 6469  ent after the di
+00012b10: 7665 7267 656e 6365 206f 6620 636f 7265  vergence of core
+00012b20: 2065 7564 6963 6f74 732c 2077 6869 6368   eudicots, which
+00012b30: 2069 7320 6c69 6b65 6c79 2073 6861 7265   is likely share
+00012b40: 6420 6279 2061 6c6c 2052 616e 756e 6375  d by all Ranuncu
+00012b50: 6c61 6c65 732e 0a0a 4669 7273 7420 6162  lales...First ab
+00012b60: 6f76 6520 616c 6c2c 206c 6574 2773 2064  ove all, let's d
+00012b70: 656c 696e 6561 7465 2074 6865 2077 686f  elineate the who
+00012b80: 6c65 2070 6172 616e 6f6d 6520 2a4b 2a3c  le paranome *K*<
+00012b90: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+00012ba0: 6469 7374 7269 6275 7469 6f6e 2061 6e64  distribution and
+00012bb0: 2068 6176 6520 6120 6261 7369 6320 6f62   have a basic ob
+00012bc0: 7365 7276 6174 696f 6e20 666f 7220 706f  servation for po
+00012bd0: 7465 6e74 6961 6c6c 7920 636f 6e63 6569  tentially concei
+00012be0: 7661 626c 6520 5747 4473 2c20 7573 696e  vable WGDs, usin
+00012bf0: 6720 7468 6520 636f 6d6d 616e 6420 6c69  g the command li
+00012c00: 6e65 2062 656c 6f77 2e0a 0a60 6060 0a77  ne below...```.w
+00012c10: 6764 2064 6d64 2041 7175 696c 6567 6961  gd dmd Aquilegia
+00012c20: 5f63 6f65 7275 6c65 610a 7767 6420 6b73  _coerulea.wgd ks
+00012c30: 6420 7767 645f 646d 642f 4171 7569 6c65  d wgd_dmd/Aquile
+00012c40: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00012c50: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+00012c60: 6c65 610a 6060 600a 0a54 6865 2063 6f6e  lea.```..The con
+00012c70: 7374 7275 6374 6564 2077 686f 6c65 2070  structed whole p
+00012c80: 6172 616e 6f6d 6520 2a4b 2a3c 7375 623e  aranome *K*<sub>
+00012c90: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
+00012ca0: 7269 6275 7469 6f6e 206f 6620 2a41 7175  ribution of *Aqu
+00012cb0: 696c 6567 6961 2063 6f65 7275 6c65 612a  ilegia coerulea*
+00012cc0: 2069 7320 6173 2062 656c 6f77 2c20 7765   is as below, we
+00012cd0: 2063 616e 2073 6565 2074 6861 7420 7468   can see that th
+00012ce0: 6572 6520 7365 656d 7320 746f 2062 6520  ere seems to be 
+00012cf0: 6120 6875 6d70 2061 7420 2a4b 2a3c 7375  a hump at *K*<su
+00012d00: 623e 533c 2f73 7562 3e20 3120 6275 7420  b>S</sub> 1 but 
+00012d10: 6e6f 7420 636c 6561 722e 0a0a 215b 5d28  not clear...![](
+00012d20: 6461 7461 2f41 7175 696c 6567 6961 5f63  data/Aquilegia_c
+00012d30: 6f65 7275 6c65 612e 7473 762e 6b73 645f  oerulea.tsv.ksd_
+00012d40: 7770 2e73 7667 290a 0a57 6520 7468 656e  wp.svg)..We then
+00012d50: 2063 6f6e 7374 7275 6374 2074 6865 2061   construct the a
+00012d60: 6e63 686f 7220 2a4b 2a3c 7375 623e 533c  nchor *K*<sub>S<
+00012d70: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+00012d80: 6275 7469 6f6e 2075 7369 6e67 2074 6865  bution using the
+00012d90: 2063 6f6d 6d61 6e64 206c 696e 6520 6265   command line be
+00012da0: 6c6f 772e 0a0a 6060 600a 7767 6420 7379  low...```.wgd sy
+00012db0: 6e20 2d66 206d 524e 4120 2d61 204e 616d  n -f mRNA -a Nam
+00012dc0: 6520 7767 645f 646d 642f 4171 7569 6c65  e wgd_dmd/Aquile
+00012dd0: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00012de0: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+00012df0: 6c65 612e 6766 6633 202d 6b73 2077 6764  lea.gff3 -ks wgd
+00012e00: 5f6b 7364 2f41 7175 696c 6567 6961 5f63  _ksd/Aquilegia_c
+00012e10: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00012e20: 7376 0a60 6060 0a0a 4173 2073 686f 776e  sv.```..As shown
+00012e30: 2062 656c 6f77 2c20 7468 6572 6520 6172   below, there ar
+00012e40: 6520 736f 6d65 2072 6574 6169 6e65 6420  e some retained 
+00012e50: 616e 6368 6f72 2070 6169 7273 2077 6974  anchor pairs wit
+00012e60: 6820 2a4b 2a3c 7375 623e 533c 2f73 7562  h *K*<sub>S</sub
+00012e70: 3e20 6265 7477 6565 6e20 3120 616e 6420  > between 1 and 
+00012e80: 322c 2077 6869 6368 2073 6565 6d73 2074  2, which seems t
+00012e90: 6f20 7375 6767 6573 7420 6120 5747 4420  o suggest a WGD 
+00012ea0: 6576 656e 742e 0a0a 215b 5d28 6461 7461  event...![](data
+00012eb0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00012ec0: 6c65 612e 7473 762e 6b73 645f 7770 5f61  lea.tsv.ksd_wp_a
+00012ed0: 702e 7376 6729 0a0a 5468 6520 6173 736f  p.svg)..The asso
+00012ee0: 6369 6174 6564 2060 6475 7053 7461 636b  ciated `dupStack
+00012ef0: 6020 706c 6f74 2073 686f 7773 2074 6861  ` plot shows tha
+00012f00: 7420 7468 6572 6520 6172 6520 6e75 6d65  t there are nume
+00012f10: 726f 7573 2064 7570 6c69 6361 7465 6420  rous duplicated 
+00012f20: 7365 676d 656e 7473 2061 6372 6f73 7320  segments across 
+00012f30: 6d6f 7374 206f 6620 7468 6520 6368 726f  most of the chro
+00012f40: 6d6f 736f 6d65 732e 0a0a 215b 5d28 6461  mosomes...![](da
+00012f50: 7461 2f73 796e 5f72 6573 756c 7473 2f41  ta/syn_results/A
+00012f60: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00012f70: 615f 4171 7569 6c65 6769 615f 636f 6572  a_Aquilegia_coer
+00012f80: 756c 6561 5f6d 756c 7469 706c 6963 6f6e  ulea_multiplicon
+00012f90: 735f 6c65 7665 6c2e 706e 6729 0a0a 5765  s_level.png)..We
+00012fa0: 2069 6d70 6c65 6d65 6e74 6564 2074 776f   implemented two
+00012fb0: 2074 7970 6573 206f 6620 646f 7420 706c   types of dot pl
+00012fc0: 6f74 7320 696e 206f 7866 6f72 6420 6772  ots in oxford gr
+00012fd0: 6964 3a20 6f6e 6520 696e 2074 6865 2075  id: one in the u
+00012fe0: 6e69 7420 6f66 2062 6173 6573 2061 6e64  nit of bases and
+00012ff0: 2074 6865 206f 7468 6572 2069 6e20 7468   the other in th
+00013000: 6520 756e 6974 206f 6620 6765 6e65 732c  e unit of genes,
+00013010: 2077 6869 6368 2063 616e 2062 6520 636f   which can be co
+00013020: 6c6f 7265 6420 6279 202a 4b2a 3c73 7562  lored by *K*<sub
+00013030: 3e53 3c2f 7375 623e 2076 616c 7565 7320  >S</sub> values 
+00013040: 6769 7665 6e20 2a4b 2a3c 7375 623e 533c  given *K*<sub>S<
+00013050: 2f73 7562 3e20 6461 7461 2e0a 0a21 5b5d  /sub> data...![]
+00013060: 2864 6174 612f 7379 6e5f 7265 7375 6c74  (data/syn_result
+00013070: 732f 4171 7569 6c65 6769 615f 636f 6572  s/Aquilegia_coer
+00013080: 756c 6561 2d76 732d 4171 7569 6c65 6769  ulea-vs-Aquilegi
+00013090: 615f 636f 6572 756c 6561 5f4b 732e 646f  a_coerulea_Ks.do
+000130a0: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
+000130b0: 0a0a 4173 2073 686f 776e 2061 626f 7665  ..As shown above
+000130c0: 2c20 7468 6520 646f 7420 706c 6f74 2069  , the dot plot i
+000130d0: 6e20 7468 6520 756e 6974 206f 6620 6765  n the unit of ge
+000130e0: 6e65 7320 7072 6573 656e 7473 206e 756d  nes presents num
+000130f0: 6572 6f75 7320 6465 6e73 656c 7920 6167  erous densely ag
+00013100: 6772 6567 6174 6564 2028 6c69 6e65 2d6c  gregated (line-l
+00013110: 696b 6529 2061 6e63 686f 7220 706f 696e  ike) anchor poin
+00013120: 7473 2061 7420 6d6f 7374 206f 6620 7468  ts at most of th
+00013130: 6520 6368 726f 6d6f 736f 6d65 7320 7769  e chromosomes wi
+00013140: 7468 2063 6f6e 7369 7374 656e 7420 2a4b  th consistent *K
+00013150: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
+00013160: 6520 6265 7477 6565 6e20 3120 616e 6420  e between 1 and 
+00013170: 322e 2054 6865 2064 6f74 2070 6c6f 7420  2. The dot plot 
+00013180: 696e 2074 6865 2075 6e69 7420 6f66 2062  in the unit of b
+00013190: 6173 6573 2073 686f 7773 2074 6865 2073  ases shows the s
+000131a0: 616d 6520 7061 7474 6572 6e2c 2061 7320  ame pattern, as 
+000131b0: 6d61 6e69 6665 7374 6564 2062 656c 6f77  manifested below
+000131c0: 2e0a 0a21 5b5d 2864 6174 612f 7379 6e5f  ...![](data/syn_
+000131d0: 7265 7375 6c74 732f 4171 7569 6c65 6769  results/Aquilegi
+000131e0: 615f 636f 6572 756c 6561 2d76 732d 4171  a_coerulea-vs-Aq
+000131f0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00013200: 5f4b 732e 646f 742e 706e 6729 0a0a 5468  _Ks.dot.png)..Th
+00013210: 6520 646f 7420 706c 6f74 7320 7769 7468  e dot plots with
+00013220: 6f75 7420 2a4b 2a3c 7375 623e 533c 2f73  out *K*<sub>S</s
+00013230: 7562 3e20 616e 6e6f 7461 7469 6f6e 2077  ub> annotation w
+00013240: 696c 6c20 616c 736f 2062 6520 6175 746f  ill also be auto
+00013250: 6d61 7465 6c79 2070 726f 6475 6365 642c  mately produced,
+00013260: 2061 7320 7368 6f77 6e20 6265 6c6f 772e   as shown below.
+00013270: 0a0a 215b 5d28 6461 7461 2f73 796e 5f72  ..![](data/syn_r
+00013280: 6573 756c 7473 2f41 7175 696c 6567 6961  esults/Aquilegia
+00013290: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
+000132a0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+000132b0: 646f 745f 756e 6974 5f67 656e 652e 706e  dot_unit_gene.pn
+000132c0: 6729 0a0a 215b 5d28 6461 7461 2f73 796e  g)..![](data/syn
+000132d0: 5f72 6573 756c 7473 2f41 7175 696c 6567  _results/Aquileg
+000132e0: 6961 5f63 6f65 7275 6c65 612d 7673 2d41  ia_coerulea-vs-A
+000132f0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00013300: 612e 646f 742e 706e 6729 0a0a 4e6f 7465  a.dot.png)..Note
+00013310: 2074 6861 7420 7468 6520 6f70 6163 6974   that the opacit
+00013320: 7920 6f66 2061 6e63 686f 7220 646f 7473  y of anchor dots
+00013330: 2061 6e64 2061 6c6c 2068 6f6d 6f6c 6f67   and all homolog
+00013340: 2064 6f74 7320 6361 6e20 6265 2073 6574   dots can be set
+00013350: 2062 7920 7468 6520 6f70 7469 6f6e 2060   by the option `
+00013360: 2d2d 6170 616c 7068 6160 2061 6e64 2060  --apalpha` and `
+00013370: 2d2d 686f 616c 7068 6160 2073 6570 6172  --hoalpha` separ
+00013380: 6174 656c 792e 2049 6620 6f6e 6520 6a75  ately. If one ju
+00013390: 7374 2077 616e 7473 2074 6f20 7365 6520  st wants to see 
+000133a0: 7468 6520 616e 6368 6f72 2064 6f74 732c  the anchor dots,
+000133b0: 2073 6574 7469 6e67 2074 6865 2060 686f   setting the `ho
+000133c0: 616c 7068 6160 2061 7320 3020 286f 7220  alpha` as 0 (or 
+000133d0: 6f74 6865 7220 6d69 6e75 7363 756c 6520  other minuscule 
+000133e0: 7661 6c75 6573 2920 7769 6c6c 2064 6f2e  values) will do.
+000133f0: 2049 6620 6f6e 6520 7761 6e74 7320 746f   If one wants to
+00013400: 2073 6565 2074 6865 2064 6973 7472 6962   see the distrib
+00013410: 7574 696f 6e20 6f66 2077 686f 6c65 2064  ution of whole d
+00013420: 6f74 7320 6265 7474 6572 2c20 7365 7474  ots better, sett
+00013430: 696e 6720 7468 6520 6068 6f61 6c70 6861  ing the `hoalpha
+00013440: 6020 6869 6768 6572 2028 616e 6420 6061  ` higher (and `a
+00013450: 7061 6c70 6861 6020 6c6f 7765 7229 2077  palpha` lower) w
+00013460: 696c 6c20 646f 2e20 5468 6520 6064 6f74  ill do. The `dot
+00013470: 7369 7a65 6020 6f70 7469 6f6e 2063 616e  size` option can
+00013480: 2062 6520 6361 6c6c 6564 2074 6f20 6164   be called to ad
+00013490: 6a75 7374 2074 6865 2073 697a 6520 6f66  just the size of
+000134a0: 2064 6f74 732e 0a0a 4120 6675 7274 6865   dots...A furthe
+000134b0: 7220 6173 736f 6369 6174 6564 2053 796e  r associated Syn
+000134c0: 6465 7074 6820 706c 6f74 2073 686f 7773  depth plot shows
+000134d0: 2074 6861 7420 7468 6572 6520 6172 6520   that there are 
+000134e0: 6d6f 7265 2074 6861 6e20 3530 2064 7570  more than 50 dup
+000134f0: 6c69 6361 7465 6420 7365 676d 656e 7473  licated segments
+00013500: 206c 6f6e 6765 7220 7468 616e 2031 3030   longer than 100
+00013510: 3030 2062 7020 616e 6420 3330 2067 656e  00 bp and 30 gen
+00013520: 6573 2028 736f 2061 7320 746f 2064 726f  es (so as to dro
+00013530: 7020 6672 6167 6d65 6e74 6172 7920 7365  p fragmentary se
+00013540: 676d 656e 7473 292c 2077 6869 6368 2064  gments), which d
+00013550: 6f6d 696e 6174 6573 2074 6865 2077 686f  ominates the who
+00013560: 6c65 2063 6f6c 6c69 6e65 6172 2072 6174  le collinear rat
+00013570: 696f 2063 6174 6567 6f72 792e 0a0a 215b  io category...![
+00013580: 5d28 6461 7461 2f73 796e 5f72 6573 756c  ](data/syn_resul
+00013590: 7473 2f53 796e 6465 7074 682e 7376 6729  ts/Syndepth.svg)
+000135a0: 0a0a 4d6f 7265 2065 7871 7569 7369 7465  ..More exquisite
+000135b0: 2070 6c6f 7473 2069 6e63 6c75 6469 6e67   plots including
+000135c0: 2062 6f74 6820 696e 7472 612d 7370 6563   both intra-spec
+000135d0: 6966 6963 2061 6e64 2069 6e74 6572 2d73  ific and inter-s
+000135e0: 7065 6369 6669 6320 636f 6d70 6172 6973  pecific comparis
+000135f0: 6f6e 7320 7573 696e 6720 7468 6520 6f72  ons using the or
+00013600: 7468 6f67 726f 7570 7320 2863 6f6d 706f  thogroups (compo
+00013610: 7365 6420 6f66 202a 4171 7569 6c65 6769  sed of *Aquilegi
+00013620: 6120 636f 6572 756c 6561 2a2c 202a 5072  a coerulea*, *Pr
+00013630: 6f74 6561 2063 796e 6172 6f69 6465 732a  otea cynaroides*
+00013640: 2c20 2a41 636f 7275 7320 616d 6572 6963  , *Acorus americ
+00013650: 616e 7573 2a20 616e 6420 2a56 6974 6973  anus* and *Vitis
+00013660: 2076 696e 6966 6572 612a 2c20 7365 6520   vinifera*, see 
+00013670: 6675 7274 6865 7220 666f 7220 7468 6520  further for the 
+00013680: 636f 6e74 6578 7429 2069 6e66 6572 7265  context) inferre
+00013690: 6420 6865 7265 696e 6166 7465 7220 6361  d hereinafter ca
+000136a0: 6e20 6265 2061 6c73 6f20 7072 6f64 7563  n be also produc
+000136b0: 6564 2075 7369 6e67 2060 7767 6420 7379  ed using `wgd sy
+000136c0: 6e60 2e20 4e6f 7465 2074 6861 7420 6469  n`. Note that di
+000136d0: 6666 6572 656e 7420 6765 6e6f 6d65 2061  fferent genome a
+000136e0: 7373 656d 626c 6965 7320 6d69 6768 7420  ssemblies might 
+000136f0: 6861 7665 2064 6966 6665 7265 6e74 2066  have different f
+00013700: 6561 7475 7265 7320 616e 6420 6174 7472  eatures and attr
+00013710: 6962 7574 6573 2077 6869 6368 2063 616e  ibutes which can
+00013720: 2062 6520 6163 636f 6d6d 6f64 6174 6564   be accommodated
+00013730: 2076 6961 2074 6865 206f 7074 696f 6e20   via the option 
+00013740: 602d 2d61 6464 6974 696f 6e61 6c67 6666  `--additionalgff
+00013750: 696e 666f 6020 666f 7220 6561 6368 2067  info` for each g
+00013760: 656e 6f6d 6520 6173 7365 6d62 6c79 2077  enome assembly w
+00013770: 686f 7365 206f 7264 6572 206e 6565 6473  hose order needs
+00013780: 2074 6f20 666f 6c6c 6f77 2074 6865 206f   to follow the o
+00013790: 7264 6572 206f 6620 6766 6633 2066 696c  rder of gff3 fil
+000137a0: 6573 2c20 666f 7220 696e 7374 616e 6365  es, for instance
+000137b0: 2027 6d4e 5241 3b4e 616d 6527 2066 6f72   'mNRA;Name' for
+000137c0: 2060 4171 7569 6c65 6769 615f 636f 6572   `Aquilegia_coer
+000137d0: 756c 6561 2e67 6666 3360 2c20 276d 4e52  ulea.gff3`, 'mNR
+000137e0: 413b 4944 2720 666f 7220 6050 726f 7465  A;ID' for `Prote
+000137f0: 615f 6379 6e61 726f 6964 6573 2e67 6666  a_cynaroides.gff
+00013800: 3360 2c20 276d 4e52 413b 4e61 6d65 2720  3`, 'mNRA;Name' 
+00013810: 666f 7220 6041 636f 7275 735f 616d 6572  for `Acorus_amer
+00013820: 6963 616e 7573 2e67 6666 3360 2061 6e64  icanus.gff3` and
+00013830: 2027 6d4e 5241 3b4e 616d 6527 2066 6f72   'mNRA;Name' for
+00013840: 2060 5669 7469 735f 7669 6e69 6665 7261   `Vitis_vinifera
+00013850: 2e67 6666 3360 2e0a 0a60 6060 0a77 6764  .gff3`...```.wgd
+00013860: 2073 796e 2077 6764 5f6f 7274 686f 2f4f   syn wgd_ortho/O
+00013870: 7274 686f 6772 6f75 7073 2e73 702e 7473  rthogroups.sp.ts
+00013880: 7620 2d6b 7320 7767 645f 6f72 7468 6f5f  v -ks wgd_ortho_
+00013890: 6b73 2f4f 7274 686f 6772 6f75 7073 2e73  ks/Orthogroups.s
+000138a0: 702e 7473 762e 6b73 2e74 7376 2041 7175  p.tsv.ks.tsv Aqu
+000138b0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+000138c0: 6766 6633 202d 2d61 6464 6974 696f 6e61  gff3 --additiona
+000138d0: 6c67 6666 696e 666f 2027 6d4e 5241 3b4e  lgffinfo 'mNRA;N
+000138e0: 616d 6527 2050 726f 7465 615f 6379 6e61  ame' Protea_cyna
+000138f0: 726f 6964 6573 2e67 6666 3320 2d2d 6164  roides.gff3 --ad
+00013900: 6469 7469 6f6e 616c 6766 6669 6e66 6f20  ditionalgffinfo 
+00013910: 276d 4e52 413b 4944 2720 4163 6f72 7573  'mNRA;ID' Acorus
+00013920: 5f61 6d65 7269 6361 6e75 732e 6766 6633  _americanus.gff3
+00013930: 202d 2d61 6464 6974 696f 6e61 6c67 6666   --additionalgff
+00013940: 696e 666f 2027 6d4e 5241 3b4e 616d 6527  info 'mNRA;Name'
+00013950: 2056 6974 6973 5f76 696e 6966 6572 612e   Vitis_vinifera.
+00013960: 6766 6633 202d 2d61 6464 6974 696f 6e61  gff3 --additiona
+00013970: 6c67 6666 696e 666f 2027 6d4e 5241 3b4e  lgffinfo 'mNRA;N
+00013980: 616d 6527 202d 6f20 7767 645f 6f72 7468  ame' -o wgd_orth
+00013990: 6f5f 7379 6e0a 6060 600a 0a55 706f 6e20  o_syn.```..Upon 
+000139a0: 7468 6520 6163 7175 6973 6974 696f 6e20  the acquisition 
+000139b0: 6f66 2074 6865 2063 6f6c 6c69 6e65 6172  of the collinear
+000139c0: 2072 6573 756c 7473 2075 7369 6e67 2060   results using `
+000139d0: 7767 6420 7379 6e60 2c20 7468 6520 7361  wgd syn`, the sa
+000139e0: 6d65 2063 6f6c 6c69 6e65 6172 2070 6c6f  me collinear plo
+000139f0: 7473 2063 616e 2062 6520 616c 736f 2070  ts can be also p
+00013a00: 726f 6475 6365 6420 6279 2060 7767 6420  roduced by `wgd 
+00013a10: 7669 7a60 2075 7369 6e67 2074 6865 2063  viz` using the c
+00013a20: 6f6d 6d61 6e64 2062 656c 6f77 2e0a 6060  ommand below..``
+00013a30: 600a 7767 6420 7669 7a20 2d2d 706c 6f74  `.wgd viz --plot
+00013a40: 7379 6e20 2d73 6d20 7767 645f 6f72 7468  syn -sm wgd_orth
+00013a50: 6f5f 7379 6e2f 6961 6468 6f72 652d 6f75  o_syn/iadhore-ou
+00013a60: 742f 7365 676d 656e 7473 2e74 7874 202d  t/segments.txt -
+00013a70: 6170 2077 6764 5f6f 7274 686f 5f73 796e  ap wgd_ortho_syn
+00013a80: 2f69 6164 686f 7265 2d6f 7574 2f61 6e63  /iadhore-out/anc
+00013a90: 686f 7270 6f69 6e74 732e 7478 7420 2d6d  horpoints.txt -m
+00013aa0: 7420 7767 645f 6f72 7468 6f5f 7379 6e2f  t wgd_ortho_syn/
+00013ab0: 6961 6468 6f72 652d 6f75 742f 6d75 6c74  iadhore-out/mult
+00013ac0: 6970 6c69 636f 6e73 2e74 7874 202d 6774  iplicons.txt -gt
+00013ad0: 2077 6764 5f6f 7274 686f 5f73 796e 2f67   wgd_ortho_syn/g
+00013ae0: 656e 652d 7461 626c 652e 6373 7620 2d64  ene-table.csv -d
+00013af0: 2077 6764 5f6f 7274 686f 5f6b 732f 4f72   wgd_ortho_ks/Or
+00013b00: 7468 6f67 726f 7570 732e 7370 2e74 7376  thogroups.sp.tsv
+00013b10: 2e6b 732e 7473 7620 2d6f 2077 6764 5f6f  .ks.tsv -o wgd_o
+00013b20: 7274 686f 5f76 697a 0a60 6060 0a0a 215b  rtho_viz.```..![
+00013b30: 5d28 6461 7461 2f6f 7274 686f 5f73 796e  ](data/ortho_syn
+00013b40: 5f72 6573 756c 7473 2f41 7175 696c 6567  _results/Aquileg
+00013b50: 6961 5f63 6f65 7275 6c65 615f 5669 7469  ia_coerulea_Viti
+00013b60: 735f 7669 6e69 6665 7261 5f6d 756c 7469  s_vinifera_multi
+00013b70: 706c 6963 6f6e 735f 6c65 7665 6c2e 706e  plicons_level.pn
+00013b80: 6729 0a54 6865 2061 626f 7665 2060 6475  g).The above `du
+00013b90: 7053 7461 636b 6020 706c 6f74 2073 686f  pStack` plot sho
+00013ba0: 7773 2074 6865 2064 6973 7472 6962 7574  ws the distribut
+00013bb0: 696f 6e20 6f66 2064 7570 6c69 6361 7465  ion of duplicate
+00013bc0: 6420 7365 676d 656e 7473 206f 6620 2a41  d segments of *A
+00013bd0: 7175 696c 6567 6961 2063 6f65 7275 6c65  quilegia coerule
+00013be0: 612a 2063 6f6d 7061 7265 6420 746f 2069  a* compared to i
+00013bf0: 7473 656c 6620 2869 6e20 6772 6565 6e29  tself (in green)
+00013c00: 2061 6e64 2063 6f6d 7061 7265 6420 746f   and compared to
+00013c10: 202a 5669 7469 7320 7669 6e69 6665 7261   *Vitis vinifera
+00013c20: 2a20 2869 6e20 626c 7565 2920 6f76 6572  * (in blue) over
+00013c30: 2074 6865 2063 6872 6f6d 6f73 6f6d 6573   the chromosomes
+00013c40: 206f 6620 2a41 2e20 636f 6572 756c 6561   of *A. coerulea
+00013c50: 2a2e 0a21 5b5d 2864 6174 612f 6f72 7468  *..![](data/orth
+00013c60: 6f5f 7379 6e5f 7265 7375 6c74 732f 4f76  o_syn_results/Ov
+00013c70: 6572 616c 6c73 7065 6369 6573 5f4b 732e  erallspecies_Ks.
+00013c80: 646f 745f 756e 6974 5f67 656e 652e 706e  dot_unit_gene.pn
+00013c90: 6729 0a54 6865 2061 626f 7665 202a 4b2a  g).The above *K*
+00013ca0: 3c73 7562 3e53 3c2f 7375 623e 2064 6f74  <sub>S</sub> dot
+00013cb0: 706c 6f74 2069 6e20 756e 6974 206f 6620  plot in unit of 
+00013cc0: 6765 6e65 2073 686f 7773 2074 6865 206f  gene shows the o
+00013cd0: 7665 7261 6c6c 2064 6973 7472 6962 7574  verall distribut
+00013ce0: 696f 6e20 6f66 2063 6f6c 6c69 6e65 6172  ion of collinear
+00013cf0: 6974 7920 6163 6f72 7373 2074 6865 2066  ity acorss the f
+00013d00: 6f75 7220 7370 6563 6965 7320 696e 766f  our species invo
+00013d10: 6c76 6564 2e0a 215b 5d28 6461 7461 2f6f  lved..![](data/o
+00013d20: 7274 686f 5f73 796e 5f72 6573 756c 7473  rtho_syn_results
+00013d30: 2f4f 7665 7261 6c6c 7370 6563 6965 732e  /Overallspecies.
+00013d40: 646f 745f 756e 6974 5f67 656e 652e 706e  dot_unit_gene.pn
+00013d50: 6729 0a54 6865 2061 626f 7665 2064 6f74  g).The above dot
+00013d60: 706c 6f74 2069 7320 7769 7468 6f75 7420  plot is without 
+00013d70: 7468 6520 616e 6e6f 7461 7469 6f6e 206f  the annotation o
+00013d80: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
+00013d90: 3e20 6167 6573 2063 6f6d 7061 7265 6420  > ages compared 
+00013da0: 746f 2074 6865 206c 6173 7420 6f6e 652e  to the last one.
+00013db0: 0a21 5b5d 2864 6174 612f 6f72 7468 6f5f  .![](data/ortho_
+00013dc0: 7379 6e5f 7265 7375 6c74 732f 5379 6e64  syn_results/Synd
+00013dd0: 6570 7468 2e73 7667 290a 5468 6520 6162  epth.svg).The ab
+00013de0: 6f76 6520 5379 6e64 6570 7468 2070 6c6f  ove Syndepth plo
+00013df0: 7420 7368 6f77 7320 7468 6520 636f 6c6c  t shows the coll
+00013e00: 696e 6561 7220 7261 7469 6f20 6163 6f72  inear ratio acor
+00013e10: 7373 2061 6c6c 2073 7065 6369 6573 2070  ss all species p
+00013e20: 6169 7273 2028 696e 7472 612d 7370 6563  airs (intra-spec
+00013e30: 6966 6963 2063 6f6d 7061 7269 736f 6e20  ific comparison 
+00013e40: 696e 2067 7265 656e 2077 6869 6c65 2069  in green while i
+00013e50: 6e74 6572 2d73 7065 6369 6669 6320 636f  nter-specific co
+00013e60: 6d70 6172 6973 6f6e 2069 6e20 626c 7565  mparison in blue
+00013e70: 292e 0a0a 5765 2063 616e 2066 6974 2061  )...We can fit a
+00013e80: 6e20 454c 4d4d 206d 6978 7475 7265 206d  n ELMM mixture m
+00013e90: 6f64 656c 2075 706f 6e20 7468 6520 7768  odel upon the wh
+00013ea0: 6f6c 6520 7061 7261 6e6f 6d65 202a 4b2a  ole paranome *K*
+00013eb0: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+00013ec0: 2064 6973 7472 6962 7574 696f 6e20 746f   distribution to
+00013ed0: 2073 6565 206d 6f72 6520 6163 6375 7261   see more accura
+00013ee0: 7465 6c79 2074 6865 2073 6967 6e69 6669  tely the signifi
+00013ef0: 6361 6e63 6520 616e 6420 6c6f 6361 7469  cance and locati
+00013f00: 6f6e 206f 6620 706f 7465 6e74 6961 6c20  on of potential 
+00013f10: 5747 4473 2c20 7573 696e 6720 7468 6520  WGDs, using the 
+00013f20: 636f 6d6d 616e 6420 6c69 6e65 2062 656c  command line bel
+00013f30: 6f77 2e0a 0a60 6060 0a77 6764 2076 697a  ow...```.wgd viz
+00013f40: 202d 6420 7767 645f 6b73 642f 4171 7569   -d wgd_ksd/Aqui
+00013f50: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+00013f60: 7376 2e6b 732e 7473 760a 6060 600a 0a54  sv.ks.tsv.```..T
+00013f70: 6865 2072 6573 756c 7420 6f66 2045 4c4d  he result of ELM
+00013f80: 4d20 6d69 7874 7572 6520 6d6f 6465 6c20  M mixture model 
+00013f90: 636c 7573 7465 7269 6e67 2073 686f 7773  clustering shows
+00013fa0: 2074 6861 7420 7468 6572 6520 6973 2061   that there is a
+00013fb0: 206c 696b 656c 7920 5747 4420 636f 6d70   likely WGD comp
+00013fc0: 6f6e 656e 7420 6174 202a 4b2a 3c73 7562  onent at *K*<sub
+00013fd0: 3e53 3c2f 7375 623e 2031 2e31 392e 0a0a  >S</sub> 1.19...
+00013fe0: 215b 5d28 6461 7461 2f65 6c6d 6d5f 4171  ![](data/elmm_Aq
+00013ff0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00014000: 2e74 7376 2e6b 732e 7473 765f 6265 7374  .tsv.ks.tsv_best
+00014010: 5f6d 6f64 656c 735f 7765 6967 6874 6564  _models_weighted
+00014020: 2e73 7667 290a 0a4c 6574 2773 2064 6f20  .svg)..Let's do 
+00014030: 6120 6d69 7874 7572 6520 6d6f 6465 6c20  a mixture model 
+00014040: 636c 7573 7465 7269 6e67 2066 6f72 2061  clustering for a
+00014050: 6e63 686f 7220 2a4b 2a3c 7375 623e 533c  nchor *K*<sub>S<
+00014060: 2f73 7562 3e20 746f 6f2c 2075 7369 6e67  /sub> too, using
+00014070: 2074 6865 2063 6f6d 6d61 6e64 206c 696e   the command lin
+00014080: 6520 6265 6c6f 772e 204e 6f74 6520 7468  e below. Note th
+00014090: 6174 2074 6869 7320 7374 6570 2077 696c  at this step wil
+000140a0: 6c20 6175 746f 6d61 7465 6c79 2063 616c  l automately cal
+000140b0: 6c20 7468 6520 7365 676d 656e 7420 2a4b  l the segment *K
+000140c0: 2a3c 7375 623e 533c 2f73 7562 3e20 636c  *<sub>S</sub> cl
+000140d0: 7573 7465 7269 6e67 2061 6e61 6c79 7369  ustering analysi
+000140e0: 7320 746f 6f2e 0a0a 6060 600a 7767 6420  s too...```.wgd 
+000140f0: 7065 616b 2077 6764 5f6b 7364 2f41 7175  peak wgd_ksd/Aqu
+00014100: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+00014110: 7473 762e 6b73 2e74 7376 202d 2d61 6e63  tsv.ks.tsv --anc
+00014120: 686f 7270 6f69 6e74 7320 7767 645f 7379  horpoints wgd_sy
+00014130: 6e2f 6961 6468 6f72 652d 6f75 742f 616e  n/iadhore-out/an
+00014140: 6368 6f72 706f 696e 7473 2e74 7874 202d  chorpoints.txt -
+00014150: 2d73 6567 6d65 6e74 7320 7767 645f 7379  -segments wgd_sy
+00014160: 6e2f 6961 6468 6f72 652d 6f75 742f 7365  n/iadhore-out/se
+00014170: 676d 656e 7473 2e74 7874 202d 2d6c 6973  gments.txt --lis
+00014180: 7465 6c65 6d65 6e74 7320 7767 645f 7379  telements wgd_sy
+00014190: 6e2f 6961 6468 6f72 652d 6f75 742f 6c69  n/iadhore-out/li
+000141a0: 7374 5f65 6c65 6d65 6e74 732e 7478 7420  st_elements.txt 
+000141b0: 2d2d 6d75 6c74 6970 6c69 636f 6e70 6169  --multipliconpai
+000141c0: 7273 2077 6764 5f73 796e 2f69 6164 686f  rs wgd_syn/iadho
+000141d0: 7265 2d6f 7574 2f6d 756c 7469 706c 6963  re-out/multiplic
+000141e0: 6f6e 5f70 6169 7273 2e74 7874 2028 2d2d  on_pairs.txt (--
+000141f0: 7765 6967 6874 6564 290a 6060 600a 0a54  weighted).```..T
+00014200: 6865 2061 6e63 686f 7220 2a4b 2a3c 7375  he anchor *K*<su
+00014210: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
+00014220: 7374 7269 6275 7469 6f6e 2061 6c73 6f20  stribution also 
+00014230: 6861 7320 6120 6c69 6b65 6c79 2057 4744  has a likely WGD
+00014240: 2063 6f6d 706f 6e65 6e74 2077 6974 6820   component with 
+00014250: 6d6f 6465 2031 2e32 382e 0a0a 215b 5d28  mode 1.28...![](
+00014260: 6461 7461 2f4f 7269 6769 6e61 6c5f 416e  data/Original_An
+00014270: 6368 6f72 4b73 5f47 4d4d 5f43 6f6d 706f  chorKs_GMM_Compo
+00014280: 6e65 6e74 335f 6e6f 6465 5f61 7665 7261  nent3_node_avera
+00014290: 6765 645f 4c6f 676e 6f72 6d61 6c2e 7376  ged_Lognormal.sv
+000142a0: 6729 0a0a 4e6f 7720 7468 6174 2077 6520  g)..Now that we 
+000142b0: 6861 7665 2073 6565 6e20 7468 6520 6576  have seen the ev
+000142c0: 6964 656e 6365 206f 6620 6e75 6d65 726f  idence of numero
+000142d0: 7573 2064 7570 6c69 6361 7465 6420 7365  us duplicated se
+000142e0: 676d 656e 7473 2061 6e64 2074 6865 2061  gments and the a
+000142f0: 6767 7265 6761 7469 6f6e 206f 6620 6475  ggregation of du
+00014300: 706c 6963 6174 6573 2061 6765 2061 7420  plicates age at 
+00014310: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00014320: 312e 3238 206f 7220 312e 3139 2066 6f72  1.28 or 1.19 for
+00014330: 2061 6e63 686f 7220 7061 6972 7320 616e   anchor pairs an
+00014340: 6420 6e6f 6e2d 616e 6368 6f72 2070 6169  d non-anchor pai
+00014350: 7273 2074 6872 6f75 6768 6f75 7420 7468  rs throughout th
+00014360: 6520 7768 6f6c 6520 6765 6e6f 6d65 2e20  e whole genome. 
+00014370: 5765 2063 616e 2063 6c61 696d 2077 6974  We can claim wit
+00014380: 6820 736f 6d65 2063 6f6e 6669 6465 6e63  h some confidenc
+00014390: 6520 7468 6174 202a 4171 7569 6c65 6769  e that *Aquilegi
+000143a0: 6120 636f 6572 756c 6561 2a20 6d69 6768  a coerulea* migh
+000143b0: 7420 6861 7665 2065 7870 6572 6965 6e63  t have experienc
+000143c0: 6564 2061 2070 616c 656f 2d70 6f6c 7970  ed a paleo-polyp
+000143d0: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
+000143e0: 742e 204e 6578 742c 204c 6574 2773 2068  t. Next, Let's h
+000143f0: 6176 6520 6120 6675 7274 6865 7220 6c6f  ave a further lo
+00014400: 6f6b 2061 626f 7574 2069 7473 2070 6879  ok about its phy
+00014410: 6c6f 6765 6e65 7469 6320 6c6f 6361 7469  logenetic locati
+00014420: 6f6e 2e20 5765 206b 6e6f 7720 7468 6174  on. We know that
+00014430: 2074 6865 7265 2061 7265 2075 6e63 6572   there are uncer
+00014440: 7461 696e 7469 6573 2061 626f 7574 2077  tainties about w
+00014450: 6865 7468 6572 2074 6869 7320 7075 7461  hether this puta
+00014460: 7469 7665 2070 616c 656f 2d70 6f6c 7970  tive paleo-polyp
+00014470: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
+00014480: 7420 6973 2073 6861 7265 6420 7769 7468  t is shared with
+00014490: 2061 6c6c 2065 7564 6963 6f74 7320 6f72   all eudicots or
+000144a0: 206e 6f74 2e20 5765 2063 616e 2063 686f   not. We can cho
+000144b0: 6f73 6520 736f 6d65 206f 7468 6572 2065  ose some other e
+000144c0: 7564 6963 6f74 2067 656e 6f6d 6573 2074  udicot genomes t
+000144d0: 6f20 7365 6520 7468 6520 6f72 6465 7269  o see the orderi
+000144e0: 6e67 206f 6620 7370 6563 6961 7469 6f6e  ng of speciation
+000144f0: 2061 6e64 2070 6f6c 7970 6c6f 6964 697a   and polyploidiz
+00014500: 6174 696f 6e20 6576 656e 7473 2e20 4865  ation events. He
+00014510: 7265 2077 6520 6368 6f6f 7365 202a 5669  re we choose *Vi
+00014520: 7469 7320 7669 6e69 6665 7261 2a2c 202a  tis vinifera*, *
+00014530: 5072 6f74 6561 2063 796e 6172 6f69 6465  Protea cynaroide
+00014540: 732a 2061 6e64 202a 4163 6f72 7573 2061  s* and *Acorus a
+00014550: 6d65 7269 6361 6e75 732a 2069 6e20 7468  mericanus* in th
+00014560: 6520 666f 6c6c 6f77 696e 6720 2a4b 2a3c  e following *K*<
+00014570: 7375 623e 533c 2f73 7562 3e20 616e 616c  sub>S</sub> anal
+00014580: 7973 6973 2e20 4669 7273 742c 2077 6520  ysis. First, we 
+00014590: 6275 696c 7420 6120 676c 6f62 616c 204d  built a global M
+000145a0: 5242 4820 6661 6d69 6c79 2075 7369 6e67  RBH family using
+000145b0: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+000145c0: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
+000145d0: 202d 2d67 6c6f 6261 6c6d 7262 6820 4171   --globalmrbh Aq
+000145e0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000145f0: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
+00014600: 6573 2041 636f 7275 735f 616d 6572 6963  es Acorus_americ
+00014610: 616e 7573 2056 6974 6973 5f76 696e 6966  anus Vitis_vinif
+00014620: 6572 6120 2d6f 2077 6764 5f67 6c6f 6261  era -o wgd_globa
+00014630: 6c6d 7262 680a 6060 600a 0a49 6e20 7468  lmrbh.```..In th
+00014640: 6520 676c 6f62 616c 204d 5242 4820 6661  e global MRBH fa
+00014650: 6d69 6c79 2c20 6576 6572 7920 7061 6972  mily, every pair
+00014660: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
+00014670: 6765 6e65 7320 6973 2074 6865 2072 6563  genes is the rec
+00014680: 6970 726f 6361 6c20 6265 7374 2068 6974  iprocal best hit
+00014690: 2c20 7375 6767 6573 7469 6e67 2074 7275  , suggesting tru
+000146a0: 6520 6f72 7468 6f6c 6f67 6f75 7320 7265  e orthologous re
+000146b0: 6c61 7469 6f6e 7368 6970 732e 2057 6520  lationships. We 
+000146c0: 776f 756c 6420 7573 6520 7468 6520 2a4b  would use the *K
+000146d0: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
+000146e0: 6c75 6573 2061 7373 6f63 6961 7465 6420  lues associated 
+000146f0: 7769 7468 2074 6865 7365 206f 7274 686f  with these ortho
+00014700: 6c6f 676f 7573 2070 6169 7273 2074 6f20  logous pairs to 
+00014710: 6465 6c69 6d69 7420 7468 6520 6469 7665  delimit the dive
+00014720: 7267 656e 6365 202a 4b2a 3c73 7562 3e53  rgence *K*<sub>S
+00014730: 3c2f 7375 623e 2070 6561 6b2e 2054 6f67  </sub> peak. Tog
+00014740: 6574 6865 7220 7769 7468 2074 6865 2077  ether with the w
+00014750: 686f 6c65 2070 6172 616e 6f6d 6520 2a4b  hole paranome *K
+00014760: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+00014770: 7374 7269 6275 7469 6f6e 2c20 7765 2063  stribution, we c
+00014780: 6f6e 6475 6374 2074 6865 2072 6174 6520  onduct the rate 
+00014790: 636f 7272 6563 7469 6f6e 2075 7369 6e67  correction using
+000147a0: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+000147b0: 6f77 2e0a 0a21 2153 696e 6365 2060 7767  ow...!!Since `wg
+000147c0: 6460 2076 6572 7369 6f6e 2032 2e30 2e32  d` version 2.0.2
+000147d0: 342c 2077 6520 7265 7772 6f74 6520 6120  4, we rewrote a 
+000147e0: 636c 6561 6e65 7220 616e 6420 7175 6963  cleaner and quic
+000147f0: 6b65 7220 7761 7920 6f66 2064 6f69 6e67  ker way of doing
+00014800: 2073 7562 7374 6974 7574 696f 6e20 7261   substitution ra
+00014810: 7465 2063 6f72 7265 6374 696f 6e2e 2049  te correction. I
+00014820: 7427 7320 6e6f 7420 7265 7175 6972 6564  t's not required
+00014830: 2074 6f20 7479 7065 2069 6e20 616e 7920   to type in any 
+00014840: 7370 6569 6365 7320 7061 6972 2061 6e64  speices pair and
+00014850: 2061 2073 6572 6965 7320 6f66 202a 4b2a   a series of *K*
+00014860: 3c73 7562 3e53 3c2f 7375 623e 2070 6c6f  <sub>S</sub> plo
+00014870: 7473 2077 696c 6c20 6265 2070 726f 6475  ts will be produ
+00014880: 6365 642e 2054 6865 2072 6571 7569 7265  ced. The require
+00014890: 6420 6669 6c65 7320 6172 6520 6f72 7468  d files are orth
+000148a0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+000148b0: 533c 2f73 7562 3e20 6461 7461 6669 6c65  S</sub> datafile
+000148c0: 2c20 7061 7261 6c6f 676f 7573 202a 4b2a  , paralogous *K*
+000148d0: 3c73 7562 3e53 3c2f 7375 623e 2064 6174  <sub>S</sub> dat
+000148e0: 6166 696c 652c 2061 2073 7065 6369 6573  afile, a species
+000148f0: 2074 7265 6520 616e 6420 6120 666f 6361   tree and a foca
+00014900: 6c20 7370 6563 6965 7320 2874 6865 206f  l species (the o
+00014910: 6e65 2069 6e70 7574 7465 6420 7769 7468  ne inputted with
+00014920: 2070 6172 616c 6f67 6f75 7320 2a4b 2a3c   paralogous *K*<
+00014930: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
+00014940: 292e 2055 7365 7273 2063 616e 2063 686f  ). Users can cho
+00014950: 6f73 6520 746f 2061 6464 206f 6e65 206d  ose to add one m
+00014960: 6f72 6520 6c61 7965 7220 6f66 2045 4c4d  ore layer of ELM
+00014970: 4d20 616e 616c 7973 6973 206f 6e20 7061  M analysis on pa
+00014980: 7261 6c6f 676f 7573 202a 4b2a 3c73 7562  ralogous *K*<sub
+00014990: 3e53 3c2f 7375 623e 2076 616c 7565 7320  >S</sub> values 
+000149a0: 616e 642f 6f72 2047 4d4d 2061 6e61 6c79  and/or GMM analy
+000149b0: 7369 7320 6f6e 2061 6e63 686f 7220 2a4b  sis on anchor *K
+000149c0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+000149d0: 7374 7269 6275 7469 6f6e 2e20 5468 6520  stribution. The 
+000149e0: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
+000149f0: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00014a00: 7269 6275 7469 6f6e 2063 616e 2062 6520  ribution can be 
+00014a10: 6361 6c63 756c 6174 6564 2075 7369 6e67  calculated using
+00014a20: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+00014a30: 6f77 2e0a 0a60 6060 0a77 6764 206b 7364  ow...```.wgd ksd
+00014a40: 2077 6764 5f67 6c6f 6261 6c6d 7262 682f   wgd_globalmrbh/
+00014a50: 676c 6f62 616c 5f4d 5242 482e 7473 7620  global_MRBH.tsv 
+00014a60: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00014a70: 6561 2050 726f 7465 615f 6379 6e61 726f  ea Protea_cynaro
+00014a80: 6964 6573 2041 636f 7275 735f 616d 6572  ides Acorus_amer
+00014a90: 6963 616e 7573 2056 6974 6973 5f76 696e  icanus Vitis_vin
+00014aa0: 6966 6572 6120 2d6f 2077 6764 5f67 6c6f  ifera -o wgd_glo
+00014ab0: 6261 6c6d 7262 685f 6b73 0a60 6060 0a0a  balmrbh_ks.```..
+00014ac0: 5769 7468 2074 6865 2063 616c 6375 6c61  With the calcula
+00014ad0: 7465 6420 6f72 7468 6f6c 6f67 6f75 7320  ted orthologous 
+00014ae0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00014af0: 6469 7374 7269 6275 7469 6f6e 2c20 7765  distribution, we
+00014b00: 2063 616e 2075 7365 2074 6865 2063 6f6d   can use the com
+00014b10: 6d61 6e64 2062 656c 6f77 2074 6f20 636f  mand below to co
+00014b20: 6e64 7563 7420 7468 6520 7261 7465 2063  nduct the rate c
+00014b30: 6f72 7265 6374 696f 6e20 616e 642f 6f72  orrection and/or
+00014b40: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
+00014b50: 6720 616e 616c 7973 6973 2e0a 0a60 6060  g analysis...```
+00014b60: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
+00014b70: 676c 6f62 616c 6d72 6268 5f6b 732f 676c  globalmrbh_ks/gl
+00014b80: 6f62 616c 5f4d 5242 482e 7473 762e 6b73  obal_MRBH.tsv.ks
+00014b90: 2e74 7376 202d 6661 2041 7175 696c 6567  .tsv -fa Aquileg
+00014ba0: 6961 5f63 6f65 7275 6c65 6120 2d65 706b  ia_coerulea -epk
+00014bb0: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
+00014bc0: 6961 5f63 6f65 7275 6c65 612e 6b73 2e74  ia_coerulea.ks.t
+00014bd0: 7376 202d 6170 2077 6764 5f73 796e 2f69  sv -ap wgd_syn/i
+00014be0: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
+00014bf0: 7270 6f69 6e74 732e 7478 7420 2d73 7020  rpoints.txt -sp 
+00014c00: 7370 6563 6965 7374 7265 652e 6e77 202d  speciestree.nw -
+00014c10: 6f20 7767 645f 7669 7a5f 6d69 7865 645f  o wgd_viz_mixed_
+00014c20: 4b73 202d 2d70 6c6f 7465 6c6d 6d20 2d2d  Ks --plotelmm --
+00014c30: 706c 6f74 6170 676d 6d20 2d2d 7265 7765  plotapgmm --rewe
+00014c40: 6967 6874 0a60 6060 0a0a 6f72 2075 7369  ight.```..or usi
+00014c50: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
+00014c60: 656c 6f77 2c20 7768 6963 6820 636f 6d62  elow, which comb
+00014c70: 696e 6573 2074 6865 2074 776f 2073 7465  ines the two ste
+00014c80: 7073 2061 626f 7665 2069 6e20 6f6e 652e  ps above in one.
+00014c90: 204e 6f74 6520 7468 6174 2077 6520 7375   Note that we su
+00014ca0: 6767 6573 7420 6f66 2074 616b 696e 6720  ggest of taking 
+00014cb0: 7477 6f20 7365 7061 7261 7465 2073 7465  two separate ste
+00014cc0: 7073 2069 6e20 7768 6963 6820 6077 6764  ps in which `wgd
+00014cd0: 206b 7364 6020 756e 6465 7274 616b 6573   ksd` undertakes
+00014ce0: 2074 6865 2063 616c 6375 6c61 7469 6f6e   the calculation
+00014cf0: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
+00014d00: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00014d10: 6469 7374 7269 6275 7469 6f6e 2077 6869  distribution whi
+00014d20: 6c65 2060 7767 6420 7669 7a60 2063 6172  le `wgd viz` car
+00014d30: 7269 6573 206f 7574 2074 6865 2072 6174  ries out the rat
+00014d40: 6520 636f 7272 6563 7469 6f6e 2061 6e64  e correction and
+00014d50: 2047 4d4d 2061 6e61 6c79 7369 7320 7375   GMM analysis su
+00014d60: 6368 2074 6861 7420 6974 2773 2065 6173  ch that it's eas
+00014d70: 6965 7220 746f 2064 6562 7567 2e0a 6060  ier to debug..``
+00014d80: 600a 7767 6420 6b73 6420 7767 645f 676c  `.wgd ksd wgd_gl
+00014d90: 6f62 616c 6d72 6268 2f67 6c6f 6261 6c5f  obalmrbh/global_
+00014da0: 4d52 4248 2e74 7376 2041 7175 696c 6567  MRBH.tsv Aquileg
+00014db0: 6961 5f63 6f65 7275 6c65 6120 5072 6f74  ia_coerulea Prot
+00014dc0: 6561 5f63 796e 6172 6f69 6465 7320 4163  ea_cynaroides Ac
+00014dd0: 6f72 7573 5f61 6d65 7269 6361 6e75 7320  orus_americanus 
+00014de0: 5669 7469 735f 7669 6e69 6665 7261 202d  Vitis_vinifera -
+00014df0: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
+00014e00: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
+00014e10: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+00014e20: 6b73 2e74 7376 202d 7370 2073 7065 6369  ks.tsv -sp speci
+00014e30: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
+00014e40: 6967 6874 202d 6f20 7767 645f 676c 6f62  ight -o wgd_glob
+00014e50: 616c 6d72 6268 5f6b 735f 7261 7465 5f63  almrbh_ks_rate_c
+00014e60: 6f72 7265 6374 696f 6e20 2d66 6120 4171  orrection -fa Aq
+00014e70: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00014e80: 202d 6170 2077 6764 5f73 796e 2f69 6164   -ap wgd_syn/iad
+00014e90: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
+00014ea0: 6f69 6e74 732e 7478 7420 2d2d 706c 6f74  oints.txt --plot
+00014eb0: 656c 6d6d 202d 2d70 6c6f 7461 7067 6d6d  elmm --plotapgmm
+00014ec0: 0a60 6060 0a0a 5468 6520 6669 6c65 2060  .```..The file `
+00014ed0: 7370 6563 6965 7374 7265 652e 6e77 6020  speciestree.nw` 
+00014ee0: 6973 2074 6865 2074 6578 7420 6669 6c65  is the text file
+00014ef0: 206f 6620 7370 6563 6965 7320 7472 6565   of species tree
+00014f00: 2069 6e20 6e65 7769 636b 2074 6861 7420   in newick that 
+00014f10: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
+00014f20: 776f 756c 6420 6265 2063 6f6e 6475 6374  would be conduct
+00014f30: 6564 206f 6e2e 2049 7473 2063 6f6e 7465  ed on. Its conte
+00014f40: 6e74 2069 7320 6173 2062 656c 6f77 2e20  nt is as below. 
+00014f50: 5573 6572 7320 6361 6e20 6f70 7469 6f6e  Users can option
+00014f60: 616c 6c79 2070 726f 7669 6465 2074 6865  ally provide the
+00014f70: 2073 7065 6369 6573 2070 6169 7273 2074   species pairs t
+00014f80: 6f20 6265 2070 6c6f 7474 6564 2062 7574  o be plotted but
+00014f90: 2077 6520 7375 6767 6573 7420 6f66 206a   we suggest of j
+00014fa0: 7573 7420 7573 696e 6720 602d 6661 2041  ust using `-fa A
+00014fb0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00014fc0: 6160 2074 6f20 706c 6f74 2061 6c6c 2070  a` to plot all p
+00014fd0: 6f73 7369 626c 6520 666f 6361 6c2d 7369  ossible focal-si
+00014fe0: 7374 6572 2073 7065 6369 6573 2070 6169  ster species pai
+00014ff0: 7273 2e20 5765 2073 7567 6765 7374 2061  rs. We suggest a
+00015000: 6464 696e 6720 7468 6520 6f70 7469 6f6e  dding the option
+00015010: 2060 2d2d 7265 7765 6967 6874 6020 746f   `--reweight` to
+00015020: 2072 6563 616c 6375 6c61 7465 2074 6865   recalculate the
+00015030: 2077 6569 6768 7420 7065 7220 7370 6563   weight per spec
+00015040: 6965 7320 7061 6972 2073 7563 6820 7468  ies pair such th
+00015050: 6174 2074 6865 2077 6569 6768 7420 6f66  at the weight of
+00015060: 206f 7274 686f 6c6f 676f 7573 2067 656e   orthologous gen
+00015070: 6520 7061 6972 7320 7769 6c6c 2062 6563  e pairs will bec
+00015080: 6f6d 6520 312e 2045 7874 7261 2063 6f6c  ome 1. Extra col
+00015090: 6c69 6e65 6172 2064 6174 6120 6361 6e20  linear data can 
+000150a0: 6265 2061 6464 6564 2062 7920 7468 6520  be added by the 
+000150b0: 6f70 7469 6f6e 2060 2d61 7060 2061 6e64  option `-ap` and
+000150c0: 2061 6464 6974 696f 6e61 6c20 636c 7573   additional clus
+000150d0: 7465 7269 6e67 2061 6e61 6c79 7369 7320  tering analysis 
+000150e0: 6361 6e20 6265 2069 6e69 7469 6174 6564  can be initiated
+000150f0: 2062 7920 7365 7474 696e 6720 7468 6520   by setting the 
+00015100: 6f70 7469 6f6e 2060 2d2d 706c 6f74 6170  option `--plotap
+00015110: 676d 6d60 2061 6e64 2060 2d2d 706c 6f74  gmm` and `--plot
+00015120: 656c 6d6d 602e 0a0a 6060 600a 2828 2856  elmm`...```.(((V
+00015130: 6974 6973 5f76 696e 6966 6572 612c 5072  itis_vinifera,Pr
+00015140: 6f74 6561 5f63 796e 6172 6f69 6465 7329  otea_cynaroides)
+00015150: 2c41 7175 696c 6567 6961 5f63 6f65 7275  ,Aquilegia_coeru
+00015160: 6c65 6129 2c41 636f 7275 735f 616d 6572  lea),Acorus_amer
+00015170: 6963 616e 7573 293b 0a60 6060 0a0a 215b  icanus);.```..![
+00015180: 5d28 6461 7461 2f4d 6978 6564 2e6b 732e  ](data/Mixed.ks.
+00015190: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000151a0: 6561 2e6e 6f64 652e 7765 6967 6874 6564  ea.node.weighted
+000151b0: 2e70 6e67 290a 0a54 6865 206d 6978 6564  .png)..The mixed
+000151c0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000151d0: 2064 6973 7472 6962 7574 696f 6e20 7368   distribution sh
+000151e0: 6f77 6e20 6162 6f76 6520 6973 2061 2070  own above is a p
+000151f0: 7562 6c69 6361 7469 6f6e 2d72 6561 6479  ublication-ready
+00015200: 2066 6967 7572 6520 7468 6174 2061 7373   figure that ass
+00015210: 656d 626c 6573 2074 6865 2072 6573 756c  embles the resul
+00015220: 7473 206f 6620 454c 4d4d 2c20 474d 4d20  ts of ELMM, GMM 
+00015230: 616e 6420 7261 7465 2063 6f72 7265 6374  and rate correct
+00015240: 696f 6e2e 2054 6865 206f 6e65 2d76 732d  ion. The one-vs-
+00015250: 6f6e 6520 6f72 7468 6f6c 6f67 6f75 7320  one orthologous 
+00015260: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00015270: 6469 7374 7269 6275 7469 6f6e 7320 6973  distributions is
+00015280: 2061 6c73 6f20 6175 746f 6d61 7469 6361   also automatica
+00015290: 6c6c 7920 7072 6f64 7563 6564 2077 6974  lly produced wit
+000152a0: 6820 7261 7465 2063 6f72 7265 6374 696f  h rate correctio
+000152b0: 6e20 7265 7375 6c74 7320 7375 7065 7269  n results superi
+000152c0: 6d70 6f73 6564 2077 6865 7265 2061 7661  mposed where ava
+000152d0: 696c 6162 6c65 2061 7320 7368 6f77 6e20  ilable as shown 
+000152e0: 6265 6c6f 772e 0a0a 215b 5d28 6461 7461  below...![](data
+000152f0: 2f46 6f63 7573 5f73 6973 7465 725f 7061  /Focus_sister_pa
+00015300: 6972 732e 6b73 2e6e 6f64 652e 7765 6967  irs.ks.node.weig
+00015310: 6874 6564 2e70 6e67 290a 215b 5d28 6461  hted.png).![](da
+00015320: 7461 2f41 6c6c 5f70 6169 7273 2e6b 732e  ta/All_pairs.ks.
+00015330: 6e6f 6465 2e77 6569 6768 7465 642e 706e  node.weighted.pn
+00015340: 6729 0a0a 4265 7369 6465 7320 7468 6520  g)..Besides the 
+00015350: 6162 6f76 6520 6b73 7261 7465 732d 6c69  above ksrates-li
+00015360: 6b65 202a 4b2a 3c73 7562 3e53 3c2f 7375  ke *K*<sub>S</su
+00015370: 623e 2070 6c6f 742c 2061 206d 6f72 6520  b> plot, a more 
+00015380: 636c 6173 7369 6320 2a4b 2a3c 7375 623e  classic *K*<sub>
+00015390: 533c 2f73 7562 3e20 706c 6f74 2063 616e  S</sub> plot can
+000153a0: 2062 6520 6d61 6465 2062 7920 6164 6469   be made by addi
+000153b0: 6e67 2074 6865 206f 7074 696f 6e20 602d  ng the option `-
+000153c0: 2d63 6c61 7373 6963 6020 746f 2074 6170  -classic` to tap
+000153d0: 206d 6f72 6520 6465 7461 696c 6564 6c79   more detailedly
+000153e0: 2069 6e74 6f20 7468 6520 7661 7269 6174   into the variat
+000153f0: 696f 6e20 6f66 2073 796e 6f6e 796d 6f75  ion of synonymou
+00015400: 7320 7375 6273 7469 7475 7469 6f6e 2072  s substitution r
+00015410: 6174 652e 0a0a 5573 696e 6720 7468 6520  ate...Using the 
+00015420: 636f 6d6d 616e 6420 6265 6c6f 772c 2074  command below, t
+00015430: 6865 2064 6972 6563 7469 6f6e 206f 6620  he direction of 
+00015440: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
+00015450: 616e 6420 6465 6772 6565 206f 6620 7261  and degree of ra
+00015460: 7465 2076 6172 6961 7469 6f6e 2063 616e  te variation can
+00015470: 2062 6520 6f62 7365 7276 6564 206d 6f72   be observed mor
+00015480: 6520 6469 7265 6374 6c79 2e0a 0a60 6060  e directly...```
+00015490: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
+000154a0: 676c 6f62 616c 6d72 6268 5f6b 732f 676c  globalmrbh_ks/gl
+000154b0: 6f62 616c 5f4d 5242 482e 7473 762e 6b73  obal_MRBH.tsv.ks
+000154c0: 2e74 7376 202d 6661 2041 7175 696c 6567  .tsv -fa Aquileg
+000154d0: 6961 5f63 6f65 7275 6c65 6120 2d65 706b  ia_coerulea -epk
+000154e0: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
+000154f0: 6961 5f63 6f65 7275 6c65 612e 6b73 2e74  ia_coerulea.ks.t
+00015500: 7376 202d 6170 2077 6764 5f73 796e 2f69  sv -ap wgd_syn/i
+00015510: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
+00015520: 7270 6f69 6e74 732e 7478 7420 2d73 7020  rpoints.txt -sp 
+00015530: 7370 6563 6965 7374 7265 652e 6e77 202d  speciestree.nw -
+00015540: 6f20 7767 645f 7669 7a5f 6d69 7865 645f  o wgd_viz_mixed_
+00015550: 4b73 202d 2d70 6c6f 7465 6c6d 6d20 2d2d  Ks --plotelmm --
+00015560: 706c 6f74 6170 676d 6d20 2d2d 7265 7765  plotapgmm --rewe
+00015570: 6967 6874 202d 2d70 6c6f 746b 6465 202d  ight --plotkde -
+00015580: 2d63 6c61 7373 6963 0a60 6060 0a0a 215b  -classic.```..![
+00015590: 5d28 6461 7461 2f41 7175 696c 6567 6961  ](data/Aquilegia
+000155a0: 5f63 6f65 7275 6c65 615f 476c 6f62 616c  _coerulea_Global
+000155b0: 6d72 6268 4b73 5f45 6c6d 6d5f 4170 676d  mrbhKs_Elmm_Apgm
+000155c0: 6d5f 436f 7272 6563 7465 642e 6b73 642e  m_Corrected.ksd.
+000155d0: 7376 6729 0a0a 4173 2073 686f 776e 2061  svg)..As shown a
+000155e0: 626f 7665 2c20 6265 6361 7573 6520 6f66  bove, because of
+000155f0: 2074 6865 2068 6967 6865 7220 7375 6273   the higher subs
+00015600: 7469 7475 7469 6f6e 2072 6174 6520 6f66  titution rate of
+00015610: 202a 4171 7569 6c65 6769 6120 636f 6572   *Aquilegia coer
+00015620: 756c 6561 2a2c 2074 6865 206f 7269 6769  ulea*, the origi
+00015630: 6e61 6c20 6f72 7468 6f6c 6f67 6f75 7320  nal orthologous 
+00015640: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00015650: 7661 6c75 6573 2077 6572 6520 6163 7475  values were actu
+00015660: 616c 6c79 2075 6e64 6572 6573 7469 6d61  ally underestima
+00015670: 7465 6420 696e 2074 6865 2074 696d 652d  ted in the time-
+00015680: 6672 616d 6520 6f66 202a 4171 7569 6c65  frame of *Aquile
+00015690: 6769 6120 636f 6572 756c 6561 2a2e 2057  gia coerulea*. W
+000156a0: 6865 6e20 7765 2072 6563 6f76 6572 6564  hen we recovered
+000156b0: 2074 6865 2064 6976 6572 6765 6e63 6520   the divergence 
+000156c0: 7375 6273 7469 7475 7469 6f6e 2064 6973  substitution dis
+000156d0: 7461 6e63 6520 696e 2074 6572 6d73 206f  tance in terms o
+000156e0: 6620 7477 6f20 7469 6d65 7320 6f66 2074  f two times of t
+000156f0: 6865 2062 7261 6e63 682d 7370 6563 6966  he branch-specif
+00015700: 6963 2063 6f6e 7472 6962 7574 696f 6e20  ic contribution 
+00015710: 6f66 202a 412e 2063 6f65 7275 6c65 612a  of *A. coerulea*
+00015720: 2073 696e 6365 2069 7473 2064 6976 6572   since its diver
+00015730: 6765 6e63 6520 7769 7468 2074 6865 2073  gence with the s
+00015740: 6973 7465 7220 7370 6563 6965 7320 706c  ister species pl
+00015750: 7573 2074 6865 2073 6861 7265 6420 7375  us the shared su
+00015760: 6273 7469 7475 7469 6f6e 2064 6973 7461  bstitution dista
+00015770: 6e63 6520 6265 666f 7265 2064 6976 6572  nce before diver
+00015780: 6765 6e63 6520 2869 6e20 7265 6c61 7469  gence (in relati
+00015790: 7665 2074 6f20 7468 6520 6f75 7467 726f  ve to the outgro
+000157a0: 7570 292c 2074 6865 2063 6f72 7265 6374  up), the correct
+000157b0: 6564 202a 4b2a 3c73 7562 3e53 3c2f 7375  ed *K*<sub>S</su
+000157c0: 623e 206d 6f64 6520 6265 6361 6d65 206c  b> mode became l
+000157d0: 6172 6765 722e 0a0a 4e6f 7465 2074 6861  arger...Note tha
+000157e0: 7420 7765 2063 616e 2065 6173 696c 7920  t we can easily 
+000157f0: 7368 6f77 2074 6861 7420 2a41 7175 696c  show that *Aquil
+00015800: 6567 6961 2063 6f65 7275 6c65 612a 2068  egia coerulea* h
+00015810: 6173 2068 6967 6865 7220 7375 6273 7469  as higher substi
+00015820: 7475 7469 6f6e 2072 6174 6520 7468 616e  tution rate than
+00015830: 202a 5072 6f74 6561 2063 796e 6172 6f69   *Protea cynaroi
+00015840: 6465 732a 2061 6e64 202a 5669 7469 7320  des* and *Vitis 
+00015850: 7669 6e69 6665 7261 2a20 6279 2063 6f6d  vinifera* by com
+00015860: 7061 7269 6e67 2074 6865 6972 2073 7562  paring their sub
+00015870: 7374 6974 7574 696f 6e20 6469 7374 616e  stitution distan
+00015880: 6365 2069 6e20 7265 6761 7264 2074 6f20  ce in regard to 
+00015890: 7468 6520 7361 6d65 2064 6976 6572 6765  the same diverge
+000158a0: 6e63 6520 6576 656e 7420 7769 7468 206f  nce event with o
+000158b0: 7574 6772 6f75 7020 7370 6563 6965 7320  utgroup species 
+000158c0: 2a41 636f 7275 735f 616d 6572 6963 616e  *Acorus_american
+000158d0: 7573 2a2c 2075 7369 6e67 2063 6f6d 6d61  us*, using comma
+000158e0: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
+000158f0: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
+00015900: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
+00015910: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
+00015920: 7376 202d 7370 2073 7065 6369 6573 7472  sv -sp speciestr
+00015930: 6565 2e6e 7720 2d2d 7265 7765 6967 6874  ee.nw --reweight
+00015940: 202d 6f20 7767 645f 7669 7a5f 436f 6d70   -o wgd_viz_Comp
+00015950: 6172 655f 7261 7465 202d 2d73 7061 6972  are_rate --spair
+00015960: 2022 4163 6f72 7573 5f61 6d65 7269 6361   "Acorus_america
+00015970: 6e75 733b 5072 6f74 6561 5f63 796e 6172  nus;Protea_cynar
+00015980: 6f69 6465 7322 202d 2d73 7061 6972 2022  oides" --spair "
+00015990: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000159a0: 6561 3b41 636f 7275 735f 616d 6572 6963  ea;Acorus_americ
+000159b0: 616e 7573 2220 2d2d 7370 6169 7220 2256  anus" --spair "V
+000159c0: 6974 6973 5f76 696e 6966 6572 613b 4163  itis_vinifera;Ac
+000159d0: 6f72 7573 5f61 6d65 7269 6361 6e75 7322  orus_americanus"
+000159e0: 202d 2d70 6c6f 746b 6465 202d 2d63 6c61   --plotkde --cla
+000159f0: 7373 6963 0a60 6060 0a0a 215b 5d28 6461  ssic.```..![](da
+00015a00: 7461 2f52 6177 5f4f 7274 686f 6c6f 6775  ta/Raw_Orthologu
+00015a10: 6573 5f43 6f6d 7061 7265 5f72 6174 652e  es_Compare_rate.
+00015a20: 6b73 642e 7376 6729 0a0a 4173 2064 6973  ksd.svg)..As dis
+00015a30: 706c 6179 6564 2061 626f 7665 2c20 7468  played above, th
+00015a40: 6520 6f72 7468 6f6c 6f67 6f75 7320 2a4b  e orthologous *K
+00015a50: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
+00015a60: 6c75 6573 2062 6577 7465 656e 202a 4171  lues bewteen *Aq
+00015a70: 7569 6c65 6769 6120 636f 6572 756c 6561  uilegia coerulea
+00015a80: 2a20 616e 6420 2a41 636f 7275 7320 616d  * and *Acorus am
+00015a90: 6572 6963 616e 7573 2a20 6861 7320 7468  ericanus* has th
+00015aa0: 6520 6869 6768 6573 7420 6d6f 6465 2c20  e highest mode, 
+00015ab0: 696e 6469 6361 7469 6e67 2074 6865 2066  indicating the f
+00015ac0: 6173 7465 7220 7375 6273 7469 7475 7469  aster substituti
+00015ad0: 6f6e 2072 6174 6520 6f66 202a 412e 2063  on rate of *A. c
+00015ae0: 6f65 7275 6c65 612a 2063 6f6d 7061 7265  oerulea* compare
+00015af0: 6420 746f 202a 5072 6f74 6561 2063 796e  d to *Protea cyn
+00015b00: 6172 6f69 6465 732a 2061 6e64 202a 5669  aroides* and *Vi
+00015b10: 7469 7320 7669 6e69 6665 7261 2a2e 0a0a  tis vinifera*...
+00015b20: 4265 666f 7265 2076 322e 302e 3231 2c20  Before v2.0.21, 
+00015b30: 7468 6520 6765 6e65 2d73 7065 6369 6573  the gene-species
+00015b40: 206d 6170 2066 696c 6520 6973 206e 6563   map file is nec
+00015b50: 6365 7373 6172 696c 7920 6e65 6564 6564  cessarily needed
+00015b60: 2066 6f72 2069 7473 2069 6d70 6c65 6d65   for its impleme
+00015b70: 6e74 6174 696f 6e20 696e 2060 7767 6420  ntation in `wgd 
+00015b80: 7669 7a60 2c20 7768 6963 6820 7368 6f75  viz`, which shou
+00015b90: 6c64 2062 6520 6175 746f 6d61 7465 6c79  ld be automately
+00015ba0: 2070 726f 6475 6365 6420 6279 2074 6865   produced by the
+00015bb0: 206c 6173 7420 6077 6764 206b 7364 6020   last `wgd ksd` 
+00015bc0: 7374 6570 2067 6976 656e 2074 6865 2060  step given the `
+00015bd0: 7370 6169 7260 2061 6e64 2060 7370 6563  spair` and `spec
+00015be0: 6965 7374 7265 6560 2070 6172 616d 6574  iestree` paramet
+00015bf0: 6572 732e 2054 6865 2060 6765 6e65 5f73  ers. The `gene_s
+00015c00: 7065 6369 6573 2e6d 6170 6020 6861 7320  pecies.map` has 
+00015c10: 636f 6e74 656e 7473 2061 7320 6265 6c6f  contents as belo
+00015c20: 7720 696e 2077 6869 6368 2065 6163 6820  w in which each 
+00015c30: 6c69 6e65 2069 7320 7468 6520 6a6f 696e  line is the join
+00015c40: 6564 2073 7472 696e 6720 6f66 2067 656e  ed string of gen
+00015c50: 6520 6e61 6d65 2061 6e64 2073 7065 6369  e name and speci
+00015c60: 6573 206e 616d 6520 6279 2073 7061 6365  es name by space
+00015c70: 2e20 4166 7465 7220 7632 2e30 2e32 3120  . After v2.0.21 
+00015c80: 2869 6e63 6c75 6465 6429 2c20 7468 6520  (included), the 
+00015c90: 6765 6e65 2d73 7065 6369 6573 206d 6170  gene-species map
+00015ca0: 2066 696c 6520 6973 206e 6f74 206e 6563   file is not nec
+00015cb0: 6365 7373 6172 696c 7920 6e65 6564 6564  cessarily needed
+00015cc0: 2061 6e79 6d6f 7265 2e0a 0a60 6060 0a41   anymore...```.A
+00015cd0: 7163 6f65 3647 3035 3738 3030 2e31 2041  qcoe6G057800.1 A
+00015ce0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00015cf0: 610a 5676 695f 5649 545f 3230 3173 3030  a.Vvi_VIT_201s00
+00015d00: 3131 6730 3135 3330 2e31 2056 6974 6973  11g01530.1 Vitis
+00015d10: 5f76 696e 6966 6572 610a 5063 795f 5072  _vinifera.Pcy_Pr
+00015d20: 6f63 7930 3167 3038 3531 3020 5072 6f74  ocy01g08510 Prot
+00015d30: 6561 5f63 796e 6172 6f69 6465 730a 4161  ea_cynaroides.Aa
+00015d40: 6d5f 4163 6f72 612e 3034 4731 3432 3930  m_Acora.04G14290
+00015d50: 302e 3120 4163 6f72 7573 5f61 6d65 7269  0.1 Acorus_ameri
+00015d60: 6361 6e75 730a 6060 600a 0a41 6e20 616c  canus.```..An al
+00015d70: 7465 726e 6174 6976 6520 7761 7920 746f  ternative way to
+00015d80: 2063 616c 6375 6c61 7465 2074 6865 206f   calculate the o
+00015d90: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
+00015da0: 7562 3e53 3c2f 7375 623e 2069 7320 746f  ub>S</sub> is to
+00015db0: 2064 6972 6563 746c 7920 7573 6520 7468   directly use th
+00015dc0: 6520 6f72 7468 6f67 726f 7570 7320 696e  e orthogroups in
+00015dd0: 7374 6561 6420 6f66 2067 6c6f 6261 6c20  stead of global 
+00015de0: 4d52 4248 2066 616d 696c 792e 2054 6861  MRBH family. Tha
+00015df0: 7420 7761 7920 7765 2064 6f6e 2774 2075  t way we don't u
+00015e00: 7365 2074 6865 2070 7265 2d69 6e66 6572  se the pre-infer
+00015e10: 7265 6420 7061 7261 6e6f 6d65 202a 4b2a  red paranome *K*
+00015e20: 3c73 7562 3e53 3c2f 7375 623e 2062 7574  <sub>S</sub> but
+00015e30: 2074 6865 2070 6172 616c 6f67 6f75 7320   the paralogous 
+00015e40: 6765 6e65 2070 6169 7273 2069 6e73 6964  gene pairs insid
+00015e50: 6520 6561 6368 206f 7274 686f 6772 6f75  e each orthogrou
+00015e60: 7020 696e 7374 6561 642e 2054 6f20 6163  p instead. To ac
+00015e70: 6869 6576 6520 7468 6174 2c20 7765 2066  hieve that, we f
+00015e80: 6972 7374 206e 6565 6420 746f 2069 6e66  irst need to inf
+00015e90: 6572 206f 7274 686f 6772 6f75 7073 2075  er orthogroups u
+00015ea0: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
+00015eb0: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
+00015ec0: 2064 6d64 2041 7175 696c 6567 6961 5f63   dmd Aquilegia_c
+00015ed0: 6f65 7275 6c65 6120 5072 6f74 6561 5f63  oerulea Protea_c
+00015ee0: 796e 6172 6f69 6465 7320 4163 6f72 7573  ynaroides Acorus
+00015ef0: 5f61 6d65 7269 6361 6e75 7320 5669 7469  _americanus Viti
+00015f00: 735f 7669 6e69 6665 7261 202d 2d6f 7274  s_vinifera --ort
+00015f10: 686f 696e 6665 7220 2d6f 2077 6764 5f6f  hoinfer -o wgd_o
+00015f20: 7274 686f 2028 2d2d 6f6e 6c79 6f72 7468  rtho (--onlyorth
+00015f30: 6f29 200a 6060 600a 0a55 7365 7273 2063  o) .```..Users c
+00015f40: 616e 2064 6563 6964 6520 746f 206f 6e6c  an decide to onl
+00015f50: 7920 636f 6e64 7563 7420 7468 6520 6f72  y conduct the or
+00015f60: 7468 6f67 726f 7570 2061 6e61 6c79 7369  thogroup analysi
+00015f70: 7320 7768 696c 6520 736b 6970 7069 6e67  s while skipping
+00015f80: 206f 7468 6572 2061 6e61 6c79 7369 7320   other analysis 
+00015f90: 6279 2061 6464 696e 6720 7468 6520 666c  by adding the fl
+00015fa0: 6167 2060 2d2d 6f6e 6c79 6f72 7468 6f60  ag `--onlyortho`
+00015fb0: 2e20 4e65 7874 2073 7465 7020 6973 2074  . Next step is t
+00015fc0: 6865 2073 616d 6520 7769 7468 2067 6c6f  he same with glo
+00015fd0: 6261 6c20 4d52 4248 2066 616d 696c 792e  bal MRBH family.
+00015fe0: 0a0a 6060 600a 7767 6420 6b73 6420 7767  ..```.wgd ksd wg
+00015ff0: 645f 6f72 7468 6f2f 4f72 7468 6f67 726f  d_ortho/Orthogro
+00016000: 7570 732e 7370 2e74 7376 2041 7175 696c  ups.sp.tsv Aquil
+00016010: 6567 6961 5f63 6f65 7275 6c65 6120 5072  egia_coerulea Pr
+00016020: 6f74 6561 5f63 796e 6172 6f69 6465 7320  otea_cynaroides 
+00016030: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
+00016040: 7320 5669 7469 735f 7669 6e69 6665 7261  s Vitis_vinifera
+00016050: 202d 6f20 7767 645f 6f72 7468 6f5f 6b73   -o wgd_ortho_ks
+00016060: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
+00016070: 6f72 7468 6f5f 6b73 2f4f 7274 686f 6772  ortho_ks/Orthogr
+00016080: 6f75 7073 2e73 702e 7473 762e 6b73 2e74  oups.sp.tsv.ks.t
+00016090: 7376 202d 6570 6b20 7767 645f 6b73 642f  sv -epk wgd_ksd/
+000160a0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000160b0: 6561 2e6b 732e 7473 7620 2d73 7020 7370  ea.ks.tsv -sp sp
+000160c0: 6563 6965 7374 7265 652e 6e77 202d 2d72  eciestree.nw --r
+000160d0: 6577 6569 6768 7420 2d61 7020 7767 645f  eweight -ap wgd_
+000160e0: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+000160f0: 616e 6368 6f72 706f 696e 7473 2e74 7874  anchorpoints.txt
+00016100: 202d 706c 6f74 656c 6d6d 202d 2d70 6c6f   -plotelmm --plo
+00016110: 7461 7067 6d6d 202d 6f20 7767 645f 6f72  tapgmm -o wgd_or
+00016120: 7468 6f5f 6b73 5f72 6174 655f 636f 7272  tho_ks_rate_corr
+00016130: 6563 7469 6f6e 0a60 6060 0a0a 215b 5d28  ection.```..![](
+00016140: 6461 7461 2f4d 6978 6564 2e6b 732e 4171  data/Mixed.ks.Aq
+00016150: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00016160: 2e6e 6f64 652e 7765 6967 6874 6564 5f4f  .node.weighted_O
+00016170: 7274 686f 5f4b 732e 706e 6729 0a21 5b5d  rtho_Ks.png).![]
+00016180: 2864 6174 612f 466f 6375 735f 7369 7374  (data/Focus_sist
+00016190: 6572 5f70 6169 7273 2e6b 732e 6e6f 6465  er_pairs.ks.node
+000161a0: 2e77 6569 6768 7465 645f 4f72 7468 6f5f  .weighted_Ortho_
+000161b0: 4b73 2e70 6e67 290a 215b 5d28 6461 7461  Ks.png).![](data
+000161c0: 2f41 6c6c 5f70 6169 7273 2e6b 732e 6e6f  /All_pairs.ks.no
+000161d0: 6465 2e77 6569 6768 7465 645f 4f72 7468  de.weighted_Orth
+000161e0: 6f5f 4b73 2e70 6e67 290a 0a41 7320 7368  o_Ks.png)..As sh
+000161f0: 6f77 6e20 6162 6f76 652c 2074 6865 206e  own above, the n
+00016200: 756d 6265 7220 6f66 206f 7274 686f 6c6f  umber of ortholo
+00016210: 676f 7573 2067 656e 6520 7061 6972 7320  gous gene pairs 
+00016220: 6973 2064 6966 6665 7265 6e74 2074 6861  is different tha
+00016230: 6e20 7468 6520 6f6e 6520 6672 6f6d 2067  n the one from g
+00016240: 6c6f 6261 6c20 4d52 4248 2066 616d 696c  lobal MRBH famil
+00016250: 6965 7320 696e 2074 6861 7420 6865 7265  ies in that here
+00016260: 2077 6520 706c 6f74 7465 6420 616c 6c20   we plotted all 
+00016270: 6f72 7468 6f6c 6f67 6f75 7320 6765 6e65  orthologous gene
+00016280: 2070 6169 7273 2069 6e73 7465 6164 206f   pairs instead o
+00016290: 6620 6f6e 6c79 2067 6c6f 6261 6c20 4d52  f only global MR
+000162a0: 4248 2066 616d 696c 6965 732c 2074 6f67  BH families, tog
+000162b0: 6574 6865 7220 7769 7468 2064 6966 6665  ether with diffe
+000162c0: 7265 6e74 2072 6563 616c 6375 6c61 7465  rent recalculate
+000162d0: 6420 7765 6967 6874 732e 0a0a 4166 7465  d weights...Afte
+000162e0: 7220 7468 6520 7068 796c 6f67 656e 6574  r the phylogenet
+000162f0: 6963 2074 696d 696e 6720 6f66 2074 6865  ic timing of the
+00016300: 2052 616e 756e 6375 6c61 6c65 7320 5747   Ranunculales WG
+00016310: 442c 2077 6520 6361 6e20 6675 7274 6865  D, we can furthe
+00016320: 7220 696e 6665 7220 6974 7320 6162 736f  r infer its abso
+00016330: 6c75 7465 2061 6765 2e20 4669 7273 7420  lute age. First 
+00016340: 7765 2069 6e66 6572 2074 6865 2063 7265  we infer the cre
+00016350: 6469 626c 6520 7261 6e67 6520 6f66 2061  dible range of a
+00016360: 6e63 686f 7220 7061 6972 7320 6279 202a  nchor pairs by *
+00016370: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2068  K*<sub>S</sub> h
+00016380: 6575 7269 7374 6963 616c 6c79 2075 7369  euristically usi
+00016390: 6e67 2074 6865 2070 726f 6772 616d 2060  ng the program `
+000163a0: 7767 6420 7065 616b 602e 0a0a 6060 600a  wgd peak`...```.
+000163b0: 7767 6420 7065 616b 202d 2d68 6575 7269  wgd peak --heuri
+000163c0: 7374 6963 2077 6764 5f6b 7364 2f41 7175  stic wgd_ksd/Aqu
+000163d0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+000163e0: 7473 762e 6b73 2e74 7376 202d 6170 2077  tsv.ks.tsv -ap w
+000163f0: 6764 5f73 796e 2f69 6164 686f 7265 2d6f  gd_syn/iadhore-o
+00016400: 7574 2f61 6e63 686f 7270 6f69 6e74 732e  ut/anchorpoints.
+00016410: 7478 7420 2d73 6d20 7767 645f 7379 6e2f  txt -sm wgd_syn/
+00016420: 6961 6468 6f72 652d 6f75 742f 7365 676d  iadhore-out/segm
+00016430: 656e 7473 2e74 7874 202d 6c65 2077 6764  ents.txt -le wgd
+00016440: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
+00016450: 2f6c 6973 745f 656c 656d 656e 7473 2e74  /list_elements.t
+00016460: 7874 202d 6d70 2077 6764 5f73 796e 2f69  xt -mp wgd_syn/i
+00016470: 6164 686f 7265 2d6f 7574 2f6d 756c 7469  adhore-out/multi
+00016480: 706c 6963 6f6e 5f70 6169 7273 2e74 7874  plicon_pairs.txt
+00016490: 202d 6f20 7767 645f 7065 616b 0a60 6060   -o wgd_peak.```
+000164a0: 0a0a 215b 5d28 6461 7461 2f41 6e63 686f  ..![](data/Ancho
+000164b0: 724b 735f 5065 616b 4349 5f41 7175 696c  rKs_PeakCI_Aquil
+000164c0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+000164d0: 762e 6b73 2e74 7376 5f6e 6f64 655f 7765  v.ks.tsv_node_we
+000164e0: 6967 6874 6564 2e73 7667 290a 0a41 7320  ighted.svg)..As 
+000164f0: 7368 6f77 6e20 6162 6f76 652c 2077 6520  shown above, we 
+00016500: 6173 7375 6d65 6420 6120 6c6f 676e 6f72  assumed a lognor
+00016510: 6d61 6c20 6469 7374 7269 6275 7469 6f6e  mal distribution
+00016520: 2061 7420 7468 6520 7065 616b 206c 6f63   at the peak loc
+00016530: 6174 696f 6e20 6465 7465 6374 6564 2062  ation detected b
+00016540: 7920 7468 6520 6073 6967 6e61 6c60 206d  y the `signal` m
+00016550: 6f64 756c 6520 6f66 2060 7363 6970 7960  odule of `scipy`
+00016560: 206c 6962 7261 7279 2e20 5468 6520 3935   library. The 95
+00016570: 2520 636f 6e66 6964 656e 6365 206c 6576  % confidence lev
+00016580: 656c 206f 6620 7468 6520 6c6f 676e 6f72  el of the lognor
+00016590: 6d61 6c20 6469 7374 7269 6275 7469 6f6e  mal distribution
+000165a0: 2077 6173 2061 7070 6c69 6564 2c20 692e   was applied, i.
+000165b0: 652e 2c20 302e 3638 2d32 2e37 342c 2069  e., 0.68-2.74, i
+000165c0: 6e20 6675 7274 6865 7220 6d6f 6c65 6375  n further molecu
+000165d0: 6c61 7220 6461 7469 6e67 2e20 5468 6520  lar dating. The 
+000165e0: 6669 6c65 2060 4171 7569 6c65 6769 615f  file `Aquilegia_
+000165f0: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00016600: 7473 765f 3935 2543 495f 4150 5f66 6f72  tsv_95%CI_AP_for
+00016610: 5f64 6174 696e 675f 7765 6967 6874 6564  _dating_weighted
+00016620: 5f66 6f72 6d61 742e 7473 7660 2069 7320  _format.tsv` is 
+00016630: 7768 6174 2077 6520 6e65 6564 2066 6f72  what we need for
+00016640: 206e 6578 7420 7374 6570 2e20 546f 2062   next step. To b
+00016650: 7569 6c64 2074 6865 206f 7274 686f 6772  uild the orthogr
+00016660: 6f75 7073 2075 7365 6420 696e 2070 6879  oups used in phy
+00016670: 6c6f 6765 6e65 7469 6320 6461 7469 6e67  logenetic dating
+00016680: 2c20 7765 206e 6565 6420 746f 2073 656c  , we need to sel
+00016690: 6563 7420 736f 6d65 2073 7065 6369 6573  ect some species
+000166a0: 2061 6e64 2066 6f72 6d20 6120 7374 6172   and form a star
+000166b0: 7469 6e67 2074 7265 6520 7769 7468 2070  ting tree with p
+000166c0: 726f 7065 7220 666f 7373 696c 2063 616c  roper fossil cal
+000166d0: 6962 7261 7469 6f6e 732e 2057 6520 7072  ibrations. We pr
+000166e0: 6f76 6964 6520 6f6e 6520 696e 206d 636d  ovide one in mcm
+000166f0: 6374 7265 6520 666f 726d 6174 2061 7320  ctree format as 
+00016700: 6265 6c6f 772e 0a0a 6060 600a 3137 2031  below...```.17 1
+00016710: 0a28 2828 2850 6f74 616d 6f67 6574 6f6e  .((((Potamogeton
+00016720: 5f61 6375 7469 666f 6c69 7573 2c28 5370  _acutifolius,(Sp
+00016730: 6972 6f64 656c 615f 696e 7465 726d 6564  irodela_intermed
+00016740: 6961 2c41 6d6f 7270 686f 7068 616c 6c75  ia,Amorphophallu
+00016750: 735f 6b6f 6e6a 6163 2929 2c28 4163 616e  s_konjac)),(Acan
+00016760: 7468 6f63 686c 616d 7973 5f62 7261 6374  thochlamys_bract
+00016770: 6561 7461 2c28 4469 6f73 636f 7265 615f  eata,(Dioscorea_
+00016780: 616c 6174 612c 4469 6f73 636f 7265 615f  alata,Dioscorea_
+00016790: 726f 7475 6e64 6174 6129 2927 3e30 2e35  rotundata))'>0.5
+000167a0: 3630 303c 312e 3238 3633 2729 273e 302e  600<1.2863')'>0.
+000167b0: 3833 3630 3c31 2e32 3836 3327 2c28 4163  8360<1.2863',(Ac
+000167c0: 6f72 7573 5f61 6d65 7269 6361 6e75 732c  orus_americanus,
+000167d0: 4163 6f72 7573 5f74 6174 6172 696e 6f77  Acorus_tatarinow
+000167e0: 6969 2929 273e 302e 3833 3630 3c31 2e32  ii))'>0.8360<1.2
+000167f0: 3836 3327 2c28 2828 2854 6574 7261 6365  863',((((Tetrace
+00016800: 6e74 726f 6e5f 7369 6e65 6e73 652c 5472  ntron_sinense,Tr
+00016810: 6f63 686f 6465 6e64 726f 6e5f 6172 616c  ochodendron_aral
+00016820: 696f 6964 6573 292c 2842 7578 7573 5f61  ioides),(Buxus_a
+00016830: 7573 7472 6f79 756e 6e61 6e65 6e73 6973  ustroyunnanensis
+00016840: 2c42 7578 7573 5f73 696e 6963 6129 2927  ,Buxus_sinica))'
+00016850: 3e31 2e31 3038 303c 312e 3238 3633 272c  >1.1080<1.2863',
+00016860: 284e 656c 756d 626f 5f6e 7563 6966 6572  (Nelumbo_nucifer
+00016870: 612c 2854 656c 6f70 6561 5f73 7065 6369  a,(Telopea_speci
+00016880: 6f73 6973 7369 6d61 2c50 726f 7465 615f  osissima,Protea_
+00016890: 6379 6e61 726f 6964 6573 2929 2927 3e31  cynaroides)))'>1
+000168a0: 2e31 3038 303c 312e 3238 3633 272c 2841  .1080<1.2863',(A
+000168b0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+000168c0: 615f 6170 312c 4171 7569 6c65 6769 615f  a_ap1,Aquilegia_
+000168d0: 636f 6572 756c 6561 5f61 7032 2929 273e  coerulea_ap2))'>
+000168e0: 312e 3130 3830 3c31 2e32 3836 3327 2927  1.1080<1.2863')'
+000168f0: 3e31 2e32 3732 303c 322e 3437 3230 273b  >1.2720<2.4720';
+00016900: 0a60 6060 0a0a 4173 2070 7265 7365 6e74  .```..As present
+00016910: 6564 2061 626f 7665 2c20 7468 6520 666f  ed above, the fo
+00016920: 6375 7320 7370 6563 6965 7320 7468 6174  cus species that
+00016930: 2069 7320 6162 6f75 7420 746f 2062 6520   is about to be 
+00016940: 6461 7465 6420 6e65 6564 7320 746f 2062  dated needs to b
+00016950: 6520 7265 706c 6163 6564 2077 6974 6820  e replaced with 
+00016960: 6028 4171 7569 6c65 6769 615f 636f 6572  `(Aquilegia_coer
+00016970: 756c 6561 5f61 7031 2c41 7175 696c 6567  ulea_ap1,Aquileg
+00016980: 6961 5f63 6f65 7275 6c65 615f 6170 3229  ia_coerulea_ap2)
+00016990: 602e 2057 6974 6820 7468 6973 2073 7461  `. With this sta
+000169a0: 7274 696e 6720 7472 6565 2061 6e64 2070  rting tree and p
+000169b0: 7265 646f 776e 6c6f 6164 6564 2063 6473  redownloaded cds
+000169c0: 2066 696c 6573 206f 6620 616c 6c20 7468   files of all th
+000169d0: 6520 7370 6563 6965 732c 2077 6520 6361  e species, we ca
+000169e0: 6e20 6275 696c 6420 7468 6520 6f72 7468  n build the orth
+000169f0: 6f67 726f 7570 2075 7365 6420 696e 2074  ogroup used in t
+00016a00: 6865 2066 696e 616c 206d 6f6c 6563 756c  he final molecul
+00016a10: 6172 2064 6174 696e 6720 7573 696e 6720  ar dating using 
+00016a20: 7468 6520 636f 6d6d 616e 6420 6173 2062  the command as b
+00016a30: 656c 6f77 2e0a 0a60 6060 0a77 6764 2064  elow...```.wgd d
+00016a40: 6d64 202d 6620 4171 7569 6c65 6769 615f  md -f Aquilegia_
+00016a50: 636f 6572 756c 6561 202d 6170 2077 6764  coerulea -ap wgd
+00016a60: 5f70 6561 6b2f 4171 7569 6c65 6769 615f  _peak/Aquilegia_
+00016a70: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00016a80: 7473 765f 3935 2543 495f 4150 5f66 6f72  tsv_95%CI_AP_for
+00016a90: 5f64 6174 696e 675f 7765 6967 6874 6564  _dating_weighted
+00016aa0: 5f66 6f72 6d61 742e 7473 7620 2d6f 2077  _format.tsv -o w
+00016ab0: 6764 5f64 6d64 5f6f 7274 686f 2050 6f74  gd_dmd_ortho Pot
+00016ac0: 616d 6f67 6574 6f6e 5f61 6375 7469 666f  amogeton_acutifo
+00016ad0: 6c69 7573 2053 7069 726f 6465 6c61 5f69  lius Spirodela_i
+00016ae0: 6e74 6572 6d65 6469 6120 416d 6f72 7068  ntermedia Amorph
+00016af0: 6f70 6861 6c6c 7573 5f6b 6f6e 6a61 6320  ophallus_konjac 
+00016b00: 4163 616e 7468 6f63 686c 616d 7973 5f62  Acanthochlamys_b
+00016b10: 7261 6374 6561 7461 2044 696f 7363 6f72  racteata Dioscor
+00016b20: 6561 5f61 6c61 7461 2044 696f 7363 6f72  ea_alata Dioscor
+00016b30: 6561 5f72 6f74 756e 6461 7461 2041 636f  ea_rotundata Aco
+00016b40: 7275 735f 616d 6572 6963 616e 7573 2041  rus_americanus A
+00016b50: 636f 7275 735f 7461 7461 7269 6e6f 7769  corus_tatarinowi
+00016b60: 6920 5465 7472 6163 656e 7472 6f6e 5f73  i Tetracentron_s
+00016b70: 696e 656e 7365 2054 726f 6368 6f64 656e  inense Trochoden
+00016b80: 6472 6f6e 5f61 7261 6c69 6f69 6465 7320  dron_aralioides 
+00016b90: 4275 7875 735f 6175 7374 726f 7975 6e6e  Buxus_austroyunn
+00016ba0: 616e 656e 7369 7320 4275 7875 735f 7369  anensis Buxus_si
+00016bb0: 6e69 6361 204e 656c 756d 626f 5f6e 7563  nica Nelumbo_nuc
+00016bc0: 6966 6572 6120 5465 6c6f 7065 615f 7370  ifera Telopea_sp
+00016bd0: 6563 696f 7369 7373 696d 6120 5072 6f74  eciosissima Prot
+00016be0: 6561 5f63 796e 6172 6f69 6465 7320 4171  ea_cynaroides Aq
+00016bf0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00016c00: 0a60 6060 0a0a 5468 6520 7265 7375 6c74  .```..The result
+00016c10: 2066 696c 6520 606d 6572 6765 5f66 6f63   file `merge_foc
+00016c20: 7573 5f61 702e 7473 7660 2069 7320 7768  us_ap.tsv` is wh
+00016c30: 6174 2077 6520 6e65 6564 2066 6f72 2074  at we need for t
+00016c40: 6865 2066 696e 616c 2073 7465 7020 6f66  he final step of
+00016c50: 206d 6f6c 6563 756c 6172 2064 6174 696e   molecular datin
+00016c60: 6720 696e 2070 726f 6772 616d 2060 7767  g in program `wg
+00016c70: 6420 666f 6375 7360 2e0a 0a60 6060 0a77  d focus`...```.w
+00016c80: 6764 2066 6f63 7573 202d 2d70 726f 7464  gd focus --protd
+00016c90: 6174 696e 6720 2d2d 6161 6d6f 6465 6c20  ating --aamodel 
+00016ca0: 6c67 2077 6764 5f64 6d64 5f6f 7274 686f  lg wgd_dmd_ortho
+00016cb0: 2f6d 6572 6765 5f66 6f63 7573 5f61 702e  /merge_focus_ap.
+00016cc0: 7473 7620 2d73 7020 6461 7469 6e67 5f74  tsv -sp dating_t
+00016cd0: 7265 652e 6e77 202d 6f20 7767 645f 6461  ree.nw -o wgd_da
+00016ce0: 7469 6e67 202d 6420 6d63 6d63 7472 6565  ting -d mcmctree
+00016cf0: 202d 6473 2027 6275 726e 696e 203d 2032   -ds 'burnin = 2
+00016d00: 3030 3027 202d 6473 2027 7361 6d70 6672  000' -ds 'sampfr
+00016d10: 6571 203d 2031 3030 3027 202d 6473 2027  eq = 1000' -ds '
+00016d20: 6e73 616d 706c 6520 3d20 3230 3030 3027  nsample = 20000'
+00016d30: 2050 6f74 616d 6f67 6574 6f6e 5f61 6375   Potamogeton_acu
+00016d40: 7469 666f 6c69 7573 2053 7069 726f 6465  tifolius Spirode
+00016d50: 6c61 5f69 6e74 6572 6d65 6469 6120 416d  la_intermedia Am
+00016d60: 6f72 7068 6f70 6861 6c6c 7573 5f6b 6f6e  orphophallus_kon
+00016d70: 6a61 6320 4163 616e 7468 6f63 686c 616d  jac Acanthochlam
+00016d80: 7973 5f62 7261 6374 6561 7461 2044 696f  ys_bracteata Dio
+00016d90: 7363 6f72 6561 5f61 6c61 7461 2044 696f  scorea_alata Dio
+00016da0: 7363 6f72 6561 5f72 6f74 756e 6461 7461  scorea_rotundata
+00016db0: 2041 636f 7275 735f 616d 6572 6963 616e   Acorus_american
+00016dc0: 7573 2041 636f 7275 735f 7461 7461 7269  us Acorus_tatari
+00016dd0: 6e6f 7769 6920 5465 7472 6163 656e 7472  nowii Tetracentr
+00016de0: 6f6e 5f73 696e 656e 7365 2054 726f 6368  on_sinense Troch
+00016df0: 6f64 656e 6472 6f6e 5f61 7261 6c69 6f69  odendron_aralioi
+00016e00: 6465 7320 4275 7875 735f 6175 7374 726f  des Buxus_austro
+00016e10: 7975 6e6e 616e 656e 7369 7320 4275 7875  yunnanensis Buxu
+00016e20: 735f 7369 6e69 6361 204e 656c 756d 626f  s_sinica Nelumbo
+00016e30: 5f6e 7563 6966 6572 6120 5465 6c6f 7065  _nucifera Telope
+00016e40: 615f 7370 6563 696f 7369 7373 696d 6120  a_speciosissima 
+00016e50: 5072 6f74 6561 5f63 796e 6172 6f69 6465  Protea_cynaroide
+00016e60: 7320 4171 7569 6c65 6769 615f 636f 6572  s Aquilegia_coer
+00016e70: 756c 6561 0a60 6060 0a0a 4865 7265 2077  ulea.```..Here w
+00016e80: 6520 6f6e 6c79 2069 6d70 6c65 6d65 6e74  e only implement
+00016e90: 6564 2074 6865 2063 6f6e 6361 7465 6e61  ed the concatena
+00016ea0: 7469 6f6e 2061 6e61 6c79 7369 7320 7573  tion analysis us
+00016eb0: 696e 6720 7072 6f74 6569 6e20 7365 7175  ing protein sequ
+00016ec0: 656e 6365 2062 7920 6164 6469 6e67 2074  ence by adding t
+00016ed0: 6865 2066 6c61 6720 602d 2d70 726f 7464  he flag `--protd
+00016ee0: 6174 696e 6760 2061 6e64 2077 6520 7365  ating` and we se
+00016ef0: 7420 7468 6520 7061 7261 6d65 7465 7220  t the parameter 
+00016f00: 666f 7220 606d 636d 6374 7265 6560 2076  for `mcmctree` v
+00016f10: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
+00016f20: 6473 602e 204e 6f74 6520 7468 6174 206f  ds`. Note that o
+00016f30: 7468 6572 2064 6174 696e 6720 7072 6f67  ther dating prog
+00016f40: 7261 6d20 7375 6368 2061 7320 6072 3873  ram such as `r8s
+00016f50: 6020 616e 6420 6062 6561 7374 6020 6172  ` and `beast` ar
+00016f60: 6520 616c 736f 2061 7661 696c 6162 6c65  e also available
+00016f70: 2067 6976 656e 2073 6f6d 6520 6d61 6e64   given some mand
+00016f80: 6174 6f72 7920 7061 7261 6d65 7465 7273  atory parameters
+00016f90: 2e20 5468 6520 6669 6e61 6c20 6c6f 6720  . The final log 
+00016fa0: 6f66 2074 6865 2073 7563 6365 7373 6675  of the successfu
+00016fb0: 6c20 7275 6e20 6973 2061 7320 6265 6c6f  l run is as belo
+00016fc0: 772e 0a0a 6060 600a 3136 3a30 343a 3235  w...```.16:04:25
+00016fd0: 2049 4e46 4f20 2020 2020 5275 6e6e 696e   INFO     Runnin
+00016fe0: 6720 6d63 6d63 7472 6565 2075 7369 6e67  g mcmctree using
+00016ff0: 2048 6573 7369 616e 206d 6174 7269 7820   Hessian matrix 
+00017000: 6f66 204c 472b 4761 6d6d 6120 2063 6f72  of LG+Gamma  cor
+00017010: 652e 7079 3a39 3637 0a20 2020 2020 2020  e.py:967.       
+00017020: 2020 2020 2020 2020 2020 2066 6f72 2070             for p
+00017030: 726f 7465 696e 206d 6f64 656c 0a32 333a  rotein model.23:
+00017040: 3439 3a33 3720 494e 464f 2020 2020 2050  49:37 INFO     P
+00017050: 6f73 7465 7269 6f72 206d 6561 6e20 666f  osterior mean fo
+00017060: 7220 7468 6520 6167 6573 206f 6620 7767  r the ages of wg
+00017070: 6420 6973 2031 3132 2e38 3934 3520 6d63  d is 112.8945 mc
+00017080: 6d63 7472 6565 2e70 793a 3239 360a 2020  mctree.py:296.  
+00017090: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000170a0: 6d69 6c6c 696f 6e20 7965 6172 7320 6672  million years fr
+000170b0: 6f6d 2043 6f6e 6361 7465 6e61 7465 6420  om Concatenated 
+000170c0: 7065 7074 6964 650a 2020 2020 2020 2020  peptide.        
+000170d0: 2020 2020 2020 2020 2020 616c 6967 6e6d            alignm
+000170e0: 656e 7420 616e 6420 3935 2520 6372 6564  ent and 95% cred
+000170f0: 6962 696c 6974 7920 696e 7465 7276 616c  ibility interval
+00017100: 7320 2843 4929 0a20 2020 2020 2020 2020  s (CI).         
+00017110: 2020 2020 2020 2020 2069 7320 3130 312e           is 101.
+00017120: 3232 342d 3132 332e 3132 3120 6d69 6c6c  224-123.121 mill
+00017130: 696f 6e20 7965 6172 730a 2020 2020 2020  ion years.      
+00017140: 2020 2049 4e46 4f20 2020 2020 546f 7461     INFO     Tota
+00017150: 6c20 7275 6e20 7469 6d65 3a20 3239 3137  l run time: 2917
+00017160: 3573 2020 2020 2020 2020 2020 2020 2020  5s              
+00017170: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00017180: 636c 692e 7079 3a32 3431 0a20 2020 2020  cli.py:241.     
+00017190: 2020 2020 494e 464f 2020 2020 2044 6f6e      INFO     Don
+000171a0: 6520 2020 2020 2020 2020 2020 2020 2020  e               
+000171b0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000171c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000171d0: 2063 6c69 2e70 793a 3234 320a 6060 600a   cli.py:242.```.
+000171e0: 0a54 6f20 7669 7375 616c 697a 6520 7468  .To visualize th
+000171f0: 6520 6461 7465 2c20 7765 2061 6c73 6f20  e date, we also 
+00017200: 7072 6f76 6964 6564 2061 2070 7974 686f  provided a pytho
+00017210: 6e20 7363 7269 7074 2074 6f20 706c 6f74  n script to plot
+00017220: 2074 6865 2057 4744 2064 6174 6573 2069   the WGD dates i
+00017230: 6e20 7468 6520 6077 6764 6020 666f 6c64  n the `wgd` fold
+00017240: 6572 2e20 5573 6572 7320 6e65 6564 2074  er. Users need t
+00017250: 6f20 6578 7472 6163 7420 7468 6520 7261  o extract the ra
+00017260: 7720 6461 7465 7320 6672 6f6d 2074 6865  w dates from the
+00017270: 2060 6d63 6d63 2e74 7874 6020 666f 7220   `mcmc.txt` for 
+00017280: 7468 6520 5747 4420 6e6f 6465 2066 6972  the WGD node fir
+00017290: 7374 2061 6e64 2073 6176 6520 6974 2061  st and save it a
+000172a0: 7320 6669 6c65 2060 6461 7465 732e 7478  s file `dates.tx
+000172b0: 7460 2028 6f72 2077 6861 7465 7665 7220  t` (or whatever 
+000172c0: 7072 6566 6572 7265 6420 6e61 6d65 292e  preferred name).
+000172d0: 2041 6e20 6578 616d 706c 6520 636f 6d6d   An example comm
+000172e0: 616e 6420 6973 2061 7320 6265 6c6f 772e  and is as below.
+000172f0: 0a0a 6060 600a 7079 7468 6f6e 2024 5041  ..```.python $PA
+00017300: 5448 2f70 6f73 7470 6c6f 742e 7079 2070  TH/postplot.py p
+00017310: 6f73 7464 6973 2064 6174 6573 2e74 7874  ostdis dates.txt
+00017320: 202d 2d70 6572 6365 6e74 696c 6520 3930   --percentile 90
+00017330: 202d 2d74 6974 6c65 2022 5747 4420 6461   --title "WGD da
+00017340: 7465 2220 2d2d 6870 6420 2d6f 2022 5261  te" --hpd -o "Ra
+00017350: 6e75 6e63 756c 616c 6573 5f57 4744 5f64  nunculales_WGD_d
+00017360: 6174 652e 7376 6722 0a60 6060 0a0a 215b  ate.svg".```..![
+00017370: 5d28 6461 7461 2f52 616e 756e 6375 6c61  ](data/Ranuncula
+00017380: 6c65 735f 5747 445f 6461 7465 2e73 7667  les_WGD_date.svg
+00017390: 290a 0a54 6865 2070 6f73 7465 7269 6f72  )..The posterior
+000173a0: 206d 6561 6e2c 206d 6564 6961 6e20 616e   mean, median an
+000173b0: 6420 6d6f 6465 206f 6620 7468 6520 5261  d mode of the Ra
+000173c0: 6e75 6e63 756c 616c 6573 2057 4744 2061  nunculales WGD a
+000173d0: 6765 2069 7320 3131 322e 3932 2c20 3131  ge is 112.92, 11
+000173e0: 332e 3434 2061 6e64 2031 3132 2e35 3420  3.44 and 112.54 
+000173f0: 6d79 612c 2077 6974 6820 3930 2520 4850  mya, with 90% HP
+00017400: 4420 3130 352e 3037 202d 2031 3232 2e33  D 105.07 - 122.3
+00017410: 3220 6d79 6120 6173 206d 616e 6966 6573  2 mya as manifes
+00017420: 7465 6420 6162 6f76 652e 0a0a 2323 2320  ted above...### 
+00017430: 4b73 7472 6565 0a0a 496e 2061 6464 6974  Kstree..In addit
+00017440: 696f 6e20 746f 2070 6169 7277 6973 6520  ion to pairwise 
+00017450: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00017460: 6573 7469 6d61 7469 6f6e 2c20 6120 2a4b  estimation, a *K
+00017470: 2a3c 7375 623e 533c 2f73 7562 3e20 7472  *<sub>S</sub> tr
+00017480: 6565 2077 6974 6820 6272 616e 6368 206c  ee with branch l
+00017490: 656e 6774 6820 696e 202a 4b2a 3c73 7562  ength in *K*<sub
+000174a0: 3e53 3c2f 7375 623e 2075 6e69 7420 6361  >S</sub> unit ca
+000174b0: 6e20 616c 736f 2062 6520 6465 7269 7665  n also be derive
+000174c0: 6420 6672 6f6d 2074 6865 2070 726f 6772  d from the progr
+000174d0: 616d 2060 7767 6420 6b73 6460 2067 6976  am `wgd ksd` giv
+000174e0: 656e 2074 6865 206f 7074 696f 6e20 602d  en the option `-
+000174f0: 2d6b 7374 7265 6560 2061 6e64 2060 2d2d  -kstree` and `--
+00017500: 7370 6563 6965 7374 7265 6560 2e20 4e6f  speciestree`. No
+00017510: 7465 2074 6861 7420 7468 6520 6164 6469  te that the addi
+00017520: 7469 6f6e 616c 206f 7074 696f 6e20 602d  tional option `-
+00017530: 2d6f 6e6c 7963 6f6e 6361 746b 7374 7265  -onlyconcatkstre
+00017540: 6560 2077 696c 6c20 6f6e 6c79 2063 616c  e` will only cal
+00017550: 6c20 7468 6520 2a4b 2a3c 7375 623e 533c  l the *K*<sub>S<
+00017560: 2f73 7562 3e20 6573 7469 6d61 7469 6f6e  /sub> estimation
+00017570: 2066 6f72 2074 6865 2063 6f6e 6361 7465   for the concate
+00017580: 6e61 7465 6420 616c 6967 6e6d 656e 7420  nated alignment 
+00017590: 7261 7468 6572 2074 6861 6e20 616c 6c20  rather than all 
+000175a0: 7468 6520 616c 6967 6e6d 656e 7473 2e20  the alignments. 
+000175b0: 5573 6572 7320 6e65 6564 2074 6f20 7072  Users need to pr
+000175c0: 6f76 6964 6520 6120 7072 6573 6574 2073  ovide a preset s
+000175d0: 7065 6369 6573 2074 7265 6520 666f 7220  pecies tree for 
+000175e0: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
+000175f0: 7562 3e20 7472 6565 2069 6e66 6572 656e  ub> tree inferen
+00017600: 6365 206f 6620 7468 6520 636f 6e63 6174  ce of the concat
+00017610: 656e 6174 6564 2061 6c69 676e 6d65 6e74  enated alignment
+00017620: 2077 6869 6c65 2074 6865 2072 656d 6169   while the remai
+00017630: 6e69 6e67 2061 6c69 676e 6d65 6e74 7320  ning alignments 
+00017640: 7769 6c6c 2062 6520 6167 6169 6e73 7420  will be against 
+00017650: 616e 2061 7574 6f6d 6174 656c 7920 696e  an automately in
+00017660: 6665 7272 6564 2074 7265 6520 6672 6f6d  ferred tree from
+00017670: 2060 6661 7374 7472 6565 6020 6f72 2060   `fasttree` or `
+00017680: 6971 7472 6565 602e 2049 6e20 7468 6520  iqtree`. In the 
+00017690: 656e 642c 2075 7365 7273 2077 696c 6c20  end, users will 
+000176a0: 6765 7420 6120 2a4b 2a3c 7375 623e 533c  get a *K*<sub>S<
+000176b0: 2f73 7562 3e20 7472 6565 2c20 6120 2a4b  /sub> tree, a *K
+000176c0: 2a3c 7375 623e 413c 2f73 7562 3e20 7472  *<sub>A</sub> tr
+000176d0: 6565 2061 6e64 2061 20cf 8920 7472 6565  ee and a .. tree
+000176e0: 2070 6572 2066 616d 2061 6e64 2066 6f72   per fam and for
+000176f0: 2074 6865 2063 6f6e 6361 7465 6e61 7465   the concatenate
+00017700: 6420 616c 6967 6e6d 656e 742e 0a0a 6060  d alignment...``
+00017710: 600a 7767 6420 6b73 6420 6461 7461 2f6b  `.wgd ksd data/k
+00017720: 7374 7265 655f 6461 7461 2f66 616d 2e74  stree_data/fam.t
+00017730: 7376 2064 6174 612f 6b73 7472 6565 5f64  sv data/kstree_d
+00017740: 6174 612f 4163 6f72 7573 5f74 6174 6172  ata/Acorus_tatar
+00017750: 696e 6f77 6969 2064 6174 612f 6b73 7472  inowii data/kstr
+00017760: 6565 5f64 6174 612f 416d 626f 7265 6c6c  ee_data/Amborell
+00017770: 615f 7472 6963 686f 706f 6461 2064 6174  a_trichopoda dat
+00017780: 612f 6b73 7472 6565 5f64 6174 612f 4171  a/kstree_data/Aq
+00017790: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000177a0: 2064 6174 612f 6b73 7472 6565 5f64 6174   data/kstree_dat
+000177b0: 612f 4172 6973 746f 6c6f 6368 6961 5f66  a/Aristolochia_f
+000177c0: 696d 6272 6961 7461 2064 6174 612f 6b73  imbriata data/ks
+000177d0: 7472 6565 5f64 6174 612f 4379 6361 735f  tree_data/Cycas_
+000177e0: 7061 6e7a 6869 6875 6165 6e73 6920 2d2d  panzhihuaensi --
+000177f0: 6b73 7472 6565 202d 2d73 7065 6369 6573  kstree --species
+00017800: 7472 6565 2064 6174 612f 6b73 7472 6565  tree data/kstree
+00017810: 5f64 6174 612f 7370 6563 6965 735f 7472  _data/species_tr
+00017820: 6565 312e 6e77 202d 2d6f 6e6c 7963 6f6e  ee1.nw --onlycon
+00017830: 6361 746b 7374 7265 6520 2d6f 2077 6764  catkstree -o wgd
+00017840: 5f6b 7374 7265 655f 746f 706f 6c6f 6779  _kstree_topology
+00017850: 310a 6060 600a 0a21 5b5d 2864 6174 612f  1.```..![](data/
+00017860: 6b73 7472 6565 5f72 6573 756c 7473 2f6b  kstree_results/k
+00017870: 7374 7265 652e 7376 6729 0a0a 4162 6f76  stree.svg)..Abov
+00017880: 6520 7765 2075 7365 6420 7468 7265 6520  e we used three 
+00017890: 616c 7465 726e 6174 6976 6520 746f 706f  alternative topo
+000178a0: 6c6f 6769 6573 2074 6f20 696e 6665 7220  logies to infer 
+000178b0: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
+000178c0: 7562 3e20 7472 6565 2077 6869 6368 206c  ub> tree which l
+000178d0: 6564 2074 6f20 6469 6666 6572 656e 7420  ed to different 
+000178e0: 6272 616e 6368 206c 656e 6774 6820 6573  branch length es
+000178f0: 7469 6d61 7469 6f6e 2e20 4e6f 7465 2074  timation. Note t
+00017900: 6861 7420 7468 6520 6661 6d69 6c69 6573  hat the families
+00017910: 2077 6520 7573 6564 2077 6572 6520 6f6e   we used were on
+00017920: 6c79 2074 776f 2067 6c6f 6261 6c20 4d52  ly two global MR
+00017930: 4248 2066 616d 696c 6965 7320 666f 7220  BH families for 
+00017940: 7468 6520 7075 7270 6f73 6520 6f66 2069  the purpose of i
+00017950: 6c6c 7573 7472 6174 696f 6e2e 2054 6f20  llustration. To 
+00017960: 6163 7175 6972 6520 616e 2061 6363 7572  acquire an accur
+00017970: 6174 6520 7072 6f66 696c 6520 6f66 2074  ate profile of t
+00017980: 6865 2073 7562 7374 6974 7574 696f 6e20  he substitution 
+00017990: 7261 7465 2076 6172 6961 7469 6f6e 2c20  rate variation, 
+000179a0: 6f72 7468 6f6c 6f67 7565 7320 6174 2074  orthologues at t
+000179b0: 6865 2077 686f 6c65 2067 656e 6f6d 6520  he whole genome 
+000179c0: 7363 616c 6520 7368 6f75 6c64 2062 6520  scale should be 
+000179d0: 7573 6564 2e0a 0a23 2320 4369 7461 7469  used...## Citati
+000179e0: 6f6e 0a20 0a50 6c65 6173 6520 6369 7465  on. .Please cite
+000179f0: 2075 7320 6174 2068 7474 7073 3a2f 2f64   us at https://d
+00017a00: 6f69 2e6f 7267 2f31 302e 3130 3037 2f39  oi.org/10.1007/9
+00017a10: 3738 2d31 2d30 3731 362d 3235 3631 2d33  78-1-0716-2561-3
+00017a20: 5f31 2061 6e64 2068 7474 7073 3a2f 2f64  _1 and https://d
+00017a30: 6f69 2e6f 7267 2f31 302e 3130 3933 2f62  oi.org/10.1093/b
+00017a40: 696f 696e 666f 726d 6174 6963 732f 6274  ioinformatics/bt
+00017a50: 6165 3237 322e 0a0a 6060 600a 4865 6e67  ae272...```.Heng
+00017a60: 6368 6920 4368 656e 2c20 4172 7468 7572  chi Chen, Arthur
+00017a70: 205a 7761 656e 6570 6f65 6c20 2832 3032   Zwaenepoel (202
+00017a80: 3329 2e20 496e 6665 7265 6e63 6520 6f66  3). Inference of
+00017a90: 2041 6e63 6965 6e74 2050 6f6c 7970 6c6f   Ancient Polyplo
+00017aa0: 6964 7920 6672 6f6d 2047 656e 6f6d 6963  idy from Genomic
+00017ab0: 2044 6174 612e 2049 6e3a 2056 616e 2064   Data. In: Van d
+00017ac0: 6520 5065 6572 2c20 592e 2028 6564 7329  e Peer, Y. (eds)
+00017ad0: 2050 6f6c 7970 6c6f 6964 792e 204d 6574   Polyploidy. Met
+00017ae0: 686f 6473 2069 6e20 4d6f 6c65 6375 6c61  hods in Molecula
+00017af0: 7220 4269 6f6c 6f67 792c 2076 6f6c 2032  r Biology, vol 2
+00017b00: 3534 352e 2048 756d 616e 612c 204e 6577  545. Humana, New
+00017b10: 2059 6f72 6b2c 204e 592e 2068 7474 7073   York, NY. https
+00017b20: 3a2f 2f64 6f69 2e6f 7267 2f31 302e 3130  ://doi.org/10.10
+00017b30: 3037 2f39 3738 2d31 2d30 3731 362d 3235  07/978-1-0716-25
+00017b40: 3631 2d33 5f31 0a48 656e 6763 6869 2043  61-3_1.Hengchi C
+00017b50: 6865 6e2c 2041 7274 6875 7220 5a77 6165  hen, Arthur Zwae
+00017b60: 6e65 706f 656c 2c20 5976 6573 2056 616e  nepoel, Yves Van
+00017b70: 2064 6520 5065 6572 2028 3230 3234 292e   de Peer (2024).
+00017b80: 2077 6764 2076 323a 2061 2073 7569 7465   wgd v2: a suite
+00017b90: 206f 6620 746f 6f6c 7320 746f 2075 6e63   of tools to unc
+00017ba0: 6f76 6572 2061 6e64 2064 6174 6520 616e  over and date an
+00017bb0: 6369 656e 7420 706f 6c79 706c 6f69 6479  cient polyploidy
+00017bc0: 2061 6e64 2077 686f 6c65 2d67 656e 6f6d   and whole-genom
+00017bd0: 6520 6475 706c 6963 6174 696f 6e2e 2042  e duplication. B
+00017be0: 696f 696e 666f 726d 6174 6963 732c 2068  ioinformatics, h
+00017bf0: 7474 7073 3a2f 2f64 6f69 2e6f 7267 2f31  ttps://doi.org/1
+00017c00: 302e 3130 3933 2f62 696f 696e 666f 726d  0.1093/bioinform
+00017c10: 6174 6963 732f 6274 6165 3237 320a 6060  atics/btae272.``
+00017c20: 600a 0a46 6f72 2063 6974 6174 696f 6e20  `..For citation 
+00017c30: 6f66 2074 6865 2074 6f6f 6c73 2075 7365  of the tools use
+00017c40: 6420 696e 2077 6764 2c20 706c 6561 7365  d in wgd, please
+00017c50: 2063 6f6e 7375 6c74 2074 6865 2064 6f63   consult the doc
+00017c60: 756d 656e 7461 7469 6f6e 2061 740a 6874  umentation at.ht
+00017c70: 7470 733a 2f2f 7767 6476 322e 7265 6164  tps://wgdv2.read
+00017c80: 7468 6564 6f63 732e 696f 2f65 6e2f 6c61  thedocs.io/en/la
+00017c90: 7465 7374 2f63 6974 6174 696f 6e2e 6874  test/citation.ht
+00017ca0: 6d6c 2e0a 0a                             ml...
```

### Comparing `wgd-2.0.34/README.md` & `wgd-2.0.35/PKG-INFO`

 * *Files 3% similar despite different names*

```diff
@@ -1,6100 +1,6106 @@
-00000000: 3c64 6976 2061 6c69 676e 3d22 6365 6e74  <div align="cent
-00000010: 6572 223e 0a0a 2320 6077 6764 2076 3260  er">..# `wgd v2`
-00000020: 203a 2061 2073 7569 7465 2074 6f6f 6c20   : a suite tool 
-00000030: 6f66 2057 4744 2069 6e66 6572 656e 6365  of WGD inference
-00000040: 2061 6e64 2074 696d 696e 670a 0a5b 215b   and timing..[![
-00000050: 4275 696c 6420 5374 6174 7573 5d28 6874  Build Status](ht
-00000060: 7470 733a 2f2f 6170 702e 7472 6176 6973  tps://app.travis
-00000070: 2d63 692e 636f 6d2f 6865 6368 652d 7073  -ci.com/heche-ps
-00000080: 622f 7767 642e 7376 673f 6272 616e 6368  b/wgd.svg?branch
-00000090: 3d70 6879 6c6f 6461 7469 6e67 295d 2868  =phylodating)](h
-000000a0: 7474 7073 3a2f 2f61 7070 2e74 7261 7669  ttps://app.travi
-000000b0: 732d 6369 2e63 6f6d 2f68 6563 6865 2d70  s-ci.com/heche-p
-000000c0: 7362 2f77 6764 290a 5b21 5b44 6f63 756d  sb/wgd).[![Docum
-000000d0: 656e 7461 7469 6f6e 2053 7461 7475 735d  entation Status]
-000000e0: 2868 7474 7073 3a2f 2f72 6561 6474 6865  (https://readthe
-000000f0: 646f 6373 2e6f 7267 2f70 726f 6a65 6374  docs.org/project
-00000100: 732f 7767 6476 322f 6261 6467 652f 3f76  s/wgdv2/badge/?v
-00000110: 6572 7369 6f6e 3d6c 6174 6573 7429 5d28  ersion=latest)](
-00000120: 6874 7470 733a 2f2f 7767 6476 322e 7265  https://wgdv2.re
-00000130: 6164 7468 6564 6f63 732e 696f 2f65 6e2f  adthedocs.io/en/
-00000140: 6c61 7465 7374 2f3f 6261 6467 653d 6c61  latest/?badge=la
-00000150: 7465 7374 290a 5b21 5b6c 6963 656e 7365  test).[![license
-00000160: 5d28 6874 7470 733a 2f2f 696d 672e 7368  ](https://img.sh
-00000170: 6965 6c64 732e 696f 2f70 7970 692f 6c2f  ields.io/pypi/l/
-00000180: 7767 642e 7376 6729 5d28 6874 7470 733a  wgd.svg)](https:
-00000190: 2f2f 7079 7069 2e70 7974 686f 6e2e 6f72  //pypi.python.or
-000001a0: 672f 7079 7069 2f77 6764 290a 5b21 5b4c  g/pypi/wgd).[![L
-000001b0: 6174 6573 7420 5079 5049 2076 6572 7369  atest PyPI versi
-000001c0: 6f6e 5d28 6874 7470 733a 2f2f 696d 672e  on](https://img.
-000001d0: 7368 6965 6c64 732e 696f 2f70 7970 692f  shields.io/pypi/
-000001e0: 762f 7767 642e 7376 6729 5d28 6874 7470  v/wgd.svg)](http
-000001f0: 733a 2f2f 7079 7069 2e70 7974 686f 6e2e  s://pypi.python.
-00000200: 6f72 672f 7079 7069 2f77 6764 290a 5b21  org/pypi/wgd).[!
-00000210: 5b69 6e73 7461 6c6c 2077 6974 6820 6269  [install with bi
-00000220: 6f63 6f6e 6461 5d28 6874 7470 733a 2f2f  oconda](https://
-00000230: 696d 672e 7368 6965 6c64 732e 696f 2f62  img.shields.io/b
-00000240: 6164 6765 2f69 6e73 7461 6c6c 2532 3077  adge/install%20w
-00000250: 6974 682d 6269 6f63 6f6e 6461 2d62 7269  ith-bioconda-bri
-00000260: 6768 7467 7265 656e 2e73 7667 3f73 7479  ghtgreen.svg?sty
-00000270: 6c65 3d66 6c61 7429 5d28 6874 7470 3a2f  le=flat)](http:/
-00000280: 2f62 696f 636f 6e64 612e 6769 7468 7562  /bioconda.github
-00000290: 2e69 6f2f 7265 6369 7065 732f 7767 642f  .io/recipes/wgd/
-000002a0: 5245 4144 4d45 2e68 746d 6c29 0a5b 215b  README.html).[![
-000002b0: 446f 776e 6c6f 6164 735d 2868 7474 7073  Downloads](https
-000002c0: 3a2f 2f70 6570 792e 7465 6368 2f62 6164  ://pepy.tech/bad
-000002d0: 6765 2f77 6764 295d 2868 7474 7073 3a2f  ge/wgd)](https:/
-000002e0: 2f70 6570 792e 7465 6368 2f70 726f 6a65  /pepy.tech/proje
-000002f0: 6374 2f77 6764 290a 0a2a 2a48 656e 6763  ct/wgd)..**Hengc
-00000300: 6869 2043 6865 6e2c 2041 7274 6875 7220  hi Chen, Arthur 
-00000310: 5a77 6165 6e65 706f 656c 2c20 5976 6573  Zwaenepoel, Yves
-00000320: 2056 616e 2064 6520 5065 6572 2a2a 0a0a   Van de Peer**..
-00000330: 5b2a 2a42 696f 696e 666f 726d 6174 6963  [**Bioinformatic
-00000340: 7320 2620 4576 6f6c 7574 696f 6e61 7279  s & Evolutionary
-00000350: 2047 656e 6f6d 6963 7320 4772 6f75 702a   Genomics Group*
-00000360: 2a5d 2868 7474 7073 3a2f 2f77 7777 2e76  *](https://www.v
-00000370: 616e 6465 7065 6572 6c61 622e 6f72 672f  andepeerlab.org/
-00000380: 292a 2a2c 2056 4942 2d55 4765 6e74 2043  )**, VIB-UGent C
-00000390: 656e 7465 7220 666f 7220 506c 616e 7420  enter for Plant 
-000003a0: 5379 7374 656d 7320 4269 6f6c 6f67 792a  Systems Biology*
-000003b0: 2a0a 0a5b 2a2a 496e 7472 6f64 7563 7469  *..[**Introducti
-000003c0: 6f6e 2a2a 5d28 2369 6e74 726f 6475 6374  on**](#introduct
-000003d0: 696f 6e29 207c 200a 5b2a 2a49 6e73 7461  ion) | .[**Insta
-000003e0: 6c6c 6174 696f 6e2a 2a5d 2823 696e 7374  llation**](#inst
-000003f0: 616c 6c61 7469 6f6e 2920 7c20 0a5b 2a2a  allation) | .[**
-00000400: 5061 7261 6d65 7465 7273 2a2a 5d28 2370  Parameters**](#p
-00000410: 6172 616d 6574 6572 7329 207c 200a 5b2a  arameters) | .[*
-00000420: 2a55 7361 6765 2a2a 5d28 2375 7361 6765  *Usage**](#usage
-00000430: 2920 7c20 0a5b 2a2a 496c 6c75 7374 7261  ) | .[**Illustra
-00000440: 7469 6f6e 2a2a 5d28 2369 6c6c 7573 7472  tion**](#illustr
-00000450: 6174 696f 6e29 207c 0a5b 2a2a 446f 6375  ation) |.[**Docu
-00000460: 6d65 6e74 6174 696f 6e2a 2a5d 2868 7474  mentation**](htt
-00000470: 7073 3a2f 2f77 6764 7632 2e72 6561 6474  ps://wgdv2.readt
-00000480: 6865 646f 6373 2e69 6f2f 656e 2f6c 6174  hedocs.io/en/lat
-00000490: 6573 742f 3f62 6164 6765 3d6c 6174 6573  est/?badge=lates
-000004a0: 7429 207c 0a5b 2a2a 4369 7461 7469 6f6e  t) |.[**Citation
-000004b0: 2a2a 5d28 2363 6974 6174 696f 6e29 0a0a  **](#citation)..
-000004c0: 3c2f 6469 763e 0a0a 6077 6764 2076 3260  </div>..`wgd v2`
-000004d0: 2069 7320 6120 7079 7468 6f6e 2070 6163   is a python pac
-000004e0: 6b61 6765 2075 7067 7261 6465 6420 6672  kage upgraded fr
-000004f0: 6f6d 2074 6865 206f 7269 6769 6e61 6c20  om the original 
-00000500: 6077 6764 6020 7061 636b 6167 6520 6169  `wgd` package ai
-00000510: 6d69 6e67 2066 6f72 2074 6865 2069 6e66  ming for the inf
-00000520: 6572 656e 6365 2061 6e64 2074 696d 696e  erence and timin
-00000530: 6720 6f66 2061 6e63 6965 6e74 2077 686f  g of ancient who
-00000540: 6c65 2d67 656e 6f6d 6520 6475 706c 6963  le-genome duplic
-00000550: 6174 696f 6e20 2857 4744 2920 6576 656e  ation (WGD) even
-00000560: 7473 2e20 466f 7220 7468 6520 7072 6f70  ts. For the prop
-00000570: 6f73 6520 6f66 2069 6c6c 7573 7472 6174  ose of illustrat
-00000580: 696e 6720 7468 6520 7072 696e 6369 706c  ing the principl
-00000590: 6520 616e 6420 7573 6167 6520 6f66 2060  e and usage of `
-000005a0: 7767 6420 7632 602c 2077 6520 636f 6d70  wgd v2`, we comp
-000005b0: 696c 6564 2074 6869 7320 646f 6375 6d65  iled this docume
-000005c0: 6e74 6174 696f 6e2e 2042 656c 6f77 2077  ntation. Below w
-000005d0: 6520 6669 7273 7420 6761 7665 2061 6e20  e first gave an 
-000005e0: 696e 7472 6f64 7563 7469 6f6e 206f 7665  introduction ove
-000005f0: 7220 7468 6520 7363 6f70 6520 616e 6420  r the scope and 
-00000600: 6d65 6368 616e 6973 6d20 6f66 2060 7767  mechanism of `wg
-00000610: 6420 7632 6020 616e 6420 7468 656e 2074  d v2` and then t
-00000620: 6865 2070 7261 6374 6963 616c 2069 6e66  he practical inf
-00000630: 6f72 6d61 7469 6f6e 2070 6572 7461 696e  ormation pertain
-00000640: 696e 6720 746f 2074 6865 2069 6e73 7461  ing to the insta
-00000650: 6c6c 6174 696f 6e20 616e 6420 7573 6167  llation and usag
-00000660: 652e 2041 6e20 6578 616d 706c 6172 2077  e. An examplar w
-00000670: 6f72 6b66 6c6f 7720 6973 2070 726f 7669  orkflow is provi
-00000680: 6465 6420 696e 2074 6865 2074 7574 6f72  ded in the tutor
-00000690: 6961 6c20 7365 6374 696f 6e20 6f6e 2068  ial section on h
-000006a0: 6f77 2074 6f20 7365 656b 2065 7669 6465  ow to seek evide
-000006b0: 6e63 6520 666f 7220 6120 7075 7461 7469  nce for a putati
-000006c0: 7665 2057 4744 2065 7665 6e74 2061 6e64  ve WGD event and
-000006d0: 2070 6572 666f 726d 2070 726f 7065 7220   perform proper 
-000006e0: 7469 6d69 6e67 2077 6974 6820 6120 6672  timing with a fr
-000006f0: 6573 686c 7920 6f62 7461 696e 6564 2067  eshly obtained g
-00000700: 656e 6f6d 6520 6173 7365 6d62 6c79 2069  enome assembly i
-00000710: 6e20 6861 6e64 2e20 466f 7220 7468 6f73  n hand. For thos
-00000720: 6520 7768 6f20 6172 6520 696e 7465 7265  e who are intere
-00000730: 7374 6564 2069 6e20 6d6f 7265 2074 6865  sted in more the
-00000740: 6f72 6574 6963 616c 2064 6574 6169 6c73  oretical details
-00000750: 2c20 7765 2072 6563 6f6d 6d65 6e64 2074  , we recommend t
-00000760: 7572 6e69 6e67 2074 6f20 6f75 7220 7061  urning to our pa
-00000770: 7065 7220 616e 6420 626f 6f6b 2063 6861  per and book cha
-00000780: 7074 6572 2066 6f72 206d 6f72 6520 6465  pter for more de
-00000790: 7461 696c 6564 2064 6573 6372 6970 7469  tailed descripti
-000007a0: 6f6e 2061 6e64 2069 6e73 6967 6874 6675  on and insightfu
-000007b0: 6c20 6469 7363 7573 7369 6f6e 732e 2054  l discussions. T
-000007c0: 6865 206b 6579 2069 6d70 726f 7665 6420  he key improved 
-000007d0: 6665 6174 7572 6573 206f 6620 6077 6764  features of `wgd
-000007e0: 2076 3260 2061 7265 2064 656d 6f6e 7374   v2` are demonst
-000007f0: 7261 7465 6420 696e 206f 7572 206c 6174  rated in our lat
-00000800: 6573 7420 6d61 6e75 7363 7269 7074 206f  est manuscript o
-00000810: 6e20 5b42 696f 696e 666f 726d 6174 6963  n [Bioinformatic
-00000820: 735d 2868 7474 7073 3a2f 2f64 6f69 2e6f  s](https://doi.o
-00000830: 7267 2f31 302e 3130 3933 2f62 696f 696e  rg/10.1093/bioin
-00000840: 666f 726d 6174 6963 732f 6274 6165 3237  formatics/btae27
-00000850: 3229 2e20 4966 2079 6f75 2075 7365 2060  2). If you use `
-00000860: 7767 6420 7632 6020 696e 2079 6f75 7220  wgd v2` in your 
-00000870: 7265 7365 6172 6368 2c20 706c 6561 7365  research, please
-00000880: 2063 6974 6520 7573 2061 7320 7375 6767   cite us as sugg
-00000890: 6573 7465 6420 696e 205b 4369 7461 7469  ested in [Citati
-000008a0: 6f6e 5d28 2363 6974 6174 696f 6e29 2073  on](#citation) s
-000008b0: 6563 7469 6f6e 2e0a 0a23 2320 496e 7472  ection...## Intr
-000008c0: 6f64 7563 7469 6f6e 0a0a 506f 6c79 706c  oduction..Polypl
-000008d0: 6f69 6469 7a61 7469 6f6e 732c 2074 6865  oidizations, the
-000008e0: 2065 766f 6c75 7469 6f6e 6172 7920 7072   evolutionary pr
-000008f0: 6f63 6573 7320 7468 6174 2074 6865 2065  ocess that the e
-00000900: 6e74 6972 6520 6765 6e6f 6d65 206f 6620  ntire genome of 
-00000910: 616e 206f 7267 616e 6973 6d20 6973 2064  an organism is d
-00000920: 7570 6c69 6361 7465 642c 2061 6c73 6f20  uplicated, also 
-00000930: 6e61 6d65 6420 6173 2077 686f 6c65 2d67  named as whole-g
-00000940: 656e 6f6d 6520 6475 706c 6963 6174 696f  enome duplicatio
-00000950: 6e73 2028 5747 4473 292c 206f 6363 7572  ns (WGDs), occur
-00000960: 2072 6563 7572 7265 6e74 6c79 2061 6372   recurrently acr
-00000970: 6f73 7320 7468 6520 7472 6565 206f 6620  oss the tree of 
-00000980: 6c69 6665 2e20 5468 6572 6520 6172 6520  life. There are 
-00000990: 7477 6f20 6d6f 6465 7320 6f66 2070 6f6c  two modes of pol
-000009a0: 7970 6c6f 6964 697a 6174 696f 6e73 2c20  yploidizations, 
-000009b0: 6175 746f 706f 6c79 706c 6f69 6469 7a61  autopolyploidiza
-000009c0: 7469 6f6e 7320 616e 6420 616c 6c6f 706f  tions and allopo
-000009d0: 6c79 706c 6f69 6469 7a61 7469 6f6e 732e  lyploidizations.
-000009e0: 2041 7574 6f70 6f6c 7970 6c6f 6964 697a   Autopolyploidiz
-000009f0: 6174 696f 6e73 2061 7265 2074 6865 2064  ations are the d
-00000a00: 7570 6c69 6361 7469 6f6e 206f 6620 7468  uplication of th
-00000a10: 6520 7361 6d65 2067 656e 6f6d 652c 2072  e same genome, r
-00000a20: 6573 756c 7469 6e67 2069 6e20 7477 6f20  esulting in two 
-00000a30: 6964 656e 7469 6361 6c20 7375 6267 656e  identical subgen
-00000a40: 6f6d 6573 2061 7420 7468 6520 7469 6d65  omes at the time
-00000a50: 2069 7420 656d 6572 6765 642e 2057 6869   it emerged. Whi
-00000a60: 6c65 2074 6865 2061 6c6c 6f70 6f6c 7970  le the allopolyp
-00000a70: 6c6f 6964 697a 6174 696f 6e73 2061 7265  loidizations are
-00000a80: 206e 6f72 6d61 6c6c 7920 6163 6869 6576   normally achiev
-00000a90: 6564 2069 6e20 7477 6f20 7374 6570 732c  ed in two steps,
-00000aa0: 2066 6972 7374 2074 6865 2068 7962 7269   first the hybri
-00000ab0: 6469 7a61 7469 6f6e 2062 6574 7765 656e  dization between
-00000ac0: 2074 776f 2064 6966 6665 7265 6e74 2073   two different s
-00000ad0: 7065 6369 6573 2c20 7265 7375 6c74 696e  pecies, resultin
-00000ae0: 6720 696e 2074 6865 2061 7269 7369 6e67  g in the arising
-00000af0: 206f 6620 7472 616e 7369 656e 7420 686f   of transient ho
-00000b00: 6d6f 706c 6f69 6479 2c73 6563 6f6e 6420  moploidy,second 
-00000b10: 7468 6520 6475 706c 6963 6174 696f 6e20  the duplication 
-00000b20: 6f66 2074 6865 2068 6f6d 6f70 6c6f 6964  of the homoploid
-00000b30: 792c 2072 6573 756c 7469 6e67 2069 6e20  y, resulting in 
-00000b40: 7468 6520 656d 6572 6765 6e63 6520 6f66  the emergence of
-00000b50: 2061 6c6c 6f70 6f6c 7970 6c6f 6964 792e   allopolyploidy.
-00000b60: 2044 7565 2074 6f20 7468 6520 756e 7374   Due to the unst
-00000b70: 6162 696c 6974 7920 616e 6420 756e 6261  ability and unba
-00000b80: 6c61 6e63 6564 2074 6574 7261 736f 6d69  lanced tetrasomi
-00000b90: 6320 696e 6865 7269 7461 6e63 652c 2066  c inheritance, f
-00000ba0: 6f72 2069 6e73 7461 6e63 6520 7468 6520  or instance the 
-00000bb0: 6e75 636c 6561 722d 6379 746f 706c 6173  nuclear-cytoplas
-00000bc0: 6d69 6320 696e 636f 6d70 6174 6962 696c  mic incompatibil
-00000bd0: 6974 792c 2074 6865 2070 6f6c 7970 6c6f  ity, the polyplo
-00000be0: 6964 7920 6765 6e6f 6d65 2077 696c 6c20  idy genome will 
-00000bf0: 7468 656e 2065 7870 6572 6965 6e63 6520  then experience 
-00000c00: 6120 7072 6f63 6573 7320 6361 6c6c 6564  a process called
-00000c10: 2064 6970 6c6f 6964 697a 6174 696f 6e2c   diploidization,
-00000c20: 2061 6c73 6f20 6e61 6d65 6420 6173 2066   also named as f
-00000c30: 7261 6374 696f 6e61 7469 6f6e 2c20 6475  ractionation, du
-00000c40: 7269 6e67 2077 6869 6368 2061 206c 6172  ring which a lar
-00000c50: 6765 2070 6f72 7469 6f6e 206f 6620 6765  ge portion of ge
-00000c60: 6e65 2064 7570 6c69 6361 7465 7320 7769  ne duplicates wi
-00000c70: 6c6c 2067 6574 206c 6f73 7420 616e 6420  ll get lost and 
-00000c80: 6f6e 6c79 2061 2066 7261 6374 696f 6e20  only a fraction 
-00000c90: 6361 6e20 6265 2072 6574 6169 6e65 642e  can be retained.
-00000ca0: 2054 6865 2074 7261 6365 7320 6f66 2070   The traces of p
-00000cb0: 6f6c 7970 6c6f 6964 697a 6174 696f 6e73  olyploidizations
-00000cc0: 2063 616e 2062 6520 7468 7573 2075 6e65   can be thus une
-00000cd0: 6172 7468 6564 2066 726f 6d20 7468 6573  arthed from thes
-00000ce0: 6520 7265 7461 696e 6564 2067 656e 6520  e retained gene 
-00000cf0: 6475 706c 6963 6174 6573 2e20 5468 7265  duplicates. Thre
-00000d00: 6520 6170 7072 6f61 6368 6573 2062 6173  e approaches bas
-00000d10: 6564 206f 6e20 6765 6e65 2064 7570 6c69  ed on gene dupli
-00000d20: 6361 7465 732c 206e 616d 656c 792c 202a  cates, namely, *
-00000d30: 4b2a 3c73 7562 3e53 3c2f 7375 623e 206d  K*<sub>S</sub> m
-00000d40: 6574 686f 642c 2067 656e 6520 7472 6565  ethod, gene tree
-00000d50: 202d 2073 7065 6369 6573 2074 7265 6520   - species tree 
-00000d60: 7265 636f 6e63 696c 6961 7469 6f6e 206d  reconciliation m
-00000d70: 6574 686f 6420 616e 6420 7379 6e74 656e  ethod and synten
-00000d80: 7920 6d65 7468 6f64 2c20 6172 6520 636f  y method, are co
-00000d90: 6d6d 6f6e 6c79 2075 7365 6420 696e 2064  mmonly used in d
-00000da0: 6574 6563 7469 6e67 2065 7669 6465 6e63  etecting evidenc
-00000db0: 6520 666f 7220 5747 4473 2e20 5468 6520  e for WGDs. The 
-00000dc0: 6765 6e65 2074 7265 6520 2d20 7370 6563  gene tree - spec
-00000dd0: 6965 7320 7472 6565 2072 6563 6f6e 6369  ies tree reconci
-00000de0: 6c69 6174 696f 6e20 6d65 7468 6f64 2069  liation method i
-00000df0: 7320 6e6f 7420 7769 7468 696e 2074 6865  s not within the
-00000e00: 2073 636f 7065 206f 6620 6077 6764 2076   scope of `wgd v
-00000e10: 3260 2c20 6275 7420 7765 206b 696e 646c  2`, but we kindl
-00000e20: 7920 7265 6665 7220 7265 6164 6572 7320  y refer readers 
-00000e30: 7768 6f20 6172 6520 696e 7465 7265 7374  who are interest
-00000e40: 6564 2074 6f20 7468 6520 7068 796c 6f67  ed to the phylog
-00000e50: 656e 6f6d 6963 2070 726f 6772 616d 2064  enomic program d
-00000e60: 6576 656c 6f70 6564 2062 7920 4172 7468  eveloped by Arth
-00000e70: 7572 205a 7761 656e 6570 6f65 6c20 6e61  ur Zwaenepoel na
-00000e80: 6d65 6420 5b57 4841 4c45 5d28 6874 7470  med [WHALE](http
-00000e90: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f61  s://github.com/a
-00000ea0: 727a 7761 2f57 6861 6c65 2e6a 6c29 2061  rzwa/Whale.jl) a
-00000eb0: 6e64 2074 6865 2061 7373 6f63 6961 7465  nd the associate
-00000ec0: 6420 5b70 6170 6572 5d28 6874 7470 733a  d [paper](https:
-00000ed0: 2f2f 646f 692e 6f72 672f 3130 2e31 3039  //doi.org/10.109
-00000ee0: 332f 6d6f 6c62 6576 2f6d 737a 3038 3829  3/molbev/msz088)
-00000ef0: 2066 6f72 206d 6f72 6520 7465 6368 6e69   for more techni
-00000f00: 6361 6c20 616e 6420 7468 656f 7265 7469  cal and theoreti
-00000f10: 6361 6c20 6465 7461 696c 732e 0a0a 5468  cal details...Th
-00000f20: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-00000f30: 3e20 6d65 7468 6f64 2069 7320 6573 7461  > method is esta
-00000f40: 626c 6973 6865 6420 6f6e 2061 206d 6f64  blished on a mod
-00000f50: 656c 206f 6620 6765 6e65 2066 616d 696c  el of gene famil
-00000f60: 7920 6576 6f6c 7574 696f 6e20 7468 6174  y evolution that
-00000f70: 2065 6163 6820 6765 6e65 2066 616d 696c   each gene famil
-00000f80: 7920 6973 2061 6c6c 6f77 6564 2074 6f20  y is allowed to 
-00000f90: 6576 6f6c 7665 2076 6961 2067 656e 6520  evolve via gene 
-00000fa0: 6475 706c 6963 6174 696f 6e20 616e 6420  duplication and 
-00000fb0: 6c6f 7373 2e20 4e6f 7465 2074 6861 7420  loss. Note that 
-00000fc0: 7468 6520 6765 6e65 2066 616d 696c 7920  the gene family 
-00000fd0: 6865 7265 2069 7320 6173 7375 6d65 6420  here is assumed 
-00000fe0: 746f 2062 6520 7468 6520 636c 7573 7465  to be the cluste
-00000ff0: 7220 6f66 2061 6c6c 2067 656e 6573 2064  r of all genes d
-00001000: 6573 6365 6e64 6564 2066 726f 6d20 616e  escended from an
-00001010: 2061 6e63 6573 7472 616c 2067 656e 6520   ancestral gene 
-00001020: 696e 2061 2073 696e 676c 6520 6765 6e6f  in a single geno
-00001030: 6d65 2e20 5265 636f 7665 7269 6e67 2074  me. Recovering t
-00001040: 6865 2067 656e 6520 7472 6565 206f 6620  he gene tree of 
-00001050: 7375 6368 2067 656e 6520 6661 6d69 6c79  such gene family
-00001060: 2069 6e66 6f72 6d73 2074 6865 2074 696d   informs the tim
-00001070: 696e 672c 2073 6369 6c69 6365 7420 7468  ing, scilicet th
-00001080: 6520 6167 652c 206f 6620 6765 6e65 2064  e age, of gene d
-00001090: 7570 6c69 6361 7469 6f6e 2065 7665 6e74  uplication event
-000010a0: 732e 2054 6865 2061 6765 2072 6566 6572  s. The age refer
-000010b0: 6564 2068 6572 652c 2069 7320 6e6f 7420  ed here, is not 
-000010c0: 696e 2072 6561 6c20 6765 6f6c 6f67 6963  in real geologic
-000010d0: 616c 2074 696d 652c 2062 7574 2069 6e20  al time, but in 
-000010e0: 7468 6520 756e 6974 206f 6620 6576 6f6c  the unit of evol
-000010f0: 7574 696f 6e61 7279 2064 6973 7461 6e63  utionary distanc
-00001100: 652c 2069 2e65 2e2c 2074 6865 206e 756d  e, i.e., the num
-00001110: 6265 7220 6f66 2073 7562 7374 6974 7574  ber of substitut
-00001120: 696f 6e73 2070 6572 2073 6974 652e 2057  ions per site. W
-00001130: 6865 6e20 7468 6520 6576 6f6c 7574 696f  hen the evolutio
-00001140: 6e61 7279 2072 6174 6520 7265 6d61 696e  nary rate remain
-00001150: 7320 6170 7072 6f78 696d 6174 656c 7920  s approximately 
-00001160: 636f 6e73 7461 6e74 2c20 7468 6520 6576  constant, the ev
-00001170: 6f6c 7574 696f 6e61 7279 2064 6973 7461  olutionary dista
-00001180: 6e63 6520 6973 2074 6865 6e20 7375 7070  nce is then supp
-00001190: 6f73 6564 2074 6f20 6265 2070 726f 706f  osed to be propo
-000011a0: 7274 696f 6e61 6c20 746f 2074 6865 2072  rtional to the r
-000011b0: 6561 6c20 6576 6f6c 7574 696f 6e61 7279  eal evolutionary
-000011c0: 2074 696d 652e 2054 6865 2073 796e 6f6e   time. The synon
-000011d0: 796d 6f75 7320 6469 7374 616e 6365 202a  ymous distance *
-000011e0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2c20  K*<sub>S</sub>, 
-000011f0: 7468 6520 6e75 6d62 6572 206f 6620 7379  the number of sy
-00001200: 6e6f 6e79 6d6f 7573 2073 7562 7374 6974  nonymous substit
-00001210: 7574 696f 6e73 2070 6572 2073 796e 6f6e  utions per synon
-00001220: 796d 6f75 7320 7369 7465 2c20 6973 2073  ymous site, is s
-00001230: 7563 6820 6361 6e64 6964 6174 6520 7468  uch candidate th
-00001240: 6174 2073 796e 6f6e 796d 6f75 7320 7375  at synonymous su
-00001250: 6273 7469 7475 7469 6f6e 7320 776f 756c  bstitutions woul
-00001260: 6420 6e6f 7420 696e 6375 7220 7468 6520  d not incur the 
-00001270: 6368 616e 6765 206f 6620 616d 696e 6f20  change of amino 
-00001280: 6163 6964 2061 6e64 2061 7265 2074 6875  acid and are thu
-00001290: 7320 7265 6761 7264 6564 2061 7320 6e65  s regarded as ne
-000012a0: 7574 7261 6c2c 2077 6869 6368 2061 6363  utral, which acc
-000012b0: 6f72 6469 6e67 2074 6f20 7468 6520 6e65  ording to the ne
-000012c0: 7574 7261 6c20 7468 656f 7279 2073 686f  utral theory sho
-000012d0: 756c 6420 6f63 6375 7220 696e 2063 6f6e  uld occur in con
-000012e0: 7374 616e 7420 7261 7465 2e20 4769 7665  stant rate. Give
-000012f0: 6e20 6120 6d6f 6465 6c20 6f66 2067 656e  n a model of gen
-00001300: 6520 6661 6d69 6c79 2074 6861 7420 616c  e family that al
-00001310: 6c6f 7773 2074 6865 2067 656e 6520 746f  lows the gene to
-00001320: 2064 7570 6c69 6361 7465 2061 6e64 2067   duplicate and g
-00001330: 6574 206c 6f73 7420 696e 2061 2066 6978  et lost in a fix
-00001340: 6564 2072 6174 652c 206f 6e65 2063 616e  ed rate, one can
-00001350: 2064 6572 6976 6520 7468 6174 2074 6865   derive that the
-00001360: 2070 726f 6261 6269 6c69 7479 2064 656e   probability den
-00001370: 7369 7479 2066 756e 6374 696f 6e20 6f66  sity function of
-00001380: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-00001390: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-000013a0: 7574 696f 6e20 6f66 2072 6574 6169 6e65  ution of retaine
-000013b0: 6420 6765 6e65 2064 7570 6c69 6361 7465  d gene duplicate
-000013c0: 7320 6973 2061 2071 7561 7369 2d65 7870  s is a quasi-exp
-000013d0: 6f6e 656e 7469 616c 2066 756e 6374 696f  onential functio
-000013e0: 6e20 7468 6174 206d 6f73 7420 7265 7461  n that most reta
-000013f0: 696e 6564 2067 656e 6520 6475 706c 6963  ined gene duplic
-00001400: 6174 6573 2061 7265 2072 6563 656e 746c  ates are recentl
-00001410: 7920 626f 726e 6564 2077 6974 6820 7e30  y borned with ~0
-00001420: 2061 6765 2077 6869 6c65 2061 7320 7468   age while as th
-00001430: 6520 6167 6520 676f 696e 6720 6f6c 6465  e age going olde
-00001440: 7220 7468 6520 6173 736f 6369 6174 6564  r the associated
-00001450: 206e 756d 6265 7220 6f66 2072 6574 6169   number of retai
-00001460: 6e65 6420 6765 6e65 2064 7570 6c69 6361  ned gene duplica
-00001470: 7465 7320 6465 6361 7920 7175 6173 692d  tes decay quasi-
-00001480: 6578 706f 6e65 6e74 6961 6c6c 792e 2054  exponentially. T
-00001490: 6865 7265 666f 7265 2c20 7468 6520 6f63  herefore, the oc
-000014a0: 6375 7261 6e63 6520 6f66 206c 6172 6765  curance of large
-000014b0: 2d73 6361 6c65 2067 656e 6520 6475 706c  -scale gene dupl
-000014c0: 6963 6174 696f 6e20 6576 656e 7473 2c20  ication events, 
-000014d0: 666f 7220 696e 7374 616e 6520 5747 4473  for instane WGDs
-000014e0: 2c20 7769 7468 2076 6172 6965 6420 7265  , with varied re
-000014f0: 7465 6e74 696f 6e20 7261 7465 2c20 7769  tention rate, wi
-00001500: 6c6c 206c 6561 7665 2061 6e20 6167 6520  ll leave an age 
-00001510: 7065 616b 2066 726f 6d20 7468 6520 6275  peak from the bu
-00001520: 7273 7420 6f66 2067 656e 6520 6475 706c  rst of gene dupl
-00001530: 6963 6174 6573 2069 6e20 6120 7368 6f72  icates in a shor
-00001540: 7420 7469 6d65 2d66 7261 6d65 2075 706f  t time-frame upo
-00001550: 6e20 7468 6520 696e 6974 6961 6c20 6167  n the initial ag
-00001560: 6520 6469 7374 7269 6275 7469 6f6e 2c20  e distribution, 
-00001570: 616e 6420 6361 6e20 6265 2075 6e76 6569  and can be unvei
-00001580: 6c65 6420 6672 6f6d 206d 6978 7475 7265  led from mixture
-00001590: 206d 6f64 656c 696e 6720 616e 616c 7973   modeling analys
-000015a0: 6973 2e20 486f 7765 7665 722c 2057 4744  is. However, WGD
-000015b0: 7320 6964 656e 7469 6669 6564 2066 726f  s identified fro
-000015c0: 6d20 7468 6520 7061 7261 6c6f 676f 7573  m the paralogous
-000015d0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000015e0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-000015f0: 6e73 2063 616e 206f 6e6c 7920 696e 666f  ns can only info
-00001600: 726d 2074 6865 2057 4744 2074 696d 696e  rm the WGD timin
-00001610: 6720 696e 2074 6865 2074 696d 652d 7363  g in the time-sc
-00001620: 616c 6520 6f66 2074 6861 7420 7370 6563  ale of that spec
-00001630: 6966 6963 2073 7065 6369 6573 2c20 7768  ific species, wh
-00001640: 6963 6820 6973 206e 6f74 2063 6f6d 7061  ich is not compa
-00001650: 7261 626c 6520 696e 2074 6865 2070 6879  rable in the phy
-00001660: 6c6f 6765 6e65 7469 6320 636f 6e74 6578  logenetic contex
-00001670: 742e 204f 6e6c 7920 7769 7468 2074 6865  t. Only with the
-00001680: 206f 7274 686f 6c6f 676f 7573 202a 4b2a   orthologous *K*
-00001690: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
-000016a0: 2064 6973 7472 6962 7574 696f 6e73 2c20   distributions, 
-000016b0: 7768 6963 6820 636f 6e76 6572 7420 7468  which convert th
-000016c0: 6520 6573 7469 6d61 7465 6420 626f 6479  e estimated body
-000016d0: 2066 726f 6d20 7061 7261 6c6f 6775 6573   from paralogues
-000016e0: 2074 6f20 6f72 7468 6f6c 6f67 7565 7320   to orthologues 
-000016f0: 616e 6420 696e 666f 726d 2074 6865 2072  and inform the r
-00001700: 656c 6174 6976 6520 7469 6d69 6e67 206f  elative timing o
-00001710: 6620 7370 6563 6961 7469 6f6e 2065 7665  f speciation eve
-00001720: 6e74 732c 2063 616e 2077 6520 6465 6369  nts, can we deci
-00001730: 7068 6572 2074 6865 2070 6879 6c6f 6765  pher the phyloge
-00001740: 6e65 7469 6320 706c 6163 656d 656e 7420  netic placement 
-00001750: 6f66 2057 4744 7320 6166 7465 7220 7072  of WGDs after pr
-00001760: 6f70 6572 2072 6174 6520 636f 7272 6563  oper rate correc
-00001770: 7469 6f6e 2e20 6077 6764 2076 3260 2069  tion. `wgd v2` i
-00001780: 7320 7375 6368 2070 726f 6772 616d 2074  s such program t
-00001790: 6861 7420 6865 6c70 7320 7573 6572 7320  hat helps users 
-000017a0: 636f 6e73 7472 7563 7420 7061 7261 6c6f  construct paralo
-000017b0: 676f 7573 2061 6e64 206f 7274 686f 6c6f  gous and ortholo
-000017c0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
-000017d0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-000017e0: 7574 696f 6e73 2061 6e64 2072 6561 6c69  utions and reali
-000017f0: 7a65 2062 6f74 6820 7468 6520 6964 656e  ze both the iden
-00001800: 7469 6669 6361 7469 6f6e 2061 6e64 2070  tification and p
-00001810: 6c61 6365 6d65 6e74 206f 6620 5747 4473  lacement of WGDs
-00001820: 2e0a 0a49 6e20 7468 6520 7072 656d 6973  ...In the premis
-00001830: 6520 6f66 2070 6879 6c6f 6765 6e65 7469  e of phylogeneti
-00001840: 6361 6c6c 7920 6c6f 6361 7465 6420 5747  cally located WG
-00001850: 4473 2c20 7468 6520 6162 736f 6c75 7465  Ds, the absolute
-00001860: 2061 6765 2028 696e 2067 656f 6c6f 6769   age (in geologi
-00001870: 6361 6c20 7469 6d65 2920 6f66 2057 4744  cal time) of WGD
-00001880: 7320 6361 6e20 616c 736f 2062 6520 696e  s can also be in
-00001890: 6665 7272 6564 2066 726f 6d20 7468 6f73  ferred from thos
-000018a0: 6520 5747 442d 7265 7461 696e 6564 2067  e WGD-retained g
-000018b0: 656e 6520 6475 706c 6963 6174 6573 2c20  ene duplicates, 
-000018c0: 616c 7468 6f75 6768 2074 6865 7265 2068  although there h
-000018d0: 6173 2062 6565 6e20 6e6f 2065 6173 7920  as been no easy 
-000018e0: 6f72 2073 7472 6169 6768 7466 6f72 7761  or straightforwa
-000018f0: 7264 2070 6970 656c 696e 6520 666f 7220  rd pipeline for 
-00001900: 7468 6973 206a 6f62 2073 6f20 6661 722e  this job so far.
-00001910: 2049 6e20 6077 6764 2076 3260 2c20 7765   In `wgd v2`, we
-00001920: 2064 6576 656c 6f70 6564 2061 2066 6561   developed a fea
-00001930: 7369 626c 6520 696e 7465 6772 6174 6564  sible integrated
-00001940: 2070 6970 6c65 6c69 6e65 2066 6f72 2061   pipleline for a
-00001950: 6273 6f6c 7574 6520 6461 7469 6e67 206f  bsolute dating o
-00001960: 6620 5747 4473 2e20 5468 6520 7069 7065  f WGDs. The pipe
-00001970: 6c69 6e65 2063 616e 2062 6520 726f 7567  line can be roug
-00001980: 686c 7920 6469 7669 6465 6420 696e 746f  hly divided into
-00001990: 2074 6872 6565 206d 6169 6e20 7374 6570   three main step
-000019a0: 732e 2031 2920 5468 6520 636f 6e73 7472  s. 1) The constr
-000019b0: 7563 7469 6f6e 206f 6620 616e 6368 6f72  uction of anchor
-000019c0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000019d0: 2064 6973 7472 6962 7574 696f 6e20 616e   distribution an
-000019e0: 6420 7468 6520 6465 6c69 6e65 6174 696f  d the delineatio
-000019f0: 6e20 6f66 2063 7265 6469 6162 6c65 202a  n of crediable *
-00001a00: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2072  K*<sub>S</sub> r
-00001a10: 616e 6765 2061 646f 7074 6564 2066 6f72  ange adopted for
-00001a20: 2070 6879 6c6f 6765 6e65 7469 6320 6461   phylogenetic da
-00001a30: 7469 6e67 2c20 7573 696e 6720 6077 6764  ting, using `wgd
-00001a40: 2064 6d64 602c 2060 7767 6420 6b73 6460   dmd`, `wgd ksd`
-00001a50: 2c20 6077 6764 2073 796e 6020 616e 6420  , `wgd syn` and 
-00001a60: 6077 6764 2070 6561 6b60 2e20 4e6f 7465  `wgd peak`. Note
-00001a70: 2074 6861 7420 6865 7265 2077 6520 6f6e   that here we on
-00001a80: 6c79 2063 6f6e 7369 6465 7220 6765 6e6f  ly consider geno
-00001a90: 6d65 2061 7373 656d 626c 7920 6265 6361  me assembly beca
-00001aa0: 7573 6520 666f 7220 7472 616e 7363 7269  use for transcri
-00001ab0: 7074 6f6d 6520 6173 7365 6d62 6c79 2069  ptome assembly i
-00001ac0: 7427 7320 696d 706f 7373 6962 6c65 2074  t's impossible t
-00001ad0: 6f20 6469 7374 696e 6775 6973 6820 5747  o distinguish WG
-00001ae0: 442d 6465 7269 7665 6420 6475 706c 6963  D-derived duplic
-00001af0: 6174 6573 2066 726f 6d20 736d 616c 6c2d  ates from small-
-00001b00: 7363 616c 6520 6475 706c 6963 6174 696f  scale duplicatio
-00001b10: 6e2d 6465 7269 7665 6420 6475 706c 6963  n-derived duplic
-00001b20: 6174 6573 2c20 7768 6963 6820 6861 7070  ates, which happ
-00001b30: 656e 6564 2069 6e20 6120 636f 6e74 696e  ened in a contin
-00001b40: 756f 7573 2074 696d 652d 6672 616d 6520  uous time-frame 
-00001b50: 696e 7374 6561 6420 6f66 206f 6e6c 7920  instead of only 
-00001b60: 6120 7365 7061 7261 7465 2073 686f 7274  a separate short
-00001b70: 2074 696d 652d 6672 616d 6520 616e 6420   time-frame and 
-00001b80: 7468 7573 2072 6566 6c65 6374 7320 7468  thus reflects th
-00001b90: 6520 6475 7261 7469 6f6e 206f 6620 7468  e duration of th
-00001ba0: 6174 2062 7261 6e63 6820 7261 7468 6572  at branch rather
-00001bb0: 2074 6861 6e20 7468 6520 7469 6d65 2061   than the time a
-00001bc0: 7420 7768 6963 6820 5747 4420 6f63 6375  t which WGD occu
-00001bd0: 7272 6564 2e20 3229 2054 6865 2066 6f72  rred. 2) The for
-00001be0: 6d75 6c61 7469 6f6e 206f 6620 6120 7374  mulation of a st
-00001bf0: 6172 7469 6e67 2074 7265 6520 7573 6564  arting tree used
-00001c00: 2069 6e20 7468 6520 7068 796c 6f67 656e   in the phylogen
-00001c10: 6574 6963 2064 6174 696e 672c 2063 6f6d  etic dating, com
-00001c20: 706f 7365 6420 6f66 2061 2066 6577 2073  posed of a few s
-00001c30: 7065 6369 6573 2061 6e64 2061 6e6e 6f74  pecies and annot
-00001c40: 6174 6564 2077 6974 6820 666f 7373 696c  ated with fossil
-00001c50: 2063 616c 6962 7261 7469 6f6e 2069 6e66   calibration inf
-00001c60: 6f72 6d61 7469 6f6e 2e20 5468 6973 2073  ormation. This s
-00001c70: 7465 7020 6973 2065 7373 656e 7469 616c  tep is essential
-00001c80: 2066 6f72 2074 6865 2072 6573 756c 7420   for the result 
-00001c90: 6f66 2061 6273 6f6c 7574 6520 5747 4420  of absolute WGD 
-00001ca0: 6461 7469 6e67 2074 6861 7420 7765 2073  dating that we s
-00001cb0: 7567 6765 7374 2075 7365 7273 206f 6620  uggest users of 
-00001cc0: 7461 6b69 6e67 2067 7265 6174 2063 6175  taking great cau
-00001cd0: 7469 6f6e 2074 6f20 6173 7375 7265 2074  tion to assure t
-00001ce0: 6865 2074 7265 6520 746f 706f 6c6f 6779  he tree topology
-00001cf0: 2061 6e64 2070 726f 7065 7220 626f 756e   and proper boun
-00001d00: 6473 2066 6f72 2066 6f73 7369 6c20 6361  ds for fossil ca
-00001d10: 6c69 6272 6174 696f 6e73 2e20 3329 2054  librations. 3) T
-00001d20: 6865 2063 6f6e 7374 7275 6374 696f 6e20  he construction 
-00001d30: 6f66 206f 7274 686f 6772 6f75 7073 2063  of orthogroups c
-00001d40: 6f6e 7369 7374 696e 6720 6f66 2063 6f6c  onsisting of col
-00001d50: 6c69 6e65 6172 2064 7570 6c69 6361 7465  linear duplicate
-00001d60: 7320 6f66 2074 6865 2066 6f63 616c 2073  s of the focal s
-00001d70: 7065 6369 6573 2061 6e64 2074 6865 6972  pecies and their
-00001d80: 2072 6563 6970 726f 6361 6c20 6265 7374   reciprocal best
-00001d90: 2068 6974 7320 2852 4248 7329 2061 6761   hits (RBHs) aga
-00001da0: 696e 7374 206f 7468 6572 2073 7065 6369  inst other speci
-00001db0: 6573 2069 6e20 7468 6520 7374 6172 7469  es in the starti
-00001dc0: 6e67 2074 7265 6520 616e 6420 7468 6520  ng tree and the 
-00001dd0: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
-00001de0: 696e 6720 7573 696e 6720 6120 6d6f 6c65  ing using a mole
-00001df0: 6375 6c61 7220 6461 7469 6e67 2070 726f  cular dating pro
-00001e00: 6772 616d 2066 6f72 2069 6e73 7461 6e63  gram for instanc
-00001e10: 6520 606d 636d 6374 7265 6560 2c20 7669  e `mcmctree`, vi
-00001e20: 6120 7072 6f67 7261 6d20 6077 6764 2064  a program `wgd d
-00001e30: 6d64 6020 616e 6420 6077 6764 2066 6f63  md` and `wgd foc
-00001e40: 7573 602e 2057 6520 7265 636f 6d6d 656e  us`. We recommen
-00001e50: 6420 7468 6520 7573 6167 6520 6f66 2042  d the usage of B
-00001e60: 6179 6573 6961 6e20 6d6f 6c65 6375 6c61  ayesian molecula
-00001e70: 7220 6461 7469 6e67 2070 726f 6772 616d  r dating program
-00001e80: 2060 6d63 6d63 7472 6565 602c 2077 6869   `mcmctree`, whi
-00001e90: 6368 2070 726f 7669 6465 7320 6120 7661  ch provides a va
-00001ea0: 7269 6574 7920 6f66 2064 6966 6665 7265  riety of differe
-00001eb0: 6e74 2073 7562 7374 6974 7574 696f 6e20  nt substitution 
-00001ec0: 616e 6420 7261 7465 206d 6f64 656c 732e  and rate models.
-00001ed0: 204e 6f6e 6574 6865 6c65 7373 2c20 7765   Nonetheless, we
-00001ee0: 2075 7267 6520 7573 6572 7320 746f 2073   urge users to s
-00001ef0: 6574 2070 7269 6f72 2064 6973 7472 6962  et prior distrib
-00001f00: 7574 696f 6e20 6f66 2064 6966 6665 7265  ution of differe
-00001f10: 6e74 2070 6172 616d 6574 6572 7320 7769  nt parameters wi
-00001f20: 7468 2063 6175 7469 6f6e 2061 6e64 2061  th caution and a
-00001f30: 7373 7572 6520 6164 6571 7561 7465 2073  ssure adequate s
-00001f40: 616d 706c 696e 6720 6f66 2064 6966 6665  ampling of diffe
-00001f50: 7265 6e74 2070 6172 616d 6574 6572 732e  rent parameters.
-00001f60: 0a0a 2323 2049 6e73 7461 6c6c 6174 696f  ..## Installatio
-00001f70: 6e0a 0a54 6865 2065 6173 6965 7374 2077  n..The easiest w
-00001f80: 6179 2074 6f20 696e 7374 616c 6c20 6077  ay to install `w
-00001f90: 6764 2076 3260 2069 7320 7573 696e 6720  gd v2` is using 
-00001fa0: 6050 5950 4960 2e20 4e6f 7465 2074 6861  `PYPI`. Note tha
-00001fb0: 7420 6966 2079 6f75 2077 616e 7420 746f  t if you want to
-00001fc0: 2067 6574 2074 6865 206c 6174 6573 7420   get the latest 
-00001fd0: 7570 6461 7465 2c20 7765 2073 7567 6765  update, we sugge
-00001fe0: 7374 2069 6e73 7461 6c6c 696e 6720 6672  st installing fr
-00001ff0: 6f6d 2074 6865 2073 6f75 7263 652c 2073  om the source, s
-00002000: 696e 6365 2074 6865 2075 7064 6174 6520  ince the update 
-00002010: 6f6e 2060 5059 5049 6020 7769 6c6c 2062  on `PYPI` will b
-00002020: 6520 6465 6c61 7965 6420 636f 6d70 6172  e delayed compar
-00002030: 6564 2074 6f20 6865 7265 206f 6620 736f  ed to here of so
-00002040: 7572 6365 2e0a 0a60 6060 0a70 6970 2069  urce...```.pip i
-00002050: 6e73 7461 6c6c 2077 6764 0a60 6060 0a0a  nstall wgd.```..
-00002060: 546f 2069 6e73 7461 6c6c 2060 7767 6420  To install `wgd 
-00002070: 7632 6020 696e 2061 2076 6972 7475 616c  v2` in a virtual
-00002080: 2065 6e76 6972 6f6e 6d65 6e74 2028 7768   environment (wh
-00002090: 6963 6820 7765 2073 7472 6f6e 676c 7920  ich we strongly 
-000020a0: 7265 636f 6d6d 656e 642c 206e 6f20 6d61  recommend, no ma
-000020b0: 7474 6572 2079 6f75 2069 6e73 7461 6c6c  tter you install
-000020c0: 2066 726f 6d20 736f 7572 6365 2c20 6050   from source, `P
-000020d0: 5950 4960 206f 7220 6062 696f 636f 6e64  YPI` or `biocond
-000020e0: 6160 292c 2074 6865 2066 6f6c 6c6f 7769  a`), the followi
-000020f0: 6e67 2063 6f6d 6d61 6e64 206c 696e 6573  ng command lines
-00002100: 2063 6f75 6c64 2062 6520 7573 6564 2e0a   could be used..
-00002110: 0a60 6060 0a67 6974 2063 6c6f 6e65 2068  .```.git clone h
-00002120: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
-00002130: 6d2f 6865 6368 652d 7073 622f 7767 640a  m/heche-psb/wgd.
-00002140: 6364 2077 6764 0a76 6972 7475 616c 656e  cd wgd.virtualen
-00002150: 7620 2d70 3d70 7974 686f 6e33 2045 4e56  v -p=python3 ENV
-00002160: 2028 6f72 2070 7974 686f 6e33 202d 6d20   (or python3 -m 
-00002170: 7665 6e76 2045 4e56 290a 736f 7572 6365  venv ENV).source
-00002180: 2045 4e56 2f62 696e 2f61 6374 6976 6174   ENV/bin/activat
-00002190: 650a 7069 7020 696e 7374 616c 6c20 2d72  e.pip install -r
-000021a0: 2072 6571 7569 7265 6d65 6e74 732e 7478   requirements.tx
-000021b0: 740a 7069 7020 696e 7374 616c 6c20 2e0a  t.pip install ..
-000021c0: 6060 600a 0a57 6865 6e20 6d65 7420 7769  ```..When met wi
-000021d0: 7468 2070 6572 6d69 7373 696f 6e20 7072  th permission pr
-000021e0: 6f62 6c65 6d20 696e 2069 6e73 7461 6c6c  oblem in install
-000021f0: 6174 696f 6e2c 2070 6c65 6173 6520 7472  ation, please tr
-00002200: 7920 7468 6520 666f 6c6c 6f77 696e 6720  y the following 
-00002210: 636f 6d6d 616e 6420 6c69 6e65 2e0a 0a60  command line...`
-00002220: 6060 0a70 6970 2069 6e73 7461 6c6c 202d  ``.pip install -
-00002230: 6520 2e0a 6060 600a 0a49 6620 6d75 6c74  e ..```..If mult
-00002240: 6970 6c79 2076 6572 7369 6f6e 7320 6f66  iply versions of
-00002250: 2060 7767 6460 2077 6572 6520 696e 7374   `wgd` were inst
-00002260: 616c 6c65 6420 696e 2074 6865 2073 7973  alled in the sys
-00002270: 7465 6d2c 2070 6c65 6173 6520 6164 6420  tem, please add 
-00002280: 7468 6520 7269 6768 7420 7061 7468 206f  the right path o
-00002290: 6620 696e 7465 7265 7374 6564 2076 6572  f interested ver
-000022a0: 7369 6f6e 2069 6e74 6f20 7468 6520 656e  sion into the en
-000022b0: 7669 726f 6e6d 656e 7420 7661 7269 6162  vironment variab
-000022c0: 6c65 732c 2066 6f72 2065 7861 6d70 6c65  les, for example
-000022d0: 0a0a 6060 600a 6578 706f 7274 2050 4154  ..```.export PAT
-000022e0: 483d 2224 5041 5448 3a7e 2f2e 6c6f 6361  H="$PATH:~/.loca
-000022f0: 6c2f 6269 6e2f 7767 6422 0a60 6060 0a0a  l/bin/wgd".```..
-00002300: 4e6f 7465 2074 6861 7420 7468 6520 7665  Note that the ve
-00002310: 7273 696f 6e20 6f66 2060 6e75 6d70 7960  rsion of `numpy`
-00002320: 2069 7320 696d 706f 7274 616e 7420 2866   is important (f
-00002330: 6f72 206d 616e 7920 6f74 6865 7220 7061  or many other pa
-00002340: 636b 6167 6573 2061 7265 2074 6865 2073  ckages are the s
-00002350: 616d 6520 6f66 2063 6f75 7273 6529 2c20  ame of course), 
-00002360: 6573 7065 6369 616c 6c79 2066 6f72 2060  especially for `
-00002370: 6661 7374 636c 7573 7465 7260 2070 6163  fastcluster` pac
-00002380: 6b61 6765 2e20 496e 206f 7572 2074 6573  kage. In our tes
-00002390: 742c 2074 6865 2060 6e75 6d70 7960 2031  t, the `numpy` 1
-000023a0: 2e31 392e 3020 776f 726b 7320 6669 6e65  .19.0 works fine
-000023b0: 206f 6e20 6070 7974 686f 6e33 2e36 2f38   on `python3.6/8
-000023c0: 602e 2049 6620 796f 7520 6d65 7420 736f  `. If you met so
-000023d0: 6d65 2065 7272 6f72 7320 6f72 2077 6172  me errors or war
-000023e0: 6e69 6e67 7320 6162 6f75 7420 606e 756d  nings about `num
-000023f0: 7079 602c 206d 6179 6265 2063 6f6e 7369  py`, maybe consi
-00002400: 6465 7269 6e67 2070 7265 2d69 6e73 7461  dering pre-insta
-00002410: 6c6c 2060 6e75 6d70 7960 2061 7320 312e  ll `numpy` as 1.
-00002420: 3139 2e30 206f 7220 6f74 6865 7220 636c  19.0 or other cl
-00002430: 6f73 652d 6279 2076 6572 7369 6f6e 7320  ose-by versions 
-00002440: 6265 666f 7265 2079 6f75 2069 6e73 7461  before you insta
-00002450: 6c6c 2060 7767 6460 2e20 6077 6764 6020  ll `wgd`. `wgd` 
-00002460: 7265 6c69 6573 206f 6e20 6578 7465 726e  relies on extern
-00002470: 616c 2073 6f66 7477 6172 6573 2069 6e63  al softwares inc
-00002480: 6c75 6469 6e67 2060 6469 616d 6f6e 6460  luding `diamond`
-00002490: 2061 6e64 2060 6d63 6c60 2066 6f72 2060   and `mcl` for `
-000024a0: 7767 6420 646d 6460 2c20 6070 616d 6c20  wgd dmd`, `paml 
-000024b0: 7634 2e39 6a60 2c20 606d 6166 6674 6020  v4.9j`, `mafft` 
-000024c0: 2860 6d75 7363 6c65 6020 616e 6420 6070  (`muscle` and `p
-000024d0: 7261 6e6b 6020 6966 2073 6574 292c 2060  rank` if set), `
-000024e0: 6661 7374 7472 6565 6020 286f 7220 6069  fasttree` (or `i
-000024f0: 7174 7265 6560 2069 6620 7365 7429 2066  qtree` if set) f
-00002500: 6f72 2060 7767 6420 6b73 6460 2061 6e64  or `wgd ksd` and
-00002510: 206f 7074 696f 6e61 6c6c 7920 606d 7262   optionally `mrb
-00002520: 6179 6573 6020 666f 7220 7468 6520 7068  ayes` for the ph
-00002530: 796c 6f67 656e 6574 6963 2069 6e66 6572  ylogenetic infer
-00002540: 656e 6365 2066 756e 6374 696f 6e20 696e  ence function in
-00002550: 2060 7767 6420 646d 6460 2061 6e64 2060   `wgd dmd` and `
-00002560: 7767 6420 666f 6375 7360 2028 606d 6166  wgd focus` (`maf
-00002570: 6674 602c 2060 6d75 7363 6c65 6020 616e  ft`, `muscle` an
-00002580: 6420 6070 7261 6e6b 6020 6173 2077 656c  d `prank` as wel
-00002590: 6c20 7768 656e 2074 6865 2061 6e61 6c79  l when the analy
-000025a0: 7369 7320 7265 7175 6972 6573 2073 6571  sis requires seq
-000025b0: 7565 6e63 6520 616c 6967 6e6d 656e 7429  uence alignment)
-000025c0: 2e20 536f 6d65 206f 7468 6572 206f 7074  . Some other opt
-000025d0: 696f 6e61 6c20 736f 6674 7761 7265 7320  ional softwares 
-000025e0: 696e 636c 7564 696e 6720 6070 616d 6c20  including `paml 
-000025f0: 7634 2e39 6a60 2c20 6072 3873 602c 2060  v4.9j`, `r8s`, `
-00002600: 6265 6173 7460 2c20 6065 6767 6e6f 6760  beast`, `eggnog`
-00002610: 2c20 6064 6961 6d6f 6e64 602c 2060 696e  , `diamond`, `in
-00002620: 7465 7270 726f 7363 616e 602c 2060 686d  terproscan`, `hm
-00002630: 6d65 7220 7633 2e31 6232 6020 616e 6420  mer v3.1b2` and 
-00002640: 6061 7374 7261 6c2d 7072 6f60 2028 6068  `astral-pro` (`h
-00002650: 6d6d 6572 2076 332e 3162 3260 2069 7320  mmer v3.1b2` is 
-00002660: 616c 736f 2066 6f72 2074 6865 205b 6f72  also for the [or
-00002670: 7468 6f67 726f 7570 2061 7373 6967 6e6d  thogroup assignm
-00002680: 656e 745d 2868 7474 7073 3a2f 2f77 6764  ent](https://wgd
-00002690: 7632 2e72 6561 6474 6865 646f 6373 2e69  v2.readthedocs.i
-000026a0: 6f2f 656e 2f6c 6174 6573 742f 7573 6167  o/en/latest/usag
-000026b0: 652e 6874 6d6c 236f 7274 686f 6772 6f75  e.html#orthogrou
-000026c0: 702d 6173 7369 676e 6d65 6e74 2920 6675  p-assignment) fu
-000026d0: 6e63 7469 6f6e 2069 6e20 6077 6764 2064  nction in `wgd d
-000026e0: 6d64 6020 616e 6420 6061 7374 7261 6c2d  md` and `astral-
-000026f0: 7072 6f60 2069 7320 616c 736f 2066 6f72  pro` is also for
-00002700: 2074 6865 2066 756e 6374 696f 6e20 602d   the function `-
-00002710: 2d63 6f6c 6c69 6e65 6172 636f 616c 6573  -collinearcoales
-00002720: 6365 6e63 6560 2069 6e20 6077 6764 2064  cence` in `wgd d
-00002730: 6d64 6029 2061 7265 2066 6f72 2074 6865  md`) are for the
-00002740: 206d 6f6c 6563 756c 6172 2064 6174 696e   molecular datin
-00002750: 672c 2067 656e 6520 6661 6d69 6c79 2066  g, gene family f
-00002760: 756e 6374 696f 6e20 616e 6e6f 7461 7469  unction annotati
-00002770: 6f6e 206f 7220 7068 796c 6f67 656e 6574  on or phylogenet
-00002780: 6963 2069 6e66 6572 656e 6365 2069 6e20  ic inference in 
-00002790: 6077 6764 2066 6f63 7573 602e 0a0a 2323  `wgd focus`...##
-000027a0: 2050 6172 616d 6574 6572 730a 0a54 6865   Parameters..The
-000027b0: 7265 2061 7265 2037 206d 6169 6e20 7072  re are 7 main pr
-000027c0: 6f67 7261 6d73 2069 6e20 6077 6764 2076  ograms in `wgd v
-000027d0: 3260 3a20 6064 6d64 602c 6066 6f63 7573  2`: `dmd`,`focus
-000027e0: 602c 606b 7364 602c 606d 6978 602c 6070  `,`ksd`,`mix`,`p
-000027f0: 6561 6b60 2c60 7379 6e60 2c60 7669 7a60  eak`,`syn`,`viz`
-00002800: 2e20 4865 7265 6166 7465 7220 7765 2077  . Hereafter we w
-00002810: 696c 6c20 7072 6f76 6964 6520 6120 6465  ill provide a de
-00002820: 7461 696c 6564 2065 6c75 6369 6461 7469  tailed elucidati
-00002830: 6f6e 206f 6e20 6561 6368 206f 6620 7468  on on each of th
-00002840: 6520 7072 6f67 7261 6d20 616e 6420 6974  e program and it
-00002850: 7320 6173 736f 6369 6174 6564 2070 6172  s associated par
-00002860: 616d 6574 6572 732e 2050 6c65 6173 6520  ameters. Please 
-00002870: 7265 6665 7220 746f 2074 6865 205b 5573  refer to the [Us
-00002880: 6167 655d 2823 7573 6167 6529 2066 6f72  age](#usage) for
-00002890: 2074 6865 2073 6365 6e61 7269 6f73 2074   the scenarios t
-000028a0: 6f20 7768 6963 6820 6561 6368 2070 6172  o which each par
-000028b0: 616d 6574 6572 2061 7070 6c69 6573 2e0a  ameter applies..
-000028c0: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
-000028d0: 6420 646d 6460 2063 616e 2072 6561 6c69  d dmd` can reali
-000028e0: 7a65 2074 6865 2064 656c 696e 6561 7469  ze the delineati
-000028f0: 6f6e 206f 6620 7768 6f6c 6520 7061 7261  on of whole para
-00002900: 6e6f 6d65 2c20 5242 4873 2028 5265 6369  nome, RBHs (Reci
-00002910: 7072 6f63 616c 2042 6573 7420 4869 7473  procal Best Hits
-00002920: 292c 204d 5242 4873 2028 4d75 6c74 6970  ), MRBHs (Multip
-00002930: 6c65 2052 6563 6970 726f 6361 6c20 4265  le Reciprocal Be
-00002940: 7374 2048 6974 7329 2c20 6f72 7468 6f67  st Hits), orthog
-00002950: 726f 7570 7320 616e 6420 736f 6d65 206f  roups and some o
-00002960: 7468 6572 206f 7274 686f 6772 6f75 702d  ther orthogroup-
-00002970: 7265 6c61 7465 6420 6675 6e63 7469 6f6e  related function
-00002980: 732c 2069 6e63 6c75 6469 6e67 2063 6972  s, including cir
-00002990: 6375 6d73 6372 6970 7469 6f6e 206f 6620  cumscription of 
-000029a0: 6e65 7374 6564 2073 696e 676c 652d 636f  nested single-co
-000029b0: 7079 206f 7274 686f 6772 6f75 7073 2028  py orthogroups (
-000029c0: 4e53 4f47 7329 2c20 756e 6269 6173 6564  NSOGs), unbiased
-000029d0: 2074 6573 7420 6f66 2073 696e 676c 652d   test of single-
-000029e0: 636f 7079 206f 7274 686f 6772 6f75 7073  copy orthogroups
-000029f0: 2028 534f 4773 2920 6f76 6572 206d 6973   (SOGs) over mis
-00002a00: 7369 6e67 2069 6e70 6172 616c 6f67 732c  sing inparalogs,
-00002a10: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
-00002a20: 2042 5553 434f 2d67 7569 6465 6420 7369   BUSCO-guided si
-00002a30: 6e67 6c65 2d63 6f70 7920 6f72 7468 6f67  ngle-copy orthog
-00002a40: 726f 7570 7320 2853 4f47 7329 2c61 6e64  roups (SOGs),and
-00002a50: 2074 6865 2063 6f6c 6c69 6e65 6172 2063   the collinear c
-00002a60: 6f61 6c65 7363 656e 6365 2069 6e66 6572  oalescence infer
-00002a70: 656e 6365 206f 6620 7068 796c 6f67 656e  ence of phylogen
-00002a80: 792e 0a60 6060 0a77 6764 2064 6d64 2073  y..```.wgd dmd s
-00002a90: 6571 7565 6e63 6573 2028 6f70 7469 6f6e  equences (option
-00002aa0: 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ).--------------
-00002ab0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002ac0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002ad0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002ae0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002af0: 2d2d 0a2d 6f2c 202d 2d6f 7574 6469 722c  --.-o, --outdir,
-00002b00: 2074 6865 206f 7574 7075 7420 6469 7265   the output dire
-00002b10: 6374 6f72 792c 2064 6566 6175 6c74 2077  ctory, default w
-00002b20: 6764 5f64 6d64 0a2d 742c 202d 2d74 6d70  gd_dmd.-t, --tmp
-00002b30: 6469 722c 2074 6865 2074 656d 706f 7261  dir, the tempora
-00002b40: 7279 2077 6f72 6b69 6e67 2064 6972 6563  ry working direc
-00002b50: 746f 7279 2c20 6465 6661 756c 7420 4e6f  tory, default No
-00002b60: 6e65 2c20 6966 204e 6f6e 6520 7761 7320  ne, if None was 
-00002b70: 6769 7665 6e2c 2074 6865 2074 6d70 6469  given, the tmpdi
-00002b80: 7220 776f 756c 6420 6265 2061 7373 6967  r would be assig
-00002b90: 6e65 6420 7261 6e64 6f6d 206e 616d 6573  ned random names
-00002ba0: 2069 6e20 6375 7272 656e 7420 6469 7265   in current dire
-00002bb0: 6374 6f72 7920 616e 6420 6175 746f 6d61  ctory and automa
-00002bc0: 7465 6c79 2072 656d 6f76 6564 2061 7420  tely removed at 
-00002bd0: 7468 6520 636f 6d70 6c65 7469 6f6e 206f  the completion o
-00002be0: 6620 7072 6f67 7261 6d2c 2065 6c73 6520  f program, else 
-00002bf0: 7468 6520 746d 7064 6972 2077 6f75 6c64  the tmpdir would
-00002c00: 2062 6520 6b65 7074 0a2d 702c 202d 2d70   be kept.-p, --p
-00002c10: 726f 742c 2066 6c61 6720 6f70 7469 6f6e  rot, flag option
-00002c20: 2c20 7768 6574 6865 7220 7573 696e 6720  , whether using 
-00002c30: 7072 6f74 6569 6e20 6f72 206e 7563 6c65  protein or nucle
-00002c40: 6f74 6964 6520 7365 7175 656e 6365 730a  otide sequences.
-00002c50: 2d63 2c20 2d2d 6373 636f 7265 2c20 7468  -c, --cscore, th
-00002c60: 6520 632d 7363 6f72 6520 746f 2072 6573  e c-score to res
-00002c70: 7472 6963 7420 7468 6520 686f 6d6f 6c6f  trict the homolo
-00002c80: 6720 7369 6d69 6c61 7269 7479 206f 6620  g similarity of 
-00002c90: 4d52 4248 732c 2064 6566 6175 6c74 204e  MRBHs, default N
-00002ca0: 6f6e 652c 2069 6620 4e6f 6e65 2077 6173  one, if None was
-00002cb0: 2067 6976 656e 2c20 7468 6520 632d 7363   given, the c-sc
-00002cc0: 6f72 6520 6675 6e63 696f 6e20 776f 756c  ore funcion woul
-00002cd0: 646e 2774 2062 6520 6163 7469 7661 7465  dn't be activate
-00002ce0: 642c 2065 6c73 6520 6578 7065 6374 696e  d, else expectin
-00002cf0: 6720 6120 6465 6369 6d61 6c20 7769 7468  g a decimal with
-00002d00: 696e 2074 6865 2072 616e 6765 206f 6620  in the range of 
-00002d10: 3020 616e 6420 310a 2d49 2c20 2d2d 696e  0 and 1.-I, --in
-00002d20: 666c 6174 696f 6e2c 2074 6865 2069 6e66  flation, the inf
-00002d30: 6c61 7469 6f6e 2066 6163 746f 7220 666f  lation factor fo
-00002d40: 7220 4d43 4c20 7072 6f67 7261 6d2c 2064  r MCL program, d
-00002d50: 6566 6175 6c74 2032 2e30 2c20 7769 7468  efault 2.0, with
-00002d60: 2068 6967 6865 7220 7661 6c75 6520 6c65   higher value le
-00002d70: 6164 696e 6720 746f 206d 6f72 6520 6275  ading to more bu
-00002d80: 7420 736d 616c 6c65 7220 636c 7573 7465  t smaller cluste
-00002d90: 7273 0a2d 652c 202d 2d65 7661 6c2c 2074  rs.-e, --eval, t
-00002da0: 6865 2065 2d76 616c 7565 2063 7574 2d6f  he e-value cut-o
-00002db0: 6666 2066 6f72 2073 696d 696c 6172 6974  ff for similarit
-00002dc0: 7920 696e 2064 6961 6d6f 6e64 2061 6e64  y in diamond and
-00002dd0: 2f6f 7220 686d 6d65 722c 2064 6566 6175  /or hmmer, defau
-00002de0: 6c74 2031 652d 3130 0a2d 2d74 6f5f 7374  lt 1e-10.--to_st
-00002df0: 6f70 2c20 666c 6167 206f 7074 696f 6e2c  op, flag option,
-00002e00: 2077 6865 7468 6572 2074 6f20 7472 616e   whether to tran
-00002e10: 736c 6174 6520 7468 726f 7567 6820 5354  slate through ST
-00002e20: 4f50 2063 6f64 6f6e 732c 2069 6620 7468  OP codons, if th
-00002e30: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00002e40: 7472 616e 736c 6174 696f 6e20 7769 6c6c  translation will
-00002e50: 2062 6520 7465 726d 696e 6174 6564 2061   be terminated a
-00002e60: 7420 7468 6520 6669 7273 7420 696e 2d66  t the first in-f
-00002e70: 7261 6d65 2073 746f 7020 636f 646f 6e2c  rame stop codon,
-00002e80: 2065 6c73 6520 6120 6675 6c6c 2074 7261   else a full tra
-00002e90: 6e73 6c61 7469 6f6e 2063 6f6e 7469 6e75  nslation continu
-00002ea0: 696e 6720 6f6e 2070 6173 7369 6e67 2061  ing on passing a
-00002eb0: 6e79 2073 746f 7020 636f 646f 6e73 2077  ny stop codons w
-00002ec0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
-00002ed0: 640a 2d2d 6364 732c 2066 6c61 6720 6f70  d.--cds, flag op
-00002ee0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00002ef0: 206f 6e6c 7920 7472 616e 736c 6174 6520   only translate 
-00002f00: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
-00002f10: 2074 6861 7420 7374 6172 7473 2077 6974   that starts wit
-00002f20: 6820 6120 7661 6c69 6420 7374 6172 7420  h a valid start 
-00002f30: 636f 646f 6e20 616e 6420 6f6e 6c79 2063  codon and only c
-00002f40: 6f6e 7461 696e 7320 6120 7369 6e67 6c65  ontains a single
-00002f50: 2069 6e2d 6672 616d 6520 7374 6f70 2063   in-frame stop c
-00002f60: 6f64 6f6e 2061 7420 7468 6520 656e 6420  odon at the end 
-00002f70: 616e 6420 6d75 7374 2062 6520 6469 7669  and must be divi
-00002f80: 6461 626c 6520 6279 2074 6872 6565 2c20  dable by three, 
-00002f90: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-00002fa0: 7365 742c 206f 6e6c 7920 7468 6520 636f  set, only the co
-00002fb0: 6d70 6c65 7465 2043 4453 2077 6f75 6c64  mplete CDS would
-00002fc0: 2062 6520 7472 616e 736c 6174 6564 0a2d   be translated.-
-00002fd0: 662c 202d 2d66 6f63 7573 2c20 7468 6520  f, --focus, the 
-00002fe0: 7370 6563 6965 7320 746f 2062 6520 6d65  species to be me
-00002ff0: 7267 6564 206f 6e20 6c6f 6361 6c20 4d52  rged on local MR
-00003000: 4248 732c 2064 6566 6175 6c74 204e 6f6e  BHs, default Non
-00003010: 652c 2069 6620 4e6f 6e65 2077 6173 2067  e, if None was g
-00003020: 6976 656e 2c20 7468 6520 6c6f 6361 6c20  iven, the local 
-00003030: 4d52 4248 7320 776f 756c 646e 2774 2062  MRBHs wouldn't b
-00003040: 6520 696e 6665 7272 6564 0a2d 6170 2c20  e inferred.-ap, 
-00003050: 2d2d 616e 6368 6f72 706f 696e 7473 2c20  --anchorpoints, 
-00003060: 7468 6520 616e 6368 6f72 2070 6f69 6e74  the anchor point
-00003070: 7320 6461 7461 2066 696c 6520 6672 6f6d  s data file from
-00003080: 2069 2d61 6468 6f72 6520 666f 7220 636f   i-adhore for co
-00003090: 6e73 7472 7563 7469 6e67 2074 6865 206f  nstructing the o
-000030a0: 7274 686f 6772 6f75 7073 2077 6974 6820  rthogroups with 
-000030b0: 616e 6368 6f72 2070 6169 7273 2c20 6465  anchor pairs, de
-000030c0: 6661 756c 7420 4e6f 6e65 0a2d 736d 2c20  fault None.-sm, 
-000030d0: 2d2d 7365 676d 656e 7473 2c20 7468 6520  --segments, the 
-000030e0: 7365 676d 656e 7473 2064 6174 6166 696c  segments datafil
-000030f0: 6520 7573 6564 2069 6e20 636f 6c6c 696e  e used in collin
-00003100: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
-00003110: 616e 616c 7973 6973 2069 6620 696e 6974  analysis if init
-00003120: 6961 7465 642c 2064 6566 6175 6c74 204e  iated, default N
-00003130: 6f6e 650a 2d6c 652c 202d 2d6c 6973 7465  one.-le, --liste
-00003140: 6c65 6d65 6e74 732c 2074 6865 206c 6973  lements, the lis
-00003150: 7473 6567 6d65 6e74 7320 6461 7461 2066  tsegments data f
-00003160: 696c 6520 7573 6564 2069 6e20 636f 6c6c  ile used in coll
-00003170: 696e 6561 7220 636f 616c 6573 6365 6e63  inear coalescenc
-00003180: 6520 616e 616c 7973 6973 2069 6620 696e  e analysis if in
-00003190: 6974 6961 7465 642c 2064 6566 6175 6c74  itiated, default
-000031a0: 204e 6f6e 650a 2d67 742c 202d 2d67 656e   None.-gt, --gen
-000031b0: 6574 6162 6c65 2c20 7468 6520 6765 6e65  etable, the gene
-000031c0: 2074 6162 6c65 2064 6174 6166 696c 6520   table datafile 
-000031d0: 7573 6564 2069 6e20 636f 6c6c 696e 6561  used in collinea
-000031e0: 7220 636f 616c 6573 6365 6e63 6520 616e  r coalescence an
-000031f0: 616c 7973 6973 2069 6620 696e 6974 6961  alysis if initia
-00003200: 7465 642c 2064 6566 6175 6c74 204e 6f6e  ted, default Non
-00003210: 650a 2d63 6f63 2c20 2d2d 636f 6c6c 696e  e.-coc, --collin
-00003220: 6561 7263 6f61 6c65 7363 656e 6365 2c20  earcoalescence, 
-00003230: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-00003240: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
-00003250: 2074 6865 2063 6f6c 6c69 6e65 6172 2063   the collinear c
-00003260: 6f61 6c65 7363 656e 6365 2061 6e61 6c79  oalescence analy
-00003270: 7369 732c 2069 6620 7468 6520 666c 6167  sis, if the flag
-00003280: 2077 6173 2073 6574 2c20 7468 6520 616e   was set, the an
-00003290: 616c 7973 6973 2077 6f75 6c64 2062 6520  alysis would be 
-000032a0: 696e 6974 6961 7465 640a 2d6b 662c 202d  initiated.-kf, -
-000032b0: 2d6b 6565 7066 6173 7461 2c20 666c 6167  -keepfasta, flag
-000032c0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-000032d0: 2074 6f20 6f75 7470 7574 2074 6865 2073   to output the s
-000032e0: 6571 7565 6e63 6520 696e 666f 726d 6174  equence informat
-000032f0: 696f 6e20 6f66 204d 5242 4873 2c20 6966  ion of MRBHs, if
-00003300: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00003310: 742c 2074 6865 2073 6571 7565 6e63 6573  t, the sequences
-00003320: 206f 6620 4d52 4248 7320 776f 756c 6420   of MRBHs would 
-00003330: 6265 2069 6e20 6f75 7470 7574 0a2d 6b64  be in output.-kd
-00003340: 2c20 2d2d 6b65 6570 6475 706c 6963 6174  , --keepduplicat
-00003350: 6573 2c20 666c 6167 206f 7074 696f 6e2c  es, flag option,
-00003360: 2077 6865 7468 6572 2074 6f20 616c 6c6f   whether to allo
-00003370: 7720 7468 6520 7361 6d65 2067 656e 6520  w the same gene 
-00003380: 746f 206f 6363 7572 2069 6e20 6469 6666  to occur in diff
-00003390: 6572 656e 7420 4d52 4248 7320 286f 6e6c  erent MRBHs (onl
-000033a0: 7920 6d65 616e 696e 6766 756c 2077 6865  y meaningful whe
-000033b0: 6e20 7468 6520 6373 636f 7265 2077 6173  n the cscore was
-000033c0: 2075 7365 6429 2c20 6966 2074 6865 2066   used), if the f
-000033d0: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
-000033e0: 2073 616d 6520 6765 6e65 2063 6f75 6c64   same gene could
-000033f0: 2062 6520 6173 7369 676e 6564 2074 6f20   be assigned to 
-00003400: 6469 6666 6572 656e 7420 4d52 4248 730a  different MRBHs.
-00003410: 2d67 6d2c 202d 2d67 6c6f 6261 6c6d 7262  -gm, --globalmrb
-00003420: 682c 2066 6c61 6720 6f70 7469 6f6e 2c20  h, flag option, 
-00003430: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
-00003440: 6174 6520 676c 6f62 616c 204d 5242 4873  ate global MRBHs
-00003450: 2063 6f6e 7374 7275 6374 696f 6e2c 2069   construction, i
-00003460: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00003470: 6574 2c20 7468 6520 2d2d 666f 6375 7320  et, the --focus 
-00003480: 6f70 7469 6f6e 2077 6f75 6c64 2062 6520  option would be 
-00003490: 6967 6e6f 7265 6420 616e 6420 6f6e 6c79  ignored and only
-000034a0: 2067 6c6f 6261 6c20 4d52 4248 7320 776f   global MRBHs wo
-000034b0: 756c 6420 6265 2062 7569 6c74 0a2d 6e2c  uld be built.-n,
-000034c0: 202d 2d6e 7468 7265 6164 732c 2074 6865   --nthreads, the
-000034d0: 206e 756d 6265 7220 6f66 2074 6872 6561   number of threa
-000034e0: 6473 2074 6f20 7573 652c 2064 6566 6175  ds to use, defau
-000034f0: 6c74 2034 0a2d 6f69 2c20 2d2d 6f72 7468  lt 4.-oi, --orth
-00003500: 6f69 6e66 6572 2c20 666c 6167 206f 7074  oinfer, flag opt
-00003510: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-00003520: 696e 6974 6961 7465 206f 7274 686f 6772  initiate orthogr
-00003530: 6f75 7020 696e 6665 726e 6563 652c 2069  oup infernece, i
-00003540: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00003550: 6574 2c20 7468 6520 6f72 7468 6f67 726f  et, the orthogro
-00003560: 7570 2069 6e66 6572 6e65 6365 2070 726f  up infernece pro
-00003570: 6772 616d 2077 6f75 6c64 2062 6520 696e  gram would be in
-00003580: 6974 6961 7465 640a 2d6f 6f2c 202d 2d6f  itiated.-oo, --o
-00003590: 6e6c 796f 7274 686f 2c20 666c 6167 206f  nlyortho, flag o
-000035a0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000035b0: 6f20 6f6e 6c79 2063 6f6e 6475 6374 206f  o only conduct o
-000035c0: 7274 686f 6772 6f75 7020 696e 6665 726e  rthogroup infern
-000035d0: 6563 652c 2069 6620 7468 6520 666c 6167  ece, if the flag
-000035e0: 2077 6173 2073 6574 2c20 6f6e 6c79 2074   was set, only t
-000035f0: 6865 206f 7274 686f 6772 6f75 7020 696e  he orthogroup in
-00003600: 6665 726e 6563 6520 7069 7065 6c69 6e65  fernece pipeline
-00003610: 2077 6f75 6c64 2062 6520 7065 7266 6f72   would be perfor
-00003620: 6d65 6420 7768 696c 6520 7468 6520 6f74  med while the ot
-00003630: 6865 7220 616e 616c 7973 6973 2077 6f75  her analysis wou
-00003640: 6c64 6e27 7420 6265 2069 6e69 7469 6174  ldn't be initiat
-00003650: 6564 0a2d 676e 2c20 2d2d 6765 746e 736f  ed.-gn, --getnso
-00003660: 672c 2066 6c61 6720 6f70 7469 6f6e 2c20  g, flag option, 
-00003670: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
-00003680: 6174 6520 7468 6520 7365 6172 6368 696e  ate the searchin
-00003690: 6720 666f 7220 6e65 7374 6564 2073 696e  g for nested sin
-000036a0: 676c 652d 636f 7079 2067 656e 6520 6661  gle-copy gene fa
-000036b0: 6d69 6c69 6573 2028 4e53 4f47 7329 2028  milies (NSOGs) (
-000036c0: 6f6e 6c79 206d 6561 6e69 6e67 6675 6c20  only meaningful 
-000036d0: 7768 656e 2074 6865 206f 7274 686f 6772  when the orthogr
-000036e0: 6f75 7020 696e 6665 726e 6563 6520 7069  oup infernece pi
-000036f0: 7065 6c69 6e65 2077 6173 2061 6374 6976  peline was activ
-00003700: 6174 6564 292c 2069 6620 7468 6520 666c  ated), if the fl
-00003710: 6167 2077 6173 2073 6574 2c20 6164 6469  ag was set, addi
-00003720: 7469 6f6e 616c 204e 534f 4773 2061 6e61  tional NSOGs ana
-00003730: 6c79 7369 7320 776f 756c 6420 6265 2070  lysis would be p
-00003740: 6572 666f 726d 6564 2062 6573 6964 6573  erformed besides
-00003750: 2074 6865 2062 6173 6963 206f 7274 686f   the basic ortho
-00003760: 6772 6f75 7020 696e 6665 726e 6563 650a  group infernece.
-00003770: 2d74 7265 652c 202d 2d74 7265 655f 6d65  -tree, --tree_me
-00003780: 7468 6f64 2c20 7768 6963 6820 6765 6e65  thod, which gene
-00003790: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
-000037a0: 7072 6f67 7261 6d20 746f 2069 6e76 6f6b  program to invok
-000037b0: 6520 286f 6e6c 7920 6d65 616e 696e 6766  e (only meaningf
-000037c0: 756c 2077 6865 6e20 7468 6520 636f 6c6c  ul when the coll
-000037d0: 696e 6561 7220 636f 616c 6573 6365 6e63  inear coalescenc
-000037e0: 652c 2067 656e 652d 746f 2d66 616d 696c  e, gene-to-famil
-000037f0: 7920 6173 7369 676e 6d65 6e74 206f 7220  y assignment or 
-00003800: 4e53 4f47 7320 616e 616c 7973 6973 2077  NSOGs analysis w
-00003810: 6572 6520 6163 7469 7661 7465 6429 2c20  ere activated), 
-00003820: 6465 6661 756c 7420 6661 7374 7472 6565  default fasttree
-00003830: 0a2d 7473 2c20 2d2d 7472 6565 7365 742c  .-ts, --treeset,
-00003840: 2074 6865 2070 6172 616d 6574 6572 7320   the parameters 
-00003850: 7365 7474 696e 6720 666f 7220 6765 6e65  setting for gene
-00003860: 2074 7265 6520 696e 6665 7265 6e63 652c   tree inference,
-00003870: 2064 6566 6175 6c74 204e 6f6e 652c 2074   default None, t
-00003880: 6869 7320 6f70 7469 6f6e 2063 616e 2062  his option can b
-00003890: 6520 7072 6f76 6964 6564 206d 756c 7469  e provided multi
-000038a0: 706c 6520 7469 6d65 730a 2d6d 632c 202d  ple times.-mc, -
-000038b0: 2d6d 736f 6763 7574 2c20 7468 6520 7261  -msogcut, the ra
-000038c0: 7469 6f20 6375 746f 6666 2066 6f72 206d  tio cutoff for m
-000038d0: 6f73 746c 7920 7369 6e67 6c65 2d63 6f70  ostly single-cop
-000038e0: 7920 6661 6d69 6c79 2028 6d65 616e 696e  y family (meanin
-000038f0: 6766 756c 2077 6865 6e20 6163 7469 7661  gful when activa
-00003900: 7469 6e67 2074 6865 206f 7274 686f 6772  ting the orthogr
-00003910: 6f75 7020 696e 6665 726e 6563 6520 7069  oup infernece pi
-00003920: 7065 6c69 6e65 2920 616e 6420 7370 6563  peline) and spec
-00003930: 6965 7320 7265 7072 6573 656e 7461 7469  ies representati
-00003940: 6f6e 2069 6e20 636f 6c6c 696e 6561 7220  on in collinear 
-00003950: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
-00003960: 7973 6973 2c20 6465 6661 756c 7420 302e  ysis, default 0.
-00003970: 382e 0a2d 6761 2c20 2d2d 6765 6e65 6173  8..-ga, --geneas
-00003980: 7369 676e 2c20 666c 6167 206f 7074 696f  sign, flag optio
-00003990: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
-000039a0: 6974 6961 7465 2074 6865 2067 656e 652d  itiate the gene-
-000039b0: 746f 2d66 616d 696c 7920 6173 7369 676e  to-family assign
-000039c0: 6d65 6e74 2061 6e61 6c79 7369 732c 2069  ment analysis, i
-000039d0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-000039e0: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
-000039f0: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00003a00: 7465 640a 2d73 612c 202d 2d73 6571 3261  ted.-sa, --seq2a
-00003a10: 7373 6967 6e2c 2074 6865 2071 7565 7269  ssign, the queri
-00003a20: 6564 2073 6571 7565 6e63 6573 2064 6174  ed sequences dat
-00003a30: 6120 6669 6c65 2069 6e20 6765 6e65 2d74  a file in gene-t
-00003a40: 6f2d 6661 6d69 6c79 2061 7373 6967 6e6d  o-family assignm
-00003a50: 656e 7420 616e 616c 7973 6973 2c20 6465  ent analysis, de
-00003a60: 6661 756c 7420 4e6f 6e65 2c20 7468 6973  fault None, this
-00003a70: 206f 7074 696f 6e20 6361 6e20 6265 2070   option can be p
-00003a80: 726f 7669 6465 6420 6d75 6c74 6970 6c65  rovided multiple
-00003a90: 2074 696d 6573 0a2d 6661 2c20 2d2d 6661   times.-fa, --fa
-00003aa0: 6d32 6173 7369 676e 2c20 7468 6520 7175  m2assign, the qu
-00003ab0: 6572 6965 6420 6661 6d69 6c69 7920 6461  eried familiy da
-00003ac0: 7461 2066 696c 6520 696e 2067 656e 652d  ta file in gene-
-00003ad0: 746f 2d66 616d 696c 7920 6173 7369 676e  to-family assign
-00003ae0: 6d65 6e74 2061 6e61 6c79 7369 732c 2064  ment analysis, d
-00003af0: 6566 6175 6c74 204e 6f6e 650a 2d63 632c  efault None.-cc,
-00003b00: 202d 2d63 6f6e 6361 742c 2066 6c61 6720   --concat, flag 
-00003b10: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00003b20: 746f 2069 6e69 7469 6174 6520 7468 6520  to initiate the 
-00003b30: 636f 6e63 6174 656e 6174 696f 6e20 7069  concatenation pi
-00003b40: 7065 6c69 6e65 2066 6f72 206f 7274 686f  peline for ortho
-00003b50: 6772 6f75 7020 696e 6665 726e 6563 652c  group infernece,
-00003b60: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00003b70: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
-00003b80: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
-00003b90: 6961 7465 640a 2d74 652c 202d 2d74 6573  iated.-te, --tes
-00003ba0: 7473 6f67 2c20 666c 6167 206f 7074 696f  tsog, flag optio
-00003bb0: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
-00003bc0: 6974 6961 7465 2074 6865 2075 6e62 6961  itiate the unbia
-00003bd0: 7365 6420 7465 7374 206f 6620 7369 6e67  sed test of sing
-00003be0: 6c65 2d63 6f70 7920 6765 6e65 2066 616d  le-copy gene fam
-00003bf0: 696c 6965 732c 2069 6620 7468 6520 666c  ilies, if the fl
-00003c00: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
-00003c10: 616e 616c 7973 6973 2077 6f75 6c64 2062  analysis would b
-00003c20: 6520 696e 6974 6961 7465 640a 2d62 732c  e initiated.-bs,
-00003c30: 202d 2d62 696e 732c 2074 6865 206e 756d   --bins, the num
-00003c40: 6265 7220 6f66 2062 696e 7320 6469 7669  ber of bins divi
-00003c50: 6465 6420 696e 2074 6865 2067 656e 6520  ded in the gene 
-00003c60: 6c65 6e67 7468 206e 6f72 6d61 6c69 7a61  length normaliza
-00003c70: 7469 6f6e 2c20 6465 6661 756c 7420 3130  tion, default 10
-00003c80: 300a 2d6e 702c 202d 2d6e 6f72 6d61 6c69  0.-np, --normali
-00003c90: 7a65 6470 6572 6365 6e74 2c20 7468 6520  zedpercent, the 
-00003ca0: 7065 7263 656e 7461 6765 206f 6620 7570  percentage of up
-00003cb0: 7065 7220 6869 7473 2075 7365 6420 666f  per hits used fo
-00003cc0: 7220 6765 6e65 206c 656e 6774 6820 6e6f  r gene length no
-00003cd0: 726d 616c 697a 6174 696f 6e2c 2064 6566  rmalization, def
-00003ce0: 6175 6c74 2035 0a2d 6e6e 2c20 2d2d 6e6f  ault 5.-nn, --no
-00003cf0: 6e6f 726d 616c 697a 6174 696f 6e2c 2066  normalization, f
-00003d00: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00003d10: 6865 7220 746f 2063 616c 6c20 6f66 6620  her to call off 
-00003d20: 7468 6520 6e6f 726d 616c 697a 6174 696f  the normalizatio
-00003d30: 6e2c 2069 6620 7468 6520 666c 6167 2077  n, if the flag w
-00003d40: 6173 2073 6574 2c20 6e6f 206e 6f72 6d61  as set, no norma
-00003d50: 6c69 7a61 7469 6f6e 2077 6f75 6c64 2062  lization would b
-00003d60: 6520 636f 6e64 7563 7465 640a 2d62 736f  e conducted.-bso
-00003d70: 672c 202d 2d62 7573 636f 736f 672c 2066  g, --buscosog, f
-00003d80: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00003d90: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
-00003da0: 7468 6520 6275 7363 6f2d 6775 6964 6564  the busco-guided
-00003db0: 2073 696e 676c 652d 636f 7079 2067 656e   single-copy gen
-00003dc0: 6520 6661 6d69 6c79 2061 6e61 6c79 7369  e family analysi
-00003dd0: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
-00003de0: 6173 2073 6574 2c20 7468 6520 616e 616c  as set, the anal
-00003df0: 7973 6973 2077 6f75 6c64 2062 6520 696e  ysis would be in
-00003e00: 6974 6961 7465 640a 2d62 686d 6d2c 202d  itiated.-bhmm, -
-00003e10: 2d62 7573 636f 686d 6d2c 2074 6865 2048  -buscohmm, the H
-00003e20: 4d4d 2070 726f 6669 6c65 2064 6174 6166  MM profile dataf
-00003e30: 696c 6520 696e 2074 6865 2062 7573 636f  ile in the busco
-00003e40: 2d67 7569 6465 6420 7369 6e67 6c65 2d63  -guided single-c
-00003e50: 6f70 7920 6765 6e65 2066 616d 696c 7920  opy gene family 
-00003e60: 616e 616c 7973 6973 2c20 6465 6661 756c  analysis, defaul
-00003e70: 7420 4e6f 6e65 0a2d 6263 7466 2c20 2d2d  t None.-bctf, --
-00003e80: 6275 7363 6f63 7574 6f66 662c 2074 6865  buscocutoff, the
-00003e90: 2048 4d4d 2073 636f 7265 2063 7574 6f66   HMM score cutof
-00003ea0: 6620 6461 7461 6669 6c65 2069 6e20 7468  f datafile in th
-00003eb0: 6520 6275 7363 6f2d 6775 6964 6564 2073  e busco-guided s
-00003ec0: 696e 676c 652d 636f 7079 2067 656e 6520  ingle-copy gene 
-00003ed0: 6661 6d69 6c79 2061 6e61 6c79 7369 732c  family analysis,
-00003ee0: 2064 6566 6175 6c74 204e 6f6e 650a 2d6f   default None.-o
-00003ef0: 6620 2c2d 2d6f 6766 6f72 6d61 742c 2066  f ,--ogformat, f
-00003f00: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00003f10: 6865 7220 746f 2061 6464 2069 6e64 6578  her to add index
-00003f20: 2074 6f20 7468 6520 5242 4820 6661 6d69   to the RBH fami
-00003f30: 6c69 6573 0a60 6060 0a0a 5468 6520 7072  lies.```..The pr
-00003f40: 6f67 7261 6d20 6077 6764 2066 6f63 7573  ogram `wgd focus
-00003f50: 6020 6361 6e20 7265 616c 697a 6520 7468  ` can realize th
-00003f60: 6520 636f 6e63 6174 656e 6174 696f 6e2d  e concatenation-
-00003f70: 6261 7365 6420 616e 6420 636f 616c 6573  based and coales
-00003f80: 6365 6e63 652d 6261 7365 6420 7068 796c  cence-based phyl
-00003f90: 6f67 656e 6574 6963 2069 6e66 6572 656e  ogenetic inferen
-00003fa0: 6365 2061 6e64 2070 6879 6c6f 6765 6e65  ce and phylogene
-00003fb0: 7469 6320 6461 7469 6e67 206f 6620 5747  tic dating of WG
-00003fc0: 4473 2065 7463 2e0a 6060 600a 7767 6420  Ds etc..```.wgd 
-00003fd0: 666f 6375 7320 6661 6d69 6c69 6573 2073  focus families s
-00003fe0: 6571 7565 6e63 6573 2028 6f70 7469 6f6e  equences (option
-00003ff0: 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ).--------------
-00004000: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004010: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004020: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004030: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004040: 2d2d 0a2d 6f2c 202d 2d6f 7574 6469 722c  --.-o, --outdir,
-00004050: 2074 6865 206f 7574 7075 7420 6469 7265   the output dire
-00004060: 6374 6f72 792c 2064 6566 6175 6c74 2077  ctory, default w
-00004070: 6764 5f66 6f63 7573 0a2d 742c 202d 2d74  gd_focus.-t, --t
-00004080: 6d70 6469 722c 2074 6865 2074 656d 706f  mpdir, the tempo
-00004090: 7261 7279 2077 6f72 6b69 6e67 2064 6972  rary working dir
-000040a0: 6563 746f 7279 2c20 6465 6661 756c 7420  ectory, default 
-000040b0: 4e6f 6e65 2c20 6966 204e 6f6e 6520 7761  None, if None wa
-000040c0: 7320 6769 7665 6e2c 2074 6865 2074 6d70  s given, the tmp
-000040d0: 6469 7220 7769 6c6c 2062 6520 6173 7369  dir will be assi
-000040e0: 676e 6564 2072 616e 646f 6d20 6e61 6d65  gned random name
-000040f0: 7320 696e 2063 7572 7265 6e74 2064 6972  s in current dir
-00004100: 6563 746f 7279 2061 6e64 2061 7574 6f6d  ectory and autom
-00004110: 6174 656c 7920 7265 6d6f 7665 6420 6174  ately removed at
-00004120: 2074 6865 2063 6f6d 706c 6574 696f 6e20   the completion 
-00004130: 6f66 2070 726f 6772 616d 2c20 656c 7365  of program, else
-00004140: 2074 6865 2074 6d70 6469 7220 776f 756c   the tmpdir woul
-00004150: 6420 6265 206b 6570 740a 2d6e 2c20 2d2d  d be kept.-n, --
-00004160: 6e74 6872 6561 6473 2c20 7468 6520 6e75  nthreads, the nu
-00004170: 6d62 6572 206f 6620 7468 7265 6164 7320  mber of threads 
-00004180: 746f 2075 7365 2c20 6465 6661 756c 7420  to use, default 
-00004190: 340a 2d2d 746f 5f73 746f 702c 2066 6c61  4.--to_stop, fla
-000041a0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-000041b0: 7220 746f 2074 7261 6e73 6c61 7465 2074  r to translate t
-000041c0: 6872 6f75 6768 2053 544f 5020 636f 646f  hrough STOP codo
-000041d0: 6e73 2c20 6966 2074 6865 2066 6c61 6720  ns, if the flag 
-000041e0: 7761 7320 7365 742c 2074 7261 6e73 6c61  was set, transla
-000041f0: 7469 6f6e 2077 696c 6c20 6265 2074 6572  tion will be ter
-00004200: 6d69 6e61 7465 6420 6174 2074 6865 2066  minated at the f
-00004210: 6972 7374 2069 6e2d 6672 616d 6520 7374  irst in-frame st
-00004220: 6f70 2063 6f64 6f6e 2c20 656c 7365 2061  op codon, else a
-00004230: 2066 756c 6c20 7472 616e 736c 6174 696f   full translatio
-00004240: 6e20 636f 6e74 696e 7569 6e67 206f 6e20  n continuing on 
-00004250: 7061 7374 2061 6e79 2073 746f 7020 636f  past any stop co
-00004260: 646f 6e73 2077 6f75 6c64 2062 6520 696e  dons would be in
-00004270: 6974 6961 7465 640a 2d2d 6364 732c 2066  itiated.--cds, f
-00004280: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00004290: 6865 7220 746f 206f 6e6c 7920 7472 616e  her to only tran
-000042a0: 736c 6174 6520 7468 6520 636f 6d70 6c65  slate the comple
-000042b0: 7465 2043 4453 2074 6861 7420 7374 6172  te CDS that star
-000042c0: 7473 2077 6974 6820 6120 7661 6c69 6420  ts with a valid 
-000042d0: 7374 6172 7420 636f 646f 6e20 616e 6420  start codon and 
-000042e0: 6f6e 6c79 2063 6f6e 7461 696e 7320 6120  only contains a 
-000042f0: 7369 6e67 6c65 2069 6e2d 6672 616d 6520  single in-frame 
-00004300: 7374 6f70 2063 6f64 6f6e 2061 7420 7468  stop codon at th
-00004310: 6520 656e 6420 616e 6420 6d75 7374 2062  e end and must b
-00004320: 6520 6469 7669 6461 626c 6520 6279 2074  e dividable by t
-00004330: 6872 6565 2c20 6966 2074 6865 2066 6c61  hree, if the fla
-00004340: 6720 7761 7320 7365 742c 206f 6e6c 7920  g was set, only 
-00004350: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
-00004360: 2077 6f75 6c64 2062 6520 7472 616e 736c   would be transl
-00004370: 6174 6564 0a2d 2d73 7472 6970 5f67 6170  ated.--strip_gap
-00004380: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
-00004390: 7768 6574 6865 7220 746f 2064 726f 7020  whether to drop 
-000043a0: 616c 6c20 6761 7073 2069 6e20 6d75 6c74  all gaps in mult
-000043b0: 6970 6c65 2073 6571 7565 6e63 6520 616c  iple sequence al
-000043c0: 6967 6e6d 656e 742c 2069 6620 7468 6520  ignment, if the 
-000043d0: 666c 6167 2077 6173 2073 6574 2c20 616c  flag was set, al
-000043e0: 6c20 6761 7073 2077 6f75 6c64 2062 6520  l gaps would be 
-000043f0: 6472 6f70 7065 640a 2d61 2c20 2d2d 616c  dropped.-a, --al
-00004400: 6967 6e65 722c 2077 6869 6368 2061 6c69  igner, which ali
-00004410: 676e 6d65 6e74 2070 726f 6772 616d 2074  gnment program t
-00004420: 6f20 7573 652c 2064 6566 6175 6c74 206d  o use, default m
-00004430: 6166 6674 0a2d 7472 6565 2c20 2d2d 7472  afft.-tree, --tr
-00004440: 6565 5f6d 6574 686f 642c 2077 6869 6368  ee_method, which
-00004450: 2067 656e 6520 7472 6565 2069 6e66 6572   gene tree infer
-00004460: 656e 6365 2070 726f 6772 616d 2074 6f20  ence program to 
-00004470: 696e 766f 6b65 2c20 6465 6661 756c 7420  invoke, default 
-00004480: 6661 7374 7472 6565 0a2d 7473 2c20 2d2d  fasttree.-ts, --
-00004490: 7472 6565 7365 742c 2074 6865 2070 6172  treeset, the par
-000044a0: 616d 6574 6572 7320 7365 7474 696e 6720  ameters setting 
-000044b0: 666f 7220 6765 6e65 2074 7265 6520 696e  for gene tree in
-000044c0: 6665 7265 6e63 652c 2064 6566 6175 6c74  ference, default
-000044d0: 204e 6f6e 652c 2074 6869 7320 6f70 7469   None, this opti
-000044e0: 6f6e 2063 616e 2062 6520 7072 6f76 6964  on can be provid
-000044f0: 6564 206d 756c 7469 706c 6520 7469 6d65  ed multiple time
-00004500: 730a 2d2d 636f 6e63 6174 656e 6174 696f  s.--concatenatio
-00004510: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
-00004520: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
-00004530: 6174 6520 7468 6520 636f 6e63 6174 656e  ate the concaten
-00004540: 6174 696f 6e2d 6261 7365 6420 7370 6563  ation-based spec
-00004550: 6965 7320 7472 6565 2069 6e66 6572 656e  ies tree inferen
-00004560: 6365 2c20 6966 2074 6865 2066 6c61 6720  ce, if the flag 
-00004570: 7761 7320 7365 742c 2063 6f6e 6361 7465  was set, concate
-00004580: 6e61 7469 6f6e 2d62 6173 6564 2073 7065  nation-based spe
-00004590: 6369 6573 2074 7265 6520 776f 756c 6420  cies tree would 
-000045a0: 6265 2069 6e66 6572 6564 0a2d 2d63 6f61  be infered.--coa
-000045b0: 6c65 7363 656e 6365 2c20 666c 6167 206f  lescence, flag o
-000045c0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000045d0: 6f20 696e 6974 6961 7465 2074 6865 2063  o initiate the c
-000045e0: 6f61 6c65 7363 656e 6365 2d62 6173 6564  oalescence-based
-000045f0: 2073 7065 6369 6573 2074 7265 6520 696e   species tree in
-00004600: 6665 7265 6e63 652c 2069 6620 7468 6520  ference, if the 
-00004610: 666c 6167 2077 6173 2073 6574 2c20 636f  flag was set, co
-00004620: 616c 6573 6365 6e63 652d 6261 7365 6420  alescence-based 
-00004630: 7370 6563 6965 7320 7472 6565 2077 6f75  species tree wou
-00004640: 6c64 2062 6520 696e 6665 7265 640a 2d73  ld be infered.-s
-00004650: 702c 202d 2d73 7065 6369 6573 7472 6565  p, --speciestree
-00004660: 2c20 7370 6563 6965 7320 7472 6565 2064  , species tree d
-00004670: 6174 6166 696c 6520 666f 7220 6461 7469  atafile for dati
-00004680: 6e67 2c20 6465 6661 756c 7420 4e6f 6e65  ng, default None
-00004690: 0a2d 642c 202d 2d64 6174 696e 672c 2077  .-d, --dating, w
-000046a0: 6869 6368 206d 6f6c 6563 756c 6172 2064  hich molecular d
-000046b0: 6174 696e 6720 7072 6f67 7261 6d20 746f  ating program to
-000046c0: 2075 7365 2c20 6465 6661 756c 7420 6e6f   use, default no
-000046d0: 6e65 0a2d 6473 2c20 2d2d 6461 7469 6e67  ne.-ds, --dating
-000046e0: 7365 742c 2074 6865 2070 6172 616d 6574  set, the paramet
-000046f0: 6572 7320 7365 7474 696e 6720 666f 7220  ers setting for 
-00004700: 6461 7469 6e67 2070 726f 6772 616d 2c20  dating program, 
-00004710: 6465 6661 756c 7420 4e6f 6e65 2c20 7468  default None, th
-00004720: 6973 206f 7074 696f 6e20 6361 6e20 6265  is option can be
-00004730: 2070 726f 7669 6465 6420 6d75 6c74 6970   provided multip
-00004740: 6c65 2074 696d 6573 0a2d 6e73 2c20 2d2d  le times.-ns, --
-00004750: 6e73 6974 6573 2c20 7468 6520 6e73 6974  nsites, the nsit
-00004760: 6573 2069 6e66 6f72 6d61 7469 6f6e 2066  es information f
-00004770: 6f72 2072 3873 2064 6174 696e 672c 2064  or r8s dating, d
-00004780: 6566 6175 6c74 204e 6f6e 650a 2d6f 742c  efault None.-ot,
-00004790: 202d 2d6f 7574 6772 6f75 702c 2074 6865   --outgroup, the
-000047a0: 206f 7574 6772 6f75 7020 696e 666f 726d   outgroup inform
-000047b0: 6174 696f 6e20 666f 7220 7238 7320 6461  ation for r8s da
-000047c0: 7469 6e67 2c20 6465 6661 756c 7420 4e6f  ting, default No
-000047d0: 6e65 0a2d 7074 2c20 2d2d 7061 7274 6974  ne.-pt, --partit
-000047e0: 696f 6e2c 2066 6c61 6720 6f70 7469 6f6e  ion, flag option
-000047f0: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
-00004800: 7469 6174 6520 7061 7274 6974 696f 6e20  tiate partition 
-00004810: 6461 7469 6e67 2061 6e61 6c79 7369 7320  dating analysis 
-00004820: 666f 7220 636f 646f 6e2c 2069 6620 7468  for codon, if th
-00004830: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00004840: 616e 2061 6464 6974 696f 6e61 6c20 7061  an additional pa
-00004850: 7274 6974 696f 6e20 6461 7469 6e67 2061  rtition dating a
-00004860: 6e61 6c79 7369 7320 776f 756c 6420 6265  nalysis would be
-00004870: 2069 6e69 7469 6174 6564 0a2d 616d 2c20   initiated.-am, 
-00004880: 2d2d 6161 6d6f 6465 6c2c 2077 6869 6368  --aamodel, which
-00004890: 2070 726f 7465 696e 206d 6f64 656c 2074   protein model t
-000048a0: 6f20 6265 2075 7365 6420 696e 206d 636d  o be used in mcm
-000048b0: 6374 7265 652c 2064 6566 6175 6c74 2070  ctree, default p
-000048c0: 6f69 7373 6f6e 0a2d 6b73 2c20 666c 6167  oisson.-ks, flag
-000048d0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-000048e0: 2074 6f20 696e 6974 6961 7465 204b 7320   to initiate Ks 
-000048f0: 6361 6c63 756c 6174 696f 6e20 666f 7220  calculation for 
-00004900: 686f 6d6f 6c6f 6775 6573 2069 6e20 7468  homologues in th
-00004910: 6520 7072 6f76 6964 6564 206f 7274 686f  e provided ortho
-00004920: 6c6f 676f 7573 2067 656e 6520 6661 6d69  logous gene fami
-00004930: 6c79 0a2d 2d61 6e6e 6f74 6174 696f 6e2c  ly.--annotation,
-00004940: 2077 6869 6368 2061 6e6e 6f74 6174 696f   which annotatio
-00004950: 6e20 7072 6f67 7261 6d20 746f 2075 7365  n program to use
-00004960: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-00004970: 2d70 6169 7277 6973 652c 2066 6c61 6720  -pairwise, flag 
-00004980: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00004990: 746f 2069 6e69 7469 6174 6520 7061 6972  to initiate pair
-000049a0: 7769 7365 204b 7320 6573 7469 6d61 7469  wise Ks estimati
-000049b0: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
-000049c0: 7761 7320 7365 742c 2070 6169 7277 6973  was set, pairwis
-000049d0: 6520 4b73 2076 616c 7565 7320 776f 756c  e Ks values woul
-000049e0: 6420 6265 2065 7374 696d 6174 6564 0a2d  d be estimated.-
-000049f0: 6564 2c20 2d2d 6567 676e 6f67 6461 7461  ed, --eggnogdata
-00004a00: 2c20 7468 6520 6567 676e 6f67 2061 6e6e  , the eggnog ann
-00004a10: 6f74 6174 696f 6e20 6461 7461 6669 6c65  otation datafile
-00004a20: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-00004a30: 2d70 6661 6d2c 2077 6869 6368 206f 7074  -pfam, which opt
-00004a40: 696f 6e20 746f 2075 7365 2066 6f72 2070  ion to use for p
-00004a50: 6661 6d20 616e 6e6f 7461 7469 6f6e 2c20  fam annotation, 
-00004a60: 6465 6661 756c 7420 4e6f 6e65 0a2d 2d64  default None.--d
-00004a70: 6d6e 622c 2074 6865 2064 6961 6d6f 6e64  mnb, the diamond
-00004a80: 2064 6174 6162 6173 6520 666f 7220 616e   database for an
-00004a90: 6e6f 7461 7469 6f6e 2c20 6465 6661 756c  notation, defaul
-00004aa0: 7420 4e6f 6e65 0a2d 2d68 6d6d 2c20 7468  t None.--hmm, th
-00004ab0: 6520 484d 4d20 7072 6f66 696c 6520 666f  e HMM profile fo
-00004ac0: 7220 616e 6e6f 7461 7469 6f6e 2c20 6465  r annotation, de
-00004ad0: 6661 756c 7420 4e6f 6e65 0a2d 2d65 7661  fault None.--eva
-00004ae0: 6c75 652c 2074 6865 2065 2d76 616c 7565  lue, the e-value
-00004af0: 2063 7574 2d6f 6666 2066 6f72 2061 6e6e   cut-off for ann
-00004b00: 6f74 6174 696f 6e2c 2064 6566 6175 6c74  otation, default
-00004b10: 2031 652d 3130 0a2d 2d65 7865 7061 7468   1e-10.--exepath
-00004b20: 2c20 7468 6520 7061 7468 2074 6f20 7468  , the path to th
-00004b30: 6520 696e 7465 7270 726f 7363 616e 2065  e interproscan e
-00004b40: 7865 6375 7461 626c 652c 2064 6566 6175  xecutable, defau
-00004b50: 6c74 204e 6f6e 650a 2d66 2c20 2d2d 666f  lt None.-f, --fo
-00004b60: 7373 696c 2c20 7468 6520 666f 7373 696c  ssil, the fossil
-00004b70: 2063 616c 6962 7261 7469 6f6e 2069 6e66   calibration inf
-00004b80: 6f72 6d61 7469 6f6e 2069 6e20 4265 6173  ormation in Beas
-00004b90: 742c 2064 6566 6175 6c74 2028 2763 6c61  t, default ('cla
-00004ba0: 6465 313b 636c 6164 6532 272c 2027 7461  de1;clade2', 'ta
-00004bb0: 7861 312c 7461 7861 323b 7461 7861 332c  xa1,taxa2;taxa3,
-00004bc0: 7461 7861 3427 2c20 2734 3b35 272c 2027  taxa4', '4;5', '
-00004bd0: 302e 353b 302e 3627 2c20 2734 3030 3b35  0.5;0.6', '400;5
-00004be0: 3030 2729 0a2d 7268 2c20 2d2d 726f 6f74  00').-rh, --root
-00004bf0: 6865 6967 6874 2c20 7468 6520 726f 6f74  height, the root
-00004c00: 2068 6569 6768 7420 6361 6c69 6272 6174   height calibrat
-00004c10: 696f 6e20 696e 666f 2069 6e20 4265 6173  ion info in Beas
-00004c20: 742c 2064 6566 6175 6c74 2028 342c 302e  t, default (4,0.
-00004c30: 352c 3430 3029 0a2d 6373 2c20 2d2d 6368  5,400).-cs, --ch
-00004c40: 6169 6e73 6574 2c20 7468 6520 7061 7261  ainset, the para
-00004c50: 6d65 7465 7273 206f 6620 4d43 4d43 2063  meters of MCMC c
-00004c60: 6861 696e 2069 6e20 4265 6173 742c 2064  hain in Beast, d
-00004c70: 6566 6175 6c74 2028 3130 3030 302c 3130  efault (10000,10
-00004c80: 3029 0a2d 2d62 6561 7374 6c67 6a61 722c  0).--beastlgjar,
-00004c90: 2074 6865 2070 6174 6820 746f 2062 6561   the path to bea
-00004ca0: 7374 4c47 2e6a 6172 2c20 6465 6661 756c  stLG.jar, defaul
-00004cb0: 7420 4e6f 6e65 0a2d 2d62 6561 676c 652c  t None.--beagle,
-00004cc0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004cd0: 6574 6865 7220 746f 2075 7365 2062 6561  ether to use bea
-00004ce0: 676c 6520 696e 2042 6561 7374 2c20 6966  gle in Beast, if
-00004cf0: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00004d00: 742c 2062 6561 676c 6520 776f 756c 6420  t, beagle would 
-00004d10: 6265 2075 7365 640a 2d2d 7072 6f74 636f  be used.--protco
-00004d20: 6364 6174 696e 672c 2066 6c61 6720 6f70  cdating, flag op
-00004d30: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00004d40: 206f 6e6c 7920 696e 6974 6961 7465 2074   only initiate t
-00004d50: 6865 2070 726f 7465 696e 2d63 6f6e 6361  he protein-conca
-00004d60: 7465 6e61 7469 6f6e 2d62 6173 6564 2064  tenation-based d
-00004d70: 6174 696e 6720 616e 616c 7973 6973 2c20  ating analysis, 
-00004d80: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-00004d90: 7365 742c 2074 6865 2061 6e61 6c79 7369  set, the analysi
-00004da0: 7320 776f 756c 6420 6265 2069 6e69 7469  s would be initi
-00004db0: 6174 6564 0a2d 2d70 726f 7464 6174 696e  ated.--protdatin
-00004dc0: 672c 2066 6c61 6720 6f70 7469 6f6e 2c20  g, flag option, 
-00004dd0: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
-00004de0: 696e 6974 6961 7465 2074 6865 2070 726f  initiate the pro
-00004df0: 7465 696e 2d62 6173 6564 2064 6174 696e  tein-based datin
-00004e00: 6720 616e 616c 7973 6973 2c20 6966 2074  g analysis, if t
-00004e10: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00004e20: 2074 6865 2061 6e61 6c79 7369 7320 776f   the analysis wo
-00004e30: 756c 6420 6265 2069 6e69 7469 6174 6564  uld be initiated
-00004e40: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
-00004e50: 6d20 6077 6764 206b 7364 6020 6361 6e20  m `wgd ksd` can 
-00004e60: 7265 616c 697a 6520 7468 6520 636f 6e73  realize the cons
-00004e70: 7472 7563 7469 6f6e 206f 6620 2a4b 2a3c  truction of *K*<
-00004e80: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-00004e90: 6469 7374 7269 6275 7469 6f6e 2061 6e64  distribution and
-00004ea0: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-00004eb0: 2e0a 6060 600a 7767 6420 6b73 6420 6661  ..```.wgd ksd fa
-00004ec0: 6d69 6c69 6573 2073 6571 7565 6e63 6573  milies sequences
-00004ed0: 2028 6f70 7469 6f6e 290a 2d2d 2d2d 2d2d   (option).------
-00004ee0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004ef0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004f00: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004f10: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004f20: 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c 202d  ----------.-o, -
-00004f30: 2d6f 7574 6469 722c 2074 6865 206f 7574  -outdir, the out
-00004f40: 7075 7420 6469 7265 6374 6f72 792c 2064  put directory, d
-00004f50: 6566 6175 6c74 2077 6764 5f6b 7364 0a2d  efault wgd_ksd.-
-00004f60: 742c 202d 2d74 6d70 6469 722c 2074 6865  t, --tmpdir, the
-00004f70: 2074 656d 706f 7261 7279 2077 6f72 6b69   temporary worki
-00004f80: 6e67 2064 6972 6563 746f 7279 2c20 6465  ng directory, de
-00004f90: 6661 756c 7420 4e6f 6e65 2c20 6966 204e  fault None, if N
-00004fa0: 6f6e 6520 7761 7320 6769 7665 6e2c 2074  one was given, t
-00004fb0: 6865 2074 6d70 6469 7220 7769 6c6c 2062  he tmpdir will b
-00004fc0: 6520 6173 7369 676e 6564 2072 616e 646f  e assigned rando
-00004fd0: 6d20 6e61 6d65 7320 696e 2063 7572 7265  m names in curre
-00004fe0: 6e74 2064 6972 6563 746f 7279 2061 6e64  nt directory and
-00004ff0: 2061 7574 6f6d 6174 656c 7920 7265 6d6f   automately remo
-00005000: 7665 6420 6174 2074 6865 2063 6f6d 706c  ved at the compl
-00005010: 6574 696f 6e20 6f66 2070 726f 6772 616d  etion of program
-00005020: 2c20 656c 7365 2074 6865 2074 6d70 6469  , else the tmpdi
-00005030: 7220 776f 756c 6420 6265 206b 6570 740a  r would be kept.
-00005040: 2d6e 2c20 2d2d 6e74 6872 6561 6473 2c20  -n, --nthreads, 
-00005050: 7468 6520 6e75 6d62 6572 206f 6620 7468  the number of th
-00005060: 7265 6164 7320 746f 2075 7365 2c20 6465  reads to use, de
-00005070: 6661 756c 7420 340a 2d2d 746f 5f73 746f  fault 4.--to_sto
-00005080: 702c 2066 6c61 6720 6f70 7469 6f6e 2c20  p, flag option, 
-00005090: 7768 6574 6865 7220 746f 2074 7261 6e73  whether to trans
-000050a0: 6c61 7465 2074 6872 6f75 6768 2053 544f  late through STO
-000050b0: 5020 636f 646f 6e73 2c20 6966 2074 6865  P codons, if the
-000050c0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-000050d0: 7261 6e73 6c61 7469 6f6e 2077 696c 6c20  ranslation will 
-000050e0: 6265 2074 6572 6d69 6e61 7465 6420 6174  be terminated at
-000050f0: 2074 6865 2066 6972 7374 2069 6e2d 6672   the first in-fr
-00005100: 616d 6520 7374 6f70 2063 6f64 6f6e 2c20  ame stop codon, 
-00005110: 656c 7365 2061 2066 756c 6c20 7472 616e  else a full tran
-00005120: 736c 6174 696f 6e20 636f 6e74 696e 7569  slation continui
-00005130: 6e67 206f 6e20 7061 7374 2061 6e79 2073  ng on past any s
-00005140: 746f 7020 636f 646f 6e73 2077 6f75 6c64  top codons would
-00005150: 2062 6520 696e 6974 6961 7465 640a 2d2d   be initiated.--
-00005160: 6364 732c 2066 6c61 6720 6f70 7469 6f6e  cds, flag option
-00005170: 2c20 7768 6574 6865 7220 746f 206f 6e6c  , whether to onl
-00005180: 7920 7472 616e 736c 6174 6520 7468 6520  y translate the 
-00005190: 636f 6d70 6c65 7465 2043 4453 2074 6861  complete CDS tha
-000051a0: 7420 7374 6172 7473 2077 6974 6820 6120  t starts with a 
-000051b0: 7661 6c69 6420 7374 6172 7420 636f 646f  valid start codo
-000051c0: 6e20 616e 6420 6f6e 6c79 2063 6f6e 7461  n and only conta
-000051d0: 696e 7320 6120 7369 6e67 6c65 2069 6e2d  ins a single in-
-000051e0: 6672 616d 6520 7374 6f70 2063 6f64 6f6e  frame stop codon
-000051f0: 2061 7420 7468 6520 656e 6420 616e 6420   at the end and 
-00005200: 6d75 7374 2062 6520 6469 7669 6461 626c  must be dividabl
-00005210: 6520 6279 2074 6872 6565 2c20 6966 2074  e by three, if t
-00005220: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00005230: 206f 6e6c 7920 7468 6520 636f 6d70 6c65   only the comple
-00005240: 7465 2043 4453 2077 6f75 6c64 2062 6520  te CDS would be 
-00005250: 7472 616e 736c 6174 6564 0a2d 2d70 6169  translated.--pai
-00005260: 7277 6973 652c 2066 6c61 6720 6f70 7469  rwise, flag opti
-00005270: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
-00005280: 6e69 7469 6174 6520 7061 6972 7769 7365  nitiate pairwise
-00005290: 204b 7320 6573 7469 6d61 7469 6f6e 2c20   Ks estimation, 
-000052a0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-000052b0: 7365 742c 2070 6169 7277 6973 6520 4b73  set, pairwise Ks
-000052c0: 2076 616c 7565 7320 776f 756c 6420 6265   values would be
-000052d0: 2065 7374 696d 6174 6564 0a2d 2d73 7472   estimated.--str
-000052e0: 6970 5f67 6170 732c 2066 6c61 6720 6f70  ip_gaps, flag op
-000052f0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00005300: 2064 726f 7020 616c 6c20 6761 7073 2069   drop all gaps i
-00005310: 6e20 6d75 6c74 6970 6c65 2073 6571 7565  n multiple seque
-00005320: 6e63 6520 616c 6967 6e6d 656e 742c 2069  nce alignment, i
-00005330: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00005340: 6574 2c20 616c 6c20 6761 7073 2077 6f75  et, all gaps wou
-00005350: 6c64 2062 6520 6472 6f70 7065 640a 2d61  ld be dropped.-a
-00005360: 2c20 2d2d 616c 6967 6e65 722c 2077 6869  , --aligner, whi
-00005370: 6368 2061 6c69 676e 6d65 6e74 2070 726f  ch alignment pro
-00005380: 6772 616d 2074 6f20 7573 652c 2064 6566  gram to use, def
-00005390: 6175 6c74 206d 6166 6674 200a 2d74 7265  ault mafft .-tre
-000053a0: 652c 202d 2d74 7265 655f 6d65 7468 6f64  e, --tree_method
-000053b0: 2c20 7768 6963 6820 6765 6e65 2074 7265  , which gene tre
-000053c0: 6520 696e 6665 7265 6e63 6520 7072 6f67  e inference prog
-000053d0: 7261 6d20 746f 2069 6e76 6f6b 652c 2064  ram to invoke, d
-000053e0: 6566 6175 6c74 2066 6173 7474 7265 650a  efault fasttree.
-000053f0: 2d2d 7472 6565 5f6f 7074 696f 6e73 2c20  --tree_options, 
-00005400: 6f70 7469 6f6e 7320 696e 2074 7265 6520  options in tree 
-00005410: 696e 6665 7265 6e63 6520 6173 2061 2063  inference as a c
-00005420: 6f6d 6d61 2073 6570 6172 6174 6564 2073  omma separated s
-00005430: 7472 696e 672c 2064 6566 6175 6c74 204e  tring, default N
-00005440: 6f6e 650a 2d2d 6e6f 6465 5f61 7665 7261  one.--node_avera
-00005450: 6765 2c20 666c 6167 206f 7074 696f 6e2c  ge, flag option,
-00005460: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
-00005470: 6961 7465 206e 6f64 652d 6176 6572 6167  iate node-averag
-00005480: 6520 7761 7920 6f66 2064 652d 7265 6475  e way of de-redu
-00005490: 6e64 616e 6379 2069 6e73 7465 6164 206f  ndancy instead o
-000054a0: 6620 6e6f 6465 2d77 6569 6768 7465 642c  f node-weighted,
-000054b0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-000054c0: 2073 6574 2c20 7468 6520 6e6f 6465 2d61   set, the node-a
-000054d0: 7665 7261 6769 6e67 2064 652d 7265 6475  veraging de-redu
-000054e0: 6e64 616e 6379 2077 6f75 6c64 2062 6520  ndancy would be 
-000054f0: 696e 6974 6961 7465 640a 2d73 722c 202d  initiated.-sr, -
-00005500: 2d73 7061 6972 2c20 7468 6520 7370 6563  -spair, the spec
-00005510: 6965 7320 7061 6972 2074 6f20 6265 2070  ies pair to be p
-00005520: 6c6f 7474 6564 2c20 6465 6661 756c 7420  lotted, default 
-00005530: 4e6f 6e65 2c20 7468 6973 206f 7074 696f  None, this optio
-00005540: 6e20 6361 6e20 6265 2070 726f 7669 6465  n can be provide
-00005550: 6420 6d75 6c74 6970 6c65 2074 696d 6573  d multiple times
-00005560: 0a2d 7370 2c20 2d2d 7370 6563 6965 7374  .-sp, --speciest
-00005570: 7265 652c 2074 6865 2073 7065 6369 6573  ree, the species
-00005580: 2074 7265 6520 746f 2070 6572 666f 726d   tree to perform
-00005590: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-000055a0: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
-000055b0: 6966 204e 6f6e 6520 7761 7320 6769 7665  if None was give
-000055c0: 6e2c 2074 6865 2072 6174 6520 636f 7272  n, the rate corr
-000055d0: 6563 7469 6f6e 2061 6e61 6c79 7369 7320  ection analysis 
-000055e0: 776f 756c 6420 6265 2063 616c 6c65 6420  would be called 
-000055f0: 6f66 660a 2d72 772c 202d 2d72 6577 6569  off.-rw, --rewei
-00005600: 6768 742c 2066 6c61 6720 6f70 7469 6f6e  ght, flag option
-00005610: 2c20 7768 6574 6865 7220 746f 2072 6563  , whether to rec
-00005620: 616c 6375 6c61 7465 2074 6865 2077 6569  alculate the wei
-00005630: 6768 7420 7065 7220 7370 6563 6965 7320  ght per species 
-00005640: 7061 6972 2c20 6966 2074 6865 2066 6c61  pair, if the fla
-00005650: 6720 7761 7320 7365 742c 2074 6865 2077  g was set, the w
-00005660: 6569 6768 7420 776f 756c 6420 6265 2072  eight would be r
-00005670: 6563 616c 6375 6c61 7465 640a 2d6f 722c  ecalculated.-or,
-00005680: 202d 2d6f 6e6c 7972 6f6f 746f 7574 2c20   --onlyrootout, 
-00005690: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-000056a0: 7468 6572 2074 6f20 6f6e 6c79 2063 6f6e  ther to only con
-000056b0: 6475 6374 2072 6174 6520 636f 7272 6563  duct rate correc
-000056c0: 7469 6f6e 2075 7369 6e67 2074 6865 206f  tion using the o
-000056d0: 7574 6772 6f75 7020 6174 2072 6f6f 7420  utgroup at root 
-000056e0: 6173 206f 7574 6772 6f75 702c 2069 6620  as outgroup, if 
-000056f0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00005700: 2c20 6f6e 6c79 2074 6865 206f 7574 6772  , only the outgr
-00005710: 6f75 7020 6174 2072 6f6f 7420 776f 756c  oup at root woul
-00005720: 6420 6265 2075 7365 6420 6173 206f 7574  d be used as out
-00005730: 6772 6f75 700a 2d65 706b 2c20 2d2d 6578  group.-epk, --ex
-00005740: 7472 6170 6172 616e 6f6d 656b 732c 2065  traparanomeks, e
-00005750: 7874 7261 2070 6172 616e 6f6d 6520 4b73  xtra paranome Ks
-00005760: 2064 6174 6120 746f 2070 6c6f 7420 696e   data to plot in
-00005770: 2074 6865 206d 6978 6564 204b 7320 6469   the mixed Ks di
-00005780: 7374 7269 6275 7469 6f6e 2c20 6465 6661  stribution, defa
-00005790: 756c 7420 4e6f 6e65 0a2d 6170 2c20 2d2d  ult None.-ap, --
-000057a0: 616e 6368 6f72 706f 696e 7473 2c20 616e  anchorpoints, an
-000057b0: 6368 6f72 706f 696e 7473 2e74 7874 2066  chorpoints.txt f
-000057c0: 696c 6520 746f 2070 6c6f 7420 616e 6368  ile to plot anch
-000057d0: 6f72 204b 7320 696e 2074 6865 206d 6978  or Ks in the mix
-000057e0: 6564 204b 7320 6469 7374 7269 6275 7469  ed Ks distributi
-000057f0: 6f6e 2c20 6465 6661 756c 7420 4e6f 6e65  on, default None
-00005800: 0a2d 706b 2c20 2d2d 706c 6f74 6b64 652c  .-pk, --plotkde,
-00005810: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00005820: 6574 6865 7220 746f 2070 6c6f 7420 6b64  ether to plot kd
-00005830: 6520 6375 7276 6520 6f66 206f 7274 686f  e curve of ortho
-00005840: 6c6f 676f 7573 204b 7320 6469 7374 7269  logous Ks distri
-00005850: 6275 7469 6f6e 206f 7665 7220 6869 7374  bution over hist
-00005860: 6f67 7261 6d20 696e 2074 6865 206d 6978  ogram in the mix
-00005870: 6564 204b 7320 6469 7374 7269 6275 7469  ed Ks distributi
-00005880: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
-00005890: 7761 7320 7365 742c 2074 6865 206b 6465  was set, the kde
-000058a0: 2063 7572 7665 2077 6f75 6c64 2062 6520   curve would be 
-000058b0: 706c 6f74 7465 640a 2d70 6167 2c20 2d2d  plotted.-pag, --
-000058c0: 706c 6f74 6170 676d 6d2c 2066 6c61 6720  plotapgmm, flag 
-000058d0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-000058e0: 746f 2070 6572 666f 726d 2061 6e64 2070  to perform and p
-000058f0: 6c6f 7420 6d69 7874 7572 6520 6d6f 6465  lot mixture mode
-00005900: 6c69 6e67 206f 6620 616e 6368 6f72 204b  ling of anchor K
-00005910: 7320 696e 2074 6865 206d 6978 6564 204b  s in the mixed K
-00005920: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
-00005930: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-00005940: 7365 742c 2074 6865 206d 6978 7475 7265  set, the mixture
-00005950: 206d 6f64 656c 696e 6720 6f66 2061 6e63   modeling of anc
-00005960: 686f 7220 4b73 2077 6f75 6c64 2062 6520  hor Ks would be 
-00005970: 706c 6f74 7465 640a 2d70 656d 2c20 2d2d  plotted.-pem, --
-00005980: 706c 6f74 656c 6d6d 2c20 666c 6167 206f  plotelmm, flag o
-00005990: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000059a0: 6f20 7065 7266 6f72 6d20 616e 6420 706c  o perform and pl
-000059b0: 6f74 2065 6c6d 6d20 6d69 7874 7572 6520  ot elmm mixture 
-000059c0: 6d6f 6465 6c69 6e67 206f 6620 7061 7261  modeling of para
-000059d0: 6e6f 6d65 204b 7320 696e 2074 6865 206d  nome Ks in the m
-000059e0: 6978 6564 204b 7320 6469 7374 7269 6275  ixed Ks distribu
-000059f0: 7469 6f6e 2c20 6966 2074 6865 2066 6c61  tion, if the fla
-00005a00: 6720 7761 7320 7365 742c 2074 6865 2065  g was set, the e
-00005a10: 6c6d 6d20 6d69 7874 7572 6520 6d6f 6465  lmm mixture mode
-00005a20: 6c69 6e67 206f 6620 7061 7261 6e6f 6d65  ling of paranome
-00005a30: 204b 7320 776f 756c 6420 6265 2070 6c6f   Ks would be plo
-00005a40: 7474 6564 0a2d 632c 202d 2d63 6f6d 706f  tted.-c, --compo
-00005a50: 6e65 6e74 732c 2074 6865 2072 616e 6765  nents, the range
-00005a60: 206f 6620 7468 6520 6e75 6d62 6572 206f   of the number o
-00005a70: 6620 636f 6d70 6f6e 656e 7473 2074 6f20  f components to 
-00005a80: 6669 7420 696e 2061 6e63 686f 7220 4b73  fit in anchor Ks
-00005a90: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
-00005aa0: 672c 2064 6566 6175 6c74 2028 312c 3429  g, default (1,4)
-00005ab0: 0a2d 786c 2c20 2d2d 786c 696d 2c20 7468  .-xl, --xlim, th
-00005ac0: 6520 7820 6178 6973 206c 696d 6974 206f  e x axis limit o
-00005ad0: 6620 4b73 2064 6973 7472 6962 7574 696f  f Ks distributio
-00005ae0: 6e0a 2d79 6c2c 202d 2d79 6c69 6d2c 2074  n.-yl, --ylim, t
-00005af0: 6865 2079 2061 7869 7320 6c69 6d69 7420  he y axis limit 
-00005b00: 6f66 204b 7320 6469 7374 7269 6275 7469  of Ks distributi
-00005b10: 6f6e 0a2d 6164 6f2c 202d 2d61 646a 7573  on.-ado, --adjus
-00005b20: 746f 7274 686f 2c20 666c 6167 206f 7074  tortho, flag opt
-00005b30: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-00005b40: 6164 6a75 7374 2074 6865 2068 6973 746f  adjust the histo
-00005b50: 6772 616d 2068 6569 6768 7420 6f66 206f  gram height of o
-00005b60: 7274 686f 6c6f 676f 7573 204b 7320 6173  rthologous Ks as
-00005b70: 2074 6f20 6d61 7463 6820 7468 6520 6865   to match the he
-00005b80: 6967 6874 206f 6620 7061 7261 6c6f 676f  ight of paralogo
-00005b90: 7573 204b 732c 2069 6620 7468 6520 666c  us Ks, if the fl
-00005ba0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
-00005bb0: 6164 6a75 7374 6d65 6e74 2077 6f75 6c64  adjustment would
-00005bc0: 2062 6520 636f 6e64 7563 7465 640a 2d61   be conducted.-a
-00005bd0: 6466 2c20 2d2d 6164 6a75 7374 6661 6374  df, --adjustfact
-00005be0: 6f72 2c20 7468 6520 6164 6a75 7374 6d65  or, the adjustme
-00005bf0: 6e74 2066 6163 746f 7220 6f66 206f 7274  nt factor of ort
-00005c00: 686f 6c6f 676f 7573 204b 732c 2064 6566  hologous Ks, def
-00005c10: 6175 6c74 2030 2e35 0a2d 6f61 2c20 2d2d  ault 0.5.-oa, --
-00005c20: 6f6b 616c 7068 612c 2074 6865 206f 7061  okalpha, the opa
-00005c30: 6369 7479 206f 6620 6f72 7468 6f6c 6f67  city of ortholog
-00005c40: 6f75 7320 4b73 2064 6973 7472 6962 7574  ous Ks distribut
-00005c50: 696f 6e20 696e 206d 6978 6564 2070 6c6f  ion in mixed plo
-00005c60: 742c 2064 6566 6175 6c74 2030 2e35 0a2d  t, default 0.5.-
-00005c70: 6661 2c20 2d2d 666f 6375 7332 616c 6c2c  fa, --focus2all,
-00005c80: 2073 6574 2066 6f63 616c 2073 7065 6369   set focal speci
-00005c90: 6573 2061 6e64 206c 6574 2073 7065 6369  es and let speci
-00005ca0: 6573 2070 6169 7220 746f 2062 6520 6265  es pair to be be
-00005cb0: 7477 6565 6e20 666f 6361 6c20 616e 6420  tween focal and 
-00005cc0: 616c 6c20 7468 6520 7265 6d61 696e 696e  all the remainin
-00005cd0: 6720 7370 6563 6965 732c 2064 6566 6175  g species, defau
-00005ce0: 6c74 204e 6f6e 650a 2d6b 732c 202d 2d6b  lt None.-ks, --k
-00005cf0: 7374 7265 652c 2066 6c61 6720 6f70 7469  stree, flag opti
-00005d00: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
-00005d10: 6e66 6572 204b 7320 7472 6565 2c20 6966  nfer Ks tree, if
-00005d20: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00005d30: 742c 2074 6865 204b 7320 7472 6565 2069  t, the Ks tree i
-00005d40: 6e66 6572 656e 6365 2061 6e61 6c79 7369  nference analysi
-00005d50: 7320 776f 756c 6420 6265 2069 6e69 7469  s would be initi
-00005d60: 6174 6564 0a2d 6f63 6b2c 202d 2d6f 6e6c  ated.-ock, --onl
-00005d70: 7963 6f6e 6361 746b 7374 7265 652c 2066  yconcatkstree, f
-00005d80: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00005d90: 6865 7220 746f 206f 6e6c 7920 696e 6665  her to only infe
-00005da0: 7220 4b73 2074 7265 6520 756e 6465 7220  r Ks tree under 
-00005db0: 636f 6e63 6174 656e 6174 6564 2061 6c69  concatenated ali
-00005dc0: 676e 6d65 6e74 2c20 6966 2074 6865 2066  gnment, if the f
-00005dd0: 6c61 6720 7761 7320 7365 742c 206f 6e6c  lag was set, onl
-00005de0: 7920 7468 6520 4b73 2074 7265 6520 756e  y the Ks tree un
-00005df0: 6465 7220 636f 6e63 6174 656e 6174 6564  der concatenated
-00005e00: 2061 6c69 676e 6d65 6e74 2077 6f75 6c64   alignment would
-00005e10: 2062 6520 6361 6c63 756c 6174 6564 0a2d   be calculated.-
-00005e20: 6373 2c20 2d2d 636c 6173 7369 632c 2066  cs, --classic, f
-00005e30: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00005e40: 6865 7220 746f 2064 7261 7720 6d69 7865  her to draw mixe
-00005e50: 6420 4b73 2070 6c6f 7420 696e 2061 2063  d Ks plot in a c
-00005e60: 6c61 7373 6963 206d 616e 6e65 7220 7768  lassic manner wh
-00005e70: 6572 6520 7468 6520 6675 6c6c 206f 7274  ere the full ort
-00005e80: 686f 6c6f 676f 7573 204b 7320 6469 7374  hologous Ks dist
-00005e90: 7269 6275 7469 6f6e 2069 7320 6472 6177  ribution is draw
-00005ea0: 6564 2c20 6966 2074 6865 2066 6c61 6720  ed, if the flag 
-00005eb0: 7761 7320 7365 742c 2074 6865 2063 6c61  was set, the cla
-00005ec0: 7373 6963 206d 6978 6564 204b 7320 706c  ssic mixed Ks pl
-00005ed0: 6f74 2077 6f75 6c64 2062 6520 6472 6177  ot would be draw
-00005ee0: 6e0a 2d74 612c 202d 2d74 6f70 6172 726f  n.-ta, --toparro
-00005ef0: 772c 2066 6c61 6720 6f70 7469 6f6e 2c20  w, flag option, 
-00005f00: 7768 6574 6865 7220 746f 2061 646a 7573  whether to adjus
-00005f10: 7420 7468 6520 6172 726f 7720 746f 2062  t the arrow to b
-00005f20: 6520 6174 2074 6865 2074 6f70 206f 6620  e at the top of 
-00005f30: 7468 6520 706c 6f74 2c20 696e 7374 6561  the plot, instea
-00005f40: 6420 6f66 2062 6569 6e67 2063 6f6f 7264  d of being coord
-00005f50: 696e 6174 6564 2061 7320 7468 6520 4b44  inated as the KD
-00005f60: 4520 6f66 2074 6865 206f 7274 686f 6c6f  E of the ortholo
-00005f70: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
-00005f80: 7469 6f6e 2c20 6966 2074 6865 2066 6c61  tion, if the fla
-00005f90: 6720 7761 7320 7365 742c 2074 6865 2061  g was set, the a
-00005fa0: 7272 6f77 2077 6f75 6c64 2062 6520 7365  rrow would be se
-00005fb0: 7420 6174 2074 6865 2074 6f70 0a2d 6273  t at the top.-bs
-00005fc0: 2c20 2d2d 626f 6f74 7374 7261 702c 2074  , --bootstrap, t
-00005fd0: 6865 206e 756d 6265 7220 6f66 2062 6f6f  he number of boo
-00005fe0: 7473 7472 6170 2072 6570 6c69 6361 7465  tstrap replicate
-00005ff0: 7320 6f66 206f 7274 686f 6c6f 6720 4b73  s of ortholog Ks
-00006000: 2064 6973 7472 6962 7574 696f 6e20 696e   distribution in
-00006010: 206d 6978 6564 2070 6c6f 740a 6060 600a   mixed plot.```.
-00006020: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
-00006030: 6420 6d69 7860 2063 616e 2072 6561 6c69  d mix` can reali
-00006040: 7a65 2074 6865 206d 6978 7475 7265 206d  ze the mixture m
-00006050: 6f64 656c 2063 6c75 7374 6572 696e 6720  odel clustering 
-00006060: 616e 616c 7973 6973 206f 6620 2a4b 2a3c  analysis of *K*<
-00006070: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-00006080: 6469 7374 7269 6275 7469 6f6e 2e0a 6060  distribution..``
-00006090: 600a 7767 6420 6d69 7820 6b73 5f64 6174  `.wgd mix ks_dat
-000060a0: 6166 696c 6520 286f 7074 696f 6e29 0a2d  afile (option).-
-000060b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000060c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000060d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000060e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000060f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d0a  ---------------.
-00006100: 2d66 2c20 2d2d 6669 6c74 6572 732c 2074  -f, --filters, t
-00006110: 6865 2063 7574 6f66 6620 616c 6967 6e6d  he cutoff alignm
-00006120: 656e 7420 6c65 6e67 7468 2c20 6465 6661  ent length, defa
-00006130: 756c 7420 3330 300a 2d72 2c20 2d2d 6b73  ult 300.-r, --ks
-00006140: 5f72 616e 6765 2c20 7468 6520 4b73 2072  _range, the Ks r
-00006150: 616e 6765 2074 6f20 6265 2063 6f6e 7369  ange to be consi
-00006160: 6465 7265 642c 2064 6566 6175 6c74 2028  dered, default (
-00006170: 302c 2035 290a 2d62 2c20 2d2d 6269 6e73  0, 5).-b, --bins
-00006180: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
-00006190: 6269 6e73 2069 6e20 4b73 2064 6973 7472  bins in Ks distr
-000061a0: 6962 7574 696f 6e2c 2064 6566 6175 6c74  ibution, default
-000061b0: 2035 300a 2d6f 2c20 2d2d 6f75 7464 6972   50.-o, --outdir
-000061c0: 2c20 7468 6520 6f75 7470 7574 2064 6972  , the output dir
-000061d0: 6563 746f 7279 2c20 6465 6661 756c 7420  ectory, default 
-000061e0: 7767 645f 6d69 780a 2d2d 6d65 7468 6f64  wgd_mix.--method
-000061f0: 2c20 7768 6963 6820 6d69 7874 7572 6520  , which mixture 
-00006200: 6d6f 6465 6c20 746f 2075 7365 2c20 6465  model to use, de
-00006210: 6661 756c 7420 676d 6d0a 2d6e 2c20 2d2d  fault gmm.-n, --
-00006220: 636f 6d70 6f6e 656e 7473 2c20 7468 6520  components, the 
-00006230: 7261 6e67 6520 6f66 2074 6865 206e 756d  range of the num
-00006240: 6265 7220 6f66 2063 6f6d 706f 6e65 6e74  ber of component
-00006250: 7320 746f 2066 6974 2c20 6465 6661 756c  s to fit, defaul
-00006260: 7420 2831 2c20 3429 0a2d 672c 202d 2d67  t (1, 4).-g, --g
-00006270: 616d 6d61 2c20 7468 6520 6761 6d6d 6120  amma, the gamma 
-00006280: 7061 7261 6d65 7465 7220 666f 7220 6267  parameter for bg
-00006290: 6d6d 206d 6f64 656c 732c 2064 6566 6175  mm models, defau
-000062a0: 6c74 2030 2e30 3031 0a2d 6e69 2c20 2d2d  lt 0.001.-ni, --
-000062b0: 6e5f 696e 6974 2c20 7468 6520 6e75 6d62  n_init, the numb
-000062c0: 6572 206f 6620 6b2d 6d65 616e 7320 696e  er of k-means in
-000062d0: 6974 6961 6c69 7a61 7469 6f6e 732c 2064  itializations, d
-000062e0: 6566 6175 6c74 2032 3030 0a2d 6d69 2c20  efault 200.-mi, 
-000062f0: 2d2d 6d61 785f 6974 6572 2c20 7468 6520  --max_iter, the 
-00006300: 6d61 7869 6d75 6d20 6e75 6d62 6572 206f  maximum number o
-00006310: 6620 6974 6572 6174 696f 6e73 2c20 6465  f iterations, de
-00006320: 6661 756c 7420 3230 300a 6060 600a 0a54  fault 200.```..T
-00006330: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-00006340: 7065 616b 6020 6361 6e20 7265 616c 697a  peak` can realiz
-00006350: 6520 7468 6520 7365 6172 6368 206f 6620  e the search of 
-00006360: 6372 6564 6961 626c 6520 2a4b 2a3c 7375  crediable *K*<su
-00006370: 623e 533c 2f73 7562 3e20 7261 6e67 6520  b>S</sub> range 
-00006380: 7573 6564 2069 6e20 5747 4420 6461 7469  used in WGD dati
-00006390: 6e67 2e0a 6060 600a 7767 6420 7065 616b  ng..```.wgd peak
-000063a0: 206b 735f 6461 7461 6669 6c65 2028 6f70   ks_datafile (op
-000063b0: 7469 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d  tion).----------
-000063c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000063d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000063e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000063f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006400: 2d2d 2d2d 2d2d 0a2d 6170 2c20 2d2d 616e  ------.-ap, --an
-00006410: 6368 6f72 706f 696e 7473 2c20 7468 6520  chorpoints, the 
-00006420: 616e 6368 6f72 2070 6f69 6e74 7320 6461  anchor points da
-00006430: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
-00006440: 4e6f 6e65 0a2d 736d 2c20 2d2d 7365 676d  None.-sm, --segm
-00006450: 656e 7473 2c20 7468 6520 7365 676d 656e  ents, the segmen
-00006460: 7473 2064 6174 6166 696c 652c 2064 6566  ts datafile, def
-00006470: 6175 6c74 204e 6f6e 650a 2d6c 652c 202d  ault None.-le, -
-00006480: 2d6c 6973 7465 6c65 6d65 6e74 732c 2074  -listelements, t
-00006490: 6865 206c 6973 7473 6567 6d65 6e74 7320  he listsegments 
-000064a0: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
-000064b0: 7420 4e6f 6e65 200a 2d6d 702c 202d 2d6d  t None .-mp, --m
-000064c0: 756c 7469 706c 6963 6f6e 7061 6972 732c  ultipliconpairs,
-000064d0: 2074 6865 206d 756c 7469 706c 6963 6f6e   the multiplicon
-000064e0: 7061 6972 7320 6461 7461 6669 6c65 2c20  pairs datafile, 
-000064f0: 6465 6661 756c 7420 4e6f 6e65 0a2d 6f2c  default None.-o,
-00006500: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
-00006510: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
-00006520: 2064 6566 6175 6c74 2077 6764 5f70 6561   default wgd_pea
-00006530: 6b0a 2d61 662c 202d 2d61 6c69 676e 6669  k.-af, --alignfi
-00006540: 6c74 6572 2c20 6375 746f 6666 2066 6f72  lter, cutoff for
-00006550: 2061 6c69 676e 6d65 6e74 2069 6465 6e74   alignment ident
-00006560: 6974 792c 206c 656e 6774 6820 616e 6420  ity, length and 
-00006570: 636f 7665 7261 6765 2c20 6465 6661 756c  coverage, defaul
-00006580: 7420 302e 302c 2030 2c20 302e 300a 2d72  t 0.0, 0, 0.0.-r
-00006590: 2c20 2d2d 6b73 7261 6e67 652c 2072 616e  , --ksrange, ran
-000065a0: 6765 206f 6620 4b73 2074 6f20 6265 2061  ge of Ks to be a
-000065b0: 6e61 6c79 7a65 642c 2064 6566 6175 6c74  nalyzed, default
-000065c0: 2028 302c 2035 290a 2d62 772c 202d 2d62   (0, 5).-bw, --b
-000065d0: 696e 5f77 6964 7468 2c20 6261 6e64 7769  in_width, bandwi
-000065e0: 6474 6820 6f66 204b 7320 6469 7374 7269  dth of Ks distri
-000065f0: 6275 7469 6f6e 2c20 6465 6661 756c 7420  bution, default 
-00006600: 302e 310a 2d69 632c 202d 2d77 6569 6768  0.1.-ic, --weigh
-00006610: 7473 5f6f 7574 6c69 6572 735f 696e 636c  ts_outliers_incl
-00006620: 7564 6564 2c20 666c 6167 206f 7074 696f  uded, flag optio
-00006630: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
-00006640: 636c 7564 6520 4b73 206f 7574 6c69 6572  clude Ks outlier
-00006650: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
-00006660: 6173 2073 6574 2c20 4b73 206f 7574 6c69  as set, Ks outli
-00006670: 6572 7320 776f 756c 6420 6265 2069 6e63  ers would be inc
-00006680: 6c75 6465 6420 696e 2074 6865 2061 6e61  luded in the ana
-00006690: 6c79 7369 730a 2d6d 2c20 2d2d 6d65 7468  lysis.-m, --meth
-000066a0: 6f64 2c20 7768 6963 6820 6d69 7874 7572  od, which mixtur
-000066b0: 6520 6d6f 6465 6c20 746f 2075 7365 2c20  e model to use, 
-000066c0: 6465 6661 756c 7420 676d 6d0a 2d2d 7365  default gmm.--se
-000066d0: 6564 2c20 7261 6e64 6f6d 2073 6565 6420  ed, random seed 
-000066e0: 6769 7665 6e20 746f 2069 6e69 7469 616c  given to initial
-000066f0: 697a 6174 696f 6e2c 2064 6566 6175 6c74  ization, default
-00006700: 2032 3335 3238 3930 0a2d 6569 2c20 2d2d   2352890.-ei, --
-00006710: 656d 5f69 7465 722c 2074 6865 206e 756d  em_iter, the num
-00006720: 6265 7220 6f66 2045 4d20 6974 6572 6174  ber of EM iterat
-00006730: 696f 6e73 2074 6f20 7065 7266 6f72 6d2c  ions to perform,
-00006740: 2064 6566 6175 6c74 2032 3030 0a2d 6e69   default 200.-ni
-00006750: 2c20 2d2d 6e5f 696e 6974 2c20 7468 6520  , --n_init, the 
-00006760: 6e75 6d62 6572 206f 6620 6b2d 6d65 616e  number of k-mean
-00006770: 7320 696e 6974 6961 6c69 7a61 7469 6f6e  s initialization
-00006780: 732c 2064 6566 6175 6c74 2032 3030 0a2d  s, default 200.-
-00006790: 6e2c 202d 2d63 6f6d 706f 6e65 6e74 732c  n, --components,
-000067a0: 2074 6865 2072 616e 6765 206f 6620 7468   the range of th
-000067b0: 6520 6e75 6d62 6572 206f 6620 636f 6d70  e number of comp
-000067c0: 6f6e 656e 7473 2074 6f20 6669 742c 2064  onents to fit, d
-000067d0: 6566 6175 6c74 2028 312c 2034 290a 2d67  efault (1, 4).-g
-000067e0: 2c20 2d2d 6761 6d6d 612c 2074 6865 2067  , --gamma, the g
-000067f0: 616d 6d61 2070 6172 616d 6574 6572 2066  amma parameter f
-00006800: 6f72 2062 676d 6d20 6d6f 6465 6c73 2c20  or bgmm models, 
-00006810: 6465 6661 756c 7420 3165 2d33 0a2d 2d62  default 1e-3.--b
-00006820: 6f6f 7473 2c20 7468 6520 6e75 6d62 6572  oots, the number
-00006830: 206f 6620 626f 6f74 7374 7261 7020 7265   of bootstrap re
-00006840: 706c 6963 6174 6573 206f 6620 6b64 652c  plicates of kde,
-00006850: 2064 6566 6175 6c74 2032 3030 0a2d 2d77   default 200.--w
-00006860: 6569 6768 7465 642c 2066 6c61 6720 6f70  eighted, flag op
-00006870: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00006880: 2075 7365 206e 6f64 652d 7765 6967 6874   use node-weight
-00006890: 6564 206d 6574 686f 6420 6f66 2064 652d  ed method of de-
-000068a0: 7265 6475 6e64 616e 6379 2c20 6966 2074  redundancy, if t
-000068b0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-000068c0: 2074 6865 206e 6f64 652d 7765 6967 6874   the node-weight
-000068d0: 6564 206d 6574 686f 6420 776f 756c 6420  ed method would 
-000068e0: 6265 2075 7365 640a 2d70 2c20 2d2d 706c  be used.-p, --pl
-000068f0: 6f74 2c20 7468 6520 706c 6f74 7469 6e67  ot, the plotting
-00006900: 206d 6574 686f 6420 746f 2062 6520 7573   method to be us
-00006910: 6564 2c20 6465 6661 756c 7420 6964 656e  ed, default iden
-00006920: 7469 6361 6c0a 2d62 6d2c 202d 2d62 775f  tical.-bm, --bw_
-00006930: 6d65 7468 6f64 2c20 7468 6520 6261 6e64  method, the band
-00006940: 7769 6474 6820 6d65 7468 6f64 2074 6f20  width method to 
-00006950: 6265 2075 7365 6420 696e 2061 6e61 6c79  be used in analy
-00006960: 7a69 6e67 2074 6865 2070 6561 6b20 6f66  zing the peak of
-00006970: 2057 4744 2064 6174 6573 2c20 6465 6661   WGD dates, defa
-00006980: 756c 7420 7369 6c76 6572 6d61 6e0a 2d2d  ult silverman.--
-00006990: 6e5f 6d65 646f 6964 732c 2074 6865 206e  n_medoids, the n
-000069a0: 756d 6265 7220 6f66 206d 6564 6f69 6473  umber of medoids
-000069b0: 2074 6f20 6669 742c 2064 6566 6175 6c74   to fit, default
-000069c0: 2032 0a2d 6b6d 2c20 2d2d 6b64 656d 6574   2.-km, --kdemet
-000069d0: 686f 642c 2074 6865 206b 6465 206d 6574  hod, the kde met
-000069e0: 686f 6420 746f 2062 6520 7573 6564 2069  hod to be used i
-000069f0: 6e20 616e 616c 797a 696e 6720 7468 6520  n analyzing the 
-00006a00: 7065 616b 206f 6620 5747 4420 6461 7465  peak of WGD date
-00006a10: 732c 206b 6d65 646f 6964 7320 616e 616c  s, kmedoids anal
-00006a20: 7973 6973 206f 7220 7468 6520 6261 7369  ysis or the basi
-00006a30: 6320 4b73 2070 6c6f 7474 696e 672c 2064  c Ks plotting, d
-00006a40: 6566 6175 6c74 2073 6369 7079 0a2d 2d6e  efault scipy.--n
-00006a50: 5f63 6c75 7374 6572 732c 2074 6865 206e  _clusters, the n
-00006a60: 756d 6265 7220 6f66 2063 6c75 7374 6572  umber of cluster
-00006a70: 7320 746f 2070 6c6f 7420 456c 626f 7720  s to plot Elbow 
-00006a80: 6c6f 7373 2066 756e 6374 696f 6e2c 2064  loss function, d
-00006a90: 6566 6175 6c74 2035 0a2d 6764 2c20 2d2d  efault 5.-gd, --
-00006aa0: 6775 6964 652c 2074 6865 2072 6567 696d  guide, the regim
-00006ab0: 6520 7265 7369 6469 6e67 2061 6e63 686f  e residing ancho
-00006ac0: 7273 2c20 6465 6661 756c 7420 5365 676d  rs, default Segm
-00006ad0: 656e 740a 2d70 7263 742c 202d 2d70 726f  ent.-prct, --pro
-00006ae0: 6d69 6e65 6e63 655f 6375 746f 6666 2c20  minence_cutoff, 
-00006af0: 7468 6520 7072 6f6d 696e 656e 6365 2063  the prominence c
-00006b00: 7574 6f66 6620 6f66 2061 6363 6570 7461  utoff of accepta
-00006b10: 626c 6520 7065 616b 7320 696e 2070 6561  ble peaks in pea
-00006b20: 6b20 6669 6e64 696e 6720 7374 6570 732c  k finding steps,
-00006b30: 2064 6566 6175 6c74 2030 2e31 0a2d 7268   default 0.1.-rh
-00006b40: 2c20 2d2d 7265 6c5f 6865 6967 6874 2c20  , --rel_height, 
-00006b50: 7468 6520 7265 6c61 7469 7665 2068 6569  the relative hei
-00006b60: 6768 7420 6174 2077 6869 6368 2074 6865  ght at which the
-00006b70: 2070 6561 6b20 7769 6474 6820 6973 206d   peak width is m
-00006b80: 6561 7375 7265 642c 2064 6566 6175 6c74  easured, default
-00006b90: 2030 2e34 0a2d 6b64 2c20 2d2d 6b73 746f   0.4.-kd, --ksto
-00006ba0: 6461 7465 2c20 7468 6520 7261 6e67 6520  date, the range 
-00006bb0: 6f66 204b 7320 746f 2062 6520 6461 7465  of Ks to be date
-00006bc0: 6420 696e 2068 6575 7269 7374 6963 2073  d in heuristic s
-00006bd0: 6561 7263 682c 2064 6566 6175 6c74 2028  earch, default (
-00006be0: 302e 352c 2031 2e35 290a 2d78 6c2c 202d  0.5, 1.5).-xl, -
-00006bf0: 2d78 6c69 6d2c 2074 6865 2078 2061 7869  -xlim, the x axi
-00006c00: 7320 6c69 6d69 7420 6f66 2047 4d4d 204b  s limit of GMM K
-00006c10: 7320 6469 7374 7269 6275 7469 6f6e 0a2d  s distribution.-
-00006c20: 796c 2c20 2d2d 796c 696d 2c20 7468 6520  yl, --ylim, the 
-00006c30: 7920 6178 6973 206c 696d 6974 206f 6620  y axis limit of 
-00006c40: 474d 4d20 4b73 2064 6973 7472 6962 7574  GMM Ks distribut
-00006c50: 696f 6e0a 2d2d 6d61 6e75 616c 7365 742c  ion.--manualset,
-00006c60: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00006c70: 6574 6865 7220 746f 206f 7574 7075 7420  ether to output 
-00006c80: 616e 6368 6f72 2070 6169 7273 2077 6974  anchor pairs wit
-00006c90: 6820 6d61 6e75 616c 6c79 2073 6574 204b  h manually set K
-00006ca0: 7320 7261 6e67 652c 2069 6620 7468 6520  s range, if the 
-00006cb0: 666c 6167 2077 6173 2073 6574 2c20 6d61  flag was set, ma
-00006cc0: 6e75 616c 6c79 2073 6574 204b 7320 7261  nually set Ks ra
-00006cd0: 6e67 6520 776f 756c 6420 6265 206f 7574  nge would be out
-00006ce0: 7075 7474 6564 0a2d 2d63 692c 2074 6865  putted.--ci, the
-00006cf0: 2063 6f6e 6669 6465 6e63 6520 6c65 7665   confidence leve
-00006d00: 6c20 6f66 206c 6f67 2d6e 6f72 6d61 6c20  l of log-normal 
-00006d10: 6469 7374 7269 6275 7469 6f6e 2074 6f20  distribution to 
-00006d20: 6461 7465 2c20 6465 6661 756c 7420 3935  date, default 95
-00006d30: 0a2d 2d68 6472 2c20 7468 6520 6869 6768  .--hdr, the high
-00006d40: 6573 7420 6465 6e73 6974 7920 7265 6769  est density regi
-00006d50: 6f6e 2028 4844 5229 2061 7070 6c69 6564  on (HDR) applied
-00006d60: 2069 6e20 7468 6520 7365 676d 656e 742d   in the segment-
-00006d70: 6775 6964 6564 2061 6e63 686f 7220 7061  guided anchor pa
-00006d80: 6972 204b 7320 6469 7374 7269 6275 7469  ir Ks distributi
-00006d90: 6f6e 2c20 6465 6661 756c 7420 3935 0a2d  on, default 95.-
-00006da0: 2d68 6575 7269 7374 6963 2c20 666c 6167  -heuristic, flag
-00006db0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-00006dc0: 2074 6f20 696e 6974 6961 7465 2068 6575   to initiate heu
-00006dd0: 7269 7374 6963 206d 6574 686f 6420 6f66  ristic method of
-00006de0: 2064 6566 696e 696e 6720 4349 2066 6f72   defining CI for
-00006df0: 2064 6174 696e 672c 2069 6620 7468 6520   dating, if the 
-00006e00: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
-00006e10: 6520 6865 7572 6973 7469 6320 6d65 7468  e heuristic meth
-00006e20: 6f64 2077 6f75 6c64 2062 6520 696e 6974  od would be init
-00006e30: 6961 7465 640a 2d6b 632c 202d 2d6b 7363  iated.-kc, --ksc
-00006e40: 7574 6f66 662c 2074 6865 204b 7320 7361  utoff, the Ks sa
-00006e50: 7475 7261 7469 6f6e 2063 7574 6f66 6620  turation cutoff 
-00006e60: 696e 2064 6174 696e 672c 2064 6566 6175  in dating, defau
-00006e70: 6c74 2035 0a2d 2d6b 6565 7074 6d70 6669  lt 5.--keeptmpfi
-00006e80: 672c 2066 6c61 6720 6f70 7469 6f6e 2c20  g, flag option, 
-00006e90: 7768 6574 6865 7220 746f 206b 6565 7020  whether to keep 
-00006ea0: 7465 6d70 6f72 6172 7920 6669 6775 7265  temporary figure
-00006eb0: 7320 696e 2070 6561 6b20 6669 6e64 696e  s in peak findin
-00006ec0: 6720 7072 6f63 6573 732c 2069 6620 7468  g process, if th
-00006ed0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00006ee0: 7468 6f73 6520 6669 6775 7265 7320 776f  those figures wo
-00006ef0: 756c 6420 6265 206b 6570 740a 6060 600a  uld be kept.```.
-00006f00: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
-00006f10: 6420 7379 6e60 2063 616e 2072 6561 6c69  d syn` can reali
-00006f20: 7a65 2074 6865 2069 6e74 7261 2d20 616e  ze the intra- an
-00006f30: 6420 696e 7465 722d 7370 6563 6966 6963  d inter-specific
-00006f40: 2073 796e 7465 6e79 2069 6e66 6572 656e   synteny inferen
-00006f50: 6365 2e0a 6060 600a 7767 6420 7379 6e20  ce..```.wgd syn 
-00006f60: 6661 6d69 6c69 6573 2067 6666 7320 286f  families gffs (o
-00006f70: 7074 696f 6e29 0a2d 2d2d 2d2d 2d2d 2d2d  ption).---------
-00006f80: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006f90: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006fa0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006fb0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006fc0: 2d2d 2d2d 2d2d 2d0a 2d6b 732c 202d 2d6b  -------.-ks, --k
-00006fd0: 735f 6469 7374 7269 6275 7469 6f6e 2c20  s_distribution, 
-00006fe0: 6b73 2064 6973 7472 6962 7574 696f 6e20  ks distribution 
-00006ff0: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
-00007000: 7420 4e6f 6e65 0a2d 6f2c 202d 2d6f 7574  t None.-o, --out
-00007010: 6469 722c 2074 6865 206f 7574 7075 7420  dir, the output 
-00007020: 6469 7265 6374 6f72 792c 2064 6566 6175  directory, defau
-00007030: 6c74 2077 6764 5f73 796e 0a2d 662c 202d  lt wgd_syn.-f, -
-00007040: 2d66 6561 7475 7265 2c20 7468 6520 6665  -feature, the fe
-00007050: 6174 7572 6520 666f 7220 7061 7273 696e  ature for parsin
-00007060: 6720 6765 6e65 2049 4473 2066 726f 6d20  g gene IDs from 
-00007070: 4746 4620 6669 6c65 732c 2064 6566 6175  GFF files, defau
-00007080: 6c74 2067 656e 650a 2d61 2c20 2d2d 6174  lt gene.-a, --at
-00007090: 7472 6962 7574 652c 2074 6865 2061 7474  tribute, the att
-000070a0: 7269 6275 7465 2066 6f72 2070 6172 7369  ribute for parsi
-000070b0: 6e67 2074 6865 2067 656e 6520 4944 7320  ng the gene IDs 
-000070c0: 6672 6f6d 2074 6865 2047 4646 2066 696c  from the GFF fil
-000070d0: 6573 2c20 6465 6661 756c 7420 4944 0a2d  es, default ID.-
-000070e0: 6174 672c 202d 2d61 6464 6974 696f 6e61  atg, --additiona
-000070f0: 6c67 6666 696e 666f 2c20 7468 6520 6665  lgffinfo, the fe
-00007100: 6174 7572 6520 616e 6420 6174 7472 6962  ature and attrib
-00007110: 7574 6520 696e 666f 726d 6174 696f 6e20  ute information 
-00007120: 6f66 2061 6464 6974 696f 6e61 6c20 6766  of additional gf
-00007130: 6633 2066 696c 6573 2069 6620 6469 6666  f3 files if diff
-00007140: 6572 656e 7420 696e 2074 6865 2066 6f72  erent in the for
-00007150: 6d61 7420 6f66 2028 6665 6174 7572 653b  mat of (feature;
-00007160: 6174 7472 6962 7574 6529 272c 2064 6566  attribute)', def
-00007170: 6175 6c74 204e 6f6e 650a 2d6d 6c2c 202d  ault None.-ml, -
-00007180: 2d6d 696e 6c65 6e2c 2074 6865 206d 696e  -minlen, the min
-00007190: 696d 756d 206c 656e 6774 6820 6f66 2061  imum length of a
-000071a0: 2073 6361 6666 6f6c 6420 746f 2062 6520   scaffold to be 
-000071b0: 696e 636c 7564 6564 2069 6e20 646f 7470  included in dotp
-000071c0: 6c6f 742c 2064 6566 6175 6c74 202d 312c  lot, default -1,
-000071d0: 2069 6620 2d31 2077 6173 2073 6574 2c20   if -1 was set, 
-000071e0: 7468 6520 3130 2520 6f66 2074 6865 206c  the 10% of the l
-000071f0: 6f6e 6765 7374 2073 6361 6666 6f6c 6420  ongest scaffold 
-00007200: 776f 756c 6420 6265 2073 6574 0a2d 6d73  would be set.-ms
-00007210: 2c20 2d2d 6d61 7873 697a 652c 2074 6865  , --maxsize, the
-00007220: 206d 6178 696d 756d 2066 616d 696c 7920   maximum family 
-00007230: 7369 7a65 2074 6f20 6265 2069 6e63 6c75  size to be inclu
-00007240: 6465 642c 2064 6566 6175 6c74 2032 3030  ded, default 200
-00007250: 0a2d 722c 202d 2d6b 735f 7261 6e67 652c  .-r, --ks_range,
-00007260: 2074 6865 204b 7320 7261 6e67 6520 696e   the Ks range in
-00007270: 2063 6f6c 6f72 6564 2064 6f74 706c 6f74   colored dotplot
-00007280: 2c20 6465 6661 756c 7420 2830 2c20 3529  , default (0, 5)
-00007290: 0a2d 2d69 6164 686f 7265 5f6f 7074 696f  .--iadhore_optio
-000072a0: 6e73 2c20 7468 6520 7061 7261 6d65 7465  ns, the paramete
-000072b0: 7220 7365 7474 696e 6720 696e 2069 6164  r setting in iad
-000072c0: 686f 7265 2c20 6465 6661 756c 7420 6173  hore, default as
-000072d0: 2061 2073 7472 696e 6720 6f66 206c 656e   a string of len
-000072e0: 6774 6820 7a65 726f 0a2d 6d67 2c20 2d2d  gth zero.-mg, --
-000072f0: 6d69 6e73 6567 6c65 6e2c 2074 6865 206d  minseglen, the m
-00007300: 696e 696d 756d 206c 656e 6774 6820 6f66  inimum length of
-00007310: 2073 6567 6d65 6e74 7320 746f 2069 6e63   segments to inc
-00007320: 6c75 6465 2069 6e20 7261 7469 6f20 6966  lude in ratio if
-00007330: 203c 3d20 312c 2064 6566 6175 6c74 2031   <= 1, default 1
-00007340: 3030 3030 0a2d 6b72 2c20 2d2d 6b65 6570  0000.-kr, --keep
-00007350: 7265 6475 6e2c 2066 6c61 6720 6f70 7469  redun, flag opti
-00007360: 6f6e 2c20 7768 6574 6865 7220 746f 206b  on, whether to k
-00007370: 6565 7020 7265 6475 6e64 616e 7420 6d75  eep redundant mu
-00007380: 6c74 6970 6c69 636f 6e73 2c20 6966 2074  ltiplicons, if t
-00007390: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-000073a0: 2074 6865 2072 6564 756e 6461 6e74 206d   the redundant m
-000073b0: 756c 7469 706c 6963 6f6e 7320 776f 756c  ultiplicons woul
-000073c0: 6420 6265 206b 6570 740a 2d6d 676e 2c20  d be kept.-mgn, 
-000073d0: 2d2d 6d69 6e67 656e 656e 756d 2c20 7468  --mingenenum, th
-000073e0: 6520 6d69 6e69 6d75 6d20 6e75 6d62 6572  e minimum number
-000073f0: 206f 6620 6765 6e65 7320 666f 7220 6120   of genes for a 
-00007400: 7365 676d 656e 7420 746f 2062 6520 636f  segment to be co
-00007410: 6e73 6964 6572 6564 2c20 6465 6661 756c  nsidered, defaul
-00007420: 7420 3330 0a2d 6473 2c20 2d2d 646f 7473  t 30.-ds, --dots
-00007430: 697a 652c 2074 6865 2064 6f74 2073 697a  ize, the dot siz
-00007440: 6520 696e 2064 6f74 2070 6c6f 742c 2064  e in dot plot, d
-00007450: 6566 6175 6c74 2030 2e33 0a2d 6161 2c20  efault 0.3.-aa, 
-00007460: 2d2d 6170 616c 7068 612c 2074 6865 206f  --apalpha, the o
-00007470: 7061 6369 7479 206f 6620 616e 6368 6f72  pacity of anchor
-00007480: 2064 6f74 732c 2064 6566 6175 6c74 2031   dots, default 1
-00007490: 0a2d 6861 2c20 2d2d 686f 616c 7068 612c  .-ha, --hoalpha,
-000074a0: 2074 6865 206f 7061 6369 7479 206f 6620   the opacity of 
-000074b0: 686f 6d6f 6c6f 6720 646f 7473 2c20 6465  homolog dots, de
-000074c0: 6661 756c 7420 300a 2d73 7274 2c20 2d2d  fault 0.-srt, --
-000074d0: 7368 6f77 7265 616c 7469 636b 2c20 666c  showrealtick, fl
-000074e0: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-000074f0: 6572 2074 6f20 7368 6f77 2074 6865 2072  er to show the r
-00007500: 6561 6c20 7469 636b 2069 6e20 6765 6e65  eal tick in gene
-00007510: 7320 6f72 2062 6173 6573 2c20 6966 2074  s or bases, if t
-00007520: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00007530: 2074 6865 2072 6561 6c20 7469 636b 2077   the real tick w
-00007540: 6f75 6c64 2062 6520 7368 6f77 6564 0a2d  ould be showed.-
-00007550: 746c 732c 202d 2d74 6963 6b6c 6162 656c  tls, --ticklabel
-00007560: 7369 7a65 2c20 7468 6520 6c61 6265 6c20  size, the label 
-00007570: 7369 7a65 206f 6620 7469 636b 2c20 6465  size of tick, de
-00007580: 6661 756c 7420 350a 2d67 722c 202d 2d67  fault 5.-gr, --g
-00007590: 6973 7472 622c 2066 6c61 6720 6f70 7469  istrb, flag opti
-000075a0: 6f6e 2c20 7768 6574 6865 7220 746f 2075  on, whether to u
-000075b0: 7365 2067 6973 745f 7261 696e 626f 7720  se gist_rainbow 
-000075c0: 6173 2063 6f6c 6f72 206d 6170 206f 6620  as color map of 
-000075d0: 646f 7470 6c6f 740a 2d6e 2c20 2d2d 6e74  dotplot.-n, --nt
-000075e0: 6872 6561 6473 2c20 7468 6520 6e75 6d62  hreads, the numb
-000075f0: 6572 206f 6620 7468 7265 6164 7320 746f  er of threads to
-00007600: 2075 7365 2069 6e20 7379 6e74 656e 7920   use in synteny 
-00007610: 696e 6665 7265 6e63 652c 2064 6566 6175  inference, defau
-00007620: 6c74 2034 0a60 6060 0a0a 5468 6520 7072  lt 4.```..The pr
-00007630: 6f67 7261 6d20 6077 6764 2076 697a 6020  ogram `wgd viz` 
-00007640: 6361 6e20 7265 616c 697a 6520 7468 6520  can realize the 
-00007650: 7669 7375 616c 697a 6174 696f 6e20 6f66  visualization of
-00007660: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00007670: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-00007680: 6e20 616e 6420 7379 6e74 656e 792e 0a60  n and synteny..`
-00007690: 6060 0a77 6764 2076 697a 2028 6f70 7469  ``.wgd viz (opti
-000076a0: 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  on).------------
-000076b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000076c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000076d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000076e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000076f0: 2d2d 2d2d 0a2d 642c 202d 2d64 6174 6166  ----.-d, --dataf
-00007700: 696c 652c 2074 6865 204b 7320 6461 7461  ile, the Ks data
-00007710: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
-00007720: 6e65 0a2d 6f2c 202d 2d6f 7574 6469 722c  ne.-o, --outdir,
-00007730: 2074 6865 206f 7574 7075 7420 6469 7265   the output dire
-00007740: 6374 6f72 792c 2064 6566 6175 6c74 2077  ctory, default w
-00007750: 6764 5f76 697a 0a2d 7372 2c20 2d2d 7370  gd_viz.-sr, --sp
-00007760: 6169 722c 2074 6865 2073 7065 6369 6573  air, the species
-00007770: 2070 6169 7220 746f 2062 6520 706c 6f74   pair to be plot
-00007780: 7465 642c 2064 6566 6175 6c74 204e 6f6e  ted, default Non
-00007790: 652c 2074 6869 7320 6f70 7469 6f6e 2063  e, this option c
-000077a0: 616e 2062 6520 7072 6f76 6964 6564 206d  an be provided m
-000077b0: 756c 7469 706c 6520 7469 6d65 730a 2d66  ultiple times.-f
-000077c0: 612c 202d 2d66 6f63 7573 3261 6c6c 2c20  a, --focus2all, 
-000077d0: 7365 7420 666f 6361 6c20 7370 6563 6965  set focal specie
-000077e0: 7320 616e 6420 6c65 7420 7370 6563 6965  s and let specie
-000077f0: 7320 7061 6972 2074 6f20 6265 2062 6574  s pair to be bet
-00007800: 7765 656e 2066 6f63 616c 2061 6e64 2061  ween focal and a
-00007810: 6c6c 2074 6865 2072 656d 6169 6e69 6e67  ll the remaining
-00007820: 2073 7065 6369 6573 2c20 6465 6661 756c   species, defaul
-00007830: 7420 4e6f 6e65 0a2d 6773 2c20 2d2d 6773  t None.-gs, --gs
-00007840: 6d61 702c 2074 6865 2067 656e 6520 6e61  map, the gene na
-00007850: 6d65 2d73 7065 6369 6573 206e 616d 6520  me-species name 
-00007860: 6d61 702c 2064 6566 6175 6c74 204e 6f6e  map, default Non
-00007870: 650a 2d73 702c 202d 2d73 7065 6369 6573  e.-sp, --species
-00007880: 7472 6565 2c20 7468 6520 7370 6563 6965  tree, the specie
-00007890: 7320 7472 6565 2074 6f20 7065 7266 6f72  s tree to perfor
-000078a0: 6d20 7261 7465 2063 6f72 7265 6374 696f  m rate correctio
-000078b0: 6e2c 2064 6566 6175 6c74 204e 6f6e 652c  n, default None,
-000078c0: 2069 6620 4e6f 6e65 2077 6173 2067 6976   if None was giv
-000078d0: 656e 2c20 7468 6520 7261 7465 2063 6f72  en, the rate cor
-000078e0: 7265 6374 696f 6e20 616e 616c 7973 6973  rection analysis
-000078f0: 2077 6f75 6c64 2062 6520 6361 6c6c 6564   would be called
-00007900: 206f 6666 0a2d 706b 2c20 2d2d 706c 6f74   off.-pk, --plot
-00007910: 6b64 652c 2066 6c61 6720 6f70 7469 6f6e  kde, flag option
-00007920: 2c20 7768 6574 6865 7220 746f 2070 6c6f  , whether to plo
-00007930: 7420 6b64 6520 6375 7276 6520 7570 6f6e  t kde curve upon
-00007940: 2068 6973 746f 6772 616d 2c20 6966 2074   histogram, if t
-00007950: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00007960: 206b 6465 2063 7572 7665 2077 6f75 6c64   kde curve would
-00007970: 2062 6520 6164 6465 640a 2d72 772c 202d   be added.-rw, -
-00007980: 2d72 6577 6569 6768 742c 2066 6c61 6720  -reweight, flag 
-00007990: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-000079a0: 746f 2072 6563 616c 6375 6c61 7465 2074  to recalculate t
-000079b0: 6865 2077 6569 6768 7420 7065 7220 7370  he weight per sp
-000079c0: 6563 6965 7320 7061 6972 2c20 6966 2074  ecies pair, if t
-000079d0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-000079e0: 2074 6865 2077 6569 6768 7420 776f 756c   the weight woul
-000079f0: 6420 6265 2072 6563 616c 6375 6c61 7465  d be recalculate
-00007a00: 640a 2d6f 722c 202d 2d6f 6e6c 7972 6f6f  d.-or, --onlyroo
-00007a10: 746f 7574 2c20 666c 6167 206f 7074 696f  tout, flag optio
-00007a20: 6e2c 2077 6865 7468 6572 2074 6f20 6f6e  n, whether to on
-00007a30: 6c79 2063 6f6e 6475 6374 2072 6174 6520  ly conduct rate 
-00007a40: 636f 7272 6563 7469 6f6e 2075 7369 6e67  correction using
-00007a50: 2074 6865 206f 7574 6772 6f75 7020 6174   the outgroup at
-00007a60: 2072 6f6f 7420 6173 206f 7574 6772 6f75   root as outgrou
-00007a70: 702c 2069 6620 7468 6520 666c 6167 2077  p, if the flag w
-00007a80: 6173 2073 6574 2c20 6f6e 6c79 2074 6865  as set, only the
-00007a90: 206f 7574 6772 6f75 7020 6174 2072 6f6f   outgroup at roo
-00007aa0: 7420 776f 756c 6420 6265 2075 7365 6420  t would be used 
-00007ab0: 6173 206f 7574 6772 6f75 700a 2d69 7465  as outgroup.-ite
-00007ac0: 722c 202d 2d65 6d5f 6974 6572 6174 696f  r, --em_iteratio
-00007ad0: 6e73 2c20 7468 6520 6d61 7869 6d75 6d20  ns, the maximum 
-00007ae0: 454d 2069 7465 7261 7469 6f6e 732c 2064  EM iterations, d
-00007af0: 6566 6175 6c74 2032 3030 0a2d 696e 6974  efault 200.-init
-00007b00: 2c20 2d2d 656d 5f69 6e69 7469 616c 697a  , --em_initializ
-00007b10: 6174 696f 6e73 2c20 7468 6520 6d61 7869  ations, the maxi
-00007b20: 6d75 6d20 454d 2069 6e69 7469 616c 697a  mum EM initializ
-00007b30: 6174 696f 6e73 2c20 6465 6661 756c 7420  ations, default 
-00007b40: 3230 300a 2d70 7263 742c 202d 2d70 726f  200.-prct, --pro
-00007b50: 6d69 6e65 6e63 655f 6375 746f 6666 2c20  minence_cutoff, 
-00007b60: 7468 6520 7072 6f6d 696e 656e 6365 2063  the prominence c
-00007b70: 7574 6f66 6620 6f66 2061 6363 6570 7461  utoff of accepta
-00007b80: 626c 6520 7065 616b 732c 2064 6566 6175  ble peaks, defau
-00007b90: 6c74 2030 2e31 0a2d 7268 2c20 2d2d 7265  lt 0.1.-rh, --re
-00007ba0: 6c5f 6865 6967 6874 2c20 7468 6520 7265  l_height, the re
-00007bb0: 6c61 7469 7665 2068 6569 6768 7420 6174  lative height at
-00007bc0: 2077 6869 6368 2074 6865 2070 6561 6b20   which the peak 
-00007bd0: 7769 6474 6820 6973 206d 6561 7375 7265  width is measure
-00007be0: 642c 2064 6566 6175 6c74 2030 2e34 0a2d  d, default 0.4.-
-00007bf0: 736d 2c20 2d2d 7365 676d 656e 7473 2c20  sm, --segments, 
-00007c00: 7468 6520 7365 676d 656e 7473 2064 6174  the segments dat
-00007c10: 6166 696c 652c 2064 6566 6175 6c74 204e  afile, default N
-00007c20: 6f6e 650a 2d6d 6c2c 202d 2d6d 696e 6c65  one.-ml, --minle
-00007c30: 6e2c 2074 6865 206d 696e 696d 756d 206c  n, the minimum l
-00007c40: 656e 6774 6820 6f66 2061 2073 6361 6666  ength of a scaff
-00007c50: 6f6c 6420 746f 2062 6520 696e 636c 7564  old to be includ
-00007c60: 6564 2069 6e20 646f 7470 6c6f 742c 2064  ed in dotplot, d
-00007c70: 6566 6175 6c74 202d 312c 2069 6620 2d31  efault -1, if -1
-00007c80: 2077 6173 2073 6574 2c20 7468 6520 3130   was set, the 10
-00007c90: 2520 6f66 2074 6865 206c 6f6e 6765 7374  % of the longest
-00007ca0: 2073 6361 6666 6f6c 6473 2077 696c 6c20   scaffolds will 
-00007cb0: 6265 2073 6574 0a2d 6d73 2c20 2d2d 6d61  be set.-ms, --ma
-00007cc0: 7873 697a 652c 2074 6865 206d 6178 696d  xsize, the maxim
-00007cd0: 756d 2066 616d 696c 7920 7369 7a65 2074  um family size t
-00007ce0: 6f20 6265 2069 6e63 6c75 6465 642c 2064  o be included, d
-00007cf0: 6566 6175 6c74 2032 3030 0a2d 6170 2c20  efault 200.-ap, 
-00007d00: 2d2d 616e 6368 6f72 706f 696e 7473 2c20  --anchorpoints, 
-00007d10: 7468 6520 616e 6368 6f72 2070 6f69 6e74  the anchor point
-00007d20: 7320 6461 7461 6669 6c65 2c20 6465 6661  s datafile, defa
-00007d30: 756c 7420 4e6f 6e65 0a2d 6d74 2c20 2d2d  ult None.-mt, --
-00007d40: 6d75 6c74 6970 6c69 636f 6e2c 2074 6865  multiplicon, the
-00007d50: 206d 756c 7469 706c 6963 6f6e 7320 6461   multiplicons da
-00007d60: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
-00007d70: 4e6f 6e65 0a2d 6774 2c20 2d2d 6765 6e65  None.-gt, --gene
-00007d80: 7461 626c 652c 2074 6865 2067 656e 6520  table, the gene 
-00007d90: 7461 626c 6520 6461 7461 6669 6c65 2c20  table datafile, 
-00007da0: 6465 6661 756c 7420 4e6f 6e65 0a2d 6d67  default None.-mg
-00007db0: 2c20 2d2d 6d69 6e73 6567 6c65 6e2c 2074  , --minseglen, t
-00007dc0: 6865 206d 696e 696d 756d 206c 656e 6774  he minimum lengt
-00007dd0: 6820 6f66 2073 6567 6d65 6e74 7320 746f  h of segments to
-00007de0: 2069 6e63 6c75 6465 2c20 696e 2072 6174   include, in rat
-00007df0: 696f 2069 6620 3c3d 2031 2c20 6465 6661  io if <= 1, defa
-00007e00: 756c 7420 3130 3030 300a 2d6d 676e 2c20  ult 10000.-mgn, 
-00007e10: 2d2d 6d69 6e67 656e 656e 756d 2c20 7468  --mingenenum, th
-00007e20: 6520 6d69 6e69 6d75 6d20 6e75 6d62 6572  e minimum number
-00007e30: 206f 6620 6765 6e65 7320 666f 7220 6120   of genes for a 
-00007e40: 7365 676d 656e 7420 746f 2062 6520 636f  segment to be co
-00007e50: 6e73 6964 6572 6564 2c20 6465 6661 756c  nsidered, defaul
-00007e60: 7420 3330 0a2d 6b72 2c20 2d2d 6b65 6570  t 30.-kr, --keep
-00007e70: 7265 6475 6e2c 2066 6c61 6720 6f70 7469  redun, flag opti
-00007e80: 6f6e 2c20 7768 6574 6865 7220 746f 206b  on, whether to k
-00007e90: 6565 7020 7265 6475 6e64 616e 7420 6d75  eep redundant mu
-00007ea0: 6c74 6970 6c69 636f 6e73 2c20 6966 2074  ltiplicons, if t
-00007eb0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00007ec0: 2074 6865 2072 6564 756e 6461 6e74 206d   the redundant m
-00007ed0: 756c 7469 706c 6963 6f6e 7320 776f 756c  ultiplicons woul
-00007ee0: 6420 6265 206b 6570 740a 2d65 706b 2c20  d be kept.-epk, 
-00007ef0: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
-00007f00: 732c 2065 7874 7261 2070 6172 616e 6f6d  s, extra paranom
-00007f10: 6520 4b73 2064 6174 6120 746f 2070 6c6f  e Ks data to plo
-00007f20: 7420 696e 2074 6865 206d 6978 6564 204b  t in the mixed K
-00007f30: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
-00007f40: 6465 6661 756c 7420 4e6f 6e65 0a2d 7061  default None.-pa
-00007f50: 672c 202d 2d70 6c6f 7461 7067 6d6d 2c20  g, --plotapgmm, 
-00007f60: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-00007f70: 7468 6572 2074 6f20 636f 6e64 7563 7420  ther to conduct 
-00007f80: 616e 6420 706c 6f74 206d 6978 7475 7265  and plot mixture
-00007f90: 206d 6f64 656c 696e 6720 6f66 2061 6e63   modeling of anc
-00007fa0: 686f 7220 4b73 2069 6e20 7468 6520 6d69  hor Ks in the mi
-00007fb0: 7865 6420 4b73 2064 6973 7472 6962 7574  xed Ks distribut
-00007fc0: 696f 6e2c 2069 6620 7468 6520 666c 6167  ion, if the flag
-00007fd0: 2077 6173 2073 6574 2c20 7468 6520 6d69   was set, the mi
-00007fe0: 7874 7572 6520 6d6f 6465 6c69 6e67 206f  xture modeling o
-00007ff0: 6620 616e 6368 6f72 204b 7320 776f 756c  f anchor Ks woul
-00008000: 6420 6265 2063 6f6e 6475 6374 6564 2061  d be conducted a
-00008010: 6e64 2070 6c6f 7474 6564 0a2d 7065 6d2c  nd plotted.-pem,
-00008020: 202d 2d70 6c6f 7465 6c6d 6d2c 2066 6c61   --plotelmm, fla
-00008030: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00008040: 7220 746f 2063 6f6e 6475 6374 2061 6e64  r to conduct and
-00008050: 2070 6c6f 7420 656c 6d6d 206d 6978 7475   plot elmm mixtu
-00008060: 7265 206d 6f64 656c 696e 6720 6f66 2070  re modeling of p
-00008070: 6172 616e 6f6d 6520 4b73 2069 6e20 7468  aranome Ks in th
-00008080: 6520 6d69 7865 6420 4b73 2064 6973 7472  e mixed Ks distr
-00008090: 6962 7574 696f 6e2c 2069 6620 7468 6520  ibution, if the 
-000080a0: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
-000080b0: 6520 656c 6d6d 206d 6978 7475 7265 206d  e elmm mixture m
-000080c0: 6f64 656c 696e 6720 6f66 2070 6172 616e  odeling of paran
-000080d0: 6f6d 6520 4b73 2077 6f75 6c64 2062 6520  ome Ks would be 
-000080e0: 636f 6e64 7563 7465 6420 616e 6420 706c  conducted and pl
-000080f0: 6f74 7465 640a 2d63 2c20 2d2d 636f 6d70  otted.-c, --comp
-00008100: 6f6e 656e 7473 2c20 7468 6520 7261 6e67  onents, the rang
-00008110: 6520 6f66 2074 6865 206e 756d 6265 7220  e of the number 
-00008120: 6f66 2063 6f6d 706f 6e65 6e74 7320 746f  of components to
-00008130: 2066 6974 2069 6e20 616e 6368 6f72 204b   fit in anchor K
-00008140: 7320 6d69 7874 7572 6520 6d6f 6465 6c69  s mixture modeli
-00008150: 6e67 2c20 6465 6661 756c 7420 2831 2c34  ng, default (1,4
-00008160: 290a 2d70 7379 2c20 2d2d 706c 6f74 7379  ).-psy, --plotsy
-00008170: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
-00008180: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
-00008190: 6174 6520 7468 6520 7379 6e74 656e 7920  ate the synteny 
-000081a0: 706c 6f74 2c20 6f6e 6c79 2077 6865 6e20  plot, only when 
-000081b0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-000081c0: 2c20 7468 6520 7379 6e74 656e 7920 706c  , the synteny pl
-000081d0: 6f74 2077 6f75 6c64 2062 6520 7072 6f64  ot would be prod
-000081e0: 7563 6564 0a2d 6473 2c20 2d2d 646f 7473  uced.-ds, --dots
-000081f0: 697a 652c 2074 6865 2064 6f74 2073 697a  ize, the dot siz
-00008200: 6520 696e 2064 6f74 2070 6c6f 742c 2064  e in dot plot, d
-00008210: 6566 6175 6c74 2030 2e33 0a2d 6161 2c20  efault 0.3.-aa, 
-00008220: 2d2d 6170 616c 7068 612c 2074 6865 206f  --apalpha, the o
-00008230: 7061 6369 7479 206f 6620 616e 6368 6f72  pacity of anchor
-00008240: 2064 6f74 732c 2064 6566 6175 6c74 2031   dots, default 1
-00008250: 0a2d 6861 2c20 2d2d 686f 616c 7068 612c  .-ha, --hoalpha,
-00008260: 2074 6865 206f 7061 6369 7479 206f 6620   the opacity of 
-00008270: 686f 6d6f 6c6f 6720 646f 7473 2c20 6465  homolog dots, de
-00008280: 6661 756c 7420 300a 2d73 7274 2c20 2d2d  fault 0.-srt, --
-00008290: 7368 6f77 7265 616c 7469 636b 2c20 666c  showrealtick, fl
-000082a0: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-000082b0: 6572 2074 6f20 7368 6f77 2074 6865 2072  er to show the r
-000082c0: 6561 6c20 7469 636b 2069 6e20 6765 6e65  eal tick in gene
-000082d0: 7320 6f72 2062 6173 6573 2c20 6966 2074  s or bases, if t
-000082e0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-000082f0: 2074 6865 2072 6561 6c20 7469 636b 2077   the real tick w
-00008300: 6f75 6c64 2062 6520 7368 6f77 6564 0a2d  ould be showed.-
-00008310: 746c 732c 202d 2d74 6963 6b6c 6162 656c  tls, --ticklabel
-00008320: 7369 7a65 2c20 7468 6520 6c61 6265 6c20  size, the label 
-00008330: 7369 7a65 206f 6620 7469 636b 2c20 6465  size of tick, de
-00008340: 6661 756c 7420 350a 2d78 6c2c 202d 2d78  fault 5.-xl, --x
-00008350: 6c69 6d2c 2074 6865 2078 2061 7869 7320  lim, the x axis 
-00008360: 6c69 6d69 7420 6f66 204b 7320 6469 7374  limit of Ks dist
-00008370: 7269 6275 7469 6f6e 0a2d 796c 2c20 2d2d  ribution.-yl, --
-00008380: 796c 696d 2c20 7468 6520 7920 6178 6973  ylim, the y axis
-00008390: 206c 696d 6974 206f 6620 4b73 2064 6973   limit of Ks dis
-000083a0: 7472 6962 7574 696f 6e0a 2d61 646f 2c20  tribution.-ado, 
-000083b0: 2d2d 6164 6a75 7374 6f72 7468 6f2c 2066  --adjustortho, f
-000083c0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-000083d0: 6865 7220 746f 2061 646a 7573 7420 7468  her to adjust th
-000083e0: 6520 6869 7374 6f67 7261 6d20 6865 6967  e histogram heig
-000083f0: 6874 206f 6620 6f72 7468 6f6c 6f67 6f75  ht of orthologou
-00008400: 7320 4b73 2061 7320 746f 206d 6174 6368  s Ks as to match
-00008410: 2074 6865 2068 6569 6768 7420 6f66 2070   the height of p
-00008420: 6172 616c 6f67 6f75 7320 4b73 2c20 6966  aralogous Ks, if
-00008430: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00008440: 742c 2074 6865 2061 646a 7573 746d 656e  t, the adjustmen
-00008450: 7420 776f 756c 6420 6265 2063 6f6e 6475  t would be condu
-00008460: 6374 6564 0a2d 6164 662c 202d 2d61 646a  cted.-adf, --adj
-00008470: 7573 7466 6163 746f 722c 2074 6865 2061  ustfactor, the a
-00008480: 646a 7573 746d 656e 7420 6661 6374 6f72  djustment factor
-00008490: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
-000084a0: 4b73 2c20 6465 6661 756c 7420 302e 350a  Ks, default 0.5.
-000084b0: 2d6f 612c 202d 2d6f 6b61 6c70 6861 2c20  -oa, --okalpha, 
-000084c0: 7468 6520 6f70 6163 6974 7920 6f66 206f  the opacity of o
-000084d0: 7274 686f 6c6f 676f 7573 204b 7320 6469  rthologous Ks di
-000084e0: 7374 7269 6275 7469 6f6e 2069 6e20 6d69  stribution in mi
-000084f0: 7865 6420 706c 6f74 2c20 6465 6661 756c  xed plot, defaul
-00008500: 7420 302e 350a 2d63 732c 202d 2d63 6c61  t 0.5.-cs, --cla
-00008510: 7373 6963 2c20 666c 6167 206f 7074 696f  ssic, flag optio
-00008520: 6e2c 2077 6865 7468 6572 2074 6f20 6472  n, whether to dr
-00008530: 6177 206d 6978 6564 204b 7320 706c 6f74  aw mixed Ks plot
-00008540: 2069 6e20 6120 636c 6173 7369 6320 6d61   in a classic ma
-00008550: 6e6e 6572 2077 6865 7265 2074 6865 2066  nner where the f
-00008560: 756c 6c20 6f72 7468 6f6c 6f67 6f75 7320  ull orthologous 
-00008570: 4b73 2064 6973 7472 6962 7574 696f 6e20  Ks distribution 
-00008580: 6973 2064 7261 7765 642c 2069 6620 7468  is drawed, if th
-00008590: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-000085a0: 7468 6520 636c 6173 7369 6320 6d69 7865  the classic mixe
-000085b0: 6420 4b73 2070 6c6f 7420 776f 756c 6420  d Ks plot would 
-000085c0: 6265 2064 7261 776e 0a2d 7461 2c20 2d2d  be drawn.-ta, --
-000085d0: 746f 7061 7272 6f77 2c20 666c 6167 206f  toparrow, flag o
-000085e0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000085f0: 6f20 6164 6a75 7374 2074 6865 2061 7272  o adjust the arr
-00008600: 6f77 2074 6f20 6265 2061 7420 7468 6520  ow to be at the 
-00008610: 746f 7020 6f66 2074 6865 2070 6c6f 742c  top of the plot,
-00008620: 2069 6e73 7465 6164 206f 6620 6265 696e   instead of bein
-00008630: 6720 636f 6f72 6469 6e61 7465 6420 6173  g coordinated as
-00008640: 2074 6865 204b 4445 206f 6620 7468 6520   the KDE of the 
-00008650: 6f72 7468 6f6c 6f67 6f75 7320 4b73 2064  orthologous Ks d
-00008660: 6973 7472 6962 7574 696f 6e2c 2069 6620  istribution, if 
-00008670: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00008680: 2c20 7468 6520 6172 726f 7720 776f 756c  , the arrow woul
-00008690: 6420 6265 2073 6574 2061 7420 7468 6520  d be set at the 
-000086a0: 746f 700a 2d6e 612c 202d 2d6e 6f64 6561  top.-na, --nodea
-000086b0: 7665 7261 6765 642c 2066 6c61 6720 6f70  veraged, flag op
-000086c0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-000086d0: 2075 7365 206e 6f64 652d 6176 6572 6167   use node-averag
-000086e0: 6564 206d 6574 686f 6420 666f 7220 6465  ed method for de
-000086f0: 2d72 6564 756e 6461 6e63 792c 2069 6620  -redundancy, if 
-00008700: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00008710: 2c20 7468 6520 6e6f 6465 2d61 7665 7261  , the node-avera
-00008720: 6765 6420 6d65 7468 6f64 2077 6f75 6c64  ged method would
-00008730: 2062 6520 696e 6974 6961 7465 640a 2d62   be initiated.-b
-00008740: 732c 202d 2d62 6f6f 7473 7472 6170 2c20  s, --bootstrap, 
-00008750: 7468 6520 6e75 6d62 6572 206f 6620 626f  the number of bo
-00008760: 6f74 7374 7261 7020 7265 706c 6963 6174  otstrap replicat
-00008770: 6573 206f 6620 6f72 7468 6f6c 6f67 204b  es of ortholog K
-00008780: 7320 6469 7374 7269 6275 7469 6f6e 2069  s distribution i
-00008790: 6e20 6d69 7865 6420 706c 6f74 0a2d 6772  n mixed plot.-gr
-000087a0: 2c20 2d2d 6769 7374 7262 2c20 666c 6167  , --gistrb, flag
-000087b0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-000087c0: 2074 6f20 7573 6520 6769 7374 5f72 6169   to use gist_rai
-000087d0: 6e62 6f77 2061 7320 636f 6c6f 7220 6d61  nbow as color ma
-000087e0: 7020 6f66 2064 6f74 706c 6f74 0a2d 6e2c  p of dotplot.-n,
-000087f0: 202d 2d6e 7468 7265 6164 732c 2074 6865   --nthreads, the
-00008800: 206e 756d 6265 7220 6f66 2074 6872 6561   number of threa
-00008810: 6473 2074 6f20 7573 6520 696e 2062 6f6f  ds to use in boo
-00008820: 7473 7472 6170 2073 616d 706c 696e 672c  tstrap sampling,
-00008830: 2064 6566 6175 6c74 2031 0a60 6060 0a0a   default 1.```..
-00008840: 2323 2055 7361 6765 0a0a 4865 7265 2077  ## Usage..Here w
-00008850: 6520 7072 6f76 6964 6564 2074 6865 2062  e provided the b
-00008860: 6173 6963 2075 7361 6765 2066 6f72 2065  asic usage for e
-00008870: 6163 6820 7072 6f67 7261 6d20 616e 6420  ach program and 
-00008880: 7468 6520 7265 6c65 7661 6e74 2070 6172  the relevant par
-00008890: 616d 6574 6572 7320 616e 6420 7375 6767  ameters and sugg
-000088a0: 6573 7469 6f6e 7320 6f6e 2070 6172 616d  estions on param
-000088b0: 6574 6572 697a 6174 696f 6e2e 2041 2072  eterization. A r
-000088c0: 656d 696e 6465 7220 7468 6174 2074 6865  eminder that the
-000088d0: 2067 6976 656e 2063 6f72 6573 2061 6e64   given cores and
-000088e0: 2074 6872 6561 6473 2063 616e 2073 6967   threads can sig
-000088f0: 6e69 6669 6361 6e74 6c79 2069 6d70 6163  nificantly impac
-00008900: 7420 7468 6520 7275 6e20 7469 6d65 2061  t the run time a
-00008910: 6e64 2074 6875 7320 7765 2061 6464 6564  nd thus we added
-00008920: 2073 6f6d 6520 7265 706f 7274 2069 6e66   some report inf
-00008930: 6f72 6d61 7469 6f6e 2070 6572 7461 696e  ormation pertain
-00008940: 696e 6720 746f 2074 6865 2073 7973 7465  ing to the syste
-00008950: 6d20 6f66 2075 7365 7273 2074 6f20 6661  m of users to fa
-00008960: 6369 6c69 7461 7465 2074 6865 2065 6666  cilitate the eff
-00008970: 6963 6965 6e74 2073 6574 7469 6e67 206f  icient setting o
-00008980: 6620 7468 7265 6164 7320 616e 6420 6d65  f threads and me
-00008990: 6d6f 7279 2e20 5468 6520 6c6f 6769 6361  mory. The logica
-000089a0: 6c20 4350 5573 2072 6570 6f72 7465 6420  l CPUs reported 
-000089b0: 7265 7072 6573 656e 7473 2074 6865 206e  represents the n
-000089c0: 756d 6265 7220 6f66 2070 6879 7369 6361  umber of physica
-000089d0: 6c20 636f 7265 7320 6d75 6c74 6970 6c69  l cores multipli
-000089e0: 6564 2062 7920 7468 6520 6e75 6d62 6572  ed by the number
-000089f0: 206f 6620 7468 7265 6164 7320 7468 6174   of threads that
-00008a00: 2063 616e 2072 756e 206f 6e20 6561 6368   can run on each
-00008a10: 2063 6f72 652c 2061 6c73 6f20 6b6e 6f77   core, also know
-00008a20: 6e20 6173 2048 7970 6572 2054 6872 6561  n as Hyper Threa
-00008a30: 6469 6e67 2e20 5468 6520 6e75 6d62 6572  ding. The number
-00008a40: 206f 6620 6c6f 6769 6361 6c20 4350 5573   of logical CPUs
-00008a50: 206d 6179 206e 6f74 206e 6563 6573 7361   may not necessa
-00008a60: 7269 6c79 2062 6520 6571 7569 7661 6c65  rily be equivale
-00008a70: 6e74 2074 6f20 7468 6520 6163 7475 616c  nt to the actual
-00008a80: 206e 756d 6265 7220 6f66 2043 5055 7320   number of CPUs 
-00008a90: 7468 6520 6375 7272 656e 7420 7072 6f63  the current proc
-00008aa0: 6573 7320 6361 6e20 7573 652e 2054 6865  ess can use. The
-00008ab0: 2061 7661 696c 6162 6c65 206d 656d 6f72   available memor
-00008ac0: 7920 7265 6665 7273 2074 6f20 7468 6520  y refers to the 
-00008ad0: 6d65 6d6f 7279 2074 6861 7420 6361 6e20  memory that can 
-00008ae0: 6265 2067 6976 656e 2069 6e73 7461 6e74  be given instant
-00008af0: 6c79 2074 6f20 7072 6f63 6573 7365 7320  ly to processes 
-00008b00: 7769 7468 6f75 7420 7468 6520 7379 7374  without the syst
-00008b10: 656d 2067 6f69 6e67 2069 6e74 6f20 7377  em going into sw
-00008b20: 6170 2061 6e64 2072 6566 6c65 6374 7320  ap and reflects 
-00008b30: 7468 6520 6163 7475 616c 206d 656d 6f72  the actual memor
-00008b40: 7920 6176 6169 6c61 626c 652e 2054 6865  y available. The
-00008b50: 2066 7265 6520 6d65 6d6f 7279 2072 6566   free memory ref
-00008b60: 6572 7320 746f 2074 6865 206d 656d 6f72  ers to the memor
-00008b70: 7920 6e6f 7420 6265 696e 6720 7573 6564  y not being used
-00008b80: 2061 7420 616c 6c20 287a 6572 6f65 6429   at all (zeroed)
-00008b90: 2074 6861 7420 6973 2072 6561 6469 6c79   that is readily
-00008ba0: 2061 7661 696c 6162 6c65 2e20 5468 6520   available. The 
-00008bb0: 6465 7363 7269 7074 696f 6e20 6162 6f76  description abov
-00008bc0: 6520 7265 6665 7273 2074 6f20 7468 6520  e refers to the 
-00008bd0: 646f 6375 6d65 6e74 6174 696f 6e20 6f66  documentation of
-00008be0: 2060 7073 7574 696c 602e 0a0a 2323 2320   `psutil`...### 
-00008bf0: 7767 6420 646d 640a 0a2a 2a54 6865 2064  wgd dmd..**The d
-00008c00: 656c 696e 6561 7469 6f6e 206f 6620 7768  elineation of wh
-00008c10: 6f6c 6520 7061 7261 6e6f 6d65 2a2a 0a60  ole paranome**.`
-00008c20: 6060 0a77 6764 2064 6d64 2041 7175 696c  ``.wgd dmd Aquil
-00008c30: 6567 6961 5f63 6f65 7275 6c65 6120 2d49  egia_coerulea -I
-00008c40: 2032 202d 6520 3165 2d31 3020 2d62 7320   2 -e 1e-10 -bs 
-00008c50: 3130 3020 2d6e 7020 3520 282d 6e6e 2920  100 -np 5 (-nn) 
-00008c60: 282d 2d74 6f5f 7374 6f70 2920 282d 2d63  (--to_stop) (--c
-00008c70: 6473 2920 282d 6e20 3429 2028 2d6f 2077  ds) (-n 4) (-o w
-00008c80: 6764 5f64 6d64 2920 282d 7420 776f 726b  gd_dmd) (-t work
-00008c90: 696e 675f 746d 7029 0a60 6060 200a 0a4e  ing_tmp).``` ..N
-00008ca0: 6f74 6520 7468 6174 2077 6520 646f 6e27  ote that we don'
-00008cb0: 7420 7072 6f76 6964 6520 7468 6520 6461  t provide the da
-00008cc0: 7461 206f 6620 7468 6973 2063 6f64 696e  ta of this codin
-00008cd0: 6720 7365 7175 656e 6365 2028 6364 7329  g sequence (cds)
-00008ce0: 2066 696c 6520 6041 7175 696c 6567 6961   file `Aquilegia
-00008cf0: 5f63 6f65 7275 6c65 6160 2062 7574 2069  _coerulea` but i
-00008d00: 7420 6361 6e20 6265 2065 6173 696c 7920  t can be easily 
-00008d10: 646f 776e 6c6f 6164 6564 2061 7420 5b50  downloaded at [P
-00008d20: 6879 746f 7a6f 6d65 5d28 6874 7470 733a  hytozome](https:
-00008d30: 2f2f 7068 7974 6f7a 6f6d 652d 6e65 7874  //phytozome-next
-00008d40: 2e6a 6769 2e64 6f65 2e67 6f76 2f69 6e66  .jgi.doe.gov/inf
-00008d50: 6f2f 4163 6f65 7275 6c65 615f 7633 5f31  o/Acoerulea_v3_1
-00008d60: 2920 2873 616d 6520 666f 7220 6f74 6865  ) (same for othe
-00008d70: 7220 5573 6167 6520 646f 6329 2e20 4120  r Usage doc). A 
-00008d80: 7265 6d69 6e64 6572 2074 6861 7420 696e  reminder that in
-00008d90: 2074 6865 2069 7373 7565 7320 736f 6d65   the issues some
-00008da0: 2075 7365 7273 2064 6964 6e27 7420 646f   users didn't do
-00008db0: 776e 6c6f 6164 2074 6865 2060 4163 6f65  wnload the `Acoe
-00008dc0: 7275 6c65 615f 3332 325f 7633 2e31 2e63  rulea_322_v3.1.c
-00008dd0: 6473 5f70 7269 6d61 7279 5472 616e 7363  ds_primaryTransc
-00008de0: 7269 7074 4f6e 6c79 2e66 612e 677a 6020  riptOnly.fa.gz` 
-00008df0: 6275 7420 696e 7374 6561 6420 7468 6520  but instead the 
-00008e00: 6041 636f 6572 756c 6561 5f33 3232 5f76  `Acoerulea_322_v
-00008e10: 332e 312e 6364 732e 6661 2e67 7a60 2066  3.1.cds.fa.gz` f
-00008e20: 696c 652e 2046 6f72 2074 6865 2063 6f6e  ile. For the con
-00008e30: 7374 7275 6374 696f 6e20 6f66 2077 686f  struction of who
-00008e40: 6c65 2070 6172 616e 6f6d 6520 2a4b 2a3c  le paranome *K*<
-00008e50: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00008e60: 7269 6275 7469 6f6e 2c20 2a2a 6f6e 6c79  ribution, **only
-00008e70: 206f 6e65 2074 7261 6e73 6372 6970 7420   one transcript 
-00008e80: 2874 6865 2070 7269 6d61 7279 206f 6e65  (the primary one
-00008e90: 2920 7065 7220 6765 6e65 2073 686f 756c  ) per gene shoul
-00008ea0: 6420 6265 2069 6e63 6c75 6465 642a 2a20  d be included** 
-00008eb0: 7375 6368 2074 6861 7420 7468 6520 2a4b  such that the *K
-00008ec0: 2a3c 7375 623e 533c 2f73 7562 3e20 6973  *<sub>S</sub> is
-00008ed0: 2072 6561 6c6c 7920 696e 6469 6361 7469   really indicati
-00008ee0: 6e67 2074 6865 2061 6765 206f 6620 6765  ng the age of ge
-00008ef0: 6e65 2064 7570 6c69 6361 7469 6f6e 2065  ne duplication e
-00008f00: 7665 6e74 2c20 696e 7374 6561 6420 6f66  vent, instead of
-00008f10: 2061 6c74 6572 6e61 7469 7665 2073 706c   alternative spl
-00008f20: 6963 696e 672e 2054 7261 6e73 6372 6970  icing. Transcrip
-00008f30: 746f 6d65 2064 6174 6120 7368 6f75 6c64  tome data should
-00008f40: 2062 6520 6361 7265 6675 6c6c 7920 7472   be carefully tr
-00008f50: 6561 7465 6420 7769 7468 2064 652d 7265  eated with de-re
-00008f60: 6475 6e64 616e 6379 2073 6f20 6173 2074  dundancy so as t
-00008f70: 6f20 7265 6475 6365 2074 6865 2066 616c  o reduce the fal
-00008f80: 7365 2070 6f73 6974 6976 6520 6475 706c  se positive dupl
-00008f90: 6963 6174 696f 6e20 6275 6d70 2063 6175  ication bump cau
-00008fa0: 7365 6420 6279 2070 6572 7661 7369 7665  sed by pervasive
-00008fb0: 2061 6c74 6572 6e61 7469 7665 2073 706c   alternative spl
-00008fc0: 6963 696e 672e 2048 6572 6520 7468 6520  icing. Here the 
-00008fd0: 696e 666c 6174 696f 6e20 6661 6374 6f72  inflation factor
-00008fe0: 2070 6172 616d 6574 6572 2c20 6769 7665   parameter, give
-00008ff0: 6e20 6279 2060 2d49 6020 6f72 2060 2d2d  n by `-I` or `--
-00009000: 696e 666c 6174 696f 6e60 2c20 6166 6665  inflation`, affe
-00009010: 6374 7320 7468 6520 6772 616e 756c 6172  cts the granular
-00009020: 6974 7920 6f72 2072 6573 6f6c 7574 696f  ity or resolutio
-00009030: 6e20 6f66 2074 6865 2063 6c75 7374 6572  n of the cluster
-00009040: 696e 6720 6f75 7463 6f6d 6520 616e 6420  ing outcome and 
-00009050: 696d 706c 6963 6974 6c79 2063 6f6e 7472  implicitly contr
-00009060: 6f6c 6c73 2074 6865 206e 756d 6265 7220  olls the number 
-00009070: 6f66 2063 6c75 7374 6572 732c 2077 6974  of clusters, wit
-00009080: 6820 6c6f 7720 7661 6c75 6573 2073 7563  h low values suc
-00009090: 6820 6173 2031 2e33 206f 7220 312e 3420  h as 1.3 or 1.4 
-000090a0: 6c65 6164 696e 6720 746f 2066 6577 6572  leading to fewer
-000090b0: 2062 7574 206c 6172 6765 7220 636c 7573   but larger clus
-000090c0: 7465 7273 2061 6e64 2068 6967 6820 7661  ters and high va
-000090d0: 6c75 6573 2073 7563 6820 6173 2035 206f  lues such as 5 o
-000090e0: 7220 3620 6c65 6164 696e 6720 746f 206d  r 6 leading to m
-000090f0: 6f72 6520 6275 7420 736d 616c 6c65 7220  ore but smaller 
-00009100: 636c 7573 7465 7273 2e20 5765 2073 6574  clusters. We set
-00009110: 2074 6865 2064 6566 6175 6c74 2076 616c   the default val
-00009120: 7565 2061 7320 3220 6173 2073 7567 6765  ue as 2 as sugge
-00009130: 7374 6564 2062 7920 5b4d 434c 5d28 6874  sted by [MCL](ht
-00009140: 7470 733a 2f2f 6d69 6361 6e73 2e6f 7267  tps://micans.org
-00009150: 2f6d 636c 2f29 2e20 5468 6520 652d 7661  /mcl/). The e-va
-00009160: 6c75 6520 6375 742d 6f66 6620 666f 7220  lue cut-off for 
-00009170: 7365 7175 656e 6365 2073 696d 696c 6172  sequence similar
-00009180: 6974 792c 2067 6976 656e 2062 7920 602d  ity, given by `-
-00009190: 6560 206f 7220 602d 2d65 7661 6c60 2c20  e` or `--eval`, 
-000091a0: 7768 6963 6820 6465 6e6f 7465 7320 7468  which denotes th
-000091b0: 6520 6578 7065 6374 6564 2076 616c 7565  e expected value
-000091c0: 206f 6620 7468 6520 6869 7420 7175 616e   of the hit quan
-000091d0: 7469 6669 6573 2074 6865 206e 756d 6265  tifies the numbe
-000091e0: 7220 6f66 2061 6c69 676e 6d65 6e74 7320  r of alignments 
-000091f0: 6f66 2073 696d 696c 6172 206f 7220 6265  of similar or be
-00009200: 7474 6572 2071 7561 6c69 7479 2074 6861  tter quality tha
-00009210: 7420 796f 7520 6578 7065 6374 2074 6f20  t you expect to 
-00009220: 6669 6e64 2073 6561 7263 6869 6e67 2074  find searching t
-00009230: 6869 7320 7175 6572 7920 6167 6169 6e73  his query agains
-00009240: 7420 6120 6461 7461 6261 7365 206f 6620  t a database of 
-00009250: 7261 6e64 6f6d 2073 6571 7565 6e63 6573  random sequences
-00009260: 2074 6865 2073 616d 6520 7369 7a65 2061   the same size a
-00009270: 7320 7468 6520 6163 7475 616c 2074 6172  s the actual tar
-00009280: 6765 7420 6461 7461 6261 7365 2c20 6973  get database, is
-00009290: 2074 6865 206b 6579 2070 6172 616d 6574   the key paramet
-000092a0: 6572 206d 6561 7375 7269 6e67 2074 6865  er measuring the
-000092b0: 2073 6967 6e69 6669 6361 6e63 6520 6f66   significance of
-000092c0: 2061 2068 6974 2c20 7768 6963 6820 6973   a hit, which is
-000092d0: 2073 6574 2068 6572 6520 6173 2064 6566   set here as def
-000092e0: 6175 6c74 2031 652d 3130 2e20 4e6f 7465  ault 1e-10. Note
-000092f0: 2074 6861 7420 5b44 4941 4d4f 4e44 5d28   that [DIAMOND](
-00009300: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-00009310: 6f6d 2f62 6275 6368 6669 6e6b 2f64 6961  om/bbuchfink/dia
-00009320: 6d6f 6e64 2f77 696b 692f 312e 2d54 7574  mond/wiki/1.-Tut
-00009330: 6f72 6961 6c29 2069 7473 656c 6620 6279  orial) itself by
-00009340: 2064 6566 6175 6c74 206f 6e6c 7920 7265   default only re
-00009350: 706f 7274 7320 616c 6c20 616c 6967 6e6d  ports all alignm
-00009360: 656e 7473 2077 6974 6820 652d 7661 6c75  ents with e-valu
-00009370: 6520 3c20 302e 3030 312e 2054 6865 2070  e < 0.001. The p
-00009380: 6572 6365 6e74 6167 6520 6f66 2075 7070  ercentage of upp
-00009390: 6572 2068 6974 7320 7573 6564 2066 6f72  er hits used for
-000093a0: 2067 656e 6520 6c65 6e67 7468 206e 6f72   gene length nor
-000093b0: 6d61 6c69 7a61 7469 6f6e 2c20 6769 7665  malization, give
-000093c0: 6e20 6279 2060 2d6e 7060 206f 7220 602d  n by `-np` or `-
-000093d0: 2d6e 6f72 6d61 6c69 7a65 6470 6572 6365  -normalizedperce
-000093e0: 6e74 602c 2077 6869 6368 2064 6574 6572  nt`, which deter
-000093f0: 6d69 6e65 7320 7468 6520 7570 7065 7220  mines the upper 
-00009400: 7065 7263 656e 7469 6c65 206f 6620 6869  percentile of hi
-00009410: 7473 2070 6572 2062 696e 2028 6361 7465  ts per bin (cate
-00009420: 676f 7269 7a65 6420 6279 2067 656e 6520  gorized by gene 
-00009430: 6c65 6e67 7468 2920 7573 6564 2069 6e20  length) used in 
-00009440: 7468 6520 6669 7420 6f66 206c 696e 6561  the fit of linea
-00009450: 7220 7265 6772 6573 7369 6f6e 2c20 636f  r regression, co
-00009460: 6e73 6964 6572 696e 6720 7468 6174 206e  nsidering that n
-00009470: 6f74 2061 6c6c 2068 6974 7320 7065 7220  ot all hits per 
-00009480: 6269 6e20 7368 6f77 2061 7070 6172 656e  bin show apparen
-00009490: 7420 6c69 6e65 6172 2072 656c 6174 696f  t linear relatio
-000094a0: 6e73 6869 702c 2069 7320 7365 7420 6173  nship, is set as
-000094b0: 2064 6566 6175 6c74 2035 2c20 696e 6469   default 5, indi
-000094c0: 6361 7469 6e67 2074 6865 2075 7361 6765  cating the usage
-000094d0: 206f 6620 746f 7020 3525 2068 6974 7320   of top 5% hits 
-000094e0: 7065 7220 6269 6e2e 2054 6865 206e 756d  per bin. The num
-000094f0: 6265 7220 6f66 2062 696e 7320 6469 7669  ber of bins divi
-00009500: 6465 6420 696e 2067 656e 6520 6c65 6e67  ded in gene leng
-00009510: 7468 206e 6f72 6d61 6c69 7a61 7469 6f6e  th normalization
-00009520: 2c20 6769 7665 6e20 6279 2060 2d62 7360  , given by `-bs`
-00009530: 206f 7220 602d 2d62 696e 7360 2c20 6465   or `--bins`, de
-00009540: 7465 726d 696e 6573 2074 6865 206e 756d  termines the num
-00009550: 6265 7220 6f66 2062 696e 7320 746f 2063  ber of bins to c
-00009560: 6174 6567 6f72 697a 6520 7468 6520 6765  ategorize the ge
-00009570: 6e65 206c 656e 6774 682c 2069 7320 7365  ne length, is se
-00009580: 7420 6173 2064 6566 6175 6c74 2031 3030  t as default 100
-00009590: 2e20 5468 6520 7061 7261 6d65 7465 7220  . The parameter 
-000095a0: 602d 6e6e 6020 6f72 2060 2d2d 6e6f 6e6f  `-nn` or `--nono
-000095b0: 726d 616c 697a 6174 696f 6e60 2063 616e  rmalization` can
-000095c0: 2062 6520 7365 7420 746f 2063 616c 6c20   be set to call 
-000095d0: 6f66 6620 7468 6520 6e6f 726d 616c 697a  off the normaliz
-000095e0: 6174 696f 6e20 7072 6f63 6573 732c 2061  ation process, a
-000095f0: 6c74 686f 7567 6820 6974 2773 2073 7567  lthough it's sug
-00009600: 6765 7374 6564 2074 6f20 636f 6e64 7563  gested to conduc
-00009610: 7420 7468 6520 6e6f 726d 616c 697a 6174  t the normalizat
-00009620: 696f 6e20 746f 2061 6371 7569 7265 206d  ion to acquire m
-00009630: 6f72 6520 6163 6375 7261 7465 2067 656e  ore accurate gen
-00009640: 6520 6661 6d69 6c79 2063 6c75 7374 6572  e family cluster
-00009650: 696e 6720 7265 7375 6c74 2e20 5468 6520  ing result. The 
-00009660: 7061 7261 6d65 7465 7273 2060 2d2d 746f  parameters `--to
-00009670: 5f73 746f 7060 2061 6e64 2060 2d2d 6364  _stop` and `--cd
-00009680: 7360 2063 6f6e 7472 6f6c 2074 6865 2062  s` control the b
-00009690: 6568 6176 696f 7572 206f 6620 7472 616e  ehaviour of tran
-000096a0: 736c 6174 696e 6720 636f 6469 6e67 2073  slating coding s
-000096b0: 6571 7565 6e63 6520 696e 746f 2061 6d69  equence into ami
-000096c0: 6e6f 2061 6369 6420 7365 7175 656e 6365  no acid sequence
-000096d0: 2e20 4966 2074 6865 2060 2d2d 746f 5f73  . If the `--to_s
-000096e0: 746f 7060 2077 6173 2073 6574 2c20 7468  top` was set, th
-000096f0: 6520 7472 616e 736c 6174 696f 6e20 776f  e translation wo
-00009700: 756c 6420 6265 2074 6572 6d69 6e61 7465  uld be terminate
-00009710: 6420 6174 2074 6865 2066 6972 7374 2069  d at the first i
-00009720: 6e2d 6672 616d 6520 7374 6f70 2063 6f64  n-frame stop cod
-00009730: 6f6e 2c20 6f74 6865 7277 6973 6520 7468  on, otherwise th
-00009740: 6520 7472 616e 736c 6174 696f 6e20 776f  e translation wo
-00009750: 756c 6420 7369 6d70 6c79 2073 6b69 7020  uld simply skip 
-00009760: 616e 7920 7374 6f70 2063 6f64 6f6e 732e  any stop codons.
-00009770: 2049 6620 7468 6520 602d 2d63 6473 6020   If the `--cds` 
-00009780: 7761 7320 7365 742c 2073 6571 7565 6e63  was set, sequenc
-00009790: 6573 2074 6861 7420 646f 6573 6e27 7420  es that doesn't 
-000097a0: 7374 6172 7420 7769 7468 2061 2076 616c  start with a val
-000097b0: 6964 2073 7461 7274 2063 6f64 6f6e 2c20  id start codon, 
-000097c0: 6f72 2063 6f6e 7461 696e 7320 6d6f 7265  or contains more
-000097d0: 2074 6861 6e20 6f6e 6520 696e 2d66 7261   than one in-fra
-000097e0: 6d65 2073 746f 7020 636f 646f 6e2c 206f  me stop codon, o
-000097f0: 7220 6973 206e 6f74 2064 6976 6964 6162  r is not dividab
-00009800: 6c65 2062 7920 332c 2077 6f75 6c64 2062  le by 3, would b
-00009810: 6520 7369 6d70 6c79 2064 726f 7070 6564  e simply dropped
-00009820: 2c20 7375 6368 2074 6861 7420 6f6e 6c79  , such that only
-00009830: 2073 7472 6963 7420 636f 6d70 6c65 7465   strict complete
-00009840: 2063 6f64 696e 6720 7365 7175 656e 6365   coding sequence
-00009850: 7320 776f 756c 6420 6265 2069 6e63 6c75  s would be inclu
-00009860: 6465 6420 696e 2074 6865 2073 7562 7365  ded in the subse
-00009870: 7175 656e 7420 616e 616c 7973 6973 2e20  quent analysis. 
-00009880: 5468 6520 6578 6163 7420 6265 6861 7669  The exact behavi
-00009890: 6f75 7220 6f66 2060 2d2d 746f 5f73 746f  our of `--to_sto
-000098a0: 7060 2061 6e64 2060 2d2d 6364 7360 2069  p` and `--cds` i
-000098b0: 7320 6465 6669 6e65 6420 616e 6420 6465  s defined and de
-000098c0: 7363 7269 6265 6420 696e 2074 6865 205b  scribed in the [
-000098d0: 6269 6f70 7974 686f 6e5d 2868 7474 7073  biopython](https
-000098e0: 3a2f 2f62 696f 7079 7468 6f6e 2e6f 7267  ://biopython.org
-000098f0: 2f64 6f63 732f 312e 3735 2f61 7069 2f42  /docs/1.75/api/B
-00009900: 696f 2e53 6571 2e68 746d 6c23 4269 6f2e  io.Seq.html#Bio.
-00009910: 5365 712e 5365 712e 7472 616e 736c 6174  Seq.Seq.translat
-00009920: 6529 206c 6962 7261 7279 2e20 5468 6520  e) library. The 
-00009930: 6e75 6d62 6572 206f 6620 7061 7261 6c6c  number of parall
-00009940: 656c 2074 6872 6561 6473 2062 7920 7468  el threads by th
-00009950: 6520 6f70 7469 6f6e 2060 2d6e 6020 6f72  e option `-n` or
-00009960: 2060 2d2d 6e74 6872 6561 6473 6020 6361   `--nthreads` ca
-00009970: 6e20 6265 2073 6574 2074 6f20 6163 6365  n be set to acce
-00009980: 6c65 7261 7465 2074 6865 2063 616c 6375  lerate the calcu
-00009990: 6c61 7469 6f6e 2077 6974 6869 6e20 6064  lation within `d
-000099a0: 6961 6d6f 6e64 602e 2054 6865 2064 6972  iamond`. The dir
-000099b0: 6563 746f 7279 206f 6620 6f75 7470 7574  ectory of output
-000099c0: 206f 7220 696e 7465 726d 6564 6961 7465   or intermediate
-000099d0: 2066 696c 6573 2069 7320 6465 7465 726d   files is determ
-000099e0: 696e 6564 2062 7920 7468 6520 7061 7261  ined by the para
-000099f0: 6d65 7465 7220 602d 6f60 206f 7220 602d  meter `-o` or `-
-00009a00: 2d6f 7574 6469 7260 2c20 616e 6420 602d  -outdir`, and `-
-00009a10: 7460 206f 7220 602d 2d74 6d70 6469 7260  t` or `--tmpdir`
-00009a20: 2c20 7768 6963 6820 7769 6c6c 2062 6520  , which will be 
-00009a30: 6372 6561 7465 6420 6279 2074 6865 2070  created by the p
-00009a40: 726f 6772 616d 2069 7473 656c 6620 616e  rogram itself an
-00009a50: 6420 6265 206f 7665 7277 7269 7474 656e  d be overwritten
-00009a60: 2069 6620 7468 6520 666f 6c64 6572 2068   if the folder h
-00009a70: 6173 2061 6c72 6561 6479 2062 6565 6e20  as already been 
-00009a80: 6372 6561 7465 642e 204e 6f74 6520 7468  created. Note th
-00009a90: 6174 2074 6865 2073 6f66 7477 6172 6520  at the software 
-00009aa0: 6064 6961 6d6f 6e64 6020 7368 6f75 6c64  `diamond` should
-00009ab0: 2062 6520 7072 652d 696e 7374 616c 6c65   be pre-installe
-00009ac0: 6420 616e 6420 7365 7420 696e 2074 6865  d and set in the
-00009ad0: 2065 6e76 6972 6f6e 6d65 6e74 2070 6174   environment pat
-00009ae0: 6820 696e 2061 6c6c 2074 6865 2061 6e61  h in all the ana
-00009af0: 6c79 7369 7320 7065 7266 6f72 6d65 6420  lysis performed 
-00009b00: 6279 2060 7767 6420 646d 6460 2065 7863  by `wgd dmd` exc
-00009b10: 6570 7420 666f 7220 7468 6520 636f 6c6c  ept for the coll
-00009b20: 696e 6561 7220 636f 616c 6573 6365 6e63  inear coalescenc
-00009b30: 6520 616e 616c 7973 6973 2e0a 0a2a 2a57  e analysis...**W
-00009b40: 6520 7375 6767 6573 7420 7468 6174 2074  e suggest that t
-00009b50: 6865 2064 6566 6175 6c74 2073 6574 7469  he default setti
-00009b60: 6e67 2069 6e20 7768 6963 6820 7468 6520  ng in which the 
-00009b70: 696e 666c 6174 696f 6e20 6661 6374 6f72  inflation factor
-00009b80: 2069 7320 7365 7420 6173 2032 2c20 652d   is set as 2, e-
-00009b90: 7661 6c75 6520 6375 742d 6f66 6620 6173  value cut-off as
-00009ba0: 2031 652d 3130 2061 6e64 206f 7468 6572   1e-10 and other
-00009bb0: 2070 6172 616d 6574 6572 7320 696e 2064   parameters in d
-00009bc0: 6566 6175 6c74 2069 7320 6120 676f 6f64  efault is a good
-00009bd0: 2073 7461 7274 696e 6720 706f 696e 742c   starting point,
-00009be0: 2075 6e6c 6573 7320 796f 7520 7370 6563   unless you spec
-00009bf0: 6966 6963 616c 6c79 2077 616e 7420 746f  ifically want to
-00009c00: 2065 7870 6c6f 7265 2074 6865 2065 6666   explore the eff
-00009c10: 6563 7473 206f 6620 6469 6666 6572 656e  ects of differen
-00009c20: 7420 7061 7261 6d65 7465 7273 2e2a 2a20  t parameters.** 
-00009c30: 5375 6368 2074 6861 7420 7468 6520 636f  Such that the co
-00009c40: 6d6d 616e 6420 666f 7220 7468 6520 6465  mmand for the de
-00009c50: 6c69 6e65 6174 696f 6e20 6f66 2077 686f  lineation of who
-00009c60: 6c65 2070 6172 616e 6f6d 6520 6973 2073  le paranome is s
-00009c70: 696d 706c 7920 6173 2062 656c 6f77 2e0a  imply as below..
-00009c80: 0a60 6060 0a77 6764 2064 6d64 2041 7175  .```.wgd dmd Aqu
-00009c90: 696c 6567 6961 5f63 6f65 7275 6c65 610a  ilegia_coerulea.
-00009ca0: 6060 600a 0a2a 2a54 6865 2064 656c 696e  ```..**The delin
-00009cb0: 6561 7469 6f6e 206f 6620 5242 4873 2a2a  eation of RBHs**
-00009cc0: 0a60 6060 0a77 6764 2064 6d64 2073 6571  .```.wgd dmd seq
-00009cd0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
-00009ce0: 202d 6520 3165 2d31 3020 2d62 7320 3130   -e 1e-10 -bs 10
-00009cf0: 3020 2d6e 7020 3520 282d 6e6e 2920 282d  0 -np 5 (-nn) (-
-00009d00: 6e20 3429 2028 2d63 2030 2e39 2920 282d  n 4) (-c 0.9) (-
-00009d10: 2d6f 6766 6f72 6d61 7429 2028 2d2d 746f  -ogformat) (--to
-00009d20: 5f73 746f 7029 2028 2d2d 6364 7329 2028  _stop) (--cds) (
-00009d30: 2d6f 2077 6764 5f64 6d64 2920 282d 7420  -o wgd_dmd) (-t 
-00009d40: 776f 726b 696e 675f 746d 7029 0a60 6060  working_tmp).```
-00009d50: 0a0a 546f 2064 656c 696e 6561 7465 2052  ..To delineate R
-00009d60: 4248 7320 6265 7477 6565 6e20 7477 6f20  BHs between two 
-00009d70: 6364 7320 7365 7175 656e 6365 2066 696c  cds sequence fil
-00009d80: 6573 2c20 7468 6520 7265 6c65 7661 6e74  es, the relevant
-00009d90: 2070 6172 616d 6574 6572 2069 7320 6d6f   parameter is mo
-00009da0: 7374 6c79 2074 6865 2073 616d 6520 6173  stly the same as
-00009db0: 2077 686f 6c65 2070 6172 616e 6f6d 6520   whole paranome 
-00009dc0: 696e 6665 7265 6e63 652c 2065 7863 6570  inference, excep
-00009dd0: 7420 666f 7220 7468 6520 7061 7261 6d65  t for the parame
-00009de0: 7465 7220 602d 6360 206f 7220 602d 2d63  ter `-c` or `--c
-00009df0: 7363 6f72 6560 2c20 7768 6963 6820 7261  score`, which ra
-00009e00: 6e67 6573 2062 6574 7765 656e 2030 2061  nges between 0 a
-00009e10: 6e64 2031 2061 6e64 2069 7320 7573 6564  nd 1 and is used
-00009e20: 2074 6f20 7265 6c61 7820 7468 6520 7369   to relax the si
-00009e30: 6d69 6c61 7269 7479 2063 7574 6f66 6620  milarity cutoff 
-00009e40: 6672 6f6d 2074 6865 2065 7863 6c75 7369  from the exclusi
-00009e50: 7665 2072 6563 6970 726f 6361 6c20 6265  ve reciprocal be
-00009e60: 7374 2068 6974 7320 746f 2061 2063 6572  st hits to a cer
-00009e70: 7461 696e 2072 6174 696f 2061 7320 746f  tain ratio as to
-00009e80: 2074 6865 2062 6573 7420 6869 7473 2e20   the best hits. 
-00009e90: 466f 7220 696e 7374 616e 6365 2c20 6966  For instance, if
-00009ea0: 2074 6865 2067 656e 6520 6231 2066 726f   the gene b1 fro
-00009eb0: 6d20 6765 6e6f 6d65 2042 2068 6173 2074  m genome B has t
-00009ec0: 6865 2062 6573 7420 6869 7420 6765 6e65  he best hit gene
-00009ed0: 2061 3120 6672 6f6d 2067 656e 6f6d 6520   a1 from genome 
-00009ee0: 4120 7769 7468 2074 6865 2062 6974 2073  A with the bit s
-00009ef0: 636f 7265 2061 7320 3130 302c 2077 6869  core as 100, whi
-00009f00: 6368 2069 7320 6120 7363 6f72 696e 6720  ch is a scoring 
-00009f10: 6d61 7472 6978 2069 6e64 6570 656e 6465  matrix independe
-00009f20: 6e74 206d 6561 7375 7265 206f 6620 7468  nt measure of th
-00009f30: 6520 286c 6f63 616c 2920 7369 6d69 6c61  e (local) simila
-00009f40: 7269 7479 206f 6620 7468 6520 7477 6f20  rity of the two 
-00009f50: 616c 6967 6e65 6420 7365 7175 656e 6365  aligned sequence
-00009f60: 732c 2077 6974 6820 6c61 7267 6572 2076  s, with larger v
-00009f70: 616c 7565 7320 696e 6469 6361 7469 6e67  alues indicating
-00009f80: 2068 6967 6865 7220 7369 6d69 6c61 7269   higher similari
-00009f90: 7469 6573 2c20 6769 7665 6e20 7468 6520  ties, given the 
-00009fa0: 602d 6320 302e 3960 2c20 6765 6e65 7320  `-c 0.9`, genes 
-00009fb0: 6672 6f6d 2067 656e 6f6d 6520 4120 7768  from genome A wh
-00009fc0: 6963 6820 6861 7320 7468 6520 6269 7420  ich has the bit 
-00009fd0: 7363 6f72 6520 7769 7468 2067 656e 6520  score with gene 
-00009fe0: 6231 2068 6967 6865 7220 7468 616e 2030  b1 higher than 0
-00009ff0: 2e39 7831 3030 2077 696c 6c20 616c 736f  .9x100 will also
-0000a000: 2062 6520 7772 6974 7465 6e20 696e 2074   be written in t
-0000a010: 6865 2072 6573 756c 7420 6669 6c65 2c20  he result file, 
-0000a020: 7768 6963 6820 696e 2061 2073 656e 7365  which in a sense
-0000a030: 2061 7265 206e 6f74 2052 4248 7320 616e   are not RBHs an
-0000a040: 796d 6f72 6520 6f66 2063 6f75 7273 652c  ymore of course,
-0000a050: 2062 7574 2074 6865 2068 6967 686c 7920   but the highly 
-0000a060: 7369 6d69 6c61 7220 686f 6d6f 6c6f 6775  similar homologu
-0000a070: 6520 7061 6972 732e 2049 6620 6d6f 7265  e pairs. If more
-0000a080: 2074 6861 6e20 3220 7365 7175 656e 6365   than 2 sequence
-0000a090: 2066 696c 6573 2077 6572 6520 7072 6f76   files were prov
-0000a0a0: 6964 6564 2c20 6576 6572 7920 7061 6972  ided, every pair
-0000a0b0: 2d77 6973 6520 5242 4873 2077 6f75 6c64  -wise RBHs would
-0000a0c0: 2062 6520 6361 6c63 756c 6174 6564 2065   be calculated e
-0000a0d0: 7863 6570 7420 666f 7220 7175 6572 7969  xcept for queryi
-0000a0e0: 6e67 2074 6865 2073 616d 6520 7365 7175  ng the same sequ
-0000a0f0: 656e 6365 2069 7473 656c 662e 2054 6865  ence itself. The
-0000a100: 206e 756d 6265 7220 6f66 2070 6172 616c   number of paral
-0000a110: 6c65 6c20 7468 7265 6164 7320 746f 2062  lel threads to b
-0000a120: 6f6f 7374 6572 2074 6865 2072 756e 6e69  ooster the runni
-0000a130: 6e67 2073 7065 6564 2063 616e 2062 6520  ng speed can be 
-0000a140: 7365 7420 6279 2074 6865 206f 7074 696f  set by the optio
-0000a150: 6e20 602d 6e60 206f 7220 602d 2d6e 7468  n `-n` or `--nth
-0000a160: 7265 6164 7360 2077 6869 6368 2069 7320  reads` which is 
-0000a170: 7375 6767 6573 7465 6420 746f 2062 6520  suggested to be 
-0000a180: 7365 7420 6173 2028 4e2d 3129 4e2f 3220  set as (N-1)N/2 
-0000a190: 7768 6572 6520 4e20 6973 2074 6865 206e  where N is the n
-0000a1a0: 756d 6265 7220 6f66 2063 6473 2066 696c  umber of cds fil
-0000a1b0: 6573 2074 6f20 6163 6869 6576 6520 7468  es to achieve th
-0000a1c0: 6520 6869 6768 6573 7420 6566 6669 6369  e highest effici
-0000a1d0: 656e 6379 2e20 5468 6520 6f70 7469 6f6e  ency. The option
-0000a1e0: 2060 2d2d 6f67 666f 726d 6174 6020 6361   `--ogformat` ca
-0000a1f0: 6e20 6265 2073 6574 2074 6f20 6164 6420  n be set to add 
-0000a200: 696e 6465 7820 2866 6f72 2069 6e73 7461  index (for insta
-0000a210: 6e63 6520 4746 3030 3030 3030 3031 2920  nce GF00000001) 
-0000a220: 746f 2074 6865 206f 7574 7075 7420 5242  to the output RB
-0000a230: 4820 6765 6e65 2066 616d 696c 6965 7320  H gene families 
-0000a240: 7768 6963 6820 6361 6e20 6265 2066 7572  which can be fur
-0000a250: 7468 6572 2075 7365 6420 696e 2074 6865  ther used in the
-0000a260: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000a270: 2063 616c 6375 6c61 7469 6f6e 2062 7920   calculation by 
-0000a280: 6077 6764 206b 7364 602e 0a0a 2a2a 5468  `wgd ksd`...**Th
-0000a290: 6520 7375 6767 6573 7465 6420 636f 6d6d  e suggested comm
-0000a2a0: 616e 6420 746f 2073 7461 7274 2077 6974  and to start wit
-0000a2b0: 6820 6973 2061 6c73 6f20 756e 6465 7220  h is also under 
-0000a2c0: 7468 6520 6465 6661 756c 7420 7365 7474  the default sett
-0000a2d0: 696e 6720 7769 7468 2074 6865 2063 6f6d  ing with the com
-0000a2e0: 6d61 6e64 2073 686f 776e 2062 656c 6f77  mand shown below
-0000a2f0: 2a2a 0a0a 6060 600a 7767 6420 646d 6420  **..```.wgd dmd 
-0000a300: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
-0000a310: 6365 320a 6060 600a 0a2a 2a54 6865 2064  ce2.```..**The d
-0000a320: 656c 696e 6561 7469 6f6e 206f 6620 6c6f  elineation of lo
-0000a330: 6361 6c20 4d52 4248 732a 2a0a 6060 600a  cal MRBHs**.```.
-0000a340: 7767 6420 646d 6420 7365 7175 656e 6365  wgd dmd sequence
-0000a350: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
-0000a360: 656e 6365 3320 2d66 2073 6571 7565 6e63  ence3 -f sequenc
-0000a370: 6531 202d 6520 3165 2d31 3020 2d62 7320  e1 -e 1e-10 -bs 
-0000a380: 3130 3020 2d6e 7020 3520 282d 6e6e 2920  100 -np 5 (-nn) 
-0000a390: 282d 6e20 3429 2028 2d6b 6629 2028 2d6b  (-n 4) (-kf) (-k
-0000a3a0: 6429 2028 2d63 2030 2e39 2920 282d 2d74  d) (-c 0.9) (--t
-0000a3b0: 6f5f 7374 6f70 2920 282d 2d63 6473 2920  o_stop) (--cds) 
-0000a3c0: 282d 6f20 7767 645f 646d 6429 2028 2d74  (-o wgd_dmd) (-t
-0000a3d0: 2077 6f72 6b69 6e67 5f74 6d70 290a 6060   working_tmp).``
-0000a3e0: 600a 0a2a 2a54 6865 2064 6973 7469 6e63  `..**The distinc
-0000a3f0: 7469 6f6e 2062 6574 7765 656e 206c 6f63  tion between loc
-0000a400: 616c 2061 6e64 2067 6c6f 6261 6c20 2868  al and global (h
-0000a410: 6572 6575 6e64 6572 2920 4d52 4248 7320  ereunder) MRBHs 
-0000a420: 6973 2074 6861 7420 6c6f 6361 6c20 4d52  is that local MR
-0000a430: 4248 7320 6172 6520 7468 6520 7265 7375  BHs are the resu
-0000a440: 6c74 7320 6f66 206d 6572 6765 6420 5242  lts of merged RB
-0000a450: 4873 206f 6e20 6120 6a6f 696e 7420 666f  Hs on a joint fo
-0000a460: 6361 6c20 7370 6563 6965 732c 2066 6f72  cal species, for
-0000a470: 2069 6e73 7461 6e63 6520 696e 2061 2074   instance in a t
-0000a480: 6872 6565 2073 7065 6369 6573 2073 7973  hree species sys
-0000a490: 7465 6d20 2841 2c28 422c 4329 292c 2074  tem (A,(B,C)), t
-0000a4a0: 6865 206c 6f63 616c 204d 5242 4873 206f  he local MRBHs o
-0000a4b0: 6620 4120 6f6e 6c79 2072 6571 7569 7265  f A only require
-0000a4c0: 2074 6865 2063 616c 6375 6c61 7469 6f6e   the calculation
-0000a4d0: 206f 6620 5242 4873 2062 6574 7765 656e   of RBHs between
-0000a4e0: 2041 2061 6e64 2043 2028 6465 6e6f 7465   A and C (denote
-0000a4f0: 6420 6173 2041 4329 2061 6e64 2041 4220  d as AC) and AB 
-0000a500: 616e 6420 7468 656e 2074 6865 206d 6572  and then the mer
-0000a510: 6769 6e67 206f 6620 4142 2061 6e64 2041  ging of AB and A
-0000a520: 4320 6174 2074 6865 2061 7869 7320 6f66  C at the axis of
-0000a530: 2041 2c20 7768 696c 6520 7468 6520 676c   A, while the gl
-0000a540: 6f61 626c 204d 5242 4873 2061 7265 2069  oabl MRBHs are i
-0000a550: 6e64 6570 656e 6465 6e74 206f 6620 666f  ndependent of fo
-0000a560: 6361 6c20 7370 6563 6965 7320 696e 2074  cal species in t
-0000a570: 6861 7420 6974 206a 7573 7420 6361 6c6c  hat it just call
-0000a580: 7320 7468 6520 6361 6c63 756c 6174 696f  s the calculatio
-0000a590: 6e73 206f 6620 616c 6c20 706f 7373 6962  ns of all possib
-0000a5a0: 6c65 2073 7065 6369 6573 2070 6169 7220  le species pair 
-0000a5b0: 286e 6f74 2073 656c 6620 746f 2073 656c  (not self to sel
-0000a5c0: 6620 7370 6563 6965 7320 7061 6972 292c  f species pair),
-0000a5d0: 2073 7563 6820 7468 6174 2041 422c 2041   such that AB, A
-0000a5e0: 432c 2061 6e64 2042 4320 6172 6520 616c  C, and BC are al
-0000a5f0: 6c20 746f 2062 6520 6361 6c63 756c 6174  l to be calculat
-0000a600: 6564 2061 6e64 206d 6572 6765 642e 2a2a  ed and merged.**
-0000a610: 0a0a 5477 6f20 7479 7065 7320 6f66 204d  ..Two types of M
-0000a620: 5242 4873 2061 7320 696e 7465 7072 6574  RBHs as intepret
-0000a630: 6174 6564 2061 626f 7665 2063 616e 2062  ated above can b
-0000a640: 6520 6465 6c69 6e65 6174 6564 2062 7920  e delineated by 
-0000a650: 6077 6764 2064 6d64 602c 2074 6865 206c  `wgd dmd`, the l
-0000a660: 6f63 616c 204d 5242 4873 2061 6e64 2074  ocal MRBHs and t
-0000a670: 6865 2067 6c6f 6261 6c20 4d52 4248 732e  he global MRBHs.
-0000a680: 2054 6865 206c 6f63 616c 204d 5242 4873   The local MRBHs
-0000a690: 2061 7265 2063 6f6e 7374 7275 6374 6564   are constructed
-0000a6a0: 2062 7920 6d65 7267 696e 6720 616c 6c20   by merging all 
-0000a6b0: 7468 6520 7265 6c65 7661 6e74 2052 4248  the relevant RBH
-0000a6c0: 7320 6f6e 6c79 2077 6974 6820 7468 6520  s only with the 
-0000a6d0: 666f 6361 6c20 7370 6563 6965 732c 2077  focal species, w
-0000a6e0: 6869 6368 2069 7320 7365 7420 6279 2060  hich is set by `
-0000a6f0: 2d66 6020 6f72 2060 2d2d 666f 6375 7360  -f` or `--focus`
-0000a700: 2e20 5468 6520 7061 7261 6d65 7465 7220  . The parameter 
-0000a710: 602d 6b66 6020 6f72 2060 2d2d 6b65 6570  `-kf` or `--keep
-0000a720: 6661 7374 6160 2063 616e 2062 6520 7365  fasta` can be se
-0000a730: 7420 746f 2072 6574 6169 6e20 7468 6520  t to retain the 
-0000a740: 7365 7175 656e 6365 2069 6e66 6f72 6d61  sequence informa
-0000a750: 7469 6f6e 206f 6620 6561 6368 204d 5242  tion of each MRB
-0000a760: 482e 2054 6865 2070 6172 616d 6574 6572  H. The parameter
-0000a770: 2060 2d6b 6460 206f 7220 602d 2d6b 6565   `-kd` or `--kee
-0000a780: 7064 7570 6c69 6361 7465 7360 2064 6574  pduplicates` det
-0000a790: 6572 6d69 6e65 7320 7768 6574 6865 7220  ermines whether 
-0000a7a0: 7468 6520 7361 6d65 2067 656e 6573 2063  the same genes c
-0000a7b0: 616e 2061 7070 6561 7220 696e 2064 6966  an appear in dif
-0000a7c0: 6665 7265 6e74 206c 6f63 616c 204d 5242  ferent local MRB
-0000a7d0: 4873 2e20 4e6f 726d 616c 6c79 2074 6865  Hs. Normally the
-0000a7e0: 7265 2077 696c 6c20 6265 206e 6f20 6475  re will be no du
-0000a7f0: 706c 6963 6174 6573 2069 6e20 7468 6520  plicates in the 
-0000a800: 6c6f 6361 6c20 4d52 4248 7320 6275 7420  local MRBHs but 
-0000a810: 6966 2075 7365 7273 2073 6574 2074 6865  if users set the
-0000a820: 2060 2d63 6020 6173 2030 2e39 2028 6f72   `-c` as 0.9 (or
-0000a830: 2061 6e79 2076 616c 7565 2073 6d61 6c6c   any value small
-0000a840: 6572 2074 6861 6e20 3129 2c20 6974 2773  er than 1), it's
-0000a850: 206c 696b 656c 7920 7468 6174 2074 6865   likely that the
-0000a860: 2073 616d 6520 6765 6e65 2077 696c 6c20   same gene will 
-0000a870: 6861 7665 2063 6861 6e63 6520 746f 2061  have chance to a
-0000a880: 7070 6561 7220 6d75 6c74 6970 6c65 2074  ppear multiple t
-0000a890: 696d 6573 2069 6e20 6469 6666 6572 656e  imes in differen
-0000a8a0: 7420 6c6f 6361 6c20 4d52 4248 732e 2054  t local MRBHs. T
-0000a8b0: 6861 7420 6973 2074 6f20 7361 792c 2074  hat is to say, t
-0000a8c0: 6865 2070 6172 616d 6574 6572 2060 2d6b  he parameter `-k
-0000a8d0: 6460 2069 7320 6d65 616e 696e 6766 756c  d` is meaningful
-0000a8e0: 206f 6e6c 7920 7768 656e 2069 7427 7320   only when it's 
-0000a8f0: 7365 7420 746f 6765 7468 6572 2077 6974  set together wit
-0000a900: 6820 7468 6520 7061 7261 6d65 7465 7220  h the parameter 
-0000a910: 602d 6360 2e20 5468 6520 6e75 6d62 6572  `-c`. The number
-0000a920: 206f 6620 7061 7261 6c6c 656c 2074 6872   of parallel thr
-0000a930: 6561 6473 2069 7320 7375 6767 6573 7465  eads is suggeste
-0000a940: 6420 746f 2062 6520 7365 7420 6173 2074  d to be set as t
-0000a950: 6865 206e 756d 6265 7220 6f66 2063 6473  he number of cds
-0000a960: 2066 696c 6573 206d 696e 7573 2031 2e0a   files minus 1..
-0000a970: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
-0000a980: 7461 7274 696e 6720 7275 6e20 6973 2075  tarting run is u
-0000a990: 6e64 6572 2074 6865 2064 6566 6175 6c74  nder the default
-0000a9a0: 2070 6172 616d 6574 6572 2077 6974 6820   parameter with 
-0000a9b0: 7468 6520 636f 6d6d 616e 6420 7368 6f77  the command show
-0000a9c0: 6e20 6173 2062 656c 6f77 2e2a 2a0a 0a60  n as below.**..`
-0000a9d0: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
-0000a9e0: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
-0000a9f0: 6571 7565 6e63 6533 202d 6620 7365 7175  equence3 -f sequ
-0000aa00: 656e 6365 310a 6060 600a 0a2a 2a54 6865  ence1.```..**The
-0000aa10: 2064 656c 696e 6561 7469 6f6e 206f 6620   delineation of 
-0000aa20: 676c 6f62 616c 204d 5242 4873 2a2a 0a60  global MRBHs**.`
-0000aa30: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
-0000aa40: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
-0000aa50: 6571 7565 6e63 6533 202d 676d 202d 6520  equence3 -gm -e 
-0000aa60: 3165 2d31 3020 2d62 7320 3130 3020 2d6e  1e-10 -bs 100 -n
-0000aa70: 7020 3520 282d 6e6e 2920 282d 6e20 3429  p 5 (-nn) (-n 4)
-0000aa80: 2028 2d6b 6629 2028 2d6b 6429 2028 2d63   (-kf) (-kd) (-c
-0000aa90: 2030 2e39 2920 282d 2d74 6f5f 7374 6f70   0.9) (--to_stop
-0000aaa0: 2920 282d 2d63 6473 2920 282d 6f20 7767  ) (--cds) (-o wg
-0000aab0: 645f 646d 6429 2028 2d74 2077 6f72 6b69  d_dmd) (-t worki
-0000aac0: 6e67 5f74 6d70 290a 6060 600a 0a54 6865  ng_tmp).```..The
-0000aad0: 2067 6c6f 6261 6c20 4d52 4248 7320 6973   global MRBHs is
-0000aae0: 2063 6f6e 7374 7275 6374 6564 2062 7920   constructed by 
-0000aaf0: 6578 6861 7573 7469 7665 6c79 206d 6572  exhaustively mer
-0000ab00: 6769 6e67 2061 6c6c 2074 6865 2070 6f73  ging all the pos
-0000ab10: 7369 626c 6520 7061 6972 2d77 6973 6520  sible pair-wise 
-0000ab20: 5242 4873 2065 7863 6570 7420 666f 7220  RBHs except for 
-0000ab30: 7175 6572 7969 6e67 2074 6865 2073 6571  querying the seq
-0000ab40: 7565 6e63 6520 6974 7365 6c66 2c20 7768  uence itself, wh
-0000ab50: 6963 6820 6361 6e20 6265 2069 6e69 7469  ich can be initi
-0000ab60: 6174 6564 2062 7920 6164 6420 7468 6520  ated by add the 
-0000ab70: 666c 6167 2060 2d67 6d60 206f 7220 602d  flag `-gm` or `-
-0000ab80: 2d67 6c6f 6261 6c6d 7262 6860 2e20 5468  -globalmrbh`. Th
-0000ab90: 6520 7265 7374 206f 6620 7265 6c65 7661  e rest of releva
-0000aba0: 6e74 2070 6172 616d 6574 6572 7320 7374  nt parameters st
-0000abb0: 6179 7320 7468 6520 7361 6d65 2061 7320  ays the same as 
-0000abc0: 7468 6520 6c6f 6361 6c20 4d52 4248 732e  the local MRBHs.
-0000abd0: 2054 6865 206e 756d 6265 7220 6f66 2070   The number of p
-0000abe0: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
-0000abf0: 6973 2073 7567 6765 7374 6564 2074 6f20  is suggested to 
-0000ac00: 6265 2073 6574 2061 7320 284e 2d31 294e  be set as (N-1)N
-0000ac10: 2f32 2074 6f6f 2077 6865 7265 204e 2069  /2 too where N i
-0000ac20: 7320 7468 6520 6e75 6d62 6572 206f 6620  s the number of 
-0000ac30: 6364 7320 6669 6c65 7320 746f 2061 6368  cds files to ach
-0000ac40: 6965 7665 2074 6865 2068 6967 6865 7374  ieve the highest
-0000ac50: 2065 6666 6963 6965 6e63 792e 0a0a 2a2a   efficiency...**
-0000ac60: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-0000ac70: 7469 6e67 2072 756e 2069 7320 756e 6465  ting run is unde
-0000ac80: 7220 7468 6520 6465 6661 756c 7420 7061  r the default pa
-0000ac90: 7261 6d65 7465 7220 7769 7468 2074 6865  rameter with the
-0000aca0: 2063 6f6d 6d61 6e64 2073 686f 776e 2061   command shown a
-0000acb0: 7320 6265 6c6f 772e 2a2a 0a0a 6060 600a  s below.**..```.
-0000acc0: 7767 6420 646d 6420 7365 7175 656e 6365  wgd dmd sequence
-0000acd0: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
-0000ace0: 656e 6365 3320 2d67 6d0a 6060 600a 0a2a  ence3 -gm.```..*
-0000acf0: 2a54 6865 2064 656c 696e 6561 7469 6f6e  *The delineation
-0000ad00: 206f 6620 6f72 7468 6f67 726f 7570 732a   of orthogroups*
-0000ad10: 2a0a 6060 600a 7767 6420 646d 6420 7365  *.```.wgd dmd se
-0000ad20: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
-0000ad30: 3220 7365 7175 656e 6365 3320 2d6f 6920  2 sequence3 -oi 
-0000ad40: 2d6f 6f20 2d65 2031 652d 3130 202d 6273  -oo -e 1e-10 -bs
-0000ad50: 2031 3030 202d 6e70 2035 2028 2d6e 6e29   100 -np 5 (-nn)
-0000ad60: 2028 2d63 6329 2028 2d74 6529 2028 2d6d   (-cc) (-te) (-m
-0000ad70: 6320 302e 3829 2028 2d67 6e29 2028 2d74  c 0.8) (-gn) (-t
-0000ad80: 7265 6520 2766 6173 7474 7265 6527 2920  ree 'fasttree') 
-0000ad90: 282d 7473 2027 2d66 6173 7465 7374 2729  (-ts '-fastest')
-0000ada0: 2028 2d6e 2034 2920 282d 2d74 6f5f 7374   (-n 4) (--to_st
-0000adb0: 6f70 2920 282d 2d63 6473 2920 282d 6f20  op) (--cds) (-o 
-0000adc0: 7767 645f 646d 6429 2028 2d74 2077 6f72  wgd_dmd) (-t wor
-0000add0: 6b69 6e67 5f74 6d70 290a 6060 600a 0a49  king_tmp).```..I
-0000ade0: 6e20 6077 6764 2076 3260 2c20 7765 2061  n `wgd v2`, we a
-0000adf0: 6c73 6f20 696d 706c 656d 656e 7465 6420  lso implemented 
-0000ae00: 616e 2061 6c67 6f72 6974 686d 206f 6620  an algorithm of 
-0000ae10: 6465 6c69 6e65 6174 696e 6720 6f72 7468  delineating orth
-0000ae20: 6f67 726f 7570 732c 2077 6869 6368 2063  ogroups, which c
-0000ae30: 616e 2062 6520 696e 6974 6961 7465 6420  an be initiated 
-0000ae40: 7769 7468 2074 6865 2070 6172 616d 6574  with the paramet
-0000ae50: 6572 2060 2d6f 6960 206f 7220 602d 2d6f  er `-oi` or `--o
-0000ae60: 7274 686f 696e 6665 7260 2e20 5477 6f20  rthoinfer`. Two 
-0000ae70: 7761 7973 206f 6620 6465 6c69 6e65 6174  ways of delineat
-0000ae80: 696f 6e20 6361 6e20 6265 2063 686f 7365  ion can be chose
-0000ae90: 6e2c 2074 6865 2063 6f6e 6361 7465 6e61  n, the concatena
-0000aea0: 7469 6f6e 2077 6179 2028 7365 7420 6279  tion way (set by
-0000aeb0: 2074 6865 2070 6172 616d 6574 6572 2060   the parameter `
-0000aec0: 2d63 6360 206f 7220 602d 2d63 6f6e 6361  -cc` or `--conca
-0000aed0: 7460 2920 6f72 2074 6865 206e 6f6e 2d63  t`) or the non-c
-0000aee0: 6f6e 6361 7465 6e61 7469 6f6e 2028 6465  oncatenation (de
-0000aef0: 6661 756c 7429 2077 6179 2e20 496e 2062  fault) way. In b
-0000af00: 7269 6566 2c20 7468 6520 636f 6e63 6174  rief, the concat
-0000af10: 656e 6174 696f 6e20 7761 7920 6f66 2064  enation way of d
-0000af20: 656c 696e 6561 7469 6e67 206f 7274 686f  elineating ortho
-0000af30: 6772 6f75 7073 2073 7461 7274 7320 7769  groups starts wi
-0000af40: 7468 2063 6f6e 6361 7465 6e61 7469 6e67  th concatenating
-0000af50: 2061 6c6c 2074 6865 2073 6571 7565 6e63   all the sequenc
-0000af60: 6573 2069 6e74 6f20 6120 7369 6e67 6c65  es into a single
-0000af70: 2073 6571 7565 6e63 6520 6669 6c65 2061   sequence file a
-0000af80: 6e64 2074 6865 6e20 696e 6665 7272 696e  nd then inferrin
-0000af90: 6720 7468 6520 7768 6f6c 6520 7061 7261  g the whole para
-0000afa0: 6e6f 6d65 206f 6620 7468 6973 2073 696e  nome of this sin
-0000afb0: 676c 6520 7365 7175 656e 6365 2066 696c  gle sequence fil
-0000afc0: 6520 7769 7468 2074 6865 2063 6c75 7374  e with the clust
-0000afd0: 6572 696e 6720 7265 7375 6c74 7320 6d61  ering results ma
-0000afe0: 7070 6564 2062 6163 6b20 746f 2074 6865  pped back to the
-0000aff0: 2062 656c 6f6e 6769 6e67 2073 7065 6369   belonging speci
-0000b000: 6573 2e20 5768 696c 6520 7468 6520 6e6f  es. While the no
-0000b010: 6e2d 636f 6e63 6174 656e 6174 696f 6e20  n-concatenation 
-0000b020: 7761 7920 7374 6172 7473 2077 6974 6820  way starts with 
-0000b030: 7265 7370 6563 7469 7665 2070 6169 722d  respective pair-
-0000b040: 7769 7365 2064 6961 6d6f 6e64 2073 6561  wise diamond sea
-0000b050: 7263 6820 2869 6e63 6c75 6469 6e67 2071  rch (including q
-0000b060: 7565 7279 696e 6720 7468 6520 7361 6d65  uerying the same
-0000b070: 2073 6571 7565 6e63 6520 6974 7365 6c66   sequence itself
-0000b080: 2920 616e 6420 7468 656e 2061 6c6c 2074  ) and then all t
-0000b090: 6865 2073 6571 7565 6e63 6520 7369 6d69  he sequence simi
-0000b0a0: 6c61 7269 7479 2074 6162 6c65 7320 7769  larity tables wi
-0000b0b0: 6c6c 2062 6520 636f 6e63 6174 656e 6174  ll be concatenat
-0000b0c0: 6564 2061 6e64 2063 6c75 7374 6572 6564  ed and clustered
-0000b0d0: 2069 6e74 6f20 6f72 7468 6f67 726f 7570   into orthogroup
-0000b0e0: 732e 2053 6f6d 6520 6f74 6865 7220 706f  s. Some other po
-0000b0f0: 7373 6962 6c79 2075 7365 6675 6c20 706f  ssibly useful po
-0000b100: 7374 2d63 6c75 7374 6572 696e 6720 6675  st-clustering fu
-0000b110: 6e63 7469 6f6e 7320 6361 6e20 6265 2069  nctions can be i
-0000b120: 6e69 7469 6174 6564 2c20 696e 636c 7564  nitiated, includ
-0000b130: 696e 6720 7468 6520 7061 7261 6d65 7465  ing the paramete
-0000b140: 7220 602d 7465 6020 6f72 2060 2d2d 7465  r `-te` or `--te
-0000b150: 7374 736f 6760 2c20 7768 6963 6820 6361  stsog`, which ca
-0000b160: 6e20 6265 2073 6574 2074 6f20 7374 6172  n be set to star
-0000b170: 7420 7468 6520 756e 6269 6173 6564 2074  t the unbiased t
-0000b180: 6573 7420 6f66 2073 696e 676c 652d 636f  est of single-co
-0000b190: 7079 2067 656e 6520 6661 6d69 6c69 6573  py gene families
-0000b1a0: 2028 6e6f 7465 2074 6861 7420 7468 6973   (note that this
-0000b1b0: 2066 756e 6374 696f 6e20 6e65 6564 7320   function needs 
-0000b1c0: 6068 6d6d 6572 6020 2876 332e 3162 3229  `hmmer` (v3.1b2)
-0000b1d0: 2074 6f20 6265 2069 6e73 7461 6c6c 6564   to be installed
-0000b1e0: 2069 6e20 7468 6520 656e 7669 726f 6e6d   in the environm
-0000b1f0: 656e 7420 7061 7468 292c 2074 6865 2070  ent path), the p
-0000b200: 6172 616d 6574 6572 2060 2d6d 6360 206f  arameter `-mc` o
-0000b210: 7220 602d 2d6d 736f 6763 7574 602c 2072  r `--msogcut`, r
-0000b220: 616e 6769 6e67 2062 6574 7765 656e 2030  anging between 0
-0000b230: 2074 6f20 312c 2077 6869 6368 2063 616e   to 1, which can
-0000b240: 2062 6520 7365 7420 746f 2073 6561 7263   be set to searc
-0000b250: 6820 7468 6520 736f 2d63 616c 6c65 6420  h the so-called 
-0000b260: 6d6f 7374 6c79 2073 696e 676c 652d 636f  mostly single-co
-0000b270: 7079 2066 616d 696c 7920 7768 6963 6820  py family which 
-0000b280: 6861 7320 6869 6768 6572 2074 6861 6e20  has higher than 
-0000b290: 6365 7274 6169 6e20 6375 742d 6f66 6620  certain cut-off 
-0000b2a0: 7065 7263 656e 7461 6765 206f 6620 7370  percentage of sp
-0000b2b0: 6563 6965 7320 636f 7665 7261 6765 2c20  ecies coverage, 
-0000b2c0: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
-0000b2d0: 676e 6020 6f72 2060 2d2d 6765 746e 736f  gn` or `--getnso
-0000b2e0: 6760 2c20 7768 6963 6820 6361 6e20 6265  g`, which can be
-0000b2f0: 2073 6574 2074 6f20 7365 6172 6368 2066   set to search f
-0000b300: 6f72 206e 6573 7465 6420 7369 6e67 6c65  or nested single
-0000b310: 2d63 6f70 7920 6765 6e65 2066 616d 696c  -copy gene famil
-0000b320: 6965 7320 284e 534f 4773 2920 7768 6963  ies (NSOGs) whic
-0000b330: 6820 6973 206f 7269 6769 6e61 6c6c 7920  h is originally 
-0000b340: 6d75 6c74 6979 2d63 6f70 7920 6275 7420  multiy-copy but 
-0000b350: 6861 7320 6120 286d 6f73 746c 7929 2073  has a (mostly) s
-0000b360: 696e 676c 652d 636f 7079 2062 7261 6e63  ingle-copy branc
-0000b370: 6820 2877 6869 6368 2072 6571 7569 7265  h (which require
-0000b380: 7320 7468 6520 6368 6f73 656e 2074 7265  s the chosen tre
-0000b390: 652d 696e 6665 7265 6e63 6520 7072 6f67  e-inference prog
-0000b3a0: 7261 6d20 7365 7420 6279 2060 2d74 7265  ram set by `-tre
-0000b3b0: 6560 206f 7220 602d 2d74 7265 655f 6d65  e` or `--tree_me
-0000b3c0: 7468 6f64 6020 746f 2062 6520 7072 652d  thod` to be pre-
-0000b3d0: 696e 7374 616c 6c65 6420 696e 2074 6865  installed in the
-0000b3e0: 2065 6e76 6972 6f6e 6d65 6e74 2070 6174   environment pat
-0000b3f0: 6820 7769 7468 2074 6865 2070 6172 616d  h with the param
-0000b400: 6574 6572 7320 7365 7474 696e 6720 666f  eters setting fo
-0000b410: 7220 6765 6e65 2074 7265 6520 696e 6665  r gene tree infe
-0000b420: 7265 6e63 6520 636f 6e74 726f 6c6c 6564  rence controlled
-0000b430: 2062 7920 602d 7473 6020 6f72 2060 2d2d   by `-ts` or `--
-0000b440: 7472 6565 7365 7460 292e 2054 6865 2070  treeset`). The p
-0000b450: 726f 6772 616d 2060 7767 6420 646d 6460  rogram `wgd dmd`
-0000b460: 2077 6f75 6c64 2073 7469 6c6c 2063 6f6e   would still con
-0000b470: 6475 6374 2074 6865 2052 4248 7320 6361  duct the RBHs ca
-0000b480: 6c63 756c 6174 696f 6e20 756e 6c65 7373  lculation unless
-0000b490: 2074 6865 2070 6172 616d 6574 6572 2060   the parameter `
-0000b4a0: 2d6f 6f60 206f 7220 602d 2d6f 6e6c 796f  -oo` or `--onlyo
-0000b4b0: 7274 686f 6020 7761 7320 7365 742e 2049  rtho` was set. I
-0000b4c0: 6620 6f6e 6520 6f6e 6c79 2077 616e 7473  f one only wants
-0000b4d0: 2074 6f20 696e 6665 7220 7468 6520 6f72   to infer the or
-0000b4e0: 7468 6f67 726f 7570 732c 2069 7427 7320  thogroups, it's 
-0000b4f0: 7375 6767 6573 7465 6420 746f 2061 6464  suggested to add
-0000b500: 2074 6865 2066 6c61 6720 602d 6f6f 6020   the flag `-oo` 
-0000b510: 746f 206a 7573 7420 696d 706c 656d 656e  to just implemen
-0000b520: 7420 7468 6520 6f72 7468 6f67 726f 7570  t the orthogroup
-0000b530: 7320 6465 6c69 6e65 6174 696f 6e20 616e  s delineation an
-0000b540: 616c 7973 6973 2e20 5468 6520 6e75 6d62  alysis. The numb
-0000b550: 6572 206f 6620 7061 7261 6c6c 656c 2074  er of parallel t
-0000b560: 6872 6561 6473 2069 7320 7375 6767 6573  hreads is sugges
-0000b570: 7465 6420 746f 2062 6520 7365 7420 6173  ted to be set as
-0000b580: 2028 4e2b 3129 4e2f 3220 7768 6572 6520   (N+1)N/2 where 
-0000b590: 4e20 6973 2074 6865 206e 756d 6265 7220  N is the number 
-0000b5a0: 6f66 2063 6473 2066 696c 6573 2074 6f20  of cds files to 
-0000b5b0: 6163 6869 6576 6520 7468 6520 6869 6768  achieve the high
-0000b5c0: 6573 7420 6566 6669 6369 656e 6379 2073  est efficiency s
-0000b5d0: 696e 6365 2074 6865 2073 656c 662d 636f  ince the self-co
-0000b5e0: 6d70 6172 6973 6f6e 2069 7320 616c 736f  mparison is also
-0000b5f0: 2069 6e63 6c75 6465 642e 0a0a 2a2a 5468   included...**Th
-0000b600: 6520 6465 6661 756c 7420 7365 7474 696e  e default settin
-0000b610: 6720 6f66 2070 6172 616d 6574 6572 7320  g of parameters 
-0000b620: 6973 2061 2072 6561 736f 6e61 626c 6520  is a reasonable 
-0000b630: 7374 6172 7469 6e67 2070 6f69 6e74 2077  starting point w
-0000b640: 6974 6820 7468 6520 636f 6d6d 616e 6420  ith the command 
-0000b650: 6173 2062 656c 6f77 2e2a 2a0a 0a60 6060  as below.**..```
-0000b660: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
-0000b670: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000b680: 7565 6e63 6533 202d 6f69 202d 6f6f 0a60  uence3 -oi -oo.`
-0000b690: 6060 0a0a 2a2a 5468 6520 636f 6c6c 696e  ``..**The collin
-0000b6a0: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
-0000b6b0: 696e 6665 7265 6e63 6520 6f66 2070 6879  inference of phy
-0000b6c0: 6c6f 6765 6e79 2a2a 0a60 6060 0a77 6764  logeny**.```.wgd
-0000b6d0: 2064 6d64 2073 6571 7565 6e63 6531 2073   dmd sequence1 s
-0000b6e0: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
-0000b6f0: 6533 202d 6170 2061 7064 6174 6120 2d73  e3 -ap apdata -s
-0000b700: 6d20 736d 6461 7461 202d 6c65 206c 6564  m smdata -le led
-0000b710: 6174 6120 2d67 7420 6774 6461 7461 202d  ata -gt gtdata -
-0000b720: 636f 6320 282d 7472 6565 2027 6661 7374  coc (-tree 'fast
-0000b730: 7472 6565 2729 2028 2d74 7320 272d 6661  tree') (-ts '-fa
-0000b740: 7374 6573 7427 2920 282d 6e20 3429 2028  stest') (-n 4) (
-0000b750: 2d2d 746f 5f73 746f 7029 2028 2d2d 6364  --to_stop) (--cd
-0000b760: 7329 2028 2d6f 2077 6764 5f64 6d64 2920  s) (-o wgd_dmd) 
-0000b770: 282d 7420 776f 726b 696e 675f 746d 7029  (-t working_tmp)
-0000b780: 0a60 6060 0a0a 4120 6e6f 7665 6c20 7068  .```..A novel ph
-0000b790: 796c 6f67 656e 6574 6963 2069 6e66 6572  ylogenetic infer
-0000b7a0: 656e 6365 206d 6574 686f 6420 6e61 6d65  ence method name
-0000b7b0: 6420 2263 6f6c 6c69 6e65 6172 2063 6f61  d "collinear coa
-0000b7c0: 6c65 7363 656e 6365 2069 6e66 6572 656e  lescence inferen
-0000b7d0: 6365 2220 6973 2061 6c73 6f20 696d 706c  ce" is also impl
-0000b7e0: 656d 656e 7465 6420 696e 2060 7767 6420  emented in `wgd 
-0000b7f0: 7632 602e 2046 6f72 2074 6869 7320 616e  v2`. For this an
-0000b800: 616c 7973 6973 2c20 7573 6572 7320 6e65  alysis, users ne
-0000b810: 6564 2074 6f20 7072 6f76 6964 6520 7468  ed to provide th
-0000b820: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
-0000b830: 6669 6c65 2062 7920 602d 6170 6020 6f72  file by `-ap` or
-0000b840: 2060 2d2d 616e 6368 6f72 706f 696e 7473   `--anchorpoints
-0000b850: 602c 2074 6865 2063 6f6c 6c69 6e65 6172  `, the collinear
-0000b860: 2073 6567 6d65 6e74 7320 6669 6c65 2062   segments file b
-0000b870: 7920 602d 736d 6020 6f72 2060 2d2d 7365  y `-sm` or `--se
-0000b880: 676d 656e 7473 602c 2074 6865 206c 6973  gments`, the lis
-0000b890: 7473 6567 6d65 6e74 7320 6669 6c65 2062  tsegments file b
-0000b8a0: 7920 602d 6c65 6020 6f72 2060 2d2d 6c69  y `-le` or `--li
-0000b8b0: 7374 656c 656d 656e 7473 602c 2061 6e64  stelements`, and
-0000b8c0: 2074 6865 2067 656e 6520 7461 626c 6520   the gene table 
-0000b8d0: 6669 6c65 2062 7920 602d 6774 6020 6f72  file by `-gt` or
-0000b8e0: 2060 2d2d 6765 6e65 7461 626c 6560 2c20   `--genetable`, 
-0000b8f0: 616c 6c20 6f66 2077 6869 6368 2063 616e  all of which can
-0000b900: 2062 6520 7072 6f64 7563 6564 2069 6e20   be produced in 
-0000b910: 7468 6520 7072 6f67 7261 6d20 6077 6764  the program `wgd
-0000b920: 2073 796e 602e 2054 6865 2070 6172 616d   syn`. The param
-0000b930: 6574 6572 2060 2d63 6f63 6020 6f72 2060  eter `-coc` or `
-0000b940: 2d2d 636f 6c6c 696e 6561 7263 6f61 6c65  --collinearcoale
-0000b950: 7363 656e 6365 6020 6e65 6564 7320 746f  scence` needs to
-0000b960: 2062 6520 7365 7420 746f 2073 7461 7274   be set to start
-0000b970: 2074 6869 7320 616e 616c 7973 6973 2e20   this analysis. 
-0000b980: 5468 6520 7472 6565 2d69 6e66 6572 656e  The tree-inferen
-0000b990: 6365 2070 726f 6772 616d 2061 6e64 2074  ce program and t
-0000b9a0: 6865 2061 7373 6f63 6961 7465 6420 7061  he associated pa
-0000b9b0: 7261 6d65 7465 7273 2063 616e 2062 6520  rameters can be 
-0000b9c0: 7365 7420 6a75 7374 2061 7320 6162 6f76  set just as abov
-0000b9d0: 6520 6279 2060 2d74 7265 6560 206f 7220  e by `-tree` or 
-0000b9e0: 602d 2d74 7265 655f 6d65 7468 6f64 6020  `--tree_method` 
-0000b9f0: 616e 6420 602d 7473 6020 6f72 2060 2d2d  and `-ts` or `--
-0000ba00: 7472 6565 7365 7460 2e20 506c 6561 7365  treeset`. Please
-0000ba10: 2061 6c73 6f20 6d61 6b65 2073 7572 6520   also make sure 
-0000ba20: 7468 6520 6368 6f73 656e 2074 7265 652d  the chosen tree-
-0000ba30: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
-0000ba40: 6d20 6973 2069 6e73 7461 6c6c 6564 2069  m is installed i
-0000ba50: 6e20 7468 6520 656e 7669 726f 6e6d 656e  n the environmen
-0000ba60: 7420 7061 7468 2e20 5468 6520 7072 6f67  t path. The prog
-0000ba70: 7261 6d20 6061 7374 7261 6c2d 7072 6f60  ram `astral-pro`
-0000ba80: 2069 7320 7265 7175 6972 6564 2074 6f20   is required to 
-0000ba90: 6265 2069 6e73 7461 6c6c 6564 2069 6e20  be installed in 
-0000baa0: 7468 6520 656e 7669 726f 6e6d 656e 7420  the environment 
-0000bab0: 7061 7468 2074 6f6f 2e20 4e6f 7465 2074  path too. Note t
-0000bac0: 6861 7420 7468 6572 6520 7368 6f75 6c64  hat there should
-0000bad0: 2062 6520 6e6f 2064 7570 6c69 6361 7465   be no duplicate
-0000bae0: 6420 6765 6e65 2049 4473 2069 6e20 7468  d gene IDs in th
-0000baf0: 6520 7365 7175 656e 6365 2066 696c 652e  e sequence file.
-0000bb00: 2054 6865 2070 6172 616c 6c65 6c20 7468   The parallel th
-0000bb10: 7265 6164 7320 6865 7265 2061 7265 2074  reads here are t
-0000bb20: 6f20 6163 6365 6c65 7261 7465 2074 6865  o accelerate the
-0000bb30: 2073 6571 7565 6e63 6520 616c 6967 6e6d   sequence alignm
-0000bb40: 656e 7420 616e 6420 6765 6e65 2074 7265  ent and gene tre
-0000bb50: 6520 696e 6665 7265 6e63 6520 666f 7220  e inference for 
-0000bb60: 6561 6368 2067 656e 6520 6661 6d69 6c79  each gene family
-0000bb70: 2061 6e64 2074 6875 7320 7375 6767 6573   and thus sugges
-0000bb80: 7465 6420 746f 2062 6520 7365 7420 6173  ted to be set as
-0000bb90: 206d 7563 6820 6173 2074 6865 206e 756d   much as the num
-0000bba0: 6265 7220 6f66 2067 656e 6520 6661 6d69  ber of gene fami
-0000bbb0: 6c69 6573 2e0a 0a2a 2a41 2073 7567 6765  lies...**A sugge
-0000bbc0: 7374 6564 2073 7461 7274 696e 6720 7275  sted starting ru
-0000bbd0: 6e20 6f66 2074 6869 7320 616e 616c 7973  n of this analys
-0000bbe0: 6973 2069 7320 7769 7468 2074 6865 2073  is is with the s
-0000bbf0: 696d 706c 6520 636f 6d6d 616e 6420 6265  imple command be
-0000bc00: 6c6f 772e 2a2a 0a0a 6060 600a 7767 6420  low.**..```.wgd 
-0000bc10: 646d 6420 7365 7175 656e 6365 3120 7365  dmd sequence1 se
-0000bc20: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
-0000bc30: 3320 2d61 7020 6170 6461 7461 202d 736d  3 -ap apdata -sm
-0000bc40: 2073 6d64 6174 6120 2d6c 6520 6c65 6461   smdata -le leda
-0000bc50: 7461 202d 6774 2067 7464 6174 6120 2d63  ta -gt gtdata -c
-0000bc60: 6f63 0a60 6060 0a0a 2323 2320 7767 6420  oc.```..### wgd 
-0000bc70: 666f 6375 730a 0a2a 2a54 6865 2063 6f6e  focus..**The con
-0000bc80: 6361 7465 6e61 7469 6f6e 2d62 6173 6564  catenation-based
-0000bc90: 2f63 6f61 6c65 7363 656e 6365 2d62 6173  /coalescence-bas
-0000bca0: 6564 2070 6879 6c6f 6765 6e65 7469 6320  ed phylogenetic 
-0000bcb0: 696e 6665 7265 6e63 652a 2a0a 6060 600a  inference**.```.
-0000bcc0: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
-0000bcd0: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
-0000bce0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000bcf0: 2028 2d2d 636f 6e63 6174 656e 6174 696f   (--concatenatio
-0000bd00: 6e29 2028 2d2d 636f 616c 6573 6365 6e63  n) (--coalescenc
-0000bd10: 6529 2028 2d74 7265 6520 2766 6173 7474  e) (-tree 'fastt
-0000bd20: 7265 6527 2920 282d 7473 2027 2d66 6173  ree') (-ts '-fas
-0000bd30: 7465 7374 2729 2028 2d6e 2034 2920 282d  test') (-n 4) (-
-0000bd40: 2d74 6f5f 7374 6f70 2920 282d 2d63 6473  -to_stop) (--cds
-0000bd50: 2920 282d 6f20 7767 645f 666f 6375 7329  ) (-o wgd_focus)
-0000bd60: 2028 2d74 2077 6f72 6b69 6e67 5f74 6d70   (-t working_tmp
-0000bd70: 290a 6060 600a 0a54 6865 2070 726f 6772  ).```..The progr
-0000bd80: 616d 2060 7767 6420 666f 6375 7360 2069  am `wgd focus` i
-0000bd90: 6d70 6c65 6d65 6e74 6564 2074 776f 2062  mplemented two b
-0000bda0: 6173 6963 2070 6879 6c6f 6765 6e65 7469  asic phylogeneti
-0000bdb0: 6320 696e 6665 7265 6e63 6520 6d65 7468  c inference meth
-0000bdc0: 6f64 732c 2069 2e65 2e2c 2063 6f6e 6361  ods, i.e., conca
-0000bdd0: 7465 6e61 7469 6f6e 2d62 6173 6564 2061  tenation-based a
-0000bde0: 6e64 2063 6f61 6c65 7363 656e 6365 2d62  nd coalescence-b
-0000bdf0: 6173 6564 206d 6574 686f 6473 2e20 546f  ased methods. To
-0000be00: 2069 6e69 7469 6174 6520 7468 6573 6520   initiate these 
-0000be10: 616e 616c 7973 6973 2c20 7573 6572 7320  analysis, users 
-0000be20: 6e65 6564 2074 6f20 7365 7420 7468 6520  need to set the 
-0000be30: 666c 6167 206f 7074 696f 6e20 602d 2d63  flag option `--c
-0000be40: 6f6e 6361 7465 6e61 7469 6f6e 6020 6f72  oncatenation` or
-0000be50: 2060 2d2d 636f 616c 6573 6365 6e63 6560   `--coalescence`
-0000be60: 2e20 5468 6520 636f 6e63 6174 656e 6174  . The concatenat
-0000be70: 696f 6e2d 6261 7365 6420 6d65 7468 6f64  ion-based method
-0000be80: 2069 6e63 6c75 6465 7320 6120 6665 7720   includes a few 
-0000be90: 6d61 6a6f 7220 7374 6570 732c 2069 2e65  major steps, i.e
-0000bea0: 2e2c 2074 6865 206d 756c 7469 706c 6520  ., the multiple 
-0000beb0: 7365 7175 656e 6365 2061 6c69 676e 6d65  sequence alignme
-0000bec0: 6e74 2028 4d53 4129 206f 6620 6561 6368  nt (MSA) of each
-0000bed0: 2067 656e 6520 6661 6d69 6c79 2c20 7468   gene family, th
-0000bee0: 6520 636f 6e63 6174 656e 6174 696f 6e20  e concatenation 
-0000bef0: 6f66 2061 6c6c 2067 656e 6520 6661 6d69  of all gene fami
-0000bf00: 6c69 6573 2061 6e64 2074 6865 6e20 7468  lies and then th
-0000bf10: 6520 6765 6e65 2074 7265 6520 2861 6c73  e gene tree (als
-0000bf20: 6f20 7370 6563 6965 7320 7472 6565 2069  o species tree i
-0000bf30: 6e20 7468 6973 2063 6173 6529 2069 6e66  n this case) inf
-0000bf40: 6572 656e 6365 2e20 5468 6520 636f 616c  erence. The coal
-0000bf50: 6573 6365 6e63 652d 6261 7365 6420 6d65  escence-based me
-0000bf60: 7468 6f64 2077 696c 6c20 696e 7374 6561  thod will instea
-0000bf70: 6420 7065 7266 6f72 6d20 7468 6520 4d53  d perform the MS
-0000bf80: 4120 6f66 2065 6163 6820 6765 6e65 2066  A of each gene f
-0000bf90: 616d 696c 7920 616e 6420 7468 6520 6765  amily and the ge
-0000bfa0: 6e65 2074 7265 6520 696e 6665 7265 6e63  ne tree inferenc
-0000bfb0: 6520 6261 7365 6420 6f6e 2065 6163 6820  e based on each 
-0000bfc0: 4d53 412c 2061 6e64 2074 6865 6e20 696e  MSA, and then in
-0000bfd0: 6665 7220 7468 6520 7370 6563 6965 7320  fer the species 
-0000bfe0: 7472 6565 2062 6173 6564 206f 6e20 7468  tree based on th
-0000bff0: 6573 6520 696e 6469 7669 6475 616c 2067  ese individual g
-0000c000: 656e 6520 7472 6565 732e 2054 6865 2074  ene trees. The t
-0000c010: 7265 652d 696e 6665 7265 6e63 6520 7072  ree-inference pr
-0000c020: 6f67 7261 6d20 616e 6420 7468 6520 6173  ogram and the as
-0000c030: 736f 6369 6174 6564 2070 6172 616d 6574  sociated paramet
-0000c040: 6572 7320 6361 6e20 6265 2073 6574 206a  ers can be set j
-0000c050: 7573 7420 6173 2061 626f 7665 2062 7920  ust as above by 
-0000c060: 602d 7472 6565 6020 6f72 2060 2d2d 7472  `-tree` or `--tr
-0000c070: 6565 5f6d 6574 686f 6460 2061 6e64 2060  ee_method` and `
-0000c080: 2d74 7360 206f 7220 602d 2d74 7265 6573  -ts` or `--trees
-0000c090: 6574 602e 2050 6c65 6173 6520 616c 736f  et`. Please also
-0000c0a0: 206d 616b 6520 7375 7265 2074 6865 2063   make sure the c
-0000c0b0: 686f 7365 6e20 7472 6565 2d69 6e66 6572  hosen tree-infer
-0000c0c0: 656e 6365 2070 726f 6772 616d 2069 7320  ence program is 
-0000c0d0: 696e 7374 616c 6c65 6420 696e 2074 6865  installed in the
-0000c0e0: 2065 6e76 6972 6f6e 6d65 6e74 2070 6174   environment pat
-0000c0f0: 682e 2054 6865 2070 726f 6772 616d 2060  h. The program `
-0000c100: 6173 7472 616c 2d70 726f 6020 6973 2072  astral-pro` is r
-0000c110: 6571 7569 7265 6420 746f 2062 6520 696e  equired to be in
-0000c120: 7374 616c 6c65 6420 696e 2074 6865 2065  stalled in the e
-0000c130: 6e76 6972 6f6e 6d65 6e74 2070 6174 6820  nvironment path 
-0000c140: 6966 2074 6865 2063 6f61 6c65 7363 656e  if the coalescen
-0000c150: 6365 2d62 6173 6564 206d 6574 686f 6420  ce-based method 
-0000c160: 6973 2063 686f 7365 6e2e 2054 6865 2070  is chosen. The p
-0000c170: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
-0000c180: 6865 7265 2061 7265 2074 6f20 6163 6365  here are to acce
-0000c190: 6c65 7261 7465 2074 6865 2073 6571 7565  lerate the seque
-0000c1a0: 6e63 6520 616c 6967 6e6d 656e 7420 616e  nce alignment an
-0000c1b0: 6420 6765 6e65 2074 7265 6520 696e 6665  d gene tree infe
-0000c1c0: 7265 6e63 6520 666f 7220 6561 6368 2067  rence for each g
-0000c1d0: 656e 6520 6661 6d69 6c79 2074 6f6f 2061  ene family too a
-0000c1e0: 6e64 2074 6875 7320 7375 6767 6573 7465  nd thus suggeste
-0000c1f0: 6420 746f 2062 6520 7365 7420 6173 206d  d to be set as m
-0000c200: 7563 6820 6173 2074 6865 206e 756d 6265  uch as the numbe
-0000c210: 7220 6f66 2067 656e 6520 6661 6d69 6c69  r of gene famili
-0000c220: 6573 2e0a 0a2a 2a41 2073 7567 6765 7374  es...**A suggest
-0000c230: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
-0000c240: 6f66 2074 6869 7320 616e 616c 7973 6973  of this analysis
-0000c250: 2069 7320 7769 7468 2074 6865 2073 696d   is with the sim
-0000c260: 706c 6520 636f 6d6d 616e 6420 6265 6c6f  ple command belo
-0000c270: 772e 2a2a 0a0a 6060 600a 7767 6420 666f  w.**..```.wgd fo
-0000c280: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
-0000c290: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
-0000c2a0: 2073 6571 7565 6e63 6533 202d 2d63 6f6e   sequence3 --con
-0000c2b0: 6361 7465 6e61 7469 6f6e 0a77 6764 2066  catenation.wgd f
-0000c2c0: 6f63 7573 2066 616d 696c 6965 7320 7365  ocus families se
-0000c2d0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
-0000c2e0: 3220 7365 7175 656e 6365 3320 2d2d 636f  2 sequence3 --co
-0000c2f0: 616c 6573 6365 6e63 650a 6060 600a 0a2a  alescence.```..*
-0000c300: 2a54 6865 2066 756e 6374 696f 6e61 6c20  *The functional 
-0000c310: 616e 6e6f 7461 7469 6f6e 206f 6620 6765  annotation of ge
-0000c320: 6e65 2066 616d 696c 6965 732a 2a0a 6060  ne families**.``
-0000c330: 600a 7767 6420 666f 6375 7320 6661 6d69  `.wgd focus fami
-0000c340: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
-0000c350: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
-0000c360: 6533 202d 2d61 6e6e 6f74 6174 696f 6e20  e3 --annotation 
-0000c370: 6567 676e 6f67 202d 6564 2065 6464 6174  eggnog -ed eddat
-0000c380: 6120 2d2d 646d 6e62 2064 6264 6174 610a  a --dmnb dbdata.
-0000c390: 6060 600a 0a54 6865 2070 726f 6772 616d  ```..The program
-0000c3a0: 2060 7767 6420 666f 6375 7360 2061 6c73   `wgd focus` als
-0000c3b0: 6f20 6164 6465 6420 736f 6d65 2077 7261  o added some wra
-0000c3c0: 7070 696e 6720 6675 6e63 7469 6f6e 7320  pping functions 
-0000c3d0: 666f 7220 6675 6e63 7469 6f6e 616c 2061  for functional a
-0000c3e0: 6e6e 6f74 6174 696f 6e20 6f66 2067 656e  nnotation of gen
-0000c3f0: 6520 6661 6d69 6c69 6573 206f 6e20 7468  e families on th
-0000c400: 6520 686f 6f64 206f 6620 6461 7461 6261  e hood of databa
-0000c410: 7365 7320 616e 6420 736f 6674 7761 7265  ses and software
-0000c420: 7320 666f 7220 696e 7374 616e 6365 2060  s for instance `
-0000c430: 4567 674e 4f47 6020 616e 6420 6045 6767  EggNOG` and `Egg
-0000c440: 4e4f 472d 6d61 7070 6572 602e 2046 6f72  NOG-mapper`. For
-0000c450: 2074 6865 2061 6e6e 6f74 6174 696f 6e20   the annotation 
-0000c460: 7573 696e 6720 6045 6767 4e4f 472d 6d61  using `EggNOG-ma
-0000c470: 7070 6572 602c 2075 7365 7273 206e 6565  pper`, users nee
-0000c480: 6420 746f 2070 726f 7669 6465 2074 6865  d to provide the
-0000c490: 2070 6174 6820 746f 2074 6865 2065 6767   path to the egg
-0000c4a0: 4e4f 4720 616e 6e6f 7461 7469 6f6e 2064  NOG annotation d
-0000c4b0: 6174 6162 6173 6520 7669 6120 7468 6520  atabase via the 
-0000c4c0: 6f70 7469 6f6e 2060 2d65 6460 2c20 7468  option `-ed`, th
-0000c4d0: 6520 7061 7468 2074 6f20 7468 6520 6469  e path to the di
-0000c4e0: 616d 6f6e 642d 636f 6d70 6174 6962 6c65  amond-compatible
-0000c4f0: 2064 6174 6162 6173 6520 7669 6120 7468   database via th
-0000c500: 6520 6f70 7469 6f6e 2060 2d2d 646d 6e64  e option `--dmnd
-0000c510: 5f64 6260 2061 6e64 2074 6865 206f 7074  _db` and the opt
-0000c520: 696f 6e20 602d 2d61 6e6e 6f74 6174 696f  ion `--annotatio
-0000c530: 6e60 2073 6574 2061 7320 2265 6767 6e6f  n` set as "eggno
-0000c540: 6722 2e20 5468 6520 6d61 6e6e 6572 2068  g". The manner h
-0000c550: 6f77 2050 4641 4d20 616e 6e6f 7461 7469  ow PFAM annotati
-0000c560: 6f6e 2077 696c 6c20 6265 2070 6572 666f  on will be perfo
-0000c570: 726d 6564 2063 616e 2062 6520 636f 6e74  rmed can be cont
-0000c580: 726f 6c6c 6564 2076 6961 2074 6865 206f  rolled via the o
-0000c590: 7074 696f 6e20 6070 6661 6d60 2c20 6569  ption `pfam`, ei
-0000c5a0: 7468 6572 2022 6e6f 6e65 222c 2022 7265  ther "none", "re
-0000c5b0: 616c 6967 6e22 206f 7220 2264 656e 6f76  align" or "denov
-0000c5c0: 6f22 2c20 7468 6520 6465 7461 696c 6564  o", the detailed
-0000c5d0: 2065 7870 6c61 6e61 7469 6f6e 2063 616e   explanation can
-0000c5e0: 2062 6520 666f 756e 6420 6174 2074 6865   be found at the
-0000c5f0: 2077 696b 6920 6f66 205b 4567 674e 4f47   wiki of [EggNOG
-0000c600: 2d6d 6170 7065 725d 2868 7474 7073 3a2f  -mapper](https:/
-0000c610: 2f67 6974 6875 622e 636f 6d2f 6567 676e  /github.com/eggn
-0000c620: 6f67 6462 2f65 6767 6e6f 672d 6d61 7070  ogdb/eggnog-mapp
-0000c630: 6572 2f77 696b 6929 2e20 506c 6561 7365  er/wiki). Please
-0000c640: 2070 7265 2d69 6e73 7461 6c6c 2074 6865   pre-install the
-0000c650: 2060 4567 674e 4f47 2d6d 6170 7065 7260   `EggNOG-mapper`
-0000c660: 2070 7974 686f 6e20 7061 636b 6167 6520   python package 
-0000c670: 6966 2075 7369 6e67 2074 6869 7320 6675  if using this fu
-0000c680: 6e63 7469 6f6e 2e20 466f 7220 7468 6520  nction. For the 
-0000c690: 616e 6e6f 7461 7469 6f6e 2075 7369 6e67  annotation using
-0000c6a0: 2060 686d 6d73 6361 6e60 2c20 7768 6174   `hmmscan`, what
-0000c6b0: 2069 7320 696d 706c 656d 656e 7465 6420   is implemented 
-0000c6c0: 696e 2060 7767 6420 7632 6020 6973 2061  in `wgd v2` is a
-0000c6d0: 2073 696d 706c 6520 6275 6e64 6c65 2066   simple bundle f
-0000c6e0: 756e 6374 696f 6e20 746f 2070 6572 666f  unction to perfo
-0000c6f0: 726d 2060 686d 6d73 6361 6e60 2061 6e61  rm `hmmscan` ana
-0000c700: 6c79 7369 7320 666f 7220 6120 6769 7665  lysis for a give
-0000c710: 6e20 686d 6d20 7072 6f66 696c 6520 616e  n hmm profile an
-0000c720: 6420 7365 7420 6f66 2067 656e 6520 6661  d set of gene fa
-0000c730: 6d69 6c69 6573 2073 7563 6820 7468 6174  milies such that
-0000c740: 2075 7365 7273 206e 6565 6420 746f 2073   users need to s
-0000c750: 6574 2074 6865 206f 7074 696f 6e20 602d  et the option `-
-0000c760: 2d61 6e6e 6f74 6174 696f 6e60 2061 7320  -annotation` as 
-0000c770: 2268 6d6d 7066 616d 2220 616e 6420 7072  "hmmpfam" and pr
-0000c780: 6f76 6964 6520 7468 6520 686d 6d70 726f  ovide the hmmpro
-0000c790: 6669 6c65 2076 6961 2074 6865 206f 7074  file via the opt
-0000c7a0: 696f 6e20 602d 2d68 6d6d 602e 2046 6f72  ion `--hmm`. For
-0000c7b0: 2074 6865 2061 6e6e 6f74 6174 696f 6e20   the annotation 
-0000c7c0: 7573 696e 6720 6069 6e74 6572 7072 6f73  using `interpros
-0000c7d0: 6361 6e60 2c20 7573 6572 7320 6e65 6564  can`, users need
-0000c7e0: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
-0000c7f0: 7061 7468 2074 6f20 7468 6520 696e 7465  path to the inte
-0000c800: 7270 726f 7363 616e 2069 6e73 7461 6c6c  rproscan install
-0000c810: 6174 696f 6e20 666f 6c64 6572 2077 6865  ation folder whe
-0000c820: 7265 2074 6865 7265 2069 7320 6120 6069  re there is a `i
-0000c830: 6e74 6572 7072 6f73 6361 6e2e 7368 6020  nterproscan.sh` 
-0000c840: 6669 6c65 2076 6961 2074 6865 206f 7074  file via the opt
-0000c850: 696f 6e20 602d 2d65 7865 7061 7468 6020  ion `--exepath` 
-0000c860: 616e 6420 7365 7420 7468 6520 7468 6520  and set the the 
-0000c870: 6f70 7469 6f6e 2060 2d2d 616e 6e6f 7461  option `--annota
-0000c880: 7469 6f6e 6020 6173 2022 696e 7465 7270  tion` as "interp
-0000c890: 726f 7363 616e 222e 2054 6865 2070 6172  roscan". The par
-0000c8a0: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
-0000c8b0: 7265 2061 7265 2074 6f20 7061 7261 6c6c  re are to parall
-0000c8c0: 656c 697a 6520 7468 6520 616e 616c 7973  elize the analys
-0000c8d0: 6973 2066 6f72 2065 6163 6820 6765 6e65  is for each gene
-0000c8e0: 2066 616d 696c 7920 616e 6420 7468 7573   family and thus
-0000c8f0: 2073 7567 6765 7374 6564 2074 6f20 6265   suggested to be
-0000c900: 2073 6574 2061 7320 6d75 6368 2061 7320   set as much as 
-0000c910: 7468 6520 6e75 6d62 6572 206f 6620 6765  the number of ge
-0000c920: 6e65 2066 616d 696c 6965 732e 0a0a 2a2a  ne families...**
-0000c930: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-0000c940: 7469 6e67 2072 756e 206f 6620 7468 6973  ting run of this
-0000c950: 2061 6e61 6c79 7369 7320 6361 6e20 6265   analysis can be
-0000c960: 2077 6974 6820 7468 6520 636f 6d6d 616e   with the comman
-0000c970: 6420 6265 6c6f 772e 2a2a 0a0a 6060 600a  d below.**..```.
-0000c980: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
-0000c990: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
-0000c9a0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000c9b0: 202d 2d61 6e6e 6f74 6174 696f 6e20 6567   --annotation eg
-0000c9c0: 676e 6f67 202d 6564 2065 6464 6174 6120  gnog -ed eddata 
-0000c9d0: 2d2d 646d 6e62 2064 6264 6174 610a 7767  --dmnb dbdata.wg
-0000c9e0: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
-0000c9f0: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000ca00: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
-0000ca10: 2d61 6e6e 6f74 6174 696f 6e20 686d 6d70  -annotation hmmp
-0000ca20: 6661 6d20 2d2d 686d 6d20 686d 6d64 6174  fam --hmm hmmdat
-0000ca30: 610a 7767 6420 666f 6375 7320 6661 6d69  a.wgd focus fami
-0000ca40: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
-0000ca50: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
-0000ca60: 6533 202d 2d61 6e6e 6f74 6174 696f 6e20  e3 --annotation 
-0000ca70: 696e 7465 7270 726f 7363 616e 202d 2d65  interproscan --e
-0000ca80: 7865 7061 7468 2024 5041 5448 0a60 6060  xepath $PATH.```
-0000ca90: 0a0a 2a2a 5468 6520 7068 796c 6f67 656e  ..**The phylogen
-0000caa0: 6574 6963 2064 6174 696e 6720 6f66 2057  etic dating of W
-0000cab0: 4744 732a 2a0a 6060 600a 7767 6420 666f  GDs**.```.wgd fo
-0000cac0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
-0000cad0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
-0000cae0: 2073 6571 7565 6e63 6533 202d 6420 6d63   sequence3 -d mc
-0000caf0: 6d63 7472 6565 202d 7370 2073 7064 6174  mctree -sp spdat
-0000cb00: 6120 282d 2d70 726f 7463 6f63 6461 7469  a (--protcocdati
-0000cb10: 6e67 2920 282d 2d70 6172 7469 7469 6f6e  ng) (--partition
-0000cb20: 2920 282d 2d61 616d 6f64 656c 206c 6729  ) (--aamodel lg)
-0000cb30: 2028 2d64 7320 2762 7572 6e69 6e20 3d20   (-ds 'burnin = 
-0000cb40: 3230 3030 2729 2028 2d64 7320 2773 616d  2000') (-ds 'sam
-0000cb50: 7066 7265 7120 3d20 3130 3030 2729 2028  pfreq = 1000') (
-0000cb60: 2d64 7320 276e 7361 6d70 6c65 203d 2032  -ds 'nsample = 2
-0000cb70: 3030 3030 2729 2020 282d 6e20 3429 2028  0000')  (-n 4) (
-0000cb80: 2d2d 746f 5f73 746f 7029 2028 2d2d 6364  --to_stop) (--cd
-0000cb90: 7329 2028 2d6f 2077 6764 5f66 6f63 7573  s) (-o wgd_focus
-0000cba0: 2920 282d 7420 776f 726b 696e 675f 746d  ) (-t working_tm
-0000cbb0: 7029 0a60 6060 0a0a 5468 6520 6162 736f  p).```..The abso
-0000cbc0: 6c75 7465 2064 6174 696e 6720 6f66 2057  lute dating of W
-0000cbd0: 4744 7320 6973 2061 2073 7065 6369 6669  GDs is a specifi
-0000cbe0: 6320 7069 7065 6c69 6e65 2069 6d70 6c65  c pipeline imple
-0000cbf0: 6d65 6e74 6564 2069 6e20 6077 6764 2076  mented in `wgd v
-0000cc00: 3260 2075 7369 6e67 2074 6865 206d 6574  2` using the met
-0000cc10: 686f 6420 6f66 2070 6879 6c6f 6765 6e65  hod of phylogene
-0000cc20: 7469 6320 6461 7469 6e67 2e20 5468 6520  tic dating. The 
-0000cc30: 6661 6d69 6c69 6573 2075 7365 6420 696e  families used in
-0000cc40: 2074 6869 7320 7374 6570 2063 616e 2062   this step can b
-0000cc50: 6520 7072 6f64 7563 6564 2066 726f 6d20  e produced from 
-0000cc60: 6077 6764 2064 6d64 6020 616e 6420 6077  `wgd dmd` and `w
-0000cc70: 6764 2070 6561 6b60 2e20 4e6f 7465 2074  gd peak`. Note t
-0000cc80: 6861 7420 6865 7265 2077 6520 6f6e 6c79  hat here we only
-0000cc90: 2064 6973 6375 7373 2068 6f77 2074 6f20   discuss how to 
-0000cca0: 6461 7465 2057 4744 7320 7769 7468 2067  date WGDs with g
-0000ccb0: 656e 6f6d 6520 6173 7365 6d62 6c79 2c20  enome assembly, 
-0000ccc0: 696e 7374 6561 6420 6f66 2074 7261 6e73  instead of trans
-0000ccd0: 6372 6970 746f 6d65 2061 7373 656d 626c  criptome assembl
-0000cce0: 7920 2877 6869 6368 2077 696c 6c20 6265  y (which will be
-0000ccf0: 2064 6973 6375 7373 6564 2069 6e20 6120   discussed in a 
-0000cd00: 7365 7061 7261 7465 2073 6563 7469 6f6e  separate section
-0000cd10: 2068 6572 6575 6e64 6572 292e 2054 6865   hereunder). The
-0000cd20: 2061 7373 756d 7074 696f 6e20 7765 206d   assumption we m
-0000cd30: 6164 6520 6865 7265 2069 7320 7468 6174  ade here is that
-0000cd40: 206e 6f74 2061 6c6c 2061 6e63 686f 7220   not all anchor 
-0000cd50: 7061 6972 7320 2863 6f6c 6c69 6e65 6172  pairs (collinear
-0000cd60: 2064 7570 6c69 6361 7465 7329 2061 7265   duplicates) are
-0000cd70: 2073 7569 7461 626c 6520 666f 7220 7068   suitable for ph
-0000cd80: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
-0000cd90: 672c 2066 6f72 2069 6e73 7461 6e63 6520  g, for instance 
-0000cda0: 7468 6f73 6520 6661 7374 6c79 206f 7220  those fastly or 
-0000cdb0: 736c 6f77 6c79 2065 766f 6c76 696e 6720  slowly evolving 
-0000cdc0: 6765 6e65 2064 7570 6c69 6361 7465 732c  gene duplicates,
-0000cdd0: 2062 6563 6175 7365 2074 6865 7927 7265   because they're
-0000cde0: 2070 726f 6e65 2074 6f20 6769 7665 2062   prone to give b
-0000cdf0: 6961 7365 6420 6461 7469 6e67 2065 7374  iased dating est
-0000ce00: 696d 6174 696f 6e2e 2053 6f20 6173 2074  imation. So as t
-0000ce10: 6f20 7265 7472 6965 7665 2074 6865 2022  o retrieve the "
-0000ce20: 7265 6c69 6162 6c65 2220 616e 6368 6f72  reliable" anchor
-0000ce30: 2070 6169 7273 2c20 7765 2069 6d70 6c65   pairs, we imple
-0000ce40: 6d65 6e74 6564 2073 6f6d 6520 6d65 7468  mented some meth
-0000ce50: 6f64 7320 6f66 2069 6465 6e74 6966 7969  ods of identifyi
-0000ce60: 6e67 2063 7265 6469 6162 6c65 2061 6e63  ng crediable anc
-0000ce70: 686f 7220 7061 6972 7320 6261 7365 6420  hor pairs based 
-0000ce80: 6f6e 2074 6865 6972 204b 7320 7661 6c75  on their Ks valu
-0000ce90: 6573 2061 6e64 2f6f 7220 7265 7369 6469  es and/or residi
-0000cea0: 6e67 2063 6f6c 6c69 6e65 6172 2073 6567  ng collinear seg
-0000ceb0: 6d65 6e74 732e 2046 6f72 2061 2067 656e  ments. For a gen
-0000cec0: 6f6d 6520 7769 7468 2063 6c65 6172 2073  ome with clear s
-0000ced0: 6967 6e61 6c73 206f 6620 7075 7461 7469  ignals of putati
-0000cee0: 7665 2057 4744 732c 2073 7563 6820 6173  ve WGDs, such as
-0000cef0: 2074 6865 202a 4171 7569 6c65 6769 6120   the *Aquilegia 
-0000cf00: 636f 6572 756c 6561 2a20 696e 2074 6865  coerulea* in the
-0000cf10: 2065 7861 6d70 6c65 2c20 6120 6865 7572   example, a heur
-0000cf20: 6973 7469 6320 6d65 7468 6f64 2074 6861  istic method tha
-0000cf30: 7420 6170 706c 6965 7320 7468 6520 7072  t applies the pr
-0000cf40: 696e 6369 706c 6520 6f66 2068 6f77 205b  inciple of how [
-0000cf50: 6b73 7261 7465 735d 2868 7474 7073 3a2f  ksrates](https:/
-0000cf60: 2f67 6974 6875 622e 636f 6d2f 5649 422d  /github.com/VIB-
-0000cf70: 5053 422f 6b73 7261 7465 7329 2066 696e  PSB/ksrates) fin
-0000cf80: 6420 7468 6520 696e 6974 6961 6c20 7065  d the initial pe
-0000cf90: 616b 7320 616e 6420 7468 6569 7220 7061  aks and their pa
-0000cfa0: 7261 6d65 7465 7273 2077 6173 2069 6d70  rameters was imp
-0000cfb0: 6c65 6d65 6e74 6564 2074 6f20 6669 6e64  lemented to find
-0000cfc0: 2074 6865 2039 3525 2063 6f6e 6669 6465   the 95% confide
-0000cfd0: 6e63 6520 6c65 7665 6c20 6f66 2074 6865  nce level of the
-0000cfe0: 2061 7373 756d 6564 206c 6f67 6e6f 726d   assumed lognorm
-0000cff0: 616c 2064 6973 7472 6962 7574 696f 6e20  al distribution 
-0000d000: 6f66 2074 6865 2061 6e63 686f 7220 7061  of the anchor pa
-0000d010: 6972 202a 4b2a 3c73 7562 3e53 3c2f 7375  ir *K*<sub>S</su
-0000d020: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-0000d030: 696f 6e20 746f 2066 696c 7465 7220 616e  ion to filter an
-0000d040: 6368 6f72 2070 6169 7273 2077 6974 6820  chor pairs with 
-0000d050: 746f 6f20 6869 6768 206f 7220 6c6f 7720  too high or low 
-0000d060: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000d070: 6167 6573 2c20 7468 6520 6578 616d 706c  ages, the exampl
-0000d080: 6172 2063 6f6d 6d61 6e64 206f 6620 7768  ar command of wh
-0000d090: 6963 6820 6973 2073 686f 7765 6420 696e  ich is showed in
-0000d0a0: 2074 6865 2060 7767 6420 7065 616b 6020   the `wgd peak` 
-0000d0b0: 7365 6374 696f 6e20 6f66 205b 496c 6c75  section of [Illu
-0000d0c0: 7374 7261 7469 6f6e 5d28 6874 7470 733a  stration](https:
-0000d0d0: 2f2f 6769 7468 7562 2e63 6f6d 2f68 6563  //github.com/hec
-0000d0e0: 6865 2d70 7362 2f77 6764 3f74 6162 3d72  he-psb/wgd?tab=r
-0000d0f0: 6561 646d 652d 6f76 2d66 696c 6523 696c  eadme-ov-file#il
-0000d100: 6c75 7374 7261 7469 6f6e 292e 2049 6620  lustration). If 
-0000d110: 7468 6973 2068 6575 7269 7374 6963 206d  this heuristic m
-0000d120: 6574 686f 6420 6661 696c 6564 2074 6f20  ethod failed to 
-0000d130: 6769 7665 2072 6561 736f 6e61 626c 6520  give reasonable 
-0000d140: 7265 7375 6c74 732c 2077 6869 6368 2069  results, which i
-0000d150: 7320 7573 7561 6c6c 7920 6475 6520 746f  s usually due to
-0000d160: 2074 6865 2065 6666 6563 7420 6f66 206d   the effect of m
-0000d170: 756c 7469 706c 6520 6164 6a61 6365 6e74  ultiple adjacent
-0000d180: 2057 4744 7320 7468 6174 2062 6c75 7273   WGDs that blurs
-0000d190: 2074 6865 2070 6561 6b2d 6669 6e64 696e   the peak-findin
-0000d1a0: 6720 7072 6f63 6573 732c 2075 7365 7273  g process, users
-0000d1b0: 2063 616e 2074 7572 6e20 746f 2074 6865   can turn to the
-0000d1c0: 2063 6f6c 6c69 6e65 6172 2073 6567 6d65   collinear segme
-0000d1d0: 6e74 732d 6775 6964 6564 2061 6e63 686f  nts-guided ancho
-0000d1e0: 7220 7061 6972 2063 6c75 7374 6572 696e  r pair clusterin
-0000d1f0: 6720 696d 706c 656d 656e 7465 6420 696e  g implemented in
-0000d200: 2060 7767 6420 7632 602c 2069 6e20 7768   `wgd v2`, in wh
-0000d210: 6963 6820 6120 636f 6c6c 696e 6561 7220  ich a collinear 
-0000d220: 7365 676d 656e 742d 7769 7365 2047 4d4d  segment-wise GMM
-0000d230: 2063 6c75 7374 6572 696e 6720 7769 6c6c   clustering will
-0000d240: 2062 6520 6669 7273 7420 636f 6e64 7563   be first conduc
-0000d250: 7465 6420 6261 7365 6420 6f6e 2074 6865  ted based on the
-0000d260: 2022 736f 2d63 616c 6c65 6422 2073 6567   "so-called" seg
-0000d270: 6d65 6e74 202a 4b2a 3c73 7562 3e53 3c2f  ment *K*<sub>S</
-0000d280: 7375 623e 2061 6765 2072 6570 7265 7365  sub> age represe
-0000d290: 6e74 6564 2062 7920 7468 6520 6d65 6469  nted by the medi
-0000d2a0: 616e 202a 4b2a 3c73 7562 3e53 3c2f 7375  an *K*<sub>S</su
-0000d2b0: 623e 2061 6765 206f 6620 616c 6c20 7468  b> age of all th
-0000d2c0: 6520 7265 7369 6469 6e67 2073 796e 7465  e residing synte
-0000d2d0: 6c6f 6773 2028 6e6f 7465 2074 6861 7420  logs (note that 
-0000d2e0: 7468 6520 6765 6e65 2064 7570 6c69 6361  the gene duplica
-0000d2f0: 7465 2070 6169 7273 2061 646f 7074 6564  te pairs adopted
-0000d300: 2069 6e20 7468 6973 2073 7465 7020 6973   in this step is
-0000d310: 2066 726f 6d20 7468 6520 6669 6c65 2060   from the file `
-0000d320: 6d75 6c74 6970 6c69 636f 6e5f 7061 6972  multiplicon_pair
-0000d330: 732e 7478 7460 2077 6869 6368 2063 6f6e  s.txt` which con
-0000d340: 7461 696e 7320 7468 6520 6675 6c6c 2073  tains the full s
-0000d350: 6574 206f 6620 7379 6e74 656c 6f67 7329  et of syntelogs)
-0000d360: 2c20 696e 7374 6561 6420 6f66 2074 6865  , instead of the
-0000d370: 2073 6d61 6c6c 6572 2067 656e 6520 7365   smaller gene se
-0000d380: 7420 6f66 2061 6e63 686f 7220 7061 6972  t of anchor pair
-0000d390: 732e 2054 6865 2064 6973 7469 6e63 7469  s. The distincti
-0000d3a0: 6f6e 2062 6574 7765 656e 2061 6e63 686f  on between ancho
-0000d3b0: 7220 7061 6972 7320 616e 6420 7379 6e74  r pairs and synt
-0000d3c0: 656c 6f67 7320 6973 2074 6861 7420 7468  elogs is that th
-0000d3d0: 6520 6c61 7474 6572 2072 6566 6572 7320  e latter refers 
-0000d3e0: 746f 206d 756c 7469 706c 6520 7365 7473  to multiple sets
-0000d3f0: 206f 6620 6765 6e65 7320 6465 7269 7665   of genes derive
-0000d400: 6420 6672 6f6d 2074 6865 2073 616d 6520  d from the same 
-0000d410: 616e 6365 7374 7261 6c20 6765 6e6f 6d69  ancestral genomi
-0000d420: 6320 7265 6769 6f6e 2077 6869 6c65 2074  c region while t
-0000d430: 6865 2066 6f72 6d65 7220 696d 706c 6965  he former implie
-0000d440: 7320 7468 6520 6c61 7474 6572 2062 7574  s the latter but
-0000d450: 2072 6571 7569 7265 7320 6164 6469 7469   requires additi
-0000d460: 6f6e 616c 6c79 2074 6865 2063 6f6e 7365  onally the conse
-0000d470: 7276 6564 2067 656e 6520 6f72 6465 722c  rved gene order,
-0000d480: 2062 6f74 6820 6f66 2077 6869 6368 2061   both of which a
-0000d490: 7265 2c20 7769 7468 696e 2061 2067 656e  re, within a gen
-0000d4a0: 6f6d 652c 2061 7373 756d 6564 2074 6f20  ome, assumed to 
-0000d4b0: 6265 206f 7269 6769 6e61 7465 6420 6672  be originated fr
-0000d4c0: 6f6d 2074 6865 2064 7570 6c69 6361 7469  om the duplicati
-0000d4d0: 6f6e 206f 6620 6120 636f 6d6d 6f6e 2061  on of a common a
-0000d4e0: 6e63 6573 7472 616c 2067 656e 6f6d 6963  ncestral genomic
-0000d4f0: 2072 6567 696f 6e20 616e 6420 6173 2073   region and as s
-0000d500: 7563 6820 6465 656d 6564 2065 7669 6465  uch deemed evide
-0000d510: 6e63 6520 666f 7220 5747 442e 2054 6865  nce for WGD. The
-0000d520: 6e20 7468 6520 7379 6e74 656c 6f67 7320  n the syntelogs 
-0000d530: 7769 6c6c 2062 6520 6d61 7070 6564 2062  will be mapped b
-0000d540: 6163 6b20 6163 636f 7264 696e 6720 746f  ack according to
-0000d550: 2074 6865 2063 6c75 7374 6572 696e 6720   the clustering 
-0000d560: 7265 7375 6c74 7320 6f66 2074 6865 6972  results of their
-0000d570: 2061 6666 696c 6961 7465 6420 7365 676d   affiliated segm
-0000d580: 656e 7473 2e20 5369 6e63 6520 7468 6520  ents. Since the 
-0000d590: 4761 7573 7369 616e 2073 6861 7065 206f  Gaussian shape o
-0000d5a0: 6620 7468 6520 7365 676d 656e 7420 636c  f the segment cl
-0000d5b0: 7573 7465 7220 646f 6573 6e27 7420 6e65  uster doesn't ne
-0000d5c0: 6365 7373 6172 696c 7920 696d 6974 6174  cessarily imitat
-0000d5d0: 6520 7468 6520 7368 6170 6520 6f66 2074  e the shape of t
-0000d5e0: 6865 2072 6573 6964 696e 6720 7379 6e74  he residing synt
-0000d5f0: 656c 6f67 732c 2073 6f20 6173 2074 6f20  elogs, so as to 
-0000d600: 7265 7472 6965 7665 2074 6865 2022 7265  retrieve the "re
-0000d610: 6c69 6162 6c65 2220 6765 6e65 2073 6574  liable" gene set
-0000d620: 7320 666f 7220 7068 796c 6f67 656e 6574  s for phylogenet
-0000d630: 6963 2064 6174 696e 672c 2077 6520 6164  ic dating, we ad
-0000d640: 706f 7465 6420 7468 6520 2839 3525 2920  poted the (95%) 
-0000d650: 6869 6768 6573 7420 6465 6e73 6974 7920  highest density 
-0000d660: 7265 6769 6f6e 2028 4844 5229 206f 6620  region (HDR) of 
-0000d670: 7468 6520 7379 6e74 656c 6f67 202a 4b2a  the syntelog *K*
-0000d680: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
-0000d690: 2064 6973 7472 6962 7574 696f 6e20 666f   distribution fo
-0000d6a0: 7220 7468 6520 7068 796c 6f67 656e 6574  r the phylogenet
-0000d6b0: 6963 2064 6174 696e 672c 2074 6865 2063  ic dating, the c
-0000d6c0: 616c 6375 6c61 7469 6f6e 206f 6620 7468  alculation of th
-0000d6d0: 6520 2839 3525 2920 4844 5220 696e 2074  e (95%) HDR in t
-0000d6e0: 6865 2066 756e 6374 696f 6e20 5b63 616c  he function [cal
-0000d6f0: 6375 6c61 7465 4850 445d 2868 7474 7073  culateHPD](https
-0000d700: 3a2f 2f67 6974 6875 622e 636f 6d2f 6865  ://github.com/he
-0000d710: 6368 652d 7073 622f 7767 642f 626c 6f62  che-psb/wgd/blob
-0000d720: 2f70 6879 6c6f 6461 7469 6e67 2f77 6764  /phylodating/wgd
-0000d730: 2f70 6561 6b2e 7079 234c 3137 3537 2920  /peak.py#L1757) 
-0000d740: 7365 656b 7320 7468 6520 7368 6f72 7465  seeks the shorte
-0000d750: 7374 202a 4b2a 3c73 7562 3e53 3c2f 7375  st *K*<sub>S</su
-0000d760: 623e 2072 616e 6765 2028 612c 6229 2077  b> range (a,b) w
-0000d770: 6869 6368 2073 6174 6973 6669 6573 2074  hich satisfies t
-0000d780: 6865 2072 6571 7569 7265 6d65 6e74 206f  he requirement o
-0000d790: 6620 7370 616e 6e69 6e67 206d 6f72 6520  f spanning more 
-0000d7a0: 7468 616e 2028 3935 2529 206f 6620 616c  than (95%) of al
-0000d7b0: 6c20 7468 6520 2a4b 2a3c 7375 623e 533c  l the *K*<sub>S<
-0000d7c0: 2f73 7562 3e20 7661 6c75 6573 2e20 4f6e  /sub> values. On
-0000d7d0: 2074 6865 2070 7265 6d69 7365 206f 6620   the premise of 
-0000d7e0: 6964 656e 7469 6669 6564 2061 6e63 686f  identified ancho
-0000d7f0: 7220 7061 6972 7320 286e 6f74 6520 7468  r pairs (note th
-0000d800: 6174 2074 6865 2073 796e 7465 6c6f 6773  at the syntelogs
-0000d810: 2061 7265 2061 6c73 6f20 7265 6665 7272   are also referr
-0000d820: 6564 2061 7320 616e 6368 6f72 2070 6169  ed as anchor pai
-0000d830: 7273 2068 6572 6561 6674 6572 292c 2077  rs hereafter), w
-0000d840: 6520 696d 706c 656d 656e 7465 6420 696e  e implemented in
-0000d850: 2074 6865 2070 726f 6772 616d 2060 7767   the program `wg
-0000d860: 6420 646d 6460 2074 6865 2073 6f2d 6361  d dmd` the so-ca
-0000d870: 6c6c 6564 2061 6e63 686f 722d 6177 6172  lled anchor-awar
-0000d880: 6520 6c6f 6361 6c20 4d52 4248 7320 6f72  e local MRBHs or
-0000d890: 206f 7274 686f 6772 6f75 7073 2c20 696e   orthogroups, in
-0000d8a0: 2077 6869 6368 2074 6865 206f 7269 6769   which the origi
-0000d8b0: 6e61 6c20 6c6f 6361 6c20 4d52 4248 7320  nal local MRBHs 
-0000d8c0: 6172 6520 6675 7274 6865 7220 6d65 7267  are further merg
-0000d8d0: 6564 2077 6974 6820 616e 6368 6f72 2070  ed with anchor p
-0000d8e0: 6169 7273 2073 7563 6820 7468 6174 2065  airs such that e
-0000d8f0: 6163 6820 6f72 7468 6f67 726f 7570 2063  ach orthogroup c
-0000d900: 6f6e 7461 696e 7320 7468 6520 616e 6368  ontains the anch
-0000d910: 6f72 2070 6169 7220 616e 6420 7468 6520  or pair and the 
-0000d920: 6f72 7468 6f6c 6f67 7565 732e 2057 6974  orthologues. Wit
-0000d930: 6820 7468 6973 206f 7274 686f 6772 6f75  h this orthogrou
-0000d940: 702c 2075 7365 7273 2074 6865 6e20 6e65  p, users then ne
-0000d950: 6564 206f 6e65 2073 7461 7274 696e 6720  ed one starting 
-0000d960: 7472 6565 2066 696c 6520 2861 7320 7368  tree file (as sh
-0000d970: 6f77 6e20 696e 2074 6865 205b 496c 6c75  own in the [Illu
-0000d980: 7374 7261 7469 6f6e 5d28 6874 7470 733a  stration](https:
-0000d990: 2f2f 6769 7468 7562 2e63 6f6d 2f68 6563  //github.com/hec
-0000d9a0: 6865 2d70 7362 2f77 6764 2f74 7265 652f  he-psb/wgd/tree/
-0000d9b0: 7068 796c 6f64 6174 696e 673f 7461 623d  phylodating?tab=
-0000d9c0: 7265 6164 6d65 2d6f 762d 6669 6c65 2369  readme-ov-file#i
-0000d9d0: 6c6c 7573 7472 6174 696f 6e29 2070 6172  llustration) par
-0000d9e0: 7429 2069 6e64 6963 6174 696e 6720 7468  t) indicating th
-0000d9f0: 6520 7472 6565 2074 6f70 6f6c 6f67 7920  e tree topology 
-0000da00: 616e 6420 666f 7373 696c 2063 616c 6962  and fossil calib
-0000da10: 7261 7469 6f6e 2069 6e66 6f72 6d61 7469  ration informati
-0000da20: 6f6e 2066 6f72 2074 6865 2066 696e 616c  on for the final
-0000da30: 2070 6879 6c6f 6765 6e65 7469 6320 6461   phylogenetic da
-0000da40: 7469 6e67 2075 7369 6e67 2074 6865 2070  ting using the p
-0000da50: 726f 6772 616d 2060 7767 6420 666f 6375  rogram `wgd focu
-0000da60: 7360 2e20 5573 6572 7320 6361 6e20 7365  s`. Users can se
-0000da70: 7420 7468 6520 666c 6167 206f 7074 696f  t the flag optio
-0000da80: 6e20 602d 2d70 726f 7463 6f63 6461 7469  n `--protcocdati
-0000da90: 6e67 6020 746f 206f 6e6c 7920 636f 6e64  ng` to only cond
-0000daa0: 7563 7420 7468 6520 6461 7469 6e67 206f  uct the dating o
-0000dab0: 6620 7468 6520 636f 6e63 6174 656e 6174  f the concatenat
-0000dac0: 696f 6e20 7072 6f74 6569 6e20 4d53 4120  ion protein MSA 
-0000dad0: 6f72 2074 6865 2066 6c61 6720 6f70 7469  or the flag opti
-0000dae0: 6f6e 2060 2d2d 7072 6f74 6461 7469 6e67  on `--protdating
-0000daf0: 6020 746f 206f 6e6c 7920 636f 6e64 7563  ` to only conduc
-0000db00: 7420 7468 6520 6461 7469 6e67 206f 6620  t the dating of 
-0000db10: 7468 6520 7072 6f74 6569 6e20 4d53 4173  the protein MSAs
-0000db20: 2c20 6e6f 7465 6420 7468 6174 2074 6865  , noted that the
-0000db30: 7365 2074 776f 206f 7074 696f 6e73 206f  se two options o
-0000db40: 6e6c 7920 776f 726b 2066 6f72 2074 6865  nly work for the
-0000db50: 2060 6d63 6d63 7472 6565 6020 6f70 7469   `mcmctree` opti
-0000db60: 6f6e 2073 6f20 6661 722e 2054 6865 2066  on so far. The f
-0000db70: 6c61 6720 6f70 7469 6f6e 2060 2d2d 7061  lag option `--pa
-0000db80: 7274 6974 696f 6e60 2063 616e 2062 6520  rtition` can be 
-0000db90: 7365 7420 746f 2070 6572 666f 726d 2060  set to perform `
-0000dba0: 6d63 6d63 7472 6565 6020 616e 616c 7973  mcmctree` analys
-0000dbb0: 6973 2075 7369 6e67 2074 6865 2070 6172  is using the par
-0000dbc0: 7469 7469 6f6e 6564 2064 6174 6120 2869  titioned data (i
-0000dbd0: 2e65 2e2c 2031 7374 2c20 326e 6420 616e  .e., 1st, 2nd an
-0000dbe0: 6420 3372 6420 706f 7369 7469 6f6e 206f  d 3rd position o
-0000dbf0: 6620 636f 646f 6e29 2069 6e73 7465 6164  f codon) instead
-0000dc00: 206f 6620 7573 696e 6720 7468 6520 636f   of using the co
-0000dc10: 646f 6e20 6173 2061 2077 686f 6c65 2e20  don as a whole. 
-0000dc20: 5468 6520 6f70 7469 6f6e 2060 2d2d 6161  The option `--aa
-0000dc30: 6d6f 6465 6c60 2063 616e 2062 6520 7365  model` can be se
-0000dc40: 7420 746f 2064 6574 6572 6d69 6e65 2074  t to determine t
-0000dc50: 6865 2061 6d69 6e6f 2061 6369 6420 6d6f  he amino acid mo
-0000dc60: 6465 6c20 6170 706c 6965 6420 696e 2060  del applied in `
-0000dc70: 6d63 6d63 7472 6565 6020 616e 616c 7973  mcmctree` analys
-0000dc80: 6973 2e20 5468 6520 6f70 7469 6f6e 2060  is. The option `
-0000dc90: 2d64 7360 2063 616e 2062 6520 7573 6564  -ds` can be used
-0000dca0: 2074 6f20 7365 7420 7061 7261 6d65 7465   to set paramete
-0000dcb0: 7273 2066 6f72 2074 6865 206d 6f6c 6563  rs for the molec
-0000dcc0: 756c 6172 2064 6174 696e 6720 7072 6f67  ular dating prog
-0000dcd0: 7261 6d2e 2054 6865 2070 6172 616c 6c65  ram. The paralle
-0000dce0: 6c20 7468 7265 6164 7320 6865 7265 2061  l threads here a
-0000dcf0: 7265 2074 6f20 7061 7261 6c6c 656c 697a  re to paralleliz
-0000dd00: 6520 7468 6520 616e 616c 7973 6973 2066  e the analysis f
-0000dd10: 6f72 2065 6163 6820 6765 6e65 2066 616d  or each gene fam
-0000dd20: 696c 7920 616e 6420 7468 7573 2073 7567  ily and thus sug
-0000dd30: 6765 7374 6564 2074 6f20 6265 2073 6574  gested to be set
-0000dd40: 2061 7320 6d75 6368 2061 7320 7468 6520   as much as the 
-0000dd50: 6e75 6d62 6572 206f 6620 6765 6e65 2066  number of gene f
-0000dd60: 616d 696c 6965 732e 0a0a 2a2a 4120 7375  amilies...**A su
-0000dd70: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
-0000dd80: 2072 756e 2063 616e 2062 6520 7769 7468   run can be with
-0000dd90: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
-0000dda0: 6f77 2e2a 2a0a 0a60 6060 0a77 6764 2066  ow.**..```.wgd f
-0000ddb0: 6f63 7573 2066 616d 696c 6965 7320 7365  ocus families se
-0000ddc0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
-0000ddd0: 3220 7365 7175 656e 6365 3320 2d64 206d  2 sequence3 -d m
-0000dde0: 636d 6374 7265 6520 2d73 7020 7370 6461  cmctree -sp spda
-0000ddf0: 7461 0a77 6764 2066 6f63 7573 2066 616d  ta.wgd focus fam
-0000de00: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
-0000de10: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
-0000de20: 6365 3320 2d64 2072 3873 202d 7370 2073  ce3 -d r8s -sp s
-0000de30: 7064 6174 6120 2d2d 6e73 6974 6573 206e  pdata --nsites n
-0000de40: 7369 7465 696e 666f 0a77 6764 2066 6f63  siteinfo.wgd foc
-0000de50: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
-0000de60: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000de70: 7365 7175 656e 6365 3320 2d64 2062 6561  sequence3 -d bea
-0000de80: 7374 202d 2d66 6f73 7369 6c20 666f 7373  st --fossil foss
-0000de90: 696c 696e 666f 202d 2d72 6f6f 7468 6569  ilinfo --roothei
-0000dea0: 6768 7420 726f 6f74 6865 6967 6874 696e  ght rootheightin
-0000deb0: 666f 202d 2d63 6861 696e 7365 7420 6368  fo --chainset ch
-0000dec0: 6169 6e73 6574 696e 666f 202d 2d62 6561  ainsetinfo --bea
-0000ded0: 7374 6c67 6a61 7220 2450 4154 480a 6060  stlgjar $PATH.``
-0000dee0: 600a 0a23 2323 2077 6764 206b 7364 0a0a  `..### wgd ksd..
-0000def0: 2a2a 5468 6520 636f 6e73 7472 7563 7469  **The constructi
-0000df00: 6f6e 206f 6620 7768 6f6c 6520 7061 7261  on of whole para
-0000df10: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
-0000df20: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-0000df30: 7574 696f 6e2a 2a0a 6060 600a 7767 6420  ution**.```.wgd 
-0000df40: 6b73 6420 6661 6d69 6c69 6573 2073 6571  ksd families seq
-0000df50: 7565 6e63 6520 282d 6f20 7767 645f 6b73  uence (-o wgd_ks
-0000df60: 6420 2d74 2077 6764 5f6b 7364 5f74 6d70  d -t wgd_ksd_tmp
-0000df70: 202d 2d6e 7468 7265 6164 7320 3420 2d2d   --nthreads 4 --
-0000df80: 746f 5f73 746f 7020 2d2d 6364 7320 2d2d  to_stop --cds --
-0000df90: 7061 6972 7769 7365 202d 2d73 7472 6970  pairwise --strip
-0000dfa0: 5f67 6170 7320 2d2d 616c 6967 6e65 7220  _gaps --aligner 
-0000dfb0: 6d61 6666 7420 2d2d 7472 6565 5f6d 6574  mafft --tree_met
-0000dfc0: 686f 6420 6661 7374 7472 6565 202d 2d6e  hod fasttree --n
-0000dfd0: 6f64 655f 6176 6572 6167 6529 0a60 6060  ode_average).```
-0000dfe0: 0a0a 5468 6520 7072 6f67 7261 6d20 6077  ..The program `w
-0000dff0: 6764 206b 7364 602c 2061 7320 696d 7069  gd ksd`, as impi
-0000e000: 6564 2062 7920 6974 7320 6e61 6d65 2c20  ed by its name, 
-0000e010: 6973 2066 6f72 2074 6865 2063 6f6e 7374  is for the const
-0000e020: 7275 6374 696f 6e20 6f66 202a 4b2a 3c73  ruction of *K*<s
-0000e030: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-0000e040: 6973 7472 6962 7574 696f 6e2e 2045 7863  istribution. Exc
-0000e050: 6570 7420 666f 7220 7468 6520 6166 6f72  ept for the afor
-0000e060: 656d 656e 7469 6f6e 6564 2070 6172 616d  ementioned param
-0000e070: 6574 6572 7320 7375 6368 2061 7320 602d  eters such as `-
-0000e080: 2d74 6f5f 7374 6f70 6020 616e 6420 602d  -to_stop` and `-
-0000e090: 2d63 6473 602c 2074 6865 7265 2061 7265  -cds`, there are
-0000e0a0: 2073 6f6d 6520 696d 706f 7274 616e 7420   some important 
-0000e0b0: 7061 7261 6d65 7465 7273 2074 6861 7420  parameters that 
-0000e0c0: 6861 7665 2063 7275 6369 616c 2069 6d70  have crucial imp
-0000e0d0: 6163 7420 6f6e 2074 6865 202a 4b2a 3c73  act on the *K*<s
-0000e0e0: 7562 3e53 3c2f 7375 623e 2065 7374 696d  ub>S</sub> estim
-0000e0f0: 6174 696f 6e2e 2054 6865 206f 7074 696f  ation. The optio
-0000e100: 6e20 602d 2d70 6169 7277 6973 6560 2069  n `--pairwise` i
-0000e110: 7320 6120 7665 7279 2069 6d70 6f72 7461  s a very importa
-0000e120: 6e74 2070 6172 616d 6574 6572 2066 6f72  nt parameter for
-0000e130: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-0000e140: 7375 623e 2065 7374 696d 6174 696f 6e2c  sub> estimation,
-0000e150: 2077 6974 6820 7768 6963 6820 7468 6520   with which the 
-0000e160: 6043 4f44 454d 4c60 2077 696c 6c20 6361  `CODEML` will ca
-0000e170: 6c63 756c 6174 6520 7468 6520 2a4b 2a3c  lculate the *K*<
-0000e180: 7375 623e 533c 2f73 7562 3e20 666f 7220  sub>S</sub> for 
-0000e190: 6561 6368 2067 656e 6520 7061 6972 2073  each gene pair s
-0000e1a0: 6570 6172 6174 656c 7920 6261 7365 6420  eparately based 
-0000e1b0: 6f6e 2074 6865 2061 6c69 676e 6d65 6e74  on the alignment
-0000e1c0: 206f 6620 6f6e 6c79 2074 6865 7365 2074   of only these t
-0000e1d0: 776f 2067 656e 6573 2069 6e73 7465 6164  wo genes instead
-0000e1e0: 206f 6620 7468 6520 7768 6f6c 6520 616c   of the whole al
-0000e1f0: 6967 6e6d 656e 7420 6f66 2074 6865 2066  ignment of the f
-0000e200: 616d 696c 792c 2073 7563 6820 7468 6174  amily, such that
-0000e210: 206c 6573 7320 6761 7073 2061 7265 2065   less gaps are e
-0000e220: 7870 6563 7465 6420 616e 6420 7468 7573  xpected and thus
-0000e230: 2074 6865 2061 6c69 676e 6d65 6e74 2069   the alignment i
-0000e240: 6e20 7468 6520 636f 6e73 6964 6572 6174  n the considerat
-0000e250: 696f 6e20 6f66 2060 434f 4445 4d4c 6020  ion of `CODEML` 
-0000e260: 7769 6c6c 2062 6520 6c6f 6e67 6572 2028  will be longer (
-0000e270: 6265 6361 7573 6520 6043 4f44 454d 4c60  because `CODEML`
-0000e280: 2077 696c 6c20 6175 746f 6d61 7465 6c79   will automately
-0000e290: 2073 6b69 7020 6576 6572 7920 636f 6c75   skip every colu
-0000e2a0: 6d6e 2077 6974 6820 6761 702c 2072 6567  mn with gap, reg
-0000e2b0: 6172 646c 6573 7320 6f66 2077 6865 7468  ardless of wheth
-0000e2c0: 6572 2069 7427 7320 616e 206f 7665 7261  er it's an overa
-0000e2d0: 6c6c 206f 7220 7061 7274 6961 6c20 6761  ll or partial ga
-0000e2e0: 7029 2c20 7769 7468 6f75 7420 7768 6963  p), without whic
-0000e2f0: 6820 7468 6520 6043 4f44 454d 4c60 2077  h the `CODEML` w
-0000e300: 696c 6c20 6361 6c63 756c 6174 6520 7468  ill calculate th
-0000e310: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-0000e320: 3e20 6261 7365 6420 6f6e 2074 6865 2077  > based on the w
-0000e330: 686f 6c65 2061 6c69 676e 6d65 6e74 206f  hole alignment o
-0000e340: 6620 7468 6520 6661 6d69 6c79 2c20 7768  f the family, wh
-0000e350: 6963 6820 6d69 6768 7420 6861 7665 206e  ich might have n
-0000e360: 6f20 2a4b 2a3c 7375 623e 533c 2f73 7562  o *K*<sub>S</sub
-0000e370: 3e20 7265 7375 6c74 2061 7420 616c 6c20  > result at all 
-0000e380: 6966 2074 6865 2073 7472 6970 7065 6420  if the stripped 
-0000e390: 616c 6967 6e6d 656e 7420 6c65 6e67 7468  alignment length
-0000e3a0: 2028 7265 6d6f 7669 6e67 2061 6c6c 2067   (removing all g
-0000e3b0: 6170 2d63 6f6e 7461 696e 696e 6720 636f  ap-containing co
-0000e3c0: 6c75 6d6e 7329 2077 6173 207a 6572 6f2c  lumns) was zero,
-0000e3d0: 2061 2063 6175 7365 206f 6620 6469 6666   a cause of diff
-0000e3e0: 6572 656e 7420 6e75 6d62 6572 206f 6620  erent number of 
-0000e3f0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000e400: 6573 7469 6d61 7465 7320 6265 7477 6565  estimates betwee
-0000e410: 6e20 2270 6169 7277 6973 6520 6d6f 6465  n "pairwise mode
-0000e420: 2220 616e 6420 226e 6f6e 2d70 6169 7277  " and "non-pairw
-0000e430: 6973 6520 6d6f 6465 222e 2049 7427 7320  ise mode". It's 
-0000e440: 6469 6666 6963 756c 7420 746f 2073 6179  difficult to say
-0000e450: 2077 6869 6368 206d 6f64 6520 6973 206d   which mode is m
-0000e460: 6f72 6520 6964 6561 6c2c 2061 6c74 686f  ore ideal, altho
-0000e470: 7567 6820 7468 6520 226e 6f6e 2d70 6169  ugh the "non-pai
-0000e480: 7277 6973 6520 6d6f 6465 2220 2864 6566  rwise mode" (def
-0000e490: 6175 6c74 2073 6574 7469 6e67 2920 7768  ault setting) wh
-0000e4a0: 6963 6820 7275 6e73 206f 6e20 7468 6520  ich runs on the 
-0000e4b0: 7768 6f6c 6520 616c 6967 6e6d 656e 7420  whole alignment 
-0000e4c0: 696e 7374 6561 6420 6f66 2061 206c 6f63  instead of a loc
-0000e4d0: 616c 2061 6c69 676e 6d65 6e74 2c20 6d69  al alignment, mi
-0000e4e0: 6768 7420 6265 206d 6f72 6520 6269 6f6c  ght be more biol
-0000e4f0: 6f67 6963 616c 6c79 2063 6f6e 7365 7276  ogically conserv
-0000e500: 6564 2069 6e20 7468 6174 2069 7420 6173  ed in that it as
-0000e510: 7375 7265 7320 7468 6520 6576 6f6c 7574  sures the evolut
-0000e520: 696f 6e20 6f66 2065 6163 6820 636f 6c75  ion of each colu
-0000e530: 6d6e 2074 6f20 6265 2073 7461 7274 6564  mn to be started
-0000e540: 2066 726f 6d20 7468 6520 726f 6f74 206f   from the root o
-0000e550: 6620 7468 6520 6661 6d69 6c79 2061 6e64  f the family and
-0000e560: 2061 6c6c 2074 6865 2067 656e 6520 6475   all the gene du
-0000e570: 706c 6963 6174 6573 2061 7265 2074 616b  plicates are tak
-0000e580: 656e 2069 6e74 6f20 6163 636f 756e 7420  en into account 
-0000e590: 696e 2074 6865 202a 4b2a 3c73 7562 3e53  in the *K*<sub>S
-0000e5a0: 3c2f 7375 623e 2065 7374 696d 6174 696f  </sub> estimatio
-0000e5b0: 6e20 7072 6f63 6573 732e 2054 6865 206f  n process. The o
-0000e5c0: 7074 696f 6e20 602d 2d73 7472 6970 5f67  ption `--strip_g
-0000e5d0: 6170 7360 2063 616e 2072 656d 6f76 6520  aps` can remove 
-0000e5e0: 616c 6c20 7468 6520 6761 702d 636f 6e74  all the gap-cont
-0000e5f0: 6169 6e69 6e67 2063 6f6c 756d 6e73 2c20  aining columns, 
-0000e600: 7769 7468 206f 7220 7769 7468 6f75 7420  with or without 
-0000e610: 7768 6963 6820 7468 6520 7265 7375 6c74  which the result
-0000e620: 206f 6620 226e 6f6e 2d70 6169 7277 6973   of "non-pairwis
-0000e630: 6520 6d6f 6465 2220 776f 6e27 7420 6265  e mode" won't be
-0000e640: 2061 6666 6563 7465 642c 2077 6869 6c65   affected, while
-0000e650: 2077 6974 6820 7768 6963 6820 7468 6520   with which the 
-0000e660: 7265 7375 6c74 206f 6620 2270 6169 7277  result of "pairw
-0000e670: 6973 6520 6d6f 6465 2220 7769 6c6c 2062  ise mode" will b
-0000e680: 6520 616c 7465 7265 642e 2054 6865 206f  e altered. The o
-0000e690: 7074 696f 6e20 602d 2d61 6c69 676e 6572  ption `--aligner
-0000e6a0: 6020 616e 6420 602d 2d61 6c6e 5f6f 7074  ` and `--aln_opt
-0000e6b0: 696f 6e73 6020 7768 6963 6820 6465 6369  ions` which deci
-0000e6c0: 6465 2077 6869 6368 2061 6c69 676e 6d65  de which alignme
-0000e6d0: 6e74 2070 726f 6772 616d 2074 6f20 6265  nt program to be
-0000e6e0: 2075 7365 6420 616e 6420 7768 6963 6820   used and which 
-0000e6f0: 7061 7261 6d65 7465 7220 746f 2062 6520  parameter to be 
-0000e700: 7365 742c 2077 696c 6c20 6861 7665 2069  set, will have i
-0000e710: 6d70 6163 7420 6f6e 2074 6865 202a 4b2a  mpact on the *K*
-0000e720: 3c73 7562 3e53 3c2f 7375 623e 2072 6573  <sub>S</sub> res
-0000e730: 756c 7473 2c20 6e6f 7465 6420 7468 6174  ults, noted that
-0000e740: 2074 6865 2064 6566 6175 6c74 2070 726f   the default pro
-0000e750: 6772 616d 2069 7320 606d 6166 6674 6020  gram is `mafft` 
-0000e760: 616e 6420 7468 6520 7061 7261 6d65 7465  and the paramete
-0000e770: 7220 6973 2060 2d2d 6175 746f 602e 2054  r is `--auto`. T
-0000e780: 6865 206f 7074 696f 6e20 602d 2d74 7265  he option `--tre
-0000e790: 655f 6d65 7468 6f64 6020 616e 6420 602d  e_method` and `-
-0000e7a0: 2d74 7265 655f 6f70 7469 6f6e 7360 2064  -tree_options` d
-0000e7b0: 6563 6964 6520 7768 6963 6820 6765 6e65  ecide which gene
-0000e7c0: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
-0000e7d0: 7072 6f67 7261 6d20 746f 2062 6520 7573  program to be us
-0000e7e0: 6564 2061 6e64 2077 6869 6368 2070 6172  ed and which par
-0000e7f0: 616d 6574 6572 2074 6f20 6265 2073 6574  ameter to be set
-0000e800: 2c20 776f 6e27 7420 6166 6665 6374 2074  , won't affect t
-0000e810: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
-0000e820: 623e 2065 7374 696d 6174 696f 6e20 6974  b> estimation it
-0000e830: 7365 6c66 2062 7574 2074 6865 2072 6573  self but the res
-0000e840: 756c 7420 6f66 2064 652d 7265 6475 6e64  ult of de-redund
-0000e850: 616e 6379 2c20 6e6f 7465 6420 7468 6174  ancy, noted that
-0000e860: 2077 6520 696d 706c 656d 656e 7465 6420   we implemented 
-0000e870: 6120 6275 696c 742d 696e 2067 656e 6520  a built-in gene 
-0000e880: 7472 6565 2069 6e66 6572 656e 6365 206d  tree inference m
-0000e890: 6574 686f 6420 6261 7365 6420 6f6e 2074  ethod based on t
-0000e8a0: 6865 2041 7665 7261 6765 204c 696e 6b61  he Average Linka
-0000e8b0: 6765 2043 6c75 7374 6572 696e 6720 2841  ge Clustering (A
-0000e8c0: 4c43 2920 2874 6875 7320 6120 6469 7374  LC) (thus a dist
-0000e8d0: 616e 6365 2d62 6173 6564 2074 7265 6529  ance-based tree)
-0000e8e0: 2077 6974 6820 7468 6520 602d 2d74 7265   with the `--tre
-0000e8f0: 655f 6d65 7468 6f64 6020 7365 7420 6173  e_method` set as
-0000e900: 2022 636c 7573 7465 7222 2e20 5477 6f20   "cluster". Two 
-0000e910: 6d65 7468 6f64 7320 6f66 2064 652d 7265  methods of de-re
-0000e920: 6475 6e64 616e 6379 2061 7265 2069 6d70  dundancy are imp
-0000e930: 6c65 6d65 6e74 6564 2069 6e20 6077 6764  lemented in `wgd
-0000e940: 2076 3260 2c20 6e61 6d65 6c79 206e 6f64   v2`, namely nod
-0000e950: 652d 7765 6967 6874 6564 2061 6e64 206e  e-weighted and n
-0000e960: 6f64 652d 6176 6572 6167 6564 206d 6574  ode-averaged met
-0000e970: 686f 6473 2e20 5468 6520 6e6f 6465 2d77  hods. The node-w
-0000e980: 6569 6768 7465 6420 6d65 7468 6f64 2061  eighted method a
-0000e990: 6368 6965 7665 7320 7468 6520 6465 2d72  chieves the de-r
-0000e9a0: 6564 756e 6461 6e63 7920 7669 6120 7765  edundancy via we
-0000e9b0: 6967 6869 6e67 2074 6865 202a 4b2a 3c73  ighing the *K*<s
-0000e9c0: 7562 3e53 3c2f 7375 623e 2076 616c 7565  ub>S</sub> value
-0000e9d0: 2061 7373 6f63 6961 7465 6420 7769 7468   associated with
-0000e9e0: 2065 6163 6820 6765 6e65 2070 6169 7220   each gene pair 
-0000e9f0: 7375 6368 2074 6861 7420 7468 6520 7765  such that the we
-0000ea00: 6967 6874 206f 6620 6120 7369 6e67 6c65  ight of a single
-0000ea10: 2064 7570 6c69 6361 7469 6f6e 2065 7665   duplication eve
-0000ea20: 6e74 2073 756d 7320 7570 2074 6f20 3120  nt sums up to 1 
-0000ea30: 286e 6f74 6564 2074 6861 7420 7468 6520  (noted that the 
-0000ea40: 6e75 6d62 6572 206f 6620 2a4b 2a3c 7375  number of *K*<su
-0000ea50: 623e 533c 2f73 7562 3e20 6573 7469 6d61  b>S</sub> estima
-0000ea60: 7469 6f6e 2072 656d 6169 6e73 2074 6865  tion remains the
-0000ea70: 2073 616d 6529 2077 6869 6c65 2074 6865   same) while the
-0000ea80: 206e 6f64 652d 6176 6572 6167 6564 206d   node-averaged m
-0000ea90: 6574 686f 6420 7265 616c 697a 6573 2074  ethod realizes t
-0000eaa0: 6865 2064 652d 7265 6475 6e64 616e 6379  he de-redundancy
-0000eab0: 2076 6961 2063 616c 6375 6c61 7469 6e67   via calculating
-0000eac0: 2070 6572 2067 656e 6520 7472 6565 206e   per gene tree n
-0000ead0: 6f64 6520 6f6e 6520 6176 6572 6167 6564  ode one averaged
-0000eae0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000eaf0: 2076 616c 7565 2074 6f20 7265 7072 6573   value to repres
-0000eb00: 656e 7420 7468 6520 6167 6520 6f66 2065  ent the age of e
-0000eb10: 6163 6820 6475 706c 6963 6174 696f 6e20  ach duplication 
-0000eb20: 6576 656e 742e 2054 6865 206f 7074 696f  event. The optio
-0000eb30: 6e20 602d 2d6e 6f64 655f 6176 6572 6167  n `--node_averag
-0000eb40: 6560 2063 616e 2062 6520 7365 7420 746f  e` can be set to
-0000eb50: 2063 686f 6f73 6520 7468 6520 6e6f 6465   choose the node
-0000eb60: 2d61 7665 7261 6765 6420 7761 7920 6f66  -averaged way of
-0000eb70: 2064 652d 7265 6475 6e64 616e 6379 2e20   de-redundancy. 
-0000eb80: 4469 6666 6572 656e 7420 6d65 7468 6f64  Different method
-0000eb90: 7320 6f66 2064 652d 7265 6475 6e64 616e  s of de-redundan
-0000eba0: 6379 2077 696c 6c20 6861 7665 2069 6d70  cy will have imp
-0000ebb0: 6163 7420 6f6e 2074 6865 2064 6574 6563  act on the detec
-0000ebc0: 7469 6f6e 206f 6620 5747 4420 7369 676e  tion of WGD sign
-0000ebd0: 616c 732c 2077 6869 6368 2068 6173 2062  als, which has b
-0000ebe0: 6565 6e20 696e 7665 7374 6967 6174 6564  een investigated
-0000ebf0: 2069 6e20 7468 6973 205b 6c69 7465 7261   in this [litera
-0000ec00: 7475 7265 5d28 6874 7470 733a 2f2f 646f  ture](https://do
-0000ec10: 692e 6f72 672f 3130 2e31 3039 332f 6762  i.org/10.1093/gb
-0000ec20: 652f 6576 7932 3030 292e 2054 6865 2070  e/evy200). The p
-0000ec30: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
-0000ec40: 6865 7265 2061 7265 2074 6f20 7061 7261  here are to para
-0000ec50: 6c6c 656c 697a 6520 7468 6520 616e 616c  llelize the anal
-0000ec60: 7973 6973 2066 6f72 2065 6163 6820 6765  ysis for each ge
-0000ec70: 6e65 2066 616d 696c 7920 616e 6420 7468  ne family and th
-0000ec80: 7573 2073 7567 6765 7374 6564 2074 6f20  us suggested to 
-0000ec90: 6265 2073 6574 2061 7320 6d75 6368 2061  be set as much a
-0000eca0: 7320 7468 6520 6e75 6d62 6572 206f 6620  s the number of 
-0000ecb0: 6765 6e65 2066 616d 696c 6965 732e 0a0a  gene families...
-0000ecc0: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
-0000ecd0: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
-0000ece0: 7365 2063 6f6d 6d61 6e64 2061 7320 6265  se command as be
-0000ecf0: 6c6f 772a 2a0a 0a60 6060 0a77 6764 206b  low**..```.wgd k
-0000ed00: 7364 2066 616d 696c 6965 7320 7365 7175  sd families sequ
-0000ed10: 656e 6365 0a60 6060 0a0a 2a2a 5468 6520  ence.```..**The 
-0000ed20: 636f 6e73 7472 7563 7469 6f6e 206f 6620  construction of 
-0000ed30: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
-0000ed40: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-0000ed50: 6469 7374 7269 6275 7469 6f6e 2a2a 0a60  distribution**.`
-0000ed60: 6060 0a77 6764 206b 7364 2066 616d 696c  ``.wgd ksd famil
-0000ed70: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
-0000ed80: 7175 656e 6365 3220 282d 2d72 6577 6569  quence2 (--rewei
-0000ed90: 6768 7429 0a60 6060 0a0a 4672 6f6d 2070  ght).```..From p
-0000eda0: 6172 616c 6f67 6f75 7320 746f 206f 7274  aralogous to ort
-0000edb0: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
-0000edc0: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-0000edd0: 7472 6962 7574 696f 6e2c 2075 7365 7273  tribution, users
-0000ede0: 206f 6e6c 7920 6e65 6564 2074 6f20 7072   only need to pr
-0000edf0: 6f76 6964 6520 6d6f 7265 2063 6473 2066  ovide more cds f
-0000ee00: 696c 6573 2e20 4e6f 7465 2074 6861 7420  iles. Note that 
-0000ee10: 7769 7468 206f 7274 686f 6c6f 676f 7573  with orthologous
-0000ee20: 2067 656e 6520 6661 6d69 6c69 6573 2074   gene families t
-0000ee30: 6865 2077 6569 6768 7469 6e67 206d 6574  he weighting met
-0000ee40: 686f 6420 6361 6e20 6265 2073 6574 2074  hod can be set t
-0000ee50: 6f20 6265 2063 616c 6375 6c61 7465 6420  o be calculated 
-0000ee60: 7065 7220 7370 6563 6965 7320 7061 6972  per species pair
-0000ee70: 2069 6e73 7465 6164 206f 6620 636f 6e73   instead of cons
-0000ee80: 6964 6572 696e 6720 7468 6520 7768 6f6c  idering the whol
-0000ee90: 6520 6661 6d69 6c79 2062 6563 6175 7365  e family because
-0000eea0: 2077 6865 6e20 706c 6f74 7469 6e67 206f   when plotting o
-0000eeb0: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
-0000eec0: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-0000eed0: 6973 7472 6962 7574 696f 6e20 6265 7477  istribution betw
-0000eee0: 6565 6e20 7477 6f20 7370 6563 6965 7320  een two species 
-0000eef0: 7468 6520 7765 6967 6874 2063 616c 6375  the weight calcu
-0000ef00: 6c61 7465 6420 6672 6f6d 2074 6869 7320  lated from this 
-0000ef10: 7370 6563 6966 6963 2073 7065 6369 6573  specific species
-0000ef20: 2070 6169 7220 7368 6f75 6c64 2062 6520   pair should be 
-0000ef30: 636f 6e73 6572 7665 6420 7768 696c 6520  conserved while 
-0000ef40: 7468 6520 6f6e 6520 6361 6c63 756c 6174  the one calculat
-0000ef50: 6564 2066 726f 6d20 7468 6520 7768 6f6c  ed from the whol
-0000ef60: 6520 6661 6d69 6c79 2077 696c 6c20 7661  e family will va
-0000ef70: 7279 2077 6974 6820 7468 6520 6e75 6d62  ry with the numb
-0000ef80: 6572 206f 6620 7370 6563 6965 732e 2054  er of species. T
-0000ef90: 6f20 696e 6974 6961 7465 2074 6865 2077  o initiate the w
-0000efa0: 6569 6768 7469 6e67 2070 6572 2073 7065  eighting per spe
-0000efb0: 6369 6573 2070 6169 722c 2074 6865 206f  cies pair, the o
-0000efc0: 7074 696f 6e20 602d 2d72 6577 6569 6768  ption `--reweigh
-0000efd0: 7460 2063 616e 2062 6520 7365 742e 0a0a  t` can be set...
-0000efe0: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
-0000eff0: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
-0000f000: 7365 2063 6f6d 6d61 6e64 2061 7320 6265  se command as be
-0000f010: 6c6f 772a 2a0a 0a60 6060 0a77 6764 206b  low**..```.wgd k
-0000f020: 7364 2066 616d 696c 6965 7320 7365 7175  sd families sequ
-0000f030: 656e 6365 3120 7365 7175 656e 6365 320a  ence1 sequence2.
-0000f040: 6060 600a 0a2a 2a54 6865 2063 6f6e 7374  ```..**The const
-0000f050: 7275 6374 696f 6e20 6f66 202a 4b2a 3c73  ruction of *K*<s
-0000f060: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-0000f070: 6973 7472 6962 7574 696f 6e20 7769 7468  istribution with
-0000f080: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-0000f090: 2a2a 0a60 6060 0a77 6764 206b 7364 2066  **.```.wgd ksd f
-0000f0a0: 616d 696c 6965 7320 7365 7175 656e 6365  amilies sequence
-0000f0b0: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
-0000f0c0: 656e 6365 3320 2d2d 666f 6375 7332 616c  ence3 --focus2al
-0000f0d0: 6c20 7365 7175 656e 6365 3120 2d73 7020  l sequence1 -sp 
-0000f0e0: 7370 6461 7461 0a60 6060 0a0a 496e 7370  spdata.```..Insp
-0000f0f0: 6972 6564 2062 7920 7468 6520 7261 7465  ired by the rate
-0000f100: 2063 6f72 7265 6374 696f 6e20 616c 676f   correction algo
-0000f110: 7269 7468 6d20 696e 205b 6b73 7261 7465  rithm in [ksrate
-0000f120: 735d 2868 7474 7073 3a2f 2f67 6974 6875  s](https://githu
-0000f130: 622e 636f 6d2f 5649 422d 5053 422f 6b73  b.com/VIB-PSB/ks
-0000f140: 7261 7465 7329 2c20 7765 2069 6d70 6c65  rates), we imple
-0000f150: 6d65 6e74 6564 2074 6865 2072 6174 6520  mented the rate 
-0000f160: 636f 7272 6563 7469 6f6e 2061 6e61 6c79  correction analy
-0000f170: 7369 7320 616c 736f 2069 6e20 6077 6764  sis also in `wgd
-0000f180: 2076 3260 2c20 7768 6963 6820 6973 206d   v2`, which is m
-0000f190: 6f73 746c 7920 7468 6520 7361 6d65 2061  ostly the same a
-0000f1a0: 7320 5b6b 7372 6174 6573 5d28 6874 7470  s [ksrates](http
-0000f1b0: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f56  s://github.com/V
-0000f1c0: 4942 2d50 5342 2f6b 7372 6174 6573 2920  IB-PSB/ksrates) 
-0000f1d0: 6275 7420 6469 6666 6572 7320 696e 2074  but differs in t
-0000f1e0: 6865 2063 616c 6375 6c61 7469 6f6e 206f  he calculation o
-0000f1f0: 6620 7468 6520 7374 616e 6461 7264 2064  f the standard d
-0000f200: 6576 6961 7469 6f6e 206f 6620 7265 7363  eviation of resc
-0000f210: 616c 6564 202a 4b2a 3c73 7562 3e53 3c2f  aled *K*<sub>S</
-0000f220: 7375 623e 2061 6765 732e 2054 6f20 7065  sub> ages. To pe
-0000f230: 7266 6f72 6d20 7468 6520 7261 7465 2063  rform the rate c
-0000f240: 6f72 7265 6374 696f 6e20 616e 616c 7973  orrection analys
-0000f250: 6973 2c20 7573 6572 7320 6361 6e20 7573  is, users can us
-0000f260: 6520 626f 7468 2074 6865 2060 7767 6420  e both the `wgd 
-0000f270: 6b73 6460 2061 6e64 2060 7767 6420 7669  ksd` and `wgd vi
-0000f280: 7a60 2070 726f 6772 616d 2e20 466f 7220  z` program. For 
-0000f290: 7468 6520 6077 6764 206b 7364 6020 7072  the `wgd ksd` pr
-0000f2a0: 6f67 7261 6d2c 2075 7365 7273 206e 6565  ogram, users nee
-0000f2b0: 6420 746f 2070 726f 7669 6465 2061 2073  d to provide a s
-0000f2c0: 7065 6369 6573 2074 7265 6520 7669 6120  pecies tree via 
-0000f2d0: 7468 6520 6f70 7469 6f6e 2060 2d2d 7370  the option `--sp
-0000f2e0: 6563 6965 7374 7265 6560 206f 6e20 7768  eciestree` on wh
-0000f2f0: 6963 6820 7261 7465 2063 6f72 7265 6374  ich rate correct
-0000f300: 696f 6e20 6361 6e20 6265 2063 6f6e 6475  ion can be condu
-0000f310: 6374 6564 2e20 4e6f 7465 2074 6861 7420  cted. Note that 
-0000f320: 756e 6e65 6365 7373 6172 7920 6272 6163  unnecessary brac
-0000f330: 6b65 7473 206d 6967 6874 206c 6561 6420  kets might lead 
-0000f340: 746f 2075 6e65 7870 6563 7465 6420 6572  to unexpected er
-0000f350: 726f 7273 2c20 666f 7220 696e 7374 616e  rors, for instan
-0000f360: 6365 2061 2074 7265 6520 6028 412c 2842  ce a tree `(A,(B
-0000f370: 2c43 2929 3b60 2073 686f 756c 6420 6e6f  ,C));` should no
-0000f380: 7420 6265 2072 6570 7265 7365 6e74 6564  t be represented
-0000f390: 2061 7320 6028 412c 2828 422c 4329 2929   as `(A,((B,C)))
-0000f3a0: 3b60 2e20 5468 6520 7365 7420 6f66 2073  ;`. The set of s
-0000f3b0: 7065 6369 6573 2070 6169 7273 2074 6f20  pecies pairs to 
-0000f3c0: 6265 2073 686f 776e 2069 7320 666c 6578  be shown is flex
-0000f3d0: 6962 6c65 2074 6861 7420 7468 6520 6d6f  ible that the mo
-0000f3e0: 7374 2063 6f6e 7665 6e69 656e 7420 6f70  st convenient op
-0000f3f0: 7469 6f6e 2069 7320 602d 2d66 6f63 7573  tion is `--focus
-0000f400: 3261 6c6c 6020 7768 6963 6820 7369 6d70  2all` which simp
-0000f410: 6c79 2073 686f 7773 2061 6c6c 2074 6865  ly shows all the
-0000f420: 2070 6f73 7369 626c 6520 666f 6361 6c2d   possible focal-
-0000f430: 7369 7374 6572 2073 7065 6369 6573 2070  sister species p
-0000f440: 6169 7273 2c20 6f72 2075 7365 7273 2063  airs, or users c
-0000f450: 616e 206d 616e 7561 6c6c 7920 7365 7420  an manually set 
-0000f460: 7468 6520 7370 6563 6965 7320 7061 6972  the species pair
-0000f470: 7320 7669 6120 7468 6520 6f70 7469 6f6e  s via the option
-0000f480: 2060 2d2d 7370 6169 7260 2e20 4e6f 7465   `--spair`. Note
-0000f490: 2074 6861 7420 6966 2074 6865 2073 7065   that if the spe
-0000f4a0: 6369 6573 2070 6169 7273 2077 6572 6520  cies pairs were 
-0000f4b0: 6d61 6e75 616c 6c79 2073 6574 2c20 6974  manually set, it
-0000f4c0: 2077 6f75 6c64 2062 6520 6e65 6564 6564   would be needed
-0000f4d0: 2074 6f20 7365 7420 7468 6520 6f70 7469   to set the opti
-0000f4e0: 6f6e 2060 2d2d 636c 6173 7369 6360 2e20  on `--classic`. 
-0000f4f0: 5468 6520 6f70 7469 6f6e 2060 6f6e 6c79  The option `only
-0000f500: 726f 6f74 6f75 7460 2063 616e 2062 6520  rootout` can be 
-0000f510: 7365 7420 746f 206f 6e6c 7920 636f 6e73  set to only cons
-0000f520: 6964 6572 206f 7574 6772 6f75 7020 6174  ider outgroup at
-0000f530: 2074 6865 2072 6f6f 742c 2069 6e73 7465   the root, inste
-0000f540: 6164 206f 6620 616c 6c20 7468 6520 706f  ad of all the po
-0000f550: 7373 6962 6c65 206f 7574 6772 6f75 7073  ssible outgroups
-0000f560: 2070 6572 2066 6f63 616c 2d73 6973 7465   per focal-siste
-0000f570: 7220 7370 6563 6965 7320 7061 6972 2c20  r species pair, 
-0000f580: 7768 6963 6820 6861 7320 696d 7061 6374  which has impact
-0000f590: 206f 6e20 7468 6520 6669 6e61 6c20 636f   on the final co
-0000f5a0: 7272 6563 7465 6420 2a4b 2a3c 7375 623e  rrected *K*<sub>
-0000f5b0: 533c 2f73 7562 3e20 6167 6573 2e20 536f  S</sub> ages. So
-0000f5c0: 6d65 206f 7468 6572 206f 7074 696f 6e73  me other options
-0000f5d0: 2077 6869 6368 2068 6176 6520 6e6f 2069   which have no i
-0000f5e0: 6d70 6163 7420 6f6e 2074 6865 2072 6174  mpact on the rat
-0000f5f0: 6520 636f 7272 6563 7469 6f6e 2062 7574  e correction but
-0000f600: 2061 6464 206d 6f72 6520 6c61 7965 7273   add more layers
-0000f610: 206f 7220 6368 616e 6765 2074 6865 2061   or change the a
-0000f620: 7070 6561 7261 6e63 6520 6f6e 2074 6865  ppearance on the
-0000f630: 206d 6978 6564 202a 4b2a 3c73 7562 3e53   mixed *K*<sub>S
-0000f640: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-0000f650: 6962 7574 696f 6e2c 2069 6e63 6c75 6465  ibution, include
-0000f660: 2060 706c 6f74 6170 676d 6d60 2061 6e64   `plotapgmm` and
-0000f670: 2060 706c 6f74 656c 6d6d 6020 6574 632e   `plotelmm` etc.
-0000f680: 2054 6865 206f 7074 696f 6e20 6070 6c6f   The option `plo
-0000f690: 7461 7067 6d6d 6020 6361 6e20 6265 2073  tapgmm` can be s
-0000f6a0: 6574 2074 6f20 6361 6c6c 2074 6865 2047  et to call the G
-0000f6b0: 4d4d 2061 6e61 6c79 7369 7320 7570 6f6e  MM analysis upon
-0000f6c0: 2074 6865 2061 6e63 686f 7220 7061 6972   the anchor pair
-0000f6d0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000f6e0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-0000f6f0: 6e20 616e 6420 706c 6f74 2074 6865 2063  n and plot the c
-0000f700: 6c75 7374 6572 696e 6720 7265 7375 6c74  lustering result
-0000f710: 2075 706f 6e20 7468 6520 6d69 7865 6420   upon the mixed 
-0000f720: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000f730: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
-0000f740: 2c20 7768 6963 6820 6861 7320 746f 2062  , which has to b
-0000f750: 6520 636f 2d73 6574 2077 6974 6820 7468  e co-set with th
-0000f760: 6520 6f70 7469 6f6e 2060 2d2d 616e 6368  e option `--anch
-0000f770: 6f72 706f 696e 7473 6020 7072 6f76 6964  orpoints` provid
-0000f780: 696e 6720 7468 6520 616e 6368 6f72 2070  ing the anchor p
-0000f790: 6169 7273 2069 6e66 6f72 6d61 7469 6f6e  airs information
-0000f7a0: 2e20 5468 6520 6f70 7469 6f6e 2060 2d2d  . The option `--
-0000f7b0: 706c 6f74 656c 6d6d 6020 6361 6e20 6265  plotelmm` can be
-0000f7c0: 2073 6574 2074 6f20 6361 6c6c 2074 6865   set to call the
-0000f7d0: 2045 4c4d 4d20 616e 616c 7973 6973 2075   ELMM analysis u
-0000f7e0: 706f 6e20 7468 6520 7061 7261 6c6f 676f  pon the paralogo
-0000f7f0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-0000f800: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-0000f810: 696f 6e2e 0a0a 5468 6572 6520 6172 6520  ion...There are 
-0000f820: 3231 2063 6f6c 756d 6e73 2069 6e20 7468  21 columns in th
-0000f830: 6520 7265 7375 6c74 2060 2a6b 732e 7473  e result `*ks.ts
-0000f840: 7660 2066 696c 6520 6265 7369 6465 7320  v` file besides 
-0000f850: 7468 6520 696e 6465 7820 636f 6c75 6d6e  the index column
-0000f860: 7320 6070 6169 7260 2061 7320 7468 6520  s `pair` as the 
-0000f870: 756e 6971 7565 2069 6465 6e74 6966 6965  unique identifie
-0000f880: 7220 666f 7220 6561 6368 2067 656e 6520  r for each gene 
-0000f890: 7061 6972 2e20 5468 6520 604e 602c 2060  pair. The `N`, `
-0000f8a0: 5360 2c20 6064 4e60 2c20 6064 4e2f 6453  S`, `dN`, `dN/dS
-0000f8b0: 602c 2060 6453 602c 2060 6c60 2061 6e64  `, `dS`, `l` and
-0000f8c0: 2060 7460 2061 7265 2066 726f 6d20 7468   `t` are from th
-0000f8d0: 6520 636f 6465 6d6c 2072 6573 756c 7473  e codeml results
-0000f8e0: 2c20 7265 7072 6573 656e 7469 6e67 2074  , representing t
-0000f8f0: 6865 204e 2065 7374 696d 6174 652c 2074  he N estimate, t
-0000f900: 6865 2053 2065 7374 696d 6174 652c 2074  he S estimate, t
-0000f910: 6865 2064 4e20 6573 7469 6d61 7465 2c20  he dN estimate, 
-0000f920: 7468 6520 644e 2f64 5320 286f 6d65 6761  the dN/dS (omega
-0000f930: 2920 6573 7469 6d61 7465 2c20 7468 6520  ) estimate, the 
-0000f940: 6453 2065 7374 696d 6174 652c 2074 6865  dS estimate, the
-0000f950: 206c 6f67 2d6c 696b 656c 6968 6f6f 6420   log-likelihood 
-0000f960: 616e 6420 7468 6520 7420 6573 7469 6d61  and the t estima
-0000f970: 7465 2c20 7265 7370 6563 7469 7665 6c79  te, respectively
-0000f980: 2e20 5468 6520 6061 6c69 676e 6d65 6e74  . The `alignment
-0000f990: 636f 7665 7261 6765 602c 2060 616c 6967  coverage`, `alig
-0000f9a0: 6e6d 656e 7469 6465 6e74 6974 7960 2061  nmentidentity` a
-0000f9b0: 6e64 2060 616c 6967 6e6d 656e 746c 656e  nd `alignmentlen
-0000f9c0: 6774 6860 2061 7265 2074 6865 2069 6e66  gth` are the inf
-0000f9d0: 6f72 6d61 7469 6f6e 2070 6572 7461 696e  ormation pertain
-0000f9e0: 696e 6720 746f 2074 6865 2061 6c69 676e  ing to the align
-0000f9f0: 6d65 6e74 2066 6f72 2065 6163 6820 6661  ment for each fa
-0000fa00: 6d69 6c79 2c20 7265 7072 6573 656e 7469  mily, representi
-0000fa10: 6e67 2074 6865 2072 6174 696f 206f 6620  ng the ratio of 
-0000fa20: 7468 6520 7374 7269 7070 6564 2061 6c69  the stripped ali
-0000fa30: 676e 6d65 6e74 206c 656e 6774 6820 636f  gnment length co
-0000fa40: 6d70 6172 6564 2074 6f20 7468 6520 6675  mpared to the fu
-0000fa50: 6c6c 2061 6c69 676e 6d65 6e74 206c 656e  ll alignment len
-0000fa60: 6774 682c 2074 6865 2072 6174 696f 206f  gth, the ratio o
-0000fa70: 6620 636f 6c75 6d6e 7320 7769 7468 2069  f columns with i
-0000fa80: 6465 6e74 6963 616c 206e 7563 6c65 6f74  dentical nucleot
-0000fa90: 6964 6573 2063 6f6d 7061 7265 6420 746f  ides compared to
-0000faa0: 2074 6865 206f 7665 7261 6c6c 2063 6f6c   the overall col
-0000fab0: 756d 6e73 206f 6620 7468 6520 7374 7269  umns of the stri
-0000fac0: 7070 6564 2061 6c69 676e 6d65 6e74 2c20  pped alignment, 
-0000fad0: 616e 6420 7468 6520 6c65 6e67 7468 206f  and the length o
-0000fae0: 6620 7468 6520 6675 6c6c 2061 6c69 676e  f the full align
-0000faf0: 6d65 6e74 2c20 7265 7370 6563 7469 7665  ment, respective
-0000fb00: 6c79 2e0a 2323 2320 7767 6420 6d69 780a  ly..### wgd mix.
-0000fb10: 0a2a 2a54 6865 206d 6978 7475 7265 206d  .**The mixture m
-0000fb20: 6f64 656c 2063 6c75 7374 6572 696e 6720  odel clustering 
-0000fb30: 616e 616c 7973 6973 206f 6620 2a4b 2a3c  analysis of *K*<
-0000fb40: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-0000fb50: 6469 7374 7269 6275 7469 6f6e 2a2a 0a60  distribution**.`
-0000fb60: 6060 0a77 6764 206d 6978 206b 7364 6174  ``.wgd mix ksdat
-0000fb70: 6120 282d 2d6e 5f69 6e69 7420 3230 3020  a (--n_init 200 
-0000fb80: 2d2d 6d61 785f 6974 6572 2032 3030 202d  --max_iter 200 -
-0000fb90: 2d6b 735f 7261 6e67 6520 3020 3520 2d2d  -ks_range 0 5 --
-0000fba0: 6669 6c74 6572 7320 3330 3020 2d2d 6269  filters 300 --bi
-0000fbb0: 6e73 2035 3020 2d2d 636f 6d70 6f6e 656e  ns 50 --componen
-0000fbc0: 7473 2031 2034 202d 2d67 616d 6d61 2030  ts 1 4 --gamma 0
-0000fbd0: 2e30 3031 290a 6060 600a 0a54 6869 7320  .001).```..This 
-0000fbe0: 7061 7274 206f 6620 4761 7573 7369 616e  part of Gaussian
-0000fbf0: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
-0000fc00: 6720 2847 4d4d 2920 616e 616c 7973 6973  g (GMM) analysis
-0000fc10: 2069 7320 696e 6865 7269 7465 6420 6672   is inherited fr
-0000fc20: 6f6d 2074 6865 206f 7269 6769 6e61 6c20  om the original 
-0000fc30: 6077 6764 6020 7072 6f67 7261 6d2c 2062  `wgd` program, b
-0000fc40: 7574 2077 7269 7465 7320 6164 6469 7469  ut writes additi
-0000fc50: 6f6e 616c 6c79 2074 6865 2070 726f 6261  onally the proba
-0000fc60: 6269 6c69 7479 206f 6620 6561 6368 202a  bility of each *
-0000fc70: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2076  K*<sub>S</sub> v
-0000fc80: 616c 7565 2069 6e74 6f20 7468 6520 6669  alue into the fi
-0000fc90: 6e61 6c20 6461 7461 6672 616d 652e 2042  nal dataframe. B
-0000fca0: 6173 6963 616c 6c79 2c20 7573 6572 7320  asically, users 
-0000fcb0: 6e65 6564 2074 6f20 7072 6f76 6964 6520  need to provide 
-0000fcc0: 7769 7468 2061 2028 6e6f 726d 616c 6c79  with a (normally
-0000fcd0: 2066 726f 6d20 7768 6f6c 652d 7061 7261   from whole-para
-0000fce0: 6e6f 6d65 206f 7220 616e 6368 6f72 2d70  nome or anchor-p
-0000fcf0: 6169 7273 2920 2a4b 2a3c 7375 623e 533c  airs) *K*<sub>S<
-0000fd00: 2f73 7562 3e20 6461 7461 6669 6c65 2061  /sub> datafile a
-0000fd10: 6e64 2074 6865 2047 4d4d 2061 6e61 6c79  nd the GMM analy
-0000fd20: 7369 7320 7769 6c6c 2062 6520 636f 6e64  sis will be cond
-0000fd30: 7563 7465 6420 7570 6f6e 2074 6865 2064  ucted upon the d
-0000fd40: 6174 6166 696c 652e 2053 6f6d 6520 7061  atafile. Some pa
-0000fd50: 7261 6d65 7465 7273 2063 616e 2061 6666  rameters can aff
-0000fd60: 6563 7420 7468 6520 7265 7375 6c74 732c  ect the results,
-0000fd70: 2069 6e63 6c75 6469 6e67 2060 2d2d 6e5f   including `--n_
-0000fd80: 696e 6974 602c 2077 6869 6368 2073 6574  init`, which set
-0000fd90: 7320 7468 6520 6e75 6d62 6572 206f 6620  s the number of 
-0000fda0: 6b2d 6d65 616e 7320 696e 6974 6961 6c69  k-means initiali
-0000fdb0: 7a61 7469 6f6e 7320 2864 6566 6175 6c74  zations (default
-0000fdc0: 2032 3030 292c 2060 2d2d 6d61 785f 6974   200), `--max_it
-0000fdd0: 6572 602c 2077 6869 6368 2073 6574 7320  er`, which sets 
-0000fde0: 7468 6520 6d61 7869 6d75 6d20 6e75 6d62  the maximum numb
-0000fdf0: 6572 206f 6620 6974 6572 6174 696f 6e73  er of iterations
-0000fe00: 2028 6465 6661 756c 7420 3230 3029 2c20   (default 200), 
-0000fe10: 602d 2d6d 6574 686f 6460 2c20 7768 6963  `--method`, whic
-0000fe20: 6820 6465 7465 726d 696e 6573 2077 6869  h determines whi
-0000fe30: 6368 2063 6c75 7374 6572 696e 6720 6d65  ch clustering me
-0000fe40: 7468 6f64 2074 6f20 7573 6520 2864 6566  thod to use (def
-0000fe50: 6175 6c74 2067 6d6d 292c 2060 2d2d 6761  ault gmm), `--ga
-0000fe60: 6d6d 6160 2c20 7768 6963 6820 7365 7473  mma`, which sets
-0000fe70: 2074 6865 2067 616d 6d61 2070 6172 616d   the gamma param
-0000fe80: 6574 6572 2066 6f72 2074 6865 2062 676d  eter for the bgm
-0000fe90: 6d20 6d6f 6465 6c20 2864 6566 6175 6c74  m model (default
-0000fea0: 2030 2e30 3031 292c 2060 2d2d 636f 6d70   0.001), `--comp
-0000feb0: 6f6e 656e 7473 602c 2077 6869 6368 2073  onents`, which s
-0000fec0: 6574 7320 7468 6520 7261 6e67 6520 6f66  ets the range of
-0000fed0: 2074 6865 206e 756d 6265 7220 6f66 2063   the number of c
-0000fee0: 6f6d 706f 6e65 6e74 7320 746f 2066 6974  omponents to fit
-0000fef0: 2028 6465 6661 756c 7420 3120 3429 2c20   (default 1 4), 
-0000ff00: 616e 6420 7468 6520 6461 7461 2066 696c  and the data fil
-0000ff10: 7465 7269 6e67 2070 6172 616d 6574 6572  tering parameter
-0000ff20: 7320 602d 2d66 696c 7465 7273 6020 7768  s `--filters` wh
-0000ff30: 6963 6820 6669 6c74 6572 7320 6461 7461  ich filters data
-0000ff40: 2062 6173 6564 206f 6e20 616c 6967 6e6d   based on alignm
-0000ff50: 656e 7420 6c65 6e67 7468 2c20 602d 2d6b  ent length, `--k
-0000ff60: 735f 7261 6e67 6560 2077 6869 6368 2066  s_range` which f
-0000ff70: 696c 7465 7273 2064 6174 6120 6261 7365  ilters data base
-0000ff80: 6420 6f6e 202a 4b2a 3c73 7562 3e53 3c2f  d on *K*<sub>S</
-0000ff90: 7375 623e 2076 616c 7565 7320 616e 6420  sub> values and 
-0000ffa0: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
-0000ffb0: 2d62 696e 7360 2077 6869 6368 2073 6574  -bins` which set
-0000ffc0: 7320 7468 6520 6e75 6d62 6572 206f 6620  s the number of 
-0000ffd0: 6269 6e73 2069 6e20 2a4b 2a3c 7375 623e  bins in *K*<sub>
-0000ffe0: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
-0000fff0: 7469 6f6e 2028 6465 6661 756c 7420 3530  tion (default 50
-00010000: 292e 0a0a 2a2a 4120 7375 6767 6573 7465  )...**A suggeste
-00010010: 6420 7374 6172 7469 6e67 2072 756e 2063  d starting run c
-00010020: 616e 2075 7365 2063 6f6d 6d61 6e64 2073  an use command s
-00010030: 696d 706c 7920 6173 2062 656c 6f77 2a2a  imply as below**
-00010040: 0a0a 6060 600a 7767 6420 6d69 7820 6b73  ..```.wgd mix ks
-00010050: 6461 7461 0a60 6060 0a0a 2323 2320 7767  data.```..### wg
-00010060: 6420 7065 616b 0a0a 2a2a 5468 6520 7365  d peak..**The se
-00010070: 6172 6368 206f 6620 6372 6564 6961 626c  arch of crediabl
-00010080: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-00010090: 3e20 7261 6e67 6520 7573 6564 2069 6e20  > range used in 
-000100a0: 5747 4420 6461 7469 6e67 2a2a 0a60 6060  WGD dating**.```
-000100b0: 0a77 6764 2070 6561 6b20 6b73 6461 7461  .wgd peak ksdata
-000100c0: 202d 6170 2061 7064 6174 6120 2d73 6d20   -ap apdata -sm 
-000100d0: 736d 6461 7461 202d 6c65 206c 6564 6174  smdata -le ledat
-000100e0: 6120 2d6d 7020 6d70 6461 7461 202d 2d68  a -mp mpdata --h
-000100f0: 6575 7269 7374 6963 2028 2d2d 616c 6967  euristic (--alig
-00010100: 6e66 696c 7465 7220 302e 3020 3020 302e  nfilter 0.0 0 0.
-00010110: 3020 2d2d 6b73 7261 6e67 6520 3020 3520  0 --ksrange 0 5 
-00010120: 2d2d 6269 6e5f 7769 6474 6820 302e 3120  --bin_width 0.1 
-00010130: 2d2d 6775 6964 6520 7365 676d 656e 7420  --guide segment 
-00010140: 2d2d 7072 6f6d 696e 656e 6365 5f63 7574  --prominence_cut
-00010150: 6f66 6620 302e 3120 2d2d 7265 6c5f 6865  off 0.1 --rel_he
-00010160: 6967 6874 2030 2e34 202d 2d63 6920 3935  ight 0.4 --ci 95
-00010170: 202d 2d68 6472 2039 3520 2d2d 6b73 6375   --hdr 95 --kscu
-00010180: 746f 6666 2035 290a 6060 600a 0a41 7320  toff 5).```..As 
-00010190: 6d65 6e74 696f 6e65 6420 7072 6576 696f  mentioned previo
-000101a0: 7573 6c79 2c20 6120 6865 7572 6973 7469  usly, a heuristi
-000101b0: 6320 6d65 7468 6f64 2061 6e64 2061 2063  c method and a c
-000101c0: 6f6c 6c69 6e65 6172 2073 6567 6d65 6e74  ollinear segment
-000101d0: 732d 6775 6964 6564 2061 6e63 686f 7220  s-guided anchor 
-000101e0: 7061 6972 2063 6c75 7374 6572 696e 6720  pair clustering 
-000101f0: 666f 7220 7468 6520 7365 6172 6368 206f  for the search o
-00010200: 6620 6372 6564 6961 626c 6520 2a4b 2a3c  f crediable *K*<
-00010210: 7375 623e 533c 2f73 7562 3e20 7261 6e67  sub>S</sub> rang
-00010220: 6520 7573 6564 2069 6e20 5747 4420 6461  e used in WGD da
-00010230: 7469 6e67 2061 7265 2069 6d70 6c65 6d65  ting are impleme
-00010240: 6e74 6564 2069 6e20 6077 6764 2076 3260  nted in `wgd v2`
-00010250: 2e20 5573 6572 7320 6e65 6564 2074 6f20  . Users need to 
-00010260: 7072 6f76 6964 6520 7468 6520 616e 6368  provide the anch
-00010270: 6f72 2070 6f69 6e74 732c 2073 6567 6d65  or points, segme
-00010280: 6e74 732c 206c 6973 7473 6567 6d65 6e74  nts, listsegment
-00010290: 732c 206d 756c 7469 706c 6963 6f6e 7061  s, multipliconpa
-000102a0: 6972 7320 6461 7461 6669 6c65 2066 726f  irs datafile fro
-000102b0: 6d20 6069 2d61 6468 6f72 6560 2074 6f20  m `i-adhore` to 
-000102c0: 6163 6869 6576 6520 7468 6520 636c 7573  achieve the clus
-000102d0: 7465 7269 6e67 2066 756e 6374 696f 6e2e  tering function.
-000102e0: 2053 6f6d 6520 7061 7261 6d65 7465 7273   Some parameters
-000102f0: 2074 6861 7420 6361 6e20 696d 7061 6374   that can impact
-00010300: 2074 6865 2072 6573 756c 7473 2069 6e63   the results inc
-00010310: 6c75 6465 2060 2d2d 616c 6967 6e66 696c  lude `--alignfil
-00010320: 7465 7260 2c20 7768 6963 6820 6669 6c74  ter`, which filt
-00010330: 6572 7320 7468 6520 6461 7461 2062 6173  ers the data bas
-00010340: 6564 206f 6e20 616c 6967 6e6d 656e 7420  ed on alignment 
-00010350: 6964 656e 7469 7479 2c20 6c65 6e67 7468  identity, length
-00010360: 2061 6e64 2063 6f76 6572 6167 652c 2060   and coverage, `
-00010370: 2d2d 6b73 7261 6e67 6560 2c20 7768 6963  --ksrange`, whic
-00010380: 6820 7365 7473 2074 6865 2072 616e 6765  h sets the range
-00010390: 206f 6620 4b73 2074 6f20 6265 2061 6e61   of Ks to be ana
-000103a0: 6c79 7a65 642c 2060 2d2d 6269 6e5f 7769  lyzed, `--bin_wi
-000103b0: 6474 6860 2c20 7768 6963 6820 7365 7473  dth`, which sets
-000103c0: 2074 6865 2062 616e 6477 6964 7468 206f   the bandwidth o
-000103d0: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
-000103e0: 3e20 6469 7374 7269 6275 7469 6f6e 2c20  > distribution, 
-000103f0: 602d 2d77 6569 6768 7473 5f6f 7574 6c69  `--weights_outli
-00010400: 6572 735f 696e 636c 7564 6564 6020 7768  ers_included` wh
-00010410: 6963 6820 6465 7465 726d 696e 6573 2077  ich determines w
-00010420: 6865 7468 6572 2074 6f20 696e 636c 7564  hether to includ
-00010430: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-00010440: 3e20 6f75 746c 6965 7273 2028 7768 6f73  > outliers (whos
-00010450: 6520 7661 6c75 6520 6973 206f 7665 7220  e value is over 
-00010460: 3529 2069 6e20 616e 616c 7973 6973 2c20  5) in analysis, 
-00010470: 602d 2d6d 6574 686f 6460 2077 6869 6368  `--method` which
-00010480: 2064 6574 6572 6d69 6e65 7320 7768 6963   determines whic
-00010490: 6820 636c 7573 7465 7269 6e67 206d 6574  h clustering met
-000104a0: 686f 6420 746f 2075 7365 2028 6465 6661  hod to use (defa
-000104b0: 756c 7420 676d 6d29 2c20 602d 2d73 6565  ult gmm), `--see
-000104c0: 6460 2077 6869 6368 2073 6574 7320 7468  d` which sets th
-000104d0: 6520 7261 6e64 6f6d 2073 6565 6420 6769  e random seed gi
-000104e0: 7665 6e20 746f 2069 6e69 7469 616c 697a  ven to initializ
-000104f0: 6174 696f 6e20 2864 6566 6175 6c74 2032  ation (default 2
-00010500: 3335 3238 3930 292c 2060 2d2d 6e5f 696e  352890), `--n_in
-00010510: 6974 602c 2077 6869 6368 2073 6574 7320  it`, which sets 
-00010520: 7468 6520 6e75 6d62 6572 206f 6620 6b2d  the number of k-
-00010530: 6d65 616e 7320 696e 6974 6961 6c69 7a61  means initializa
-00010540: 7469 6f6e 7320 2864 6566 6175 6c74 2032  tions (default 2
-00010550: 3030 292c 2060 2d2d 656d 5f69 7465 7260  00), `--em_iter`
-00010560: 2c20 7768 6963 6820 7365 7473 2074 6865  , which sets the
-00010570: 206d 6178 696d 756d 206e 756d 6265 7220   maximum number 
-00010580: 6f66 2069 7465 7261 7469 6f6e 7320 2864  of iterations (d
-00010590: 6566 6175 6c74 2032 3030 292c 2060 2d2d  efault 200), `--
-000105a0: 6761 6d6d 6160 2c20 7768 6963 6820 7365  gamma`, which se
-000105b0: 7473 2074 6865 2067 616d 6d61 2070 6172  ts the gamma par
-000105c0: 616d 6574 6572 2066 6f72 2074 6865 2062  ameter for the b
-000105d0: 676d 6d20 6d6f 6465 6c20 2864 6566 6175  gmm model (defau
-000105e0: 6c74 2030 2e30 3031 292c 2060 2d2d 636f  lt 0.001), `--co
-000105f0: 6d70 6f6e 656e 7473 602c 2077 6869 6368  mponents`, which
-00010600: 2073 6574 7320 7468 6520 7261 6e67 6520   sets the range 
-00010610: 6f66 2074 6865 206e 756d 6265 7220 6f66  of the number of
-00010620: 2063 6f6d 706f 6e65 6e74 7320 746f 2066   components to f
-00010630: 6974 2028 6465 6661 756c 7420 3120 3429  it (default 1 4)
-00010640: 2c20 602d 2d77 6569 6768 7465 6460 2077  , `--weighted` w
-00010650: 6869 6368 2064 6574 6572 6d69 6e65 7320  hich determines 
-00010660: 7768 6574 6865 7220 746f 2075 7365 206e  whether to use n
-00010670: 6f64 652d 7765 6967 6874 6564 206d 6574  ode-weighted met
-00010680: 686f 6420 666f 7220 6465 2d72 6564 756e  hod for de-redun
-00010690: 6461 6e63 792c 2060 2d2d 6775 6964 6560  dancy, `--guide`
-000106a0: 2077 6869 6368 2064 6574 6572 6d69 6e65   which determine
-000106b0: 7320 7768 6963 6820 7265 6769 6d65 2072  s which regime r
-000106c0: 6573 6964 696e 6720 616e 6368 6f72 7320  esiding anchors 
-000106d0: 746f 2062 6520 7573 6564 2028 6465 6661  to be used (defa
-000106e0: 756c 7420 5365 676d 656e 7429 2c20 602d  ult Segment), `-
-000106f0: 2d70 726f 6d69 6e65 6e63 655f 6375 746f  -prominence_cuto
-00010700: 6666 6020 7768 6963 6820 7365 7473 2074  ff` which sets t
-00010710: 6865 2070 726f 6d69 6e65 6e63 6520 6375  he prominence cu
-00010720: 746f 6666 206f 6620 6163 6365 7074 6162  toff of acceptab
-00010730: 6c65 2070 6561 6b73 2069 6e20 7065 616b  le peaks in peak
-00010740: 2066 696e 6469 6e67 2070 726f 6365 7373   finding process
-00010750: 2c20 602d 2d72 656c 5f68 6569 6768 7460  , `--rel_height`
-00010760: 2077 6869 6368 2073 6574 7320 7468 6520   which sets the 
-00010770: 7265 6c61 7469 7665 2068 6569 6768 7420  relative height 
-00010780: 6174 2077 6869 6368 2074 6865 2070 6561  at which the pea
-00010790: 6b20 7769 6474 6820 6973 206d 6561 7375  k width is measu
-000107a0: 7265 642c 2060 2d2d 6b73 746f 6461 7465  red, `--kstodate
-000107b0: 6020 7768 6963 6820 6d61 6e75 616c 6c79  ` which manually
-000107c0: 2073 6574 7320 7468 6520 7261 6e67 6520   sets the range 
-000107d0: 6f66 202a 4b2a 3c73 7562 3e53 3c2f 7375  of *K*<sub>S</su
-000107e0: 623e 2074 6f20 6265 2064 6174 6564 2069  b> to be dated i
-000107f0: 6e20 6865 7572 6973 7469 6320 7365 6172  n heuristic sear
-00010800: 6368 2061 6e64 206e 6565 6473 2074 6f20  ch and needs to 
-00010810: 6265 2063 6f2d 7365 7420 7769 7468 206f  be co-set with o
-00010820: 7074 696f 6e20 602d 2d6d 616e 7561 6c73  ption `--manuals
-00010830: 6574 602c 2060 2d2d 786c 696d 6020 616e  et`, `--xlim` an
-00010840: 6420 602d 2d79 6c69 6d60 2064 6574 6572  d `--ylim` deter
-00010850: 6d69 6e69 6e67 2074 6865 2078 2061 6e64  mining the x and
-00010860: 2079 2061 7869 7320 6c69 6d69 7420 6f66   y axis limit of
-00010870: 2047 4d4d 202a 4b2a 3c73 7562 3e53 3c2f   GMM *K*<sub>S</
-00010880: 7375 623e 2064 6973 7472 6962 7574 696f  sub> distributio
-00010890: 6e2c 2060 2d2d 6369 6020 7365 7474 696e  n, `--ci` settin
-000108a0: 6720 7468 6520 636f 6e66 6964 656e 6365  g the confidence
-000108b0: 206c 6576 656c 206f 6620 6c6f 672d 6e6f   level of log-no
-000108c0: 726d 616c 2064 6973 7472 6962 7574 696f  rmal distributio
-000108d0: 6e20 746f 2064 6174 6520 2864 6566 6175  n to date (defau
-000108e0: 6c74 2039 3529 2c20 602d 2d68 6472 6020  lt 95), `--hdr` 
-000108f0: 7365 7474 696e 6720 7468 6520 6869 6768  setting the high
-00010900: 6573 7420 6465 6e73 6974 7920 7265 6769  est density regi
-00010910: 6f6e 2028 4844 5229 2061 7070 6c69 6564  on (HDR) applied
-00010920: 2069 6e20 7468 6520 7365 676d 656e 742d   in the segment-
-00010930: 6775 6964 6564 2061 6e63 686f 7220 7061  guided anchor pa
-00010940: 6972 202a 4b2a 3c73 7562 3e53 3c2f 7375  ir *K*<sub>S</su
-00010950: 623e 2064 6973 7472 6962 7574 696f 6e20  b> distribution 
-00010960: 2864 6566 6175 6c74 2039 3529 2c20 602d  (default 95), `-
-00010970: 2d68 6575 7269 7374 6963 6020 6465 7465  -heuristic` dete
-00010980: 726d 696e 696e 6720 7768 6574 6865 7220  rmining whether 
-00010990: 746f 2069 6e69 7469 6174 6520 6865 7572  to initiate heur
-000109a0: 6973 7469 6320 6d65 7468 6f64 206f 6620  istic method of 
-000109b0: 6465 6669 6e69 6e67 2043 4920 666f 7220  defining CI for 
-000109c0: 6461 7469 6e67 2c20 602d 2d6b 7363 7574  dating, `--kscut
-000109d0: 6f66 6660 2073 6574 7469 6e67 2074 6865  off` setting the
-000109e0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000109f0: 2073 6174 7572 6174 696f 6e20 6375 746f   saturation cuto
-00010a00: 6666 2069 6e20 6461 7469 6e67 2028 6465  ff in dating (de
-00010a10: 6661 756c 7420 3529 2e0a 0a46 6f75 7220  fault 5)...Four 
-00010a20: 7265 7375 6c74 2073 7562 666c 6f64 6572  result subfloder
-00010a30: 7320 7769 6c6c 2062 6520 7072 6f64 7563  s will be produc
-00010a40: 6564 2c20 6e61 6d65 6c79 2060 416e 6368  ed, namely `Anch
-00010a50: 6f72 4b73 5f46 696e 6450 6561 6b60 2c20  orKs_FindPeak`, 
-00010a60: 6041 6e63 686f 724b 735f 474d 4d60 2c20  `AnchorKs_GMM`, 
-00010a70: 6053 6567 6d65 6e74 4775 6964 654b 735f  `SegmentGuideKs_
-00010a80: 474d 4d60 2061 6e64 2060 5365 676d 656e  GMM` and `Segmen
-00010a90: 744b 735f 4669 6e64 5065 616b 602e 2054  tKs_FindPeak`. T
-00010aa0: 6865 2060 416e 6368 6f72 4b73 5f46 696e  he `AnchorKs_Fin
-00010ab0: 6450 6561 6b60 2073 7562 666c 6f64 6572  dPeak` subfloder
-00010ac0: 2063 6f6e 7461 696e 7320 7265 7375 6c74   contains result
-00010ad0: 7320 6f66 2074 6865 2064 6574 6563 7465  s of the detecte
-00010ae0: 6420 7065 616b 7320 6279 2074 6865 2060  d peaks by the `
-00010af0: 7369 676e 616c 6020 6d6f 6475 6c65 206f  signal` module o
-00010b00: 6620 6053 6369 5079 6020 6c69 6272 6172  f `SciPy` librar
-00010b10: 7920 616e 6420 7468 6520 6173 7375 6d65  y and the assume
-00010b20: 6420 6869 6768 6573 7420 6d61 7373 2070  d highest mass p
-00010b30: 6172 7420 2872 6566 6572 7265 6420 746f  art (referred to
-00010b40: 2061 7320 4869 6768 4d61 7373 2068 6572   as HighMass her
-00010b50: 6561 6674 6572 2920 6f66 2065 6163 6820  eafter) of each 
-00010b60: 7065 616b 2c20 7768 6963 6820 6361 6e20  peak, which can 
-00010b70: 6265 2075 7365 6420 666f 7220 6675 7274  be used for furt
-00010b80: 6865 7220 5747 4420 6461 7469 6e67 2e20  her WGD dating. 
-00010b90: 5468 6520 6041 6e63 686f 724b 735f 474d  The `AnchorKs_GM
-00010ba0: 4d60 2073 686f 7773 2074 6865 2047 4d4d  M` shows the GMM
-00010bb0: 2072 6573 756c 7473 2075 706f 6e20 7468   results upon th
-00010bc0: 6520 6f72 6967 696e 616c 2061 6e63 686f  e original ancho
-00010bd0: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
-00010be0: 3e20 6469 7374 7269 6275 7469 6f6e 2062  > distribution b
-00010bf0: 7920 7468 6520 606d 6978 7475 7265 6020  y the `mixture` 
-00010c00: 6d6f 6475 6c65 206f 6620 6073 6369 6b69  module of `sciki
-00010c10: 742d 6c65 6172 6e60 206c 6962 7261 7279  t-learn` library
-00010c20: 2061 6e64 2074 776f 2073 7562 666c 6f64   and two subflod
-00010c30: 6572 732c 2060 4c6f 6747 4d4d 5f43 4960  ers, `LogGMM_CI`
-00010c40: 2063 6f6e 7461 696e 696e 6720 7468 6520   containing the 
-00010c50: 7265 7375 6c74 7320 6f66 2039 3525 2043  results of 95% C
-00010c60: 4920 6f66 2065 6163 6820 636f 6d70 6f6e  I of each compon
-00010c70: 656e 742c 2060 4869 6768 4d61 7373 5f43  ent, `HighMass_C
-00010c80: 4960 2063 6f6e 7461 696e 696e 6720 7468  I` containing th
-00010c90: 6520 4869 6768 4d61 7373 206f 6620 6561  e HighMass of ea
-00010ca0: 6368 2063 6f6d 706f 6e65 6e74 2c20 7768  ch component, wh
-00010cb0: 6963 6820 6361 6e20 6265 2075 7365 6420  ich can be used 
-00010cc0: 666f 7220 6675 7274 6865 7220 5747 4420  for further WGD 
-00010cd0: 6461 7469 6e67 2e20 5468 6520 6053 6567  dating. The `Seg
-00010ce0: 6d65 6e74 4775 6964 654b 735f 474d 4d60  mentGuideKs_GMM`
-00010cf0: 2073 7562 666f 6c64 6572 2070 7265 7365   subfolder prese
-00010d00: 6e74 7320 7265 7375 6c74 7320 6f66 2073  nts results of s
-00010d10: 6567 6d65 6e74 202a 4b2a 3c73 7562 3e53  egment *K*<sub>S
-00010d20: 3c2f 7375 623e 2047 4d4d 2077 6869 6368  </sub> GMM which
-00010d30: 2061 7265 206d 6170 7065 6420 6261 636b   are mapped back
-00010d40: 2074 6f20 7468 6520 7265 7369 6469 6e67   to the residing
-00010d50: 2061 6e63 686f 7220 7061 6972 7320 616e   anchor pairs an
-00010d60: 6420 7468 6520 6173 736f 6369 6174 6564  d the associated
-00010d70: 2039 3525 2048 4452 2061 6e64 2048 6967   95% HDR and Hig
-00010d80: 684d 6173 7320 6f66 2065 6163 6820 7365  hMass of each se
-00010d90: 676d 656e 7420 636c 7573 7465 7220 696e  gment cluster in
-00010da0: 2073 7562 666c 6f64 6572 7320 6f66 2060   subfloders of `
-00010db0: 4844 525f 4349 6020 616e 6420 6048 6967  HDR_CI` and `Hig
-00010dc0: 684d 6173 735f 4349 602e 2054 6865 2060  hMass_CI`. The `
-00010dd0: 5365 676d 656e 744b 735f 4669 6e64 5065  SegmentKs_FindPe
-00010de0: 616b 6020 7375 6266 6f6c 6465 7220 6973  ak` subfolder is
-00010df0: 2073 696d 696c 6172 2074 6f20 6041 6e63   similar to `Anc
-00010e00: 686f 724b 735f 4669 6e64 5065 616b 6020  horKs_FindPeak` 
-00010e10: 6275 7420 7769 7468 2073 6567 6d65 6e74  but with segment
-00010e20: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00010e30: 2069 6e73 7465 6164 2e20 5468 6520 2a4b   instead. The *K
-00010e40: 2a3c 7375 623e 533c 2f73 7562 3e20 696e  *<sub>S</sub> in
-00010e50: 2060 4d75 6c74 6970 6c69 636f 6e60 2063   `Multiplicon` c
-00010e60: 616e 2061 6c73 6f20 6265 2063 616c 6375  an also be calcu
-00010e70: 6c61 7465 6420 696e 2070 6c61 6365 206f  lated in place o
-00010e80: 6620 6053 6567 6d65 6e74 6020 7573 696e  f `Segment` usin
-00010e90: 6720 7468 6520 6f70 7469 6f6e 2060 2d2d  g the option `--
-00010ea0: 6775 6964 6560 2061 7320 7375 6368 2074  guide` as such t
-00010eb0: 6865 2072 6573 756c 7420 7469 746c 652c  he result title,
-00010ec0: 206c 6162 656c 2c20 6669 6c65 2061 6e64   label, file and
-00010ed0: 2066 6f6c 6465 7220 6e61 6d65 7320 7769   folder names wi
-00010ee0: 6c6c 2062 6520 6368 616e 6765 6420 6163  ll be changed ac
-00010ef0: 636f 7264 696e 676c 792e 0a0a 2a2a 4120  cordingly...**A 
-00010f00: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
-00010f10: 6e67 2072 756e 2063 616e 2075 7365 2063  ng run can use c
-00010f20: 6f6d 6d61 6e64 2073 696d 706c 7920 6173  ommand simply as
-00010f30: 2062 656c 6f77 2a2a 0a0a 6060 600a 7767   below**..```.wg
-00010f40: 6420 7065 616b 206b 7364 6174 6120 2d61  d peak ksdata -a
-00010f50: 7020 6170 6461 7461 202d 736d 2073 6d64  p apdata -sm smd
-00010f60: 6174 6120 2d6c 6520 6c65 6461 7461 202d  ata -le ledata -
-00010f70: 6d70 206d 7064 6174 6120 2d2d 6865 7572  mp mpdata --heur
-00010f80: 6973 7469 630a 6060 600a 0a23 2323 2077  istic.```..### w
-00010f90: 6764 2073 796e 0a0a 2a2a 5468 6520 696e  gd syn..**The in
-00010fa0: 7472 612d 7370 6563 6966 6963 2073 796e  tra-specific syn
-00010fb0: 7465 6e79 2069 6e66 6572 656e 6365 2a2a  teny inference**
-00010fc0: 0a60 6060 0a77 6764 2073 796e 2066 616d  .```.wgd syn fam
-00010fd0: 696c 6965 7320 6766 6620 282d 2d6b 735f  ilies gff (--ks_
-00010fe0: 6469 7374 7269 6275 7469 6f6e 206b 7364  distribution ksd
-00010ff0: 6174 6120 2d66 2067 656e 6520 2d61 2049  ata -f gene -a I
-00011000: 4420 2d2d 6d69 6e6c 656e 202d 3120 2d2d  D --minlen -1 --
-00011010: 6d69 6e73 6567 6c65 6e20 3130 3030 3020  minseglen 10000 
-00011020: 2d2d 6d69 6e67 656e 656e 756d 2033 3029  --mingenenum 30)
-00011030: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
-00011040: 6d20 6077 6764 2073 796e 6020 6973 206d  m `wgd syn` is m
-00011050: 6169 6e6c 7920 6465 616c 696e 6720 7769  ainly dealing wi
-00011060: 7468 2063 6f6c 6c69 6e65 6172 6974 7920  th collinearity 
-00011070: 6f72 2073 796e 7465 6e79 2028 626f 7468  or synteny (both
-00011080: 2072 6566 6572 7265 6420 746f 2061 7320   referred to as 
-00011090: 7379 6e74 656e 7920 6865 7265 6166 7465  synteny hereafte
-000110a0: 7229 2061 6e61 6c79 7369 732e 2054 776f  r) analysis. Two
-000110b0: 2069 6e70 7574 2066 696c 6573 2061 7265   input files are
-000110c0: 2065 7373 656e 7469 616c 2c20 7468 6520   essential, the 
-000110d0: 6765 6e65 2066 616d 696c 7920 6669 6c65  gene family file
-000110e0: 2061 6e64 2074 6865 2067 6666 3320 6669   and the gff3 fi
-000110f0: 6c65 2e20 5468 6520 6765 6e65 2066 616d  le. The gene fam
-00011100: 696c 7920 6669 6c65 2069 7320 696e 2074  ily file is in t
-00011110: 6865 2066 6f72 6d61 7420 6173 2060 4f72  he format as `Or
-00011120: 7468 6f46 696e 6465 7260 2e20 5468 6520  thoFinder`. The 
-00011130: 736f 6674 7761 7265 2060 692d 6164 686f  software `i-adho
-00011140: 7265 6020 6973 2061 2070 7265 7265 7175  re` is a prerequ
-00011150: 6973 6974 652e 2057 6974 6820 6465 6661  isite. With defa
-00011160: 756c 7420 7061 7261 6d65 7465 7273 2c20  ult parameters, 
-00011170: 7468 6520 7072 6f67 7261 6d20 6261 7369  the program basi
-00011180: 6361 6c6c 7920 636f 6e64 7563 7473 2031  cally conducts 1
-00011190: 2920 6669 6c74 6572 696e 6720 6765 6e65  ) filtering gene
-000111a0: 2066 616d 696c 6965 7320 6261 7365 6420   families based 
-000111b0: 6f6e 206d 6178 696d 756d 2066 616d 696c  on maximum famil
-000111c0: 7920 7369 7a65 2032 2920 7265 7472 6965  y size 2) retrie
-000111d0: 7669 6e67 2067 656e 6520 706f 7369 7469  ving gene positi
-000111e0: 6f6e 2061 6e64 2073 6361 6666 6f6c 6420  on and scaffold 
-000111f0: 696e 666f 726d 6174 696f 6e20 6672 6f6d  information from
-00011200: 2067 6666 3320 6669 6c65 2033 2920 7072   gff3 file 3) pr
-00011210: 6f64 7563 696e 6720 7468 6520 636f 6e66  oducing the conf
-00011220: 6967 7572 6174 696f 6e20 6669 6c65 2061  iguration file a
-00011230: 6e64 2061 7373 6f63 6961 7465 6420 6461  nd associated da
-00011240: 7461 6669 6c65 7320 666f 7220 6069 2d61  tafiles for `i-a
-00011250: 6468 6f72 6560 2034 2920 6361 6c6c 696e  dhore` 4) callin
-00011260: 6720 6069 2d61 6468 6f72 6560 2067 6976  g `i-adhore` giv
-00011270: 656e 2074 6865 2070 6172 616d 6574 6572  en the parameter
-00011280: 7320 7365 7420 746f 2069 6e66 6572 2073  s set to infer s
-00011290: 796e 7465 6e79 2035 2920 7669 7375 616c  ynteny 5) visual
-000112a0: 697a 696e 6720 7468 6520 7379 6e74 656e  izing the synten
-000112b0: 7920 696e 2022 646f 7470 6c6f 7422 2069  y in "dotplot" i
-000112c0: 6e20 7468 6520 756e 6974 206f 6620 6765  n the unit of ge
-000112d0: 6e65 7320 616e 6420 6261 7365 732c 2069  nes and bases, i
-000112e0: 6e20 2253 796e 6465 7074 6822 2070 6c6f  n "Syndepth" plo
-000112f0: 7420 7368 6f77 696e 6720 7468 6520 6469  t showing the di
-00011300: 7374 7269 6275 7469 6f6e 206f 6620 6469  stribution of di
-00011310: 6666 6572 656e 7420 6361 7465 676f 7269  fferent categori
-00011320: 6573 206f 6620 636f 6c6c 696e 6561 7269  es of collineari
-00011330: 7479 2072 6174 696f 7320 7769 7468 696e  ty ratios within
-00011340: 2061 6e64 2062 6574 7765 656e 2073 7065   and between spe
-00011350: 6369 6573 2c20 696e 2022 6475 7053 7461  cies, in "dupSta
-00011360: 636b 2220 706c 6f74 2073 686f 7769 6e67  ck" plot showing
-00011370: 206d 756c 7469 706c 6963 6f6e 7320 7769   multiplicons wi
-00011380: 7468 2064 6966 6665 7265 6e74 206d 756c  th different mul
-00011390: 7469 706c 6963 6174 696f 6e20 6c65 7665  tiplication leve
-000113a0: 6c73 2e20 3629 2069 6620 7769 7468 202a  ls. 6) if with *
-000113b0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
-000113c0: 6174 612c 2061 2022 2a4b 2a3c 7375 623e  ata, a "*K*<sub>
-000113d0: 533c 2f73 7562 3e20 646f 7470 6c6f 7422  S</sub> dotplot"
-000113e0: 2077 6974 6820 646f 7473 2061 6e6e 6f74   with dots annot
-000113f0: 6174 6564 2069 6e20 2a4b 2a3c 7375 623e  ated in *K*<sub>
-00011400: 533c 2f73 7562 3e20 7661 6c75 6573 2061  S</sub> values a
-00011410: 6e64 2061 202a 4b2a 3c73 7562 3e53 3c2f  nd a *K*<sub>S</
-00011420: 7375 623e 2064 6973 7472 6962 7574 696f  sub> distributio
-00011430: 6e20 7769 7468 2061 6e63 686f 7220 7061  n with anchor pa
-00011440: 6972 7320 6465 6e6f 7465 6420 7769 6c6c  irs denoted will
-00011450: 2062 6520 7072 6f64 7563 6564 2e20 5468   be produced. Th
-00011460: 6520 6765 6e65 2069 6e66 6f72 6d61 7469  e gene informati
-00011470: 6f6e 2069 6e20 7468 6520 6765 6e65 2066  on in the gene f
-00011480: 616d 696c 7920 6669 6c65 2061 6e64 2067  amily file and g
-00011490: 6666 3320 6669 6c65 2073 686f 756c 6420  ff3 file should 
-000114a0: 6265 206d 6174 6368 6564 2077 6869 6368  be matched which
-000114b0: 2072 6571 7569 7265 7320 7573 6572 7320   requires users 
-000114c0: 746f 2073 6574 2070 726f 7065 7220 602d  to set proper `-
-000114d0: 2d66 6561 7475 7265 6020 616e 6420 602d  -feature` and `-
-000114e0: 2d61 7474 7269 6275 7465 602e 2054 6865  -attribute`. The
-000114f0: 206d 6178 696d 756d 2066 616d 696c 7920   maximum family 
-00011500: 7369 7a65 2074 6f20 6265 2069 6e63 6c75  size to be inclu
-00011510: 6465 6420 6361 6e20 6265 2073 6574 2076  ded can be set v
-00011520: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
-00011530: 2d6d 6178 7369 7a65 602c 206e 6f74 6564  -maxsize`, noted
-00011540: 2074 6861 7420 7468 6973 2066 696c 7465   that this filte
-00011550: 7269 6e67 2069 7320 6d61 696e 6c79 2074  ring is mainly t
-00011560: 6f20 6472 6f70 2074 686f 7365 2068 7567  o drop those hug
-00011570: 6520 7461 6e64 656d 2064 7570 6c69 6361  e tandem duplica
-00011580: 7465 7320 6661 6d69 6c79 2061 6e64 2074  tes family and t
-00011590: 7261 6e73 706f 7361 626c 6520 656c 656d  ransposable elem
-000115a0: 656e 7473 2028 5445 7329 2066 616d 696c  ents (TEs) famil
-000115b0: 792c 2061 6e64 206e 6f74 206d 616e 6461  y, and not manda
-000115c0: 746f 7279 2e20 5573 6572 7320 6361 6e20  tory. Users can 
-000115d0: 6669 6c74 6572 2074 686f 7365 2066 7261  filter those fra
-000115e0: 676d 656e 7461 7279 2073 6361 6666 6f6c  gmentary scaffol
-000115f0: 6473 2076 6961 2074 6865 206f 7074 696f  ds via the optio
-00011600: 6e20 602d 2d6d 696e 6c65 6e60 2e20 5468  n `--minlen`. Th
-00011610: 6520 6d69 6e69 6d75 6d20 6c65 6e67 7468  e minimum length
-00011620: 2061 6e64 206e 756d 6265 7220 6f66 2067   and number of g
-00011630: 656e 6573 2066 6f72 2061 2073 6567 6d65  enes for a segme
-00011640: 6e74 2074 6f20 6265 2063 6f6e 7369 6465  nt to be conside
-00011650: 7265 6420 6361 6e20 6265 2073 6574 2076  red can be set v
-00011660: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
-00011670: 2d6d 696e 7365 676c 656e 6020 616e 6420  -minseglen` and 
-00011680: 602d 2d6d 696e 6765 6e65 6e75 6d60 2e20  `--mingenenum`. 
-00011690: 5265 6475 6e64 616e 7420 6d75 6c74 6970  Redundant multip
-000116a0: 6c69 636f 6e73 2063 616e 2062 6520 6b65  licons can be ke
-000116b0: 7074 2062 7920 7365 7420 7468 6520 666c  pt by set the fl
-000116c0: 6167 206f 7074 696f 6e20 602d 2d6b 6565  ag option `--kee
-000116d0: 7072 6564 756e 602e 0a0a 2a2a 4120 7375  predun`...**A su
-000116e0: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
-000116f0: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
-00011700: 6d61 6e64 2073 696d 706c 7920 6173 2062  mand simply as b
-00011710: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
-00011720: 7379 6e20 6661 6d69 6c69 6573 2067 6666  syn families gff
-00011730: 0a60 6060 0a0a 2a2a 5468 6520 696e 7465  .```..**The inte
-00011740: 722d 7370 6563 6966 6963 2073 796e 7465  r-specific synte
-00011750: 6e79 2069 6e66 6572 656e 6365 2a2a 0a60  ny inference**.`
-00011760: 6060 0a77 6764 2073 796e 2066 616d 696c  ``.wgd syn famil
-00011770: 6965 7320 6766 6631 2067 6666 3220 282d  ies gff1 gff2 (-
-00011780: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
-00011790: 666f 2022 6d52 4e41 3b4e 616d 6522 202d  fo "mRNA;Name" -
-000117a0: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
-000117b0: 666f 2022 6765 6e65 3b49 4422 290a 6060  fo "gene;ID").``
-000117c0: 600a 0a46 6f72 206d 756c 7469 2d73 7065  `..For multi-spe
-000117d0: 6369 6573 2073 796e 7465 6e79 2069 6e66  cies synteny inf
-000117e0: 6572 656e 6365 2c20 6966 2075 7365 7273  erence, if users
-000117f0: 2068 6176 6520 6766 6633 2066 696c 6573   have gff3 files
-00011800: 2077 6869 6368 2068 6176 6520 6469 6666   which have diff
-00011810: 6572 656e 7420 6665 6174 7572 6573 206f  erent features o
-00011820: 7220 6174 7472 6962 7574 6573 2066 6f72  r attributes for
-00011830: 2067 656e 6520 706f 7369 7469 6f6e 2069   gene position i
-00011840: 6e66 6f72 6d61 7469 6f6e 2072 6574 7269  nformation retri
-00011850: 6576 616c 2c20 7468 6520 6f70 7469 6f6e  eval, the option
-00011860: 2060 2d2d 6164 6469 7469 6f6e 616c 6766   `--additionalgf
-00011870: 6669 6e66 6f60 2063 616e 2062 6520 7365  finfo` can be se
-00011880: 7420 746f 2070 726f 7669 6465 2074 6865  t to provide the
-00011890: 2061 6464 6974 696f 6e61 6c20 696e 666f   additional info
-000118a0: 726d 6174 696f 6e2e 2054 6865 2072 656d  rmation. The rem
-000118b0: 6169 6e69 6e67 2070 6172 616d 6574 6572  aining parameter
-000118c0: 2073 6574 7469 6e67 2069 7320 7468 6520   setting is the 
-000118d0: 7361 6d65 2061 7320 7468 6520 696e 7472  same as the intr
-000118e0: 612d 7370 6563 6966 6963 2073 796e 7465  a-specific synte
-000118f0: 6e79 2069 6e66 6572 656e 6365 2e0a 0a2a  ny inference...*
-00011900: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
-00011910: 7274 696e 6720 7275 6e20 6361 6e20 7573  rting run can us
-00011920: 6520 636f 6d6d 616e 6420 7369 6d70 6c79  e command simply
-00011930: 2061 7320 6265 6c6f 772a 2a0a 0a60 6060   as below**..```
-00011940: 0a77 6764 2073 796e 2066 616d 696c 6965  .wgd syn familie
-00011950: 7320 6766 6631 2067 6666 320a 6060 600a  s gff1 gff2.```.
-00011960: 0a23 2323 2077 6764 2076 697a 0a0a 2a2a  .### wgd viz..**
-00011970: 5468 6520 7669 7375 616c 697a 6174 696f  The visualizatio
-00011980: 6e20 6f66 202a 4b2a 3c73 7562 3e53 3c2f  n of *K*<sub>S</
-00011990: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-000119a0: 7574 696f 6e20 616e 6420 454c 4d4d 2061  ution and ELMM a
-000119b0: 6e61 6c79 7369 732a 2a0a 6060 600a 7767  nalysis**.```.wg
-000119c0: 6420 7669 7a20 2d64 206b 7364 6174 610a  d viz -d ksdata.
-000119d0: 6060 600a 0a54 6865 2070 726f 6772 616d  ```..The program
-000119e0: 2060 7767 6420 7669 7a60 2069 7320 6d61   `wgd viz` is ma
-000119f0: 696e 6c79 2066 6f72 2074 6865 2070 7572  inly for the pur
-00011a00: 706f 7365 206f 6620 2a4b 2a3c 7375 623e  pose of *K*<sub>
-00011a10: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
-00011a20: 7469 6f6e 2061 6e64 2073 796e 7465 6e79  tion and synteny
-00011a30: 2076 6973 7561 6c69 7a61 7469 6f6e 2c20   visualization, 
-00011a40: 7769 7468 2073 6f6d 6520 6f70 7469 6f6e  with some option
-00011a50: 616c 206d 6978 7475 7265 206d 6f64 656c  al mixture model
-00011a60: 696e 6720 616e 616c 7973 6973 2e20 5468  ing analysis. Th
-00011a70: 6520 6261 7369 6320 6675 6e63 7469 6f6e  e basic function
-00011a80: 2069 7320 6a75 7374 2074 6f20 706c 6f74   is just to plot
-00011a90: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-00011aa0: 7375 623e 2064 6973 7472 6962 7574 696f  sub> distributio
-00011ab0: 6e20 616e 6420 636f 6e64 7563 7420 616e  n and conduct an
-00011ac0: 2045 4c4d 4d20 616e 616c 7973 6973 2069   ELMM analysis i
-00011ad0: 6e20 7365 6172 6368 206f 6620 706f 7465  n search of pote
-00011ae0: 6e74 6961 6c20 5747 4420 636f 6d70 6f6e  ntial WGD compon
-00011af0: 656e 7473 2e20 536f 6d65 206b 6579 2070  ents. Some key p
-00011b00: 6172 616d 6574 6572 7320 6166 6665 6374  arameters affect
-00011b10: 696e 6720 7468 6520 454c 4d4d 2072 6573  ing the ELMM res
-00011b20: 756c 7420 696e 636c 7564 6520 602d 2d70  ult include `--p
-00011b30: 726f 6d69 6e65 6e63 655f 6375 746f 6666  rominence_cutoff
-00011b40: 602c 2060 2d2d 7265 6c5f 6865 6967 6874  `, `--rel_height
-00011b50: 602c 2060 2d2d 656d 5f69 7465 7261 7469  `, `--em_iterati
-00011b60: 6f6e 7360 2061 6e64 2060 2d2d 656d 5f69  ons` and `--em_i
-00011b70: 6e69 7469 616c 697a 6174 696f 6e73 602c  nitializations`,
-00011b80: 2061 6c6c 206f 6620 7768 6963 6820 6861   all of which ha
-00011b90: 7665 2062 6565 6e20 6578 706c 6169 6e65  ve been explaine
-00011ba0: 6420 6168 6561 642e 0a0a 2a2a 4120 7375  d ahead...**A su
-00011bb0: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
-00011bc0: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
-00011bd0: 6d61 6e64 2073 696d 706c 7920 6173 2062  mand simply as b
-00011be0: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
-00011bf0: 7669 7a20 2d64 206b 7364 6174 610a 6060  viz -d ksdata.``
-00011c00: 600a 0a2a 2a54 6865 2076 6973 7561 6c69  `..**The visuali
-00011c10: 7a61 7469 6f6e 206f 6620 2a4b 2a3c 7375  zation of *K*<su
-00011c20: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-00011c30: 7374 7269 6275 7469 6f6e 2077 6974 6820  stribution with 
-00011c40: 7261 7465 2063 6f72 7265 6374 696f 6e2a  rate correction*
-00011c50: 2a0a 6060 600a 7767 6420 7669 7a20 2d64  *.```.wgd viz -d
-00011c60: 206b 7364 6174 6120 2d73 7020 7370 6461   ksdata -sp spda
-00011c70: 7461 202d 2d66 6f63 7573 3261 6c6c 2066  ta --focus2all f
-00011c80: 6f63 616c 5f73 7065 6369 6573 202d 2d65  ocal_species --e
-00011c90: 7874 7261 7061 7261 6e6f 6d65 6b73 206b  xtraparanomeks k
-00011ca0: 7364 6174 6120 282d 2d61 6e63 686f 7270  sdata (--anchorp
-00011cb0: 6f69 6e74 7320 6170 6461 7461 202d 2d70  oints apdata --p
-00011cc0: 6c6f 7461 7067 6d6d 202d 2d70 6c6f 7465  lotapgmm --plote
-00011cd0: 6c6d 6d29 0a60 6060 0a0a 4265 7369 6465  lmm).```..Beside
-00011ce0: 7320 7468 6520 6261 7369 6320 2a4b 2a3c  s the basic *K*<
-00011cf0: 7375 623e 533c 2f73 7562 3e20 706c 6f74  sub>S</sub> plot
-00011d00: 2c20 7375 6273 7469 7475 7469 6f6e 2072  , substitution r
-00011d10: 6174 6520 636f 7272 6563 7469 6f6e 2063  ate correction c
-00011d20: 616e 2061 6c73 6f20 6265 2061 6368 6965  an also be achie
-00011d30: 7665 6420 6769 7665 6e20 6174 206c 6561  ved given at lea
-00011d40: 7374 2061 2073 7065 6369 6573 2074 7265  st a species tre
-00011d50: 6520 2876 6961 2074 6865 206f 7074 696f  e (via the optio
-00011d60: 6e20 602d 2d73 7065 6369 6573 7472 6565  n `--speciestree
-00011d70: 6029 2061 6e64 2061 2066 6f63 616c 2073  `) and a focal s
-00011d80: 7065 6369 6573 2028 6569 7468 6572 2076  pecies (either v
-00011d90: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
-00011da0: 2d66 6f63 7573 3261 6c6c 6020 6f72 2076  -focus2all` or v
-00011db0: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
-00011dc0: 2d73 7061 6972 6020 696e 2074 6865 2066  -spair` in the f
-00011dd0: 6f72 6d20 6f66 2022 2466 6f63 616c 5f73  orm of "$focal_s
-00011de0: 7065 6369 6573 3b24 666f 6361 6c5f 7370  pecies;$focal_sp
-00011df0: 6563 6965 7322 292e 2049 7427 7320 7375  ecies"). It's su
-00011e00: 6767 6573 7465 6420 7468 6174 2074 6865  ggested that the
-00011e10: 206f 7274 686f 6c6f 676f 7573 202a 4b2a   orthologous *K*
-00011e20: 3c73 7562 3e53 3c2f 7375 623e 2064 6174  <sub>S</sub> dat
-00011e30: 6120 6973 2070 726f 7669 6465 6420 6279  a is provided by
-00011e40: 2074 6865 2060 2d2d 6461 7461 6669 6c65   the `--datafile
-00011e50: 6020 6f70 7469 6f6e 2061 6e64 2074 6865  ` option and the
-00011e60: 2070 6172 616c 6f67 6f75 7320 2a4b 2a3c   paralogous *K*<
-00011e70: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
-00011e80: 2069 7320 7072 6f76 6964 6564 2062 7920   is provided by 
-00011e90: 7468 6520 602d 2d65 7874 7261 7061 7261  the `--extrapara
-00011ea0: 6e6f 6d65 6b73 6020 6f70 7469 6f6e 2c20  nomeks` option, 
-00011eb0: 616c 7468 6f75 6768 2069 7427 7320 616c  although it's al
-00011ec0: 6c6f 7765 6420 746f 206f 6e6c 7920 7072  lowed to only pr
-00011ed0: 6f76 6964 6520 2a4b 2a3c 7375 623e 533c  ovide *K*<sub>S<
-00011ee0: 2f73 7562 3e20 6461 7461 2076 6961 2074  /sub> data via t
-00011ef0: 6865 2060 2d2d 6461 7461 6669 6c65 6020  he `--datafile` 
-00011f00: 6f70 7469 6f6e 2061 6e64 2064 6570 6f73  option and depos
-00011f10: 6974 2062 6f74 6820 6f72 7468 6f6c 6f67  it both ortholog
-00011f20: 6f75 7326 7061 7261 6c6f 676f 7573 202a  ous&paralogous *
-00011f30: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
-00011f40: 6174 6120 7468 6572 656f 6e2e 2054 6865  ata thereon. The
-00011f50: 7265 2061 7265 2074 776f 2074 7970 6573  re are two types
-00011f60: 206f 6620 6d69 7865 6420 706c 6f74 7320   of mixed plots 
-00011f70: 2874 6865 2022 6d69 7865 6422 2068 6572  (the "mixed" her
-00011f80: 6520 7265 6665 7273 2074 6f20 6d69 7865  e refers to mixe
-00011f90: 6420 6f72 7468 6f6c 6f67 6f75 7320 616e  d orthologous an
-00011fa0: 6420 7061 7261 6c6f 676f 7573 202a 4b2a  d paralogous *K*
-00011fb0: 3c73 7562 3e53 3c2f 7375 623e 2064 6973  <sub>S</sub> dis
-00011fc0: 7472 6962 7574 696f 6e73 292c 206f 6e65  tributions), one
-00011fd0: 206f 6620 7768 6963 6820 6973 2073 696d   of which is sim
-00011fe0: 696c 6172 2074 6f20 7768 6174 2060 6b73  ilar to what `ks
-00011ff0: 7261 7465 7360 2070 6c6f 7473 2c20 7768  rates` plots, wh
-00012000: 696c 6520 7468 6520 6f74 6865 7220 6f66  ile the other of
-00012010: 2077 6869 6368 2069 7320 6c69 6b65 2074   which is like t
-00012020: 6865 2063 6f6e 7665 6e74 696f 6e61 6c20  he conventional 
-00012030: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00012040: 706c 6f74 2074 6861 7420 626f 7468 2074  plot that both t
-00012050: 6865 206f 7269 6769 6e61 6c20 6f72 7468  he original orth
-00012060: 6f6c 6f67 6f75 7320 616e 6420 7061 7261  ologous and para
-00012070: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-00012080: 3c2f 7375 623e 2064 6973 7472 6962 7574  </sub> distribut
-00012090: 696f 6e73 2061 7265 2074 7275 7468 6675  ions are truthfu
-000120a0: 6c6c 7920 706c 6f74 7465 6420 696e 7374  lly plotted inst
-000120b0: 6561 6420 6f66 206a 7573 7420 6265 696e  ead of just bein
-000120c0: 6720 7265 7072 6573 656e 7465 6420 6279  g represented by
-000120d0: 2073 6f6d 6520 7665 7274 6963 616c 206c   some vertical l
-000120e0: 696e 6573 2e20 5765 2073 7567 6765 7374  ines. We suggest
-000120f0: 2075 7365 7273 2074 6f20 6164 6f70 7420   users to adopt 
-00012100: 7468 6520 606b 7372 6174 6573 602d 6c69  the `ksrates`-li
-00012110: 6b65 2070 6c6f 7473 2c20 7768 6963 6820  ke plots, which 
-00012120: 6973 2074 6865 2064 6566 6175 6c74 2e20  is the default. 
-00012130: 4f74 6865 7277 6973 6520 7468 6520 6f70  Otherwise the op
-00012140: 7469 6f6e 2060 2d2d 636c 6173 7369 6360  tion `--classic`
-00012150: 2077 696c 6c20 6265 206e 6565 6465 6420   will be needed 
-00012160: 746f 2073 6574 2e20 4578 7472 6120 6d69  to set. Extra mi
-00012170: 7874 7572 6520 6d6f 6465 6c69 6e67 2061  xture modeling a
-00012180: 6e61 6c79 7369 7320 6361 6e20 6265 2069  nalysis can be i
-00012190: 6e69 7469 6174 6564 2076 6961 2074 6865  nitiated via the
-000121a0: 206f 7074 696f 6e20 602d 2d70 6c6f 7465   option `--plote
-000121b0: 6c6d 6d60 2061 6e64 2060 2d2d 706c 6f74  lmm` and `--plot
-000121c0: 6170 676d 6d60 2077 6974 6820 7468 6520  apgmm` with the 
-000121d0: 616e 6368 6f72 2070 6f69 6e74 7320 6461  anchor points da
-000121e0: 7461 6669 6c65 2070 726f 7669 6465 6420  tafile provided 
-000121f0: 6279 2074 6865 2060 2d2d 616e 6368 6f72  by the `--anchor
-00012200: 706f 696e 7473 6020 6f70 7469 6f6e 2e0a  points` option..
-00012210: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
-00012220: 7461 7274 696e 6720 7275 6e20 6361 6e20  tarting run can 
-00012230: 7573 6520 636f 6d6d 616e 6420 7369 6d70  use command simp
-00012240: 6c79 2061 7320 6265 6c6f 772a 2a0a 0a60  ly as below**..`
-00012250: 6060 0a77 6764 2076 697a 202d 6420 6b73  ``.wgd viz -d ks
-00012260: 6461 7461 202d 7370 2073 7064 6174 6120  data -sp spdata 
-00012270: 2d2d 666f 6375 7332 616c 6c20 666f 6361  --focus2all foca
-00012280: 6c5f 7370 6563 6965 7320 2d2d 6578 7472  l_species --extr
-00012290: 6170 6172 616e 6f6d 656b 7320 6b73 6461  aparanomeks ksda
-000122a0: 7461 0a60 6060 0a0a 2a2a 5468 6520 7669  ta.```..**The vi
-000122b0: 7375 616c 697a 6174 696f 6e20 6f66 2073  sualization of s
-000122c0: 796e 7465 6e79 2a2a 0a60 6060 0a77 6764  ynteny**.```.wgd
-000122d0: 2076 697a 202d 6170 2061 7064 6174 6120   viz -ap apdata 
-000122e0: 2d73 6d20 736d 6461 7461 202d 6d74 206d  -sm smdata -mt m
-000122f0: 7464 6174 6120 2d67 7420 6774 6461 7461  tdata -gt gtdata
-00012300: 202d 2d70 6c6f 7473 796e 2028 2d2d 6d69   --plotsyn (--mi
-00012310: 6e6c 656e 202d 3120 2d2d 6d69 6e73 6567  nlen -1 --minseg
-00012320: 6c65 6e20 3130 3030 3020 2d2d 6d69 6e67  len 10000 --ming
-00012330: 656e 656e 756d 2033 3029 0a60 6060 0a0a  enenum 30).```..
-00012340: 436f 6d70 6172 6564 2074 6f20 7468 6520  Compared to the 
-00012350: 6f72 6967 696e 616c 2060 7767 6460 2c20  original `wgd`, 
-00012360: 7468 6520 6077 6764 2076 697a 6020 7072  the `wgd viz` pr
-00012370: 6f67 7261 6d20 6164 6465 6420 7379 6e74  ogram added synt
-00012380: 656e 7920 7669 7375 616c 697a 6174 696f  eny visualizatio
-00012390: 6e20 7069 7065 6c69 6e65 2e20 5573 6572  n pipeline. User
-000123a0: 7320 6e65 6564 2074 6f20 7072 6f76 6964  s need to provid
-000123b0: 6520 7468 6520 666c 6167 206f 7074 696f  e the flag optio
-000123c0: 6e20 602d 2d70 6c6f 7473 796e 6020 746f  n `--plotsyn` to
-000123d0: 2069 6e69 7469 6174 6520 7468 6973 2070   initiate this p
-000123e0: 6172 7420 6f66 2070 6970 656c 696e 652e  art of pipeline.
-000123f0: 2054 6869 7320 7374 6570 2061 7373 756d   This step assum
-00012400: 6573 2074 6861 7420 7573 6572 7320 6861  es that users ha
-00012410: 7665 206f 6274 6169 6e65 6420 616c 7265  ve obtained alre
-00012420: 6164 7920 7468 6520 7379 6e74 656e 6963  ady the syntenic
-00012430: 2072 6573 756c 7473 2066 726f 6d20 6069   results from `i
-00012440: 2d61 6468 6f72 6560 2061 6e64 2075 7365  -adhore` and use
-00012450: 7320 7468 6f73 6520 7265 7375 6c74 2066  s those result f
-00012460: 696c 6573 2074 6f20 7265 616c 697a 6520  iles to realize 
-00012470: 7468 6520 7379 6e74 656e 7920 7669 7375  the synteny visu
-00012480: 616c 697a 6174 696f 6e2e 2053 696d 6c69  alization. Simli
-00012490: 6172 2074 6f20 6077 6764 2073 796e 602c  ar to `wgd syn`,
-000124a0: 2074 6865 2065 7874 7261 202a 4b2a 3c73   the extra *K*<s
-000124b0: 7562 3e53 3c2f 7375 623e 2064 6174 6120  ub>S</sub> data 
-000124c0: 6669 6c65 2063 616e 2062 6520 7472 616e  file can be tran
-000124d0: 736d 6974 7465 6420 7669 6120 7468 6520  smitted via the 
-000124e0: 6f70 7469 6f6e 2060 2d2d 6461 7461 6669  option `--datafi
-000124f0: 6c65 6020 2869 6e73 7465 6164 206f 6620  le` (instead of 
-00012500: 602d 2d65 7874 7261 7061 7261 6e6f 6d65  `--extraparanome
-00012510: 6b73 6020 6f70 7469 6f6e 292e 2042 6173  ks` option). Bas
-00012520: 6963 616c 6c79 2c20 7468 6520 7379 6e74  ically, the synt
-00012530: 656e 6963 2072 6573 756c 7420 6669 6c65  enic result file
-00012540: 7320 7265 7175 6972 6564 2061 7265 2074  s required are t
-00012550: 6865 2061 6e63 686f 7220 706f 696e 7473  he anchor points
-00012560: 2064 6174 6166 696c 652c 206d 756c 7469   datafile, multi
-00012570: 706c 6963 6f6e 7320 6461 7461 6669 6c65  plicons datafile
-00012580: 2c20 6765 6e65 2074 6162 6c65 2064 6174  , gene table dat
-00012590: 6166 696c 6520 2861 7574 6f6d 6174 656c  afile (automatel
-000125a0: 7920 7072 6f64 7563 6564 2062 7920 6077  y produced by `w
-000125b0: 6764 2073 796e 6029 2c20 616e 6420 7365  gd syn`), and se
-000125c0: 676d 656e 7473 2064 6174 6166 696c 652e  gments datafile.
-000125d0: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
-000125e0: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
-000125f0: 2075 7365 2063 6f6d 6d61 6e64 2073 696d   use command sim
-00012600: 706c 7920 6173 2062 656c 6f77 2a2a 0a0a  ply as below**..
-00012610: 6060 600a 7767 6420 7669 7a20 2d61 7020  ```.wgd viz -ap 
-00012620: 6170 6461 7461 202d 736d 2073 6d64 6174  apdata -sm smdat
-00012630: 6120 2d6d 7420 6d74 6461 7461 202d 6774  a -mt mtdata -gt
-00012640: 2067 7464 6174 6120 2d2d 706c 6f74 7379   gtdata --plotsy
-00012650: 6e0a 6060 600a 0a23 2320 496c 6c75 7374  n.```..## Illust
-00012660: 7261 7469 6f6e 0a0a 5765 2069 6c6c 7573  ration..We illus
-00012670: 7472 6174 6520 6f75 7220 7072 6f67 7261  trate our progra
-00012680: 6d20 6f6e 2061 6e20 6578 656d 706c 6172  m on an exemplar
-00012690: 7920 5747 4420 696e 6665 7265 6e63 6520  y WGD inference 
-000126a0: 616e 6420 6461 7469 6e67 2075 706f 6e20  and dating upon 
-000126b0: 7370 6563 6965 7320 2a41 7175 696c 6567  species *Aquileg
-000126c0: 6961 2063 6f65 7275 6c65 612a 2e0a 0a54  ia coerulea*...T
-000126d0: 6865 202a 4171 7569 6c65 6769 6120 636f  he *Aquilegia co
-000126e0: 6572 756c 6561 2a20 7761 7320 7265 706f  erulea* was repo
-000126f0: 7274 6564 2074 6f20 6578 7065 7269 656e  rted to experien
-00012700: 6365 2061 6e20 7061 6c65 6f2d 706f 6c79  ce an paleo-poly
-00012710: 706c 6f69 6469 7a61 7469 6f6e 2065 7665  ploidization eve
-00012720: 6e74 2061 6674 6572 2074 6865 2064 6976  nt after the div
-00012730: 6572 6765 6e63 6520 6f66 2063 6f72 6520  ergence of core 
-00012740: 6575 6469 636f 7473 2c20 7768 6963 6820  eudicots, which 
-00012750: 6973 206c 696b 656c 7920 7368 6172 6564  is likely shared
-00012760: 2062 7920 616c 6c20 5261 6e75 6e63 756c   by all Ranuncul
-00012770: 616c 6573 2e0a 0a46 6972 7374 2061 626f  ales...First abo
-00012780: 7665 2061 6c6c 2c20 6c65 7427 7320 6465  ve all, let's de
-00012790: 6c69 6e65 6174 6520 7468 6520 7768 6f6c  lineate the whol
-000127a0: 6520 7061 7261 6e6f 6d65 202a 4b2a 3c73  e paranome *K*<s
-000127b0: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-000127c0: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
-000127d0: 6861 7665 2061 2062 6173 6963 206f 6273  have a basic obs
-000127e0: 6572 7661 7469 6f6e 2066 6f72 2070 6f74  ervation for pot
-000127f0: 656e 7469 616c 6c79 2063 6f6e 6365 6976  entially conceiv
-00012800: 6162 6c65 2057 4744 732c 2075 7369 6e67  able WGDs, using
-00012810: 2074 6865 2063 6f6d 6d61 6e64 206c 696e   the command lin
-00012820: 6520 6265 6c6f 772e 0a0a 6060 600a 7767  e below...```.wg
-00012830: 6420 646d 6420 4171 7569 6c65 6769 615f  d dmd Aquilegia_
-00012840: 636f 6572 756c 6561 0a77 6764 206b 7364  coerulea.wgd ksd
-00012850: 2077 6764 5f64 6d64 2f41 7175 696c 6567   wgd_dmd/Aquileg
-00012860: 6961 5f63 6f65 7275 6c65 612e 7473 7620  ia_coerulea.tsv 
-00012870: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00012880: 6561 0a60 6060 0a0a 5468 6520 636f 6e73  ea.```..The cons
-00012890: 7472 7563 7465 6420 7768 6f6c 6520 7061  tructed whole pa
-000128a0: 7261 6e6f 6d65 202a 4b2a 3c73 7562 3e53  ranome *K*<sub>S
-000128b0: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-000128c0: 6962 7574 696f 6e20 6f66 202a 4171 7569  ibution of *Aqui
-000128d0: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
-000128e0: 6973 2061 7320 6265 6c6f 772c 2077 6520  is as below, we 
-000128f0: 6361 6e20 7365 6520 7468 6174 2074 6865  can see that the
-00012900: 7265 2073 6565 6d73 2074 6f20 6265 2061  re seems to be a
-00012910: 2068 756d 7020 6174 202a 4b2a 3c73 7562   hump at *K*<sub
-00012920: 3e53 3c2f 7375 623e 2031 2062 7574 206e  >S</sub> 1 but n
-00012930: 6f74 2063 6c65 6172 2e0a 0a21 5b5d 2864  ot clear...![](d
-00012940: 6174 612f 4171 7569 6c65 6769 615f 636f  ata/Aquilegia_co
-00012950: 6572 756c 6561 2e74 7376 2e6b 7364 5f77  erulea.tsv.ksd_w
-00012960: 702e 7376 6729 0a0a 5765 2074 6865 6e20  p.svg)..We then 
-00012970: 636f 6e73 7472 7563 7420 7468 6520 616e  construct the an
-00012980: 6368 6f72 202a 4b2a 3c73 7562 3e53 3c2f  chor *K*<sub>S</
-00012990: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-000129a0: 7574 696f 6e20 7573 696e 6720 7468 6520  ution using the 
-000129b0: 636f 6d6d 616e 6420 6c69 6e65 2062 656c  command line bel
-000129c0: 6f77 2e0a 0a60 6060 0a77 6764 2073 796e  ow...```.wgd syn
-000129d0: 202d 6620 6d52 4e41 202d 6120 4e61 6d65   -f mRNA -a Name
-000129e0: 2077 6764 5f64 6d64 2f41 7175 696c 6567   wgd_dmd/Aquileg
-000129f0: 6961 5f63 6f65 7275 6c65 612e 7473 7620  ia_coerulea.tsv 
-00012a00: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00012a10: 6561 2e67 6666 3320 2d6b 7320 7767 645f  ea.gff3 -ks wgd_
-00012a20: 6b73 642f 4171 7569 6c65 6769 615f 636f  ksd/Aquilegia_co
-00012a30: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
-00012a40: 760a 6060 600a 0a41 7320 7368 6f77 6e20  v.```..As shown 
-00012a50: 6265 6c6f 772c 2074 6865 7265 2061 7265  below, there are
-00012a60: 2073 6f6d 6520 7265 7461 696e 6564 2061   some retained a
-00012a70: 6e63 686f 7220 7061 6972 7320 7769 7468  nchor pairs with
-00012a80: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00012a90: 2062 6574 7765 656e 2031 2061 6e64 2032   between 1 and 2
-00012aa0: 2c20 7768 6963 6820 7365 656d 7320 746f  , which seems to
-00012ab0: 2073 7567 6765 7374 2061 2057 4744 2065   suggest a WGD e
-00012ac0: 7665 6e74 2e0a 0a21 5b5d 2864 6174 612f  vent...![](data/
-00012ad0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00012ae0: 6561 2e74 7376 2e6b 7364 5f77 705f 6170  ea.tsv.ksd_wp_ap
-00012af0: 2e73 7667 290a 0a54 6865 2061 7373 6f63  .svg)..The assoc
-00012b00: 6961 7465 6420 6064 7570 5374 6163 6b60  iated `dupStack`
-00012b10: 2070 6c6f 7420 7368 6f77 7320 7468 6174   plot shows that
-00012b20: 2074 6865 7265 2061 7265 206e 756d 6572   there are numer
-00012b30: 6f75 7320 6475 706c 6963 6174 6564 2073  ous duplicated s
-00012b40: 6567 6d65 6e74 7320 6163 726f 7373 206d  egments across m
-00012b50: 6f73 7420 6f66 2074 6865 2063 6872 6f6d  ost of the chrom
-00012b60: 6f73 6f6d 6573 2e0a 0a21 5b5d 2864 6174  osomes...![](dat
-00012b70: 612f 4171 7569 6c65 6769 615f 636f 6572  a/Aquilegia_coer
-00012b80: 756c 6561 5f41 7175 696c 6567 6961 5f63  ulea_Aquilegia_c
-00012b90: 6f65 7275 6c65 615f 6d75 6c74 6970 6c69  oerulea_multipli
-00012ba0: 636f 6e73 5f6c 6576 656c 2e73 7667 290a  cons_level.svg).
-00012bb0: 0a57 6520 696d 706c 656d 656e 7465 6420  .We implemented 
-00012bc0: 7477 6f20 7479 7065 7320 6f66 2064 6f74  two types of dot
-00012bd0: 2070 6c6f 7473 2069 6e20 6f78 666f 7264   plots in oxford
-00012be0: 2067 7269 643a 206f 6e65 2069 6e20 7468   grid: one in th
-00012bf0: 6520 756e 6974 206f 6620 6261 7365 7320  e unit of bases 
-00012c00: 616e 6420 7468 6520 6f74 6865 7220 696e  and the other in
-00012c10: 2074 6865 2075 6e69 7420 6f66 2067 656e   the unit of gen
-00012c20: 6573 2c20 7768 6963 6820 6361 6e20 6265  es, which can be
-00012c30: 2063 6f6c 6f72 6564 2062 7920 2a4b 2a3c   colored by *K*<
-00012c40: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
-00012c50: 6573 2067 6976 656e 202a 4b2a 3c73 7562  es given *K*<sub
-00012c60: 3e53 3c2f 7375 623e 2064 6174 612e 0a0a  >S</sub> data...
-00012c70: 215b 5d28 6461 7461 2f41 7175 696c 6567  ![](data/Aquileg
-00012c80: 6961 5f63 6f65 7275 6c65 612d 7673 2d41  ia_coerulea-vs-A
-00012c90: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00012ca0: 615f 4b73 2e64 6f74 5f75 6e69 745f 6765  a_Ks.dot_unit_ge
-00012cb0: 6e65 2e70 6e67 290a 0a41 7320 7368 6f77  ne.png)..As show
-00012cc0: 6e20 6162 6f76 652c 2074 6865 2064 6f74  n above, the dot
-00012cd0: 2070 6c6f 7420 696e 2074 6865 2075 6e69   plot in the uni
-00012ce0: 7420 6f66 2067 656e 6573 2070 7265 7365  t of genes prese
-00012cf0: 6e74 7320 6e75 6d65 726f 7573 2064 656e  nts numerous den
-00012d00: 7365 6c79 2061 6767 7265 6761 7465 6420  sely aggregated 
-00012d10: 286c 696e 652d 6c69 6b65 2920 616e 6368  (line-like) anch
-00012d20: 6f72 2070 6f69 6e74 7320 6174 206d 6f73  or points at mos
-00012d30: 7420 6f66 2074 6865 2063 6872 6f6d 6f73  t of the chromos
-00012d40: 6f6d 6573 2077 6974 6820 636f 6e73 6973  omes with consis
-00012d50: 7465 6e74 202a 4b2a 3c73 7562 3e53 3c2f  tent *K*<sub>S</
-00012d60: 7375 623e 2061 6765 2062 6574 7765 656e  sub> age between
-00012d70: 2031 2061 6e64 2032 2e20 5468 6520 646f   1 and 2. The do
-00012d80: 7420 706c 6f74 2069 6e20 7468 6520 756e  t plot in the un
-00012d90: 6974 206f 6620 6261 7365 7320 7368 6f77  it of bases show
-00012da0: 7320 7468 6520 7361 6d65 2070 6174 7465  s the same patte
-00012db0: 726e 2c20 6173 206d 616e 6966 6573 7465  rn, as manifeste
-00012dc0: 6420 6265 6c6f 772e 0a0a 215b 5d28 6461  d below...![](da
-00012dd0: 7461 2f41 7175 696c 6567 6961 5f63 6f65  ta/Aquilegia_coe
-00012de0: 7275 6c65 612d 7673 2d41 7175 696c 6567  rulea-vs-Aquileg
-00012df0: 6961 5f63 6f65 7275 6c65 615f 4b73 2e64  ia_coerulea_Ks.d
-00012e00: 6f74 2e70 6e67 290a 0a54 6865 2064 6f74  ot.png)..The dot
-00012e10: 2070 6c6f 7473 2077 6974 686f 7574 202a   plots without *
-00012e20: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-00012e30: 6e6e 6f74 6174 696f 6e20 7769 6c6c 2061  nnotation will a
-00012e40: 6c73 6f20 6265 2061 7574 6f6d 6174 656c  lso be automatel
-00012e50: 7920 7072 6f64 7563 6564 2c20 6173 2073  y produced, as s
-00012e60: 686f 776e 2062 656c 6f77 2e0a 0a21 5b5d  hown below...![]
-00012e70: 2864 6174 612f 4171 7569 6c65 6769 615f  (data/Aquilegia_
-00012e80: 636f 6572 756c 6561 2d76 732d 4171 7569  coerulea-vs-Aqui
-00012e90: 6c65 6769 615f 636f 6572 756c 6561 2e64  legia_coerulea.d
-00012ea0: 6f74 5f75 6e69 745f 6765 6e65 2e70 6e67  ot_unit_gene.png
-00012eb0: 290a 0a21 5b5d 2864 6174 612f 4171 7569  )..![](data/Aqui
-00012ec0: 6c65 6769 615f 636f 6572 756c 6561 2d76  legia_coerulea-v
-00012ed0: 732d 4171 7569 6c65 6769 615f 636f 6572  s-Aquilegia_coer
-00012ee0: 756c 6561 2e64 6f74 2e70 6e67 290a 0a4e  ulea.dot.png)..N
-00012ef0: 6f74 6520 7468 6174 2074 6865 206f 7061  ote that the opa
-00012f00: 6369 7479 206f 6620 616e 6368 6f72 2064  city of anchor d
-00012f10: 6f74 7320 616e 6420 616c 6c20 686f 6d6f  ots and all homo
-00012f20: 6c6f 6720 646f 7473 2063 616e 2062 6520  log dots can be 
-00012f30: 7365 7420 6279 2074 6865 206f 7074 696f  set by the optio
-00012f40: 6e20 6061 7061 6c70 6861 6020 616e 6420  n `apalpha` and 
-00012f50: 6068 6f61 6c70 6861 6020 7365 7061 7261  `hoalpha` separa
-00012f60: 7465 6c79 2e20 4966 206f 6e65 206a 7573  tely. If one jus
-00012f70: 7420 7761 6e74 7320 746f 2073 6565 2074  t wants to see t
-00012f80: 6865 2061 6e63 686f 7220 646f 7473 2c20  he anchor dots, 
-00012f90: 7365 7474 696e 6720 7468 6520 6068 6f61  setting the `hoa
-00012fa0: 6c70 6861 6020 6173 2030 2028 6f72 206f  lpha` as 0 (or o
-00012fb0: 7468 6572 206d 696e 7573 6375 6c65 2076  ther minuscule v
-00012fc0: 616c 7565 7329 2077 696c 6c20 646f 2e20  alues) will do. 
-00012fd0: 4966 206f 6e65 2077 616e 7473 2074 6f20  If one wants to 
-00012fe0: 7365 6520 7468 6520 6469 7374 7269 6275  see the distribu
-00012ff0: 7469 6f6e 206f 6620 7768 6f6c 6520 646f  tion of whole do
-00013000: 7473 2062 6574 7465 722c 2073 6574 7469  ts better, setti
-00013010: 6e67 2074 6865 2060 686f 616c 7068 6160  ng the `hoalpha`
-00013020: 2068 6967 6865 7220 2861 6e64 2060 6170   higher (and `ap
-00013030: 616c 7068 6160 206c 6f77 6572 2920 7769  alpha` lower) wi
-00013040: 6c6c 2064 6f2e 2054 6865 2060 646f 7473  ll do. The `dots
-00013050: 697a 6560 206f 7074 696f 6e20 6361 6e20  ize` option can 
-00013060: 6265 2063 616c 6c65 6420 746f 2061 646a  be called to adj
-00013070: 7573 7420 7468 6520 7369 7a65 206f 6620  ust the size of 
-00013080: 646f 7473 2e0a 0a41 2066 7572 7468 6572  dots...A further
-00013090: 2061 7373 6f63 6961 7465 6420 5379 6e64   associated Synd
-000130a0: 6570 7468 2070 6c6f 7420 7368 6f77 7320  epth plot shows 
-000130b0: 7468 6174 2074 6865 7265 2061 7265 206d  that there are m
-000130c0: 6f72 6520 7468 616e 2038 3020 6475 706c  ore than 80 dupl
-000130d0: 6963 6174 6564 2073 6567 6d65 6e74 732c  icated segments,
-000130e0: 2077 6869 6368 2064 6f6d 696e 6174 6573   which dominates
-000130f0: 2074 6865 2077 686f 6c65 2063 6f6c 6c69   the whole colli
-00013100: 6e65 6172 2072 6174 696f 2063 6174 6567  near ratio categ
-00013110: 6f72 792e 0a0a 215b 5d28 6461 7461 2f53  ory...![](data/S
-00013120: 796e 6465 7074 682e 7376 6729 0a0a 5765  yndepth.svg)..We
-00013130: 2063 616e 2066 6974 2061 6e20 454c 4d4d   can fit an ELMM
-00013140: 206d 6978 7475 7265 206d 6f64 656c 2075   mixture model u
-00013150: 706f 6e20 7468 6520 7768 6f6c 6520 7061  pon the whole pa
-00013160: 7261 6e6f 6d65 202a 4b2a 3c73 7562 3e53  ranome *K*<sub>S
-00013170: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-00013180: 6962 7574 696f 6e20 746f 2073 6565 206d  ibution to see m
-00013190: 6f72 6520 6163 6375 7261 7465 6c79 2074  ore accurately t
-000131a0: 6865 2073 6967 6e69 6669 6361 6e63 6520  he significance 
-000131b0: 616e 6420 6c6f 6361 7469 6f6e 206f 6620  and location of 
-000131c0: 706f 7465 6e74 6961 6c20 5747 4473 2c20  potential WGDs, 
-000131d0: 7573 696e 6720 7468 6520 636f 6d6d 616e  using the comman
-000131e0: 6420 6c69 6e65 2062 656c 6f77 2e0a 0a60  d line below...`
-000131f0: 6060 0a77 6764 2076 697a 202d 6420 7767  ``.wgd viz -d wg
-00013200: 645f 6b73 642f 4171 7569 6c65 6769 615f  d_ksd/Aquilegia_
-00013210: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
-00013220: 7473 760a 6060 600a 0a54 6865 2072 6573  tsv.```..The res
-00013230: 756c 7420 6f66 2045 4c4d 4d20 6d69 7874  ult of ELMM mixt
-00013240: 7572 6520 6d6f 6465 6c20 636c 7573 7465  ure model cluste
-00013250: 7269 6e67 2073 686f 7773 2074 6861 7420  ring shows that 
-00013260: 7468 6572 6520 6973 2061 206c 696b 656c  there is a likel
-00013270: 7920 5747 4420 636f 6d70 6f6e 656e 7420  y WGD component 
-00013280: 6174 202a 4b2a 3c73 7562 3e53 3c2f 7375  at *K*<sub>S</su
-00013290: 623e 2031 2e31 392e 0a0a 215b 5d28 6461  b> 1.19...![](da
-000132a0: 7461 2f65 6c6d 6d5f 4171 7569 6c65 6769  ta/elmm_Aquilegi
-000132b0: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
-000132c0: 732e 7473 765f 6265 7374 5f6d 6f64 656c  s.tsv_best_model
-000132d0: 735f 7765 6967 6874 6564 2e73 7667 290a  s_weighted.svg).
-000132e0: 0a4c 6574 2773 2064 6f20 6120 6d69 7874  .Let's do a mixt
-000132f0: 7572 6520 6d6f 6465 6c20 636c 7573 7465  ure model cluste
-00013300: 7269 6e67 2066 6f72 2061 6e63 686f 7220  ring for anchor 
-00013310: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013320: 746f 6f2c 2075 7369 6e67 2074 6865 2063  too, using the c
-00013330: 6f6d 6d61 6e64 206c 696e 6520 6265 6c6f  ommand line belo
-00013340: 772e 0a0a 6060 600a 7767 6420 7065 616b  w...```.wgd peak
-00013350: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
-00013360: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-00013370: 6b73 2e74 7376 202d 2d61 6e63 686f 7270  ks.tsv --anchorp
-00013380: 6f69 6e74 7320 7767 645f 7379 6e2f 6961  oints wgd_syn/ia
-00013390: 6468 6f72 652d 6f75 742f 616e 6368 6f72  dhore-out/anchor
-000133a0: 706f 696e 7473 2e74 7874 202d 2d73 6567  points.txt --seg
-000133b0: 6d65 6e74 7320 7767 645f 7379 6e2f 6961  ments wgd_syn/ia
-000133c0: 6468 6f72 652d 6f75 742f 7365 676d 656e  dhore-out/segmen
-000133d0: 7473 2e74 7874 202d 2d6c 6973 7465 6c65  ts.txt --listele
-000133e0: 6d65 6e74 7320 7767 645f 7379 6e2f 6961  ments wgd_syn/ia
-000133f0: 6468 6f72 652d 6f75 742f 6c69 7374 5f65  dhore-out/list_e
-00013400: 6c65 6d65 6e74 732e 7478 7420 2d2d 6d75  lements.txt --mu
-00013410: 6c74 6970 6c69 636f 6e70 6169 7273 2077  ltipliconpairs w
-00013420: 6764 5f73 796e 2f69 6164 686f 7265 2d6f  gd_syn/iadhore-o
-00013430: 7574 2f6d 756c 7469 706c 6963 6f6e 5f70  ut/multiplicon_p
-00013440: 6169 7273 2e74 7874 202d 2d77 6569 6768  airs.txt --weigh
-00013450: 7465 640a 6060 600a 0a54 6865 2061 6e63  ted.```..The anc
-00013460: 686f 7220 2a4b 2a3c 7375 623e 533c 2f73  hor *K*<sub>S</s
-00013470: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
-00013480: 7469 6f6e 2061 6c73 6f20 6861 7320 6120  tion also has a 
-00013490: 6c69 6b65 6c79 2057 4744 2063 6f6d 706f  likely WGD compo
-000134a0: 6e65 6e74 2077 6974 6820 6d6f 6465 2031  nent with mode 1
-000134b0: 2e32 382e 0a0a 215b 5d28 6461 7461 2f4f  .28...![](data/O
-000134c0: 7269 6769 6e61 6c5f 416e 6368 6f72 4b73  riginal_AnchorKs
-000134d0: 5f47 4d4d 5f43 6f6d 706f 6e65 6e74 335f  _GMM_Component3_
-000134e0: 6e6f 6465 5f77 6569 6768 7465 645f 4c6f  node_weighted_Lo
-000134f0: 676e 6f72 6d61 6c2e 7376 6729 0a0a 4e6f  gnormal.svg)..No
-00013500: 7720 7468 6174 2077 6520 6861 7665 2073  w that we have s
-00013510: 6565 6e20 7468 6520 6576 6964 656e 6365  een the evidence
-00013520: 206f 6620 6e75 6d65 726f 7573 2064 7570   of numerous dup
-00013530: 6c69 6361 7465 6420 7365 676d 656e 7473  licated segments
-00013540: 2061 6e64 2074 6865 2061 6767 7265 6761   and the aggrega
-00013550: 7469 6f6e 206f 6620 6475 706c 6963 6174  tion of duplicat
-00013560: 6573 2061 6765 2069 6e20 2a4b 2a3c 7375  es age in *K*<su
-00013570: 623e 533c 2f73 7562 3e20 6172 6f75 6e64  b>S</sub> around
-00013580: 2031 2e32 2066 6f72 2061 6e63 686f 7220   1.2 for anchor 
-00013590: 7061 6972 7320 616e 6420 6e6f 6e2d 616e  pairs and non-an
-000135a0: 6368 6f72 2070 6169 7273 2074 6872 6f75  chor pairs throu
-000135b0: 6768 6f75 7420 7468 6520 7768 6f6c 6520  ghout the whole 
-000135c0: 6765 6e6f 6d65 2e20 5765 2063 616e 2063  genome. We can c
-000135d0: 6c61 696d 2077 6974 6820 736f 6d65 2063  laim with some c
-000135e0: 6f6e 6669 6465 6e63 6520 7468 6174 202a  onfidence that *
-000135f0: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
-00013600: 6561 2a20 6d69 6768 7420 6861 7665 2065  ea* might have e
-00013610: 7870 6572 6965 6e63 6564 2061 2070 616c  xperienced a pal
-00013620: 656f 2d70 6f6c 7970 6c6f 6964 697a 6174  eo-polyploidizat
-00013630: 696f 6e20 6576 656e 742e 204e 6578 742c  ion event. Next,
-00013640: 204c 6574 2773 2068 6176 6520 6120 6675   Let's have a fu
-00013650: 7274 6865 7220 6c6f 6f6b 2061 626f 7574  rther look about
-00013660: 2069 7473 2070 6879 6c6f 6765 6e65 7469   its phylogeneti
-00013670: 6320 6c6f 6361 7469 6f6e 2e20 5765 206b  c location. We k
-00013680: 6e6f 7720 7468 6174 2074 6865 7265 2061  now that there a
-00013690: 7265 2075 6e63 6572 7461 696e 7469 6573  re uncertainties
-000136a0: 2061 626f 7574 2077 6865 7468 6572 2074   about whether t
-000136b0: 6869 7320 7075 7461 7469 7665 2070 616c  his putative pal
-000136c0: 656f 2d70 6f6c 7970 6c6f 6964 697a 6174  eo-polyploidizat
-000136d0: 696f 6e20 6576 656e 7420 6973 2073 6861  ion event is sha
-000136e0: 7265 6420 7769 7468 2061 6c6c 2065 7564  red with all eud
-000136f0: 6963 6f74 7320 6f72 206e 6f74 2e20 5765  icots or not. We
-00013700: 2063 616e 2063 686f 6f73 6520 736f 6d65   can choose some
-00013710: 206f 7468 6572 2065 7564 6963 6f74 2067   other eudicot g
-00013720: 656e 6f6d 6573 2074 6f20 7365 6520 7468  enomes to see th
-00013730: 6520 6f72 6465 7269 6e67 206f 6620 7370  e ordering of sp
-00013740: 6563 6961 7469 6f6e 2061 6e64 2070 6f6c  eciation and pol
-00013750: 7970 6c6f 6964 697a 6174 696f 6e20 6576  yploidization ev
-00013760: 656e 7473 2e20 4865 7265 2077 6520 6368  ents. Here we ch
-00013770: 6f6f 7365 202a 5669 7469 7320 7669 6e69  oose *Vitis vini
-00013780: 6665 7261 2a2c 202a 5072 6f74 6561 2063  fera*, *Protea c
-00013790: 796e 6172 6f69 6465 732a 2061 6e64 202a  ynaroides* and *
-000137a0: 4163 6f72 7573 2061 6d65 7269 6361 6e75  Acorus americanu
-000137b0: 732a 2069 6e20 7468 6520 666f 6c6c 6f77  s* in the follow
-000137c0: 696e 6720 2a4b 2a3c 7375 623e 533c 2f73  ing *K*<sub>S</s
-000137d0: 7562 3e20 616e 616c 7973 6973 2e20 4669  ub> analysis. Fi
-000137e0: 7273 742c 2077 6520 6275 696c 7420 6120  rst, we built a 
-000137f0: 676c 6f62 616c 204d 5242 4820 6661 6d69  global MRBH fami
-00013800: 6c79 2075 7369 6e67 2074 6865 2063 6f6d  ly using the com
-00013810: 6d61 6e64 2062 656c 6f77 2e0a 0a60 6060  mand below...```
-00013820: 0a77 6764 2064 6d64 202d 2d67 6c6f 6261  .wgd dmd --globa
-00013830: 6c6d 7262 6820 4171 7569 6c65 6769 615f  lmrbh Aquilegia_
-00013840: 636f 6572 756c 6561 2050 726f 7465 615f  coerulea Protea_
-00013850: 6379 6e61 726f 6964 6573 2041 636f 7275  cynaroides Acoru
-00013860: 735f 616d 6572 6963 616e 7573 2056 6974  s_americanus Vit
-00013870: 6973 5f76 696e 6966 6572 6120 2d6f 2077  is_vinifera -o w
-00013880: 6764 5f67 6c6f 6261 6c6d 7262 680a 6060  gd_globalmrbh.``
-00013890: 600a 0a49 6e20 7468 6520 676c 6f62 616c  `..In the global
-000138a0: 204d 5242 4820 6661 6d69 6c79 2c20 6576   MRBH family, ev
-000138b0: 6572 7920 7061 6972 206f 6620 6f72 7468  ery pair of orth
-000138c0: 6f6c 6f67 6f75 7320 6765 6e65 7320 6973  ologous genes is
-000138d0: 2074 6865 2072 6563 6970 726f 6361 6c20   the reciprocal 
-000138e0: 6265 7374 2068 6974 2c20 7375 6767 6573  best hit, sugges
-000138f0: 7469 6e67 2074 7275 6520 6f72 7468 6f6c  ting true orthol
-00013900: 6f67 6f75 7320 7265 6c61 7469 6f6e 7368  ogous relationsh
-00013910: 6970 732e 2057 6520 776f 756c 6420 7573  ips. We would us
-00013920: 6520 7468 6520 2a4b 2a3c 7375 623e 533c  e the *K*<sub>S<
-00013930: 2f73 7562 3e20 7661 6c75 6573 2061 7373  /sub> values ass
-00013940: 6f63 6961 7465 6420 7769 7468 2074 6865  ociated with the
-00013950: 7365 206f 7274 686f 6c6f 676f 7573 2070  se orthologous p
-00013960: 6169 7273 2074 6f20 6465 6c69 6d69 7420  airs to delimit 
-00013970: 7468 6520 6469 7665 7267 656e 6365 202a  the divergence *
-00013980: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2070  K*<sub>S</sub> p
-00013990: 6561 6b2e 2054 6f67 6574 6865 7220 7769  eak. Together wi
-000139a0: 7468 2074 6865 2077 686f 6c65 2070 6172  th the whole par
-000139b0: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
-000139c0: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-000139d0: 6f6e 2c20 7765 2063 6f6e 6475 6374 2074  on, we conduct t
-000139e0: 6865 2072 6174 6520 636f 7272 6563 7469  he rate correcti
-000139f0: 6f6e 2075 7369 6e67 2074 6865 2063 6f6d  on using the com
-00013a00: 6d61 6e64 2062 656c 6f77 2e0a 0a21 2153  mand below...!!S
-00013a10: 696e 6365 2060 7767 6460 2076 6572 7369  ince `wgd` versi
-00013a20: 6f6e 2032 2e30 2e32 342c 2077 6520 7265  on 2.0.24, we re
-00013a30: 7772 6f74 6520 6120 636c 6561 6e65 7220  wrote a cleaner 
-00013a40: 616e 6420 7175 6963 6b65 7220 7761 7920  and quicker way 
-00013a50: 6f66 2064 6f69 6e67 2073 7562 7374 6974  of doing substit
-00013a60: 7574 696f 6e20 7261 7465 2063 6f72 7265  ution rate corre
-00013a70: 6374 696f 6e2e 2049 7427 7320 6e6f 7420  ction. It's not 
-00013a80: 7265 7175 6972 6564 2074 6f20 7479 7065  required to type
-00013a90: 2069 6e20 616e 7920 7370 6569 6365 7320   in any speices 
-00013aa0: 7061 6972 2061 6e64 2061 2073 6572 6965  pair and a serie
-00013ab0: 7320 6f66 202a 4b2a 3c73 7562 3e53 3c2f  s of *K*<sub>S</
-00013ac0: 7375 623e 2070 6c6f 7473 2077 696c 6c20  sub> plots will 
-00013ad0: 6265 2070 726f 6475 6365 642e 2054 6865  be produced. The
-00013ae0: 2072 6571 7569 7265 6420 6669 6c65 7320   required files 
-00013af0: 6172 6520 6f72 7468 6f6c 6f67 6f75 7320  are orthologous 
-00013b00: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013b10: 7461 626c 652c 2070 6172 616c 6f67 6f75  table, paralogou
-00013b20: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
-00013b30: 3e20 7461 626c 652c 2061 2073 7065 6369  > table, a speci
-00013b40: 6573 2074 7265 6520 616e 6420 6120 666f  es tree and a fo
-00013b50: 6375 7365 6420 7370 6563 6965 7320 2874  cused species (t
-00013b60: 6865 206f 6e65 2069 6e70 7574 7465 6420  he one inputted 
-00013b70: 7769 7468 2070 6172 616c 6f67 6f75 7320  with paralogous 
-00013b80: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013b90: 6461 7461 292e 2055 7365 7273 2063 616e  data). Users can
-00013ba0: 2063 686f 6f73 6520 746f 2061 6464 206f   choose to add o
-00013bb0: 6e65 206d 6f72 6520 6c61 7965 7220 6f66  ne more layer of
-00013bc0: 2065 6c6d 6d20 6d6f 6465 6c69 6e67 206f   elmm modeling o
-00013bd0: 6e20 7061 7261 6c6f 676f 7573 202a 4b2a  n paralogous *K*
-00013be0: 3c73 7562 3e53 3c2f 7375 623e 2076 616c  <sub>S</sub> val
-00013bf0: 7565 7320 616e 642f 6f72 2067 6d6d 206d  ues and/or gmm m
-00013c00: 6f64 656c 696e 6720 6f6e 2061 6e63 686f  odeling on ancho
-00013c10: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
-00013c20: 3e20 7661 6c75 6573 2e20 5468 6520 6f72  > values. The or
-00013c30: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
-00013c40: 623e 533c 2f73 7562 3e20 7661 6c75 6573  b>S</sub> values
-00013c50: 2063 616e 2062 6520 6361 6c63 756c 6174   can be calculat
-00013c60: 6564 2075 7369 6e67 2074 6865 2063 6f6d  ed using the com
-00013c70: 6d61 6e64 2062 656c 6f77 2e0a 0a60 6060  mand below...```
-00013c80: 0a77 6764 206b 7364 2077 6764 5f67 6c6f  .wgd ksd wgd_glo
-00013c90: 6261 6c6d 7262 682f 676c 6f62 616c 5f4d  balmrbh/global_M
-00013ca0: 5242 482e 7473 7620 7365 7173 2a20 2d6f  RBH.tsv seqs* -o
-00013cb0: 2077 6764 5f67 6c6f 6261 6c6d 7262 685f   wgd_globalmrbh_
-00013cc0: 6b73 0a60 6060 0a0a 5769 7468 2074 6865  ks.```..With the
-00013cd0: 2063 616c 6375 6c61 7465 6420 6f72 7468   calculated orth
-00013ce0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
-00013cf0: 533c 2f73 7562 3e20 7461 626c 652c 2077  S</sub> table, w
-00013d00: 6520 6361 6e20 7573 6520 7468 6520 636f  e can use the co
-00013d10: 6d6d 616e 6420 6265 6c6f 7720 746f 2063  mmand below to c
-00013d20: 6f6e 6475 6374 2074 6865 2072 6174 6520  onduct the rate 
-00013d30: 636f 7272 6563 7469 6f6e 2061 6e64 2f6f  correction and/o
-00013d40: 7220 6d69 7874 7572 6520 6d6f 6465 6c69  r mixture modeli
-00013d50: 6e67 2061 6e61 6c79 7369 732e 0a0a 6060  ng analysis...``
-00013d60: 600a 7767 6420 7669 7a20 2d64 2077 6764  `.wgd viz -d wgd
-00013d70: 5f67 6c6f 6261 6c6d 7262 685f 6b73 2f67  _globalmrbh_ks/g
-00013d80: 6c6f 6261 6c5f 4d52 4248 2e74 7376 2e6b  lobal_MRBH.tsv.k
-00013d90: 732e 7473 7620 2d66 6120 4171 7569 6c65  s.tsv -fa Aquile
-00013da0: 6769 615f 636f 6572 756c 6561 202d 6570  gia_coerulea -ep
-00013db0: 6b20 4171 7569 6c65 6769 615f 636f 6572  k Aquilegia_coer
-00013dc0: 756c 6561 2e6b 732e 7473 7620 2d61 7020  ulea.ks.tsv -ap 
-00013dd0: 616e 6368 6f72 706f 696e 7473 2e74 7874  anchorpoints.txt
-00013de0: 202d 7370 2073 7065 6369 6573 7472 6565   -sp speciestree
-00013df0: 2e6e 7720 2d6f 2077 6764 5f76 697a 5f6d  .nw -o wgd_viz_m
-00013e00: 6978 6564 5f4b 7320 2d2d 706c 6f74 656c  ixed_Ks --plotel
-00013e10: 6d6d 202d 2d70 6c6f 7461 7067 6d6d 0a60  mm --plotapgmm.`
-00013e20: 6060 0a0a 6060 600a 7767 6420 6b73 6420  ``..```.wgd ksd 
-00013e30: 7767 645f 676c 6f62 616c 6d72 6268 2f67  wgd_globalmrbh/g
-00013e40: 6c6f 6261 6c5f 4d52 4248 2e74 7376 2073  lobal_MRBH.tsv s
-00013e50: 6571 732a 202d 2d65 7874 7261 7061 7261  eqs* --extrapara
-00013e60: 6e6f 6d65 6b73 2077 6764 5f6b 7364 2f41  nomeks wgd_ksd/A
-00013e70: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00013e80: 612e 7473 762e 6b73 2e74 7376 202d 7370  a.tsv.ks.tsv -sp
-00013e90: 2073 7065 6369 6573 7472 6565 2e6e 7720   speciestree.nw 
-00013ea0: 2d2d 7265 7765 6967 6874 202d 6f20 7767  --reweight -o wg
-00013eb0: 645f 676c 6f62 616c 6d72 6268 5f6b 7320  d_globalmrbh_ks 
-00013ec0: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
-00013ed0: 6961 5f63 6f65 7275 6c65 613b 5072 6f74  ia_coerulea;Prot
-00013ee0: 6561 5f63 796e 6172 6f69 6465 7322 202d  ea_cynaroides" -
-00013ef0: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
-00013f00: 615f 636f 6572 756c 6561 3b56 6974 6973  a_coerulea;Vitis
-00013f10: 5f76 696e 6966 6572 6122 202d 2d73 7061  _vinifera" --spa
-00013f20: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-00013f30: 6572 756c 6561 3b41 636f 7275 735f 616d  erulea;Acorus_am
-00013f40: 6572 6963 616e 7573 2220 2d2d 7370 6169  ericanus" --spai
-00013f50: 7220 2241 7175 696c 6567 6961 5f63 6f65  r "Aquilegia_coe
-00013f60: 7275 6c65 613b 4171 7569 6c65 6769 615f  rulea;Aquilegia_
-00013f70: 636f 6572 756c 6561 2220 2d2d 706c 6f74  coerulea" --plot
-00013f80: 6b64 6520 282d 6170 2077 6764 5f73 796e  kde (-ap wgd_syn
-00013f90: 2f69 6164 686f 7265 2d6f 7574 2f61 6e63  /iadhore-out/anc
-00013fa0: 686f 7270 6f69 6e74 732e 7478 7429 0a60  horpoints.txt).`
-00013fb0: 6060 0a0a 5468 6520 6669 6c65 2060 7370  ``..The file `sp
-00013fc0: 6563 6965 7374 7265 652e 6e77 6020 6973  eciestree.nw` is
-00013fd0: 2074 6865 2074 6578 7420 6669 6c65 206f   the text file o
-00013fe0: 6620 7370 6563 6965 7320 7472 6565 2069  f species tree i
-00013ff0: 6e20 6e65 7769 636b 2074 6861 7420 7261  n newick that ra
-00014000: 7465 2063 6f72 7265 6374 696f 6e20 776f  te correction wo
-00014010: 756c 6420 6265 2063 6f6e 6475 6374 6564  uld be conducted
-00014020: 206f 6e2e 2049 7473 2063 6f6e 7465 6e74   on. Its content
-00014030: 2069 7320 6173 2062 656c 6f77 2e20 5573   is as below. Us
-00014040: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
-00014050: 6964 6520 7468 6520 7370 6563 6965 7320  ide the species 
-00014060: 7061 6972 7320 746f 2062 6520 706c 6f74  pairs to be plot
-00014070: 7465 642e 2057 6520 7375 6767 6573 7420  ted. We suggest 
-00014080: 6164 6469 6e67 2074 6865 206f 7074 696f  adding the optio
-00014090: 6e20 602d 2d72 6577 6569 6768 7460 2074  n `--reweight` t
-000140a0: 6f20 7265 6361 6c63 756c 6174 6520 7468  o recalculate th
-000140b0: 6520 7765 6967 6874 2070 6572 2073 7065  e weight per spe
-000140c0: 6369 6573 2070 6169 7220 7375 6368 2074  cies pair such t
-000140d0: 6861 7420 7468 6520 7765 6967 6874 206f  hat the weight o
-000140e0: 6620 6f72 7468 6f6c 6f67 6f75 7320 6765  f orthologous ge
-000140f0: 6e65 2070 6169 7273 2077 696c 6c20 6265  ne pairs will be
-00014100: 636f 6d65 2031 2061 7320 7468 6520 7061  come 1 as the pa
-00014110: 7261 6c6f 676f 7573 2067 656e 6520 7061  ralogous gene pa
-00014120: 6972 732e 2054 6865 2066 6c61 6720 602d  irs. The flag `-
-00014130: 2d70 6c6f 746b 6465 6020 6361 6e20 6265  -plotkde` can be
-00014140: 2061 6464 6564 2077 6865 6e20 7468 6520   added when the 
-00014150: 6b64 6520 6375 7276 6520 6f66 206f 7274  kde curve of ort
-00014160: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
-00014170: 3e53 3c2f 7375 623e 2069 7320 6465 7369  >S</sub> is desi
-00014180: 7265 642e 2045 7874 7261 2063 6f6c 6c69  red. Extra colli
-00014190: 6e65 6172 2064 6174 6120 6361 6e20 6265  near data can be
-000141a0: 2061 6464 6564 2062 7920 7468 6520 6f70   added by the op
-000141b0: 7469 6f6e 2060 2d61 7060 2e0a 0a60 6060  tion `-ap`...```
-000141c0: 0a28 2828 5669 7469 735f 7669 6e69 6665  .(((Vitis_vinife
-000141d0: 7261 2c50 726f 7465 615f 6379 6e61 726f  ra,Protea_cynaro
-000141e0: 6964 6573 292c 4171 7569 6c65 6769 615f  ides),Aquilegia_
-000141f0: 636f 6572 756c 6561 292c 4163 6f72 7573  coerulea),Acorus
-00014200: 5f61 6d65 7269 6361 6e75 7329 3b0a 6060  _americanus);.``
-00014210: 600a 0a21 5b5d 2864 6174 612f 4171 7569  `..![](data/Aqui
-00014220: 6c65 6769 615f 636f 6572 756c 6561 5f47  legia_coerulea_G
-00014230: 6c6f 6261 6c6d 7262 684b 735f 436f 7272  lobalmrbhKs_Corr
-00014240: 6563 7465 642e 6b73 642e 7376 6729 0a0a  ected.ksd.svg)..
-00014250: 4173 2073 686f 776e 2061 626f 7665 2c20  As shown above, 
-00014260: 6265 6361 7573 6520 6f66 2074 6865 2068  because of the h
-00014270: 6967 6865 7220 7375 6273 7469 7475 7469  igher substituti
-00014280: 6f6e 2072 6174 6520 6f66 202a 4171 7569  on rate of *Aqui
-00014290: 6c65 6769 6120 636f 6572 756c 6561 2a2c  legia coerulea*,
-000142a0: 2074 6865 206f 7269 6769 6e61 6c20 6f72   the original or
-000142b0: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
-000142c0: 623e 533c 2f73 7562 3e20 7661 6c75 6573  b>S</sub> values
-000142d0: 2077 6572 6520 6163 7475 616c 6c79 2075   were actually u
-000142e0: 6e64 6572 6573 7469 6d61 7465 6420 696e  nderestimated in
-000142f0: 2074 6865 2074 696d 652d 6672 616d 6520   the time-frame 
-00014300: 6f66 202a 4171 7569 6c65 6769 6120 636f  of *Aquilegia co
-00014310: 6572 756c 6561 2a2e 2057 6865 6e20 7765  erulea*. When we
-00014320: 2072 6563 6f76 6572 6564 2074 6865 2064   recovered the d
-00014330: 6976 6572 6765 6e63 6520 7375 6273 7469  ivergence substi
-00014340: 7475 7469 6f6e 2064 6973 7461 6e63 6520  tution distance 
-00014350: 696e 2074 6572 6d73 206f 6620 7477 6f20  in terms of two 
-00014360: 7469 6d65 7320 6f66 2074 6865 2062 7261  times of the bra
-00014370: 6e63 682d 7370 6563 6966 6963 2063 6f6e  nch-specific con
-00014380: 7472 6962 7574 696f 6e20 6f66 202a 4171  tribution of *Aq
-00014390: 7569 6c65 6769 6120 636f 6572 756c 6561  uilegia coerulea
-000143a0: 2a20 7369 6e63 6520 6974 7320 6469 7665  * since its dive
-000143b0: 7267 656e 6365 2077 6974 6820 7468 6520  rgence with the 
-000143c0: 7369 7374 6572 2073 7065 6369 6573 2070  sister species p
-000143d0: 6c75 7320 7468 6520 7368 6172 6564 2073  lus the shared s
-000143e0: 7562 7374 6974 7574 696f 6e20 6469 7374  ubstitution dist
-000143f0: 616e 6365 2062 6566 6f72 6520 6469 7665  ance before dive
-00014400: 7267 656e 6365 2028 696e 2072 656c 6174  rgence (in relat
-00014410: 6976 6520 746f 2074 6865 206f 7574 6772  ive to the outgr
-00014420: 6f75 7029 2c20 7468 6520 636f 7272 6563  oup), the correc
-00014430: 7465 6420 2a4b 2a3c 7375 623e 533c 2f73  ted *K*<sub>S</s
-00014440: 7562 3e20 6d6f 6465 2062 6563 616d 6520  ub> mode became 
-00014450: 6c61 7267 6572 2e0a 0a49 6620 6f6e 6520  larger...If one 
-00014460: 6861 6420 7468 6520 6f72 7468 6f6c 6f67  had the ortholog
-00014470: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
-00014480: 7562 3e20 6461 7461 2061 6c72 6561 6479  ub> data already
-00014490: 2c20 6f6e 6520 636f 756c 6420 616c 736f  , one could also
-000144a0: 2061 7070 6c79 2074 6865 2070 726f 6772   apply the progr
-000144b0: 616d 2060 7767 6420 7669 7a60 2074 6f20  am `wgd viz` to 
-000144c0: 636f 6e64 7563 7420 7468 6520 7261 7465  conduct the rate
-000144d0: 2063 6f72 7265 6374 696f 6e20 616e 616c   correction anal
-000144e0: 7973 6973 2075 7369 6e67 2074 6865 2063  ysis using the c
-000144f0: 6f6d 6d61 6e64 2062 656c 6f77 2e20 4e6f  ommand below. No
-00014500: 7465 2074 6861 7420 7468 6520 6f72 6465  te that the orde
-00014510: 7220 6f66 2067 6976 656e 2060 7370 6169  r of given `spai
-00014520: 7260 206f 7074 696f 6e73 2064 6563 6964  r` options decid
-00014530: 6573 2074 6865 2063 6f6c 6f72 206f 6620  es the color of 
-00014540: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
-00014550: 7562 3e20 6469 7374 7269 6275 7469 6f6e  ub> distribution
-00014560: 206f 6620 6561 6368 2073 7065 6369 6573   of each species
-00014570: 2070 6169 722e 2054 6865 2061 6464 6974   pair. The addit
-00014580: 696f 6e61 6c20 6f70 7469 6f6e 2060 666f  ional option `fo
-00014590: 6375 7332 616c 6c60 2069 6e20 6077 6764  cus2all` in `wgd
-000145a0: 206b 7364 6020 616e 6420 6077 6764 2076   ksd` and `wgd v
-000145b0: 697a 6020 6361 6e20 6265 2075 7365 6420  iz` can be used 
-000145c0: 746f 2074 656c 6c20 7468 6520 7072 6f67  to tell the prog
-000145d0: 7261 6d20 7468 6174 2074 6865 2073 7065  ram that the spe
-000145e0: 6369 6573 2070 6169 7273 2061 7265 2062  cies pairs are b
-000145f0: 6574 7765 656e 2074 6865 2066 6f63 7573  etween the focus
-00014600: 2073 7065 6369 6573 2061 6e64 2061 6c6c   species and all
-00014610: 2074 6865 206f 7468 6572 2073 7065 6369   the other speci
-00014620: 6573 2073 7563 6820 7468 6174 2075 7365  es such that use
-00014630: 7273 2064 6f6e 2774 206e 6565 6420 746f  rs don't need to
-00014640: 2074 7970 6520 696e 2065 6163 6820 7061   type in each pa
-00014650: 6972 2069 6e64 6976 6964 7561 6c6c 792e  ir individually.
-00014660: 0a0a 6060 600a 7767 6420 7669 7a20 2d64  ..```.wgd viz -d
-00014670: 2077 6764 5f67 6c6f 6261 6c6d 7262 685f   wgd_globalmrbh_
-00014680: 6b73 2f67 6c6f 6261 6c5f 4d52 4248 2e74  ks/global_MRBH.t
-00014690: 7376 2e6b 732e 7473 7620 2d2d 6578 7472  sv.ks.tsv --extr
-000146a0: 6170 6172 616e 6f6d 656b 7320 7767 645f  aparanomeks wgd_
-000146b0: 6b73 642f 4171 7569 6c65 6769 615f 636f  ksd/Aquilegia_co
-000146c0: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
-000146d0: 7620 2d73 7020 7370 6563 6965 7374 7265  v -sp speciestre
-000146e0: 652e 6e77 202d 2d72 6577 6569 6768 7420  e.nw --reweight 
-000146f0: 2d61 7020 7767 645f 7379 6e2f 6961 6468  -ap wgd_syn/iadh
-00014700: 6f72 652d 6f75 742f 616e 6368 6f72 706f  ore-out/anchorpo
-00014710: 696e 7473 2e74 7874 202d 6f20 7767 645f  ints.txt -o wgd_
-00014720: 7669 7a5f 6d69 7865 645f 4b73 202d 2d73  viz_mixed_Ks --s
-00014730: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
-00014740: 636f 6572 756c 6561 3b50 726f 7465 615f  coerulea;Protea_
-00014750: 6379 6e61 726f 6964 6573 2220 2d2d 7370  cynaroides" --sp
-00014760: 6169 7220 2241 7175 696c 6567 6961 5f63  air "Aquilegia_c
-00014770: 6f65 7275 6c65 613b 5669 7469 735f 7669  oerulea;Vitis_vi
-00014780: 6e69 6665 7261 2220 2d2d 7370 6169 7220  nifera" --spair 
-00014790: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-000147a0: 6c65 613b 4163 6f72 7573 5f61 6d65 7269  lea;Acorus_ameri
-000147b0: 6361 6e75 7322 202d 2d73 7061 6972 2022  canus" --spair "
-000147c0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000147d0: 6561 3b41 7175 696c 6567 6961 5f63 6f65  ea;Aquilegia_coe
-000147e0: 7275 6c65 6122 202d 2d70 6c6f 746b 6465  rulea" --plotkde
-000147f0: 2028 2d2d 6773 6d61 7020 6765 6e65 5f73   (--gsmap gene_s
-00014800: 7065 6369 6573 2e6d 6170 290a 6060 600a  pecies.map).```.
-00014810: 0a57 6974 6820 7468 6520 6f70 7469 6f6e  .With the option
-00014820: 2060 666f 6375 7332 616c 6c60 2c20 7468   `focus2all`, th
-00014830: 6520 636f 6d6d 616e 6420 6361 6e20 6265  e command can be
-00014840: 2061 6c73 6f20 6c69 6b65 2074 6869 732c   also like this,
-00014850: 2073 616d 6520 7769 7468 2060 7767 6420   same with `wgd 
-00014860: 6b73 6460 2e0a 0a60 6060 0a77 6764 2076  ksd`...```.wgd v
-00014870: 697a 202d 6420 7767 645f 676c 6f62 616c  iz -d wgd_global
-00014880: 6d72 6268 5f6b 732f 676c 6f62 616c 5f4d  mrbh_ks/global_M
-00014890: 5242 482e 7473 762e 6b73 2e74 7376 202d  RBH.tsv.ks.tsv -
-000148a0: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
-000148b0: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
-000148c0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-000148d0: 6b73 2e74 7376 202d 7370 2073 7065 6369  ks.tsv -sp speci
-000148e0: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
-000148f0: 6967 6874 202d 6170 2077 6764 5f73 796e  ight -ap wgd_syn
-00014900: 2f69 6164 686f 7265 2d6f 7574 2f61 6e63  /iadhore-out/anc
-00014910: 686f 7270 6f69 6e74 732e 7478 7420 2d6f  horpoints.txt -o
-00014920: 2077 6764 5f76 697a 5f6d 6978 6564 5f4b   wgd_viz_mixed_K
-00014930: 7320 2d2d 666f 6375 7332 616c 6c20 4171  s --focus2all Aq
-00014940: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00014950: 202d 2d70 6c6f 746b 6465 0a60 6060 0a0a   --plotkde.```..
-00014960: 4e6f 7465 2074 6861 7420 7765 2063 616e  Note that we can
-00014970: 2065 6173 696c 7920 7368 6f77 2074 6861   easily show tha
-00014980: 7420 2a41 7175 696c 6567 6961 2063 6f65  t *Aquilegia coe
-00014990: 7275 6c65 612a 2068 6173 2068 6967 6865  rulea* has highe
-000149a0: 7220 7375 6273 7469 7475 7469 6f6e 2072  r substitution r
-000149b0: 6174 6520 7468 616e 202a 5072 6f74 6561  ate than *Protea
-000149c0: 2063 796e 6172 6f69 6465 732a 2061 6e64   cynaroides* and
-000149d0: 202a 5669 7469 7320 7669 6e69 6665 7261   *Vitis vinifera
-000149e0: 2a20 6279 2063 6f6d 7061 7269 6e67 2074  * by comparing t
-000149f0: 6865 6972 2073 7562 7374 6974 7574 696f  heir substitutio
-00014a00: 6e20 6469 7374 616e 6365 2069 6e20 7265  n distance in re
-00014a10: 6761 7264 2074 6f20 7468 6520 7361 6d65  gard to the same
-00014a20: 2064 6976 6572 6765 6e63 6520 6576 656e   divergence even
-00014a30: 7420 7769 7468 206f 7574 6772 6f75 7020  t with outgroup 
-00014a40: 7370 6563 6965 7320 2a41 636f 7275 735f  species *Acorus_
-00014a50: 616d 6572 6963 616e 7573 2a2c 2075 7369  americanus*, usi
-00014a60: 6e67 2063 6f6d 6d61 6e64 2062 656c 6f77  ng command below
-00014a70: 2e0a 0a60 6060 0a77 6764 2076 697a 202d  ...```.wgd viz -
-00014a80: 6420 7767 645f 676c 6f62 616c 6d72 6268  d wgd_globalmrbh
-00014a90: 5f6b 732f 676c 6f62 616c 5f4d 5242 482e  _ks/global_MRBH.
-00014aa0: 7473 762e 6b73 2e74 7376 202d 7370 2073  tsv.ks.tsv -sp s
-00014ab0: 7065 6369 6573 7472 6565 2e6e 7720 2d2d  peciestree.nw --
-00014ac0: 7265 7765 6967 6874 202d 6f20 7767 645f  reweight -o wgd_
-00014ad0: 7669 7a5f 436f 6d70 6172 655f 7261 7465  viz_Compare_rate
-00014ae0: 202d 2d73 7061 6972 2022 4163 6f72 7573   --spair "Acorus
-00014af0: 5f61 6d65 7269 6361 6e75 733b 5072 6f74  _americanus;Prot
-00014b00: 6561 5f63 796e 6172 6f69 6465 7322 202d  ea_cynaroides" -
-00014b10: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
-00014b20: 615f 636f 6572 756c 6561 3b41 636f 7275  a_coerulea;Acoru
-00014b30: 735f 616d 6572 6963 616e 7573 2220 2d2d  s_americanus" --
-00014b40: 7370 6169 7220 2256 6974 6973 5f76 696e  spair "Vitis_vin
-00014b50: 6966 6572 613b 4163 6f72 7573 5f61 6d65  ifera;Acorus_ame
-00014b60: 7269 6361 6e75 7322 202d 2d67 736d 6170  ricanus" --gsmap
-00014b70: 2067 656e 655f 7370 6563 6965 732e 6d61   gene_species.ma
-00014b80: 7020 2d2d 706c 6f74 6b64 650a 6060 600a  p --plotkde.```.
-00014b90: 0a21 5b5d 2864 6174 612f 5261 775f 4f72  .![](data/Raw_Or
-00014ba0: 7468 6f6c 6f67 7565 735f 436f 6d70 6172  thologues_Compar
-00014bb0: 655f 7261 7465 2e6b 7364 2e73 7667 290a  e_rate.ksd.svg).
-00014bc0: 0a41 7320 6469 7370 6c61 7965 6420 6162  .As displayed ab
-00014bd0: 6f76 652c 2074 6865 206f 7274 686f 6c6f  ove, the ortholo
-00014be0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
-00014bf0: 7375 623e 2076 616c 7565 7320 6265 7774  sub> values bewt
-00014c00: 6565 6e20 2a41 7175 696c 6567 6961 2063  een *Aquilegia c
-00014c10: 6f65 7275 6c65 612a 2061 6e64 202a 4163  oerulea* and *Ac
-00014c20: 6f72 7573 2061 6d65 7269 6361 6e75 732a  orus americanus*
-00014c30: 2068 6173 2074 6865 2068 6967 6865 7374   has the highest
-00014c40: 206d 6f64 652c 2069 6e64 6963 6174 696e   mode, indicatin
-00014c50: 6774 6865 2066 6173 7465 7220 7375 6273  gthe faster subs
-00014c60: 7469 7475 7469 6f6e 2072 6174 6520 6f66  titution rate of
-00014c70: 202a 4171 7569 6c65 6769 6120 636f 6572   *Aquilegia coer
-00014c80: 756c 6561 2a20 636f 6d70 6172 6564 2074  ulea* compared t
-00014c90: 6f20 2a50 726f 7465 6120 6379 6e61 726f  o *Protea cynaro
-00014ca0: 6964 6573 2a20 616e 6420 2a56 6974 6973  ides* and *Vitis
-00014cb0: 2076 696e 6966 6572 612a 2e0a 0a42 6566   vinifera*...Bef
-00014cc0: 6f72 6520 7632 2e30 2e32 312c 2074 6865  ore v2.0.21, the
-00014cd0: 2067 656e 652d 7370 6563 6965 7320 6d61   gene-species ma
-00014ce0: 7020 6669 6c65 2069 7320 6e65 6363 6573  p file is necces
-00014cf0: 7361 7269 6c79 206e 6565 6465 6420 666f  sarily needed fo
-00014d00: 7220 6974 7320 696d 706c 656d 656e 7461  r its implementa
-00014d10: 7469 6f6e 2069 6e20 6077 6764 2076 697a  tion in `wgd viz
-00014d20: 602c 2077 6869 6368 2073 686f 756c 6420  `, which should 
-00014d30: 6265 2061 7574 6f6d 6174 656c 7920 7072  be automately pr
-00014d40: 6f64 7563 6564 2062 7920 7468 6520 6c61  oduced by the la
-00014d50: 7374 2060 7767 6420 6b73 6460 2073 7465  st `wgd ksd` ste
-00014d60: 7020 6769 7665 6e20 7468 6520 6073 7061  p given the `spa
-00014d70: 6972 6020 616e 6420 6073 7065 6369 6573  ir` and `species
-00014d80: 7472 6565 6020 7061 7261 6d65 7465 7273  tree` parameters
-00014d90: 2e20 5468 6520 6067 656e 655f 7370 6563  . The `gene_spec
-00014da0: 6965 732e 6d61 7060 2068 6173 2063 6f6e  ies.map` has con
-00014db0: 7465 6e74 7320 6173 2062 656c 6f77 2069  tents as below i
-00014dc0: 6e20 7768 6963 6820 6561 6368 206c 696e  n which each lin
-00014dd0: 6520 6973 2074 6865 206a 6f69 6e65 6420  e is the joined 
-00014de0: 7374 7269 6e67 206f 6620 6765 6e65 206e  string of gene n
-00014df0: 616d 6520 616e 6420 7370 6563 6965 7320  ame and species 
-00014e00: 6e61 6d65 2062 7920 7370 6163 652e 2041  name by space. A
-00014e10: 6674 6572 2076 322e 302e 3231 2028 696e  fter v2.0.21 (in
-00014e20: 636c 7564 6564 292c 2074 6865 2067 656e  cluded), the gen
-00014e30: 652d 7370 6563 6965 7320 6d61 7020 6669  e-species map fi
-00014e40: 6c65 2069 7320 6e6f 7420 6e65 6363 6573  le is not necces
-00014e50: 7361 7269 6c79 206e 6565 6465 6420 616e  sarily needed an
-00014e60: 796d 6f72 652e 0a0a 6060 600a 4171 636f  ymore...```.Aqco
-00014e70: 6536 4730 3537 3830 302e 3120 4171 7569  e6G057800.1 Aqui
-00014e80: 6c65 6769 615f 636f 6572 756c 6561 0a56  legia_coerulea.V
-00014e90: 7669 5f56 4954 5f32 3031 7330 3031 3167  vi_VIT_201s0011g
-00014ea0: 3031 3533 302e 3120 5669 7469 735f 7669  01530.1 Vitis_vi
-00014eb0: 6e69 6665 7261 0a50 6379 5f50 726f 6379  nifera.Pcy_Procy
-00014ec0: 3031 6730 3835 3130 2050 726f 7465 615f  01g08510 Protea_
-00014ed0: 6379 6e61 726f 6964 6573 0a41 616d 5f41  cynaroides.Aam_A
-00014ee0: 636f 7261 2e30 3447 3134 3239 3030 2e31  cora.04G142900.1
-00014ef0: 2041 636f 7275 735f 616d 6572 6963 616e   Acorus_american
-00014f00: 7573 0a60 6060 0a0a 4120 6d6f 7265 2063  us.```..A more c
-00014f10: 6f6d 706c 6578 2070 6c6f 7420 6361 6e20  omplex plot can 
-00014f20: 6265 206d 6164 6520 6279 2061 6464 2074  be made by add t
-00014f30: 6865 2066 6c61 6720 602d 2d70 6c6f 7465  he flag `--plote
-00014f40: 6c6d 6d60 2073 7563 6820 7468 6174 2074  lmm` such that t
-00014f50: 6865 2045 4c4d 4d20 6d69 7874 7572 6520  he ELMM mixture 
-00014f60: 6d6f 6465 6c69 6e67 206f 6620 7072 6f76  modeling of prov
-00014f70: 6964 6564 2070 6172 616e 6f6d 6520 2a4b  ided paranome *K
-00014f80: 2a3c 7375 623e 533c 2f73 7562 3e20 6361  *<sub>S</sub> ca
-00014f90: 6e20 6265 2073 7570 6572 696d 706f 7365  n be superimpose
-00014fa0: 642c 2075 7369 6e67 2074 6865 2063 6f6d  d, using the com
-00014fb0: 6d61 6e64 2062 656c 6f77 2e0a 0a60 6060  mand below...```
-00014fc0: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
-00014fd0: 676c 6f62 616c 6d72 6268 5f6b 732f 676c  globalmrbh_ks/gl
-00014fe0: 6f62 616c 5f4d 5242 482e 7473 762e 6b73  obal_MRBH.tsv.ks
-00014ff0: 2e74 7376 202d 2d65 7874 7261 7061 7261  .tsv --extrapara
-00015000: 6e6f 6d65 6b73 2077 6764 5f6b 7364 2f41  nomeks wgd_ksd/A
-00015010: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00015020: 612e 7473 762e 6b73 2e74 7376 202d 7370  a.tsv.ks.tsv -sp
-00015030: 2073 7065 6369 6573 7472 6565 2e6e 7720   speciestree.nw 
-00015040: 2d2d 7265 7765 6967 6874 202d 6170 2077  --reweight -ap w
-00015050: 6764 5f73 796e 2f69 6164 686f 7265 2d6f  gd_syn/iadhore-o
-00015060: 7574 2f61 6e63 686f 7270 6f69 6e74 732e  ut/anchorpoints.
-00015070: 7478 7420 2d6f 2077 6764 5f76 697a 5f6d  txt -o wgd_viz_m
-00015080: 6978 6564 5f4b 735f 656c 6d6d 202d 2d73  ixed_Ks_elmm --s
-00015090: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
-000150a0: 636f 6572 756c 6561 3b50 726f 7465 615f  coerulea;Protea_
-000150b0: 6379 6e61 726f 6964 6573 2220 2d2d 7370  cynaroides" --sp
-000150c0: 6169 7220 2241 7175 696c 6567 6961 5f63  air "Aquilegia_c
-000150d0: 6f65 7275 6c65 613b 5669 7469 735f 7669  oerulea;Vitis_vi
-000150e0: 6e69 6665 7261 2220 2d2d 7370 6169 7220  nifera" --spair 
-000150f0: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-00015100: 6c65 613b 4163 6f72 7573 5f61 6d65 7269  lea;Acorus_ameri
-00015110: 6361 6e75 7322 202d 2d73 7061 6972 2022  canus" --spair "
-00015120: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00015130: 6561 3b41 7175 696c 6567 6961 5f63 6f65  ea;Aquilegia_coe
-00015140: 7275 6c65 6122 202d 2d67 736d 6170 2067  rulea" --gsmap g
-00015150: 656e 655f 7370 6563 6965 732e 6d61 7020  ene_species.map 
-00015160: 2d2d 706c 6f74 6b64 6520 2d2d 706c 6f74  --plotkde --plot
-00015170: 656c 6d6d 0a60 6060 0a0a 215b 5d28 6461  elmm.```..![](da
-00015180: 7461 2f41 7175 696c 6567 6961 5f63 6f65  ta/Aquilegia_coe
-00015190: 7275 6c65 615f 476c 6f62 616c 6d72 6268  rulea_Globalmrbh
-000151a0: 4b73 5f45 6c6d 6d5f 436f 7272 6563 7465  Ks_Elmm_Correcte
-000151b0: 642e 6b73 642e 7376 6729 0a0a 4672 6f6d  d.ksd.svg)..From
-000151c0: 2074 6865 206d 6978 6564 202a 4b2a 3c73   the mixed *K*<s
-000151d0: 7562 3e53 3c2f 7375 623e 2070 6c6f 7420  ub>S</sub> plot 
-000151e0: 6162 6f76 652c 2077 6520 6361 6e20 7365  above, we can se
-000151f0: 6520 7468 6174 2074 6865 206f 7074 696d  e that the optim
-00015200: 697a 6564 206c 6f67 6e6f 726d 616c 2063  ized lognormal c
-00015210: 6f6d 706f 6e65 6e74 2062 2077 6974 6820  omponent b with 
-00015220: 6d6f 6465 2031 2e32 2069 7320 796f 756e  mode 1.2 is youn
-00015230: 6765 7220 7468 616e 2074 6865 2063 6f72  ger than the cor
-00015240: 7265 6374 6564 206f 7274 686f 6c6f 676f  rected orthologo
-00015250: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-00015260: 623e 206d 6f64 6520 7769 7468 202a 5072  b> mode with *Pr
-00015270: 6f74 6561 2063 796e 6172 6f69 6465 732a  otea cynaroides*
-00015280: 2061 6e64 202a 5669 7469 7320 7669 6e69   and *Vitis vini
-00015290: 6665 7261 2a20 2831 2e33 3920 616e 6420  fera* (1.39 and 
-000152a0: 312e 3437 2c20 7265 7370 6563 7469 7665  1.47, respective
-000152b0: 6c79 292e 0a0a 4265 7369 6465 732c 2077  ly)...Besides, w
-000152c0: 6520 6361 6e20 616c 736f 2061 6464 2074  e can also add t
-000152d0: 6865 2047 4d4d 206d 6978 7475 7265 206d  he GMM mixture m
-000152e0: 6f64 656c 696e 6720 6f66 2061 6e63 686f  odeling of ancho
-000152f0: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
-00015300: 3e20 7661 6c75 6573 2077 6974 6820 7468  > values with th
-00015310: 6520 666c 6167 2060 2d2d 706c 6f74 6170  e flag `--plotap
-00015320: 676d 6d60 2c20 7573 696e 6720 7468 6520  gmm`, using the 
-00015330: 636f 6d6d 616e 6420 6265 6c6f 772e 0a0a  command below...
-00015340: 6060 600a 7767 6420 7669 7a20 2d64 2077  ```.wgd viz -d w
-00015350: 6764 5f67 6c6f 6261 6c6d 7262 685f 6b73  gd_globalmrbh_ks
-00015360: 2f67 6c6f 6261 6c5f 4d52 4248 2e74 7376  /global_MRBH.tsv
-00015370: 2e6b 732e 7473 7620 2d2d 6578 7472 6170  .ks.tsv --extrap
-00015380: 6172 616e 6f6d 656b 7320 7767 645f 6b73  aranomeks wgd_ks
-00015390: 642f 4171 7569 6c65 6769 615f 636f 6572  d/Aquilegia_coer
-000153a0: 756c 6561 2e74 7376 2e6b 732e 7473 7620  ulea.tsv.ks.tsv 
-000153b0: 2d73 7020 7370 6563 6965 7374 7265 652e  -sp speciestree.
-000153c0: 6e77 202d 2d72 6577 6569 6768 7420 2d61  nw --reweight -a
-000153d0: 7020 7767 645f 7379 6e2f 6961 6468 6f72  p wgd_syn/iadhor
-000153e0: 652d 6f75 742f 616e 6368 6f72 706f 696e  e-out/anchorpoin
-000153f0: 7473 2e74 7874 202d 6f20 7767 645f 7669  ts.txt -o wgd_vi
-00015400: 7a5f 6d69 7865 645f 4b73 5f65 6c6d 6d20  z_mixed_Ks_elmm 
-00015410: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
-00015420: 6961 5f63 6f65 7275 6c65 613b 5072 6f74  ia_coerulea;Prot
-00015430: 6561 5f63 796e 6172 6f69 6465 7322 202d  ea_cynaroides" -
-00015440: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
-00015450: 615f 636f 6572 756c 6561 3b56 6974 6973  a_coerulea;Vitis
-00015460: 5f76 696e 6966 6572 6122 202d 2d73 7061  _vinifera" --spa
-00015470: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-00015480: 6572 756c 6561 3b41 636f 7275 735f 616d  erulea;Acorus_am
-00015490: 6572 6963 616e 7573 2220 2d2d 7370 6169  ericanus" --spai
-000154a0: 7220 2241 7175 696c 6567 6961 5f63 6f65  r "Aquilegia_coe
-000154b0: 7275 6c65 613b 4171 7569 6c65 6769 615f  rulea;Aquilegia_
-000154c0: 636f 6572 756c 6561 2220 2d2d 6773 6d61  coerulea" --gsma
-000154d0: 7020 6765 6e65 5f73 7065 6369 6573 2e6d  p gene_species.m
-000154e0: 6170 202d 2d70 6c6f 746b 6465 202d 2d70  ap --plotkde --p
-000154f0: 6c6f 7461 7067 6d6d 6d0a 6060 600a 0a21  lotapgmmm.```..!
-00015500: 5b5d 2864 6174 612f 4171 7569 6c65 6769  [](data/Aquilegi
-00015510: 615f 636f 6572 756c 6561 5f47 6c6f 6261  a_coerulea_Globa
-00015520: 6c6d 7262 684b 735f 4170 676d 6d5f 436f  lmrbhKs_Apgmm_Co
-00015530: 7272 6563 7465 642e 6b73 642e 7376 6729  rrected.ksd.svg)
-00015540: 0a0a 4173 206d 616e 6966 6573 7465 6420  ..As manifested 
-00015550: 6162 6f76 652c 2074 6865 2061 6e63 686f  above, the ancho
-00015560: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
-00015570: 3e20 636f 6d70 6f6e 656e 7420 3220 7769  > component 2 wi
-00015580: 7468 206d 6f64 6520 312e 3238 2069 7320  th mode 1.28 is 
-00015590: 616c 736f 2079 6f75 6e67 6572 2074 6861  also younger tha
-000155a0: 6e20 7468 6520 636f 7272 6563 7465 6420  n the corrected 
-000155b0: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
-000155c0: 7375 623e 533c 2f73 7562 3e20 6d6f 6465  sub>S</sub> mode
-000155d0: 2077 6974 6820 2a50 726f 7465 6120 6379   with *Protea cy
-000155e0: 6e61 726f 6964 6573 2a20 616e 6420 2a56  naroides* and *V
-000155f0: 6974 6973 2076 696e 6966 6572 612a 2e20  itis vinifera*. 
-00015600: 4275 7420 7765 206e 6565 6420 746f 2062  But we need to b
-00015610: 6520 6f66 2063 6f75 7273 6520 6361 7574  e of course caut
-00015620: 696f 7573 2074 6861 7420 7375 6368 2064  ious that such d
-00015630: 6973 7469 6e63 7469 6f6e 2063 6f6d 6573  istinction comes
-00015640: 2077 6974 6820 7468 6520 756e 6365 7274   with the uncert
-00015650: 6169 6e74 6965 7320 696e 7472 6f64 7563  ainties introduc
-00015660: 6564 2066 726f 6d20 7468 6520 6170 706c  ed from the appl
-00015670: 6965 6420 6d69 7874 7572 6520 6d6f 6465  ied mixture mode
-00015680: 6c69 6e67 206d 6574 686f 646f 6c6f 6779  ling methodology
-00015690: 2069 6e20 7465 726d 7320 6f66 2066 6f72   in terms of for
-000156a0: 2069 6e73 7461 6e63 6520 6469 6666 6572   instance differ
-000156b0: 656e 7420 696e 6974 6961 6c69 7a61 7469  ent initializati
-000156c0: 6f6e 2070 6f69 6e74 7320 616e 6420 7468  on points and th
-000156d0: 6520 6973 7375 6520 6f66 206f 7665 7266  e issue of overf
-000156e0: 6974 7469 6e67 2061 6e64 2074 6865 2073  itting and the s
-000156f0: 6973 7465 7220 7370 6563 6965 7373 2061  ister speciess a
-00015700: 646f 7074 6564 2069 6e20 7468 6174 2074  dopted in that t
-00015710: 6865 7265 206d 6967 6874 2062 6520 7370  here might be sp
-00015720: 6563 6965 7320 7769 7468 206d 6f72 6520  ecies with more 
-00015730: 6469 7370 6172 6174 6520 7375 6273 7469  disparate substi
-00015740: 7475 7469 6f6e 2072 6174 6573 2074 6861  tution rates tha
-00015750: 6e20 7468 6520 6f6e 6520 7765 2063 686f  n the one we cho
-00015760: 7365 2e0a 0a41 6464 696e 6720 626f 7468  se...Adding both
-00015770: 2045 4c4d 4d20 7265 7375 6c74 2066 6f72   ELMM result for
-00015780: 2070 6172 616e 6f6d 6520 616e 6420 474d   paranome and GM
-00015790: 4d20 7265 7375 6c74 2066 6f72 2061 6e63  M result for anc
-000157a0: 686f 7220 2a4b 2a3c 7375 623e 533c 2f73  hor *K*<sub>S</s
-000157b0: 7562 3e20 6361 6e20 6265 2061 6368 6965  ub> can be achie
-000157c0: 7665 6420 6a75 7374 2062 7920 6164 6420  ved just by add 
-000157d0: 7468 6520 7477 6f20 666c 6167 7320 6d65  the two flags me
-000157e0: 6e74 696f 6e65 6420 6162 6f76 652c 2075  ntioned above, u
-000157f0: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
-00015800: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
-00015810: 2076 697a 202d 6420 7767 645f 676c 6f62   viz -d wgd_glob
-00015820: 616c 6d72 6268 5f6b 732f 676c 6f62 616c  almrbh_ks/global
-00015830: 5f4d 5242 482e 7473 762e 6b73 2e74 7376  _MRBH.tsv.ks.tsv
-00015840: 202d 2d65 7874 7261 7061 7261 6e6f 6d65   --extraparanome
-00015850: 6b73 2077 6764 5f6b 7364 2f41 7175 696c  ks wgd_ksd/Aquil
-00015860: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
-00015870: 762e 6b73 2e74 7376 202d 7370 2073 7065  v.ks.tsv -sp spe
-00015880: 6369 6573 7472 6565 2e6e 7720 2d2d 7265  ciestree.nw --re
-00015890: 7765 6967 6874 202d 6170 2077 6764 5f73  weight -ap wgd_s
-000158a0: 796e 2f69 6164 686f 7265 2d6f 7574 2f61  yn/iadhore-out/a
-000158b0: 6e63 686f 7270 6f69 6e74 732e 7478 7420  nchorpoints.txt 
-000158c0: 2d6f 2077 6764 5f76 697a 5f6d 6978 6564  -o wgd_viz_mixed
-000158d0: 5f4b 735f 656c 6d6d 202d 2d73 7061 6972  _Ks_elmm --spair
-000158e0: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
-000158f0: 756c 6561 3b50 726f 7465 615f 6379 6e61  ulea;Protea_cyna
-00015900: 726f 6964 6573 2220 2d2d 7370 6169 7220  roides" --spair 
-00015910: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-00015920: 6c65 613b 5669 7469 735f 7669 6e69 6665  lea;Vitis_vinife
-00015930: 7261 2220 2d2d 7370 6169 7220 2241 7175  ra" --spair "Aqu
-00015940: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-00015950: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
-00015960: 7322 202d 2d73 7061 6972 2022 4171 7569  s" --spair "Aqui
-00015970: 6c65 6769 615f 636f 6572 756c 6561 3b41  legia_coerulea;A
-00015980: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00015990: 6122 202d 2d67 736d 6170 2067 656e 655f  a" --gsmap gene_
-000159a0: 7370 6563 6965 732e 6d61 7020 2d2d 706c  species.map --pl
-000159b0: 6f74 6b64 6520 2d2d 706c 6f74 656c 6d6d  otkde --plotelmm
-000159c0: 202d 2d70 6c6f 7461 7067 6d6d 6d0a 6060   --plotapgmmm.``
-000159d0: 600a 0a21 5b5d 2864 6174 612f 4171 7569  `..![](data/Aqui
-000159e0: 6c65 6769 615f 636f 6572 756c 6561 5f47  legia_coerulea_G
-000159f0: 6c6f 6261 6c6d 7262 684b 735f 456c 6d6d  lobalmrbhKs_Elmm
-00015a00: 5f41 7067 6d6d 5f43 6f72 7265 6374 6564  _Apgmm_Corrected
-00015a10: 2e6b 7364 2e73 7667 290a 0a41 6e20 616c  .ksd.svg)..An al
-00015a20: 7465 726e 6174 6976 6520 7761 7920 746f  ternative way to
-00015a30: 2063 616c 6375 6c61 7465 2074 6865 206f   calculate the o
-00015a40: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
-00015a50: 7562 3e53 3c2f 7375 623e 2069 7320 746f  ub>S</sub> is to
-00015a60: 2064 6972 6563 746c 7920 7573 6520 7468   directly use th
-00015a70: 6520 6f72 7468 6f67 726f 7570 7320 696e  e orthogroups in
-00015a80: 7374 6561 6420 6f66 2067 6c6f 6261 6c20  stead of global 
-00015a90: 4d52 4248 2066 616d 696c 792e 2054 6861  MRBH family. Tha
-00015aa0: 7420 7761 7920 7765 2064 6f6e 2774 2075  t way we don't u
-00015ab0: 7365 2074 6865 2070 7265 2d69 6e66 6572  se the pre-infer
-00015ac0: 7265 6420 7061 7261 6e6f 6d65 202a 4b2a  red paranome *K*
-00015ad0: 3c73 7562 3e53 3c2f 7375 623e 2062 7574  <sub>S</sub> but
-00015ae0: 2074 6865 2070 6172 616c 6f67 6f75 7320   the paralogous 
-00015af0: 6765 6e65 2070 6169 7273 2069 6e73 6964  gene pairs insid
-00015b00: 6520 6561 6368 206f 7274 686f 6772 6f75  e each orthogrou
-00015b10: 7020 696e 7374 6561 642e 2054 6f20 6163  p instead. To ac
-00015b20: 6869 6576 6520 7468 6174 2c20 7765 2066  hieve that, we f
-00015b30: 6972 7374 206e 6565 6420 746f 2069 6e66  irst need to inf
-00015b40: 6572 206f 7274 686f 6772 6f75 7073 2075  er orthogroups u
-00015b50: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
-00015b60: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
-00015b70: 2064 6d64 2041 7175 696c 6567 6961 5f63   dmd Aquilegia_c
-00015b80: 6f65 7275 6c65 6120 5072 6f74 6561 5f63  oerulea Protea_c
-00015b90: 796e 6172 6f69 6465 7320 4163 6f72 7573  ynaroides Acorus
-00015ba0: 5f61 6d65 7269 6361 6e75 7320 5669 7469  _americanus Viti
-00015bb0: 735f 7669 6e69 6665 7261 202d 2d6f 7274  s_vinifera --ort
-00015bc0: 686f 696e 6665 7220 2d6f 2077 6764 5f6f  hoinfer -o wgd_o
-00015bd0: 7274 686f 2028 2d2d 6f6e 6c79 6f72 7468  rtho (--onlyorth
-00015be0: 6f29 200a 6060 600a 0a55 7365 7273 2063  o) .```..Users c
-00015bf0: 616e 2064 6563 6964 6520 746f 206f 6e6c  an decide to onl
-00015c00: 7920 636f 6e64 7563 7420 7468 6520 6f72  y conduct the or
-00015c10: 7468 6f67 726f 7570 2061 6e61 6c79 7369  thogroup analysi
-00015c20: 7320 7768 696c 6520 736b 6970 7069 6e67  s while skipping
-00015c30: 206f 7468 6572 2061 6e61 6c79 7369 7320   other analysis 
-00015c40: 6279 2061 6464 696e 6720 7468 6520 666c  by adding the fl
-00015c50: 6167 2060 2d2d 6f6e 6c79 6f72 7468 6f60  ag `--onlyortho`
-00015c60: 2e20 4e65 7874 2073 7465 7020 6973 2074  . Next step is t
-00015c70: 6865 2073 616d 6520 7769 7468 2067 6c6f  he same with glo
-00015c80: 6261 6c20 4d52 4248 2066 616d 696c 7920  bal MRBH family 
-00015c90: 6578 6365 7074 2074 6861 7420 7765 2064  except that we d
-00015ca0: 6f6e 2774 2075 7365 2074 6865 2065 7874  on't use the ext
-00015cb0: 7261 2070 7265 2d69 6e66 6572 7265 6420  ra pre-inferred 
-00015cc0: 7061 7261 6e6f 6d65 202a 4b2a 3c73 7562  paranome *K*<sub
-00015cd0: 3e53 3c2f 7375 623e 2061 6e79 6d6f 7265  >S</sub> anymore
-00015ce0: 2e20 5765 2069 6e66 6572 2074 6865 202a  . We infer the *
-00015cf0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2075  K*<sub>S</sub> u
-00015d00: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
-00015d10: 2062 656c 6f77 2e20 4e6f 7465 2074 6861   below. Note tha
-00015d20: 7420 7468 6520 7072 6f67 7261 6d20 6077  t the program `w
-00015d30: 6764 2076 697a 6020 6361 6e20 706c 6f74  gd viz` can plot
-00015d40: 2074 6865 2073 616d 6520 6a75 7374 2061   the same just a
-00015d50: 7320 7368 6f77 6e20 6162 6f76 652e 0a0a  s shown above...
-00015d60: 6060 600a 7767 6420 6b73 6420 7767 645f  ```.wgd ksd wgd_
-00015d70: 6f72 7468 6f2f 4f72 7468 6f67 726f 7570  ortho/Orthogroup
-00015d80: 732e 7370 2e74 7376 202d 7370 2073 7065  s.sp.tsv -sp spe
-00015d90: 6369 6573 7472 6565 2e6e 7720 2d2d 7265  ciestree.nw --re
-00015da0: 7765 6967 6874 202d 6f20 7767 645f 6f72  weight -o wgd_or
-00015db0: 7468 6f5f 6b73 202d 2d73 7061 6972 2022  tho_ks --spair "
-00015dc0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00015dd0: 6561 3b50 726f 7465 615f 6379 6e61 726f  ea;Protea_cynaro
-00015de0: 6964 6573 2220 2d2d 7370 6169 7220 2241  ides" --spair "A
-00015df0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00015e00: 613b 5669 7469 735f 7669 6e69 6665 7261  a;Vitis_vinifera
-00015e10: 2220 2d2d 7370 6169 7220 2241 7175 696c  " --spair "Aquil
-00015e20: 6567 6961 5f63 6f65 7275 6c65 613b 4163  egia_coerulea;Ac
-00015e30: 6f72 7573 5f61 6d65 7269 6361 6e75 7322  orus_americanus"
-00015e40: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
-00015e50: 6769 615f 636f 6572 756c 6561 3b41 7175  gia_coerulea;Aqu
-00015e60: 696c 6567 6961 5f63 6f65 7275 6c65 6122  ilegia_coerulea"
-00015e70: 202d 2d70 6c6f 746b 6465 2028 2d61 7020   --plotkde (-ap 
-00015e80: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
-00015e90: 6f75 742f 616e 6368 6f72 706f 696e 7473  out/anchorpoints
-00015ea0: 2e74 7874 290a 6060 600a 0a21 5b5d 2864  .txt).```..![](d
-00015eb0: 6174 612f 4171 7569 6c65 6769 615f 636f  ata/Aquilegia_co
-00015ec0: 6572 756c 6561 5f4f 7274 686f 4b73 5f43  erulea_OrthoKs_C
-00015ed0: 6f72 7265 6374 6564 2e6b 7364 2e73 7667  orrected.ksd.svg
-00015ee0: 290a 0a41 7320 7368 6f77 6e20 6162 6f76  )..As shown abov
-00015ef0: 652c 2074 6865 206e 756d 6265 7220 6f66  e, the number of
-00015f00: 2062 6f74 6820 7061 7261 6c6f 676f 7573   both paralogous
-00015f10: 2067 656e 6520 7061 6972 7320 616e 6420   gene pairs and 
-00015f20: 6f72 7468 6f6c 6f67 6f75 7320 6765 6e65  orthologous gene
-00015f30: 2070 6169 7273 2069 7320 6469 6666 6572   pairs is differ
-00015f40: 656e 7420 7468 616e 2074 6865 206f 6e65  ent than the one
-00015f50: 2066 726f 6d20 676c 6f62 616c 204d 5242   from global MRB
-00015f60: 4820 6661 6d69 6c79 2069 6e20 7468 6174  H family in that
-00015f70: 2068 6572 6520 7765 2070 6c6f 7474 6564   here we plotted
-00015f80: 2061 6c6c 206f 7274 686f 6c6f 676f 7573   all orthologous
-00015f90: 2067 656e 6520 7061 6972 7320 696e 7374   gene pairs inst
-00015fa0: 6561 6420 6f66 206f 6e6c 7920 676c 6f62  ead of only glob
-00015fb0: 616c 204d 5242 4820 616e 6420 706f 7465  al MRBH and pote
-00015fc0: 6e74 6961 6c6c 7920 6e65 7720 7061 7261  ntially new para
-00015fd0: 6c6f 676f 7573 2067 656e 6520 7061 6972  logous gene pair
-00015fe0: 7320 6d69 6768 7420 6265 2070 726f 6475  s might be produ
-00015ff0: 6365 6420 696e 2074 6865 206f 7274 686f  ced in the ortho
-00016000: 6772 6f75 7020 696e 6665 7265 6e63 6520  group inference 
-00016010: 7374 6570 2c20 746f 6765 7468 6572 2077  step, together w
-00016020: 6974 6820 6469 6666 6572 656e 7420 7265  ith different re
-00016030: 6361 6c63 756c 6174 6564 2077 6569 6768  calculated weigh
-00016040: 7473 2e0a 0a49 6620 6f6e 6520 6f6e 6c79  ts...If one only
-00016050: 2077 616e 7465 6420 746f 2070 6c6f 7420   wanted to plot 
-00016060: 7468 6520 6f72 7468 6f6c 6f67 6f75 7320  the orthologous 
-00016070: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00016080: 6469 7374 7269 6275 7469 6f6e 2c20 6173  distribution, as
-00016090: 2077 6520 6861 7665 2061 6c72 6561 6479   we have already
-000160a0: 2073 686f 776e 2061 626f 7665 2c20 6974   shown above, it
-000160b0: 2063 616e 2062 6520 6561 7369 6c79 2061   can be easily a
-000160c0: 6368 6965 7665 6420 6279 2072 656d 6f76  chieved by remov
-000160d0: 696e 6720 7468 6520 7061 7261 6c6f 676f  ing the paralogo
-000160e0: 7573 2073 7065 6369 6573 2070 6169 7220  us species pair 
-000160f0: 6041 7175 696c 6567 6961 5f63 6f65 7275  `Aquilegia_coeru
-00016100: 6c65 613b 4171 7569 6c65 6769 615f 636f  lea;Aquilegia_co
-00016110: 6572 756c 6561 602c 2075 7369 6e67 2074  erulea`, using t
-00016120: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
-00016130: 2e0a 0a60 6060 0a77 6764 2076 697a 202d  ...```.wgd viz -
-00016140: 6420 7767 645f 676c 6f62 616c 6d72 6268  d wgd_globalmrbh
-00016150: 5f6b 732f 676c 6f62 616c 5f4d 5242 482e  _ks/global_MRBH.
-00016160: 7473 762e 6b73 2e74 7376 202d 7370 2073  tsv.ks.tsv -sp s
-00016170: 7065 6369 6573 7472 6565 2e6e 7720 2d2d  peciestree.nw --
-00016180: 7265 7765 6967 6874 202d 6f20 7767 645f  reweight -o wgd_
-00016190: 676c 6f62 616c 6d72 6268 5f6f 6e6c 796f  globalmrbh_onlyo
-000161a0: 7274 686f 5f6b 7320 2d2d 7370 6169 7220  rtho_ks --spair 
-000161b0: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-000161c0: 6c65 613b 5072 6f74 6561 5f63 796e 6172  lea;Protea_cynar
-000161d0: 6f69 6465 7322 202d 2d73 7061 6972 2022  oides" --spair "
-000161e0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000161f0: 6561 3b56 6974 6973 5f76 696e 6966 6572  ea;Vitis_vinifer
-00016200: 6122 202d 2d73 7061 6972 2022 4171 7569  a" --spair "Aqui
-00016210: 6c65 6769 615f 636f 6572 756c 6561 3b41  legia_coerulea;A
-00016220: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
-00016230: 2220 2d2d 6773 6d61 7020 6765 6e65 5f73  " --gsmap gene_s
-00016240: 7065 6369 6573 2e6d 6170 0a60 6060 0a0a  pecies.map.```..
-00016250: 5765 2063 616e 2063 6c65 6172 6c79 2073  We can clearly s
-00016260: 6565 2074 6861 7420 2a56 6974 6973 2076  ee that *Vitis v
-00016270: 696e 6966 6572 612a 2068 6173 2068 6967  inifera* has hig
-00016280: 6865 7220 7375 6273 7469 7475 7469 6f6e  her substitution
-00016290: 2072 6174 6520 7468 616e 202a 5072 6f74   rate than *Prot
-000162a0: 6561 2063 796e 6172 6f69 6465 732a 2069  ea cynaroides* i
-000162b0: 6e20 7468 6174 2074 6865 6972 206f 7274  n that their ort
-000162c0: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
-000162d0: 3e53 3c2f 7375 623e 2070 6561 6b73 2077  >S</sub> peaks w
-000162e0: 6974 6820 2a41 7175 696c 6567 6961 2063  ith *Aquilegia c
-000162f0: 6f65 7275 6c65 612a 2c20 616c 7468 6f75  oerulea*, althou
-00016300: 6768 2072 6570 7265 7365 6e74 696e 6720  gh representing 
-00016310: 7468 6520 7361 6d65 2064 6976 6572 6765  the same diverge
-00016320: 6e63 6520 6576 656e 742c 2064 6966 6665  nce event, diffe
-00016330: 7265 6420 696e 2073 7562 7374 6974 7574  red in substitut
-00016340: 696f 6e20 6469 7374 616e 6365 2e0a 215b  ion distance..![
-00016350: 5d28 6461 7461 2f52 6177 5f4f 7274 686f  ](data/Raw_Ortho
-00016360: 6c6f 6775 6573 2e6b 7364 2e73 7667 290a  logues.ksd.svg).
-00016370: 0a41 6674 6572 2074 6865 2070 6879 6c6f  .After the phylo
-00016380: 6765 6e65 7469 6320 7469 6d69 6e67 206f  genetic timing o
-00016390: 6620 7468 6520 5261 6e75 6e63 756c 616c  f the Ranunculal
-000163a0: 6573 2057 4744 2c20 7765 2063 616e 2066  es WGD, we can f
-000163b0: 7572 7468 6572 2069 6e66 6572 2069 7473  urther infer its
-000163c0: 2061 6273 6f6c 7574 6520 6167 652e 2046   absolute age. F
-000163d0: 6972 7374 2077 6520 696e 6665 7220 7468  irst we infer th
-000163e0: 6520 6372 6564 6962 6c65 2072 616e 6765  e credible range
-000163f0: 206f 6620 616e 6368 6f72 2070 6169 7273   of anchor pairs
-00016400: 2062 7920 2a4b 2a3c 7375 623e 533c 2f73   by *K*<sub>S</s
-00016410: 7562 3e20 6865 7572 6973 7469 6361 6c6c  ub> heuristicall
-00016420: 7920 7573 696e 6720 7468 6520 7072 6f67  y using the prog
-00016430: 7261 6d20 6077 6764 2070 6561 6b60 2e0a  ram `wgd peak`..
-00016440: 0a60 6060 0a77 6764 2070 6561 6b20 2d2d  .```.wgd peak --
-00016450: 6865 7572 6973 7469 6320 7767 645f 6b73  heuristic wgd_ks
-00016460: 642f 4171 7569 6c65 6769 615f 636f 6572  d/Aquilegia_coer
-00016470: 756c 6561 2e74 7376 2e6b 732e 7473 7620  ulea.tsv.ks.tsv 
-00016480: 2d61 7020 7767 645f 7379 6e2f 6961 6468  -ap wgd_syn/iadh
-00016490: 6f72 652d 6f75 742f 616e 6368 6f72 706f  ore-out/anchorpo
-000164a0: 696e 7473 2e74 7874 202d 736d 2077 6764  ints.txt -sm wgd
-000164b0: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
-000164c0: 2f73 6567 6d65 6e74 732e 7478 7420 2d6c  /segments.txt -l
-000164d0: 6520 7767 645f 7379 6e2f 6961 6468 6f72  e wgd_syn/iadhor
-000164e0: 652d 6f75 742f 6c69 7374 5f65 6c65 6d65  e-out/list_eleme
-000164f0: 6e74 732e 7478 7420 2d6d 7020 7767 645f  nts.txt -mp wgd_
-00016500: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
-00016510: 6d75 6c74 6970 6c69 636f 6e5f 7061 6972  multiplicon_pair
-00016520: 732e 7478 7420 2d6f 2077 6764 5f70 6561  s.txt -o wgd_pea
-00016530: 6b0a 6060 600a 0a21 5b5d 2864 6174 612f  k.```..![](data/
-00016540: 416e 6368 6f72 4b73 5f50 6561 6b43 495f  AnchorKs_PeakCI_
-00016550: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00016560: 6561 2e74 7376 2e6b 732e 7473 765f 6e6f  ea.tsv.ks.tsv_no
-00016570: 6465 5f77 6569 6768 7465 642e 7376 6729  de_weighted.svg)
-00016580: 0a0a 4173 2073 686f 776e 2061 626f 7665  ..As shown above
-00016590: 2c20 7765 2061 7373 756d 6564 2061 206c  , we assumed a l
-000165a0: 6f67 6e6f 726d 616c 2064 6973 7472 6962  ognormal distrib
-000165b0: 7574 696f 6e20 6174 2074 6865 2070 6561  ution at the pea
-000165c0: 6b20 6c6f 6361 7469 6f6e 2064 6574 6563  k location detec
-000165d0: 7465 6420 6279 2074 6865 2060 7369 676e  ted by the `sign
-000165e0: 616c 6020 6d6f 6475 6c65 206f 6620 6073  al` module of `s
-000165f0: 6369 7079 6020 6c69 6272 6172 792e 2054  cipy` library. T
-00016600: 6865 2039 3525 2063 6f6e 6669 6465 6e63  he 95% confidenc
-00016610: 6520 6c65 7665 6c20 6f66 2074 6865 206c  e level of the l
-00016620: 6f67 6e6f 726d 616c 2064 6973 7472 6962  ognormal distrib
-00016630: 7574 696f 6e20 7761 7320 6170 706c 6965  ution was applie
-00016640: 642c 2069 2e65 2e2c 2030 2e36 382d 322e  d, i.e., 0.68-2.
-00016650: 3734 2c20 696e 2066 7572 7468 6572 206d  74, in further m
-00016660: 6f6c 6563 756c 6172 2064 6174 696e 672e  olecular dating.
-00016670: 2054 6865 2066 696c 6520 6041 7175 696c   The file `Aquil
-00016680: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
-00016690: 762e 6b73 2e74 7376 5f39 3525 4349 5f41  v.ks.tsv_95%CI_A
-000166a0: 505f 666f 725f 6461 7469 6e67 5f77 6569  P_for_dating_wei
-000166b0: 6768 7465 645f 666f 726d 6174 2e74 7376  ghted_format.tsv
-000166c0: 6020 6973 2077 6861 7420 7765 206e 6565  ` is what we nee
-000166d0: 6420 666f 7220 6e65 7874 2073 7465 702e  d for next step.
-000166e0: 2054 6f20 6275 696c 6420 7468 6520 6f72   To build the or
-000166f0: 7468 6f67 726f 7570 7320 7573 6564 2069  thogroups used i
-00016700: 6e20 7068 796c 6f67 656e 6574 6963 2064  n phylogenetic d
-00016710: 6174 696e 672c 2077 6520 6e65 6564 2074  ating, we need t
-00016720: 6f20 7365 6c65 6374 2073 6f6d 6520 7370  o select some sp
-00016730: 6563 6965 7320 616e 6420 666f 726d 2061  ecies and form a
-00016740: 2073 7461 7274 696e 6720 7472 6565 2077   starting tree w
-00016750: 6974 6820 7072 6f70 6572 2066 6f73 7369  ith proper fossi
-00016760: 6c20 6361 6c69 6272 6174 696f 6e73 2e20  l calibrations. 
-00016770: 5765 2070 726f 7669 6465 206f 6e65 2069  We provide one i
-00016780: 6e20 6d63 6d63 7472 6565 2066 6f72 6d61  n mcmctree forma
-00016790: 7420 6173 2062 656c 6f77 2e0a 0a60 6060  t as below...```
-000167a0: 0a31 3720 310a 2828 2828 506f 7461 6d6f  .17 1.((((Potamo
-000167b0: 6765 746f 6e5f 6163 7574 6966 6f6c 6975  geton_acutifoliu
-000167c0: 732c 2853 7069 726f 6465 6c61 5f69 6e74  s,(Spirodela_int
-000167d0: 6572 6d65 6469 612c 416d 6f72 7068 6f70  ermedia,Amorphop
-000167e0: 6861 6c6c 7573 5f6b 6f6e 6a61 6329 292c  hallus_konjac)),
-000167f0: 2841 6361 6e74 686f 6368 6c61 6d79 735f  (Acanthochlamys_
-00016800: 6272 6163 7465 6174 612c 2844 696f 7363  bracteata,(Diosc
-00016810: 6f72 6561 5f61 6c61 7461 2c44 696f 7363  orea_alata,Diosc
-00016820: 6f72 6561 5f72 6f74 756e 6461 7461 2929  orea_rotundata))
-00016830: 273e 302e 3536 3030 3c31 2e32 3836 3327  '>0.5600<1.2863'
-00016840: 2927 3e30 2e38 3336 303c 312e 3238 3633  )'>0.8360<1.2863
-00016850: 272c 2841 636f 7275 735f 616d 6572 6963  ',(Acorus_americ
-00016860: 616e 7573 2c41 636f 7275 735f 7461 7461  anus,Acorus_tata
-00016870: 7269 6e6f 7769 6929 2927 3e30 2e38 3336  rinowii))'>0.836
-00016880: 303c 312e 3238 3633 272c 2828 2828 5465  0<1.2863',((((Te
-00016890: 7472 6163 656e 7472 6f6e 5f73 696e 656e  tracentron_sinen
-000168a0: 7365 2c54 726f 6368 6f64 656e 6472 6f6e  se,Trochodendron
-000168b0: 5f61 7261 6c69 6f69 6465 7329 2c28 4275  _aralioides),(Bu
-000168c0: 7875 735f 6175 7374 726f 7975 6e6e 616e  xus_austroyunnan
-000168d0: 656e 7369 732c 4275 7875 735f 7369 6e69  ensis,Buxus_sini
-000168e0: 6361 2929 273e 312e 3130 3830 3c31 2e32  ca))'>1.1080<1.2
-000168f0: 3836 3327 2c28 4e65 6c75 6d62 6f5f 6e75  863',(Nelumbo_nu
-00016900: 6369 6665 7261 2c28 5465 6c6f 7065 615f  cifera,(Telopea_
-00016910: 7370 6563 696f 7369 7373 696d 612c 5072  speciosissima,Pr
-00016920: 6f74 6561 5f63 796e 6172 6f69 6465 7329  otea_cynaroides)
-00016930: 2929 273e 312e 3130 3830 3c31 2e32 3836  ))'>1.1080<1.286
-00016940: 3327 2c28 4171 7569 6c65 6769 615f 636f  3',(Aquilegia_co
-00016950: 6572 756c 6561 5f61 7031 2c41 7175 696c  erulea_ap1,Aquil
-00016960: 6567 6961 5f63 6f65 7275 6c65 615f 6170  egia_coerulea_ap
-00016970: 3229 2927 3e31 2e31 3038 303c 312e 3238  2))'>1.1080<1.28
-00016980: 3633 2729 273e 312e 3237 3230 3c32 2e34  63')'>1.2720<2.4
-00016990: 3732 3027 3b0a 6060 600a 0a41 7320 7072  720';.```..As pr
-000169a0: 6573 656e 7465 6420 6162 6f76 652c 2074  esented above, t
-000169b0: 6865 2066 6f63 7573 2073 7065 6369 6573  he focus species
-000169c0: 2074 6861 7420 6973 2061 626f 7574 2074   that is about t
-000169d0: 6f20 6265 2064 6174 6564 206e 6565 6473  o be dated needs
-000169e0: 2074 6f20 6265 2072 6570 6c61 6365 6420   to be replaced 
-000169f0: 7769 7468 2060 2841 7175 696c 6567 6961  with `(Aquilegia
-00016a00: 5f63 6f65 7275 6c65 615f 6170 312c 4171  _coerulea_ap1,Aq
-00016a10: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00016a20: 5f61 7032 2960 2e20 5769 7468 2074 6869  _ap2)`. With thi
-00016a30: 7320 7374 6172 7469 6e67 2074 7265 6520  s starting tree 
-00016a40: 616e 6420 7072 6564 6f77 6e6c 6f61 6465  and predownloade
-00016a50: 6420 6364 7320 6669 6c65 7320 6f66 2061  d cds files of a
-00016a60: 6c6c 2074 6865 2073 7065 6369 6573 2c20  ll the species, 
-00016a70: 7765 2063 616e 2062 7569 6c64 2074 6865  we can build the
-00016a80: 206f 7274 686f 6772 6f75 7020 7573 6564   orthogroup used
-00016a90: 2069 6e20 7468 6520 6669 6e61 6c20 6d6f   in the final mo
-00016aa0: 6c65 6375 6c61 7220 6461 7469 6e67 2075  lecular dating u
-00016ab0: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
-00016ac0: 2061 7320 6265 6c6f 772e 0a0a 6060 600a   as below...```.
-00016ad0: 7767 6420 646d 6420 2d66 2041 7175 696c  wgd dmd -f Aquil
-00016ae0: 6567 6961 5f63 6f65 7275 6c65 6120 2d61  egia_coerulea -a
-00016af0: 7020 7767 645f 7065 616b 2f41 7175 696c  p wgd_peak/Aquil
-00016b00: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
-00016b10: 762e 6b73 2e74 7376 5f39 3525 4349 5f41  v.ks.tsv_95%CI_A
-00016b20: 505f 666f 725f 6461 7469 6e67 5f77 6569  P_for_dating_wei
-00016b30: 6768 7465 645f 666f 726d 6174 2e74 7376  ghted_format.tsv
-00016b40: 202d 6f20 7767 645f 646d 645f 6f72 7468   -o wgd_dmd_orth
-00016b50: 6f20 506f 7461 6d6f 6765 746f 6e5f 6163  o Potamogeton_ac
-00016b60: 7574 6966 6f6c 6975 7320 5370 6972 6f64  utifolius Spirod
-00016b70: 656c 615f 696e 7465 726d 6564 6961 2041  ela_intermedia A
-00016b80: 6d6f 7270 686f 7068 616c 6c75 735f 6b6f  morphophallus_ko
-00016b90: 6e6a 6163 2041 6361 6e74 686f 6368 6c61  njac Acanthochla
-00016ba0: 6d79 735f 6272 6163 7465 6174 6120 4469  mys_bracteata Di
-00016bb0: 6f73 636f 7265 615f 616c 6174 6120 4469  oscorea_alata Di
-00016bc0: 6f73 636f 7265 615f 726f 7475 6e64 6174  oscorea_rotundat
-00016bd0: 6120 4163 6f72 7573 5f61 6d65 7269 6361  a Acorus_america
-00016be0: 6e75 7320 4163 6f72 7573 5f74 6174 6172  nus Acorus_tatar
-00016bf0: 696e 6f77 6969 2054 6574 7261 6365 6e74  inowii Tetracent
-00016c00: 726f 6e5f 7369 6e65 6e73 6520 5472 6f63  ron_sinense Troc
-00016c10: 686f 6465 6e64 726f 6e5f 6172 616c 696f  hodendron_aralio
-00016c20: 6964 6573 2042 7578 7573 5f61 7573 7472  ides Buxus_austr
-00016c30: 6f79 756e 6e61 6e65 6e73 6973 2042 7578  oyunnanensis Bux
-00016c40: 7573 5f73 696e 6963 6120 4e65 6c75 6d62  us_sinica Nelumb
-00016c50: 6f5f 6e75 6369 6665 7261 2054 656c 6f70  o_nucifera Telop
-00016c60: 6561 5f73 7065 6369 6f73 6973 7369 6d61  ea_speciosissima
-00016c70: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
-00016c80: 6573 2041 7175 696c 6567 6961 5f63 6f65  es Aquilegia_coe
-00016c90: 7275 6c65 610a 6060 600a 0a54 6865 2072  rulea.```..The r
-00016ca0: 6573 756c 7420 6669 6c65 2060 6d65 7267  esult file `merg
-00016cb0: 655f 666f 6375 735f 6170 2e74 7376 6020  e_focus_ap.tsv` 
-00016cc0: 6973 2077 6861 7420 7765 206e 6565 6420  is what we need 
-00016cd0: 666f 7220 7468 6520 6669 6e61 6c20 7374  for the final st
-00016ce0: 6570 206f 6620 6d6f 6c65 6375 6c61 7220  ep of molecular 
-00016cf0: 6461 7469 6e67 2069 6e20 7072 6f67 7261  dating in progra
-00016d00: 6d20 6077 6764 2066 6f63 7573 602e 0a0a  m `wgd focus`...
-00016d10: 6060 600a 7767 6420 666f 6375 7320 2d2d  ```.wgd focus --
-00016d20: 7072 6f74 6461 7469 6e67 202d 2d61 616d  protdating --aam
-00016d30: 6f64 656c 206c 6720 7767 645f 646d 645f  odel lg wgd_dmd_
-00016d40: 6f72 7468 6f2f 6d65 7267 655f 666f 6375  ortho/merge_focu
-00016d50: 735f 6170 2e74 7376 202d 7370 2064 6174  s_ap.tsv -sp dat
-00016d60: 696e 675f 7472 6565 2e6e 7720 2d6f 2077  ing_tree.nw -o w
-00016d70: 6764 5f64 6174 696e 6720 2d64 206d 636d  gd_dating -d mcm
-00016d80: 6374 7265 6520 2d64 7320 2762 7572 6e69  ctree -ds 'burni
-00016d90: 6e20 3d20 3230 3030 2720 2d64 7320 2773  n = 2000' -ds 's
-00016da0: 616d 7066 7265 7120 3d20 3130 3030 2720  ampfreq = 1000' 
-00016db0: 2d64 7320 276e 7361 6d70 6c65 203d 2032  -ds 'nsample = 2
-00016dc0: 3030 3030 2720 506f 7461 6d6f 6765 746f  0000' Potamogeto
-00016dd0: 6e5f 6163 7574 6966 6f6c 6975 7320 5370  n_acutifolius Sp
-00016de0: 6972 6f64 656c 615f 696e 7465 726d 6564  irodela_intermed
-00016df0: 6961 2041 6d6f 7270 686f 7068 616c 6c75  ia Amorphophallu
-00016e00: 735f 6b6f 6e6a 6163 2041 6361 6e74 686f  s_konjac Acantho
-00016e10: 6368 6c61 6d79 735f 6272 6163 7465 6174  chlamys_bracteat
-00016e20: 6120 4469 6f73 636f 7265 615f 616c 6174  a Dioscorea_alat
-00016e30: 6120 4469 6f73 636f 7265 615f 726f 7475  a Dioscorea_rotu
-00016e40: 6e64 6174 6120 4163 6f72 7573 5f61 6d65  ndata Acorus_ame
-00016e50: 7269 6361 6e75 7320 4163 6f72 7573 5f74  ricanus Acorus_t
-00016e60: 6174 6172 696e 6f77 6969 2054 6574 7261  atarinowii Tetra
-00016e70: 6365 6e74 726f 6e5f 7369 6e65 6e73 6520  centron_sinense 
-00016e80: 5472 6f63 686f 6465 6e64 726f 6e5f 6172  Trochodendron_ar
-00016e90: 616c 696f 6964 6573 2042 7578 7573 5f61  alioides Buxus_a
-00016ea0: 7573 7472 6f79 756e 6e61 6e65 6e73 6973  ustroyunnanensis
-00016eb0: 2042 7578 7573 5f73 696e 6963 6120 4e65   Buxus_sinica Ne
-00016ec0: 6c75 6d62 6f5f 6e75 6369 6665 7261 2054  lumbo_nucifera T
-00016ed0: 656c 6f70 6561 5f73 7065 6369 6f73 6973  elopea_speciosis
-00016ee0: 7369 6d61 2050 726f 7465 615f 6379 6e61  sima Protea_cyna
-00016ef0: 726f 6964 6573 2041 7175 696c 6567 6961  roides Aquilegia
-00016f00: 5f63 6f65 7275 6c65 610a 6060 600a 0a48  _coerulea.```..H
-00016f10: 6572 6520 7765 206f 6e6c 7920 696d 706c  ere we only impl
-00016f20: 656d 656e 7465 6420 7468 6520 636f 6e63  emented the conc
-00016f30: 6174 656e 6174 696f 6e20 616e 616c 7973  atenation analys
-00016f40: 6973 2075 7369 6e67 2070 726f 7465 696e  is using protein
-00016f50: 2073 6571 7565 6e63 6520 6279 2061 6464   sequence by add
-00016f60: 696e 6720 7468 6520 666c 6167 2060 2d2d  ing the flag `--
-00016f70: 7072 6f74 6461 7469 6e67 6020 616e 6420  protdating` and 
-00016f80: 7765 2073 6574 2074 6865 2070 6172 616d  we set the param
-00016f90: 6574 6572 2066 6f72 2060 6d63 6d63 7472  eter for `mcmctr
-00016fa0: 6565 6020 7669 6120 7468 6520 6f70 7469  ee` via the opti
-00016fb0: 6f6e 2060 2d64 7360 2e20 4e6f 7465 2074  on `-ds`. Note t
-00016fc0: 6861 7420 6f74 6865 7220 6461 7469 6e67  hat other dating
-00016fd0: 2070 726f 6772 616d 2073 7563 6820 6173   program such as
-00016fe0: 2060 7238 7360 2061 6e64 2060 6265 6173   `r8s` and `beas
-00016ff0: 7460 2061 7265 2061 6c73 6f20 6176 6169  t` are also avai
-00017000: 6c61 626c 6520 6769 7665 6e20 736f 6d65  lable given some
-00017010: 206d 616e 6461 746f 7279 2070 6172 616d   mandatory param
-00017020: 6574 6572 732e 2054 6865 2066 696e 616c  eters. The final
-00017030: 206c 6f67 206f 6620 7468 6520 7375 6363   log of the succ
-00017040: 6573 7366 756c 2072 756e 2069 7320 6173  essful run is as
-00017050: 2062 656c 6f77 2e0a 0a60 6060 0a31 363a   below...```.16:
-00017060: 3034 3a32 3520 494e 464f 2020 2020 2052  04:25 INFO     R
-00017070: 756e 6e69 6e67 206d 636d 6374 7265 6520  unning mcmctree 
-00017080: 7573 696e 6720 4865 7373 6961 6e20 6d61  using Hessian ma
-00017090: 7472 6978 206f 6620 4c47 2b47 616d 6d61  trix of LG+Gamma
-000170a0: 2020 636f 7265 2e70 793a 3936 370a 2020    core.py:967.  
-000170b0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000170c0: 666f 7220 7072 6f74 6569 6e20 6d6f 6465  for protein mode
-000170d0: 6c0a 3233 3a34 393a 3337 2049 4e46 4f20  l.23:49:37 INFO 
-000170e0: 2020 2020 506f 7374 6572 696f 7220 6d65      Posterior me
-000170f0: 616e 2066 6f72 2074 6865 2061 6765 7320  an for the ages 
-00017100: 6f66 2077 6764 2069 7320 3131 322e 3839  of wgd is 112.89
-00017110: 3435 206d 636d 6374 7265 652e 7079 3a32  45 mcmctree.py:2
-00017120: 3936 0a20 2020 2020 2020 2020 2020 2020  96.             
-00017130: 2020 2020 206d 696c 6c69 6f6e 2079 6561       million yea
-00017140: 7273 2066 726f 6d20 436f 6e63 6174 656e  rs from Concaten
-00017150: 6174 6564 2070 6570 7469 6465 0a20 2020  ated peptide.   
-00017160: 2020 2020 2020 2020 2020 2020 2020 2061                 a
-00017170: 6c69 676e 6d65 6e74 2061 6e64 2039 3525  lignment and 95%
-00017180: 2063 7265 6469 6269 6c69 7479 2069 6e74   credibility int
-00017190: 6572 7661 6c73 2028 4349 290a 2020 2020  ervals (CI).    
-000171a0: 2020 2020 2020 2020 2020 2020 2020 6973                is
-000171b0: 2031 3031 2e32 3234 2d31 3233 2e31 3231   101.224-123.121
-000171c0: 206d 696c 6c69 6f6e 2079 6561 7273 0a20   million years. 
-000171d0: 2020 2020 2020 2020 494e 464f 2020 2020          INFO    
-000171e0: 2054 6f74 616c 2072 756e 2074 696d 653a   Total run time:
-000171f0: 2032 3931 3735 7320 2020 2020 2020 2020   29175s         
-00017200: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00017210: 2020 2020 2063 6c69 2e70 793a 3234 310a       cli.py:241.
-00017220: 2020 2020 2020 2020 2049 4e46 4f20 2020           INFO   
-00017230: 2020 446f 6e65 2020 2020 2020 2020 2020    Done          
-00017240: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00000000: 4d65 7461 6461 7461 2d56 6572 7369 6f6e  Metadata-Version
+00000010: 3a20 322e 310a 4e61 6d65 3a20 7767 640a  : 2.1.Name: wgd.
+00000020: 5665 7273 696f 6e3a 2032 2e30 2e33 350a  Version: 2.0.35.
+00000030: 5375 6d6d 6172 793a 2077 6764 0a48 6f6d  Summary: wgd.Hom
+00000040: 652d 7061 6765 3a20 6874 7470 3a2f 2f67  e-page: http://g
+00000050: 6974 6875 622e 636f 6d2f 6865 6368 652d  ithub.com/heche-
+00000060: 7073 622f 7767 640a 4175 7468 6f72 3a20  psb/wgd.Author: 
+00000070: 4865 6e67 6368 6920 4368 656e 0a41 7574  Hengchi Chen.Aut
+00000080: 686f 722d 656d 6169 6c3a 2068 6563 6865  hor-email: heche
+00000090: 4070 7362 2e76 6962 2d75 6765 6e74 2e62  @psb.vib-ugent.b
+000000a0: 650a 4c69 6365 6e73 653a 2047 504c 0a44  e.License: GPL.D
+000000b0: 6573 6372 6970 7469 6f6e 2d43 6f6e 7465  escription-Conte
+000000c0: 6e74 2d54 7970 653a 2074 6578 742f 6d61  nt-Type: text/ma
+000000d0: 726b 646f 776e 0a4c 6963 656e 7365 2d46  rkdown.License-F
+000000e0: 696c 653a 204c 4943 454e 5345 0a0a 3c64  ile: LICENSE..<d
+000000f0: 6976 2061 6c69 676e 3d22 6365 6e74 6572  iv align="center
+00000100: 223e 0a0a 2320 6077 6764 2076 3260 203a  ">..# `wgd v2` :
+00000110: 2061 2073 7569 7465 2074 6f6f 6c20 6f66   a suite tool of
+00000120: 2057 4744 2069 6e66 6572 656e 6365 2061   WGD inference a
+00000130: 6e64 2074 696d 696e 670a 0a5b 215b 4275  nd timing..[![Bu
+00000140: 696c 6420 5374 6174 7573 5d28 6874 7470  ild Status](http
+00000150: 733a 2f2f 6170 702e 7472 6176 6973 2d63  s://app.travis-c
+00000160: 692e 636f 6d2f 6865 6368 652d 7073 622f  i.com/heche-psb/
+00000170: 7767 642e 7376 673f 6272 616e 6368 3d70  wgd.svg?branch=p
+00000180: 6879 6c6f 6461 7469 6e67 295d 2868 7474  hylodating)](htt
+00000190: 7073 3a2f 2f61 7070 2e74 7261 7669 732d  ps://app.travis-
+000001a0: 6369 2e63 6f6d 2f68 6563 6865 2d70 7362  ci.com/heche-psb
+000001b0: 2f77 6764 290a 5b21 5b44 6f63 756d 656e  /wgd).[![Documen
+000001c0: 7461 7469 6f6e 2053 7461 7475 735d 2868  tation Status](h
+000001d0: 7474 7073 3a2f 2f72 6561 6474 6865 646f  ttps://readthedo
+000001e0: 6373 2e6f 7267 2f70 726f 6a65 6374 732f  cs.org/projects/
+000001f0: 7767 6476 322f 6261 6467 652f 3f76 6572  wgdv2/badge/?ver
+00000200: 7369 6f6e 3d6c 6174 6573 7429 5d28 6874  sion=latest)](ht
+00000210: 7470 733a 2f2f 7767 6476 322e 7265 6164  tps://wgdv2.read
+00000220: 7468 6564 6f63 732e 696f 2f65 6e2f 6c61  thedocs.io/en/la
+00000230: 7465 7374 2f3f 6261 6467 653d 6c61 7465  test/?badge=late
+00000240: 7374 290a 5b21 5b6c 6963 656e 7365 5d28  st).[![license](
+00000250: 6874 7470 733a 2f2f 696d 672e 7368 6965  https://img.shie
+00000260: 6c64 732e 696f 2f70 7970 692f 6c2f 7767  lds.io/pypi/l/wg
+00000270: 642e 7376 6729 5d28 6874 7470 733a 2f2f  d.svg)](https://
+00000280: 7079 7069 2e70 7974 686f 6e2e 6f72 672f  pypi.python.org/
+00000290: 7079 7069 2f77 6764 290a 5b21 5b4c 6174  pypi/wgd).[![Lat
+000002a0: 6573 7420 5079 5049 2076 6572 7369 6f6e  est PyPI version
+000002b0: 5d28 6874 7470 733a 2f2f 696d 672e 7368  ](https://img.sh
+000002c0: 6965 6c64 732e 696f 2f70 7970 692f 762f  ields.io/pypi/v/
+000002d0: 7767 642e 7376 6729 5d28 6874 7470 733a  wgd.svg)](https:
+000002e0: 2f2f 7079 7069 2e70 7974 686f 6e2e 6f72  //pypi.python.or
+000002f0: 672f 7079 7069 2f77 6764 290a 5b21 5b69  g/pypi/wgd).[![i
+00000300: 6e73 7461 6c6c 2077 6974 6820 6269 6f63  nstall with bioc
+00000310: 6f6e 6461 5d28 6874 7470 733a 2f2f 696d  onda](https://im
+00000320: 672e 7368 6965 6c64 732e 696f 2f62 6164  g.shields.io/bad
+00000330: 6765 2f69 6e73 7461 6c6c 2532 3077 6974  ge/install%20wit
+00000340: 682d 6269 6f63 6f6e 6461 2d62 7269 6768  h-bioconda-brigh
+00000350: 7467 7265 656e 2e73 7667 3f73 7479 6c65  tgreen.svg?style
+00000360: 3d66 6c61 7429 5d28 6874 7470 3a2f 2f62  =flat)](http://b
+00000370: 696f 636f 6e64 612e 6769 7468 7562 2e69  ioconda.github.i
+00000380: 6f2f 7265 6369 7065 732f 7767 642f 5245  o/recipes/wgd/RE
+00000390: 4144 4d45 2e68 746d 6c29 0a5b 215b 446f  ADME.html).[![Do
+000003a0: 776e 6c6f 6164 735d 2868 7474 7073 3a2f  wnloads](https:/
+000003b0: 2f70 6570 792e 7465 6368 2f62 6164 6765  /pepy.tech/badge
+000003c0: 2f77 6764 295d 2868 7474 7073 3a2f 2f70  /wgd)](https://p
+000003d0: 6570 792e 7465 6368 2f70 726f 6a65 6374  epy.tech/project
+000003e0: 2f77 6764 290a 0a2a 2a48 656e 6763 6869  /wgd)..**Hengchi
+000003f0: 2043 6865 6e2c 2041 7274 6875 7220 5a77   Chen, Arthur Zw
+00000400: 6165 6e65 706f 656c 2c20 5976 6573 2056  aenepoel, Yves V
+00000410: 616e 2064 6520 5065 6572 2a2a 0a0a 5b2a  an de Peer**..[*
+00000420: 2a42 696f 696e 666f 726d 6174 6963 7320  *Bioinformatics 
+00000430: 2620 4576 6f6c 7574 696f 6e61 7279 2047  & Evolutionary G
+00000440: 656e 6f6d 6963 7320 4772 6f75 702a 2a5d  enomics Group**]
+00000450: 2868 7474 7073 3a2f 2f77 7777 2e76 616e  (https://www.van
+00000460: 6465 7065 6572 6c61 622e 6f72 672f 292a  depeerlab.org/)*
+00000470: 2a2c 2056 4942 2d55 4765 6e74 2043 656e  *, VIB-UGent Cen
+00000480: 7465 7220 666f 7220 506c 616e 7420 5379  ter for Plant Sy
+00000490: 7374 656d 7320 4269 6f6c 6f67 792a 2a0a  stems Biology**.
+000004a0: 0a5b 2a2a 496e 7472 6f64 7563 7469 6f6e  .[**Introduction
+000004b0: 2a2a 5d28 2369 6e74 726f 6475 6374 696f  **](#introductio
+000004c0: 6e29 207c 200a 5b2a 2a49 6e73 7461 6c6c  n) | .[**Install
+000004d0: 6174 696f 6e2a 2a5d 2823 696e 7374 616c  ation**](#instal
+000004e0: 6c61 7469 6f6e 2920 7c20 0a5b 2a2a 5061  lation) | .[**Pa
+000004f0: 7261 6d65 7465 7273 2a2a 5d28 2370 6172  rameters**](#par
+00000500: 616d 6574 6572 7329 207c 200a 5b2a 2a55  ameters) | .[**U
+00000510: 7361 6765 2a2a 5d28 2375 7361 6765 2920  sage**](#usage) 
+00000520: 7c20 0a5b 2a2a 496c 6c75 7374 7261 7469  | .[**Illustrati
+00000530: 6f6e 2a2a 5d28 2369 6c6c 7573 7472 6174  on**](#illustrat
+00000540: 696f 6e29 207c 0a5b 2a2a 446f 6375 6d65  ion) |.[**Docume
+00000550: 6e74 6174 696f 6e2a 2a5d 2868 7474 7073  ntation**](https
+00000560: 3a2f 2f77 6764 7632 2e72 6561 6474 6865  ://wgdv2.readthe
+00000570: 646f 6373 2e69 6f2f 656e 2f6c 6174 6573  docs.io/en/lates
+00000580: 742f 3f62 6164 6765 3d6c 6174 6573 7429  t/?badge=latest)
+00000590: 207c 0a5b 2a2a 4369 7461 7469 6f6e 2a2a   |.[**Citation**
+000005a0: 5d28 2363 6974 6174 696f 6e29 0a0a 3c2f  ](#citation)..</
+000005b0: 6469 763e 0a0a 6077 6764 2076 3260 2069  div>..`wgd v2` i
+000005c0: 7320 6120 7079 7468 6f6e 2070 6163 6b61  s a python packa
+000005d0: 6765 2075 7067 7261 6465 6420 6672 6f6d  ge upgraded from
+000005e0: 2074 6865 206f 7269 6769 6e61 6c20 6077   the original `w
+000005f0: 6764 6020 7061 636b 6167 6520 6169 6d69  gd` package aimi
+00000600: 6e67 2066 6f72 2074 6865 2069 6e66 6572  ng for the infer
+00000610: 656e 6365 2061 6e64 2074 696d 696e 6720  ence and timing 
+00000620: 6f66 2061 6e63 6965 6e74 2077 686f 6c65  of ancient whole
+00000630: 2d67 656e 6f6d 6520 6475 706c 6963 6174  -genome duplicat
+00000640: 696f 6e20 2857 4744 2920 6576 656e 7473  ion (WGD) events
+00000650: 2e20 466f 7220 7468 6520 7072 6f70 6f73  . For the propos
+00000660: 6520 6f66 2069 6c6c 7573 7472 6174 696e  e of illustratin
+00000670: 6720 7468 6520 7072 696e 6369 706c 6520  g the principle 
+00000680: 616e 6420 7573 6167 6520 6f66 2060 7767  and usage of `wg
+00000690: 6420 7632 602c 2077 6520 636f 6d70 696c  d v2`, we compil
+000006a0: 6564 2074 6869 7320 646f 6375 6d65 6e74  ed this document
+000006b0: 6174 696f 6e2e 2042 656c 6f77 2077 6520  ation. Below we 
+000006c0: 6669 7273 7420 6761 7665 2061 6e20 696e  first gave an in
+000006d0: 7472 6f64 7563 7469 6f6e 206f 7665 7220  troduction over 
+000006e0: 7468 6520 7363 6f70 6520 616e 6420 6d65  the scope and me
+000006f0: 6368 616e 6973 6d20 6f66 2060 7767 6420  chanism of `wgd 
+00000700: 7632 6020 616e 6420 7468 656e 2074 6865  v2` and then the
+00000710: 2070 7261 6374 6963 616c 2069 6e66 6f72   practical infor
+00000720: 6d61 7469 6f6e 2070 6572 7461 696e 696e  mation pertainin
+00000730: 6720 746f 2074 6865 2069 6e73 7461 6c6c  g to the install
+00000740: 6174 696f 6e20 616e 6420 7573 6167 652e  ation and usage.
+00000750: 2041 6e20 6578 616d 706c 6172 2077 6f72   An examplar wor
+00000760: 6b66 6c6f 7720 6973 2070 726f 7669 6465  kflow is provide
+00000770: 6420 696e 2074 6865 2074 7574 6f72 6961  d in the tutoria
+00000780: 6c20 7365 6374 696f 6e20 6f6e 2068 6f77  l section on how
+00000790: 2074 6f20 7365 656b 2065 7669 6465 6e63   to seek evidenc
+000007a0: 6520 666f 7220 6120 7075 7461 7469 7665  e for a putative
+000007b0: 2057 4744 2065 7665 6e74 2061 6e64 2070   WGD event and p
+000007c0: 6572 666f 726d 2070 726f 7065 7220 7469  erform proper ti
+000007d0: 6d69 6e67 2077 6974 6820 6120 6672 6573  ming with a fres
+000007e0: 686c 7920 6f62 7461 696e 6564 2067 656e  hly obtained gen
+000007f0: 6f6d 6520 6173 7365 6d62 6c79 2069 6e20  ome assembly in 
+00000800: 6861 6e64 2e20 466f 7220 7468 6f73 6520  hand. For those 
+00000810: 7768 6f20 6172 6520 696e 7465 7265 7374  who are interest
+00000820: 6564 2069 6e20 6d6f 7265 2074 6865 6f72  ed in more theor
+00000830: 6574 6963 616c 2064 6574 6169 6c73 2c20  etical details, 
+00000840: 7765 2072 6563 6f6d 6d65 6e64 2074 7572  we recommend tur
+00000850: 6e69 6e67 2074 6f20 6f75 7220 7061 7065  ning to our pape
+00000860: 7220 616e 6420 626f 6f6b 2063 6861 7074  r and book chapt
+00000870: 6572 2066 6f72 206d 6f72 6520 6465 7461  er for more deta
+00000880: 696c 6564 2064 6573 6372 6970 7469 6f6e  iled description
+00000890: 2061 6e64 2069 6e73 6967 6874 6675 6c20   and insightful 
+000008a0: 6469 7363 7573 7369 6f6e 732e 2054 6865  discussions. The
+000008b0: 206b 6579 2069 6d70 726f 7665 6420 6665   key improved fe
+000008c0: 6174 7572 6573 206f 6620 6077 6764 2076  atures of `wgd v
+000008d0: 3260 2061 7265 2064 656d 6f6e 7374 7261  2` are demonstra
+000008e0: 7465 6420 696e 206f 7572 206c 6174 6573  ted in our lates
+000008f0: 7420 6d61 6e75 7363 7269 7074 206f 6e20  t manuscript on 
+00000900: 5b42 696f 696e 666f 726d 6174 6963 735d  [Bioinformatics]
+00000910: 2868 7474 7073 3a2f 2f64 6f69 2e6f 7267  (https://doi.org
+00000920: 2f31 302e 3130 3933 2f62 696f 696e 666f  /10.1093/bioinfo
+00000930: 726d 6174 6963 732f 6274 6165 3237 3229  rmatics/btae272)
+00000940: 2e20 4966 2079 6f75 2075 7365 2060 7767  . If you use `wg
+00000950: 6420 7632 6020 696e 2079 6f75 7220 7265  d v2` in your re
+00000960: 7365 6172 6368 2c20 706c 6561 7365 2063  search, please c
+00000970: 6974 6520 7573 2061 7320 7375 6767 6573  ite us as sugges
+00000980: 7465 6420 696e 205b 4369 7461 7469 6f6e  ted in [Citation
+00000990: 5d28 2363 6974 6174 696f 6e29 2073 6563  ](#citation) sec
+000009a0: 7469 6f6e 2e0a 0a23 2320 496e 7472 6f64  tion...## Introd
+000009b0: 7563 7469 6f6e 0a0a 506f 6c79 706c 6f69  uction..Polyploi
+000009c0: 6469 7a61 7469 6f6e 732c 2074 6865 2065  dizations, the e
+000009d0: 766f 6c75 7469 6f6e 6172 7920 7072 6f63  volutionary proc
+000009e0: 6573 7320 7468 6174 2074 6865 2065 6e74  ess that the ent
+000009f0: 6972 6520 6765 6e6f 6d65 206f 6620 616e  ire genome of an
+00000a00: 206f 7267 616e 6973 6d20 6973 2064 7570   organism is dup
+00000a10: 6c69 6361 7465 642c 2061 6c73 6f20 6e61  licated, also na
+00000a20: 6d65 6420 6173 2077 686f 6c65 2d67 656e  med as whole-gen
+00000a30: 6f6d 6520 6475 706c 6963 6174 696f 6e73  ome duplications
+00000a40: 2028 5747 4473 292c 206f 6363 7572 2072   (WGDs), occur r
+00000a50: 6563 7572 7265 6e74 6c79 2061 6372 6f73  ecurrently acros
+00000a60: 7320 7468 6520 7472 6565 206f 6620 6c69  s the tree of li
+00000a70: 6665 2e20 5468 6572 6520 6172 6520 7477  fe. There are tw
+00000a80: 6f20 6d6f 6465 7320 6f66 2070 6f6c 7970  o modes of polyp
+00000a90: 6c6f 6964 697a 6174 696f 6e73 2c20 6175  loidizations, au
+00000aa0: 746f 706f 6c79 706c 6f69 6469 7a61 7469  topolyploidizati
+00000ab0: 6f6e 7320 616e 6420 616c 6c6f 706f 6c79  ons and allopoly
+00000ac0: 706c 6f69 6469 7a61 7469 6f6e 732e 2041  ploidizations. A
+00000ad0: 7574 6f70 6f6c 7970 6c6f 6964 697a 6174  utopolyploidizat
+00000ae0: 696f 6e73 2061 7265 2074 6865 2064 7570  ions are the dup
+00000af0: 6c69 6361 7469 6f6e 206f 6620 7468 6520  lication of the 
+00000b00: 7361 6d65 2067 656e 6f6d 652c 2072 6573  same genome, res
+00000b10: 756c 7469 6e67 2069 6e20 7477 6f20 6964  ulting in two id
+00000b20: 656e 7469 6361 6c20 7375 6267 656e 6f6d  entical subgenom
+00000b30: 6573 2061 7420 7468 6520 7469 6d65 2069  es at the time i
+00000b40: 7420 656d 6572 6765 642e 2057 6869 6c65  t emerged. While
+00000b50: 2074 6865 2061 6c6c 6f70 6f6c 7970 6c6f   the allopolyplo
+00000b60: 6964 697a 6174 696f 6e73 2061 7265 206e  idizations are n
+00000b70: 6f72 6d61 6c6c 7920 6163 6869 6576 6564  ormally achieved
+00000b80: 2069 6e20 7477 6f20 7374 6570 732c 2066   in two steps, f
+00000b90: 6972 7374 2074 6865 2068 7962 7269 6469  irst the hybridi
+00000ba0: 7a61 7469 6f6e 2062 6574 7765 656e 2074  zation between t
+00000bb0: 776f 2064 6966 6665 7265 6e74 2073 7065  wo different spe
+00000bc0: 6369 6573 2c20 7265 7375 6c74 696e 6720  cies, resulting 
+00000bd0: 696e 2074 6865 2061 7269 7369 6e67 206f  in the arising o
+00000be0: 6620 7472 616e 7369 656e 7420 686f 6d6f  f transient homo
+00000bf0: 706c 6f69 6479 2c73 6563 6f6e 6420 7468  ploidy,second th
+00000c00: 6520 6475 706c 6963 6174 696f 6e20 6f66  e duplication of
+00000c10: 2074 6865 2068 6f6d 6f70 6c6f 6964 792c   the homoploidy,
+00000c20: 2072 6573 756c 7469 6e67 2069 6e20 7468   resulting in th
+00000c30: 6520 656d 6572 6765 6e63 6520 6f66 2061  e emergence of a
+00000c40: 6c6c 6f70 6f6c 7970 6c6f 6964 792e 2044  llopolyploidy. D
+00000c50: 7565 2074 6f20 7468 6520 756e 7374 6162  ue to the unstab
+00000c60: 696c 6974 7920 616e 6420 756e 6261 6c61  ility and unbala
+00000c70: 6e63 6564 2074 6574 7261 736f 6d69 6320  nced tetrasomic 
+00000c80: 696e 6865 7269 7461 6e63 652c 2066 6f72  inheritance, for
+00000c90: 2069 6e73 7461 6e63 6520 7468 6520 6e75   instance the nu
+00000ca0: 636c 6561 722d 6379 746f 706c 6173 6d69  clear-cytoplasmi
+00000cb0: 6320 696e 636f 6d70 6174 6962 696c 6974  c incompatibilit
+00000cc0: 792c 2074 6865 2070 6f6c 7970 6c6f 6964  y, the polyploid
+00000cd0: 7920 6765 6e6f 6d65 2077 696c 6c20 7468  y genome will th
+00000ce0: 656e 2065 7870 6572 6965 6e63 6520 6120  en experience a 
+00000cf0: 7072 6f63 6573 7320 6361 6c6c 6564 2064  process called d
+00000d00: 6970 6c6f 6964 697a 6174 696f 6e2c 2061  iploidization, a
+00000d10: 6c73 6f20 6e61 6d65 6420 6173 2066 7261  lso named as fra
+00000d20: 6374 696f 6e61 7469 6f6e 2c20 6475 7269  ctionation, duri
+00000d30: 6e67 2077 6869 6368 2061 206c 6172 6765  ng which a large
+00000d40: 2070 6f72 7469 6f6e 206f 6620 6765 6e65   portion of gene
+00000d50: 2064 7570 6c69 6361 7465 7320 7769 6c6c   duplicates will
+00000d60: 2067 6574 206c 6f73 7420 616e 6420 6f6e   get lost and on
+00000d70: 6c79 2061 2066 7261 6374 696f 6e20 6361  ly a fraction ca
+00000d80: 6e20 6265 2072 6574 6169 6e65 642e 2054  n be retained. T
+00000d90: 6865 2074 7261 6365 7320 6f66 2070 6f6c  he traces of pol
+00000da0: 7970 6c6f 6964 697a 6174 696f 6e73 2063  yploidizations c
+00000db0: 616e 2062 6520 7468 7573 2075 6e65 6172  an be thus unear
+00000dc0: 7468 6564 2066 726f 6d20 7468 6573 6520  thed from these 
+00000dd0: 7265 7461 696e 6564 2067 656e 6520 6475  retained gene du
+00000de0: 706c 6963 6174 6573 2e20 5468 7265 6520  plicates. Three 
+00000df0: 6170 7072 6f61 6368 6573 2062 6173 6564  approaches based
+00000e00: 206f 6e20 6765 6e65 2064 7570 6c69 6361   on gene duplica
+00000e10: 7465 732c 206e 616d 656c 792c 202a 4b2a  tes, namely, *K*
+00000e20: 3c73 7562 3e53 3c2f 7375 623e 206d 6574  <sub>S</sub> met
+00000e30: 686f 642c 2067 656e 6520 7472 6565 202d  hod, gene tree -
+00000e40: 2073 7065 6369 6573 2074 7265 6520 7265   species tree re
+00000e50: 636f 6e63 696c 6961 7469 6f6e 206d 6574  conciliation met
+00000e60: 686f 6420 616e 6420 7379 6e74 656e 7920  hod and synteny 
+00000e70: 6d65 7468 6f64 2c20 6172 6520 636f 6d6d  method, are comm
+00000e80: 6f6e 6c79 2075 7365 6420 696e 2064 6574  only used in det
+00000e90: 6563 7469 6e67 2065 7669 6465 6e63 6520  ecting evidence 
+00000ea0: 666f 7220 5747 4473 2e20 5468 6520 6765  for WGDs. The ge
+00000eb0: 6e65 2074 7265 6520 2d20 7370 6563 6965  ne tree - specie
+00000ec0: 7320 7472 6565 2072 6563 6f6e 6369 6c69  s tree reconcili
+00000ed0: 6174 696f 6e20 6d65 7468 6f64 2069 7320  ation method is 
+00000ee0: 6e6f 7420 7769 7468 696e 2074 6865 2073  not within the s
+00000ef0: 636f 7065 206f 6620 6077 6764 2076 3260  cope of `wgd v2`
+00000f00: 2c20 6275 7420 7765 206b 696e 646c 7920  , but we kindly 
+00000f10: 7265 6665 7220 7265 6164 6572 7320 7768  refer readers wh
+00000f20: 6f20 6172 6520 696e 7465 7265 7374 6564  o are interested
+00000f30: 2074 6f20 7468 6520 7068 796c 6f67 656e   to the phylogen
+00000f40: 6f6d 6963 2070 726f 6772 616d 2064 6576  omic program dev
+00000f50: 656c 6f70 6564 2062 7920 4172 7468 7572  eloped by Arthur
+00000f60: 205a 7761 656e 6570 6f65 6c20 6e61 6d65   Zwaenepoel name
+00000f70: 6420 5b57 4841 4c45 5d28 6874 7470 733a  d [WHALE](https:
+00000f80: 2f2f 6769 7468 7562 2e63 6f6d 2f61 727a  //github.com/arz
+00000f90: 7761 2f57 6861 6c65 2e6a 6c29 2061 6e64  wa/Whale.jl) and
+00000fa0: 2074 6865 2061 7373 6f63 6961 7465 6420   the associated 
+00000fb0: 5b70 6170 6572 5d28 6874 7470 733a 2f2f  [paper](https://
+00000fc0: 646f 692e 6f72 672f 3130 2e31 3039 332f  doi.org/10.1093/
+00000fd0: 6d6f 6c62 6576 2f6d 737a 3038 3829 2066  molbev/msz088) f
+00000fe0: 6f72 206d 6f72 6520 7465 6368 6e69 6361  or more technica
+00000ff0: 6c20 616e 6420 7468 656f 7265 7469 6361  l and theoretica
+00001000: 6c20 6465 7461 696c 732e 0a0a 5468 6520  l details...The 
+00001010: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00001020: 6d65 7468 6f64 2069 7320 6573 7461 626c  method is establ
+00001030: 6973 6865 6420 6f6e 2061 206d 6f64 656c  ished on a model
+00001040: 206f 6620 6765 6e65 2066 616d 696c 7920   of gene family 
+00001050: 6576 6f6c 7574 696f 6e20 7468 6174 2065  evolution that e
+00001060: 6163 6820 6765 6e65 2066 616d 696c 7920  ach gene family 
+00001070: 6973 2061 6c6c 6f77 6564 2074 6f20 6576  is allowed to ev
+00001080: 6f6c 7665 2076 6961 2067 656e 6520 6475  olve via gene du
+00001090: 706c 6963 6174 696f 6e20 616e 6420 6c6f  plication and lo
+000010a0: 7373 2e20 4e6f 7465 2074 6861 7420 7468  ss. Note that th
+000010b0: 6520 6765 6e65 2066 616d 696c 7920 6865  e gene family he
+000010c0: 7265 2069 7320 6173 7375 6d65 6420 746f  re is assumed to
+000010d0: 2062 6520 7468 6520 636c 7573 7465 7220   be the cluster 
+000010e0: 6f66 2061 6c6c 2067 656e 6573 2064 6573  of all genes des
+000010f0: 6365 6e64 6564 2066 726f 6d20 616e 2061  cended from an a
+00001100: 6e63 6573 7472 616c 2067 656e 6520 696e  ncestral gene in
+00001110: 2061 2073 696e 676c 6520 6765 6e6f 6d65   a single genome
+00001120: 2e20 5265 636f 7665 7269 6e67 2074 6865  . Recovering the
+00001130: 2067 656e 6520 7472 6565 206f 6620 7375   gene tree of su
+00001140: 6368 2067 656e 6520 6661 6d69 6c79 2069  ch gene family i
+00001150: 6e66 6f72 6d73 2074 6865 2074 696d 696e  nforms the timin
+00001160: 672c 2073 6369 6c69 6365 7420 7468 6520  g, scilicet the 
+00001170: 6167 652c 206f 6620 6765 6e65 2064 7570  age, of gene dup
+00001180: 6c69 6361 7469 6f6e 2065 7665 6e74 732e  lication events.
+00001190: 2054 6865 2061 6765 2072 6566 6572 6564   The age refered
+000011a0: 2068 6572 652c 2069 7320 6e6f 7420 696e   here, is not in
+000011b0: 2072 6561 6c20 6765 6f6c 6f67 6963 616c   real geological
+000011c0: 2074 696d 652c 2062 7574 2069 6e20 7468   time, but in th
+000011d0: 6520 756e 6974 206f 6620 6576 6f6c 7574  e unit of evolut
+000011e0: 696f 6e61 7279 2064 6973 7461 6e63 652c  ionary distance,
+000011f0: 2069 2e65 2e2c 2074 6865 206e 756d 6265   i.e., the numbe
+00001200: 7220 6f66 2073 7562 7374 6974 7574 696f  r of substitutio
+00001210: 6e73 2070 6572 2073 6974 652e 2057 6865  ns per site. Whe
+00001220: 6e20 7468 6520 6576 6f6c 7574 696f 6e61  n the evolutiona
+00001230: 7279 2072 6174 6520 7265 6d61 696e 7320  ry rate remains 
+00001240: 6170 7072 6f78 696d 6174 656c 7920 636f  approximately co
+00001250: 6e73 7461 6e74 2c20 7468 6520 6576 6f6c  nstant, the evol
+00001260: 7574 696f 6e61 7279 2064 6973 7461 6e63  utionary distanc
+00001270: 6520 6973 2074 6865 6e20 7375 7070 6f73  e is then suppos
+00001280: 6564 2074 6f20 6265 2070 726f 706f 7274  ed to be proport
+00001290: 696f 6e61 6c20 746f 2074 6865 2072 6561  ional to the rea
+000012a0: 6c20 6576 6f6c 7574 696f 6e61 7279 2074  l evolutionary t
+000012b0: 696d 652e 2054 6865 2073 796e 6f6e 796d  ime. The synonym
+000012c0: 6f75 7320 6469 7374 616e 6365 202a 4b2a  ous distance *K*
+000012d0: 3c73 7562 3e53 3c2f 7375 623e 2c20 7468  <sub>S</sub>, th
+000012e0: 6520 6e75 6d62 6572 206f 6620 7379 6e6f  e number of syno
+000012f0: 6e79 6d6f 7573 2073 7562 7374 6974 7574  nymous substitut
+00001300: 696f 6e73 2070 6572 2073 796e 6f6e 796d  ions per synonym
+00001310: 6f75 7320 7369 7465 2c20 6973 2073 7563  ous site, is suc
+00001320: 6820 6361 6e64 6964 6174 6520 7468 6174  h candidate that
+00001330: 2073 796e 6f6e 796d 6f75 7320 7375 6273   synonymous subs
+00001340: 7469 7475 7469 6f6e 7320 776f 756c 6420  titutions would 
+00001350: 6e6f 7420 696e 6375 7220 7468 6520 6368  not incur the ch
+00001360: 616e 6765 206f 6620 616d 696e 6f20 6163  ange of amino ac
+00001370: 6964 2061 6e64 2061 7265 2074 6875 7320  id and are thus 
+00001380: 7265 6761 7264 6564 2061 7320 6e65 7574  regarded as neut
+00001390: 7261 6c2c 2077 6869 6368 2061 6363 6f72  ral, which accor
+000013a0: 6469 6e67 2074 6f20 7468 6520 6e65 7574  ding to the neut
+000013b0: 7261 6c20 7468 656f 7279 2073 686f 756c  ral theory shoul
+000013c0: 6420 6f63 6375 7220 696e 2063 6f6e 7374  d occur in const
+000013d0: 616e 7420 7261 7465 2e20 4769 7665 6e20  ant rate. Given 
+000013e0: 6120 6d6f 6465 6c20 6f66 2067 656e 6520  a model of gene 
+000013f0: 6661 6d69 6c79 2074 6861 7420 616c 6c6f  family that allo
+00001400: 7773 2074 6865 2067 656e 6520 746f 2064  ws the gene to d
+00001410: 7570 6c69 6361 7465 2061 6e64 2067 6574  uplicate and get
+00001420: 206c 6f73 7420 696e 2061 2066 6978 6564   lost in a fixed
+00001430: 2072 6174 652c 206f 6e65 2063 616e 2064   rate, one can d
+00001440: 6572 6976 6520 7468 6174 2074 6865 2070  erive that the p
+00001450: 726f 6261 6269 6c69 7479 2064 656e 7369  robability densi
+00001460: 7479 2066 756e 6374 696f 6e20 6f66 2074  ty function of t
+00001470: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+00001480: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+00001490: 696f 6e20 6f66 2072 6574 6169 6e65 6420  ion of retained 
+000014a0: 6765 6e65 2064 7570 6c69 6361 7465 7320  gene duplicates 
+000014b0: 6973 2061 2071 7561 7369 2d65 7870 6f6e  is a quasi-expon
+000014c0: 656e 7469 616c 2066 756e 6374 696f 6e20  ential function 
+000014d0: 7468 6174 206d 6f73 7420 7265 7461 696e  that most retain
+000014e0: 6564 2067 656e 6520 6475 706c 6963 6174  ed gene duplicat
+000014f0: 6573 2061 7265 2072 6563 656e 746c 7920  es are recently 
+00001500: 626f 726e 6564 2077 6974 6820 7e30 2061  borned with ~0 a
+00001510: 6765 2077 6869 6c65 2061 7320 7468 6520  ge while as the 
+00001520: 6167 6520 676f 696e 6720 6f6c 6465 7220  age going older 
+00001530: 7468 6520 6173 736f 6369 6174 6564 206e  the associated n
+00001540: 756d 6265 7220 6f66 2072 6574 6169 6e65  umber of retaine
+00001550: 6420 6765 6e65 2064 7570 6c69 6361 7465  d gene duplicate
+00001560: 7320 6465 6361 7920 7175 6173 692d 6578  s decay quasi-ex
+00001570: 706f 6e65 6e74 6961 6c6c 792e 2054 6865  ponentially. The
+00001580: 7265 666f 7265 2c20 7468 6520 6f63 6375  refore, the occu
+00001590: 7261 6e63 6520 6f66 206c 6172 6765 2d73  rance of large-s
+000015a0: 6361 6c65 2067 656e 6520 6475 706c 6963  cale gene duplic
+000015b0: 6174 696f 6e20 6576 656e 7473 2c20 666f  ation events, fo
+000015c0: 7220 696e 7374 616e 6520 5747 4473 2c20  r instane WGDs, 
+000015d0: 7769 7468 2076 6172 6965 6420 7265 7465  with varied rete
+000015e0: 6e74 696f 6e20 7261 7465 2c20 7769 6c6c  ntion rate, will
+000015f0: 206c 6561 7665 2061 6e20 6167 6520 7065   leave an age pe
+00001600: 616b 2066 726f 6d20 7468 6520 6275 7273  ak from the burs
+00001610: 7420 6f66 2067 656e 6520 6475 706c 6963  t of gene duplic
+00001620: 6174 6573 2069 6e20 6120 7368 6f72 7420  ates in a short 
+00001630: 7469 6d65 2d66 7261 6d65 2075 706f 6e20  time-frame upon 
+00001640: 7468 6520 696e 6974 6961 6c20 6167 6520  the initial age 
+00001650: 6469 7374 7269 6275 7469 6f6e 2c20 616e  distribution, an
+00001660: 6420 6361 6e20 6265 2075 6e76 6569 6c65  d can be unveile
+00001670: 6420 6672 6f6d 206d 6978 7475 7265 206d  d from mixture m
+00001680: 6f64 656c 696e 6720 616e 616c 7973 6973  odeling analysis
+00001690: 2e20 486f 7765 7665 722c 2057 4744 7320  . However, WGDs 
+000016a0: 6964 656e 7469 6669 6564 2066 726f 6d20  identified from 
+000016b0: 7468 6520 7061 7261 6c6f 676f 7573 202a  the paralogous *
+000016c0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+000016d0: 6765 2064 6973 7472 6962 7574 696f 6e73  ge distributions
+000016e0: 2063 616e 206f 6e6c 7920 696e 666f 726d   can only inform
+000016f0: 2074 6865 2057 4744 2074 696d 696e 6720   the WGD timing 
+00001700: 696e 2074 6865 2074 696d 652d 7363 616c  in the time-scal
+00001710: 6520 6f66 2074 6861 7420 7370 6563 6966  e of that specif
+00001720: 6963 2073 7065 6369 6573 2c20 7768 6963  ic species, whic
+00001730: 6820 6973 206e 6f74 2063 6f6d 7061 7261  h is not compara
+00001740: 626c 6520 696e 2074 6865 2070 6879 6c6f  ble in the phylo
+00001750: 6765 6e65 7469 6320 636f 6e74 6578 742e  genetic context.
+00001760: 204f 6e6c 7920 7769 7468 2074 6865 206f   Only with the o
+00001770: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
+00001780: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+00001790: 6973 7472 6962 7574 696f 6e73 2c20 7768  istributions, wh
+000017a0: 6963 6820 636f 6e76 6572 7420 7468 6520  ich convert the 
+000017b0: 6573 7469 6d61 7465 6420 626f 6479 2066  estimated body f
+000017c0: 726f 6d20 7061 7261 6c6f 6775 6573 2074  rom paralogues t
+000017d0: 6f20 6f72 7468 6f6c 6f67 7565 7320 616e  o orthologues an
+000017e0: 6420 696e 666f 726d 2074 6865 2072 656c  d inform the rel
+000017f0: 6174 6976 6520 7469 6d69 6e67 206f 6620  ative timing of 
+00001800: 7370 6563 6961 7469 6f6e 2065 7665 6e74  speciation event
+00001810: 732c 2063 616e 2077 6520 6465 6369 7068  s, can we deciph
+00001820: 6572 2074 6865 2070 6879 6c6f 6765 6e65  er the phylogene
+00001830: 7469 6320 706c 6163 656d 656e 7420 6f66  tic placement of
+00001840: 2057 4744 7320 6166 7465 7220 7072 6f70   WGDs after prop
+00001850: 6572 2072 6174 6520 636f 7272 6563 7469  er rate correcti
+00001860: 6f6e 2e20 6077 6764 2076 3260 2069 7320  on. `wgd v2` is 
+00001870: 7375 6368 2070 726f 6772 616d 2074 6861  such program tha
+00001880: 7420 6865 6c70 7320 7573 6572 7320 636f  t helps users co
+00001890: 6e73 7472 7563 7420 7061 7261 6c6f 676f  nstruct paralogo
+000018a0: 7573 2061 6e64 206f 7274 686f 6c6f 676f  us and orthologo
+000018b0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+000018c0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+000018d0: 696f 6e73 2061 6e64 2072 6561 6c69 7a65  ions and realize
+000018e0: 2062 6f74 6820 7468 6520 6964 656e 7469   both the identi
+000018f0: 6669 6361 7469 6f6e 2061 6e64 2070 6c61  fication and pla
+00001900: 6365 6d65 6e74 206f 6620 5747 4473 2e0a  cement of WGDs..
+00001910: 0a49 6e20 7468 6520 7072 656d 6973 6520  .In the premise 
+00001920: 6f66 2070 6879 6c6f 6765 6e65 7469 6361  of phylogenetica
+00001930: 6c6c 7920 6c6f 6361 7465 6420 5747 4473  lly located WGDs
+00001940: 2c20 7468 6520 6162 736f 6c75 7465 2061  , the absolute a
+00001950: 6765 2028 696e 2067 656f 6c6f 6769 6361  ge (in geologica
+00001960: 6c20 7469 6d65 2920 6f66 2057 4744 7320  l time) of WGDs 
+00001970: 6361 6e20 616c 736f 2062 6520 696e 6665  can also be infe
+00001980: 7272 6564 2066 726f 6d20 7468 6f73 6520  rred from those 
+00001990: 5747 442d 7265 7461 696e 6564 2067 656e  WGD-retained gen
+000019a0: 6520 6475 706c 6963 6174 6573 2c20 616c  e duplicates, al
+000019b0: 7468 6f75 6768 2074 6865 7265 2068 6173  though there has
+000019c0: 2062 6565 6e20 6e6f 2065 6173 7920 6f72   been no easy or
+000019d0: 2073 7472 6169 6768 7466 6f72 7761 7264   straightforward
+000019e0: 2070 6970 656c 696e 6520 666f 7220 7468   pipeline for th
+000019f0: 6973 206a 6f62 2073 6f20 6661 722e 2049  is job so far. I
+00001a00: 6e20 6077 6764 2076 3260 2c20 7765 2064  n `wgd v2`, we d
+00001a10: 6576 656c 6f70 6564 2061 2066 6561 7369  eveloped a feasi
+00001a20: 626c 6520 696e 7465 6772 6174 6564 2070  ble integrated p
+00001a30: 6970 6c65 6c69 6e65 2066 6f72 2061 6273  ipleline for abs
+00001a40: 6f6c 7574 6520 6461 7469 6e67 206f 6620  olute dating of 
+00001a50: 5747 4473 2e20 5468 6520 7069 7065 6c69  WGDs. The pipeli
+00001a60: 6e65 2063 616e 2062 6520 726f 7567 686c  ne can be roughl
+00001a70: 7920 6469 7669 6465 6420 696e 746f 2074  y divided into t
+00001a80: 6872 6565 206d 6169 6e20 7374 6570 732e  hree main steps.
+00001a90: 2031 2920 5468 6520 636f 6e73 7472 7563   1) The construc
+00001aa0: 7469 6f6e 206f 6620 616e 6368 6f72 202a  tion of anchor *
+00001ab0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
+00001ac0: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
+00001ad0: 7468 6520 6465 6c69 6e65 6174 696f 6e20  the delineation 
+00001ae0: 6f66 2063 7265 6469 6162 6c65 202a 4b2a  of crediable *K*
+00001af0: 3c73 7562 3e53 3c2f 7375 623e 2072 616e  <sub>S</sub> ran
+00001b00: 6765 2061 646f 7074 6564 2066 6f72 2070  ge adopted for p
+00001b10: 6879 6c6f 6765 6e65 7469 6320 6461 7469  hylogenetic dati
+00001b20: 6e67 2c20 7573 696e 6720 6077 6764 2064  ng, using `wgd d
+00001b30: 6d64 602c 2060 7767 6420 6b73 6460 2c20  md`, `wgd ksd`, 
+00001b40: 6077 6764 2073 796e 6020 616e 6420 6077  `wgd syn` and `w
+00001b50: 6764 2070 6561 6b60 2e20 4e6f 7465 2074  gd peak`. Note t
+00001b60: 6861 7420 6865 7265 2077 6520 6f6e 6c79  hat here we only
+00001b70: 2063 6f6e 7369 6465 7220 6765 6e6f 6d65   consider genome
+00001b80: 2061 7373 656d 626c 7920 6265 6361 7573   assembly becaus
+00001b90: 6520 666f 7220 7472 616e 7363 7269 7074  e for transcript
+00001ba0: 6f6d 6520 6173 7365 6d62 6c79 2069 7427  ome assembly it'
+00001bb0: 7320 696d 706f 7373 6962 6c65 2074 6f20  s impossible to 
+00001bc0: 6469 7374 696e 6775 6973 6820 5747 442d  distinguish WGD-
+00001bd0: 6465 7269 7665 6420 6475 706c 6963 6174  derived duplicat
+00001be0: 6573 2066 726f 6d20 736d 616c 6c2d 7363  es from small-sc
+00001bf0: 616c 6520 6475 706c 6963 6174 696f 6e2d  ale duplication-
+00001c00: 6465 7269 7665 6420 6475 706c 6963 6174  derived duplicat
+00001c10: 6573 2c20 7768 6963 6820 6861 7070 656e  es, which happen
+00001c20: 6564 2069 6e20 6120 636f 6e74 696e 756f  ed in a continuo
+00001c30: 7573 2074 696d 652d 6672 616d 6520 696e  us time-frame in
+00001c40: 7374 6561 6420 6f66 206f 6e6c 7920 6120  stead of only a 
+00001c50: 7365 7061 7261 7465 2073 686f 7274 2074  separate short t
+00001c60: 696d 652d 6672 616d 6520 616e 6420 7468  ime-frame and th
+00001c70: 7573 2072 6566 6c65 6374 7320 7468 6520  us reflects the 
+00001c80: 6475 7261 7469 6f6e 206f 6620 7468 6174  duration of that
+00001c90: 2062 7261 6e63 6820 7261 7468 6572 2074   branch rather t
+00001ca0: 6861 6e20 7468 6520 7469 6d65 2061 7420  han the time at 
+00001cb0: 7768 6963 6820 5747 4420 6f63 6375 7272  which WGD occurr
+00001cc0: 6564 2e20 3229 2054 6865 2066 6f72 6d75  ed. 2) The formu
+00001cd0: 6c61 7469 6f6e 206f 6620 6120 7374 6172  lation of a star
+00001ce0: 7469 6e67 2074 7265 6520 7573 6564 2069  ting tree used i
+00001cf0: 6e20 7468 6520 7068 796c 6f67 656e 6574  n the phylogenet
+00001d00: 6963 2064 6174 696e 672c 2063 6f6d 706f  ic dating, compo
+00001d10: 7365 6420 6f66 2061 2066 6577 2073 7065  sed of a few spe
+00001d20: 6369 6573 2061 6e64 2061 6e6e 6f74 6174  cies and annotat
+00001d30: 6564 2077 6974 6820 666f 7373 696c 2063  ed with fossil c
+00001d40: 616c 6962 7261 7469 6f6e 2069 6e66 6f72  alibration infor
+00001d50: 6d61 7469 6f6e 2e20 5468 6973 2073 7465  mation. This ste
+00001d60: 7020 6973 2065 7373 656e 7469 616c 2066  p is essential f
+00001d70: 6f72 2074 6865 2072 6573 756c 7420 6f66  or the result of
+00001d80: 2061 6273 6f6c 7574 6520 5747 4420 6461   absolute WGD da
+00001d90: 7469 6e67 2074 6861 7420 7765 2073 7567  ting that we sug
+00001da0: 6765 7374 2075 7365 7273 206f 6620 7461  gest users of ta
+00001db0: 6b69 6e67 2067 7265 6174 2063 6175 7469  king great cauti
+00001dc0: 6f6e 2074 6f20 6173 7375 7265 2074 6865  on to assure the
+00001dd0: 2074 7265 6520 746f 706f 6c6f 6779 2061   tree topology a
+00001de0: 6e64 2070 726f 7065 7220 626f 756e 6473  nd proper bounds
+00001df0: 2066 6f72 2066 6f73 7369 6c20 6361 6c69   for fossil cali
+00001e00: 6272 6174 696f 6e73 2e20 3329 2054 6865  brations. 3) The
+00001e10: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
+00001e20: 206f 7274 686f 6772 6f75 7073 2063 6f6e   orthogroups con
+00001e30: 7369 7374 696e 6720 6f66 2063 6f6c 6c69  sisting of colli
+00001e40: 6e65 6172 2064 7570 6c69 6361 7465 7320  near duplicates 
+00001e50: 6f66 2074 6865 2066 6f63 616c 2073 7065  of the focal spe
+00001e60: 6369 6573 2061 6e64 2074 6865 6972 2072  cies and their r
+00001e70: 6563 6970 726f 6361 6c20 6265 7374 2068  eciprocal best h
+00001e80: 6974 7320 2852 4248 7329 2061 6761 696e  its (RBHs) again
+00001e90: 7374 206f 7468 6572 2073 7065 6369 6573  st other species
+00001ea0: 2069 6e20 7468 6520 7374 6172 7469 6e67   in the starting
+00001eb0: 2074 7265 6520 616e 6420 7468 6520 7068   tree and the ph
+00001ec0: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
+00001ed0: 6720 7573 696e 6720 6120 6d6f 6c65 6375  g using a molecu
+00001ee0: 6c61 7220 6461 7469 6e67 2070 726f 6772  lar dating progr
+00001ef0: 616d 2066 6f72 2069 6e73 7461 6e63 6520  am for instance 
+00001f00: 606d 636d 6374 7265 6560 2c20 7669 6120  `mcmctree`, via 
+00001f10: 7072 6f67 7261 6d20 6077 6764 2064 6d64  program `wgd dmd
+00001f20: 6020 616e 6420 6077 6764 2066 6f63 7573  ` and `wgd focus
+00001f30: 602e 2057 6520 7265 636f 6d6d 656e 6420  `. We recommend 
+00001f40: 7468 6520 7573 6167 6520 6f66 2042 6179  the usage of Bay
+00001f50: 6573 6961 6e20 6d6f 6c65 6375 6c61 7220  esian molecular 
+00001f60: 6461 7469 6e67 2070 726f 6772 616d 2060  dating program `
+00001f70: 6d63 6d63 7472 6565 602c 2077 6869 6368  mcmctree`, which
+00001f80: 2070 726f 7669 6465 7320 6120 7661 7269   provides a vari
+00001f90: 6574 7920 6f66 2064 6966 6665 7265 6e74  ety of different
+00001fa0: 2073 7562 7374 6974 7574 696f 6e20 616e   substitution an
+00001fb0: 6420 7261 7465 206d 6f64 656c 732e 204e  d rate models. N
+00001fc0: 6f6e 6574 6865 6c65 7373 2c20 7765 2075  onetheless, we u
+00001fd0: 7267 6520 7573 6572 7320 746f 2073 6574  rge users to set
+00001fe0: 2070 7269 6f72 2064 6973 7472 6962 7574   prior distribut
+00001ff0: 696f 6e20 6f66 2064 6966 6665 7265 6e74  ion of different
+00002000: 2070 6172 616d 6574 6572 7320 7769 7468   parameters with
+00002010: 2063 6175 7469 6f6e 2061 6e64 2061 7373   caution and ass
+00002020: 7572 6520 6164 6571 7561 7465 2073 616d  ure adequate sam
+00002030: 706c 696e 6720 6f66 2064 6966 6665 7265  pling of differe
+00002040: 6e74 2070 6172 616d 6574 6572 732e 0a0a  nt parameters...
+00002050: 2323 2049 6e73 7461 6c6c 6174 696f 6e0a  ## Installation.
+00002060: 0a54 6865 2065 6173 6965 7374 2077 6179  .The easiest way
+00002070: 2074 6f20 696e 7374 616c 6c20 6077 6764   to install `wgd
+00002080: 2076 3260 2069 7320 7573 696e 6720 6050   v2` is using `P
+00002090: 5950 4960 2e20 4e6f 7465 2074 6861 7420  YPI`. Note that 
+000020a0: 6966 2079 6f75 2077 616e 7420 746f 2067  if you want to g
+000020b0: 6574 2074 6865 206c 6174 6573 7420 7570  et the latest up
+000020c0: 6461 7465 2c20 7765 2073 7567 6765 7374  date, we suggest
+000020d0: 2069 6e73 7461 6c6c 696e 6720 6672 6f6d   installing from
+000020e0: 2074 6865 2073 6f75 7263 652c 2073 696e   the source, sin
+000020f0: 6365 2074 6865 2075 7064 6174 6520 6f6e  ce the update on
+00002100: 2060 5059 5049 6020 7769 6c6c 2062 6520   `PYPI` will be 
+00002110: 6465 6c61 7965 6420 636f 6d70 6172 6564  delayed compared
+00002120: 2074 6f20 6865 7265 206f 6620 736f 7572   to here of sour
+00002130: 6365 2e0a 0a60 6060 0a70 6970 2069 6e73  ce...```.pip ins
+00002140: 7461 6c6c 2077 6764 0a60 6060 0a0a 546f  tall wgd.```..To
+00002150: 2069 6e73 7461 6c6c 2060 7767 6420 7632   install `wgd v2
+00002160: 6020 696e 2061 2076 6972 7475 616c 2065  ` in a virtual e
+00002170: 6e76 6972 6f6e 6d65 6e74 2028 7768 6963  nvironment (whic
+00002180: 6820 7765 2073 7472 6f6e 676c 7920 7265  h we strongly re
+00002190: 636f 6d6d 656e 642c 206e 6f20 6d61 7474  commend, no matt
+000021a0: 6572 2079 6f75 2069 6e73 7461 6c6c 2066  er you install f
+000021b0: 726f 6d20 736f 7572 6365 2c20 6050 5950  rom source, `PYP
+000021c0: 4960 206f 7220 6062 696f 636f 6e64 6160  I` or `bioconda`
+000021d0: 292c 2074 6865 2066 6f6c 6c6f 7769 6e67  ), the following
+000021e0: 2063 6f6d 6d61 6e64 206c 696e 6573 2063   command lines c
+000021f0: 6f75 6c64 2062 6520 7573 6564 2e0a 0a60  ould be used...`
+00002200: 6060 0a67 6974 2063 6c6f 6e65 2068 7474  ``.git clone htt
+00002210: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
+00002220: 6865 6368 652d 7073 622f 7767 640a 6364  heche-psb/wgd.cd
+00002230: 2077 6764 0a76 6972 7475 616c 656e 7620   wgd.virtualenv 
+00002240: 2d70 3d70 7974 686f 6e33 2045 4e56 2028  -p=python3 ENV (
+00002250: 6f72 2070 7974 686f 6e33 202d 6d20 7665  or python3 -m ve
+00002260: 6e76 2045 4e56 290a 736f 7572 6365 2045  nv ENV).source E
+00002270: 4e56 2f62 696e 2f61 6374 6976 6174 650a  NV/bin/activate.
+00002280: 7069 7020 696e 7374 616c 6c20 2d72 2072  pip install -r r
+00002290: 6571 7569 7265 6d65 6e74 732e 7478 740a  equirements.txt.
+000022a0: 7069 7020 696e 7374 616c 6c20 2e0a 6060  pip install ..``
+000022b0: 600a 0a57 6865 6e20 6d65 7420 7769 7468  `..When met with
+000022c0: 2070 6572 6d69 7373 696f 6e20 7072 6f62   permission prob
+000022d0: 6c65 6d20 696e 2069 6e73 7461 6c6c 6174  lem in installat
+000022e0: 696f 6e2c 2070 6c65 6173 6520 7472 7920  ion, please try 
+000022f0: 7468 6520 666f 6c6c 6f77 696e 6720 636f  the following co
+00002300: 6d6d 616e 6420 6c69 6e65 2e0a 0a60 6060  mmand line...```
+00002310: 0a70 6970 2069 6e73 7461 6c6c 202d 6520  .pip install -e 
+00002320: 2e0a 6060 600a 0a49 6620 6d75 6c74 6970  ..```..If multip
+00002330: 6c79 2076 6572 7369 6f6e 7320 6f66 2060  ly versions of `
+00002340: 7767 6460 2077 6572 6520 696e 7374 616c  wgd` were instal
+00002350: 6c65 6420 696e 2074 6865 2073 7973 7465  led in the syste
+00002360: 6d2c 2070 6c65 6173 6520 6164 6420 7468  m, please add th
+00002370: 6520 7269 6768 7420 7061 7468 206f 6620  e right path of 
+00002380: 696e 7465 7265 7374 6564 2076 6572 7369  interested versi
+00002390: 6f6e 2069 6e74 6f20 7468 6520 656e 7669  on into the envi
+000023a0: 726f 6e6d 656e 7420 7661 7269 6162 6c65  ronment variable
+000023b0: 732c 2066 6f72 2065 7861 6d70 6c65 0a0a  s, for example..
+000023c0: 6060 600a 6578 706f 7274 2050 4154 483d  ```.export PATH=
+000023d0: 2224 5041 5448 3a7e 2f2e 6c6f 6361 6c2f  "$PATH:~/.local/
+000023e0: 6269 6e2f 7767 6422 0a60 6060 0a0a 4e6f  bin/wgd".```..No
+000023f0: 7465 2074 6861 7420 7468 6520 7665 7273  te that the vers
+00002400: 696f 6e20 6f66 2060 6e75 6d70 7960 2069  ion of `numpy` i
+00002410: 7320 696d 706f 7274 616e 7420 2866 6f72  s important (for
+00002420: 206d 616e 7920 6f74 6865 7220 7061 636b   many other pack
+00002430: 6167 6573 2061 7265 2074 6865 2073 616d  ages are the sam
+00002440: 6520 6f66 2063 6f75 7273 6529 2c20 6573  e of course), es
+00002450: 7065 6369 616c 6c79 2066 6f72 2060 6661  pecially for `fa
+00002460: 7374 636c 7573 7465 7260 2070 6163 6b61  stcluster` packa
+00002470: 6765 2e20 496e 206f 7572 2074 6573 742c  ge. In our test,
+00002480: 2074 6865 2060 6e75 6d70 7960 2031 2e31   the `numpy` 1.1
+00002490: 392e 3020 776f 726b 7320 6669 6e65 206f  9.0 works fine o
+000024a0: 6e20 6070 7974 686f 6e33 2e36 2f38 602e  n `python3.6/8`.
+000024b0: 2049 6620 796f 7520 6d65 7420 736f 6d65   If you met some
+000024c0: 2065 7272 6f72 7320 6f72 2077 6172 6e69   errors or warni
+000024d0: 6e67 7320 6162 6f75 7420 606e 756d 7079  ngs about `numpy
+000024e0: 602c 206d 6179 6265 2063 6f6e 7369 6465  `, maybe conside
+000024f0: 7269 6e67 2070 7265 2d69 6e73 7461 6c6c  ring pre-install
+00002500: 2060 6e75 6d70 7960 2061 7320 312e 3139   `numpy` as 1.19
+00002510: 2e30 206f 7220 6f74 6865 7220 636c 6f73  .0 or other clos
+00002520: 652d 6279 2076 6572 7369 6f6e 7320 6265  e-by versions be
+00002530: 666f 7265 2079 6f75 2069 6e73 7461 6c6c  fore you install
+00002540: 2060 7767 6460 2e20 6077 6764 6020 7265   `wgd`. `wgd` re
+00002550: 6c69 6573 206f 6e20 6578 7465 726e 616c  lies on external
+00002560: 2073 6f66 7477 6172 6573 2069 6e63 6c75   softwares inclu
+00002570: 6469 6e67 2060 6469 616d 6f6e 6460 2061  ding `diamond` a
+00002580: 6e64 2060 6d63 6c60 2066 6f72 2060 7767  nd `mcl` for `wg
+00002590: 6420 646d 6460 2c20 6070 616d 6c20 7634  d dmd`, `paml v4
+000025a0: 2e39 6a60 2c20 606d 6166 6674 6020 2860  .9j`, `mafft` (`
+000025b0: 6d75 7363 6c65 6020 616e 6420 6070 7261  muscle` and `pra
+000025c0: 6e6b 6020 6966 2073 6574 292c 2060 6661  nk` if set), `fa
+000025d0: 7374 7472 6565 6020 286f 7220 6069 7174  sttree` (or `iqt
+000025e0: 7265 6560 2069 6620 7365 7429 2066 6f72  ree` if set) for
+000025f0: 2060 7767 6420 6b73 6460 2061 6e64 206f   `wgd ksd` and o
+00002600: 7074 696f 6e61 6c6c 7920 606d 7262 6179  ptionally `mrbay
+00002610: 6573 6020 666f 7220 7468 6520 7068 796c  es` for the phyl
+00002620: 6f67 656e 6574 6963 2069 6e66 6572 656e  ogenetic inferen
+00002630: 6365 2066 756e 6374 696f 6e20 696e 2060  ce function in `
+00002640: 7767 6420 646d 6460 2061 6e64 2060 7767  wgd dmd` and `wg
+00002650: 6420 666f 6375 7360 2028 606d 6166 6674  d focus` (`mafft
+00002660: 602c 2060 6d75 7363 6c65 6020 616e 6420  `, `muscle` and 
+00002670: 6070 7261 6e6b 6020 6173 2077 656c 6c20  `prank` as well 
+00002680: 7768 656e 2074 6865 2061 6e61 6c79 7369  when the analysi
+00002690: 7320 7265 7175 6972 6573 2073 6571 7565  s requires seque
+000026a0: 6e63 6520 616c 6967 6e6d 656e 7429 2e20  nce alignment). 
+000026b0: 536f 6d65 206f 7468 6572 206f 7074 696f  Some other optio
+000026c0: 6e61 6c20 736f 6674 7761 7265 7320 696e  nal softwares in
+000026d0: 636c 7564 696e 6720 6070 616d 6c20 7634  cluding `paml v4
+000026e0: 2e39 6a60 2c20 6072 3873 602c 2060 6265  .9j`, `r8s`, `be
+000026f0: 6173 7460 2c20 6065 6767 6e6f 6760 2c20  ast`, `eggnog`, 
+00002700: 6064 6961 6d6f 6e64 602c 2060 696e 7465  `diamond`, `inte
+00002710: 7270 726f 7363 616e 602c 2060 686d 6d65  rproscan`, `hmme
+00002720: 7220 7633 2e31 6232 6020 616e 6420 6061  r v3.1b2` and `a
+00002730: 7374 7261 6c2d 7072 6f60 2028 6068 6d6d  stral-pro` (`hmm
+00002740: 6572 2076 332e 3162 3260 2069 7320 616c  er v3.1b2` is al
+00002750: 736f 2066 6f72 2074 6865 205b 6f72 7468  so for the [orth
+00002760: 6f67 726f 7570 2061 7373 6967 6e6d 656e  ogroup assignmen
+00002770: 745d 2868 7474 7073 3a2f 2f77 6764 7632  t](https://wgdv2
+00002780: 2e72 6561 6474 6865 646f 6373 2e69 6f2f  .readthedocs.io/
+00002790: 656e 2f6c 6174 6573 742f 7573 6167 652e  en/latest/usage.
+000027a0: 6874 6d6c 236f 7274 686f 6772 6f75 702d  html#orthogroup-
+000027b0: 6173 7369 676e 6d65 6e74 2920 6675 6e63  assignment) func
+000027c0: 7469 6f6e 2069 6e20 6077 6764 2064 6d64  tion in `wgd dmd
+000027d0: 6020 616e 6420 6061 7374 7261 6c2d 7072  ` and `astral-pr
+000027e0: 6f60 2069 7320 616c 736f 2066 6f72 2074  o` is also for t
+000027f0: 6865 2066 756e 6374 696f 6e20 602d 2d63  he function `--c
+00002800: 6f6c 6c69 6e65 6172 636f 616c 6573 6365  ollinearcoalesce
+00002810: 6e63 6560 2069 6e20 6077 6764 2064 6d64  nce` in `wgd dmd
+00002820: 6029 2061 7265 2066 6f72 2074 6865 206d  `) are for the m
+00002830: 6f6c 6563 756c 6172 2064 6174 696e 672c  olecular dating,
+00002840: 2067 656e 6520 6661 6d69 6c79 2066 756e   gene family fun
+00002850: 6374 696f 6e20 616e 6e6f 7461 7469 6f6e  ction annotation
+00002860: 206f 7220 7068 796c 6f67 656e 6574 6963   or phylogenetic
+00002870: 2069 6e66 6572 656e 6365 2069 6e20 6077   inference in `w
+00002880: 6764 2066 6f63 7573 602e 0a0a 2323 2050  gd focus`...## P
+00002890: 6172 616d 6574 6572 730a 0a54 6865 7265  arameters..There
+000028a0: 2061 7265 2037 206d 6169 6e20 7072 6f67   are 7 main prog
+000028b0: 7261 6d73 2069 6e20 6077 6764 2076 3260  rams in `wgd v2`
+000028c0: 3a20 6064 6d64 602c 6066 6f63 7573 602c  : `dmd`,`focus`,
+000028d0: 606b 7364 602c 606d 6978 602c 6070 6561  `ksd`,`mix`,`pea
+000028e0: 6b60 2c60 7379 6e60 2c60 7669 7a60 2e20  k`,`syn`,`viz`. 
+000028f0: 4865 7265 6166 7465 7220 7765 2077 696c  Hereafter we wil
+00002900: 6c20 7072 6f76 6964 6520 6120 6465 7461  l provide a deta
+00002910: 696c 6564 2065 6c75 6369 6461 7469 6f6e  iled elucidation
+00002920: 206f 6e20 6561 6368 206f 6620 7468 6520   on each of the 
+00002930: 7072 6f67 7261 6d20 616e 6420 6974 7320  program and its 
+00002940: 6173 736f 6369 6174 6564 2070 6172 616d  associated param
+00002950: 6574 6572 732e 2050 6c65 6173 6520 7265  eters. Please re
+00002960: 6665 7220 746f 2074 6865 205b 5573 6167  fer to the [Usag
+00002970: 655d 2823 7573 6167 6529 2066 6f72 2074  e](#usage) for t
+00002980: 6865 2073 6365 6e61 7269 6f73 2074 6f20  he scenarios to 
+00002990: 7768 6963 6820 6561 6368 2070 6172 616d  which each param
+000029a0: 6574 6572 2061 7070 6c69 6573 2e0a 0a54  eter applies...T
+000029b0: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
+000029c0: 646d 6460 2063 616e 2072 6561 6c69 7a65  dmd` can realize
+000029d0: 2074 6865 2064 656c 696e 6561 7469 6f6e   the delineation
+000029e0: 206f 6620 7768 6f6c 6520 7061 7261 6e6f   of whole parano
+000029f0: 6d65 2c20 5242 4873 2028 5265 6369 7072  me, RBHs (Recipr
+00002a00: 6f63 616c 2042 6573 7420 4869 7473 292c  ocal Best Hits),
+00002a10: 204d 5242 4873 2028 4d75 6c74 6970 6c65   MRBHs (Multiple
+00002a20: 2052 6563 6970 726f 6361 6c20 4265 7374   Reciprocal Best
+00002a30: 2048 6974 7329 2c20 6f72 7468 6f67 726f   Hits), orthogro
+00002a40: 7570 7320 616e 6420 736f 6d65 206f 7468  ups and some oth
+00002a50: 6572 206f 7274 686f 6772 6f75 702d 7265  er orthogroup-re
+00002a60: 6c61 7465 6420 6675 6e63 7469 6f6e 732c  lated functions,
+00002a70: 2069 6e63 6c75 6469 6e67 2063 6972 6375   including circu
+00002a80: 6d73 6372 6970 7469 6f6e 206f 6620 6e65  mscription of ne
+00002a90: 7374 6564 2073 696e 676c 652d 636f 7079  sted single-copy
+00002aa0: 206f 7274 686f 6772 6f75 7073 2028 4e53   orthogroups (NS
+00002ab0: 4f47 7329 2c20 756e 6269 6173 6564 2074  OGs), unbiased t
+00002ac0: 6573 7420 6f66 2073 696e 676c 652d 636f  est of single-co
+00002ad0: 7079 206f 7274 686f 6772 6f75 7073 2028  py orthogroups (
+00002ae0: 534f 4773 2920 6f76 6572 206d 6973 7369  SOGs) over missi
+00002af0: 6e67 2069 6e70 6172 616c 6f67 732c 2063  ng inparalogs, c
+00002b00: 6f6e 7374 7275 6374 696f 6e20 6f66 2042  onstruction of B
+00002b10: 5553 434f 2d67 7569 6465 6420 7369 6e67  USCO-guided sing
+00002b20: 6c65 2d63 6f70 7920 6f72 7468 6f67 726f  le-copy orthogro
+00002b30: 7570 7320 2853 4f47 7329 2c61 6e64 2074  ups (SOGs),and t
+00002b40: 6865 2063 6f6c 6c69 6e65 6172 2063 6f61  he collinear coa
+00002b50: 6c65 7363 656e 6365 2069 6e66 6572 656e  lescence inferen
+00002b60: 6365 206f 6620 7068 796c 6f67 656e 792e  ce of phylogeny.
+00002b70: 0a60 6060 0a77 6764 2064 6d64 2073 6571  .```.wgd dmd seq
+00002b80: 7565 6e63 6573 2028 6f70 7469 6f6e 290a  uences (option).
+00002b90: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002ba0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002bb0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002bc0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002bd0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00002be0: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
+00002bf0: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
+00002c00: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
+00002c10: 5f64 6d64 0a2d 742c 202d 2d74 6d70 6469  _dmd.-t, --tmpdi
+00002c20: 722c 2074 6865 2074 656d 706f 7261 7279  r, the temporary
+00002c30: 2077 6f72 6b69 6e67 2064 6972 6563 746f   working directo
+00002c40: 7279 2c20 6465 6661 756c 7420 4e6f 6e65  ry, default None
+00002c50: 2c20 6966 204e 6f6e 6520 7761 7320 6769  , if None was gi
+00002c60: 7665 6e2c 2074 6865 2074 6d70 6469 7220  ven, the tmpdir 
+00002c70: 776f 756c 6420 6265 2061 7373 6967 6e65  would be assigne
+00002c80: 6420 7261 6e64 6f6d 206e 616d 6573 2069  d random names i
+00002c90: 6e20 6375 7272 656e 7420 6469 7265 6374  n current direct
+00002ca0: 6f72 7920 616e 6420 6175 746f 6d61 7465  ory and automate
+00002cb0: 6c79 2072 656d 6f76 6564 2061 7420 7468  ly removed at th
+00002cc0: 6520 636f 6d70 6c65 7469 6f6e 206f 6620  e completion of 
+00002cd0: 7072 6f67 7261 6d2c 2065 6c73 6520 7468  program, else th
+00002ce0: 6520 746d 7064 6972 2077 6f75 6c64 2062  e tmpdir would b
+00002cf0: 6520 6b65 7074 0a2d 702c 202d 2d70 726f  e kept.-p, --pro
+00002d00: 742c 2066 6c61 6720 6f70 7469 6f6e 2c20  t, flag option, 
+00002d10: 7768 6574 6865 7220 7573 696e 6720 7072  whether using pr
+00002d20: 6f74 6569 6e20 6f72 206e 7563 6c65 6f74  otein or nucleot
+00002d30: 6964 6520 7365 7175 656e 6365 730a 2d63  ide sequences.-c
+00002d40: 2c20 2d2d 6373 636f 7265 2c20 7468 6520  , --cscore, the 
+00002d50: 632d 7363 6f72 6520 746f 2072 6573 7472  c-score to restr
+00002d60: 6963 7420 7468 6520 686f 6d6f 6c6f 6720  ict the homolog 
+00002d70: 7369 6d69 6c61 7269 7479 206f 6620 4d52  similarity of MR
+00002d80: 4248 732c 2064 6566 6175 6c74 204e 6f6e  BHs, default Non
+00002d90: 652c 2069 6620 4e6f 6e65 2077 6173 2067  e, if None was g
+00002da0: 6976 656e 2c20 7468 6520 632d 7363 6f72  iven, the c-scor
+00002db0: 6520 6675 6e63 696f 6e20 776f 756c 646e  e funcion wouldn
+00002dc0: 2774 2062 6520 6163 7469 7661 7465 642c  't be activated,
+00002dd0: 2065 6c73 6520 6578 7065 6374 696e 6720   else expecting 
+00002de0: 6120 6465 6369 6d61 6c20 7769 7468 696e  a decimal within
+00002df0: 2074 6865 2072 616e 6765 206f 6620 3020   the range of 0 
+00002e00: 616e 6420 310a 2d49 2c20 2d2d 696e 666c  and 1.-I, --infl
+00002e10: 6174 696f 6e2c 2074 6865 2069 6e66 6c61  ation, the infla
+00002e20: 7469 6f6e 2066 6163 746f 7220 666f 7220  tion factor for 
+00002e30: 4d43 4c20 7072 6f67 7261 6d2c 2064 6566  MCL program, def
+00002e40: 6175 6c74 2032 2e30 2c20 7769 7468 2068  ault 2.0, with h
+00002e50: 6967 6865 7220 7661 6c75 6520 6c65 6164  igher value lead
+00002e60: 696e 6720 746f 206d 6f72 6520 6275 7420  ing to more but 
+00002e70: 736d 616c 6c65 7220 636c 7573 7465 7273  smaller clusters
+00002e80: 0a2d 652c 202d 2d65 7661 6c2c 2074 6865  .-e, --eval, the
+00002e90: 2065 2d76 616c 7565 2063 7574 2d6f 6666   e-value cut-off
+00002ea0: 2066 6f72 2073 696d 696c 6172 6974 7920   for similarity 
+00002eb0: 696e 2064 6961 6d6f 6e64 2061 6e64 2f6f  in diamond and/o
+00002ec0: 7220 686d 6d65 722c 2064 6566 6175 6c74  r hmmer, default
+00002ed0: 2031 652d 3130 0a2d 2d74 6f5f 7374 6f70   1e-10.--to_stop
+00002ee0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00002ef0: 6865 7468 6572 2074 6f20 7472 616e 736c  hether to transl
+00002f00: 6174 6520 7468 726f 7567 6820 5354 4f50  ate through STOP
+00002f10: 2063 6f64 6f6e 732c 2069 6620 7468 6520   codons, if the 
+00002f20: 666c 6167 2077 6173 2073 6574 2c20 7472  flag was set, tr
+00002f30: 616e 736c 6174 696f 6e20 7769 6c6c 2062  anslation will b
+00002f40: 6520 7465 726d 696e 6174 6564 2061 7420  e terminated at 
+00002f50: 7468 6520 6669 7273 7420 696e 2d66 7261  the first in-fra
+00002f60: 6d65 2073 746f 7020 636f 646f 6e2c 2065  me stop codon, e
+00002f70: 6c73 6520 6120 6675 6c6c 2074 7261 6e73  lse a full trans
+00002f80: 6c61 7469 6f6e 2063 6f6e 7469 6e75 696e  lation continuin
+00002f90: 6720 6f6e 2070 6173 7369 6e67 2061 6e79  g on passing any
+00002fa0: 2073 746f 7020 636f 646f 6e73 2077 6f75   stop codons wou
+00002fb0: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
+00002fc0: 2d2d 6364 732c 2066 6c61 6720 6f70 7469  --cds, flag opti
+00002fd0: 6f6e 2c20 7768 6574 6865 7220 746f 206f  on, whether to o
+00002fe0: 6e6c 7920 7472 616e 736c 6174 6520 7468  nly translate th
+00002ff0: 6520 636f 6d70 6c65 7465 2043 4453 2074  e complete CDS t
+00003000: 6861 7420 7374 6172 7473 2077 6974 6820  hat starts with 
+00003010: 6120 7661 6c69 6420 7374 6172 7420 636f  a valid start co
+00003020: 646f 6e20 616e 6420 6f6e 6c79 2063 6f6e  don and only con
+00003030: 7461 696e 7320 6120 7369 6e67 6c65 2069  tains a single i
+00003040: 6e2d 6672 616d 6520 7374 6f70 2063 6f64  n-frame stop cod
+00003050: 6f6e 2061 7420 7468 6520 656e 6420 616e  on at the end an
+00003060: 6420 6d75 7374 2062 6520 6469 7669 6461  d must be divida
+00003070: 626c 6520 6279 2074 6872 6565 2c20 6966  ble by three, if
+00003080: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00003090: 742c 206f 6e6c 7920 7468 6520 636f 6d70  t, only the comp
+000030a0: 6c65 7465 2043 4453 2077 6f75 6c64 2062  lete CDS would b
+000030b0: 6520 7472 616e 736c 6174 6564 0a2d 662c  e translated.-f,
+000030c0: 202d 2d66 6f63 7573 2c20 7468 6520 7370   --focus, the sp
+000030d0: 6563 6965 7320 746f 2062 6520 6d65 7267  ecies to be merg
+000030e0: 6564 206f 6e20 6c6f 6361 6c20 4d52 4248  ed on local MRBH
+000030f0: 732c 2064 6566 6175 6c74 204e 6f6e 652c  s, default None,
+00003100: 2069 6620 4e6f 6e65 2077 6173 2067 6976   if None was giv
+00003110: 656e 2c20 7468 6520 6c6f 6361 6c20 4d52  en, the local MR
+00003120: 4248 7320 776f 756c 646e 2774 2062 6520  BHs wouldn't be 
+00003130: 696e 6665 7272 6564 0a2d 6170 2c20 2d2d  inferred.-ap, --
+00003140: 616e 6368 6f72 706f 696e 7473 2c20 7468  anchorpoints, th
+00003150: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
+00003160: 6461 7461 2066 696c 6520 6672 6f6d 2069  data file from i
+00003170: 2d61 6468 6f72 6520 666f 7220 636f 6e73  -adhore for cons
+00003180: 7472 7563 7469 6e67 2074 6865 206f 7274  tructing the ort
+00003190: 686f 6772 6f75 7073 2077 6974 6820 616e  hogroups with an
+000031a0: 6368 6f72 2070 6169 7273 2c20 6465 6661  chor pairs, defa
+000031b0: 756c 7420 4e6f 6e65 0a2d 736d 2c20 2d2d  ult None.-sm, --
+000031c0: 7365 676d 656e 7473 2c20 7468 6520 7365  segments, the se
+000031d0: 676d 656e 7473 2064 6174 6166 696c 6520  gments datafile 
+000031e0: 7573 6564 2069 6e20 636f 6c6c 696e 6561  used in collinea
+000031f0: 7220 636f 616c 6573 6365 6e63 6520 616e  r coalescence an
+00003200: 616c 7973 6973 2069 6620 696e 6974 6961  alysis if initia
+00003210: 7465 642c 2064 6566 6175 6c74 204e 6f6e  ted, default Non
+00003220: 650a 2d6c 652c 202d 2d6c 6973 7465 6c65  e.-le, --listele
+00003230: 6d65 6e74 732c 2074 6865 206c 6973 7473  ments, the lists
+00003240: 6567 6d65 6e74 7320 6461 7461 2066 696c  egments data fil
+00003250: 6520 7573 6564 2069 6e20 636f 6c6c 696e  e used in collin
+00003260: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
+00003270: 616e 616c 7973 6973 2069 6620 696e 6974  analysis if init
+00003280: 6961 7465 642c 2064 6566 6175 6c74 204e  iated, default N
+00003290: 6f6e 650a 2d67 742c 202d 2d67 656e 6574  one.-gt, --genet
+000032a0: 6162 6c65 2c20 7468 6520 6765 6e65 2074  able, the gene t
+000032b0: 6162 6c65 2064 6174 6166 696c 6520 7573  able datafile us
+000032c0: 6564 2069 6e20 636f 6c6c 696e 6561 7220  ed in collinear 
+000032d0: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
+000032e0: 7973 6973 2069 6620 696e 6974 6961 7465  ysis if initiate
+000032f0: 642c 2064 6566 6175 6c74 204e 6f6e 650a  d, default None.
+00003300: 2d63 6f63 2c20 2d2d 636f 6c6c 696e 6561  -coc, --collinea
+00003310: 7263 6f61 6c65 7363 656e 6365 2c20 666c  rcoalescence, fl
+00003320: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+00003330: 6572 2074 6f20 696e 6974 6961 7465 2074  er to initiate t
+00003340: 6865 2063 6f6c 6c69 6e65 6172 2063 6f61  he collinear coa
+00003350: 6c65 7363 656e 6365 2061 6e61 6c79 7369  lescence analysi
+00003360: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
+00003370: 6173 2073 6574 2c20 7468 6520 616e 616c  as set, the anal
+00003380: 7973 6973 2077 6f75 6c64 2062 6520 696e  ysis would be in
+00003390: 6974 6961 7465 640a 2d6b 662c 202d 2d6b  itiated.-kf, --k
+000033a0: 6565 7066 6173 7461 2c20 666c 6167 206f  eepfasta, flag o
+000033b0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+000033c0: 6f20 6f75 7470 7574 2074 6865 2073 6571  o output the seq
+000033d0: 7565 6e63 6520 696e 666f 726d 6174 696f  uence informatio
+000033e0: 6e20 6f66 204d 5242 4873 2c20 6966 2074  n of MRBHs, if t
+000033f0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00003400: 2074 6865 2073 6571 7565 6e63 6573 206f   the sequences o
+00003410: 6620 4d52 4248 7320 776f 756c 6420 6265  f MRBHs would be
+00003420: 2069 6e20 6f75 7470 7574 0a2d 6b64 2c20   in output.-kd, 
+00003430: 2d2d 6b65 6570 6475 706c 6963 6174 6573  --keepduplicates
+00003440: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00003450: 6865 7468 6572 2074 6f20 616c 6c6f 7720  hether to allow 
+00003460: 7468 6520 7361 6d65 2067 656e 6520 746f  the same gene to
+00003470: 206f 6363 7572 2069 6e20 6469 6666 6572   occur in differ
+00003480: 656e 7420 4d52 4248 7320 286f 6e6c 7920  ent MRBHs (only 
+00003490: 6d65 616e 696e 6766 756c 2077 6865 6e20  meaningful when 
+000034a0: 7468 6520 6373 636f 7265 2077 6173 2075  the cscore was u
+000034b0: 7365 6429 2c20 6966 2074 6865 2066 6c61  sed), if the fla
+000034c0: 6720 7761 7320 7365 742c 2074 6865 2073  g was set, the s
+000034d0: 616d 6520 6765 6e65 2063 6f75 6c64 2062  ame gene could b
+000034e0: 6520 6173 7369 676e 6564 2074 6f20 6469  e assigned to di
+000034f0: 6666 6572 656e 7420 4d52 4248 730a 2d67  fferent MRBHs.-g
+00003500: 6d2c 202d 2d67 6c6f 6261 6c6d 7262 682c  m, --globalmrbh,
+00003510: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00003520: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
+00003530: 6520 676c 6f62 616c 204d 5242 4873 2063  e global MRBHs c
+00003540: 6f6e 7374 7275 6374 696f 6e2c 2069 6620  onstruction, if 
+00003550: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00003560: 2c20 7468 6520 2d2d 666f 6375 7320 6f70  , the --focus op
+00003570: 7469 6f6e 2077 6f75 6c64 2062 6520 6967  tion would be ig
+00003580: 6e6f 7265 6420 616e 6420 6f6e 6c79 2067  nored and only g
+00003590: 6c6f 6261 6c20 4d52 4248 7320 776f 756c  lobal MRBHs woul
+000035a0: 6420 6265 2062 7569 6c74 0a2d 6e2c 202d  d be built.-n, -
+000035b0: 2d6e 7468 7265 6164 732c 2074 6865 206e  -nthreads, the n
+000035c0: 756d 6265 7220 6f66 2074 6872 6561 6473  umber of threads
+000035d0: 2074 6f20 7573 652c 2064 6566 6175 6c74   to use, default
+000035e0: 2034 0a2d 6f69 2c20 2d2d 6f72 7468 6f69   4.-oi, --orthoi
+000035f0: 6e66 6572 2c20 666c 6167 206f 7074 696f  nfer, flag optio
+00003600: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
+00003610: 6974 6961 7465 206f 7274 686f 6772 6f75  itiate orthogrou
+00003620: 7020 696e 6665 726e 6563 652c 2069 6620  p infernece, if 
+00003630: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00003640: 2c20 7468 6520 6f72 7468 6f67 726f 7570  , the orthogroup
+00003650: 2069 6e66 6572 6e65 6365 2070 726f 6772   infernece progr
+00003660: 616d 2077 6f75 6c64 2062 6520 696e 6974  am would be init
+00003670: 6961 7465 640a 2d6f 6f2c 202d 2d6f 6e6c  iated.-oo, --onl
+00003680: 796f 7274 686f 2c20 666c 6167 206f 7074  yortho, flag opt
+00003690: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+000036a0: 6f6e 6c79 2063 6f6e 6475 6374 206f 7274  only conduct ort
+000036b0: 686f 6772 6f75 7020 696e 6665 726e 6563  hogroup infernec
+000036c0: 652c 2069 6620 7468 6520 666c 6167 2077  e, if the flag w
+000036d0: 6173 2073 6574 2c20 6f6e 6c79 2074 6865  as set, only the
+000036e0: 206f 7274 686f 6772 6f75 7020 696e 6665   orthogroup infe
+000036f0: 726e 6563 6520 7069 7065 6c69 6e65 2077  rnece pipeline w
+00003700: 6f75 6c64 2062 6520 7065 7266 6f72 6d65  ould be performe
+00003710: 6420 7768 696c 6520 7468 6520 6f74 6865  d while the othe
+00003720: 7220 616e 616c 7973 6973 2077 6f75 6c64  r analysis would
+00003730: 6e27 7420 6265 2069 6e69 7469 6174 6564  n't be initiated
+00003740: 0a2d 676e 2c20 2d2d 6765 746e 736f 672c  .-gn, --getnsog,
+00003750: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00003760: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
+00003770: 6520 7468 6520 7365 6172 6368 696e 6720  e the searching 
+00003780: 666f 7220 6e65 7374 6564 2073 696e 676c  for nested singl
+00003790: 652d 636f 7079 2067 656e 6520 6661 6d69  e-copy gene fami
+000037a0: 6c69 6573 2028 4e53 4f47 7329 2028 6f6e  lies (NSOGs) (on
+000037b0: 6c79 206d 6561 6e69 6e67 6675 6c20 7768  ly meaningful wh
+000037c0: 656e 2074 6865 206f 7274 686f 6772 6f75  en the orthogrou
+000037d0: 7020 696e 6665 726e 6563 6520 7069 7065  p infernece pipe
+000037e0: 6c69 6e65 2077 6173 2061 6374 6976 6174  line was activat
+000037f0: 6564 292c 2069 6620 7468 6520 666c 6167  ed), if the flag
+00003800: 2077 6173 2073 6574 2c20 6164 6469 7469   was set, additi
+00003810: 6f6e 616c 204e 534f 4773 2061 6e61 6c79  onal NSOGs analy
+00003820: 7369 7320 776f 756c 6420 6265 2070 6572  sis would be per
+00003830: 666f 726d 6564 2062 6573 6964 6573 2074  formed besides t
+00003840: 6865 2062 6173 6963 206f 7274 686f 6772  he basic orthogr
+00003850: 6f75 7020 696e 6665 726e 6563 650a 2d74  oup infernece.-t
+00003860: 7265 652c 202d 2d74 7265 655f 6d65 7468  ree, --tree_meth
+00003870: 6f64 2c20 7768 6963 6820 6765 6e65 2074  od, which gene t
+00003880: 7265 6520 696e 6665 7265 6e63 6520 7072  ree inference pr
+00003890: 6f67 7261 6d20 746f 2069 6e76 6f6b 6520  ogram to invoke 
+000038a0: 286f 6e6c 7920 6d65 616e 696e 6766 756c  (only meaningful
+000038b0: 2077 6865 6e20 7468 6520 636f 6c6c 696e   when the collin
+000038c0: 6561 7220 636f 616c 6573 6365 6e63 652c  ear coalescence,
+000038d0: 2067 656e 652d 746f 2d66 616d 696c 7920   gene-to-family 
+000038e0: 6173 7369 676e 6d65 6e74 206f 7220 4e53  assignment or NS
+000038f0: 4f47 7320 616e 616c 7973 6973 2077 6572  OGs analysis wer
+00003900: 6520 6163 7469 7661 7465 6429 2c20 6465  e activated), de
+00003910: 6661 756c 7420 6661 7374 7472 6565 0a2d  fault fasttree.-
+00003920: 7473 2c20 2d2d 7472 6565 7365 742c 2074  ts, --treeset, t
+00003930: 6865 2070 6172 616d 6574 6572 7320 7365  he parameters se
+00003940: 7474 696e 6720 666f 7220 6765 6e65 2074  tting for gene t
+00003950: 7265 6520 696e 6665 7265 6e63 652c 2064  ree inference, d
+00003960: 6566 6175 6c74 204e 6f6e 652c 2074 6869  efault None, thi
+00003970: 7320 6f70 7469 6f6e 2063 616e 2062 6520  s option can be 
+00003980: 7072 6f76 6964 6564 206d 756c 7469 706c  provided multipl
+00003990: 6520 7469 6d65 730a 2d6d 632c 202d 2d6d  e times.-mc, --m
+000039a0: 736f 6763 7574 2c20 7468 6520 7261 7469  sogcut, the rati
+000039b0: 6f20 6375 746f 6666 2066 6f72 206d 6f73  o cutoff for mos
+000039c0: 746c 7920 7369 6e67 6c65 2d63 6f70 7920  tly single-copy 
+000039d0: 6661 6d69 6c79 2028 6d65 616e 696e 6766  family (meaningf
+000039e0: 756c 2077 6865 6e20 6163 7469 7661 7469  ul when activati
+000039f0: 6e67 2074 6865 206f 7274 686f 6772 6f75  ng the orthogrou
+00003a00: 7020 696e 6665 726e 6563 6520 7069 7065  p infernece pipe
+00003a10: 6c69 6e65 2920 616e 6420 7370 6563 6965  line) and specie
+00003a20: 7320 7265 7072 6573 656e 7461 7469 6f6e  s representation
+00003a30: 2069 6e20 636f 6c6c 696e 6561 7220 636f   in collinear co
+00003a40: 616c 6573 6365 6e63 6520 616e 616c 7973  alescence analys
+00003a50: 6973 2c20 6465 6661 756c 7420 302e 382e  is, default 0.8.
+00003a60: 0a2d 6761 2c20 2d2d 6765 6e65 6173 7369  .-ga, --geneassi
+00003a70: 676e 2c20 666c 6167 206f 7074 696f 6e2c  gn, flag option,
+00003a80: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
+00003a90: 6961 7465 2074 6865 2067 656e 652d 746f  iate the gene-to
+00003aa0: 2d66 616d 696c 7920 6173 7369 676e 6d65  -family assignme
+00003ab0: 6e74 2061 6e61 6c79 7369 732c 2069 6620  nt analysis, if 
+00003ac0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00003ad0: 2c20 7468 6520 616e 616c 7973 6973 2077  , the analysis w
+00003ae0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+00003af0: 640a 2d73 612c 202d 2d73 6571 3261 7373  d.-sa, --seq2ass
+00003b00: 6967 6e2c 2074 6865 2071 7565 7269 6564  ign, the queried
+00003b10: 2073 6571 7565 6e63 6573 2064 6174 6120   sequences data 
+00003b20: 6669 6c65 2069 6e20 6765 6e65 2d74 6f2d  file in gene-to-
+00003b30: 6661 6d69 6c79 2061 7373 6967 6e6d 656e  family assignmen
+00003b40: 7420 616e 616c 7973 6973 2c20 6465 6661  t analysis, defa
+00003b50: 756c 7420 4e6f 6e65 2c20 7468 6973 206f  ult None, this o
+00003b60: 7074 696f 6e20 6361 6e20 6265 2070 726f  ption can be pro
+00003b70: 7669 6465 6420 6d75 6c74 6970 6c65 2074  vided multiple t
+00003b80: 696d 6573 0a2d 6661 2c20 2d2d 6661 6d32  imes.-fa, --fam2
+00003b90: 6173 7369 676e 2c20 7468 6520 7175 6572  assign, the quer
+00003ba0: 6965 6420 6661 6d69 6c69 7920 6461 7461  ied familiy data
+00003bb0: 2066 696c 6520 696e 2067 656e 652d 746f   file in gene-to
+00003bc0: 2d66 616d 696c 7920 6173 7369 676e 6d65  -family assignme
+00003bd0: 6e74 2061 6e61 6c79 7369 732c 2064 6566  nt analysis, def
+00003be0: 6175 6c74 204e 6f6e 650a 2d63 632c 202d  ault None.-cc, -
+00003bf0: 2d63 6f6e 6361 742c 2066 6c61 6720 6f70  -concat, flag op
+00003c00: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00003c10: 2069 6e69 7469 6174 6520 7468 6520 636f   initiate the co
+00003c20: 6e63 6174 656e 6174 696f 6e20 7069 7065  ncatenation pipe
+00003c30: 6c69 6e65 2066 6f72 206f 7274 686f 6772  line for orthogr
+00003c40: 6f75 7020 696e 6665 726e 6563 652c 2069  oup infernece, i
+00003c50: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00003c60: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
+00003c70: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+00003c80: 7465 640a 2d74 652c 202d 2d74 6573 7473  ted.-te, --tests
+00003c90: 6f67 2c20 666c 6167 206f 7074 696f 6e2c  og, flag option,
+00003ca0: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
+00003cb0: 6961 7465 2074 6865 2075 6e62 6961 7365  iate the unbiase
+00003cc0: 6420 7465 7374 206f 6620 7369 6e67 6c65  d test of single
+00003cd0: 2d63 6f70 7920 6765 6e65 2066 616d 696c  -copy gene famil
+00003ce0: 6965 732c 2069 6620 7468 6520 666c 6167  ies, if the flag
+00003cf0: 2077 6173 2073 6574 2c20 7468 6520 616e   was set, the an
+00003d00: 616c 7973 6973 2077 6f75 6c64 2062 6520  alysis would be 
+00003d10: 696e 6974 6961 7465 640a 2d62 732c 202d  initiated.-bs, -
+00003d20: 2d62 696e 732c 2074 6865 206e 756d 6265  -bins, the numbe
+00003d30: 7220 6f66 2062 696e 7320 6469 7669 6465  r of bins divide
+00003d40: 6420 696e 2074 6865 2067 656e 6520 6c65  d in the gene le
+00003d50: 6e67 7468 206e 6f72 6d61 6c69 7a61 7469  ngth normalizati
+00003d60: 6f6e 2c20 6465 6661 756c 7420 3130 300a  on, default 100.
+00003d70: 2d6e 702c 202d 2d6e 6f72 6d61 6c69 7a65  -np, --normalize
+00003d80: 6470 6572 6365 6e74 2c20 7468 6520 7065  dpercent, the pe
+00003d90: 7263 656e 7461 6765 206f 6620 7570 7065  rcentage of uppe
+00003da0: 7220 6869 7473 2075 7365 6420 666f 7220  r hits used for 
+00003db0: 6765 6e65 206c 656e 6774 6820 6e6f 726d  gene length norm
+00003dc0: 616c 697a 6174 696f 6e2c 2064 6566 6175  alization, defau
+00003dd0: 6c74 2035 0a2d 6e6e 2c20 2d2d 6e6f 6e6f  lt 5.-nn, --nono
+00003de0: 726d 616c 697a 6174 696f 6e2c 2066 6c61  rmalization, fla
+00003df0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00003e00: 7220 746f 2063 616c 6c20 6f66 6620 7468  r to call off th
+00003e10: 6520 6e6f 726d 616c 697a 6174 696f 6e2c  e normalization,
+00003e20: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00003e30: 2073 6574 2c20 6e6f 206e 6f72 6d61 6c69   set, no normali
+00003e40: 7a61 7469 6f6e 2077 6f75 6c64 2062 6520  zation would be 
+00003e50: 636f 6e64 7563 7465 640a 2d62 736f 672c  conducted.-bsog,
+00003e60: 202d 2d62 7573 636f 736f 672c 2066 6c61   --buscosog, fla
+00003e70: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00003e80: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
+00003e90: 6520 6275 7363 6f2d 6775 6964 6564 2073  e busco-guided s
+00003ea0: 696e 676c 652d 636f 7079 2067 656e 6520  ingle-copy gene 
+00003eb0: 6661 6d69 6c79 2061 6e61 6c79 7369 732c  family analysis,
+00003ec0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00003ed0: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
+00003ee0: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
+00003ef0: 6961 7465 640a 2d62 686d 6d2c 202d 2d62  iated.-bhmm, --b
+00003f00: 7573 636f 686d 6d2c 2074 6865 2048 4d4d  uscohmm, the HMM
+00003f10: 2070 726f 6669 6c65 2064 6174 6166 696c   profile datafil
+00003f20: 6520 696e 2074 6865 2062 7573 636f 2d67  e in the busco-g
+00003f30: 7569 6465 6420 7369 6e67 6c65 2d63 6f70  uided single-cop
+00003f40: 7920 6765 6e65 2066 616d 696c 7920 616e  y gene family an
+00003f50: 616c 7973 6973 2c20 6465 6661 756c 7420  alysis, default 
+00003f60: 4e6f 6e65 0a2d 6263 7466 2c20 2d2d 6275  None.-bctf, --bu
+00003f70: 7363 6f63 7574 6f66 662c 2074 6865 2048  scocutoff, the H
+00003f80: 4d4d 2073 636f 7265 2063 7574 6f66 6620  MM score cutoff 
+00003f90: 6461 7461 6669 6c65 2069 6e20 7468 6520  datafile in the 
+00003fa0: 6275 7363 6f2d 6775 6964 6564 2073 696e  busco-guided sin
+00003fb0: 676c 652d 636f 7079 2067 656e 6520 6661  gle-copy gene fa
+00003fc0: 6d69 6c79 2061 6e61 6c79 7369 732c 2064  mily analysis, d
+00003fd0: 6566 6175 6c74 204e 6f6e 650a 2d6f 6620  efault None.-of 
+00003fe0: 2c2d 2d6f 6766 6f72 6d61 742c 2066 6c61  ,--ogformat, fla
+00003ff0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00004000: 7220 746f 2061 6464 2069 6e64 6578 2074  r to add index t
+00004010: 6f20 7468 6520 5242 4820 6661 6d69 6c69  o the RBH famili
+00004020: 6573 0a60 6060 0a0a 5468 6520 7072 6f67  es.```..The prog
+00004030: 7261 6d20 6077 6764 2066 6f63 7573 6020  ram `wgd focus` 
+00004040: 6361 6e20 7265 616c 697a 6520 7468 6520  can realize the 
+00004050: 636f 6e63 6174 656e 6174 696f 6e2d 6261  concatenation-ba
+00004060: 7365 6420 616e 6420 636f 616c 6573 6365  sed and coalesce
+00004070: 6e63 652d 6261 7365 6420 7068 796c 6f67  nce-based phylog
+00004080: 656e 6574 6963 2069 6e66 6572 656e 6365  enetic inference
+00004090: 2061 6e64 2070 6879 6c6f 6765 6e65 7469   and phylogeneti
+000040a0: 6320 6461 7469 6e67 206f 6620 5747 4473  c dating of WGDs
+000040b0: 2065 7463 2e0a 6060 600a 7767 6420 666f   etc..```.wgd fo
+000040c0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+000040d0: 7565 6e63 6573 2028 6f70 7469 6f6e 290a  uences (option).
+000040e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000040f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004100: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004110: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004120: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004130: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
+00004140: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
+00004150: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
+00004160: 5f66 6f63 7573 0a2d 742c 202d 2d74 6d70  _focus.-t, --tmp
+00004170: 6469 722c 2074 6865 2074 656d 706f 7261  dir, the tempora
+00004180: 7279 2077 6f72 6b69 6e67 2064 6972 6563  ry working direc
+00004190: 746f 7279 2c20 6465 6661 756c 7420 4e6f  tory, default No
+000041a0: 6e65 2c20 6966 204e 6f6e 6520 7761 7320  ne, if None was 
+000041b0: 6769 7665 6e2c 2074 6865 2074 6d70 6469  given, the tmpdi
+000041c0: 7220 7769 6c6c 2062 6520 6173 7369 676e  r will be assign
+000041d0: 6564 2072 616e 646f 6d20 6e61 6d65 7320  ed random names 
+000041e0: 696e 2063 7572 7265 6e74 2064 6972 6563  in current direc
+000041f0: 746f 7279 2061 6e64 2061 7574 6f6d 6174  tory and automat
+00004200: 656c 7920 7265 6d6f 7665 6420 6174 2074  ely removed at t
+00004210: 6865 2063 6f6d 706c 6574 696f 6e20 6f66  he completion of
+00004220: 2070 726f 6772 616d 2c20 656c 7365 2074   program, else t
+00004230: 6865 2074 6d70 6469 7220 776f 756c 6420  he tmpdir would 
+00004240: 6265 206b 6570 740a 2d6e 2c20 2d2d 6e74  be kept.-n, --nt
+00004250: 6872 6561 6473 2c20 7468 6520 6e75 6d62  hreads, the numb
+00004260: 6572 206f 6620 7468 7265 6164 7320 746f  er of threads to
+00004270: 2075 7365 2c20 6465 6661 756c 7420 340a   use, default 4.
+00004280: 2d2d 746f 5f73 746f 702c 2066 6c61 6720  --to_stop, flag 
+00004290: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+000042a0: 746f 2074 7261 6e73 6c61 7465 2074 6872  to translate thr
+000042b0: 6f75 6768 2053 544f 5020 636f 646f 6e73  ough STOP codons
+000042c0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+000042d0: 7320 7365 742c 2074 7261 6e73 6c61 7469  s set, translati
+000042e0: 6f6e 2077 696c 6c20 6265 2074 6572 6d69  on will be termi
+000042f0: 6e61 7465 6420 6174 2074 6865 2066 6972  nated at the fir
+00004300: 7374 2069 6e2d 6672 616d 6520 7374 6f70  st in-frame stop
+00004310: 2063 6f64 6f6e 2c20 656c 7365 2061 2066   codon, else a f
+00004320: 756c 6c20 7472 616e 736c 6174 696f 6e20  ull translation 
+00004330: 636f 6e74 696e 7569 6e67 206f 6e20 7061  continuing on pa
+00004340: 7374 2061 6e79 2073 746f 7020 636f 646f  st any stop codo
+00004350: 6e73 2077 6f75 6c64 2062 6520 696e 6974  ns would be init
+00004360: 6961 7465 640a 2d2d 6364 732c 2066 6c61  iated.--cds, fla
+00004370: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00004380: 7220 746f 206f 6e6c 7920 7472 616e 736c  r to only transl
+00004390: 6174 6520 7468 6520 636f 6d70 6c65 7465  ate the complete
+000043a0: 2043 4453 2074 6861 7420 7374 6172 7473   CDS that starts
+000043b0: 2077 6974 6820 6120 7661 6c69 6420 7374   with a valid st
+000043c0: 6172 7420 636f 646f 6e20 616e 6420 6f6e  art codon and on
+000043d0: 6c79 2063 6f6e 7461 696e 7320 6120 7369  ly contains a si
+000043e0: 6e67 6c65 2069 6e2d 6672 616d 6520 7374  ngle in-frame st
+000043f0: 6f70 2063 6f64 6f6e 2061 7420 7468 6520  op codon at the 
+00004400: 656e 6420 616e 6420 6d75 7374 2062 6520  end and must be 
+00004410: 6469 7669 6461 626c 6520 6279 2074 6872  dividable by thr
+00004420: 6565 2c20 6966 2074 6865 2066 6c61 6720  ee, if the flag 
+00004430: 7761 7320 7365 742c 206f 6e6c 7920 7468  was set, only th
+00004440: 6520 636f 6d70 6c65 7465 2043 4453 2077  e complete CDS w
+00004450: 6f75 6c64 2062 6520 7472 616e 736c 6174  ould be translat
+00004460: 6564 0a2d 2d73 7472 6970 5f67 6170 732c  ed.--strip_gaps,
+00004470: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00004480: 6574 6865 7220 746f 2064 726f 7020 616c  ether to drop al
+00004490: 6c20 6761 7073 2069 6e20 6d75 6c74 6970  l gaps in multip
+000044a0: 6c65 2073 6571 7565 6e63 6520 616c 6967  le sequence alig
+000044b0: 6e6d 656e 742c 2069 6620 7468 6520 666c  nment, if the fl
+000044c0: 6167 2077 6173 2073 6574 2c20 616c 6c20  ag was set, all 
+000044d0: 6761 7073 2077 6f75 6c64 2062 6520 6472  gaps would be dr
+000044e0: 6f70 7065 640a 2d61 2c20 2d2d 616c 6967  opped.-a, --alig
+000044f0: 6e65 722c 2077 6869 6368 2061 6c69 676e  ner, which align
+00004500: 6d65 6e74 2070 726f 6772 616d 2074 6f20  ment program to 
+00004510: 7573 652c 2064 6566 6175 6c74 206d 6166  use, default maf
+00004520: 6674 0a2d 7472 6565 2c20 2d2d 7472 6565  ft.-tree, --tree
+00004530: 5f6d 6574 686f 642c 2077 6869 6368 2067  _method, which g
+00004540: 656e 6520 7472 6565 2069 6e66 6572 656e  ene tree inferen
+00004550: 6365 2070 726f 6772 616d 2074 6f20 696e  ce program to in
+00004560: 766f 6b65 2c20 6465 6661 756c 7420 6661  voke, default fa
+00004570: 7374 7472 6565 0a2d 7473 2c20 2d2d 7472  sttree.-ts, --tr
+00004580: 6565 7365 742c 2074 6865 2070 6172 616d  eeset, the param
+00004590: 6574 6572 7320 7365 7474 696e 6720 666f  eters setting fo
+000045a0: 7220 6765 6e65 2074 7265 6520 696e 6665  r gene tree infe
+000045b0: 7265 6e63 652c 2064 6566 6175 6c74 204e  rence, default N
+000045c0: 6f6e 652c 2074 6869 7320 6f70 7469 6f6e  one, this option
+000045d0: 2063 616e 2062 6520 7072 6f76 6964 6564   can be provided
+000045e0: 206d 756c 7469 706c 6520 7469 6d65 730a   multiple times.
+000045f0: 2d2d 636f 6e63 6174 656e 6174 696f 6e2c  --concatenation,
+00004600: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00004610: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
+00004620: 6520 7468 6520 636f 6e63 6174 656e 6174  e the concatenat
+00004630: 696f 6e2d 6261 7365 6420 7370 6563 6965  ion-based specie
+00004640: 7320 7472 6565 2069 6e66 6572 656e 6365  s tree inference
+00004650: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00004660: 7320 7365 742c 2063 6f6e 6361 7465 6e61  s set, concatena
+00004670: 7469 6f6e 2d62 6173 6564 2073 7065 6369  tion-based speci
+00004680: 6573 2074 7265 6520 776f 756c 6420 6265  es tree would be
+00004690: 2069 6e66 6572 6564 0a2d 2d63 6f61 6c65   infered.--coale
+000046a0: 7363 656e 6365 2c20 666c 6167 206f 7074  scence, flag opt
+000046b0: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+000046c0: 696e 6974 6961 7465 2074 6865 2063 6f61  initiate the coa
+000046d0: 6c65 7363 656e 6365 2d62 6173 6564 2073  lescence-based s
+000046e0: 7065 6369 6573 2074 7265 6520 696e 6665  pecies tree infe
+000046f0: 7265 6e63 652c 2069 6620 7468 6520 666c  rence, if the fl
+00004700: 6167 2077 6173 2073 6574 2c20 636f 616c  ag was set, coal
+00004710: 6573 6365 6e63 652d 6261 7365 6420 7370  escence-based sp
+00004720: 6563 6965 7320 7472 6565 2077 6f75 6c64  ecies tree would
+00004730: 2062 6520 696e 6665 7265 640a 2d73 702c   be infered.-sp,
+00004740: 202d 2d73 7065 6369 6573 7472 6565 2c20   --speciestree, 
+00004750: 7370 6563 6965 7320 7472 6565 2064 6174  species tree dat
+00004760: 6166 696c 6520 666f 7220 6461 7469 6e67  afile for dating
+00004770: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+00004780: 642c 202d 2d64 6174 696e 672c 2077 6869  d, --dating, whi
+00004790: 6368 206d 6f6c 6563 756c 6172 2064 6174  ch molecular dat
+000047a0: 696e 6720 7072 6f67 7261 6d20 746f 2075  ing program to u
+000047b0: 7365 2c20 6465 6661 756c 7420 6e6f 6e65  se, default none
+000047c0: 0a2d 6473 2c20 2d2d 6461 7469 6e67 7365  .-ds, --datingse
+000047d0: 742c 2074 6865 2070 6172 616d 6574 6572  t, the parameter
+000047e0: 7320 7365 7474 696e 6720 666f 7220 6461  s setting for da
+000047f0: 7469 6e67 2070 726f 6772 616d 2c20 6465  ting program, de
+00004800: 6661 756c 7420 4e6f 6e65 2c20 7468 6973  fault None, this
+00004810: 206f 7074 696f 6e20 6361 6e20 6265 2070   option can be p
+00004820: 726f 7669 6465 6420 6d75 6c74 6970 6c65  rovided multiple
+00004830: 2074 696d 6573 0a2d 6e73 2c20 2d2d 6e73   times.-ns, --ns
+00004840: 6974 6573 2c20 7468 6520 6e73 6974 6573  ites, the nsites
+00004850: 2069 6e66 6f72 6d61 7469 6f6e 2066 6f72   information for
+00004860: 2072 3873 2064 6174 696e 672c 2064 6566   r8s dating, def
+00004870: 6175 6c74 204e 6f6e 650a 2d6f 742c 202d  ault None.-ot, -
+00004880: 2d6f 7574 6772 6f75 702c 2074 6865 206f  -outgroup, the o
+00004890: 7574 6772 6f75 7020 696e 666f 726d 6174  utgroup informat
+000048a0: 696f 6e20 666f 7220 7238 7320 6461 7469  ion for r8s dati
+000048b0: 6e67 2c20 6465 6661 756c 7420 4e6f 6e65  ng, default None
+000048c0: 0a2d 7074 2c20 2d2d 7061 7274 6974 696f  .-pt, --partitio
+000048d0: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
+000048e0: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
+000048f0: 6174 6520 7061 7274 6974 696f 6e20 6461  ate partition da
+00004900: 7469 6e67 2061 6e61 6c79 7369 7320 666f  ting analysis fo
+00004910: 7220 636f 646f 6e2c 2069 6620 7468 6520  r codon, if the 
+00004920: 666c 6167 2077 6173 2073 6574 2c20 616e  flag was set, an
+00004930: 2061 6464 6974 696f 6e61 6c20 7061 7274   additional part
+00004940: 6974 696f 6e20 6461 7469 6e67 2061 6e61  ition dating ana
+00004950: 6c79 7369 7320 776f 756c 6420 6265 2069  lysis would be i
+00004960: 6e69 7469 6174 6564 0a2d 616d 2c20 2d2d  nitiated.-am, --
+00004970: 6161 6d6f 6465 6c2c 2077 6869 6368 2070  aamodel, which p
+00004980: 726f 7465 696e 206d 6f64 656c 2074 6f20  rotein model to 
+00004990: 6265 2075 7365 6420 696e 206d 636d 6374  be used in mcmct
+000049a0: 7265 652c 2064 6566 6175 6c74 2070 6f69  ree, default poi
+000049b0: 7373 6f6e 0a2d 6b73 2c20 666c 6167 206f  sson.-ks, flag o
+000049c0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+000049d0: 6f20 696e 6974 6961 7465 204b 7320 6361  o initiate Ks ca
+000049e0: 6c63 756c 6174 696f 6e20 666f 7220 686f  lculation for ho
+000049f0: 6d6f 6c6f 6775 6573 2069 6e20 7468 6520  mologues in the 
+00004a00: 7072 6f76 6964 6564 206f 7274 686f 6c6f  provided ortholo
+00004a10: 676f 7573 2067 656e 6520 6661 6d69 6c79  gous gene family
+00004a20: 0a2d 2d61 6e6e 6f74 6174 696f 6e2c 2077  .--annotation, w
+00004a30: 6869 6368 2061 6e6e 6f74 6174 696f 6e20  hich annotation 
+00004a40: 7072 6f67 7261 6d20 746f 2075 7365 2c20  program to use, 
+00004a50: 6465 6661 756c 7420 4e6f 6e65 0a2d 2d70  default None.--p
+00004a60: 6169 7277 6973 652c 2066 6c61 6720 6f70  airwise, flag op
+00004a70: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00004a80: 2069 6e69 7469 6174 6520 7061 6972 7769   initiate pairwi
+00004a90: 7365 204b 7320 6573 7469 6d61 7469 6f6e  se Ks estimation
+00004aa0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00004ab0: 7320 7365 742c 2070 6169 7277 6973 6520  s set, pairwise 
+00004ac0: 4b73 2076 616c 7565 7320 776f 756c 6420  Ks values would 
+00004ad0: 6265 2065 7374 696d 6174 6564 0a2d 6564  be estimated.-ed
+00004ae0: 2c20 2d2d 6567 676e 6f67 6461 7461 2c20  , --eggnogdata, 
+00004af0: 7468 6520 6567 676e 6f67 2061 6e6e 6f74  the eggnog annot
+00004b00: 6174 696f 6e20 6461 7461 6669 6c65 2c20  ation datafile, 
+00004b10: 6465 6661 756c 7420 4e6f 6e65 0a2d 2d70  default None.--p
+00004b20: 6661 6d2c 2077 6869 6368 206f 7074 696f  fam, which optio
+00004b30: 6e20 746f 2075 7365 2066 6f72 2070 6661  n to use for pfa
+00004b40: 6d20 616e 6e6f 7461 7469 6f6e 2c20 6465  m annotation, de
+00004b50: 6661 756c 7420 4e6f 6e65 0a2d 2d64 6d6e  fault None.--dmn
+00004b60: 622c 2074 6865 2064 6961 6d6f 6e64 2064  b, the diamond d
+00004b70: 6174 6162 6173 6520 666f 7220 616e 6e6f  atabase for anno
+00004b80: 7461 7469 6f6e 2c20 6465 6661 756c 7420  tation, default 
+00004b90: 4e6f 6e65 0a2d 2d68 6d6d 2c20 7468 6520  None.--hmm, the 
+00004ba0: 484d 4d20 7072 6f66 696c 6520 666f 7220  HMM profile for 
+00004bb0: 616e 6e6f 7461 7469 6f6e 2c20 6465 6661  annotation, defa
+00004bc0: 756c 7420 4e6f 6e65 0a2d 2d65 7661 6c75  ult None.--evalu
+00004bd0: 652c 2074 6865 2065 2d76 616c 7565 2063  e, the e-value c
+00004be0: 7574 2d6f 6666 2066 6f72 2061 6e6e 6f74  ut-off for annot
+00004bf0: 6174 696f 6e2c 2064 6566 6175 6c74 2031  ation, default 1
+00004c00: 652d 3130 0a2d 2d65 7865 7061 7468 2c20  e-10.--exepath, 
+00004c10: 7468 6520 7061 7468 2074 6f20 7468 6520  the path to the 
+00004c20: 696e 7465 7270 726f 7363 616e 2065 7865  interproscan exe
+00004c30: 6375 7461 626c 652c 2064 6566 6175 6c74  cutable, default
+00004c40: 204e 6f6e 650a 2d66 2c20 2d2d 666f 7373   None.-f, --foss
+00004c50: 696c 2c20 7468 6520 666f 7373 696c 2063  il, the fossil c
+00004c60: 616c 6962 7261 7469 6f6e 2069 6e66 6f72  alibration infor
+00004c70: 6d61 7469 6f6e 2069 6e20 4265 6173 742c  mation in Beast,
+00004c80: 2064 6566 6175 6c74 2028 2763 6c61 6465   default ('clade
+00004c90: 313b 636c 6164 6532 272c 2027 7461 7861  1;clade2', 'taxa
+00004ca0: 312c 7461 7861 323b 7461 7861 332c 7461  1,taxa2;taxa3,ta
+00004cb0: 7861 3427 2c20 2734 3b35 272c 2027 302e  xa4', '4;5', '0.
+00004cc0: 353b 302e 3627 2c20 2734 3030 3b35 3030  5;0.6', '400;500
+00004cd0: 2729 0a2d 7268 2c20 2d2d 726f 6f74 6865  ').-rh, --roothe
+00004ce0: 6967 6874 2c20 7468 6520 726f 6f74 2068  ight, the root h
+00004cf0: 6569 6768 7420 6361 6c69 6272 6174 696f  eight calibratio
+00004d00: 6e20 696e 666f 2069 6e20 4265 6173 742c  n info in Beast,
+00004d10: 2064 6566 6175 6c74 2028 342c 302e 352c   default (4,0.5,
+00004d20: 3430 3029 0a2d 6373 2c20 2d2d 6368 6169  400).-cs, --chai
+00004d30: 6e73 6574 2c20 7468 6520 7061 7261 6d65  nset, the parame
+00004d40: 7465 7273 206f 6620 4d43 4d43 2063 6861  ters of MCMC cha
+00004d50: 696e 2069 6e20 4265 6173 742c 2064 6566  in in Beast, def
+00004d60: 6175 6c74 2028 3130 3030 302c 3130 3029  ault (10000,100)
+00004d70: 0a2d 2d62 6561 7374 6c67 6a61 722c 2074  .--beastlgjar, t
+00004d80: 6865 2070 6174 6820 746f 2062 6561 7374  he path to beast
+00004d90: 4c47 2e6a 6172 2c20 6465 6661 756c 7420  LG.jar, default 
+00004da0: 4e6f 6e65 0a2d 2d62 6561 676c 652c 2066  None.--beagle, f
+00004db0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00004dc0: 6865 7220 746f 2075 7365 2062 6561 676c  her to use beagl
+00004dd0: 6520 696e 2042 6561 7374 2c20 6966 2074  e in Beast, if t
+00004de0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00004df0: 2062 6561 676c 6520 776f 756c 6420 6265   beagle would be
+00004e00: 2075 7365 640a 2d2d 7072 6f74 636f 6364   used.--protcocd
+00004e10: 6174 696e 672c 2066 6c61 6720 6f70 7469  ating, flag opti
+00004e20: 6f6e 2c20 7768 6574 6865 7220 746f 206f  on, whether to o
+00004e30: 6e6c 7920 696e 6974 6961 7465 2074 6865  nly initiate the
+00004e40: 2070 726f 7465 696e 2d63 6f6e 6361 7465   protein-concate
+00004e50: 6e61 7469 6f6e 2d62 6173 6564 2064 6174  nation-based dat
+00004e60: 696e 6720 616e 616c 7973 6973 2c20 6966  ing analysis, if
+00004e70: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00004e80: 742c 2074 6865 2061 6e61 6c79 7369 7320  t, the analysis 
+00004e90: 776f 756c 6420 6265 2069 6e69 7469 6174  would be initiat
+00004ea0: 6564 0a2d 2d70 726f 7464 6174 696e 672c  ed.--protdating,
+00004eb0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00004ec0: 6574 6865 7220 746f 206f 6e6c 7920 696e  ether to only in
+00004ed0: 6974 6961 7465 2074 6865 2070 726f 7465  itiate the prote
+00004ee0: 696e 2d62 6173 6564 2064 6174 696e 6720  in-based dating 
+00004ef0: 616e 616c 7973 6973 2c20 6966 2074 6865  analysis, if the
+00004f00: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+00004f10: 6865 2061 6e61 6c79 7369 7320 776f 756c  he analysis woul
+00004f20: 6420 6265 2069 6e69 7469 6174 6564 0a60  d be initiated.`
+00004f30: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
+00004f40: 6077 6764 206b 7364 6020 6361 6e20 7265  `wgd ksd` can re
+00004f50: 616c 697a 6520 7468 6520 636f 6e73 7472  alize the constr
+00004f60: 7563 7469 6f6e 206f 6620 2a4b 2a3c 7375  uction of *K*<su
+00004f70: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
+00004f80: 7374 7269 6275 7469 6f6e 2061 6e64 2072  stribution and r
+00004f90: 6174 6520 636f 7272 6563 7469 6f6e 2e0a  ate correction..
+00004fa0: 6060 600a 7767 6420 6b73 6420 6661 6d69  ```.wgd ksd fami
+00004fb0: 6c69 6573 2073 6571 7565 6e63 6573 2028  lies sequences (
+00004fc0: 6f70 7469 6f6e 290a 2d2d 2d2d 2d2d 2d2d  option).--------
+00004fd0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004fe0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00004ff0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00005000: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00005010: 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c 202d 2d6f  --------.-o, --o
+00005020: 7574 6469 722c 2074 6865 206f 7574 7075  utdir, the outpu
+00005030: 7420 6469 7265 6374 6f72 792c 2064 6566  t directory, def
+00005040: 6175 6c74 2077 6764 5f6b 7364 0a2d 742c  ault wgd_ksd.-t,
+00005050: 202d 2d74 6d70 6469 722c 2074 6865 2074   --tmpdir, the t
+00005060: 656d 706f 7261 7279 2077 6f72 6b69 6e67  emporary working
+00005070: 2064 6972 6563 746f 7279 2c20 6465 6661   directory, defa
+00005080: 756c 7420 4e6f 6e65 2c20 6966 204e 6f6e  ult None, if Non
+00005090: 6520 7761 7320 6769 7665 6e2c 2074 6865  e was given, the
+000050a0: 2074 6d70 6469 7220 7769 6c6c 2062 6520   tmpdir will be 
+000050b0: 6173 7369 676e 6564 2072 616e 646f 6d20  assigned random 
+000050c0: 6e61 6d65 7320 696e 2063 7572 7265 6e74  names in current
+000050d0: 2064 6972 6563 746f 7279 2061 6e64 2061   directory and a
+000050e0: 7574 6f6d 6174 656c 7920 7265 6d6f 7665  utomately remove
+000050f0: 6420 6174 2074 6865 2063 6f6d 706c 6574  d at the complet
+00005100: 696f 6e20 6f66 2070 726f 6772 616d 2c20  ion of program, 
+00005110: 656c 7365 2074 6865 2074 6d70 6469 7220  else the tmpdir 
+00005120: 776f 756c 6420 6265 206b 6570 740a 2d6e  would be kept.-n
+00005130: 2c20 2d2d 6e74 6872 6561 6473 2c20 7468  , --nthreads, th
+00005140: 6520 6e75 6d62 6572 206f 6620 7468 7265  e number of thre
+00005150: 6164 7320 746f 2075 7365 2c20 6465 6661  ads to use, defa
+00005160: 756c 7420 340a 2d2d 746f 5f73 746f 702c  ult 4.--to_stop,
+00005170: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00005180: 6574 6865 7220 746f 2074 7261 6e73 6c61  ether to transla
+00005190: 7465 2074 6872 6f75 6768 2053 544f 5020  te through STOP 
+000051a0: 636f 646f 6e73 2c20 6966 2074 6865 2066  codons, if the f
+000051b0: 6c61 6720 7761 7320 7365 742c 2074 7261  lag was set, tra
+000051c0: 6e73 6c61 7469 6f6e 2077 696c 6c20 6265  nslation will be
+000051d0: 2074 6572 6d69 6e61 7465 6420 6174 2074   terminated at t
+000051e0: 6865 2066 6972 7374 2069 6e2d 6672 616d  he first in-fram
+000051f0: 6520 7374 6f70 2063 6f64 6f6e 2c20 656c  e stop codon, el
+00005200: 7365 2061 2066 756c 6c20 7472 616e 736c  se a full transl
+00005210: 6174 696f 6e20 636f 6e74 696e 7569 6e67  ation continuing
+00005220: 206f 6e20 7061 7374 2061 6e79 2073 746f   on past any sto
+00005230: 7020 636f 646f 6e73 2077 6f75 6c64 2062  p codons would b
+00005240: 6520 696e 6974 6961 7465 640a 2d2d 6364  e initiated.--cd
+00005250: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
+00005260: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
+00005270: 7472 616e 736c 6174 6520 7468 6520 636f  translate the co
+00005280: 6d70 6c65 7465 2043 4453 2074 6861 7420  mplete CDS that 
+00005290: 7374 6172 7473 2077 6974 6820 6120 7661  starts with a va
+000052a0: 6c69 6420 7374 6172 7420 636f 646f 6e20  lid start codon 
+000052b0: 616e 6420 6f6e 6c79 2063 6f6e 7461 696e  and only contain
+000052c0: 7320 6120 7369 6e67 6c65 2069 6e2d 6672  s a single in-fr
+000052d0: 616d 6520 7374 6f70 2063 6f64 6f6e 2061  ame stop codon a
+000052e0: 7420 7468 6520 656e 6420 616e 6420 6d75  t the end and mu
+000052f0: 7374 2062 6520 6469 7669 6461 626c 6520  st be dividable 
+00005300: 6279 2074 6872 6565 2c20 6966 2074 6865  by three, if the
+00005310: 2066 6c61 6720 7761 7320 7365 742c 206f   flag was set, o
+00005320: 6e6c 7920 7468 6520 636f 6d70 6c65 7465  nly the complete
+00005330: 2043 4453 2077 6f75 6c64 2062 6520 7472   CDS would be tr
+00005340: 616e 736c 6174 6564 0a2d 2d70 6169 7277  anslated.--pairw
+00005350: 6973 652c 2066 6c61 6720 6f70 7469 6f6e  ise, flag option
+00005360: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
+00005370: 7469 6174 6520 7061 6972 7769 7365 204b  tiate pairwise K
+00005380: 7320 6573 7469 6d61 7469 6f6e 2c20 6966  s estimation, if
+00005390: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+000053a0: 742c 2070 6169 7277 6973 6520 4b73 2076  t, pairwise Ks v
+000053b0: 616c 7565 7320 776f 756c 6420 6265 2065  alues would be e
+000053c0: 7374 696d 6174 6564 0a2d 2d73 7472 6970  stimated.--strip
+000053d0: 5f67 6170 732c 2066 6c61 6720 6f70 7469  _gaps, flag opti
+000053e0: 6f6e 2c20 7768 6574 6865 7220 746f 2064  on, whether to d
+000053f0: 726f 7020 616c 6c20 6761 7073 2069 6e20  rop all gaps in 
+00005400: 6d75 6c74 6970 6c65 2073 6571 7565 6e63  multiple sequenc
+00005410: 6520 616c 6967 6e6d 656e 742c 2069 6620  e alignment, if 
+00005420: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00005430: 2c20 616c 6c20 6761 7073 2077 6f75 6c64  , all gaps would
+00005440: 2062 6520 6472 6f70 7065 640a 2d61 2c20   be dropped.-a, 
+00005450: 2d2d 616c 6967 6e65 722c 2077 6869 6368  --aligner, which
+00005460: 2061 6c69 676e 6d65 6e74 2070 726f 6772   alignment progr
+00005470: 616d 2074 6f20 7573 652c 2064 6566 6175  am to use, defau
+00005480: 6c74 206d 6166 6674 200a 2d74 7265 652c  lt mafft .-tree,
+00005490: 202d 2d74 7265 655f 6d65 7468 6f64 2c20   --tree_method, 
+000054a0: 7768 6963 6820 6765 6e65 2074 7265 6520  which gene tree 
+000054b0: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
+000054c0: 6d20 746f 2069 6e76 6f6b 652c 2064 6566  m to invoke, def
+000054d0: 6175 6c74 2066 6173 7474 7265 650a 2d2d  ault fasttree.--
+000054e0: 7472 6565 5f6f 7074 696f 6e73 2c20 6f70  tree_options, op
+000054f0: 7469 6f6e 7320 696e 2074 7265 6520 696e  tions in tree in
+00005500: 6665 7265 6e63 6520 6173 2061 2063 6f6d  ference as a com
+00005510: 6d61 2073 6570 6172 6174 6564 2073 7472  ma separated str
+00005520: 696e 672c 2064 6566 6175 6c74 204e 6f6e  ing, default Non
+00005530: 650a 2d2d 6e6f 6465 5f61 7665 7261 6765  e.--node_average
+00005540: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00005550: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
+00005560: 7465 206e 6f64 652d 6176 6572 6167 6520  te node-average 
+00005570: 7761 7920 6f66 2064 652d 7265 6475 6e64  way of de-redund
+00005580: 616e 6379 2069 6e73 7465 6164 206f 6620  ancy instead of 
+00005590: 6e6f 6465 2d77 6569 6768 7465 642c 2069  node-weighted, i
+000055a0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+000055b0: 6574 2c20 7468 6520 6e6f 6465 2d61 7665  et, the node-ave
+000055c0: 7261 6769 6e67 2064 652d 7265 6475 6e64  raging de-redund
+000055d0: 616e 6379 2077 6f75 6c64 2062 6520 696e  ancy would be in
+000055e0: 6974 6961 7465 640a 2d73 722c 202d 2d73  itiated.-sr, --s
+000055f0: 7061 6972 2c20 7468 6520 7370 6563 6965  pair, the specie
+00005600: 7320 7061 6972 2074 6f20 6265 2070 6c6f  s pair to be plo
+00005610: 7474 6564 2c20 6465 6661 756c 7420 4e6f  tted, default No
+00005620: 6e65 2c20 7468 6973 206f 7074 696f 6e20  ne, this option 
+00005630: 6361 6e20 6265 2070 726f 7669 6465 6420  can be provided 
+00005640: 6d75 6c74 6970 6c65 2074 696d 6573 0a2d  multiple times.-
+00005650: 7370 2c20 2d2d 7370 6563 6965 7374 7265  sp, --speciestre
+00005660: 652c 2074 6865 2073 7065 6369 6573 2074  e, the species t
+00005670: 7265 6520 746f 2070 6572 666f 726d 2072  ree to perform r
+00005680: 6174 6520 636f 7272 6563 7469 6f6e 2c20  ate correction, 
+00005690: 6465 6661 756c 7420 4e6f 6e65 2c20 6966  default None, if
+000056a0: 204e 6f6e 6520 7761 7320 6769 7665 6e2c   None was given,
+000056b0: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
+000056c0: 7469 6f6e 2061 6e61 6c79 7369 7320 776f  tion analysis wo
+000056d0: 756c 6420 6265 2063 616c 6c65 6420 6f66  uld be called of
+000056e0: 660a 2d72 772c 202d 2d72 6577 6569 6768  f.-rw, --reweigh
+000056f0: 742c 2066 6c61 6720 6f70 7469 6f6e 2c20  t, flag option, 
+00005700: 7768 6574 6865 7220 746f 2072 6563 616c  whether to recal
+00005710: 6375 6c61 7465 2074 6865 2077 6569 6768  culate the weigh
+00005720: 7420 7065 7220 7370 6563 6965 7320 7061  t per species pa
+00005730: 6972 2c20 6966 2074 6865 2066 6c61 6720  ir, if the flag 
+00005740: 7761 7320 7365 742c 2074 6865 2077 6569  was set, the wei
+00005750: 6768 7420 776f 756c 6420 6265 2072 6563  ght would be rec
+00005760: 616c 6375 6c61 7465 640a 2d6f 722c 202d  alculated.-or, -
+00005770: 2d6f 6e6c 7972 6f6f 746f 7574 2c20 666c  -onlyrootout, fl
+00005780: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+00005790: 6572 2074 6f20 6f6e 6c79 2063 6f6e 6475  er to only condu
+000057a0: 6374 2072 6174 6520 636f 7272 6563 7469  ct rate correcti
+000057b0: 6f6e 2075 7369 6e67 2074 6865 206f 7574  on using the out
+000057c0: 6772 6f75 7020 6174 2072 6f6f 7420 6173  group at root as
+000057d0: 206f 7574 6772 6f75 702c 2069 6620 7468   outgroup, if th
+000057e0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+000057f0: 6f6e 6c79 2074 6865 206f 7574 6772 6f75  only the outgrou
+00005800: 7020 6174 2072 6f6f 7420 776f 756c 6420  p at root would 
+00005810: 6265 2075 7365 6420 6173 206f 7574 6772  be used as outgr
+00005820: 6f75 700a 2d65 706b 2c20 2d2d 6578 7472  oup.-epk, --extr
+00005830: 6170 6172 616e 6f6d 656b 732c 2065 7874  aparanomeks, ext
+00005840: 7261 2070 6172 616e 6f6d 6520 4b73 2064  ra paranome Ks d
+00005850: 6174 6120 746f 2070 6c6f 7420 696e 2074  ata to plot in t
+00005860: 6865 206d 6978 6564 204b 7320 6469 7374  he mixed Ks dist
+00005870: 7269 6275 7469 6f6e 2c20 6465 6661 756c  ribution, defaul
+00005880: 7420 4e6f 6e65 0a2d 6170 2c20 2d2d 616e  t None.-ap, --an
+00005890: 6368 6f72 706f 696e 7473 2c20 616e 6368  chorpoints, anch
+000058a0: 6f72 706f 696e 7473 2e74 7874 2066 696c  orpoints.txt fil
+000058b0: 6520 746f 2070 6c6f 7420 616e 6368 6f72  e to plot anchor
+000058c0: 204b 7320 696e 2074 6865 206d 6978 6564   Ks in the mixed
+000058d0: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+000058e0: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+000058f0: 706b 2c20 2d2d 706c 6f74 6b64 652c 2066  pk, --plotkde, f
+00005900: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00005910: 6865 7220 746f 2070 6c6f 7420 6b64 6520  her to plot kde 
+00005920: 6375 7276 6520 6f66 206f 7274 686f 6c6f  curve of ortholo
+00005930: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
+00005940: 7469 6f6e 206f 7665 7220 6869 7374 6f67  tion over histog
+00005950: 7261 6d20 696e 2074 6865 206d 6978 6564  ram in the mixed
+00005960: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+00005970: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00005980: 7320 7365 742c 2074 6865 206b 6465 2063  s set, the kde c
+00005990: 7572 7665 2077 6f75 6c64 2062 6520 706c  urve would be pl
+000059a0: 6f74 7465 640a 2d70 6167 2c20 2d2d 706c  otted.-pag, --pl
+000059b0: 6f74 6170 676d 6d2c 2066 6c61 6720 6f70  otapgmm, flag op
+000059c0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+000059d0: 2070 6572 666f 726d 2061 6e64 2070 6c6f   perform and plo
+000059e0: 7420 6d69 7874 7572 6520 6d6f 6465 6c69  t mixture modeli
+000059f0: 6e67 206f 6620 616e 6368 6f72 204b 7320  ng of anchor Ks 
+00005a00: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
+00005a10: 6469 7374 7269 6275 7469 6f6e 2c20 6966  distribution, if
+00005a20: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00005a30: 742c 2074 6865 206d 6978 7475 7265 206d  t, the mixture m
+00005a40: 6f64 656c 696e 6720 6f66 2061 6e63 686f  odeling of ancho
+00005a50: 7220 4b73 2077 6f75 6c64 2062 6520 706c  r Ks would be pl
+00005a60: 6f74 7465 640a 2d70 656d 2c20 2d2d 706c  otted.-pem, --pl
+00005a70: 6f74 656c 6d6d 2c20 666c 6167 206f 7074  otelmm, flag opt
+00005a80: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+00005a90: 7065 7266 6f72 6d20 616e 6420 706c 6f74  perform and plot
+00005aa0: 2065 6c6d 6d20 6d69 7874 7572 6520 6d6f   elmm mixture mo
+00005ab0: 6465 6c69 6e67 206f 6620 7061 7261 6e6f  deling of parano
+00005ac0: 6d65 204b 7320 696e 2074 6865 206d 6978  me Ks in the mix
+00005ad0: 6564 204b 7320 6469 7374 7269 6275 7469  ed Ks distributi
+00005ae0: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
+00005af0: 7761 7320 7365 742c 2074 6865 2065 6c6d  was set, the elm
+00005b00: 6d20 6d69 7874 7572 6520 6d6f 6465 6c69  m mixture modeli
+00005b10: 6e67 206f 6620 7061 7261 6e6f 6d65 204b  ng of paranome K
+00005b20: 7320 776f 756c 6420 6265 2070 6c6f 7474  s would be plott
+00005b30: 6564 0a2d 632c 202d 2d63 6f6d 706f 6e65  ed.-c, --compone
+00005b40: 6e74 732c 2074 6865 2072 616e 6765 206f  nts, the range o
+00005b50: 6620 7468 6520 6e75 6d62 6572 206f 6620  f the number of 
+00005b60: 636f 6d70 6f6e 656e 7473 2074 6f20 6669  components to fi
+00005b70: 7420 696e 2061 6e63 686f 7220 4b73 206d  t in anchor Ks m
+00005b80: 6978 7475 7265 206d 6f64 656c 696e 672c  ixture modeling,
+00005b90: 2064 6566 6175 6c74 2028 312c 3429 0a2d   default (1,4).-
+00005ba0: 786c 2c20 2d2d 786c 696d 2c20 7468 6520  xl, --xlim, the 
+00005bb0: 7820 6178 6973 206c 696d 6974 206f 6620  x axis limit of 
+00005bc0: 4b73 2064 6973 7472 6962 7574 696f 6e0a  Ks distribution.
+00005bd0: 2d79 6c2c 202d 2d79 6c69 6d2c 2074 6865  -yl, --ylim, the
+00005be0: 2079 2061 7869 7320 6c69 6d69 7420 6f66   y axis limit of
+00005bf0: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+00005c00: 0a2d 6164 6f2c 202d 2d61 646a 7573 746f  .-ado, --adjusto
+00005c10: 7274 686f 2c20 666c 6167 206f 7074 696f  rtho, flag optio
+00005c20: 6e2c 2077 6865 7468 6572 2074 6f20 6164  n, whether to ad
+00005c30: 6a75 7374 2074 6865 2068 6973 746f 6772  just the histogr
+00005c40: 616d 2068 6569 6768 7420 6f66 206f 7274  am height of ort
+00005c50: 686f 6c6f 676f 7573 204b 7320 6173 2074  hologous Ks as t
+00005c60: 6f20 6d61 7463 6820 7468 6520 6865 6967  o match the heig
+00005c70: 6874 206f 6620 7061 7261 6c6f 676f 7573  ht of paralogous
+00005c80: 204b 732c 2069 6620 7468 6520 666c 6167   Ks, if the flag
+00005c90: 2077 6173 2073 6574 2c20 7468 6520 6164   was set, the ad
+00005ca0: 6a75 7374 6d65 6e74 2077 6f75 6c64 2062  justment would b
+00005cb0: 6520 636f 6e64 7563 7465 640a 2d61 6466  e conducted.-adf
+00005cc0: 2c20 2d2d 6164 6a75 7374 6661 6374 6f72  , --adjustfactor
+00005cd0: 2c20 7468 6520 6164 6a75 7374 6d65 6e74  , the adjustment
+00005ce0: 2066 6163 746f 7220 6f66 206f 7274 686f   factor of ortho
+00005cf0: 6c6f 676f 7573 204b 732c 2064 6566 6175  logous Ks, defau
+00005d00: 6c74 2030 2e35 0a2d 6f61 2c20 2d2d 6f6b  lt 0.5.-oa, --ok
+00005d10: 616c 7068 612c 2074 6865 206f 7061 6369  alpha, the opaci
+00005d20: 7479 206f 6620 6f72 7468 6f6c 6f67 6f75  ty of orthologou
+00005d30: 7320 4b73 2064 6973 7472 6962 7574 696f  s Ks distributio
+00005d40: 6e20 696e 206d 6978 6564 2070 6c6f 742c  n in mixed plot,
+00005d50: 2064 6566 6175 6c74 2030 2e35 0a2d 6661   default 0.5.-fa
+00005d60: 2c20 2d2d 666f 6375 7332 616c 6c2c 2073  , --focus2all, s
+00005d70: 6574 2066 6f63 616c 2073 7065 6369 6573  et focal species
+00005d80: 2061 6e64 206c 6574 2073 7065 6369 6573   and let species
+00005d90: 2070 6169 7220 746f 2062 6520 6265 7477   pair to be betw
+00005da0: 6565 6e20 666f 6361 6c20 616e 6420 616c  een focal and al
+00005db0: 6c20 7468 6520 7265 6d61 696e 696e 6720  l the remaining 
+00005dc0: 7370 6563 6965 732c 2064 6566 6175 6c74  species, default
+00005dd0: 204e 6f6e 650a 2d6b 732c 202d 2d6b 7374   None.-ks, --kst
+00005de0: 7265 652c 2066 6c61 6720 6f70 7469 6f6e  ree, flag option
+00005df0: 2c20 7768 6574 6865 7220 746f 2069 6e66  , whether to inf
+00005e00: 6572 204b 7320 7472 6565 2c20 6966 2074  er Ks tree, if t
+00005e10: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00005e20: 2074 6865 204b 7320 7472 6565 2069 6e66   the Ks tree inf
+00005e30: 6572 656e 6365 2061 6e61 6c79 7369 7320  erence analysis 
+00005e40: 776f 756c 6420 6265 2069 6e69 7469 6174  would be initiat
+00005e50: 6564 0a2d 6f63 6b2c 202d 2d6f 6e6c 7963  ed.-ock, --onlyc
+00005e60: 6f6e 6361 746b 7374 7265 652c 2066 6c61  oncatkstree, fla
+00005e70: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00005e80: 7220 746f 206f 6e6c 7920 696e 6665 7220  r to only infer 
+00005e90: 4b73 2074 7265 6520 756e 6465 7220 636f  Ks tree under co
+00005ea0: 6e63 6174 656e 6174 6564 2061 6c69 676e  ncatenated align
+00005eb0: 6d65 6e74 2c20 6966 2074 6865 2066 6c61  ment, if the fla
+00005ec0: 6720 7761 7320 7365 742c 206f 6e6c 7920  g was set, only 
+00005ed0: 7468 6520 4b73 2074 7265 6520 756e 6465  the Ks tree unde
+00005ee0: 7220 636f 6e63 6174 656e 6174 6564 2061  r concatenated a
+00005ef0: 6c69 676e 6d65 6e74 2077 6f75 6c64 2062  lignment would b
+00005f00: 6520 6361 6c63 756c 6174 6564 0a2d 6373  e calculated.-cs
+00005f10: 2c20 2d2d 636c 6173 7369 632c 2066 6c61  , --classic, fla
+00005f20: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00005f30: 7220 746f 2064 7261 7720 6d69 7865 6420  r to draw mixed 
+00005f40: 4b73 2070 6c6f 7420 696e 2061 2063 6c61  Ks plot in a cla
+00005f50: 7373 6963 206d 616e 6e65 7220 7768 6572  ssic manner wher
+00005f60: 6520 7468 6520 6675 6c6c 206f 7274 686f  e the full ortho
+00005f70: 6c6f 676f 7573 204b 7320 6469 7374 7269  logous Ks distri
+00005f80: 6275 7469 6f6e 2069 7320 6472 6177 6564  bution is drawed
+00005f90: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00005fa0: 7320 7365 742c 2074 6865 2063 6c61 7373  s set, the class
+00005fb0: 6963 206d 6978 6564 204b 7320 706c 6f74  ic mixed Ks plot
+00005fc0: 2077 6f75 6c64 2062 6520 6472 6177 6e0a   would be drawn.
+00005fd0: 2d74 612c 202d 2d74 6f70 6172 726f 772c  -ta, --toparrow,
+00005fe0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00005ff0: 6574 6865 7220 746f 2061 646a 7573 7420  ether to adjust 
+00006000: 7468 6520 6172 726f 7720 746f 2062 6520  the arrow to be 
+00006010: 6174 2074 6865 2074 6f70 206f 6620 7468  at the top of th
+00006020: 6520 706c 6f74 2c20 696e 7374 6561 6420  e plot, instead 
+00006030: 6f66 2062 6569 6e67 2063 6f6f 7264 696e  of being coordin
+00006040: 6174 6564 2061 7320 7468 6520 4b44 4520  ated as the KDE 
+00006050: 6f66 2074 6865 206f 7274 686f 6c6f 676f  of the orthologo
+00006060: 7573 204b 7320 6469 7374 7269 6275 7469  us Ks distributi
+00006070: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
+00006080: 7761 7320 7365 742c 2074 6865 2061 7272  was set, the arr
+00006090: 6f77 2077 6f75 6c64 2062 6520 7365 7420  ow would be set 
+000060a0: 6174 2074 6865 2074 6f70 0a2d 6273 2c20  at the top.-bs, 
+000060b0: 2d2d 626f 6f74 7374 7261 702c 2074 6865  --bootstrap, the
+000060c0: 206e 756d 6265 7220 6f66 2062 6f6f 7473   number of boots
+000060d0: 7472 6170 2072 6570 6c69 6361 7465 7320  trap replicates 
+000060e0: 6f66 206f 7274 686f 6c6f 6720 4b73 2064  of ortholog Ks d
+000060f0: 6973 7472 6962 7574 696f 6e20 696e 206d  istribution in m
+00006100: 6978 6564 2070 6c6f 740a 6060 600a 0a54  ixed plot.```..T
+00006110: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
+00006120: 6d69 7860 2063 616e 2072 6561 6c69 7a65  mix` can realize
+00006130: 2074 6865 206d 6978 7475 7265 206d 6f64   the mixture mod
+00006140: 656c 2063 6c75 7374 6572 696e 6720 616e  el clustering an
+00006150: 616c 7973 6973 206f 6620 2a4b 2a3c 7375  alysis of *K*<su
+00006160: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
+00006170: 7374 7269 6275 7469 6f6e 2e0a 6060 600a  stribution..```.
+00006180: 7767 6420 6d69 7820 6b73 5f64 6174 6166  wgd mix ks_dataf
+00006190: 696c 6520 286f 7074 696f 6e29 0a2d 2d2d  ile (option).---
+000061a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000061b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000061c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000061d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000061e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d0a 2d66  -------------.-f
+000061f0: 2c20 2d2d 6669 6c74 6572 732c 2074 6865  , --filters, the
+00006200: 2063 7574 6f66 6620 616c 6967 6e6d 656e   cutoff alignmen
+00006210: 7420 6c65 6e67 7468 2c20 6465 6661 756c  t length, defaul
+00006220: 7420 3330 300a 2d72 2c20 2d2d 6b73 5f72  t 300.-r, --ks_r
+00006230: 616e 6765 2c20 7468 6520 4b73 2072 616e  ange, the Ks ran
+00006240: 6765 2074 6f20 6265 2063 6f6e 7369 6465  ge to be conside
+00006250: 7265 642c 2064 6566 6175 6c74 2028 302c  red, default (0,
+00006260: 2035 290a 2d62 2c20 2d2d 6269 6e73 2c20   5).-b, --bins, 
+00006270: 7468 6520 6e75 6d62 6572 206f 6620 6269  the number of bi
+00006280: 6e73 2069 6e20 4b73 2064 6973 7472 6962  ns in Ks distrib
+00006290: 7574 696f 6e2c 2064 6566 6175 6c74 2035  ution, default 5
+000062a0: 300a 2d6f 2c20 2d2d 6f75 7464 6972 2c20  0.-o, --outdir, 
+000062b0: 7468 6520 6f75 7470 7574 2064 6972 6563  the output direc
+000062c0: 746f 7279 2c20 6465 6661 756c 7420 7767  tory, default wg
+000062d0: 645f 6d69 780a 2d2d 6d65 7468 6f64 2c20  d_mix.--method, 
+000062e0: 7768 6963 6820 6d69 7874 7572 6520 6d6f  which mixture mo
+000062f0: 6465 6c20 746f 2075 7365 2c20 6465 6661  del to use, defa
+00006300: 756c 7420 676d 6d0a 2d6e 2c20 2d2d 636f  ult gmm.-n, --co
+00006310: 6d70 6f6e 656e 7473 2c20 7468 6520 7261  mponents, the ra
+00006320: 6e67 6520 6f66 2074 6865 206e 756d 6265  nge of the numbe
+00006330: 7220 6f66 2063 6f6d 706f 6e65 6e74 7320  r of components 
+00006340: 746f 2066 6974 2c20 6465 6661 756c 7420  to fit, default 
+00006350: 2831 2c20 3429 0a2d 672c 202d 2d67 616d  (1, 4).-g, --gam
+00006360: 6d61 2c20 7468 6520 6761 6d6d 6120 7061  ma, the gamma pa
+00006370: 7261 6d65 7465 7220 666f 7220 6267 6d6d  rameter for bgmm
+00006380: 206d 6f64 656c 732c 2064 6566 6175 6c74   models, default
+00006390: 2030 2e30 3031 0a2d 6e69 2c20 2d2d 6e5f   0.001.-ni, --n_
+000063a0: 696e 6974 2c20 7468 6520 6e75 6d62 6572  init, the number
+000063b0: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
+000063c0: 6961 6c69 7a61 7469 6f6e 732c 2064 6566  ializations, def
+000063d0: 6175 6c74 2032 3030 0a2d 6d69 2c20 2d2d  ault 200.-mi, --
+000063e0: 6d61 785f 6974 6572 2c20 7468 6520 6d61  max_iter, the ma
+000063f0: 7869 6d75 6d20 6e75 6d62 6572 206f 6620  ximum number of 
+00006400: 6974 6572 6174 696f 6e73 2c20 6465 6661  iterations, defa
+00006410: 756c 7420 3230 300a 6060 600a 0a54 6865  ult 200.```..The
+00006420: 2070 726f 6772 616d 2060 7767 6420 7065   program `wgd pe
+00006430: 616b 6020 6361 6e20 7265 616c 697a 6520  ak` can realize 
+00006440: 7468 6520 7365 6172 6368 206f 6620 6372  the search of cr
+00006450: 6564 6961 626c 6520 2a4b 2a3c 7375 623e  ediable *K*<sub>
+00006460: 533c 2f73 7562 3e20 7261 6e67 6520 7573  S</sub> range us
+00006470: 6564 2069 6e20 5747 4420 6461 7469 6e67  ed in WGD dating
+00006480: 2e0a 6060 600a 7767 6420 7065 616b 206b  ..```.wgd peak k
+00006490: 735f 6461 7461 6669 6c65 2028 6f70 7469  s_datafile (opti
+000064a0: 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  on).------------
+000064b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000064c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000064d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000064e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000064f0: 2d2d 2d2d 0a2d 6170 2c20 2d2d 616e 6368  ----.-ap, --anch
+00006500: 6f72 706f 696e 7473 2c20 7468 6520 616e  orpoints, the an
+00006510: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
+00006520: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
+00006530: 6e65 0a2d 736d 2c20 2d2d 7365 676d 656e  ne.-sm, --segmen
+00006540: 7473 2c20 7468 6520 7365 676d 656e 7473  ts, the segments
+00006550: 2064 6174 6166 696c 652c 2064 6566 6175   datafile, defau
+00006560: 6c74 204e 6f6e 650a 2d6c 652c 202d 2d6c  lt None.-le, --l
+00006570: 6973 7465 6c65 6d65 6e74 732c 2074 6865  istelements, the
+00006580: 206c 6973 7473 6567 6d65 6e74 7320 6461   listsegments da
+00006590: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
+000065a0: 4e6f 6e65 200a 2d6d 702c 202d 2d6d 756c  None .-mp, --mul
+000065b0: 7469 706c 6963 6f6e 7061 6972 732c 2074  tipliconpairs, t
+000065c0: 6865 206d 756c 7469 706c 6963 6f6e 7061  he multipliconpa
+000065d0: 6972 7320 6461 7461 6669 6c65 2c20 6465  irs datafile, de
+000065e0: 6661 756c 7420 4e6f 6e65 0a2d 6f2c 202d  fault None.-o, -
+000065f0: 2d6f 7574 6469 722c 2074 6865 206f 7574  -outdir, the out
+00006600: 7075 7420 6469 7265 6374 6f72 792c 2064  put directory, d
+00006610: 6566 6175 6c74 2077 6764 5f70 6561 6b0a  efault wgd_peak.
+00006620: 2d61 662c 202d 2d61 6c69 676e 6669 6c74  -af, --alignfilt
+00006630: 6572 2c20 6375 746f 6666 2066 6f72 2061  er, cutoff for a
+00006640: 6c69 676e 6d65 6e74 2069 6465 6e74 6974  lignment identit
+00006650: 792c 206c 656e 6774 6820 616e 6420 636f  y, length and co
+00006660: 7665 7261 6765 2c20 6465 6661 756c 7420  verage, default 
+00006670: 302e 302c 2030 2c20 302e 300a 2d72 2c20  0.0, 0, 0.0.-r, 
+00006680: 2d2d 6b73 7261 6e67 652c 2072 616e 6765  --ksrange, range
+00006690: 206f 6620 4b73 2074 6f20 6265 2061 6e61   of Ks to be ana
+000066a0: 6c79 7a65 642c 2064 6566 6175 6c74 2028  lyzed, default (
+000066b0: 302c 2035 290a 2d62 772c 202d 2d62 696e  0, 5).-bw, --bin
+000066c0: 5f77 6964 7468 2c20 6261 6e64 7769 6474  _width, bandwidt
+000066d0: 6820 6f66 204b 7320 6469 7374 7269 6275  h of Ks distribu
+000066e0: 7469 6f6e 2c20 6465 6661 756c 7420 302e  tion, default 0.
+000066f0: 310a 2d69 632c 202d 2d77 6569 6768 7473  1.-ic, --weights
+00006700: 5f6f 7574 6c69 6572 735f 696e 636c 7564  _outliers_includ
+00006710: 6564 2c20 666c 6167 206f 7074 696f 6e2c  ed, flag option,
+00006720: 2077 6865 7468 6572 2074 6f20 696e 636c   whether to incl
+00006730: 7564 6520 4b73 206f 7574 6c69 6572 732c  ude Ks outliers,
+00006740: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00006750: 2073 6574 2c20 4b73 206f 7574 6c69 6572   set, Ks outlier
+00006760: 7320 776f 756c 6420 6265 2069 6e63 6c75  s would be inclu
+00006770: 6465 6420 696e 2074 6865 2061 6e61 6c79  ded in the analy
+00006780: 7369 730a 2d6d 2c20 2d2d 6d65 7468 6f64  sis.-m, --method
+00006790: 2c20 7768 6963 6820 6d69 7874 7572 6520  , which mixture 
+000067a0: 6d6f 6465 6c20 746f 2075 7365 2c20 6465  model to use, de
+000067b0: 6661 756c 7420 676d 6d0a 2d2d 7365 6564  fault gmm.--seed
+000067c0: 2c20 7261 6e64 6f6d 2073 6565 6420 6769  , random seed gi
+000067d0: 7665 6e20 746f 2069 6e69 7469 616c 697a  ven to initializ
+000067e0: 6174 696f 6e2c 2064 6566 6175 6c74 2032  ation, default 2
+000067f0: 3335 3238 3930 0a2d 6569 2c20 2d2d 656d  352890.-ei, --em
+00006800: 5f69 7465 722c 2074 6865 206e 756d 6265  _iter, the numbe
+00006810: 7220 6f66 2045 4d20 6974 6572 6174 696f  r of EM iteratio
+00006820: 6e73 2074 6f20 7065 7266 6f72 6d2c 2064  ns to perform, d
+00006830: 6566 6175 6c74 2032 3030 0a2d 6e69 2c20  efault 200.-ni, 
+00006840: 2d2d 6e5f 696e 6974 2c20 7468 6520 6e75  --n_init, the nu
+00006850: 6d62 6572 206f 6620 6b2d 6d65 616e 7320  mber of k-means 
+00006860: 696e 6974 6961 6c69 7a61 7469 6f6e 732c  initializations,
+00006870: 2064 6566 6175 6c74 2032 3030 0a2d 6e2c   default 200.-n,
+00006880: 202d 2d63 6f6d 706f 6e65 6e74 732c 2074   --components, t
+00006890: 6865 2072 616e 6765 206f 6620 7468 6520  he range of the 
+000068a0: 6e75 6d62 6572 206f 6620 636f 6d70 6f6e  number of compon
+000068b0: 656e 7473 2074 6f20 6669 742c 2064 6566  ents to fit, def
+000068c0: 6175 6c74 2028 312c 2034 290a 2d67 2c20  ault (1, 4).-g, 
+000068d0: 2d2d 6761 6d6d 612c 2074 6865 2067 616d  --gamma, the gam
+000068e0: 6d61 2070 6172 616d 6574 6572 2066 6f72  ma parameter for
+000068f0: 2062 676d 6d20 6d6f 6465 6c73 2c20 6465   bgmm models, de
+00006900: 6661 756c 7420 3165 2d33 0a2d 2d62 6f6f  fault 1e-3.--boo
+00006910: 7473 2c20 7468 6520 6e75 6d62 6572 206f  ts, the number o
+00006920: 6620 626f 6f74 7374 7261 7020 7265 706c  f bootstrap repl
+00006930: 6963 6174 6573 206f 6620 6b64 652c 2064  icates of kde, d
+00006940: 6566 6175 6c74 2032 3030 0a2d 2d77 6569  efault 200.--wei
+00006950: 6768 7465 642c 2066 6c61 6720 6f70 7469  ghted, flag opti
+00006960: 6f6e 2c20 7768 6574 6865 7220 746f 2075  on, whether to u
+00006970: 7365 206e 6f64 652d 7765 6967 6874 6564  se node-weighted
+00006980: 206d 6574 686f 6420 6f66 2064 652d 7265   method of de-re
+00006990: 6475 6e64 616e 6379 2c20 6966 2074 6865  dundancy, if the
+000069a0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+000069b0: 6865 206e 6f64 652d 7765 6967 6874 6564  he node-weighted
+000069c0: 206d 6574 686f 6420 776f 756c 6420 6265   method would be
+000069d0: 2075 7365 640a 2d70 2c20 2d2d 706c 6f74   used.-p, --plot
+000069e0: 2c20 7468 6520 706c 6f74 7469 6e67 206d  , the plotting m
+000069f0: 6574 686f 6420 746f 2062 6520 7573 6564  ethod to be used
+00006a00: 2c20 6465 6661 756c 7420 6964 656e 7469  , default identi
+00006a10: 6361 6c0a 2d62 6d2c 202d 2d62 775f 6d65  cal.-bm, --bw_me
+00006a20: 7468 6f64 2c20 7468 6520 6261 6e64 7769  thod, the bandwi
+00006a30: 6474 6820 6d65 7468 6f64 2074 6f20 6265  dth method to be
+00006a40: 2075 7365 6420 696e 2061 6e61 6c79 7a69   used in analyzi
+00006a50: 6e67 2074 6865 2070 6561 6b20 6f66 2057  ng the peak of W
+00006a60: 4744 2064 6174 6573 2c20 6465 6661 756c  GD dates, defaul
+00006a70: 7420 7369 6c76 6572 6d61 6e0a 2d2d 6e5f  t silverman.--n_
+00006a80: 6d65 646f 6964 732c 2074 6865 206e 756d  medoids, the num
+00006a90: 6265 7220 6f66 206d 6564 6f69 6473 2074  ber of medoids t
+00006aa0: 6f20 6669 742c 2064 6566 6175 6c74 2032  o fit, default 2
+00006ab0: 0a2d 6b6d 2c20 2d2d 6b64 656d 6574 686f  .-km, --kdemetho
+00006ac0: 642c 2074 6865 206b 6465 206d 6574 686f  d, the kde metho
+00006ad0: 6420 746f 2062 6520 7573 6564 2069 6e20  d to be used in 
+00006ae0: 616e 616c 797a 696e 6720 7468 6520 7065  analyzing the pe
+00006af0: 616b 206f 6620 5747 4420 6461 7465 732c  ak of WGD dates,
+00006b00: 206b 6d65 646f 6964 7320 616e 616c 7973   kmedoids analys
+00006b10: 6973 206f 7220 7468 6520 6261 7369 6320  is or the basic 
+00006b20: 4b73 2070 6c6f 7474 696e 672c 2064 6566  Ks plotting, def
+00006b30: 6175 6c74 2073 6369 7079 0a2d 2d6e 5f63  ault scipy.--n_c
+00006b40: 6c75 7374 6572 732c 2074 6865 206e 756d  lusters, the num
+00006b50: 6265 7220 6f66 2063 6c75 7374 6572 7320  ber of clusters 
+00006b60: 746f 2070 6c6f 7420 456c 626f 7720 6c6f  to plot Elbow lo
+00006b70: 7373 2066 756e 6374 696f 6e2c 2064 6566  ss function, def
+00006b80: 6175 6c74 2035 0a2d 6764 2c20 2d2d 6775  ault 5.-gd, --gu
+00006b90: 6964 652c 2074 6865 2072 6567 696d 6520  ide, the regime 
+00006ba0: 7265 7369 6469 6e67 2061 6e63 686f 7273  residing anchors
+00006bb0: 2c20 6465 6661 756c 7420 5365 676d 656e  , default Segmen
+00006bc0: 740a 2d70 7263 742c 202d 2d70 726f 6d69  t.-prct, --promi
+00006bd0: 6e65 6e63 655f 6375 746f 6666 2c20 7468  nence_cutoff, th
+00006be0: 6520 7072 6f6d 696e 656e 6365 2063 7574  e prominence cut
+00006bf0: 6f66 6620 6f66 2061 6363 6570 7461 626c  off of acceptabl
+00006c00: 6520 7065 616b 7320 696e 2070 6561 6b20  e peaks in peak 
+00006c10: 6669 6e64 696e 6720 7374 6570 732c 2064  finding steps, d
+00006c20: 6566 6175 6c74 2030 2e31 0a2d 7268 2c20  efault 0.1.-rh, 
+00006c30: 2d2d 7265 6c5f 6865 6967 6874 2c20 7468  --rel_height, th
+00006c40: 6520 7265 6c61 7469 7665 2068 6569 6768  e relative heigh
+00006c50: 7420 6174 2077 6869 6368 2074 6865 2070  t at which the p
+00006c60: 6561 6b20 7769 6474 6820 6973 206d 6561  eak width is mea
+00006c70: 7375 7265 642c 2064 6566 6175 6c74 2030  sured, default 0
+00006c80: 2e34 0a2d 6b64 2c20 2d2d 6b73 746f 6461  .4.-kd, --kstoda
+00006c90: 7465 2c20 7468 6520 7261 6e67 6520 6f66  te, the range of
+00006ca0: 204b 7320 746f 2062 6520 6461 7465 6420   Ks to be dated 
+00006cb0: 696e 2068 6575 7269 7374 6963 2073 6561  in heuristic sea
+00006cc0: 7263 682c 2064 6566 6175 6c74 2028 302e  rch, default (0.
+00006cd0: 352c 2031 2e35 290a 2d78 6c2c 202d 2d78  5, 1.5).-xl, --x
+00006ce0: 6c69 6d2c 2074 6865 2078 2061 7869 7320  lim, the x axis 
+00006cf0: 6c69 6d69 7420 6f66 2047 4d4d 204b 7320  limit of GMM Ks 
+00006d00: 6469 7374 7269 6275 7469 6f6e 0a2d 796c  distribution.-yl
+00006d10: 2c20 2d2d 796c 696d 2c20 7468 6520 7920  , --ylim, the y 
+00006d20: 6178 6973 206c 696d 6974 206f 6620 474d  axis limit of GM
+00006d30: 4d20 4b73 2064 6973 7472 6962 7574 696f  M Ks distributio
+00006d40: 6e0a 2d2d 6d61 6e75 616c 7365 742c 2066  n.--manualset, f
+00006d50: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00006d60: 6865 7220 746f 206f 7574 7075 7420 616e  her to output an
+00006d70: 6368 6f72 2070 6169 7273 2077 6974 6820  chor pairs with 
+00006d80: 6d61 6e75 616c 6c79 2073 6574 204b 7320  manually set Ks 
+00006d90: 7261 6e67 652c 2069 6620 7468 6520 666c  range, if the fl
+00006da0: 6167 2077 6173 2073 6574 2c20 6d61 6e75  ag was set, manu
+00006db0: 616c 6c79 2073 6574 204b 7320 7261 6e67  ally set Ks rang
+00006dc0: 6520 776f 756c 6420 6265 206f 7574 7075  e would be outpu
+00006dd0: 7474 6564 0a2d 2d63 692c 2074 6865 2063  tted.--ci, the c
+00006de0: 6f6e 6669 6465 6e63 6520 6c65 7665 6c20  onfidence level 
+00006df0: 6f66 206c 6f67 2d6e 6f72 6d61 6c20 6469  of log-normal di
+00006e00: 7374 7269 6275 7469 6f6e 2074 6f20 6461  stribution to da
+00006e10: 7465 2c20 6465 6661 756c 7420 3935 0a2d  te, default 95.-
+00006e20: 2d68 6472 2c20 7468 6520 6869 6768 6573  -hdr, the highes
+00006e30: 7420 6465 6e73 6974 7920 7265 6769 6f6e  t density region
+00006e40: 2028 4844 5229 2061 7070 6c69 6564 2069   (HDR) applied i
+00006e50: 6e20 7468 6520 7365 676d 656e 742d 6775  n the segment-gu
+00006e60: 6964 6564 2061 6e63 686f 7220 7061 6972  ided anchor pair
+00006e70: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+00006e80: 2c20 6465 6661 756c 7420 3935 0a2d 2d68  , default 95.--h
+00006e90: 6575 7269 7374 6963 2c20 666c 6167 206f  euristic, flag o
+00006ea0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+00006eb0: 6f20 696e 6974 6961 7465 2068 6575 7269  o initiate heuri
+00006ec0: 7374 6963 206d 6574 686f 6420 6f66 2064  stic method of d
+00006ed0: 6566 696e 696e 6720 4349 2066 6f72 2064  efining CI for d
+00006ee0: 6174 696e 672c 2069 6620 7468 6520 666c  ating, if the fl
+00006ef0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+00006f00: 6865 7572 6973 7469 6320 6d65 7468 6f64  heuristic method
+00006f10: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+00006f20: 7465 640a 2d6b 632c 202d 2d6b 7363 7574  ted.-kc, --kscut
+00006f30: 6f66 662c 2074 6865 204b 7320 7361 7475  off, the Ks satu
+00006f40: 7261 7469 6f6e 2063 7574 6f66 6620 696e  ration cutoff in
+00006f50: 2064 6174 696e 672c 2064 6566 6175 6c74   dating, default
+00006f60: 2035 0a2d 2d6b 6565 7074 6d70 6669 672c   5.--keeptmpfig,
+00006f70: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00006f80: 6574 6865 7220 746f 206b 6565 7020 7465  ether to keep te
+00006f90: 6d70 6f72 6172 7920 6669 6775 7265 7320  mporary figures 
+00006fa0: 696e 2070 6561 6b20 6669 6e64 696e 6720  in peak finding 
+00006fb0: 7072 6f63 6573 732c 2069 6620 7468 6520  process, if the 
+00006fc0: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+00006fd0: 6f73 6520 6669 6775 7265 7320 776f 756c  ose figures woul
+00006fe0: 6420 6265 206b 6570 740a 6060 600a 0a54  d be kept.```..T
+00006ff0: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
+00007000: 7379 6e60 2063 616e 2072 6561 6c69 7a65  syn` can realize
+00007010: 2074 6865 2069 6e74 7261 2d20 616e 6420   the intra- and 
+00007020: 696e 7465 722d 7370 6563 6966 6963 2073  inter-specific s
+00007030: 796e 7465 6e79 2069 6e66 6572 656e 6365  ynteny inference
+00007040: 2e0a 6060 600a 7767 6420 7379 6e20 6661  ..```.wgd syn fa
+00007050: 6d69 6c69 6573 2067 6666 7320 286f 7074  milies gffs (opt
+00007060: 696f 6e29 0a2d 2d2d 2d2d 2d2d 2d2d 2d2d  ion).-----------
+00007070: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00007080: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00007090: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000070a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000070b0: 2d2d 2d2d 2d0a 2d6b 732c 202d 2d6b 735f  -----.-ks, --ks_
+000070c0: 6469 7374 7269 6275 7469 6f6e 2c20 6b73  distribution, ks
+000070d0: 2064 6973 7472 6962 7574 696f 6e20 6461   distribution da
+000070e0: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
+000070f0: 4e6f 6e65 0a2d 6f2c 202d 2d6f 7574 6469  None.-o, --outdi
+00007100: 722c 2074 6865 206f 7574 7075 7420 6469  r, the output di
+00007110: 7265 6374 6f72 792c 2064 6566 6175 6c74  rectory, default
+00007120: 2077 6764 5f73 796e 0a2d 662c 202d 2d66   wgd_syn.-f, --f
+00007130: 6561 7475 7265 2c20 7468 6520 6665 6174  eature, the feat
+00007140: 7572 6520 666f 7220 7061 7273 696e 6720  ure for parsing 
+00007150: 6765 6e65 2049 4473 2066 726f 6d20 4746  gene IDs from GF
+00007160: 4620 6669 6c65 732c 2064 6566 6175 6c74  F files, default
+00007170: 2067 656e 650a 2d61 2c20 2d2d 6174 7472   gene.-a, --attr
+00007180: 6962 7574 652c 2074 6865 2061 7474 7269  ibute, the attri
+00007190: 6275 7465 2066 6f72 2070 6172 7369 6e67  bute for parsing
+000071a0: 2074 6865 2067 656e 6520 4944 7320 6672   the gene IDs fr
+000071b0: 6f6d 2074 6865 2047 4646 2066 696c 6573  om the GFF files
+000071c0: 2c20 6465 6661 756c 7420 4944 0a2d 6174  , default ID.-at
+000071d0: 672c 202d 2d61 6464 6974 696f 6e61 6c67  g, --additionalg
+000071e0: 6666 696e 666f 2c20 7468 6520 6665 6174  ffinfo, the feat
+000071f0: 7572 6520 616e 6420 6174 7472 6962 7574  ure and attribut
+00007200: 6520 696e 666f 726d 6174 696f 6e20 6f66  e information of
+00007210: 2061 6464 6974 696f 6e61 6c20 6766 6633   additional gff3
+00007220: 2066 696c 6573 2069 6620 6469 6666 6572   files if differ
+00007230: 656e 7420 696e 2074 6865 2066 6f72 6d61  ent in the forma
+00007240: 7420 6f66 2028 6665 6174 7572 653b 6174  t of (feature;at
+00007250: 7472 6962 7574 6529 272c 2064 6566 6175  tribute)', defau
+00007260: 6c74 204e 6f6e 650a 2d6d 6c2c 202d 2d6d  lt None.-ml, --m
+00007270: 696e 6c65 6e2c 2074 6865 206d 696e 696d  inlen, the minim
+00007280: 756d 206c 656e 6774 6820 6f66 2061 2073  um length of a s
+00007290: 6361 6666 6f6c 6420 746f 2062 6520 696e  caffold to be in
+000072a0: 636c 7564 6564 2069 6e20 646f 7470 6c6f  cluded in dotplo
+000072b0: 742c 2064 6566 6175 6c74 202d 312c 2069  t, default -1, i
+000072c0: 6620 2d31 2077 6173 2073 6574 2c20 7468  f -1 was set, th
+000072d0: 6520 3130 2520 6f66 2074 6865 206c 6f6e  e 10% of the lon
+000072e0: 6765 7374 2073 6361 6666 6f6c 6420 776f  gest scaffold wo
+000072f0: 756c 6420 6265 2073 6574 0a2d 6d73 2c20  uld be set.-ms, 
+00007300: 2d2d 6d61 7873 697a 652c 2074 6865 206d  --maxsize, the m
+00007310: 6178 696d 756d 2066 616d 696c 7920 7369  aximum family si
+00007320: 7a65 2074 6f20 6265 2069 6e63 6c75 6465  ze to be include
+00007330: 642c 2064 6566 6175 6c74 2032 3030 0a2d  d, default 200.-
+00007340: 722c 202d 2d6b 735f 7261 6e67 652c 2074  r, --ks_range, t
+00007350: 6865 204b 7320 7261 6e67 6520 696e 2063  he Ks range in c
+00007360: 6f6c 6f72 6564 2064 6f74 706c 6f74 2c20  olored dotplot, 
+00007370: 6465 6661 756c 7420 2830 2c20 3529 0a2d  default (0, 5).-
+00007380: 2d69 6164 686f 7265 5f6f 7074 696f 6e73  -iadhore_options
+00007390: 2c20 7468 6520 7061 7261 6d65 7465 7220  , the parameter 
+000073a0: 7365 7474 696e 6720 696e 2069 6164 686f  setting in iadho
+000073b0: 7265 2c20 6465 6661 756c 7420 6173 2061  re, default as a
+000073c0: 2073 7472 696e 6720 6f66 206c 656e 6774   string of lengt
+000073d0: 6820 7a65 726f 0a2d 6d67 2c20 2d2d 6d69  h zero.-mg, --mi
+000073e0: 6e73 6567 6c65 6e2c 2074 6865 206d 696e  nseglen, the min
+000073f0: 696d 756d 206c 656e 6774 6820 6f66 2073  imum length of s
+00007400: 6567 6d65 6e74 7320 746f 2069 6e63 6c75  egments to inclu
+00007410: 6465 2069 6e20 7261 7469 6f20 6966 203c  de in ratio if <
+00007420: 3d20 312c 2064 6566 6175 6c74 2031 3030  = 1, default 100
+00007430: 3030 0a2d 6b72 2c20 2d2d 6b65 6570 7265  00.-kr, --keepre
+00007440: 6475 6e2c 2066 6c61 6720 6f70 7469 6f6e  dun, flag option
+00007450: 2c20 7768 6574 6865 7220 746f 206b 6565  , whether to kee
+00007460: 7020 7265 6475 6e64 616e 7420 6d75 6c74  p redundant mult
+00007470: 6970 6c69 636f 6e73 2c20 6966 2074 6865  iplicons, if the
+00007480: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+00007490: 6865 2072 6564 756e 6461 6e74 206d 756c  he redundant mul
+000074a0: 7469 706c 6963 6f6e 7320 776f 756c 6420  tiplicons would 
+000074b0: 6265 206b 6570 740a 2d6d 676e 2c20 2d2d  be kept.-mgn, --
+000074c0: 6d69 6e67 656e 656e 756d 2c20 7468 6520  mingenenum, the 
+000074d0: 6d69 6e69 6d75 6d20 6e75 6d62 6572 206f  minimum number o
+000074e0: 6620 6765 6e65 7320 666f 7220 6120 7365  f genes for a se
+000074f0: 676d 656e 7420 746f 2062 6520 636f 6e73  gment to be cons
+00007500: 6964 6572 6564 2c20 6465 6661 756c 7420  idered, default 
+00007510: 3330 0a2d 6473 2c20 2d2d 646f 7473 697a  30.-ds, --dotsiz
+00007520: 652c 2074 6865 2064 6f74 2073 697a 6520  e, the dot size 
+00007530: 696e 2064 6f74 2070 6c6f 742c 2064 6566  in dot plot, def
+00007540: 6175 6c74 2030 2e33 0a2d 6161 2c20 2d2d  ault 0.3.-aa, --
+00007550: 6170 616c 7068 612c 2074 6865 206f 7061  apalpha, the opa
+00007560: 6369 7479 206f 6620 616e 6368 6f72 2064  city of anchor d
+00007570: 6f74 732c 2064 6566 6175 6c74 2031 0a2d  ots, default 1.-
+00007580: 6861 2c20 2d2d 686f 616c 7068 612c 2074  ha, --hoalpha, t
+00007590: 6865 206f 7061 6369 7479 206f 6620 686f  he opacity of ho
+000075a0: 6d6f 6c6f 6720 646f 7473 2c20 6465 6661  molog dots, defa
+000075b0: 756c 7420 300a 2d73 7274 2c20 2d2d 7368  ult 0.-srt, --sh
+000075c0: 6f77 7265 616c 7469 636b 2c20 666c 6167  owrealtick, flag
+000075d0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
+000075e0: 2074 6f20 7368 6f77 2074 6865 2072 6561   to show the rea
+000075f0: 6c20 7469 636b 2069 6e20 6765 6e65 7320  l tick in genes 
+00007600: 6f72 2062 6173 6573 2c20 6966 2074 6865  or bases, if the
+00007610: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+00007620: 6865 2072 6561 6c20 7469 636b 2077 6f75  he real tick wou
+00007630: 6c64 2062 6520 7368 6f77 6564 0a2d 746c  ld be showed.-tl
+00007640: 732c 202d 2d74 6963 6b6c 6162 656c 7369  s, --ticklabelsi
+00007650: 7a65 2c20 7468 6520 6c61 6265 6c20 7369  ze, the label si
+00007660: 7a65 206f 6620 7469 636b 2c20 6465 6661  ze of tick, defa
+00007670: 756c 7420 350a 2d67 722c 202d 2d67 6973  ult 5.-gr, --gis
+00007680: 7472 622c 2066 6c61 6720 6f70 7469 6f6e  trb, flag option
+00007690: 2c20 7768 6574 6865 7220 746f 2075 7365  , whether to use
+000076a0: 2067 6973 745f 7261 696e 626f 7720 6173   gist_rainbow as
+000076b0: 2063 6f6c 6f72 206d 6170 206f 6620 646f   color map of do
+000076c0: 7470 6c6f 740a 2d6e 2c20 2d2d 6e74 6872  tplot.-n, --nthr
+000076d0: 6561 6473 2c20 7468 6520 6e75 6d62 6572  eads, the number
+000076e0: 206f 6620 7468 7265 6164 7320 746f 2075   of threads to u
+000076f0: 7365 2069 6e20 7379 6e74 656e 7920 696e  se in synteny in
+00007700: 6665 7265 6e63 652c 2064 6566 6175 6c74  ference, default
+00007710: 2034 0a60 6060 0a0a 5468 6520 7072 6f67   4.```..The prog
+00007720: 7261 6d20 6077 6764 2076 697a 6020 6361  ram `wgd viz` ca
+00007730: 6e20 7265 616c 697a 6520 7468 6520 7669  n realize the vi
+00007740: 7375 616c 697a 6174 696f 6e20 6f66 202a  sualization of *
+00007750: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+00007760: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
+00007770: 616e 6420 7379 6e74 656e 792e 0a60 6060  and synteny..```
+00007780: 0a77 6764 2076 697a 2028 6f70 7469 6f6e  .wgd viz (option
+00007790: 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ).--------------
+000077a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000077b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000077c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000077d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000077e0: 2d2d 0a2d 642c 202d 2d64 6174 6166 696c  --.-d, --datafil
+000077f0: 652c 2074 6865 204b 7320 6461 7461 6669  e, the Ks datafi
+00007800: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
+00007810: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
+00007820: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
+00007830: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
+00007840: 5f76 697a 0a2d 7372 2c20 2d2d 7370 6169  _viz.-sr, --spai
+00007850: 722c 2074 6865 2073 7065 6369 6573 2070  r, the species p
+00007860: 6169 7220 746f 2062 6520 706c 6f74 7465  air to be plotte
+00007870: 642c 2064 6566 6175 6c74 204e 6f6e 652c  d, default None,
+00007880: 2074 6869 7320 6f70 7469 6f6e 2063 616e   this option can
+00007890: 2062 6520 7072 6f76 6964 6564 206d 756c   be provided mul
+000078a0: 7469 706c 6520 7469 6d65 730a 2d66 612c  tiple times.-fa,
+000078b0: 202d 2d66 6f63 7573 3261 6c6c 2c20 7365   --focus2all, se
+000078c0: 7420 666f 6361 6c20 7370 6563 6965 7320  t focal species 
+000078d0: 616e 6420 6c65 7420 7370 6563 6965 7320  and let species 
+000078e0: 7061 6972 2074 6f20 6265 2062 6574 7765  pair to be betwe
+000078f0: 656e 2066 6f63 616c 2061 6e64 2061 6c6c  en focal and all
+00007900: 2074 6865 2072 656d 6169 6e69 6e67 2073   the remaining s
+00007910: 7065 6369 6573 2c20 6465 6661 756c 7420  pecies, default 
+00007920: 4e6f 6e65 0a2d 6773 2c20 2d2d 6773 6d61  None.-gs, --gsma
+00007930: 702c 2074 6865 2067 656e 6520 6e61 6d65  p, the gene name
+00007940: 2d73 7065 6369 6573 206e 616d 6520 6d61  -species name ma
+00007950: 702c 2064 6566 6175 6c74 204e 6f6e 650a  p, default None.
+00007960: 2d73 702c 202d 2d73 7065 6369 6573 7472  -sp, --speciestr
+00007970: 6565 2c20 7468 6520 7370 6563 6965 7320  ee, the species 
+00007980: 7472 6565 2074 6f20 7065 7266 6f72 6d20  tree to perform 
+00007990: 7261 7465 2063 6f72 7265 6374 696f 6e2c  rate correction,
+000079a0: 2064 6566 6175 6c74 204e 6f6e 652c 2069   default None, i
+000079b0: 6620 4e6f 6e65 2077 6173 2067 6976 656e  f None was given
+000079c0: 2c20 7468 6520 7261 7465 2063 6f72 7265  , the rate corre
+000079d0: 6374 696f 6e20 616e 616c 7973 6973 2077  ction analysis w
+000079e0: 6f75 6c64 2062 6520 6361 6c6c 6564 206f  ould be called o
+000079f0: 6666 0a2d 706b 2c20 2d2d 706c 6f74 6b64  ff.-pk, --plotkd
+00007a00: 652c 2066 6c61 6720 6f70 7469 6f6e 2c20  e, flag option, 
+00007a10: 7768 6574 6865 7220 746f 2070 6c6f 7420  whether to plot 
+00007a20: 6b64 6520 6375 7276 6520 7570 6f6e 2068  kde curve upon h
+00007a30: 6973 746f 6772 616d 2c20 6966 2074 6865  istogram, if the
+00007a40: 2066 6c61 6720 7761 7320 7365 742c 206b   flag was set, k
+00007a50: 6465 2063 7572 7665 2077 6f75 6c64 2062  de curve would b
+00007a60: 6520 6164 6465 640a 2d72 772c 202d 2d72  e added.-rw, --r
+00007a70: 6577 6569 6768 742c 2066 6c61 6720 6f70  eweight, flag op
+00007a80: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00007a90: 2072 6563 616c 6375 6c61 7465 2074 6865   recalculate the
+00007aa0: 2077 6569 6768 7420 7065 7220 7370 6563   weight per spec
+00007ab0: 6965 7320 7061 6972 2c20 6966 2074 6865  ies pair, if the
+00007ac0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+00007ad0: 6865 2077 6569 6768 7420 776f 756c 6420  he weight would 
+00007ae0: 6265 2072 6563 616c 6375 6c61 7465 640a  be recalculated.
+00007af0: 2d6f 722c 202d 2d6f 6e6c 7972 6f6f 746f  -or, --onlyrooto
+00007b00: 7574 2c20 666c 6167 206f 7074 696f 6e2c  ut, flag option,
+00007b10: 2077 6865 7468 6572 2074 6f20 6f6e 6c79   whether to only
+00007b20: 2063 6f6e 6475 6374 2072 6174 6520 636f   conduct rate co
+00007b30: 7272 6563 7469 6f6e 2075 7369 6e67 2074  rrection using t
+00007b40: 6865 206f 7574 6772 6f75 7020 6174 2072  he outgroup at r
+00007b50: 6f6f 7420 6173 206f 7574 6772 6f75 702c  oot as outgroup,
+00007b60: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00007b70: 2073 6574 2c20 6f6e 6c79 2074 6865 206f   set, only the o
+00007b80: 7574 6772 6f75 7020 6174 2072 6f6f 7420  utgroup at root 
+00007b90: 776f 756c 6420 6265 2075 7365 6420 6173  would be used as
+00007ba0: 206f 7574 6772 6f75 700a 2d69 7465 722c   outgroup.-iter,
+00007bb0: 202d 2d65 6d5f 6974 6572 6174 696f 6e73   --em_iterations
+00007bc0: 2c20 7468 6520 6d61 7869 6d75 6d20 454d  , the maximum EM
+00007bd0: 2069 7465 7261 7469 6f6e 732c 2064 6566   iterations, def
+00007be0: 6175 6c74 2032 3030 0a2d 696e 6974 2c20  ault 200.-init, 
+00007bf0: 2d2d 656d 5f69 6e69 7469 616c 697a 6174  --em_initializat
+00007c00: 696f 6e73 2c20 7468 6520 6d61 7869 6d75  ions, the maximu
+00007c10: 6d20 454d 2069 6e69 7469 616c 697a 6174  m EM initializat
+00007c20: 696f 6e73 2c20 6465 6661 756c 7420 3230  ions, default 20
+00007c30: 300a 2d70 7263 742c 202d 2d70 726f 6d69  0.-prct, --promi
+00007c40: 6e65 6e63 655f 6375 746f 6666 2c20 7468  nence_cutoff, th
+00007c50: 6520 7072 6f6d 696e 656e 6365 2063 7574  e prominence cut
+00007c60: 6f66 6620 6f66 2061 6363 6570 7461 626c  off of acceptabl
+00007c70: 6520 7065 616b 732c 2064 6566 6175 6c74  e peaks, default
+00007c80: 2030 2e31 0a2d 7268 2c20 2d2d 7265 6c5f   0.1.-rh, --rel_
+00007c90: 6865 6967 6874 2c20 7468 6520 7265 6c61  height, the rela
+00007ca0: 7469 7665 2068 6569 6768 7420 6174 2077  tive height at w
+00007cb0: 6869 6368 2074 6865 2070 6561 6b20 7769  hich the peak wi
+00007cc0: 6474 6820 6973 206d 6561 7375 7265 642c  dth is measured,
+00007cd0: 2064 6566 6175 6c74 2030 2e34 0a2d 736d   default 0.4.-sm
+00007ce0: 2c20 2d2d 7365 676d 656e 7473 2c20 7468  , --segments, th
+00007cf0: 6520 7365 676d 656e 7473 2064 6174 6166  e segments dataf
+00007d00: 696c 652c 2064 6566 6175 6c74 204e 6f6e  ile, default Non
+00007d10: 650a 2d6d 6c2c 202d 2d6d 696e 6c65 6e2c  e.-ml, --minlen,
+00007d20: 2074 6865 206d 696e 696d 756d 206c 656e   the minimum len
+00007d30: 6774 6820 6f66 2061 2073 6361 6666 6f6c  gth of a scaffol
+00007d40: 6420 746f 2062 6520 696e 636c 7564 6564  d to be included
+00007d50: 2069 6e20 646f 7470 6c6f 742c 2064 6566   in dotplot, def
+00007d60: 6175 6c74 202d 312c 2069 6620 2d31 2077  ault -1, if -1 w
+00007d70: 6173 2073 6574 2c20 7468 6520 3130 2520  as set, the 10% 
+00007d80: 6f66 2074 6865 206c 6f6e 6765 7374 2073  of the longest s
+00007d90: 6361 6666 6f6c 6473 2077 696c 6c20 6265  caffolds will be
+00007da0: 2073 6574 0a2d 6d73 2c20 2d2d 6d61 7873   set.-ms, --maxs
+00007db0: 697a 652c 2074 6865 206d 6178 696d 756d  ize, the maximum
+00007dc0: 2066 616d 696c 7920 7369 7a65 2074 6f20   family size to 
+00007dd0: 6265 2069 6e63 6c75 6465 642c 2064 6566  be included, def
+00007de0: 6175 6c74 2032 3030 0a2d 6170 2c20 2d2d  ault 200.-ap, --
+00007df0: 616e 6368 6f72 706f 696e 7473 2c20 7468  anchorpoints, th
+00007e00: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
+00007e10: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
+00007e20: 7420 4e6f 6e65 0a2d 6d74 2c20 2d2d 6d75  t None.-mt, --mu
+00007e30: 6c74 6970 6c69 636f 6e2c 2074 6865 206d  ltiplicon, the m
+00007e40: 756c 7469 706c 6963 6f6e 7320 6461 7461  ultiplicons data
+00007e50: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
+00007e60: 6e65 0a2d 6774 2c20 2d2d 6765 6e65 7461  ne.-gt, --geneta
+00007e70: 626c 652c 2074 6865 2067 656e 6520 7461  ble, the gene ta
+00007e80: 626c 6520 6461 7461 6669 6c65 2c20 6465  ble datafile, de
+00007e90: 6661 756c 7420 4e6f 6e65 0a2d 6d67 2c20  fault None.-mg, 
+00007ea0: 2d2d 6d69 6e73 6567 6c65 6e2c 2074 6865  --minseglen, the
+00007eb0: 206d 696e 696d 756d 206c 656e 6774 6820   minimum length 
+00007ec0: 6f66 2073 6567 6d65 6e74 7320 746f 2069  of segments to i
+00007ed0: 6e63 6c75 6465 2c20 696e 2072 6174 696f  nclude, in ratio
+00007ee0: 2069 6620 3c3d 2031 2c20 6465 6661 756c   if <= 1, defaul
+00007ef0: 7420 3130 3030 300a 2d6d 676e 2c20 2d2d  t 10000.-mgn, --
+00007f00: 6d69 6e67 656e 656e 756d 2c20 7468 6520  mingenenum, the 
+00007f10: 6d69 6e69 6d75 6d20 6e75 6d62 6572 206f  minimum number o
+00007f20: 6620 6765 6e65 7320 666f 7220 6120 7365  f genes for a se
+00007f30: 676d 656e 7420 746f 2062 6520 636f 6e73  gment to be cons
+00007f40: 6964 6572 6564 2c20 6465 6661 756c 7420  idered, default 
+00007f50: 3330 0a2d 6b72 2c20 2d2d 6b65 6570 7265  30.-kr, --keepre
+00007f60: 6475 6e2c 2066 6c61 6720 6f70 7469 6f6e  dun, flag option
+00007f70: 2c20 7768 6574 6865 7220 746f 206b 6565  , whether to kee
+00007f80: 7020 7265 6475 6e64 616e 7420 6d75 6c74  p redundant mult
+00007f90: 6970 6c69 636f 6e73 2c20 6966 2074 6865  iplicons, if the
+00007fa0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+00007fb0: 6865 2072 6564 756e 6461 6e74 206d 756c  he redundant mul
+00007fc0: 7469 706c 6963 6f6e 7320 776f 756c 6420  tiplicons would 
+00007fd0: 6265 206b 6570 740a 2d65 706b 2c20 2d2d  be kept.-epk, --
+00007fe0: 6578 7472 6170 6172 616e 6f6d 656b 732c  extraparanomeks,
+00007ff0: 2065 7874 7261 2070 6172 616e 6f6d 6520   extra paranome 
+00008000: 4b73 2064 6174 6120 746f 2070 6c6f 7420  Ks data to plot 
+00008010: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
+00008020: 6469 7374 7269 6275 7469 6f6e 2c20 6465  distribution, de
+00008030: 6661 756c 7420 4e6f 6e65 0a2d 7061 672c  fault None.-pag,
+00008040: 202d 2d70 6c6f 7461 7067 6d6d 2c20 666c   --plotapgmm, fl
+00008050: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+00008060: 6572 2074 6f20 636f 6e64 7563 7420 616e  er to conduct an
+00008070: 6420 706c 6f74 206d 6978 7475 7265 206d  d plot mixture m
+00008080: 6f64 656c 696e 6720 6f66 2061 6e63 686f  odeling of ancho
+00008090: 7220 4b73 2069 6e20 7468 6520 6d69 7865  r Ks in the mixe
+000080a0: 6420 4b73 2064 6973 7472 6962 7574 696f  d Ks distributio
+000080b0: 6e2c 2069 6620 7468 6520 666c 6167 2077  n, if the flag w
+000080c0: 6173 2073 6574 2c20 7468 6520 6d69 7874  as set, the mixt
+000080d0: 7572 6520 6d6f 6465 6c69 6e67 206f 6620  ure modeling of 
+000080e0: 616e 6368 6f72 204b 7320 776f 756c 6420  anchor Ks would 
+000080f0: 6265 2063 6f6e 6475 6374 6564 2061 6e64  be conducted and
+00008100: 2070 6c6f 7474 6564 0a2d 7065 6d2c 202d   plotted.-pem, -
+00008110: 2d70 6c6f 7465 6c6d 6d2c 2066 6c61 6720  -plotelmm, flag 
+00008120: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00008130: 746f 2063 6f6e 6475 6374 2061 6e64 2070  to conduct and p
+00008140: 6c6f 7420 656c 6d6d 206d 6978 7475 7265  lot elmm mixture
+00008150: 206d 6f64 656c 696e 6720 6f66 2070 6172   modeling of par
+00008160: 616e 6f6d 6520 4b73 2069 6e20 7468 6520  anome Ks in the 
+00008170: 6d69 7865 6420 4b73 2064 6973 7472 6962  mixed Ks distrib
+00008180: 7574 696f 6e2c 2069 6620 7468 6520 666c  ution, if the fl
+00008190: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+000081a0: 656c 6d6d 206d 6978 7475 7265 206d 6f64  elmm mixture mod
+000081b0: 656c 696e 6720 6f66 2070 6172 616e 6f6d  eling of paranom
+000081c0: 6520 4b73 2077 6f75 6c64 2062 6520 636f  e Ks would be co
+000081d0: 6e64 7563 7465 6420 616e 6420 706c 6f74  nducted and plot
+000081e0: 7465 640a 2d63 2c20 2d2d 636f 6d70 6f6e  ted.-c, --compon
+000081f0: 656e 7473 2c20 7468 6520 7261 6e67 6520  ents, the range 
+00008200: 6f66 2074 6865 206e 756d 6265 7220 6f66  of the number of
+00008210: 2063 6f6d 706f 6e65 6e74 7320 746f 2066   components to f
+00008220: 6974 2069 6e20 616e 6368 6f72 204b 7320  it in anchor Ks 
+00008230: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
+00008240: 2c20 6465 6661 756c 7420 2831 2c34 290a  , default (1,4).
+00008250: 2d70 7379 2c20 2d2d 706c 6f74 7379 6e2c  -psy, --plotsyn,
+00008260: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00008270: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
+00008280: 6520 7468 6520 7379 6e74 656e 7920 706c  e the synteny pl
+00008290: 6f74 2c20 6f6e 6c79 2077 6865 6e20 7468  ot, only when th
+000082a0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+000082b0: 7468 6520 7379 6e74 656e 7920 706c 6f74  the synteny plot
+000082c0: 2077 6f75 6c64 2062 6520 7072 6f64 7563   would be produc
+000082d0: 6564 0a2d 6473 2c20 2d2d 646f 7473 697a  ed.-ds, --dotsiz
+000082e0: 652c 2074 6865 2064 6f74 2073 697a 6520  e, the dot size 
+000082f0: 696e 2064 6f74 2070 6c6f 742c 2064 6566  in dot plot, def
+00008300: 6175 6c74 2030 2e33 0a2d 6161 2c20 2d2d  ault 0.3.-aa, --
+00008310: 6170 616c 7068 612c 2074 6865 206f 7061  apalpha, the opa
+00008320: 6369 7479 206f 6620 616e 6368 6f72 2064  city of anchor d
+00008330: 6f74 732c 2064 6566 6175 6c74 2031 0a2d  ots, default 1.-
+00008340: 6861 2c20 2d2d 686f 616c 7068 612c 2074  ha, --hoalpha, t
+00008350: 6865 206f 7061 6369 7479 206f 6620 686f  he opacity of ho
+00008360: 6d6f 6c6f 6720 646f 7473 2c20 6465 6661  molog dots, defa
+00008370: 756c 7420 300a 2d73 7274 2c20 2d2d 7368  ult 0.-srt, --sh
+00008380: 6f77 7265 616c 7469 636b 2c20 666c 6167  owrealtick, flag
+00008390: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
+000083a0: 2074 6f20 7368 6f77 2074 6865 2072 6561   to show the rea
+000083b0: 6c20 7469 636b 2069 6e20 6765 6e65 7320  l tick in genes 
+000083c0: 6f72 2062 6173 6573 2c20 6966 2074 6865  or bases, if the
+000083d0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+000083e0: 6865 2072 6561 6c20 7469 636b 2077 6f75  he real tick wou
+000083f0: 6c64 2062 6520 7368 6f77 6564 0a2d 746c  ld be showed.-tl
+00008400: 732c 202d 2d74 6963 6b6c 6162 656c 7369  s, --ticklabelsi
+00008410: 7a65 2c20 7468 6520 6c61 6265 6c20 7369  ze, the label si
+00008420: 7a65 206f 6620 7469 636b 2c20 6465 6661  ze of tick, defa
+00008430: 756c 7420 350a 2d78 6c2c 202d 2d78 6c69  ult 5.-xl, --xli
+00008440: 6d2c 2074 6865 2078 2061 7869 7320 6c69  m, the x axis li
+00008450: 6d69 7420 6f66 204b 7320 6469 7374 7269  mit of Ks distri
+00008460: 6275 7469 6f6e 0a2d 796c 2c20 2d2d 796c  bution.-yl, --yl
+00008470: 696d 2c20 7468 6520 7920 6178 6973 206c  im, the y axis l
+00008480: 696d 6974 206f 6620 4b73 2064 6973 7472  imit of Ks distr
+00008490: 6962 7574 696f 6e0a 2d61 646f 2c20 2d2d  ibution.-ado, --
+000084a0: 6164 6a75 7374 6f72 7468 6f2c 2066 6c61  adjustortho, fla
+000084b0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+000084c0: 7220 746f 2061 646a 7573 7420 7468 6520  r to adjust the 
+000084d0: 6869 7374 6f67 7261 6d20 6865 6967 6874  histogram height
+000084e0: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
+000084f0: 4b73 2061 7320 746f 206d 6174 6368 2074  Ks as to match t
+00008500: 6865 2068 6569 6768 7420 6f66 2070 6172  he height of par
+00008510: 616c 6f67 6f75 7320 4b73 2c20 6966 2074  alogous Ks, if t
+00008520: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00008530: 2074 6865 2061 646a 7573 746d 656e 7420   the adjustment 
+00008540: 776f 756c 6420 6265 2063 6f6e 6475 6374  would be conduct
+00008550: 6564 0a2d 6164 662c 202d 2d61 646a 7573  ed.-adf, --adjus
+00008560: 7466 6163 746f 722c 2074 6865 2061 646a  tfactor, the adj
+00008570: 7573 746d 656e 7420 6661 6374 6f72 206f  ustment factor o
+00008580: 6620 6f72 7468 6f6c 6f67 6f75 7320 4b73  f orthologous Ks
+00008590: 2c20 6465 6661 756c 7420 302e 350a 2d6f  , default 0.5.-o
+000085a0: 612c 202d 2d6f 6b61 6c70 6861 2c20 7468  a, --okalpha, th
+000085b0: 6520 6f70 6163 6974 7920 6f66 206f 7274  e opacity of ort
+000085c0: 686f 6c6f 676f 7573 204b 7320 6469 7374  hologous Ks dist
+000085d0: 7269 6275 7469 6f6e 2069 6e20 6d69 7865  ribution in mixe
+000085e0: 6420 706c 6f74 2c20 6465 6661 756c 7420  d plot, default 
+000085f0: 302e 350a 2d63 732c 202d 2d63 6c61 7373  0.5.-cs, --class
+00008600: 6963 2c20 666c 6167 206f 7074 696f 6e2c  ic, flag option,
+00008610: 2077 6865 7468 6572 2074 6f20 6472 6177   whether to draw
+00008620: 206d 6978 6564 204b 7320 706c 6f74 2069   mixed Ks plot i
+00008630: 6e20 6120 636c 6173 7369 6320 6d61 6e6e  n a classic mann
+00008640: 6572 2077 6865 7265 2074 6865 2066 756c  er where the ful
+00008650: 6c20 6f72 7468 6f6c 6f67 6f75 7320 4b73  l orthologous Ks
+00008660: 2064 6973 7472 6962 7574 696f 6e20 6973   distribution is
+00008670: 2064 7261 7765 642c 2069 6620 7468 6520   drawed, if the 
+00008680: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+00008690: 6520 636c 6173 7369 6320 6d69 7865 6420  e classic mixed 
+000086a0: 4b73 2070 6c6f 7420 776f 756c 6420 6265  Ks plot would be
+000086b0: 2064 7261 776e 0a2d 7461 2c20 2d2d 746f   drawn.-ta, --to
+000086c0: 7061 7272 6f77 2c20 666c 6167 206f 7074  parrow, flag opt
+000086d0: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+000086e0: 6164 6a75 7374 2074 6865 2061 7272 6f77  adjust the arrow
+000086f0: 2074 6f20 6265 2061 7420 7468 6520 746f   to be at the to
+00008700: 7020 6f66 2074 6865 2070 6c6f 742c 2069  p of the plot, i
+00008710: 6e73 7465 6164 206f 6620 6265 696e 6720  nstead of being 
+00008720: 636f 6f72 6469 6e61 7465 6420 6173 2074  coordinated as t
+00008730: 6865 204b 4445 206f 6620 7468 6520 6f72  he KDE of the or
+00008740: 7468 6f6c 6f67 6f75 7320 4b73 2064 6973  thologous Ks dis
+00008750: 7472 6962 7574 696f 6e2c 2069 6620 7468  tribution, if th
+00008760: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00008770: 7468 6520 6172 726f 7720 776f 756c 6420  the arrow would 
+00008780: 6265 2073 6574 2061 7420 7468 6520 746f  be set at the to
+00008790: 700a 2d6e 612c 202d 2d6e 6f64 6561 7665  p.-na, --nodeave
+000087a0: 7261 6765 642c 2066 6c61 6720 6f70 7469  raged, flag opti
+000087b0: 6f6e 2c20 7768 6574 6865 7220 746f 2075  on, whether to u
+000087c0: 7365 206e 6f64 652d 6176 6572 6167 6564  se node-averaged
+000087d0: 206d 6574 686f 6420 666f 7220 6465 2d72   method for de-r
+000087e0: 6564 756e 6461 6e63 792c 2069 6620 7468  edundancy, if th
+000087f0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00008800: 7468 6520 6e6f 6465 2d61 7665 7261 6765  the node-average
+00008810: 6420 6d65 7468 6f64 2077 6f75 6c64 2062  d method would b
+00008820: 6520 696e 6974 6961 7465 640a 2d62 732c  e initiated.-bs,
+00008830: 202d 2d62 6f6f 7473 7472 6170 2c20 7468   --bootstrap, th
+00008840: 6520 6e75 6d62 6572 206f 6620 626f 6f74  e number of boot
+00008850: 7374 7261 7020 7265 706c 6963 6174 6573  strap replicates
+00008860: 206f 6620 6f72 7468 6f6c 6f67 204b 7320   of ortholog Ks 
+00008870: 6469 7374 7269 6275 7469 6f6e 2069 6e20  distribution in 
+00008880: 6d69 7865 6420 706c 6f74 0a2d 6772 2c20  mixed plot.-gr, 
+00008890: 2d2d 6769 7374 7262 2c20 666c 6167 206f  --gistrb, flag o
+000088a0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+000088b0: 6f20 7573 6520 6769 7374 5f72 6169 6e62  o use gist_rainb
+000088c0: 6f77 2061 7320 636f 6c6f 7220 6d61 7020  ow as color map 
+000088d0: 6f66 2064 6f74 706c 6f74 0a2d 6e2c 202d  of dotplot.-n, -
+000088e0: 2d6e 7468 7265 6164 732c 2074 6865 206e  -nthreads, the n
+000088f0: 756d 6265 7220 6f66 2074 6872 6561 6473  umber of threads
+00008900: 2074 6f20 7573 6520 696e 2062 6f6f 7473   to use in boots
+00008910: 7472 6170 2073 616d 706c 696e 672c 2064  trap sampling, d
+00008920: 6566 6175 6c74 2031 0a60 6060 0a0a 2323  efault 1.```..##
+00008930: 2055 7361 6765 0a0a 4865 7265 2077 6520   Usage..Here we 
+00008940: 7072 6f76 6964 6564 2074 6865 2062 6173  provided the bas
+00008950: 6963 2075 7361 6765 2066 6f72 2065 6163  ic usage for eac
+00008960: 6820 7072 6f67 7261 6d20 616e 6420 7468  h program and th
+00008970: 6520 7265 6c65 7661 6e74 2070 6172 616d  e relevant param
+00008980: 6574 6572 7320 616e 6420 7375 6767 6573  eters and sugges
+00008990: 7469 6f6e 7320 6f6e 2070 6172 616d 6574  tions on paramet
+000089a0: 6572 697a 6174 696f 6e2e 2041 2072 656d  erization. A rem
+000089b0: 696e 6465 7220 7468 6174 2074 6865 2067  inder that the g
+000089c0: 6976 656e 2063 6f72 6573 2061 6e64 2074  iven cores and t
+000089d0: 6872 6561 6473 2063 616e 2073 6967 6e69  hreads can signi
+000089e0: 6669 6361 6e74 6c79 2069 6d70 6163 7420  ficantly impact 
+000089f0: 7468 6520 7275 6e20 7469 6d65 2061 6e64  the run time and
+00008a00: 2074 6875 7320 7765 2061 6464 6564 2073   thus we added s
+00008a10: 6f6d 6520 7265 706f 7274 2069 6e66 6f72  ome report infor
+00008a20: 6d61 7469 6f6e 2070 6572 7461 696e 696e  mation pertainin
+00008a30: 6720 746f 2074 6865 2073 7973 7465 6d20  g to the system 
+00008a40: 6f66 2075 7365 7273 2074 6f20 6661 6369  of users to faci
+00008a50: 6c69 7461 7465 2074 6865 2065 6666 6963  litate the effic
+00008a60: 6965 6e74 2073 6574 7469 6e67 206f 6620  ient setting of 
+00008a70: 7468 7265 6164 7320 616e 6420 6d65 6d6f  threads and memo
+00008a80: 7279 2e20 5468 6520 6c6f 6769 6361 6c20  ry. The logical 
+00008a90: 4350 5573 2072 6570 6f72 7465 6420 7265  CPUs reported re
+00008aa0: 7072 6573 656e 7473 2074 6865 206e 756d  presents the num
+00008ab0: 6265 7220 6f66 2070 6879 7369 6361 6c20  ber of physical 
+00008ac0: 636f 7265 7320 6d75 6c74 6970 6c69 6564  cores multiplied
+00008ad0: 2062 7920 7468 6520 6e75 6d62 6572 206f   by the number o
+00008ae0: 6620 7468 7265 6164 7320 7468 6174 2063  f threads that c
+00008af0: 616e 2072 756e 206f 6e20 6561 6368 2063  an run on each c
+00008b00: 6f72 652c 2061 6c73 6f20 6b6e 6f77 6e20  ore, also known 
+00008b10: 6173 2048 7970 6572 2054 6872 6561 6469  as Hyper Threadi
+00008b20: 6e67 2e20 5468 6520 6e75 6d62 6572 206f  ng. The number o
+00008b30: 6620 6c6f 6769 6361 6c20 4350 5573 206d  f logical CPUs m
+00008b40: 6179 206e 6f74 206e 6563 6573 7361 7269  ay not necessari
+00008b50: 6c79 2062 6520 6571 7569 7661 6c65 6e74  ly be equivalent
+00008b60: 2074 6f20 7468 6520 6163 7475 616c 206e   to the actual n
+00008b70: 756d 6265 7220 6f66 2043 5055 7320 7468  umber of CPUs th
+00008b80: 6520 6375 7272 656e 7420 7072 6f63 6573  e current proces
+00008b90: 7320 6361 6e20 7573 652e 2054 6865 2061  s can use. The a
+00008ba0: 7661 696c 6162 6c65 206d 656d 6f72 7920  vailable memory 
+00008bb0: 7265 6665 7273 2074 6f20 7468 6520 6d65  refers to the me
+00008bc0: 6d6f 7279 2074 6861 7420 6361 6e20 6265  mory that can be
+00008bd0: 2067 6976 656e 2069 6e73 7461 6e74 6c79   given instantly
+00008be0: 2074 6f20 7072 6f63 6573 7365 7320 7769   to processes wi
+00008bf0: 7468 6f75 7420 7468 6520 7379 7374 656d  thout the system
+00008c00: 2067 6f69 6e67 2069 6e74 6f20 7377 6170   going into swap
+00008c10: 2061 6e64 2072 6566 6c65 6374 7320 7468   and reflects th
+00008c20: 6520 6163 7475 616c 206d 656d 6f72 7920  e actual memory 
+00008c30: 6176 6169 6c61 626c 652e 2054 6865 2066  available. The f
+00008c40: 7265 6520 6d65 6d6f 7279 2072 6566 6572  ree memory refer
+00008c50: 7320 746f 2074 6865 206d 656d 6f72 7920  s to the memory 
+00008c60: 6e6f 7420 6265 696e 6720 7573 6564 2061  not being used a
+00008c70: 7420 616c 6c20 287a 6572 6f65 6429 2074  t all (zeroed) t
+00008c80: 6861 7420 6973 2072 6561 6469 6c79 2061  hat is readily a
+00008c90: 7661 696c 6162 6c65 2e20 5468 6520 6465  vailable. The de
+00008ca0: 7363 7269 7074 696f 6e20 6162 6f76 6520  scription above 
+00008cb0: 7265 6665 7273 2074 6f20 7468 6520 646f  refers to the do
+00008cc0: 6375 6d65 6e74 6174 696f 6e20 6f66 2060  cumentation of `
+00008cd0: 7073 7574 696c 602e 0a0a 2323 2320 7767  psutil`...### wg
+00008ce0: 6420 646d 640a 0a2a 2a54 6865 2064 656c  d dmd..**The del
+00008cf0: 696e 6561 7469 6f6e 206f 6620 7768 6f6c  ineation of whol
+00008d00: 6520 7061 7261 6e6f 6d65 2a2a 0a60 6060  e paranome**.```
+00008d10: 0a77 6764 2064 6d64 2041 7175 696c 6567  .wgd dmd Aquileg
+00008d20: 6961 5f63 6f65 7275 6c65 6120 2d49 2032  ia_coerulea -I 2
+00008d30: 202d 6520 3165 2d31 3020 2d62 7320 3130   -e 1e-10 -bs 10
+00008d40: 3020 2d6e 7020 3520 282d 6e6e 2920 282d  0 -np 5 (-nn) (-
+00008d50: 2d74 6f5f 7374 6f70 2920 282d 2d63 6473  -to_stop) (--cds
+00008d60: 2920 282d 6e20 3429 2028 2d6f 2077 6764  ) (-n 4) (-o wgd
+00008d70: 5f64 6d64 2920 282d 7420 776f 726b 696e  _dmd) (-t workin
+00008d80: 675f 746d 7029 0a60 6060 200a 0a4e 6f74  g_tmp).``` ..Not
+00008d90: 6520 7468 6174 2077 6520 646f 6e27 7420  e that we don't 
+00008da0: 7072 6f76 6964 6520 7468 6520 6461 7461  provide the data
+00008db0: 206f 6620 7468 6973 2063 6f64 696e 6720   of this coding 
+00008dc0: 7365 7175 656e 6365 2028 6364 7329 2066  sequence (cds) f
+00008dd0: 696c 6520 6041 7175 696c 6567 6961 5f63  ile `Aquilegia_c
+00008de0: 6f65 7275 6c65 6160 2062 7574 2069 7420  oerulea` but it 
+00008df0: 6361 6e20 6265 2065 6173 696c 7920 646f  can be easily do
+00008e00: 776e 6c6f 6164 6564 2061 7420 5b50 6879  wnloaded at [Phy
+00008e10: 746f 7a6f 6d65 5d28 6874 7470 733a 2f2f  tozome](https://
+00008e20: 7068 7974 6f7a 6f6d 652d 6e65 7874 2e6a  phytozome-next.j
+00008e30: 6769 2e64 6f65 2e67 6f76 2f69 6e66 6f2f  gi.doe.gov/info/
+00008e40: 4163 6f65 7275 6c65 615f 7633 5f31 2920  Acoerulea_v3_1) 
+00008e50: 2873 616d 6520 666f 7220 6f74 6865 7220  (same for other 
+00008e60: 5573 6167 6520 646f 6329 2e20 4120 7265  Usage doc). A re
+00008e70: 6d69 6e64 6572 2074 6861 7420 696e 2074  minder that in t
+00008e80: 6865 2069 7373 7565 7320 736f 6d65 2075  he issues some u
+00008e90: 7365 7273 2064 6964 6e27 7420 646f 776e  sers didn't down
+00008ea0: 6c6f 6164 2074 6865 2060 4163 6f65 7275  load the `Acoeru
+00008eb0: 6c65 615f 3332 325f 7633 2e31 2e63 6473  lea_322_v3.1.cds
+00008ec0: 5f70 7269 6d61 7279 5472 616e 7363 7269  _primaryTranscri
+00008ed0: 7074 4f6e 6c79 2e66 612e 677a 6020 6275  ptOnly.fa.gz` bu
+00008ee0: 7420 696e 7374 6561 6420 7468 6520 6041  t instead the `A
+00008ef0: 636f 6572 756c 6561 5f33 3232 5f76 332e  coerulea_322_v3.
+00008f00: 312e 6364 732e 6661 2e67 7a60 2066 696c  1.cds.fa.gz` fil
+00008f10: 652e 2046 6f72 2074 6865 2063 6f6e 7374  e. For the const
+00008f20: 7275 6374 696f 6e20 6f66 2077 686f 6c65  ruction of whole
+00008f30: 2070 6172 616e 6f6d 6520 2a4b 2a3c 7375   paranome *K*<su
+00008f40: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
+00008f50: 6275 7469 6f6e 2c20 2a2a 6f6e 6c79 206f  bution, **only o
+00008f60: 6e65 2074 7261 6e73 6372 6970 7420 2874  ne transcript (t
+00008f70: 6865 2070 7269 6d61 7279 206f 6e65 2920  he primary one) 
+00008f80: 7065 7220 6765 6e65 2073 686f 756c 6420  per gene should 
+00008f90: 6265 2069 6e63 6c75 6465 642a 2a20 7375  be included** su
+00008fa0: 6368 2074 6861 7420 7468 6520 2a4b 2a3c  ch that the *K*<
+00008fb0: 7375 623e 533c 2f73 7562 3e20 6973 2072  sub>S</sub> is r
+00008fc0: 6561 6c6c 7920 696e 6469 6361 7469 6e67  eally indicating
+00008fd0: 2074 6865 2061 6765 206f 6620 6765 6e65   the age of gene
+00008fe0: 2064 7570 6c69 6361 7469 6f6e 2065 7665   duplication eve
+00008ff0: 6e74 2c20 696e 7374 6561 6420 6f66 2061  nt, instead of a
+00009000: 6c74 6572 6e61 7469 7665 2073 706c 6963  lternative splic
+00009010: 696e 672e 2054 7261 6e73 6372 6970 746f  ing. Transcripto
+00009020: 6d65 2064 6174 6120 7368 6f75 6c64 2062  me data should b
+00009030: 6520 6361 7265 6675 6c6c 7920 7472 6561  e carefully trea
+00009040: 7465 6420 7769 7468 2064 652d 7265 6475  ted with de-redu
+00009050: 6e64 616e 6379 2073 6f20 6173 2074 6f20  ndancy so as to 
+00009060: 7265 6475 6365 2074 6865 2066 616c 7365  reduce the false
+00009070: 2070 6f73 6974 6976 6520 6475 706c 6963   positive duplic
+00009080: 6174 696f 6e20 6275 6d70 2063 6175 7365  ation bump cause
+00009090: 6420 6279 2070 6572 7661 7369 7665 2061  d by pervasive a
+000090a0: 6c74 6572 6e61 7469 7665 2073 706c 6963  lternative splic
+000090b0: 696e 672e 2048 6572 6520 7468 6520 696e  ing. Here the in
+000090c0: 666c 6174 696f 6e20 6661 6374 6f72 2070  flation factor p
+000090d0: 6172 616d 6574 6572 2c20 6769 7665 6e20  arameter, given 
+000090e0: 6279 2060 2d49 6020 6f72 2060 2d2d 696e  by `-I` or `--in
+000090f0: 666c 6174 696f 6e60 2c20 6166 6665 6374  flation`, affect
+00009100: 7320 7468 6520 6772 616e 756c 6172 6974  s the granularit
+00009110: 7920 6f72 2072 6573 6f6c 7574 696f 6e20  y or resolution 
+00009120: 6f66 2074 6865 2063 6c75 7374 6572 696e  of the clusterin
+00009130: 6720 6f75 7463 6f6d 6520 616e 6420 696d  g outcome and im
+00009140: 706c 6963 6974 6c79 2063 6f6e 7472 6f6c  plicitly control
+00009150: 6c73 2074 6865 206e 756d 6265 7220 6f66  ls the number of
+00009160: 2063 6c75 7374 6572 732c 2077 6974 6820   clusters, with 
+00009170: 6c6f 7720 7661 6c75 6573 2073 7563 6820  low values such 
+00009180: 6173 2031 2e33 206f 7220 312e 3420 6c65  as 1.3 or 1.4 le
+00009190: 6164 696e 6720 746f 2066 6577 6572 2062  ading to fewer b
+000091a0: 7574 206c 6172 6765 7220 636c 7573 7465  ut larger cluste
+000091b0: 7273 2061 6e64 2068 6967 6820 7661 6c75  rs and high valu
+000091c0: 6573 2073 7563 6820 6173 2035 206f 7220  es such as 5 or 
+000091d0: 3620 6c65 6164 696e 6720 746f 206d 6f72  6 leading to mor
+000091e0: 6520 6275 7420 736d 616c 6c65 7220 636c  e but smaller cl
+000091f0: 7573 7465 7273 2e20 5765 2073 6574 2074  usters. We set t
+00009200: 6865 2064 6566 6175 6c74 2076 616c 7565  he default value
+00009210: 2061 7320 3220 6173 2073 7567 6765 7374   as 2 as suggest
+00009220: 6564 2062 7920 5b4d 434c 5d28 6874 7470  ed by [MCL](http
+00009230: 733a 2f2f 6d69 6361 6e73 2e6f 7267 2f6d  s://micans.org/m
+00009240: 636c 2f29 2e20 5468 6520 652d 7661 6c75  cl/). The e-valu
+00009250: 6520 6375 742d 6f66 6620 666f 7220 7365  e cut-off for se
+00009260: 7175 656e 6365 2073 696d 696c 6172 6974  quence similarit
+00009270: 792c 2067 6976 656e 2062 7920 602d 6560  y, given by `-e`
+00009280: 206f 7220 602d 2d65 7661 6c60 2c20 7768   or `--eval`, wh
+00009290: 6963 6820 6465 6e6f 7465 7320 7468 6520  ich denotes the 
+000092a0: 6578 7065 6374 6564 2076 616c 7565 206f  expected value o
+000092b0: 6620 7468 6520 6869 7420 7175 616e 7469  f the hit quanti
+000092c0: 6669 6573 2074 6865 206e 756d 6265 7220  fies the number 
+000092d0: 6f66 2061 6c69 676e 6d65 6e74 7320 6f66  of alignments of
+000092e0: 2073 696d 696c 6172 206f 7220 6265 7474   similar or bett
+000092f0: 6572 2071 7561 6c69 7479 2074 6861 7420  er quality that 
+00009300: 796f 7520 6578 7065 6374 2074 6f20 6669  you expect to fi
+00009310: 6e64 2073 6561 7263 6869 6e67 2074 6869  nd searching thi
+00009320: 7320 7175 6572 7920 6167 6169 6e73 7420  s query against 
+00009330: 6120 6461 7461 6261 7365 206f 6620 7261  a database of ra
+00009340: 6e64 6f6d 2073 6571 7565 6e63 6573 2074  ndom sequences t
+00009350: 6865 2073 616d 6520 7369 7a65 2061 7320  he same size as 
+00009360: 7468 6520 6163 7475 616c 2074 6172 6765  the actual targe
+00009370: 7420 6461 7461 6261 7365 2c20 6973 2074  t database, is t
+00009380: 6865 206b 6579 2070 6172 616d 6574 6572  he key parameter
+00009390: 206d 6561 7375 7269 6e67 2074 6865 2073   measuring the s
+000093a0: 6967 6e69 6669 6361 6e63 6520 6f66 2061  ignificance of a
+000093b0: 2068 6974 2c20 7768 6963 6820 6973 2073   hit, which is s
+000093c0: 6574 2068 6572 6520 6173 2064 6566 6175  et here as defau
+000093d0: 6c74 2031 652d 3130 2e20 4e6f 7465 2074  lt 1e-10. Note t
+000093e0: 6861 7420 5b44 4941 4d4f 4e44 5d28 6874  hat [DIAMOND](ht
+000093f0: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
+00009400: 2f62 6275 6368 6669 6e6b 2f64 6961 6d6f  /bbuchfink/diamo
+00009410: 6e64 2f77 696b 692f 312e 2d54 7574 6f72  nd/wiki/1.-Tutor
+00009420: 6961 6c29 2069 7473 656c 6620 6279 2064  ial) itself by d
+00009430: 6566 6175 6c74 206f 6e6c 7920 7265 706f  efault only repo
+00009440: 7274 7320 616c 6c20 616c 6967 6e6d 656e  rts all alignmen
+00009450: 7473 2077 6974 6820 652d 7661 6c75 6520  ts with e-value 
+00009460: 3c20 302e 3030 312e 2054 6865 2070 6572  < 0.001. The per
+00009470: 6365 6e74 6167 6520 6f66 2075 7070 6572  centage of upper
+00009480: 2068 6974 7320 7573 6564 2066 6f72 2067   hits used for g
+00009490: 656e 6520 6c65 6e67 7468 206e 6f72 6d61  ene length norma
+000094a0: 6c69 7a61 7469 6f6e 2c20 6769 7665 6e20  lization, given 
+000094b0: 6279 2060 2d6e 7060 206f 7220 602d 2d6e  by `-np` or `--n
+000094c0: 6f72 6d61 6c69 7a65 6470 6572 6365 6e74  ormalizedpercent
+000094d0: 602c 2077 6869 6368 2064 6574 6572 6d69  `, which determi
+000094e0: 6e65 7320 7468 6520 7570 7065 7220 7065  nes the upper pe
+000094f0: 7263 656e 7469 6c65 206f 6620 6869 7473  rcentile of hits
+00009500: 2070 6572 2062 696e 2028 6361 7465 676f   per bin (catego
+00009510: 7269 7a65 6420 6279 2067 656e 6520 6c65  rized by gene le
+00009520: 6e67 7468 2920 7573 6564 2069 6e20 7468  ngth) used in th
+00009530: 6520 6669 7420 6f66 206c 696e 6561 7220  e fit of linear 
+00009540: 7265 6772 6573 7369 6f6e 2c20 636f 6e73  regression, cons
+00009550: 6964 6572 696e 6720 7468 6174 206e 6f74  idering that not
+00009560: 2061 6c6c 2068 6974 7320 7065 7220 6269   all hits per bi
+00009570: 6e20 7368 6f77 2061 7070 6172 656e 7420  n show apparent 
+00009580: 6c69 6e65 6172 2072 656c 6174 696f 6e73  linear relations
+00009590: 6869 702c 2069 7320 7365 7420 6173 2064  hip, is set as d
+000095a0: 6566 6175 6c74 2035 2c20 696e 6469 6361  efault 5, indica
+000095b0: 7469 6e67 2074 6865 2075 7361 6765 206f  ting the usage o
+000095c0: 6620 746f 7020 3525 2068 6974 7320 7065  f top 5% hits pe
+000095d0: 7220 6269 6e2e 2054 6865 206e 756d 6265  r bin. The numbe
+000095e0: 7220 6f66 2062 696e 7320 6469 7669 6465  r of bins divide
+000095f0: 6420 696e 2067 656e 6520 6c65 6e67 7468  d in gene length
+00009600: 206e 6f72 6d61 6c69 7a61 7469 6f6e 2c20   normalization, 
+00009610: 6769 7665 6e20 6279 2060 2d62 7360 206f  given by `-bs` o
+00009620: 7220 602d 2d62 696e 7360 2c20 6465 7465  r `--bins`, dete
+00009630: 726d 696e 6573 2074 6865 206e 756d 6265  rmines the numbe
+00009640: 7220 6f66 2062 696e 7320 746f 2063 6174  r of bins to cat
+00009650: 6567 6f72 697a 6520 7468 6520 6765 6e65  egorize the gene
+00009660: 206c 656e 6774 682c 2069 7320 7365 7420   length, is set 
+00009670: 6173 2064 6566 6175 6c74 2031 3030 2e20  as default 100. 
+00009680: 5468 6520 7061 7261 6d65 7465 7220 602d  The parameter `-
+00009690: 6e6e 6020 6f72 2060 2d2d 6e6f 6e6f 726d  nn` or `--nonorm
+000096a0: 616c 697a 6174 696f 6e60 2063 616e 2062  alization` can b
+000096b0: 6520 7365 7420 746f 2063 616c 6c20 6f66  e set to call of
+000096c0: 6620 7468 6520 6e6f 726d 616c 697a 6174  f the normalizat
+000096d0: 696f 6e20 7072 6f63 6573 732c 2061 6c74  ion process, alt
+000096e0: 686f 7567 6820 6974 2773 2073 7567 6765  hough it's sugge
+000096f0: 7374 6564 2074 6f20 636f 6e64 7563 7420  sted to conduct 
+00009700: 7468 6520 6e6f 726d 616c 697a 6174 696f  the normalizatio
+00009710: 6e20 746f 2061 6371 7569 7265 206d 6f72  n to acquire mor
+00009720: 6520 6163 6375 7261 7465 2067 656e 6520  e accurate gene 
+00009730: 6661 6d69 6c79 2063 6c75 7374 6572 696e  family clusterin
+00009740: 6720 7265 7375 6c74 2e20 5468 6520 7061  g result. The pa
+00009750: 7261 6d65 7465 7273 2060 2d2d 746f 5f73  rameters `--to_s
+00009760: 746f 7060 2061 6e64 2060 2d2d 6364 7360  top` and `--cds`
+00009770: 2063 6f6e 7472 6f6c 2074 6865 2062 6568   control the beh
+00009780: 6176 696f 7572 206f 6620 7472 616e 736c  aviour of transl
+00009790: 6174 696e 6720 636f 6469 6e67 2073 6571  ating coding seq
+000097a0: 7565 6e63 6520 696e 746f 2061 6d69 6e6f  uence into amino
+000097b0: 2061 6369 6420 7365 7175 656e 6365 2e20   acid sequence. 
+000097c0: 4966 2074 6865 2060 2d2d 746f 5f73 746f  If the `--to_sto
+000097d0: 7060 2077 6173 2073 6574 2c20 7468 6520  p` was set, the 
+000097e0: 7472 616e 736c 6174 696f 6e20 776f 756c  translation woul
+000097f0: 6420 6265 2074 6572 6d69 6e61 7465 6420  d be terminated 
+00009800: 6174 2074 6865 2066 6972 7374 2069 6e2d  at the first in-
+00009810: 6672 616d 6520 7374 6f70 2063 6f64 6f6e  frame stop codon
+00009820: 2c20 6f74 6865 7277 6973 6520 7468 6520  , otherwise the 
+00009830: 7472 616e 736c 6174 696f 6e20 776f 756c  translation woul
+00009840: 6420 7369 6d70 6c79 2073 6b69 7020 616e  d simply skip an
+00009850: 7920 7374 6f70 2063 6f64 6f6e 732e 2049  y stop codons. I
+00009860: 6620 7468 6520 602d 2d63 6473 6020 7761  f the `--cds` wa
+00009870: 7320 7365 742c 2073 6571 7565 6e63 6573  s set, sequences
+00009880: 2074 6861 7420 646f 6573 6e27 7420 7374   that doesn't st
+00009890: 6172 7420 7769 7468 2061 2076 616c 6964  art with a valid
+000098a0: 2073 7461 7274 2063 6f64 6f6e 2c20 6f72   start codon, or
+000098b0: 2063 6f6e 7461 696e 7320 6d6f 7265 2074   contains more t
+000098c0: 6861 6e20 6f6e 6520 696e 2d66 7261 6d65  han one in-frame
+000098d0: 2073 746f 7020 636f 646f 6e2c 206f 7220   stop codon, or 
+000098e0: 6973 206e 6f74 2064 6976 6964 6162 6c65  is not dividable
+000098f0: 2062 7920 332c 2077 6f75 6c64 2062 6520   by 3, would be 
+00009900: 7369 6d70 6c79 2064 726f 7070 6564 2c20  simply dropped, 
+00009910: 7375 6368 2074 6861 7420 6f6e 6c79 2073  such that only s
+00009920: 7472 6963 7420 636f 6d70 6c65 7465 2063  trict complete c
+00009930: 6f64 696e 6720 7365 7175 656e 6365 7320  oding sequences 
+00009940: 776f 756c 6420 6265 2069 6e63 6c75 6465  would be include
+00009950: 6420 696e 2074 6865 2073 7562 7365 7175  d in the subsequ
+00009960: 656e 7420 616e 616c 7973 6973 2e20 5468  ent analysis. Th
+00009970: 6520 6578 6163 7420 6265 6861 7669 6f75  e exact behaviou
+00009980: 7220 6f66 2060 2d2d 746f 5f73 746f 7060  r of `--to_stop`
+00009990: 2061 6e64 2060 2d2d 6364 7360 2069 7320   and `--cds` is 
+000099a0: 6465 6669 6e65 6420 616e 6420 6465 7363  defined and desc
+000099b0: 7269 6265 6420 696e 2074 6865 205b 6269  ribed in the [bi
+000099c0: 6f70 7974 686f 6e5d 2868 7474 7073 3a2f  opython](https:/
+000099d0: 2f62 696f 7079 7468 6f6e 2e6f 7267 2f64  /biopython.org/d
+000099e0: 6f63 732f 312e 3735 2f61 7069 2f42 696f  ocs/1.75/api/Bio
+000099f0: 2e53 6571 2e68 746d 6c23 4269 6f2e 5365  .Seq.html#Bio.Se
+00009a00: 712e 5365 712e 7472 616e 736c 6174 6529  q.Seq.translate)
+00009a10: 206c 6962 7261 7279 2e20 5468 6520 6e75   library. The nu
+00009a20: 6d62 6572 206f 6620 7061 7261 6c6c 656c  mber of parallel
+00009a30: 2074 6872 6561 6473 2062 7920 7468 6520   threads by the 
+00009a40: 6f70 7469 6f6e 2060 2d6e 6020 6f72 2060  option `-n` or `
+00009a50: 2d2d 6e74 6872 6561 6473 6020 6361 6e20  --nthreads` can 
+00009a60: 6265 2073 6574 2074 6f20 6163 6365 6c65  be set to accele
+00009a70: 7261 7465 2074 6865 2063 616c 6375 6c61  rate the calcula
+00009a80: 7469 6f6e 2077 6974 6869 6e20 6064 6961  tion within `dia
+00009a90: 6d6f 6e64 602e 2054 6865 2064 6972 6563  mond`. The direc
+00009aa0: 746f 7279 206f 6620 6f75 7470 7574 206f  tory of output o
+00009ab0: 7220 696e 7465 726d 6564 6961 7465 2066  r intermediate f
+00009ac0: 696c 6573 2069 7320 6465 7465 726d 696e  iles is determin
+00009ad0: 6564 2062 7920 7468 6520 7061 7261 6d65  ed by the parame
+00009ae0: 7465 7220 602d 6f60 206f 7220 602d 2d6f  ter `-o` or `--o
+00009af0: 7574 6469 7260 2c20 616e 6420 602d 7460  utdir`, and `-t`
+00009b00: 206f 7220 602d 2d74 6d70 6469 7260 2c20   or `--tmpdir`, 
+00009b10: 7768 6963 6820 7769 6c6c 2062 6520 6372  which will be cr
+00009b20: 6561 7465 6420 6279 2074 6865 2070 726f  eated by the pro
+00009b30: 6772 616d 2069 7473 656c 6620 616e 6420  gram itself and 
+00009b40: 6265 206f 7665 7277 7269 7474 656e 2069  be overwritten i
+00009b50: 6620 7468 6520 666f 6c64 6572 2068 6173  f the folder has
+00009b60: 2061 6c72 6561 6479 2062 6565 6e20 6372   already been cr
+00009b70: 6561 7465 642e 204e 6f74 6520 7468 6174  eated. Note that
+00009b80: 2074 6865 2073 6f66 7477 6172 6520 6064   the software `d
+00009b90: 6961 6d6f 6e64 6020 7368 6f75 6c64 2062  iamond` should b
+00009ba0: 6520 7072 652d 696e 7374 616c 6c65 6420  e pre-installed 
+00009bb0: 616e 6420 7365 7420 696e 2074 6865 2065  and set in the e
+00009bc0: 6e76 6972 6f6e 6d65 6e74 2070 6174 6820  nvironment path 
+00009bd0: 696e 2061 6c6c 2074 6865 2061 6e61 6c79  in all the analy
+00009be0: 7369 7320 7065 7266 6f72 6d65 6420 6279  sis performed by
+00009bf0: 2060 7767 6420 646d 6460 2065 7863 6570   `wgd dmd` excep
+00009c00: 7420 666f 7220 7468 6520 636f 6c6c 696e  t for the collin
+00009c10: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
+00009c20: 616e 616c 7973 6973 2e0a 0a2a 2a57 6520  analysis...**We 
+00009c30: 7375 6767 6573 7420 7468 6174 2074 6865  suggest that the
+00009c40: 2064 6566 6175 6c74 2073 6574 7469 6e67   default setting
+00009c50: 2069 6e20 7768 6963 6820 7468 6520 696e   in which the in
+00009c60: 666c 6174 696f 6e20 6661 6374 6f72 2069  flation factor i
+00009c70: 7320 7365 7420 6173 2032 2c20 652d 7661  s set as 2, e-va
+00009c80: 6c75 6520 6375 742d 6f66 6620 6173 2031  lue cut-off as 1
+00009c90: 652d 3130 2061 6e64 206f 7468 6572 2070  e-10 and other p
+00009ca0: 6172 616d 6574 6572 7320 696e 2064 6566  arameters in def
+00009cb0: 6175 6c74 2069 7320 6120 676f 6f64 2073  ault is a good s
+00009cc0: 7461 7274 696e 6720 706f 696e 742c 2075  tarting point, u
+00009cd0: 6e6c 6573 7320 796f 7520 7370 6563 6966  nless you specif
+00009ce0: 6963 616c 6c79 2077 616e 7420 746f 2065  ically want to e
+00009cf0: 7870 6c6f 7265 2074 6865 2065 6666 6563  xplore the effec
+00009d00: 7473 206f 6620 6469 6666 6572 656e 7420  ts of different 
+00009d10: 7061 7261 6d65 7465 7273 2e2a 2a20 5375  parameters.** Su
+00009d20: 6368 2074 6861 7420 7468 6520 636f 6d6d  ch that the comm
+00009d30: 616e 6420 666f 7220 7468 6520 6465 6c69  and for the deli
+00009d40: 6e65 6174 696f 6e20 6f66 2077 686f 6c65  neation of whole
+00009d50: 2070 6172 616e 6f6d 6520 6973 2073 696d   paranome is sim
+00009d60: 706c 7920 6173 2062 656c 6f77 2e0a 0a60  ply as below...`
+00009d70: 6060 0a77 6764 2064 6d64 2041 7175 696c  ``.wgd dmd Aquil
+00009d80: 6567 6961 5f63 6f65 7275 6c65 610a 6060  egia_coerulea.``
+00009d90: 600a 0a2a 2a54 6865 2064 656c 696e 6561  `..**The delinea
+00009da0: 7469 6f6e 206f 6620 5242 4873 2a2a 0a60  tion of RBHs**.`
+00009db0: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
+00009dc0: 6e63 6531 2073 6571 7565 6e63 6532 202d  nce1 sequence2 -
+00009dd0: 6520 3165 2d31 3020 2d62 7320 3130 3020  e 1e-10 -bs 100 
+00009de0: 2d6e 7020 3520 282d 6e6e 2920 282d 6e20  -np 5 (-nn) (-n 
+00009df0: 3429 2028 2d63 2030 2e39 2920 282d 2d6f  4) (-c 0.9) (--o
+00009e00: 6766 6f72 6d61 7429 2028 2d2d 746f 5f73  gformat) (--to_s
+00009e10: 746f 7029 2028 2d2d 6364 7329 2028 2d6f  top) (--cds) (-o
+00009e20: 2077 6764 5f64 6d64 2920 282d 7420 776f   wgd_dmd) (-t wo
+00009e30: 726b 696e 675f 746d 7029 0a60 6060 0a0a  rking_tmp).```..
+00009e40: 546f 2064 656c 696e 6561 7465 2052 4248  To delineate RBH
+00009e50: 7320 6265 7477 6565 6e20 7477 6f20 6364  s between two cd
+00009e60: 7320 7365 7175 656e 6365 2066 696c 6573  s sequence files
+00009e70: 2c20 7468 6520 7265 6c65 7661 6e74 2070  , the relevant p
+00009e80: 6172 616d 6574 6572 2069 7320 6d6f 7374  arameter is most
+00009e90: 6c79 2074 6865 2073 616d 6520 6173 2077  ly the same as w
+00009ea0: 686f 6c65 2070 6172 616e 6f6d 6520 696e  hole paranome in
+00009eb0: 6665 7265 6e63 652c 2065 7863 6570 7420  ference, except 
+00009ec0: 666f 7220 7468 6520 7061 7261 6d65 7465  for the paramete
+00009ed0: 7220 602d 6360 206f 7220 602d 2d63 7363  r `-c` or `--csc
+00009ee0: 6f72 6560 2c20 7768 6963 6820 7261 6e67  ore`, which rang
+00009ef0: 6573 2062 6574 7765 656e 2030 2061 6e64  es between 0 and
+00009f00: 2031 2061 6e64 2069 7320 7573 6564 2074   1 and is used t
+00009f10: 6f20 7265 6c61 7820 7468 6520 7369 6d69  o relax the simi
+00009f20: 6c61 7269 7479 2063 7574 6f66 6620 6672  larity cutoff fr
+00009f30: 6f6d 2074 6865 2065 7863 6c75 7369 7665  om the exclusive
+00009f40: 2072 6563 6970 726f 6361 6c20 6265 7374   reciprocal best
+00009f50: 2068 6974 7320 746f 2061 2063 6572 7461   hits to a certa
+00009f60: 696e 2072 6174 696f 2061 7320 746f 2074  in ratio as to t
+00009f70: 6865 2062 6573 7420 6869 7473 2e20 466f  he best hits. Fo
+00009f80: 7220 696e 7374 616e 6365 2c20 6966 2074  r instance, if t
+00009f90: 6865 2067 656e 6520 6231 2066 726f 6d20  he gene b1 from 
+00009fa0: 6765 6e6f 6d65 2042 2068 6173 2074 6865  genome B has the
+00009fb0: 2062 6573 7420 6869 7420 6765 6e65 2061   best hit gene a
+00009fc0: 3120 6672 6f6d 2067 656e 6f6d 6520 4120  1 from genome A 
+00009fd0: 7769 7468 2074 6865 2062 6974 2073 636f  with the bit sco
+00009fe0: 7265 2061 7320 3130 302c 2077 6869 6368  re as 100, which
+00009ff0: 2069 7320 6120 7363 6f72 696e 6720 6d61   is a scoring ma
+0000a000: 7472 6978 2069 6e64 6570 656e 6465 6e74  trix independent
+0000a010: 206d 6561 7375 7265 206f 6620 7468 6520   measure of the 
+0000a020: 286c 6f63 616c 2920 7369 6d69 6c61 7269  (local) similari
+0000a030: 7479 206f 6620 7468 6520 7477 6f20 616c  ty of the two al
+0000a040: 6967 6e65 6420 7365 7175 656e 6365 732c  igned sequences,
+0000a050: 2077 6974 6820 6c61 7267 6572 2076 616c   with larger val
+0000a060: 7565 7320 696e 6469 6361 7469 6e67 2068  ues indicating h
+0000a070: 6967 6865 7220 7369 6d69 6c61 7269 7469  igher similariti
+0000a080: 6573 2c20 6769 7665 6e20 7468 6520 602d  es, given the `-
+0000a090: 6320 302e 3960 2c20 6765 6e65 7320 6672  c 0.9`, genes fr
+0000a0a0: 6f6d 2067 656e 6f6d 6520 4120 7768 6963  om genome A whic
+0000a0b0: 6820 6861 7320 7468 6520 6269 7420 7363  h has the bit sc
+0000a0c0: 6f72 6520 7769 7468 2067 656e 6520 6231  ore with gene b1
+0000a0d0: 2068 6967 6865 7220 7468 616e 2030 2e39   higher than 0.9
+0000a0e0: 7831 3030 2077 696c 6c20 616c 736f 2062  x100 will also b
+0000a0f0: 6520 7772 6974 7465 6e20 696e 2074 6865  e written in the
+0000a100: 2072 6573 756c 7420 6669 6c65 2c20 7768   result file, wh
+0000a110: 6963 6820 696e 2061 2073 656e 7365 2061  ich in a sense a
+0000a120: 7265 206e 6f74 2052 4248 7320 616e 796d  re not RBHs anym
+0000a130: 6f72 6520 6f66 2063 6f75 7273 652c 2062  ore of course, b
+0000a140: 7574 2074 6865 2068 6967 686c 7920 7369  ut the highly si
+0000a150: 6d69 6c61 7220 686f 6d6f 6c6f 6775 6520  milar homologue 
+0000a160: 7061 6972 732e 2049 6620 6d6f 7265 2074  pairs. If more t
+0000a170: 6861 6e20 3220 7365 7175 656e 6365 2066  han 2 sequence f
+0000a180: 696c 6573 2077 6572 6520 7072 6f76 6964  iles were provid
+0000a190: 6564 2c20 6576 6572 7920 7061 6972 2d77  ed, every pair-w
+0000a1a0: 6973 6520 5242 4873 2077 6f75 6c64 2062  ise RBHs would b
+0000a1b0: 6520 6361 6c63 756c 6174 6564 2065 7863  e calculated exc
+0000a1c0: 6570 7420 666f 7220 7175 6572 7969 6e67  ept for querying
+0000a1d0: 2074 6865 2073 616d 6520 7365 7175 656e   the same sequen
+0000a1e0: 6365 2069 7473 656c 662e 2054 6865 206e  ce itself. The n
+0000a1f0: 756d 6265 7220 6f66 2070 6172 616c 6c65  umber of paralle
+0000a200: 6c20 7468 7265 6164 7320 746f 2062 6f6f  l threads to boo
+0000a210: 7374 6572 2074 6865 2072 756e 6e69 6e67  ster the running
+0000a220: 2073 7065 6564 2063 616e 2062 6520 7365   speed can be se
+0000a230: 7420 6279 2074 6865 206f 7074 696f 6e20  t by the option 
+0000a240: 602d 6e60 206f 7220 602d 2d6e 7468 7265  `-n` or `--nthre
+0000a250: 6164 7360 2077 6869 6368 2069 7320 7375  ads` which is su
+0000a260: 6767 6573 7465 6420 746f 2062 6520 7365  ggested to be se
+0000a270: 7420 6173 2028 4e2d 3129 4e2f 3220 7768  t as (N-1)N/2 wh
+0000a280: 6572 6520 4e20 6973 2074 6865 206e 756d  ere N is the num
+0000a290: 6265 7220 6f66 2063 6473 2066 696c 6573  ber of cds files
+0000a2a0: 2074 6f20 6163 6869 6576 6520 7468 6520   to achieve the 
+0000a2b0: 6869 6768 6573 7420 6566 6669 6369 656e  highest efficien
+0000a2c0: 6379 2e20 5468 6520 6f70 7469 6f6e 2060  cy. The option `
+0000a2d0: 2d2d 6f67 666f 726d 6174 6020 6361 6e20  --ogformat` can 
+0000a2e0: 6265 2073 6574 2074 6f20 6164 6420 696e  be set to add in
+0000a2f0: 6465 7820 2866 6f72 2069 6e73 7461 6e63  dex (for instanc
+0000a300: 6520 4746 3030 3030 3030 3031 2920 746f  e GF00000001) to
+0000a310: 2074 6865 206f 7574 7075 7420 5242 4820   the output RBH 
+0000a320: 6765 6e65 2066 616d 696c 6965 7320 7768  gene families wh
+0000a330: 6963 6820 6361 6e20 6265 2066 7572 7468  ich can be furth
+0000a340: 6572 2075 7365 6420 696e 2074 6865 202a  er used in the *
+0000a350: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2063  K*<sub>S</sub> c
+0000a360: 616c 6375 6c61 7469 6f6e 2062 7920 6077  alculation by `w
+0000a370: 6764 206b 7364 602e 0a0a 2a2a 5468 6520  gd ksd`...**The 
+0000a380: 7375 6767 6573 7465 6420 636f 6d6d 616e  suggested comman
+0000a390: 6420 746f 2073 7461 7274 2077 6974 6820  d to start with 
+0000a3a0: 6973 2061 6c73 6f20 756e 6465 7220 7468  is also under th
+0000a3b0: 6520 6465 6661 756c 7420 7365 7474 696e  e default settin
+0000a3c0: 6720 7769 7468 2074 6865 2063 6f6d 6d61  g with the comma
+0000a3d0: 6e64 2073 686f 776e 2062 656c 6f77 2a2a  nd shown below**
+0000a3e0: 0a0a 6060 600a 7767 6420 646d 6420 7365  ..```.wgd dmd se
+0000a3f0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
+0000a400: 320a 6060 600a 0a2a 2a54 6865 2064 656c  2.```..**The del
+0000a410: 696e 6561 7469 6f6e 206f 6620 6c6f 6361  ineation of loca
+0000a420: 6c20 4d52 4248 732a 2a0a 6060 600a 7767  l MRBHs**.```.wg
+0000a430: 6420 646d 6420 7365 7175 656e 6365 3120  d dmd sequence1 
+0000a440: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
+0000a450: 6365 3320 2d66 2073 6571 7565 6e63 6531  ce3 -f sequence1
+0000a460: 202d 6520 3165 2d31 3020 2d62 7320 3130   -e 1e-10 -bs 10
+0000a470: 3020 2d6e 7020 3520 282d 6e6e 2920 282d  0 -np 5 (-nn) (-
+0000a480: 6e20 3429 2028 2d6b 6629 2028 2d6b 6429  n 4) (-kf) (-kd)
+0000a490: 2028 2d63 2030 2e39 2920 282d 2d74 6f5f   (-c 0.9) (--to_
+0000a4a0: 7374 6f70 2920 282d 2d63 6473 2920 282d  stop) (--cds) (-
+0000a4b0: 6f20 7767 645f 646d 6429 2028 2d74 2077  o wgd_dmd) (-t w
+0000a4c0: 6f72 6b69 6e67 5f74 6d70 290a 6060 600a  orking_tmp).```.
+0000a4d0: 0a2a 2a54 6865 2064 6973 7469 6e63 7469  .**The distincti
+0000a4e0: 6f6e 2062 6574 7765 656e 206c 6f63 616c  on between local
+0000a4f0: 2061 6e64 2067 6c6f 6261 6c20 2868 6572   and global (her
+0000a500: 6575 6e64 6572 2920 4d52 4248 7320 6973  eunder) MRBHs is
+0000a510: 2074 6861 7420 6c6f 6361 6c20 4d52 4248   that local MRBH
+0000a520: 7320 6172 6520 7468 6520 7265 7375 6c74  s are the result
+0000a530: 7320 6f66 206d 6572 6765 6420 5242 4873  s of merged RBHs
+0000a540: 206f 6e20 6120 6a6f 696e 7420 666f 6361   on a joint foca
+0000a550: 6c20 7370 6563 6965 732c 2066 6f72 2069  l species, for i
+0000a560: 6e73 7461 6e63 6520 696e 2061 2074 6872  nstance in a thr
+0000a570: 6565 2073 7065 6369 6573 2073 7973 7465  ee species syste
+0000a580: 6d20 2841 2c28 422c 4329 292c 2074 6865  m (A,(B,C)), the
+0000a590: 206c 6f63 616c 204d 5242 4873 206f 6620   local MRBHs of 
+0000a5a0: 4120 6f6e 6c79 2072 6571 7569 7265 2074  A only require t
+0000a5b0: 6865 2063 616c 6375 6c61 7469 6f6e 206f  he calculation o
+0000a5c0: 6620 5242 4873 2062 6574 7765 656e 2041  f RBHs between A
+0000a5d0: 2061 6e64 2043 2028 6465 6e6f 7465 6420   and C (denoted 
+0000a5e0: 6173 2041 4329 2061 6e64 2041 4220 616e  as AC) and AB an
+0000a5f0: 6420 7468 656e 2074 6865 206d 6572 6769  d then the mergi
+0000a600: 6e67 206f 6620 4142 2061 6e64 2041 4320  ng of AB and AC 
+0000a610: 6174 2074 6865 2061 7869 7320 6f66 2041  at the axis of A
+0000a620: 2c20 7768 696c 6520 7468 6520 676c 6f61  , while the gloa
+0000a630: 626c 204d 5242 4873 2061 7265 2069 6e64  bl MRBHs are ind
+0000a640: 6570 656e 6465 6e74 206f 6620 666f 6361  ependent of foca
+0000a650: 6c20 7370 6563 6965 7320 696e 2074 6861  l species in tha
+0000a660: 7420 6974 206a 7573 7420 6361 6c6c 7320  t it just calls 
+0000a670: 7468 6520 6361 6c63 756c 6174 696f 6e73  the calculations
+0000a680: 206f 6620 616c 6c20 706f 7373 6962 6c65   of all possible
+0000a690: 2073 7065 6369 6573 2070 6169 7220 286e   species pair (n
+0000a6a0: 6f74 2073 656c 6620 746f 2073 656c 6620  ot self to self 
+0000a6b0: 7370 6563 6965 7320 7061 6972 292c 2073  species pair), s
+0000a6c0: 7563 6820 7468 6174 2041 422c 2041 432c  uch that AB, AC,
+0000a6d0: 2061 6e64 2042 4320 6172 6520 616c 6c20   and BC are all 
+0000a6e0: 746f 2062 6520 6361 6c63 756c 6174 6564  to be calculated
+0000a6f0: 2061 6e64 206d 6572 6765 642e 2a2a 0a0a   and merged.**..
+0000a700: 5477 6f20 7479 7065 7320 6f66 204d 5242  Two types of MRB
+0000a710: 4873 2061 7320 696e 7465 7072 6574 6174  Hs as intepretat
+0000a720: 6564 2061 626f 7665 2063 616e 2062 6520  ed above can be 
+0000a730: 6465 6c69 6e65 6174 6564 2062 7920 6077  delineated by `w
+0000a740: 6764 2064 6d64 602c 2074 6865 206c 6f63  gd dmd`, the loc
+0000a750: 616c 204d 5242 4873 2061 6e64 2074 6865  al MRBHs and the
+0000a760: 2067 6c6f 6261 6c20 4d52 4248 732e 2054   global MRBHs. T
+0000a770: 6865 206c 6f63 616c 204d 5242 4873 2061  he local MRBHs a
+0000a780: 7265 2063 6f6e 7374 7275 6374 6564 2062  re constructed b
+0000a790: 7920 6d65 7267 696e 6720 616c 6c20 7468  y merging all th
+0000a7a0: 6520 7265 6c65 7661 6e74 2052 4248 7320  e relevant RBHs 
+0000a7b0: 6f6e 6c79 2077 6974 6820 7468 6520 666f  only with the fo
+0000a7c0: 6361 6c20 7370 6563 6965 732c 2077 6869  cal species, whi
+0000a7d0: 6368 2069 7320 7365 7420 6279 2060 2d66  ch is set by `-f
+0000a7e0: 6020 6f72 2060 2d2d 666f 6375 7360 2e20  ` or `--focus`. 
+0000a7f0: 5468 6520 7061 7261 6d65 7465 7220 602d  The parameter `-
+0000a800: 6b66 6020 6f72 2060 2d2d 6b65 6570 6661  kf` or `--keepfa
+0000a810: 7374 6160 2063 616e 2062 6520 7365 7420  sta` can be set 
+0000a820: 746f 2072 6574 6169 6e20 7468 6520 7365  to retain the se
+0000a830: 7175 656e 6365 2069 6e66 6f72 6d61 7469  quence informati
+0000a840: 6f6e 206f 6620 6561 6368 204d 5242 482e  on of each MRBH.
+0000a850: 2054 6865 2070 6172 616d 6574 6572 2060   The parameter `
+0000a860: 2d6b 6460 206f 7220 602d 2d6b 6565 7064  -kd` or `--keepd
+0000a870: 7570 6c69 6361 7465 7360 2064 6574 6572  uplicates` deter
+0000a880: 6d69 6e65 7320 7768 6574 6865 7220 7468  mines whether th
+0000a890: 6520 7361 6d65 2067 656e 6573 2063 616e  e same genes can
+0000a8a0: 2061 7070 6561 7220 696e 2064 6966 6665   appear in diffe
+0000a8b0: 7265 6e74 206c 6f63 616c 204d 5242 4873  rent local MRBHs
+0000a8c0: 2e20 4e6f 726d 616c 6c79 2074 6865 7265  . Normally there
+0000a8d0: 2077 696c 6c20 6265 206e 6f20 6475 706c   will be no dupl
+0000a8e0: 6963 6174 6573 2069 6e20 7468 6520 6c6f  icates in the lo
+0000a8f0: 6361 6c20 4d52 4248 7320 6275 7420 6966  cal MRBHs but if
+0000a900: 2075 7365 7273 2073 6574 2074 6865 2060   users set the `
+0000a910: 2d63 6020 6173 2030 2e39 2028 6f72 2061  -c` as 0.9 (or a
+0000a920: 6e79 2076 616c 7565 2073 6d61 6c6c 6572  ny value smaller
+0000a930: 2074 6861 6e20 3129 2c20 6974 2773 206c   than 1), it's l
+0000a940: 696b 656c 7920 7468 6174 2074 6865 2073  ikely that the s
+0000a950: 616d 6520 6765 6e65 2077 696c 6c20 6861  ame gene will ha
+0000a960: 7665 2063 6861 6e63 6520 746f 2061 7070  ve chance to app
+0000a970: 6561 7220 6d75 6c74 6970 6c65 2074 696d  ear multiple tim
+0000a980: 6573 2069 6e20 6469 6666 6572 656e 7420  es in different 
+0000a990: 6c6f 6361 6c20 4d52 4248 732e 2054 6861  local MRBHs. Tha
+0000a9a0: 7420 6973 2074 6f20 7361 792c 2074 6865  t is to say, the
+0000a9b0: 2070 6172 616d 6574 6572 2060 2d6b 6460   parameter `-kd`
+0000a9c0: 2069 7320 6d65 616e 696e 6766 756c 206f   is meaningful o
+0000a9d0: 6e6c 7920 7768 656e 2069 7427 7320 7365  nly when it's se
+0000a9e0: 7420 746f 6765 7468 6572 2077 6974 6820  t together with 
+0000a9f0: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
+0000aa00: 6360 2e20 5468 6520 6e75 6d62 6572 206f  c`. The number o
+0000aa10: 6620 7061 7261 6c6c 656c 2074 6872 6561  f parallel threa
+0000aa20: 6473 2069 7320 7375 6767 6573 7465 6420  ds is suggested 
+0000aa30: 746f 2062 6520 7365 7420 6173 2074 6865  to be set as the
+0000aa40: 206e 756d 6265 7220 6f66 2063 6473 2066   number of cds f
+0000aa50: 696c 6573 206d 696e 7573 2031 2e0a 0a2a  iles minus 1...*
+0000aa60: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
+0000aa70: 7274 696e 6720 7275 6e20 6973 2075 6e64  rting run is und
+0000aa80: 6572 2074 6865 2064 6566 6175 6c74 2070  er the default p
+0000aa90: 6172 616d 6574 6572 2077 6974 6820 7468  arameter with th
+0000aaa0: 6520 636f 6d6d 616e 6420 7368 6f77 6e20  e command shown 
+0000aab0: 6173 2062 656c 6f77 2e2a 2a0a 0a60 6060  as below.**..```
+0000aac0: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
+0000aad0: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
+0000aae0: 7565 6e63 6533 202d 6620 7365 7175 656e  uence3 -f sequen
+0000aaf0: 6365 310a 6060 600a 0a2a 2a54 6865 2064  ce1.```..**The d
+0000ab00: 656c 696e 6561 7469 6f6e 206f 6620 676c  elineation of gl
+0000ab10: 6f62 616c 204d 5242 4873 2a2a 0a60 6060  obal MRBHs**.```
+0000ab20: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
+0000ab30: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
+0000ab40: 7565 6e63 6533 202d 676d 202d 6520 3165  uence3 -gm -e 1e
+0000ab50: 2d31 3020 2d62 7320 3130 3020 2d6e 7020  -10 -bs 100 -np 
+0000ab60: 3520 282d 6e6e 2920 282d 6e20 3429 2028  5 (-nn) (-n 4) (
+0000ab70: 2d6b 6629 2028 2d6b 6429 2028 2d63 2030  -kf) (-kd) (-c 0
+0000ab80: 2e39 2920 282d 2d74 6f5f 7374 6f70 2920  .9) (--to_stop) 
+0000ab90: 282d 2d63 6473 2920 282d 6f20 7767 645f  (--cds) (-o wgd_
+0000aba0: 646d 6429 2028 2d74 2077 6f72 6b69 6e67  dmd) (-t working
+0000abb0: 5f74 6d70 290a 6060 600a 0a54 6865 2067  _tmp).```..The g
+0000abc0: 6c6f 6261 6c20 4d52 4248 7320 6973 2063  lobal MRBHs is c
+0000abd0: 6f6e 7374 7275 6374 6564 2062 7920 6578  onstructed by ex
+0000abe0: 6861 7573 7469 7665 6c79 206d 6572 6769  haustively mergi
+0000abf0: 6e67 2061 6c6c 2074 6865 2070 6f73 7369  ng all the possi
+0000ac00: 626c 6520 7061 6972 2d77 6973 6520 5242  ble pair-wise RB
+0000ac10: 4873 2065 7863 6570 7420 666f 7220 7175  Hs except for qu
+0000ac20: 6572 7969 6e67 2074 6865 2073 6571 7565  erying the seque
+0000ac30: 6e63 6520 6974 7365 6c66 2c20 7768 6963  nce itself, whic
+0000ac40: 6820 6361 6e20 6265 2069 6e69 7469 6174  h can be initiat
+0000ac50: 6564 2062 7920 6164 6420 7468 6520 666c  ed by add the fl
+0000ac60: 6167 2060 2d67 6d60 206f 7220 602d 2d67  ag `-gm` or `--g
+0000ac70: 6c6f 6261 6c6d 7262 6860 2e20 5468 6520  lobalmrbh`. The 
+0000ac80: 7265 7374 206f 6620 7265 6c65 7661 6e74  rest of relevant
+0000ac90: 2070 6172 616d 6574 6572 7320 7374 6179   parameters stay
+0000aca0: 7320 7468 6520 7361 6d65 2061 7320 7468  s the same as th
+0000acb0: 6520 6c6f 6361 6c20 4d52 4248 732e 2054  e local MRBHs. T
+0000acc0: 6865 206e 756d 6265 7220 6f66 2070 6172  he number of par
+0000acd0: 616c 6c65 6c20 7468 7265 6164 7320 6973  allel threads is
+0000ace0: 2073 7567 6765 7374 6564 2074 6f20 6265   suggested to be
+0000acf0: 2073 6574 2061 7320 284e 2d31 294e 2f32   set as (N-1)N/2
+0000ad00: 2074 6f6f 2077 6865 7265 204e 2069 7320   too where N is 
+0000ad10: 7468 6520 6e75 6d62 6572 206f 6620 6364  the number of cd
+0000ad20: 7320 6669 6c65 7320 746f 2061 6368 6965  s files to achie
+0000ad30: 7665 2074 6865 2068 6967 6865 7374 2065  ve the highest e
+0000ad40: 6666 6963 6965 6e63 792e 0a0a 2a2a 4120  fficiency...**A 
+0000ad50: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
+0000ad60: 6e67 2072 756e 2069 7320 756e 6465 7220  ng run is under 
+0000ad70: 7468 6520 6465 6661 756c 7420 7061 7261  the default para
+0000ad80: 6d65 7465 7220 7769 7468 2074 6865 2063  meter with the c
+0000ad90: 6f6d 6d61 6e64 2073 686f 776e 2061 7320  ommand shown as 
+0000ada0: 6265 6c6f 772e 2a2a 0a0a 6060 600a 7767  below.**..```.wg
+0000adb0: 6420 646d 6420 7365 7175 656e 6365 3120  d dmd sequence1 
+0000adc0: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
+0000add0: 6365 3320 2d67 6d0a 6060 600a 0a2a 2a54  ce3 -gm.```..**T
+0000ade0: 6865 2064 656c 696e 6561 7469 6f6e 206f  he delineation o
+0000adf0: 6620 6f72 7468 6f67 726f 7570 732a 2a0a  f orthogroups**.
+0000ae00: 6060 600a 7767 6420 646d 6420 7365 7175  ```.wgd dmd sequ
+0000ae10: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
+0000ae20: 7365 7175 656e 6365 3320 2d6f 6920 2d6f  sequence3 -oi -o
+0000ae30: 6f20 2d65 2031 652d 3130 202d 6273 2031  o -e 1e-10 -bs 1
+0000ae40: 3030 202d 6e70 2035 2028 2d6e 6e29 2028  00 -np 5 (-nn) (
+0000ae50: 2d63 6329 2028 2d74 6529 2028 2d6d 6320  -cc) (-te) (-mc 
+0000ae60: 302e 3829 2028 2d67 6e29 2028 2d74 7265  0.8) (-gn) (-tre
+0000ae70: 6520 2766 6173 7474 7265 6527 2920 282d  e 'fasttree') (-
+0000ae80: 7473 2027 2d66 6173 7465 7374 2729 2028  ts '-fastest') (
+0000ae90: 2d6e 2034 2920 282d 2d74 6f5f 7374 6f70  -n 4) (--to_stop
+0000aea0: 2920 282d 2d63 6473 2920 282d 6f20 7767  ) (--cds) (-o wg
+0000aeb0: 645f 646d 6429 2028 2d74 2077 6f72 6b69  d_dmd) (-t worki
+0000aec0: 6e67 5f74 6d70 290a 6060 600a 0a49 6e20  ng_tmp).```..In 
+0000aed0: 6077 6764 2076 3260 2c20 7765 2061 6c73  `wgd v2`, we als
+0000aee0: 6f20 696d 706c 656d 656e 7465 6420 616e  o implemented an
+0000aef0: 2061 6c67 6f72 6974 686d 206f 6620 6465   algorithm of de
+0000af00: 6c69 6e65 6174 696e 6720 6f72 7468 6f67  lineating orthog
+0000af10: 726f 7570 732c 2077 6869 6368 2063 616e  roups, which can
+0000af20: 2062 6520 696e 6974 6961 7465 6420 7769   be initiated wi
+0000af30: 7468 2074 6865 2070 6172 616d 6574 6572  th the parameter
+0000af40: 2060 2d6f 6960 206f 7220 602d 2d6f 7274   `-oi` or `--ort
+0000af50: 686f 696e 6665 7260 2e20 5477 6f20 7761  hoinfer`. Two wa
+0000af60: 7973 206f 6620 6465 6c69 6e65 6174 696f  ys of delineatio
+0000af70: 6e20 6361 6e20 6265 2063 686f 7365 6e2c  n can be chosen,
+0000af80: 2074 6865 2063 6f6e 6361 7465 6e61 7469   the concatenati
+0000af90: 6f6e 2077 6179 2028 7365 7420 6279 2074  on way (set by t
+0000afa0: 6865 2070 6172 616d 6574 6572 2060 2d63  he parameter `-c
+0000afb0: 6360 206f 7220 602d 2d63 6f6e 6361 7460  c` or `--concat`
+0000afc0: 2920 6f72 2074 6865 206e 6f6e 2d63 6f6e  ) or the non-con
+0000afd0: 6361 7465 6e61 7469 6f6e 2028 6465 6661  catenation (defa
+0000afe0: 756c 7429 2077 6179 2e20 496e 2062 7269  ult) way. In bri
+0000aff0: 6566 2c20 7468 6520 636f 6e63 6174 656e  ef, the concaten
+0000b000: 6174 696f 6e20 7761 7920 6f66 2064 656c  ation way of del
+0000b010: 696e 6561 7469 6e67 206f 7274 686f 6772  ineating orthogr
+0000b020: 6f75 7073 2073 7461 7274 7320 7769 7468  oups starts with
+0000b030: 2063 6f6e 6361 7465 6e61 7469 6e67 2061   concatenating a
+0000b040: 6c6c 2074 6865 2073 6571 7565 6e63 6573  ll the sequences
+0000b050: 2069 6e74 6f20 6120 7369 6e67 6c65 2073   into a single s
+0000b060: 6571 7565 6e63 6520 6669 6c65 2061 6e64  equence file and
+0000b070: 2074 6865 6e20 696e 6665 7272 696e 6720   then inferring 
+0000b080: 7468 6520 7768 6f6c 6520 7061 7261 6e6f  the whole parano
+0000b090: 6d65 206f 6620 7468 6973 2073 696e 676c  me of this singl
+0000b0a0: 6520 7365 7175 656e 6365 2066 696c 6520  e sequence file 
+0000b0b0: 7769 7468 2074 6865 2063 6c75 7374 6572  with the cluster
+0000b0c0: 696e 6720 7265 7375 6c74 7320 6d61 7070  ing results mapp
+0000b0d0: 6564 2062 6163 6b20 746f 2074 6865 2062  ed back to the b
+0000b0e0: 656c 6f6e 6769 6e67 2073 7065 6369 6573  elonging species
+0000b0f0: 2e20 5768 696c 6520 7468 6520 6e6f 6e2d  . While the non-
+0000b100: 636f 6e63 6174 656e 6174 696f 6e20 7761  concatenation wa
+0000b110: 7920 7374 6172 7473 2077 6974 6820 7265  y starts with re
+0000b120: 7370 6563 7469 7665 2070 6169 722d 7769  spective pair-wi
+0000b130: 7365 2064 6961 6d6f 6e64 2073 6561 7263  se diamond searc
+0000b140: 6820 2869 6e63 6c75 6469 6e67 2071 7565  h (including que
+0000b150: 7279 696e 6720 7468 6520 7361 6d65 2073  rying the same s
+0000b160: 6571 7565 6e63 6520 6974 7365 6c66 2920  equence itself) 
+0000b170: 616e 6420 7468 656e 2061 6c6c 2074 6865  and then all the
+0000b180: 2073 6571 7565 6e63 6520 7369 6d69 6c61   sequence simila
+0000b190: 7269 7479 2074 6162 6c65 7320 7769 6c6c  rity tables will
+0000b1a0: 2062 6520 636f 6e63 6174 656e 6174 6564   be concatenated
+0000b1b0: 2061 6e64 2063 6c75 7374 6572 6564 2069   and clustered i
+0000b1c0: 6e74 6f20 6f72 7468 6f67 726f 7570 732e  nto orthogroups.
+0000b1d0: 2053 6f6d 6520 6f74 6865 7220 706f 7373   Some other poss
+0000b1e0: 6962 6c79 2075 7365 6675 6c20 706f 7374  ibly useful post
+0000b1f0: 2d63 6c75 7374 6572 696e 6720 6675 6e63  -clustering func
+0000b200: 7469 6f6e 7320 6361 6e20 6265 2069 6e69  tions can be ini
+0000b210: 7469 6174 6564 2c20 696e 636c 7564 696e  tiated, includin
+0000b220: 6720 7468 6520 7061 7261 6d65 7465 7220  g the parameter 
+0000b230: 602d 7465 6020 6f72 2060 2d2d 7465 7374  `-te` or `--test
+0000b240: 736f 6760 2c20 7768 6963 6820 6361 6e20  sog`, which can 
+0000b250: 6265 2073 6574 2074 6f20 7374 6172 7420  be set to start 
+0000b260: 7468 6520 756e 6269 6173 6564 2074 6573  the unbiased tes
+0000b270: 7420 6f66 2073 696e 676c 652d 636f 7079  t of single-copy
+0000b280: 2067 656e 6520 6661 6d69 6c69 6573 2028   gene families (
+0000b290: 6e6f 7465 2074 6861 7420 7468 6973 2066  note that this f
+0000b2a0: 756e 6374 696f 6e20 6e65 6564 7320 6068  unction needs `h
+0000b2b0: 6d6d 6572 6020 2876 332e 3162 3229 2074  mmer` (v3.1b2) t
+0000b2c0: 6f20 6265 2069 6e73 7461 6c6c 6564 2069  o be installed i
+0000b2d0: 6e20 7468 6520 656e 7669 726f 6e6d 656e  n the environmen
+0000b2e0: 7420 7061 7468 292c 2074 6865 2070 6172  t path), the par
+0000b2f0: 616d 6574 6572 2060 2d6d 6360 206f 7220  ameter `-mc` or 
+0000b300: 602d 2d6d 736f 6763 7574 602c 2072 616e  `--msogcut`, ran
+0000b310: 6769 6e67 2062 6574 7765 656e 2030 2074  ging between 0 t
+0000b320: 6f20 312c 2077 6869 6368 2063 616e 2062  o 1, which can b
+0000b330: 6520 7365 7420 746f 2073 6561 7263 6820  e set to search 
+0000b340: 7468 6520 736f 2d63 616c 6c65 6420 6d6f  the so-called mo
+0000b350: 7374 6c79 2073 696e 676c 652d 636f 7079  stly single-copy
+0000b360: 2066 616d 696c 7920 7768 6963 6820 6861   family which ha
+0000b370: 7320 6869 6768 6572 2074 6861 6e20 6365  s higher than ce
+0000b380: 7274 6169 6e20 6375 742d 6f66 6620 7065  rtain cut-off pe
+0000b390: 7263 656e 7461 6765 206f 6620 7370 6563  rcentage of spec
+0000b3a0: 6965 7320 636f 7665 7261 6765 2c20 7468  ies coverage, th
+0000b3b0: 6520 7061 7261 6d65 7465 7220 602d 676e  e parameter `-gn
+0000b3c0: 6020 6f72 2060 2d2d 6765 746e 736f 6760  ` or `--getnsog`
+0000b3d0: 2c20 7768 6963 6820 6361 6e20 6265 2073  , which can be s
+0000b3e0: 6574 2074 6f20 7365 6172 6368 2066 6f72  et to search for
+0000b3f0: 206e 6573 7465 6420 7369 6e67 6c65 2d63   nested single-c
+0000b400: 6f70 7920 6765 6e65 2066 616d 696c 6965  opy gene familie
+0000b410: 7320 284e 534f 4773 2920 7768 6963 6820  s (NSOGs) which 
+0000b420: 6973 206f 7269 6769 6e61 6c6c 7920 6d75  is originally mu
+0000b430: 6c74 6979 2d63 6f70 7920 6275 7420 6861  ltiy-copy but ha
+0000b440: 7320 6120 286d 6f73 746c 7929 2073 696e  s a (mostly) sin
+0000b450: 676c 652d 636f 7079 2062 7261 6e63 6820  gle-copy branch 
+0000b460: 2877 6869 6368 2072 6571 7569 7265 7320  (which requires 
+0000b470: 7468 6520 6368 6f73 656e 2074 7265 652d  the chosen tree-
+0000b480: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
+0000b490: 6d20 7365 7420 6279 2060 2d74 7265 6560  m set by `-tree`
+0000b4a0: 206f 7220 602d 2d74 7265 655f 6d65 7468   or `--tree_meth
+0000b4b0: 6f64 6020 746f 2062 6520 7072 652d 696e  od` to be pre-in
+0000b4c0: 7374 616c 6c65 6420 696e 2074 6865 2065  stalled in the e
+0000b4d0: 6e76 6972 6f6e 6d65 6e74 2070 6174 6820  nvironment path 
+0000b4e0: 7769 7468 2074 6865 2070 6172 616d 6574  with the paramet
+0000b4f0: 6572 7320 7365 7474 696e 6720 666f 7220  ers setting for 
+0000b500: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
+0000b510: 6e63 6520 636f 6e74 726f 6c6c 6564 2062  nce controlled b
+0000b520: 7920 602d 7473 6020 6f72 2060 2d2d 7472  y `-ts` or `--tr
+0000b530: 6565 7365 7460 292e 2054 6865 2070 726f  eeset`). The pro
+0000b540: 6772 616d 2060 7767 6420 646d 6460 2077  gram `wgd dmd` w
+0000b550: 6f75 6c64 2073 7469 6c6c 2063 6f6e 6475  ould still condu
+0000b560: 6374 2074 6865 2052 4248 7320 6361 6c63  ct the RBHs calc
+0000b570: 756c 6174 696f 6e20 756e 6c65 7373 2074  ulation unless t
+0000b580: 6865 2070 6172 616d 6574 6572 2060 2d6f  he parameter `-o
+0000b590: 6f60 206f 7220 602d 2d6f 6e6c 796f 7274  o` or `--onlyort
+0000b5a0: 686f 6020 7761 7320 7365 742e 2049 6620  ho` was set. If 
+0000b5b0: 6f6e 6520 6f6e 6c79 2077 616e 7473 2074  one only wants t
+0000b5c0: 6f20 696e 6665 7220 7468 6520 6f72 7468  o infer the orth
+0000b5d0: 6f67 726f 7570 732c 2069 7427 7320 7375  ogroups, it's su
+0000b5e0: 6767 6573 7465 6420 746f 2061 6464 2074  ggested to add t
+0000b5f0: 6865 2066 6c61 6720 602d 6f6f 6020 746f  he flag `-oo` to
+0000b600: 206a 7573 7420 696d 706c 656d 656e 7420   just implement 
+0000b610: 7468 6520 6f72 7468 6f67 726f 7570 7320  the orthogroups 
+0000b620: 6465 6c69 6e65 6174 696f 6e20 616e 616c  delineation anal
+0000b630: 7973 6973 2e20 5468 6520 6e75 6d62 6572  ysis. The number
+0000b640: 206f 6620 7061 7261 6c6c 656c 2074 6872   of parallel thr
+0000b650: 6561 6473 2069 7320 7375 6767 6573 7465  eads is suggeste
+0000b660: 6420 746f 2062 6520 7365 7420 6173 2028  d to be set as (
+0000b670: 4e2b 3129 4e2f 3220 7768 6572 6520 4e20  N+1)N/2 where N 
+0000b680: 6973 2074 6865 206e 756d 6265 7220 6f66  is the number of
+0000b690: 2063 6473 2066 696c 6573 2074 6f20 6163   cds files to ac
+0000b6a0: 6869 6576 6520 7468 6520 6869 6768 6573  hieve the highes
+0000b6b0: 7420 6566 6669 6369 656e 6379 2073 696e  t efficiency sin
+0000b6c0: 6365 2074 6865 2073 656c 662d 636f 6d70  ce the self-comp
+0000b6d0: 6172 6973 6f6e 2069 7320 616c 736f 2069  arison is also i
+0000b6e0: 6e63 6c75 6465 642e 0a0a 2a2a 5468 6520  ncluded...**The 
+0000b6f0: 6465 6661 756c 7420 7365 7474 696e 6720  default setting 
+0000b700: 6f66 2070 6172 616d 6574 6572 7320 6973  of parameters is
+0000b710: 2061 2072 6561 736f 6e61 626c 6520 7374   a reasonable st
+0000b720: 6172 7469 6e67 2070 6f69 6e74 2077 6974  arting point wit
+0000b730: 6820 7468 6520 636f 6d6d 616e 6420 6173  h the command as
+0000b740: 2062 656c 6f77 2e2a 2a0a 0a60 6060 0a77   below.**..```.w
+0000b750: 6764 2064 6d64 2073 6571 7565 6e63 6531  gd dmd sequence1
+0000b760: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
+0000b770: 6e63 6533 202d 6f69 202d 6f6f 0a60 6060  nce3 -oi -oo.```
+0000b780: 0a0a 2a2a 5468 6520 636f 6c6c 696e 6561  ..**The collinea
+0000b790: 7220 636f 616c 6573 6365 6e63 6520 696e  r coalescence in
+0000b7a0: 6665 7265 6e63 6520 6f66 2070 6879 6c6f  ference of phylo
+0000b7b0: 6765 6e79 2a2a 0a60 6060 0a77 6764 2064  geny**.```.wgd d
+0000b7c0: 6d64 2073 6571 7565 6e63 6531 2073 6571  md sequence1 seq
+0000b7d0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+0000b7e0: 202d 6170 2061 7064 6174 6120 2d73 6d20   -ap apdata -sm 
+0000b7f0: 736d 6461 7461 202d 6c65 206c 6564 6174  smdata -le ledat
+0000b800: 6120 2d67 7420 6774 6461 7461 202d 636f  a -gt gtdata -co
+0000b810: 6320 282d 7472 6565 2027 6661 7374 7472  c (-tree 'fasttr
+0000b820: 6565 2729 2028 2d74 7320 272d 6661 7374  ee') (-ts '-fast
+0000b830: 6573 7427 2920 282d 6e20 3429 2028 2d2d  est') (-n 4) (--
+0000b840: 746f 5f73 746f 7029 2028 2d2d 6364 7329  to_stop) (--cds)
+0000b850: 2028 2d6f 2077 6764 5f64 6d64 2920 282d   (-o wgd_dmd) (-
+0000b860: 7420 776f 726b 696e 675f 746d 7029 0a60  t working_tmp).`
+0000b870: 6060 0a0a 4120 6e6f 7665 6c20 7068 796c  ``..A novel phyl
+0000b880: 6f67 656e 6574 6963 2069 6e66 6572 656e  ogenetic inferen
+0000b890: 6365 206d 6574 686f 6420 6e61 6d65 6420  ce method named 
+0000b8a0: 2263 6f6c 6c69 6e65 6172 2063 6f61 6c65  "collinear coale
+0000b8b0: 7363 656e 6365 2069 6e66 6572 656e 6365  scence inference
+0000b8c0: 2220 6973 2061 6c73 6f20 696d 706c 656d  " is also implem
+0000b8d0: 656e 7465 6420 696e 2060 7767 6420 7632  ented in `wgd v2
+0000b8e0: 602e 2046 6f72 2074 6869 7320 616e 616c  `. For this anal
+0000b8f0: 7973 6973 2c20 7573 6572 7320 6e65 6564  ysis, users need
+0000b900: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
+0000b910: 616e 6368 6f72 2070 6f69 6e74 7320 6669  anchor points fi
+0000b920: 6c65 2062 7920 602d 6170 6020 6f72 2060  le by `-ap` or `
+0000b930: 2d2d 616e 6368 6f72 706f 696e 7473 602c  --anchorpoints`,
+0000b940: 2074 6865 2063 6f6c 6c69 6e65 6172 2073   the collinear s
+0000b950: 6567 6d65 6e74 7320 6669 6c65 2062 7920  egments file by 
+0000b960: 602d 736d 6020 6f72 2060 2d2d 7365 676d  `-sm` or `--segm
+0000b970: 656e 7473 602c 2074 6865 206c 6973 7473  ents`, the lists
+0000b980: 6567 6d65 6e74 7320 6669 6c65 2062 7920  egments file by 
+0000b990: 602d 6c65 6020 6f72 2060 2d2d 6c69 7374  `-le` or `--list
+0000b9a0: 656c 656d 656e 7473 602c 2061 6e64 2074  elements`, and t
+0000b9b0: 6865 2067 656e 6520 7461 626c 6520 6669  he gene table fi
+0000b9c0: 6c65 2062 7920 602d 6774 6020 6f72 2060  le by `-gt` or `
+0000b9d0: 2d2d 6765 6e65 7461 626c 6560 2c20 616c  --genetable`, al
+0000b9e0: 6c20 6f66 2077 6869 6368 2063 616e 2062  l of which can b
+0000b9f0: 6520 7072 6f64 7563 6564 2069 6e20 7468  e produced in th
+0000ba00: 6520 7072 6f67 7261 6d20 6077 6764 2073  e program `wgd s
+0000ba10: 796e 602e 2054 6865 2070 6172 616d 6574  yn`. The paramet
+0000ba20: 6572 2060 2d63 6f63 6020 6f72 2060 2d2d  er `-coc` or `--
+0000ba30: 636f 6c6c 696e 6561 7263 6f61 6c65 7363  collinearcoalesc
+0000ba40: 656e 6365 6020 6e65 6564 7320 746f 2062  ence` needs to b
+0000ba50: 6520 7365 7420 746f 2073 7461 7274 2074  e set to start t
+0000ba60: 6869 7320 616e 616c 7973 6973 2e20 5468  his analysis. Th
+0000ba70: 6520 7472 6565 2d69 6e66 6572 656e 6365  e tree-inference
+0000ba80: 2070 726f 6772 616d 2061 6e64 2074 6865   program and the
+0000ba90: 2061 7373 6f63 6961 7465 6420 7061 7261   associated para
+0000baa0: 6d65 7465 7273 2063 616e 2062 6520 7365  meters can be se
+0000bab0: 7420 6a75 7374 2061 7320 6162 6f76 6520  t just as above 
+0000bac0: 6279 2060 2d74 7265 6560 206f 7220 602d  by `-tree` or `-
+0000bad0: 2d74 7265 655f 6d65 7468 6f64 6020 616e  -tree_method` an
+0000bae0: 6420 602d 7473 6020 6f72 2060 2d2d 7472  d `-ts` or `--tr
+0000baf0: 6565 7365 7460 2e20 506c 6561 7365 2061  eeset`. Please a
+0000bb00: 6c73 6f20 6d61 6b65 2073 7572 6520 7468  lso make sure th
+0000bb10: 6520 6368 6f73 656e 2074 7265 652d 696e  e chosen tree-in
+0000bb20: 6665 7265 6e63 6520 7072 6f67 7261 6d20  ference program 
+0000bb30: 6973 2069 6e73 7461 6c6c 6564 2069 6e20  is installed in 
+0000bb40: 7468 6520 656e 7669 726f 6e6d 656e 7420  the environment 
+0000bb50: 7061 7468 2e20 5468 6520 7072 6f67 7261  path. The progra
+0000bb60: 6d20 6061 7374 7261 6c2d 7072 6f60 2069  m `astral-pro` i
+0000bb70: 7320 7265 7175 6972 6564 2074 6f20 6265  s required to be
+0000bb80: 2069 6e73 7461 6c6c 6564 2069 6e20 7468   installed in th
+0000bb90: 6520 656e 7669 726f 6e6d 656e 7420 7061  e environment pa
+0000bba0: 7468 2074 6f6f 2e20 4e6f 7465 2074 6861  th too. Note tha
+0000bbb0: 7420 7468 6572 6520 7368 6f75 6c64 2062  t there should b
+0000bbc0: 6520 6e6f 2064 7570 6c69 6361 7465 6420  e no duplicated 
+0000bbd0: 6765 6e65 2049 4473 2069 6e20 7468 6520  gene IDs in the 
+0000bbe0: 7365 7175 656e 6365 2066 696c 652e 2054  sequence file. T
+0000bbf0: 6865 2070 6172 616c 6c65 6c20 7468 7265  he parallel thre
+0000bc00: 6164 7320 6865 7265 2061 7265 2074 6f20  ads here are to 
+0000bc10: 6163 6365 6c65 7261 7465 2074 6865 2073  accelerate the s
+0000bc20: 6571 7565 6e63 6520 616c 6967 6e6d 656e  equence alignmen
+0000bc30: 7420 616e 6420 6765 6e65 2074 7265 6520  t and gene tree 
+0000bc40: 696e 6665 7265 6e63 6520 666f 7220 6561  inference for ea
+0000bc50: 6368 2067 656e 6520 6661 6d69 6c79 2061  ch gene family a
+0000bc60: 6e64 2074 6875 7320 7375 6767 6573 7465  nd thus suggeste
+0000bc70: 6420 746f 2062 6520 7365 7420 6173 206d  d to be set as m
+0000bc80: 7563 6820 6173 2074 6865 206e 756d 6265  uch as the numbe
+0000bc90: 7220 6f66 2067 656e 6520 6661 6d69 6c69  r of gene famili
+0000bca0: 6573 2e0a 0a2a 2a41 2073 7567 6765 7374  es...**A suggest
+0000bcb0: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
+0000bcc0: 6f66 2074 6869 7320 616e 616c 7973 6973  of this analysis
+0000bcd0: 2069 7320 7769 7468 2074 6865 2073 696d   is with the sim
+0000bce0: 706c 6520 636f 6d6d 616e 6420 6265 6c6f  ple command belo
+0000bcf0: 772e 2a2a 0a0a 6060 600a 7767 6420 646d  w.**..```.wgd dm
+0000bd00: 6420 7365 7175 656e 6365 3120 7365 7175  d sequence1 sequ
+0000bd10: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+0000bd20: 2d61 7020 6170 6461 7461 202d 736d 2073  -ap apdata -sm s
+0000bd30: 6d64 6174 6120 2d6c 6520 6c65 6461 7461  mdata -le ledata
+0000bd40: 202d 6774 2067 7464 6174 6120 2d63 6f63   -gt gtdata -coc
+0000bd50: 0a60 6060 0a0a 2323 2320 7767 6420 666f  .```..### wgd fo
+0000bd60: 6375 730a 0a2a 2a54 6865 2063 6f6e 6361  cus..**The conca
+0000bd70: 7465 6e61 7469 6f6e 2d62 6173 6564 2f63  tenation-based/c
+0000bd80: 6f61 6c65 7363 656e 6365 2d62 6173 6564  oalescence-based
+0000bd90: 2070 6879 6c6f 6765 6e65 7469 6320 696e   phylogenetic in
+0000bda0: 6665 7265 6e63 652a 2a0a 6060 600a 7767  ference**.```.wg
+0000bdb0: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
+0000bdc0: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
+0000bdd0: 6e63 6532 2073 6571 7565 6e63 6533 2028  nce2 sequence3 (
+0000bde0: 2d2d 636f 6e63 6174 656e 6174 696f 6e29  --concatenation)
+0000bdf0: 2028 2d2d 636f 616c 6573 6365 6e63 6529   (--coalescence)
+0000be00: 2028 2d74 7265 6520 2766 6173 7474 7265   (-tree 'fasttre
+0000be10: 6527 2920 282d 7473 2027 2d66 6173 7465  e') (-ts '-faste
+0000be20: 7374 2729 2028 2d6e 2034 2920 282d 2d74  st') (-n 4) (--t
+0000be30: 6f5f 7374 6f70 2920 282d 2d63 6473 2920  o_stop) (--cds) 
+0000be40: 282d 6f20 7767 645f 666f 6375 7329 2028  (-o wgd_focus) (
+0000be50: 2d74 2077 6f72 6b69 6e67 5f74 6d70 290a  -t working_tmp).
+0000be60: 6060 600a 0a54 6865 2070 726f 6772 616d  ```..The program
+0000be70: 2060 7767 6420 666f 6375 7360 2069 6d70   `wgd focus` imp
+0000be80: 6c65 6d65 6e74 6564 2074 776f 2062 6173  lemented two bas
+0000be90: 6963 2070 6879 6c6f 6765 6e65 7469 6320  ic phylogenetic 
+0000bea0: 696e 6665 7265 6e63 6520 6d65 7468 6f64  inference method
+0000beb0: 732c 2069 2e65 2e2c 2063 6f6e 6361 7465  s, i.e., concate
+0000bec0: 6e61 7469 6f6e 2d62 6173 6564 2061 6e64  nation-based and
+0000bed0: 2063 6f61 6c65 7363 656e 6365 2d62 6173   coalescence-bas
+0000bee0: 6564 206d 6574 686f 6473 2e20 546f 2069  ed methods. To i
+0000bef0: 6e69 7469 6174 6520 7468 6573 6520 616e  nitiate these an
+0000bf00: 616c 7973 6973 2c20 7573 6572 7320 6e65  alysis, users ne
+0000bf10: 6564 2074 6f20 7365 7420 7468 6520 666c  ed to set the fl
+0000bf20: 6167 206f 7074 696f 6e20 602d 2d63 6f6e  ag option `--con
+0000bf30: 6361 7465 6e61 7469 6f6e 6020 6f72 2060  catenation` or `
+0000bf40: 2d2d 636f 616c 6573 6365 6e63 6560 2e20  --coalescence`. 
+0000bf50: 5468 6520 636f 6e63 6174 656e 6174 696f  The concatenatio
+0000bf60: 6e2d 6261 7365 6420 6d65 7468 6f64 2069  n-based method i
+0000bf70: 6e63 6c75 6465 7320 6120 6665 7720 6d61  ncludes a few ma
+0000bf80: 6a6f 7220 7374 6570 732c 2069 2e65 2e2c  jor steps, i.e.,
+0000bf90: 2074 6865 206d 756c 7469 706c 6520 7365   the multiple se
+0000bfa0: 7175 656e 6365 2061 6c69 676e 6d65 6e74  quence alignment
+0000bfb0: 2028 4d53 4129 206f 6620 6561 6368 2067   (MSA) of each g
+0000bfc0: 656e 6520 6661 6d69 6c79 2c20 7468 6520  ene family, the 
+0000bfd0: 636f 6e63 6174 656e 6174 696f 6e20 6f66  concatenation of
+0000bfe0: 2061 6c6c 2067 656e 6520 6661 6d69 6c69   all gene famili
+0000bff0: 6573 2061 6e64 2074 6865 6e20 7468 6520  es and then the 
+0000c000: 6765 6e65 2074 7265 6520 2861 6c73 6f20  gene tree (also 
+0000c010: 7370 6563 6965 7320 7472 6565 2069 6e20  species tree in 
+0000c020: 7468 6973 2063 6173 6529 2069 6e66 6572  this case) infer
+0000c030: 656e 6365 2e20 5468 6520 636f 616c 6573  ence. The coales
+0000c040: 6365 6e63 652d 6261 7365 6420 6d65 7468  cence-based meth
+0000c050: 6f64 2077 696c 6c20 696e 7374 6561 6420  od will instead 
+0000c060: 7065 7266 6f72 6d20 7468 6520 4d53 4120  perform the MSA 
+0000c070: 6f66 2065 6163 6820 6765 6e65 2066 616d  of each gene fam
+0000c080: 696c 7920 616e 6420 7468 6520 6765 6e65  ily and the gene
+0000c090: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
+0000c0a0: 6261 7365 6420 6f6e 2065 6163 6820 4d53  based on each MS
+0000c0b0: 412c 2061 6e64 2074 6865 6e20 696e 6665  A, and then infe
+0000c0c0: 7220 7468 6520 7370 6563 6965 7320 7472  r the species tr
+0000c0d0: 6565 2062 6173 6564 206f 6e20 7468 6573  ee based on thes
+0000c0e0: 6520 696e 6469 7669 6475 616c 2067 656e  e individual gen
+0000c0f0: 6520 7472 6565 732e 2054 6865 2074 7265  e trees. The tre
+0000c100: 652d 696e 6665 7265 6e63 6520 7072 6f67  e-inference prog
+0000c110: 7261 6d20 616e 6420 7468 6520 6173 736f  ram and the asso
+0000c120: 6369 6174 6564 2070 6172 616d 6574 6572  ciated parameter
+0000c130: 7320 6361 6e20 6265 2073 6574 206a 7573  s can be set jus
+0000c140: 7420 6173 2061 626f 7665 2062 7920 602d  t as above by `-
+0000c150: 7472 6565 6020 6f72 2060 2d2d 7472 6565  tree` or `--tree
+0000c160: 5f6d 6574 686f 6460 2061 6e64 2060 2d74  _method` and `-t
+0000c170: 7360 206f 7220 602d 2d74 7265 6573 6574  s` or `--treeset
+0000c180: 602e 2050 6c65 6173 6520 616c 736f 206d  `. Please also m
+0000c190: 616b 6520 7375 7265 2074 6865 2063 686f  ake sure the cho
+0000c1a0: 7365 6e20 7472 6565 2d69 6e66 6572 656e  sen tree-inferen
+0000c1b0: 6365 2070 726f 6772 616d 2069 7320 696e  ce program is in
+0000c1c0: 7374 616c 6c65 6420 696e 2074 6865 2065  stalled in the e
+0000c1d0: 6e76 6972 6f6e 6d65 6e74 2070 6174 682e  nvironment path.
+0000c1e0: 2054 6865 2070 726f 6772 616d 2060 6173   The program `as
+0000c1f0: 7472 616c 2d70 726f 6020 6973 2072 6571  tral-pro` is req
+0000c200: 7569 7265 6420 746f 2062 6520 696e 7374  uired to be inst
+0000c210: 616c 6c65 6420 696e 2074 6865 2065 6e76  alled in the env
+0000c220: 6972 6f6e 6d65 6e74 2070 6174 6820 6966  ironment path if
+0000c230: 2074 6865 2063 6f61 6c65 7363 656e 6365   the coalescence
+0000c240: 2d62 6173 6564 206d 6574 686f 6420 6973  -based method is
+0000c250: 2063 686f 7365 6e2e 2054 6865 2070 6172   chosen. The par
+0000c260: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
+0000c270: 7265 2061 7265 2074 6f20 6163 6365 6c65  re are to accele
+0000c280: 7261 7465 2074 6865 2073 6571 7565 6e63  rate the sequenc
+0000c290: 6520 616c 6967 6e6d 656e 7420 616e 6420  e alignment and 
+0000c2a0: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
+0000c2b0: 6e63 6520 666f 7220 6561 6368 2067 656e  nce for each gen
+0000c2c0: 6520 6661 6d69 6c79 2074 6f6f 2061 6e64  e family too and
+0000c2d0: 2074 6875 7320 7375 6767 6573 7465 6420   thus suggested 
+0000c2e0: 746f 2062 6520 7365 7420 6173 206d 7563  to be set as muc
+0000c2f0: 6820 6173 2074 6865 206e 756d 6265 7220  h as the number 
+0000c300: 6f66 2067 656e 6520 6661 6d69 6c69 6573  of gene families
+0000c310: 2e0a 0a2a 2a41 2073 7567 6765 7374 6564  ...**A suggested
+0000c320: 2073 7461 7274 696e 6720 7275 6e20 6f66   starting run of
+0000c330: 2074 6869 7320 616e 616c 7973 6973 2069   this analysis i
+0000c340: 7320 7769 7468 2074 6865 2073 696d 706c  s with the simpl
+0000c350: 6520 636f 6d6d 616e 6420 6265 6c6f 772e  e command below.
+0000c360: 2a2a 0a0a 6060 600a 7767 6420 666f 6375  **..```.wgd focu
+0000c370: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
+0000c380: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
+0000c390: 6571 7565 6e63 6533 202d 2d63 6f6e 6361  equence3 --conca
+0000c3a0: 7465 6e61 7469 6f6e 0a77 6764 2066 6f63  tenation.wgd foc
+0000c3b0: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
+0000c3c0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
+0000c3d0: 7365 7175 656e 6365 3320 2d2d 636f 616c  sequence3 --coal
+0000c3e0: 6573 6365 6e63 650a 6060 600a 0a2a 2a54  escence.```..**T
+0000c3f0: 6865 2066 756e 6374 696f 6e61 6c20 616e  he functional an
+0000c400: 6e6f 7461 7469 6f6e 206f 6620 6765 6e65  notation of gene
+0000c410: 2066 616d 696c 6965 732a 2a0a 6060 600a   families**.```.
+0000c420: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
+0000c430: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
+0000c440: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+0000c450: 202d 2d61 6e6e 6f74 6174 696f 6e20 6567   --annotation eg
+0000c460: 676e 6f67 202d 6564 2065 6464 6174 6120  gnog -ed eddata 
+0000c470: 2d2d 646d 6e62 2064 6264 6174 610a 6060  --dmnb dbdata.``
+0000c480: 600a 0a54 6865 2070 726f 6772 616d 2060  `..The program `
+0000c490: 7767 6420 666f 6375 7360 2061 6c73 6f20  wgd focus` also 
+0000c4a0: 6164 6465 6420 736f 6d65 2077 7261 7070  added some wrapp
+0000c4b0: 696e 6720 6675 6e63 7469 6f6e 7320 666f  ing functions fo
+0000c4c0: 7220 6675 6e63 7469 6f6e 616c 2061 6e6e  r functional ann
+0000c4d0: 6f74 6174 696f 6e20 6f66 2067 656e 6520  otation of gene 
+0000c4e0: 6661 6d69 6c69 6573 206f 6e20 7468 6520  families on the 
+0000c4f0: 686f 6f64 206f 6620 6461 7461 6261 7365  hood of database
+0000c500: 7320 616e 6420 736f 6674 7761 7265 7320  s and softwares 
+0000c510: 666f 7220 696e 7374 616e 6365 2060 4567  for instance `Eg
+0000c520: 674e 4f47 6020 616e 6420 6045 6767 4e4f  gNOG` and `EggNO
+0000c530: 472d 6d61 7070 6572 602e 2046 6f72 2074  G-mapper`. For t
+0000c540: 6865 2061 6e6e 6f74 6174 696f 6e20 7573  he annotation us
+0000c550: 696e 6720 6045 6767 4e4f 472d 6d61 7070  ing `EggNOG-mapp
+0000c560: 6572 602c 2075 7365 7273 206e 6565 6420  er`, users need 
+0000c570: 746f 2070 726f 7669 6465 2074 6865 2070  to provide the p
+0000c580: 6174 6820 746f 2074 6865 2065 6767 4e4f  ath to the eggNO
+0000c590: 4720 616e 6e6f 7461 7469 6f6e 2064 6174  G annotation dat
+0000c5a0: 6162 6173 6520 7669 6120 7468 6520 6f70  abase via the op
+0000c5b0: 7469 6f6e 2060 2d65 6460 2c20 7468 6520  tion `-ed`, the 
+0000c5c0: 7061 7468 2074 6f20 7468 6520 6469 616d  path to the diam
+0000c5d0: 6f6e 642d 636f 6d70 6174 6962 6c65 2064  ond-compatible d
+0000c5e0: 6174 6162 6173 6520 7669 6120 7468 6520  atabase via the 
+0000c5f0: 6f70 7469 6f6e 2060 2d2d 646d 6e64 5f64  option `--dmnd_d
+0000c600: 6260 2061 6e64 2074 6865 206f 7074 696f  b` and the optio
+0000c610: 6e20 602d 2d61 6e6e 6f74 6174 696f 6e60  n `--annotation`
+0000c620: 2073 6574 2061 7320 2265 6767 6e6f 6722   set as "eggnog"
+0000c630: 2e20 5468 6520 6d61 6e6e 6572 2068 6f77  . The manner how
+0000c640: 2050 4641 4d20 616e 6e6f 7461 7469 6f6e   PFAM annotation
+0000c650: 2077 696c 6c20 6265 2070 6572 666f 726d   will be perform
+0000c660: 6564 2063 616e 2062 6520 636f 6e74 726f  ed can be contro
+0000c670: 6c6c 6564 2076 6961 2074 6865 206f 7074  lled via the opt
+0000c680: 696f 6e20 6070 6661 6d60 2c20 6569 7468  ion `pfam`, eith
+0000c690: 6572 2022 6e6f 6e65 222c 2022 7265 616c  er "none", "real
+0000c6a0: 6967 6e22 206f 7220 2264 656e 6f76 6f22  ign" or "denovo"
+0000c6b0: 2c20 7468 6520 6465 7461 696c 6564 2065  , the detailed e
+0000c6c0: 7870 6c61 6e61 7469 6f6e 2063 616e 2062  xplanation can b
+0000c6d0: 6520 666f 756e 6420 6174 2074 6865 2077  e found at the w
+0000c6e0: 696b 6920 6f66 205b 4567 674e 4f47 2d6d  iki of [EggNOG-m
+0000c6f0: 6170 7065 725d 2868 7474 7073 3a2f 2f67  apper](https://g
+0000c700: 6974 6875 622e 636f 6d2f 6567 676e 6f67  ithub.com/eggnog
+0000c710: 6462 2f65 6767 6e6f 672d 6d61 7070 6572  db/eggnog-mapper
+0000c720: 2f77 696b 6929 2e20 506c 6561 7365 2070  /wiki). Please p
+0000c730: 7265 2d69 6e73 7461 6c6c 2074 6865 2060  re-install the `
+0000c740: 4567 674e 4f47 2d6d 6170 7065 7260 2070  EggNOG-mapper` p
+0000c750: 7974 686f 6e20 7061 636b 6167 6520 6966  ython package if
+0000c760: 2075 7369 6e67 2074 6869 7320 6675 6e63   using this func
+0000c770: 7469 6f6e 2e20 466f 7220 7468 6520 616e  tion. For the an
+0000c780: 6e6f 7461 7469 6f6e 2075 7369 6e67 2060  notation using `
+0000c790: 686d 6d73 6361 6e60 2c20 7768 6174 2069  hmmscan`, what i
+0000c7a0: 7320 696d 706c 656d 656e 7465 6420 696e  s implemented in
+0000c7b0: 2060 7767 6420 7632 6020 6973 2061 2073   `wgd v2` is a s
+0000c7c0: 696d 706c 6520 6275 6e64 6c65 2066 756e  imple bundle fun
+0000c7d0: 6374 696f 6e20 746f 2070 6572 666f 726d  ction to perform
+0000c7e0: 2060 686d 6d73 6361 6e60 2061 6e61 6c79   `hmmscan` analy
+0000c7f0: 7369 7320 666f 7220 6120 6769 7665 6e20  sis for a given 
+0000c800: 686d 6d20 7072 6f66 696c 6520 616e 6420  hmm profile and 
+0000c810: 7365 7420 6f66 2067 656e 6520 6661 6d69  set of gene fami
+0000c820: 6c69 6573 2073 7563 6820 7468 6174 2075  lies such that u
+0000c830: 7365 7273 206e 6565 6420 746f 2073 6574  sers need to set
+0000c840: 2074 6865 206f 7074 696f 6e20 602d 2d61   the option `--a
+0000c850: 6e6e 6f74 6174 696f 6e60 2061 7320 2268  nnotation` as "h
+0000c860: 6d6d 7066 616d 2220 616e 6420 7072 6f76  mmpfam" and prov
+0000c870: 6964 6520 7468 6520 686d 6d70 726f 6669  ide the hmmprofi
+0000c880: 6c65 2076 6961 2074 6865 206f 7074 696f  le via the optio
+0000c890: 6e20 602d 2d68 6d6d 602e 2046 6f72 2074  n `--hmm`. For t
+0000c8a0: 6865 2061 6e6e 6f74 6174 696f 6e20 7573  he annotation us
+0000c8b0: 696e 6720 6069 6e74 6572 7072 6f73 6361  ing `interprosca
+0000c8c0: 6e60 2c20 7573 6572 7320 6e65 6564 2074  n`, users need t
+0000c8d0: 6f20 7072 6f76 6964 6520 7468 6520 7061  o provide the pa
+0000c8e0: 7468 2074 6f20 7468 6520 696e 7465 7270  th to the interp
+0000c8f0: 726f 7363 616e 2069 6e73 7461 6c6c 6174  roscan installat
+0000c900: 696f 6e20 666f 6c64 6572 2077 6865 7265  ion folder where
+0000c910: 2074 6865 7265 2069 7320 6120 6069 6e74   there is a `int
+0000c920: 6572 7072 6f73 6361 6e2e 7368 6020 6669  erproscan.sh` fi
+0000c930: 6c65 2076 6961 2074 6865 206f 7074 696f  le via the optio
+0000c940: 6e20 602d 2d65 7865 7061 7468 6020 616e  n `--exepath` an
+0000c950: 6420 7365 7420 7468 6520 7468 6520 6f70  d set the the op
+0000c960: 7469 6f6e 2060 2d2d 616e 6e6f 7461 7469  tion `--annotati
+0000c970: 6f6e 6020 6173 2022 696e 7465 7270 726f  on` as "interpro
+0000c980: 7363 616e 222e 2054 6865 2070 6172 616c  scan". The paral
+0000c990: 6c65 6c20 7468 7265 6164 7320 6865 7265  lel threads here
+0000c9a0: 2061 7265 2074 6f20 7061 7261 6c6c 656c   are to parallel
+0000c9b0: 697a 6520 7468 6520 616e 616c 7973 6973  ize the analysis
+0000c9c0: 2066 6f72 2065 6163 6820 6765 6e65 2066   for each gene f
+0000c9d0: 616d 696c 7920 616e 6420 7468 7573 2073  amily and thus s
+0000c9e0: 7567 6765 7374 6564 2074 6f20 6265 2073  uggested to be s
+0000c9f0: 6574 2061 7320 6d75 6368 2061 7320 7468  et as much as th
+0000ca00: 6520 6e75 6d62 6572 206f 6620 6765 6e65  e number of gene
+0000ca10: 2066 616d 696c 6965 732e 0a0a 2a2a 4120   families...**A 
+0000ca20: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
+0000ca30: 6e67 2072 756e 206f 6620 7468 6973 2061  ng run of this a
+0000ca40: 6e61 6c79 7369 7320 6361 6e20 6265 2077  nalysis can be w
+0000ca50: 6974 6820 7468 6520 636f 6d6d 616e 6420  ith the command 
+0000ca60: 6265 6c6f 772e 2a2a 0a0a 6060 600a 7767  below.**..```.wg
+0000ca70: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
+0000ca80: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
+0000ca90: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
+0000caa0: 2d61 6e6e 6f74 6174 696f 6e20 6567 676e  -annotation eggn
+0000cab0: 6f67 202d 6564 2065 6464 6174 6120 2d2d  og -ed eddata --
+0000cac0: 646d 6e62 2064 6264 6174 610a 7767 6420  dmnb dbdata.wgd 
+0000cad0: 666f 6375 7320 6661 6d69 6c69 6573 2073  focus families s
+0000cae0: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
+0000caf0: 6532 2073 6571 7565 6e63 6533 202d 2d61  e2 sequence3 --a
+0000cb00: 6e6e 6f74 6174 696f 6e20 686d 6d70 6661  nnotation hmmpfa
+0000cb10: 6d20 2d2d 686d 6d20 686d 6d64 6174 610a  m --hmm hmmdata.
+0000cb20: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
+0000cb30: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
+0000cb40: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+0000cb50: 202d 2d61 6e6e 6f74 6174 696f 6e20 696e   --annotation in
+0000cb60: 7465 7270 726f 7363 616e 202d 2d65 7865  terproscan --exe
+0000cb70: 7061 7468 2024 5041 5448 0a60 6060 0a0a  path $PATH.```..
+0000cb80: 2a2a 5468 6520 7068 796c 6f67 656e 6574  **The phylogenet
+0000cb90: 6963 2064 6174 696e 6720 6f66 2057 4744  ic dating of WGD
+0000cba0: 732a 2a0a 6060 600a 7767 6420 666f 6375  s**.```.wgd focu
+0000cbb0: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
+0000cbc0: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
+0000cbd0: 6571 7565 6e63 6533 202d 6420 6d63 6d63  equence3 -d mcmc
+0000cbe0: 7472 6565 202d 7370 2073 7064 6174 6120  tree -sp spdata 
+0000cbf0: 282d 2d70 726f 7463 6f63 6461 7469 6e67  (--protcocdating
+0000cc00: 2920 282d 2d70 6172 7469 7469 6f6e 2920  ) (--partition) 
+0000cc10: 282d 2d61 616d 6f64 656c 206c 6729 2028  (--aamodel lg) (
+0000cc20: 2d64 7320 2762 7572 6e69 6e20 3d20 3230  -ds 'burnin = 20
+0000cc30: 3030 2729 2028 2d64 7320 2773 616d 7066  00') (-ds 'sampf
+0000cc40: 7265 7120 3d20 3130 3030 2729 2028 2d64  req = 1000') (-d
+0000cc50: 7320 276e 7361 6d70 6c65 203d 2032 3030  s 'nsample = 200
+0000cc60: 3030 2729 2020 282d 6e20 3429 2028 2d2d  00')  (-n 4) (--
+0000cc70: 746f 5f73 746f 7029 2028 2d2d 6364 7329  to_stop) (--cds)
+0000cc80: 2028 2d6f 2077 6764 5f66 6f63 7573 2920   (-o wgd_focus) 
+0000cc90: 282d 7420 776f 726b 696e 675f 746d 7029  (-t working_tmp)
+0000cca0: 0a60 6060 0a0a 5468 6520 6162 736f 6c75  .```..The absolu
+0000ccb0: 7465 2064 6174 696e 6720 6f66 2057 4744  te dating of WGD
+0000ccc0: 7320 6973 2061 2073 7065 6369 6669 6320  s is a specific 
+0000ccd0: 7069 7065 6c69 6e65 2069 6d70 6c65 6d65  pipeline impleme
+0000cce0: 6e74 6564 2069 6e20 6077 6764 2076 3260  nted in `wgd v2`
+0000ccf0: 2075 7369 6e67 2074 6865 206d 6574 686f   using the metho
+0000cd00: 6420 6f66 2070 6879 6c6f 6765 6e65 7469  d of phylogeneti
+0000cd10: 6320 6461 7469 6e67 2e20 5468 6520 6661  c dating. The fa
+0000cd20: 6d69 6c69 6573 2075 7365 6420 696e 2074  milies used in t
+0000cd30: 6869 7320 7374 6570 2063 616e 2062 6520  his step can be 
+0000cd40: 7072 6f64 7563 6564 2066 726f 6d20 6077  produced from `w
+0000cd50: 6764 2064 6d64 6020 616e 6420 6077 6764  gd dmd` and `wgd
+0000cd60: 2070 6561 6b60 2e20 4e6f 7465 2074 6861   peak`. Note tha
+0000cd70: 7420 6865 7265 2077 6520 6f6e 6c79 2064  t here we only d
+0000cd80: 6973 6375 7373 2068 6f77 2074 6f20 6461  iscuss how to da
+0000cd90: 7465 2057 4744 7320 7769 7468 2067 656e  te WGDs with gen
+0000cda0: 6f6d 6520 6173 7365 6d62 6c79 2c20 696e  ome assembly, in
+0000cdb0: 7374 6561 6420 6f66 2074 7261 6e73 6372  stead of transcr
+0000cdc0: 6970 746f 6d65 2061 7373 656d 626c 7920  iptome assembly 
+0000cdd0: 2877 6869 6368 2077 696c 6c20 6265 2064  (which will be d
+0000cde0: 6973 6375 7373 6564 2069 6e20 6120 7365  iscussed in a se
+0000cdf0: 7061 7261 7465 2073 6563 7469 6f6e 2068  parate section h
+0000ce00: 6572 6575 6e64 6572 292e 2054 6865 2061  ereunder). The a
+0000ce10: 7373 756d 7074 696f 6e20 7765 206d 6164  ssumption we mad
+0000ce20: 6520 6865 7265 2069 7320 7468 6174 206e  e here is that n
+0000ce30: 6f74 2061 6c6c 2061 6e63 686f 7220 7061  ot all anchor pa
+0000ce40: 6972 7320 2863 6f6c 6c69 6e65 6172 2064  irs (collinear d
+0000ce50: 7570 6c69 6361 7465 7329 2061 7265 2073  uplicates) are s
+0000ce60: 7569 7461 626c 6520 666f 7220 7068 796c  uitable for phyl
+0000ce70: 6f67 656e 6574 6963 2064 6174 696e 672c  ogenetic dating,
+0000ce80: 2066 6f72 2069 6e73 7461 6e63 6520 7468   for instance th
+0000ce90: 6f73 6520 6661 7374 6c79 206f 7220 736c  ose fastly or sl
+0000cea0: 6f77 6c79 2065 766f 6c76 696e 6720 6765  owly evolving ge
+0000ceb0: 6e65 2064 7570 6c69 6361 7465 732c 2062  ne duplicates, b
+0000cec0: 6563 6175 7365 2074 6865 7927 7265 2070  ecause they're p
+0000ced0: 726f 6e65 2074 6f20 6769 7665 2062 6961  rone to give bia
+0000cee0: 7365 6420 6461 7469 6e67 2065 7374 696d  sed dating estim
+0000cef0: 6174 696f 6e2e 2053 6f20 6173 2074 6f20  ation. So as to 
+0000cf00: 7265 7472 6965 7665 2074 6865 2022 7265  retrieve the "re
+0000cf10: 6c69 6162 6c65 2220 616e 6368 6f72 2070  liable" anchor p
+0000cf20: 6169 7273 2c20 7765 2069 6d70 6c65 6d65  airs, we impleme
+0000cf30: 6e74 6564 2073 6f6d 6520 6d65 7468 6f64  nted some method
+0000cf40: 7320 6f66 2069 6465 6e74 6966 7969 6e67  s of identifying
+0000cf50: 2063 7265 6469 6162 6c65 2061 6e63 686f   crediable ancho
+0000cf60: 7220 7061 6972 7320 6261 7365 6420 6f6e  r pairs based on
+0000cf70: 2074 6865 6972 204b 7320 7661 6c75 6573   their Ks values
+0000cf80: 2061 6e64 2f6f 7220 7265 7369 6469 6e67   and/or residing
+0000cf90: 2063 6f6c 6c69 6e65 6172 2073 6567 6d65   collinear segme
+0000cfa0: 6e74 732e 2046 6f72 2061 2067 656e 6f6d  nts. For a genom
+0000cfb0: 6520 7769 7468 2063 6c65 6172 2073 6967  e with clear sig
+0000cfc0: 6e61 6c73 206f 6620 7075 7461 7469 7665  nals of putative
+0000cfd0: 2057 4744 732c 2073 7563 6820 6173 2074   WGDs, such as t
+0000cfe0: 6865 202a 4171 7569 6c65 6769 6120 636f  he *Aquilegia co
+0000cff0: 6572 756c 6561 2a20 696e 2074 6865 2065  erulea* in the e
+0000d000: 7861 6d70 6c65 2c20 6120 6865 7572 6973  xample, a heuris
+0000d010: 7469 6320 6d65 7468 6f64 2074 6861 7420  tic method that 
+0000d020: 6170 706c 6965 7320 7468 6520 7072 696e  applies the prin
+0000d030: 6369 706c 6520 6f66 2068 6f77 205b 6b73  ciple of how [ks
+0000d040: 7261 7465 735d 2868 7474 7073 3a2f 2f67  rates](https://g
+0000d050: 6974 6875 622e 636f 6d2f 5649 422d 5053  ithub.com/VIB-PS
+0000d060: 422f 6b73 7261 7465 7329 2066 696e 6420  B/ksrates) find 
+0000d070: 7468 6520 696e 6974 6961 6c20 7065 616b  the initial peak
+0000d080: 7320 616e 6420 7468 6569 7220 7061 7261  s and their para
+0000d090: 6d65 7465 7273 2077 6173 2069 6d70 6c65  meters was imple
+0000d0a0: 6d65 6e74 6564 2074 6f20 6669 6e64 2074  mented to find t
+0000d0b0: 6865 2039 3525 2063 6f6e 6669 6465 6e63  he 95% confidenc
+0000d0c0: 6520 6c65 7665 6c20 6f66 2074 6865 2061  e level of the a
+0000d0d0: 7373 756d 6564 206c 6f67 6e6f 726d 616c  ssumed lognormal
+0000d0e0: 2064 6973 7472 6962 7574 696f 6e20 6f66   distribution of
+0000d0f0: 2074 6865 2061 6e63 686f 7220 7061 6972   the anchor pair
+0000d100: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000d110: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
+0000d120: 6e20 746f 2066 696c 7465 7220 616e 6368  n to filter anch
+0000d130: 6f72 2070 6169 7273 2077 6974 6820 746f  or pairs with to
+0000d140: 6f20 6869 6768 206f 7220 6c6f 7720 2a4b  o high or low *K
+0000d150: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
+0000d160: 6573 2c20 7468 6520 6578 616d 706c 6172  es, the examplar
+0000d170: 2063 6f6d 6d61 6e64 206f 6620 7768 6963   command of whic
+0000d180: 6820 6973 2073 686f 7765 6420 696e 2074  h is showed in t
+0000d190: 6865 2060 7767 6420 7065 616b 6020 7365  he `wgd peak` se
+0000d1a0: 6374 696f 6e20 6f66 205b 496c 6c75 7374  ction of [Illust
+0000d1b0: 7261 7469 6f6e 5d28 6874 7470 733a 2f2f  ration](https://
+0000d1c0: 6769 7468 7562 2e63 6f6d 2f68 6563 6865  github.com/heche
+0000d1d0: 2d70 7362 2f77 6764 3f74 6162 3d72 6561  -psb/wgd?tab=rea
+0000d1e0: 646d 652d 6f76 2d66 696c 6523 696c 6c75  dme-ov-file#illu
+0000d1f0: 7374 7261 7469 6f6e 292e 2049 6620 7468  stration). If th
+0000d200: 6973 2068 6575 7269 7374 6963 206d 6574  is heuristic met
+0000d210: 686f 6420 6661 696c 6564 2074 6f20 6769  hod failed to gi
+0000d220: 7665 2072 6561 736f 6e61 626c 6520 7265  ve reasonable re
+0000d230: 7375 6c74 732c 2077 6869 6368 2069 7320  sults, which is 
+0000d240: 7573 7561 6c6c 7920 6475 6520 746f 2074  usually due to t
+0000d250: 6865 2065 6666 6563 7420 6f66 206d 756c  he effect of mul
+0000d260: 7469 706c 6520 6164 6a61 6365 6e74 2057  tiple adjacent W
+0000d270: 4744 7320 7468 6174 2062 6c75 7273 2074  GDs that blurs t
+0000d280: 6865 2070 6561 6b2d 6669 6e64 696e 6720  he peak-finding 
+0000d290: 7072 6f63 6573 732c 2075 7365 7273 2063  process, users c
+0000d2a0: 616e 2074 7572 6e20 746f 2074 6865 2063  an turn to the c
+0000d2b0: 6f6c 6c69 6e65 6172 2073 6567 6d65 6e74  ollinear segment
+0000d2c0: 732d 6775 6964 6564 2061 6e63 686f 7220  s-guided anchor 
+0000d2d0: 7061 6972 2063 6c75 7374 6572 696e 6720  pair clustering 
+0000d2e0: 696d 706c 656d 656e 7465 6420 696e 2060  implemented in `
+0000d2f0: 7767 6420 7632 602c 2069 6e20 7768 6963  wgd v2`, in whic
+0000d300: 6820 6120 636f 6c6c 696e 6561 7220 7365  h a collinear se
+0000d310: 676d 656e 742d 7769 7365 2047 4d4d 2063  gment-wise GMM c
+0000d320: 6c75 7374 6572 696e 6720 7769 6c6c 2062  lustering will b
+0000d330: 6520 6669 7273 7420 636f 6e64 7563 7465  e first conducte
+0000d340: 6420 6261 7365 6420 6f6e 2074 6865 2022  d based on the "
+0000d350: 736f 2d63 616c 6c65 6422 2073 6567 6d65  so-called" segme
+0000d360: 6e74 202a 4b2a 3c73 7562 3e53 3c2f 7375  nt *K*<sub>S</su
+0000d370: 623e 2061 6765 2072 6570 7265 7365 6e74  b> age represent
+0000d380: 6564 2062 7920 7468 6520 6d65 6469 616e  ed by the median
+0000d390: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000d3a0: 2061 6765 206f 6620 616c 6c20 7468 6520   age of all the 
+0000d3b0: 7265 7369 6469 6e67 2073 796e 7465 6c6f  residing syntelo
+0000d3c0: 6773 2028 6e6f 7465 2074 6861 7420 7468  gs (note that th
+0000d3d0: 6520 6765 6e65 2064 7570 6c69 6361 7465  e gene duplicate
+0000d3e0: 2070 6169 7273 2061 646f 7074 6564 2069   pairs adopted i
+0000d3f0: 6e20 7468 6973 2073 7465 7020 6973 2066  n this step is f
+0000d400: 726f 6d20 7468 6520 6669 6c65 2060 6d75  rom the file `mu
+0000d410: 6c74 6970 6c69 636f 6e5f 7061 6972 732e  ltiplicon_pairs.
+0000d420: 7478 7460 2077 6869 6368 2063 6f6e 7461  txt` which conta
+0000d430: 696e 7320 7468 6520 6675 6c6c 2073 6574  ins the full set
+0000d440: 206f 6620 7379 6e74 656c 6f67 7329 2c20   of syntelogs), 
+0000d450: 696e 7374 6561 6420 6f66 2074 6865 2073  instead of the s
+0000d460: 6d61 6c6c 6572 2067 656e 6520 7365 7420  maller gene set 
+0000d470: 6f66 2061 6e63 686f 7220 7061 6972 732e  of anchor pairs.
+0000d480: 2054 6865 2064 6973 7469 6e63 7469 6f6e   The distinction
+0000d490: 2062 6574 7765 656e 2061 6e63 686f 7220   between anchor 
+0000d4a0: 7061 6972 7320 616e 6420 7379 6e74 656c  pairs and syntel
+0000d4b0: 6f67 7320 6973 2074 6861 7420 7468 6520  ogs is that the 
+0000d4c0: 6c61 7474 6572 2072 6566 6572 7320 746f  latter refers to
+0000d4d0: 206d 756c 7469 706c 6520 7365 7473 206f   multiple sets o
+0000d4e0: 6620 6765 6e65 7320 6465 7269 7665 6420  f genes derived 
+0000d4f0: 6672 6f6d 2074 6865 2073 616d 6520 616e  from the same an
+0000d500: 6365 7374 7261 6c20 6765 6e6f 6d69 6320  cestral genomic 
+0000d510: 7265 6769 6f6e 2077 6869 6c65 2074 6865  region while the
+0000d520: 2066 6f72 6d65 7220 696d 706c 6965 7320   former implies 
+0000d530: 7468 6520 6c61 7474 6572 2062 7574 2072  the latter but r
+0000d540: 6571 7569 7265 7320 6164 6469 7469 6f6e  equires addition
+0000d550: 616c 6c79 2074 6865 2063 6f6e 7365 7276  ally the conserv
+0000d560: 6564 2067 656e 6520 6f72 6465 722c 2062  ed gene order, b
+0000d570: 6f74 6820 6f66 2077 6869 6368 2061 7265  oth of which are
+0000d580: 2c20 7769 7468 696e 2061 2067 656e 6f6d  , within a genom
+0000d590: 652c 2061 7373 756d 6564 2074 6f20 6265  e, assumed to be
+0000d5a0: 206f 7269 6769 6e61 7465 6420 6672 6f6d   originated from
+0000d5b0: 2074 6865 2064 7570 6c69 6361 7469 6f6e   the duplication
+0000d5c0: 206f 6620 6120 636f 6d6d 6f6e 2061 6e63   of a common anc
+0000d5d0: 6573 7472 616c 2067 656e 6f6d 6963 2072  estral genomic r
+0000d5e0: 6567 696f 6e20 616e 6420 6173 2073 7563  egion and as suc
+0000d5f0: 6820 6465 656d 6564 2065 7669 6465 6e63  h deemed evidenc
+0000d600: 6520 666f 7220 5747 442e 2054 6865 6e20  e for WGD. Then 
+0000d610: 7468 6520 7379 6e74 656c 6f67 7320 7769  the syntelogs wi
+0000d620: 6c6c 2062 6520 6d61 7070 6564 2062 6163  ll be mapped bac
+0000d630: 6b20 6163 636f 7264 696e 6720 746f 2074  k according to t
+0000d640: 6865 2063 6c75 7374 6572 696e 6720 7265  he clustering re
+0000d650: 7375 6c74 7320 6f66 2074 6865 6972 2061  sults of their a
+0000d660: 6666 696c 6961 7465 6420 7365 676d 656e  ffiliated segmen
+0000d670: 7473 2e20 5369 6e63 6520 7468 6520 4761  ts. Since the Ga
+0000d680: 7573 7369 616e 2073 6861 7065 206f 6620  ussian shape of 
+0000d690: 7468 6520 7365 676d 656e 7420 636c 7573  the segment clus
+0000d6a0: 7465 7220 646f 6573 6e27 7420 6e65 6365  ter doesn't nece
+0000d6b0: 7373 6172 696c 7920 696d 6974 6174 6520  ssarily imitate 
+0000d6c0: 7468 6520 7368 6170 6520 6f66 2074 6865  the shape of the
+0000d6d0: 2072 6573 6964 696e 6720 7379 6e74 656c   residing syntel
+0000d6e0: 6f67 732c 2073 6f20 6173 2074 6f20 7265  ogs, so as to re
+0000d6f0: 7472 6965 7665 2074 6865 2022 7265 6c69  trieve the "reli
+0000d700: 6162 6c65 2220 6765 6e65 2073 6574 7320  able" gene sets 
+0000d710: 666f 7220 7068 796c 6f67 656e 6574 6963  for phylogenetic
+0000d720: 2064 6174 696e 672c 2077 6520 6164 706f   dating, we adpo
+0000d730: 7465 6420 7468 6520 2839 3525 2920 6869  ted the (95%) hi
+0000d740: 6768 6573 7420 6465 6e73 6974 7920 7265  ghest density re
+0000d750: 6769 6f6e 2028 4844 5229 206f 6620 7468  gion (HDR) of th
+0000d760: 6520 7379 6e74 656c 6f67 202a 4b2a 3c73  e syntelog *K*<s
+0000d770: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+0000d780: 6973 7472 6962 7574 696f 6e20 666f 7220  istribution for 
+0000d790: 7468 6520 7068 796c 6f67 656e 6574 6963  the phylogenetic
+0000d7a0: 2064 6174 696e 672c 2074 6865 2063 616c   dating, the cal
+0000d7b0: 6375 6c61 7469 6f6e 206f 6620 7468 6520  culation of the 
+0000d7c0: 2839 3525 2920 4844 5220 696e 2074 6865  (95%) HDR in the
+0000d7d0: 2066 756e 6374 696f 6e20 5b63 616c 6375   function [calcu
+0000d7e0: 6c61 7465 4850 445d 2868 7474 7073 3a2f  lateHPD](https:/
+0000d7f0: 2f67 6974 6875 622e 636f 6d2f 6865 6368  /github.com/hech
+0000d800: 652d 7073 622f 7767 642f 626c 6f62 2f70  e-psb/wgd/blob/p
+0000d810: 6879 6c6f 6461 7469 6e67 2f77 6764 2f70  hylodating/wgd/p
+0000d820: 6561 6b2e 7079 234c 3137 3537 2920 7365  eak.py#L1757) se
+0000d830: 656b 7320 7468 6520 7368 6f72 7465 7374  eks the shortest
+0000d840: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000d850: 2072 616e 6765 2028 612c 6229 2077 6869   range (a,b) whi
+0000d860: 6368 2073 6174 6973 6669 6573 2074 6865  ch satisfies the
+0000d870: 2072 6571 7569 7265 6d65 6e74 206f 6620   requirement of 
+0000d880: 7370 616e 6e69 6e67 206d 6f72 6520 7468  spanning more th
+0000d890: 616e 2028 3935 2529 206f 6620 616c 6c20  an (95%) of all 
+0000d8a0: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
+0000d8b0: 7562 3e20 7661 6c75 6573 2e20 4f6e 2074  ub> values. On t
+0000d8c0: 6865 2070 7265 6d69 7365 206f 6620 6964  he premise of id
+0000d8d0: 656e 7469 6669 6564 2061 6e63 686f 7220  entified anchor 
+0000d8e0: 7061 6972 7320 286e 6f74 6520 7468 6174  pairs (note that
+0000d8f0: 2074 6865 2073 796e 7465 6c6f 6773 2061   the syntelogs a
+0000d900: 7265 2061 6c73 6f20 7265 6665 7272 6564  re also referred
+0000d910: 2061 7320 616e 6368 6f72 2070 6169 7273   as anchor pairs
+0000d920: 2068 6572 6561 6674 6572 292c 2077 6520   hereafter), we 
+0000d930: 696d 706c 656d 656e 7465 6420 696e 2074  implemented in t
+0000d940: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
+0000d950: 646d 6460 2074 6865 2073 6f2d 6361 6c6c  dmd` the so-call
+0000d960: 6564 2061 6e63 686f 722d 6177 6172 6520  ed anchor-aware 
+0000d970: 6c6f 6361 6c20 4d52 4248 7320 6f72 206f  local MRBHs or o
+0000d980: 7274 686f 6772 6f75 7073 2c20 696e 2077  rthogroups, in w
+0000d990: 6869 6368 2074 6865 206f 7269 6769 6e61  hich the origina
+0000d9a0: 6c20 6c6f 6361 6c20 4d52 4248 7320 6172  l local MRBHs ar
+0000d9b0: 6520 6675 7274 6865 7220 6d65 7267 6564  e further merged
+0000d9c0: 2077 6974 6820 616e 6368 6f72 2070 6169   with anchor pai
+0000d9d0: 7273 2073 7563 6820 7468 6174 2065 6163  rs such that eac
+0000d9e0: 6820 6f72 7468 6f67 726f 7570 2063 6f6e  h orthogroup con
+0000d9f0: 7461 696e 7320 7468 6520 616e 6368 6f72  tains the anchor
+0000da00: 2070 6169 7220 616e 6420 7468 6520 6f72   pair and the or
+0000da10: 7468 6f6c 6f67 7565 732e 2057 6974 6820  thologues. With 
+0000da20: 7468 6973 206f 7274 686f 6772 6f75 702c  this orthogroup,
+0000da30: 2075 7365 7273 2074 6865 6e20 6e65 6564   users then need
+0000da40: 206f 6e65 2073 7461 7274 696e 6720 7472   one starting tr
+0000da50: 6565 2066 696c 6520 2861 7320 7368 6f77  ee file (as show
+0000da60: 6e20 696e 2074 6865 205b 496c 6c75 7374  n in the [Illust
+0000da70: 7261 7469 6f6e 5d28 6874 7470 733a 2f2f  ration](https://
+0000da80: 6769 7468 7562 2e63 6f6d 2f68 6563 6865  github.com/heche
+0000da90: 2d70 7362 2f77 6764 2f74 7265 652f 7068  -psb/wgd/tree/ph
+0000daa0: 796c 6f64 6174 696e 673f 7461 623d 7265  ylodating?tab=re
+0000dab0: 6164 6d65 2d6f 762d 6669 6c65 2369 6c6c  adme-ov-file#ill
+0000dac0: 7573 7472 6174 696f 6e29 2070 6172 7429  ustration) part)
+0000dad0: 2069 6e64 6963 6174 696e 6720 7468 6520   indicating the 
+0000dae0: 7472 6565 2074 6f70 6f6c 6f67 7920 616e  tree topology an
+0000daf0: 6420 666f 7373 696c 2063 616c 6962 7261  d fossil calibra
+0000db00: 7469 6f6e 2069 6e66 6f72 6d61 7469 6f6e  tion information
+0000db10: 2066 6f72 2074 6865 2066 696e 616c 2070   for the final p
+0000db20: 6879 6c6f 6765 6e65 7469 6320 6461 7469  hylogenetic dati
+0000db30: 6e67 2075 7369 6e67 2074 6865 2070 726f  ng using the pro
+0000db40: 6772 616d 2060 7767 6420 666f 6375 7360  gram `wgd focus`
+0000db50: 2e20 5573 6572 7320 6361 6e20 7365 7420  . Users can set 
+0000db60: 7468 6520 666c 6167 206f 7074 696f 6e20  the flag option 
+0000db70: 602d 2d70 726f 7463 6f63 6461 7469 6e67  `--protcocdating
+0000db80: 6020 746f 206f 6e6c 7920 636f 6e64 7563  ` to only conduc
+0000db90: 7420 7468 6520 6461 7469 6e67 206f 6620  t the dating of 
+0000dba0: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
+0000dbb0: 6e20 7072 6f74 6569 6e20 4d53 4120 6f72  n protein MSA or
+0000dbc0: 2074 6865 2066 6c61 6720 6f70 7469 6f6e   the flag option
+0000dbd0: 2060 2d2d 7072 6f74 6461 7469 6e67 6020   `--protdating` 
+0000dbe0: 746f 206f 6e6c 7920 636f 6e64 7563 7420  to only conduct 
+0000dbf0: 7468 6520 6461 7469 6e67 206f 6620 7468  the dating of th
+0000dc00: 6520 7072 6f74 6569 6e20 4d53 4173 2c20  e protein MSAs, 
+0000dc10: 6e6f 7465 6420 7468 6174 2074 6865 7365  noted that these
+0000dc20: 2074 776f 206f 7074 696f 6e73 206f 6e6c   two options onl
+0000dc30: 7920 776f 726b 2066 6f72 2074 6865 2060  y work for the `
+0000dc40: 6d63 6d63 7472 6565 6020 6f70 7469 6f6e  mcmctree` option
+0000dc50: 2073 6f20 6661 722e 2054 6865 2066 6c61   so far. The fla
+0000dc60: 6720 6f70 7469 6f6e 2060 2d2d 7061 7274  g option `--part
+0000dc70: 6974 696f 6e60 2063 616e 2062 6520 7365  ition` can be se
+0000dc80: 7420 746f 2070 6572 666f 726d 2060 6d63  t to perform `mc
+0000dc90: 6d63 7472 6565 6020 616e 616c 7973 6973  mctree` analysis
+0000dca0: 2075 7369 6e67 2074 6865 2070 6172 7469   using the parti
+0000dcb0: 7469 6f6e 6564 2064 6174 6120 2869 2e65  tioned data (i.e
+0000dcc0: 2e2c 2031 7374 2c20 326e 6420 616e 6420  ., 1st, 2nd and 
+0000dcd0: 3372 6420 706f 7369 7469 6f6e 206f 6620  3rd position of 
+0000dce0: 636f 646f 6e29 2069 6e73 7465 6164 206f  codon) instead o
+0000dcf0: 6620 7573 696e 6720 7468 6520 636f 646f  f using the codo
+0000dd00: 6e20 6173 2061 2077 686f 6c65 2e20 5468  n as a whole. Th
+0000dd10: 6520 6f70 7469 6f6e 2060 2d2d 6161 6d6f  e option `--aamo
+0000dd20: 6465 6c60 2063 616e 2062 6520 7365 7420  del` can be set 
+0000dd30: 746f 2064 6574 6572 6d69 6e65 2074 6865  to determine the
+0000dd40: 2061 6d69 6e6f 2061 6369 6420 6d6f 6465   amino acid mode
+0000dd50: 6c20 6170 706c 6965 6420 696e 2060 6d63  l applied in `mc
+0000dd60: 6d63 7472 6565 6020 616e 616c 7973 6973  mctree` analysis
+0000dd70: 2e20 5468 6520 6f70 7469 6f6e 2060 2d64  . The option `-d
+0000dd80: 7360 2063 616e 2062 6520 7573 6564 2074  s` can be used t
+0000dd90: 6f20 7365 7420 7061 7261 6d65 7465 7273  o set parameters
+0000dda0: 2066 6f72 2074 6865 206d 6f6c 6563 756c   for the molecul
+0000ddb0: 6172 2064 6174 696e 6720 7072 6f67 7261  ar dating progra
+0000ddc0: 6d2e 2054 6865 2070 6172 616c 6c65 6c20  m. The parallel 
+0000ddd0: 7468 7265 6164 7320 6865 7265 2061 7265  threads here are
+0000dde0: 2074 6f20 7061 7261 6c6c 656c 697a 6520   to parallelize 
+0000ddf0: 7468 6520 616e 616c 7973 6973 2066 6f72  the analysis for
+0000de00: 2065 6163 6820 6765 6e65 2066 616d 696c   each gene famil
+0000de10: 7920 616e 6420 7468 7573 2073 7567 6765  y and thus sugge
+0000de20: 7374 6564 2074 6f20 6265 2073 6574 2061  sted to be set a
+0000de30: 7320 6d75 6368 2061 7320 7468 6520 6e75  s much as the nu
+0000de40: 6d62 6572 206f 6620 6765 6e65 2066 616d  mber of gene fam
+0000de50: 696c 6965 732e 0a0a 2a2a 4120 7375 6767  ilies...**A sugg
+0000de60: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+0000de70: 756e 2063 616e 2062 6520 7769 7468 2074  un can be with t
+0000de80: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
+0000de90: 2e2a 2a0a 0a60 6060 0a77 6764 2066 6f63  .**..```.wgd foc
+0000dea0: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
+0000deb0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
+0000dec0: 7365 7175 656e 6365 3320 2d64 206d 636d  sequence3 -d mcm
+0000ded0: 6374 7265 6520 2d73 7020 7370 6461 7461  ctree -sp spdata
+0000dee0: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
+0000def0: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+0000df00: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+0000df10: 3320 2d64 2072 3873 202d 7370 2073 7064  3 -d r8s -sp spd
+0000df20: 6174 6120 2d2d 6e73 6974 6573 206e 7369  ata --nsites nsi
+0000df30: 7465 696e 666f 0a77 6764 2066 6f63 7573  teinfo.wgd focus
+0000df40: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
+0000df50: 6365 3120 7365 7175 656e 6365 3220 7365  ce1 sequence2 se
+0000df60: 7175 656e 6365 3320 2d64 2062 6561 7374  quence3 -d beast
+0000df70: 202d 2d66 6f73 7369 6c20 666f 7373 696c   --fossil fossil
+0000df80: 696e 666f 202d 2d72 6f6f 7468 6569 6768  info --rootheigh
+0000df90: 7420 726f 6f74 6865 6967 6874 696e 666f  t rootheightinfo
+0000dfa0: 202d 2d63 6861 696e 7365 7420 6368 6169   --chainset chai
+0000dfb0: 6e73 6574 696e 666f 202d 2d62 6561 7374  nsetinfo --beast
+0000dfc0: 6c67 6a61 7220 2450 4154 480a 6060 600a  lgjar $PATH.```.
+0000dfd0: 0a23 2323 2077 6764 206b 7364 0a0a 2a2a  .### wgd ksd..**
+0000dfe0: 5468 6520 636f 6e73 7472 7563 7469 6f6e  The construction
+0000dff0: 206f 6620 7768 6f6c 6520 7061 7261 6e6f   of whole parano
+0000e000: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
+0000e010: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+0000e020: 696f 6e2a 2a0a 6060 600a 7767 6420 6b73  ion**.```.wgd ks
+0000e030: 6420 6661 6d69 6c69 6573 2073 6571 7565  d families seque
+0000e040: 6e63 6520 282d 6f20 7767 645f 6b73 6420  nce (-o wgd_ksd 
+0000e050: 2d74 2077 6764 5f6b 7364 5f74 6d70 202d  -t wgd_ksd_tmp -
+0000e060: 2d6e 7468 7265 6164 7320 3420 2d2d 746f  -nthreads 4 --to
+0000e070: 5f73 746f 7020 2d2d 6364 7320 2d2d 7061  _stop --cds --pa
+0000e080: 6972 7769 7365 202d 2d73 7472 6970 5f67  irwise --strip_g
+0000e090: 6170 7320 2d2d 616c 6967 6e65 7220 6d61  aps --aligner ma
+0000e0a0: 6666 7420 2d2d 7472 6565 5f6d 6574 686f  fft --tree_metho
+0000e0b0: 6420 6661 7374 7472 6565 202d 2d6e 6f64  d fasttree --nod
+0000e0c0: 655f 6176 6572 6167 6529 0a60 6060 0a0a  e_average).```..
+0000e0d0: 5468 6520 7072 6f67 7261 6d20 6077 6764  The program `wgd
+0000e0e0: 206b 7364 602c 2061 7320 696d 7069 6564   ksd`, as impied
+0000e0f0: 2062 7920 6974 7320 6e61 6d65 2c20 6973   by its name, is
+0000e100: 2066 6f72 2074 6865 2063 6f6e 7374 7275   for the constru
+0000e110: 6374 696f 6e20 6f66 202a 4b2a 3c73 7562  ction of *K*<sub
+0000e120: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
+0000e130: 7472 6962 7574 696f 6e2e 2045 7863 6570  tribution. Excep
+0000e140: 7420 666f 7220 7468 6520 6166 6f72 656d  t for the aforem
+0000e150: 656e 7469 6f6e 6564 2070 6172 616d 6574  entioned paramet
+0000e160: 6572 7320 7375 6368 2061 7320 602d 2d74  ers such as `--t
+0000e170: 6f5f 7374 6f70 6020 616e 6420 602d 2d63  o_stop` and `--c
+0000e180: 6473 602c 2074 6865 7265 2061 7265 2073  ds`, there are s
+0000e190: 6f6d 6520 696d 706f 7274 616e 7420 7061  ome important pa
+0000e1a0: 7261 6d65 7465 7273 2074 6861 7420 6861  rameters that ha
+0000e1b0: 7665 2063 7275 6369 616c 2069 6d70 6163  ve crucial impac
+0000e1c0: 7420 6f6e 2074 6865 202a 4b2a 3c73 7562  t on the *K*<sub
+0000e1d0: 3e53 3c2f 7375 623e 2065 7374 696d 6174  >S</sub> estimat
+0000e1e0: 696f 6e2e 2054 6865 206f 7074 696f 6e20  ion. The option 
+0000e1f0: 602d 2d70 6169 7277 6973 6560 2069 7320  `--pairwise` is 
+0000e200: 6120 7665 7279 2069 6d70 6f72 7461 6e74  a very important
+0000e210: 2070 6172 616d 6574 6572 2066 6f72 2074   parameter for t
+0000e220: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+0000e230: 623e 2065 7374 696d 6174 696f 6e2c 2077  b> estimation, w
+0000e240: 6974 6820 7768 6963 6820 7468 6520 6043  ith which the `C
+0000e250: 4f44 454d 4c60 2077 696c 6c20 6361 6c63  ODEML` will calc
+0000e260: 756c 6174 6520 7468 6520 2a4b 2a3c 7375  ulate the *K*<su
+0000e270: 623e 533c 2f73 7562 3e20 666f 7220 6561  b>S</sub> for ea
+0000e280: 6368 2067 656e 6520 7061 6972 2073 6570  ch gene pair sep
+0000e290: 6172 6174 656c 7920 6261 7365 6420 6f6e  arately based on
+0000e2a0: 2074 6865 2061 6c69 676e 6d65 6e74 206f   the alignment o
+0000e2b0: 6620 6f6e 6c79 2074 6865 7365 2074 776f  f only these two
+0000e2c0: 2067 656e 6573 2069 6e73 7465 6164 206f   genes instead o
+0000e2d0: 6620 7468 6520 7768 6f6c 6520 616c 6967  f the whole alig
+0000e2e0: 6e6d 656e 7420 6f66 2074 6865 2066 616d  nment of the fam
+0000e2f0: 696c 792c 2073 7563 6820 7468 6174 206c  ily, such that l
+0000e300: 6573 7320 6761 7073 2061 7265 2065 7870  ess gaps are exp
+0000e310: 6563 7465 6420 616e 6420 7468 7573 2074  ected and thus t
+0000e320: 6865 2061 6c69 676e 6d65 6e74 2069 6e20  he alignment in 
+0000e330: 7468 6520 636f 6e73 6964 6572 6174 696f  the consideratio
+0000e340: 6e20 6f66 2060 434f 4445 4d4c 6020 7769  n of `CODEML` wi
+0000e350: 6c6c 2062 6520 6c6f 6e67 6572 2028 6265  ll be longer (be
+0000e360: 6361 7573 6520 6043 4f44 454d 4c60 2077  cause `CODEML` w
+0000e370: 696c 6c20 6175 746f 6d61 7465 6c79 2073  ill automately s
+0000e380: 6b69 7020 6576 6572 7920 636f 6c75 6d6e  kip every column
+0000e390: 2077 6974 6820 6761 702c 2072 6567 6172   with gap, regar
+0000e3a0: 646c 6573 7320 6f66 2077 6865 7468 6572  dless of whether
+0000e3b0: 2069 7427 7320 616e 206f 7665 7261 6c6c   it's an overall
+0000e3c0: 206f 7220 7061 7274 6961 6c20 6761 7029   or partial gap)
+0000e3d0: 2c20 7769 7468 6f75 7420 7768 6963 6820  , without which 
+0000e3e0: 7468 6520 6043 4f44 454d 4c60 2077 696c  the `CODEML` wil
+0000e3f0: 6c20 6361 6c63 756c 6174 6520 7468 6520  l calculate the 
+0000e400: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000e410: 6261 7365 6420 6f6e 2074 6865 2077 686f  based on the who
+0000e420: 6c65 2061 6c69 676e 6d65 6e74 206f 6620  le alignment of 
+0000e430: 7468 6520 6661 6d69 6c79 2c20 7768 6963  the family, whic
+0000e440: 6820 6d69 6768 7420 6861 7665 206e 6f20  h might have no 
+0000e450: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000e460: 7265 7375 6c74 2061 7420 616c 6c20 6966  result at all if
+0000e470: 2074 6865 2073 7472 6970 7065 6420 616c   the stripped al
+0000e480: 6967 6e6d 656e 7420 6c65 6e67 7468 2028  ignment length (
+0000e490: 7265 6d6f 7669 6e67 2061 6c6c 2067 6170  removing all gap
+0000e4a0: 2d63 6f6e 7461 696e 696e 6720 636f 6c75  -containing colu
+0000e4b0: 6d6e 7329 2077 6173 207a 6572 6f2c 2061  mns) was zero, a
+0000e4c0: 2063 6175 7365 206f 6620 6469 6666 6572   cause of differ
+0000e4d0: 656e 7420 6e75 6d62 6572 206f 6620 2a4b  ent number of *K
+0000e4e0: 2a3c 7375 623e 533c 2f73 7562 3e20 6573  *<sub>S</sub> es
+0000e4f0: 7469 6d61 7465 7320 6265 7477 6565 6e20  timates between 
+0000e500: 2270 6169 7277 6973 6520 6d6f 6465 2220  "pairwise mode" 
+0000e510: 616e 6420 226e 6f6e 2d70 6169 7277 6973  and "non-pairwis
+0000e520: 6520 6d6f 6465 222e 2049 7427 7320 6469  e mode". It's di
+0000e530: 6666 6963 756c 7420 746f 2073 6179 2077  fficult to say w
+0000e540: 6869 6368 206d 6f64 6520 6973 206d 6f72  hich mode is mor
+0000e550: 6520 6964 6561 6c2c 2061 6c74 686f 7567  e ideal, althoug
+0000e560: 6820 7468 6520 226e 6f6e 2d70 6169 7277  h the "non-pairw
+0000e570: 6973 6520 6d6f 6465 2220 2864 6566 6175  ise mode" (defau
+0000e580: 6c74 2073 6574 7469 6e67 2920 7768 6963  lt setting) whic
+0000e590: 6820 7275 6e73 206f 6e20 7468 6520 7768  h runs on the wh
+0000e5a0: 6f6c 6520 616c 6967 6e6d 656e 7420 696e  ole alignment in
+0000e5b0: 7374 6561 6420 6f66 2061 206c 6f63 616c  stead of a local
+0000e5c0: 2061 6c69 676e 6d65 6e74 2c20 6d69 6768   alignment, migh
+0000e5d0: 7420 6265 206d 6f72 6520 6269 6f6c 6f67  t be more biolog
+0000e5e0: 6963 616c 6c79 2063 6f6e 7365 7276 6564  ically conserved
+0000e5f0: 2069 6e20 7468 6174 2069 7420 6173 7375   in that it assu
+0000e600: 7265 7320 7468 6520 6576 6f6c 7574 696f  res the evolutio
+0000e610: 6e20 6f66 2065 6163 6820 636f 6c75 6d6e  n of each column
+0000e620: 2074 6f20 6265 2073 7461 7274 6564 2066   to be started f
+0000e630: 726f 6d20 7468 6520 726f 6f74 206f 6620  rom the root of 
+0000e640: 7468 6520 6661 6d69 6c79 2061 6e64 2061  the family and a
+0000e650: 6c6c 2074 6865 2067 656e 6520 6475 706c  ll the gene dupl
+0000e660: 6963 6174 6573 2061 7265 2074 616b 656e  icates are taken
+0000e670: 2069 6e74 6f20 6163 636f 756e 7420 696e   into account in
+0000e680: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+0000e690: 7375 623e 2065 7374 696d 6174 696f 6e20  sub> estimation 
+0000e6a0: 7072 6f63 6573 732e 2054 6865 206f 7074  process. The opt
+0000e6b0: 696f 6e20 602d 2d73 7472 6970 5f67 6170  ion `--strip_gap
+0000e6c0: 7360 2063 616e 2072 656d 6f76 6520 616c  s` can remove al
+0000e6d0: 6c20 7468 6520 6761 702d 636f 6e74 6169  l the gap-contai
+0000e6e0: 6e69 6e67 2063 6f6c 756d 6e73 2c20 7769  ning columns, wi
+0000e6f0: 7468 206f 7220 7769 7468 6f75 7420 7768  th or without wh
+0000e700: 6963 6820 7468 6520 7265 7375 6c74 206f  ich the result o
+0000e710: 6620 226e 6f6e 2d70 6169 7277 6973 6520  f "non-pairwise 
+0000e720: 6d6f 6465 2220 776f 6e27 7420 6265 2061  mode" won't be a
+0000e730: 6666 6563 7465 642c 2077 6869 6c65 2077  ffected, while w
+0000e740: 6974 6820 7768 6963 6820 7468 6520 7265  ith which the re
+0000e750: 7375 6c74 206f 6620 2270 6169 7277 6973  sult of "pairwis
+0000e760: 6520 6d6f 6465 2220 7769 6c6c 2062 6520  e mode" will be 
+0000e770: 616c 7465 7265 642e 2054 6865 206f 7074  altered. The opt
+0000e780: 696f 6e20 602d 2d61 6c69 676e 6572 6020  ion `--aligner` 
+0000e790: 616e 6420 602d 2d61 6c6e 5f6f 7074 696f  and `--aln_optio
+0000e7a0: 6e73 6020 7768 6963 6820 6465 6369 6465  ns` which decide
+0000e7b0: 2077 6869 6368 2061 6c69 676e 6d65 6e74   which alignment
+0000e7c0: 2070 726f 6772 616d 2074 6f20 6265 2075   program to be u
+0000e7d0: 7365 6420 616e 6420 7768 6963 6820 7061  sed and which pa
+0000e7e0: 7261 6d65 7465 7220 746f 2062 6520 7365  rameter to be se
+0000e7f0: 742c 2077 696c 6c20 6861 7665 2069 6d70  t, will have imp
+0000e800: 6163 7420 6f6e 2074 6865 202a 4b2a 3c73  act on the *K*<s
+0000e810: 7562 3e53 3c2f 7375 623e 2072 6573 756c  ub>S</sub> resul
+0000e820: 7473 2c20 6e6f 7465 6420 7468 6174 2074  ts, noted that t
+0000e830: 6865 2064 6566 6175 6c74 2070 726f 6772  he default progr
+0000e840: 616d 2069 7320 606d 6166 6674 6020 616e  am is `mafft` an
+0000e850: 6420 7468 6520 7061 7261 6d65 7465 7220  d the parameter 
+0000e860: 6973 2060 2d2d 6175 746f 602e 2054 6865  is `--auto`. The
+0000e870: 206f 7074 696f 6e20 602d 2d74 7265 655f   option `--tree_
+0000e880: 6d65 7468 6f64 6020 616e 6420 602d 2d74  method` and `--t
+0000e890: 7265 655f 6f70 7469 6f6e 7360 2064 6563  ree_options` dec
+0000e8a0: 6964 6520 7768 6963 6820 6765 6e65 2074  ide which gene t
+0000e8b0: 7265 6520 696e 6665 7265 6e63 6520 7072  ree inference pr
+0000e8c0: 6f67 7261 6d20 746f 2062 6520 7573 6564  ogram to be used
+0000e8d0: 2061 6e64 2077 6869 6368 2070 6172 616d   and which param
+0000e8e0: 6574 6572 2074 6f20 6265 2073 6574 2c20  eter to be set, 
+0000e8f0: 776f 6e27 7420 6166 6665 6374 2074 6865  won't affect the
+0000e900: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000e910: 2065 7374 696d 6174 696f 6e20 6974 7365   estimation itse
+0000e920: 6c66 2062 7574 2074 6865 2072 6573 756c  lf but the resul
+0000e930: 7420 6f66 2064 652d 7265 6475 6e64 616e  t of de-redundan
+0000e940: 6379 2c20 6e6f 7465 6420 7468 6174 2077  cy, noted that w
+0000e950: 6520 696d 706c 656d 656e 7465 6420 6120  e implemented a 
+0000e960: 6275 696c 742d 696e 2067 656e 6520 7472  built-in gene tr
+0000e970: 6565 2069 6e66 6572 656e 6365 206d 6574  ee inference met
+0000e980: 686f 6420 6261 7365 6420 6f6e 2074 6865  hod based on the
+0000e990: 2041 7665 7261 6765 204c 696e 6b61 6765   Average Linkage
+0000e9a0: 2043 6c75 7374 6572 696e 6720 2841 4c43   Clustering (ALC
+0000e9b0: 2920 2874 6875 7320 6120 6469 7374 616e  ) (thus a distan
+0000e9c0: 6365 2d62 6173 6564 2074 7265 6529 2077  ce-based tree) w
+0000e9d0: 6974 6820 7468 6520 602d 2d74 7265 655f  ith the `--tree_
+0000e9e0: 6d65 7468 6f64 6020 7365 7420 6173 2022  method` set as "
+0000e9f0: 636c 7573 7465 7222 2e20 5477 6f20 6d65  cluster". Two me
+0000ea00: 7468 6f64 7320 6f66 2064 652d 7265 6475  thods of de-redu
+0000ea10: 6e64 616e 6379 2061 7265 2069 6d70 6c65  ndancy are imple
+0000ea20: 6d65 6e74 6564 2069 6e20 6077 6764 2076  mented in `wgd v
+0000ea30: 3260 2c20 6e61 6d65 6c79 206e 6f64 652d  2`, namely node-
+0000ea40: 7765 6967 6874 6564 2061 6e64 206e 6f64  weighted and nod
+0000ea50: 652d 6176 6572 6167 6564 206d 6574 686f  e-averaged metho
+0000ea60: 6473 2e20 5468 6520 6e6f 6465 2d77 6569  ds. The node-wei
+0000ea70: 6768 7465 6420 6d65 7468 6f64 2061 6368  ghted method ach
+0000ea80: 6965 7665 7320 7468 6520 6465 2d72 6564  ieves the de-red
+0000ea90: 756e 6461 6e63 7920 7669 6120 7765 6967  undancy via weig
+0000eaa0: 6869 6e67 2074 6865 202a 4b2a 3c73 7562  hing the *K*<sub
+0000eab0: 3e53 3c2f 7375 623e 2076 616c 7565 2061  >S</sub> value a
+0000eac0: 7373 6f63 6961 7465 6420 7769 7468 2065  ssociated with e
+0000ead0: 6163 6820 6765 6e65 2070 6169 7220 7375  ach gene pair su
+0000eae0: 6368 2074 6861 7420 7468 6520 7765 6967  ch that the weig
+0000eaf0: 6874 206f 6620 6120 7369 6e67 6c65 2064  ht of a single d
+0000eb00: 7570 6c69 6361 7469 6f6e 2065 7665 6e74  uplication event
+0000eb10: 2073 756d 7320 7570 2074 6f20 3120 286e   sums up to 1 (n
+0000eb20: 6f74 6564 2074 6861 7420 7468 6520 6e75  oted that the nu
+0000eb30: 6d62 6572 206f 6620 2a4b 2a3c 7375 623e  mber of *K*<sub>
+0000eb40: 533c 2f73 7562 3e20 6573 7469 6d61 7469  S</sub> estimati
+0000eb50: 6f6e 2072 656d 6169 6e73 2074 6865 2073  on remains the s
+0000eb60: 616d 6529 2077 6869 6c65 2074 6865 206e  ame) while the n
+0000eb70: 6f64 652d 6176 6572 6167 6564 206d 6574  ode-averaged met
+0000eb80: 686f 6420 7265 616c 697a 6573 2074 6865  hod realizes the
+0000eb90: 2064 652d 7265 6475 6e64 616e 6379 2076   de-redundancy v
+0000eba0: 6961 2063 616c 6375 6c61 7469 6e67 2070  ia calculating p
+0000ebb0: 6572 2067 656e 6520 7472 6565 206e 6f64  er gene tree nod
+0000ebc0: 6520 6f6e 6520 6176 6572 6167 6564 202a  e one averaged *
+0000ebd0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2076  K*<sub>S</sub> v
+0000ebe0: 616c 7565 2074 6f20 7265 7072 6573 656e  alue to represen
+0000ebf0: 7420 7468 6520 6167 6520 6f66 2065 6163  t the age of eac
+0000ec00: 6820 6475 706c 6963 6174 696f 6e20 6576  h duplication ev
+0000ec10: 656e 742e 2054 6865 206f 7074 696f 6e20  ent. The option 
+0000ec20: 602d 2d6e 6f64 655f 6176 6572 6167 6560  `--node_average`
+0000ec30: 2063 616e 2062 6520 7365 7420 746f 2063   can be set to c
+0000ec40: 686f 6f73 6520 7468 6520 6e6f 6465 2d61  hoose the node-a
+0000ec50: 7665 7261 6765 6420 7761 7920 6f66 2064  veraged way of d
+0000ec60: 652d 7265 6475 6e64 616e 6379 2e20 4469  e-redundancy. Di
+0000ec70: 6666 6572 656e 7420 6d65 7468 6f64 7320  fferent methods 
+0000ec80: 6f66 2064 652d 7265 6475 6e64 616e 6379  of de-redundancy
+0000ec90: 2077 696c 6c20 6861 7665 2069 6d70 6163   will have impac
+0000eca0: 7420 6f6e 2074 6865 2064 6574 6563 7469  t on the detecti
+0000ecb0: 6f6e 206f 6620 5747 4420 7369 676e 616c  on of WGD signal
+0000ecc0: 732c 2077 6869 6368 2068 6173 2062 6565  s, which has bee
+0000ecd0: 6e20 696e 7665 7374 6967 6174 6564 2069  n investigated i
+0000ece0: 6e20 7468 6973 205b 6c69 7465 7261 7475  n this [literatu
+0000ecf0: 7265 5d28 6874 7470 733a 2f2f 646f 692e  re](https://doi.
+0000ed00: 6f72 672f 3130 2e31 3039 332f 6762 652f  org/10.1093/gbe/
+0000ed10: 6576 7932 3030 292e 2054 6865 2070 6172  evy200). The par
+0000ed20: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
+0000ed30: 7265 2061 7265 2074 6f20 7061 7261 6c6c  re are to parall
+0000ed40: 656c 697a 6520 7468 6520 616e 616c 7973  elize the analys
+0000ed50: 6973 2066 6f72 2065 6163 6820 6765 6e65  is for each gene
+0000ed60: 2066 616d 696c 7920 616e 6420 7468 7573   family and thus
+0000ed70: 2073 7567 6765 7374 6564 2074 6f20 6265   suggested to be
+0000ed80: 2073 6574 2061 7320 6d75 6368 2061 7320   set as much as 
+0000ed90: 7468 6520 6e75 6d62 6572 206f 6620 6765  the number of ge
+0000eda0: 6e65 2066 616d 696c 6965 732e 0a0a 2a2a  ne families...**
+0000edb0: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
+0000edc0: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
+0000edd0: 2063 6f6d 6d61 6e64 2061 7320 6265 6c6f   command as belo
+0000ede0: 772a 2a0a 0a60 6060 0a77 6764 206b 7364  w**..```.wgd ksd
+0000edf0: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
+0000ee00: 6365 0a60 6060 0a0a 2a2a 5468 6520 636f  ce.```..**The co
+0000ee10: 6e73 7472 7563 7469 6f6e 206f 6620 6f72  nstruction of or
+0000ee20: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
+0000ee30: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
+0000ee40: 7374 7269 6275 7469 6f6e 2a2a 0a60 6060  stribution**.```
+0000ee50: 0a77 6764 206b 7364 2066 616d 696c 6965  .wgd ksd familie
+0000ee60: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
+0000ee70: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+0000ee80: 282d 2d72 6577 6569 6768 7429 0a60 6060  (--reweight).```
+0000ee90: 0a0a 4672 6f6d 2070 6172 616c 6f67 6f75  ..From paralogou
+0000eea0: 7320 746f 206f 7274 686f 6c6f 676f 7573  s to orthologous
+0000eeb0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000eec0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
+0000eed0: 6e2c 2075 7365 7273 206f 6e6c 7920 6e65  n, users only ne
+0000eee0: 6564 2074 6f20 7072 6f76 6964 6520 6d6f  ed to provide mo
+0000eef0: 7265 2063 6473 2066 696c 6573 2e20 4e6f  re cds files. No
+0000ef00: 7465 2074 6861 7420 7769 7468 206f 7274  te that with ort
+0000ef10: 686f 6c6f 676f 7573 2067 656e 6520 6661  hologous gene fa
+0000ef20: 6d69 6c69 6573 2074 6865 2077 6569 6768  milies the weigh
+0000ef30: 7469 6e67 206d 6574 686f 6420 6361 6e20  ting method can 
+0000ef40: 6265 2073 6574 2074 6f20 6265 2063 616c  be set to be cal
+0000ef50: 6375 6c61 7465 6420 7065 7220 7370 6563  culated per spec
+0000ef60: 6965 7320 7061 6972 2069 6e73 7465 6164  ies pair instead
+0000ef70: 206f 6620 636f 6e73 6964 6572 696e 6720   of considering 
+0000ef80: 7468 6520 7768 6f6c 6520 6661 6d69 6c79  the whole family
+0000ef90: 2062 6563 6175 7365 2077 6865 6e20 706c   because when pl
+0000efa0: 6f74 7469 6e67 206f 7274 686f 6c6f 676f  otting orthologo
+0000efb0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+0000efc0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+0000efd0: 696f 6e20 6265 7477 6565 6e20 7477 6f20  ion between two 
+0000efe0: 7370 6563 6965 7320 7468 6520 7765 6967  species the weig
+0000eff0: 6874 2063 616c 6375 6c61 7465 6420 6672  ht calculated fr
+0000f000: 6f6d 2074 6869 7320 7370 6563 6966 6963  om this specific
+0000f010: 2073 7065 6369 6573 2070 6169 7220 7368   species pair sh
+0000f020: 6f75 6c64 2062 6520 636f 6e73 6572 7665  ould be conserve
+0000f030: 6420 7768 696c 6520 7468 6520 6f6e 6520  d while the one 
+0000f040: 6361 6c63 756c 6174 6564 2066 726f 6d20  calculated from 
+0000f050: 7468 6520 7768 6f6c 6520 6661 6d69 6c79  the whole family
+0000f060: 2077 696c 6c20 7661 7279 2077 6974 6820   will vary with 
+0000f070: 7468 6520 6e75 6d62 6572 206f 6620 7370  the number of sp
+0000f080: 6563 6965 732e 2054 6f20 696e 6974 6961  ecies. To initia
+0000f090: 7465 2074 6865 2077 6569 6768 7469 6e67  te the weighting
+0000f0a0: 2070 6572 2073 7065 6369 6573 2070 6169   per species pai
+0000f0b0: 722c 2074 6865 206f 7074 696f 6e20 602d  r, the option `-
+0000f0c0: 2d72 6577 6569 6768 7460 2063 616e 2062  -reweight` can b
+0000f0d0: 6520 7365 742e 0a0a 2a2a 4120 7375 6767  e set...**A sugg
+0000f0e0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+0000f0f0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
+0000f100: 6e64 2061 7320 6265 6c6f 772a 2a0a 0a60  nd as below**..`
+0000f110: 6060 0a77 6764 206b 7364 2066 616d 696c  ``.wgd ksd famil
+0000f120: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+0000f130: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+0000f140: 330a 6060 600a 0a2a 2a54 6865 2063 6f6e  3.```..**The con
+0000f150: 7374 7275 6374 696f 6e20 6f66 202a 4b2a  struction of *K*
+0000f160: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000f170: 2064 6973 7472 6962 7574 696f 6e20 7769   distribution wi
+0000f180: 7468 2072 6174 6520 636f 7272 6563 7469  th rate correcti
+0000f190: 6f6e 2a2a 0a60 6060 0a77 6764 206b 7364  on**.```.wgd ksd
+0000f1a0: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
+0000f1b0: 6365 3120 7365 7175 656e 6365 3220 7365  ce1 sequence2 se
+0000f1c0: 7175 656e 6365 3320 2d2d 666f 6375 7332  quence3 --focus2
+0000f1d0: 616c 6c20 7365 7175 656e 6365 3120 2d73  all sequence1 -s
+0000f1e0: 7020 7370 6461 7461 202d 2d65 7874 7261  p spdata --extra
+0000f1f0: 7061 7261 6e6f 6d65 6b73 2070 6172 616e  paranomeks paran
+0000f200: 6f6d 654b 7364 6174 6120 282d 2d70 6c6f  omeKsdata (--plo
+0000f210: 7465 6c6d 6d20 2d2d 706c 6f74 6170 676d  telmm --plotapgm
+0000f220: 6d20 2d2d 616e 6368 6f72 706f 696e 7473  m --anchorpoints
+0000f230: 2061 7064 6174 6120 2d2d 7265 7765 6967   apdata --reweig
+0000f240: 6874 202d 2d6f 6e6c 7972 6f6f 746f 7574  ht --onlyrootout
+0000f250: 290a 6060 600a 0a49 6e73 7069 7265 6420  ).```..Inspired 
+0000f260: 6279 2074 6865 2072 6174 6520 636f 7272  by the rate corr
+0000f270: 6563 7469 6f6e 2061 6c67 6f72 6974 686d  ection algorithm
+0000f280: 2069 6e20 5b6b 7372 6174 6573 5d28 6874   in [ksrates](ht
+0000f290: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
+0000f2a0: 2f56 4942 2d50 5342 2f6b 7372 6174 6573  /VIB-PSB/ksrates
+0000f2b0: 292c 2077 6520 696d 706c 656d 656e 7465  ), we implemente
+0000f2c0: 6420 7468 6520 7261 7465 2063 6f72 7265  d the rate corre
+0000f2d0: 6374 696f 6e20 616e 616c 7973 6973 2061  ction analysis a
+0000f2e0: 6c73 6f20 696e 2060 7767 6420 7632 602c  lso in `wgd v2`,
+0000f2f0: 2077 6869 6368 2069 7320 6d6f 7374 6c79   which is mostly
+0000f300: 2074 6865 2073 616d 6520 6173 205b 6b73   the same as [ks
+0000f310: 7261 7465 735d 2868 7474 7073 3a2f 2f67  rates](https://g
+0000f320: 6974 6875 622e 636f 6d2f 5649 422d 5053  ithub.com/VIB-PS
+0000f330: 422f 6b73 7261 7465 7329 2062 7574 2064  B/ksrates) but d
+0000f340: 6966 6665 7273 2069 6e20 7468 6520 6361  iffers in the ca
+0000f350: 6c63 756c 6174 696f 6e20 6f66 2074 6865  lculation of the
+0000f360: 2073 7461 6e64 6172 6420 6465 7669 6174   standard deviat
+0000f370: 696f 6e20 6f66 2072 6573 6361 6c65 6420  ion of rescaled 
+0000f380: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000f390: 6167 6573 2e20 546f 2070 6572 666f 726d  ages. To perform
+0000f3a0: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
+0000f3b0: 7469 6f6e 2061 6e61 6c79 7369 732c 2075  tion analysis, u
+0000f3c0: 7365 7273 2063 616e 2075 7365 2062 6f74  sers can use bot
+0000f3d0: 6820 7468 6520 6077 6764 206b 7364 6020  h the `wgd ksd` 
+0000f3e0: 616e 6420 6077 6764 2076 697a 6020 7072  and `wgd viz` pr
+0000f3f0: 6f67 7261 6d2e 2046 6f72 2074 6865 2060  ogram. For the `
+0000f400: 7767 6420 6b73 6460 2070 726f 6772 616d  wgd ksd` program
+0000f410: 2c20 7573 6572 7320 6e65 6564 2074 6f20  , users need to 
+0000f420: 7072 6f76 6964 6520 6120 7370 6563 6965  provide a specie
+0000f430: 7320 7472 6565 2076 6961 2074 6865 206f  s tree via the o
+0000f440: 7074 696f 6e20 602d 2d73 7065 6369 6573  ption `--species
+0000f450: 7472 6565 6020 6f6e 2077 6869 6368 2072  tree` on which r
+0000f460: 6174 6520 636f 7272 6563 7469 6f6e 2063  ate correction c
+0000f470: 616e 2062 6520 636f 6e64 7563 7465 642e  an be conducted.
+0000f480: 204e 6f74 6520 7468 6174 2075 6e6e 6563   Note that unnec
+0000f490: 6573 7361 7279 2062 7261 636b 6574 7320  essary brackets 
+0000f4a0: 6d69 6768 7420 6c65 6164 2074 6f20 756e  might lead to un
+0000f4b0: 6578 7065 6374 6564 2065 7272 6f72 732c  expected errors,
+0000f4c0: 2066 6f72 2069 6e73 7461 6e63 6520 6120   for instance a 
+0000f4d0: 7472 6565 2060 2841 2c28 422c 4329 293b  tree `(A,(B,C));
+0000f4e0: 6020 7368 6f75 6c64 206e 6f74 2062 6520  ` should not be 
+0000f4f0: 7265 7072 6573 656e 7465 6420 6173 2060  represented as `
+0000f500: 2841 2c28 2842 2c43 2929 293b 602e 2054  (A,((B,C)));`. T
+0000f510: 6865 2073 6574 206f 6620 7370 6563 6965  he set of specie
+0000f520: 7320 7061 6972 7320 746f 2062 6520 7368  s pairs to be sh
+0000f530: 6f77 6e20 6973 2066 6c65 7869 626c 6520  own is flexible 
+0000f540: 7468 6174 2074 6865 206d 6f73 7420 636f  that the most co
+0000f550: 6e76 656e 6965 6e74 206f 7074 696f 6e20  nvenient option 
+0000f560: 6973 2060 2d2d 666f 6375 7332 616c 6c60  is `--focus2all`
+0000f570: 2077 6869 6368 2073 696d 706c 7920 7368   which simply sh
+0000f580: 6f77 7320 616c 6c20 7468 6520 706f 7373  ows all the poss
+0000f590: 6962 6c65 2066 6f63 616c 2d73 6973 7465  ible focal-siste
+0000f5a0: 7220 7370 6563 6965 7320 7061 6972 732c  r species pairs,
+0000f5b0: 206f 7220 7573 6572 7320 6361 6e20 6d61   or users can ma
+0000f5c0: 6e75 616c 6c79 2073 6574 2074 6865 2073  nually set the s
+0000f5d0: 7065 6369 6573 2070 6169 7273 2076 6961  pecies pairs via
+0000f5e0: 2074 6865 206f 7074 696f 6e20 602d 2d73   the option `--s
+0000f5f0: 7061 6972 602e 204e 6f74 6520 7468 6174  pair`. Note that
+0000f600: 2069 6620 7468 6520 7370 6563 6965 7320   if the species 
+0000f610: 7061 6972 7320 7765 7265 206d 616e 7561  pairs were manua
+0000f620: 6c6c 7920 7365 742c 2069 7420 776f 756c  lly set, it woul
+0000f630: 6420 6265 206e 6565 6465 6420 746f 2063  d be needed to c
+0000f640: 6f2d 7365 7420 7468 6520 6f70 7469 6f6e  o-set the option
+0000f650: 2060 2d2d 636c 6173 7369 6360 2e20 5468   `--classic`. Th
+0000f660: 6520 6f70 7469 6f6e 2060 2d2d 6f6e 6c79  e option `--only
+0000f670: 726f 6f74 6f75 7460 2063 616e 2062 6520  rootout` can be 
+0000f680: 7365 7420 746f 206f 6e6c 7920 636f 6e73  set to only cons
+0000f690: 6964 6572 206f 7574 6772 6f75 7020 6174  ider outgroup at
+0000f6a0: 2074 6865 2072 6f6f 742c 2069 6e73 7465   the root, inste
+0000f6b0: 6164 206f 6620 616c 6c20 7468 6520 706f  ad of all the po
+0000f6c0: 7373 6962 6c65 206f 7574 6772 6f75 7073  ssible outgroups
+0000f6d0: 2070 6572 2066 6f63 616c 2d73 6973 7465   per focal-siste
+0000f6e0: 7220 7370 6563 6965 7320 7061 6972 2c20  r species pair, 
+0000f6f0: 7768 6963 6820 6861 7320 696d 7061 6374  which has impact
+0000f700: 206f 6e20 7468 6520 6669 6e61 6c20 636f   on the final co
+0000f710: 7272 6563 7465 6420 2a4b 2a3c 7375 623e  rrected *K*<sub>
+0000f720: 533c 2f73 7562 3e20 6167 6573 2e20 5765  S</sub> ages. We
+0000f730: 2073 7567 6765 7374 206f 6620 7573 696e   suggest of usin
+0000f740: 6720 616c 6c20 7468 6520 706f 7373 6962  g all the possib
+0000f750: 6c65 206f 7574 6772 6f75 7073 2070 6572  le outgroups per
+0000f760: 2066 6f63 616c 2d73 6973 7465 7220 7370   focal-sister sp
+0000f770: 6563 6965 7320 7061 6972 2061 7320 666f  ecies pair as fo
+0000f780: 7220 6120 6c65 7373 2062 6961 7365 6420  r a less biased 
+0000f790: 7265 7375 6c74 2e20 5468 6520 7061 7261  result. The para
+0000f7a0: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
+0000f7b0: 7375 623e 2064 6174 6120 7368 6f75 6c64  sub> data should
+0000f7c0: 2062 6520 7072 6f76 6964 6564 2076 6961   be provided via
+0000f7d0: 2074 6865 206f 7074 696f 6e20 602d 2d65   the option `--e
+0000f7e0: 7874 7261 7061 7261 6e6f 6d65 6b73 602e  xtraparanomeks`.
+0000f7f0: 200a 0a53 6f6d 6520 6f74 6865 7220 6f70   ..Some other op
+0000f800: 7469 6f6e 7320 7768 6963 6820 6861 7665  tions which have
+0000f810: 206e 6f20 696d 7061 6374 206f 6e20 7468   no impact on th
+0000f820: 6520 7261 7465 2063 6f72 7265 6374 696f  e rate correctio
+0000f830: 6e20 6275 7420 6164 6420 6d6f 7265 206c  n but add more l
+0000f840: 6179 6572 7320 6f72 2063 6861 6e67 6520  ayers or change 
+0000f850: 7468 6520 6170 7065 6172 616e 6365 206f  the appearance o
+0000f860: 6e20 7468 6520 6d69 7865 6420 2a4b 2a3c  n the mixed *K*<
+0000f870: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+0000f880: 6469 7374 7269 6275 7469 6f6e 2c20 696e  distribution, in
+0000f890: 636c 7564 6520 602d 2d70 6c6f 7461 7067  clude `--plotapg
+0000f8a0: 6d6d 6020 616e 6420 602d 2d70 6c6f 7465  mm` and `--plote
+0000f8b0: 6c6d 6d60 2065 7463 2e20 5468 6520 6f70  lmm` etc. The op
+0000f8c0: 7469 6f6e 2060 2d2d 706c 6f74 6170 676d  tion `--plotapgm
+0000f8d0: 6d60 2063 616e 2062 6520 7365 7420 746f  m` can be set to
+0000f8e0: 2063 616c 6c20 7468 6520 474d 4d20 616e   call the GMM an
+0000f8f0: 616c 7973 6973 2075 706f 6e20 7468 6520  alysis upon the 
+0000f900: 616e 6368 6f72 2070 6169 7220 2a4b 2a3c  anchor pair *K*<
+0000f910: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+0000f920: 6469 7374 7269 6275 7469 6f6e 2061 6e64  distribution and
+0000f930: 2070 6c6f 7420 7468 6520 636c 7573 7465   plot the cluste
+0000f940: 7269 6e67 2072 6573 756c 7420 7570 6f6e  ring result upon
+0000f950: 2074 6865 206d 6978 6564 202a 4b2a 3c73   the mixed *K*<s
+0000f960: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+0000f970: 6973 7472 6962 7574 696f 6e2c 2077 6869  istribution, whi
+0000f980: 6368 2068 6173 2074 6f20 6265 2063 6f2d  ch has to be co-
+0000f990: 7365 7420 7769 7468 2074 6865 206f 7074  set with the opt
+0000f9a0: 696f 6e20 602d 2d61 6e63 686f 7270 6f69  ion `--anchorpoi
+0000f9b0: 6e74 7360 2070 726f 7669 6469 6e67 2074  nts` providing t
+0000f9c0: 6865 2061 6e63 686f 7220 7061 6972 7320  he anchor pairs 
+0000f9d0: 696e 666f 726d 6174 696f 6e2e 2054 6865  information. The
+0000f9e0: 206f 7074 696f 6e20 602d 2d70 6c6f 7465   option `--plote
+0000f9f0: 6c6d 6d60 2063 616e 2062 6520 7365 7420  lmm` can be set 
+0000fa00: 746f 2063 616c 6c20 7468 6520 454c 4d4d  to call the ELMM
+0000fa10: 2061 6e61 6c79 7369 7320 7570 6f6e 2074   analysis upon t
+0000fa20: 6865 2077 686f 6c65 2070 6172 616e 6f6d  he whole paranom
+0000fa30: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+0000fa40: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+0000fa50: 6f6e 2e20 2a2a 4e6f 7465 2074 6861 7420  on. **Note that 
+0000fa60: 7468 6520 7370 6563 6965 7320 6e61 6d65  the species name
+0000fa70: 7320 7072 6573 656e 7420 696e 2074 6865  s present in the
+0000fa80: 2073 7065 6369 6573 2074 7265 6520 6669   species tree fi
+0000fa90: 6c65 2073 686f 756c 6420 6d61 7463 6820  le should match 
+0000faa0: 7468 6520 6e61 6d65 7320 6f66 2074 6865  the names of the
+0000fab0: 2063 6f72 7265 7370 6f6e 6469 6e67 2073   corresponding s
+0000fac0: 6571 7565 6e63 6520 6669 6c65 732a 2a20  equence files** 
+0000fad0: 466f 7220 696e 7374 616e 6365 2c20 6769  For instance, gi
+0000fae0: 7665 6e20 7468 6520 6364 7320 6669 6c65  ven the cds file
+0000faf0: 206e 616d 6573 2027 412e 6364 7327 2c27   names 'A.cds','
+0000fb00: 422e 6364 7327 2c27 432e 6364 7327 2c20  B.cds','C.cds', 
+0000fb10: 7468 6520 7370 6563 6965 7320 7472 6565  the species tree
+0000fb20: 2063 6f75 6c64 2062 6520 2728 412e 6364   could be '(A.cd
+0000fb30: 732c 2842 2e63 6473 2c43 2e63 6473 2929  s,(B.cds,C.cds))
+0000fb40: 3b27 2072 6174 6865 7220 7468 616e 2027  ;' rather than '
+0000fb50: 2841 2c28 422c 4329 293b 272e 2054 6865  (A,(B,C));'. The
+0000fb60: 7265 2069 7320 6e6f 2072 6571 7569 7265  re is no require
+0000fb70: 6d65 6e74 2066 6f72 2074 6865 206e 616d  ment for the nam
+0000fb80: 6520 6f66 2074 6865 2070 6172 616e 6f6d  e of the paranom
+0000fb90: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+0000fba0: 3e20 6461 7461 6669 6c65 2077 6869 6368  > datafile which
+0000fbb0: 2063 616e 2062 6520 6e61 6d65 6420 696e   can be named in
+0000fbc0: 2077 6861 7465 7665 7220 6d61 6e6e 6572   whatever manner
+0000fbd0: 2075 7365 7273 2070 7265 6665 722e 2054   users prefer. T
+0000fbe0: 6865 2027 474d 4d27 2069 7320 7468 6520  he 'GMM' is the 
+0000fbf0: 6162 6272 6576 6961 7469 6f6e 206f 6620  abbreviation of 
+0000fc00: 4761 7573 7369 616e 204d 6978 7475 7265  Gaussian Mixture
+0000fc10: 204d 6f64 656c 696e 6720 7768 696c 6520   Modeling while 
+0000fc20: 7468 6520 2745 4c4d 4d27 2072 6566 6572  the 'ELMM' refer
+0000fc30: 7320 746f 2045 7870 6f6e 656e 7469 616c  s to Exponential
+0000fc40: 2d4c 6f67 6e6f 726d 616c 204d 6978 7475  -Lognormal Mixtu
+0000fc50: 7265 204d 6f64 656c 696e 6720 6173 205b  re Modeling as [
+0000fc60: 6b73 7261 7465 735d 2868 7474 7073 3a2f  ksrates](https:/
+0000fc70: 2f67 6974 6875 622e 636f 6d2f 5649 422d  /github.com/VIB-
+0000fc80: 5053 422f 6b73 7261 7465 7329 2069 6e74  PSB/ksrates) int
+0000fc90: 6572 7072 6574 732e 0a0a 5468 6572 6520  erprets...There 
+0000fca0: 6172 6520 3231 2063 6f6c 756d 6e73 2069  are 21 columns i
+0000fcb0: 6e20 7468 6520 7265 7375 6c74 2060 2e6b  n the result `.k
+0000fcc0: 732e 7473 7660 2066 696c 6520 6265 7369  s.tsv` file besi
+0000fcd0: 6465 7320 7468 6520 696e 6465 7820 636f  des the index co
+0000fce0: 6c75 6d6e 7320 6070 6169 7260 2061 7320  lumns `pair` as 
+0000fcf0: 7468 6520 756e 6971 7565 2069 6465 6e74  the unique ident
+0000fd00: 6966 6965 7220 666f 7220 6561 6368 2067  ifier for each g
+0000fd10: 656e 6520 7061 6972 2e20 5468 6520 604e  ene pair. The `N
+0000fd20: 602c 2060 5360 2c20 6064 4e60 2c20 6064  `, `S`, `dN`, `d
+0000fd30: 4e2f 6453 602c 2060 6453 602c 2060 6c60  N/dS`, `dS`, `l`
+0000fd40: 2061 6e64 2060 7460 2061 7265 2066 726f   and `t` are fro
+0000fd50: 6d20 7468 6520 636f 6465 6d6c 2072 6573  m the codeml res
+0000fd60: 756c 7473 2c20 7265 7072 6573 656e 7469  ults, representi
+0000fd70: 6e67 2074 6865 204e 2065 7374 696d 6174  ng the N estimat
+0000fd80: 652c 2074 6865 2053 2065 7374 696d 6174  e, the S estimat
+0000fd90: 652c 2074 6865 2064 4e20 6573 7469 6d61  e, the dN estima
+0000fda0: 7465 2c20 7468 6520 644e 2f64 5320 286f  te, the dN/dS (o
+0000fdb0: 6d65 6761 2920 6573 7469 6d61 7465 2c20  mega) estimate, 
+0000fdc0: 7468 6520 6453 2065 7374 696d 6174 652c  the dS estimate,
+0000fdd0: 2074 6865 206c 6f67 2d6c 696b 656c 6968   the log-likelih
+0000fde0: 6f6f 6420 616e 6420 7468 6520 7420 6573  ood and the t es
+0000fdf0: 7469 6d61 7465 2c20 7265 7370 6563 7469  timate, respecti
+0000fe00: 7665 6c79 2e20 5468 6520 6061 6c69 676e  vely. The `align
+0000fe10: 6d65 6e74 636f 7665 7261 6765 602c 2060  mentcoverage`, `
+0000fe20: 616c 6967 6e6d 656e 7469 6465 6e74 6974  alignmentidentit
+0000fe30: 7960 2061 6e64 2060 616c 6967 6e6d 656e  y` and `alignmen
+0000fe40: 746c 656e 6774 6860 2061 7265 2074 6865  tlength` are the
+0000fe50: 2069 6e66 6f72 6d61 7469 6f6e 2070 6572   information per
+0000fe60: 7461 696e 696e 6720 746f 2074 6865 2061  taining to the a
+0000fe70: 6c69 676e 6d65 6e74 2066 6f72 2065 6163  lignment for eac
+0000fe80: 6820 6661 6d69 6c79 2c20 7265 7072 6573  h family, repres
+0000fe90: 656e 7469 6e67 2074 6865 2072 6174 696f  enting the ratio
+0000fea0: 206f 6620 7468 6520 7374 7269 7070 6564   of the stripped
+0000feb0: 2061 6c69 676e 6d65 6e74 206c 656e 6774   alignment lengt
+0000fec0: 6820 636f 6d70 6172 6564 2074 6f20 7468  h compared to th
+0000fed0: 6520 6675 6c6c 2061 6c69 676e 6d65 6e74  e full alignment
+0000fee0: 206c 656e 6774 682c 2074 6865 2072 6174   length, the rat
+0000fef0: 696f 206f 6620 636f 6c75 6d6e 7320 7769  io of columns wi
+0000ff00: 7468 2069 6465 6e74 6963 616c 206e 7563  th identical nuc
+0000ff10: 6c65 6f74 6964 6573 2063 6f6d 7061 7265  leotides compare
+0000ff20: 6420 746f 2074 6865 206f 7665 7261 6c6c  d to the overall
+0000ff30: 2063 6f6c 756d 6e73 206f 6620 7468 6520   columns of the 
+0000ff40: 7374 7269 7070 6564 2061 6c69 676e 6d65  stripped alignme
+0000ff50: 6e74 2c20 616e 6420 7468 6520 6c65 6e67  nt, and the leng
+0000ff60: 7468 206f 6620 7468 6520 6675 6c6c 2061  th of the full a
+0000ff70: 6c69 676e 6d65 6e74 2c20 7265 7370 6563  lignment, respec
+0000ff80: 7469 7665 6c79 2e0a 2323 2320 7767 6420  tively..### wgd 
+0000ff90: 6d69 780a 0a2a 2a54 6865 206d 6978 7475  mix..**The mixtu
+0000ffa0: 7265 206d 6f64 656c 2063 6c75 7374 6572  re model cluster
+0000ffb0: 696e 6720 616e 616c 7973 6973 206f 6620  ing analysis of 
+0000ffc0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000ffd0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
+0000ffe0: 2a2a 0a60 6060 0a77 6764 206d 6978 206b  **.```.wgd mix k
+0000fff0: 7364 6174 6120 282d 2d6e 5f69 6e69 7420  sdata (--n_init 
+00010000: 3230 3020 2d2d 6d61 785f 6974 6572 2032  200 --max_iter 2
+00010010: 3030 202d 2d6b 735f 7261 6e67 6520 3020  00 --ks_range 0 
+00010020: 3520 2d2d 6669 6c74 6572 7320 3330 3020  5 --filters 300 
+00010030: 2d2d 6269 6e73 2035 3020 2d2d 636f 6d70  --bins 50 --comp
+00010040: 6f6e 656e 7473 2031 2034 202d 2d67 616d  onents 1 4 --gam
+00010050: 6d61 2030 2e30 3031 290a 6060 600a 0a54  ma 0.001).```..T
+00010060: 6869 7320 7061 7274 206f 6620 4761 7573  his part of Gaus
+00010070: 7369 616e 206d 6978 7475 7265 206d 6f64  sian mixture mod
+00010080: 656c 696e 6720 2847 4d4d 2920 616e 616c  eling (GMM) anal
+00010090: 7973 6973 2069 7320 696e 6865 7269 7465  ysis is inherite
+000100a0: 6420 6672 6f6d 2074 6865 206f 7269 6769  d from the origi
+000100b0: 6e61 6c20 6077 6764 6020 7072 6f67 7261  nal `wgd` progra
+000100c0: 6d2c 2062 7574 2077 7269 7465 7320 6164  m, but writes ad
+000100d0: 6469 7469 6f6e 616c 6c79 2074 6865 2070  ditionally the p
+000100e0: 726f 6261 6269 6c69 7479 206f 6620 6561  robability of ea
+000100f0: 6368 202a 4b2a 3c73 7562 3e53 3c2f 7375  ch *K*<sub>S</su
+00010100: 623e 2076 616c 7565 2069 6e74 6f20 7468  b> value into th
+00010110: 6520 6669 6e61 6c20 6461 7461 6672 616d  e final datafram
+00010120: 652e 2042 6173 6963 616c 6c79 2c20 7573  e. Basically, us
+00010130: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
+00010140: 6964 6520 7769 7468 2061 2028 6e6f 726d  ide with a (norm
+00010150: 616c 6c79 2066 726f 6d20 7768 6f6c 652d  ally from whole-
+00010160: 7061 7261 6e6f 6d65 206f 7220 616e 6368  paranome or anch
+00010170: 6f72 2d70 6169 7273 2920 2a4b 2a3c 7375  or-pairs) *K*<su
+00010180: 623e 533c 2f73 7562 3e20 6461 7461 6669  b>S</sub> datafi
+00010190: 6c65 2061 6e64 2074 6865 2047 4d4d 2061  le and the GMM a
+000101a0: 6e61 6c79 7369 7320 7769 6c6c 2062 6520  nalysis will be 
+000101b0: 636f 6e64 7563 7465 6420 7570 6f6e 2074  conducted upon t
+000101c0: 6865 2064 6174 6166 696c 652e 2053 6f6d  he datafile. Som
+000101d0: 6520 7061 7261 6d65 7465 7273 2063 616e  e parameters can
+000101e0: 2061 6666 6563 7420 7468 6520 7265 7375   affect the resu
+000101f0: 6c74 732c 2069 6e63 6c75 6469 6e67 2060  lts, including `
+00010200: 2d2d 6e5f 696e 6974 602c 2077 6869 6368  --n_init`, which
+00010210: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
+00010220: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
+00010230: 6961 6c69 7a61 7469 6f6e 7320 2864 6566  ializations (def
+00010240: 6175 6c74 2032 3030 292c 2060 2d2d 6d61  ault 200), `--ma
+00010250: 785f 6974 6572 602c 2077 6869 6368 2073  x_iter`, which s
+00010260: 6574 7320 7468 6520 6d61 7869 6d75 6d20  ets the maximum 
+00010270: 6e75 6d62 6572 206f 6620 6974 6572 6174  number of iterat
+00010280: 696f 6e73 2028 6465 6661 756c 7420 3230  ions (default 20
+00010290: 3029 2c20 602d 2d6d 6574 686f 6460 2c20  0), `--method`, 
+000102a0: 7768 6963 6820 6465 7465 726d 696e 6573  which determines
+000102b0: 2077 6869 6368 2063 6c75 7374 6572 696e   which clusterin
+000102c0: 6720 6d65 7468 6f64 2074 6f20 7573 6520  g method to use 
+000102d0: 2864 6566 6175 6c74 2067 6d6d 292c 2060  (default gmm), `
+000102e0: 2d2d 6761 6d6d 6160 2c20 7768 6963 6820  --gamma`, which 
+000102f0: 7365 7473 2074 6865 2067 616d 6d61 2070  sets the gamma p
+00010300: 6172 616d 6574 6572 2066 6f72 2074 6865  arameter for the
+00010310: 2062 676d 6d20 6d6f 6465 6c20 2864 6566   bgmm model (def
+00010320: 6175 6c74 2030 2e30 3031 292c 2060 2d2d  ault 0.001), `--
+00010330: 636f 6d70 6f6e 656e 7473 602c 2077 6869  components`, whi
+00010340: 6368 2073 6574 7320 7468 6520 7261 6e67  ch sets the rang
+00010350: 6520 6f66 2074 6865 206e 756d 6265 7220  e of the number 
+00010360: 6f66 2063 6f6d 706f 6e65 6e74 7320 746f  of components to
+00010370: 2066 6974 2028 6465 6661 756c 7420 3120   fit (default 1 
+00010380: 3429 2c20 616e 6420 7468 6520 6461 7461  4), and the data
+00010390: 2066 696c 7465 7269 6e67 2070 6172 616d   filtering param
+000103a0: 6574 6572 7320 602d 2d66 696c 7465 7273  eters `--filters
+000103b0: 6020 7768 6963 6820 6669 6c74 6572 7320  ` which filters 
+000103c0: 6461 7461 2062 6173 6564 206f 6e20 616c  data based on al
+000103d0: 6967 6e6d 656e 7420 6c65 6e67 7468 2c20  ignment length, 
+000103e0: 602d 2d6b 735f 7261 6e67 6560 2077 6869  `--ks_range` whi
+000103f0: 6368 2066 696c 7465 7273 2064 6174 6120  ch filters data 
+00010400: 6261 7365 6420 6f6e 202a 4b2a 3c73 7562  based on *K*<sub
+00010410: 3e53 3c2f 7375 623e 2076 616c 7565 7320  >S</sub> values 
+00010420: 616e 6420 7468 6520 7061 7261 6d65 7465  and the paramete
+00010430: 7220 602d 2d62 696e 7360 2077 6869 6368  r `--bins` which
+00010440: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
+00010450: 206f 6620 6269 6e73 2069 6e20 2a4b 2a3c   of bins in *K*<
+00010460: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00010470: 7269 6275 7469 6f6e 2028 6465 6661 756c  ribution (defaul
+00010480: 7420 3530 292e 0a0a 2a2a 4120 7375 6767  t 50)...**A sugg
+00010490: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+000104a0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
+000104b0: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
+000104c0: 6f77 2a2a 0a0a 6060 600a 7767 6420 6d69  ow**..```.wgd mi
+000104d0: 7820 6b73 6461 7461 0a60 6060 0a0a 2323  x ksdata.```..##
+000104e0: 2320 7767 6420 7065 616b 0a0a 2a2a 5468  # wgd peak..**Th
+000104f0: 6520 7365 6172 6368 206f 6620 6372 6564  e search of cred
+00010500: 6961 626c 6520 2a4b 2a3c 7375 623e 533c  iable *K*<sub>S<
+00010510: 2f73 7562 3e20 7261 6e67 6520 7573 6564  /sub> range used
+00010520: 2069 6e20 5747 4420 6461 7469 6e67 2a2a   in WGD dating**
+00010530: 0a60 6060 0a77 6764 2070 6561 6b20 6b73  .```.wgd peak ks
+00010540: 6461 7461 202d 6170 2061 7064 6174 6120  data -ap apdata 
+00010550: 2d73 6d20 736d 6461 7461 202d 6c65 206c  -sm smdata -le l
+00010560: 6564 6174 6120 2d6d 7020 6d70 6461 7461  edata -mp mpdata
+00010570: 202d 2d68 6575 7269 7374 6963 2028 2d2d   --heuristic (--
+00010580: 616c 6967 6e66 696c 7465 7220 302e 3020  alignfilter 0.0 
+00010590: 3020 302e 3020 2d2d 6b73 7261 6e67 6520  0 0.0 --ksrange 
+000105a0: 3020 3520 2d2d 6269 6e5f 7769 6474 6820  0 5 --bin_width 
+000105b0: 302e 3120 2d2d 6775 6964 6520 7365 676d  0.1 --guide segm
+000105c0: 656e 7420 2d2d 7072 6f6d 696e 656e 6365  ent --prominence
+000105d0: 5f63 7574 6f66 6620 302e 3120 2d2d 7265  _cutoff 0.1 --re
+000105e0: 6c5f 6865 6967 6874 2030 2e34 202d 2d63  l_height 0.4 --c
+000105f0: 6920 3935 202d 2d68 6472 2039 3520 2d2d  i 95 --hdr 95 --
+00010600: 6b73 6375 746f 6666 2035 290a 6060 600a  kscutoff 5).```.
+00010610: 0a41 7320 6d65 6e74 696f 6e65 6420 7072  .As mentioned pr
+00010620: 6576 696f 7573 6c79 2c20 6120 6865 7572  eviously, a heur
+00010630: 6973 7469 6320 6d65 7468 6f64 2061 6e64  istic method and
+00010640: 2061 2063 6f6c 6c69 6e65 6172 2073 6567   a collinear seg
+00010650: 6d65 6e74 732d 6775 6964 6564 2061 6e63  ments-guided anc
+00010660: 686f 7220 7061 6972 2063 6c75 7374 6572  hor pair cluster
+00010670: 696e 6720 666f 7220 7468 6520 7365 6172  ing for the sear
+00010680: 6368 206f 6620 6372 6564 6961 626c 6520  ch of crediable 
+00010690: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000106a0: 7261 6e67 6520 7573 6564 2069 6e20 5747  range used in WG
+000106b0: 4420 6461 7469 6e67 2061 7265 2069 6d70  D dating are imp
+000106c0: 6c65 6d65 6e74 6564 2069 6e20 6077 6764  lemented in `wgd
+000106d0: 2076 3260 2e20 5573 6572 7320 6e65 6564   v2`. Users need
+000106e0: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
+000106f0: 616e 6368 6f72 2070 6f69 6e74 732c 2073  anchor points, s
+00010700: 6567 6d65 6e74 732c 206c 6973 7473 6567  egments, listseg
+00010710: 6d65 6e74 732c 206d 756c 7469 706c 6963  ments, multiplic
+00010720: 6f6e 7061 6972 7320 6461 7461 6669 6c65  onpairs datafile
+00010730: 2066 726f 6d20 6069 2d61 6468 6f72 6560   from `i-adhore`
+00010740: 2074 6f20 6163 6869 6576 6520 7468 6520   to achieve the 
+00010750: 636c 7573 7465 7269 6e67 2066 756e 6374  clustering funct
+00010760: 696f 6e2e 2053 6f6d 6520 7061 7261 6d65  ion. Some parame
+00010770: 7465 7273 2074 6861 7420 6361 6e20 696d  ters that can im
+00010780: 7061 6374 2074 6865 2072 6573 756c 7473  pact the results
+00010790: 2069 6e63 6c75 6465 2060 2d2d 616c 6967   include `--alig
+000107a0: 6e66 696c 7465 7260 2c20 7768 6963 6820  nfilter`, which 
+000107b0: 6669 6c74 6572 7320 7468 6520 6461 7461  filters the data
+000107c0: 2062 6173 6564 206f 6e20 616c 6967 6e6d   based on alignm
+000107d0: 656e 7420 6964 656e 7469 7479 2c20 6c65  ent identity, le
+000107e0: 6e67 7468 2061 6e64 2063 6f76 6572 6167  ngth and coverag
+000107f0: 652c 2060 2d2d 6b73 7261 6e67 6560 2c20  e, `--ksrange`, 
+00010800: 7768 6963 6820 7365 7473 2074 6865 2072  which sets the r
+00010810: 616e 6765 206f 6620 4b73 2074 6f20 6265  ange of Ks to be
+00010820: 2061 6e61 6c79 7a65 642c 2060 2d2d 6269   analyzed, `--bi
+00010830: 6e5f 7769 6474 6860 2c20 7768 6963 6820  n_width`, which 
+00010840: 7365 7473 2074 6865 2062 616e 6477 6964  sets the bandwid
+00010850: 7468 206f 6620 2a4b 2a3c 7375 623e 533c  th of *K*<sub>S<
+00010860: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+00010870: 6f6e 2c20 602d 2d77 6569 6768 7473 5f6f  on, `--weights_o
+00010880: 7574 6c69 6572 735f 696e 636c 7564 6564  utliers_included
+00010890: 6020 7768 6963 6820 6465 7465 726d 696e  ` which determin
+000108a0: 6573 2077 6865 7468 6572 2074 6f20 696e  es whether to in
+000108b0: 636c 7564 6520 2a4b 2a3c 7375 623e 533c  clude *K*<sub>S<
+000108c0: 2f73 7562 3e20 6f75 746c 6965 7273 2028  /sub> outliers (
+000108d0: 7768 6f73 6520 7661 6c75 6520 6973 206f  whose value is o
+000108e0: 7665 7220 3529 2069 6e20 616e 616c 7973  ver 5) in analys
+000108f0: 6973 2c20 602d 2d6d 6574 686f 6460 2077  is, `--method` w
+00010900: 6869 6368 2064 6574 6572 6d69 6e65 7320  hich determines 
+00010910: 7768 6963 6820 636c 7573 7465 7269 6e67  which clustering
+00010920: 206d 6574 686f 6420 746f 2075 7365 2028   method to use (
+00010930: 6465 6661 756c 7420 676d 6d29 2c20 602d  default gmm), `-
+00010940: 2d73 6565 6460 2077 6869 6368 2073 6574  -seed` which set
+00010950: 7320 7468 6520 7261 6e64 6f6d 2073 6565  s the random see
+00010960: 6420 6769 7665 6e20 746f 2069 6e69 7469  d given to initi
+00010970: 616c 697a 6174 696f 6e20 2864 6566 6175  alization (defau
+00010980: 6c74 2032 3335 3238 3930 292c 2060 2d2d  lt 2352890), `--
+00010990: 6e5f 696e 6974 602c 2077 6869 6368 2073  n_init`, which s
+000109a0: 6574 7320 7468 6520 6e75 6d62 6572 206f  ets the number o
+000109b0: 6620 6b2d 6d65 616e 7320 696e 6974 6961  f k-means initia
+000109c0: 6c69 7a61 7469 6f6e 7320 2864 6566 6175  lizations (defau
+000109d0: 6c74 2032 3030 292c 2060 2d2d 656d 5f69  lt 200), `--em_i
+000109e0: 7465 7260 2c20 7768 6963 6820 7365 7473  ter`, which sets
+000109f0: 2074 6865 206d 6178 696d 756d 206e 756d   the maximum num
+00010a00: 6265 7220 6f66 2069 7465 7261 7469 6f6e  ber of iteration
+00010a10: 7320 2864 6566 6175 6c74 2032 3030 292c  s (default 200),
+00010a20: 2060 2d2d 6761 6d6d 6160 2c20 7768 6963   `--gamma`, whic
+00010a30: 6820 7365 7473 2074 6865 2067 616d 6d61  h sets the gamma
+00010a40: 2070 6172 616d 6574 6572 2066 6f72 2074   parameter for t
+00010a50: 6865 2062 676d 6d20 6d6f 6465 6c20 2864  he bgmm model (d
+00010a60: 6566 6175 6c74 2030 2e30 3031 292c 2060  efault 0.001), `
+00010a70: 2d2d 636f 6d70 6f6e 656e 7473 602c 2077  --components`, w
+00010a80: 6869 6368 2073 6574 7320 7468 6520 7261  hich sets the ra
+00010a90: 6e67 6520 6f66 2074 6865 206e 756d 6265  nge of the numbe
+00010aa0: 7220 6f66 2063 6f6d 706f 6e65 6e74 7320  r of components 
+00010ab0: 746f 2066 6974 2028 6465 6661 756c 7420  to fit (default 
+00010ac0: 3120 3429 2c20 602d 2d77 6569 6768 7465  1 4), `--weighte
+00010ad0: 6460 2077 6869 6368 2064 6574 6572 6d69  d` which determi
+00010ae0: 6e65 7320 7768 6574 6865 7220 746f 2075  nes whether to u
+00010af0: 7365 206e 6f64 652d 7765 6967 6874 6564  se node-weighted
+00010b00: 206d 6574 686f 6420 666f 7220 6465 2d72   method for de-r
+00010b10: 6564 756e 6461 6e63 792c 2060 2d2d 6775  edundancy, `--gu
+00010b20: 6964 6560 2077 6869 6368 2064 6574 6572  ide` which deter
+00010b30: 6d69 6e65 7320 7768 6963 6820 7265 6769  mines which regi
+00010b40: 6d65 2072 6573 6964 696e 6720 616e 6368  me residing anch
+00010b50: 6f72 7320 746f 2062 6520 7573 6564 2028  ors to be used (
+00010b60: 6465 6661 756c 7420 5365 676d 656e 7429  default Segment)
+00010b70: 2c20 602d 2d70 726f 6d69 6e65 6e63 655f  , `--prominence_
+00010b80: 6375 746f 6666 6020 7768 6963 6820 7365  cutoff` which se
+00010b90: 7473 2074 6865 2070 726f 6d69 6e65 6e63  ts the prominenc
+00010ba0: 6520 6375 746f 6666 206f 6620 6163 6365  e cutoff of acce
+00010bb0: 7074 6162 6c65 2070 6561 6b73 2069 6e20  ptable peaks in 
+00010bc0: 7065 616b 2066 696e 6469 6e67 2070 726f  peak finding pro
+00010bd0: 6365 7373 2c20 602d 2d72 656c 5f68 6569  cess, `--rel_hei
+00010be0: 6768 7460 2077 6869 6368 2073 6574 7320  ght` which sets 
+00010bf0: 7468 6520 7265 6c61 7469 7665 2068 6569  the relative hei
+00010c00: 6768 7420 6174 2077 6869 6368 2074 6865  ght at which the
+00010c10: 2070 6561 6b20 7769 6474 6820 6973 206d   peak width is m
+00010c20: 6561 7375 7265 642c 2060 2d2d 6b73 746f  easured, `--ksto
+00010c30: 6461 7465 6020 7768 6963 6820 6d61 6e75  date` which manu
+00010c40: 616c 6c79 2073 6574 7320 7468 6520 7261  ally sets the ra
+00010c50: 6e67 6520 6f66 202a 4b2a 3c73 7562 3e53  nge of *K*<sub>S
+00010c60: 3c2f 7375 623e 2074 6f20 6265 2064 6174  </sub> to be dat
+00010c70: 6564 2069 6e20 6865 7572 6973 7469 6320  ed in heuristic 
+00010c80: 7365 6172 6368 2061 6e64 206e 6565 6473  search and needs
+00010c90: 2074 6f20 6265 2063 6f2d 7365 7420 7769   to be co-set wi
+00010ca0: 7468 206f 7074 696f 6e20 602d 2d6d 616e  th option `--man
+00010cb0: 7561 6c73 6574 602c 2060 2d2d 786c 696d  ualset`, `--xlim
+00010cc0: 6020 616e 6420 602d 2d79 6c69 6d60 2064  ` and `--ylim` d
+00010cd0: 6574 6572 6d69 6e69 6e67 2074 6865 2078  etermining the x
+00010ce0: 2061 6e64 2079 2061 7869 7320 6c69 6d69   and y axis limi
+00010cf0: 7420 6f66 2047 4d4d 202a 4b2a 3c73 7562  t of GMM *K*<sub
+00010d00: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
+00010d10: 7574 696f 6e2c 2060 2d2d 6369 6020 7365  ution, `--ci` se
+00010d20: 7474 696e 6720 7468 6520 636f 6e66 6964  tting the confid
+00010d30: 656e 6365 206c 6576 656c 206f 6620 6c6f  ence level of lo
+00010d40: 672d 6e6f 726d 616c 2064 6973 7472 6962  g-normal distrib
+00010d50: 7574 696f 6e20 746f 2064 6174 6520 2864  ution to date (d
+00010d60: 6566 6175 6c74 2039 3529 2c20 602d 2d68  efault 95), `--h
+00010d70: 6472 6020 7365 7474 696e 6720 7468 6520  dr` setting the 
+00010d80: 6869 6768 6573 7420 6465 6e73 6974 7920  highest density 
+00010d90: 7265 6769 6f6e 2028 4844 5229 2061 7070  region (HDR) app
+00010da0: 6c69 6564 2069 6e20 7468 6520 7365 676d  lied in the segm
+00010db0: 656e 742d 6775 6964 6564 2061 6e63 686f  ent-guided ancho
+00010dc0: 7220 7061 6972 202a 4b2a 3c73 7562 3e53  r pair *K*<sub>S
+00010dd0: 3c2f 7375 623e 2064 6973 7472 6962 7574  </sub> distribut
+00010de0: 696f 6e20 2864 6566 6175 6c74 2039 3529  ion (default 95)
+00010df0: 2c20 602d 2d68 6575 7269 7374 6963 6020  , `--heuristic` 
+00010e00: 6465 7465 726d 696e 696e 6720 7768 6574  determining whet
+00010e10: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+00010e20: 6865 7572 6973 7469 6320 6d65 7468 6f64  heuristic method
+00010e30: 206f 6620 6465 6669 6e69 6e67 2043 4920   of defining CI 
+00010e40: 666f 7220 6461 7469 6e67 2c20 602d 2d6b  for dating, `--k
+00010e50: 7363 7574 6f66 6660 2073 6574 7469 6e67  scutoff` setting
+00010e60: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+00010e70: 7375 623e 2073 6174 7572 6174 696f 6e20  sub> saturation 
+00010e80: 6375 746f 6666 2069 6e20 6461 7469 6e67  cutoff in dating
+00010e90: 2028 6465 6661 756c 7420 3529 2e0a 0a46   (default 5)...F
+00010ea0: 6f75 7220 7265 7375 6c74 2073 7562 666c  our result subfl
+00010eb0: 6f64 6572 7320 7769 6c6c 2062 6520 7072  oders will be pr
+00010ec0: 6f64 7563 6564 2c20 6e61 6d65 6c79 2060  oduced, namely `
+00010ed0: 416e 6368 6f72 4b73 5f46 696e 6450 6561  AnchorKs_FindPea
+00010ee0: 6b60 2c20 6041 6e63 686f 724b 735f 474d  k`, `AnchorKs_GM
+00010ef0: 4d60 2c20 6053 6567 6d65 6e74 4775 6964  M`, `SegmentGuid
+00010f00: 654b 735f 474d 4d60 2061 6e64 2060 5365  eKs_GMM` and `Se
+00010f10: 676d 656e 744b 735f 4669 6e64 5065 616b  gmentKs_FindPeak
+00010f20: 602e 2054 6865 2060 416e 6368 6f72 4b73  `. The `AnchorKs
+00010f30: 5f46 696e 6450 6561 6b60 2073 7562 666c  _FindPeak` subfl
+00010f40: 6f64 6572 2063 6f6e 7461 696e 7320 7265  oder contains re
+00010f50: 7375 6c74 7320 6f66 2074 6865 2064 6574  sults of the det
+00010f60: 6563 7465 6420 7065 616b 7320 6279 2074  ected peaks by t
+00010f70: 6865 2060 7369 676e 616c 6020 6d6f 6475  he `signal` modu
+00010f80: 6c65 206f 6620 6053 6369 5079 6020 6c69  le of `SciPy` li
+00010f90: 6272 6172 7920 616e 6420 7468 6520 6173  brary and the as
+00010fa0: 7375 6d65 6420 6869 6768 6573 7420 6d61  sumed highest ma
+00010fb0: 7373 2070 6172 7420 2872 6566 6572 7265  ss part (referre
+00010fc0: 6420 746f 2061 7320 4869 6768 4d61 7373  d to as HighMass
+00010fd0: 2068 6572 6561 6674 6572 2920 6f66 2065   hereafter) of e
+00010fe0: 6163 6820 7065 616b 2c20 7768 6963 6820  ach peak, which 
+00010ff0: 6361 6e20 6265 2075 7365 6420 666f 7220  can be used for 
+00011000: 6675 7274 6865 7220 5747 4420 6461 7469  further WGD dati
+00011010: 6e67 2e20 5468 6520 6041 6e63 686f 724b  ng. The `AnchorK
+00011020: 735f 474d 4d60 2073 686f 7773 2074 6865  s_GMM` shows the
+00011030: 2047 4d4d 2072 6573 756c 7473 2075 706f   GMM results upo
+00011040: 6e20 7468 6520 6f72 6967 696e 616c 2061  n the original a
+00011050: 6e63 686f 7220 2a4b 2a3c 7375 623e 533c  nchor *K*<sub>S<
+00011060: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+00011070: 6f6e 2062 7920 7468 6520 606d 6978 7475  on by the `mixtu
+00011080: 7265 6020 6d6f 6475 6c65 206f 6620 6073  re` module of `s
+00011090: 6369 6b69 742d 6c65 6172 6e60 206c 6962  cikit-learn` lib
+000110a0: 7261 7279 2061 6e64 2074 776f 2073 7562  rary and two sub
+000110b0: 666c 6f64 6572 732c 2060 4c6f 6747 4d4d  floders, `LogGMM
+000110c0: 5f43 4960 2063 6f6e 7461 696e 696e 6720  _CI` containing 
+000110d0: 7468 6520 7265 7375 6c74 7320 6f66 2039  the results of 9
+000110e0: 3525 2043 4920 6f66 2065 6163 6820 636f  5% CI of each co
+000110f0: 6d70 6f6e 656e 742c 2060 4869 6768 4d61  mponent, `HighMa
+00011100: 7373 5f43 4960 2063 6f6e 7461 696e 696e  ss_CI` containin
+00011110: 6720 7468 6520 4869 6768 4d61 7373 206f  g the HighMass o
+00011120: 6620 6561 6368 2063 6f6d 706f 6e65 6e74  f each component
+00011130: 2c20 7768 6963 6820 6361 6e20 6265 2075  , which can be u
+00011140: 7365 6420 666f 7220 6675 7274 6865 7220  sed for further 
+00011150: 5747 4420 6461 7469 6e67 2e20 5468 6520  WGD dating. The 
+00011160: 6053 6567 6d65 6e74 4775 6964 654b 735f  `SegmentGuideKs_
+00011170: 474d 4d60 2073 7562 666f 6c64 6572 2070  GMM` subfolder p
+00011180: 7265 7365 6e74 7320 7265 7375 6c74 7320  resents results 
+00011190: 6f66 2073 6567 6d65 6e74 202a 4b2a 3c73  of segment *K*<s
+000111a0: 7562 3e53 3c2f 7375 623e 2047 4d4d 2077  ub>S</sub> GMM w
+000111b0: 6869 6368 2061 7265 206d 6170 7065 6420  hich are mapped 
+000111c0: 6261 636b 2074 6f20 7468 6520 7265 7369  back to the resi
+000111d0: 6469 6e67 2061 6e63 686f 7220 7061 6972  ding anchor pair
+000111e0: 7320 616e 6420 7468 6520 6173 736f 6369  s and the associ
+000111f0: 6174 6564 2039 3525 2048 4452 2061 6e64  ated 95% HDR and
+00011200: 2048 6967 684d 6173 7320 6f66 2065 6163   HighMass of eac
+00011210: 6820 7365 676d 656e 7420 636c 7573 7465  h segment cluste
+00011220: 7220 696e 2073 7562 666c 6f64 6572 7320  r in subfloders 
+00011230: 6f66 2060 4844 525f 4349 6020 616e 6420  of `HDR_CI` and 
+00011240: 6048 6967 684d 6173 735f 4349 602e 2054  `HighMass_CI`. T
+00011250: 6865 2060 5365 676d 656e 744b 735f 4669  he `SegmentKs_Fi
+00011260: 6e64 5065 616b 6020 7375 6266 6f6c 6465  ndPeak` subfolde
+00011270: 7220 6973 2073 696d 696c 6172 2074 6f20  r is similar to 
+00011280: 6041 6e63 686f 724b 735f 4669 6e64 5065  `AnchorKs_FindPe
+00011290: 616b 6020 6275 7420 7769 7468 2073 6567  ak` but with seg
+000112a0: 6d65 6e74 202a 4b2a 3c73 7562 3e53 3c2f  ment *K*<sub>S</
+000112b0: 7375 623e 2069 6e73 7465 6164 2e20 5468  sub> instead. Th
+000112c0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+000112d0: 3e20 696e 2060 4d75 6c74 6970 6c69 636f  > in `Multiplico
+000112e0: 6e60 2063 616e 2061 6c73 6f20 6265 2063  n` can also be c
+000112f0: 616c 6375 6c61 7465 6420 696e 2070 6c61  alculated in pla
+00011300: 6365 206f 6620 6053 6567 6d65 6e74 6020  ce of `Segment` 
+00011310: 7573 696e 6720 7468 6520 6f70 7469 6f6e  using the option
+00011320: 2060 2d2d 6775 6964 6560 2061 7320 7375   `--guide` as su
+00011330: 6368 2074 6865 2072 6573 756c 7420 7469  ch the result ti
+00011340: 746c 652c 206c 6162 656c 2c20 6669 6c65  tle, label, file
+00011350: 2061 6e64 2066 6f6c 6465 7220 6e61 6d65   and folder name
+00011360: 7320 7769 6c6c 2062 6520 6368 616e 6765  s will be change
+00011370: 6420 6163 636f 7264 696e 676c 792e 0a0a  d accordingly...
+00011380: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
+00011390: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
+000113a0: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
+000113b0: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
+000113c0: 600a 7767 6420 7065 616b 206b 7364 6174  `.wgd peak ksdat
+000113d0: 6120 2d61 7020 6170 6461 7461 202d 736d  a -ap apdata -sm
+000113e0: 2073 6d64 6174 6120 2d6c 6520 6c65 6461   smdata -le leda
+000113f0: 7461 202d 6d70 206d 7064 6174 6120 2d2d  ta -mp mpdata --
+00011400: 6865 7572 6973 7469 630a 6060 600a 0a23  heuristic.```..#
+00011410: 2323 2077 6764 2073 796e 0a0a 2a2a 5468  ## wgd syn..**Th
+00011420: 6520 696e 7472 612d 7370 6563 6966 6963  e intra-specific
+00011430: 2073 796e 7465 6e79 2069 6e66 6572 656e   synteny inferen
+00011440: 6365 2a2a 0a60 6060 0a77 6764 2073 796e  ce**.```.wgd syn
+00011450: 2066 616d 696c 6965 7320 6766 6620 282d   families gff (-
+00011460: 2d6b 735f 6469 7374 7269 6275 7469 6f6e  -ks_distribution
+00011470: 206b 7364 6174 6120 2d66 2067 656e 6520   ksdata -f gene 
+00011480: 2d61 2049 4420 2d2d 6d69 6e6c 656e 202d  -a ID --minlen -
+00011490: 3120 2d2d 6d69 6e73 6567 6c65 6e20 3130  1 --minseglen 10
+000114a0: 3030 3020 2d2d 6d69 6e67 656e 656e 756d  000 --mingenenum
+000114b0: 2033 3029 0a60 6060 0a0a 5468 6520 7072   30).```..The pr
+000114c0: 6f67 7261 6d20 6077 6764 2073 796e 6020  ogram `wgd syn` 
+000114d0: 6973 206d 6169 6e6c 7920 6465 616c 696e  is mainly dealin
+000114e0: 6720 7769 7468 2063 6f6c 6c69 6e65 6172  g with collinear
+000114f0: 6974 7920 6f72 2073 796e 7465 6e79 2028  ity or synteny (
+00011500: 626f 7468 2072 6566 6572 7265 6420 746f  both referred to
+00011510: 2061 7320 7379 6e74 656e 7920 6865 7265   as synteny here
+00011520: 6166 7465 7229 2061 6e61 6c79 7369 732e  after) analysis.
+00011530: 2054 776f 2069 6e70 7574 2066 696c 6573   Two input files
+00011540: 2061 7265 2065 7373 656e 7469 616c 2c20   are essential, 
+00011550: 7468 6520 6765 6e65 2066 616d 696c 7920  the gene family 
+00011560: 6669 6c65 2061 6e64 2074 6865 2067 6666  file and the gff
+00011570: 3320 6669 6c65 2e20 5468 6520 6765 6e65  3 file. The gene
+00011580: 2066 616d 696c 7920 6669 6c65 2069 7320   family file is 
+00011590: 696e 2074 6865 2066 6f72 6d61 7420 6173  in the format as
+000115a0: 2060 4f72 7468 6f46 696e 6465 7260 2e20   `OrthoFinder`. 
+000115b0: 5468 6520 736f 6674 7761 7265 2060 692d  The software `i-
+000115c0: 6164 686f 7265 6020 6973 2061 2070 7265  adhore` is a pre
+000115d0: 7265 7175 6973 6974 652e 2057 6974 6820  requisite. With 
+000115e0: 6465 6661 756c 7420 7061 7261 6d65 7465  default paramete
+000115f0: 7273 2c20 7468 6520 7072 6f67 7261 6d20  rs, the program 
+00011600: 6261 7369 6361 6c6c 7920 636f 6e64 7563  basically conduc
+00011610: 7473 2031 2920 6669 6c74 6572 696e 6720  ts 1) filtering 
+00011620: 6765 6e65 2066 616d 696c 6965 7320 6261  gene families ba
+00011630: 7365 6420 6f6e 206d 6178 696d 756d 2066  sed on maximum f
+00011640: 616d 696c 7920 7369 7a65 2032 2920 7265  amily size 2) re
+00011650: 7472 6965 7669 6e67 2067 656e 6520 706f  trieving gene po
+00011660: 7369 7469 6f6e 2061 6e64 2073 6361 6666  sition and scaff
+00011670: 6f6c 6420 696e 666f 726d 6174 696f 6e20  old information 
+00011680: 6672 6f6d 2067 6666 3320 6669 6c65 2033  from gff3 file 3
+00011690: 2920 7072 6f64 7563 696e 6720 7468 6520  ) producing the 
+000116a0: 636f 6e66 6967 7572 6174 696f 6e20 6669  configuration fi
+000116b0: 6c65 2061 6e64 2061 7373 6f63 6961 7465  le and associate
+000116c0: 6420 6461 7461 6669 6c65 7320 666f 7220  d datafiles for 
+000116d0: 6069 2d61 6468 6f72 6560 2034 2920 6361  `i-adhore` 4) ca
+000116e0: 6c6c 696e 6720 6069 2d61 6468 6f72 6560  lling `i-adhore`
+000116f0: 2067 6976 656e 2074 6865 2070 6172 616d   given the param
+00011700: 6574 6572 7320 7365 7420 746f 2069 6e66  eters set to inf
+00011710: 6572 2073 796e 7465 6e79 2035 2920 7669  er synteny 5) vi
+00011720: 7375 616c 697a 696e 6720 7468 6520 7379  sualizing the sy
+00011730: 6e74 656e 7920 696e 2022 646f 7470 6c6f  nteny in "dotplo
+00011740: 7422 2069 6e20 7468 6520 756e 6974 206f  t" in the unit o
+00011750: 6620 6765 6e65 7320 616e 6420 6261 7365  f genes and base
+00011760: 732c 2069 6e20 2253 796e 6465 7074 6822  s, in "Syndepth"
+00011770: 2070 6c6f 7420 7368 6f77 696e 6720 7468   plot showing th
+00011780: 6520 6469 7374 7269 6275 7469 6f6e 206f  e distribution o
+00011790: 6620 6469 6666 6572 656e 7420 6361 7465  f different cate
+000117a0: 676f 7269 6573 206f 6620 636f 6c6c 696e  gories of collin
+000117b0: 6561 7269 7479 2072 6174 696f 7320 7769  earity ratios wi
+000117c0: 7468 696e 2061 6e64 2062 6574 7765 656e  thin and between
+000117d0: 2073 7065 6369 6573 2c20 696e 2022 6475   species, in "du
+000117e0: 7053 7461 636b 2220 706c 6f74 2073 686f  pStack" plot sho
+000117f0: 7769 6e67 206d 756c 7469 706c 6963 6f6e  wing multiplicon
+00011800: 7320 7769 7468 2064 6966 6665 7265 6e74  s with different
+00011810: 206d 756c 7469 706c 6963 6174 696f 6e20   multiplication 
+00011820: 6c65 7665 6c73 2e20 3629 2069 6620 7769  levels. 6) if wi
+00011830: 7468 202a 4b2a 3c73 7562 3e53 3c2f 7375  th *K*<sub>S</su
+00011840: 623e 2064 6174 612c 2061 2022 2a4b 2a3c  b> data, a "*K*<
+00011850: 7375 623e 533c 2f73 7562 3e20 646f 7470  sub>S</sub> dotp
+00011860: 6c6f 7422 2077 6974 6820 646f 7473 2061  lot" with dots a
+00011870: 6e6e 6f74 6174 6564 2069 6e20 2a4b 2a3c  nnotated in *K*<
+00011880: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
+00011890: 6573 2061 6e64 2061 202a 4b2a 3c73 7562  es and a *K*<sub
+000118a0: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
+000118b0: 7574 696f 6e20 7769 7468 2061 6e63 686f  ution with ancho
+000118c0: 7220 7061 6972 7320 6465 6e6f 7465 6420  r pairs denoted 
+000118d0: 7769 6c6c 2062 6520 7072 6f64 7563 6564  will be produced
+000118e0: 2e20 5468 6520 6765 6e65 2069 6e66 6f72  . The gene infor
+000118f0: 6d61 7469 6f6e 2069 6e20 7468 6520 6765  mation in the ge
+00011900: 6e65 2066 616d 696c 7920 6669 6c65 2061  ne family file a
+00011910: 6e64 2067 6666 3320 6669 6c65 2073 686f  nd gff3 file sho
+00011920: 756c 6420 6265 206d 6174 6368 6564 2077  uld be matched w
+00011930: 6869 6368 2072 6571 7569 7265 7320 7573  hich requires us
+00011940: 6572 7320 746f 2073 6574 2070 726f 7065  ers to set prope
+00011950: 7220 602d 2d66 6561 7475 7265 6020 616e  r `--feature` an
+00011960: 6420 602d 2d61 7474 7269 6275 7465 602e  d `--attribute`.
+00011970: 2054 6865 206d 6178 696d 756d 2066 616d   The maximum fam
+00011980: 696c 7920 7369 7a65 2074 6f20 6265 2069  ily size to be i
+00011990: 6e63 6c75 6465 6420 6361 6e20 6265 2073  ncluded can be s
+000119a0: 6574 2076 6961 2074 6865 206f 7074 696f  et via the optio
+000119b0: 6e20 602d 2d6d 6178 7369 7a65 602c 206e  n `--maxsize`, n
+000119c0: 6f74 6564 2074 6861 7420 7468 6973 2066  oted that this f
+000119d0: 696c 7465 7269 6e67 2069 7320 6d61 696e  iltering is main
+000119e0: 6c79 2074 6f20 6472 6f70 2074 686f 7365  ly to drop those
+000119f0: 2068 7567 6520 7461 6e64 656d 2064 7570   huge tandem dup
+00011a00: 6c69 6361 7465 7320 6661 6d69 6c79 2061  licates family a
+00011a10: 6e64 2074 7261 6e73 706f 7361 626c 6520  nd transposable 
+00011a20: 656c 656d 656e 7473 2028 5445 7329 2066  elements (TEs) f
+00011a30: 616d 696c 792c 2061 6e64 206e 6f74 206d  amily, and not m
+00011a40: 616e 6461 746f 7279 2e20 5573 6572 7320  andatory. Users 
+00011a50: 6361 6e20 6669 6c74 6572 2074 686f 7365  can filter those
+00011a60: 2066 7261 676d 656e 7461 7279 2073 6361   fragmentary sca
+00011a70: 6666 6f6c 6473 2076 6961 2074 6865 206f  ffolds via the o
+00011a80: 7074 696f 6e20 602d 2d6d 696e 6c65 6e60  ption `--minlen`
+00011a90: 2e20 5468 6520 6d69 6e69 6d75 6d20 6c65  . The minimum le
+00011aa0: 6e67 7468 2061 6e64 206e 756d 6265 7220  ngth and number 
+00011ab0: 6f66 2067 656e 6573 2066 6f72 2061 2073  of genes for a s
+00011ac0: 6567 6d65 6e74 2074 6f20 6265 2063 6f6e  egment to be con
+00011ad0: 7369 6465 7265 6420 6361 6e20 6265 2073  sidered can be s
+00011ae0: 6574 2076 6961 2074 6865 206f 7074 696f  et via the optio
+00011af0: 6e20 602d 2d6d 696e 7365 676c 656e 6020  n `--minseglen` 
+00011b00: 616e 6420 602d 2d6d 696e 6765 6e65 6e75  and `--mingenenu
+00011b10: 6d60 2e20 5265 6475 6e64 616e 7420 6d75  m`. Redundant mu
+00011b20: 6c74 6970 6c69 636f 6e73 2063 616e 2062  ltiplicons can b
+00011b30: 6520 6b65 7074 2062 7920 7365 7420 7468  e kept by set th
+00011b40: 6520 666c 6167 206f 7074 696f 6e20 602d  e flag option `-
+00011b50: 2d6b 6565 7072 6564 756e 602e 0a0a 2a2a  -keepredun`...**
+00011b60: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
+00011b70: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
+00011b80: 2063 6f6d 6d61 6e64 2073 696d 706c 7920   command simply 
+00011b90: 6173 2062 656c 6f77 2a2a 0a0a 6060 600a  as below**..```.
+00011ba0: 7767 6420 7379 6e20 6661 6d69 6c69 6573  wgd syn families
+00011bb0: 2067 6666 0a60 6060 0a0a 2a2a 5468 6520   gff.```..**The 
+00011bc0: 696e 7465 722d 7370 6563 6966 6963 2073  inter-specific s
+00011bd0: 796e 7465 6e79 2069 6e66 6572 656e 6365  ynteny inference
+00011be0: 2a2a 0a60 6060 0a77 6764 2073 796e 2066  **.```.wgd syn f
+00011bf0: 616d 696c 6965 7320 6766 6631 2067 6666  amilies gff1 gff
+00011c00: 3220 282d 2d61 6464 6974 696f 6e61 6c67  2 (--additionalg
+00011c10: 6666 696e 666f 2022 6d52 4e41 3b4e 616d  ffinfo "mRNA;Nam
+00011c20: 6522 202d 2d61 6464 6974 696f 6e61 6c67  e" --additionalg
+00011c30: 6666 696e 666f 2022 6765 6e65 3b49 4422  ffinfo "gene;ID"
+00011c40: 290a 6060 600a 0a46 6f72 206d 756c 7469  ).```..For multi
+00011c50: 2d73 7065 6369 6573 2073 796e 7465 6e79  -species synteny
+00011c60: 2069 6e66 6572 656e 6365 2c20 6966 2075   inference, if u
+00011c70: 7365 7273 2068 6176 6520 6766 6633 2066  sers have gff3 f
+00011c80: 696c 6573 2077 6869 6368 2068 6176 6520  iles which have 
+00011c90: 6469 6666 6572 656e 7420 6665 6174 7572  different featur
+00011ca0: 6573 206f 7220 6174 7472 6962 7574 6573  es or attributes
+00011cb0: 2066 6f72 2067 656e 6520 706f 7369 7469   for gene positi
+00011cc0: 6f6e 2069 6e66 6f72 6d61 7469 6f6e 2072  on information r
+00011cd0: 6574 7269 6576 616c 2c20 7468 6520 6f70  etrieval, the op
+00011ce0: 7469 6f6e 2060 2d2d 6164 6469 7469 6f6e  tion `--addition
+00011cf0: 616c 6766 6669 6e66 6f60 2063 616e 2062  algffinfo` can b
+00011d00: 6520 7365 7420 746f 2070 726f 7669 6465  e set to provide
+00011d10: 2074 6865 2061 6464 6974 696f 6e61 6c20   the additional 
+00011d20: 696e 666f 726d 6174 696f 6e2e 2054 6865  information. The
+00011d30: 2072 656d 6169 6e69 6e67 2070 6172 616d   remaining param
+00011d40: 6574 6572 2073 6574 7469 6e67 2069 7320  eter setting is 
+00011d50: 7468 6520 7361 6d65 2061 7320 7468 6520  the same as the 
+00011d60: 696e 7472 612d 7370 6563 6966 6963 2073  intra-specific s
+00011d70: 796e 7465 6e79 2069 6e66 6572 656e 6365  ynteny inference
+00011d80: 2e0a 0a2a 2a41 2073 7567 6765 7374 6564  ...**A suggested
+00011d90: 2073 7461 7274 696e 6720 7275 6e20 6361   starting run ca
+00011da0: 6e20 7573 6520 636f 6d6d 616e 6420 7369  n use command si
+00011db0: 6d70 6c79 2061 7320 6265 6c6f 772a 2a0a  mply as below**.
+00011dc0: 0a60 6060 0a77 6764 2073 796e 2066 616d  .```.wgd syn fam
+00011dd0: 696c 6965 7320 6766 6631 2067 6666 320a  ilies gff1 gff2.
+00011de0: 6060 600a 0a23 2323 2077 6764 2076 697a  ```..### wgd viz
+00011df0: 0a0a 2a2a 5468 6520 7669 7375 616c 697a  ..**The visualiz
+00011e00: 6174 696f 6e20 6f66 202a 4b2a 3c73 7562  ation of *K*<sub
+00011e10: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
+00011e20: 7472 6962 7574 696f 6e20 616e 6420 454c  tribution and EL
+00011e30: 4d4d 2061 6e61 6c79 7369 732a 2a0a 6060  MM analysis**.``
+00011e40: 600a 7767 6420 7669 7a20 2d64 206b 7364  `.wgd viz -d ksd
+00011e50: 6174 610a 6060 600a 0a54 6865 2070 726f  ata.```..The pro
+00011e60: 6772 616d 2060 7767 6420 7669 7a60 2069  gram `wgd viz` i
+00011e70: 7320 6d61 696e 6c79 2066 6f72 2074 6865  s mainly for the
+00011e80: 2070 7572 706f 7365 206f 6620 2a4b 2a3c   purpose of *K*<
+00011e90: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00011ea0: 7269 6275 7469 6f6e 2061 6e64 2073 796e  ribution and syn
+00011eb0: 7465 6e79 2076 6973 7561 6c69 7a61 7469  teny visualizati
+00011ec0: 6f6e 2c20 7769 7468 2073 6f6d 6520 6f70  on, with some op
+00011ed0: 7469 6f6e 616c 206d 6978 7475 7265 206d  tional mixture m
+00011ee0: 6f64 656c 696e 6720 616e 616c 7973 6973  odeling analysis
+00011ef0: 2e20 5468 6520 6261 7369 6320 6675 6e63  . The basic func
+00011f00: 7469 6f6e 2069 7320 6a75 7374 2074 6f20  tion is just to 
+00011f10: 706c 6f74 2074 6865 202a 4b2a 3c73 7562  plot the *K*<sub
+00011f20: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
+00011f30: 7574 696f 6e20 616e 6420 636f 6e64 7563  ution and conduc
+00011f40: 7420 616e 2045 4c4d 4d20 616e 616c 7973  t an ELMM analys
+00011f50: 6973 2069 6e20 7365 6172 6368 206f 6620  is in search of 
+00011f60: 706f 7465 6e74 6961 6c20 5747 4420 636f  potential WGD co
+00011f70: 6d70 6f6e 656e 7473 2e20 536f 6d65 206b  mponents. Some k
+00011f80: 6579 2070 6172 616d 6574 6572 7320 6166  ey parameters af
+00011f90: 6665 6374 696e 6720 7468 6520 454c 4d4d  fecting the ELMM
+00011fa0: 2072 6573 756c 7420 696e 636c 7564 6520   result include 
+00011fb0: 602d 2d70 726f 6d69 6e65 6e63 655f 6375  `--prominence_cu
+00011fc0: 746f 6666 6020 616e 6420 602d 2d72 656c  toff` and `--rel
+00011fd0: 5f68 6569 6768 7460 2c20 7768 6963 6820  _height`, which 
+00011fe0: 6861 7665 2062 6565 6e20 6578 706c 6169  have been explai
+00011ff0: 6e65 6420 6168 6561 642c 2060 2d2d 656d  ned ahead, `--em
+00012000: 5f69 7465 7261 7469 6f6e 7360 2061 6e64  _iterations` and
+00012010: 2060 2d2d 656d 5f69 6e69 7469 616c 697a   `--em_initializ
+00012020: 6174 696f 6e73 6020 6465 7465 726d 696e  ations` determin
+00012030: 696e 6720 7468 6520 6d61 7869 6d75 6d20  ing the maximum 
+00012040: 6974 6572 6174 696f 6e73 2061 6e64 2069  iterations and i
+00012050: 6e69 7469 616c 697a 6174 696f 6e73 2069  nitializations i
+00012060: 6e20 7468 6520 454d 2061 6c67 6f72 6974  n the EM algorit
+00012070: 686d 2e0a 0a2a 2a41 2073 7567 6765 7374  hm...**A suggest
+00012080: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
+00012090: 6361 6e20 7573 6520 636f 6d6d 616e 6420  can use command 
+000120a0: 7369 6d70 6c79 2061 7320 6265 6c6f 772a  simply as below*
+000120b0: 2a0a 0a60 6060 0a77 6764 2076 697a 202d  *..```.wgd viz -
+000120c0: 6420 6b73 6461 7461 0a60 6060 0a0a 2a2a  d ksdata.```..**
+000120d0: 5468 6520 7669 7375 616c 697a 6174 696f  The visualizatio
+000120e0: 6e20 6f66 202a 4b2a 3c73 7562 3e53 3c2f  n of *K*<sub>S</
+000120f0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+00012100: 7574 696f 6e20 7769 7468 2072 6174 6520  ution with rate 
+00012110: 636f 7272 6563 7469 6f6e 2a2a 0a60 6060  correction**.```
+00012120: 0a77 6764 2076 697a 202d 6420 6b73 6461  .wgd viz -d ksda
+00012130: 7461 202d 7370 2073 7064 6174 6120 2d2d  ta -sp spdata --
+00012140: 666f 6375 7332 616c 6c20 666f 6361 6c5f  focus2all focal_
+00012150: 7370 6563 6965 7320 2d2d 6578 7472 6170  species --extrap
+00012160: 6172 616e 6f6d 656b 7320 6b73 6461 7461  aranomeks ksdata
+00012170: 2028 2d2d 616e 6368 6f72 706f 696e 7473   (--anchorpoints
+00012180: 2061 7064 6174 6120 2d2d 706c 6f74 6170   apdata --plotap
+00012190: 676d 6d20 2d2d 706c 6f74 656c 6d6d 290a  gmm --plotelmm).
+000121a0: 6060 600a 0a42 6573 6964 6573 2074 6865  ```..Besides the
+000121b0: 2062 6173 6963 202a 4b2a 3c73 7562 3e53   basic *K*<sub>S
+000121c0: 3c2f 7375 623e 2070 6c6f 742c 2073 7562  </sub> plot, sub
+000121d0: 7374 6974 7574 696f 6e20 7261 7465 2063  stitution rate c
+000121e0: 6f72 7265 6374 696f 6e20 6361 6e20 616c  orrection can al
+000121f0: 736f 2062 6520 6163 6869 6576 6564 2067  so be achieved g
+00012200: 6976 656e 2061 7420 6c65 6173 7420 6120  iven at least a 
+00012210: 7370 6563 6965 7320 7472 6565 2028 7669  species tree (vi
+00012220: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
+00012230: 7370 6563 6965 7374 7265 6560 2920 616e  speciestree`) an
+00012240: 6420 6120 666f 6361 6c20 7370 6563 6965  d a focal specie
+00012250: 7320 2865 6974 6865 7220 7669 6120 7468  s (either via th
+00012260: 6520 6f70 7469 6f6e 2060 2d2d 666f 6375  e option `--focu
+00012270: 7332 616c 6c60 206f 7220 7669 6120 7468  s2all` or via th
+00012280: 6520 6f70 7469 6f6e 2060 2d2d 7370 6169  e option `--spai
+00012290: 7260 2069 6e20 7468 6520 666f 726d 206f  r` in the form o
+000122a0: 6620 2224 666f 6361 6c5f 7370 6563 6965  f "$focal_specie
+000122b0: 733b 2466 6f63 616c 5f73 7065 6369 6573  s;$focal_species
+000122c0: 2229 2e20 4974 2773 2073 7567 6765 7374  "). It's suggest
+000122d0: 6564 2074 6861 7420 7468 6520 6f72 7468  ed that the orth
+000122e0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+000122f0: 533c 2f73 7562 3e20 6461 7461 2069 7320  S</sub> data is 
+00012300: 7072 6f76 6964 6564 2062 7920 7468 6520  provided by the 
+00012310: 602d 2d64 6174 6166 696c 6560 206f 7074  `--datafile` opt
+00012320: 696f 6e20 616e 6420 7468 6520 7061 7261  ion and the para
+00012330: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
+00012340: 3c2f 7375 623e 2064 6174 6120 6973 2070  </sub> data is p
+00012350: 726f 7669 6465 6420 6279 2074 6865 2060  rovided by the `
+00012360: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
+00012370: 7360 206f 7074 696f 6e2c 2061 6c74 686f  s` option, altho
+00012380: 7567 6820 6974 2773 2061 6c6c 6f77 6564  ugh it's allowed
+00012390: 2074 6f20 6f6e 6c79 2070 726f 7669 6465   to only provide
+000123a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000123b0: 2064 6174 6120 7669 6120 7468 6520 602d   data via the `-
+000123c0: 2d64 6174 6166 696c 6560 206f 7074 696f  -datafile` optio
+000123d0: 6e20 616e 6420 6465 706f 7369 7420 626f  n and deposit bo
+000123e0: 7468 206f 7274 686f 6c6f 676f 7573 2061  th orthologous a
+000123f0: 6e64 2070 6172 616c 6f67 6f75 7320 2a4b  nd paralogous *K
+00012400: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
+00012410: 7461 2074 6865 7265 6f6e 2e20 5468 6572  ta thereon. Ther
+00012420: 6520 6172 6520 7477 6f20 7479 7065 7320  e are two types 
+00012430: 6f66 206d 6978 6564 2070 6c6f 7473 2028  of mixed plots (
+00012440: 7468 6520 226d 6978 6564 2220 6865 7265  the "mixed" here
+00012450: 2072 6566 6572 7320 746f 206d 6978 6564   refers to mixed
+00012460: 206f 7274 686f 6c6f 676f 7573 2061 6e64   orthologous and
+00012470: 2070 6172 616c 6f67 6f75 7320 2a4b 2a3c   paralogous *K*<
+00012480: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00012490: 7269 6275 7469 6f6e 7329 2c20 6f6e 6520  ributions), one 
+000124a0: 6f66 2077 6869 6368 2069 7320 7369 6d69  of which is simi
+000124b0: 6c61 7220 746f 2077 6861 7420 606b 7372  lar to what `ksr
+000124c0: 6174 6573 6020 706c 6f74 732c 2077 6869  ates` plots, whi
+000124d0: 6c65 2074 6865 206f 7468 6572 206f 6620  le the other of 
+000124e0: 7768 6963 6820 6973 206c 696b 6520 7468  which is like th
+000124f0: 6520 636f 6e76 656e 7469 6f6e 616c 202a  e conventional *
+00012500: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2070  K*<sub>S</sub> p
+00012510: 6c6f 7420 7468 6174 2062 6f74 6820 7468  lot that both th
+00012520: 6520 6f72 6967 696e 616c 206f 7274 686f  e original ortho
+00012530: 6c6f 676f 7573 2061 6e64 2070 6172 616c  logous and paral
+00012540: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
+00012550: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+00012560: 6f6e 7320 6172 6520 7472 7574 6866 756c  ons are truthful
+00012570: 6c79 2070 6c6f 7474 6564 2069 6e73 7465  ly plotted inste
+00012580: 6164 206f 6620 6a75 7374 2062 6569 6e67  ad of just being
+00012590: 2072 6570 7265 7365 6e74 6564 2062 7920   represented by 
+000125a0: 736f 6d65 2076 6572 7469 6361 6c20 6c69  some vertical li
+000125b0: 6e65 732e 2057 6520 7375 6767 6573 7420  nes. We suggest 
+000125c0: 7573 6572 7320 746f 2061 646f 7074 2074  users to adopt t
+000125d0: 6865 2060 6b73 7261 7465 7360 2d6c 696b  he `ksrates`-lik
+000125e0: 6520 706c 6f74 732c 2077 6869 6368 2069  e plots, which i
+000125f0: 7320 7468 6520 6465 6661 756c 742e 204f  s the default. O
+00012600: 7468 6572 7769 7365 2074 6865 206f 7074  therwise the opt
+00012610: 696f 6e20 602d 2d63 6c61 7373 6963 6020  ion `--classic` 
+00012620: 7769 6c6c 2062 6520 6e65 6564 6564 2074  will be needed t
+00012630: 6f20 7365 742e 2045 7874 7261 206d 6978  o set. Extra mix
+00012640: 7475 7265 206d 6f64 656c 696e 6720 616e  ture modeling an
+00012650: 616c 7973 6973 2063 616e 2062 6520 696e  alysis can be in
+00012660: 6974 6961 7465 6420 7669 6120 7468 6520  itiated via the 
+00012670: 6f70 7469 6f6e 2060 2d2d 706c 6f74 656c  option `--plotel
+00012680: 6d6d 6020 616e 6420 602d 2d70 6c6f 7461  mm` and `--plota
+00012690: 7067 6d6d 6020 7769 7468 2074 6865 2061  pgmm` with the a
+000126a0: 6e63 686f 7220 706f 696e 7473 2064 6174  nchor points dat
+000126b0: 6166 696c 6520 7072 6f76 6964 6564 2062  afile provided b
+000126c0: 7920 7468 6520 602d 2d61 6e63 686f 7270  y the `--anchorp
+000126d0: 6f69 6e74 7360 206f 7074 696f 6e2e 0a0a  oints` option...
+000126e0: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
+000126f0: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
+00012700: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
+00012710: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
+00012720: 600a 7767 6420 7669 7a20 2d64 206b 7364  `.wgd viz -d ksd
+00012730: 6174 6120 2d73 7020 7370 6461 7461 202d  ata -sp spdata -
+00012740: 2d66 6f63 7573 3261 6c6c 2066 6f63 616c  -focus2all focal
+00012750: 5f73 7065 6369 6573 202d 2d65 7874 7261  _species --extra
+00012760: 7061 7261 6e6f 6d65 6b73 206b 7364 6174  paranomeks ksdat
+00012770: 610a 6060 600a 0a2a 2a54 6865 2076 6973  a.```..**The vis
+00012780: 7561 6c69 7a61 7469 6f6e 206f 6620 7379  ualization of sy
+00012790: 6e74 656e 792a 2a0a 6060 600a 7767 6420  nteny**.```.wgd 
+000127a0: 7669 7a20 2d61 7020 6170 6461 7461 202d  viz -ap apdata -
+000127b0: 736d 2073 6d64 6174 6120 2d6d 7420 6d74  sm smdata -mt mt
+000127c0: 6461 7461 202d 6774 2067 7464 6174 6120  data -gt gtdata 
+000127d0: 2d2d 706c 6f74 7379 6e20 282d 2d6d 696e  --plotsyn (--min
+000127e0: 6c65 6e20 2d31 202d 2d6d 696e 7365 676c  len -1 --minsegl
+000127f0: 656e 2031 3030 3030 202d 2d6d 696e 6765  en 10000 --minge
+00012800: 6e65 6e75 6d20 3330 290a 6060 600a 0a43  nenum 30).```..C
+00012810: 6f6d 7061 7265 6420 746f 2074 6865 206f  ompared to the o
+00012820: 7269 6769 6e61 6c20 6077 6764 602c 2074  riginal `wgd`, t
+00012830: 6865 2060 7767 6420 7669 7a60 2070 726f  he `wgd viz` pro
+00012840: 6772 616d 2061 6464 6564 2073 796e 7465  gram added synte
+00012850: 6e79 2076 6973 7561 6c69 7a61 7469 6f6e  ny visualization
+00012860: 2070 6970 656c 696e 652e 2055 7365 7273   pipeline. Users
+00012870: 206e 6565 6420 746f 2070 726f 7669 6465   need to provide
+00012880: 2074 6865 2066 6c61 6720 6f70 7469 6f6e   the flag option
+00012890: 2060 2d2d 706c 6f74 7379 6e60 2074 6f20   `--plotsyn` to 
+000128a0: 696e 6974 6961 7465 2074 6869 7320 7061  initiate this pa
+000128b0: 7274 206f 6620 7069 7065 6c69 6e65 2e20  rt of pipeline. 
+000128c0: 5468 6973 2073 7465 7020 6173 7375 6d65  This step assume
+000128d0: 7320 7468 6174 2075 7365 7273 2068 6176  s that users hav
+000128e0: 6520 6f62 7461 696e 6564 2061 6c72 6561  e obtained alrea
+000128f0: 6479 2074 6865 2073 796e 7465 6e69 6320  dy the syntenic 
+00012900: 7265 7375 6c74 7320 6672 6f6d 2060 692d  results from `i-
+00012910: 6164 686f 7265 6020 616e 6420 7573 6573  adhore` and uses
+00012920: 2074 686f 7365 2072 6573 756c 7420 6669   those result fi
+00012930: 6c65 7320 746f 2072 6561 6c69 7a65 2074  les to realize t
+00012940: 6865 2073 796e 7465 6e79 2076 6973 7561  he synteny visua
+00012950: 6c69 7a61 7469 6f6e 2e20 5369 6d6c 6961  lization. Simlia
+00012960: 7220 746f 2060 7767 6420 7379 6e60 2c20  r to `wgd syn`, 
+00012970: 7468 6520 6578 7472 6120 2a4b 2a3c 7375  the extra *K*<su
+00012980: 623e 533c 2f73 7562 3e20 6461 7461 2066  b>S</sub> data f
+00012990: 696c 6520 6361 6e20 6265 2074 7261 6e73  ile can be trans
+000129a0: 6d69 7474 6564 2076 6961 2074 6865 206f  mitted via the o
+000129b0: 7074 696f 6e20 602d 2d64 6174 6166 696c  ption `--datafil
+000129c0: 6560 2028 696e 7374 6561 6420 6f66 2060  e` (instead of `
+000129d0: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
+000129e0: 7360 206f 7074 696f 6e29 2e20 4261 7369  s` option). Basi
+000129f0: 6361 6c6c 792c 2074 6865 2073 796e 7465  cally, the synte
+00012a00: 6e69 6320 7265 7375 6c74 2066 696c 6573  nic result files
+00012a10: 2072 6571 7569 7265 6420 6172 6520 7468   required are th
+00012a20: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
+00012a30: 6461 7461 6669 6c65 2c20 6d75 6c74 6970  datafile, multip
+00012a40: 6c69 636f 6e73 2064 6174 6166 696c 652c  licons datafile,
+00012a50: 2067 656e 6520 7461 626c 6520 6461 7461   gene table data
+00012a60: 6669 6c65 2028 6175 746f 6d61 7465 6c79  file (automately
+00012a70: 2070 726f 6475 6365 6420 6279 2060 7767   produced by `wg
+00012a80: 6420 7379 6e60 292c 2061 6e64 2073 6567  d syn`), and seg
+00012a90: 6d65 6e74 7320 6461 7461 6669 6c65 2e0a  ments datafile..
+00012aa0: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
+00012ab0: 7461 7274 696e 6720 7275 6e20 6361 6e20  tarting run can 
+00012ac0: 7573 6520 636f 6d6d 616e 6420 7369 6d70  use command simp
+00012ad0: 6c79 2061 7320 6265 6c6f 772a 2a0a 0a60  ly as below**..`
+00012ae0: 6060 0a77 6764 2076 697a 202d 6170 2061  ``.wgd viz -ap a
+00012af0: 7064 6174 6120 2d73 6d20 736d 6461 7461  pdata -sm smdata
+00012b00: 202d 6d74 206d 7464 6174 6120 2d67 7420   -mt mtdata -gt 
+00012b10: 6774 6461 7461 202d 2d70 6c6f 7473 796e  gtdata --plotsyn
+00012b20: 0a60 6060 0a0a 2323 2049 6c6c 7573 7472  .```..## Illustr
+00012b30: 6174 696f 6e0a 0a57 6520 696c 6c75 7374  ation..We illust
+00012b40: 7261 7465 206f 7572 2070 726f 6772 616d  rate our program
+00012b50: 206f 6e20 616e 2065 7865 6d70 6c61 7279   on an exemplary
+00012b60: 2057 4744 2069 6e66 6572 656e 6365 2061   WGD inference a
+00012b70: 6e64 2064 6174 696e 6720 7570 6f6e 2073  nd dating upon s
+00012b80: 7065 6369 6573 202a 4171 7569 6c65 6769  pecies *Aquilegi
+00012b90: 6120 636f 6572 756c 6561 2a2e 0a0a 5468  a coerulea*...Th
+00012ba0: 6520 2a41 7175 696c 6567 6961 2063 6f65  e *Aquilegia coe
+00012bb0: 7275 6c65 612a 2077 6173 2072 6570 6f72  rulea* was repor
+00012bc0: 7465 6420 746f 2065 7870 6572 6965 6e63  ted to experienc
+00012bd0: 6520 616e 2070 616c 656f 2d70 6f6c 7970  e an paleo-polyp
+00012be0: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
+00012bf0: 7420 6166 7465 7220 7468 6520 6469 7665  t after the dive
+00012c00: 7267 656e 6365 206f 6620 636f 7265 2065  rgence of core e
+00012c10: 7564 6963 6f74 732c 2077 6869 6368 2069  udicots, which i
+00012c20: 7320 6c69 6b65 6c79 2073 6861 7265 6420  s likely shared 
+00012c30: 6279 2061 6c6c 2052 616e 756e 6375 6c61  by all Ranuncula
+00012c40: 6c65 732e 0a0a 4669 7273 7420 6162 6f76  les...First abov
+00012c50: 6520 616c 6c2c 206c 6574 2773 2064 656c  e all, let's del
+00012c60: 696e 6561 7465 2074 6865 2077 686f 6c65  ineate the whole
+00012c70: 2070 6172 616e 6f6d 6520 2a4b 2a3c 7375   paranome *K*<su
+00012c80: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
+00012c90: 7374 7269 6275 7469 6f6e 2061 6e64 2068  stribution and h
+00012ca0: 6176 6520 6120 6261 7369 6320 6f62 7365  ave a basic obse
+00012cb0: 7276 6174 696f 6e20 666f 7220 706f 7465  rvation for pote
+00012cc0: 6e74 6961 6c6c 7920 636f 6e63 6569 7661  ntially conceiva
+00012cd0: 626c 6520 5747 4473 2c20 7573 696e 6720  ble WGDs, using 
+00012ce0: 7468 6520 636f 6d6d 616e 6420 6c69 6e65  the command line
+00012cf0: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
+00012d00: 2064 6d64 2041 7175 696c 6567 6961 5f63   dmd Aquilegia_c
+00012d10: 6f65 7275 6c65 610a 7767 6420 6b73 6420  oerulea.wgd ksd 
+00012d20: 7767 645f 646d 642f 4171 7569 6c65 6769  wgd_dmd/Aquilegi
+00012d30: 615f 636f 6572 756c 6561 2e74 7376 2041  a_coerulea.tsv A
+00012d40: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00012d50: 610a 6060 600a 0a54 6865 2063 6f6e 7374  a.```..The const
+00012d60: 7275 6374 6564 2077 686f 6c65 2070 6172  ructed whole par
+00012d70: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
+00012d80: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+00012d90: 6275 7469 6f6e 206f 6620 2a41 7175 696c  bution of *Aquil
+00012da0: 6567 6961 2063 6f65 7275 6c65 612a 2069  egia coerulea* i
+00012db0: 7320 6173 2062 656c 6f77 2c20 7765 2063  s as below, we c
+00012dc0: 616e 2073 6565 2074 6861 7420 7468 6572  an see that ther
+00012dd0: 6520 7365 656d 7320 746f 2062 6520 6120  e seems to be a 
+00012de0: 6875 6d70 2061 7420 2a4b 2a3c 7375 623e  hump at *K*<sub>
+00012df0: 533c 2f73 7562 3e20 3120 6275 7420 6e6f  S</sub> 1 but no
+00012e00: 7420 636c 6561 722e 0a0a 215b 5d28 6461  t clear...![](da
+00012e10: 7461 2f41 7175 696c 6567 6961 5f63 6f65  ta/Aquilegia_coe
+00012e20: 7275 6c65 612e 7473 762e 6b73 645f 7770  rulea.tsv.ksd_wp
+00012e30: 2e73 7667 290a 0a57 6520 7468 656e 2063  .svg)..We then c
+00012e40: 6f6e 7374 7275 6374 2074 6865 2061 6e63  onstruct the anc
+00012e50: 686f 7220 2a4b 2a3c 7375 623e 533c 2f73  hor *K*<sub>S</s
+00012e60: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+00012e70: 7469 6f6e 2075 7369 6e67 2074 6865 2063  tion using the c
+00012e80: 6f6d 6d61 6e64 206c 696e 6520 6265 6c6f  ommand line belo
+00012e90: 772e 0a0a 6060 600a 7767 6420 7379 6e20  w...```.wgd syn 
+00012ea0: 2d66 206d 524e 4120 2d61 204e 616d 6520  -f mRNA -a Name 
+00012eb0: 7767 645f 646d 642f 4171 7569 6c65 6769  wgd_dmd/Aquilegi
+00012ec0: 615f 636f 6572 756c 6561 2e74 7376 2041  a_coerulea.tsv A
+00012ed0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00012ee0: 612e 6766 6633 202d 6b73 2077 6764 5f6b  a.gff3 -ks wgd_k
+00012ef0: 7364 2f41 7175 696c 6567 6961 5f63 6f65  sd/Aquilegia_coe
+00012f00: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00012f10: 0a60 6060 0a0a 4173 2073 686f 776e 2062  .```..As shown b
+00012f20: 656c 6f77 2c20 7468 6572 6520 6172 6520  elow, there are 
+00012f30: 736f 6d65 2072 6574 6169 6e65 6420 616e  some retained an
+00012f40: 6368 6f72 2070 6169 7273 2077 6974 6820  chor pairs with 
+00012f50: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00012f60: 6265 7477 6565 6e20 3120 616e 6420 322c  between 1 and 2,
+00012f70: 2077 6869 6368 2073 6565 6d73 2074 6f20   which seems to 
+00012f80: 7375 6767 6573 7420 6120 5747 4420 6576  suggest a WGD ev
+00012f90: 656e 742e 0a0a 215b 5d28 6461 7461 2f41  ent...![](data/A
+00012fa0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00012fb0: 612e 7473 762e 6b73 645f 7770 5f61 702e  a.tsv.ksd_wp_ap.
+00012fc0: 7376 6729 0a0a 5468 6520 6173 736f 6369  svg)..The associ
+00012fd0: 6174 6564 2060 6475 7053 7461 636b 6020  ated `dupStack` 
+00012fe0: 706c 6f74 2073 686f 7773 2074 6861 7420  plot shows that 
+00012ff0: 7468 6572 6520 6172 6520 6e75 6d65 726f  there are numero
+00013000: 7573 2064 7570 6c69 6361 7465 6420 7365  us duplicated se
+00013010: 676d 656e 7473 2061 6372 6f73 7320 6d6f  gments across mo
+00013020: 7374 206f 6620 7468 6520 6368 726f 6d6f  st of the chromo
+00013030: 736f 6d65 732e 0a0a 215b 5d28 6461 7461  somes...![](data
+00013040: 2f73 796e 5f72 6573 756c 7473 2f41 7175  /syn_results/Aqu
+00013050: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
+00013060: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00013070: 6561 5f6d 756c 7469 706c 6963 6f6e 735f  ea_multiplicons_
+00013080: 6c65 7665 6c2e 706e 6729 0a0a 5765 2069  level.png)..We i
+00013090: 6d70 6c65 6d65 6e74 6564 2074 776f 2074  mplemented two t
+000130a0: 7970 6573 206f 6620 646f 7420 706c 6f74  ypes of dot plot
+000130b0: 7320 696e 206f 7866 6f72 6420 6772 6964  s in oxford grid
+000130c0: 3a20 6f6e 6520 696e 2074 6865 2075 6e69  : one in the uni
+000130d0: 7420 6f66 2062 6173 6573 2061 6e64 2074  t of bases and t
+000130e0: 6865 206f 7468 6572 2069 6e20 7468 6520  he other in the 
+000130f0: 756e 6974 206f 6620 6765 6e65 732c 2077  unit of genes, w
+00013100: 6869 6368 2063 616e 2062 6520 636f 6c6f  hich can be colo
+00013110: 7265 6420 6279 202a 4b2a 3c73 7562 3e53  red by *K*<sub>S
+00013120: 3c2f 7375 623e 2076 616c 7565 7320 6769  </sub> values gi
+00013130: 7665 6e20 2a4b 2a3c 7375 623e 533c 2f73  ven *K*<sub>S</s
+00013140: 7562 3e20 6461 7461 2e0a 0a21 5b5d 2864  ub> data...![](d
+00013150: 6174 612f 7379 6e5f 7265 7375 6c74 732f  ata/syn_results/
+00013160: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00013170: 6561 2d76 732d 4171 7569 6c65 6769 615f  ea-vs-Aquilegia_
+00013180: 636f 6572 756c 6561 5f4b 732e 646f 745f  coerulea_Ks.dot_
+00013190: 756e 6974 5f67 656e 652e 706e 6729 0a0a  unit_gene.png)..
+000131a0: 4173 2073 686f 776e 2061 626f 7665 2c20  As shown above, 
+000131b0: 7468 6520 646f 7420 706c 6f74 2069 6e20  the dot plot in 
+000131c0: 7468 6520 756e 6974 206f 6620 6765 6e65  the unit of gene
+000131d0: 7320 7072 6573 656e 7473 206e 756d 6572  s presents numer
+000131e0: 6f75 7320 6465 6e73 656c 7920 6167 6772  ous densely aggr
+000131f0: 6567 6174 6564 2028 6c69 6e65 2d6c 696b  egated (line-lik
+00013200: 6529 2061 6e63 686f 7220 706f 696e 7473  e) anchor points
+00013210: 2061 7420 6d6f 7374 206f 6620 7468 6520   at most of the 
+00013220: 6368 726f 6d6f 736f 6d65 7320 7769 7468  chromosomes with
+00013230: 2063 6f6e 7369 7374 656e 7420 2a4b 2a3c   consistent *K*<
+00013240: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+00013250: 6265 7477 6565 6e20 3120 616e 6420 322e  between 1 and 2.
+00013260: 2054 6865 2064 6f74 2070 6c6f 7420 696e   The dot plot in
+00013270: 2074 6865 2075 6e69 7420 6f66 2062 6173   the unit of bas
+00013280: 6573 2073 686f 7773 2074 6865 2073 616d  es shows the sam
+00013290: 6520 7061 7474 6572 6e2c 2061 7320 6d61  e pattern, as ma
+000132a0: 6e69 6665 7374 6564 2062 656c 6f77 2e0a  nifested below..
+000132b0: 0a21 5b5d 2864 6174 612f 7379 6e5f 7265  .![](data/syn_re
+000132c0: 7375 6c74 732f 4171 7569 6c65 6769 615f  sults/Aquilegia_
+000132d0: 636f 6572 756c 6561 2d76 732d 4171 7569  coerulea-vs-Aqui
+000132e0: 6c65 6769 615f 636f 6572 756c 6561 5f4b  legia_coerulea_K
+000132f0: 732e 646f 742e 706e 6729 0a0a 5468 6520  s.dot.png)..The 
+00013300: 646f 7420 706c 6f74 7320 7769 7468 6f75  dot plots withou
+00013310: 7420 2a4b 2a3c 7375 623e 533c 2f73 7562  t *K*<sub>S</sub
+00013320: 3e20 616e 6e6f 7461 7469 6f6e 2077 696c  > annotation wil
+00013330: 6c20 616c 736f 2062 6520 6175 746f 6d61  l also be automa
+00013340: 7465 6c79 2070 726f 6475 6365 642c 2061  tely produced, a
+00013350: 7320 7368 6f77 6e20 6265 6c6f 772e 0a0a  s shown below...
+00013360: 215b 5d28 6461 7461 2f73 796e 5f72 6573  ![](data/syn_res
+00013370: 756c 7473 2f41 7175 696c 6567 6961 5f63  ults/Aquilegia_c
+00013380: 6f65 7275 6c65 612d 7673 2d41 7175 696c  oerulea-vs-Aquil
+00013390: 6567 6961 5f63 6f65 7275 6c65 612e 646f  egia_coerulea.do
+000133a0: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
+000133b0: 0a0a 215b 5d28 6461 7461 2f73 796e 5f72  ..![](data/syn_r
+000133c0: 6573 756c 7473 2f41 7175 696c 6567 6961  esults/Aquilegia
+000133d0: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
+000133e0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+000133f0: 646f 742e 706e 6729 0a0a 4e6f 7465 2074  dot.png)..Note t
+00013400: 6861 7420 7468 6520 6f70 6163 6974 7920  hat the opacity 
+00013410: 6f66 2061 6e63 686f 7220 646f 7473 2061  of anchor dots a
+00013420: 6e64 2061 6c6c 2068 6f6d 6f6c 6f67 2064  nd all homolog d
+00013430: 6f74 7320 6361 6e20 6265 2073 6574 2062  ots can be set b
+00013440: 7920 7468 6520 6f70 7469 6f6e 2060 2d2d  y the option `--
+00013450: 6170 616c 7068 6160 2061 6e64 2060 2d2d  apalpha` and `--
+00013460: 686f 616c 7068 6160 2073 6570 6172 6174  hoalpha` separat
+00013470: 656c 792e 2049 6620 6f6e 6520 6a75 7374  ely. If one just
+00013480: 2077 616e 7473 2074 6f20 7365 6520 7468   wants to see th
+00013490: 6520 616e 6368 6f72 2064 6f74 732c 2073  e anchor dots, s
+000134a0: 6574 7469 6e67 2074 6865 2060 686f 616c  etting the `hoal
+000134b0: 7068 6160 2061 7320 3020 286f 7220 6f74  pha` as 0 (or ot
+000134c0: 6865 7220 6d69 6e75 7363 756c 6520 7661  her minuscule va
+000134d0: 6c75 6573 2920 7769 6c6c 2064 6f2e 2049  lues) will do. I
+000134e0: 6620 6f6e 6520 7761 6e74 7320 746f 2073  f one wants to s
+000134f0: 6565 2074 6865 2064 6973 7472 6962 7574  ee the distribut
+00013500: 696f 6e20 6f66 2077 686f 6c65 2064 6f74  ion of whole dot
+00013510: 7320 6265 7474 6572 2c20 7365 7474 696e  s better, settin
+00013520: 6720 7468 6520 6068 6f61 6c70 6861 6020  g the `hoalpha` 
+00013530: 6869 6768 6572 2028 616e 6420 6061 7061  higher (and `apa
+00013540: 6c70 6861 6020 6c6f 7765 7229 2077 696c  lpha` lower) wil
+00013550: 6c20 646f 2e20 5468 6520 6064 6f74 7369  l do. The `dotsi
+00013560: 7a65 6020 6f70 7469 6f6e 2063 616e 2062  ze` option can b
+00013570: 6520 6361 6c6c 6564 2074 6f20 6164 6a75  e called to adju
+00013580: 7374 2074 6865 2073 697a 6520 6f66 2064  st the size of d
+00013590: 6f74 732e 0a0a 4120 6675 7274 6865 7220  ots...A further 
+000135a0: 6173 736f 6369 6174 6564 2053 796e 6465  associated Synde
+000135b0: 7074 6820 706c 6f74 2073 686f 7773 2074  pth plot shows t
+000135c0: 6861 7420 7468 6572 6520 6172 6520 6d6f  hat there are mo
+000135d0: 7265 2074 6861 6e20 3530 2064 7570 6c69  re than 50 dupli
+000135e0: 6361 7465 6420 7365 676d 656e 7473 206c  cated segments l
+000135f0: 6f6e 6765 7220 7468 616e 2031 3030 3030  onger than 10000
+00013600: 2062 7020 616e 6420 3330 2067 656e 6573   bp and 30 genes
+00013610: 2028 736f 2061 7320 746f 2064 726f 7020   (so as to drop 
+00013620: 6672 6167 6d65 6e74 6172 7920 7365 676d  fragmentary segm
+00013630: 656e 7473 292c 2077 6869 6368 2064 6f6d  ents), which dom
+00013640: 696e 6174 6573 2074 6865 2077 686f 6c65  inates the whole
+00013650: 2063 6f6c 6c69 6e65 6172 2072 6174 696f   collinear ratio
+00013660: 2063 6174 6567 6f72 792e 0a0a 215b 5d28   category...![](
+00013670: 6461 7461 2f73 796e 5f72 6573 756c 7473  data/syn_results
+00013680: 2f53 796e 6465 7074 682e 7376 6729 0a0a  /Syndepth.svg)..
+00013690: 4d6f 7265 2065 7871 7569 7369 7465 2070  More exquisite p
+000136a0: 6c6f 7473 2069 6e63 6c75 6469 6e67 2062  lots including b
+000136b0: 6f74 6820 696e 7472 612d 7370 6563 6966  oth intra-specif
+000136c0: 6963 2061 6e64 2069 6e74 6572 2d73 7065  ic and inter-spe
+000136d0: 6369 6669 6320 636f 6d70 6172 6973 6f6e  cific comparison
+000136e0: 7320 7573 696e 6720 7468 6520 6f72 7468  s using the orth
+000136f0: 6f67 726f 7570 7320 2863 6f6d 706f 7365  ogroups (compose
+00013700: 6420 6f66 202a 4171 7569 6c65 6769 6120  d of *Aquilegia 
+00013710: 636f 6572 756c 6561 2a2c 202a 5072 6f74  coerulea*, *Prot
+00013720: 6561 2063 796e 6172 6f69 6465 732a 2c20  ea cynaroides*, 
+00013730: 2a41 636f 7275 7320 616d 6572 6963 616e  *Acorus american
+00013740: 7573 2a20 616e 6420 2a56 6974 6973 2076  us* and *Vitis v
+00013750: 696e 6966 6572 612a 2c20 7365 6520 6675  inifera*, see fu
+00013760: 7274 6865 7220 666f 7220 7468 6520 636f  rther for the co
+00013770: 6e74 6578 7429 2069 6e66 6572 7265 6420  ntext) inferred 
+00013780: 6865 7265 696e 6166 7465 7220 6361 6e20  hereinafter can 
+00013790: 6265 2061 6c73 6f20 7072 6f64 7563 6564  be also produced
+000137a0: 2075 7369 6e67 2060 7767 6420 7379 6e60   using `wgd syn`
+000137b0: 2e20 4e6f 7465 2074 6861 7420 6469 6666  . Note that diff
+000137c0: 6572 656e 7420 6765 6e6f 6d65 2061 7373  erent genome ass
+000137d0: 656d 626c 6965 7320 6d69 6768 7420 6861  emblies might ha
+000137e0: 7665 2064 6966 6665 7265 6e74 2066 6561  ve different fea
+000137f0: 7475 7265 7320 616e 6420 6174 7472 6962  tures and attrib
+00013800: 7574 6573 2077 6869 6368 2063 616e 2062  utes which can b
+00013810: 6520 6163 636f 6d6d 6f64 6174 6564 2076  e accommodated v
+00013820: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
+00013830: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
+00013840: 666f 6020 666f 7220 6561 6368 2067 656e  fo` for each gen
+00013850: 6f6d 6520 6173 7365 6d62 6c79 2077 686f  ome assembly who
+00013860: 7365 206f 7264 6572 206e 6565 6473 2074  se order needs t
+00013870: 6f20 666f 6c6c 6f77 2074 6865 206f 7264  o follow the ord
+00013880: 6572 206f 6620 6766 6633 2066 696c 6573  er of gff3 files
+00013890: 2c20 666f 7220 696e 7374 616e 6365 2027  , for instance '
+000138a0: 6d4e 5241 3b4e 616d 6527 2066 6f72 2060  mNRA;Name' for `
+000138b0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000138c0: 6561 2e67 6666 3360 2c20 276d 4e52 413b  ea.gff3`, 'mNRA;
+000138d0: 4944 2720 666f 7220 6050 726f 7465 615f  ID' for `Protea_
+000138e0: 6379 6e61 726f 6964 6573 2e67 6666 3360  cynaroides.gff3`
+000138f0: 2c20 276d 4e52 413b 4e61 6d65 2720 666f  , 'mNRA;Name' fo
+00013900: 7220 6041 636f 7275 735f 616d 6572 6963  r `Acorus_americ
+00013910: 616e 7573 2e67 6666 3360 2061 6e64 2027  anus.gff3` and '
+00013920: 6d4e 5241 3b4e 616d 6527 2066 6f72 2060  mNRA;Name' for `
+00013930: 5669 7469 735f 7669 6e69 6665 7261 2e67  Vitis_vinifera.g
+00013940: 6666 3360 2e0a 0a60 6060 0a77 6764 2073  ff3`...```.wgd s
+00013950: 796e 2077 6764 5f6f 7274 686f 2f4f 7274  yn wgd_ortho/Ort
+00013960: 686f 6772 6f75 7073 2e73 702e 7473 7620  hogroups.sp.tsv 
+00013970: 2d6b 7320 7767 645f 6f72 7468 6f5f 6b73  -ks wgd_ortho_ks
+00013980: 2f4f 7274 686f 6772 6f75 7073 2e73 702e  /Orthogroups.sp.
+00013990: 7473 762e 6b73 2e74 7376 2041 7175 696c  tsv.ks.tsv Aquil
+000139a0: 6567 6961 5f63 6f65 7275 6c65 612e 6766  egia_coerulea.gf
+000139b0: 6633 202d 2d61 6464 6974 696f 6e61 6c67  f3 --additionalg
+000139c0: 6666 696e 666f 2027 6d4e 5241 3b4e 616d  ffinfo 'mNRA;Nam
+000139d0: 6527 2050 726f 7465 615f 6379 6e61 726f  e' Protea_cynaro
+000139e0: 6964 6573 2e67 6666 3320 2d2d 6164 6469  ides.gff3 --addi
+000139f0: 7469 6f6e 616c 6766 6669 6e66 6f20 276d  tionalgffinfo 'm
+00013a00: 4e52 413b 4944 2720 4163 6f72 7573 5f61  NRA;ID' Acorus_a
+00013a10: 6d65 7269 6361 6e75 732e 6766 6633 202d  mericanus.gff3 -
+00013a20: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
+00013a30: 666f 2027 6d4e 5241 3b4e 616d 6527 2056  fo 'mNRA;Name' V
+00013a40: 6974 6973 5f76 696e 6966 6572 612e 6766  itis_vinifera.gf
+00013a50: 6633 202d 2d61 6464 6974 696f 6e61 6c67  f3 --additionalg
+00013a60: 6666 696e 666f 2027 6d4e 5241 3b4e 616d  ffinfo 'mNRA;Nam
+00013a70: 6527 202d 6f20 7767 645f 6f72 7468 6f5f  e' -o wgd_ortho_
+00013a80: 7379 6e0a 6060 600a 0a55 706f 6e20 7468  syn.```..Upon th
+00013a90: 6520 6163 7175 6973 6974 696f 6e20 6f66  e acquisition of
+00013aa0: 2074 6865 2063 6f6c 6c69 6e65 6172 2072   the collinear r
+00013ab0: 6573 756c 7473 2075 7369 6e67 2060 7767  esults using `wg
+00013ac0: 6420 7379 6e60 2c20 7468 6520 7361 6d65  d syn`, the same
+00013ad0: 2063 6f6c 6c69 6e65 6172 2070 6c6f 7473   collinear plots
+00013ae0: 2063 616e 2062 6520 616c 736f 2070 726f   can be also pro
+00013af0: 6475 6365 6420 6279 2060 7767 6420 7669  duced by `wgd vi
+00013b00: 7a60 2075 7369 6e67 2074 6865 2063 6f6d  z` using the com
+00013b10: 6d61 6e64 2062 656c 6f77 2e0a 6060 600a  mand below..```.
+00013b20: 7767 6420 7669 7a20 2d2d 706c 6f74 7379  wgd viz --plotsy
+00013b30: 6e20 2d73 6d20 7767 645f 6f72 7468 6f5f  n -sm wgd_ortho_
+00013b40: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+00013b50: 7365 676d 656e 7473 2e74 7874 202d 6170  segments.txt -ap
+00013b60: 2077 6764 5f6f 7274 686f 5f73 796e 2f69   wgd_ortho_syn/i
+00013b70: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
+00013b80: 7270 6f69 6e74 732e 7478 7420 2d6d 7420  rpoints.txt -mt 
+00013b90: 7767 645f 6f72 7468 6f5f 7379 6e2f 6961  wgd_ortho_syn/ia
+00013ba0: 6468 6f72 652d 6f75 742f 6d75 6c74 6970  dhore-out/multip
+00013bb0: 6c69 636f 6e73 2e74 7874 202d 6774 2077  licons.txt -gt w
+00013bc0: 6764 5f6f 7274 686f 5f73 796e 2f67 656e  gd_ortho_syn/gen
+00013bd0: 652d 7461 626c 652e 6373 7620 2d64 2077  e-table.csv -d w
+00013be0: 6764 5f6f 7274 686f 5f6b 732f 4f72 7468  gd_ortho_ks/Orth
+00013bf0: 6f67 726f 7570 732e 7370 2e74 7376 2e6b  ogroups.sp.tsv.k
+00013c00: 732e 7473 7620 2d6f 2077 6764 5f6f 7274  s.tsv -o wgd_ort
+00013c10: 686f 5f76 697a 0a60 6060 0a0a 215b 5d28  ho_viz.```..![](
+00013c20: 6461 7461 2f6f 7274 686f 5f73 796e 5f72  data/ortho_syn_r
+00013c30: 6573 756c 7473 2f41 7175 696c 6567 6961  esults/Aquilegia
+00013c40: 5f63 6f65 7275 6c65 615f 5669 7469 735f  _coerulea_Vitis_
+00013c50: 7669 6e69 6665 7261 5f6d 756c 7469 706c  vinifera_multipl
+00013c60: 6963 6f6e 735f 6c65 7665 6c2e 706e 6729  icons_level.png)
+00013c70: 0a54 6865 2061 626f 7665 2060 6475 7053  .The above `dupS
+00013c80: 7461 636b 6020 706c 6f74 2073 686f 7773  tack` plot shows
+00013c90: 2074 6865 2064 6973 7472 6962 7574 696f   the distributio
+00013ca0: 6e20 6f66 2064 7570 6c69 6361 7465 6420  n of duplicated 
+00013cb0: 7365 676d 656e 7473 206f 6620 2a41 7175  segments of *Aqu
+00013cc0: 696c 6567 6961 2063 6f65 7275 6c65 612a  ilegia coerulea*
+00013cd0: 2063 6f6d 7061 7265 6420 746f 2069 7473   compared to its
+00013ce0: 656c 6620 2869 6e20 6772 6565 6e29 2061  elf (in green) a
+00013cf0: 6e64 2063 6f6d 7061 7265 6420 746f 202a  nd compared to *
+00013d00: 5669 7469 7320 7669 6e69 6665 7261 2a20  Vitis vinifera* 
+00013d10: 2869 6e20 626c 7565 2920 6f76 6572 2074  (in blue) over t
+00013d20: 6865 2063 6872 6f6d 6f73 6f6d 6573 206f  he chromosomes o
+00013d30: 6620 2a41 2e20 636f 6572 756c 6561 2a2e  f *A. coerulea*.
+00013d40: 0a21 5b5d 2864 6174 612f 6f72 7468 6f5f  .![](data/ortho_
+00013d50: 7379 6e5f 7265 7375 6c74 732f 4f76 6572  syn_results/Over
+00013d60: 616c 6c73 7065 6369 6573 5f4b 732e 646f  allspecies_Ks.do
+00013d70: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
+00013d80: 0a54 6865 2061 626f 7665 202a 4b2a 3c73  .The above *K*<s
+00013d90: 7562 3e53 3c2f 7375 623e 2064 6f74 706c  ub>S</sub> dotpl
+00013da0: 6f74 2069 6e20 756e 6974 206f 6620 6765  ot in unit of ge
+00013db0: 6e65 2073 686f 7773 2074 6865 206f 7665  ne shows the ove
+00013dc0: 7261 6c6c 2064 6973 7472 6962 7574 696f  rall distributio
+00013dd0: 6e20 6f66 2063 6f6c 6c69 6e65 6172 6974  n of collinearit
+00013de0: 7920 6163 6f72 7373 2074 6865 2066 6f75  y acorss the fou
+00013df0: 7220 7370 6563 6965 7320 696e 766f 6c76  r species involv
+00013e00: 6564 2e0a 215b 5d28 6461 7461 2f6f 7274  ed..![](data/ort
+00013e10: 686f 5f73 796e 5f72 6573 756c 7473 2f4f  ho_syn_results/O
+00013e20: 7665 7261 6c6c 7370 6563 6965 732e 646f  verallspecies.do
+00013e30: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
+00013e40: 0a54 6865 2061 626f 7665 2064 6f74 706c  .The above dotpl
+00013e50: 6f74 2069 7320 7769 7468 6f75 7420 7468  ot is without th
+00013e60: 6520 616e 6e6f 7461 7469 6f6e 206f 6620  e annotation of 
+00013e70: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00013e80: 6167 6573 2063 6f6d 7061 7265 6420 746f  ages compared to
+00013e90: 2074 6865 206c 6173 7420 6f6e 652e 0a21   the last one..!
+00013ea0: 5b5d 2864 6174 612f 6f72 7468 6f5f 7379  [](data/ortho_sy
+00013eb0: 6e5f 7265 7375 6c74 732f 5379 6e64 6570  n_results/Syndep
+00013ec0: 7468 2e73 7667 290a 5468 6520 6162 6f76  th.svg).The abov
+00013ed0: 6520 5379 6e64 6570 7468 2070 6c6f 7420  e Syndepth plot 
+00013ee0: 7368 6f77 7320 7468 6520 636f 6c6c 696e  shows the collin
+00013ef0: 6561 7220 7261 7469 6f20 6163 6f72 7373  ear ratio acorss
+00013f00: 2061 6c6c 2073 7065 6369 6573 2070 6169   all species pai
+00013f10: 7273 2028 696e 7472 612d 7370 6563 6966  rs (intra-specif
+00013f20: 6963 2063 6f6d 7061 7269 736f 6e20 696e  ic comparison in
+00013f30: 2067 7265 656e 2077 6869 6c65 2069 6e74   green while int
+00013f40: 6572 2d73 7065 6369 6669 6320 636f 6d70  er-specific comp
+00013f50: 6172 6973 6f6e 2069 6e20 626c 7565 292e  arison in blue).
+00013f60: 0a0a 5765 2063 616e 2066 6974 2061 6e20  ..We can fit an 
+00013f70: 454c 4d4d 206d 6978 7475 7265 206d 6f64  ELMM mixture mod
+00013f80: 656c 2075 706f 6e20 7468 6520 7768 6f6c  el upon the whol
+00013f90: 6520 7061 7261 6e6f 6d65 202a 4b2a 3c73  e paranome *K*<s
+00013fa0: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+00013fb0: 6973 7472 6962 7574 696f 6e20 746f 2073  istribution to s
+00013fc0: 6565 206d 6f72 6520 6163 6375 7261 7465  ee more accurate
+00013fd0: 6c79 2074 6865 2073 6967 6e69 6669 6361  ly the significa
+00013fe0: 6e63 6520 616e 6420 6c6f 6361 7469 6f6e  nce and location
+00013ff0: 206f 6620 706f 7465 6e74 6961 6c20 5747   of potential WG
+00014000: 4473 2c20 7573 696e 6720 7468 6520 636f  Ds, using the co
+00014010: 6d6d 616e 6420 6c69 6e65 2062 656c 6f77  mmand line below
+00014020: 2e0a 0a60 6060 0a77 6764 2076 697a 202d  ...```.wgd viz -
+00014030: 6420 7767 645f 6b73 642f 4171 7569 6c65  d wgd_ksd/Aquile
+00014040: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00014050: 2e6b 732e 7473 760a 6060 600a 0a54 6865  .ks.tsv.```..The
+00014060: 2072 6573 756c 7420 6f66 2045 4c4d 4d20   result of ELMM 
+00014070: 6d69 7874 7572 6520 6d6f 6465 6c20 636c  mixture model cl
+00014080: 7573 7465 7269 6e67 2073 686f 7773 2074  ustering shows t
+00014090: 6861 7420 7468 6572 6520 6973 2061 206c  hat there is a l
+000140a0: 696b 656c 7920 5747 4420 636f 6d70 6f6e  ikely WGD compon
+000140b0: 656e 7420 6174 202a 4b2a 3c73 7562 3e53  ent at *K*<sub>S
+000140c0: 3c2f 7375 623e 2031 2e31 392e 0a0a 215b  </sub> 1.19...![
+000140d0: 5d28 6461 7461 2f65 6c6d 6d5f 4171 7569  ](data/elmm_Aqui
+000140e0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+000140f0: 7376 2e6b 732e 7473 765f 6265 7374 5f6d  sv.ks.tsv_best_m
+00014100: 6f64 656c 735f 7765 6967 6874 6564 2e73  odels_weighted.s
+00014110: 7667 290a 0a4c 6574 2773 2064 6f20 6120  vg)..Let's do a 
+00014120: 6d69 7874 7572 6520 6d6f 6465 6c20 636c  mixture model cl
+00014130: 7573 7465 7269 6e67 2066 6f72 2061 6e63  ustering for anc
+00014140: 686f 7220 2a4b 2a3c 7375 623e 533c 2f73  hor *K*<sub>S</s
+00014150: 7562 3e20 746f 6f2c 2075 7369 6e67 2074  ub> too, using t
+00014160: 6865 2063 6f6d 6d61 6e64 206c 696e 6520  he command line 
+00014170: 6265 6c6f 772e 204e 6f74 6520 7468 6174  below. Note that
+00014180: 2074 6869 7320 7374 6570 2077 696c 6c20   this step will 
+00014190: 6175 746f 6d61 7465 6c79 2063 616c 6c20  automately call 
+000141a0: 7468 6520 7365 676d 656e 7420 2a4b 2a3c  the segment *K*<
+000141b0: 7375 623e 533c 2f73 7562 3e20 636c 7573  sub>S</sub> clus
+000141c0: 7465 7269 6e67 2061 6e61 6c79 7369 7320  tering analysis 
+000141d0: 746f 6f2e 0a0a 6060 600a 7767 6420 7065  too...```.wgd pe
+000141e0: 616b 2077 6764 5f6b 7364 2f41 7175 696c  ak wgd_ksd/Aquil
+000141f0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00014200: 762e 6b73 2e74 7376 202d 2d61 6e63 686f  v.ks.tsv --ancho
+00014210: 7270 6f69 6e74 7320 7767 645f 7379 6e2f  rpoints wgd_syn/
+00014220: 6961 6468 6f72 652d 6f75 742f 616e 6368  iadhore-out/anch
+00014230: 6f72 706f 696e 7473 2e74 7874 202d 2d73  orpoints.txt --s
+00014240: 6567 6d65 6e74 7320 7767 645f 7379 6e2f  egments wgd_syn/
+00014250: 6961 6468 6f72 652d 6f75 742f 7365 676d  iadhore-out/segm
+00014260: 656e 7473 2e74 7874 202d 2d6c 6973 7465  ents.txt --liste
+00014270: 6c65 6d65 6e74 7320 7767 645f 7379 6e2f  lements wgd_syn/
+00014280: 6961 6468 6f72 652d 6f75 742f 6c69 7374  iadhore-out/list
+00014290: 5f65 6c65 6d65 6e74 732e 7478 7420 2d2d  _elements.txt --
+000142a0: 6d75 6c74 6970 6c69 636f 6e70 6169 7273  multipliconpairs
+000142b0: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
+000142c0: 2d6f 7574 2f6d 756c 7469 706c 6963 6f6e  -out/multiplicon
+000142d0: 5f70 6169 7273 2e74 7874 2028 2d2d 7765  _pairs.txt (--we
+000142e0: 6967 6874 6564 290a 6060 600a 0a54 6865  ighted).```..The
+000142f0: 2061 6e63 686f 7220 2a4b 2a3c 7375 623e   anchor *K*<sub>
+00014300: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
+00014310: 7269 6275 7469 6f6e 2061 6c73 6f20 6861  ribution also ha
+00014320: 7320 6120 6c69 6b65 6c79 2057 4744 2063  s a likely WGD c
+00014330: 6f6d 706f 6e65 6e74 2077 6974 6820 6d6f  omponent with mo
+00014340: 6465 2031 2e32 382e 0a0a 215b 5d28 6461  de 1.28...![](da
+00014350: 7461 2f4f 7269 6769 6e61 6c5f 416e 6368  ta/Original_Anch
+00014360: 6f72 4b73 5f47 4d4d 5f43 6f6d 706f 6e65  orKs_GMM_Compone
+00014370: 6e74 335f 6e6f 6465 5f61 7665 7261 6765  nt3_node_average
+00014380: 645f 4c6f 676e 6f72 6d61 6c2e 7376 6729  d_Lognormal.svg)
+00014390: 0a0a 4e6f 7720 7468 6174 2077 6520 6861  ..Now that we ha
+000143a0: 7665 2073 6565 6e20 7468 6520 6576 6964  ve seen the evid
+000143b0: 656e 6365 206f 6620 6e75 6d65 726f 7573  ence of numerous
+000143c0: 2064 7570 6c69 6361 7465 6420 7365 676d   duplicated segm
+000143d0: 656e 7473 2061 6e64 2074 6865 2061 6767  ents and the agg
+000143e0: 7265 6761 7469 6f6e 206f 6620 6475 706c  regation of dupl
+000143f0: 6963 6174 6573 2061 6765 2061 7420 2a4b  icates age at *K
+00014400: 2a3c 7375 623e 533c 2f73 7562 3e20 312e  *<sub>S</sub> 1.
+00014410: 3238 206f 7220 312e 3139 2066 6f72 2061  28 or 1.19 for a
+00014420: 6e63 686f 7220 7061 6972 7320 616e 6420  nchor pairs and 
+00014430: 6e6f 6e2d 616e 6368 6f72 2070 6169 7273  non-anchor pairs
+00014440: 2074 6872 6f75 6768 6f75 7420 7468 6520   throughout the 
+00014450: 7768 6f6c 6520 6765 6e6f 6d65 2e20 5765  whole genome. We
+00014460: 2063 616e 2063 6c61 696d 2077 6974 6820   can claim with 
+00014470: 736f 6d65 2063 6f6e 6669 6465 6e63 6520  some confidence 
+00014480: 7468 6174 202a 4171 7569 6c65 6769 6120  that *Aquilegia 
+00014490: 636f 6572 756c 6561 2a20 6d69 6768 7420  coerulea* might 
+000144a0: 6861 7665 2065 7870 6572 6965 6e63 6564  have experienced
+000144b0: 2061 2070 616c 656f 2d70 6f6c 7970 6c6f   a paleo-polyplo
+000144c0: 6964 697a 6174 696f 6e20 6576 656e 742e  idization event.
+000144d0: 204e 6578 742c 204c 6574 2773 2068 6176   Next, Let's hav
+000144e0: 6520 6120 6675 7274 6865 7220 6c6f 6f6b  e a further look
+000144f0: 2061 626f 7574 2069 7473 2070 6879 6c6f   about its phylo
+00014500: 6765 6e65 7469 6320 6c6f 6361 7469 6f6e  genetic location
+00014510: 2e20 5765 206b 6e6f 7720 7468 6174 2074  . We know that t
+00014520: 6865 7265 2061 7265 2075 6e63 6572 7461  here are uncerta
+00014530: 696e 7469 6573 2061 626f 7574 2077 6865  inties about whe
+00014540: 7468 6572 2074 6869 7320 7075 7461 7469  ther this putati
+00014550: 7665 2070 616c 656f 2d70 6f6c 7970 6c6f  ve paleo-polyplo
+00014560: 6964 697a 6174 696f 6e20 6576 656e 7420  idization event 
+00014570: 6973 2073 6861 7265 6420 7769 7468 2061  is shared with a
+00014580: 6c6c 2065 7564 6963 6f74 7320 6f72 206e  ll eudicots or n
+00014590: 6f74 2e20 5765 2063 616e 2063 686f 6f73  ot. We can choos
+000145a0: 6520 736f 6d65 206f 7468 6572 2065 7564  e some other eud
+000145b0: 6963 6f74 2067 656e 6f6d 6573 2074 6f20  icot genomes to 
+000145c0: 7365 6520 7468 6520 6f72 6465 7269 6e67  see the ordering
+000145d0: 206f 6620 7370 6563 6961 7469 6f6e 2061   of speciation a
+000145e0: 6e64 2070 6f6c 7970 6c6f 6964 697a 6174  nd polyploidizat
+000145f0: 696f 6e20 6576 656e 7473 2e20 4865 7265  ion events. Here
+00014600: 2077 6520 6368 6f6f 7365 202a 5669 7469   we choose *Viti
+00014610: 7320 7669 6e69 6665 7261 2a2c 202a 5072  s vinifera*, *Pr
+00014620: 6f74 6561 2063 796e 6172 6f69 6465 732a  otea cynaroides*
+00014630: 2061 6e64 202a 4163 6f72 7573 2061 6d65   and *Acorus ame
+00014640: 7269 6361 6e75 732a 2069 6e20 7468 6520  ricanus* in the 
+00014650: 666f 6c6c 6f77 696e 6720 2a4b 2a3c 7375  following *K*<su
+00014660: 623e 533c 2f73 7562 3e20 616e 616c 7973  b>S</sub> analys
+00014670: 6973 2e20 4669 7273 742c 2077 6520 6275  is. First, we bu
+00014680: 696c 7420 6120 676c 6f62 616c 204d 5242  ilt a global MRB
+00014690: 4820 6661 6d69 6c79 2075 7369 6e67 2074  H family using t
+000146a0: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
+000146b0: 2e0a 0a60 6060 0a77 6764 2064 6d64 202d  ...```.wgd dmd -
+000146c0: 2d67 6c6f 6261 6c6d 7262 6820 4171 7569  -globalmrbh Aqui
+000146d0: 6c65 6769 615f 636f 6572 756c 6561 2050  legia_coerulea P
+000146e0: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
+000146f0: 2041 636f 7275 735f 616d 6572 6963 616e   Acorus_american
+00014700: 7573 2056 6974 6973 5f76 696e 6966 6572  us Vitis_vinifer
+00014710: 6120 2d6f 2077 6764 5f67 6c6f 6261 6c6d  a -o wgd_globalm
+00014720: 7262 680a 6060 600a 0a49 6e20 7468 6520  rbh.```..In the 
+00014730: 676c 6f62 616c 204d 5242 4820 6661 6d69  global MRBH fami
+00014740: 6c79 2c20 6576 6572 7920 7061 6972 206f  ly, every pair o
+00014750: 6620 6f72 7468 6f6c 6f67 6f75 7320 6765  f orthologous ge
+00014760: 6e65 7320 6973 2074 6865 2072 6563 6970  nes is the recip
+00014770: 726f 6361 6c20 6265 7374 2068 6974 2c20  rocal best hit, 
+00014780: 7375 6767 6573 7469 6e67 2074 7275 6520  suggesting true 
+00014790: 6f72 7468 6f6c 6f67 6f75 7320 7265 6c61  orthologous rela
+000147a0: 7469 6f6e 7368 6970 732e 2057 6520 776f  tionships. We wo
+000147b0: 756c 6420 7573 6520 7468 6520 2a4b 2a3c  uld use the *K*<
+000147c0: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
+000147d0: 6573 2061 7373 6f63 6961 7465 6420 7769  es associated wi
+000147e0: 7468 2074 6865 7365 206f 7274 686f 6c6f  th these ortholo
+000147f0: 676f 7573 2070 6169 7273 2074 6f20 6465  gous pairs to de
+00014800: 6c69 6d69 7420 7468 6520 6469 7665 7267  limit the diverg
+00014810: 656e 6365 202a 4b2a 3c73 7562 3e53 3c2f  ence *K*<sub>S</
+00014820: 7375 623e 2070 6561 6b2e 2054 6f67 6574  sub> peak. Toget
+00014830: 6865 7220 7769 7468 2074 6865 2077 686f  her with the who
+00014840: 6c65 2070 6172 616e 6f6d 6520 2a4b 2a3c  le paranome *K*<
+00014850: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00014860: 7269 6275 7469 6f6e 2c20 7765 2063 6f6e  ribution, we con
+00014870: 6475 6374 2074 6865 2072 6174 6520 636f  duct the rate co
+00014880: 7272 6563 7469 6f6e 2075 7369 6e67 2074  rrection using t
+00014890: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
+000148a0: 2e0a 0a21 2153 696e 6365 2060 7767 6460  ...!!Since `wgd`
+000148b0: 2076 6572 7369 6f6e 2032 2e30 2e32 342c   version 2.0.24,
+000148c0: 2077 6520 7265 7772 6f74 6520 6120 636c   we rewrote a cl
+000148d0: 6561 6e65 7220 616e 6420 7175 6963 6b65  eaner and quicke
+000148e0: 7220 7761 7920 6f66 2064 6f69 6e67 2073  r way of doing s
+000148f0: 7562 7374 6974 7574 696f 6e20 7261 7465  ubstitution rate
+00014900: 2063 6f72 7265 6374 696f 6e2e 2049 7427   correction. It'
+00014910: 7320 6e6f 7420 7265 7175 6972 6564 2074  s not required t
+00014920: 6f20 7479 7065 2069 6e20 616e 7920 7370  o type in any sp
+00014930: 6569 6365 7320 7061 6972 2061 6e64 2061  eices pair and a
+00014940: 2073 6572 6965 7320 6f66 202a 4b2a 3c73   series of *K*<s
+00014950: 7562 3e53 3c2f 7375 623e 2070 6c6f 7473  ub>S</sub> plots
+00014960: 2077 696c 6c20 6265 2070 726f 6475 6365   will be produce
+00014970: 642e 2054 6865 2072 6571 7569 7265 6420  d. The required 
+00014980: 6669 6c65 7320 6172 6520 6f72 7468 6f6c  files are orthol
+00014990: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
+000149a0: 2f73 7562 3e20 6461 7461 6669 6c65 2c20  /sub> datafile, 
+000149b0: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
+000149c0: 7562 3e53 3c2f 7375 623e 2064 6174 6166  ub>S</sub> dataf
+000149d0: 696c 652c 2061 2073 7065 6369 6573 2074  ile, a species t
+000149e0: 7265 6520 616e 6420 6120 666f 6361 6c20  ree and a focal 
+000149f0: 7370 6563 6965 7320 2874 6865 206f 6e65  species (the one
+00014a00: 2069 6e70 7574 7465 6420 7769 7468 2070   inputted with p
+00014a10: 6172 616c 6f67 6f75 7320 2a4b 2a3c 7375  aralogous *K*<su
+00014a20: 623e 533c 2f73 7562 3e20 6461 7461 292e  b>S</sub> data).
+00014a30: 2055 7365 7273 2063 616e 2063 686f 6f73   Users can choos
+00014a40: 6520 746f 2061 6464 206f 6e65 206d 6f72  e to add one mor
+00014a50: 6520 6c61 7965 7220 6f66 2045 4c4d 4d20  e layer of ELMM 
+00014a60: 616e 616c 7973 6973 206f 6e20 7061 7261  analysis on para
+00014a70: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
+00014a80: 3c2f 7375 623e 2076 616c 7565 7320 616e  </sub> values an
+00014a90: 642f 6f72 2047 4d4d 2061 6e61 6c79 7369  d/or GMM analysi
+00014aa0: 7320 6f6e 2061 6e63 686f 7220 2a4b 2a3c  s on anchor *K*<
+00014ab0: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00014ac0: 7269 6275 7469 6f6e 2e20 5468 6520 6f72  ribution. The or
+00014ad0: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
+00014ae0: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
+00014af0: 6275 7469 6f6e 2063 616e 2062 6520 6361  bution can be ca
+00014b00: 6c63 756c 6174 6564 2075 7369 6e67 2074  lculated using t
+00014b10: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
+00014b20: 2e0a 0a60 6060 0a77 6764 206b 7364 2077  ...```.wgd ksd w
+00014b30: 6764 5f67 6c6f 6261 6c6d 7262 682f 676c  gd_globalmrbh/gl
+00014b40: 6f62 616c 5f4d 5242 482e 7473 7620 4171  obal_MRBH.tsv Aq
+00014b50: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00014b60: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
+00014b70: 6573 2041 636f 7275 735f 616d 6572 6963  es Acorus_americ
+00014b80: 616e 7573 2056 6974 6973 5f76 696e 6966  anus Vitis_vinif
+00014b90: 6572 6120 2d6f 2077 6764 5f67 6c6f 6261  era -o wgd_globa
+00014ba0: 6c6d 7262 685f 6b73 0a60 6060 0a0a 5769  lmrbh_ks.```..Wi
+00014bb0: 7468 2074 6865 2063 616c 6375 6c61 7465  th the calculate
+00014bc0: 6420 6f72 7468 6f6c 6f67 6f75 7320 2a4b  d orthologous *K
+00014bd0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+00014be0: 7374 7269 6275 7469 6f6e 2c20 7765 2063  stribution, we c
+00014bf0: 616e 2075 7365 2074 6865 2063 6f6d 6d61  an use the comma
+00014c00: 6e64 2062 656c 6f77 2074 6f20 636f 6e64  nd below to cond
+00014c10: 7563 7420 7468 6520 7261 7465 2063 6f72  uct the rate cor
+00014c20: 7265 6374 696f 6e20 616e 642f 6f72 206d  rection and/or m
+00014c30: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
+00014c40: 616e 616c 7973 6973 2e0a 0a60 6060 0a77  analysis...```.w
+00014c50: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
+00014c60: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
+00014c70: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
+00014c80: 7376 202d 6661 2041 7175 696c 6567 6961  sv -fa Aquilegia
+00014c90: 5f63 6f65 7275 6c65 6120 2d65 706b 2077  _coerulea -epk w
+00014ca0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
+00014cb0: 5f63 6f65 7275 6c65 612e 6b73 2e74 7376  _coerulea.ks.tsv
+00014cc0: 202d 6170 2077 6764 5f73 796e 2f69 6164   -ap wgd_syn/iad
+00014cd0: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
+00014ce0: 6f69 6e74 732e 7478 7420 2d73 7020 7370  oints.txt -sp sp
+00014cf0: 6563 6965 7374 7265 652e 6e77 202d 6f20  eciestree.nw -o 
+00014d00: 7767 645f 7669 7a5f 6d69 7865 645f 4b73  wgd_viz_mixed_Ks
+00014d10: 202d 2d70 6c6f 7465 6c6d 6d20 2d2d 706c   --plotelmm --pl
+00014d20: 6f74 6170 676d 6d20 2d2d 7265 7765 6967  otapgmm --reweig
+00014d30: 6874 0a60 6060 0a0a 6f72 2075 7369 6e67  ht.```..or using
+00014d40: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+00014d50: 6f77 2c20 7768 6963 6820 636f 6d62 696e  ow, which combin
+00014d60: 6573 2074 6865 2074 776f 2073 7465 7073  es the two steps
+00014d70: 2061 626f 7665 2069 6e20 6f6e 652e 204e   above in one. N
+00014d80: 6f74 6520 7468 6174 2077 6520 7375 6767  ote that we sugg
+00014d90: 6573 7420 6f66 2074 616b 696e 6720 7477  est of taking tw
+00014da0: 6f20 7365 7061 7261 7465 2073 7465 7073  o separate steps
+00014db0: 2069 6e20 7768 6963 6820 6077 6764 206b   in which `wgd k
+00014dc0: 7364 6020 756e 6465 7274 616b 6573 2074  sd` undertakes t
+00014dd0: 6865 2063 616c 6375 6c61 7469 6f6e 206f  he calculation o
+00014de0: 6620 6f72 7468 6f6c 6f67 6f75 7320 2a4b  f orthologous *K
+00014df0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+00014e00: 7374 7269 6275 7469 6f6e 2077 6869 6c65  stribution while
+00014e10: 2060 7767 6420 7669 7a60 2063 6172 7269   `wgd viz` carri
+00014e20: 6573 206f 7574 2074 6865 2072 6174 6520  es out the rate 
+00014e30: 636f 7272 6563 7469 6f6e 2061 6e64 2047  correction and G
+00014e40: 4d4d 2061 6e61 6c79 7369 7320 7375 6368  MM analysis such
+00014e50: 2074 6861 7420 6974 2773 2065 6173 6965   that it's easie
+00014e60: 7220 746f 2064 6562 7567 2e0a 6060 600a  r to debug..```.
+00014e70: 7767 6420 6b73 6420 7767 645f 676c 6f62  wgd ksd wgd_glob
+00014e80: 616c 6d72 6268 2f67 6c6f 6261 6c5f 4d52  almrbh/global_MR
+00014e90: 4248 2e74 7376 2041 7175 696c 6567 6961  BH.tsv Aquilegia
+00014ea0: 5f63 6f65 7275 6c65 6120 5072 6f74 6561  _coerulea Protea
+00014eb0: 5f63 796e 6172 6f69 6465 7320 4163 6f72  _cynaroides Acor
+00014ec0: 7573 5f61 6d65 7269 6361 6e75 7320 5669  us_americanus Vi
+00014ed0: 7469 735f 7669 6e69 6665 7261 202d 2d65  tis_vinifera --e
+00014ee0: 7874 7261 7061 7261 6e6f 6d65 6b73 2077  xtraparanomeks w
+00014ef0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
+00014f00: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00014f10: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
+00014f20: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
+00014f30: 6874 202d 6f20 7767 645f 676c 6f62 616c  ht -o wgd_global
+00014f40: 6d72 6268 5f6b 735f 7261 7465 5f63 6f72  mrbh_ks_rate_cor
+00014f50: 7265 6374 696f 6e20 2d66 6120 4171 7569  rection -fa Aqui
+00014f60: 6c65 6769 615f 636f 6572 756c 6561 202d  legia_coerulea -
+00014f70: 6170 2077 6764 5f73 796e 2f69 6164 686f  ap wgd_syn/iadho
+00014f80: 7265 2d6f 7574 2f61 6e63 686f 7270 6f69  re-out/anchorpoi
+00014f90: 6e74 732e 7478 7420 2d2d 706c 6f74 656c  nts.txt --plotel
+00014fa0: 6d6d 202d 2d70 6c6f 7461 7067 6d6d 0a60  mm --plotapgmm.`
+00014fb0: 6060 0a0a 5468 6520 6669 6c65 2060 7370  ``..The file `sp
+00014fc0: 6563 6965 7374 7265 652e 6e77 6020 6973  eciestree.nw` is
+00014fd0: 2074 6865 2074 6578 7420 6669 6c65 206f   the text file o
+00014fe0: 6620 7370 6563 6965 7320 7472 6565 2069  f species tree i
+00014ff0: 6e20 6e65 7769 636b 2074 6861 7420 7261  n newick that ra
+00015000: 7465 2063 6f72 7265 6374 696f 6e20 776f  te correction wo
+00015010: 756c 6420 6265 2063 6f6e 6475 6374 6564  uld be conducted
+00015020: 206f 6e2e 2049 7473 2063 6f6e 7465 6e74   on. Its content
+00015030: 2069 7320 6173 2062 656c 6f77 2e20 5573   is as below. Us
+00015040: 6572 7320 6361 6e20 6f70 7469 6f6e 616c  ers can optional
+00015050: 6c79 2070 726f 7669 6465 2074 6865 2073  ly provide the s
+00015060: 7065 6369 6573 2070 6169 7273 2074 6f20  pecies pairs to 
+00015070: 6265 2070 6c6f 7474 6564 2062 7574 2077  be plotted but w
+00015080: 6520 7375 6767 6573 7420 6f66 206a 7573  e suggest of jus
+00015090: 7420 7573 696e 6720 602d 6661 2041 7175  t using `-fa Aqu
+000150a0: 696c 6567 6961 5f63 6f65 7275 6c65 6160  ilegia_coerulea`
+000150b0: 2074 6f20 706c 6f74 2061 6c6c 2070 6f73   to plot all pos
+000150c0: 7369 626c 6520 666f 6361 6c2d 7369 7374  sible focal-sist
+000150d0: 6572 2073 7065 6369 6573 2070 6169 7273  er species pairs
+000150e0: 2e20 5765 2073 7567 6765 7374 2061 6464  . We suggest add
+000150f0: 696e 6720 7468 6520 6f70 7469 6f6e 2060  ing the option `
+00015100: 2d2d 7265 7765 6967 6874 6020 746f 2072  --reweight` to r
+00015110: 6563 616c 6375 6c61 7465 2074 6865 2077  ecalculate the w
+00015120: 6569 6768 7420 7065 7220 7370 6563 6965  eight per specie
+00015130: 7320 7061 6972 2073 7563 6820 7468 6174  s pair such that
+00015140: 2074 6865 2077 6569 6768 7420 6f66 206f   the weight of o
+00015150: 7274 686f 6c6f 676f 7573 2067 656e 6520  rthologous gene 
+00015160: 7061 6972 7320 7769 6c6c 2062 6563 6f6d  pairs will becom
+00015170: 6520 312e 2045 7874 7261 2063 6f6c 6c69  e 1. Extra colli
+00015180: 6e65 6172 2064 6174 6120 6361 6e20 6265  near data can be
+00015190: 2061 6464 6564 2062 7920 7468 6520 6f70   added by the op
+000151a0: 7469 6f6e 2060 2d61 7060 2061 6e64 2061  tion `-ap` and a
+000151b0: 6464 6974 696f 6e61 6c20 636c 7573 7465  dditional cluste
+000151c0: 7269 6e67 2061 6e61 6c79 7369 7320 6361  ring analysis ca
+000151d0: 6e20 6265 2069 6e69 7469 6174 6564 2062  n be initiated b
+000151e0: 7920 7365 7474 696e 6720 7468 6520 6f70  y setting the op
+000151f0: 7469 6f6e 2060 2d2d 706c 6f74 6170 676d  tion `--plotapgm
+00015200: 6d60 2061 6e64 2060 2d2d 706c 6f74 656c  m` and `--plotel
+00015210: 6d6d 602e 0a0a 6060 600a 2828 2856 6974  mm`...```.(((Vit
+00015220: 6973 5f76 696e 6966 6572 612c 5072 6f74  is_vinifera,Prot
+00015230: 6561 5f63 796e 6172 6f69 6465 7329 2c41  ea_cynaroides),A
+00015240: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00015250: 6129 2c41 636f 7275 735f 616d 6572 6963  a),Acorus_americ
+00015260: 616e 7573 293b 0a60 6060 0a0a 215b 5d28  anus);.```..![](
+00015270: 6461 7461 2f4d 6978 6564 2e6b 732e 4171  data/Mixed.ks.Aq
+00015280: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00015290: 2e6e 6f64 652e 7765 6967 6874 6564 2e70  .node.weighted.p
+000152a0: 6e67 290a 0a54 6865 206d 6978 6564 202a  ng)..The mixed *
+000152b0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
+000152c0: 6973 7472 6962 7574 696f 6e20 7368 6f77  istribution show
+000152d0: 6e20 6162 6f76 6520 6973 2061 2070 7562  n above is a pub
+000152e0: 6c69 6361 7469 6f6e 2d72 6561 6479 2066  lication-ready f
+000152f0: 6967 7572 6520 7468 6174 2061 7373 656d  igure that assem
+00015300: 626c 6573 2074 6865 2072 6573 756c 7473  bles the results
+00015310: 206f 6620 454c 4d4d 2c20 474d 4d20 616e   of ELMM, GMM an
+00015320: 6420 7261 7465 2063 6f72 7265 6374 696f  d rate correctio
+00015330: 6e2e 2054 6865 206f 6e65 2d76 732d 6f6e  n. The one-vs-on
+00015340: 6520 6f72 7468 6f6c 6f67 6f75 7320 2a4b  e orthologous *K
+00015350: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+00015360: 7374 7269 6275 7469 6f6e 7320 6973 2061  stributions is a
+00015370: 6c73 6f20 6175 746f 6d61 7469 6361 6c6c  lso automaticall
+00015380: 7920 7072 6f64 7563 6564 2077 6974 6820  y produced with 
+00015390: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
+000153a0: 7265 7375 6c74 7320 7375 7065 7269 6d70  results superimp
+000153b0: 6f73 6564 2077 6865 7265 2061 7661 696c  osed where avail
+000153c0: 6162 6c65 2061 7320 7368 6f77 6e20 6265  able as shown be
+000153d0: 6c6f 772e 0a0a 215b 5d28 6461 7461 2f46  low...![](data/F
+000153e0: 6f63 7573 5f73 6973 7465 725f 7061 6972  ocus_sister_pair
+000153f0: 732e 6b73 2e6e 6f64 652e 7765 6967 6874  s.ks.node.weight
+00015400: 6564 2e70 6e67 290a 215b 5d28 6461 7461  ed.png).![](data
+00015410: 2f41 6c6c 5f70 6169 7273 2e6b 732e 6e6f  /All_pairs.ks.no
+00015420: 6465 2e77 6569 6768 7465 642e 706e 6729  de.weighted.png)
+00015430: 0a0a 4265 7369 6465 7320 7468 6520 6162  ..Besides the ab
+00015440: 6f76 6520 6b73 7261 7465 732d 6c69 6b65  ove ksrates-like
+00015450: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00015460: 2070 6c6f 742c 2061 206d 6f72 6520 636c   plot, a more cl
+00015470: 6173 7369 6320 2a4b 2a3c 7375 623e 533c  assic *K*<sub>S<
+00015480: 2f73 7562 3e20 706c 6f74 2063 616e 2062  /sub> plot can b
+00015490: 6520 6d61 6465 2062 7920 6164 6469 6e67  e made by adding
+000154a0: 2074 6865 206f 7074 696f 6e20 602d 2d63   the option `--c
+000154b0: 6c61 7373 6963 6020 746f 2074 6170 206d  lassic` to tap m
+000154c0: 6f72 6520 6465 7461 696c 6564 6c79 2069  ore detailedly i
+000154d0: 6e74 6f20 7468 6520 7661 7269 6174 696f  nto the variatio
+000154e0: 6e20 6f66 2073 796e 6f6e 796d 6f75 7320  n of synonymous 
+000154f0: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
+00015500: 652e 0a0a 5573 696e 6720 7468 6520 636f  e...Using the co
+00015510: 6d6d 616e 6420 6265 6c6f 772c 2074 6865  mmand below, the
+00015520: 2064 6972 6563 7469 6f6e 206f 6620 7261   direction of ra
+00015530: 7465 2063 6f72 7265 6374 696f 6e20 616e  te correction an
+00015540: 6420 6465 6772 6565 206f 6620 7261 7465  d degree of rate
+00015550: 2076 6172 6961 7469 6f6e 2063 616e 2062   variation can b
+00015560: 6520 6f62 7365 7276 6564 206d 6f72 6520  e observed more 
+00015570: 6469 7265 6374 6c79 2e0a 0a60 6060 0a77  directly...```.w
+00015580: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
+00015590: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
+000155a0: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
+000155b0: 7376 202d 6661 2041 7175 696c 6567 6961  sv -fa Aquilegia
+000155c0: 5f63 6f65 7275 6c65 6120 2d65 706b 2077  _coerulea -epk w
+000155d0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
+000155e0: 5f63 6f65 7275 6c65 612e 6b73 2e74 7376  _coerulea.ks.tsv
+000155f0: 202d 6170 2077 6764 5f73 796e 2f69 6164   -ap wgd_syn/iad
+00015600: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
+00015610: 6f69 6e74 732e 7478 7420 2d73 7020 7370  oints.txt -sp sp
+00015620: 6563 6965 7374 7265 652e 6e77 202d 6f20  eciestree.nw -o 
+00015630: 7767 645f 7669 7a5f 6d69 7865 645f 4b73  wgd_viz_mixed_Ks
+00015640: 202d 2d70 6c6f 7465 6c6d 6d20 2d2d 706c   --plotelmm --pl
+00015650: 6f74 6170 676d 6d20 2d2d 7265 7765 6967  otapgmm --reweig
+00015660: 6874 202d 2d70 6c6f 746b 6465 202d 2d63  ht --plotkde --c
+00015670: 6c61 7373 6963 0a60 6060 0a0a 215b 5d28  lassic.```..![](
+00015680: 6461 7461 2f41 7175 696c 6567 6961 5f63  data/Aquilegia_c
+00015690: 6f65 7275 6c65 615f 476c 6f62 616c 6d72  oerulea_Globalmr
+000156a0: 6268 4b73 5f45 6c6d 6d5f 4170 676d 6d5f  bhKs_Elmm_Apgmm_
+000156b0: 436f 7272 6563 7465 642e 6b73 642e 7376  Corrected.ksd.sv
+000156c0: 6729 0a0a 4173 2073 686f 776e 2061 626f  g)..As shown abo
+000156d0: 7665 2c20 6265 6361 7573 6520 6f66 2074  ve, because of t
+000156e0: 6865 2068 6967 6865 7220 7375 6273 7469  he higher substi
+000156f0: 7475 7469 6f6e 2072 6174 6520 6f66 202a  tution rate of *
+00015700: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
+00015710: 6561 2a2c 2074 6865 206f 7269 6769 6e61  ea*, the origina
+00015720: 6c20 6f72 7468 6f6c 6f67 6f75 7320 2a4b  l orthologous *K
+00015730: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
+00015740: 6c75 6573 2077 6572 6520 6163 7475 616c  lues were actual
+00015750: 6c79 2075 6e64 6572 6573 7469 6d61 7465  ly underestimate
+00015760: 6420 696e 2074 6865 2074 696d 652d 6672  d in the time-fr
+00015770: 616d 6520 6f66 202a 4171 7569 6c65 6769  ame of *Aquilegi
+00015780: 6120 636f 6572 756c 6561 2a2e 2057 6865  a coerulea*. Whe
+00015790: 6e20 7765 2072 6563 6f76 6572 6564 2074  n we recovered t
+000157a0: 6865 2064 6976 6572 6765 6e63 6520 7375  he divergence su
+000157b0: 6273 7469 7475 7469 6f6e 2064 6973 7461  bstitution dista
+000157c0: 6e63 6520 696e 2074 6572 6d73 206f 6620  nce in terms of 
+000157d0: 7477 6f20 7469 6d65 7320 6f66 2074 6865  two times of the
+000157e0: 2062 7261 6e63 682d 7370 6563 6966 6963   branch-specific
+000157f0: 2063 6f6e 7472 6962 7574 696f 6e20 6f66   contribution of
+00015800: 202a 412e 2063 6f65 7275 6c65 612a 2073   *A. coerulea* s
+00015810: 696e 6365 2069 7473 2064 6976 6572 6765  ince its diverge
+00015820: 6e63 6520 7769 7468 2074 6865 2073 6973  nce with the sis
+00015830: 7465 7220 7370 6563 6965 7320 706c 7573  ter species plus
+00015840: 2074 6865 2073 6861 7265 6420 7375 6273   the shared subs
+00015850: 7469 7475 7469 6f6e 2064 6973 7461 6e63  titution distanc
+00015860: 6520 6265 666f 7265 2064 6976 6572 6765  e before diverge
+00015870: 6e63 6520 2869 6e20 7265 6c61 7469 7665  nce (in relative
+00015880: 2074 6f20 7468 6520 6f75 7467 726f 7570   to the outgroup
+00015890: 292c 2074 6865 2063 6f72 7265 6374 6564  ), the corrected
+000158a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000158b0: 206d 6f64 6520 6265 6361 6d65 206c 6172   mode became lar
+000158c0: 6765 722e 0a0a 4e6f 7465 2074 6861 7420  ger...Note that 
+000158d0: 7765 2063 616e 2065 6173 696c 7920 7368  we can easily sh
+000158e0: 6f77 2074 6861 7420 2a41 7175 696c 6567  ow that *Aquileg
+000158f0: 6961 2063 6f65 7275 6c65 612a 2068 6173  ia coerulea* has
+00015900: 2068 6967 6865 7220 7375 6273 7469 7475   higher substitu
+00015910: 7469 6f6e 2072 6174 6520 7468 616e 202a  tion rate than *
+00015920: 5072 6f74 6561 2063 796e 6172 6f69 6465  Protea cynaroide
+00015930: 732a 2061 6e64 202a 5669 7469 7320 7669  s* and *Vitis vi
+00015940: 6e69 6665 7261 2a20 6279 2063 6f6d 7061  nifera* by compa
+00015950: 7269 6e67 2074 6865 6972 2073 7562 7374  ring their subst
+00015960: 6974 7574 696f 6e20 6469 7374 616e 6365  itution distance
+00015970: 2069 6e20 7265 6761 7264 2074 6f20 7468   in regard to th
+00015980: 6520 7361 6d65 2064 6976 6572 6765 6e63  e same divergenc
+00015990: 6520 6576 656e 7420 7769 7468 206f 7574  e event with out
+000159a0: 6772 6f75 7020 7370 6563 6965 7320 2a41  group species *A
+000159b0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
+000159c0: 2a2c 2075 7369 6e67 2063 6f6d 6d61 6e64  *, using command
+000159d0: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
+000159e0: 2076 697a 202d 6420 7767 645f 676c 6f62   viz -d wgd_glob
+000159f0: 616c 6d72 6268 5f6b 732f 676c 6f62 616c  almrbh_ks/global
+00015a00: 5f4d 5242 482e 7473 762e 6b73 2e74 7376  _MRBH.tsv.ks.tsv
+00015a10: 202d 7370 2073 7065 6369 6573 7472 6565   -sp speciestree
+00015a20: 2e6e 7720 2d2d 7265 7765 6967 6874 202d  .nw --reweight -
+00015a30: 6f20 7767 645f 7669 7a5f 436f 6d70 6172  o wgd_viz_Compar
+00015a40: 655f 7261 7465 202d 2d73 7061 6972 2022  e_rate --spair "
+00015a50: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
+00015a60: 733b 5072 6f74 6561 5f63 796e 6172 6f69  s;Protea_cynaroi
+00015a70: 6465 7322 202d 2d73 7061 6972 2022 4171  des" --spair "Aq
+00015a80: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00015a90: 3b41 636f 7275 735f 616d 6572 6963 616e  ;Acorus_american
+00015aa0: 7573 2220 2d2d 7370 6169 7220 2256 6974  us" --spair "Vit
+00015ab0: 6973 5f76 696e 6966 6572 613b 4163 6f72  is_vinifera;Acor
+00015ac0: 7573 5f61 6d65 7269 6361 6e75 7322 202d  us_americanus" -
+00015ad0: 2d70 6c6f 746b 6465 202d 2d63 6c61 7373  -plotkde --class
+00015ae0: 6963 0a60 6060 0a0a 215b 5d28 6461 7461  ic.```..![](data
+00015af0: 2f52 6177 5f4f 7274 686f 6c6f 6775 6573  /Raw_Orthologues
+00015b00: 5f43 6f6d 7061 7265 5f72 6174 652e 6b73  _Compare_rate.ks
+00015b10: 642e 7376 6729 0a0a 4173 2064 6973 706c  d.svg)..As displ
+00015b20: 6179 6564 2061 626f 7665 2c20 7468 6520  ayed above, the 
+00015b30: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
+00015b40: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
+00015b50: 6573 2062 6577 7465 656e 202a 4171 7569  es bewteen *Aqui
+00015b60: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
+00015b70: 616e 6420 2a41 636f 7275 7320 616d 6572  and *Acorus amer
+00015b80: 6963 616e 7573 2a20 6861 7320 7468 6520  icanus* has the 
+00015b90: 6869 6768 6573 7420 6d6f 6465 2c20 696e  highest mode, in
+00015ba0: 6469 6361 7469 6e67 2074 6865 2066 6173  dicating the fas
+00015bb0: 7465 7220 7375 6273 7469 7475 7469 6f6e  ter substitution
+00015bc0: 2072 6174 6520 6f66 202a 412e 2063 6f65   rate of *A. coe
+00015bd0: 7275 6c65 612a 2063 6f6d 7061 7265 6420  rulea* compared 
+00015be0: 746f 202a 5072 6f74 6561 2063 796e 6172  to *Protea cynar
+00015bf0: 6f69 6465 732a 2061 6e64 202a 5669 7469  oides* and *Viti
+00015c00: 7320 7669 6e69 6665 7261 2a2e 0a0a 4265  s vinifera*...Be
+00015c10: 666f 7265 2076 322e 302e 3231 2c20 7468  fore v2.0.21, th
+00015c20: 6520 6765 6e65 2d73 7065 6369 6573 206d  e gene-species m
+00015c30: 6170 2066 696c 6520 6973 206e 6563 6365  ap file is necce
+00015c40: 7373 6172 696c 7920 6e65 6564 6564 2066  ssarily needed f
+00015c50: 6f72 2069 7473 2069 6d70 6c65 6d65 6e74  or its implement
+00015c60: 6174 696f 6e20 696e 2060 7767 6420 7669  ation in `wgd vi
+00015c70: 7a60 2c20 7768 6963 6820 7368 6f75 6c64  z`, which should
+00015c80: 2062 6520 6175 746f 6d61 7465 6c79 2070   be automately p
+00015c90: 726f 6475 6365 6420 6279 2074 6865 206c  roduced by the l
+00015ca0: 6173 7420 6077 6764 206b 7364 6020 7374  ast `wgd ksd` st
+00015cb0: 6570 2067 6976 656e 2074 6865 2060 7370  ep given the `sp
+00015cc0: 6169 7260 2061 6e64 2060 7370 6563 6965  air` and `specie
+00015cd0: 7374 7265 6560 2070 6172 616d 6574 6572  stree` parameter
+00015ce0: 732e 2054 6865 2060 6765 6e65 5f73 7065  s. The `gene_spe
+00015cf0: 6369 6573 2e6d 6170 6020 6861 7320 636f  cies.map` has co
+00015d00: 6e74 656e 7473 2061 7320 6265 6c6f 7720  ntents as below 
+00015d10: 696e 2077 6869 6368 2065 6163 6820 6c69  in which each li
+00015d20: 6e65 2069 7320 7468 6520 6a6f 696e 6564  ne is the joined
+00015d30: 2073 7472 696e 6720 6f66 2067 656e 6520   string of gene 
+00015d40: 6e61 6d65 2061 6e64 2073 7065 6369 6573  name and species
+00015d50: 206e 616d 6520 6279 2073 7061 6365 2e20   name by space. 
+00015d60: 4166 7465 7220 7632 2e30 2e32 3120 2869  After v2.0.21 (i
+00015d70: 6e63 6c75 6465 6429 2c20 7468 6520 6765  ncluded), the ge
+00015d80: 6e65 2d73 7065 6369 6573 206d 6170 2066  ne-species map f
+00015d90: 696c 6520 6973 206e 6f74 206e 6563 6365  ile is not necce
+00015da0: 7373 6172 696c 7920 6e65 6564 6564 2061  ssarily needed a
+00015db0: 6e79 6d6f 7265 2e0a 0a60 6060 0a41 7163  nymore...```.Aqc
+00015dc0: 6f65 3647 3035 3738 3030 2e31 2041 7175  oe6G057800.1 Aqu
+00015dd0: 696c 6567 6961 5f63 6f65 7275 6c65 610a  ilegia_coerulea.
+00015de0: 5676 695f 5649 545f 3230 3173 3030 3131  Vvi_VIT_201s0011
+00015df0: 6730 3135 3330 2e31 2056 6974 6973 5f76  g01530.1 Vitis_v
+00015e00: 696e 6966 6572 610a 5063 795f 5072 6f63  inifera.Pcy_Proc
+00015e10: 7930 3167 3038 3531 3020 5072 6f74 6561  y01g08510 Protea
+00015e20: 5f63 796e 6172 6f69 6465 730a 4161 6d5f  _cynaroides.Aam_
+00015e30: 4163 6f72 612e 3034 4731 3432 3930 302e  Acora.04G142900.
+00015e40: 3120 4163 6f72 7573 5f61 6d65 7269 6361  1 Acorus_america
+00015e50: 6e75 730a 6060 600a 0a41 6e20 616c 7465  nus.```..An alte
+00015e60: 726e 6174 6976 6520 7761 7920 746f 2063  rnative way to c
+00015e70: 616c 6375 6c61 7465 2074 6865 206f 7274  alculate the ort
+00015e80: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
+00015e90: 3e53 3c2f 7375 623e 2069 7320 746f 2064  >S</sub> is to d
+00015ea0: 6972 6563 746c 7920 7573 6520 7468 6520  irectly use the 
+00015eb0: 6f72 7468 6f67 726f 7570 7320 696e 7374  orthogroups inst
+00015ec0: 6561 6420 6f66 2067 6c6f 6261 6c20 4d52  ead of global MR
+00015ed0: 4248 2066 616d 696c 792e 2054 6861 7420  BH family. That 
+00015ee0: 7761 7920 7765 2064 6f6e 2774 2075 7365  way we don't use
+00015ef0: 2074 6865 2070 7265 2d69 6e66 6572 7265   the pre-inferre
+00015f00: 6420 7061 7261 6e6f 6d65 202a 4b2a 3c73  d paranome *K*<s
+00015f10: 7562 3e53 3c2f 7375 623e 2062 7574 2074  ub>S</sub> but t
+00015f20: 6865 2070 6172 616c 6f67 6f75 7320 6765  he paralogous ge
+00015f30: 6e65 2070 6169 7273 2069 6e73 6964 6520  ne pairs inside 
+00015f40: 6561 6368 206f 7274 686f 6772 6f75 7020  each orthogroup 
+00015f50: 696e 7374 6561 642e 2054 6f20 6163 6869  instead. To achi
+00015f60: 6576 6520 7468 6174 2c20 7765 2066 6972  eve that, we fir
+00015f70: 7374 206e 6565 6420 746f 2069 6e66 6572  st need to infer
+00015f80: 206f 7274 686f 6772 6f75 7073 2075 7369   orthogroups usi
+00015f90: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
+00015fa0: 656c 6f77 2e0a 0a60 6060 0a77 6764 2064  elow...```.wgd d
+00015fb0: 6d64 2041 7175 696c 6567 6961 5f63 6f65  md Aquilegia_coe
+00015fc0: 7275 6c65 6120 5072 6f74 6561 5f63 796e  rulea Protea_cyn
+00015fd0: 6172 6f69 6465 7320 4163 6f72 7573 5f61  aroides Acorus_a
+00015fe0: 6d65 7269 6361 6e75 7320 5669 7469 735f  mericanus Vitis_
+00015ff0: 7669 6e69 6665 7261 202d 2d6f 7274 686f  vinifera --ortho
+00016000: 696e 6665 7220 2d6f 2077 6764 5f6f 7274  infer -o wgd_ort
+00016010: 686f 2028 2d2d 6f6e 6c79 6f72 7468 6f29  ho (--onlyortho)
+00016020: 200a 6060 600a 0a55 7365 7273 2063 616e   .```..Users can
+00016030: 2064 6563 6964 6520 746f 206f 6e6c 7920   decide to only 
+00016040: 636f 6e64 7563 7420 7468 6520 6f72 7468  conduct the orth
+00016050: 6f67 726f 7570 2061 6e61 6c79 7369 7320  ogroup analysis 
+00016060: 7768 696c 6520 736b 6970 7069 6e67 206f  while skipping o
+00016070: 7468 6572 2061 6e61 6c79 7369 7320 6279  ther analysis by
+00016080: 2061 6464 696e 6720 7468 6520 666c 6167   adding the flag
+00016090: 2060 2d2d 6f6e 6c79 6f72 7468 6f60 2e20   `--onlyortho`. 
+000160a0: 4e65 7874 2073 7465 7020 6973 2074 6865  Next step is the
+000160b0: 2073 616d 6520 7769 7468 2067 6c6f 6261   same with globa
+000160c0: 6c20 4d52 4248 2066 616d 696c 792e 0a0a  l MRBH family...
+000160d0: 6060 600a 7767 6420 6b73 6420 7767 645f  ```.wgd ksd wgd_
+000160e0: 6f72 7468 6f2f 4f72 7468 6f67 726f 7570  ortho/Orthogroup
+000160f0: 732e 7370 2e74 7376 2041 7175 696c 6567  s.sp.tsv Aquileg
+00016100: 6961 5f63 6f65 7275 6c65 6120 5072 6f74  ia_coerulea Prot
+00016110: 6561 5f63 796e 6172 6f69 6465 7320 4163  ea_cynaroides Ac
+00016120: 6f72 7573 5f61 6d65 7269 6361 6e75 7320  orus_americanus 
+00016130: 5669 7469 735f 7669 6e69 6665 7261 202d  Vitis_vinifera -
+00016140: 6f20 7767 645f 6f72 7468 6f5f 6b73 0a77  o wgd_ortho_ks.w
+00016150: 6764 2076 697a 202d 6420 7767 645f 6f72  gd viz -d wgd_or
+00016160: 7468 6f5f 6b73 2f4f 7274 686f 6772 6f75  tho_ks/Orthogrou
+00016170: 7073 2e73 702e 7473 762e 6b73 2e74 7376  ps.sp.tsv.ks.tsv
+00016180: 202d 6570 6b20 7767 645f 6b73 642f 4171   -epk wgd_ksd/Aq
+00016190: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000161a0: 2e6b 732e 7473 7620 2d73 7020 7370 6563  .ks.tsv -sp spec
+000161b0: 6965 7374 7265 652e 6e77 202d 2d72 6577  iestree.nw --rew
+000161c0: 6569 6768 7420 2d61 7020 7767 645f 7379  eight -ap wgd_sy
+000161d0: 6e2f 6961 6468 6f72 652d 6f75 742f 616e  n/iadhore-out/an
+000161e0: 6368 6f72 706f 696e 7473 2e74 7874 202d  chorpoints.txt -
+000161f0: 706c 6f74 656c 6d6d 202d 2d70 6c6f 7461  plotelmm --plota
+00016200: 7067 6d6d 202d 6f20 7767 645f 6f72 7468  pgmm -o wgd_orth
+00016210: 6f5f 6b73 5f72 6174 655f 636f 7272 6563  o_ks_rate_correc
+00016220: 7469 6f6e 0a60 6060 0a0a 215b 5d28 6461  tion.```..![](da
+00016230: 7461 2f4d 6978 6564 2e6b 732e 4171 7569  ta/Mixed.ks.Aqui
+00016240: 6c65 6769 615f 636f 6572 756c 6561 2e6e  legia_coerulea.n
+00016250: 6f64 652e 7765 6967 6874 6564 5f4f 7274  ode.weighted_Ort
+00016260: 686f 5f4b 732e 706e 6729 0a21 5b5d 2864  ho_Ks.png).![](d
+00016270: 6174 612f 466f 6375 735f 7369 7374 6572  ata/Focus_sister
+00016280: 5f70 6169 7273 2e6b 732e 6e6f 6465 2e77  _pairs.ks.node.w
+00016290: 6569 6768 7465 645f 4f72 7468 6f5f 4b73  eighted_Ortho_Ks
+000162a0: 2e70 6e67 290a 215b 5d28 6461 7461 2f41  .png).![](data/A
+000162b0: 6c6c 5f70 6169 7273 2e6b 732e 6e6f 6465  ll_pairs.ks.node
+000162c0: 2e77 6569 6768 7465 645f 4f72 7468 6f5f  .weighted_Ortho_
+000162d0: 4b73 2e70 6e67 290a 0a41 7320 7368 6f77  Ks.png)..As show
+000162e0: 6e20 6162 6f76 652c 2074 6865 206e 756d  n above, the num
+000162f0: 6265 7220 6f66 206f 7274 686f 6c6f 676f  ber of orthologo
+00016300: 7573 2067 656e 6520 7061 6972 7320 6973  us gene pairs is
+00016310: 2064 6966 6665 7265 6e74 2074 6861 6e20   different than 
+00016320: 7468 6520 6f6e 6520 6672 6f6d 2067 6c6f  the one from glo
+00016330: 6261 6c20 4d52 4248 2066 616d 696c 6965  bal MRBH familie
+00016340: 7320 696e 2074 6861 7420 6865 7265 2077  s in that here w
+00016350: 6520 706c 6f74 7465 6420 616c 6c20 6f72  e plotted all or
+00016360: 7468 6f6c 6f67 6f75 7320 6765 6e65 2070  thologous gene p
+00016370: 6169 7273 2069 6e73 7465 6164 206f 6620  airs instead of 
+00016380: 6f6e 6c79 2067 6c6f 6261 6c20 4d52 4248  only global MRBH
+00016390: 2066 616d 696c 6965 732c 2074 6f67 6574   families, toget
+000163a0: 6865 7220 7769 7468 2064 6966 6665 7265  her with differe
+000163b0: 6e74 2072 6563 616c 6375 6c61 7465 6420  nt recalculated 
+000163c0: 7765 6967 6874 732e 0a0a 4166 7465 7220  weights...After 
+000163d0: 7468 6520 7068 796c 6f67 656e 6574 6963  the phylogenetic
+000163e0: 2074 696d 696e 6720 6f66 2074 6865 2052   timing of the R
+000163f0: 616e 756e 6375 6c61 6c65 7320 5747 442c  anunculales WGD,
+00016400: 2077 6520 6361 6e20 6675 7274 6865 7220   we can further 
+00016410: 696e 6665 7220 6974 7320 6162 736f 6c75  infer its absolu
+00016420: 7465 2061 6765 2e20 4669 7273 7420 7765  te age. First we
+00016430: 2069 6e66 6572 2074 6865 2063 7265 6469   infer the credi
+00016440: 626c 6520 7261 6e67 6520 6f66 2061 6e63  ble range of anc
+00016450: 686f 7220 7061 6972 7320 6279 202a 4b2a  hor pairs by *K*
+00016460: 3c73 7562 3e53 3c2f 7375 623e 2068 6575  <sub>S</sub> heu
+00016470: 7269 7374 6963 616c 6c79 2075 7369 6e67  ristically using
+00016480: 2074 6865 2070 726f 6772 616d 2060 7767   the program `wg
+00016490: 6420 7065 616b 602e 0a0a 6060 600a 7767  d peak`...```.wg
+000164a0: 6420 7065 616b 202d 2d68 6575 7269 7374  d peak --heurist
+000164b0: 6963 2077 6764 5f6b 7364 2f41 7175 696c  ic wgd_ksd/Aquil
+000164c0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+000164d0: 762e 6b73 2e74 7376 202d 6170 2077 6764  v.ks.tsv -ap wgd
+000164e0: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
+000164f0: 2f61 6e63 686f 7270 6f69 6e74 732e 7478  /anchorpoints.tx
+00016500: 7420 2d73 6d20 7767 645f 7379 6e2f 6961  t -sm wgd_syn/ia
+00016510: 6468 6f72 652d 6f75 742f 7365 676d 656e  dhore-out/segmen
+00016520: 7473 2e74 7874 202d 6c65 2077 6764 5f73  ts.txt -le wgd_s
+00016530: 796e 2f69 6164 686f 7265 2d6f 7574 2f6c  yn/iadhore-out/l
+00016540: 6973 745f 656c 656d 656e 7473 2e74 7874  ist_elements.txt
+00016550: 202d 6d70 2077 6764 5f73 796e 2f69 6164   -mp wgd_syn/iad
+00016560: 686f 7265 2d6f 7574 2f6d 756c 7469 706c  hore-out/multipl
+00016570: 6963 6f6e 5f70 6169 7273 2e74 7874 202d  icon_pairs.txt -
+00016580: 6f20 7767 645f 7065 616b 0a60 6060 0a0a  o wgd_peak.```..
+00016590: 215b 5d28 6461 7461 2f41 6e63 686f 724b  ![](data/AnchorK
+000165a0: 735f 5065 616b 4349 5f41 7175 696c 6567  s_PeakCI_Aquileg
+000165b0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+000165c0: 6b73 2e74 7376 5f6e 6f64 655f 7765 6967  ks.tsv_node_weig
+000165d0: 6874 6564 2e73 7667 290a 0a41 7320 7368  hted.svg)..As sh
+000165e0: 6f77 6e20 6162 6f76 652c 2077 6520 6173  own above, we as
+000165f0: 7375 6d65 6420 6120 6c6f 676e 6f72 6d61  sumed a lognorma
+00016600: 6c20 6469 7374 7269 6275 7469 6f6e 2061  l distribution a
+00016610: 7420 7468 6520 7065 616b 206c 6f63 6174  t the peak locat
+00016620: 696f 6e20 6465 7465 6374 6564 2062 7920  ion detected by 
+00016630: 7468 6520 6073 6967 6e61 6c60 206d 6f64  the `signal` mod
+00016640: 756c 6520 6f66 2060 7363 6970 7960 206c  ule of `scipy` l
+00016650: 6962 7261 7279 2e20 5468 6520 3935 2520  ibrary. The 95% 
+00016660: 636f 6e66 6964 656e 6365 206c 6576 656c  confidence level
+00016670: 206f 6620 7468 6520 6c6f 676e 6f72 6d61   of the lognorma
+00016680: 6c20 6469 7374 7269 6275 7469 6f6e 2077  l distribution w
+00016690: 6173 2061 7070 6c69 6564 2c20 692e 652e  as applied, i.e.
+000166a0: 2c20 302e 3638 2d32 2e37 342c 2069 6e20  , 0.68-2.74, in 
+000166b0: 6675 7274 6865 7220 6d6f 6c65 6375 6c61  further molecula
+000166c0: 7220 6461 7469 6e67 2e20 5468 6520 6669  r dating. The fi
+000166d0: 6c65 2060 4171 7569 6c65 6769 615f 636f  le `Aquilegia_co
+000166e0: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
+000166f0: 765f 3935 2543 495f 4150 5f66 6f72 5f64  v_95%CI_AP_for_d
+00016700: 6174 696e 675f 7765 6967 6874 6564 5f66  ating_weighted_f
+00016710: 6f72 6d61 742e 7473 7660 2069 7320 7768  ormat.tsv` is wh
+00016720: 6174 2077 6520 6e65 6564 2066 6f72 206e  at we need for n
+00016730: 6578 7420 7374 6570 2e20 546f 2062 7569  ext step. To bui
+00016740: 6c64 2074 6865 206f 7274 686f 6772 6f75  ld the orthogrou
+00016750: 7073 2075 7365 6420 696e 2070 6879 6c6f  ps used in phylo
+00016760: 6765 6e65 7469 6320 6461 7469 6e67 2c20  genetic dating, 
+00016770: 7765 206e 6565 6420 746f 2073 656c 6563  we need to selec
+00016780: 7420 736f 6d65 2073 7065 6369 6573 2061  t some species a
+00016790: 6e64 2066 6f72 6d20 6120 7374 6172 7469  nd form a starti
+000167a0: 6e67 2074 7265 6520 7769 7468 2070 726f  ng tree with pro
+000167b0: 7065 7220 666f 7373 696c 2063 616c 6962  per fossil calib
+000167c0: 7261 7469 6f6e 732e 2057 6520 7072 6f76  rations. We prov
+000167d0: 6964 6520 6f6e 6520 696e 206d 636d 6374  ide one in mcmct
+000167e0: 7265 6520 666f 726d 6174 2061 7320 6265  ree format as be
+000167f0: 6c6f 772e 0a0a 6060 600a 3137 2031 0a28  low...```.17 1.(
+00016800: 2828 2850 6f74 616d 6f67 6574 6f6e 5f61  (((Potamogeton_a
+00016810: 6375 7469 666f 6c69 7573 2c28 5370 6972  cutifolius,(Spir
+00016820: 6f64 656c 615f 696e 7465 726d 6564 6961  odela_intermedia
+00016830: 2c41 6d6f 7270 686f 7068 616c 6c75 735f  ,Amorphophallus_
+00016840: 6b6f 6e6a 6163 2929 2c28 4163 616e 7468  konjac)),(Acanth
+00016850: 6f63 686c 616d 7973 5f62 7261 6374 6561  ochlamys_bractea
+00016860: 7461 2c28 4469 6f73 636f 7265 615f 616c  ta,(Dioscorea_al
+00016870: 6174 612c 4469 6f73 636f 7265 615f 726f  ata,Dioscorea_ro
+00016880: 7475 6e64 6174 6129 2927 3e30 2e35 3630  tundata))'>0.560
+00016890: 303c 312e 3238 3633 2729 273e 302e 3833  0<1.2863')'>0.83
+000168a0: 3630 3c31 2e32 3836 3327 2c28 4163 6f72  60<1.2863',(Acor
+000168b0: 7573 5f61 6d65 7269 6361 6e75 732c 4163  us_americanus,Ac
+000168c0: 6f72 7573 5f74 6174 6172 696e 6f77 6969  orus_tatarinowii
+000168d0: 2929 273e 302e 3833 3630 3c31 2e32 3836  ))'>0.8360<1.286
+000168e0: 3327 2c28 2828 2854 6574 7261 6365 6e74  3',((((Tetracent
+000168f0: 726f 6e5f 7369 6e65 6e73 652c 5472 6f63  ron_sinense,Troc
+00016900: 686f 6465 6e64 726f 6e5f 6172 616c 696f  hodendron_aralio
+00016910: 6964 6573 292c 2842 7578 7573 5f61 7573  ides),(Buxus_aus
+00016920: 7472 6f79 756e 6e61 6e65 6e73 6973 2c42  troyunnanensis,B
+00016930: 7578 7573 5f73 696e 6963 6129 2927 3e31  uxus_sinica))'>1
+00016940: 2e31 3038 303c 312e 3238 3633 272c 284e  .1080<1.2863',(N
+00016950: 656c 756d 626f 5f6e 7563 6966 6572 612c  elumbo_nucifera,
+00016960: 2854 656c 6f70 6561 5f73 7065 6369 6f73  (Telopea_specios
+00016970: 6973 7369 6d61 2c50 726f 7465 615f 6379  issima,Protea_cy
+00016980: 6e61 726f 6964 6573 2929 2927 3e31 2e31  naroides)))'>1.1
+00016990: 3038 303c 312e 3238 3633 272c 2841 7175  080<1.2863',(Aqu
+000169a0: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
+000169b0: 6170 312c 4171 7569 6c65 6769 615f 636f  ap1,Aquilegia_co
+000169c0: 6572 756c 6561 5f61 7032 2929 273e 312e  erulea_ap2))'>1.
+000169d0: 3130 3830 3c31 2e32 3836 3327 2927 3e31  1080<1.2863')'>1
+000169e0: 2e32 3732 303c 322e 3437 3230 273b 0a60  .2720<2.4720';.`
+000169f0: 6060 0a0a 4173 2070 7265 7365 6e74 6564  ``..As presented
+00016a00: 2061 626f 7665 2c20 7468 6520 666f 6375   above, the focu
+00016a10: 7320 7370 6563 6965 7320 7468 6174 2069  s species that i
+00016a20: 7320 6162 6f75 7420 746f 2062 6520 6461  s about to be da
+00016a30: 7465 6420 6e65 6564 7320 746f 2062 6520  ted needs to be 
+00016a40: 7265 706c 6163 6564 2077 6974 6820 6028  replaced with `(
+00016a50: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00016a60: 6561 5f61 7031 2c41 7175 696c 6567 6961  ea_ap1,Aquilegia
+00016a70: 5f63 6f65 7275 6c65 615f 6170 3229 602e  _coerulea_ap2)`.
+00016a80: 2057 6974 6820 7468 6973 2073 7461 7274   With this start
+00016a90: 696e 6720 7472 6565 2061 6e64 2070 7265  ing tree and pre
+00016aa0: 646f 776e 6c6f 6164 6564 2063 6473 2066  downloaded cds f
+00016ab0: 696c 6573 206f 6620 616c 6c20 7468 6520  iles of all the 
+00016ac0: 7370 6563 6965 732c 2077 6520 6361 6e20  species, we can 
+00016ad0: 6275 696c 6420 7468 6520 6f72 7468 6f67  build the orthog
+00016ae0: 726f 7570 2075 7365 6420 696e 2074 6865  roup used in the
+00016af0: 2066 696e 616c 206d 6f6c 6563 756c 6172   final molecular
+00016b00: 2064 6174 696e 6720 7573 696e 6720 7468   dating using th
+00016b10: 6520 636f 6d6d 616e 6420 6173 2062 656c  e command as bel
+00016b20: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
+00016b30: 202d 6620 4171 7569 6c65 6769 615f 636f   -f Aquilegia_co
+00016b40: 6572 756c 6561 202d 6170 2077 6764 5f70  erulea -ap wgd_p
+00016b50: 6561 6b2f 4171 7569 6c65 6769 615f 636f  eak/Aquilegia_co
+00016b60: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
+00016b70: 765f 3935 2543 495f 4150 5f66 6f72 5f64  v_95%CI_AP_for_d
+00016b80: 6174 696e 675f 7765 6967 6874 6564 5f66  ating_weighted_f
+00016b90: 6f72 6d61 742e 7473 7620 2d6f 2077 6764  ormat.tsv -o wgd
+00016ba0: 5f64 6d64 5f6f 7274 686f 2050 6f74 616d  _dmd_ortho Potam
+00016bb0: 6f67 6574 6f6e 5f61 6375 7469 666f 6c69  ogeton_acutifoli
+00016bc0: 7573 2053 7069 726f 6465 6c61 5f69 6e74  us Spirodela_int
+00016bd0: 6572 6d65 6469 6120 416d 6f72 7068 6f70  ermedia Amorphop
+00016be0: 6861 6c6c 7573 5f6b 6f6e 6a61 6320 4163  hallus_konjac Ac
+00016bf0: 616e 7468 6f63 686c 616d 7973 5f62 7261  anthochlamys_bra
+00016c00: 6374 6561 7461 2044 696f 7363 6f72 6561  cteata Dioscorea
+00016c10: 5f61 6c61 7461 2044 696f 7363 6f72 6561  _alata Dioscorea
+00016c20: 5f72 6f74 756e 6461 7461 2041 636f 7275  _rotundata Acoru
+00016c30: 735f 616d 6572 6963 616e 7573 2041 636f  s_americanus Aco
+00016c40: 7275 735f 7461 7461 7269 6e6f 7769 6920  rus_tatarinowii 
+00016c50: 5465 7472 6163 656e 7472 6f6e 5f73 696e  Tetracentron_sin
+00016c60: 656e 7365 2054 726f 6368 6f64 656e 6472  ense Trochodendr
+00016c70: 6f6e 5f61 7261 6c69 6f69 6465 7320 4275  on_aralioides Bu
+00016c80: 7875 735f 6175 7374 726f 7975 6e6e 616e  xus_austroyunnan
+00016c90: 656e 7369 7320 4275 7875 735f 7369 6e69  ensis Buxus_sini
+00016ca0: 6361 204e 656c 756d 626f 5f6e 7563 6966  ca Nelumbo_nucif
+00016cb0: 6572 6120 5465 6c6f 7065 615f 7370 6563  era Telopea_spec
+00016cc0: 696f 7369 7373 696d 6120 5072 6f74 6561  iosissima Protea
+00016cd0: 5f63 796e 6172 6f69 6465 7320 4171 7569  _cynaroides Aqui
+00016ce0: 6c65 6769 615f 636f 6572 756c 6561 0a60  legia_coerulea.`
+00016cf0: 6060 0a0a 5468 6520 7265 7375 6c74 2066  ``..The result f
+00016d00: 696c 6520 606d 6572 6765 5f66 6f63 7573  ile `merge_focus
+00016d10: 5f61 702e 7473 7660 2069 7320 7768 6174  _ap.tsv` is what
+00016d20: 2077 6520 6e65 6564 2066 6f72 2074 6865   we need for the
+00016d30: 2066 696e 616c 2073 7465 7020 6f66 206d   final step of m
+00016d40: 6f6c 6563 756c 6172 2064 6174 696e 6720  olecular dating 
+00016d50: 696e 2070 726f 6772 616d 2060 7767 6420  in program `wgd 
+00016d60: 666f 6375 7360 2e0a 0a60 6060 0a77 6764  focus`...```.wgd
+00016d70: 2066 6f63 7573 202d 2d70 726f 7464 6174   focus --protdat
+00016d80: 696e 6720 2d2d 6161 6d6f 6465 6c20 6c67  ing --aamodel lg
+00016d90: 2077 6764 5f64 6d64 5f6f 7274 686f 2f6d   wgd_dmd_ortho/m
+00016da0: 6572 6765 5f66 6f63 7573 5f61 702e 7473  erge_focus_ap.ts
+00016db0: 7620 2d73 7020 6461 7469 6e67 5f74 7265  v -sp dating_tre
+00016dc0: 652e 6e77 202d 6f20 7767 645f 6461 7469  e.nw -o wgd_dati
+00016dd0: 6e67 202d 6420 6d63 6d63 7472 6565 202d  ng -d mcmctree -
+00016de0: 6473 2027 6275 726e 696e 203d 2032 3030  ds 'burnin = 200
+00016df0: 3027 202d 6473 2027 7361 6d70 6672 6571  0' -ds 'sampfreq
+00016e00: 203d 2031 3030 3027 202d 6473 2027 6e73   = 1000' -ds 'ns
+00016e10: 616d 706c 6520 3d20 3230 3030 3027 2050  ample = 20000' P
+00016e20: 6f74 616d 6f67 6574 6f6e 5f61 6375 7469  otamogeton_acuti
+00016e30: 666f 6c69 7573 2053 7069 726f 6465 6c61  folius Spirodela
+00016e40: 5f69 6e74 6572 6d65 6469 6120 416d 6f72  _intermedia Amor
+00016e50: 7068 6f70 6861 6c6c 7573 5f6b 6f6e 6a61  phophallus_konja
+00016e60: 6320 4163 616e 7468 6f63 686c 616d 7973  c Acanthochlamys
+00016e70: 5f62 7261 6374 6561 7461 2044 696f 7363  _bracteata Diosc
+00016e80: 6f72 6561 5f61 6c61 7461 2044 696f 7363  orea_alata Diosc
+00016e90: 6f72 6561 5f72 6f74 756e 6461 7461 2041  orea_rotundata A
+00016ea0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
+00016eb0: 2041 636f 7275 735f 7461 7461 7269 6e6f   Acorus_tatarino
+00016ec0: 7769 6920 5465 7472 6163 656e 7472 6f6e  wii Tetracentron
+00016ed0: 5f73 696e 656e 7365 2054 726f 6368 6f64  _sinense Trochod
+00016ee0: 656e 6472 6f6e 5f61 7261 6c69 6f69 6465  endron_aralioide
+00016ef0: 7320 4275 7875 735f 6175 7374 726f 7975  s Buxus_austroyu
+00016f00: 6e6e 616e 656e 7369 7320 4275 7875 735f  nnanensis Buxus_
+00016f10: 7369 6e69 6361 204e 656c 756d 626f 5f6e  sinica Nelumbo_n
+00016f20: 7563 6966 6572 6120 5465 6c6f 7065 615f  ucifera Telopea_
+00016f30: 7370 6563 696f 7369 7373 696d 6120 5072  speciosissima Pr
+00016f40: 6f74 6561 5f63 796e 6172 6f69 6465 7320  otea_cynaroides 
+00016f50: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00016f60: 6561 0a60 6060 0a0a 4865 7265 2077 6520  ea.```..Here we 
+00016f70: 6f6e 6c79 2069 6d70 6c65 6d65 6e74 6564  only implemented
+00016f80: 2074 6865 2063 6f6e 6361 7465 6e61 7469   the concatenati
+00016f90: 6f6e 2061 6e61 6c79 7369 7320 7573 696e  on analysis usin
+00016fa0: 6720 7072 6f74 6569 6e20 7365 7175 656e  g protein sequen
+00016fb0: 6365 2062 7920 6164 6469 6e67 2074 6865  ce by adding the
+00016fc0: 2066 6c61 6720 602d 2d70 726f 7464 6174   flag `--protdat
+00016fd0: 696e 6760 2061 6e64 2077 6520 7365 7420  ing` and we set 
+00016fe0: 7468 6520 7061 7261 6d65 7465 7220 666f  the parameter fo
+00016ff0: 7220 606d 636d 6374 7265 6560 2076 6961  r `mcmctree` via
+00017000: 2074 6865 206f 7074 696f 6e20 602d 6473   the option `-ds
+00017010: 602e 204e 6f74 6520 7468 6174 206f 7468  `. Note that oth
+00017020: 6572 2064 6174 696e 6720 7072 6f67 7261  er dating progra
+00017030: 6d20 7375 6368 2061 7320 6072 3873 6020  m such as `r8s` 
+00017040: 616e 6420 6062 6561 7374 6020 6172 6520  and `beast` are 
+00017050: 616c 736f 2061 7661 696c 6162 6c65 2067  also available g
+00017060: 6976 656e 2073 6f6d 6520 6d61 6e64 6174  iven some mandat
+00017070: 6f72 7920 7061 7261 6d65 7465 7273 2e20  ory parameters. 
+00017080: 5468 6520 6669 6e61 6c20 6c6f 6720 6f66  The final log of
+00017090: 2074 6865 2073 7563 6365 7373 6675 6c20   the successful 
+000170a0: 7275 6e20 6973 2061 7320 6265 6c6f 772e  run is as below.
+000170b0: 0a0a 6060 600a 3136 3a30 343a 3235 2049  ..```.16:04:25 I
+000170c0: 4e46 4f20 2020 2020 5275 6e6e 696e 6720  NFO     Running 
+000170d0: 6d63 6d63 7472 6565 2075 7369 6e67 2048  mcmctree using H
+000170e0: 6573 7369 616e 206d 6174 7269 7820 6f66  essian matrix of
+000170f0: 204c 472b 4761 6d6d 6120 2063 6f72 652e   LG+Gamma  core.
+00017100: 7079 3a39 3637 0a20 2020 2020 2020 2020  py:967.         
+00017110: 2020 2020 2020 2020 2066 6f72 2070 726f           for pro
+00017120: 7465 696e 206d 6f64 656c 0a32 333a 3439  tein model.23:49
+00017130: 3a33 3720 494e 464f 2020 2020 2050 6f73  :37 INFO     Pos
+00017140: 7465 7269 6f72 206d 6561 6e20 666f 7220  terior mean for 
+00017150: 7468 6520 6167 6573 206f 6620 7767 6420  the ages of wgd 
+00017160: 6973 2031 3132 2e38 3934 3520 6d63 6d63  is 112.8945 mcmc
+00017170: 7472 6565 2e70 793a 3239 360a 2020 2020  tree.py:296.    
+00017180: 2020 2020 2020 2020 2020 2020 2020 6d69                mi
+00017190: 6c6c 696f 6e20 7965 6172 7320 6672 6f6d  llion years from
+000171a0: 2043 6f6e 6361 7465 6e61 7465 6420 7065   Concatenated pe
+000171b0: 7074 6964 650a 2020 2020 2020 2020 2020  ptide.          
+000171c0: 2020 2020 2020 2020 616c 6967 6e6d 656e          alignmen
+000171d0: 7420 616e 6420 3935 2520 6372 6564 6962  t and 95% credib
+000171e0: 696c 6974 7920 696e 7465 7276 616c 7320  ility intervals 
+000171f0: 2843 4929 0a20 2020 2020 2020 2020 2020  (CI).           
+00017200: 2020 2020 2020 2069 7320 3130 312e 3232         is 101.22
+00017210: 342d 3132 332e 3132 3120 6d69 6c6c 696f  4-123.121 millio
+00017220: 6e20 7965 6172 730a 2020 2020 2020 2020  n years.        
+00017230: 2049 4e46 4f20 2020 2020 546f 7461 6c20   INFO     Total 
+00017240: 7275 6e20 7469 6d65 3a20 3239 3137 3573  run time: 29175s
 00017250: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00017260: 2020 2020 2020 636c 692e 7079 3a32 3432        cli.py:242
-00017270: 0a60 6060 0a0a 546f 2076 6973 7561 6c69  .```..To visuali
-00017280: 7a65 2074 6865 2064 6174 652c 2077 6520  ze the date, we 
-00017290: 616c 736f 2070 726f 7669 6465 6420 6120  also provided a 
-000172a0: 7079 7468 6f6e 2073 6372 6970 7420 746f  python script to
-000172b0: 2070 6c6f 7420 7468 6520 5747 4420 6461   plot the WGD da
-000172c0: 7465 7320 696e 2074 6865 2060 7767 6460  tes in the `wgd`
-000172d0: 2066 6f6c 6465 722e 2055 7365 7273 206e   folder. Users n
-000172e0: 6565 6420 746f 2065 7874 7261 6374 2074  eed to extract t
-000172f0: 6865 2072 6177 2064 6174 6573 2066 726f  he raw dates fro
-00017300: 6d20 7468 6520 606d 636d 632e 7478 7460  m the `mcmc.txt`
-00017310: 2066 6f72 2074 6865 2057 4744 206e 6f64   for the WGD nod
-00017320: 6520 6669 7273 7420 616e 6420 7361 7665  e first and save
-00017330: 2069 7420 6173 2066 696c 6520 6064 6174   it as file `dat
-00017340: 6573 2e74 7874 6020 286f 7220 7768 6174  es.txt` (or what
-00017350: 6576 6572 2070 7265 6665 7272 6564 206e  ever preferred n
-00017360: 616d 6529 2e20 416e 2065 7861 6d70 6c65  ame). An example
-00017370: 2063 6f6d 6d61 6e64 2069 7320 6173 2062   command is as b
-00017380: 656c 6f77 2e0a 0a60 6060 0a70 7974 686f  elow...```.pytho
-00017390: 6e20 2450 4154 482f 706f 7374 706c 6f74  n $PATH/postplot
-000173a0: 2e70 7920 706f 7374 6469 7320 6461 7465  .py postdis date
-000173b0: 732e 7478 7420 2d2d 7065 7263 656e 7469  s.txt --percenti
-000173c0: 6c65 2039 3020 2d2d 7469 746c 6520 2257  le 90 --title "W
-000173d0: 4744 2064 6174 6522 202d 2d68 7064 202d  GD date" --hpd -
-000173e0: 6f20 2252 616e 756e 6375 6c61 6c65 735f  o "Ranunculales_
-000173f0: 5747 445f 6461 7465 2e73 7667 220a 6060  WGD_date.svg".``
-00017400: 600a 0a21 5b5d 2864 6174 612f 5261 6e75  `..![](data/Ranu
-00017410: 6e63 756c 616c 6573 5f57 4744 5f64 6174  nculales_WGD_dat
-00017420: 652e 7376 6729 0a0a 5468 6520 706f 7374  e.svg)..The post
-00017430: 6572 696f 7220 6d65 616e 2c20 6d65 6469  erior mean, medi
-00017440: 616e 2061 6e64 206d 6f64 6520 6f66 2074  an and mode of t
-00017450: 6865 2052 616e 756e 6375 6c61 6c65 7320  he Ranunculales 
-00017460: 5747 4420 6167 6520 6973 2031 3132 2e39  WGD age is 112.9
-00017470: 322c 2031 3133 2e34 3420 616e 6420 3131  2, 113.44 and 11
-00017480: 322e 3534 206d 7961 2c20 7769 7468 2039  2.54 mya, with 9
-00017490: 3025 2048 5044 2031 3035 2e30 3720 2d20  0% HPD 105.07 - 
-000174a0: 3132 322e 3332 206d 7961 2061 7320 6d61  122.32 mya as ma
-000174b0: 6e69 6665 7374 6564 2061 626f 7665 2e0a  nifested above..
-000174c0: 0a23 2323 204b 7374 7265 650a 0a49 6e20  .### Kstree..In 
-000174d0: 6164 6469 7469 6f6e 2074 6f20 7061 6972  addition to pair
-000174e0: 7769 7365 202a 4b2a 3c73 7562 3e53 3c2f  wise *K*<sub>S</
-000174f0: 7375 623e 2065 7374 696d 6174 696f 6e2c  sub> estimation,
-00017500: 2061 202a 4b2a 3c73 7562 3e53 3c2f 7375   a *K*<sub>S</su
-00017510: 623e 2074 7265 6520 7769 7468 2062 7261  b> tree with bra
-00017520: 6e63 6820 6c65 6e67 7468 2069 6e20 2a4b  nch length in *K
-00017530: 2a3c 7375 623e 533c 2f73 7562 3e20 756e  *<sub>S</sub> un
-00017540: 6974 2063 616e 2061 6c73 6f20 6265 2064  it can also be d
-00017550: 6572 6976 6564 2066 726f 6d20 7468 6520  erived from the 
-00017560: 7072 6f67 7261 6d20 6077 6764 206b 7364  program `wgd ksd
-00017570: 6020 6769 7665 6e20 7468 6520 6f70 7469  ` given the opti
-00017580: 6f6e 2060 2d2d 6b73 7472 6565 6020 616e  on `--kstree` an
-00017590: 6420 602d 2d73 7065 6369 6573 7472 6565  d `--speciestree
-000175a0: 602e 204e 6f74 6520 7468 6174 2074 6865  `. Note that the
-000175b0: 2061 6464 6974 696f 6e61 6c20 6f70 7469   additional opti
-000175c0: 6f6e 2060 2d2d 6f6e 6c79 636f 6e63 6174  on `--onlyconcat
-000175d0: 6b73 7472 6565 6020 7769 6c6c 206f 6e6c  kstree` will onl
-000175e0: 7920 6361 6c6c 2074 6865 202a 4b2a 3c73  y call the *K*<s
-000175f0: 7562 3e53 3c2f 7375 623e 2065 7374 696d  ub>S</sub> estim
-00017600: 6174 696f 6e20 666f 7220 7468 6520 636f  ation for the co
-00017610: 6e63 6174 656e 6174 6564 2061 6c69 676e  ncatenated align
-00017620: 6d65 6e74 2072 6174 6865 7220 7468 616e  ment rather than
-00017630: 2061 6c6c 2074 6865 2061 6c69 676e 6d65   all the alignme
-00017640: 6e74 732e 2055 7365 7273 206e 6565 6420  nts. Users need 
-00017650: 746f 2070 726f 7669 6465 2061 2070 7265  to provide a pre
-00017660: 7365 7420 7370 6563 6965 7320 7472 6565  set species tree
-00017670: 2066 6f72 2074 6865 202a 4b2a 3c73 7562   for the *K*<sub
-00017680: 3e53 3c2f 7375 623e 2074 7265 6520 696e  >S</sub> tree in
-00017690: 6665 7265 6e63 6520 6f66 2074 6865 2063  ference of the c
-000176a0: 6f6e 6361 7465 6e61 7465 6420 616c 6967  oncatenated alig
-000176b0: 6e6d 656e 7420 7768 696c 6520 7468 6520  nment while the 
-000176c0: 7265 6d61 696e 696e 6720 616c 6967 6e6d  remaining alignm
-000176d0: 656e 7473 2077 696c 6c20 6265 2061 6761  ents will be aga
-000176e0: 696e 7374 2061 6e20 6175 746f 6d61 7465  inst an automate
-000176f0: 6c79 2069 6e66 6572 7265 6420 7472 6565  ly inferred tree
-00017700: 2066 726f 6d20 6066 6173 7474 7265 6560   from `fasttree`
-00017710: 206f 7220 6069 7174 7265 6560 2e20 496e   or `iqtree`. In
-00017720: 2074 6865 2065 6e64 2c20 7573 6572 7320   the end, users 
-00017730: 7769 6c6c 2067 6574 2061 202a 4b2a 3c73  will get a *K*<s
-00017740: 7562 3e53 3c2f 7375 623e 2074 7265 652c  ub>S</sub> tree,
-00017750: 2061 202a 4b2a 3c73 7562 3e41 3c2f 7375   a *K*<sub>A</su
-00017760: 623e 2074 7265 6520 616e 6420 6120 cf89  b> tree and a ..
-00017770: 2074 7265 6520 7065 7220 6661 6d20 616e   tree per fam an
-00017780: 6420 666f 7220 7468 6520 636f 6e63 6174  d for the concat
-00017790: 656e 6174 6564 2061 6c69 676e 6d65 6e74  enated alignment
-000177a0: 2e0a 0a60 6060 0a77 6764 206b 7364 2064  ...```.wgd ksd d
-000177b0: 6174 612f 6b73 7472 6565 5f64 6174 612f  ata/kstree_data/
-000177c0: 6661 6d2e 7473 7620 6461 7461 2f6b 7374  fam.tsv data/kst
-000177d0: 7265 655f 6461 7461 2f41 636f 7275 735f  ree_data/Acorus_
-000177e0: 7461 7461 7269 6e6f 7769 6920 6461 7461  tatarinowii data
-000177f0: 2f6b 7374 7265 655f 6461 7461 2f41 6d62  /kstree_data/Amb
-00017800: 6f72 656c 6c61 5f74 7269 6368 6f70 6f64  orella_trichopod
-00017810: 6120 6461 7461 2f6b 7374 7265 655f 6461  a data/kstree_da
-00017820: 7461 2f41 7175 696c 6567 6961 5f63 6f65  ta/Aquilegia_coe
-00017830: 7275 6c65 6120 6461 7461 2f6b 7374 7265  rulea data/kstre
-00017840: 655f 6461 7461 2f41 7269 7374 6f6c 6f63  e_data/Aristoloc
-00017850: 6869 615f 6669 6d62 7269 6174 6120 6461  hia_fimbriata da
-00017860: 7461 2f6b 7374 7265 655f 6461 7461 2f43  ta/kstree_data/C
-00017870: 7963 6173 5f70 616e 7a68 6968 7561 656e  ycas_panzhihuaen
-00017880: 7369 202d 2d6b 7374 7265 6520 2d2d 7370  si --kstree --sp
-00017890: 6563 6965 7374 7265 6520 6461 7461 2f6b  eciestree data/k
-000178a0: 7374 7265 655f 6461 7461 2f73 7065 6369  stree_data/speci
-000178b0: 6573 5f74 7265 6531 2e6e 7720 2d2d 6f6e  es_tree1.nw --on
-000178c0: 6c79 636f 6e63 6174 6b73 7472 6565 202d  lyconcatkstree -
-000178d0: 6f20 7767 645f 6b73 7472 6565 5f74 6f70  o wgd_kstree_top
-000178e0: 6f6c 6f67 7931 0a60 6060 0a0a 215b 5d28  ology1.```..![](
-000178f0: 6461 7461 2f6b 7374 7265 655f 7265 7375  data/kstree_resu
-00017900: 6c74 732f 6b73 7472 6565 2e73 7667 290a  lts/kstree.svg).
-00017910: 0a41 626f 7665 2077 6520 7573 6564 2074  .Above we used t
-00017920: 6872 6565 2061 6c74 6572 6e61 7469 7665  hree alternative
-00017930: 2074 6f70 6f6c 6f67 6965 7320 746f 2069   topologies to i
-00017940: 6e66 6572 2074 6865 202a 4b2a 3c73 7562  nfer the *K*<sub
-00017950: 3e53 3c2f 7375 623e 2074 7265 6520 7768  >S</sub> tree wh
-00017960: 6963 6820 6c65 6420 746f 2064 6966 6665  ich led to diffe
-00017970: 7265 6e74 2062 7261 6e63 6820 6c65 6e67  rent branch leng
-00017980: 7468 2065 7374 696d 6174 696f 6e2e 204e  th estimation. N
-00017990: 6f74 6520 7468 6174 2074 6865 2066 616d  ote that the fam
-000179a0: 696c 6965 7320 7765 2075 7365 6420 7765  ilies we used we
-000179b0: 7265 206f 6e6c 7920 7477 6f20 676c 6f62  re only two glob
-000179c0: 616c 204d 5242 4820 6661 6d69 6c69 6573  al MRBH families
-000179d0: 2066 6f72 2074 6865 2070 7572 706f 7365   for the purpose
-000179e0: 206f 6620 696c 6c75 7374 7261 7469 6f6e   of illustration
-000179f0: 2e20 546f 2061 6371 7569 7265 2061 6e20  . To acquire an 
-00017a00: 6163 6375 7261 7465 2070 726f 6669 6c65  accurate profile
-00017a10: 206f 6620 7468 6520 7375 6273 7469 7475   of the substitu
-00017a20: 7469 6f6e 2072 6174 6520 7661 7269 6174  tion rate variat
-00017a30: 696f 6e2c 206f 7274 686f 6c6f 6775 6573  ion, orthologues
-00017a40: 2061 7420 7468 6520 7768 6f6c 6520 6765   at the whole ge
-00017a50: 6e6f 6d65 2073 6361 6c65 2073 686f 756c  nome scale shoul
-00017a60: 6420 6265 2075 7365 642e 0a0a 2323 2043  d be used...## C
-00017a70: 6974 6174 696f 6e0a 200a 506c 6561 7365  itation. .Please
-00017a80: 2063 6974 6520 7573 2061 7420 6874 7470   cite us at http
-00017a90: 733a 2f2f 646f 692e 6f72 672f 3130 2e31  s://doi.org/10.1
-00017aa0: 3030 372f 3937 382d 312d 3037 3136 2d32  007/978-1-0716-2
-00017ab0: 3536 312d 335f 3120 616e 6420 6874 7470  561-3_1 and http
-00017ac0: 733a 2f2f 646f 692e 6f72 672f 3130 2e31  s://doi.org/10.1
-00017ad0: 3039 332f 6269 6f69 6e66 6f72 6d61 7469  093/bioinformati
-00017ae0: 6373 2f62 7461 6532 3732 2e0a 0a60 6060  cs/btae272...```
-00017af0: 0a48 656e 6763 6869 2043 6865 6e2c 2041  .Hengchi Chen, A
-00017b00: 7274 6875 7220 5a77 6165 6e65 706f 656c  rthur Zwaenepoel
-00017b10: 2028 3230 3233 292e 2049 6e66 6572 656e   (2023). Inferen
-00017b20: 6365 206f 6620 416e 6369 656e 7420 506f  ce of Ancient Po
-00017b30: 6c79 706c 6f69 6479 2066 726f 6d20 4765  lyploidy from Ge
-00017b40: 6e6f 6d69 6320 4461 7461 2e20 496e 3a20  nomic Data. In: 
-00017b50: 5661 6e20 6465 2050 6565 722c 2059 2e20  Van de Peer, Y. 
-00017b60: 2865 6473 2920 506f 6c79 706c 6f69 6479  (eds) Polyploidy
-00017b70: 2e20 4d65 7468 6f64 7320 696e 204d 6f6c  . Methods in Mol
-00017b80: 6563 756c 6172 2042 696f 6c6f 6779 2c20  ecular Biology, 
-00017b90: 766f 6c20 3235 3435 2e20 4875 6d61 6e61  vol 2545. Humana
-00017ba0: 2c20 4e65 7720 596f 726b 2c20 4e59 2e20  , New York, NY. 
-00017bb0: 6874 7470 733a 2f2f 646f 692e 6f72 672f  https://doi.org/
-00017bc0: 3130 2e31 3030 372f 3937 382d 312d 3037  10.1007/978-1-07
-00017bd0: 3136 2d32 3536 312d 335f 310a 4865 6e67  16-2561-3_1.Heng
-00017be0: 6368 6920 4368 656e 2c20 4172 7468 7572  chi Chen, Arthur
-00017bf0: 205a 7761 656e 6570 6f65 6c2c 2059 7665   Zwaenepoel, Yve
-00017c00: 7320 5661 6e20 6465 2050 6565 7220 2832  s Van de Peer (2
-00017c10: 3032 3429 2e20 7767 6420 7632 3a20 6120  024). wgd v2: a 
-00017c20: 7375 6974 6520 6f66 2074 6f6f 6c73 2074  suite of tools t
-00017c30: 6f20 756e 636f 7665 7220 616e 6420 6461  o uncover and da
-00017c40: 7465 2061 6e63 6965 6e74 2070 6f6c 7970  te ancient polyp
-00017c50: 6c6f 6964 7920 616e 6420 7768 6f6c 652d  loidy and whole-
-00017c60: 6765 6e6f 6d65 2064 7570 6c69 6361 7469  genome duplicati
-00017c70: 6f6e 2e20 4269 6f69 6e66 6f72 6d61 7469  on. Bioinformati
-00017c80: 6373 2c20 6874 7470 733a 2f2f 646f 692e  cs, https://doi.
-00017c90: 6f72 672f 3130 2e31 3039 332f 6269 6f69  org/10.1093/bioi
-00017ca0: 6e66 6f72 6d61 7469 6373 2f62 7461 6532  nformatics/btae2
-00017cb0: 3732 0a60 6060 0a0a 466f 7220 6369 7461  72.```..For cita
-00017cc0: 7469 6f6e 206f 6620 7468 6520 746f 6f6c  tion of the tool
-00017cd0: 7320 7573 6564 2069 6e20 7767 642c 2070  s used in wgd, p
-00017ce0: 6c65 6173 6520 636f 6e73 756c 7420 7468  lease consult th
-00017cf0: 6520 646f 6375 6d65 6e74 6174 696f 6e20  e documentation 
-00017d00: 6174 0a68 7474 7073 3a2f 2f77 6764 7632  at.https://wgdv2
-00017d10: 2e72 6561 6474 6865 646f 6373 2e69 6f2f  .readthedocs.io/
-00017d20: 656e 2f6c 6174 6573 742f 6369 7461 7469  en/latest/citati
-00017d30: 6f6e 2e68 746d 6c2e 0a0a                 on.html...
+00017260: 2020 2020 2020 2020 2020 2020 2020 636c                cl
+00017270: 692e 7079 3a32 3431 0a20 2020 2020 2020  i.py:241.       
+00017280: 2020 494e 464f 2020 2020 2044 6f6e 6520    INFO     Done 
+00017290: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000172a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000172b0: 2020 2020 2020 2020 2020 2020 2020 2063                 c
+000172c0: 6c69 2e70 793a 3234 320a 6060 600a 0a54  li.py:242.```..T
+000172d0: 6f20 7669 7375 616c 697a 6520 7468 6520  o visualize the 
+000172e0: 6461 7465 2c20 7765 2061 6c73 6f20 7072  date, we also pr
+000172f0: 6f76 6964 6564 2061 2070 7974 686f 6e20  ovided a python 
+00017300: 7363 7269 7074 2074 6f20 706c 6f74 2074  script to plot t
+00017310: 6865 2057 4744 2064 6174 6573 2069 6e20  he WGD dates in 
+00017320: 7468 6520 6077 6764 6020 666f 6c64 6572  the `wgd` folder
+00017330: 2e20 5573 6572 7320 6e65 6564 2074 6f20  . Users need to 
+00017340: 6578 7472 6163 7420 7468 6520 7261 7720  extract the raw 
+00017350: 6461 7465 7320 6672 6f6d 2074 6865 2060  dates from the `
+00017360: 6d63 6d63 2e74 7874 6020 666f 7220 7468  mcmc.txt` for th
+00017370: 6520 5747 4420 6e6f 6465 2066 6972 7374  e WGD node first
+00017380: 2061 6e64 2073 6176 6520 6974 2061 7320   and save it as 
+00017390: 6669 6c65 2060 6461 7465 732e 7478 7460  file `dates.txt`
+000173a0: 2028 6f72 2077 6861 7465 7665 7220 7072   (or whatever pr
+000173b0: 6566 6572 7265 6420 6e61 6d65 292e 2041  eferred name). A
+000173c0: 6e20 6578 616d 706c 6520 636f 6d6d 616e  n example comman
+000173d0: 6420 6973 2061 7320 6265 6c6f 772e 0a0a  d is as below...
+000173e0: 6060 600a 7079 7468 6f6e 2024 5041 5448  ```.python $PATH
+000173f0: 2f70 6f73 7470 6c6f 742e 7079 2070 6f73  /postplot.py pos
+00017400: 7464 6973 2064 6174 6573 2e74 7874 202d  tdis dates.txt -
+00017410: 2d70 6572 6365 6e74 696c 6520 3930 202d  -percentile 90 -
+00017420: 2d74 6974 6c65 2022 5747 4420 6461 7465  -title "WGD date
+00017430: 2220 2d2d 6870 6420 2d6f 2022 5261 6e75  " --hpd -o "Ranu
+00017440: 6e63 756c 616c 6573 5f57 4744 5f64 6174  nculales_WGD_dat
+00017450: 652e 7376 6722 0a60 6060 0a0a 215b 5d28  e.svg".```..![](
+00017460: 6461 7461 2f52 616e 756e 6375 6c61 6c65  data/Ranunculale
+00017470: 735f 5747 445f 6461 7465 2e73 7667 290a  s_WGD_date.svg).
+00017480: 0a54 6865 2070 6f73 7465 7269 6f72 206d  .The posterior m
+00017490: 6561 6e2c 206d 6564 6961 6e20 616e 6420  ean, median and 
+000174a0: 6d6f 6465 206f 6620 7468 6520 5261 6e75  mode of the Ranu
+000174b0: 6e63 756c 616c 6573 2057 4744 2061 6765  nculales WGD age
+000174c0: 2069 7320 3131 322e 3932 2c20 3131 332e   is 112.92, 113.
+000174d0: 3434 2061 6e64 2031 3132 2e35 3420 6d79  44 and 112.54 my
+000174e0: 612c 2077 6974 6820 3930 2520 4850 4420  a, with 90% HPD 
+000174f0: 3130 352e 3037 202d 2031 3232 2e33 3220  105.07 - 122.32 
+00017500: 6d79 6120 6173 206d 616e 6966 6573 7465  mya as manifeste
+00017510: 6420 6162 6f76 652e 0a0a 2323 2320 4b73  d above...### Ks
+00017520: 7472 6565 0a0a 496e 2061 6464 6974 696f  tree..In additio
+00017530: 6e20 746f 2070 6169 7277 6973 6520 2a4b  n to pairwise *K
+00017540: 2a3c 7375 623e 533c 2f73 7562 3e20 6573  *<sub>S</sub> es
+00017550: 7469 6d61 7469 6f6e 2c20 6120 2a4b 2a3c  timation, a *K*<
+00017560: 7375 623e 533c 2f73 7562 3e20 7472 6565  sub>S</sub> tree
+00017570: 2077 6974 6820 6272 616e 6368 206c 656e   with branch len
+00017580: 6774 6820 696e 202a 4b2a 3c73 7562 3e53  gth in *K*<sub>S
+00017590: 3c2f 7375 623e 2075 6e69 7420 6361 6e20  </sub> unit can 
+000175a0: 616c 736f 2062 6520 6465 7269 7665 6420  also be derived 
+000175b0: 6672 6f6d 2074 6865 2070 726f 6772 616d  from the program
+000175c0: 2060 7767 6420 6b73 6460 2067 6976 656e   `wgd ksd` given
+000175d0: 2074 6865 206f 7074 696f 6e20 602d 2d6b   the option `--k
+000175e0: 7374 7265 6560 2061 6e64 2060 2d2d 7370  stree` and `--sp
+000175f0: 6563 6965 7374 7265 6560 2e20 4e6f 7465  eciestree`. Note
+00017600: 2074 6861 7420 7468 6520 6164 6469 7469   that the additi
+00017610: 6f6e 616c 206f 7074 696f 6e20 602d 2d6f  onal option `--o
+00017620: 6e6c 7963 6f6e 6361 746b 7374 7265 6560  nlyconcatkstree`
+00017630: 2077 696c 6c20 6f6e 6c79 2063 616c 6c20   will only call 
+00017640: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
+00017650: 7562 3e20 6573 7469 6d61 7469 6f6e 2066  ub> estimation f
+00017660: 6f72 2074 6865 2063 6f6e 6361 7465 6e61  or the concatena
+00017670: 7465 6420 616c 6967 6e6d 656e 7420 7261  ted alignment ra
+00017680: 7468 6572 2074 6861 6e20 616c 6c20 7468  ther than all th
+00017690: 6520 616c 6967 6e6d 656e 7473 2e20 5573  e alignments. Us
+000176a0: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
+000176b0: 6964 6520 6120 7072 6573 6574 2073 7065  ide a preset spe
+000176c0: 6369 6573 2074 7265 6520 666f 7220 7468  cies tree for th
+000176d0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+000176e0: 3e20 7472 6565 2069 6e66 6572 656e 6365  > tree inference
+000176f0: 206f 6620 7468 6520 636f 6e63 6174 656e   of the concaten
+00017700: 6174 6564 2061 6c69 676e 6d65 6e74 2077  ated alignment w
+00017710: 6869 6c65 2074 6865 2072 656d 6169 6e69  hile the remaini
+00017720: 6e67 2061 6c69 676e 6d65 6e74 7320 7769  ng alignments wi
+00017730: 6c6c 2062 6520 6167 6169 6e73 7420 616e  ll be against an
+00017740: 2061 7574 6f6d 6174 656c 7920 696e 6665   automately infe
+00017750: 7272 6564 2074 7265 6520 6672 6f6d 2060  rred tree from `
+00017760: 6661 7374 7472 6565 6020 6f72 2060 6971  fasttree` or `iq
+00017770: 7472 6565 602e 2049 6e20 7468 6520 656e  tree`. In the en
+00017780: 642c 2075 7365 7273 2077 696c 6c20 6765  d, users will ge
+00017790: 7420 6120 2a4b 2a3c 7375 623e 533c 2f73  t a *K*<sub>S</s
+000177a0: 7562 3e20 7472 6565 2c20 6120 2a4b 2a3c  ub> tree, a *K*<
+000177b0: 7375 623e 413c 2f73 7562 3e20 7472 6565  sub>A</sub> tree
+000177c0: 2061 6e64 2061 20cf 8920 7472 6565 2070   and a .. tree p
+000177d0: 6572 2066 616d 2061 6e64 2066 6f72 2074  er fam and for t
+000177e0: 6865 2063 6f6e 6361 7465 6e61 7465 6420  he concatenated 
+000177f0: 616c 6967 6e6d 656e 742e 0a0a 6060 600a  alignment...```.
+00017800: 7767 6420 6b73 6420 6461 7461 2f6b 7374  wgd ksd data/kst
+00017810: 7265 655f 6461 7461 2f66 616d 2e74 7376  ree_data/fam.tsv
+00017820: 2064 6174 612f 6b73 7472 6565 5f64 6174   data/kstree_dat
+00017830: 612f 4163 6f72 7573 5f74 6174 6172 696e  a/Acorus_tatarin
+00017840: 6f77 6969 2064 6174 612f 6b73 7472 6565  owii data/kstree
+00017850: 5f64 6174 612f 416d 626f 7265 6c6c 615f  _data/Amborella_
+00017860: 7472 6963 686f 706f 6461 2064 6174 612f  trichopoda data/
+00017870: 6b73 7472 6565 5f64 6174 612f 4171 7569  kstree_data/Aqui
+00017880: 6c65 6769 615f 636f 6572 756c 6561 2064  legia_coerulea d
+00017890: 6174 612f 6b73 7472 6565 5f64 6174 612f  ata/kstree_data/
+000178a0: 4172 6973 746f 6c6f 6368 6961 5f66 696d  Aristolochia_fim
+000178b0: 6272 6961 7461 2064 6174 612f 6b73 7472  briata data/kstr
+000178c0: 6565 5f64 6174 612f 4379 6361 735f 7061  ee_data/Cycas_pa
+000178d0: 6e7a 6869 6875 6165 6e73 6920 2d2d 6b73  nzhihuaensi --ks
+000178e0: 7472 6565 202d 2d73 7065 6369 6573 7472  tree --speciestr
+000178f0: 6565 2064 6174 612f 6b73 7472 6565 5f64  ee data/kstree_d
+00017900: 6174 612f 7370 6563 6965 735f 7472 6565  ata/species_tree
+00017910: 312e 6e77 202d 2d6f 6e6c 7963 6f6e 6361  1.nw --onlyconca
+00017920: 746b 7374 7265 6520 2d6f 2077 6764 5f6b  tkstree -o wgd_k
+00017930: 7374 7265 655f 746f 706f 6c6f 6779 310a  stree_topology1.
+00017940: 6060 600a 0a21 5b5d 2864 6174 612f 6b73  ```..![](data/ks
+00017950: 7472 6565 5f72 6573 756c 7473 2f6b 7374  tree_results/kst
+00017960: 7265 652e 7376 6729 0a0a 4162 6f76 6520  ree.svg)..Above 
+00017970: 7765 2075 7365 6420 7468 7265 6520 616c  we used three al
+00017980: 7465 726e 6174 6976 6520 746f 706f 6c6f  ternative topolo
+00017990: 6769 6573 2074 6f20 696e 6665 7220 7468  gies to infer th
+000179a0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+000179b0: 3e20 7472 6565 2077 6869 6368 206c 6564  > tree which led
+000179c0: 2074 6f20 6469 6666 6572 656e 7420 6272   to different br
+000179d0: 616e 6368 206c 656e 6774 6820 6573 7469  anch length esti
+000179e0: 6d61 7469 6f6e 2e20 4e6f 7465 2074 6861  mation. Note tha
+000179f0: 7420 7468 6520 6661 6d69 6c69 6573 2077  t the families w
+00017a00: 6520 7573 6564 2077 6572 6520 6f6e 6c79  e used were only
+00017a10: 2074 776f 2067 6c6f 6261 6c20 4d52 4248   two global MRBH
+00017a20: 2066 616d 696c 6965 7320 666f 7220 7468   families for th
+00017a30: 6520 7075 7270 6f73 6520 6f66 2069 6c6c  e purpose of ill
+00017a40: 7573 7472 6174 696f 6e2e 2054 6f20 6163  ustration. To ac
+00017a50: 7175 6972 6520 616e 2061 6363 7572 6174  quire an accurat
+00017a60: 6520 7072 6f66 696c 6520 6f66 2074 6865  e profile of the
+00017a70: 2073 7562 7374 6974 7574 696f 6e20 7261   substitution ra
+00017a80: 7465 2076 6172 6961 7469 6f6e 2c20 6f72  te variation, or
+00017a90: 7468 6f6c 6f67 7565 7320 6174 2074 6865  thologues at the
+00017aa0: 2077 686f 6c65 2067 656e 6f6d 6520 7363   whole genome sc
+00017ab0: 616c 6520 7368 6f75 6c64 2062 6520 7573  ale should be us
+00017ac0: 6564 2e0a 0a23 2320 4369 7461 7469 6f6e  ed...## Citation
+00017ad0: 0a20 0a50 6c65 6173 6520 6369 7465 2075  . .Please cite u
+00017ae0: 7320 6174 2068 7474 7073 3a2f 2f64 6f69  s at https://doi
+00017af0: 2e6f 7267 2f31 302e 3130 3037 2f39 3738  .org/10.1007/978
+00017b00: 2d31 2d30 3731 362d 3235 3631 2d33 5f31  -1-0716-2561-3_1
+00017b10: 2061 6e64 2068 7474 7073 3a2f 2f64 6f69   and https://doi
+00017b20: 2e6f 7267 2f31 302e 3130 3933 2f62 696f  .org/10.1093/bio
+00017b30: 696e 666f 726d 6174 6963 732f 6274 6165  informatics/btae
+00017b40: 3237 322e 0a0a 6060 600a 4865 6e67 6368  272...```.Hengch
+00017b50: 6920 4368 656e 2c20 4172 7468 7572 205a  i Chen, Arthur Z
+00017b60: 7761 656e 6570 6f65 6c20 2832 3032 3329  waenepoel (2023)
+00017b70: 2e20 496e 6665 7265 6e63 6520 6f66 2041  . Inference of A
+00017b80: 6e63 6965 6e74 2050 6f6c 7970 6c6f 6964  ncient Polyploid
+00017b90: 7920 6672 6f6d 2047 656e 6f6d 6963 2044  y from Genomic D
+00017ba0: 6174 612e 2049 6e3a 2056 616e 2064 6520  ata. In: Van de 
+00017bb0: 5065 6572 2c20 592e 2028 6564 7329 2050  Peer, Y. (eds) P
+00017bc0: 6f6c 7970 6c6f 6964 792e 204d 6574 686f  olyploidy. Metho
+00017bd0: 6473 2069 6e20 4d6f 6c65 6375 6c61 7220  ds in Molecular 
+00017be0: 4269 6f6c 6f67 792c 2076 6f6c 2032 3534  Biology, vol 254
+00017bf0: 352e 2048 756d 616e 612c 204e 6577 2059  5. Humana, New Y
+00017c00: 6f72 6b2c 204e 592e 2068 7474 7073 3a2f  ork, NY. https:/
+00017c10: 2f64 6f69 2e6f 7267 2f31 302e 3130 3037  /doi.org/10.1007
+00017c20: 2f39 3738 2d31 2d30 3731 362d 3235 3631  /978-1-0716-2561
+00017c30: 2d33 5f31 0a48 656e 6763 6869 2043 6865  -3_1.Hengchi Che
+00017c40: 6e2c 2041 7274 6875 7220 5a77 6165 6e65  n, Arthur Zwaene
+00017c50: 706f 656c 2c20 5976 6573 2056 616e 2064  poel, Yves Van d
+00017c60: 6520 5065 6572 2028 3230 3234 292e 2077  e Peer (2024). w
+00017c70: 6764 2076 323a 2061 2073 7569 7465 206f  gd v2: a suite o
+00017c80: 6620 746f 6f6c 7320 746f 2075 6e63 6f76  f tools to uncov
+00017c90: 6572 2061 6e64 2064 6174 6520 616e 6369  er and date anci
+00017ca0: 656e 7420 706f 6c79 706c 6f69 6479 2061  ent polyploidy a
+00017cb0: 6e64 2077 686f 6c65 2d67 656e 6f6d 6520  nd whole-genome 
+00017cc0: 6475 706c 6963 6174 696f 6e2e 2042 696f  duplication. Bio
+00017cd0: 696e 666f 726d 6174 6963 732c 2068 7474  informatics, htt
+00017ce0: 7073 3a2f 2f64 6f69 2e6f 7267 2f31 302e  ps://doi.org/10.
+00017cf0: 3130 3933 2f62 696f 696e 666f 726d 6174  1093/bioinformat
+00017d00: 6963 732f 6274 6165 3237 320a 6060 600a  ics/btae272.```.
+00017d10: 0a46 6f72 2063 6974 6174 696f 6e20 6f66  .For citation of
+00017d20: 2074 6865 2074 6f6f 6c73 2075 7365 6420   the tools used 
+00017d30: 696e 2077 6764 2c20 706c 6561 7365 2063  in wgd, please c
+00017d40: 6f6e 7375 6c74 2074 6865 2064 6f63 756d  onsult the docum
+00017d50: 656e 7461 7469 6f6e 2061 740a 6874 7470  entation at.http
+00017d60: 733a 2f2f 7767 6476 322e 7265 6164 7468  s://wgdv2.readth
+00017d70: 6564 6f63 732e 696f 2f65 6e2f 6c61 7465  edocs.io/en/late
+00017d80: 7374 2f63 6974 6174 696f 6e2e 6874 6d6c  st/citation.html
+00017d90: 2e0a 0a                                  ...
```

### Comparing `wgd-2.0.34/cli.py` & `wgd-2.0.35/cli.py`

 * *Files 0% similar despite different names*

```diff
@@ -598,15 +598,15 @@
         df_anchor,orig_anchors = get_anchors('',userdf=anchorpoints)
         df_multi = get_multi('',userdf2=multiplicon)
         ordered_genes_perchrom_allsp, gene_orders = get_chrom_gene(table,outdir)
         #ordered_mp = get_mp_geneorder(gene_orders,'',outdir,table,userdf4=multipliconpairs)
         segs = get_segments_profile(df_multi,keepredun,'',userdf3=segments)
         segs,table,df_multi,removed_scfa = filter_by_minlength(table,segs,minlen,df_multi,keepredun,outdir,minseglen)
         segs_gene_unit, gene_order_dict_allsp = transformunit(segs,ordered_genes_perchrom_allsp,outdir)
-        segs = filter_mingenumber(segs_gene_unit,mingenenum,outdir,len(gene_order_dict_allsp))
+        segs = filter_mingenumber(segs_gene_unit,mingenenum,outdir,len(gene_order_dict_allsp),start)
         dotplotingene(ordered_genes_perchrom_allsp,removed_scfa,outdir,table,gene_orders,anchor=df_anchor,ksdf=df,maxsize=maxsize,dotsize=dotsize, apalpha=apalpha, hoalpha=hoalpha,showrealtick=showrealtick, las=ticklabelsize, gistrb=gistrb)
         if len(ordered_genes_perchrom_allsp)>1: dotplotingeneoverall(ordered_genes_perchrom_allsp,removed_scfa,outdir,table,gene_orders,anchor=df_anchor,ksdf=df,maxsize=maxsize,dotsize=dotsize, apalpha=apalpha, hoalpha=hoalpha,showrealtick=showrealtick, las=ticklabelsize, gistrb=gistrb)
         #dotplotunitgene(ordered_genes_perchrom_allsp,segs_gene_unit,removed_scfa,outdir,mingenenum,table_orig,ordered_mp,ksdf=df)
         figs = all_dotplots(table, segs, df_multi, minseglen, anchors=df_anchor, maxsize=maxsize, minlen=minlen, outdir=outdir, Ks = df, dotsize=dotsize, apalpha=apalpha, hoalpha=hoalpha, showrealtick=showrealtick, las=ticklabelsize, gistrb=gistrb)
         for k, v in figs.items():
             v.savefig(os.path.join(outdir, "{}.dot.svg".format(k)))
             v.savefig(os.path.join(outdir, "{}.dot.pdf".format(k)))
@@ -711,22 +711,23 @@
     logging.info("Processing I-ADHoRe output")
     #ordered_mp = get_mp_geneorder(gene_orders,out_path,outdir,table)
     anchors,orig_anchors = get_anchors(out_path)
     multi = get_multi(out_path)
     if anchors is None:
         logging.warning("No anchors found, terminating! Please inspect your input files "
                 "and the I-ADHoRe results in `{}`".format(out_path))
+        endtime(start)
         exit(1)
     anchors.to_csv(os.path.join(outdir, "anchors.csv"))
     #ap_order_permlt = getmltorder(orig_anchors,multi,gene_orders)
     segs = get_segments_profile(multi,keepredun,out_path)
     #segmentpair_order = get_segmentpair_order(orig_anchors,segs,table,gene_orders)
     segs,table,multi,removed_scfa = filter_by_minlength(table,segs,minlen,multi,keepredun,outdir,minseglen)
     segs_gene_unit, gene_order_dict_allsp = transformunit(segs,ordered_genes_perchrom_allsp,outdir)
-    segs = filter_mingenumber(segs_gene_unit,mingenenum,outdir,len(gene_order_dict_allsp))
+    segs = filter_mingenumber(segs_gene_unit,mingenenum,outdir,len(gene_order_dict_allsp),start)
     #if ks_distribution: segs_gene_unit_ks = getsegks(segs_gene_unit,ks_distribution,ordered_genes_perchrom_allsp)
     df_ks = None
     if ks_distribution!=None:
         ksdb_df = pd.read_csv(ks_distribution,header=0,index_col=0,sep='\t')
         ksdb_df = formatv2(ksdb_df)
         df_ks = apply_filters(ksdb_df, [("dS", 0., 5.)])
     dotplotingene(ordered_genes_perchrom_allsp,removed_scfa,outdir,table,gene_orders,anchor=anchors,ksdf=df_ks,maxsize=maxsize,dotsize=dotsize,apalpha=apalpha, hoalpha=hoalpha, showrealtick=showrealtick, las = ticklabelsize, gistrb=gistrb)
```

### Comparing `wgd-2.0.34/setup.py` & `wgd-2.0.35/setup.py`

 * *Files 6% similar despite different names*

```diff
@@ -5,15 +5,15 @@
 from setuptools import setup
 
 with open("README.md", "r") as fh:
     long_description = fh.read()
 
 setup(
     name='wgd',
-    version='2.0.34',
+    version='2.0.35',
     packages=['wgd'],
     url='http://github.com/heche-psb/wgd',
     license='GPL',
     author='Hengchi Chen',
     author_email='heche@psb.vib-ugent.be',
     description='wgd',
     long_description=long_description,
```

### Comparing `wgd-2.0.34/test/test_core.py` & `wgd-2.0.35/test/test_core.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/wgd/__init__.py` & `wgd-2.0.35/wgd/__init__.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/wgd/beast.py` & `wgd-2.0.35/wgd/beast.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/wgd/cluster.py` & `wgd-2.0.35/wgd/cluster.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/wgd/codeml.py` & `wgd-2.0.35/wgd/codeml.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/wgd/core.py` & `wgd-2.0.35/wgd/core.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/wgd/mcmctree.py` & `wgd-2.0.35/wgd/mcmctree.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/wgd/mix.py` & `wgd-2.0.35/wgd/mix.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/wgd/peak.py` & `wgd-2.0.35/wgd/peak.py`

 * *Files 0% similar despite different names*

```diff
@@ -1889,14 +1889,17 @@
     na=False if weighted else True
     #Components = sorted(list(set(df_c['AnchorKs_GMM_Component'])))
     colors = cm.viridis(np.linspace(0, 1, n))
     for cp,color,scaling in zip(range(n),colors,scalings):
         mean,std = means[cp][0],np.sqrt(stds[cp][0][0])
         df = df_c[df_c['AnchorKs_GMM_Component']==cp]
         lower_CI, upper_CI = find_apeak(df,None,None,outdir,peak_threshold=peak_threshold,na=na,rel_height=rel_height,withfig=False,loglabel="{}-Components Model Peak {}".format(n,cp),keeptmp=keeptmp)
+        if lower_CI is None:
+            logging.warning("No qualified peak detected")
+            continue
         if upper_CI > 5: upper_CI=5
         #ax.axvline(lower_CI, color = color, alpha = 0.8, ls = '--', lw = 0.8,label='Peak of c{} {}%CI {:.2f}-{:.2f}'.format(cp,95,lower_CI,upper_CI))
         #ax.axvline(upper_CI, color = color, alpha = 0.8, ls = '--', lw = 0.8)
         lower_CI_y0 = scaling*stats.lognorm.pdf(lower_CI, scale=np.exp(mean),s=std)
         upper_CI_y1 = scaling*stats.lognorm.pdf(upper_CI, scale=np.exp(mean),s=std)
         ax.plot([lower_CI,lower_CI],[0,lower_CI_y0], color = color, alpha = 0.8, ls = '--', lw = 1,label='HighMass of c{} {}%CI {:.2f}-{:.2f}'.format(cp,95,lower_CI,upper_CI))
         ax.plot([upper_CI,upper_CI],[0,upper_CI_y1], color = color, alpha = 0.8, ls = '--', lw = 1)
```

### Comparing `wgd-2.0.34/wgd/postplot.py` & `wgd-2.0.35/wgd/postplot.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/wgd/ratecorrect.py` & `wgd-2.0.35/wgd/ratecorrect.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/wgd/syn.py` & `wgd-2.0.35/wgd/syn.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/wgd/tree.py` & `wgd-2.0.35/wgd/tree.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/wgd/utils.py` & `wgd-2.0.35/wgd/utils.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.34/wgd/viz.py` & `wgd-2.0.35/wgd/viz.py`

 * *Files 1% similar despite different names*

```diff
@@ -47,25 +47,30 @@
     return df
 
 _labels = {
         "dS" : "$K_\mathrm{S}$",
         "dN" : "$K_\mathrm{A}$",
         "dN/dS": "$\omega$"}
 
-def getspair_ks(spair,df,reweight,onlyrootout,sptree=None,na=False,spgenemap=None,focus2all=None):
+def getspair_ks(spair,df,reweight,onlyrootout,sptree=None,na=False,spgenemap=None,focus2all=None, classic=False):
     df_perspair = {}
     allspair = []
     paralog_pair = []
-    if not (focus2all is None):
+    if not (focus2all is None) and not classic:
         tree = Phylo.read(sptree,'newick')
         spair = [";".join([focus2all,clade.name]) for clade in tree.get_terminals()]
     for i in spair:
         pair = '__'.join(sorted([j.strip() for j in i.split(';')]))
         if i.split(';')[0].strip() == i.split(';')[1].strip(): paralog_pair.append(pair)
         if pair not in allspair: allspair.append(pair)
+    if len(paralog_pair) == 0:
+        if not (focus2all is None):
+            pair = '__'.join([focus2all,focus2all])
+            paralog_pair.append(pair)
+            if pair not in allspair: allspair.append(pair)
     #If users provide various paralogous pair, we still only consider the first one in correction
     if not (spgenemap is None):
         df['spair'] = ['__'.join(sorted([spgenemap[g1],spgenemap[g2]])) for g1,g2 in zip(df['gene1'],df['gene2'])]
     #else:
     #for x in zip(df['g1'],df['']):
     #    if not type(x) is float:
     #        print(x)
@@ -640,15 +645,15 @@
         df_para["family"] = df_para["family"].apply(lambda x:"ExtraParalog_"+x)
         df = pd.concat([df,df_para])
     # I add this dropduplicates to prevent the same paralogous pair from occuring twice and also use preferentially paranome
     df = df[~df.index.duplicated(keep='last')]
     if not ksd and not (gsmap is None): spgenemap = getgsmap(gsmap)
     else: spgenemap = gsmap
     if not viz: writespgenemap(spgenemap,outdir)
-    df_perspair,allspair,paralog_pair,corrected_ks_spair,Outgroup_spnames,Outgroup_spair_ordered = getspair_ks(spair,df,reweight,onlyrootout,sptree=sptree,na=na,spgenemap=spgenemap,focus2all=focus2all)
+    df_perspair,allspair,paralog_pair,corrected_ks_spair,Outgroup_spnames,Outgroup_spair_ordered = getspair_ks(spair,df,reweight,onlyrootout,sptree=sptree,na=na,spgenemap=spgenemap,focus2all=focus2all,classic=clean)
     if len(paralog_pair) == 1:
         if len(df_perspair) == 1:
             if na:
                 fnames = os.path.join(outdir,'{}.ksd.averaged.svg'.format(paralog_pair[0].split('__')[0]))
                 fnamep = os.path.join(outdir,'{}.ksd.averaged.pdf'.format(paralog_pair[0].split('__')[0]))
             else:
                 fnames = os.path.join(outdir,'{}.ksd.weighted.svg'.format(paralog_pair[0].split('__')[0]))
@@ -772,15 +777,15 @@
                     Hs, Bins, patches = ax.hist(y, bins = np.linspace(0, 50, num=51,dtype=int)/10, weights=w, color=cs[i], alpha=okalpha, rwidth=0.8,label=pair)
                 y_lim_beforekde = ax.get_ylim()[1]
                 y_lim_beforekdes.append(y_lim_beforekde)
                 Hs_maxs.append(max(Hs))
                 kde = stats.gaussian_kde(y,weights=w,bw_method=0.1)
                 kde_y = kde(kde_x)
                 mode, maxim = kde_mode(kde_x, kde_y)
-                logging.info('The mode of species pair {} is {:.3f}'.format(pair,mode))
+                logging.info('The mode of species pair {} is {:.2f}'.format(pair,mode))
                 CHF = get_totalH(Hs)
                 scaling = CHF*0.1
                 if plotkde: ax.plot(kde_x, kde_y*scaling, color=cs[i],alpha=0.4, ls = '--')
                 #ax.plot([mode,mode], [0,maxim*scaling], color=cs[i], ls=':', lw=1, label='Original mode {:.2f} of {}'.format(mode,pair))
                 #ax.plot([mode,mode], [0,y_lim_beforekde], color=cs[i], ls=':', lw=1, label='Original mode {:.2f} of {}'.format(mode,pair))
                 if toparrow: ax.axvline(x = mode, ymin=0, ymax=ax.get_ylim()[1], color = cs[i], alpha = 0.8, ls = ':', lw = 1,label = 'Original mode {:.2f} of {}'.format(mode,pair))
                 else: ax.plot([mode,mode], [0,maxim*scaling], color=cs[i], ls=':', lw=1, label='Original mode {:.2f} of {}'.format(mode,pair))
@@ -1142,23 +1147,18 @@
     for i in np.arange(5+0.01, max_ks_tail + 0.01, 0.01):
         midpoint = round((i - 0.01 / 2), 3)
         d = np.append(d,np.repeat(midpoint,round(np.mean(hist_data[0][-51:]))))
     return d
 
 def e_step(num_comp, ks, means, stdevs, weights, lambd):
     products = []
-    #print(ks.shape)
     for k in range(num_comp):
         # at the very beginning, the weight are randomly given as equal
-        #print((k,lambd,np.exp(means[k-1]),stdevs[k-1],weights[k]))
         if k == 0: prod = weights[k] * stats.expon.pdf(ks, scale=1/lambd)
         else:
-            #print(len(ks))
-            #print(type(ks))
-            #print(stats.lognorm.pdf(ks, scale=np.exp(means[k-1]), s=stdevs[k-1]))
             prod = weights[k] * stats.lognorm.pdf(ks, scale=np.exp(means[k-1]), s=stdevs[k-1])
         # The addition of all pdf with weight
         products.append(prod)
     sum_comp_perpoint = sum(products)
     log_sum_comp_perpoint = np.log(sum_comp_perpoint)
     fit_loglikelihood = sum(log_sum_comp_perpoint)
     posteriors = [products[i] / sum_comp_perpoint for i in range(num_comp)]
@@ -1345,17 +1345,22 @@
     if not (user_xlim is None): axs[0,0].set_xlim(user_xlim[0],user_xlim[1])
     sns.despine(offset=1)
     fig.suptitle(title, x=0.125, y=0.9, ha="left", va="top")
     fig.tight_layout()
     plt.subplots_adjust(top=0.85)  # prevent suptitle from overlapping
     return fig
 
-def syntenic_depth_plot(segprofile):
+def syntenic_depth_plot(segprofile,start):
+    from wgd.core import endtime
     cols = segprofile.columns
     n = len(cols)
+    if n == 0:
+        logging.error("No eligible multiplicon discovered in terms of segment length and/or gene number!")
+        endtime(start)
+        exit()
     #fig, axs = plt.subplots(1, int(n + n*(n-1)/2))
     fig, axs = plt.subplots(n, n)
     fig.set_size_inches(n*3.2, n*2.4)
     #if n == 1:
     #    axs = [axs]  # HACK
     #k = 0
     for i in range(n):
@@ -1448,15 +1453,15 @@
         segs_info = seg.groupby(["multiplicon", "genome"])["segment"].aggregate(lambda x: len(set(x)))
         profile = segs_info.unstack(level=-1).fillna(0)
         profile_good = profile.loc[(profile[spx]>0) & (profile[ancestor]>0)]
         if len(profile_good) == 0: logging.info('No multiplicon contained both genome of {0} and {1}'.format(spx,spy))
         else:
             seg_good = seg.merge(profile_good.reset_index(),on='multiplicon')
             gl_tocolor = seg_good.loc[seg_good['genome']==spx,'list']
-            for gl in gl_tocolor: print(gl)
+            #for gl in gl_tocolor: print(gl)
         plot_descendant(spx,df1x.len,df1x.index,outdir,colortable,gl_tocolor,seg_good.loc[:,['genome','list','first','last']])
 
 def plot_descendant(sp,scafflength,scafflabel,outdir,colortable,gl_tocolor,segs_tocolor):
     scafflength_normalized = [i/max(scafflength) for i in scafflength]
     fname = os.path.join(outdir, "{}_descendant_karyotype.png".format(sp))
     fnamep = os.path.join(outdir, "{}_descendant_karyotype.pdf".format(sp))
     fnames = os.path.join(outdir, "{}_descendant_karyotype.svg".format(sp))
@@ -2602,21 +2607,17 @@
                 ax2 = ax.twinx()
                 ax3 = ax.twiny()
             spx, dfx = gdf[ii]
             spy, dfy = gdf[jj]
             logging.info("{} vs. {}".format(spx, spy))
             #get_marco(dfx, dfy, segs, multi, minseglen, **kwargs)
             df, xs, ys, scaffxlabels, scaffylabels, scaffxtick, scaffytick = get_dots(dfx, dfy, segs, multi, minseglen, dupStack = False, **kwargs)
-            #print(xs)
-            #print(ys)
-            #for i,j in zip(df.x,df.y): print((i,j))
             if df is None:  # HACK, in case we're dealing with RBH orthologs...
                 continue
             #for x,y in zip(df.x,df.y): ax.scatter(x, y, s=1, color="k", alpha=0.1)
-            #print((len(list(itertools.chain(df.x))),len(list(itertools.chain(df.y)))))
             ax.scatter(list(itertools.chain(df.x)), list(itertools.chain(df.y)), s=dotsize, color="k", alpha=hoalpha)
             if not (Ks is None):
                 for i,x,y in zip(df.index,df['x'],df['y']):
                     ksage = ks_dict.get(i,None)
                     if ksage == None:
                         continue
                     ksages.append(ksage)
@@ -2734,34 +2735,27 @@
     if not keepredun:
         Mul_to_rm = list(multi[multi['is_redundant']==-1].loc[:,'id'])
         Segs_to_rm = segs.loc[segs['multiplicon'].isin(Mul_to_rm),:]
         for i in Segs_to_rm.index: I3.append(i)
         segs = segs.drop(I3)
         multi = multi[multi['is_redundant']==0]
     segs = Filter_miniseglen(segs,Sf_len_lab_persp,minseglen,genetable)
-    #counted = segs.groupby(["multiplicon", "genome"])["segment"].aggregate(lambda x: len(set(x)))
-    #profile = counted.unstack(level=-1).fillna(0)
-    #if len(gdf) <=5 :
-    #    fig = syntenic_depth_plot(profile)
-    #    fig.savefig(os.path.join(outdir, "Syndepth.svg"),bbox_inches='tight')
-    #    fig.savefig(os.path.join(outdir, "Syndepth.pdf"),bbox_inches='tight')
-    #profile.to_csv(os.path.join(outdir, "Segprofile.csv"))
     return segs,genetable,multi,removed_scfa
 
-def filter_mingenumber(segs,mingenenum,outdir,N):
+def filter_mingenumber(segs,mingenenum,outdir,N,start):
     rm_indice = []
     for indice, f, l in zip(segs.index,segs['first_coordinate'],segs['last_coordinate']):
         if (l-f+1) < mingenenum:
             rm_indice.append(indice)
     segs = segs.drop(rm_indice)
     counted = segs.groupby(["multiplicon", "genome"])["segment"].aggregate(lambda x: len(set(x)))
     profile = counted.unstack(level=-1).fillna(0)
     #if N <=5 :
     logging.info("Making Syndepth plot")
-    fig = syntenic_depth_plot(profile)
+    fig = syntenic_depth_plot(profile,start)
     fig.savefig(os.path.join(outdir, "Syndepth.svg"),bbox_inches='tight')
     fig.savefig(os.path.join(outdir, "Syndepth.pdf"),bbox_inches='tight')
     profile.to_csv(os.path.join(outdir, "Segprofile.csv"))
     return segs
 
 def Filter_miniseglen(segs,scaf_info,minseglen,genetable):
     rm_indice = []
@@ -2834,15 +2828,14 @@
     lens = df.groupby("scaffold")["start"].agg(max)
     lens.name = "len"
     lens = pd.DataFrame(lens).sort_values("len", ascending=False)
     scaffstart = [0] + list(np.cumsum(lens.len))[0:-1]
     scafftick = list(np.cumsum(lens.len))
     lens["scaffstart"] = scaffstart
     df = df.join(lens, on="scaffold").sort_values("len", ascending=False).dropna()
-    #print(set(df['scaffold']))
     # df now contains scaffold lengths
     #if minlen < 0:  # find a reasonable threshold, 5% of longest scaffold?
     #    minlen = df.len.max() * 0.1
     #    logging.info("`minlen` not set, taking 10% of longest scaffold ({})".format(minlen))
     #noriginal = len(df.index)
     #df = df.loc[df.len > minlen]
     #logging.info("Dropped {} genes because they are on scaffolds shorter "
```

### Comparing `wgd-2.0.34/wgd.egg-info/PKG-INFO` & `wgd-2.0.35/wgd.egg-info/PKG-INFO`

 * *Files 2% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 00000000: 4d65 7461 6461 7461 2d56 6572 7369 6f6e  Metadata-Version
 00000010: 3a20 322e 310a 4e61 6d65 3a20 7767 640a  : 2.1.Name: wgd.
-00000020: 5665 7273 696f 6e3a 2032 2e30 2e33 340a  Version: 2.0.34.
+00000020: 5665 7273 696f 6e3a 2032 2e30 2e33 350a  Version: 2.0.35.
 00000030: 5375 6d6d 6172 793a 2077 6764 0a48 6f6d  Summary: wgd.Hom
 00000040: 652d 7061 6765 3a20 6874 7470 3a2f 2f67  e-page: http://g
 00000050: 6974 6875 622e 636f 6d2f 6865 6368 652d  ithub.com/heche-
 00000060: 7073 622f 7767 640a 4175 7468 6f72 3a20  psb/wgd.Author: 
 00000070: 4865 6e67 6368 6920 4368 656e 0a41 7574  Hengchi Chen.Aut
 00000080: 686f 722d 656d 6169 6c3a 2068 6563 6865  hor-email: heche
 00000090: 4070 7362 2e76 6962 2d75 6765 6e74 2e62  @psb.vib-ugent.b
@@ -3809,2307 +3809,2298 @@
 0000ee00: 6365 0a60 6060 0a0a 2a2a 5468 6520 636f  ce.```..**The co
 0000ee10: 6e73 7472 7563 7469 6f6e 206f 6620 6f72  nstruction of or
 0000ee20: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
 0000ee30: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
 0000ee40: 7374 7269 6275 7469 6f6e 2a2a 0a60 6060  stribution**.```
 0000ee50: 0a77 6764 206b 7364 2066 616d 696c 6965  .wgd ksd familie
 0000ee60: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
-0000ee70: 656e 6365 3220 282d 2d72 6577 6569 6768  ence2 (--reweigh
-0000ee80: 7429 0a60 6060 0a0a 4672 6f6d 2070 6172  t).```..From par
-0000ee90: 616c 6f67 6f75 7320 746f 206f 7274 686f  alogous to ortho
-0000eea0: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-0000eeb0: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-0000eec0: 6962 7574 696f 6e2c 2075 7365 7273 206f  ibution, users o
-0000eed0: 6e6c 7920 6e65 6564 2074 6f20 7072 6f76  nly need to prov
-0000eee0: 6964 6520 6d6f 7265 2063 6473 2066 696c  ide more cds fil
-0000eef0: 6573 2e20 4e6f 7465 2074 6861 7420 7769  es. Note that wi
-0000ef00: 7468 206f 7274 686f 6c6f 676f 7573 2067  th orthologous g
-0000ef10: 656e 6520 6661 6d69 6c69 6573 2074 6865  ene families the
-0000ef20: 2077 6569 6768 7469 6e67 206d 6574 686f   weighting metho
-0000ef30: 6420 6361 6e20 6265 2073 6574 2074 6f20  d can be set to 
-0000ef40: 6265 2063 616c 6375 6c61 7465 6420 7065  be calculated pe
-0000ef50: 7220 7370 6563 6965 7320 7061 6972 2069  r species pair i
-0000ef60: 6e73 7465 6164 206f 6620 636f 6e73 6964  nstead of consid
-0000ef70: 6572 696e 6720 7468 6520 7768 6f6c 6520  ering the whole 
-0000ef80: 6661 6d69 6c79 2062 6563 6175 7365 2077  family because w
-0000ef90: 6865 6e20 706c 6f74 7469 6e67 206f 7274  hen plotting ort
-0000efa0: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
-0000efb0: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-0000efc0: 7472 6962 7574 696f 6e20 6265 7477 6565  tribution betwee
-0000efd0: 6e20 7477 6f20 7370 6563 6965 7320 7468  n two species th
-0000efe0: 6520 7765 6967 6874 2063 616c 6375 6c61  e weight calcula
-0000eff0: 7465 6420 6672 6f6d 2074 6869 7320 7370  ted from this sp
-0000f000: 6563 6966 6963 2073 7065 6369 6573 2070  ecific species p
-0000f010: 6169 7220 7368 6f75 6c64 2062 6520 636f  air should be co
-0000f020: 6e73 6572 7665 6420 7768 696c 6520 7468  nserved while th
-0000f030: 6520 6f6e 6520 6361 6c63 756c 6174 6564  e one calculated
-0000f040: 2066 726f 6d20 7468 6520 7768 6f6c 6520   from the whole 
-0000f050: 6661 6d69 6c79 2077 696c 6c20 7661 7279  family will vary
-0000f060: 2077 6974 6820 7468 6520 6e75 6d62 6572   with the number
-0000f070: 206f 6620 7370 6563 6965 732e 2054 6f20   of species. To 
-0000f080: 696e 6974 6961 7465 2074 6865 2077 6569  initiate the wei
-0000f090: 6768 7469 6e67 2070 6572 2073 7065 6369  ghting per speci
-0000f0a0: 6573 2070 6169 722c 2074 6865 206f 7074  es pair, the opt
-0000f0b0: 696f 6e20 602d 2d72 6577 6569 6768 7460  ion `--reweight`
-0000f0c0: 2063 616e 2062 6520 7365 742e 0a0a 2a2a   can be set...**
-0000f0d0: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-0000f0e0: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
-0000f0f0: 2063 6f6d 6d61 6e64 2061 7320 6265 6c6f   command as belo
-0000f100: 772a 2a0a 0a60 6060 0a77 6764 206b 7364  w**..```.wgd ksd
-0000f110: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
-0000f120: 6365 3120 7365 7175 656e 6365 320a 6060  ce1 sequence2.``
-0000f130: 600a 0a2a 2a54 6865 2063 6f6e 7374 7275  `..**The constru
-0000f140: 6374 696f 6e20 6f66 202a 4b2a 3c73 7562  ction of *K*<sub
-0000f150: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-0000f160: 7472 6962 7574 696f 6e20 7769 7468 2072  tribution with r
-0000f170: 6174 6520 636f 7272 6563 7469 6f6e 2a2a  ate correction**
-0000f180: 0a60 6060 0a77 6764 206b 7364 2066 616d  .```.wgd ksd fam
-0000f190: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
-0000f1a0: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
-0000f1b0: 6365 3320 2d2d 666f 6375 7332 616c 6c20  ce3 --focus2all 
-0000f1c0: 7365 7175 656e 6365 3120 2d73 7020 7370  sequence1 -sp sp
-0000f1d0: 6461 7461 0a60 6060 0a0a 496e 7370 6972  data.```..Inspir
-0000f1e0: 6564 2062 7920 7468 6520 7261 7465 2063  ed by the rate c
-0000f1f0: 6f72 7265 6374 696f 6e20 616c 676f 7269  orrection algori
-0000f200: 7468 6d20 696e 205b 6b73 7261 7465 735d  thm in [ksrates]
-0000f210: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
-0000f220: 636f 6d2f 5649 422d 5053 422f 6b73 7261  com/VIB-PSB/ksra
-0000f230: 7465 7329 2c20 7765 2069 6d70 6c65 6d65  tes), we impleme
-0000f240: 6e74 6564 2074 6865 2072 6174 6520 636f  nted the rate co
-0000f250: 7272 6563 7469 6f6e 2061 6e61 6c79 7369  rrection analysi
-0000f260: 7320 616c 736f 2069 6e20 6077 6764 2076  s also in `wgd v
-0000f270: 3260 2c20 7768 6963 6820 6973 206d 6f73  2`, which is mos
-0000f280: 746c 7920 7468 6520 7361 6d65 2061 7320  tly the same as 
-0000f290: 5b6b 7372 6174 6573 5d28 6874 7470 733a  [ksrates](https:
-0000f2a0: 2f2f 6769 7468 7562 2e63 6f6d 2f56 4942  //github.com/VIB
-0000f2b0: 2d50 5342 2f6b 7372 6174 6573 2920 6275  -PSB/ksrates) bu
-0000f2c0: 7420 6469 6666 6572 7320 696e 2074 6865  t differs in the
-0000f2d0: 2063 616c 6375 6c61 7469 6f6e 206f 6620   calculation of 
-0000f2e0: 7468 6520 7374 616e 6461 7264 2064 6576  the standard dev
-0000f2f0: 6961 7469 6f6e 206f 6620 7265 7363 616c  iation of rescal
-0000f300: 6564 202a 4b2a 3c73 7562 3e53 3c2f 7375  ed *K*<sub>S</su
-0000f310: 623e 2061 6765 732e 2054 6f20 7065 7266  b> ages. To perf
-0000f320: 6f72 6d20 7468 6520 7261 7465 2063 6f72  orm the rate cor
-0000f330: 7265 6374 696f 6e20 616e 616c 7973 6973  rection analysis
-0000f340: 2c20 7573 6572 7320 6361 6e20 7573 6520  , users can use 
-0000f350: 626f 7468 2074 6865 2060 7767 6420 6b73  both the `wgd ks
-0000f360: 6460 2061 6e64 2060 7767 6420 7669 7a60  d` and `wgd viz`
-0000f370: 2070 726f 6772 616d 2e20 466f 7220 7468   program. For th
-0000f380: 6520 6077 6764 206b 7364 6020 7072 6f67  e `wgd ksd` prog
-0000f390: 7261 6d2c 2075 7365 7273 206e 6565 6420  ram, users need 
-0000f3a0: 746f 2070 726f 7669 6465 2061 2073 7065  to provide a spe
-0000f3b0: 6369 6573 2074 7265 6520 7669 6120 7468  cies tree via th
-0000f3c0: 6520 6f70 7469 6f6e 2060 2d2d 7370 6563  e option `--spec
-0000f3d0: 6965 7374 7265 6560 206f 6e20 7768 6963  iestree` on whic
-0000f3e0: 6820 7261 7465 2063 6f72 7265 6374 696f  h rate correctio
-0000f3f0: 6e20 6361 6e20 6265 2063 6f6e 6475 6374  n can be conduct
-0000f400: 6564 2e20 4e6f 7465 2074 6861 7420 756e  ed. Note that un
-0000f410: 6e65 6365 7373 6172 7920 6272 6163 6b65  necessary bracke
-0000f420: 7473 206d 6967 6874 206c 6561 6420 746f  ts might lead to
-0000f430: 2075 6e65 7870 6563 7465 6420 6572 726f   unexpected erro
-0000f440: 7273 2c20 666f 7220 696e 7374 616e 6365  rs, for instance
-0000f450: 2061 2074 7265 6520 6028 412c 2842 2c43   a tree `(A,(B,C
-0000f460: 2929 3b60 2073 686f 756c 6420 6e6f 7420  ));` should not 
-0000f470: 6265 2072 6570 7265 7365 6e74 6564 2061  be represented a
-0000f480: 7320 6028 412c 2828 422c 4329 2929 3b60  s `(A,((B,C)));`
-0000f490: 2e20 5468 6520 7365 7420 6f66 2073 7065  . The set of spe
-0000f4a0: 6369 6573 2070 6169 7273 2074 6f20 6265  cies pairs to be
-0000f4b0: 2073 686f 776e 2069 7320 666c 6578 6962   shown is flexib
-0000f4c0: 6c65 2074 6861 7420 7468 6520 6d6f 7374  le that the most
-0000f4d0: 2063 6f6e 7665 6e69 656e 7420 6f70 7469   convenient opti
-0000f4e0: 6f6e 2069 7320 602d 2d66 6f63 7573 3261  on is `--focus2a
-0000f4f0: 6c6c 6020 7768 6963 6820 7369 6d70 6c79  ll` which simply
-0000f500: 2073 686f 7773 2061 6c6c 2074 6865 2070   shows all the p
-0000f510: 6f73 7369 626c 6520 666f 6361 6c2d 7369  ossible focal-si
-0000f520: 7374 6572 2073 7065 6369 6573 2070 6169  ster species pai
-0000f530: 7273 2c20 6f72 2075 7365 7273 2063 616e  rs, or users can
-0000f540: 206d 616e 7561 6c6c 7920 7365 7420 7468   manually set th
-0000f550: 6520 7370 6563 6965 7320 7061 6972 7320  e species pairs 
-0000f560: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
-0000f570: 2d2d 7370 6169 7260 2e20 4e6f 7465 2074  --spair`. Note t
-0000f580: 6861 7420 6966 2074 6865 2073 7065 6369  hat if the speci
-0000f590: 6573 2070 6169 7273 2077 6572 6520 6d61  es pairs were ma
-0000f5a0: 6e75 616c 6c79 2073 6574 2c20 6974 2077  nually set, it w
-0000f5b0: 6f75 6c64 2062 6520 6e65 6564 6564 2074  ould be needed t
-0000f5c0: 6f20 7365 7420 7468 6520 6f70 7469 6f6e  o set the option
-0000f5d0: 2060 2d2d 636c 6173 7369 6360 2e20 5468   `--classic`. Th
-0000f5e0: 6520 6f70 7469 6f6e 2060 6f6e 6c79 726f  e option `onlyro
-0000f5f0: 6f74 6f75 7460 2063 616e 2062 6520 7365  otout` can be se
-0000f600: 7420 746f 206f 6e6c 7920 636f 6e73 6964  t to only consid
-0000f610: 6572 206f 7574 6772 6f75 7020 6174 2074  er outgroup at t
-0000f620: 6865 2072 6f6f 742c 2069 6e73 7465 6164  he root, instead
-0000f630: 206f 6620 616c 6c20 7468 6520 706f 7373   of all the poss
-0000f640: 6962 6c65 206f 7574 6772 6f75 7073 2070  ible outgroups p
-0000f650: 6572 2066 6f63 616c 2d73 6973 7465 7220  er focal-sister 
-0000f660: 7370 6563 6965 7320 7061 6972 2c20 7768  species pair, wh
-0000f670: 6963 6820 6861 7320 696d 7061 6374 206f  ich has impact o
-0000f680: 6e20 7468 6520 6669 6e61 6c20 636f 7272  n the final corr
-0000f690: 6563 7465 6420 2a4b 2a3c 7375 623e 533c  ected *K*<sub>S<
-0000f6a0: 2f73 7562 3e20 6167 6573 2e20 536f 6d65  /sub> ages. Some
-0000f6b0: 206f 7468 6572 206f 7074 696f 6e73 2077   other options w
-0000f6c0: 6869 6368 2068 6176 6520 6e6f 2069 6d70  hich have no imp
-0000f6d0: 6163 7420 6f6e 2074 6865 2072 6174 6520  act on the rate 
-0000f6e0: 636f 7272 6563 7469 6f6e 2062 7574 2061  correction but a
-0000f6f0: 6464 206d 6f72 6520 6c61 7965 7273 206f  dd more layers o
-0000f700: 7220 6368 616e 6765 2074 6865 2061 7070  r change the app
-0000f710: 6561 7261 6e63 6520 6f6e 2074 6865 206d  earance on the m
-0000f720: 6978 6564 202a 4b2a 3c73 7562 3e53 3c2f  ixed *K*<sub>S</
-0000f730: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-0000f740: 7574 696f 6e2c 2069 6e63 6c75 6465 2060  ution, include `
-0000f750: 706c 6f74 6170 676d 6d60 2061 6e64 2060  plotapgmm` and `
-0000f760: 706c 6f74 656c 6d6d 6020 6574 632e 2054  plotelmm` etc. T
-0000f770: 6865 206f 7074 696f 6e20 6070 6c6f 7461  he option `plota
-0000f780: 7067 6d6d 6020 6361 6e20 6265 2073 6574  pgmm` can be set
-0000f790: 2074 6f20 6361 6c6c 2074 6865 2047 4d4d   to call the GMM
-0000f7a0: 2061 6e61 6c79 7369 7320 7570 6f6e 2074   analysis upon t
-0000f7b0: 6865 2061 6e63 686f 7220 7061 6972 202a  he anchor pair *
-0000f7c0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-0000f7d0: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
-0000f7e0: 616e 6420 706c 6f74 2074 6865 2063 6c75  and plot the clu
-0000f7f0: 7374 6572 696e 6720 7265 7375 6c74 2075  stering result u
-0000f800: 706f 6e20 7468 6520 6d69 7865 6420 2a4b  pon the mixed *K
-0000f810: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
-0000f820: 6520 6469 7374 7269 6275 7469 6f6e 2c20  e distribution, 
-0000f830: 7768 6963 6820 6861 7320 746f 2062 6520  which has to be 
-0000f840: 636f 2d73 6574 2077 6974 6820 7468 6520  co-set with the 
-0000f850: 6f70 7469 6f6e 2060 2d2d 616e 6368 6f72  option `--anchor
-0000f860: 706f 696e 7473 6020 7072 6f76 6964 696e  points` providin
-0000f870: 6720 7468 6520 616e 6368 6f72 2070 6169  g the anchor pai
-0000f880: 7273 2069 6e66 6f72 6d61 7469 6f6e 2e20  rs information. 
-0000f890: 5468 6520 6f70 7469 6f6e 2060 2d2d 706c  The option `--pl
-0000f8a0: 6f74 656c 6d6d 6020 6361 6e20 6265 2073  otelmm` can be s
-0000f8b0: 6574 2074 6f20 6361 6c6c 2074 6865 2045  et to call the E
-0000f8c0: 4c4d 4d20 616e 616c 7973 6973 2075 706f  LMM analysis upo
-0000f8d0: 6e20 7468 6520 7061 7261 6c6f 676f 7573  n the paralogous
-0000f8e0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000f8f0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-0000f900: 6e2e 0a0a 5468 6572 6520 6172 6520 3231  n...There are 21
-0000f910: 2063 6f6c 756d 6e73 2069 6e20 7468 6520   columns in the 
-0000f920: 7265 7375 6c74 2060 2a6b 732e 7473 7660  result `*ks.tsv`
-0000f930: 2066 696c 6520 6265 7369 6465 7320 7468   file besides th
-0000f940: 6520 696e 6465 7820 636f 6c75 6d6e 7320  e index columns 
-0000f950: 6070 6169 7260 2061 7320 7468 6520 756e  `pair` as the un
-0000f960: 6971 7565 2069 6465 6e74 6966 6965 7220  ique identifier 
-0000f970: 666f 7220 6561 6368 2067 656e 6520 7061  for each gene pa
-0000f980: 6972 2e20 5468 6520 604e 602c 2060 5360  ir. The `N`, `S`
-0000f990: 2c20 6064 4e60 2c20 6064 4e2f 6453 602c  , `dN`, `dN/dS`,
-0000f9a0: 2060 6453 602c 2060 6c60 2061 6e64 2060   `dS`, `l` and `
-0000f9b0: 7460 2061 7265 2066 726f 6d20 7468 6520  t` are from the 
-0000f9c0: 636f 6465 6d6c 2072 6573 756c 7473 2c20  codeml results, 
-0000f9d0: 7265 7072 6573 656e 7469 6e67 2074 6865  representing the
-0000f9e0: 204e 2065 7374 696d 6174 652c 2074 6865   N estimate, the
-0000f9f0: 2053 2065 7374 696d 6174 652c 2074 6865   S estimate, the
-0000fa00: 2064 4e20 6573 7469 6d61 7465 2c20 7468   dN estimate, th
-0000fa10: 6520 644e 2f64 5320 286f 6d65 6761 2920  e dN/dS (omega) 
-0000fa20: 6573 7469 6d61 7465 2c20 7468 6520 6453  estimate, the dS
-0000fa30: 2065 7374 696d 6174 652c 2074 6865 206c   estimate, the l
-0000fa40: 6f67 2d6c 696b 656c 6968 6f6f 6420 616e  og-likelihood an
-0000fa50: 6420 7468 6520 7420 6573 7469 6d61 7465  d the t estimate
-0000fa60: 2c20 7265 7370 6563 7469 7665 6c79 2e20  , respectively. 
-0000fa70: 5468 6520 6061 6c69 676e 6d65 6e74 636f  The `alignmentco
-0000fa80: 7665 7261 6765 602c 2060 616c 6967 6e6d  verage`, `alignm
-0000fa90: 656e 7469 6465 6e74 6974 7960 2061 6e64  entidentity` and
-0000faa0: 2060 616c 6967 6e6d 656e 746c 656e 6774   `alignmentlengt
-0000fab0: 6860 2061 7265 2074 6865 2069 6e66 6f72  h` are the infor
-0000fac0: 6d61 7469 6f6e 2070 6572 7461 696e 696e  mation pertainin
-0000fad0: 6720 746f 2074 6865 2061 6c69 676e 6d65  g to the alignme
-0000fae0: 6e74 2066 6f72 2065 6163 6820 6661 6d69  nt for each fami
-0000faf0: 6c79 2c20 7265 7072 6573 656e 7469 6e67  ly, representing
-0000fb00: 2074 6865 2072 6174 696f 206f 6620 7468   the ratio of th
-0000fb10: 6520 7374 7269 7070 6564 2061 6c69 676e  e stripped align
-0000fb20: 6d65 6e74 206c 656e 6774 6820 636f 6d70  ment length comp
-0000fb30: 6172 6564 2074 6f20 7468 6520 6675 6c6c  ared to the full
-0000fb40: 2061 6c69 676e 6d65 6e74 206c 656e 6774   alignment lengt
-0000fb50: 682c 2074 6865 2072 6174 696f 206f 6620  h, the ratio of 
-0000fb60: 636f 6c75 6d6e 7320 7769 7468 2069 6465  columns with ide
-0000fb70: 6e74 6963 616c 206e 7563 6c65 6f74 6964  ntical nucleotid
-0000fb80: 6573 2063 6f6d 7061 7265 6420 746f 2074  es compared to t
-0000fb90: 6865 206f 7665 7261 6c6c 2063 6f6c 756d  he overall colum
-0000fba0: 6e73 206f 6620 7468 6520 7374 7269 7070  ns of the stripp
-0000fbb0: 6564 2061 6c69 676e 6d65 6e74 2c20 616e  ed alignment, an
-0000fbc0: 6420 7468 6520 6c65 6e67 7468 206f 6620  d the length of 
-0000fbd0: 7468 6520 6675 6c6c 2061 6c69 676e 6d65  the full alignme
-0000fbe0: 6e74 2c20 7265 7370 6563 7469 7665 6c79  nt, respectively
-0000fbf0: 2e0a 2323 2320 7767 6420 6d69 780a 0a2a  ..### wgd mix..*
-0000fc00: 2a54 6865 206d 6978 7475 7265 206d 6f64  *The mixture mod
-0000fc10: 656c 2063 6c75 7374 6572 696e 6720 616e  el clustering an
-0000fc20: 616c 7973 6973 206f 6620 2a4b 2a3c 7375  alysis of *K*<su
-0000fc30: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-0000fc40: 7374 7269 6275 7469 6f6e 2a2a 0a60 6060  stribution**.```
-0000fc50: 0a77 6764 206d 6978 206b 7364 6174 6120  .wgd mix ksdata 
-0000fc60: 282d 2d6e 5f69 6e69 7420 3230 3020 2d2d  (--n_init 200 --
-0000fc70: 6d61 785f 6974 6572 2032 3030 202d 2d6b  max_iter 200 --k
-0000fc80: 735f 7261 6e67 6520 3020 3520 2d2d 6669  s_range 0 5 --fi
-0000fc90: 6c74 6572 7320 3330 3020 2d2d 6269 6e73  lters 300 --bins
-0000fca0: 2035 3020 2d2d 636f 6d70 6f6e 656e 7473   50 --components
-0000fcb0: 2031 2034 202d 2d67 616d 6d61 2030 2e30   1 4 --gamma 0.0
-0000fcc0: 3031 290a 6060 600a 0a54 6869 7320 7061  01).```..This pa
-0000fcd0: 7274 206f 6620 4761 7573 7369 616e 206d  rt of Gaussian m
-0000fce0: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
-0000fcf0: 2847 4d4d 2920 616e 616c 7973 6973 2069  (GMM) analysis i
-0000fd00: 7320 696e 6865 7269 7465 6420 6672 6f6d  s inherited from
-0000fd10: 2074 6865 206f 7269 6769 6e61 6c20 6077   the original `w
-0000fd20: 6764 6020 7072 6f67 7261 6d2c 2062 7574  gd` program, but
-0000fd30: 2077 7269 7465 7320 6164 6469 7469 6f6e   writes addition
-0000fd40: 616c 6c79 2074 6865 2070 726f 6261 6269  ally the probabi
-0000fd50: 6c69 7479 206f 6620 6561 6368 202a 4b2a  lity of each *K*
-0000fd60: 3c73 7562 3e53 3c2f 7375 623e 2076 616c  <sub>S</sub> val
-0000fd70: 7565 2069 6e74 6f20 7468 6520 6669 6e61  ue into the fina
-0000fd80: 6c20 6461 7461 6672 616d 652e 2042 6173  l dataframe. Bas
-0000fd90: 6963 616c 6c79 2c20 7573 6572 7320 6e65  ically, users ne
-0000fda0: 6564 2074 6f20 7072 6f76 6964 6520 7769  ed to provide wi
-0000fdb0: 7468 2061 2028 6e6f 726d 616c 6c79 2066  th a (normally f
-0000fdc0: 726f 6d20 7768 6f6c 652d 7061 7261 6e6f  rom whole-parano
-0000fdd0: 6d65 206f 7220 616e 6368 6f72 2d70 6169  me or anchor-pai
-0000fde0: 7273 2920 2a4b 2a3c 7375 623e 533c 2f73  rs) *K*<sub>S</s
-0000fdf0: 7562 3e20 6461 7461 6669 6c65 2061 6e64  ub> datafile and
-0000fe00: 2074 6865 2047 4d4d 2061 6e61 6c79 7369   the GMM analysi
-0000fe10: 7320 7769 6c6c 2062 6520 636f 6e64 7563  s will be conduc
-0000fe20: 7465 6420 7570 6f6e 2074 6865 2064 6174  ted upon the dat
-0000fe30: 6166 696c 652e 2053 6f6d 6520 7061 7261  afile. Some para
-0000fe40: 6d65 7465 7273 2063 616e 2061 6666 6563  meters can affec
-0000fe50: 7420 7468 6520 7265 7375 6c74 732c 2069  t the results, i
-0000fe60: 6e63 6c75 6469 6e67 2060 2d2d 6e5f 696e  ncluding `--n_in
-0000fe70: 6974 602c 2077 6869 6368 2073 6574 7320  it`, which sets 
-0000fe80: 7468 6520 6e75 6d62 6572 206f 6620 6b2d  the number of k-
-0000fe90: 6d65 616e 7320 696e 6974 6961 6c69 7a61  means initializa
-0000fea0: 7469 6f6e 7320 2864 6566 6175 6c74 2032  tions (default 2
-0000feb0: 3030 292c 2060 2d2d 6d61 785f 6974 6572  00), `--max_iter
-0000fec0: 602c 2077 6869 6368 2073 6574 7320 7468  `, which sets th
-0000fed0: 6520 6d61 7869 6d75 6d20 6e75 6d62 6572  e maximum number
-0000fee0: 206f 6620 6974 6572 6174 696f 6e73 2028   of iterations (
-0000fef0: 6465 6661 756c 7420 3230 3029 2c20 602d  default 200), `-
-0000ff00: 2d6d 6574 686f 6460 2c20 7768 6963 6820  -method`, which 
-0000ff10: 6465 7465 726d 696e 6573 2077 6869 6368  determines which
-0000ff20: 2063 6c75 7374 6572 696e 6720 6d65 7468   clustering meth
-0000ff30: 6f64 2074 6f20 7573 6520 2864 6566 6175  od to use (defau
-0000ff40: 6c74 2067 6d6d 292c 2060 2d2d 6761 6d6d  lt gmm), `--gamm
-0000ff50: 6160 2c20 7768 6963 6820 7365 7473 2074  a`, which sets t
-0000ff60: 6865 2067 616d 6d61 2070 6172 616d 6574  he gamma paramet
-0000ff70: 6572 2066 6f72 2074 6865 2062 676d 6d20  er for the bgmm 
-0000ff80: 6d6f 6465 6c20 2864 6566 6175 6c74 2030  model (default 0
-0000ff90: 2e30 3031 292c 2060 2d2d 636f 6d70 6f6e  .001), `--compon
-0000ffa0: 656e 7473 602c 2077 6869 6368 2073 6574  ents`, which set
-0000ffb0: 7320 7468 6520 7261 6e67 6520 6f66 2074  s the range of t
-0000ffc0: 6865 206e 756d 6265 7220 6f66 2063 6f6d  he number of com
-0000ffd0: 706f 6e65 6e74 7320 746f 2066 6974 2028  ponents to fit (
-0000ffe0: 6465 6661 756c 7420 3120 3429 2c20 616e  default 1 4), an
-0000fff0: 6420 7468 6520 6461 7461 2066 696c 7465  d the data filte
-00010000: 7269 6e67 2070 6172 616d 6574 6572 7320  ring parameters 
-00010010: 602d 2d66 696c 7465 7273 6020 7768 6963  `--filters` whic
-00010020: 6820 6669 6c74 6572 7320 6461 7461 2062  h filters data b
-00010030: 6173 6564 206f 6e20 616c 6967 6e6d 656e  ased on alignmen
-00010040: 7420 6c65 6e67 7468 2c20 602d 2d6b 735f  t length, `--ks_
-00010050: 7261 6e67 6560 2077 6869 6368 2066 696c  range` which fil
-00010060: 7465 7273 2064 6174 6120 6261 7365 6420  ters data based 
-00010070: 6f6e 202a 4b2a 3c73 7562 3e53 3c2f 7375  on *K*<sub>S</su
-00010080: 623e 2076 616c 7565 7320 616e 6420 7468  b> values and th
-00010090: 6520 7061 7261 6d65 7465 7220 602d 2d62  e parameter `--b
-000100a0: 696e 7360 2077 6869 6368 2073 6574 7320  ins` which sets 
-000100b0: 7468 6520 6e75 6d62 6572 206f 6620 6269  the number of bi
-000100c0: 6e73 2069 6e20 2a4b 2a3c 7375 623e 533c  ns in *K*<sub>S<
-000100d0: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-000100e0: 6f6e 2028 6465 6661 756c 7420 3530 292e  on (default 50).
-000100f0: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
-00010100: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
-00010110: 2075 7365 2063 6f6d 6d61 6e64 2073 696d   use command sim
-00010120: 706c 7920 6173 2062 656c 6f77 2a2a 0a0a  ply as below**..
-00010130: 6060 600a 7767 6420 6d69 7820 6b73 6461  ```.wgd mix ksda
-00010140: 7461 0a60 6060 0a0a 2323 2320 7767 6420  ta.```..### wgd 
-00010150: 7065 616b 0a0a 2a2a 5468 6520 7365 6172  peak..**The sear
-00010160: 6368 206f 6620 6372 6564 6961 626c 6520  ch of crediable 
-00010170: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00010180: 7261 6e67 6520 7573 6564 2069 6e20 5747  range used in WG
-00010190: 4420 6461 7469 6e67 2a2a 0a60 6060 0a77  D dating**.```.w
-000101a0: 6764 2070 6561 6b20 6b73 6461 7461 202d  gd peak ksdata -
-000101b0: 6170 2061 7064 6174 6120 2d73 6d20 736d  ap apdata -sm sm
-000101c0: 6461 7461 202d 6c65 206c 6564 6174 6120  data -le ledata 
-000101d0: 2d6d 7020 6d70 6461 7461 202d 2d68 6575  -mp mpdata --heu
-000101e0: 7269 7374 6963 2028 2d2d 616c 6967 6e66  ristic (--alignf
-000101f0: 696c 7465 7220 302e 3020 3020 302e 3020  ilter 0.0 0 0.0 
-00010200: 2d2d 6b73 7261 6e67 6520 3020 3520 2d2d  --ksrange 0 5 --
-00010210: 6269 6e5f 7769 6474 6820 302e 3120 2d2d  bin_width 0.1 --
-00010220: 6775 6964 6520 7365 676d 656e 7420 2d2d  guide segment --
-00010230: 7072 6f6d 696e 656e 6365 5f63 7574 6f66  prominence_cutof
-00010240: 6620 302e 3120 2d2d 7265 6c5f 6865 6967  f 0.1 --rel_heig
-00010250: 6874 2030 2e34 202d 2d63 6920 3935 202d  ht 0.4 --ci 95 -
-00010260: 2d68 6472 2039 3520 2d2d 6b73 6375 746f  -hdr 95 --kscuto
-00010270: 6666 2035 290a 6060 600a 0a41 7320 6d65  ff 5).```..As me
-00010280: 6e74 696f 6e65 6420 7072 6576 696f 7573  ntioned previous
-00010290: 6c79 2c20 6120 6865 7572 6973 7469 6320  ly, a heuristic 
-000102a0: 6d65 7468 6f64 2061 6e64 2061 2063 6f6c  method and a col
-000102b0: 6c69 6e65 6172 2073 6567 6d65 6e74 732d  linear segments-
-000102c0: 6775 6964 6564 2061 6e63 686f 7220 7061  guided anchor pa
-000102d0: 6972 2063 6c75 7374 6572 696e 6720 666f  ir clustering fo
-000102e0: 7220 7468 6520 7365 6172 6368 206f 6620  r the search of 
-000102f0: 6372 6564 6961 626c 6520 2a4b 2a3c 7375  crediable *K*<su
-00010300: 623e 533c 2f73 7562 3e20 7261 6e67 6520  b>S</sub> range 
-00010310: 7573 6564 2069 6e20 5747 4420 6461 7469  used in WGD dati
-00010320: 6e67 2061 7265 2069 6d70 6c65 6d65 6e74  ng are implement
-00010330: 6564 2069 6e20 6077 6764 2076 3260 2e20  ed in `wgd v2`. 
-00010340: 5573 6572 7320 6e65 6564 2074 6f20 7072  Users need to pr
-00010350: 6f76 6964 6520 7468 6520 616e 6368 6f72  ovide the anchor
-00010360: 2070 6f69 6e74 732c 2073 6567 6d65 6e74   points, segment
-00010370: 732c 206c 6973 7473 6567 6d65 6e74 732c  s, listsegments,
-00010380: 206d 756c 7469 706c 6963 6f6e 7061 6972   multipliconpair
-00010390: 7320 6461 7461 6669 6c65 2066 726f 6d20  s datafile from 
-000103a0: 6069 2d61 6468 6f72 6560 2074 6f20 6163  `i-adhore` to ac
-000103b0: 6869 6576 6520 7468 6520 636c 7573 7465  hieve the cluste
-000103c0: 7269 6e67 2066 756e 6374 696f 6e2e 2053  ring function. S
-000103d0: 6f6d 6520 7061 7261 6d65 7465 7273 2074  ome parameters t
-000103e0: 6861 7420 6361 6e20 696d 7061 6374 2074  hat can impact t
-000103f0: 6865 2072 6573 756c 7473 2069 6e63 6c75  he results inclu
-00010400: 6465 2060 2d2d 616c 6967 6e66 696c 7465  de `--alignfilte
-00010410: 7260 2c20 7768 6963 6820 6669 6c74 6572  r`, which filter
-00010420: 7320 7468 6520 6461 7461 2062 6173 6564  s the data based
-00010430: 206f 6e20 616c 6967 6e6d 656e 7420 6964   on alignment id
-00010440: 656e 7469 7479 2c20 6c65 6e67 7468 2061  entity, length a
-00010450: 6e64 2063 6f76 6572 6167 652c 2060 2d2d  nd coverage, `--
-00010460: 6b73 7261 6e67 6560 2c20 7768 6963 6820  ksrange`, which 
-00010470: 7365 7473 2074 6865 2072 616e 6765 206f  sets the range o
-00010480: 6620 4b73 2074 6f20 6265 2061 6e61 6c79  f Ks to be analy
-00010490: 7a65 642c 2060 2d2d 6269 6e5f 7769 6474  zed, `--bin_widt
-000104a0: 6860 2c20 7768 6963 6820 7365 7473 2074  h`, which sets t
-000104b0: 6865 2062 616e 6477 6964 7468 206f 6620  he bandwidth of 
-000104c0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-000104d0: 6469 7374 7269 6275 7469 6f6e 2c20 602d  distribution, `-
-000104e0: 2d77 6569 6768 7473 5f6f 7574 6c69 6572  -weights_outlier
-000104f0: 735f 696e 636c 7564 6564 6020 7768 6963  s_included` whic
-00010500: 6820 6465 7465 726d 696e 6573 2077 6865  h determines whe
-00010510: 7468 6572 2074 6f20 696e 636c 7564 6520  ther to include 
-00010520: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00010530: 6f75 746c 6965 7273 2028 7768 6f73 6520  outliers (whose 
-00010540: 7661 6c75 6520 6973 206f 7665 7220 3529  value is over 5)
-00010550: 2069 6e20 616e 616c 7973 6973 2c20 602d   in analysis, `-
-00010560: 2d6d 6574 686f 6460 2077 6869 6368 2064  -method` which d
-00010570: 6574 6572 6d69 6e65 7320 7768 6963 6820  etermines which 
-00010580: 636c 7573 7465 7269 6e67 206d 6574 686f  clustering metho
-00010590: 6420 746f 2075 7365 2028 6465 6661 756c  d to use (defaul
-000105a0: 7420 676d 6d29 2c20 602d 2d73 6565 6460  t gmm), `--seed`
-000105b0: 2077 6869 6368 2073 6574 7320 7468 6520   which sets the 
-000105c0: 7261 6e64 6f6d 2073 6565 6420 6769 7665  random seed give
-000105d0: 6e20 746f 2069 6e69 7469 616c 697a 6174  n to initializat
-000105e0: 696f 6e20 2864 6566 6175 6c74 2032 3335  ion (default 235
-000105f0: 3238 3930 292c 2060 2d2d 6e5f 696e 6974  2890), `--n_init
-00010600: 602c 2077 6869 6368 2073 6574 7320 7468  `, which sets th
-00010610: 6520 6e75 6d62 6572 206f 6620 6b2d 6d65  e number of k-me
-00010620: 616e 7320 696e 6974 6961 6c69 7a61 7469  ans initializati
-00010630: 6f6e 7320 2864 6566 6175 6c74 2032 3030  ons (default 200
-00010640: 292c 2060 2d2d 656d 5f69 7465 7260 2c20  ), `--em_iter`, 
-00010650: 7768 6963 6820 7365 7473 2074 6865 206d  which sets the m
-00010660: 6178 696d 756d 206e 756d 6265 7220 6f66  aximum number of
-00010670: 2069 7465 7261 7469 6f6e 7320 2864 6566   iterations (def
-00010680: 6175 6c74 2032 3030 292c 2060 2d2d 6761  ault 200), `--ga
-00010690: 6d6d 6160 2c20 7768 6963 6820 7365 7473  mma`, which sets
-000106a0: 2074 6865 2067 616d 6d61 2070 6172 616d   the gamma param
-000106b0: 6574 6572 2066 6f72 2074 6865 2062 676d  eter for the bgm
-000106c0: 6d20 6d6f 6465 6c20 2864 6566 6175 6c74  m model (default
-000106d0: 2030 2e30 3031 292c 2060 2d2d 636f 6d70   0.001), `--comp
-000106e0: 6f6e 656e 7473 602c 2077 6869 6368 2073  onents`, which s
-000106f0: 6574 7320 7468 6520 7261 6e67 6520 6f66  ets the range of
-00010700: 2074 6865 206e 756d 6265 7220 6f66 2063   the number of c
-00010710: 6f6d 706f 6e65 6e74 7320 746f 2066 6974  omponents to fit
-00010720: 2028 6465 6661 756c 7420 3120 3429 2c20   (default 1 4), 
-00010730: 602d 2d77 6569 6768 7465 6460 2077 6869  `--weighted` whi
-00010740: 6368 2064 6574 6572 6d69 6e65 7320 7768  ch determines wh
-00010750: 6574 6865 7220 746f 2075 7365 206e 6f64  ether to use nod
-00010760: 652d 7765 6967 6874 6564 206d 6574 686f  e-weighted metho
-00010770: 6420 666f 7220 6465 2d72 6564 756e 6461  d for de-redunda
-00010780: 6e63 792c 2060 2d2d 6775 6964 6560 2077  ncy, `--guide` w
-00010790: 6869 6368 2064 6574 6572 6d69 6e65 7320  hich determines 
-000107a0: 7768 6963 6820 7265 6769 6d65 2072 6573  which regime res
-000107b0: 6964 696e 6720 616e 6368 6f72 7320 746f  iding anchors to
-000107c0: 2062 6520 7573 6564 2028 6465 6661 756c   be used (defaul
-000107d0: 7420 5365 676d 656e 7429 2c20 602d 2d70  t Segment), `--p
-000107e0: 726f 6d69 6e65 6e63 655f 6375 746f 6666  rominence_cutoff
-000107f0: 6020 7768 6963 6820 7365 7473 2074 6865  ` which sets the
-00010800: 2070 726f 6d69 6e65 6e63 6520 6375 746f   prominence cuto
-00010810: 6666 206f 6620 6163 6365 7074 6162 6c65  ff of acceptable
-00010820: 2070 6561 6b73 2069 6e20 7065 616b 2066   peaks in peak f
-00010830: 696e 6469 6e67 2070 726f 6365 7373 2c20  inding process, 
-00010840: 602d 2d72 656c 5f68 6569 6768 7460 2077  `--rel_height` w
-00010850: 6869 6368 2073 6574 7320 7468 6520 7265  hich sets the re
-00010860: 6c61 7469 7665 2068 6569 6768 7420 6174  lative height at
-00010870: 2077 6869 6368 2074 6865 2070 6561 6b20   which the peak 
-00010880: 7769 6474 6820 6973 206d 6561 7375 7265  width is measure
-00010890: 642c 2060 2d2d 6b73 746f 6461 7465 6020  d, `--kstodate` 
-000108a0: 7768 6963 6820 6d61 6e75 616c 6c79 2073  which manually s
-000108b0: 6574 7320 7468 6520 7261 6e67 6520 6f66  ets the range of
-000108c0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000108d0: 2074 6f20 6265 2064 6174 6564 2069 6e20   to be dated in 
-000108e0: 6865 7572 6973 7469 6320 7365 6172 6368  heuristic search
-000108f0: 2061 6e64 206e 6565 6473 2074 6f20 6265   and needs to be
-00010900: 2063 6f2d 7365 7420 7769 7468 206f 7074   co-set with opt
-00010910: 696f 6e20 602d 2d6d 616e 7561 6c73 6574  ion `--manualset
-00010920: 602c 2060 2d2d 786c 696d 6020 616e 6420  `, `--xlim` and 
-00010930: 602d 2d79 6c69 6d60 2064 6574 6572 6d69  `--ylim` determi
-00010940: 6e69 6e67 2074 6865 2078 2061 6e64 2079  ning the x and y
-00010950: 2061 7869 7320 6c69 6d69 7420 6f66 2047   axis limit of G
-00010960: 4d4d 202a 4b2a 3c73 7562 3e53 3c2f 7375  MM *K*<sub>S</su
-00010970: 623e 2064 6973 7472 6962 7574 696f 6e2c  b> distribution,
-00010980: 2060 2d2d 6369 6020 7365 7474 696e 6720   `--ci` setting 
-00010990: 7468 6520 636f 6e66 6964 656e 6365 206c  the confidence l
-000109a0: 6576 656c 206f 6620 6c6f 672d 6e6f 726d  evel of log-norm
-000109b0: 616c 2064 6973 7472 6962 7574 696f 6e20  al distribution 
-000109c0: 746f 2064 6174 6520 2864 6566 6175 6c74  to date (default
-000109d0: 2039 3529 2c20 602d 2d68 6472 6020 7365   95), `--hdr` se
-000109e0: 7474 696e 6720 7468 6520 6869 6768 6573  tting the highes
-000109f0: 7420 6465 6e73 6974 7920 7265 6769 6f6e  t density region
-00010a00: 2028 4844 5229 2061 7070 6c69 6564 2069   (HDR) applied i
-00010a10: 6e20 7468 6520 7365 676d 656e 742d 6775  n the segment-gu
-00010a20: 6964 6564 2061 6e63 686f 7220 7061 6972  ided anchor pair
-00010a30: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00010a40: 2064 6973 7472 6962 7574 696f 6e20 2864   distribution (d
-00010a50: 6566 6175 6c74 2039 3529 2c20 602d 2d68  efault 95), `--h
-00010a60: 6575 7269 7374 6963 6020 6465 7465 726d  euristic` determ
-00010a70: 696e 696e 6720 7768 6574 6865 7220 746f  ining whether to
-00010a80: 2069 6e69 7469 6174 6520 6865 7572 6973   initiate heuris
-00010a90: 7469 6320 6d65 7468 6f64 206f 6620 6465  tic method of de
-00010aa0: 6669 6e69 6e67 2043 4920 666f 7220 6461  fining CI for da
-00010ab0: 7469 6e67 2c20 602d 2d6b 7363 7574 6f66  ting, `--kscutof
-00010ac0: 6660 2073 6574 7469 6e67 2074 6865 202a  f` setting the *
-00010ad0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2073  K*<sub>S</sub> s
-00010ae0: 6174 7572 6174 696f 6e20 6375 746f 6666  aturation cutoff
-00010af0: 2069 6e20 6461 7469 6e67 2028 6465 6661   in dating (defa
-00010b00: 756c 7420 3529 2e0a 0a46 6f75 7220 7265  ult 5)...Four re
-00010b10: 7375 6c74 2073 7562 666c 6f64 6572 7320  sult subfloders 
-00010b20: 7769 6c6c 2062 6520 7072 6f64 7563 6564  will be produced
-00010b30: 2c20 6e61 6d65 6c79 2060 416e 6368 6f72  , namely `Anchor
-00010b40: 4b73 5f46 696e 6450 6561 6b60 2c20 6041  Ks_FindPeak`, `A
-00010b50: 6e63 686f 724b 735f 474d 4d60 2c20 6053  nchorKs_GMM`, `S
-00010b60: 6567 6d65 6e74 4775 6964 654b 735f 474d  egmentGuideKs_GM
-00010b70: 4d60 2061 6e64 2060 5365 676d 656e 744b  M` and `SegmentK
-00010b80: 735f 4669 6e64 5065 616b 602e 2054 6865  s_FindPeak`. The
-00010b90: 2060 416e 6368 6f72 4b73 5f46 696e 6450   `AnchorKs_FindP
-00010ba0: 6561 6b60 2073 7562 666c 6f64 6572 2063  eak` subfloder c
-00010bb0: 6f6e 7461 696e 7320 7265 7375 6c74 7320  ontains results 
-00010bc0: 6f66 2074 6865 2064 6574 6563 7465 6420  of the detected 
-00010bd0: 7065 616b 7320 6279 2074 6865 2060 7369  peaks by the `si
-00010be0: 676e 616c 6020 6d6f 6475 6c65 206f 6620  gnal` module of 
-00010bf0: 6053 6369 5079 6020 6c69 6272 6172 7920  `SciPy` library 
-00010c00: 616e 6420 7468 6520 6173 7375 6d65 6420  and the assumed 
-00010c10: 6869 6768 6573 7420 6d61 7373 2070 6172  highest mass par
-00010c20: 7420 2872 6566 6572 7265 6420 746f 2061  t (referred to a
-00010c30: 7320 4869 6768 4d61 7373 2068 6572 6561  s HighMass herea
-00010c40: 6674 6572 2920 6f66 2065 6163 6820 7065  fter) of each pe
-00010c50: 616b 2c20 7768 6963 6820 6361 6e20 6265  ak, which can be
-00010c60: 2075 7365 6420 666f 7220 6675 7274 6865   used for furthe
-00010c70: 7220 5747 4420 6461 7469 6e67 2e20 5468  r WGD dating. Th
-00010c80: 6520 6041 6e63 686f 724b 735f 474d 4d60  e `AnchorKs_GMM`
-00010c90: 2073 686f 7773 2074 6865 2047 4d4d 2072   shows the GMM r
-00010ca0: 6573 756c 7473 2075 706f 6e20 7468 6520  esults upon the 
-00010cb0: 6f72 6967 696e 616c 2061 6e63 686f 7220  original anchor 
-00010cc0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00010cd0: 6469 7374 7269 6275 7469 6f6e 2062 7920  distribution by 
-00010ce0: 7468 6520 606d 6978 7475 7265 6020 6d6f  the `mixture` mo
-00010cf0: 6475 6c65 206f 6620 6073 6369 6b69 742d  dule of `scikit-
-00010d00: 6c65 6172 6e60 206c 6962 7261 7279 2061  learn` library a
-00010d10: 6e64 2074 776f 2073 7562 666c 6f64 6572  nd two subfloder
-00010d20: 732c 2060 4c6f 6747 4d4d 5f43 4960 2063  s, `LogGMM_CI` c
-00010d30: 6f6e 7461 696e 696e 6720 7468 6520 7265  ontaining the re
-00010d40: 7375 6c74 7320 6f66 2039 3525 2043 4920  sults of 95% CI 
-00010d50: 6f66 2065 6163 6820 636f 6d70 6f6e 656e  of each componen
-00010d60: 742c 2060 4869 6768 4d61 7373 5f43 4960  t, `HighMass_CI`
-00010d70: 2063 6f6e 7461 696e 696e 6720 7468 6520   containing the 
-00010d80: 4869 6768 4d61 7373 206f 6620 6561 6368  HighMass of each
-00010d90: 2063 6f6d 706f 6e65 6e74 2c20 7768 6963   component, whic
-00010da0: 6820 6361 6e20 6265 2075 7365 6420 666f  h can be used fo
-00010db0: 7220 6675 7274 6865 7220 5747 4420 6461  r further WGD da
-00010dc0: 7469 6e67 2e20 5468 6520 6053 6567 6d65  ting. The `Segme
-00010dd0: 6e74 4775 6964 654b 735f 474d 4d60 2073  ntGuideKs_GMM` s
-00010de0: 7562 666f 6c64 6572 2070 7265 7365 6e74  ubfolder present
-00010df0: 7320 7265 7375 6c74 7320 6f66 2073 6567  s results of seg
-00010e00: 6d65 6e74 202a 4b2a 3c73 7562 3e53 3c2f  ment *K*<sub>S</
-00010e10: 7375 623e 2047 4d4d 2077 6869 6368 2061  sub> GMM which a
-00010e20: 7265 206d 6170 7065 6420 6261 636b 2074  re mapped back t
-00010e30: 6f20 7468 6520 7265 7369 6469 6e67 2061  o the residing a
-00010e40: 6e63 686f 7220 7061 6972 7320 616e 6420  nchor pairs and 
-00010e50: 7468 6520 6173 736f 6369 6174 6564 2039  the associated 9
-00010e60: 3525 2048 4452 2061 6e64 2048 6967 684d  5% HDR and HighM
-00010e70: 6173 7320 6f66 2065 6163 6820 7365 676d  ass of each segm
-00010e80: 656e 7420 636c 7573 7465 7220 696e 2073  ent cluster in s
-00010e90: 7562 666c 6f64 6572 7320 6f66 2060 4844  ubfloders of `HD
-00010ea0: 525f 4349 6020 616e 6420 6048 6967 684d  R_CI` and `HighM
-00010eb0: 6173 735f 4349 602e 2054 6865 2060 5365  ass_CI`. The `Se
-00010ec0: 676d 656e 744b 735f 4669 6e64 5065 616b  gmentKs_FindPeak
-00010ed0: 6020 7375 6266 6f6c 6465 7220 6973 2073  ` subfolder is s
-00010ee0: 696d 696c 6172 2074 6f20 6041 6e63 686f  imilar to `Ancho
-00010ef0: 724b 735f 4669 6e64 5065 616b 6020 6275  rKs_FindPeak` bu
-00010f00: 7420 7769 7468 2073 6567 6d65 6e74 202a  t with segment *
-00010f10: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2069  K*<sub>S</sub> i
-00010f20: 6e73 7465 6164 2e20 5468 6520 2a4b 2a3c  nstead. The *K*<
-00010f30: 7375 623e 533c 2f73 7562 3e20 696e 2060  sub>S</sub> in `
-00010f40: 4d75 6c74 6970 6c69 636f 6e60 2063 616e  Multiplicon` can
-00010f50: 2061 6c73 6f20 6265 2063 616c 6375 6c61   also be calcula
-00010f60: 7465 6420 696e 2070 6c61 6365 206f 6620  ted in place of 
-00010f70: 6053 6567 6d65 6e74 6020 7573 696e 6720  `Segment` using 
-00010f80: 7468 6520 6f70 7469 6f6e 2060 2d2d 6775  the option `--gu
-00010f90: 6964 6560 2061 7320 7375 6368 2074 6865  ide` as such the
-00010fa0: 2072 6573 756c 7420 7469 746c 652c 206c   result title, l
-00010fb0: 6162 656c 2c20 6669 6c65 2061 6e64 2066  abel, file and f
-00010fc0: 6f6c 6465 7220 6e61 6d65 7320 7769 6c6c  older names will
-00010fd0: 2062 6520 6368 616e 6765 6420 6163 636f   be changed acco
-00010fe0: 7264 696e 676c 792e 0a0a 2a2a 4120 7375  rdingly...**A su
-00010ff0: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
-00011000: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
-00011010: 6d61 6e64 2073 696d 706c 7920 6173 2062  mand simply as b
-00011020: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
-00011030: 7065 616b 206b 7364 6174 6120 2d61 7020  peak ksdata -ap 
-00011040: 6170 6461 7461 202d 736d 2073 6d64 6174  apdata -sm smdat
-00011050: 6120 2d6c 6520 6c65 6461 7461 202d 6d70  a -le ledata -mp
-00011060: 206d 7064 6174 6120 2d2d 6865 7572 6973   mpdata --heuris
-00011070: 7469 630a 6060 600a 0a23 2323 2077 6764  tic.```..### wgd
-00011080: 2073 796e 0a0a 2a2a 5468 6520 696e 7472   syn..**The intr
-00011090: 612d 7370 6563 6966 6963 2073 796e 7465  a-specific synte
-000110a0: 6e79 2069 6e66 6572 656e 6365 2a2a 0a60  ny inference**.`
-000110b0: 6060 0a77 6764 2073 796e 2066 616d 696c  ``.wgd syn famil
-000110c0: 6965 7320 6766 6620 282d 2d6b 735f 6469  ies gff (--ks_di
-000110d0: 7374 7269 6275 7469 6f6e 206b 7364 6174  stribution ksdat
-000110e0: 6120 2d66 2067 656e 6520 2d61 2049 4420  a -f gene -a ID 
-000110f0: 2d2d 6d69 6e6c 656e 202d 3120 2d2d 6d69  --minlen -1 --mi
-00011100: 6e73 6567 6c65 6e20 3130 3030 3020 2d2d  nseglen 10000 --
-00011110: 6d69 6e67 656e 656e 756d 2033 3029 0a60  mingenenum 30).`
-00011120: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
-00011130: 6077 6764 2073 796e 6020 6973 206d 6169  `wgd syn` is mai
-00011140: 6e6c 7920 6465 616c 696e 6720 7769 7468  nly dealing with
-00011150: 2063 6f6c 6c69 6e65 6172 6974 7920 6f72   collinearity or
-00011160: 2073 796e 7465 6e79 2028 626f 7468 2072   synteny (both r
-00011170: 6566 6572 7265 6420 746f 2061 7320 7379  eferred to as sy
-00011180: 6e74 656e 7920 6865 7265 6166 7465 7229  nteny hereafter)
-00011190: 2061 6e61 6c79 7369 732e 2054 776f 2069   analysis. Two i
-000111a0: 6e70 7574 2066 696c 6573 2061 7265 2065  nput files are e
-000111b0: 7373 656e 7469 616c 2c20 7468 6520 6765  ssential, the ge
-000111c0: 6e65 2066 616d 696c 7920 6669 6c65 2061  ne family file a
-000111d0: 6e64 2074 6865 2067 6666 3320 6669 6c65  nd the gff3 file
-000111e0: 2e20 5468 6520 6765 6e65 2066 616d 696c  . The gene famil
-000111f0: 7920 6669 6c65 2069 7320 696e 2074 6865  y file is in the
-00011200: 2066 6f72 6d61 7420 6173 2060 4f72 7468   format as `Orth
-00011210: 6f46 696e 6465 7260 2e20 5468 6520 736f  oFinder`. The so
-00011220: 6674 7761 7265 2060 692d 6164 686f 7265  ftware `i-adhore
-00011230: 6020 6973 2061 2070 7265 7265 7175 6973  ` is a prerequis
-00011240: 6974 652e 2057 6974 6820 6465 6661 756c  ite. With defaul
-00011250: 7420 7061 7261 6d65 7465 7273 2c20 7468  t parameters, th
-00011260: 6520 7072 6f67 7261 6d20 6261 7369 6361  e program basica
-00011270: 6c6c 7920 636f 6e64 7563 7473 2031 2920  lly conducts 1) 
-00011280: 6669 6c74 6572 696e 6720 6765 6e65 2066  filtering gene f
-00011290: 616d 696c 6965 7320 6261 7365 6420 6f6e  amilies based on
-000112a0: 206d 6178 696d 756d 2066 616d 696c 7920   maximum family 
-000112b0: 7369 7a65 2032 2920 7265 7472 6965 7669  size 2) retrievi
-000112c0: 6e67 2067 656e 6520 706f 7369 7469 6f6e  ng gene position
-000112d0: 2061 6e64 2073 6361 6666 6f6c 6420 696e   and scaffold in
-000112e0: 666f 726d 6174 696f 6e20 6672 6f6d 2067  formation from g
-000112f0: 6666 3320 6669 6c65 2033 2920 7072 6f64  ff3 file 3) prod
-00011300: 7563 696e 6720 7468 6520 636f 6e66 6967  ucing the config
-00011310: 7572 6174 696f 6e20 6669 6c65 2061 6e64  uration file and
-00011320: 2061 7373 6f63 6961 7465 6420 6461 7461   associated data
-00011330: 6669 6c65 7320 666f 7220 6069 2d61 6468  files for `i-adh
-00011340: 6f72 6560 2034 2920 6361 6c6c 696e 6720  ore` 4) calling 
-00011350: 6069 2d61 6468 6f72 6560 2067 6976 656e  `i-adhore` given
-00011360: 2074 6865 2070 6172 616d 6574 6572 7320   the parameters 
-00011370: 7365 7420 746f 2069 6e66 6572 2073 796e  set to infer syn
-00011380: 7465 6e79 2035 2920 7669 7375 616c 697a  teny 5) visualiz
-00011390: 696e 6720 7468 6520 7379 6e74 656e 7920  ing the synteny 
-000113a0: 696e 2022 646f 7470 6c6f 7422 2069 6e20  in "dotplot" in 
-000113b0: 7468 6520 756e 6974 206f 6620 6765 6e65  the unit of gene
-000113c0: 7320 616e 6420 6261 7365 732c 2069 6e20  s and bases, in 
-000113d0: 2253 796e 6465 7074 6822 2070 6c6f 7420  "Syndepth" plot 
-000113e0: 7368 6f77 696e 6720 7468 6520 6469 7374  showing the dist
-000113f0: 7269 6275 7469 6f6e 206f 6620 6469 6666  ribution of diff
-00011400: 6572 656e 7420 6361 7465 676f 7269 6573  erent categories
-00011410: 206f 6620 636f 6c6c 696e 6561 7269 7479   of collinearity
-00011420: 2072 6174 696f 7320 7769 7468 696e 2061   ratios within a
-00011430: 6e64 2062 6574 7765 656e 2073 7065 6369  nd between speci
-00011440: 6573 2c20 696e 2022 6475 7053 7461 636b  es, in "dupStack
-00011450: 2220 706c 6f74 2073 686f 7769 6e67 206d  " plot showing m
-00011460: 756c 7469 706c 6963 6f6e 7320 7769 7468  ultiplicons with
-00011470: 2064 6966 6665 7265 6e74 206d 756c 7469   different multi
-00011480: 706c 6963 6174 696f 6e20 6c65 7665 6c73  plication levels
-00011490: 2e20 3629 2069 6620 7769 7468 202a 4b2a  . 6) if with *K*
-000114a0: 3c73 7562 3e53 3c2f 7375 623e 2064 6174  <sub>S</sub> dat
-000114b0: 612c 2061 2022 2a4b 2a3c 7375 623e 533c  a, a "*K*<sub>S<
-000114c0: 2f73 7562 3e20 646f 7470 6c6f 7422 2077  /sub> dotplot" w
-000114d0: 6974 6820 646f 7473 2061 6e6e 6f74 6174  ith dots annotat
-000114e0: 6564 2069 6e20 2a4b 2a3c 7375 623e 533c  ed in *K*<sub>S<
-000114f0: 2f73 7562 3e20 7661 6c75 6573 2061 6e64  /sub> values and
-00011500: 2061 202a 4b2a 3c73 7562 3e53 3c2f 7375   a *K*<sub>S</su
-00011510: 623e 2064 6973 7472 6962 7574 696f 6e20  b> distribution 
-00011520: 7769 7468 2061 6e63 686f 7220 7061 6972  with anchor pair
-00011530: 7320 6465 6e6f 7465 6420 7769 6c6c 2062  s denoted will b
-00011540: 6520 7072 6f64 7563 6564 2e20 5468 6520  e produced. The 
-00011550: 6765 6e65 2069 6e66 6f72 6d61 7469 6f6e  gene information
-00011560: 2069 6e20 7468 6520 6765 6e65 2066 616d   in the gene fam
-00011570: 696c 7920 6669 6c65 2061 6e64 2067 6666  ily file and gff
-00011580: 3320 6669 6c65 2073 686f 756c 6420 6265  3 file should be
-00011590: 206d 6174 6368 6564 2077 6869 6368 2072   matched which r
-000115a0: 6571 7569 7265 7320 7573 6572 7320 746f  equires users to
-000115b0: 2073 6574 2070 726f 7065 7220 602d 2d66   set proper `--f
-000115c0: 6561 7475 7265 6020 616e 6420 602d 2d61  eature` and `--a
-000115d0: 7474 7269 6275 7465 602e 2054 6865 206d  ttribute`. The m
-000115e0: 6178 696d 756d 2066 616d 696c 7920 7369  aximum family si
-000115f0: 7a65 2074 6f20 6265 2069 6e63 6c75 6465  ze to be include
-00011600: 6420 6361 6e20 6265 2073 6574 2076 6961  d can be set via
-00011610: 2074 6865 206f 7074 696f 6e20 602d 2d6d   the option `--m
-00011620: 6178 7369 7a65 602c 206e 6f74 6564 2074  axsize`, noted t
-00011630: 6861 7420 7468 6973 2066 696c 7465 7269  hat this filteri
-00011640: 6e67 2069 7320 6d61 696e 6c79 2074 6f20  ng is mainly to 
-00011650: 6472 6f70 2074 686f 7365 2068 7567 6520  drop those huge 
-00011660: 7461 6e64 656d 2064 7570 6c69 6361 7465  tandem duplicate
-00011670: 7320 6661 6d69 6c79 2061 6e64 2074 7261  s family and tra
-00011680: 6e73 706f 7361 626c 6520 656c 656d 656e  nsposable elemen
-00011690: 7473 2028 5445 7329 2066 616d 696c 792c  ts (TEs) family,
-000116a0: 2061 6e64 206e 6f74 206d 616e 6461 746f   and not mandato
-000116b0: 7279 2e20 5573 6572 7320 6361 6e20 6669  ry. Users can fi
-000116c0: 6c74 6572 2074 686f 7365 2066 7261 676d  lter those fragm
-000116d0: 656e 7461 7279 2073 6361 6666 6f6c 6473  entary scaffolds
-000116e0: 2076 6961 2074 6865 206f 7074 696f 6e20   via the option 
-000116f0: 602d 2d6d 696e 6c65 6e60 2e20 5468 6520  `--minlen`. The 
-00011700: 6d69 6e69 6d75 6d20 6c65 6e67 7468 2061  minimum length a
-00011710: 6e64 206e 756d 6265 7220 6f66 2067 656e  nd number of gen
-00011720: 6573 2066 6f72 2061 2073 6567 6d65 6e74  es for a segment
-00011730: 2074 6f20 6265 2063 6f6e 7369 6465 7265   to be considere
-00011740: 6420 6361 6e20 6265 2073 6574 2076 6961  d can be set via
-00011750: 2074 6865 206f 7074 696f 6e20 602d 2d6d   the option `--m
-00011760: 696e 7365 676c 656e 6020 616e 6420 602d  inseglen` and `-
-00011770: 2d6d 696e 6765 6e65 6e75 6d60 2e20 5265  -mingenenum`. Re
-00011780: 6475 6e64 616e 7420 6d75 6c74 6970 6c69  dundant multipli
-00011790: 636f 6e73 2063 616e 2062 6520 6b65 7074  cons can be kept
-000117a0: 2062 7920 7365 7420 7468 6520 666c 6167   by set the flag
-000117b0: 206f 7074 696f 6e20 602d 2d6b 6565 7072   option `--keepr
-000117c0: 6564 756e 602e 0a0a 2a2a 4120 7375 6767  edun`...**A sugg
-000117d0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-000117e0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-000117f0: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
-00011800: 6f77 2a2a 0a0a 6060 600a 7767 6420 7379  ow**..```.wgd sy
-00011810: 6e20 6661 6d69 6c69 6573 2067 6666 0a60  n families gff.`
-00011820: 6060 0a0a 2a2a 5468 6520 696e 7465 722d  ``..**The inter-
-00011830: 7370 6563 6966 6963 2073 796e 7465 6e79  specific synteny
-00011840: 2069 6e66 6572 656e 6365 2a2a 0a60 6060   inference**.```
-00011850: 0a77 6764 2073 796e 2066 616d 696c 6965  .wgd syn familie
-00011860: 7320 6766 6631 2067 6666 3220 282d 2d61  s gff1 gff2 (--a
-00011870: 6464 6974 696f 6e61 6c67 6666 696e 666f  dditionalgffinfo
-00011880: 2022 6d52 4e41 3b4e 616d 6522 202d 2d61   "mRNA;Name" --a
-00011890: 6464 6974 696f 6e61 6c67 6666 696e 666f  dditionalgffinfo
-000118a0: 2022 6765 6e65 3b49 4422 290a 6060 600a   "gene;ID").```.
-000118b0: 0a46 6f72 206d 756c 7469 2d73 7065 6369  .For multi-speci
-000118c0: 6573 2073 796e 7465 6e79 2069 6e66 6572  es synteny infer
-000118d0: 656e 6365 2c20 6966 2075 7365 7273 2068  ence, if users h
-000118e0: 6176 6520 6766 6633 2066 696c 6573 2077  ave gff3 files w
-000118f0: 6869 6368 2068 6176 6520 6469 6666 6572  hich have differ
-00011900: 656e 7420 6665 6174 7572 6573 206f 7220  ent features or 
-00011910: 6174 7472 6962 7574 6573 2066 6f72 2067  attributes for g
-00011920: 656e 6520 706f 7369 7469 6f6e 2069 6e66  ene position inf
-00011930: 6f72 6d61 7469 6f6e 2072 6574 7269 6576  ormation retriev
-00011940: 616c 2c20 7468 6520 6f70 7469 6f6e 2060  al, the option `
-00011950: 2d2d 6164 6469 7469 6f6e 616c 6766 6669  --additionalgffi
-00011960: 6e66 6f60 2063 616e 2062 6520 7365 7420  nfo` can be set 
-00011970: 746f 2070 726f 7669 6465 2074 6865 2061  to provide the a
-00011980: 6464 6974 696f 6e61 6c20 696e 666f 726d  dditional inform
-00011990: 6174 696f 6e2e 2054 6865 2072 656d 6169  ation. The remai
-000119a0: 6e69 6e67 2070 6172 616d 6574 6572 2073  ning parameter s
-000119b0: 6574 7469 6e67 2069 7320 7468 6520 7361  etting is the sa
-000119c0: 6d65 2061 7320 7468 6520 696e 7472 612d  me as the intra-
-000119d0: 7370 6563 6966 6963 2073 796e 7465 6e79  specific synteny
-000119e0: 2069 6e66 6572 656e 6365 2e0a 0a2a 2a41   inference...**A
-000119f0: 2073 7567 6765 7374 6564 2073 7461 7274   suggested start
-00011a00: 696e 6720 7275 6e20 6361 6e20 7573 6520  ing run can use 
-00011a10: 636f 6d6d 616e 6420 7369 6d70 6c79 2061  command simply a
-00011a20: 7320 6265 6c6f 772a 2a0a 0a60 6060 0a77  s below**..```.w
-00011a30: 6764 2073 796e 2066 616d 696c 6965 7320  gd syn families 
-00011a40: 6766 6631 2067 6666 320a 6060 600a 0a23  gff1 gff2.```..#
-00011a50: 2323 2077 6764 2076 697a 0a0a 2a2a 5468  ## wgd viz..**Th
-00011a60: 6520 7669 7375 616c 697a 6174 696f 6e20  e visualization 
-00011a70: 6f66 202a 4b2a 3c73 7562 3e53 3c2f 7375  of *K*<sub>S</su
-00011a80: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-00011a90: 696f 6e20 616e 6420 454c 4d4d 2061 6e61  ion and ELMM ana
-00011aa0: 6c79 7369 732a 2a0a 6060 600a 7767 6420  lysis**.```.wgd 
-00011ab0: 7669 7a20 2d64 206b 7364 6174 610a 6060  viz -d ksdata.``
-00011ac0: 600a 0a54 6865 2070 726f 6772 616d 2060  `..The program `
-00011ad0: 7767 6420 7669 7a60 2069 7320 6d61 696e  wgd viz` is main
-00011ae0: 6c79 2066 6f72 2074 6865 2070 7572 706f  ly for the purpo
-00011af0: 7365 206f 6620 2a4b 2a3c 7375 623e 533c  se of *K*<sub>S<
-00011b00: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-00011b10: 6f6e 2061 6e64 2073 796e 7465 6e79 2076  on and synteny v
-00011b20: 6973 7561 6c69 7a61 7469 6f6e 2c20 7769  isualization, wi
-00011b30: 7468 2073 6f6d 6520 6f70 7469 6f6e 616c  th some optional
-00011b40: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
-00011b50: 6720 616e 616c 7973 6973 2e20 5468 6520  g analysis. The 
-00011b60: 6261 7369 6320 6675 6e63 7469 6f6e 2069  basic function i
-00011b70: 7320 6a75 7374 2074 6f20 706c 6f74 2074  s just to plot t
-00011b80: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
-00011b90: 623e 2064 6973 7472 6962 7574 696f 6e20  b> distribution 
-00011ba0: 616e 6420 636f 6e64 7563 7420 616e 2045  and conduct an E
-00011bb0: 4c4d 4d20 616e 616c 7973 6973 2069 6e20  LMM analysis in 
-00011bc0: 7365 6172 6368 206f 6620 706f 7465 6e74  search of potent
-00011bd0: 6961 6c20 5747 4420 636f 6d70 6f6e 656e  ial WGD componen
-00011be0: 7473 2e20 536f 6d65 206b 6579 2070 6172  ts. Some key par
-00011bf0: 616d 6574 6572 7320 6166 6665 6374 696e  ameters affectin
-00011c00: 6720 7468 6520 454c 4d4d 2072 6573 756c  g the ELMM resul
-00011c10: 7420 696e 636c 7564 6520 602d 2d70 726f  t include `--pro
-00011c20: 6d69 6e65 6e63 655f 6375 746f 6666 602c  minence_cutoff`,
-00011c30: 2060 2d2d 7265 6c5f 6865 6967 6874 602c   `--rel_height`,
-00011c40: 2060 2d2d 656d 5f69 7465 7261 7469 6f6e   `--em_iteration
-00011c50: 7360 2061 6e64 2060 2d2d 656d 5f69 6e69  s` and `--em_ini
-00011c60: 7469 616c 697a 6174 696f 6e73 602c 2061  tializations`, a
-00011c70: 6c6c 206f 6620 7768 6963 6820 6861 7665  ll of which have
-00011c80: 2062 6565 6e20 6578 706c 6169 6e65 6420   been explained 
-00011c90: 6168 6561 642e 0a0a 2a2a 4120 7375 6767  ahead...**A sugg
-00011ca0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-00011cb0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-00011cc0: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
-00011cd0: 6f77 2a2a 0a0a 6060 600a 7767 6420 7669  ow**..```.wgd vi
-00011ce0: 7a20 2d64 206b 7364 6174 610a 6060 600a  z -d ksdata.```.
-00011cf0: 0a2a 2a54 6865 2076 6973 7561 6c69 7a61  .**The visualiza
-00011d00: 7469 6f6e 206f 6620 2a4b 2a3c 7375 623e  tion of *K*<sub>
-00011d10: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
-00011d20: 7269 6275 7469 6f6e 2077 6974 6820 7261  ribution with ra
-00011d30: 7465 2063 6f72 7265 6374 696f 6e2a 2a0a  te correction**.
-00011d40: 6060 600a 7767 6420 7669 7a20 2d64 206b  ```.wgd viz -d k
-00011d50: 7364 6174 6120 2d73 7020 7370 6461 7461  sdata -sp spdata
-00011d60: 202d 2d66 6f63 7573 3261 6c6c 2066 6f63   --focus2all foc
-00011d70: 616c 5f73 7065 6369 6573 202d 2d65 7874  al_species --ext
-00011d80: 7261 7061 7261 6e6f 6d65 6b73 206b 7364  raparanomeks ksd
-00011d90: 6174 6120 282d 2d61 6e63 686f 7270 6f69  ata (--anchorpoi
-00011da0: 6e74 7320 6170 6461 7461 202d 2d70 6c6f  nts apdata --plo
-00011db0: 7461 7067 6d6d 202d 2d70 6c6f 7465 6c6d  tapgmm --plotelm
-00011dc0: 6d29 0a60 6060 0a0a 4265 7369 6465 7320  m).```..Besides 
-00011dd0: 7468 6520 6261 7369 6320 2a4b 2a3c 7375  the basic *K*<su
-00011de0: 623e 533c 2f73 7562 3e20 706c 6f74 2c20  b>S</sub> plot, 
-00011df0: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
-00011e00: 6520 636f 7272 6563 7469 6f6e 2063 616e  e correction can
-00011e10: 2061 6c73 6f20 6265 2061 6368 6965 7665   also be achieve
-00011e20: 6420 6769 7665 6e20 6174 206c 6561 7374  d given at least
-00011e30: 2061 2073 7065 6369 6573 2074 7265 6520   a species tree 
-00011e40: 2876 6961 2074 6865 206f 7074 696f 6e20  (via the option 
-00011e50: 602d 2d73 7065 6369 6573 7472 6565 6029  `--speciestree`)
-00011e60: 2061 6e64 2061 2066 6f63 616c 2073 7065   and a focal spe
-00011e70: 6369 6573 2028 6569 7468 6572 2076 6961  cies (either via
-00011e80: 2074 6865 206f 7074 696f 6e20 602d 2d66   the option `--f
-00011e90: 6f63 7573 3261 6c6c 6020 6f72 2076 6961  ocus2all` or via
-00011ea0: 2074 6865 206f 7074 696f 6e20 602d 2d73   the option `--s
-00011eb0: 7061 6972 6020 696e 2074 6865 2066 6f72  pair` in the for
-00011ec0: 6d20 6f66 2022 2466 6f63 616c 5f73 7065  m of "$focal_spe
-00011ed0: 6369 6573 3b24 666f 6361 6c5f 7370 6563  cies;$focal_spec
-00011ee0: 6965 7322 292e 2049 7427 7320 7375 6767  ies"). It's sugg
-00011ef0: 6573 7465 6420 7468 6174 2074 6865 206f  ested that the o
-00011f00: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
-00011f10: 7562 3e53 3c2f 7375 623e 2064 6174 6120  ub>S</sub> data 
-00011f20: 6973 2070 726f 7669 6465 6420 6279 2074  is provided by t
-00011f30: 6865 2060 2d2d 6461 7461 6669 6c65 6020  he `--datafile` 
-00011f40: 6f70 7469 6f6e 2061 6e64 2074 6865 2070  option and the p
-00011f50: 6172 616c 6f67 6f75 7320 2a4b 2a3c 7375  aralogous *K*<su
-00011f60: 623e 533c 2f73 7562 3e20 6461 7461 2069  b>S</sub> data i
-00011f70: 7320 7072 6f76 6964 6564 2062 7920 7468  s provided by th
-00011f80: 6520 602d 2d65 7874 7261 7061 7261 6e6f  e `--extraparano
-00011f90: 6d65 6b73 6020 6f70 7469 6f6e 2c20 616c  meks` option, al
-00011fa0: 7468 6f75 6768 2069 7427 7320 616c 6c6f  though it's allo
-00011fb0: 7765 6420 746f 206f 6e6c 7920 7072 6f76  wed to only prov
-00011fc0: 6964 6520 2a4b 2a3c 7375 623e 533c 2f73  ide *K*<sub>S</s
-00011fd0: 7562 3e20 6461 7461 2076 6961 2074 6865  ub> data via the
-00011fe0: 2060 2d2d 6461 7461 6669 6c65 6020 6f70   `--datafile` op
-00011ff0: 7469 6f6e 2061 6e64 2064 6570 6f73 6974  tion and deposit
-00012000: 2062 6f74 6820 6f72 7468 6f6c 6f67 6f75   both orthologou
-00012010: 7326 7061 7261 6c6f 676f 7573 202a 4b2a  s&paralogous *K*
-00012020: 3c73 7562 3e53 3c2f 7375 623e 2064 6174  <sub>S</sub> dat
-00012030: 6120 7468 6572 656f 6e2e 2054 6865 7265  a thereon. There
-00012040: 2061 7265 2074 776f 2074 7970 6573 206f   are two types o
-00012050: 6620 6d69 7865 6420 706c 6f74 7320 2874  f mixed plots (t
-00012060: 6865 2022 6d69 7865 6422 2068 6572 6520  he "mixed" here 
-00012070: 7265 6665 7273 2074 6f20 6d69 7865 6420  refers to mixed 
-00012080: 6f72 7468 6f6c 6f67 6f75 7320 616e 6420  orthologous and 
-00012090: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
-000120a0: 7562 3e53 3c2f 7375 623e 2064 6973 7472  ub>S</sub> distr
-000120b0: 6962 7574 696f 6e73 292c 206f 6e65 206f  ibutions), one o
-000120c0: 6620 7768 6963 6820 6973 2073 696d 696c  f which is simil
-000120d0: 6172 2074 6f20 7768 6174 2060 6b73 7261  ar to what `ksra
-000120e0: 7465 7360 2070 6c6f 7473 2c20 7768 696c  tes` plots, whil
-000120f0: 6520 7468 6520 6f74 6865 7220 6f66 2077  e the other of w
-00012100: 6869 6368 2069 7320 6c69 6b65 2074 6865  hich is like the
-00012110: 2063 6f6e 7665 6e74 696f 6e61 6c20 2a4b   conventional *K
-00012120: 2a3c 7375 623e 533c 2f73 7562 3e20 706c  *<sub>S</sub> pl
-00012130: 6f74 2074 6861 7420 626f 7468 2074 6865  ot that both the
-00012140: 206f 7269 6769 6e61 6c20 6f72 7468 6f6c   original orthol
-00012150: 6f67 6f75 7320 616e 6420 7061 7261 6c6f  ogous and paralo
-00012160: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
-00012170: 7375 623e 2064 6973 7472 6962 7574 696f  sub> distributio
-00012180: 6e73 2061 7265 2074 7275 7468 6675 6c6c  ns are truthfull
-00012190: 7920 706c 6f74 7465 6420 696e 7374 6561  y plotted instea
-000121a0: 6420 6f66 206a 7573 7420 6265 696e 6720  d of just being 
-000121b0: 7265 7072 6573 656e 7465 6420 6279 2073  represented by s
-000121c0: 6f6d 6520 7665 7274 6963 616c 206c 696e  ome vertical lin
-000121d0: 6573 2e20 5765 2073 7567 6765 7374 2075  es. We suggest u
-000121e0: 7365 7273 2074 6f20 6164 6f70 7420 7468  sers to adopt th
-000121f0: 6520 606b 7372 6174 6573 602d 6c69 6b65  e `ksrates`-like
-00012200: 2070 6c6f 7473 2c20 7768 6963 6820 6973   plots, which is
-00012210: 2074 6865 2064 6566 6175 6c74 2e20 4f74   the default. Ot
-00012220: 6865 7277 6973 6520 7468 6520 6f70 7469  herwise the opti
-00012230: 6f6e 2060 2d2d 636c 6173 7369 6360 2077  on `--classic` w
-00012240: 696c 6c20 6265 206e 6565 6465 6420 746f  ill be needed to
-00012250: 2073 6574 2e20 4578 7472 6120 6d69 7874   set. Extra mixt
-00012260: 7572 6520 6d6f 6465 6c69 6e67 2061 6e61  ure modeling ana
-00012270: 6c79 7369 7320 6361 6e20 6265 2069 6e69  lysis can be ini
-00012280: 7469 6174 6564 2076 6961 2074 6865 206f  tiated via the o
-00012290: 7074 696f 6e20 602d 2d70 6c6f 7465 6c6d  ption `--plotelm
-000122a0: 6d60 2061 6e64 2060 2d2d 706c 6f74 6170  m` and `--plotap
-000122b0: 676d 6d60 2077 6974 6820 7468 6520 616e  gmm` with the an
-000122c0: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
-000122d0: 6669 6c65 2070 726f 7669 6465 6420 6279  file provided by
-000122e0: 2074 6865 2060 2d2d 616e 6368 6f72 706f   the `--anchorpo
-000122f0: 696e 7473 6020 6f70 7469 6f6e 2e0a 0a2a  ints` option...*
-00012300: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
-00012310: 7274 696e 6720 7275 6e20 6361 6e20 7573  rting run can us
-00012320: 6520 636f 6d6d 616e 6420 7369 6d70 6c79  e command simply
-00012330: 2061 7320 6265 6c6f 772a 2a0a 0a60 6060   as below**..```
-00012340: 0a77 6764 2076 697a 202d 6420 6b73 6461  .wgd viz -d ksda
-00012350: 7461 202d 7370 2073 7064 6174 6120 2d2d  ta -sp spdata --
-00012360: 666f 6375 7332 616c 6c20 666f 6361 6c5f  focus2all focal_
-00012370: 7370 6563 6965 7320 2d2d 6578 7472 6170  species --extrap
-00012380: 6172 616e 6f6d 656b 7320 6b73 6461 7461  aranomeks ksdata
-00012390: 0a60 6060 0a0a 2a2a 5468 6520 7669 7375  .```..**The visu
-000123a0: 616c 697a 6174 696f 6e20 6f66 2073 796e  alization of syn
-000123b0: 7465 6e79 2a2a 0a60 6060 0a77 6764 2076  teny**.```.wgd v
-000123c0: 697a 202d 6170 2061 7064 6174 6120 2d73  iz -ap apdata -s
-000123d0: 6d20 736d 6461 7461 202d 6d74 206d 7464  m smdata -mt mtd
-000123e0: 6174 6120 2d67 7420 6774 6461 7461 202d  ata -gt gtdata -
-000123f0: 2d70 6c6f 7473 796e 2028 2d2d 6d69 6e6c  -plotsyn (--minl
-00012400: 656e 202d 3120 2d2d 6d69 6e73 6567 6c65  en -1 --minsegle
-00012410: 6e20 3130 3030 3020 2d2d 6d69 6e67 656e  n 10000 --mingen
-00012420: 656e 756d 2033 3029 0a60 6060 0a0a 436f  enum 30).```..Co
-00012430: 6d70 6172 6564 2074 6f20 7468 6520 6f72  mpared to the or
-00012440: 6967 696e 616c 2060 7767 6460 2c20 7468  iginal `wgd`, th
-00012450: 6520 6077 6764 2076 697a 6020 7072 6f67  e `wgd viz` prog
-00012460: 7261 6d20 6164 6465 6420 7379 6e74 656e  ram added synten
-00012470: 7920 7669 7375 616c 697a 6174 696f 6e20  y visualization 
-00012480: 7069 7065 6c69 6e65 2e20 5573 6572 7320  pipeline. Users 
-00012490: 6e65 6564 2074 6f20 7072 6f76 6964 6520  need to provide 
-000124a0: 7468 6520 666c 6167 206f 7074 696f 6e20  the flag option 
-000124b0: 602d 2d70 6c6f 7473 796e 6020 746f 2069  `--plotsyn` to i
-000124c0: 6e69 7469 6174 6520 7468 6973 2070 6172  nitiate this par
-000124d0: 7420 6f66 2070 6970 656c 696e 652e 2054  t of pipeline. T
-000124e0: 6869 7320 7374 6570 2061 7373 756d 6573  his step assumes
-000124f0: 2074 6861 7420 7573 6572 7320 6861 7665   that users have
-00012500: 206f 6274 6169 6e65 6420 616c 7265 6164   obtained alread
-00012510: 7920 7468 6520 7379 6e74 656e 6963 2072  y the syntenic r
-00012520: 6573 756c 7473 2066 726f 6d20 6069 2d61  esults from `i-a
-00012530: 6468 6f72 6560 2061 6e64 2075 7365 7320  dhore` and uses 
-00012540: 7468 6f73 6520 7265 7375 6c74 2066 696c  those result fil
-00012550: 6573 2074 6f20 7265 616c 697a 6520 7468  es to realize th
-00012560: 6520 7379 6e74 656e 7920 7669 7375 616c  e synteny visual
-00012570: 697a 6174 696f 6e2e 2053 696d 6c69 6172  ization. Simliar
-00012580: 2074 6f20 6077 6764 2073 796e 602c 2074   to `wgd syn`, t
-00012590: 6865 2065 7874 7261 202a 4b2a 3c73 7562  he extra *K*<sub
-000125a0: 3e53 3c2f 7375 623e 2064 6174 6120 6669  >S</sub> data fi
-000125b0: 6c65 2063 616e 2062 6520 7472 616e 736d  le can be transm
-000125c0: 6974 7465 6420 7669 6120 7468 6520 6f70  itted via the op
-000125d0: 7469 6f6e 2060 2d2d 6461 7461 6669 6c65  tion `--datafile
-000125e0: 6020 2869 6e73 7465 6164 206f 6620 602d  ` (instead of `-
-000125f0: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
-00012600: 6020 6f70 7469 6f6e 292e 2042 6173 6963  ` option). Basic
-00012610: 616c 6c79 2c20 7468 6520 7379 6e74 656e  ally, the synten
-00012620: 6963 2072 6573 756c 7420 6669 6c65 7320  ic result files 
-00012630: 7265 7175 6972 6564 2061 7265 2074 6865  required are the
-00012640: 2061 6e63 686f 7220 706f 696e 7473 2064   anchor points d
-00012650: 6174 6166 696c 652c 206d 756c 7469 706c  atafile, multipl
-00012660: 6963 6f6e 7320 6461 7461 6669 6c65 2c20  icons datafile, 
-00012670: 6765 6e65 2074 6162 6c65 2064 6174 6166  gene table dataf
-00012680: 696c 6520 2861 7574 6f6d 6174 656c 7920  ile (automately 
-00012690: 7072 6f64 7563 6564 2062 7920 6077 6764  produced by `wgd
-000126a0: 2073 796e 6029 2c20 616e 6420 7365 676d   syn`), and segm
-000126b0: 656e 7473 2064 6174 6166 696c 652e 0a0a  ents datafile...
-000126c0: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
-000126d0: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
-000126e0: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
-000126f0: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
-00012700: 600a 7767 6420 7669 7a20 2d61 7020 6170  `.wgd viz -ap ap
-00012710: 6461 7461 202d 736d 2073 6d64 6174 6120  data -sm smdata 
-00012720: 2d6d 7420 6d74 6461 7461 202d 6774 2067  -mt mtdata -gt g
-00012730: 7464 6174 6120 2d2d 706c 6f74 7379 6e0a  tdata --plotsyn.
-00012740: 6060 600a 0a23 2320 496c 6c75 7374 7261  ```..## Illustra
-00012750: 7469 6f6e 0a0a 5765 2069 6c6c 7573 7472  tion..We illustr
-00012760: 6174 6520 6f75 7220 7072 6f67 7261 6d20  ate our program 
-00012770: 6f6e 2061 6e20 6578 656d 706c 6172 7920  on an exemplary 
-00012780: 5747 4420 696e 6665 7265 6e63 6520 616e  WGD inference an
-00012790: 6420 6461 7469 6e67 2075 706f 6e20 7370  d dating upon sp
-000127a0: 6563 6965 7320 2a41 7175 696c 6567 6961  ecies *Aquilegia
-000127b0: 2063 6f65 7275 6c65 612a 2e0a 0a54 6865   coerulea*...The
-000127c0: 202a 4171 7569 6c65 6769 6120 636f 6572   *Aquilegia coer
-000127d0: 756c 6561 2a20 7761 7320 7265 706f 7274  ulea* was report
-000127e0: 6564 2074 6f20 6578 7065 7269 656e 6365  ed to experience
-000127f0: 2061 6e20 7061 6c65 6f2d 706f 6c79 706c   an paleo-polypl
-00012800: 6f69 6469 7a61 7469 6f6e 2065 7665 6e74  oidization event
-00012810: 2061 6674 6572 2074 6865 2064 6976 6572   after the diver
-00012820: 6765 6e63 6520 6f66 2063 6f72 6520 6575  gence of core eu
-00012830: 6469 636f 7473 2c20 7768 6963 6820 6973  dicots, which is
-00012840: 206c 696b 656c 7920 7368 6172 6564 2062   likely shared b
-00012850: 7920 616c 6c20 5261 6e75 6e63 756c 616c  y all Ranunculal
-00012860: 6573 2e0a 0a46 6972 7374 2061 626f 7665  es...First above
-00012870: 2061 6c6c 2c20 6c65 7427 7320 6465 6c69   all, let's deli
-00012880: 6e65 6174 6520 7468 6520 7768 6f6c 6520  neate the whole 
-00012890: 7061 7261 6e6f 6d65 202a 4b2a 3c73 7562  paranome *K*<sub
-000128a0: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-000128b0: 7472 6962 7574 696f 6e20 616e 6420 6861  tribution and ha
-000128c0: 7665 2061 2062 6173 6963 206f 6273 6572  ve a basic obser
-000128d0: 7661 7469 6f6e 2066 6f72 2070 6f74 656e  vation for poten
-000128e0: 7469 616c 6c79 2063 6f6e 6365 6976 6162  tially conceivab
-000128f0: 6c65 2057 4744 732c 2075 7369 6e67 2074  le WGDs, using t
-00012900: 6865 2063 6f6d 6d61 6e64 206c 696e 6520  he command line 
-00012910: 6265 6c6f 772e 0a0a 6060 600a 7767 6420  below...```.wgd 
-00012920: 646d 6420 4171 7569 6c65 6769 615f 636f  dmd Aquilegia_co
-00012930: 6572 756c 6561 0a77 6764 206b 7364 2077  erulea.wgd ksd w
-00012940: 6764 5f64 6d64 2f41 7175 696c 6567 6961  gd_dmd/Aquilegia
-00012950: 5f63 6f65 7275 6c65 612e 7473 7620 4171  _coerulea.tsv Aq
-00012960: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00012970: 0a60 6060 0a0a 5468 6520 636f 6e73 7472  .```..The constr
-00012980: 7563 7465 6420 7768 6f6c 6520 7061 7261  ucted whole para
-00012990: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
-000129a0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-000129b0: 7574 696f 6e20 6f66 202a 4171 7569 6c65  ution of *Aquile
-000129c0: 6769 6120 636f 6572 756c 6561 2a20 6973  gia coerulea* is
-000129d0: 2061 7320 6265 6c6f 772c 2077 6520 6361   as below, we ca
-000129e0: 6e20 7365 6520 7468 6174 2074 6865 7265  n see that there
-000129f0: 2073 6565 6d73 2074 6f20 6265 2061 2068   seems to be a h
-00012a00: 756d 7020 6174 202a 4b2a 3c73 7562 3e53  ump at *K*<sub>S
-00012a10: 3c2f 7375 623e 2031 2062 7574 206e 6f74  </sub> 1 but not
-00012a20: 2063 6c65 6172 2e0a 0a21 5b5d 2864 6174   clear...![](dat
-00012a30: 612f 4171 7569 6c65 6769 615f 636f 6572  a/Aquilegia_coer
-00012a40: 756c 6561 2e74 7376 2e6b 7364 5f77 702e  ulea.tsv.ksd_wp.
-00012a50: 7376 6729 0a0a 5765 2074 6865 6e20 636f  svg)..We then co
-00012a60: 6e73 7472 7563 7420 7468 6520 616e 6368  nstruct the anch
-00012a70: 6f72 202a 4b2a 3c73 7562 3e53 3c2f 7375  or *K*<sub>S</su
-00012a80: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-00012a90: 696f 6e20 7573 696e 6720 7468 6520 636f  ion using the co
-00012aa0: 6d6d 616e 6420 6c69 6e65 2062 656c 6f77  mmand line below
-00012ab0: 2e0a 0a60 6060 0a77 6764 2073 796e 202d  ...```.wgd syn -
-00012ac0: 6620 6d52 4e41 202d 6120 4e61 6d65 2077  f mRNA -a Name w
-00012ad0: 6764 5f64 6d64 2f41 7175 696c 6567 6961  gd_dmd/Aquilegia
-00012ae0: 5f63 6f65 7275 6c65 612e 7473 7620 4171  _coerulea.tsv Aq
-00012af0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00012b00: 2e67 6666 3320 2d6b 7320 7767 645f 6b73  .gff3 -ks wgd_ks
-00012b10: 642f 4171 7569 6c65 6769 615f 636f 6572  d/Aquilegia_coer
-00012b20: 756c 6561 2e74 7376 2e6b 732e 7473 760a  ulea.tsv.ks.tsv.
-00012b30: 6060 600a 0a41 7320 7368 6f77 6e20 6265  ```..As shown be
-00012b40: 6c6f 772c 2074 6865 7265 2061 7265 2073  low, there are s
-00012b50: 6f6d 6520 7265 7461 696e 6564 2061 6e63  ome retained anc
-00012b60: 686f 7220 7061 6972 7320 7769 7468 202a  hor pairs with *
-00012b70: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2062  K*<sub>S</sub> b
-00012b80: 6574 7765 656e 2031 2061 6e64 2032 2c20  etween 1 and 2, 
-00012b90: 7768 6963 6820 7365 656d 7320 746f 2073  which seems to s
-00012ba0: 7567 6765 7374 2061 2057 4744 2065 7665  uggest a WGD eve
-00012bb0: 6e74 2e0a 0a21 5b5d 2864 6174 612f 4171  nt...![](data/Aq
-00012bc0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00012bd0: 2e74 7376 2e6b 7364 5f77 705f 6170 2e73  .tsv.ksd_wp_ap.s
-00012be0: 7667 290a 0a54 6865 2061 7373 6f63 6961  vg)..The associa
-00012bf0: 7465 6420 6064 7570 5374 6163 6b60 2070  ted `dupStack` p
-00012c00: 6c6f 7420 7368 6f77 7320 7468 6174 2074  lot shows that t
-00012c10: 6865 7265 2061 7265 206e 756d 6572 6f75  here are numerou
-00012c20: 7320 6475 706c 6963 6174 6564 2073 6567  s duplicated seg
-00012c30: 6d65 6e74 7320 6163 726f 7373 206d 6f73  ments across mos
-00012c40: 7420 6f66 2074 6865 2063 6872 6f6d 6f73  t of the chromos
-00012c50: 6f6d 6573 2e0a 0a21 5b5d 2864 6174 612f  omes...![](data/
-00012c60: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00012c70: 6561 5f41 7175 696c 6567 6961 5f63 6f65  ea_Aquilegia_coe
-00012c80: 7275 6c65 615f 6d75 6c74 6970 6c69 636f  rulea_multiplico
-00012c90: 6e73 5f6c 6576 656c 2e73 7667 290a 0a57  ns_level.svg)..W
-00012ca0: 6520 696d 706c 656d 656e 7465 6420 7477  e implemented tw
-00012cb0: 6f20 7479 7065 7320 6f66 2064 6f74 2070  o types of dot p
-00012cc0: 6c6f 7473 2069 6e20 6f78 666f 7264 2067  lots in oxford g
-00012cd0: 7269 643a 206f 6e65 2069 6e20 7468 6520  rid: one in the 
-00012ce0: 756e 6974 206f 6620 6261 7365 7320 616e  unit of bases an
-00012cf0: 6420 7468 6520 6f74 6865 7220 696e 2074  d the other in t
-00012d00: 6865 2075 6e69 7420 6f66 2067 656e 6573  he unit of genes
-00012d10: 2c20 7768 6963 6820 6361 6e20 6265 2063  , which can be c
-00012d20: 6f6c 6f72 6564 2062 7920 2a4b 2a3c 7375  olored by *K*<su
-00012d30: 623e 533c 2f73 7562 3e20 7661 6c75 6573  b>S</sub> values
-00012d40: 2067 6976 656e 202a 4b2a 3c73 7562 3e53   given *K*<sub>S
-00012d50: 3c2f 7375 623e 2064 6174 612e 0a0a 215b  </sub> data...![
-00012d60: 5d28 6461 7461 2f41 7175 696c 6567 6961  ](data/Aquilegia
-00012d70: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
-00012d80: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
-00012d90: 4b73 2e64 6f74 5f75 6e69 745f 6765 6e65  Ks.dot_unit_gene
-00012da0: 2e70 6e67 290a 0a41 7320 7368 6f77 6e20  .png)..As shown 
-00012db0: 6162 6f76 652c 2074 6865 2064 6f74 2070  above, the dot p
-00012dc0: 6c6f 7420 696e 2074 6865 2075 6e69 7420  lot in the unit 
-00012dd0: 6f66 2067 656e 6573 2070 7265 7365 6e74  of genes present
-00012de0: 7320 6e75 6d65 726f 7573 2064 656e 7365  s numerous dense
-00012df0: 6c79 2061 6767 7265 6761 7465 6420 286c  ly aggregated (l
-00012e00: 696e 652d 6c69 6b65 2920 616e 6368 6f72  ine-like) anchor
-00012e10: 2070 6f69 6e74 7320 6174 206d 6f73 7420   points at most 
-00012e20: 6f66 2074 6865 2063 6872 6f6d 6f73 6f6d  of the chromosom
-00012e30: 6573 2077 6974 6820 636f 6e73 6973 7465  es with consiste
-00012e40: 6e74 202a 4b2a 3c73 7562 3e53 3c2f 7375  nt *K*<sub>S</su
-00012e50: 623e 2061 6765 2062 6574 7765 656e 2031  b> age between 1
-00012e60: 2061 6e64 2032 2e20 5468 6520 646f 7420   and 2. The dot 
-00012e70: 706c 6f74 2069 6e20 7468 6520 756e 6974  plot in the unit
-00012e80: 206f 6620 6261 7365 7320 7368 6f77 7320   of bases shows 
-00012e90: 7468 6520 7361 6d65 2070 6174 7465 726e  the same pattern
-00012ea0: 2c20 6173 206d 616e 6966 6573 7465 6420  , as manifested 
-00012eb0: 6265 6c6f 772e 0a0a 215b 5d28 6461 7461  below...![](data
-00012ec0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
-00012ed0: 6c65 612d 7673 2d41 7175 696c 6567 6961  lea-vs-Aquilegia
-00012ee0: 5f63 6f65 7275 6c65 615f 4b73 2e64 6f74  _coerulea_Ks.dot
-00012ef0: 2e70 6e67 290a 0a54 6865 2064 6f74 2070  .png)..The dot p
-00012f00: 6c6f 7473 2077 6974 686f 7574 202a 4b2a  lots without *K*
-00012f10: 3c73 7562 3e53 3c2f 7375 623e 2061 6e6e  <sub>S</sub> ann
-00012f20: 6f74 6174 696f 6e20 7769 6c6c 2061 6c73  otation will als
-00012f30: 6f20 6265 2061 7574 6f6d 6174 656c 7920  o be automately 
-00012f40: 7072 6f64 7563 6564 2c20 6173 2073 686f  produced, as sho
-00012f50: 776e 2062 656c 6f77 2e0a 0a21 5b5d 2864  wn below...![](d
-00012f60: 6174 612f 4171 7569 6c65 6769 615f 636f  ata/Aquilegia_co
-00012f70: 6572 756c 6561 2d76 732d 4171 7569 6c65  erulea-vs-Aquile
-00012f80: 6769 615f 636f 6572 756c 6561 2e64 6f74  gia_coerulea.dot
-00012f90: 5f75 6e69 745f 6765 6e65 2e70 6e67 290a  _unit_gene.png).
-00012fa0: 0a21 5b5d 2864 6174 612f 4171 7569 6c65  .![](data/Aquile
-00012fb0: 6769 615f 636f 6572 756c 6561 2d76 732d  gia_coerulea-vs-
-00012fc0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00012fd0: 6561 2e64 6f74 2e70 6e67 290a 0a4e 6f74  ea.dot.png)..Not
-00012fe0: 6520 7468 6174 2074 6865 206f 7061 6369  e that the opaci
-00012ff0: 7479 206f 6620 616e 6368 6f72 2064 6f74  ty of anchor dot
-00013000: 7320 616e 6420 616c 6c20 686f 6d6f 6c6f  s and all homolo
-00013010: 6720 646f 7473 2063 616e 2062 6520 7365  g dots can be se
-00013020: 7420 6279 2074 6865 206f 7074 696f 6e20  t by the option 
-00013030: 6061 7061 6c70 6861 6020 616e 6420 6068  `apalpha` and `h
-00013040: 6f61 6c70 6861 6020 7365 7061 7261 7465  oalpha` separate
-00013050: 6c79 2e20 4966 206f 6e65 206a 7573 7420  ly. If one just 
-00013060: 7761 6e74 7320 746f 2073 6565 2074 6865  wants to see the
-00013070: 2061 6e63 686f 7220 646f 7473 2c20 7365   anchor dots, se
-00013080: 7474 696e 6720 7468 6520 6068 6f61 6c70  tting the `hoalp
-00013090: 6861 6020 6173 2030 2028 6f72 206f 7468  ha` as 0 (or oth
-000130a0: 6572 206d 696e 7573 6375 6c65 2076 616c  er minuscule val
-000130b0: 7565 7329 2077 696c 6c20 646f 2e20 4966  ues) will do. If
-000130c0: 206f 6e65 2077 616e 7473 2074 6f20 7365   one wants to se
-000130d0: 6520 7468 6520 6469 7374 7269 6275 7469  e the distributi
-000130e0: 6f6e 206f 6620 7768 6f6c 6520 646f 7473  on of whole dots
-000130f0: 2062 6574 7465 722c 2073 6574 7469 6e67   better, setting
-00013100: 2074 6865 2060 686f 616c 7068 6160 2068   the `hoalpha` h
-00013110: 6967 6865 7220 2861 6e64 2060 6170 616c  igher (and `apal
-00013120: 7068 6160 206c 6f77 6572 2920 7769 6c6c  pha` lower) will
-00013130: 2064 6f2e 2054 6865 2060 646f 7473 697a   do. The `dotsiz
-00013140: 6560 206f 7074 696f 6e20 6361 6e20 6265  e` option can be
-00013150: 2063 616c 6c65 6420 746f 2061 646a 7573   called to adjus
-00013160: 7420 7468 6520 7369 7a65 206f 6620 646f  t the size of do
-00013170: 7473 2e0a 0a41 2066 7572 7468 6572 2061  ts...A further a
-00013180: 7373 6f63 6961 7465 6420 5379 6e64 6570  ssociated Syndep
-00013190: 7468 2070 6c6f 7420 7368 6f77 7320 7468  th plot shows th
-000131a0: 6174 2074 6865 7265 2061 7265 206d 6f72  at there are mor
-000131b0: 6520 7468 616e 2038 3020 6475 706c 6963  e than 80 duplic
-000131c0: 6174 6564 2073 6567 6d65 6e74 732c 2077  ated segments, w
-000131d0: 6869 6368 2064 6f6d 696e 6174 6573 2074  hich dominates t
-000131e0: 6865 2077 686f 6c65 2063 6f6c 6c69 6e65  he whole colline
-000131f0: 6172 2072 6174 696f 2063 6174 6567 6f72  ar ratio categor
-00013200: 792e 0a0a 215b 5d28 6461 7461 2f53 796e  y...![](data/Syn
-00013210: 6465 7074 682e 7376 6729 0a0a 5765 2063  depth.svg)..We c
-00013220: 616e 2066 6974 2061 6e20 454c 4d4d 206d  an fit an ELMM m
-00013230: 6978 7475 7265 206d 6f64 656c 2075 706f  ixture model upo
-00013240: 6e20 7468 6520 7768 6f6c 6520 7061 7261  n the whole para
-00013250: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
-00013260: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-00013270: 7574 696f 6e20 746f 2073 6565 206d 6f72  ution to see mor
-00013280: 6520 6163 6375 7261 7465 6c79 2074 6865  e accurately the
-00013290: 2073 6967 6e69 6669 6361 6e63 6520 616e   significance an
-000132a0: 6420 6c6f 6361 7469 6f6e 206f 6620 706f  d location of po
-000132b0: 7465 6e74 6961 6c20 5747 4473 2c20 7573  tential WGDs, us
-000132c0: 696e 6720 7468 6520 636f 6d6d 616e 6420  ing the command 
-000132d0: 6c69 6e65 2062 656c 6f77 2e0a 0a60 6060  line below...```
-000132e0: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
-000132f0: 6b73 642f 4171 7569 6c65 6769 615f 636f  ksd/Aquilegia_co
-00013300: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
-00013310: 760a 6060 600a 0a54 6865 2072 6573 756c  v.```..The resul
-00013320: 7420 6f66 2045 4c4d 4d20 6d69 7874 7572  t of ELMM mixtur
-00013330: 6520 6d6f 6465 6c20 636c 7573 7465 7269  e model clusteri
-00013340: 6e67 2073 686f 7773 2074 6861 7420 7468  ng shows that th
-00013350: 6572 6520 6973 2061 206c 696b 656c 7920  ere is a likely 
-00013360: 5747 4420 636f 6d70 6f6e 656e 7420 6174  WGD component at
-00013370: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00013380: 2031 2e31 392e 0a0a 215b 5d28 6461 7461   1.19...![](data
-00013390: 2f65 6c6d 6d5f 4171 7569 6c65 6769 615f  /elmm_Aquilegia_
-000133a0: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
-000133b0: 7473 765f 6265 7374 5f6d 6f64 656c 735f  tsv_best_models_
-000133c0: 7765 6967 6874 6564 2e73 7667 290a 0a4c  weighted.svg)..L
-000133d0: 6574 2773 2064 6f20 6120 6d69 7874 7572  et's do a mixtur
-000133e0: 6520 6d6f 6465 6c20 636c 7573 7465 7269  e model clusteri
-000133f0: 6e67 2066 6f72 2061 6e63 686f 7220 2a4b  ng for anchor *K
-00013400: 2a3c 7375 623e 533c 2f73 7562 3e20 746f  *<sub>S</sub> to
-00013410: 6f2c 2075 7369 6e67 2074 6865 2063 6f6d  o, using the com
-00013420: 6d61 6e64 206c 696e 6520 6265 6c6f 772e  mand line below.
-00013430: 0a0a 6060 600a 7767 6420 7065 616b 2077  ..```.wgd peak w
-00013440: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-00013450: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-00013460: 2e74 7376 202d 2d61 6e63 686f 7270 6f69  .tsv --anchorpoi
-00013470: 6e74 7320 7767 645f 7379 6e2f 6961 6468  nts wgd_syn/iadh
-00013480: 6f72 652d 6f75 742f 616e 6368 6f72 706f  ore-out/anchorpo
-00013490: 696e 7473 2e74 7874 202d 2d73 6567 6d65  ints.txt --segme
-000134a0: 6e74 7320 7767 645f 7379 6e2f 6961 6468  nts wgd_syn/iadh
-000134b0: 6f72 652d 6f75 742f 7365 676d 656e 7473  ore-out/segments
-000134c0: 2e74 7874 202d 2d6c 6973 7465 6c65 6d65  .txt --listeleme
-000134d0: 6e74 7320 7767 645f 7379 6e2f 6961 6468  nts wgd_syn/iadh
-000134e0: 6f72 652d 6f75 742f 6c69 7374 5f65 6c65  ore-out/list_ele
-000134f0: 6d65 6e74 732e 7478 7420 2d2d 6d75 6c74  ments.txt --mult
-00013500: 6970 6c69 636f 6e70 6169 7273 2077 6764  ipliconpairs wgd
-00013510: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
-00013520: 2f6d 756c 7469 706c 6963 6f6e 5f70 6169  /multiplicon_pai
-00013530: 7273 2e74 7874 202d 2d77 6569 6768 7465  rs.txt --weighte
-00013540: 640a 6060 600a 0a54 6865 2061 6e63 686f  d.```..The ancho
-00013550: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
-00013560: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
-00013570: 6f6e 2061 6c73 6f20 6861 7320 6120 6c69  on also has a li
-00013580: 6b65 6c79 2057 4744 2063 6f6d 706f 6e65  kely WGD compone
-00013590: 6e74 2077 6974 6820 6d6f 6465 2031 2e32  nt with mode 1.2
-000135a0: 382e 0a0a 215b 5d28 6461 7461 2f4f 7269  8...![](data/Ori
-000135b0: 6769 6e61 6c5f 416e 6368 6f72 4b73 5f47  ginal_AnchorKs_G
-000135c0: 4d4d 5f43 6f6d 706f 6e65 6e74 335f 6e6f  MM_Component3_no
-000135d0: 6465 5f77 6569 6768 7465 645f 4c6f 676e  de_weighted_Logn
-000135e0: 6f72 6d61 6c2e 7376 6729 0a0a 4e6f 7720  ormal.svg)..Now 
-000135f0: 7468 6174 2077 6520 6861 7665 2073 6565  that we have see
-00013600: 6e20 7468 6520 6576 6964 656e 6365 206f  n the evidence o
-00013610: 6620 6e75 6d65 726f 7573 2064 7570 6c69  f numerous dupli
-00013620: 6361 7465 6420 7365 676d 656e 7473 2061  cated segments a
-00013630: 6e64 2074 6865 2061 6767 7265 6761 7469  nd the aggregati
-00013640: 6f6e 206f 6620 6475 706c 6963 6174 6573  on of duplicates
-00013650: 2061 6765 2069 6e20 2a4b 2a3c 7375 623e   age in *K*<sub>
-00013660: 533c 2f73 7562 3e20 6172 6f75 6e64 2031  S</sub> around 1
-00013670: 2e32 2066 6f72 2061 6e63 686f 7220 7061  .2 for anchor pa
-00013680: 6972 7320 616e 6420 6e6f 6e2d 616e 6368  irs and non-anch
-00013690: 6f72 2070 6169 7273 2074 6872 6f75 6768  or pairs through
-000136a0: 6f75 7420 7468 6520 7768 6f6c 6520 6765  out the whole ge
-000136b0: 6e6f 6d65 2e20 5765 2063 616e 2063 6c61  nome. We can cla
-000136c0: 696d 2077 6974 6820 736f 6d65 2063 6f6e  im with some con
-000136d0: 6669 6465 6e63 6520 7468 6174 202a 4171  fidence that *Aq
-000136e0: 7569 6c65 6769 6120 636f 6572 756c 6561  uilegia coerulea
-000136f0: 2a20 6d69 6768 7420 6861 7665 2065 7870  * might have exp
-00013700: 6572 6965 6e63 6564 2061 2070 616c 656f  erienced a paleo
-00013710: 2d70 6f6c 7970 6c6f 6964 697a 6174 696f  -polyploidizatio
-00013720: 6e20 6576 656e 742e 204e 6578 742c 204c  n event. Next, L
-00013730: 6574 2773 2068 6176 6520 6120 6675 7274  et's have a furt
-00013740: 6865 7220 6c6f 6f6b 2061 626f 7574 2069  her look about i
-00013750: 7473 2070 6879 6c6f 6765 6e65 7469 6320  ts phylogenetic 
-00013760: 6c6f 6361 7469 6f6e 2e20 5765 206b 6e6f  location. We kno
-00013770: 7720 7468 6174 2074 6865 7265 2061 7265  w that there are
-00013780: 2075 6e63 6572 7461 696e 7469 6573 2061   uncertainties a
-00013790: 626f 7574 2077 6865 7468 6572 2074 6869  bout whether thi
-000137a0: 7320 7075 7461 7469 7665 2070 616c 656f  s putative paleo
-000137b0: 2d70 6f6c 7970 6c6f 6964 697a 6174 696f  -polyploidizatio
-000137c0: 6e20 6576 656e 7420 6973 2073 6861 7265  n event is share
-000137d0: 6420 7769 7468 2061 6c6c 2065 7564 6963  d with all eudic
-000137e0: 6f74 7320 6f72 206e 6f74 2e20 5765 2063  ots or not. We c
-000137f0: 616e 2063 686f 6f73 6520 736f 6d65 206f  an choose some o
-00013800: 7468 6572 2065 7564 6963 6f74 2067 656e  ther eudicot gen
-00013810: 6f6d 6573 2074 6f20 7365 6520 7468 6520  omes to see the 
-00013820: 6f72 6465 7269 6e67 206f 6620 7370 6563  ordering of spec
-00013830: 6961 7469 6f6e 2061 6e64 2070 6f6c 7970  iation and polyp
-00013840: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
-00013850: 7473 2e20 4865 7265 2077 6520 6368 6f6f  ts. Here we choo
-00013860: 7365 202a 5669 7469 7320 7669 6e69 6665  se *Vitis vinife
-00013870: 7261 2a2c 202a 5072 6f74 6561 2063 796e  ra*, *Protea cyn
-00013880: 6172 6f69 6465 732a 2061 6e64 202a 4163  aroides* and *Ac
-00013890: 6f72 7573 2061 6d65 7269 6361 6e75 732a  orus americanus*
-000138a0: 2069 6e20 7468 6520 666f 6c6c 6f77 696e   in the followin
-000138b0: 6720 2a4b 2a3c 7375 623e 533c 2f73 7562  g *K*<sub>S</sub
-000138c0: 3e20 616e 616c 7973 6973 2e20 4669 7273  > analysis. Firs
-000138d0: 742c 2077 6520 6275 696c 7420 6120 676c  t, we built a gl
-000138e0: 6f62 616c 204d 5242 4820 6661 6d69 6c79  obal MRBH family
-000138f0: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-00013900: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
-00013910: 6764 2064 6d64 202d 2d67 6c6f 6261 6c6d  gd dmd --globalm
-00013920: 7262 6820 4171 7569 6c65 6769 615f 636f  rbh Aquilegia_co
-00013930: 6572 756c 6561 2050 726f 7465 615f 6379  erulea Protea_cy
-00013940: 6e61 726f 6964 6573 2041 636f 7275 735f  naroides Acorus_
-00013950: 616d 6572 6963 616e 7573 2056 6974 6973  americanus Vitis
-00013960: 5f76 696e 6966 6572 6120 2d6f 2077 6764  _vinifera -o wgd
-00013970: 5f67 6c6f 6261 6c6d 7262 680a 6060 600a  _globalmrbh.```.
-00013980: 0a49 6e20 7468 6520 676c 6f62 616c 204d  .In the global M
-00013990: 5242 4820 6661 6d69 6c79 2c20 6576 6572  RBH family, ever
-000139a0: 7920 7061 6972 206f 6620 6f72 7468 6f6c  y pair of orthol
-000139b0: 6f67 6f75 7320 6765 6e65 7320 6973 2074  ogous genes is t
-000139c0: 6865 2072 6563 6970 726f 6361 6c20 6265  he reciprocal be
-000139d0: 7374 2068 6974 2c20 7375 6767 6573 7469  st hit, suggesti
-000139e0: 6e67 2074 7275 6520 6f72 7468 6f6c 6f67  ng true ortholog
-000139f0: 6f75 7320 7265 6c61 7469 6f6e 7368 6970  ous relationship
-00013a00: 732e 2057 6520 776f 756c 6420 7573 6520  s. We would use 
-00013a10: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
-00013a20: 7562 3e20 7661 6c75 6573 2061 7373 6f63  ub> values assoc
-00013a30: 6961 7465 6420 7769 7468 2074 6865 7365  iated with these
-00013a40: 206f 7274 686f 6c6f 676f 7573 2070 6169   orthologous pai
-00013a50: 7273 2074 6f20 6465 6c69 6d69 7420 7468  rs to delimit th
-00013a60: 6520 6469 7665 7267 656e 6365 202a 4b2a  e divergence *K*
-00013a70: 3c73 7562 3e53 3c2f 7375 623e 2070 6561  <sub>S</sub> pea
-00013a80: 6b2e 2054 6f67 6574 6865 7220 7769 7468  k. Together with
-00013a90: 2074 6865 2077 686f 6c65 2070 6172 616e   the whole paran
-00013aa0: 6f6d 6520 2a4b 2a3c 7375 623e 533c 2f73  ome *K*<sub>S</s
-00013ab0: 7562 3e20 6469 7374 7269 6275 7469 6f6e  ub> distribution
-00013ac0: 2c20 7765 2063 6f6e 6475 6374 2074 6865  , we conduct the
-00013ad0: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-00013ae0: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-00013af0: 6e64 2062 656c 6f77 2e0a 0a21 2153 696e  nd below...!!Sin
-00013b00: 6365 2060 7767 6460 2076 6572 7369 6f6e  ce `wgd` version
-00013b10: 2032 2e30 2e32 342c 2077 6520 7265 7772   2.0.24, we rewr
-00013b20: 6f74 6520 6120 636c 6561 6e65 7220 616e  ote a cleaner an
-00013b30: 6420 7175 6963 6b65 7220 7761 7920 6f66  d quicker way of
-00013b40: 2064 6f69 6e67 2073 7562 7374 6974 7574   doing substitut
-00013b50: 696f 6e20 7261 7465 2063 6f72 7265 6374  ion rate correct
-00013b60: 696f 6e2e 2049 7427 7320 6e6f 7420 7265  ion. It's not re
-00013b70: 7175 6972 6564 2074 6f20 7479 7065 2069  quired to type i
-00013b80: 6e20 616e 7920 7370 6569 6365 7320 7061  n any speices pa
-00013b90: 6972 2061 6e64 2061 2073 6572 6965 7320  ir and a series 
-00013ba0: 6f66 202a 4b2a 3c73 7562 3e53 3c2f 7375  of *K*<sub>S</su
-00013bb0: 623e 2070 6c6f 7473 2077 696c 6c20 6265  b> plots will be
-00013bc0: 2070 726f 6475 6365 642e 2054 6865 2072   produced. The r
-00013bd0: 6571 7569 7265 6420 6669 6c65 7320 6172  equired files ar
-00013be0: 6520 6f72 7468 6f6c 6f67 6f75 7320 2a4b  e orthologous *K
-00013bf0: 2a3c 7375 623e 533c 2f73 7562 3e20 7461  *<sub>S</sub> ta
-00013c00: 626c 652c 2070 6172 616c 6f67 6f75 7320  ble, paralogous 
-00013c10: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013c20: 7461 626c 652c 2061 2073 7065 6369 6573  table, a species
-00013c30: 2074 7265 6520 616e 6420 6120 666f 6375   tree and a focu
-00013c40: 7365 6420 7370 6563 6965 7320 2874 6865  sed species (the
-00013c50: 206f 6e65 2069 6e70 7574 7465 6420 7769   one inputted wi
-00013c60: 7468 2070 6172 616c 6f67 6f75 7320 2a4b  th paralogous *K
-00013c70: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
-00013c80: 7461 292e 2055 7365 7273 2063 616e 2063  ta). Users can c
-00013c90: 686f 6f73 6520 746f 2061 6464 206f 6e65  hoose to add one
-00013ca0: 206d 6f72 6520 6c61 7965 7220 6f66 2065   more layer of e
-00013cb0: 6c6d 6d20 6d6f 6465 6c69 6e67 206f 6e20  lmm modeling on 
-00013cc0: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
-00013cd0: 7562 3e53 3c2f 7375 623e 2076 616c 7565  ub>S</sub> value
-00013ce0: 7320 616e 642f 6f72 2067 6d6d 206d 6f64  s and/or gmm mod
-00013cf0: 656c 696e 6720 6f6e 2061 6e63 686f 7220  eling on anchor 
-00013d00: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013d10: 7661 6c75 6573 2e20 5468 6520 6f72 7468  values. The orth
-00013d20: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
-00013d30: 533c 2f73 7562 3e20 7661 6c75 6573 2063  S</sub> values c
-00013d40: 616e 2062 6520 6361 6c63 756c 6174 6564  an be calculated
-00013d50: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-00013d60: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
-00013d70: 6764 206b 7364 2077 6764 5f67 6c6f 6261  gd ksd wgd_globa
-00013d80: 6c6d 7262 682f 676c 6f62 616c 5f4d 5242  lmrbh/global_MRB
-00013d90: 482e 7473 7620 7365 7173 2a20 2d6f 2077  H.tsv seqs* -o w
-00013da0: 6764 5f67 6c6f 6261 6c6d 7262 685f 6b73  gd_globalmrbh_ks
-00013db0: 0a60 6060 0a0a 5769 7468 2074 6865 2063  .```..With the c
-00013dc0: 616c 6375 6c61 7465 6420 6f72 7468 6f6c  alculated orthol
-00013dd0: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
-00013de0: 2f73 7562 3e20 7461 626c 652c 2077 6520  /sub> table, we 
-00013df0: 6361 6e20 7573 6520 7468 6520 636f 6d6d  can use the comm
-00013e00: 616e 6420 6265 6c6f 7720 746f 2063 6f6e  and below to con
-00013e10: 6475 6374 2074 6865 2072 6174 6520 636f  duct the rate co
-00013e20: 7272 6563 7469 6f6e 2061 6e64 2f6f 7220  rrection and/or 
-00013e30: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
-00013e40: 2061 6e61 6c79 7369 732e 0a0a 6060 600a   analysis...```.
-00013e50: 7767 6420 7669 7a20 2d64 2077 6764 5f67  wgd viz -d wgd_g
-00013e60: 6c6f 6261 6c6d 7262 685f 6b73 2f67 6c6f  lobalmrbh_ks/glo
-00013e70: 6261 6c5f 4d52 4248 2e74 7376 2e6b 732e  bal_MRBH.tsv.ks.
-00013e80: 7473 7620 2d66 6120 4171 7569 6c65 6769  tsv -fa Aquilegi
-00013e90: 615f 636f 6572 756c 6561 202d 6570 6b20  a_coerulea -epk 
-00013ea0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00013eb0: 6561 2e6b 732e 7473 7620 2d61 7020 616e  ea.ks.tsv -ap an
-00013ec0: 6368 6f72 706f 696e 7473 2e74 7874 202d  chorpoints.txt -
-00013ed0: 7370 2073 7065 6369 6573 7472 6565 2e6e  sp speciestree.n
-00013ee0: 7720 2d6f 2077 6764 5f76 697a 5f6d 6978  w -o wgd_viz_mix
-00013ef0: 6564 5f4b 7320 2d2d 706c 6f74 656c 6d6d  ed_Ks --plotelmm
-00013f00: 202d 2d70 6c6f 7461 7067 6d6d 0a60 6060   --plotapgmm.```
-00013f10: 0a0a 6060 600a 7767 6420 6b73 6420 7767  ..```.wgd ksd wg
-00013f20: 645f 676c 6f62 616c 6d72 6268 2f67 6c6f  d_globalmrbh/glo
-00013f30: 6261 6c5f 4d52 4248 2e74 7376 2073 6571  bal_MRBH.tsv seq
-00013f40: 732a 202d 2d65 7874 7261 7061 7261 6e6f  s* --extraparano
-00013f50: 6d65 6b73 2077 6764 5f6b 7364 2f41 7175  meks wgd_ksd/Aqu
-00013f60: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
-00013f70: 7473 762e 6b73 2e74 7376 202d 7370 2073  tsv.ks.tsv -sp s
-00013f80: 7065 6369 6573 7472 6565 2e6e 7720 2d2d  peciestree.nw --
-00013f90: 7265 7765 6967 6874 202d 6f20 7767 645f  reweight -o wgd_
-00013fa0: 676c 6f62 616c 6d72 6268 5f6b 7320 2d2d  globalmrbh_ks --
-00013fb0: 7370 6169 7220 2241 7175 696c 6567 6961  spair "Aquilegia
-00013fc0: 5f63 6f65 7275 6c65 613b 5072 6f74 6561  _coerulea;Protea
-00013fd0: 5f63 796e 6172 6f69 6465 7322 202d 2d73  _cynaroides" --s
-00013fe0: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
-00013ff0: 636f 6572 756c 6561 3b56 6974 6973 5f76  coerulea;Vitis_v
-00014000: 696e 6966 6572 6122 202d 2d73 7061 6972  inifera" --spair
-00014010: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
-00014020: 756c 6561 3b41 636f 7275 735f 616d 6572  ulea;Acorus_amer
-00014030: 6963 616e 7573 2220 2d2d 7370 6169 7220  icanus" --spair 
-00014040: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-00014050: 6c65 613b 4171 7569 6c65 6769 615f 636f  lea;Aquilegia_co
-00014060: 6572 756c 6561 2220 2d2d 706c 6f74 6b64  erulea" --plotkd
-00014070: 6520 282d 6170 2077 6764 5f73 796e 2f69  e (-ap wgd_syn/i
-00014080: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
-00014090: 7270 6f69 6e74 732e 7478 7429 0a60 6060  rpoints.txt).```
-000140a0: 0a0a 5468 6520 6669 6c65 2060 7370 6563  ..The file `spec
-000140b0: 6965 7374 7265 652e 6e77 6020 6973 2074  iestree.nw` is t
-000140c0: 6865 2074 6578 7420 6669 6c65 206f 6620  he text file of 
-000140d0: 7370 6563 6965 7320 7472 6565 2069 6e20  species tree in 
-000140e0: 6e65 7769 636b 2074 6861 7420 7261 7465  newick that rate
-000140f0: 2063 6f72 7265 6374 696f 6e20 776f 756c   correction woul
-00014100: 6420 6265 2063 6f6e 6475 6374 6564 206f  d be conducted o
-00014110: 6e2e 2049 7473 2063 6f6e 7465 6e74 2069  n. Its content i
-00014120: 7320 6173 2062 656c 6f77 2e20 5573 6572  s as below. User
-00014130: 7320 6e65 6564 2074 6f20 7072 6f76 6964  s need to provid
-00014140: 6520 7468 6520 7370 6563 6965 7320 7061  e the species pa
-00014150: 6972 7320 746f 2062 6520 706c 6f74 7465  irs to be plotte
-00014160: 642e 2057 6520 7375 6767 6573 7420 6164  d. We suggest ad
-00014170: 6469 6e67 2074 6865 206f 7074 696f 6e20  ding the option 
-00014180: 602d 2d72 6577 6569 6768 7460 2074 6f20  `--reweight` to 
-00014190: 7265 6361 6c63 756c 6174 6520 7468 6520  recalculate the 
-000141a0: 7765 6967 6874 2070 6572 2073 7065 6369  weight per speci
-000141b0: 6573 2070 6169 7220 7375 6368 2074 6861  es pair such tha
-000141c0: 7420 7468 6520 7765 6967 6874 206f 6620  t the weight of 
-000141d0: 6f72 7468 6f6c 6f67 6f75 7320 6765 6e65  orthologous gene
-000141e0: 2070 6169 7273 2077 696c 6c20 6265 636f   pairs will beco
-000141f0: 6d65 2031 2061 7320 7468 6520 7061 7261  me 1 as the para
-00014200: 6c6f 676f 7573 2067 656e 6520 7061 6972  logous gene pair
-00014210: 732e 2054 6865 2066 6c61 6720 602d 2d70  s. The flag `--p
-00014220: 6c6f 746b 6465 6020 6361 6e20 6265 2061  lotkde` can be a
-00014230: 6464 6564 2077 6865 6e20 7468 6520 6b64  dded when the kd
-00014240: 6520 6375 7276 6520 6f66 206f 7274 686f  e curve of ortho
-00014250: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-00014260: 3c2f 7375 623e 2069 7320 6465 7369 7265  </sub> is desire
-00014270: 642e 2045 7874 7261 2063 6f6c 6c69 6e65  d. Extra colline
-00014280: 6172 2064 6174 6120 6361 6e20 6265 2061  ar data can be a
-00014290: 6464 6564 2062 7920 7468 6520 6f70 7469  dded by the opti
-000142a0: 6f6e 2060 2d61 7060 2e0a 0a60 6060 0a28  on `-ap`...```.(
-000142b0: 2828 5669 7469 735f 7669 6e69 6665 7261  ((Vitis_vinifera
-000142c0: 2c50 726f 7465 615f 6379 6e61 726f 6964  ,Protea_cynaroid
-000142d0: 6573 292c 4171 7569 6c65 6769 615f 636f  es),Aquilegia_co
-000142e0: 6572 756c 6561 292c 4163 6f72 7573 5f61  erulea),Acorus_a
-000142f0: 6d65 7269 6361 6e75 7329 3b0a 6060 600a  mericanus);.```.
-00014300: 0a21 5b5d 2864 6174 612f 4171 7569 6c65  .![](data/Aquile
-00014310: 6769 615f 636f 6572 756c 6561 5f47 6c6f  gia_coerulea_Glo
-00014320: 6261 6c6d 7262 684b 735f 436f 7272 6563  balmrbhKs_Correc
-00014330: 7465 642e 6b73 642e 7376 6729 0a0a 4173  ted.ksd.svg)..As
-00014340: 2073 686f 776e 2061 626f 7665 2c20 6265   shown above, be
-00014350: 6361 7573 6520 6f66 2074 6865 2068 6967  cause of the hig
-00014360: 6865 7220 7375 6273 7469 7475 7469 6f6e  her substitution
-00014370: 2072 6174 6520 6f66 202a 4171 7569 6c65   rate of *Aquile
-00014380: 6769 6120 636f 6572 756c 6561 2a2c 2074  gia coerulea*, t
-00014390: 6865 206f 7269 6769 6e61 6c20 6f72 7468  he original orth
-000143a0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
-000143b0: 533c 2f73 7562 3e20 7661 6c75 6573 2077  S</sub> values w
-000143c0: 6572 6520 6163 7475 616c 6c79 2075 6e64  ere actually und
-000143d0: 6572 6573 7469 6d61 7465 6420 696e 2074  erestimated in t
-000143e0: 6865 2074 696d 652d 6672 616d 6520 6f66  he time-frame of
-000143f0: 202a 4171 7569 6c65 6769 6120 636f 6572   *Aquilegia coer
-00014400: 756c 6561 2a2e 2057 6865 6e20 7765 2072  ulea*. When we r
-00014410: 6563 6f76 6572 6564 2074 6865 2064 6976  ecovered the div
-00014420: 6572 6765 6e63 6520 7375 6273 7469 7475  ergence substitu
-00014430: 7469 6f6e 2064 6973 7461 6e63 6520 696e  tion distance in
-00014440: 2074 6572 6d73 206f 6620 7477 6f20 7469   terms of two ti
-00014450: 6d65 7320 6f66 2074 6865 2062 7261 6e63  mes of the branc
-00014460: 682d 7370 6563 6966 6963 2063 6f6e 7472  h-specific contr
-00014470: 6962 7574 696f 6e20 6f66 202a 4171 7569  ibution of *Aqui
-00014480: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
-00014490: 7369 6e63 6520 6974 7320 6469 7665 7267  since its diverg
-000144a0: 656e 6365 2077 6974 6820 7468 6520 7369  ence with the si
-000144b0: 7374 6572 2073 7065 6369 6573 2070 6c75  ster species plu
-000144c0: 7320 7468 6520 7368 6172 6564 2073 7562  s the shared sub
-000144d0: 7374 6974 7574 696f 6e20 6469 7374 616e  stitution distan
-000144e0: 6365 2062 6566 6f72 6520 6469 7665 7267  ce before diverg
-000144f0: 656e 6365 2028 696e 2072 656c 6174 6976  ence (in relativ
-00014500: 6520 746f 2074 6865 206f 7574 6772 6f75  e to the outgrou
-00014510: 7029 2c20 7468 6520 636f 7272 6563 7465  p), the correcte
-00014520: 6420 2a4b 2a3c 7375 623e 533c 2f73 7562  d *K*<sub>S</sub
-00014530: 3e20 6d6f 6465 2062 6563 616d 6520 6c61  > mode became la
-00014540: 7267 6572 2e0a 0a49 6620 6f6e 6520 6861  rger...If one ha
-00014550: 6420 7468 6520 6f72 7468 6f6c 6f67 6f75  d the orthologou
-00014560: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
-00014570: 3e20 6461 7461 2061 6c72 6561 6479 2c20  > data already, 
-00014580: 6f6e 6520 636f 756c 6420 616c 736f 2061  one could also a
-00014590: 7070 6c79 2074 6865 2070 726f 6772 616d  pply the program
-000145a0: 2060 7767 6420 7669 7a60 2074 6f20 636f   `wgd viz` to co
-000145b0: 6e64 7563 7420 7468 6520 7261 7465 2063  nduct the rate c
-000145c0: 6f72 7265 6374 696f 6e20 616e 616c 7973  orrection analys
-000145d0: 6973 2075 7369 6e67 2074 6865 2063 6f6d  is using the com
-000145e0: 6d61 6e64 2062 656c 6f77 2e20 4e6f 7465  mand below. Note
-000145f0: 2074 6861 7420 7468 6520 6f72 6465 7220   that the order 
-00014600: 6f66 2067 6976 656e 2060 7370 6169 7260  of given `spair`
-00014610: 206f 7074 696f 6e73 2064 6563 6964 6573   options decides
-00014620: 2074 6865 2063 6f6c 6f72 206f 6620 7468   the color of th
-00014630: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-00014640: 3e20 6469 7374 7269 6275 7469 6f6e 206f  > distribution o
-00014650: 6620 6561 6368 2073 7065 6369 6573 2070  f each species p
-00014660: 6169 722e 2054 6865 2061 6464 6974 696f  air. The additio
-00014670: 6e61 6c20 6f70 7469 6f6e 2060 666f 6375  nal option `focu
-00014680: 7332 616c 6c60 2069 6e20 6077 6764 206b  s2all` in `wgd k
-00014690: 7364 6020 616e 6420 6077 6764 2076 697a  sd` and `wgd viz
-000146a0: 6020 6361 6e20 6265 2075 7365 6420 746f  ` can be used to
-000146b0: 2074 656c 6c20 7468 6520 7072 6f67 7261   tell the progra
-000146c0: 6d20 7468 6174 2074 6865 2073 7065 6369  m that the speci
-000146d0: 6573 2070 6169 7273 2061 7265 2062 6574  es pairs are bet
-000146e0: 7765 656e 2074 6865 2066 6f63 7573 2073  ween the focus s
-000146f0: 7065 6369 6573 2061 6e64 2061 6c6c 2074  pecies and all t
-00014700: 6865 206f 7468 6572 2073 7065 6369 6573  he other species
-00014710: 2073 7563 6820 7468 6174 2075 7365 7273   such that users
-00014720: 2064 6f6e 2774 206e 6565 6420 746f 2074   don't need to t
-00014730: 7970 6520 696e 2065 6163 6820 7061 6972  ype in each pair
-00014740: 2069 6e64 6976 6964 7561 6c6c 792e 0a0a   individually...
-00014750: 6060 600a 7767 6420 7669 7a20 2d64 2077  ```.wgd viz -d w
-00014760: 6764 5f67 6c6f 6261 6c6d 7262 685f 6b73  gd_globalmrbh_ks
-00014770: 2f67 6c6f 6261 6c5f 4d52 4248 2e74 7376  /global_MRBH.tsv
-00014780: 2e6b 732e 7473 7620 2d2d 6578 7472 6170  .ks.tsv --extrap
-00014790: 6172 616e 6f6d 656b 7320 7767 645f 6b73  aranomeks wgd_ks
-000147a0: 642f 4171 7569 6c65 6769 615f 636f 6572  d/Aquilegia_coer
-000147b0: 756c 6561 2e74 7376 2e6b 732e 7473 7620  ulea.tsv.ks.tsv 
-000147c0: 2d73 7020 7370 6563 6965 7374 7265 652e  -sp speciestree.
-000147d0: 6e77 202d 2d72 6577 6569 6768 7420 2d61  nw --reweight -a
-000147e0: 7020 7767 645f 7379 6e2f 6961 6468 6f72  p wgd_syn/iadhor
-000147f0: 652d 6f75 742f 616e 6368 6f72 706f 696e  e-out/anchorpoin
-00014800: 7473 2e74 7874 202d 6f20 7767 645f 7669  ts.txt -o wgd_vi
-00014810: 7a5f 6d69 7865 645f 4b73 202d 2d73 7061  z_mixed_Ks --spa
-00014820: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-00014830: 6572 756c 6561 3b50 726f 7465 615f 6379  erulea;Protea_cy
-00014840: 6e61 726f 6964 6573 2220 2d2d 7370 6169  naroides" --spai
-00014850: 7220 2241 7175 696c 6567 6961 5f63 6f65  r "Aquilegia_coe
-00014860: 7275 6c65 613b 5669 7469 735f 7669 6e69  rulea;Vitis_vini
-00014870: 6665 7261 2220 2d2d 7370 6169 7220 2241  fera" --spair "A
-00014880: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00014890: 613b 4163 6f72 7573 5f61 6d65 7269 6361  a;Acorus_america
-000148a0: 6e75 7322 202d 2d73 7061 6972 2022 4171  nus" --spair "Aq
-000148b0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000148c0: 3b41 7175 696c 6567 6961 5f63 6f65 7275  ;Aquilegia_coeru
-000148d0: 6c65 6122 202d 2d70 6c6f 746b 6465 2028  lea" --plotkde (
-000148e0: 2d2d 6773 6d61 7020 6765 6e65 5f73 7065  --gsmap gene_spe
-000148f0: 6369 6573 2e6d 6170 290a 6060 600a 0a57  cies.map).```..W
-00014900: 6974 6820 7468 6520 6f70 7469 6f6e 2060  ith the option `
-00014910: 666f 6375 7332 616c 6c60 2c20 7468 6520  focus2all`, the 
-00014920: 636f 6d6d 616e 6420 6361 6e20 6265 2061  command can be a
-00014930: 6c73 6f20 6c69 6b65 2074 6869 732c 2073  lso like this, s
-00014940: 616d 6520 7769 7468 2060 7767 6420 6b73  ame with `wgd ks
-00014950: 6460 2e0a 0a60 6060 0a77 6764 2076 697a  d`...```.wgd viz
-00014960: 202d 6420 7767 645f 676c 6f62 616c 6d72   -d wgd_globalmr
-00014970: 6268 5f6b 732f 676c 6f62 616c 5f4d 5242  bh_ks/global_MRB
-00014980: 482e 7473 762e 6b73 2e74 7376 202d 2d65  H.tsv.ks.tsv --e
-00014990: 7874 7261 7061 7261 6e6f 6d65 6b73 2077  xtraparanomeks w
-000149a0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-000149b0: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-000149c0: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
-000149d0: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
-000149e0: 6874 202d 6170 2077 6764 5f73 796e 2f69  ht -ap wgd_syn/i
-000149f0: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
-00014a00: 7270 6f69 6e74 732e 7478 7420 2d6f 2077  rpoints.txt -o w
-00014a10: 6764 5f76 697a 5f6d 6978 6564 5f4b 7320  gd_viz_mixed_Ks 
-00014a20: 2d2d 666f 6375 7332 616c 6c20 4171 7569  --focus2all Aqui
-00014a30: 6c65 6769 615f 636f 6572 756c 6561 202d  legia_coerulea -
-00014a40: 2d70 6c6f 746b 6465 0a60 6060 0a0a 4e6f  -plotkde.```..No
-00014a50: 7465 2074 6861 7420 7765 2063 616e 2065  te that we can e
-00014a60: 6173 696c 7920 7368 6f77 2074 6861 7420  asily show that 
-00014a70: 2a41 7175 696c 6567 6961 2063 6f65 7275  *Aquilegia coeru
-00014a80: 6c65 612a 2068 6173 2068 6967 6865 7220  lea* has higher 
-00014a90: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
-00014aa0: 6520 7468 616e 202a 5072 6f74 6561 2063  e than *Protea c
-00014ab0: 796e 6172 6f69 6465 732a 2061 6e64 202a  ynaroides* and *
-00014ac0: 5669 7469 7320 7669 6e69 6665 7261 2a20  Vitis vinifera* 
-00014ad0: 6279 2063 6f6d 7061 7269 6e67 2074 6865  by comparing the
-00014ae0: 6972 2073 7562 7374 6974 7574 696f 6e20  ir substitution 
-00014af0: 6469 7374 616e 6365 2069 6e20 7265 6761  distance in rega
-00014b00: 7264 2074 6f20 7468 6520 7361 6d65 2064  rd to the same d
-00014b10: 6976 6572 6765 6e63 6520 6576 656e 7420  ivergence event 
-00014b20: 7769 7468 206f 7574 6772 6f75 7020 7370  with outgroup sp
-00014b30: 6563 6965 7320 2a41 636f 7275 735f 616d  ecies *Acorus_am
-00014b40: 6572 6963 616e 7573 2a2c 2075 7369 6e67  ericanus*, using
-00014b50: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e0a   command below..
-00014b60: 0a60 6060 0a77 6764 2076 697a 202d 6420  .```.wgd viz -d 
-00014b70: 7767 645f 676c 6f62 616c 6d72 6268 5f6b  wgd_globalmrbh_k
-00014b80: 732f 676c 6f62 616c 5f4d 5242 482e 7473  s/global_MRBH.ts
-00014b90: 762e 6b73 2e74 7376 202d 7370 2073 7065  v.ks.tsv -sp spe
-00014ba0: 6369 6573 7472 6565 2e6e 7720 2d2d 7265  ciestree.nw --re
-00014bb0: 7765 6967 6874 202d 6f20 7767 645f 7669  weight -o wgd_vi
-00014bc0: 7a5f 436f 6d70 6172 655f 7261 7465 202d  z_Compare_rate -
-00014bd0: 2d73 7061 6972 2022 4163 6f72 7573 5f61  -spair "Acorus_a
-00014be0: 6d65 7269 6361 6e75 733b 5072 6f74 6561  mericanus;Protea
-00014bf0: 5f63 796e 6172 6f69 6465 7322 202d 2d73  _cynaroides" --s
-00014c00: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
-00014c10: 636f 6572 756c 6561 3b41 636f 7275 735f  coerulea;Acorus_
-00014c20: 616d 6572 6963 616e 7573 2220 2d2d 7370  americanus" --sp
-00014c30: 6169 7220 2256 6974 6973 5f76 696e 6966  air "Vitis_vinif
-00014c40: 6572 613b 4163 6f72 7573 5f61 6d65 7269  era;Acorus_ameri
-00014c50: 6361 6e75 7322 202d 2d67 736d 6170 2067  canus" --gsmap g
-00014c60: 656e 655f 7370 6563 6965 732e 6d61 7020  ene_species.map 
-00014c70: 2d2d 706c 6f74 6b64 650a 6060 600a 0a21  --plotkde.```..!
-00014c80: 5b5d 2864 6174 612f 5261 775f 4f72 7468  [](data/Raw_Orth
-00014c90: 6f6c 6f67 7565 735f 436f 6d70 6172 655f  ologues_Compare_
-00014ca0: 7261 7465 2e6b 7364 2e73 7667 290a 0a41  rate.ksd.svg)..A
-00014cb0: 7320 6469 7370 6c61 7965 6420 6162 6f76  s displayed abov
-00014cc0: 652c 2074 6865 206f 7274 686f 6c6f 676f  e, the orthologo
-00014cd0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-00014ce0: 623e 2076 616c 7565 7320 6265 7774 6565  b> values bewtee
-00014cf0: 6e20 2a41 7175 696c 6567 6961 2063 6f65  n *Aquilegia coe
-00014d00: 7275 6c65 612a 2061 6e64 202a 4163 6f72  rulea* and *Acor
-00014d10: 7573 2061 6d65 7269 6361 6e75 732a 2068  us americanus* h
-00014d20: 6173 2074 6865 2068 6967 6865 7374 206d  as the highest m
-00014d30: 6f64 652c 2069 6e64 6963 6174 696e 6774  ode, indicatingt
-00014d40: 6865 2066 6173 7465 7220 7375 6273 7469  he faster substi
-00014d50: 7475 7469 6f6e 2072 6174 6520 6f66 202a  tution rate of *
-00014d60: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
-00014d70: 6561 2a20 636f 6d70 6172 6564 2074 6f20  ea* compared to 
-00014d80: 2a50 726f 7465 6120 6379 6e61 726f 6964  *Protea cynaroid
-00014d90: 6573 2a20 616e 6420 2a56 6974 6973 2076  es* and *Vitis v
-00014da0: 696e 6966 6572 612a 2e0a 0a42 6566 6f72  inifera*...Befor
-00014db0: 6520 7632 2e30 2e32 312c 2074 6865 2067  e v2.0.21, the g
-00014dc0: 656e 652d 7370 6563 6965 7320 6d61 7020  ene-species map 
-00014dd0: 6669 6c65 2069 7320 6e65 6363 6573 7361  file is neccessa
-00014de0: 7269 6c79 206e 6565 6465 6420 666f 7220  rily needed for 
-00014df0: 6974 7320 696d 706c 656d 656e 7461 7469  its implementati
-00014e00: 6f6e 2069 6e20 6077 6764 2076 697a 602c  on in `wgd viz`,
-00014e10: 2077 6869 6368 2073 686f 756c 6420 6265   which should be
-00014e20: 2061 7574 6f6d 6174 656c 7920 7072 6f64   automately prod
-00014e30: 7563 6564 2062 7920 7468 6520 6c61 7374  uced by the last
-00014e40: 2060 7767 6420 6b73 6460 2073 7465 7020   `wgd ksd` step 
-00014e50: 6769 7665 6e20 7468 6520 6073 7061 6972  given the `spair
-00014e60: 6020 616e 6420 6073 7065 6369 6573 7472  ` and `speciestr
-00014e70: 6565 6020 7061 7261 6d65 7465 7273 2e20  ee` parameters. 
-00014e80: 5468 6520 6067 656e 655f 7370 6563 6965  The `gene_specie
-00014e90: 732e 6d61 7060 2068 6173 2063 6f6e 7465  s.map` has conte
-00014ea0: 6e74 7320 6173 2062 656c 6f77 2069 6e20  nts as below in 
-00014eb0: 7768 6963 6820 6561 6368 206c 696e 6520  which each line 
-00014ec0: 6973 2074 6865 206a 6f69 6e65 6420 7374  is the joined st
-00014ed0: 7269 6e67 206f 6620 6765 6e65 206e 616d  ring of gene nam
-00014ee0: 6520 616e 6420 7370 6563 6965 7320 6e61  e and species na
-00014ef0: 6d65 2062 7920 7370 6163 652e 2041 6674  me by space. Aft
-00014f00: 6572 2076 322e 302e 3231 2028 696e 636c  er v2.0.21 (incl
-00014f10: 7564 6564 292c 2074 6865 2067 656e 652d  uded), the gene-
-00014f20: 7370 6563 6965 7320 6d61 7020 6669 6c65  species map file
-00014f30: 2069 7320 6e6f 7420 6e65 6363 6573 7361   is not neccessa
-00014f40: 7269 6c79 206e 6565 6465 6420 616e 796d  rily needed anym
-00014f50: 6f72 652e 0a0a 6060 600a 4171 636f 6536  ore...```.Aqcoe6
-00014f60: 4730 3537 3830 302e 3120 4171 7569 6c65  G057800.1 Aquile
-00014f70: 6769 615f 636f 6572 756c 6561 0a56 7669  gia_coerulea.Vvi
-00014f80: 5f56 4954 5f32 3031 7330 3031 3167 3031  _VIT_201s0011g01
-00014f90: 3533 302e 3120 5669 7469 735f 7669 6e69  530.1 Vitis_vini
-00014fa0: 6665 7261 0a50 6379 5f50 726f 6379 3031  fera.Pcy_Procy01
-00014fb0: 6730 3835 3130 2050 726f 7465 615f 6379  g08510 Protea_cy
-00014fc0: 6e61 726f 6964 6573 0a41 616d 5f41 636f  naroides.Aam_Aco
-00014fd0: 7261 2e30 3447 3134 3239 3030 2e31 2041  ra.04G142900.1 A
-00014fe0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
-00014ff0: 0a60 6060 0a0a 4120 6d6f 7265 2063 6f6d  .```..A more com
-00015000: 706c 6578 2070 6c6f 7420 6361 6e20 6265  plex plot can be
-00015010: 206d 6164 6520 6279 2061 6464 2074 6865   made by add the
-00015020: 2066 6c61 6720 602d 2d70 6c6f 7465 6c6d   flag `--plotelm
-00015030: 6d60 2073 7563 6820 7468 6174 2074 6865  m` such that the
-00015040: 2045 4c4d 4d20 6d69 7874 7572 6520 6d6f   ELMM mixture mo
-00015050: 6465 6c69 6e67 206f 6620 7072 6f76 6964  deling of provid
-00015060: 6564 2070 6172 616e 6f6d 6520 2a4b 2a3c  ed paranome *K*<
-00015070: 7375 623e 533c 2f73 7562 3e20 6361 6e20  sub>S</sub> can 
-00015080: 6265 2073 7570 6572 696d 706f 7365 642c  be superimposed,
-00015090: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
-000150a0: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
-000150b0: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
-000150c0: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
-000150d0: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
-000150e0: 7376 202d 2d65 7874 7261 7061 7261 6e6f  sv --extraparano
-000150f0: 6d65 6b73 2077 6764 5f6b 7364 2f41 7175  meks wgd_ksd/Aqu
-00015100: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
-00015110: 7473 762e 6b73 2e74 7376 202d 7370 2073  tsv.ks.tsv -sp s
-00015120: 7065 6369 6573 7472 6565 2e6e 7720 2d2d  peciestree.nw --
-00015130: 7265 7765 6967 6874 202d 6170 2077 6764  reweight -ap wgd
-00015140: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
-00015150: 2f61 6e63 686f 7270 6f69 6e74 732e 7478  /anchorpoints.tx
-00015160: 7420 2d6f 2077 6764 5f76 697a 5f6d 6978  t -o wgd_viz_mix
-00015170: 6564 5f4b 735f 656c 6d6d 202d 2d73 7061  ed_Ks_elmm --spa
-00015180: 6972 2022 4171 7569 6c65 6769 615f 636f  ir "Aquilegia_co
-00015190: 6572 756c 6561 3b50 726f 7465 615f 6379  erulea;Protea_cy
-000151a0: 6e61 726f 6964 6573 2220 2d2d 7370 6169  naroides" --spai
-000151b0: 7220 2241 7175 696c 6567 6961 5f63 6f65  r "Aquilegia_coe
-000151c0: 7275 6c65 613b 5669 7469 735f 7669 6e69  rulea;Vitis_vini
-000151d0: 6665 7261 2220 2d2d 7370 6169 7220 2241  fera" --spair "A
-000151e0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-000151f0: 613b 4163 6f72 7573 5f61 6d65 7269 6361  a;Acorus_america
-00015200: 6e75 7322 202d 2d73 7061 6972 2022 4171  nus" --spair "Aq
-00015210: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00015220: 3b41 7175 696c 6567 6961 5f63 6f65 7275  ;Aquilegia_coeru
-00015230: 6c65 6122 202d 2d67 736d 6170 2067 656e  lea" --gsmap gen
-00015240: 655f 7370 6563 6965 732e 6d61 7020 2d2d  e_species.map --
-00015250: 706c 6f74 6b64 6520 2d2d 706c 6f74 656c  plotkde --plotel
-00015260: 6d6d 0a60 6060 0a0a 215b 5d28 6461 7461  mm.```..![](data
-00015270: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
-00015280: 6c65 615f 476c 6f62 616c 6d72 6268 4b73  lea_GlobalmrbhKs
-00015290: 5f45 6c6d 6d5f 436f 7272 6563 7465 642e  _Elmm_Corrected.
-000152a0: 6b73 642e 7376 6729 0a0a 4672 6f6d 2074  ksd.svg)..From t
-000152b0: 6865 206d 6978 6564 202a 4b2a 3c73 7562  he mixed *K*<sub
-000152c0: 3e53 3c2f 7375 623e 2070 6c6f 7420 6162  >S</sub> plot ab
-000152d0: 6f76 652c 2077 6520 6361 6e20 7365 6520  ove, we can see 
-000152e0: 7468 6174 2074 6865 206f 7074 696d 697a  that the optimiz
-000152f0: 6564 206c 6f67 6e6f 726d 616c 2063 6f6d  ed lognormal com
-00015300: 706f 6e65 6e74 2062 2077 6974 6820 6d6f  ponent b with mo
-00015310: 6465 2031 2e32 2069 7320 796f 756e 6765  de 1.2 is younge
-00015320: 7220 7468 616e 2074 6865 2063 6f72 7265  r than the corre
-00015330: 6374 6564 206f 7274 686f 6c6f 676f 7573  cted orthologous
-00015340: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00015350: 206d 6f64 6520 7769 7468 202a 5072 6f74   mode with *Prot
-00015360: 6561 2063 796e 6172 6f69 6465 732a 2061  ea cynaroides* a
-00015370: 6e64 202a 5669 7469 7320 7669 6e69 6665  nd *Vitis vinife
-00015380: 7261 2a20 2831 2e33 3920 616e 6420 312e  ra* (1.39 and 1.
-00015390: 3437 2c20 7265 7370 6563 7469 7665 6c79  47, respectively
-000153a0: 292e 0a0a 4265 7369 6465 732c 2077 6520  )...Besides, we 
-000153b0: 6361 6e20 616c 736f 2061 6464 2074 6865  can also add the
-000153c0: 2047 4d4d 206d 6978 7475 7265 206d 6f64   GMM mixture mod
-000153d0: 656c 696e 6720 6f66 2061 6e63 686f 7220  eling of anchor 
-000153e0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-000153f0: 7661 6c75 6573 2077 6974 6820 7468 6520  values with the 
-00015400: 666c 6167 2060 2d2d 706c 6f74 6170 676d  flag `--plotapgm
-00015410: 6d60 2c20 7573 696e 6720 7468 6520 636f  m`, using the co
-00015420: 6d6d 616e 6420 6265 6c6f 772e 0a0a 6060  mmand below...``
-00015430: 600a 7767 6420 7669 7a20 2d64 2077 6764  `.wgd viz -d wgd
-00015440: 5f67 6c6f 6261 6c6d 7262 685f 6b73 2f67  _globalmrbh_ks/g
-00015450: 6c6f 6261 6c5f 4d52 4248 2e74 7376 2e6b  lobal_MRBH.tsv.k
-00015460: 732e 7473 7620 2d2d 6578 7472 6170 6172  s.tsv --extrapar
-00015470: 616e 6f6d 656b 7320 7767 645f 6b73 642f  anomeks wgd_ksd/
-00015480: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00015490: 6561 2e74 7376 2e6b 732e 7473 7620 2d73  ea.tsv.ks.tsv -s
-000154a0: 7020 7370 6563 6965 7374 7265 652e 6e77  p speciestree.nw
-000154b0: 202d 2d72 6577 6569 6768 7420 2d61 7020   --reweight -ap 
-000154c0: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
-000154d0: 6f75 742f 616e 6368 6f72 706f 696e 7473  out/anchorpoints
-000154e0: 2e74 7874 202d 6f20 7767 645f 7669 7a5f  .txt -o wgd_viz_
-000154f0: 6d69 7865 645f 4b73 5f65 6c6d 6d20 2d2d  mixed_Ks_elmm --
-00015500: 7370 6169 7220 2241 7175 696c 6567 6961  spair "Aquilegia
-00015510: 5f63 6f65 7275 6c65 613b 5072 6f74 6561  _coerulea;Protea
-00015520: 5f63 796e 6172 6f69 6465 7322 202d 2d73  _cynaroides" --s
-00015530: 7061 6972 2022 4171 7569 6c65 6769 615f  pair "Aquilegia_
-00015540: 636f 6572 756c 6561 3b56 6974 6973 5f76  coerulea;Vitis_v
-00015550: 696e 6966 6572 6122 202d 2d73 7061 6972  inifera" --spair
-00015560: 2022 4171 7569 6c65 6769 615f 636f 6572   "Aquilegia_coer
-00015570: 756c 6561 3b41 636f 7275 735f 616d 6572  ulea;Acorus_amer
-00015580: 6963 616e 7573 2220 2d2d 7370 6169 7220  icanus" --spair 
-00015590: 2241 7175 696c 6567 6961 5f63 6f65 7275  "Aquilegia_coeru
-000155a0: 6c65 613b 4171 7569 6c65 6769 615f 636f  lea;Aquilegia_co
-000155b0: 6572 756c 6561 2220 2d2d 6773 6d61 7020  erulea" --gsmap 
-000155c0: 6765 6e65 5f73 7065 6369 6573 2e6d 6170  gene_species.map
-000155d0: 202d 2d70 6c6f 746b 6465 202d 2d70 6c6f   --plotkde --plo
-000155e0: 7461 7067 6d6d 6d0a 6060 600a 0a21 5b5d  tapgmmm.```..![]
-000155f0: 2864 6174 612f 4171 7569 6c65 6769 615f  (data/Aquilegia_
-00015600: 636f 6572 756c 6561 5f47 6c6f 6261 6c6d  coerulea_Globalm
-00015610: 7262 684b 735f 4170 676d 6d5f 436f 7272  rbhKs_Apgmm_Corr
-00015620: 6563 7465 642e 6b73 642e 7376 6729 0a0a  ected.ksd.svg)..
-00015630: 4173 206d 616e 6966 6573 7465 6420 6162  As manifested ab
-00015640: 6f76 652c 2074 6865 2061 6e63 686f 7220  ove, the anchor 
-00015650: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00015660: 636f 6d70 6f6e 656e 7420 3220 7769 7468  component 2 with
-00015670: 206d 6f64 6520 312e 3238 2069 7320 616c   mode 1.28 is al
-00015680: 736f 2079 6f75 6e67 6572 2074 6861 6e20  so younger than 
-00015690: 7468 6520 636f 7272 6563 7465 6420 6f72  the corrected or
-000156a0: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
-000156b0: 623e 533c 2f73 7562 3e20 6d6f 6465 2077  b>S</sub> mode w
-000156c0: 6974 6820 2a50 726f 7465 6120 6379 6e61  ith *Protea cyna
-000156d0: 726f 6964 6573 2a20 616e 6420 2a56 6974  roides* and *Vit
-000156e0: 6973 2076 696e 6966 6572 612a 2e20 4275  is vinifera*. Bu
-000156f0: 7420 7765 206e 6565 6420 746f 2062 6520  t we need to be 
-00015700: 6f66 2063 6f75 7273 6520 6361 7574 696f  of course cautio
-00015710: 7573 2074 6861 7420 7375 6368 2064 6973  us that such dis
-00015720: 7469 6e63 7469 6f6e 2063 6f6d 6573 2077  tinction comes w
-00015730: 6974 6820 7468 6520 756e 6365 7274 6169  ith the uncertai
-00015740: 6e74 6965 7320 696e 7472 6f64 7563 6564  nties introduced
-00015750: 2066 726f 6d20 7468 6520 6170 706c 6965   from the applie
-00015760: 6420 6d69 7874 7572 6520 6d6f 6465 6c69  d mixture modeli
-00015770: 6e67 206d 6574 686f 646f 6c6f 6779 2069  ng methodology i
-00015780: 6e20 7465 726d 7320 6f66 2066 6f72 2069  n terms of for i
-00015790: 6e73 7461 6e63 6520 6469 6666 6572 656e  nstance differen
-000157a0: 7420 696e 6974 6961 6c69 7a61 7469 6f6e  t initialization
-000157b0: 2070 6f69 6e74 7320 616e 6420 7468 6520   points and the 
-000157c0: 6973 7375 6520 6f66 206f 7665 7266 6974  issue of overfit
-000157d0: 7469 6e67 2061 6e64 2074 6865 2073 6973  ting and the sis
-000157e0: 7465 7220 7370 6563 6965 7373 2061 646f  ter speciess ado
-000157f0: 7074 6564 2069 6e20 7468 6174 2074 6865  pted in that the
-00015800: 7265 206d 6967 6874 2062 6520 7370 6563  re might be spec
-00015810: 6965 7320 7769 7468 206d 6f72 6520 6469  ies with more di
-00015820: 7370 6172 6174 6520 7375 6273 7469 7475  sparate substitu
-00015830: 7469 6f6e 2072 6174 6573 2074 6861 6e20  tion rates than 
-00015840: 7468 6520 6f6e 6520 7765 2063 686f 7365  the one we chose
-00015850: 2e0a 0a41 6464 696e 6720 626f 7468 2045  ...Adding both E
-00015860: 4c4d 4d20 7265 7375 6c74 2066 6f72 2070  LMM result for p
-00015870: 6172 616e 6f6d 6520 616e 6420 474d 4d20  aranome and GMM 
-00015880: 7265 7375 6c74 2066 6f72 2061 6e63 686f  result for ancho
-00015890: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
-000158a0: 3e20 6361 6e20 6265 2061 6368 6965 7665  > can be achieve
-000158b0: 6420 6a75 7374 2062 7920 6164 6420 7468  d just by add th
-000158c0: 6520 7477 6f20 666c 6167 7320 6d65 6e74  e two flags ment
-000158d0: 696f 6e65 6420 6162 6f76 652c 2075 7369  ioned above, usi
-000158e0: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
-000158f0: 656c 6f77 2e0a 0a60 6060 0a77 6764 2076  elow...```.wgd v
-00015900: 697a 202d 6420 7767 645f 676c 6f62 616c  iz -d wgd_global
-00015910: 6d72 6268 5f6b 732f 676c 6f62 616c 5f4d  mrbh_ks/global_M
-00015920: 5242 482e 7473 762e 6b73 2e74 7376 202d  RBH.tsv.ks.tsv -
-00015930: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
-00015940: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
-00015950: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-00015960: 6b73 2e74 7376 202d 7370 2073 7065 6369  ks.tsv -sp speci
-00015970: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
-00015980: 6967 6874 202d 6170 2077 6764 5f73 796e  ight -ap wgd_syn
-00015990: 2f69 6164 686f 7265 2d6f 7574 2f61 6e63  /iadhore-out/anc
-000159a0: 686f 7270 6f69 6e74 732e 7478 7420 2d6f  horpoints.txt -o
-000159b0: 2077 6764 5f76 697a 5f6d 6978 6564 5f4b   wgd_viz_mixed_K
-000159c0: 735f 656c 6d6d 202d 2d73 7061 6972 2022  s_elmm --spair "
-000159d0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000159e0: 6561 3b50 726f 7465 615f 6379 6e61 726f  ea;Protea_cynaro
-000159f0: 6964 6573 2220 2d2d 7370 6169 7220 2241  ides" --spair "A
-00015a00: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00015a10: 613b 5669 7469 735f 7669 6e69 6665 7261  a;Vitis_vinifera
-00015a20: 2220 2d2d 7370 6169 7220 2241 7175 696c  " --spair "Aquil
-00015a30: 6567 6961 5f63 6f65 7275 6c65 613b 4163  egia_coerulea;Ac
-00015a40: 6f72 7573 5f61 6d65 7269 6361 6e75 7322  orus_americanus"
-00015a50: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
-00015a60: 6769 615f 636f 6572 756c 6561 3b41 7175  gia_coerulea;Aqu
-00015a70: 696c 6567 6961 5f63 6f65 7275 6c65 6122  ilegia_coerulea"
-00015a80: 202d 2d67 736d 6170 2067 656e 655f 7370   --gsmap gene_sp
-00015a90: 6563 6965 732e 6d61 7020 2d2d 706c 6f74  ecies.map --plot
-00015aa0: 6b64 6520 2d2d 706c 6f74 656c 6d6d 202d  kde --plotelmm -
-00015ab0: 2d70 6c6f 7461 7067 6d6d 6d0a 6060 600a  -plotapgmmm.```.
-00015ac0: 0a21 5b5d 2864 6174 612f 4171 7569 6c65  .![](data/Aquile
-00015ad0: 6769 615f 636f 6572 756c 6561 5f47 6c6f  gia_coerulea_Glo
-00015ae0: 6261 6c6d 7262 684b 735f 456c 6d6d 5f41  balmrbhKs_Elmm_A
-00015af0: 7067 6d6d 5f43 6f72 7265 6374 6564 2e6b  pgmm_Corrected.k
-00015b00: 7364 2e73 7667 290a 0a41 6e20 616c 7465  sd.svg)..An alte
-00015b10: 726e 6174 6976 6520 7761 7920 746f 2063  rnative way to c
-00015b20: 616c 6375 6c61 7465 2074 6865 206f 7274  alculate the ort
-00015b30: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
-00015b40: 3e53 3c2f 7375 623e 2069 7320 746f 2064  >S</sub> is to d
-00015b50: 6972 6563 746c 7920 7573 6520 7468 6520  irectly use the 
-00015b60: 6f72 7468 6f67 726f 7570 7320 696e 7374  orthogroups inst
-00015b70: 6561 6420 6f66 2067 6c6f 6261 6c20 4d52  ead of global MR
-00015b80: 4248 2066 616d 696c 792e 2054 6861 7420  BH family. That 
-00015b90: 7761 7920 7765 2064 6f6e 2774 2075 7365  way we don't use
-00015ba0: 2074 6865 2070 7265 2d69 6e66 6572 7265   the pre-inferre
-00015bb0: 6420 7061 7261 6e6f 6d65 202a 4b2a 3c73  d paranome *K*<s
-00015bc0: 7562 3e53 3c2f 7375 623e 2062 7574 2074  ub>S</sub> but t
-00015bd0: 6865 2070 6172 616c 6f67 6f75 7320 6765  he paralogous ge
-00015be0: 6e65 2070 6169 7273 2069 6e73 6964 6520  ne pairs inside 
-00015bf0: 6561 6368 206f 7274 686f 6772 6f75 7020  each orthogroup 
-00015c00: 696e 7374 6561 642e 2054 6f20 6163 6869  instead. To achi
-00015c10: 6576 6520 7468 6174 2c20 7765 2066 6972  eve that, we fir
-00015c20: 7374 206e 6565 6420 746f 2069 6e66 6572  st need to infer
-00015c30: 206f 7274 686f 6772 6f75 7073 2075 7369   orthogroups usi
-00015c40: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
-00015c50: 656c 6f77 2e0a 0a60 6060 0a77 6764 2064  elow...```.wgd d
-00015c60: 6d64 2041 7175 696c 6567 6961 5f63 6f65  md Aquilegia_coe
-00015c70: 7275 6c65 6120 5072 6f74 6561 5f63 796e  rulea Protea_cyn
-00015c80: 6172 6f69 6465 7320 4163 6f72 7573 5f61  aroides Acorus_a
-00015c90: 6d65 7269 6361 6e75 7320 5669 7469 735f  mericanus Vitis_
-00015ca0: 7669 6e69 6665 7261 202d 2d6f 7274 686f  vinifera --ortho
-00015cb0: 696e 6665 7220 2d6f 2077 6764 5f6f 7274  infer -o wgd_ort
-00015cc0: 686f 2028 2d2d 6f6e 6c79 6f72 7468 6f29  ho (--onlyortho)
-00015cd0: 200a 6060 600a 0a55 7365 7273 2063 616e   .```..Users can
-00015ce0: 2064 6563 6964 6520 746f 206f 6e6c 7920   decide to only 
-00015cf0: 636f 6e64 7563 7420 7468 6520 6f72 7468  conduct the orth
-00015d00: 6f67 726f 7570 2061 6e61 6c79 7369 7320  ogroup analysis 
-00015d10: 7768 696c 6520 736b 6970 7069 6e67 206f  while skipping o
-00015d20: 7468 6572 2061 6e61 6c79 7369 7320 6279  ther analysis by
-00015d30: 2061 6464 696e 6720 7468 6520 666c 6167   adding the flag
-00015d40: 2060 2d2d 6f6e 6c79 6f72 7468 6f60 2e20   `--onlyortho`. 
-00015d50: 4e65 7874 2073 7465 7020 6973 2074 6865  Next step is the
-00015d60: 2073 616d 6520 7769 7468 2067 6c6f 6261   same with globa
-00015d70: 6c20 4d52 4248 2066 616d 696c 7920 6578  l MRBH family ex
-00015d80: 6365 7074 2074 6861 7420 7765 2064 6f6e  cept that we don
-00015d90: 2774 2075 7365 2074 6865 2065 7874 7261  't use the extra
-00015da0: 2070 7265 2d69 6e66 6572 7265 6420 7061   pre-inferred pa
-00015db0: 7261 6e6f 6d65 202a 4b2a 3c73 7562 3e53  ranome *K*<sub>S
-00015dc0: 3c2f 7375 623e 2061 6e79 6d6f 7265 2e20  </sub> anymore. 
-00015dd0: 5765 2069 6e66 6572 2074 6865 202a 4b2a  We infer the *K*
-00015de0: 3c73 7562 3e53 3c2f 7375 623e 2075 7369  <sub>S</sub> usi
-00015df0: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
-00015e00: 656c 6f77 2e20 4e6f 7465 2074 6861 7420  elow. Note that 
-00015e10: 7468 6520 7072 6f67 7261 6d20 6077 6764  the program `wgd
-00015e20: 2076 697a 6020 6361 6e20 706c 6f74 2074   viz` can plot t
-00015e30: 6865 2073 616d 6520 6a75 7374 2061 7320  he same just as 
-00015e40: 7368 6f77 6e20 6162 6f76 652e 0a0a 6060  shown above...``
-00015e50: 600a 7767 6420 6b73 6420 7767 645f 6f72  `.wgd ksd wgd_or
-00015e60: 7468 6f2f 4f72 7468 6f67 726f 7570 732e  tho/Orthogroups.
-00015e70: 7370 2e74 7376 202d 7370 2073 7065 6369  sp.tsv -sp speci
-00015e80: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
-00015e90: 6967 6874 202d 6f20 7767 645f 6f72 7468  ight -o wgd_orth
-00015ea0: 6f5f 6b73 202d 2d73 7061 6972 2022 4171  o_ks --spair "Aq
-00015eb0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00015ec0: 3b50 726f 7465 615f 6379 6e61 726f 6964  ;Protea_cynaroid
-00015ed0: 6573 2220 2d2d 7370 6169 7220 2241 7175  es" --spair "Aqu
-00015ee0: 696c 6567 6961 5f63 6f65 7275 6c65 613b  ilegia_coerulea;
-00015ef0: 5669 7469 735f 7669 6e69 6665 7261 2220  Vitis_vinifera" 
-00015f00: 2d2d 7370 6169 7220 2241 7175 696c 6567  --spair "Aquileg
-00015f10: 6961 5f63 6f65 7275 6c65 613b 4163 6f72  ia_coerulea;Acor
-00015f20: 7573 5f61 6d65 7269 6361 6e75 7322 202d  us_americanus" -
-00015f30: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
-00015f40: 615f 636f 6572 756c 6561 3b41 7175 696c  a_coerulea;Aquil
-00015f50: 6567 6961 5f63 6f65 7275 6c65 6122 202d  egia_coerulea" -
-00015f60: 2d70 6c6f 746b 6465 2028 2d61 7020 7767  -plotkde (-ap wg
-00015f70: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
-00015f80: 742f 616e 6368 6f72 706f 696e 7473 2e74  t/anchorpoints.t
-00015f90: 7874 290a 6060 600a 0a21 5b5d 2864 6174  xt).```..![](dat
-00015fa0: 612f 4171 7569 6c65 6769 615f 636f 6572  a/Aquilegia_coer
-00015fb0: 756c 6561 5f4f 7274 686f 4b73 5f43 6f72  ulea_OrthoKs_Cor
-00015fc0: 7265 6374 6564 2e6b 7364 2e73 7667 290a  rected.ksd.svg).
-00015fd0: 0a41 7320 7368 6f77 6e20 6162 6f76 652c  .As shown above,
-00015fe0: 2074 6865 206e 756d 6265 7220 6f66 2062   the number of b
-00015ff0: 6f74 6820 7061 7261 6c6f 676f 7573 2067  oth paralogous g
-00016000: 656e 6520 7061 6972 7320 616e 6420 6f72  ene pairs and or
-00016010: 7468 6f6c 6f67 6f75 7320 6765 6e65 2070  thologous gene p
-00016020: 6169 7273 2069 7320 6469 6666 6572 656e  airs is differen
-00016030: 7420 7468 616e 2074 6865 206f 6e65 2066  t than the one f
-00016040: 726f 6d20 676c 6f62 616c 204d 5242 4820  rom global MRBH 
-00016050: 6661 6d69 6c79 2069 6e20 7468 6174 2068  family in that h
-00016060: 6572 6520 7765 2070 6c6f 7474 6564 2061  ere we plotted a
-00016070: 6c6c 206f 7274 686f 6c6f 676f 7573 2067  ll orthologous g
-00016080: 656e 6520 7061 6972 7320 696e 7374 6561  ene pairs instea
-00016090: 6420 6f66 206f 6e6c 7920 676c 6f62 616c  d of only global
-000160a0: 204d 5242 4820 616e 6420 706f 7465 6e74   MRBH and potent
-000160b0: 6961 6c6c 7920 6e65 7720 7061 7261 6c6f  ially new paralo
-000160c0: 676f 7573 2067 656e 6520 7061 6972 7320  gous gene pairs 
-000160d0: 6d69 6768 7420 6265 2070 726f 6475 6365  might be produce
-000160e0: 6420 696e 2074 6865 206f 7274 686f 6772  d in the orthogr
-000160f0: 6f75 7020 696e 6665 7265 6e63 6520 7374  oup inference st
-00016100: 6570 2c20 746f 6765 7468 6572 2077 6974  ep, together wit
-00016110: 6820 6469 6666 6572 656e 7420 7265 6361  h different reca
-00016120: 6c63 756c 6174 6564 2077 6569 6768 7473  lculated weights
-00016130: 2e0a 0a49 6620 6f6e 6520 6f6e 6c79 2077  ...If one only w
-00016140: 616e 7465 6420 746f 2070 6c6f 7420 7468  anted to plot th
-00016150: 6520 6f72 7468 6f6c 6f67 6f75 7320 2a4b  e orthologous *K
-00016160: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-00016170: 7374 7269 6275 7469 6f6e 2c20 6173 2077  stribution, as w
-00016180: 6520 6861 7665 2061 6c72 6561 6479 2073  e have already s
-00016190: 686f 776e 2061 626f 7665 2c20 6974 2063  hown above, it c
-000161a0: 616e 2062 6520 6561 7369 6c79 2061 6368  an be easily ach
-000161b0: 6965 7665 6420 6279 2072 656d 6f76 696e  ieved by removin
-000161c0: 6720 7468 6520 7061 7261 6c6f 676f 7573  g the paralogous
-000161d0: 2073 7065 6369 6573 2070 6169 7220 6041   species pair `A
-000161e0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-000161f0: 613b 4171 7569 6c65 6769 615f 636f 6572  a;Aquilegia_coer
-00016200: 756c 6561 602c 2075 7369 6e67 2074 6865  ulea`, using the
-00016210: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e0a   command below..
-00016220: 0a60 6060 0a77 6764 2076 697a 202d 6420  .```.wgd viz -d 
-00016230: 7767 645f 676c 6f62 616c 6d72 6268 5f6b  wgd_globalmrbh_k
-00016240: 732f 676c 6f62 616c 5f4d 5242 482e 7473  s/global_MRBH.ts
-00016250: 762e 6b73 2e74 7376 202d 7370 2073 7065  v.ks.tsv -sp spe
-00016260: 6369 6573 7472 6565 2e6e 7720 2d2d 7265  ciestree.nw --re
-00016270: 7765 6967 6874 202d 6f20 7767 645f 676c  weight -o wgd_gl
-00016280: 6f62 616c 6d72 6268 5f6f 6e6c 796f 7274  obalmrbh_onlyort
-00016290: 686f 5f6b 7320 2d2d 7370 6169 7220 2241  ho_ks --spair "A
-000162a0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-000162b0: 613b 5072 6f74 6561 5f63 796e 6172 6f69  a;Protea_cynaroi
-000162c0: 6465 7322 202d 2d73 7061 6972 2022 4171  des" --spair "Aq
-000162d0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000162e0: 3b56 6974 6973 5f76 696e 6966 6572 6122  ;Vitis_vinifera"
-000162f0: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
-00016300: 6769 615f 636f 6572 756c 6561 3b41 636f  gia_coerulea;Aco
-00016310: 7275 735f 616d 6572 6963 616e 7573 2220  rus_americanus" 
-00016320: 2d2d 6773 6d61 7020 6765 6e65 5f73 7065  --gsmap gene_spe
-00016330: 6369 6573 2e6d 6170 0a60 6060 0a0a 5765  cies.map.```..We
-00016340: 2063 616e 2063 6c65 6172 6c79 2073 6565   can clearly see
-00016350: 2074 6861 7420 2a56 6974 6973 2076 696e   that *Vitis vin
-00016360: 6966 6572 612a 2068 6173 2068 6967 6865  ifera* has highe
-00016370: 7220 7375 6273 7469 7475 7469 6f6e 2072  r substitution r
-00016380: 6174 6520 7468 616e 202a 5072 6f74 6561  ate than *Protea
-00016390: 2063 796e 6172 6f69 6465 732a 2069 6e20   cynaroides* in 
-000163a0: 7468 6174 2074 6865 6972 206f 7274 686f  that their ortho
-000163b0: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-000163c0: 3c2f 7375 623e 2070 6561 6b73 2077 6974  </sub> peaks wit
-000163d0: 6820 2a41 7175 696c 6567 6961 2063 6f65  h *Aquilegia coe
-000163e0: 7275 6c65 612a 2c20 616c 7468 6f75 6768  rulea*, although
-000163f0: 2072 6570 7265 7365 6e74 696e 6720 7468   representing th
-00016400: 6520 7361 6d65 2064 6976 6572 6765 6e63  e same divergenc
-00016410: 6520 6576 656e 742c 2064 6966 6665 7265  e event, differe
-00016420: 6420 696e 2073 7562 7374 6974 7574 696f  d in substitutio
-00016430: 6e20 6469 7374 616e 6365 2e0a 215b 5d28  n distance..![](
-00016440: 6461 7461 2f52 6177 5f4f 7274 686f 6c6f  data/Raw_Ortholo
-00016450: 6775 6573 2e6b 7364 2e73 7667 290a 0a41  gues.ksd.svg)..A
-00016460: 6674 6572 2074 6865 2070 6879 6c6f 6765  fter the phyloge
-00016470: 6e65 7469 6320 7469 6d69 6e67 206f 6620  netic timing of 
-00016480: 7468 6520 5261 6e75 6e63 756c 616c 6573  the Ranunculales
-00016490: 2057 4744 2c20 7765 2063 616e 2066 7572   WGD, we can fur
-000164a0: 7468 6572 2069 6e66 6572 2069 7473 2061  ther infer its a
-000164b0: 6273 6f6c 7574 6520 6167 652e 2046 6972  bsolute age. Fir
-000164c0: 7374 2077 6520 696e 6665 7220 7468 6520  st we infer the 
-000164d0: 6372 6564 6962 6c65 2072 616e 6765 206f  credible range o
-000164e0: 6620 616e 6368 6f72 2070 6169 7273 2062  f anchor pairs b
-000164f0: 7920 2a4b 2a3c 7375 623e 533c 2f73 7562  y *K*<sub>S</sub
-00016500: 3e20 6865 7572 6973 7469 6361 6c6c 7920  > heuristically 
-00016510: 7573 696e 6720 7468 6520 7072 6f67 7261  using the progra
-00016520: 6d20 6077 6764 2070 6561 6b60 2e0a 0a60  m `wgd peak`...`
-00016530: 6060 0a77 6764 2070 6561 6b20 2d2d 6865  ``.wgd peak --he
-00016540: 7572 6973 7469 6320 7767 645f 6b73 642f  uristic wgd_ksd/
-00016550: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00016560: 6561 2e74 7376 2e6b 732e 7473 7620 2d61  ea.tsv.ks.tsv -a
-00016570: 7020 7767 645f 7379 6e2f 6961 6468 6f72  p wgd_syn/iadhor
-00016580: 652d 6f75 742f 616e 6368 6f72 706f 696e  e-out/anchorpoin
-00016590: 7473 2e74 7874 202d 736d 2077 6764 5f73  ts.txt -sm wgd_s
-000165a0: 796e 2f69 6164 686f 7265 2d6f 7574 2f73  yn/iadhore-out/s
-000165b0: 6567 6d65 6e74 732e 7478 7420 2d6c 6520  egments.txt -le 
-000165c0: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
-000165d0: 6f75 742f 6c69 7374 5f65 6c65 6d65 6e74  out/list_element
-000165e0: 732e 7478 7420 2d6d 7020 7767 645f 7379  s.txt -mp wgd_sy
-000165f0: 6e2f 6961 6468 6f72 652d 6f75 742f 6d75  n/iadhore-out/mu
-00016600: 6c74 6970 6c69 636f 6e5f 7061 6972 732e  ltiplicon_pairs.
-00016610: 7478 7420 2d6f 2077 6764 5f70 6561 6b0a  txt -o wgd_peak.
-00016620: 6060 600a 0a21 5b5d 2864 6174 612f 416e  ```..![](data/An
-00016630: 6368 6f72 4b73 5f50 6561 6b43 495f 4171  chorKs_PeakCI_Aq
-00016640: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00016650: 2e74 7376 2e6b 732e 7473 765f 6e6f 6465  .tsv.ks.tsv_node
-00016660: 5f77 6569 6768 7465 642e 7376 6729 0a0a  _weighted.svg)..
-00016670: 4173 2073 686f 776e 2061 626f 7665 2c20  As shown above, 
-00016680: 7765 2061 7373 756d 6564 2061 206c 6f67  we assumed a log
-00016690: 6e6f 726d 616c 2064 6973 7472 6962 7574  normal distribut
-000166a0: 696f 6e20 6174 2074 6865 2070 6561 6b20  ion at the peak 
-000166b0: 6c6f 6361 7469 6f6e 2064 6574 6563 7465  location detecte
-000166c0: 6420 6279 2074 6865 2060 7369 676e 616c  d by the `signal
-000166d0: 6020 6d6f 6475 6c65 206f 6620 6073 6369  ` module of `sci
-000166e0: 7079 6020 6c69 6272 6172 792e 2054 6865  py` library. The
-000166f0: 2039 3525 2063 6f6e 6669 6465 6e63 6520   95% confidence 
-00016700: 6c65 7665 6c20 6f66 2074 6865 206c 6f67  level of the log
-00016710: 6e6f 726d 616c 2064 6973 7472 6962 7574  normal distribut
-00016720: 696f 6e20 7761 7320 6170 706c 6965 642c  ion was applied,
-00016730: 2069 2e65 2e2c 2030 2e36 382d 322e 3734   i.e., 0.68-2.74
-00016740: 2c20 696e 2066 7572 7468 6572 206d 6f6c  , in further mol
-00016750: 6563 756c 6172 2064 6174 696e 672e 2054  ecular dating. T
-00016760: 6865 2066 696c 6520 6041 7175 696c 6567  he file `Aquileg
-00016770: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-00016780: 6b73 2e74 7376 5f39 3525 4349 5f41 505f  ks.tsv_95%CI_AP_
-00016790: 666f 725f 6461 7469 6e67 5f77 6569 6768  for_dating_weigh
-000167a0: 7465 645f 666f 726d 6174 2e74 7376 6020  ted_format.tsv` 
-000167b0: 6973 2077 6861 7420 7765 206e 6565 6420  is what we need 
-000167c0: 666f 7220 6e65 7874 2073 7465 702e 2054  for next step. T
-000167d0: 6f20 6275 696c 6420 7468 6520 6f72 7468  o build the orth
-000167e0: 6f67 726f 7570 7320 7573 6564 2069 6e20  ogroups used in 
-000167f0: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
-00016800: 696e 672c 2077 6520 6e65 6564 2074 6f20  ing, we need to 
-00016810: 7365 6c65 6374 2073 6f6d 6520 7370 6563  select some spec
-00016820: 6965 7320 616e 6420 666f 726d 2061 2073  ies and form a s
-00016830: 7461 7274 696e 6720 7472 6565 2077 6974  tarting tree wit
-00016840: 6820 7072 6f70 6572 2066 6f73 7369 6c20  h proper fossil 
-00016850: 6361 6c69 6272 6174 696f 6e73 2e20 5765  calibrations. We
-00016860: 2070 726f 7669 6465 206f 6e65 2069 6e20   provide one in 
-00016870: 6d63 6d63 7472 6565 2066 6f72 6d61 7420  mcmctree format 
-00016880: 6173 2062 656c 6f77 2e0a 0a60 6060 0a31  as below...```.1
-00016890: 3720 310a 2828 2828 506f 7461 6d6f 6765  7 1.((((Potamoge
-000168a0: 746f 6e5f 6163 7574 6966 6f6c 6975 732c  ton_acutifolius,
-000168b0: 2853 7069 726f 6465 6c61 5f69 6e74 6572  (Spirodela_inter
-000168c0: 6d65 6469 612c 416d 6f72 7068 6f70 6861  media,Amorphopha
-000168d0: 6c6c 7573 5f6b 6f6e 6a61 6329 292c 2841  llus_konjac)),(A
-000168e0: 6361 6e74 686f 6368 6c61 6d79 735f 6272  canthochlamys_br
-000168f0: 6163 7465 6174 612c 2844 696f 7363 6f72  acteata,(Dioscor
-00016900: 6561 5f61 6c61 7461 2c44 696f 7363 6f72  ea_alata,Dioscor
-00016910: 6561 5f72 6f74 756e 6461 7461 2929 273e  ea_rotundata))'>
-00016920: 302e 3536 3030 3c31 2e32 3836 3327 2927  0.5600<1.2863')'
-00016930: 3e30 2e38 3336 303c 312e 3238 3633 272c  >0.8360<1.2863',
-00016940: 2841 636f 7275 735f 616d 6572 6963 616e  (Acorus_american
-00016950: 7573 2c41 636f 7275 735f 7461 7461 7269  us,Acorus_tatari
-00016960: 6e6f 7769 6929 2927 3e30 2e38 3336 303c  nowii))'>0.8360<
-00016970: 312e 3238 3633 272c 2828 2828 5465 7472  1.2863',((((Tetr
-00016980: 6163 656e 7472 6f6e 5f73 696e 656e 7365  acentron_sinense
-00016990: 2c54 726f 6368 6f64 656e 6472 6f6e 5f61  ,Trochodendron_a
-000169a0: 7261 6c69 6f69 6465 7329 2c28 4275 7875  ralioides),(Buxu
-000169b0: 735f 6175 7374 726f 7975 6e6e 616e 656e  s_austroyunnanen
-000169c0: 7369 732c 4275 7875 735f 7369 6e69 6361  sis,Buxus_sinica
-000169d0: 2929 273e 312e 3130 3830 3c31 2e32 3836  ))'>1.1080<1.286
-000169e0: 3327 2c28 4e65 6c75 6d62 6f5f 6e75 6369  3',(Nelumbo_nuci
-000169f0: 6665 7261 2c28 5465 6c6f 7065 615f 7370  fera,(Telopea_sp
-00016a00: 6563 696f 7369 7373 696d 612c 5072 6f74  eciosissima,Prot
-00016a10: 6561 5f63 796e 6172 6f69 6465 7329 2929  ea_cynaroides)))
-00016a20: 273e 312e 3130 3830 3c31 2e32 3836 3327  '>1.1080<1.2863'
-00016a30: 2c28 4171 7569 6c65 6769 615f 636f 6572  ,(Aquilegia_coer
-00016a40: 756c 6561 5f61 7031 2c41 7175 696c 6567  ulea_ap1,Aquileg
-00016a50: 6961 5f63 6f65 7275 6c65 615f 6170 3229  ia_coerulea_ap2)
-00016a60: 2927 3e31 2e31 3038 303c 312e 3238 3633  )'>1.1080<1.2863
-00016a70: 2729 273e 312e 3237 3230 3c32 2e34 3732  ')'>1.2720<2.472
-00016a80: 3027 3b0a 6060 600a 0a41 7320 7072 6573  0';.```..As pres
-00016a90: 656e 7465 6420 6162 6f76 652c 2074 6865  ented above, the
-00016aa0: 2066 6f63 7573 2073 7065 6369 6573 2074   focus species t
-00016ab0: 6861 7420 6973 2061 626f 7574 2074 6f20  hat is about to 
-00016ac0: 6265 2064 6174 6564 206e 6565 6473 2074  be dated needs t
-00016ad0: 6f20 6265 2072 6570 6c61 6365 6420 7769  o be replaced wi
-00016ae0: 7468 2060 2841 7175 696c 6567 6961 5f63  th `(Aquilegia_c
-00016af0: 6f65 7275 6c65 615f 6170 312c 4171 7569  oerulea_ap1,Aqui
-00016b00: 6c65 6769 615f 636f 6572 756c 6561 5f61  legia_coerulea_a
-00016b10: 7032 2960 2e20 5769 7468 2074 6869 7320  p2)`. With this 
-00016b20: 7374 6172 7469 6e67 2074 7265 6520 616e  starting tree an
-00016b30: 6420 7072 6564 6f77 6e6c 6f61 6465 6420  d predownloaded 
-00016b40: 6364 7320 6669 6c65 7320 6f66 2061 6c6c  cds files of all
-00016b50: 2074 6865 2073 7065 6369 6573 2c20 7765   the species, we
-00016b60: 2063 616e 2062 7569 6c64 2074 6865 206f   can build the o
-00016b70: 7274 686f 6772 6f75 7020 7573 6564 2069  rthogroup used i
-00016b80: 6e20 7468 6520 6669 6e61 6c20 6d6f 6c65  n the final mole
-00016b90: 6375 6c61 7220 6461 7469 6e67 2075 7369  cular dating usi
-00016ba0: 6e67 2074 6865 2063 6f6d 6d61 6e64 2061  ng the command a
-00016bb0: 7320 6265 6c6f 772e 0a0a 6060 600a 7767  s below...```.wg
-00016bc0: 6420 646d 6420 2d66 2041 7175 696c 6567  d dmd -f Aquileg
-00016bd0: 6961 5f63 6f65 7275 6c65 6120 2d61 7020  ia_coerulea -ap 
-00016be0: 7767 645f 7065 616b 2f41 7175 696c 6567  wgd_peak/Aquileg
-00016bf0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-00016c00: 6b73 2e74 7376 5f39 3525 4349 5f41 505f  ks.tsv_95%CI_AP_
-00016c10: 666f 725f 6461 7469 6e67 5f77 6569 6768  for_dating_weigh
-00016c20: 7465 645f 666f 726d 6174 2e74 7376 202d  ted_format.tsv -
-00016c30: 6f20 7767 645f 646d 645f 6f72 7468 6f20  o wgd_dmd_ortho 
-00016c40: 506f 7461 6d6f 6765 746f 6e5f 6163 7574  Potamogeton_acut
-00016c50: 6966 6f6c 6975 7320 5370 6972 6f64 656c  ifolius Spirodel
-00016c60: 615f 696e 7465 726d 6564 6961 2041 6d6f  a_intermedia Amo
-00016c70: 7270 686f 7068 616c 6c75 735f 6b6f 6e6a  rphophallus_konj
-00016c80: 6163 2041 6361 6e74 686f 6368 6c61 6d79  ac Acanthochlamy
-00016c90: 735f 6272 6163 7465 6174 6120 4469 6f73  s_bracteata Dios
-00016ca0: 636f 7265 615f 616c 6174 6120 4469 6f73  corea_alata Dios
-00016cb0: 636f 7265 615f 726f 7475 6e64 6174 6120  corea_rotundata 
-00016cc0: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
-00016cd0: 7320 4163 6f72 7573 5f74 6174 6172 696e  s Acorus_tatarin
-00016ce0: 6f77 6969 2054 6574 7261 6365 6e74 726f  owii Tetracentro
-00016cf0: 6e5f 7369 6e65 6e73 6520 5472 6f63 686f  n_sinense Trocho
-00016d00: 6465 6e64 726f 6e5f 6172 616c 696f 6964  dendron_aralioid
-00016d10: 6573 2042 7578 7573 5f61 7573 7472 6f79  es Buxus_austroy
-00016d20: 756e 6e61 6e65 6e73 6973 2042 7578 7573  unnanensis Buxus
-00016d30: 5f73 696e 6963 6120 4e65 6c75 6d62 6f5f  _sinica Nelumbo_
-00016d40: 6e75 6369 6665 7261 2054 656c 6f70 6561  nucifera Telopea
-00016d50: 5f73 7065 6369 6f73 6973 7369 6d61 2050  _speciosissima P
-00016d60: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
-00016d70: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
-00016d80: 6c65 610a 6060 600a 0a54 6865 2072 6573  lea.```..The res
-00016d90: 756c 7420 6669 6c65 2060 6d65 7267 655f  ult file `merge_
-00016da0: 666f 6375 735f 6170 2e74 7376 6020 6973  focus_ap.tsv` is
-00016db0: 2077 6861 7420 7765 206e 6565 6420 666f   what we need fo
-00016dc0: 7220 7468 6520 6669 6e61 6c20 7374 6570  r the final step
-00016dd0: 206f 6620 6d6f 6c65 6375 6c61 7220 6461   of molecular da
-00016de0: 7469 6e67 2069 6e20 7072 6f67 7261 6d20  ting in program 
-00016df0: 6077 6764 2066 6f63 7573 602e 0a0a 6060  `wgd focus`...``
-00016e00: 600a 7767 6420 666f 6375 7320 2d2d 7072  `.wgd focus --pr
-00016e10: 6f74 6461 7469 6e67 202d 2d61 616d 6f64  otdating --aamod
-00016e20: 656c 206c 6720 7767 645f 646d 645f 6f72  el lg wgd_dmd_or
-00016e30: 7468 6f2f 6d65 7267 655f 666f 6375 735f  tho/merge_focus_
-00016e40: 6170 2e74 7376 202d 7370 2064 6174 696e  ap.tsv -sp datin
-00016e50: 675f 7472 6565 2e6e 7720 2d6f 2077 6764  g_tree.nw -o wgd
-00016e60: 5f64 6174 696e 6720 2d64 206d 636d 6374  _dating -d mcmct
-00016e70: 7265 6520 2d64 7320 2762 7572 6e69 6e20  ree -ds 'burnin 
-00016e80: 3d20 3230 3030 2720 2d64 7320 2773 616d  = 2000' -ds 'sam
-00016e90: 7066 7265 7120 3d20 3130 3030 2720 2d64  pfreq = 1000' -d
-00016ea0: 7320 276e 7361 6d70 6c65 203d 2032 3030  s 'nsample = 200
-00016eb0: 3030 2720 506f 7461 6d6f 6765 746f 6e5f  00' Potamogeton_
-00016ec0: 6163 7574 6966 6f6c 6975 7320 5370 6972  acutifolius Spir
-00016ed0: 6f64 656c 615f 696e 7465 726d 6564 6961  odela_intermedia
-00016ee0: 2041 6d6f 7270 686f 7068 616c 6c75 735f   Amorphophallus_
-00016ef0: 6b6f 6e6a 6163 2041 6361 6e74 686f 6368  konjac Acanthoch
-00016f00: 6c61 6d79 735f 6272 6163 7465 6174 6120  lamys_bracteata 
-00016f10: 4469 6f73 636f 7265 615f 616c 6174 6120  Dioscorea_alata 
-00016f20: 4469 6f73 636f 7265 615f 726f 7475 6e64  Dioscorea_rotund
-00016f30: 6174 6120 4163 6f72 7573 5f61 6d65 7269  ata Acorus_ameri
-00016f40: 6361 6e75 7320 4163 6f72 7573 5f74 6174  canus Acorus_tat
-00016f50: 6172 696e 6f77 6969 2054 6574 7261 6365  arinowii Tetrace
-00016f60: 6e74 726f 6e5f 7369 6e65 6e73 6520 5472  ntron_sinense Tr
-00016f70: 6f63 686f 6465 6e64 726f 6e5f 6172 616c  ochodendron_aral
-00016f80: 696f 6964 6573 2042 7578 7573 5f61 7573  ioides Buxus_aus
-00016f90: 7472 6f79 756e 6e61 6e65 6e73 6973 2042  troyunnanensis B
-00016fa0: 7578 7573 5f73 696e 6963 6120 4e65 6c75  uxus_sinica Nelu
-00016fb0: 6d62 6f5f 6e75 6369 6665 7261 2054 656c  mbo_nucifera Tel
-00016fc0: 6f70 6561 5f73 7065 6369 6f73 6973 7369  opea_speciosissi
-00016fd0: 6d61 2050 726f 7465 615f 6379 6e61 726f  ma Protea_cynaro
-00016fe0: 6964 6573 2041 7175 696c 6567 6961 5f63  ides Aquilegia_c
-00016ff0: 6f65 7275 6c65 610a 6060 600a 0a48 6572  oerulea.```..Her
-00017000: 6520 7765 206f 6e6c 7920 696d 706c 656d  e we only implem
-00017010: 656e 7465 6420 7468 6520 636f 6e63 6174  ented the concat
-00017020: 656e 6174 696f 6e20 616e 616c 7973 6973  enation analysis
-00017030: 2075 7369 6e67 2070 726f 7465 696e 2073   using protein s
-00017040: 6571 7565 6e63 6520 6279 2061 6464 696e  equence by addin
-00017050: 6720 7468 6520 666c 6167 2060 2d2d 7072  g the flag `--pr
-00017060: 6f74 6461 7469 6e67 6020 616e 6420 7765  otdating` and we
-00017070: 2073 6574 2074 6865 2070 6172 616d 6574   set the paramet
-00017080: 6572 2066 6f72 2060 6d63 6d63 7472 6565  er for `mcmctree
-00017090: 6020 7669 6120 7468 6520 6f70 7469 6f6e  ` via the option
-000170a0: 2060 2d64 7360 2e20 4e6f 7465 2074 6861   `-ds`. Note tha
-000170b0: 7420 6f74 6865 7220 6461 7469 6e67 2070  t other dating p
-000170c0: 726f 6772 616d 2073 7563 6820 6173 2060  rogram such as `
-000170d0: 7238 7360 2061 6e64 2060 6265 6173 7460  r8s` and `beast`
-000170e0: 2061 7265 2061 6c73 6f20 6176 6169 6c61   are also availa
-000170f0: 626c 6520 6769 7665 6e20 736f 6d65 206d  ble given some m
-00017100: 616e 6461 746f 7279 2070 6172 616d 6574  andatory paramet
-00017110: 6572 732e 2054 6865 2066 696e 616c 206c  ers. The final l
-00017120: 6f67 206f 6620 7468 6520 7375 6363 6573  og of the succes
-00017130: 7366 756c 2072 756e 2069 7320 6173 2062  sful run is as b
-00017140: 656c 6f77 2e0a 0a60 6060 0a31 363a 3034  elow...```.16:04
-00017150: 3a32 3520 494e 464f 2020 2020 2052 756e  :25 INFO     Run
-00017160: 6e69 6e67 206d 636d 6374 7265 6520 7573  ning mcmctree us
-00017170: 696e 6720 4865 7373 6961 6e20 6d61 7472  ing Hessian matr
-00017180: 6978 206f 6620 4c47 2b47 616d 6d61 2020  ix of LG+Gamma  
-00017190: 636f 7265 2e70 793a 3936 370a 2020 2020  core.py:967.    
-000171a0: 2020 2020 2020 2020 2020 2020 2020 666f                fo
-000171b0: 7220 7072 6f74 6569 6e20 6d6f 6465 6c0a  r protein model.
-000171c0: 3233 3a34 393a 3337 2049 4e46 4f20 2020  23:49:37 INFO   
-000171d0: 2020 506f 7374 6572 696f 7220 6d65 616e    Posterior mean
-000171e0: 2066 6f72 2074 6865 2061 6765 7320 6f66   for the ages of
-000171f0: 2077 6764 2069 7320 3131 322e 3839 3435   wgd is 112.8945
-00017200: 206d 636d 6374 7265 652e 7079 3a32 3936   mcmctree.py:296
-00017210: 0a20 2020 2020 2020 2020 2020 2020 2020  .               
-00017220: 2020 206d 696c 6c69 6f6e 2079 6561 7273     million years
-00017230: 2066 726f 6d20 436f 6e63 6174 656e 6174   from Concatenat
-00017240: 6564 2070 6570 7469 6465 0a20 2020 2020  ed peptide.     
-00017250: 2020 2020 2020 2020 2020 2020 2061 6c69               ali
-00017260: 676e 6d65 6e74 2061 6e64 2039 3525 2063  gnment and 95% c
-00017270: 7265 6469 6269 6c69 7479 2069 6e74 6572  redibility inter
-00017280: 7661 6c73 2028 4349 290a 2020 2020 2020  vals (CI).      
-00017290: 2020 2020 2020 2020 2020 2020 6973 2031              is 1
-000172a0: 3031 2e32 3234 2d31 3233 2e31 3231 206d  01.224-123.121 m
-000172b0: 696c 6c69 6f6e 2079 6561 7273 0a20 2020  illion years.   
-000172c0: 2020 2020 2020 494e 464f 2020 2020 2054        INFO     T
-000172d0: 6f74 616c 2072 756e 2074 696d 653a 2032  otal run time: 2
-000172e0: 3931 3735 7320 2020 2020 2020 2020 2020  9175s           
-000172f0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00017300: 2020 2063 6c69 2e70 793a 3234 310a 2020     cli.py:241.  
-00017310: 2020 2020 2020 2049 4e46 4f20 2020 2020         INFO     
-00017320: 446f 6e65 2020 2020 2020 2020 2020 2020  Done            
-00017330: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00017340: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00017350: 2020 2020 636c 692e 7079 3a32 3432 0a60      cli.py:242.`
-00017360: 6060 0a0a 546f 2076 6973 7561 6c69 7a65  ``..To visualize
-00017370: 2074 6865 2064 6174 652c 2077 6520 616c   the date, we al
-00017380: 736f 2070 726f 7669 6465 6420 6120 7079  so provided a py
-00017390: 7468 6f6e 2073 6372 6970 7420 746f 2070  thon script to p
-000173a0: 6c6f 7420 7468 6520 5747 4420 6461 7465  lot the WGD date
-000173b0: 7320 696e 2074 6865 2060 7767 6460 2066  s in the `wgd` f
-000173c0: 6f6c 6465 722e 2055 7365 7273 206e 6565  older. Users nee
-000173d0: 6420 746f 2065 7874 7261 6374 2074 6865  d to extract the
-000173e0: 2072 6177 2064 6174 6573 2066 726f 6d20   raw dates from 
-000173f0: 7468 6520 606d 636d 632e 7478 7460 2066  the `mcmc.txt` f
-00017400: 6f72 2074 6865 2057 4744 206e 6f64 6520  or the WGD node 
-00017410: 6669 7273 7420 616e 6420 7361 7665 2069  first and save i
-00017420: 7420 6173 2066 696c 6520 6064 6174 6573  t as file `dates
-00017430: 2e74 7874 6020 286f 7220 7768 6174 6576  .txt` (or whatev
-00017440: 6572 2070 7265 6665 7272 6564 206e 616d  er preferred nam
-00017450: 6529 2e20 416e 2065 7861 6d70 6c65 2063  e). An example c
-00017460: 6f6d 6d61 6e64 2069 7320 6173 2062 656c  ommand is as bel
-00017470: 6f77 2e0a 0a60 6060 0a70 7974 686f 6e20  ow...```.python 
-00017480: 2450 4154 482f 706f 7374 706c 6f74 2e70  $PATH/postplot.p
-00017490: 7920 706f 7374 6469 7320 6461 7465 732e  y postdis dates.
-000174a0: 7478 7420 2d2d 7065 7263 656e 7469 6c65  txt --percentile
-000174b0: 2039 3020 2d2d 7469 746c 6520 2257 4744   90 --title "WGD
-000174c0: 2064 6174 6522 202d 2d68 7064 202d 6f20   date" --hpd -o 
-000174d0: 2252 616e 756e 6375 6c61 6c65 735f 5747  "Ranunculales_WG
-000174e0: 445f 6461 7465 2e73 7667 220a 6060 600a  D_date.svg".```.
-000174f0: 0a21 5b5d 2864 6174 612f 5261 6e75 6e63  .![](data/Ranunc
-00017500: 756c 616c 6573 5f57 4744 5f64 6174 652e  ulales_WGD_date.
-00017510: 7376 6729 0a0a 5468 6520 706f 7374 6572  svg)..The poster
-00017520: 696f 7220 6d65 616e 2c20 6d65 6469 616e  ior mean, median
-00017530: 2061 6e64 206d 6f64 6520 6f66 2074 6865   and mode of the
-00017540: 2052 616e 756e 6375 6c61 6c65 7320 5747   Ranunculales WG
-00017550: 4420 6167 6520 6973 2031 3132 2e39 322c  D age is 112.92,
-00017560: 2031 3133 2e34 3420 616e 6420 3131 322e   113.44 and 112.
-00017570: 3534 206d 7961 2c20 7769 7468 2039 3025  54 mya, with 90%
-00017580: 2048 5044 2031 3035 2e30 3720 2d20 3132   HPD 105.07 - 12
-00017590: 322e 3332 206d 7961 2061 7320 6d61 6e69  2.32 mya as mani
-000175a0: 6665 7374 6564 2061 626f 7665 2e0a 0a23  fested above...#
-000175b0: 2323 204b 7374 7265 650a 0a49 6e20 6164  ## Kstree..In ad
-000175c0: 6469 7469 6f6e 2074 6f20 7061 6972 7769  dition to pairwi
-000175d0: 7365 202a 4b2a 3c73 7562 3e53 3c2f 7375  se *K*<sub>S</su
-000175e0: 623e 2065 7374 696d 6174 696f 6e2c 2061  b> estimation, a
-000175f0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00017600: 2074 7265 6520 7769 7468 2062 7261 6e63   tree with branc
-00017610: 6820 6c65 6e67 7468 2069 6e20 2a4b 2a3c  h length in *K*<
-00017620: 7375 623e 533c 2f73 7562 3e20 756e 6974  sub>S</sub> unit
-00017630: 2063 616e 2061 6c73 6f20 6265 2064 6572   can also be der
-00017640: 6976 6564 2066 726f 6d20 7468 6520 7072  ived from the pr
-00017650: 6f67 7261 6d20 6077 6764 206b 7364 6020  ogram `wgd ksd` 
-00017660: 6769 7665 6e20 7468 6520 6f70 7469 6f6e  given the option
-00017670: 2060 2d2d 6b73 7472 6565 6020 616e 6420   `--kstree` and 
-00017680: 602d 2d73 7065 6369 6573 7472 6565 602e  `--speciestree`.
-00017690: 204e 6f74 6520 7468 6174 2074 6865 2061   Note that the a
-000176a0: 6464 6974 696f 6e61 6c20 6f70 7469 6f6e  dditional option
-000176b0: 2060 2d2d 6f6e 6c79 636f 6e63 6174 6b73   `--onlyconcatks
-000176c0: 7472 6565 6020 7769 6c6c 206f 6e6c 7920  tree` will only 
-000176d0: 6361 6c6c 2074 6865 202a 4b2a 3c73 7562  call the *K*<sub
-000176e0: 3e53 3c2f 7375 623e 2065 7374 696d 6174  >S</sub> estimat
-000176f0: 696f 6e20 666f 7220 7468 6520 636f 6e63  ion for the conc
-00017700: 6174 656e 6174 6564 2061 6c69 676e 6d65  atenated alignme
-00017710: 6e74 2072 6174 6865 7220 7468 616e 2061  nt rather than a
-00017720: 6c6c 2074 6865 2061 6c69 676e 6d65 6e74  ll the alignment
-00017730: 732e 2055 7365 7273 206e 6565 6420 746f  s. Users need to
-00017740: 2070 726f 7669 6465 2061 2070 7265 7365   provide a prese
-00017750: 7420 7370 6563 6965 7320 7472 6565 2066  t species tree f
-00017760: 6f72 2074 6865 202a 4b2a 3c73 7562 3e53  or the *K*<sub>S
-00017770: 3c2f 7375 623e 2074 7265 6520 696e 6665  </sub> tree infe
-00017780: 7265 6e63 6520 6f66 2074 6865 2063 6f6e  rence of the con
-00017790: 6361 7465 6e61 7465 6420 616c 6967 6e6d  catenated alignm
-000177a0: 656e 7420 7768 696c 6520 7468 6520 7265  ent while the re
-000177b0: 6d61 696e 696e 6720 616c 6967 6e6d 656e  maining alignmen
-000177c0: 7473 2077 696c 6c20 6265 2061 6761 696e  ts will be again
-000177d0: 7374 2061 6e20 6175 746f 6d61 7465 6c79  st an automately
-000177e0: 2069 6e66 6572 7265 6420 7472 6565 2066   inferred tree f
-000177f0: 726f 6d20 6066 6173 7474 7265 6560 206f  rom `fasttree` o
-00017800: 7220 6069 7174 7265 6560 2e20 496e 2074  r `iqtree`. In t
-00017810: 6865 2065 6e64 2c20 7573 6572 7320 7769  he end, users wi
-00017820: 6c6c 2067 6574 2061 202a 4b2a 3c73 7562  ll get a *K*<sub
-00017830: 3e53 3c2f 7375 623e 2074 7265 652c 2061  >S</sub> tree, a
-00017840: 202a 4b2a 3c73 7562 3e41 3c2f 7375 623e   *K*<sub>A</sub>
-00017850: 2074 7265 6520 616e 6420 6120 cf89 2074   tree and a .. t
-00017860: 7265 6520 7065 7220 6661 6d20 616e 6420  ree per fam and 
-00017870: 666f 7220 7468 6520 636f 6e63 6174 656e  for the concaten
-00017880: 6174 6564 2061 6c69 676e 6d65 6e74 2e0a  ated alignment..
-00017890: 0a60 6060 0a77 6764 206b 7364 2064 6174  .```.wgd ksd dat
-000178a0: 612f 6b73 7472 6565 5f64 6174 612f 6661  a/kstree_data/fa
-000178b0: 6d2e 7473 7620 6461 7461 2f6b 7374 7265  m.tsv data/kstre
-000178c0: 655f 6461 7461 2f41 636f 7275 735f 7461  e_data/Acorus_ta
-000178d0: 7461 7269 6e6f 7769 6920 6461 7461 2f6b  tarinowii data/k
-000178e0: 7374 7265 655f 6461 7461 2f41 6d62 6f72  stree_data/Ambor
-000178f0: 656c 6c61 5f74 7269 6368 6f70 6f64 6120  ella_trichopoda 
-00017900: 6461 7461 2f6b 7374 7265 655f 6461 7461  data/kstree_data
-00017910: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
-00017920: 6c65 6120 6461 7461 2f6b 7374 7265 655f  lea data/kstree_
-00017930: 6461 7461 2f41 7269 7374 6f6c 6f63 6869  data/Aristolochi
-00017940: 615f 6669 6d62 7269 6174 6120 6461 7461  a_fimbriata data
-00017950: 2f6b 7374 7265 655f 6461 7461 2f43 7963  /kstree_data/Cyc
-00017960: 6173 5f70 616e 7a68 6968 7561 656e 7369  as_panzhihuaensi
-00017970: 202d 2d6b 7374 7265 6520 2d2d 7370 6563   --kstree --spec
-00017980: 6965 7374 7265 6520 6461 7461 2f6b 7374  iestree data/kst
-00017990: 7265 655f 6461 7461 2f73 7065 6369 6573  ree_data/species
-000179a0: 5f74 7265 6531 2e6e 7720 2d2d 6f6e 6c79  _tree1.nw --only
-000179b0: 636f 6e63 6174 6b73 7472 6565 202d 6f20  concatkstree -o 
-000179c0: 7767 645f 6b73 7472 6565 5f74 6f70 6f6c  wgd_kstree_topol
-000179d0: 6f67 7931 0a60 6060 0a0a 215b 5d28 6461  ogy1.```..![](da
-000179e0: 7461 2f6b 7374 7265 655f 7265 7375 6c74  ta/kstree_result
-000179f0: 732f 6b73 7472 6565 2e73 7667 290a 0a41  s/kstree.svg)..A
-00017a00: 626f 7665 2077 6520 7573 6564 2074 6872  bove we used thr
-00017a10: 6565 2061 6c74 6572 6e61 7469 7665 2074  ee alternative t
-00017a20: 6f70 6f6c 6f67 6965 7320 746f 2069 6e66  opologies to inf
-00017a30: 6572 2074 6865 202a 4b2a 3c73 7562 3e53  er the *K*<sub>S
-00017a40: 3c2f 7375 623e 2074 7265 6520 7768 6963  </sub> tree whic
-00017a50: 6820 6c65 6420 746f 2064 6966 6665 7265  h led to differe
-00017a60: 6e74 2062 7261 6e63 6820 6c65 6e67 7468  nt branch length
-00017a70: 2065 7374 696d 6174 696f 6e2e 204e 6f74   estimation. Not
-00017a80: 6520 7468 6174 2074 6865 2066 616d 696c  e that the famil
-00017a90: 6965 7320 7765 2075 7365 6420 7765 7265  ies we used were
-00017aa0: 206f 6e6c 7920 7477 6f20 676c 6f62 616c   only two global
-00017ab0: 204d 5242 4820 6661 6d69 6c69 6573 2066   MRBH families f
-00017ac0: 6f72 2074 6865 2070 7572 706f 7365 206f  or the purpose o
-00017ad0: 6620 696c 6c75 7374 7261 7469 6f6e 2e20  f illustration. 
-00017ae0: 546f 2061 6371 7569 7265 2061 6e20 6163  To acquire an ac
-00017af0: 6375 7261 7465 2070 726f 6669 6c65 206f  curate profile o
-00017b00: 6620 7468 6520 7375 6273 7469 7475 7469  f the substituti
-00017b10: 6f6e 2072 6174 6520 7661 7269 6174 696f  on rate variatio
-00017b20: 6e2c 206f 7274 686f 6c6f 6775 6573 2061  n, orthologues a
-00017b30: 7420 7468 6520 7768 6f6c 6520 6765 6e6f  t the whole geno
-00017b40: 6d65 2073 6361 6c65 2073 686f 756c 6420  me scale should 
-00017b50: 6265 2075 7365 642e 0a0a 2323 2043 6974  be used...## Cit
-00017b60: 6174 696f 6e0a 200a 506c 6561 7365 2063  ation. .Please c
-00017b70: 6974 6520 7573 2061 7420 6874 7470 733a  ite us at https:
-00017b80: 2f2f 646f 692e 6f72 672f 3130 2e31 3030  //doi.org/10.100
-00017b90: 372f 3937 382d 312d 3037 3136 2d32 3536  7/978-1-0716-256
-00017ba0: 312d 335f 3120 616e 6420 6874 7470 733a  1-3_1 and https:
-00017bb0: 2f2f 646f 692e 6f72 672f 3130 2e31 3039  //doi.org/10.109
-00017bc0: 332f 6269 6f69 6e66 6f72 6d61 7469 6373  3/bioinformatics
-00017bd0: 2f62 7461 6532 3732 2e0a 0a60 6060 0a48  /btae272...```.H
-00017be0: 656e 6763 6869 2043 6865 6e2c 2041 7274  engchi Chen, Art
-00017bf0: 6875 7220 5a77 6165 6e65 706f 656c 2028  hur Zwaenepoel (
-00017c00: 3230 3233 292e 2049 6e66 6572 656e 6365  2023). Inference
-00017c10: 206f 6620 416e 6369 656e 7420 506f 6c79   of Ancient Poly
-00017c20: 706c 6f69 6479 2066 726f 6d20 4765 6e6f  ploidy from Geno
-00017c30: 6d69 6320 4461 7461 2e20 496e 3a20 5661  mic Data. In: Va
-00017c40: 6e20 6465 2050 6565 722c 2059 2e20 2865  n de Peer, Y. (e
-00017c50: 6473 2920 506f 6c79 706c 6f69 6479 2e20  ds) Polyploidy. 
-00017c60: 4d65 7468 6f64 7320 696e 204d 6f6c 6563  Methods in Molec
-00017c70: 756c 6172 2042 696f 6c6f 6779 2c20 766f  ular Biology, vo
-00017c80: 6c20 3235 3435 2e20 4875 6d61 6e61 2c20  l 2545. Humana, 
-00017c90: 4e65 7720 596f 726b 2c20 4e59 2e20 6874  New York, NY. ht
-00017ca0: 7470 733a 2f2f 646f 692e 6f72 672f 3130  tps://doi.org/10
-00017cb0: 2e31 3030 372f 3937 382d 312d 3037 3136  .1007/978-1-0716
-00017cc0: 2d32 3536 312d 335f 310a 4865 6e67 6368  -2561-3_1.Hengch
-00017cd0: 6920 4368 656e 2c20 4172 7468 7572 205a  i Chen, Arthur Z
-00017ce0: 7761 656e 6570 6f65 6c2c 2059 7665 7320  waenepoel, Yves 
-00017cf0: 5661 6e20 6465 2050 6565 7220 2832 3032  Van de Peer (202
-00017d00: 3429 2e20 7767 6420 7632 3a20 6120 7375  4). wgd v2: a su
-00017d10: 6974 6520 6f66 2074 6f6f 6c73 2074 6f20  ite of tools to 
-00017d20: 756e 636f 7665 7220 616e 6420 6461 7465  uncover and date
-00017d30: 2061 6e63 6965 6e74 2070 6f6c 7970 6c6f   ancient polyplo
-00017d40: 6964 7920 616e 6420 7768 6f6c 652d 6765  idy and whole-ge
-00017d50: 6e6f 6d65 2064 7570 6c69 6361 7469 6f6e  nome duplication
-00017d60: 2e20 4269 6f69 6e66 6f72 6d61 7469 6373  . Bioinformatics
-00017d70: 2c20 6874 7470 733a 2f2f 646f 692e 6f72  , https://doi.or
-00017d80: 672f 3130 2e31 3039 332f 6269 6f69 6e66  g/10.1093/bioinf
-00017d90: 6f72 6d61 7469 6373 2f62 7461 6532 3732  ormatics/btae272
-00017da0: 0a60 6060 0a0a 466f 7220 6369 7461 7469  .```..For citati
-00017db0: 6f6e 206f 6620 7468 6520 746f 6f6c 7320  on of the tools 
-00017dc0: 7573 6564 2069 6e20 7767 642c 2070 6c65  used in wgd, ple
-00017dd0: 6173 6520 636f 6e73 756c 7420 7468 6520  ase consult the 
-00017de0: 646f 6375 6d65 6e74 6174 696f 6e20 6174  documentation at
-00017df0: 0a68 7474 7073 3a2f 2f77 6764 7632 2e72  .https://wgdv2.r
-00017e00: 6561 6474 6865 646f 6373 2e69 6f2f 656e  eadthedocs.io/en
-00017e10: 2f6c 6174 6573 742f 6369 7461 7469 6f6e  /latest/citation
-00017e20: 2e68 746d 6c2e 0a0a                      .html...
+0000ee70: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+0000ee80: 282d 2d72 6577 6569 6768 7429 0a60 6060  (--reweight).```
+0000ee90: 0a0a 4672 6f6d 2070 6172 616c 6f67 6f75  ..From paralogou
+0000eea0: 7320 746f 206f 7274 686f 6c6f 676f 7573  s to orthologous
+0000eeb0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000eec0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
+0000eed0: 6e2c 2075 7365 7273 206f 6e6c 7920 6e65  n, users only ne
+0000eee0: 6564 2074 6f20 7072 6f76 6964 6520 6d6f  ed to provide mo
+0000eef0: 7265 2063 6473 2066 696c 6573 2e20 4e6f  re cds files. No
+0000ef00: 7465 2074 6861 7420 7769 7468 206f 7274  te that with ort
+0000ef10: 686f 6c6f 676f 7573 2067 656e 6520 6661  hologous gene fa
+0000ef20: 6d69 6c69 6573 2074 6865 2077 6569 6768  milies the weigh
+0000ef30: 7469 6e67 206d 6574 686f 6420 6361 6e20  ting method can 
+0000ef40: 6265 2073 6574 2074 6f20 6265 2063 616c  be set to be cal
+0000ef50: 6375 6c61 7465 6420 7065 7220 7370 6563  culated per spec
+0000ef60: 6965 7320 7061 6972 2069 6e73 7465 6164  ies pair instead
+0000ef70: 206f 6620 636f 6e73 6964 6572 696e 6720   of considering 
+0000ef80: 7468 6520 7768 6f6c 6520 6661 6d69 6c79  the whole family
+0000ef90: 2062 6563 6175 7365 2077 6865 6e20 706c   because when pl
+0000efa0: 6f74 7469 6e67 206f 7274 686f 6c6f 676f  otting orthologo
+0000efb0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+0000efc0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+0000efd0: 696f 6e20 6265 7477 6565 6e20 7477 6f20  ion between two 
+0000efe0: 7370 6563 6965 7320 7468 6520 7765 6967  species the weig
+0000eff0: 6874 2063 616c 6375 6c61 7465 6420 6672  ht calculated fr
+0000f000: 6f6d 2074 6869 7320 7370 6563 6966 6963  om this specific
+0000f010: 2073 7065 6369 6573 2070 6169 7220 7368   species pair sh
+0000f020: 6f75 6c64 2062 6520 636f 6e73 6572 7665  ould be conserve
+0000f030: 6420 7768 696c 6520 7468 6520 6f6e 6520  d while the one 
+0000f040: 6361 6c63 756c 6174 6564 2066 726f 6d20  calculated from 
+0000f050: 7468 6520 7768 6f6c 6520 6661 6d69 6c79  the whole family
+0000f060: 2077 696c 6c20 7661 7279 2077 6974 6820   will vary with 
+0000f070: 7468 6520 6e75 6d62 6572 206f 6620 7370  the number of sp
+0000f080: 6563 6965 732e 2054 6f20 696e 6974 6961  ecies. To initia
+0000f090: 7465 2074 6865 2077 6569 6768 7469 6e67  te the weighting
+0000f0a0: 2070 6572 2073 7065 6369 6573 2070 6169   per species pai
+0000f0b0: 722c 2074 6865 206f 7074 696f 6e20 602d  r, the option `-
+0000f0c0: 2d72 6577 6569 6768 7460 2063 616e 2062  -reweight` can b
+0000f0d0: 6520 7365 742e 0a0a 2a2a 4120 7375 6767  e set...**A sugg
+0000f0e0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+0000f0f0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
+0000f100: 6e64 2061 7320 6265 6c6f 772a 2a0a 0a60  nd as below**..`
+0000f110: 6060 0a77 6764 206b 7364 2066 616d 696c  ``.wgd ksd famil
+0000f120: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+0000f130: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+0000f140: 330a 6060 600a 0a2a 2a54 6865 2063 6f6e  3.```..**The con
+0000f150: 7374 7275 6374 696f 6e20 6f66 202a 4b2a  struction of *K*
+0000f160: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000f170: 2064 6973 7472 6962 7574 696f 6e20 7769   distribution wi
+0000f180: 7468 2072 6174 6520 636f 7272 6563 7469  th rate correcti
+0000f190: 6f6e 2a2a 0a60 6060 0a77 6764 206b 7364  on**.```.wgd ksd
+0000f1a0: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
+0000f1b0: 6365 3120 7365 7175 656e 6365 3220 7365  ce1 sequence2 se
+0000f1c0: 7175 656e 6365 3320 2d2d 666f 6375 7332  quence3 --focus2
+0000f1d0: 616c 6c20 7365 7175 656e 6365 3120 2d73  all sequence1 -s
+0000f1e0: 7020 7370 6461 7461 202d 2d65 7874 7261  p spdata --extra
+0000f1f0: 7061 7261 6e6f 6d65 6b73 2070 6172 616e  paranomeks paran
+0000f200: 6f6d 654b 7364 6174 6120 282d 2d70 6c6f  omeKsdata (--plo
+0000f210: 7465 6c6d 6d20 2d2d 706c 6f74 6170 676d  telmm --plotapgm
+0000f220: 6d20 2d2d 616e 6368 6f72 706f 696e 7473  m --anchorpoints
+0000f230: 2061 7064 6174 6120 2d2d 7265 7765 6967   apdata --reweig
+0000f240: 6874 202d 2d6f 6e6c 7972 6f6f 746f 7574  ht --onlyrootout
+0000f250: 290a 6060 600a 0a49 6e73 7069 7265 6420  ).```..Inspired 
+0000f260: 6279 2074 6865 2072 6174 6520 636f 7272  by the rate corr
+0000f270: 6563 7469 6f6e 2061 6c67 6f72 6974 686d  ection algorithm
+0000f280: 2069 6e20 5b6b 7372 6174 6573 5d28 6874   in [ksrates](ht
+0000f290: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
+0000f2a0: 2f56 4942 2d50 5342 2f6b 7372 6174 6573  /VIB-PSB/ksrates
+0000f2b0: 292c 2077 6520 696d 706c 656d 656e 7465  ), we implemente
+0000f2c0: 6420 7468 6520 7261 7465 2063 6f72 7265  d the rate corre
+0000f2d0: 6374 696f 6e20 616e 616c 7973 6973 2061  ction analysis a
+0000f2e0: 6c73 6f20 696e 2060 7767 6420 7632 602c  lso in `wgd v2`,
+0000f2f0: 2077 6869 6368 2069 7320 6d6f 7374 6c79   which is mostly
+0000f300: 2074 6865 2073 616d 6520 6173 205b 6b73   the same as [ks
+0000f310: 7261 7465 735d 2868 7474 7073 3a2f 2f67  rates](https://g
+0000f320: 6974 6875 622e 636f 6d2f 5649 422d 5053  ithub.com/VIB-PS
+0000f330: 422f 6b73 7261 7465 7329 2062 7574 2064  B/ksrates) but d
+0000f340: 6966 6665 7273 2069 6e20 7468 6520 6361  iffers in the ca
+0000f350: 6c63 756c 6174 696f 6e20 6f66 2074 6865  lculation of the
+0000f360: 2073 7461 6e64 6172 6420 6465 7669 6174   standard deviat
+0000f370: 696f 6e20 6f66 2072 6573 6361 6c65 6420  ion of rescaled 
+0000f380: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000f390: 6167 6573 2e20 546f 2070 6572 666f 726d  ages. To perform
+0000f3a0: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
+0000f3b0: 7469 6f6e 2061 6e61 6c79 7369 732c 2075  tion analysis, u
+0000f3c0: 7365 7273 2063 616e 2075 7365 2062 6f74  sers can use bot
+0000f3d0: 6820 7468 6520 6077 6764 206b 7364 6020  h the `wgd ksd` 
+0000f3e0: 616e 6420 6077 6764 2076 697a 6020 7072  and `wgd viz` pr
+0000f3f0: 6f67 7261 6d2e 2046 6f72 2074 6865 2060  ogram. For the `
+0000f400: 7767 6420 6b73 6460 2070 726f 6772 616d  wgd ksd` program
+0000f410: 2c20 7573 6572 7320 6e65 6564 2074 6f20  , users need to 
+0000f420: 7072 6f76 6964 6520 6120 7370 6563 6965  provide a specie
+0000f430: 7320 7472 6565 2076 6961 2074 6865 206f  s tree via the o
+0000f440: 7074 696f 6e20 602d 2d73 7065 6369 6573  ption `--species
+0000f450: 7472 6565 6020 6f6e 2077 6869 6368 2072  tree` on which r
+0000f460: 6174 6520 636f 7272 6563 7469 6f6e 2063  ate correction c
+0000f470: 616e 2062 6520 636f 6e64 7563 7465 642e  an be conducted.
+0000f480: 204e 6f74 6520 7468 6174 2075 6e6e 6563   Note that unnec
+0000f490: 6573 7361 7279 2062 7261 636b 6574 7320  essary brackets 
+0000f4a0: 6d69 6768 7420 6c65 6164 2074 6f20 756e  might lead to un
+0000f4b0: 6578 7065 6374 6564 2065 7272 6f72 732c  expected errors,
+0000f4c0: 2066 6f72 2069 6e73 7461 6e63 6520 6120   for instance a 
+0000f4d0: 7472 6565 2060 2841 2c28 422c 4329 293b  tree `(A,(B,C));
+0000f4e0: 6020 7368 6f75 6c64 206e 6f74 2062 6520  ` should not be 
+0000f4f0: 7265 7072 6573 656e 7465 6420 6173 2060  represented as `
+0000f500: 2841 2c28 2842 2c43 2929 293b 602e 2054  (A,((B,C)));`. T
+0000f510: 6865 2073 6574 206f 6620 7370 6563 6965  he set of specie
+0000f520: 7320 7061 6972 7320 746f 2062 6520 7368  s pairs to be sh
+0000f530: 6f77 6e20 6973 2066 6c65 7869 626c 6520  own is flexible 
+0000f540: 7468 6174 2074 6865 206d 6f73 7420 636f  that the most co
+0000f550: 6e76 656e 6965 6e74 206f 7074 696f 6e20  nvenient option 
+0000f560: 6973 2060 2d2d 666f 6375 7332 616c 6c60  is `--focus2all`
+0000f570: 2077 6869 6368 2073 696d 706c 7920 7368   which simply sh
+0000f580: 6f77 7320 616c 6c20 7468 6520 706f 7373  ows all the poss
+0000f590: 6962 6c65 2066 6f63 616c 2d73 6973 7465  ible focal-siste
+0000f5a0: 7220 7370 6563 6965 7320 7061 6972 732c  r species pairs,
+0000f5b0: 206f 7220 7573 6572 7320 6361 6e20 6d61   or users can ma
+0000f5c0: 6e75 616c 6c79 2073 6574 2074 6865 2073  nually set the s
+0000f5d0: 7065 6369 6573 2070 6169 7273 2076 6961  pecies pairs via
+0000f5e0: 2074 6865 206f 7074 696f 6e20 602d 2d73   the option `--s
+0000f5f0: 7061 6972 602e 204e 6f74 6520 7468 6174  pair`. Note that
+0000f600: 2069 6620 7468 6520 7370 6563 6965 7320   if the species 
+0000f610: 7061 6972 7320 7765 7265 206d 616e 7561  pairs were manua
+0000f620: 6c6c 7920 7365 742c 2069 7420 776f 756c  lly set, it woul
+0000f630: 6420 6265 206e 6565 6465 6420 746f 2063  d be needed to c
+0000f640: 6f2d 7365 7420 7468 6520 6f70 7469 6f6e  o-set the option
+0000f650: 2060 2d2d 636c 6173 7369 6360 2e20 5468   `--classic`. Th
+0000f660: 6520 6f70 7469 6f6e 2060 2d2d 6f6e 6c79  e option `--only
+0000f670: 726f 6f74 6f75 7460 2063 616e 2062 6520  rootout` can be 
+0000f680: 7365 7420 746f 206f 6e6c 7920 636f 6e73  set to only cons
+0000f690: 6964 6572 206f 7574 6772 6f75 7020 6174  ider outgroup at
+0000f6a0: 2074 6865 2072 6f6f 742c 2069 6e73 7465   the root, inste
+0000f6b0: 6164 206f 6620 616c 6c20 7468 6520 706f  ad of all the po
+0000f6c0: 7373 6962 6c65 206f 7574 6772 6f75 7073  ssible outgroups
+0000f6d0: 2070 6572 2066 6f63 616c 2d73 6973 7465   per focal-siste
+0000f6e0: 7220 7370 6563 6965 7320 7061 6972 2c20  r species pair, 
+0000f6f0: 7768 6963 6820 6861 7320 696d 7061 6374  which has impact
+0000f700: 206f 6e20 7468 6520 6669 6e61 6c20 636f   on the final co
+0000f710: 7272 6563 7465 6420 2a4b 2a3c 7375 623e  rrected *K*<sub>
+0000f720: 533c 2f73 7562 3e20 6167 6573 2e20 5765  S</sub> ages. We
+0000f730: 2073 7567 6765 7374 206f 6620 7573 696e   suggest of usin
+0000f740: 6720 616c 6c20 7468 6520 706f 7373 6962  g all the possib
+0000f750: 6c65 206f 7574 6772 6f75 7073 2070 6572  le outgroups per
+0000f760: 2066 6f63 616c 2d73 6973 7465 7220 7370   focal-sister sp
+0000f770: 6563 6965 7320 7061 6972 2061 7320 666f  ecies pair as fo
+0000f780: 7220 6120 6c65 7373 2062 6961 7365 6420  r a less biased 
+0000f790: 7265 7375 6c74 2e20 5468 6520 7061 7261  result. The para
+0000f7a0: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
+0000f7b0: 7375 623e 2064 6174 6120 7368 6f75 6c64  sub> data should
+0000f7c0: 2062 6520 7072 6f76 6964 6564 2076 6961   be provided via
+0000f7d0: 2074 6865 206f 7074 696f 6e20 602d 2d65   the option `--e
+0000f7e0: 7874 7261 7061 7261 6e6f 6d65 6b73 602e  xtraparanomeks`.
+0000f7f0: 200a 0a53 6f6d 6520 6f74 6865 7220 6f70   ..Some other op
+0000f800: 7469 6f6e 7320 7768 6963 6820 6861 7665  tions which have
+0000f810: 206e 6f20 696d 7061 6374 206f 6e20 7468   no impact on th
+0000f820: 6520 7261 7465 2063 6f72 7265 6374 696f  e rate correctio
+0000f830: 6e20 6275 7420 6164 6420 6d6f 7265 206c  n but add more l
+0000f840: 6179 6572 7320 6f72 2063 6861 6e67 6520  ayers or change 
+0000f850: 7468 6520 6170 7065 6172 616e 6365 206f  the appearance o
+0000f860: 6e20 7468 6520 6d69 7865 6420 2a4b 2a3c  n the mixed *K*<
+0000f870: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+0000f880: 6469 7374 7269 6275 7469 6f6e 2c20 696e  distribution, in
+0000f890: 636c 7564 6520 602d 2d70 6c6f 7461 7067  clude `--plotapg
+0000f8a0: 6d6d 6020 616e 6420 602d 2d70 6c6f 7465  mm` and `--plote
+0000f8b0: 6c6d 6d60 2065 7463 2e20 5468 6520 6f70  lmm` etc. The op
+0000f8c0: 7469 6f6e 2060 2d2d 706c 6f74 6170 676d  tion `--plotapgm
+0000f8d0: 6d60 2063 616e 2062 6520 7365 7420 746f  m` can be set to
+0000f8e0: 2063 616c 6c20 7468 6520 474d 4d20 616e   call the GMM an
+0000f8f0: 616c 7973 6973 2075 706f 6e20 7468 6520  alysis upon the 
+0000f900: 616e 6368 6f72 2070 6169 7220 2a4b 2a3c  anchor pair *K*<
+0000f910: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+0000f920: 6469 7374 7269 6275 7469 6f6e 2061 6e64  distribution and
+0000f930: 2070 6c6f 7420 7468 6520 636c 7573 7465   plot the cluste
+0000f940: 7269 6e67 2072 6573 756c 7420 7570 6f6e  ring result upon
+0000f950: 2074 6865 206d 6978 6564 202a 4b2a 3c73   the mixed *K*<s
+0000f960: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+0000f970: 6973 7472 6962 7574 696f 6e2c 2077 6869  istribution, whi
+0000f980: 6368 2068 6173 2074 6f20 6265 2063 6f2d  ch has to be co-
+0000f990: 7365 7420 7769 7468 2074 6865 206f 7074  set with the opt
+0000f9a0: 696f 6e20 602d 2d61 6e63 686f 7270 6f69  ion `--anchorpoi
+0000f9b0: 6e74 7360 2070 726f 7669 6469 6e67 2074  nts` providing t
+0000f9c0: 6865 2061 6e63 686f 7220 7061 6972 7320  he anchor pairs 
+0000f9d0: 696e 666f 726d 6174 696f 6e2e 2054 6865  information. The
+0000f9e0: 206f 7074 696f 6e20 602d 2d70 6c6f 7465   option `--plote
+0000f9f0: 6c6d 6d60 2063 616e 2062 6520 7365 7420  lmm` can be set 
+0000fa00: 746f 2063 616c 6c20 7468 6520 454c 4d4d  to call the ELMM
+0000fa10: 2061 6e61 6c79 7369 7320 7570 6f6e 2074   analysis upon t
+0000fa20: 6865 2077 686f 6c65 2070 6172 616e 6f6d  he whole paranom
+0000fa30: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+0000fa40: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+0000fa50: 6f6e 2e20 2a2a 4e6f 7465 2074 6861 7420  on. **Note that 
+0000fa60: 7468 6520 7370 6563 6965 7320 6e61 6d65  the species name
+0000fa70: 7320 7072 6573 656e 7420 696e 2074 6865  s present in the
+0000fa80: 2073 7065 6369 6573 2074 7265 6520 6669   species tree fi
+0000fa90: 6c65 2073 686f 756c 6420 6d61 7463 6820  le should match 
+0000faa0: 7468 6520 6e61 6d65 7320 6f66 2074 6865  the names of the
+0000fab0: 2063 6f72 7265 7370 6f6e 6469 6e67 2073   corresponding s
+0000fac0: 6571 7565 6e63 6520 6669 6c65 732a 2a20  equence files** 
+0000fad0: 466f 7220 696e 7374 616e 6365 2c20 6769  For instance, gi
+0000fae0: 7665 6e20 7468 6520 6364 7320 6669 6c65  ven the cds file
+0000faf0: 206e 616d 6573 2027 412e 6364 7327 2c27   names 'A.cds','
+0000fb00: 422e 6364 7327 2c27 432e 6364 7327 2c20  B.cds','C.cds', 
+0000fb10: 7468 6520 7370 6563 6965 7320 7472 6565  the species tree
+0000fb20: 2063 6f75 6c64 2062 6520 2728 412e 6364   could be '(A.cd
+0000fb30: 732c 2842 2e63 6473 2c43 2e63 6473 2929  s,(B.cds,C.cds))
+0000fb40: 3b27 2072 6174 6865 7220 7468 616e 2027  ;' rather than '
+0000fb50: 2841 2c28 422c 4329 293b 272e 2054 6865  (A,(B,C));'. The
+0000fb60: 7265 2069 7320 6e6f 2072 6571 7569 7265  re is no require
+0000fb70: 6d65 6e74 2066 6f72 2074 6865 206e 616d  ment for the nam
+0000fb80: 6520 6f66 2074 6865 2070 6172 616e 6f6d  e of the paranom
+0000fb90: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+0000fba0: 3e20 6461 7461 6669 6c65 2077 6869 6368  > datafile which
+0000fbb0: 2063 616e 2062 6520 6e61 6d65 6420 696e   can be named in
+0000fbc0: 2077 6861 7465 7665 7220 6d61 6e6e 6572   whatever manner
+0000fbd0: 2075 7365 7273 2070 7265 6665 722e 2054   users prefer. T
+0000fbe0: 6865 2027 474d 4d27 2069 7320 7468 6520  he 'GMM' is the 
+0000fbf0: 6162 6272 6576 6961 7469 6f6e 206f 6620  abbreviation of 
+0000fc00: 4761 7573 7369 616e 204d 6978 7475 7265  Gaussian Mixture
+0000fc10: 204d 6f64 656c 696e 6720 7768 696c 6520   Modeling while 
+0000fc20: 7468 6520 2745 4c4d 4d27 2072 6566 6572  the 'ELMM' refer
+0000fc30: 7320 746f 2045 7870 6f6e 656e 7469 616c  s to Exponential
+0000fc40: 2d4c 6f67 6e6f 726d 616c 204d 6978 7475  -Lognormal Mixtu
+0000fc50: 7265 204d 6f64 656c 696e 6720 6173 205b  re Modeling as [
+0000fc60: 6b73 7261 7465 735d 2868 7474 7073 3a2f  ksrates](https:/
+0000fc70: 2f67 6974 6875 622e 636f 6d2f 5649 422d  /github.com/VIB-
+0000fc80: 5053 422f 6b73 7261 7465 7329 2069 6e74  PSB/ksrates) int
+0000fc90: 6572 7072 6574 732e 0a0a 5468 6572 6520  erprets...There 
+0000fca0: 6172 6520 3231 2063 6f6c 756d 6e73 2069  are 21 columns i
+0000fcb0: 6e20 7468 6520 7265 7375 6c74 2060 2e6b  n the result `.k
+0000fcc0: 732e 7473 7660 2066 696c 6520 6265 7369  s.tsv` file besi
+0000fcd0: 6465 7320 7468 6520 696e 6465 7820 636f  des the index co
+0000fce0: 6c75 6d6e 7320 6070 6169 7260 2061 7320  lumns `pair` as 
+0000fcf0: 7468 6520 756e 6971 7565 2069 6465 6e74  the unique ident
+0000fd00: 6966 6965 7220 666f 7220 6561 6368 2067  ifier for each g
+0000fd10: 656e 6520 7061 6972 2e20 5468 6520 604e  ene pair. The `N
+0000fd20: 602c 2060 5360 2c20 6064 4e60 2c20 6064  `, `S`, `dN`, `d
+0000fd30: 4e2f 6453 602c 2060 6453 602c 2060 6c60  N/dS`, `dS`, `l`
+0000fd40: 2061 6e64 2060 7460 2061 7265 2066 726f   and `t` are fro
+0000fd50: 6d20 7468 6520 636f 6465 6d6c 2072 6573  m the codeml res
+0000fd60: 756c 7473 2c20 7265 7072 6573 656e 7469  ults, representi
+0000fd70: 6e67 2074 6865 204e 2065 7374 696d 6174  ng the N estimat
+0000fd80: 652c 2074 6865 2053 2065 7374 696d 6174  e, the S estimat
+0000fd90: 652c 2074 6865 2064 4e20 6573 7469 6d61  e, the dN estima
+0000fda0: 7465 2c20 7468 6520 644e 2f64 5320 286f  te, the dN/dS (o
+0000fdb0: 6d65 6761 2920 6573 7469 6d61 7465 2c20  mega) estimate, 
+0000fdc0: 7468 6520 6453 2065 7374 696d 6174 652c  the dS estimate,
+0000fdd0: 2074 6865 206c 6f67 2d6c 696b 656c 6968   the log-likelih
+0000fde0: 6f6f 6420 616e 6420 7468 6520 7420 6573  ood and the t es
+0000fdf0: 7469 6d61 7465 2c20 7265 7370 6563 7469  timate, respecti
+0000fe00: 7665 6c79 2e20 5468 6520 6061 6c69 676e  vely. The `align
+0000fe10: 6d65 6e74 636f 7665 7261 6765 602c 2060  mentcoverage`, `
+0000fe20: 616c 6967 6e6d 656e 7469 6465 6e74 6974  alignmentidentit
+0000fe30: 7960 2061 6e64 2060 616c 6967 6e6d 656e  y` and `alignmen
+0000fe40: 746c 656e 6774 6860 2061 7265 2074 6865  tlength` are the
+0000fe50: 2069 6e66 6f72 6d61 7469 6f6e 2070 6572   information per
+0000fe60: 7461 696e 696e 6720 746f 2074 6865 2061  taining to the a
+0000fe70: 6c69 676e 6d65 6e74 2066 6f72 2065 6163  lignment for eac
+0000fe80: 6820 6661 6d69 6c79 2c20 7265 7072 6573  h family, repres
+0000fe90: 656e 7469 6e67 2074 6865 2072 6174 696f  enting the ratio
+0000fea0: 206f 6620 7468 6520 7374 7269 7070 6564   of the stripped
+0000feb0: 2061 6c69 676e 6d65 6e74 206c 656e 6774   alignment lengt
+0000fec0: 6820 636f 6d70 6172 6564 2074 6f20 7468  h compared to th
+0000fed0: 6520 6675 6c6c 2061 6c69 676e 6d65 6e74  e full alignment
+0000fee0: 206c 656e 6774 682c 2074 6865 2072 6174   length, the rat
+0000fef0: 696f 206f 6620 636f 6c75 6d6e 7320 7769  io of columns wi
+0000ff00: 7468 2069 6465 6e74 6963 616c 206e 7563  th identical nuc
+0000ff10: 6c65 6f74 6964 6573 2063 6f6d 7061 7265  leotides compare
+0000ff20: 6420 746f 2074 6865 206f 7665 7261 6c6c  d to the overall
+0000ff30: 2063 6f6c 756d 6e73 206f 6620 7468 6520   columns of the 
+0000ff40: 7374 7269 7070 6564 2061 6c69 676e 6d65  stripped alignme
+0000ff50: 6e74 2c20 616e 6420 7468 6520 6c65 6e67  nt, and the leng
+0000ff60: 7468 206f 6620 7468 6520 6675 6c6c 2061  th of the full a
+0000ff70: 6c69 676e 6d65 6e74 2c20 7265 7370 6563  lignment, respec
+0000ff80: 7469 7665 6c79 2e0a 2323 2320 7767 6420  tively..### wgd 
+0000ff90: 6d69 780a 0a2a 2a54 6865 206d 6978 7475  mix..**The mixtu
+0000ffa0: 7265 206d 6f64 656c 2063 6c75 7374 6572  re model cluster
+0000ffb0: 696e 6720 616e 616c 7973 6973 206f 6620  ing analysis of 
+0000ffc0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0000ffd0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
+0000ffe0: 2a2a 0a60 6060 0a77 6764 206d 6978 206b  **.```.wgd mix k
+0000fff0: 7364 6174 6120 282d 2d6e 5f69 6e69 7420  sdata (--n_init 
+00010000: 3230 3020 2d2d 6d61 785f 6974 6572 2032  200 --max_iter 2
+00010010: 3030 202d 2d6b 735f 7261 6e67 6520 3020  00 --ks_range 0 
+00010020: 3520 2d2d 6669 6c74 6572 7320 3330 3020  5 --filters 300 
+00010030: 2d2d 6269 6e73 2035 3020 2d2d 636f 6d70  --bins 50 --comp
+00010040: 6f6e 656e 7473 2031 2034 202d 2d67 616d  onents 1 4 --gam
+00010050: 6d61 2030 2e30 3031 290a 6060 600a 0a54  ma 0.001).```..T
+00010060: 6869 7320 7061 7274 206f 6620 4761 7573  his part of Gaus
+00010070: 7369 616e 206d 6978 7475 7265 206d 6f64  sian mixture mod
+00010080: 656c 696e 6720 2847 4d4d 2920 616e 616c  eling (GMM) anal
+00010090: 7973 6973 2069 7320 696e 6865 7269 7465  ysis is inherite
+000100a0: 6420 6672 6f6d 2074 6865 206f 7269 6769  d from the origi
+000100b0: 6e61 6c20 6077 6764 6020 7072 6f67 7261  nal `wgd` progra
+000100c0: 6d2c 2062 7574 2077 7269 7465 7320 6164  m, but writes ad
+000100d0: 6469 7469 6f6e 616c 6c79 2074 6865 2070  ditionally the p
+000100e0: 726f 6261 6269 6c69 7479 206f 6620 6561  robability of ea
+000100f0: 6368 202a 4b2a 3c73 7562 3e53 3c2f 7375  ch *K*<sub>S</su
+00010100: 623e 2076 616c 7565 2069 6e74 6f20 7468  b> value into th
+00010110: 6520 6669 6e61 6c20 6461 7461 6672 616d  e final datafram
+00010120: 652e 2042 6173 6963 616c 6c79 2c20 7573  e. Basically, us
+00010130: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
+00010140: 6964 6520 7769 7468 2061 2028 6e6f 726d  ide with a (norm
+00010150: 616c 6c79 2066 726f 6d20 7768 6f6c 652d  ally from whole-
+00010160: 7061 7261 6e6f 6d65 206f 7220 616e 6368  paranome or anch
+00010170: 6f72 2d70 6169 7273 2920 2a4b 2a3c 7375  or-pairs) *K*<su
+00010180: 623e 533c 2f73 7562 3e20 6461 7461 6669  b>S</sub> datafi
+00010190: 6c65 2061 6e64 2074 6865 2047 4d4d 2061  le and the GMM a
+000101a0: 6e61 6c79 7369 7320 7769 6c6c 2062 6520  nalysis will be 
+000101b0: 636f 6e64 7563 7465 6420 7570 6f6e 2074  conducted upon t
+000101c0: 6865 2064 6174 6166 696c 652e 2053 6f6d  he datafile. Som
+000101d0: 6520 7061 7261 6d65 7465 7273 2063 616e  e parameters can
+000101e0: 2061 6666 6563 7420 7468 6520 7265 7375   affect the resu
+000101f0: 6c74 732c 2069 6e63 6c75 6469 6e67 2060  lts, including `
+00010200: 2d2d 6e5f 696e 6974 602c 2077 6869 6368  --n_init`, which
+00010210: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
+00010220: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
+00010230: 6961 6c69 7a61 7469 6f6e 7320 2864 6566  ializations (def
+00010240: 6175 6c74 2032 3030 292c 2060 2d2d 6d61  ault 200), `--ma
+00010250: 785f 6974 6572 602c 2077 6869 6368 2073  x_iter`, which s
+00010260: 6574 7320 7468 6520 6d61 7869 6d75 6d20  ets the maximum 
+00010270: 6e75 6d62 6572 206f 6620 6974 6572 6174  number of iterat
+00010280: 696f 6e73 2028 6465 6661 756c 7420 3230  ions (default 20
+00010290: 3029 2c20 602d 2d6d 6574 686f 6460 2c20  0), `--method`, 
+000102a0: 7768 6963 6820 6465 7465 726d 696e 6573  which determines
+000102b0: 2077 6869 6368 2063 6c75 7374 6572 696e   which clusterin
+000102c0: 6720 6d65 7468 6f64 2074 6f20 7573 6520  g method to use 
+000102d0: 2864 6566 6175 6c74 2067 6d6d 292c 2060  (default gmm), `
+000102e0: 2d2d 6761 6d6d 6160 2c20 7768 6963 6820  --gamma`, which 
+000102f0: 7365 7473 2074 6865 2067 616d 6d61 2070  sets the gamma p
+00010300: 6172 616d 6574 6572 2066 6f72 2074 6865  arameter for the
+00010310: 2062 676d 6d20 6d6f 6465 6c20 2864 6566   bgmm model (def
+00010320: 6175 6c74 2030 2e30 3031 292c 2060 2d2d  ault 0.001), `--
+00010330: 636f 6d70 6f6e 656e 7473 602c 2077 6869  components`, whi
+00010340: 6368 2073 6574 7320 7468 6520 7261 6e67  ch sets the rang
+00010350: 6520 6f66 2074 6865 206e 756d 6265 7220  e of the number 
+00010360: 6f66 2063 6f6d 706f 6e65 6e74 7320 746f  of components to
+00010370: 2066 6974 2028 6465 6661 756c 7420 3120   fit (default 1 
+00010380: 3429 2c20 616e 6420 7468 6520 6461 7461  4), and the data
+00010390: 2066 696c 7465 7269 6e67 2070 6172 616d   filtering param
+000103a0: 6574 6572 7320 602d 2d66 696c 7465 7273  eters `--filters
+000103b0: 6020 7768 6963 6820 6669 6c74 6572 7320  ` which filters 
+000103c0: 6461 7461 2062 6173 6564 206f 6e20 616c  data based on al
+000103d0: 6967 6e6d 656e 7420 6c65 6e67 7468 2c20  ignment length, 
+000103e0: 602d 2d6b 735f 7261 6e67 6560 2077 6869  `--ks_range` whi
+000103f0: 6368 2066 696c 7465 7273 2064 6174 6120  ch filters data 
+00010400: 6261 7365 6420 6f6e 202a 4b2a 3c73 7562  based on *K*<sub
+00010410: 3e53 3c2f 7375 623e 2076 616c 7565 7320  >S</sub> values 
+00010420: 616e 6420 7468 6520 7061 7261 6d65 7465  and the paramete
+00010430: 7220 602d 2d62 696e 7360 2077 6869 6368  r `--bins` which
+00010440: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
+00010450: 206f 6620 6269 6e73 2069 6e20 2a4b 2a3c   of bins in *K*<
+00010460: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00010470: 7269 6275 7469 6f6e 2028 6465 6661 756c  ribution (defaul
+00010480: 7420 3530 292e 0a0a 2a2a 4120 7375 6767  t 50)...**A sugg
+00010490: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+000104a0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
+000104b0: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
+000104c0: 6f77 2a2a 0a0a 6060 600a 7767 6420 6d69  ow**..```.wgd mi
+000104d0: 7820 6b73 6461 7461 0a60 6060 0a0a 2323  x ksdata.```..##
+000104e0: 2320 7767 6420 7065 616b 0a0a 2a2a 5468  # wgd peak..**Th
+000104f0: 6520 7365 6172 6368 206f 6620 6372 6564  e search of cred
+00010500: 6961 626c 6520 2a4b 2a3c 7375 623e 533c  iable *K*<sub>S<
+00010510: 2f73 7562 3e20 7261 6e67 6520 7573 6564  /sub> range used
+00010520: 2069 6e20 5747 4420 6461 7469 6e67 2a2a   in WGD dating**
+00010530: 0a60 6060 0a77 6764 2070 6561 6b20 6b73  .```.wgd peak ks
+00010540: 6461 7461 202d 6170 2061 7064 6174 6120  data -ap apdata 
+00010550: 2d73 6d20 736d 6461 7461 202d 6c65 206c  -sm smdata -le l
+00010560: 6564 6174 6120 2d6d 7020 6d70 6461 7461  edata -mp mpdata
+00010570: 202d 2d68 6575 7269 7374 6963 2028 2d2d   --heuristic (--
+00010580: 616c 6967 6e66 696c 7465 7220 302e 3020  alignfilter 0.0 
+00010590: 3020 302e 3020 2d2d 6b73 7261 6e67 6520  0 0.0 --ksrange 
+000105a0: 3020 3520 2d2d 6269 6e5f 7769 6474 6820  0 5 --bin_width 
+000105b0: 302e 3120 2d2d 6775 6964 6520 7365 676d  0.1 --guide segm
+000105c0: 656e 7420 2d2d 7072 6f6d 696e 656e 6365  ent --prominence
+000105d0: 5f63 7574 6f66 6620 302e 3120 2d2d 7265  _cutoff 0.1 --re
+000105e0: 6c5f 6865 6967 6874 2030 2e34 202d 2d63  l_height 0.4 --c
+000105f0: 6920 3935 202d 2d68 6472 2039 3520 2d2d  i 95 --hdr 95 --
+00010600: 6b73 6375 746f 6666 2035 290a 6060 600a  kscutoff 5).```.
+00010610: 0a41 7320 6d65 6e74 696f 6e65 6420 7072  .As mentioned pr
+00010620: 6576 696f 7573 6c79 2c20 6120 6865 7572  eviously, a heur
+00010630: 6973 7469 6320 6d65 7468 6f64 2061 6e64  istic method and
+00010640: 2061 2063 6f6c 6c69 6e65 6172 2073 6567   a collinear seg
+00010650: 6d65 6e74 732d 6775 6964 6564 2061 6e63  ments-guided anc
+00010660: 686f 7220 7061 6972 2063 6c75 7374 6572  hor pair cluster
+00010670: 696e 6720 666f 7220 7468 6520 7365 6172  ing for the sear
+00010680: 6368 206f 6620 6372 6564 6961 626c 6520  ch of crediable 
+00010690: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000106a0: 7261 6e67 6520 7573 6564 2069 6e20 5747  range used in WG
+000106b0: 4420 6461 7469 6e67 2061 7265 2069 6d70  D dating are imp
+000106c0: 6c65 6d65 6e74 6564 2069 6e20 6077 6764  lemented in `wgd
+000106d0: 2076 3260 2e20 5573 6572 7320 6e65 6564   v2`. Users need
+000106e0: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
+000106f0: 616e 6368 6f72 2070 6f69 6e74 732c 2073  anchor points, s
+00010700: 6567 6d65 6e74 732c 206c 6973 7473 6567  egments, listseg
+00010710: 6d65 6e74 732c 206d 756c 7469 706c 6963  ments, multiplic
+00010720: 6f6e 7061 6972 7320 6461 7461 6669 6c65  onpairs datafile
+00010730: 2066 726f 6d20 6069 2d61 6468 6f72 6560   from `i-adhore`
+00010740: 2074 6f20 6163 6869 6576 6520 7468 6520   to achieve the 
+00010750: 636c 7573 7465 7269 6e67 2066 756e 6374  clustering funct
+00010760: 696f 6e2e 2053 6f6d 6520 7061 7261 6d65  ion. Some parame
+00010770: 7465 7273 2074 6861 7420 6361 6e20 696d  ters that can im
+00010780: 7061 6374 2074 6865 2072 6573 756c 7473  pact the results
+00010790: 2069 6e63 6c75 6465 2060 2d2d 616c 6967   include `--alig
+000107a0: 6e66 696c 7465 7260 2c20 7768 6963 6820  nfilter`, which 
+000107b0: 6669 6c74 6572 7320 7468 6520 6461 7461  filters the data
+000107c0: 2062 6173 6564 206f 6e20 616c 6967 6e6d   based on alignm
+000107d0: 656e 7420 6964 656e 7469 7479 2c20 6c65  ent identity, le
+000107e0: 6e67 7468 2061 6e64 2063 6f76 6572 6167  ngth and coverag
+000107f0: 652c 2060 2d2d 6b73 7261 6e67 6560 2c20  e, `--ksrange`, 
+00010800: 7768 6963 6820 7365 7473 2074 6865 2072  which sets the r
+00010810: 616e 6765 206f 6620 4b73 2074 6f20 6265  ange of Ks to be
+00010820: 2061 6e61 6c79 7a65 642c 2060 2d2d 6269   analyzed, `--bi
+00010830: 6e5f 7769 6474 6860 2c20 7768 6963 6820  n_width`, which 
+00010840: 7365 7473 2074 6865 2062 616e 6477 6964  sets the bandwid
+00010850: 7468 206f 6620 2a4b 2a3c 7375 623e 533c  th of *K*<sub>S<
+00010860: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+00010870: 6f6e 2c20 602d 2d77 6569 6768 7473 5f6f  on, `--weights_o
+00010880: 7574 6c69 6572 735f 696e 636c 7564 6564  utliers_included
+00010890: 6020 7768 6963 6820 6465 7465 726d 696e  ` which determin
+000108a0: 6573 2077 6865 7468 6572 2074 6f20 696e  es whether to in
+000108b0: 636c 7564 6520 2a4b 2a3c 7375 623e 533c  clude *K*<sub>S<
+000108c0: 2f73 7562 3e20 6f75 746c 6965 7273 2028  /sub> outliers (
+000108d0: 7768 6f73 6520 7661 6c75 6520 6973 206f  whose value is o
+000108e0: 7665 7220 3529 2069 6e20 616e 616c 7973  ver 5) in analys
+000108f0: 6973 2c20 602d 2d6d 6574 686f 6460 2077  is, `--method` w
+00010900: 6869 6368 2064 6574 6572 6d69 6e65 7320  hich determines 
+00010910: 7768 6963 6820 636c 7573 7465 7269 6e67  which clustering
+00010920: 206d 6574 686f 6420 746f 2075 7365 2028   method to use (
+00010930: 6465 6661 756c 7420 676d 6d29 2c20 602d  default gmm), `-
+00010940: 2d73 6565 6460 2077 6869 6368 2073 6574  -seed` which set
+00010950: 7320 7468 6520 7261 6e64 6f6d 2073 6565  s the random see
+00010960: 6420 6769 7665 6e20 746f 2069 6e69 7469  d given to initi
+00010970: 616c 697a 6174 696f 6e20 2864 6566 6175  alization (defau
+00010980: 6c74 2032 3335 3238 3930 292c 2060 2d2d  lt 2352890), `--
+00010990: 6e5f 696e 6974 602c 2077 6869 6368 2073  n_init`, which s
+000109a0: 6574 7320 7468 6520 6e75 6d62 6572 206f  ets the number o
+000109b0: 6620 6b2d 6d65 616e 7320 696e 6974 6961  f k-means initia
+000109c0: 6c69 7a61 7469 6f6e 7320 2864 6566 6175  lizations (defau
+000109d0: 6c74 2032 3030 292c 2060 2d2d 656d 5f69  lt 200), `--em_i
+000109e0: 7465 7260 2c20 7768 6963 6820 7365 7473  ter`, which sets
+000109f0: 2074 6865 206d 6178 696d 756d 206e 756d   the maximum num
+00010a00: 6265 7220 6f66 2069 7465 7261 7469 6f6e  ber of iteration
+00010a10: 7320 2864 6566 6175 6c74 2032 3030 292c  s (default 200),
+00010a20: 2060 2d2d 6761 6d6d 6160 2c20 7768 6963   `--gamma`, whic
+00010a30: 6820 7365 7473 2074 6865 2067 616d 6d61  h sets the gamma
+00010a40: 2070 6172 616d 6574 6572 2066 6f72 2074   parameter for t
+00010a50: 6865 2062 676d 6d20 6d6f 6465 6c20 2864  he bgmm model (d
+00010a60: 6566 6175 6c74 2030 2e30 3031 292c 2060  efault 0.001), `
+00010a70: 2d2d 636f 6d70 6f6e 656e 7473 602c 2077  --components`, w
+00010a80: 6869 6368 2073 6574 7320 7468 6520 7261  hich sets the ra
+00010a90: 6e67 6520 6f66 2074 6865 206e 756d 6265  nge of the numbe
+00010aa0: 7220 6f66 2063 6f6d 706f 6e65 6e74 7320  r of components 
+00010ab0: 746f 2066 6974 2028 6465 6661 756c 7420  to fit (default 
+00010ac0: 3120 3429 2c20 602d 2d77 6569 6768 7465  1 4), `--weighte
+00010ad0: 6460 2077 6869 6368 2064 6574 6572 6d69  d` which determi
+00010ae0: 6e65 7320 7768 6574 6865 7220 746f 2075  nes whether to u
+00010af0: 7365 206e 6f64 652d 7765 6967 6874 6564  se node-weighted
+00010b00: 206d 6574 686f 6420 666f 7220 6465 2d72   method for de-r
+00010b10: 6564 756e 6461 6e63 792c 2060 2d2d 6775  edundancy, `--gu
+00010b20: 6964 6560 2077 6869 6368 2064 6574 6572  ide` which deter
+00010b30: 6d69 6e65 7320 7768 6963 6820 7265 6769  mines which regi
+00010b40: 6d65 2072 6573 6964 696e 6720 616e 6368  me residing anch
+00010b50: 6f72 7320 746f 2062 6520 7573 6564 2028  ors to be used (
+00010b60: 6465 6661 756c 7420 5365 676d 656e 7429  default Segment)
+00010b70: 2c20 602d 2d70 726f 6d69 6e65 6e63 655f  , `--prominence_
+00010b80: 6375 746f 6666 6020 7768 6963 6820 7365  cutoff` which se
+00010b90: 7473 2074 6865 2070 726f 6d69 6e65 6e63  ts the prominenc
+00010ba0: 6520 6375 746f 6666 206f 6620 6163 6365  e cutoff of acce
+00010bb0: 7074 6162 6c65 2070 6561 6b73 2069 6e20  ptable peaks in 
+00010bc0: 7065 616b 2066 696e 6469 6e67 2070 726f  peak finding pro
+00010bd0: 6365 7373 2c20 602d 2d72 656c 5f68 6569  cess, `--rel_hei
+00010be0: 6768 7460 2077 6869 6368 2073 6574 7320  ght` which sets 
+00010bf0: 7468 6520 7265 6c61 7469 7665 2068 6569  the relative hei
+00010c00: 6768 7420 6174 2077 6869 6368 2074 6865  ght at which the
+00010c10: 2070 6561 6b20 7769 6474 6820 6973 206d   peak width is m
+00010c20: 6561 7375 7265 642c 2060 2d2d 6b73 746f  easured, `--ksto
+00010c30: 6461 7465 6020 7768 6963 6820 6d61 6e75  date` which manu
+00010c40: 616c 6c79 2073 6574 7320 7468 6520 7261  ally sets the ra
+00010c50: 6e67 6520 6f66 202a 4b2a 3c73 7562 3e53  nge of *K*<sub>S
+00010c60: 3c2f 7375 623e 2074 6f20 6265 2064 6174  </sub> to be dat
+00010c70: 6564 2069 6e20 6865 7572 6973 7469 6320  ed in heuristic 
+00010c80: 7365 6172 6368 2061 6e64 206e 6565 6473  search and needs
+00010c90: 2074 6f20 6265 2063 6f2d 7365 7420 7769   to be co-set wi
+00010ca0: 7468 206f 7074 696f 6e20 602d 2d6d 616e  th option `--man
+00010cb0: 7561 6c73 6574 602c 2060 2d2d 786c 696d  ualset`, `--xlim
+00010cc0: 6020 616e 6420 602d 2d79 6c69 6d60 2064  ` and `--ylim` d
+00010cd0: 6574 6572 6d69 6e69 6e67 2074 6865 2078  etermining the x
+00010ce0: 2061 6e64 2079 2061 7869 7320 6c69 6d69   and y axis limi
+00010cf0: 7420 6f66 2047 4d4d 202a 4b2a 3c73 7562  t of GMM *K*<sub
+00010d00: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
+00010d10: 7574 696f 6e2c 2060 2d2d 6369 6020 7365  ution, `--ci` se
+00010d20: 7474 696e 6720 7468 6520 636f 6e66 6964  tting the confid
+00010d30: 656e 6365 206c 6576 656c 206f 6620 6c6f  ence level of lo
+00010d40: 672d 6e6f 726d 616c 2064 6973 7472 6962  g-normal distrib
+00010d50: 7574 696f 6e20 746f 2064 6174 6520 2864  ution to date (d
+00010d60: 6566 6175 6c74 2039 3529 2c20 602d 2d68  efault 95), `--h
+00010d70: 6472 6020 7365 7474 696e 6720 7468 6520  dr` setting the 
+00010d80: 6869 6768 6573 7420 6465 6e73 6974 7920  highest density 
+00010d90: 7265 6769 6f6e 2028 4844 5229 2061 7070  region (HDR) app
+00010da0: 6c69 6564 2069 6e20 7468 6520 7365 676d  lied in the segm
+00010db0: 656e 742d 6775 6964 6564 2061 6e63 686f  ent-guided ancho
+00010dc0: 7220 7061 6972 202a 4b2a 3c73 7562 3e53  r pair *K*<sub>S
+00010dd0: 3c2f 7375 623e 2064 6973 7472 6962 7574  </sub> distribut
+00010de0: 696f 6e20 2864 6566 6175 6c74 2039 3529  ion (default 95)
+00010df0: 2c20 602d 2d68 6575 7269 7374 6963 6020  , `--heuristic` 
+00010e00: 6465 7465 726d 696e 696e 6720 7768 6574  determining whet
+00010e10: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+00010e20: 6865 7572 6973 7469 6320 6d65 7468 6f64  heuristic method
+00010e30: 206f 6620 6465 6669 6e69 6e67 2043 4920   of defining CI 
+00010e40: 666f 7220 6461 7469 6e67 2c20 602d 2d6b  for dating, `--k
+00010e50: 7363 7574 6f66 6660 2073 6574 7469 6e67  scutoff` setting
+00010e60: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+00010e70: 7375 623e 2073 6174 7572 6174 696f 6e20  sub> saturation 
+00010e80: 6375 746f 6666 2069 6e20 6461 7469 6e67  cutoff in dating
+00010e90: 2028 6465 6661 756c 7420 3529 2e0a 0a46   (default 5)...F
+00010ea0: 6f75 7220 7265 7375 6c74 2073 7562 666c  our result subfl
+00010eb0: 6f64 6572 7320 7769 6c6c 2062 6520 7072  oders will be pr
+00010ec0: 6f64 7563 6564 2c20 6e61 6d65 6c79 2060  oduced, namely `
+00010ed0: 416e 6368 6f72 4b73 5f46 696e 6450 6561  AnchorKs_FindPea
+00010ee0: 6b60 2c20 6041 6e63 686f 724b 735f 474d  k`, `AnchorKs_GM
+00010ef0: 4d60 2c20 6053 6567 6d65 6e74 4775 6964  M`, `SegmentGuid
+00010f00: 654b 735f 474d 4d60 2061 6e64 2060 5365  eKs_GMM` and `Se
+00010f10: 676d 656e 744b 735f 4669 6e64 5065 616b  gmentKs_FindPeak
+00010f20: 602e 2054 6865 2060 416e 6368 6f72 4b73  `. The `AnchorKs
+00010f30: 5f46 696e 6450 6561 6b60 2073 7562 666c  _FindPeak` subfl
+00010f40: 6f64 6572 2063 6f6e 7461 696e 7320 7265  oder contains re
+00010f50: 7375 6c74 7320 6f66 2074 6865 2064 6574  sults of the det
+00010f60: 6563 7465 6420 7065 616b 7320 6279 2074  ected peaks by t
+00010f70: 6865 2060 7369 676e 616c 6020 6d6f 6475  he `signal` modu
+00010f80: 6c65 206f 6620 6053 6369 5079 6020 6c69  le of `SciPy` li
+00010f90: 6272 6172 7920 616e 6420 7468 6520 6173  brary and the as
+00010fa0: 7375 6d65 6420 6869 6768 6573 7420 6d61  sumed highest ma
+00010fb0: 7373 2070 6172 7420 2872 6566 6572 7265  ss part (referre
+00010fc0: 6420 746f 2061 7320 4869 6768 4d61 7373  d to as HighMass
+00010fd0: 2068 6572 6561 6674 6572 2920 6f66 2065   hereafter) of e
+00010fe0: 6163 6820 7065 616b 2c20 7768 6963 6820  ach peak, which 
+00010ff0: 6361 6e20 6265 2075 7365 6420 666f 7220  can be used for 
+00011000: 6675 7274 6865 7220 5747 4420 6461 7469  further WGD dati
+00011010: 6e67 2e20 5468 6520 6041 6e63 686f 724b  ng. The `AnchorK
+00011020: 735f 474d 4d60 2073 686f 7773 2074 6865  s_GMM` shows the
+00011030: 2047 4d4d 2072 6573 756c 7473 2075 706f   GMM results upo
+00011040: 6e20 7468 6520 6f72 6967 696e 616c 2061  n the original a
+00011050: 6e63 686f 7220 2a4b 2a3c 7375 623e 533c  nchor *K*<sub>S<
+00011060: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+00011070: 6f6e 2062 7920 7468 6520 606d 6978 7475  on by the `mixtu
+00011080: 7265 6020 6d6f 6475 6c65 206f 6620 6073  re` module of `s
+00011090: 6369 6b69 742d 6c65 6172 6e60 206c 6962  cikit-learn` lib
+000110a0: 7261 7279 2061 6e64 2074 776f 2073 7562  rary and two sub
+000110b0: 666c 6f64 6572 732c 2060 4c6f 6747 4d4d  floders, `LogGMM
+000110c0: 5f43 4960 2063 6f6e 7461 696e 696e 6720  _CI` containing 
+000110d0: 7468 6520 7265 7375 6c74 7320 6f66 2039  the results of 9
+000110e0: 3525 2043 4920 6f66 2065 6163 6820 636f  5% CI of each co
+000110f0: 6d70 6f6e 656e 742c 2060 4869 6768 4d61  mponent, `HighMa
+00011100: 7373 5f43 4960 2063 6f6e 7461 696e 696e  ss_CI` containin
+00011110: 6720 7468 6520 4869 6768 4d61 7373 206f  g the HighMass o
+00011120: 6620 6561 6368 2063 6f6d 706f 6e65 6e74  f each component
+00011130: 2c20 7768 6963 6820 6361 6e20 6265 2075  , which can be u
+00011140: 7365 6420 666f 7220 6675 7274 6865 7220  sed for further 
+00011150: 5747 4420 6461 7469 6e67 2e20 5468 6520  WGD dating. The 
+00011160: 6053 6567 6d65 6e74 4775 6964 654b 735f  `SegmentGuideKs_
+00011170: 474d 4d60 2073 7562 666f 6c64 6572 2070  GMM` subfolder p
+00011180: 7265 7365 6e74 7320 7265 7375 6c74 7320  resents results 
+00011190: 6f66 2073 6567 6d65 6e74 202a 4b2a 3c73  of segment *K*<s
+000111a0: 7562 3e53 3c2f 7375 623e 2047 4d4d 2077  ub>S</sub> GMM w
+000111b0: 6869 6368 2061 7265 206d 6170 7065 6420  hich are mapped 
+000111c0: 6261 636b 2074 6f20 7468 6520 7265 7369  back to the resi
+000111d0: 6469 6e67 2061 6e63 686f 7220 7061 6972  ding anchor pair
+000111e0: 7320 616e 6420 7468 6520 6173 736f 6369  s and the associ
+000111f0: 6174 6564 2039 3525 2048 4452 2061 6e64  ated 95% HDR and
+00011200: 2048 6967 684d 6173 7320 6f66 2065 6163   HighMass of eac
+00011210: 6820 7365 676d 656e 7420 636c 7573 7465  h segment cluste
+00011220: 7220 696e 2073 7562 666c 6f64 6572 7320  r in subfloders 
+00011230: 6f66 2060 4844 525f 4349 6020 616e 6420  of `HDR_CI` and 
+00011240: 6048 6967 684d 6173 735f 4349 602e 2054  `HighMass_CI`. T
+00011250: 6865 2060 5365 676d 656e 744b 735f 4669  he `SegmentKs_Fi
+00011260: 6e64 5065 616b 6020 7375 6266 6f6c 6465  ndPeak` subfolde
+00011270: 7220 6973 2073 696d 696c 6172 2074 6f20  r is similar to 
+00011280: 6041 6e63 686f 724b 735f 4669 6e64 5065  `AnchorKs_FindPe
+00011290: 616b 6020 6275 7420 7769 7468 2073 6567  ak` but with seg
+000112a0: 6d65 6e74 202a 4b2a 3c73 7562 3e53 3c2f  ment *K*<sub>S</
+000112b0: 7375 623e 2069 6e73 7465 6164 2e20 5468  sub> instead. Th
+000112c0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+000112d0: 3e20 696e 2060 4d75 6c74 6970 6c69 636f  > in `Multiplico
+000112e0: 6e60 2063 616e 2061 6c73 6f20 6265 2063  n` can also be c
+000112f0: 616c 6375 6c61 7465 6420 696e 2070 6c61  alculated in pla
+00011300: 6365 206f 6620 6053 6567 6d65 6e74 6020  ce of `Segment` 
+00011310: 7573 696e 6720 7468 6520 6f70 7469 6f6e  using the option
+00011320: 2060 2d2d 6775 6964 6560 2061 7320 7375   `--guide` as su
+00011330: 6368 2074 6865 2072 6573 756c 7420 7469  ch the result ti
+00011340: 746c 652c 206c 6162 656c 2c20 6669 6c65  tle, label, file
+00011350: 2061 6e64 2066 6f6c 6465 7220 6e61 6d65   and folder name
+00011360: 7320 7769 6c6c 2062 6520 6368 616e 6765  s will be change
+00011370: 6420 6163 636f 7264 696e 676c 792e 0a0a  d accordingly...
+00011380: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
+00011390: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
+000113a0: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
+000113b0: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
+000113c0: 600a 7767 6420 7065 616b 206b 7364 6174  `.wgd peak ksdat
+000113d0: 6120 2d61 7020 6170 6461 7461 202d 736d  a -ap apdata -sm
+000113e0: 2073 6d64 6174 6120 2d6c 6520 6c65 6461   smdata -le leda
+000113f0: 7461 202d 6d70 206d 7064 6174 6120 2d2d  ta -mp mpdata --
+00011400: 6865 7572 6973 7469 630a 6060 600a 0a23  heuristic.```..#
+00011410: 2323 2077 6764 2073 796e 0a0a 2a2a 5468  ## wgd syn..**Th
+00011420: 6520 696e 7472 612d 7370 6563 6966 6963  e intra-specific
+00011430: 2073 796e 7465 6e79 2069 6e66 6572 656e   synteny inferen
+00011440: 6365 2a2a 0a60 6060 0a77 6764 2073 796e  ce**.```.wgd syn
+00011450: 2066 616d 696c 6965 7320 6766 6620 282d   families gff (-
+00011460: 2d6b 735f 6469 7374 7269 6275 7469 6f6e  -ks_distribution
+00011470: 206b 7364 6174 6120 2d66 2067 656e 6520   ksdata -f gene 
+00011480: 2d61 2049 4420 2d2d 6d69 6e6c 656e 202d  -a ID --minlen -
+00011490: 3120 2d2d 6d69 6e73 6567 6c65 6e20 3130  1 --minseglen 10
+000114a0: 3030 3020 2d2d 6d69 6e67 656e 656e 756d  000 --mingenenum
+000114b0: 2033 3029 0a60 6060 0a0a 5468 6520 7072   30).```..The pr
+000114c0: 6f67 7261 6d20 6077 6764 2073 796e 6020  ogram `wgd syn` 
+000114d0: 6973 206d 6169 6e6c 7920 6465 616c 696e  is mainly dealin
+000114e0: 6720 7769 7468 2063 6f6c 6c69 6e65 6172  g with collinear
+000114f0: 6974 7920 6f72 2073 796e 7465 6e79 2028  ity or synteny (
+00011500: 626f 7468 2072 6566 6572 7265 6420 746f  both referred to
+00011510: 2061 7320 7379 6e74 656e 7920 6865 7265   as synteny here
+00011520: 6166 7465 7229 2061 6e61 6c79 7369 732e  after) analysis.
+00011530: 2054 776f 2069 6e70 7574 2066 696c 6573   Two input files
+00011540: 2061 7265 2065 7373 656e 7469 616c 2c20   are essential, 
+00011550: 7468 6520 6765 6e65 2066 616d 696c 7920  the gene family 
+00011560: 6669 6c65 2061 6e64 2074 6865 2067 6666  file and the gff
+00011570: 3320 6669 6c65 2e20 5468 6520 6765 6e65  3 file. The gene
+00011580: 2066 616d 696c 7920 6669 6c65 2069 7320   family file is 
+00011590: 696e 2074 6865 2066 6f72 6d61 7420 6173  in the format as
+000115a0: 2060 4f72 7468 6f46 696e 6465 7260 2e20   `OrthoFinder`. 
+000115b0: 5468 6520 736f 6674 7761 7265 2060 692d  The software `i-
+000115c0: 6164 686f 7265 6020 6973 2061 2070 7265  adhore` is a pre
+000115d0: 7265 7175 6973 6974 652e 2057 6974 6820  requisite. With 
+000115e0: 6465 6661 756c 7420 7061 7261 6d65 7465  default paramete
+000115f0: 7273 2c20 7468 6520 7072 6f67 7261 6d20  rs, the program 
+00011600: 6261 7369 6361 6c6c 7920 636f 6e64 7563  basically conduc
+00011610: 7473 2031 2920 6669 6c74 6572 696e 6720  ts 1) filtering 
+00011620: 6765 6e65 2066 616d 696c 6965 7320 6261  gene families ba
+00011630: 7365 6420 6f6e 206d 6178 696d 756d 2066  sed on maximum f
+00011640: 616d 696c 7920 7369 7a65 2032 2920 7265  amily size 2) re
+00011650: 7472 6965 7669 6e67 2067 656e 6520 706f  trieving gene po
+00011660: 7369 7469 6f6e 2061 6e64 2073 6361 6666  sition and scaff
+00011670: 6f6c 6420 696e 666f 726d 6174 696f 6e20  old information 
+00011680: 6672 6f6d 2067 6666 3320 6669 6c65 2033  from gff3 file 3
+00011690: 2920 7072 6f64 7563 696e 6720 7468 6520  ) producing the 
+000116a0: 636f 6e66 6967 7572 6174 696f 6e20 6669  configuration fi
+000116b0: 6c65 2061 6e64 2061 7373 6f63 6961 7465  le and associate
+000116c0: 6420 6461 7461 6669 6c65 7320 666f 7220  d datafiles for 
+000116d0: 6069 2d61 6468 6f72 6560 2034 2920 6361  `i-adhore` 4) ca
+000116e0: 6c6c 696e 6720 6069 2d61 6468 6f72 6560  lling `i-adhore`
+000116f0: 2067 6976 656e 2074 6865 2070 6172 616d   given the param
+00011700: 6574 6572 7320 7365 7420 746f 2069 6e66  eters set to inf
+00011710: 6572 2073 796e 7465 6e79 2035 2920 7669  er synteny 5) vi
+00011720: 7375 616c 697a 696e 6720 7468 6520 7379  sualizing the sy
+00011730: 6e74 656e 7920 696e 2022 646f 7470 6c6f  nteny in "dotplo
+00011740: 7422 2069 6e20 7468 6520 756e 6974 206f  t" in the unit o
+00011750: 6620 6765 6e65 7320 616e 6420 6261 7365  f genes and base
+00011760: 732c 2069 6e20 2253 796e 6465 7074 6822  s, in "Syndepth"
+00011770: 2070 6c6f 7420 7368 6f77 696e 6720 7468   plot showing th
+00011780: 6520 6469 7374 7269 6275 7469 6f6e 206f  e distribution o
+00011790: 6620 6469 6666 6572 656e 7420 6361 7465  f different cate
+000117a0: 676f 7269 6573 206f 6620 636f 6c6c 696e  gories of collin
+000117b0: 6561 7269 7479 2072 6174 696f 7320 7769  earity ratios wi
+000117c0: 7468 696e 2061 6e64 2062 6574 7765 656e  thin and between
+000117d0: 2073 7065 6369 6573 2c20 696e 2022 6475   species, in "du
+000117e0: 7053 7461 636b 2220 706c 6f74 2073 686f  pStack" plot sho
+000117f0: 7769 6e67 206d 756c 7469 706c 6963 6f6e  wing multiplicon
+00011800: 7320 7769 7468 2064 6966 6665 7265 6e74  s with different
+00011810: 206d 756c 7469 706c 6963 6174 696f 6e20   multiplication 
+00011820: 6c65 7665 6c73 2e20 3629 2069 6620 7769  levels. 6) if wi
+00011830: 7468 202a 4b2a 3c73 7562 3e53 3c2f 7375  th *K*<sub>S</su
+00011840: 623e 2064 6174 612c 2061 2022 2a4b 2a3c  b> data, a "*K*<
+00011850: 7375 623e 533c 2f73 7562 3e20 646f 7470  sub>S</sub> dotp
+00011860: 6c6f 7422 2077 6974 6820 646f 7473 2061  lot" with dots a
+00011870: 6e6e 6f74 6174 6564 2069 6e20 2a4b 2a3c  nnotated in *K*<
+00011880: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
+00011890: 6573 2061 6e64 2061 202a 4b2a 3c73 7562  es and a *K*<sub
+000118a0: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
+000118b0: 7574 696f 6e20 7769 7468 2061 6e63 686f  ution with ancho
+000118c0: 7220 7061 6972 7320 6465 6e6f 7465 6420  r pairs denoted 
+000118d0: 7769 6c6c 2062 6520 7072 6f64 7563 6564  will be produced
+000118e0: 2e20 5468 6520 6765 6e65 2069 6e66 6f72  . The gene infor
+000118f0: 6d61 7469 6f6e 2069 6e20 7468 6520 6765  mation in the ge
+00011900: 6e65 2066 616d 696c 7920 6669 6c65 2061  ne family file a
+00011910: 6e64 2067 6666 3320 6669 6c65 2073 686f  nd gff3 file sho
+00011920: 756c 6420 6265 206d 6174 6368 6564 2077  uld be matched w
+00011930: 6869 6368 2072 6571 7569 7265 7320 7573  hich requires us
+00011940: 6572 7320 746f 2073 6574 2070 726f 7065  ers to set prope
+00011950: 7220 602d 2d66 6561 7475 7265 6020 616e  r `--feature` an
+00011960: 6420 602d 2d61 7474 7269 6275 7465 602e  d `--attribute`.
+00011970: 2054 6865 206d 6178 696d 756d 2066 616d   The maximum fam
+00011980: 696c 7920 7369 7a65 2074 6f20 6265 2069  ily size to be i
+00011990: 6e63 6c75 6465 6420 6361 6e20 6265 2073  ncluded can be s
+000119a0: 6574 2076 6961 2074 6865 206f 7074 696f  et via the optio
+000119b0: 6e20 602d 2d6d 6178 7369 7a65 602c 206e  n `--maxsize`, n
+000119c0: 6f74 6564 2074 6861 7420 7468 6973 2066  oted that this f
+000119d0: 696c 7465 7269 6e67 2069 7320 6d61 696e  iltering is main
+000119e0: 6c79 2074 6f20 6472 6f70 2074 686f 7365  ly to drop those
+000119f0: 2068 7567 6520 7461 6e64 656d 2064 7570   huge tandem dup
+00011a00: 6c69 6361 7465 7320 6661 6d69 6c79 2061  licates family a
+00011a10: 6e64 2074 7261 6e73 706f 7361 626c 6520  nd transposable 
+00011a20: 656c 656d 656e 7473 2028 5445 7329 2066  elements (TEs) f
+00011a30: 616d 696c 792c 2061 6e64 206e 6f74 206d  amily, and not m
+00011a40: 616e 6461 746f 7279 2e20 5573 6572 7320  andatory. Users 
+00011a50: 6361 6e20 6669 6c74 6572 2074 686f 7365  can filter those
+00011a60: 2066 7261 676d 656e 7461 7279 2073 6361   fragmentary sca
+00011a70: 6666 6f6c 6473 2076 6961 2074 6865 206f  ffolds via the o
+00011a80: 7074 696f 6e20 602d 2d6d 696e 6c65 6e60  ption `--minlen`
+00011a90: 2e20 5468 6520 6d69 6e69 6d75 6d20 6c65  . The minimum le
+00011aa0: 6e67 7468 2061 6e64 206e 756d 6265 7220  ngth and number 
+00011ab0: 6f66 2067 656e 6573 2066 6f72 2061 2073  of genes for a s
+00011ac0: 6567 6d65 6e74 2074 6f20 6265 2063 6f6e  egment to be con
+00011ad0: 7369 6465 7265 6420 6361 6e20 6265 2073  sidered can be s
+00011ae0: 6574 2076 6961 2074 6865 206f 7074 696f  et via the optio
+00011af0: 6e20 602d 2d6d 696e 7365 676c 656e 6020  n `--minseglen` 
+00011b00: 616e 6420 602d 2d6d 696e 6765 6e65 6e75  and `--mingenenu
+00011b10: 6d60 2e20 5265 6475 6e64 616e 7420 6d75  m`. Redundant mu
+00011b20: 6c74 6970 6c69 636f 6e73 2063 616e 2062  ltiplicons can b
+00011b30: 6520 6b65 7074 2062 7920 7365 7420 7468  e kept by set th
+00011b40: 6520 666c 6167 206f 7074 696f 6e20 602d  e flag option `-
+00011b50: 2d6b 6565 7072 6564 756e 602e 0a0a 2a2a  -keepredun`...**
+00011b60: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
+00011b70: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
+00011b80: 2063 6f6d 6d61 6e64 2073 696d 706c 7920   command simply 
+00011b90: 6173 2062 656c 6f77 2a2a 0a0a 6060 600a  as below**..```.
+00011ba0: 7767 6420 7379 6e20 6661 6d69 6c69 6573  wgd syn families
+00011bb0: 2067 6666 0a60 6060 0a0a 2a2a 5468 6520   gff.```..**The 
+00011bc0: 696e 7465 722d 7370 6563 6966 6963 2073  inter-specific s
+00011bd0: 796e 7465 6e79 2069 6e66 6572 656e 6365  ynteny inference
+00011be0: 2a2a 0a60 6060 0a77 6764 2073 796e 2066  **.```.wgd syn f
+00011bf0: 616d 696c 6965 7320 6766 6631 2067 6666  amilies gff1 gff
+00011c00: 3220 282d 2d61 6464 6974 696f 6e61 6c67  2 (--additionalg
+00011c10: 6666 696e 666f 2022 6d52 4e41 3b4e 616d  ffinfo "mRNA;Nam
+00011c20: 6522 202d 2d61 6464 6974 696f 6e61 6c67  e" --additionalg
+00011c30: 6666 696e 666f 2022 6765 6e65 3b49 4422  ffinfo "gene;ID"
+00011c40: 290a 6060 600a 0a46 6f72 206d 756c 7469  ).```..For multi
+00011c50: 2d73 7065 6369 6573 2073 796e 7465 6e79  -species synteny
+00011c60: 2069 6e66 6572 656e 6365 2c20 6966 2075   inference, if u
+00011c70: 7365 7273 2068 6176 6520 6766 6633 2066  sers have gff3 f
+00011c80: 696c 6573 2077 6869 6368 2068 6176 6520  iles which have 
+00011c90: 6469 6666 6572 656e 7420 6665 6174 7572  different featur
+00011ca0: 6573 206f 7220 6174 7472 6962 7574 6573  es or attributes
+00011cb0: 2066 6f72 2067 656e 6520 706f 7369 7469   for gene positi
+00011cc0: 6f6e 2069 6e66 6f72 6d61 7469 6f6e 2072  on information r
+00011cd0: 6574 7269 6576 616c 2c20 7468 6520 6f70  etrieval, the op
+00011ce0: 7469 6f6e 2060 2d2d 6164 6469 7469 6f6e  tion `--addition
+00011cf0: 616c 6766 6669 6e66 6f60 2063 616e 2062  algffinfo` can b
+00011d00: 6520 7365 7420 746f 2070 726f 7669 6465  e set to provide
+00011d10: 2074 6865 2061 6464 6974 696f 6e61 6c20   the additional 
+00011d20: 696e 666f 726d 6174 696f 6e2e 2054 6865  information. The
+00011d30: 2072 656d 6169 6e69 6e67 2070 6172 616d   remaining param
+00011d40: 6574 6572 2073 6574 7469 6e67 2069 7320  eter setting is 
+00011d50: 7468 6520 7361 6d65 2061 7320 7468 6520  the same as the 
+00011d60: 696e 7472 612d 7370 6563 6966 6963 2073  intra-specific s
+00011d70: 796e 7465 6e79 2069 6e66 6572 656e 6365  ynteny inference
+00011d80: 2e0a 0a2a 2a41 2073 7567 6765 7374 6564  ...**A suggested
+00011d90: 2073 7461 7274 696e 6720 7275 6e20 6361   starting run ca
+00011da0: 6e20 7573 6520 636f 6d6d 616e 6420 7369  n use command si
+00011db0: 6d70 6c79 2061 7320 6265 6c6f 772a 2a0a  mply as below**.
+00011dc0: 0a60 6060 0a77 6764 2073 796e 2066 616d  .```.wgd syn fam
+00011dd0: 696c 6965 7320 6766 6631 2067 6666 320a  ilies gff1 gff2.
+00011de0: 6060 600a 0a23 2323 2077 6764 2076 697a  ```..### wgd viz
+00011df0: 0a0a 2a2a 5468 6520 7669 7375 616c 697a  ..**The visualiz
+00011e00: 6174 696f 6e20 6f66 202a 4b2a 3c73 7562  ation of *K*<sub
+00011e10: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
+00011e20: 7472 6962 7574 696f 6e20 616e 6420 454c  tribution and EL
+00011e30: 4d4d 2061 6e61 6c79 7369 732a 2a0a 6060  MM analysis**.``
+00011e40: 600a 7767 6420 7669 7a20 2d64 206b 7364  `.wgd viz -d ksd
+00011e50: 6174 610a 6060 600a 0a54 6865 2070 726f  ata.```..The pro
+00011e60: 6772 616d 2060 7767 6420 7669 7a60 2069  gram `wgd viz` i
+00011e70: 7320 6d61 696e 6c79 2066 6f72 2074 6865  s mainly for the
+00011e80: 2070 7572 706f 7365 206f 6620 2a4b 2a3c   purpose of *K*<
+00011e90: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00011ea0: 7269 6275 7469 6f6e 2061 6e64 2073 796e  ribution and syn
+00011eb0: 7465 6e79 2076 6973 7561 6c69 7a61 7469  teny visualizati
+00011ec0: 6f6e 2c20 7769 7468 2073 6f6d 6520 6f70  on, with some op
+00011ed0: 7469 6f6e 616c 206d 6978 7475 7265 206d  tional mixture m
+00011ee0: 6f64 656c 696e 6720 616e 616c 7973 6973  odeling analysis
+00011ef0: 2e20 5468 6520 6261 7369 6320 6675 6e63  . The basic func
+00011f00: 7469 6f6e 2069 7320 6a75 7374 2074 6f20  tion is just to 
+00011f10: 706c 6f74 2074 6865 202a 4b2a 3c73 7562  plot the *K*<sub
+00011f20: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
+00011f30: 7574 696f 6e20 616e 6420 636f 6e64 7563  ution and conduc
+00011f40: 7420 616e 2045 4c4d 4d20 616e 616c 7973  t an ELMM analys
+00011f50: 6973 2069 6e20 7365 6172 6368 206f 6620  is in search of 
+00011f60: 706f 7465 6e74 6961 6c20 5747 4420 636f  potential WGD co
+00011f70: 6d70 6f6e 656e 7473 2e20 536f 6d65 206b  mponents. Some k
+00011f80: 6579 2070 6172 616d 6574 6572 7320 6166  ey parameters af
+00011f90: 6665 6374 696e 6720 7468 6520 454c 4d4d  fecting the ELMM
+00011fa0: 2072 6573 756c 7420 696e 636c 7564 6520   result include 
+00011fb0: 602d 2d70 726f 6d69 6e65 6e63 655f 6375  `--prominence_cu
+00011fc0: 746f 6666 6020 616e 6420 602d 2d72 656c  toff` and `--rel
+00011fd0: 5f68 6569 6768 7460 2c20 7768 6963 6820  _height`, which 
+00011fe0: 6861 7665 2062 6565 6e20 6578 706c 6169  have been explai
+00011ff0: 6e65 6420 6168 6561 642c 2060 2d2d 656d  ned ahead, `--em
+00012000: 5f69 7465 7261 7469 6f6e 7360 2061 6e64  _iterations` and
+00012010: 2060 2d2d 656d 5f69 6e69 7469 616c 697a   `--em_initializ
+00012020: 6174 696f 6e73 6020 6465 7465 726d 696e  ations` determin
+00012030: 696e 6720 7468 6520 6d61 7869 6d75 6d20  ing the maximum 
+00012040: 6974 6572 6174 696f 6e73 2061 6e64 2069  iterations and i
+00012050: 6e69 7469 616c 697a 6174 696f 6e73 2069  nitializations i
+00012060: 6e20 7468 6520 454d 2061 6c67 6f72 6974  n the EM algorit
+00012070: 686d 2e0a 0a2a 2a41 2073 7567 6765 7374  hm...**A suggest
+00012080: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
+00012090: 6361 6e20 7573 6520 636f 6d6d 616e 6420  can use command 
+000120a0: 7369 6d70 6c79 2061 7320 6265 6c6f 772a  simply as below*
+000120b0: 2a0a 0a60 6060 0a77 6764 2076 697a 202d  *..```.wgd viz -
+000120c0: 6420 6b73 6461 7461 0a60 6060 0a0a 2a2a  d ksdata.```..**
+000120d0: 5468 6520 7669 7375 616c 697a 6174 696f  The visualizatio
+000120e0: 6e20 6f66 202a 4b2a 3c73 7562 3e53 3c2f  n of *K*<sub>S</
+000120f0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+00012100: 7574 696f 6e20 7769 7468 2072 6174 6520  ution with rate 
+00012110: 636f 7272 6563 7469 6f6e 2a2a 0a60 6060  correction**.```
+00012120: 0a77 6764 2076 697a 202d 6420 6b73 6461  .wgd viz -d ksda
+00012130: 7461 202d 7370 2073 7064 6174 6120 2d2d  ta -sp spdata --
+00012140: 666f 6375 7332 616c 6c20 666f 6361 6c5f  focus2all focal_
+00012150: 7370 6563 6965 7320 2d2d 6578 7472 6170  species --extrap
+00012160: 6172 616e 6f6d 656b 7320 6b73 6461 7461  aranomeks ksdata
+00012170: 2028 2d2d 616e 6368 6f72 706f 696e 7473   (--anchorpoints
+00012180: 2061 7064 6174 6120 2d2d 706c 6f74 6170   apdata --plotap
+00012190: 676d 6d20 2d2d 706c 6f74 656c 6d6d 290a  gmm --plotelmm).
+000121a0: 6060 600a 0a42 6573 6964 6573 2074 6865  ```..Besides the
+000121b0: 2062 6173 6963 202a 4b2a 3c73 7562 3e53   basic *K*<sub>S
+000121c0: 3c2f 7375 623e 2070 6c6f 742c 2073 7562  </sub> plot, sub
+000121d0: 7374 6974 7574 696f 6e20 7261 7465 2063  stitution rate c
+000121e0: 6f72 7265 6374 696f 6e20 6361 6e20 616c  orrection can al
+000121f0: 736f 2062 6520 6163 6869 6576 6564 2067  so be achieved g
+00012200: 6976 656e 2061 7420 6c65 6173 7420 6120  iven at least a 
+00012210: 7370 6563 6965 7320 7472 6565 2028 7669  species tree (vi
+00012220: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
+00012230: 7370 6563 6965 7374 7265 6560 2920 616e  speciestree`) an
+00012240: 6420 6120 666f 6361 6c20 7370 6563 6965  d a focal specie
+00012250: 7320 2865 6974 6865 7220 7669 6120 7468  s (either via th
+00012260: 6520 6f70 7469 6f6e 2060 2d2d 666f 6375  e option `--focu
+00012270: 7332 616c 6c60 206f 7220 7669 6120 7468  s2all` or via th
+00012280: 6520 6f70 7469 6f6e 2060 2d2d 7370 6169  e option `--spai
+00012290: 7260 2069 6e20 7468 6520 666f 726d 206f  r` in the form o
+000122a0: 6620 2224 666f 6361 6c5f 7370 6563 6965  f "$focal_specie
+000122b0: 733b 2466 6f63 616c 5f73 7065 6369 6573  s;$focal_species
+000122c0: 2229 2e20 4974 2773 2073 7567 6765 7374  "). It's suggest
+000122d0: 6564 2074 6861 7420 7468 6520 6f72 7468  ed that the orth
+000122e0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+000122f0: 533c 2f73 7562 3e20 6461 7461 2069 7320  S</sub> data is 
+00012300: 7072 6f76 6964 6564 2062 7920 7468 6520  provided by the 
+00012310: 602d 2d64 6174 6166 696c 6560 206f 7074  `--datafile` opt
+00012320: 696f 6e20 616e 6420 7468 6520 7061 7261  ion and the para
+00012330: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
+00012340: 3c2f 7375 623e 2064 6174 6120 6973 2070  </sub> data is p
+00012350: 726f 7669 6465 6420 6279 2074 6865 2060  rovided by the `
+00012360: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
+00012370: 7360 206f 7074 696f 6e2c 2061 6c74 686f  s` option, altho
+00012380: 7567 6820 6974 2773 2061 6c6c 6f77 6564  ugh it's allowed
+00012390: 2074 6f20 6f6e 6c79 2070 726f 7669 6465   to only provide
+000123a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000123b0: 2064 6174 6120 7669 6120 7468 6520 602d   data via the `-
+000123c0: 2d64 6174 6166 696c 6560 206f 7074 696f  -datafile` optio
+000123d0: 6e20 616e 6420 6465 706f 7369 7420 626f  n and deposit bo
+000123e0: 7468 206f 7274 686f 6c6f 676f 7573 2061  th orthologous a
+000123f0: 6e64 2070 6172 616c 6f67 6f75 7320 2a4b  nd paralogous *K
+00012400: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
+00012410: 7461 2074 6865 7265 6f6e 2e20 5468 6572  ta thereon. Ther
+00012420: 6520 6172 6520 7477 6f20 7479 7065 7320  e are two types 
+00012430: 6f66 206d 6978 6564 2070 6c6f 7473 2028  of mixed plots (
+00012440: 7468 6520 226d 6978 6564 2220 6865 7265  the "mixed" here
+00012450: 2072 6566 6572 7320 746f 206d 6978 6564   refers to mixed
+00012460: 206f 7274 686f 6c6f 676f 7573 2061 6e64   orthologous and
+00012470: 2070 6172 616c 6f67 6f75 7320 2a4b 2a3c   paralogous *K*<
+00012480: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00012490: 7269 6275 7469 6f6e 7329 2c20 6f6e 6520  ributions), one 
+000124a0: 6f66 2077 6869 6368 2069 7320 7369 6d69  of which is simi
+000124b0: 6c61 7220 746f 2077 6861 7420 606b 7372  lar to what `ksr
+000124c0: 6174 6573 6020 706c 6f74 732c 2077 6869  ates` plots, whi
+000124d0: 6c65 2074 6865 206f 7468 6572 206f 6620  le the other of 
+000124e0: 7768 6963 6820 6973 206c 696b 6520 7468  which is like th
+000124f0: 6520 636f 6e76 656e 7469 6f6e 616c 202a  e conventional *
+00012500: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2070  K*<sub>S</sub> p
+00012510: 6c6f 7420 7468 6174 2062 6f74 6820 7468  lot that both th
+00012520: 6520 6f72 6967 696e 616c 206f 7274 686f  e original ortho
+00012530: 6c6f 676f 7573 2061 6e64 2070 6172 616c  logous and paral
+00012540: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
+00012550: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+00012560: 6f6e 7320 6172 6520 7472 7574 6866 756c  ons are truthful
+00012570: 6c79 2070 6c6f 7474 6564 2069 6e73 7465  ly plotted inste
+00012580: 6164 206f 6620 6a75 7374 2062 6569 6e67  ad of just being
+00012590: 2072 6570 7265 7365 6e74 6564 2062 7920   represented by 
+000125a0: 736f 6d65 2076 6572 7469 6361 6c20 6c69  some vertical li
+000125b0: 6e65 732e 2057 6520 7375 6767 6573 7420  nes. We suggest 
+000125c0: 7573 6572 7320 746f 2061 646f 7074 2074  users to adopt t
+000125d0: 6865 2060 6b73 7261 7465 7360 2d6c 696b  he `ksrates`-lik
+000125e0: 6520 706c 6f74 732c 2077 6869 6368 2069  e plots, which i
+000125f0: 7320 7468 6520 6465 6661 756c 742e 204f  s the default. O
+00012600: 7468 6572 7769 7365 2074 6865 206f 7074  therwise the opt
+00012610: 696f 6e20 602d 2d63 6c61 7373 6963 6020  ion `--classic` 
+00012620: 7769 6c6c 2062 6520 6e65 6564 6564 2074  will be needed t
+00012630: 6f20 7365 742e 2045 7874 7261 206d 6978  o set. Extra mix
+00012640: 7475 7265 206d 6f64 656c 696e 6720 616e  ture modeling an
+00012650: 616c 7973 6973 2063 616e 2062 6520 696e  alysis can be in
+00012660: 6974 6961 7465 6420 7669 6120 7468 6520  itiated via the 
+00012670: 6f70 7469 6f6e 2060 2d2d 706c 6f74 656c  option `--plotel
+00012680: 6d6d 6020 616e 6420 602d 2d70 6c6f 7461  mm` and `--plota
+00012690: 7067 6d6d 6020 7769 7468 2074 6865 2061  pgmm` with the a
+000126a0: 6e63 686f 7220 706f 696e 7473 2064 6174  nchor points dat
+000126b0: 6166 696c 6520 7072 6f76 6964 6564 2062  afile provided b
+000126c0: 7920 7468 6520 602d 2d61 6e63 686f 7270  y the `--anchorp
+000126d0: 6f69 6e74 7360 206f 7074 696f 6e2e 0a0a  oints` option...
+000126e0: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
+000126f0: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
+00012700: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
+00012710: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
+00012720: 600a 7767 6420 7669 7a20 2d64 206b 7364  `.wgd viz -d ksd
+00012730: 6174 6120 2d73 7020 7370 6461 7461 202d  ata -sp spdata -
+00012740: 2d66 6f63 7573 3261 6c6c 2066 6f63 616c  -focus2all focal
+00012750: 5f73 7065 6369 6573 202d 2d65 7874 7261  _species --extra
+00012760: 7061 7261 6e6f 6d65 6b73 206b 7364 6174  paranomeks ksdat
+00012770: 610a 6060 600a 0a2a 2a54 6865 2076 6973  a.```..**The vis
+00012780: 7561 6c69 7a61 7469 6f6e 206f 6620 7379  ualization of sy
+00012790: 6e74 656e 792a 2a0a 6060 600a 7767 6420  nteny**.```.wgd 
+000127a0: 7669 7a20 2d61 7020 6170 6461 7461 202d  viz -ap apdata -
+000127b0: 736d 2073 6d64 6174 6120 2d6d 7420 6d74  sm smdata -mt mt
+000127c0: 6461 7461 202d 6774 2067 7464 6174 6120  data -gt gtdata 
+000127d0: 2d2d 706c 6f74 7379 6e20 282d 2d6d 696e  --plotsyn (--min
+000127e0: 6c65 6e20 2d31 202d 2d6d 696e 7365 676c  len -1 --minsegl
+000127f0: 656e 2031 3030 3030 202d 2d6d 696e 6765  en 10000 --minge
+00012800: 6e65 6e75 6d20 3330 290a 6060 600a 0a43  nenum 30).```..C
+00012810: 6f6d 7061 7265 6420 746f 2074 6865 206f  ompared to the o
+00012820: 7269 6769 6e61 6c20 6077 6764 602c 2074  riginal `wgd`, t
+00012830: 6865 2060 7767 6420 7669 7a60 2070 726f  he `wgd viz` pro
+00012840: 6772 616d 2061 6464 6564 2073 796e 7465  gram added synte
+00012850: 6e79 2076 6973 7561 6c69 7a61 7469 6f6e  ny visualization
+00012860: 2070 6970 656c 696e 652e 2055 7365 7273   pipeline. Users
+00012870: 206e 6565 6420 746f 2070 726f 7669 6465   need to provide
+00012880: 2074 6865 2066 6c61 6720 6f70 7469 6f6e   the flag option
+00012890: 2060 2d2d 706c 6f74 7379 6e60 2074 6f20   `--plotsyn` to 
+000128a0: 696e 6974 6961 7465 2074 6869 7320 7061  initiate this pa
+000128b0: 7274 206f 6620 7069 7065 6c69 6e65 2e20  rt of pipeline. 
+000128c0: 5468 6973 2073 7465 7020 6173 7375 6d65  This step assume
+000128d0: 7320 7468 6174 2075 7365 7273 2068 6176  s that users hav
+000128e0: 6520 6f62 7461 696e 6564 2061 6c72 6561  e obtained alrea
+000128f0: 6479 2074 6865 2073 796e 7465 6e69 6320  dy the syntenic 
+00012900: 7265 7375 6c74 7320 6672 6f6d 2060 692d  results from `i-
+00012910: 6164 686f 7265 6020 616e 6420 7573 6573  adhore` and uses
+00012920: 2074 686f 7365 2072 6573 756c 7420 6669   those result fi
+00012930: 6c65 7320 746f 2072 6561 6c69 7a65 2074  les to realize t
+00012940: 6865 2073 796e 7465 6e79 2076 6973 7561  he synteny visua
+00012950: 6c69 7a61 7469 6f6e 2e20 5369 6d6c 6961  lization. Simlia
+00012960: 7220 746f 2060 7767 6420 7379 6e60 2c20  r to `wgd syn`, 
+00012970: 7468 6520 6578 7472 6120 2a4b 2a3c 7375  the extra *K*<su
+00012980: 623e 533c 2f73 7562 3e20 6461 7461 2066  b>S</sub> data f
+00012990: 696c 6520 6361 6e20 6265 2074 7261 6e73  ile can be trans
+000129a0: 6d69 7474 6564 2076 6961 2074 6865 206f  mitted via the o
+000129b0: 7074 696f 6e20 602d 2d64 6174 6166 696c  ption `--datafil
+000129c0: 6560 2028 696e 7374 6561 6420 6f66 2060  e` (instead of `
+000129d0: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
+000129e0: 7360 206f 7074 696f 6e29 2e20 4261 7369  s` option). Basi
+000129f0: 6361 6c6c 792c 2074 6865 2073 796e 7465  cally, the synte
+00012a00: 6e69 6320 7265 7375 6c74 2066 696c 6573  nic result files
+00012a10: 2072 6571 7569 7265 6420 6172 6520 7468   required are th
+00012a20: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
+00012a30: 6461 7461 6669 6c65 2c20 6d75 6c74 6970  datafile, multip
+00012a40: 6c69 636f 6e73 2064 6174 6166 696c 652c  licons datafile,
+00012a50: 2067 656e 6520 7461 626c 6520 6461 7461   gene table data
+00012a60: 6669 6c65 2028 6175 746f 6d61 7465 6c79  file (automately
+00012a70: 2070 726f 6475 6365 6420 6279 2060 7767   produced by `wg
+00012a80: 6420 7379 6e60 292c 2061 6e64 2073 6567  d syn`), and seg
+00012a90: 6d65 6e74 7320 6461 7461 6669 6c65 2e0a  ments datafile..
+00012aa0: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
+00012ab0: 7461 7274 696e 6720 7275 6e20 6361 6e20  tarting run can 
+00012ac0: 7573 6520 636f 6d6d 616e 6420 7369 6d70  use command simp
+00012ad0: 6c79 2061 7320 6265 6c6f 772a 2a0a 0a60  ly as below**..`
+00012ae0: 6060 0a77 6764 2076 697a 202d 6170 2061  ``.wgd viz -ap a
+00012af0: 7064 6174 6120 2d73 6d20 736d 6461 7461  pdata -sm smdata
+00012b00: 202d 6d74 206d 7464 6174 6120 2d67 7420   -mt mtdata -gt 
+00012b10: 6774 6461 7461 202d 2d70 6c6f 7473 796e  gtdata --plotsyn
+00012b20: 0a60 6060 0a0a 2323 2049 6c6c 7573 7472  .```..## Illustr
+00012b30: 6174 696f 6e0a 0a57 6520 696c 6c75 7374  ation..We illust
+00012b40: 7261 7465 206f 7572 2070 726f 6772 616d  rate our program
+00012b50: 206f 6e20 616e 2065 7865 6d70 6c61 7279   on an exemplary
+00012b60: 2057 4744 2069 6e66 6572 656e 6365 2061   WGD inference a
+00012b70: 6e64 2064 6174 696e 6720 7570 6f6e 2073  nd dating upon s
+00012b80: 7065 6369 6573 202a 4171 7569 6c65 6769  pecies *Aquilegi
+00012b90: 6120 636f 6572 756c 6561 2a2e 0a0a 5468  a coerulea*...Th
+00012ba0: 6520 2a41 7175 696c 6567 6961 2063 6f65  e *Aquilegia coe
+00012bb0: 7275 6c65 612a 2077 6173 2072 6570 6f72  rulea* was repor
+00012bc0: 7465 6420 746f 2065 7870 6572 6965 6e63  ted to experienc
+00012bd0: 6520 616e 2070 616c 656f 2d70 6f6c 7970  e an paleo-polyp
+00012be0: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
+00012bf0: 7420 6166 7465 7220 7468 6520 6469 7665  t after the dive
+00012c00: 7267 656e 6365 206f 6620 636f 7265 2065  rgence of core e
+00012c10: 7564 6963 6f74 732c 2077 6869 6368 2069  udicots, which i
+00012c20: 7320 6c69 6b65 6c79 2073 6861 7265 6420  s likely shared 
+00012c30: 6279 2061 6c6c 2052 616e 756e 6375 6c61  by all Ranuncula
+00012c40: 6c65 732e 0a0a 4669 7273 7420 6162 6f76  les...First abov
+00012c50: 6520 616c 6c2c 206c 6574 2773 2064 656c  e all, let's del
+00012c60: 696e 6561 7465 2074 6865 2077 686f 6c65  ineate the whole
+00012c70: 2070 6172 616e 6f6d 6520 2a4b 2a3c 7375   paranome *K*<su
+00012c80: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
+00012c90: 7374 7269 6275 7469 6f6e 2061 6e64 2068  stribution and h
+00012ca0: 6176 6520 6120 6261 7369 6320 6f62 7365  ave a basic obse
+00012cb0: 7276 6174 696f 6e20 666f 7220 706f 7465  rvation for pote
+00012cc0: 6e74 6961 6c6c 7920 636f 6e63 6569 7661  ntially conceiva
+00012cd0: 626c 6520 5747 4473 2c20 7573 696e 6720  ble WGDs, using 
+00012ce0: 7468 6520 636f 6d6d 616e 6420 6c69 6e65  the command line
+00012cf0: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
+00012d00: 2064 6d64 2041 7175 696c 6567 6961 5f63   dmd Aquilegia_c
+00012d10: 6f65 7275 6c65 610a 7767 6420 6b73 6420  oerulea.wgd ksd 
+00012d20: 7767 645f 646d 642f 4171 7569 6c65 6769  wgd_dmd/Aquilegi
+00012d30: 615f 636f 6572 756c 6561 2e74 7376 2041  a_coerulea.tsv A
+00012d40: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00012d50: 610a 6060 600a 0a54 6865 2063 6f6e 7374  a.```..The const
+00012d60: 7275 6374 6564 2077 686f 6c65 2070 6172  ructed whole par
+00012d70: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
+00012d80: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+00012d90: 6275 7469 6f6e 206f 6620 2a41 7175 696c  bution of *Aquil
+00012da0: 6567 6961 2063 6f65 7275 6c65 612a 2069  egia coerulea* i
+00012db0: 7320 6173 2062 656c 6f77 2c20 7765 2063  s as below, we c
+00012dc0: 616e 2073 6565 2074 6861 7420 7468 6572  an see that ther
+00012dd0: 6520 7365 656d 7320 746f 2062 6520 6120  e seems to be a 
+00012de0: 6875 6d70 2061 7420 2a4b 2a3c 7375 623e  hump at *K*<sub>
+00012df0: 533c 2f73 7562 3e20 3120 6275 7420 6e6f  S</sub> 1 but no
+00012e00: 7420 636c 6561 722e 0a0a 215b 5d28 6461  t clear...![](da
+00012e10: 7461 2f41 7175 696c 6567 6961 5f63 6f65  ta/Aquilegia_coe
+00012e20: 7275 6c65 612e 7473 762e 6b73 645f 7770  rulea.tsv.ksd_wp
+00012e30: 2e73 7667 290a 0a57 6520 7468 656e 2063  .svg)..We then c
+00012e40: 6f6e 7374 7275 6374 2074 6865 2061 6e63  onstruct the anc
+00012e50: 686f 7220 2a4b 2a3c 7375 623e 533c 2f73  hor *K*<sub>S</s
+00012e60: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+00012e70: 7469 6f6e 2075 7369 6e67 2074 6865 2063  tion using the c
+00012e80: 6f6d 6d61 6e64 206c 696e 6520 6265 6c6f  ommand line belo
+00012e90: 772e 0a0a 6060 600a 7767 6420 7379 6e20  w...```.wgd syn 
+00012ea0: 2d66 206d 524e 4120 2d61 204e 616d 6520  -f mRNA -a Name 
+00012eb0: 7767 645f 646d 642f 4171 7569 6c65 6769  wgd_dmd/Aquilegi
+00012ec0: 615f 636f 6572 756c 6561 2e74 7376 2041  a_coerulea.tsv A
+00012ed0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00012ee0: 612e 6766 6633 202d 6b73 2077 6764 5f6b  a.gff3 -ks wgd_k
+00012ef0: 7364 2f41 7175 696c 6567 6961 5f63 6f65  sd/Aquilegia_coe
+00012f00: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00012f10: 0a60 6060 0a0a 4173 2073 686f 776e 2062  .```..As shown b
+00012f20: 656c 6f77 2c20 7468 6572 6520 6172 6520  elow, there are 
+00012f30: 736f 6d65 2072 6574 6169 6e65 6420 616e  some retained an
+00012f40: 6368 6f72 2070 6169 7273 2077 6974 6820  chor pairs with 
+00012f50: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00012f60: 6265 7477 6565 6e20 3120 616e 6420 322c  between 1 and 2,
+00012f70: 2077 6869 6368 2073 6565 6d73 2074 6f20   which seems to 
+00012f80: 7375 6767 6573 7420 6120 5747 4420 6576  suggest a WGD ev
+00012f90: 656e 742e 0a0a 215b 5d28 6461 7461 2f41  ent...![](data/A
+00012fa0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00012fb0: 612e 7473 762e 6b73 645f 7770 5f61 702e  a.tsv.ksd_wp_ap.
+00012fc0: 7376 6729 0a0a 5468 6520 6173 736f 6369  svg)..The associ
+00012fd0: 6174 6564 2060 6475 7053 7461 636b 6020  ated `dupStack` 
+00012fe0: 706c 6f74 2073 686f 7773 2074 6861 7420  plot shows that 
+00012ff0: 7468 6572 6520 6172 6520 6e75 6d65 726f  there are numero
+00013000: 7573 2064 7570 6c69 6361 7465 6420 7365  us duplicated se
+00013010: 676d 656e 7473 2061 6372 6f73 7320 6d6f  gments across mo
+00013020: 7374 206f 6620 7468 6520 6368 726f 6d6f  st of the chromo
+00013030: 736f 6d65 732e 0a0a 215b 5d28 6461 7461  somes...![](data
+00013040: 2f73 796e 5f72 6573 756c 7473 2f41 7175  /syn_results/Aqu
+00013050: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
+00013060: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00013070: 6561 5f6d 756c 7469 706c 6963 6f6e 735f  ea_multiplicons_
+00013080: 6c65 7665 6c2e 706e 6729 0a0a 5765 2069  level.png)..We i
+00013090: 6d70 6c65 6d65 6e74 6564 2074 776f 2074  mplemented two t
+000130a0: 7970 6573 206f 6620 646f 7420 706c 6f74  ypes of dot plot
+000130b0: 7320 696e 206f 7866 6f72 6420 6772 6964  s in oxford grid
+000130c0: 3a20 6f6e 6520 696e 2074 6865 2075 6e69  : one in the uni
+000130d0: 7420 6f66 2062 6173 6573 2061 6e64 2074  t of bases and t
+000130e0: 6865 206f 7468 6572 2069 6e20 7468 6520  he other in the 
+000130f0: 756e 6974 206f 6620 6765 6e65 732c 2077  unit of genes, w
+00013100: 6869 6368 2063 616e 2062 6520 636f 6c6f  hich can be colo
+00013110: 7265 6420 6279 202a 4b2a 3c73 7562 3e53  red by *K*<sub>S
+00013120: 3c2f 7375 623e 2076 616c 7565 7320 6769  </sub> values gi
+00013130: 7665 6e20 2a4b 2a3c 7375 623e 533c 2f73  ven *K*<sub>S</s
+00013140: 7562 3e20 6461 7461 2e0a 0a21 5b5d 2864  ub> data...![](d
+00013150: 6174 612f 7379 6e5f 7265 7375 6c74 732f  ata/syn_results/
+00013160: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00013170: 6561 2d76 732d 4171 7569 6c65 6769 615f  ea-vs-Aquilegia_
+00013180: 636f 6572 756c 6561 5f4b 732e 646f 745f  coerulea_Ks.dot_
+00013190: 756e 6974 5f67 656e 652e 706e 6729 0a0a  unit_gene.png)..
+000131a0: 4173 2073 686f 776e 2061 626f 7665 2c20  As shown above, 
+000131b0: 7468 6520 646f 7420 706c 6f74 2069 6e20  the dot plot in 
+000131c0: 7468 6520 756e 6974 206f 6620 6765 6e65  the unit of gene
+000131d0: 7320 7072 6573 656e 7473 206e 756d 6572  s presents numer
+000131e0: 6f75 7320 6465 6e73 656c 7920 6167 6772  ous densely aggr
+000131f0: 6567 6174 6564 2028 6c69 6e65 2d6c 696b  egated (line-lik
+00013200: 6529 2061 6e63 686f 7220 706f 696e 7473  e) anchor points
+00013210: 2061 7420 6d6f 7374 206f 6620 7468 6520   at most of the 
+00013220: 6368 726f 6d6f 736f 6d65 7320 7769 7468  chromosomes with
+00013230: 2063 6f6e 7369 7374 656e 7420 2a4b 2a3c   consistent *K*<
+00013240: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+00013250: 6265 7477 6565 6e20 3120 616e 6420 322e  between 1 and 2.
+00013260: 2054 6865 2064 6f74 2070 6c6f 7420 696e   The dot plot in
+00013270: 2074 6865 2075 6e69 7420 6f66 2062 6173   the unit of bas
+00013280: 6573 2073 686f 7773 2074 6865 2073 616d  es shows the sam
+00013290: 6520 7061 7474 6572 6e2c 2061 7320 6d61  e pattern, as ma
+000132a0: 6e69 6665 7374 6564 2062 656c 6f77 2e0a  nifested below..
+000132b0: 0a21 5b5d 2864 6174 612f 7379 6e5f 7265  .![](data/syn_re
+000132c0: 7375 6c74 732f 4171 7569 6c65 6769 615f  sults/Aquilegia_
+000132d0: 636f 6572 756c 6561 2d76 732d 4171 7569  coerulea-vs-Aqui
+000132e0: 6c65 6769 615f 636f 6572 756c 6561 5f4b  legia_coerulea_K
+000132f0: 732e 646f 742e 706e 6729 0a0a 5468 6520  s.dot.png)..The 
+00013300: 646f 7420 706c 6f74 7320 7769 7468 6f75  dot plots withou
+00013310: 7420 2a4b 2a3c 7375 623e 533c 2f73 7562  t *K*<sub>S</sub
+00013320: 3e20 616e 6e6f 7461 7469 6f6e 2077 696c  > annotation wil
+00013330: 6c20 616c 736f 2062 6520 6175 746f 6d61  l also be automa
+00013340: 7465 6c79 2070 726f 6475 6365 642c 2061  tely produced, a
+00013350: 7320 7368 6f77 6e20 6265 6c6f 772e 0a0a  s shown below...
+00013360: 215b 5d28 6461 7461 2f73 796e 5f72 6573  ![](data/syn_res
+00013370: 756c 7473 2f41 7175 696c 6567 6961 5f63  ults/Aquilegia_c
+00013380: 6f65 7275 6c65 612d 7673 2d41 7175 696c  oerulea-vs-Aquil
+00013390: 6567 6961 5f63 6f65 7275 6c65 612e 646f  egia_coerulea.do
+000133a0: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
+000133b0: 0a0a 215b 5d28 6461 7461 2f73 796e 5f72  ..![](data/syn_r
+000133c0: 6573 756c 7473 2f41 7175 696c 6567 6961  esults/Aquilegia
+000133d0: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
+000133e0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+000133f0: 646f 742e 706e 6729 0a0a 4e6f 7465 2074  dot.png)..Note t
+00013400: 6861 7420 7468 6520 6f70 6163 6974 7920  hat the opacity 
+00013410: 6f66 2061 6e63 686f 7220 646f 7473 2061  of anchor dots a
+00013420: 6e64 2061 6c6c 2068 6f6d 6f6c 6f67 2064  nd all homolog d
+00013430: 6f74 7320 6361 6e20 6265 2073 6574 2062  ots can be set b
+00013440: 7920 7468 6520 6f70 7469 6f6e 2060 2d2d  y the option `--
+00013450: 6170 616c 7068 6160 2061 6e64 2060 2d2d  apalpha` and `--
+00013460: 686f 616c 7068 6160 2073 6570 6172 6174  hoalpha` separat
+00013470: 656c 792e 2049 6620 6f6e 6520 6a75 7374  ely. If one just
+00013480: 2077 616e 7473 2074 6f20 7365 6520 7468   wants to see th
+00013490: 6520 616e 6368 6f72 2064 6f74 732c 2073  e anchor dots, s
+000134a0: 6574 7469 6e67 2074 6865 2060 686f 616c  etting the `hoal
+000134b0: 7068 6160 2061 7320 3020 286f 7220 6f74  pha` as 0 (or ot
+000134c0: 6865 7220 6d69 6e75 7363 756c 6520 7661  her minuscule va
+000134d0: 6c75 6573 2920 7769 6c6c 2064 6f2e 2049  lues) will do. I
+000134e0: 6620 6f6e 6520 7761 6e74 7320 746f 2073  f one wants to s
+000134f0: 6565 2074 6865 2064 6973 7472 6962 7574  ee the distribut
+00013500: 696f 6e20 6f66 2077 686f 6c65 2064 6f74  ion of whole dot
+00013510: 7320 6265 7474 6572 2c20 7365 7474 696e  s better, settin
+00013520: 6720 7468 6520 6068 6f61 6c70 6861 6020  g the `hoalpha` 
+00013530: 6869 6768 6572 2028 616e 6420 6061 7061  higher (and `apa
+00013540: 6c70 6861 6020 6c6f 7765 7229 2077 696c  lpha` lower) wil
+00013550: 6c20 646f 2e20 5468 6520 6064 6f74 7369  l do. The `dotsi
+00013560: 7a65 6020 6f70 7469 6f6e 2063 616e 2062  ze` option can b
+00013570: 6520 6361 6c6c 6564 2074 6f20 6164 6a75  e called to adju
+00013580: 7374 2074 6865 2073 697a 6520 6f66 2064  st the size of d
+00013590: 6f74 732e 0a0a 4120 6675 7274 6865 7220  ots...A further 
+000135a0: 6173 736f 6369 6174 6564 2053 796e 6465  associated Synde
+000135b0: 7074 6820 706c 6f74 2073 686f 7773 2074  pth plot shows t
+000135c0: 6861 7420 7468 6572 6520 6172 6520 6d6f  hat there are mo
+000135d0: 7265 2074 6861 6e20 3530 2064 7570 6c69  re than 50 dupli
+000135e0: 6361 7465 6420 7365 676d 656e 7473 206c  cated segments l
+000135f0: 6f6e 6765 7220 7468 616e 2031 3030 3030  onger than 10000
+00013600: 2062 7020 616e 6420 3330 2067 656e 6573   bp and 30 genes
+00013610: 2028 736f 2061 7320 746f 2064 726f 7020   (so as to drop 
+00013620: 6672 6167 6d65 6e74 6172 7920 7365 676d  fragmentary segm
+00013630: 656e 7473 292c 2077 6869 6368 2064 6f6d  ents), which dom
+00013640: 696e 6174 6573 2074 6865 2077 686f 6c65  inates the whole
+00013650: 2063 6f6c 6c69 6e65 6172 2072 6174 696f   collinear ratio
+00013660: 2063 6174 6567 6f72 792e 0a0a 215b 5d28   category...![](
+00013670: 6461 7461 2f73 796e 5f72 6573 756c 7473  data/syn_results
+00013680: 2f53 796e 6465 7074 682e 7376 6729 0a0a  /Syndepth.svg)..
+00013690: 4d6f 7265 2065 7871 7569 7369 7465 2070  More exquisite p
+000136a0: 6c6f 7473 2069 6e63 6c75 6469 6e67 2062  lots including b
+000136b0: 6f74 6820 696e 7472 612d 7370 6563 6966  oth intra-specif
+000136c0: 6963 2061 6e64 2069 6e74 6572 2d73 7065  ic and inter-spe
+000136d0: 6369 6669 6320 636f 6d70 6172 6973 6f6e  cific comparison
+000136e0: 7320 7573 696e 6720 7468 6520 6f72 7468  s using the orth
+000136f0: 6f67 726f 7570 7320 2863 6f6d 706f 7365  ogroups (compose
+00013700: 6420 6f66 202a 4171 7569 6c65 6769 6120  d of *Aquilegia 
+00013710: 636f 6572 756c 6561 2a2c 202a 5072 6f74  coerulea*, *Prot
+00013720: 6561 2063 796e 6172 6f69 6465 732a 2c20  ea cynaroides*, 
+00013730: 2a41 636f 7275 7320 616d 6572 6963 616e  *Acorus american
+00013740: 7573 2a20 616e 6420 2a56 6974 6973 2076  us* and *Vitis v
+00013750: 696e 6966 6572 612a 2c20 7365 6520 6675  inifera*, see fu
+00013760: 7274 6865 7220 666f 7220 7468 6520 636f  rther for the co
+00013770: 6e74 6578 7429 2069 6e66 6572 7265 6420  ntext) inferred 
+00013780: 6865 7265 696e 6166 7465 7220 6361 6e20  hereinafter can 
+00013790: 6265 2061 6c73 6f20 7072 6f64 7563 6564  be also produced
+000137a0: 2075 7369 6e67 2060 7767 6420 7379 6e60   using `wgd syn`
+000137b0: 2e20 4e6f 7465 2074 6861 7420 6469 6666  . Note that diff
+000137c0: 6572 656e 7420 6765 6e6f 6d65 2061 7373  erent genome ass
+000137d0: 656d 626c 6965 7320 6d69 6768 7420 6861  emblies might ha
+000137e0: 7665 2064 6966 6665 7265 6e74 2066 6561  ve different fea
+000137f0: 7475 7265 7320 616e 6420 6174 7472 6962  tures and attrib
+00013800: 7574 6573 2077 6869 6368 2063 616e 2062  utes which can b
+00013810: 6520 6163 636f 6d6d 6f64 6174 6564 2076  e accommodated v
+00013820: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
+00013830: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
+00013840: 666f 6020 666f 7220 6561 6368 2067 656e  fo` for each gen
+00013850: 6f6d 6520 6173 7365 6d62 6c79 2077 686f  ome assembly who
+00013860: 7365 206f 7264 6572 206e 6565 6473 2074  se order needs t
+00013870: 6f20 666f 6c6c 6f77 2074 6865 206f 7264  o follow the ord
+00013880: 6572 206f 6620 6766 6633 2066 696c 6573  er of gff3 files
+00013890: 2c20 666f 7220 696e 7374 616e 6365 2027  , for instance '
+000138a0: 6d4e 5241 3b4e 616d 6527 2066 6f72 2060  mNRA;Name' for `
+000138b0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000138c0: 6561 2e67 6666 3360 2c20 276d 4e52 413b  ea.gff3`, 'mNRA;
+000138d0: 4944 2720 666f 7220 6050 726f 7465 615f  ID' for `Protea_
+000138e0: 6379 6e61 726f 6964 6573 2e67 6666 3360  cynaroides.gff3`
+000138f0: 2c20 276d 4e52 413b 4e61 6d65 2720 666f  , 'mNRA;Name' fo
+00013900: 7220 6041 636f 7275 735f 616d 6572 6963  r `Acorus_americ
+00013910: 616e 7573 2e67 6666 3360 2061 6e64 2027  anus.gff3` and '
+00013920: 6d4e 5241 3b4e 616d 6527 2066 6f72 2060  mNRA;Name' for `
+00013930: 5669 7469 735f 7669 6e69 6665 7261 2e67  Vitis_vinifera.g
+00013940: 6666 3360 2e0a 0a60 6060 0a77 6764 2073  ff3`...```.wgd s
+00013950: 796e 2077 6764 5f6f 7274 686f 2f4f 7274  yn wgd_ortho/Ort
+00013960: 686f 6772 6f75 7073 2e73 702e 7473 7620  hogroups.sp.tsv 
+00013970: 2d6b 7320 7767 645f 6f72 7468 6f5f 6b73  -ks wgd_ortho_ks
+00013980: 2f4f 7274 686f 6772 6f75 7073 2e73 702e  /Orthogroups.sp.
+00013990: 7473 762e 6b73 2e74 7376 2041 7175 696c  tsv.ks.tsv Aquil
+000139a0: 6567 6961 5f63 6f65 7275 6c65 612e 6766  egia_coerulea.gf
+000139b0: 6633 202d 2d61 6464 6974 696f 6e61 6c67  f3 --additionalg
+000139c0: 6666 696e 666f 2027 6d4e 5241 3b4e 616d  ffinfo 'mNRA;Nam
+000139d0: 6527 2050 726f 7465 615f 6379 6e61 726f  e' Protea_cynaro
+000139e0: 6964 6573 2e67 6666 3320 2d2d 6164 6469  ides.gff3 --addi
+000139f0: 7469 6f6e 616c 6766 6669 6e66 6f20 276d  tionalgffinfo 'm
+00013a00: 4e52 413b 4944 2720 4163 6f72 7573 5f61  NRA;ID' Acorus_a
+00013a10: 6d65 7269 6361 6e75 732e 6766 6633 202d  mericanus.gff3 -
+00013a20: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
+00013a30: 666f 2027 6d4e 5241 3b4e 616d 6527 2056  fo 'mNRA;Name' V
+00013a40: 6974 6973 5f76 696e 6966 6572 612e 6766  itis_vinifera.gf
+00013a50: 6633 202d 2d61 6464 6974 696f 6e61 6c67  f3 --additionalg
+00013a60: 6666 696e 666f 2027 6d4e 5241 3b4e 616d  ffinfo 'mNRA;Nam
+00013a70: 6527 202d 6f20 7767 645f 6f72 7468 6f5f  e' -o wgd_ortho_
+00013a80: 7379 6e0a 6060 600a 0a55 706f 6e20 7468  syn.```..Upon th
+00013a90: 6520 6163 7175 6973 6974 696f 6e20 6f66  e acquisition of
+00013aa0: 2074 6865 2063 6f6c 6c69 6e65 6172 2072   the collinear r
+00013ab0: 6573 756c 7473 2075 7369 6e67 2060 7767  esults using `wg
+00013ac0: 6420 7379 6e60 2c20 7468 6520 7361 6d65  d syn`, the same
+00013ad0: 2063 6f6c 6c69 6e65 6172 2070 6c6f 7473   collinear plots
+00013ae0: 2063 616e 2062 6520 616c 736f 2070 726f   can be also pro
+00013af0: 6475 6365 6420 6279 2060 7767 6420 7669  duced by `wgd vi
+00013b00: 7a60 2075 7369 6e67 2074 6865 2063 6f6d  z` using the com
+00013b10: 6d61 6e64 2062 656c 6f77 2e0a 6060 600a  mand below..```.
+00013b20: 7767 6420 7669 7a20 2d2d 706c 6f74 7379  wgd viz --plotsy
+00013b30: 6e20 2d73 6d20 7767 645f 6f72 7468 6f5f  n -sm wgd_ortho_
+00013b40: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+00013b50: 7365 676d 656e 7473 2e74 7874 202d 6170  segments.txt -ap
+00013b60: 2077 6764 5f6f 7274 686f 5f73 796e 2f69   wgd_ortho_syn/i
+00013b70: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
+00013b80: 7270 6f69 6e74 732e 7478 7420 2d6d 7420  rpoints.txt -mt 
+00013b90: 7767 645f 6f72 7468 6f5f 7379 6e2f 6961  wgd_ortho_syn/ia
+00013ba0: 6468 6f72 652d 6f75 742f 6d75 6c74 6970  dhore-out/multip
+00013bb0: 6c69 636f 6e73 2e74 7874 202d 6774 2077  licons.txt -gt w
+00013bc0: 6764 5f6f 7274 686f 5f73 796e 2f67 656e  gd_ortho_syn/gen
+00013bd0: 652d 7461 626c 652e 6373 7620 2d64 2077  e-table.csv -d w
+00013be0: 6764 5f6f 7274 686f 5f6b 732f 4f72 7468  gd_ortho_ks/Orth
+00013bf0: 6f67 726f 7570 732e 7370 2e74 7376 2e6b  ogroups.sp.tsv.k
+00013c00: 732e 7473 7620 2d6f 2077 6764 5f6f 7274  s.tsv -o wgd_ort
+00013c10: 686f 5f76 697a 0a60 6060 0a0a 215b 5d28  ho_viz.```..![](
+00013c20: 6461 7461 2f6f 7274 686f 5f73 796e 5f72  data/ortho_syn_r
+00013c30: 6573 756c 7473 2f41 7175 696c 6567 6961  esults/Aquilegia
+00013c40: 5f63 6f65 7275 6c65 615f 5669 7469 735f  _coerulea_Vitis_
+00013c50: 7669 6e69 6665 7261 5f6d 756c 7469 706c  vinifera_multipl
+00013c60: 6963 6f6e 735f 6c65 7665 6c2e 706e 6729  icons_level.png)
+00013c70: 0a54 6865 2061 626f 7665 2060 6475 7053  .The above `dupS
+00013c80: 7461 636b 6020 706c 6f74 2073 686f 7773  tack` plot shows
+00013c90: 2074 6865 2064 6973 7472 6962 7574 696f   the distributio
+00013ca0: 6e20 6f66 2064 7570 6c69 6361 7465 6420  n of duplicated 
+00013cb0: 7365 676d 656e 7473 206f 6620 2a41 7175  segments of *Aqu
+00013cc0: 696c 6567 6961 2063 6f65 7275 6c65 612a  ilegia coerulea*
+00013cd0: 2063 6f6d 7061 7265 6420 746f 2069 7473   compared to its
+00013ce0: 656c 6620 2869 6e20 6772 6565 6e29 2061  elf (in green) a
+00013cf0: 6e64 2063 6f6d 7061 7265 6420 746f 202a  nd compared to *
+00013d00: 5669 7469 7320 7669 6e69 6665 7261 2a20  Vitis vinifera* 
+00013d10: 2869 6e20 626c 7565 2920 6f76 6572 2074  (in blue) over t
+00013d20: 6865 2063 6872 6f6d 6f73 6f6d 6573 206f  he chromosomes o
+00013d30: 6620 2a41 2e20 636f 6572 756c 6561 2a2e  f *A. coerulea*.
+00013d40: 0a21 5b5d 2864 6174 612f 6f72 7468 6f5f  .![](data/ortho_
+00013d50: 7379 6e5f 7265 7375 6c74 732f 4f76 6572  syn_results/Over
+00013d60: 616c 6c73 7065 6369 6573 5f4b 732e 646f  allspecies_Ks.do
+00013d70: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
+00013d80: 0a54 6865 2061 626f 7665 202a 4b2a 3c73  .The above *K*<s
+00013d90: 7562 3e53 3c2f 7375 623e 2064 6f74 706c  ub>S</sub> dotpl
+00013da0: 6f74 2069 6e20 756e 6974 206f 6620 6765  ot in unit of ge
+00013db0: 6e65 2073 686f 7773 2074 6865 206f 7665  ne shows the ove
+00013dc0: 7261 6c6c 2064 6973 7472 6962 7574 696f  rall distributio
+00013dd0: 6e20 6f66 2063 6f6c 6c69 6e65 6172 6974  n of collinearit
+00013de0: 7920 6163 6f72 7373 2074 6865 2066 6f75  y acorss the fou
+00013df0: 7220 7370 6563 6965 7320 696e 766f 6c76  r species involv
+00013e00: 6564 2e0a 215b 5d28 6461 7461 2f6f 7274  ed..![](data/ort
+00013e10: 686f 5f73 796e 5f72 6573 756c 7473 2f4f  ho_syn_results/O
+00013e20: 7665 7261 6c6c 7370 6563 6965 732e 646f  verallspecies.do
+00013e30: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
+00013e40: 0a54 6865 2061 626f 7665 2064 6f74 706c  .The above dotpl
+00013e50: 6f74 2069 7320 7769 7468 6f75 7420 7468  ot is without th
+00013e60: 6520 616e 6e6f 7461 7469 6f6e 206f 6620  e annotation of 
+00013e70: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00013e80: 6167 6573 2063 6f6d 7061 7265 6420 746f  ages compared to
+00013e90: 2074 6865 206c 6173 7420 6f6e 652e 0a21   the last one..!
+00013ea0: 5b5d 2864 6174 612f 6f72 7468 6f5f 7379  [](data/ortho_sy
+00013eb0: 6e5f 7265 7375 6c74 732f 5379 6e64 6570  n_results/Syndep
+00013ec0: 7468 2e73 7667 290a 5468 6520 6162 6f76  th.svg).The abov
+00013ed0: 6520 5379 6e64 6570 7468 2070 6c6f 7420  e Syndepth plot 
+00013ee0: 7368 6f77 7320 7468 6520 636f 6c6c 696e  shows the collin
+00013ef0: 6561 7220 7261 7469 6f20 6163 6f72 7373  ear ratio acorss
+00013f00: 2061 6c6c 2073 7065 6369 6573 2070 6169   all species pai
+00013f10: 7273 2028 696e 7472 612d 7370 6563 6966  rs (intra-specif
+00013f20: 6963 2063 6f6d 7061 7269 736f 6e20 696e  ic comparison in
+00013f30: 2067 7265 656e 2077 6869 6c65 2069 6e74   green while int
+00013f40: 6572 2d73 7065 6369 6669 6320 636f 6d70  er-specific comp
+00013f50: 6172 6973 6f6e 2069 6e20 626c 7565 292e  arison in blue).
+00013f60: 0a0a 5765 2063 616e 2066 6974 2061 6e20  ..We can fit an 
+00013f70: 454c 4d4d 206d 6978 7475 7265 206d 6f64  ELMM mixture mod
+00013f80: 656c 2075 706f 6e20 7468 6520 7768 6f6c  el upon the whol
+00013f90: 6520 7061 7261 6e6f 6d65 202a 4b2a 3c73  e paranome *K*<s
+00013fa0: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+00013fb0: 6973 7472 6962 7574 696f 6e20 746f 2073  istribution to s
+00013fc0: 6565 206d 6f72 6520 6163 6375 7261 7465  ee more accurate
+00013fd0: 6c79 2074 6865 2073 6967 6e69 6669 6361  ly the significa
+00013fe0: 6e63 6520 616e 6420 6c6f 6361 7469 6f6e  nce and location
+00013ff0: 206f 6620 706f 7465 6e74 6961 6c20 5747   of potential WG
+00014000: 4473 2c20 7573 696e 6720 7468 6520 636f  Ds, using the co
+00014010: 6d6d 616e 6420 6c69 6e65 2062 656c 6f77  mmand line below
+00014020: 2e0a 0a60 6060 0a77 6764 2076 697a 202d  ...```.wgd viz -
+00014030: 6420 7767 645f 6b73 642f 4171 7569 6c65  d wgd_ksd/Aquile
+00014040: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00014050: 2e6b 732e 7473 760a 6060 600a 0a54 6865  .ks.tsv.```..The
+00014060: 2072 6573 756c 7420 6f66 2045 4c4d 4d20   result of ELMM 
+00014070: 6d69 7874 7572 6520 6d6f 6465 6c20 636c  mixture model cl
+00014080: 7573 7465 7269 6e67 2073 686f 7773 2074  ustering shows t
+00014090: 6861 7420 7468 6572 6520 6973 2061 206c  hat there is a l
+000140a0: 696b 656c 7920 5747 4420 636f 6d70 6f6e  ikely WGD compon
+000140b0: 656e 7420 6174 202a 4b2a 3c73 7562 3e53  ent at *K*<sub>S
+000140c0: 3c2f 7375 623e 2031 2e31 392e 0a0a 215b  </sub> 1.19...![
+000140d0: 5d28 6461 7461 2f65 6c6d 6d5f 4171 7569  ](data/elmm_Aqui
+000140e0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+000140f0: 7376 2e6b 732e 7473 765f 6265 7374 5f6d  sv.ks.tsv_best_m
+00014100: 6f64 656c 735f 7765 6967 6874 6564 2e73  odels_weighted.s
+00014110: 7667 290a 0a4c 6574 2773 2064 6f20 6120  vg)..Let's do a 
+00014120: 6d69 7874 7572 6520 6d6f 6465 6c20 636c  mixture model cl
+00014130: 7573 7465 7269 6e67 2066 6f72 2061 6e63  ustering for anc
+00014140: 686f 7220 2a4b 2a3c 7375 623e 533c 2f73  hor *K*<sub>S</s
+00014150: 7562 3e20 746f 6f2c 2075 7369 6e67 2074  ub> too, using t
+00014160: 6865 2063 6f6d 6d61 6e64 206c 696e 6520  he command line 
+00014170: 6265 6c6f 772e 204e 6f74 6520 7468 6174  below. Note that
+00014180: 2074 6869 7320 7374 6570 2077 696c 6c20   this step will 
+00014190: 6175 746f 6d61 7465 6c79 2063 616c 6c20  automately call 
+000141a0: 7468 6520 7365 676d 656e 7420 2a4b 2a3c  the segment *K*<
+000141b0: 7375 623e 533c 2f73 7562 3e20 636c 7573  sub>S</sub> clus
+000141c0: 7465 7269 6e67 2061 6e61 6c79 7369 7320  tering analysis 
+000141d0: 746f 6f2e 0a0a 6060 600a 7767 6420 7065  too...```.wgd pe
+000141e0: 616b 2077 6764 5f6b 7364 2f41 7175 696c  ak wgd_ksd/Aquil
+000141f0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00014200: 762e 6b73 2e74 7376 202d 2d61 6e63 686f  v.ks.tsv --ancho
+00014210: 7270 6f69 6e74 7320 7767 645f 7379 6e2f  rpoints wgd_syn/
+00014220: 6961 6468 6f72 652d 6f75 742f 616e 6368  iadhore-out/anch
+00014230: 6f72 706f 696e 7473 2e74 7874 202d 2d73  orpoints.txt --s
+00014240: 6567 6d65 6e74 7320 7767 645f 7379 6e2f  egments wgd_syn/
+00014250: 6961 6468 6f72 652d 6f75 742f 7365 676d  iadhore-out/segm
+00014260: 656e 7473 2e74 7874 202d 2d6c 6973 7465  ents.txt --liste
+00014270: 6c65 6d65 6e74 7320 7767 645f 7379 6e2f  lements wgd_syn/
+00014280: 6961 6468 6f72 652d 6f75 742f 6c69 7374  iadhore-out/list
+00014290: 5f65 6c65 6d65 6e74 732e 7478 7420 2d2d  _elements.txt --
+000142a0: 6d75 6c74 6970 6c69 636f 6e70 6169 7273  multipliconpairs
+000142b0: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
+000142c0: 2d6f 7574 2f6d 756c 7469 706c 6963 6f6e  -out/multiplicon
+000142d0: 5f70 6169 7273 2e74 7874 2028 2d2d 7765  _pairs.txt (--we
+000142e0: 6967 6874 6564 290a 6060 600a 0a54 6865  ighted).```..The
+000142f0: 2061 6e63 686f 7220 2a4b 2a3c 7375 623e   anchor *K*<sub>
+00014300: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
+00014310: 7269 6275 7469 6f6e 2061 6c73 6f20 6861  ribution also ha
+00014320: 7320 6120 6c69 6b65 6c79 2057 4744 2063  s a likely WGD c
+00014330: 6f6d 706f 6e65 6e74 2077 6974 6820 6d6f  omponent with mo
+00014340: 6465 2031 2e32 382e 0a0a 215b 5d28 6461  de 1.28...![](da
+00014350: 7461 2f4f 7269 6769 6e61 6c5f 416e 6368  ta/Original_Anch
+00014360: 6f72 4b73 5f47 4d4d 5f43 6f6d 706f 6e65  orKs_GMM_Compone
+00014370: 6e74 335f 6e6f 6465 5f61 7665 7261 6765  nt3_node_average
+00014380: 645f 4c6f 676e 6f72 6d61 6c2e 7376 6729  d_Lognormal.svg)
+00014390: 0a0a 4e6f 7720 7468 6174 2077 6520 6861  ..Now that we ha
+000143a0: 7665 2073 6565 6e20 7468 6520 6576 6964  ve seen the evid
+000143b0: 656e 6365 206f 6620 6e75 6d65 726f 7573  ence of numerous
+000143c0: 2064 7570 6c69 6361 7465 6420 7365 676d   duplicated segm
+000143d0: 656e 7473 2061 6e64 2074 6865 2061 6767  ents and the agg
+000143e0: 7265 6761 7469 6f6e 206f 6620 6475 706c  regation of dupl
+000143f0: 6963 6174 6573 2061 6765 2061 7420 2a4b  icates age at *K
+00014400: 2a3c 7375 623e 533c 2f73 7562 3e20 312e  *<sub>S</sub> 1.
+00014410: 3238 206f 7220 312e 3139 2066 6f72 2061  28 or 1.19 for a
+00014420: 6e63 686f 7220 7061 6972 7320 616e 6420  nchor pairs and 
+00014430: 6e6f 6e2d 616e 6368 6f72 2070 6169 7273  non-anchor pairs
+00014440: 2074 6872 6f75 6768 6f75 7420 7468 6520   throughout the 
+00014450: 7768 6f6c 6520 6765 6e6f 6d65 2e20 5765  whole genome. We
+00014460: 2063 616e 2063 6c61 696d 2077 6974 6820   can claim with 
+00014470: 736f 6d65 2063 6f6e 6669 6465 6e63 6520  some confidence 
+00014480: 7468 6174 202a 4171 7569 6c65 6769 6120  that *Aquilegia 
+00014490: 636f 6572 756c 6561 2a20 6d69 6768 7420  coerulea* might 
+000144a0: 6861 7665 2065 7870 6572 6965 6e63 6564  have experienced
+000144b0: 2061 2070 616c 656f 2d70 6f6c 7970 6c6f   a paleo-polyplo
+000144c0: 6964 697a 6174 696f 6e20 6576 656e 742e  idization event.
+000144d0: 204e 6578 742c 204c 6574 2773 2068 6176   Next, Let's hav
+000144e0: 6520 6120 6675 7274 6865 7220 6c6f 6f6b  e a further look
+000144f0: 2061 626f 7574 2069 7473 2070 6879 6c6f   about its phylo
+00014500: 6765 6e65 7469 6320 6c6f 6361 7469 6f6e  genetic location
+00014510: 2e20 5765 206b 6e6f 7720 7468 6174 2074  . We know that t
+00014520: 6865 7265 2061 7265 2075 6e63 6572 7461  here are uncerta
+00014530: 696e 7469 6573 2061 626f 7574 2077 6865  inties about whe
+00014540: 7468 6572 2074 6869 7320 7075 7461 7469  ther this putati
+00014550: 7665 2070 616c 656f 2d70 6f6c 7970 6c6f  ve paleo-polyplo
+00014560: 6964 697a 6174 696f 6e20 6576 656e 7420  idization event 
+00014570: 6973 2073 6861 7265 6420 7769 7468 2061  is shared with a
+00014580: 6c6c 2065 7564 6963 6f74 7320 6f72 206e  ll eudicots or n
+00014590: 6f74 2e20 5765 2063 616e 2063 686f 6f73  ot. We can choos
+000145a0: 6520 736f 6d65 206f 7468 6572 2065 7564  e some other eud
+000145b0: 6963 6f74 2067 656e 6f6d 6573 2074 6f20  icot genomes to 
+000145c0: 7365 6520 7468 6520 6f72 6465 7269 6e67  see the ordering
+000145d0: 206f 6620 7370 6563 6961 7469 6f6e 2061   of speciation a
+000145e0: 6e64 2070 6f6c 7970 6c6f 6964 697a 6174  nd polyploidizat
+000145f0: 696f 6e20 6576 656e 7473 2e20 4865 7265  ion events. Here
+00014600: 2077 6520 6368 6f6f 7365 202a 5669 7469   we choose *Viti
+00014610: 7320 7669 6e69 6665 7261 2a2c 202a 5072  s vinifera*, *Pr
+00014620: 6f74 6561 2063 796e 6172 6f69 6465 732a  otea cynaroides*
+00014630: 2061 6e64 202a 4163 6f72 7573 2061 6d65   and *Acorus ame
+00014640: 7269 6361 6e75 732a 2069 6e20 7468 6520  ricanus* in the 
+00014650: 666f 6c6c 6f77 696e 6720 2a4b 2a3c 7375  following *K*<su
+00014660: 623e 533c 2f73 7562 3e20 616e 616c 7973  b>S</sub> analys
+00014670: 6973 2e20 4669 7273 742c 2077 6520 6275  is. First, we bu
+00014680: 696c 7420 6120 676c 6f62 616c 204d 5242  ilt a global MRB
+00014690: 4820 6661 6d69 6c79 2075 7369 6e67 2074  H family using t
+000146a0: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
+000146b0: 2e0a 0a60 6060 0a77 6764 2064 6d64 202d  ...```.wgd dmd -
+000146c0: 2d67 6c6f 6261 6c6d 7262 6820 4171 7569  -globalmrbh Aqui
+000146d0: 6c65 6769 615f 636f 6572 756c 6561 2050  legia_coerulea P
+000146e0: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
+000146f0: 2041 636f 7275 735f 616d 6572 6963 616e   Acorus_american
+00014700: 7573 2056 6974 6973 5f76 696e 6966 6572  us Vitis_vinifer
+00014710: 6120 2d6f 2077 6764 5f67 6c6f 6261 6c6d  a -o wgd_globalm
+00014720: 7262 680a 6060 600a 0a49 6e20 7468 6520  rbh.```..In the 
+00014730: 676c 6f62 616c 204d 5242 4820 6661 6d69  global MRBH fami
+00014740: 6c79 2c20 6576 6572 7920 7061 6972 206f  ly, every pair o
+00014750: 6620 6f72 7468 6f6c 6f67 6f75 7320 6765  f orthologous ge
+00014760: 6e65 7320 6973 2074 6865 2072 6563 6970  nes is the recip
+00014770: 726f 6361 6c20 6265 7374 2068 6974 2c20  rocal best hit, 
+00014780: 7375 6767 6573 7469 6e67 2074 7275 6520  suggesting true 
+00014790: 6f72 7468 6f6c 6f67 6f75 7320 7265 6c61  orthologous rela
+000147a0: 7469 6f6e 7368 6970 732e 2057 6520 776f  tionships. We wo
+000147b0: 756c 6420 7573 6520 7468 6520 2a4b 2a3c  uld use the *K*<
+000147c0: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
+000147d0: 6573 2061 7373 6f63 6961 7465 6420 7769  es associated wi
+000147e0: 7468 2074 6865 7365 206f 7274 686f 6c6f  th these ortholo
+000147f0: 676f 7573 2070 6169 7273 2074 6f20 6465  gous pairs to de
+00014800: 6c69 6d69 7420 7468 6520 6469 7665 7267  limit the diverg
+00014810: 656e 6365 202a 4b2a 3c73 7562 3e53 3c2f  ence *K*<sub>S</
+00014820: 7375 623e 2070 6561 6b2e 2054 6f67 6574  sub> peak. Toget
+00014830: 6865 7220 7769 7468 2074 6865 2077 686f  her with the who
+00014840: 6c65 2070 6172 616e 6f6d 6520 2a4b 2a3c  le paranome *K*<
+00014850: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00014860: 7269 6275 7469 6f6e 2c20 7765 2063 6f6e  ribution, we con
+00014870: 6475 6374 2074 6865 2072 6174 6520 636f  duct the rate co
+00014880: 7272 6563 7469 6f6e 2075 7369 6e67 2074  rrection using t
+00014890: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
+000148a0: 2e0a 0a21 2153 696e 6365 2060 7767 6460  ...!!Since `wgd`
+000148b0: 2076 6572 7369 6f6e 2032 2e30 2e32 342c   version 2.0.24,
+000148c0: 2077 6520 7265 7772 6f74 6520 6120 636c   we rewrote a cl
+000148d0: 6561 6e65 7220 616e 6420 7175 6963 6b65  eaner and quicke
+000148e0: 7220 7761 7920 6f66 2064 6f69 6e67 2073  r way of doing s
+000148f0: 7562 7374 6974 7574 696f 6e20 7261 7465  ubstitution rate
+00014900: 2063 6f72 7265 6374 696f 6e2e 2049 7427   correction. It'
+00014910: 7320 6e6f 7420 7265 7175 6972 6564 2074  s not required t
+00014920: 6f20 7479 7065 2069 6e20 616e 7920 7370  o type in any sp
+00014930: 6569 6365 7320 7061 6972 2061 6e64 2061  eices pair and a
+00014940: 2073 6572 6965 7320 6f66 202a 4b2a 3c73   series of *K*<s
+00014950: 7562 3e53 3c2f 7375 623e 2070 6c6f 7473  ub>S</sub> plots
+00014960: 2077 696c 6c20 6265 2070 726f 6475 6365   will be produce
+00014970: 642e 2054 6865 2072 6571 7569 7265 6420  d. The required 
+00014980: 6669 6c65 7320 6172 6520 6f72 7468 6f6c  files are orthol
+00014990: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
+000149a0: 2f73 7562 3e20 6461 7461 6669 6c65 2c20  /sub> datafile, 
+000149b0: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
+000149c0: 7562 3e53 3c2f 7375 623e 2064 6174 6166  ub>S</sub> dataf
+000149d0: 696c 652c 2061 2073 7065 6369 6573 2074  ile, a species t
+000149e0: 7265 6520 616e 6420 6120 666f 6361 6c20  ree and a focal 
+000149f0: 7370 6563 6965 7320 2874 6865 206f 6e65  species (the one
+00014a00: 2069 6e70 7574 7465 6420 7769 7468 2070   inputted with p
+00014a10: 6172 616c 6f67 6f75 7320 2a4b 2a3c 7375  aralogous *K*<su
+00014a20: 623e 533c 2f73 7562 3e20 6461 7461 292e  b>S</sub> data).
+00014a30: 2055 7365 7273 2063 616e 2063 686f 6f73   Users can choos
+00014a40: 6520 746f 2061 6464 206f 6e65 206d 6f72  e to add one mor
+00014a50: 6520 6c61 7965 7220 6f66 2045 4c4d 4d20  e layer of ELMM 
+00014a60: 616e 616c 7973 6973 206f 6e20 7061 7261  analysis on para
+00014a70: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
+00014a80: 3c2f 7375 623e 2076 616c 7565 7320 616e  </sub> values an
+00014a90: 642f 6f72 2047 4d4d 2061 6e61 6c79 7369  d/or GMM analysi
+00014aa0: 7320 6f6e 2061 6e63 686f 7220 2a4b 2a3c  s on anchor *K*<
+00014ab0: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00014ac0: 7269 6275 7469 6f6e 2e20 5468 6520 6f72  ribution. The or
+00014ad0: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
+00014ae0: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
+00014af0: 6275 7469 6f6e 2063 616e 2062 6520 6361  bution can be ca
+00014b00: 6c63 756c 6174 6564 2075 7369 6e67 2074  lculated using t
+00014b10: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
+00014b20: 2e0a 0a60 6060 0a77 6764 206b 7364 2077  ...```.wgd ksd w
+00014b30: 6764 5f67 6c6f 6261 6c6d 7262 682f 676c  gd_globalmrbh/gl
+00014b40: 6f62 616c 5f4d 5242 482e 7473 7620 4171  obal_MRBH.tsv Aq
+00014b50: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00014b60: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
+00014b70: 6573 2041 636f 7275 735f 616d 6572 6963  es Acorus_americ
+00014b80: 616e 7573 2056 6974 6973 5f76 696e 6966  anus Vitis_vinif
+00014b90: 6572 6120 2d6f 2077 6764 5f67 6c6f 6261  era -o wgd_globa
+00014ba0: 6c6d 7262 685f 6b73 0a60 6060 0a0a 5769  lmrbh_ks.```..Wi
+00014bb0: 7468 2074 6865 2063 616c 6375 6c61 7465  th the calculate
+00014bc0: 6420 6f72 7468 6f6c 6f67 6f75 7320 2a4b  d orthologous *K
+00014bd0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+00014be0: 7374 7269 6275 7469 6f6e 2c20 7765 2063  stribution, we c
+00014bf0: 616e 2075 7365 2074 6865 2063 6f6d 6d61  an use the comma
+00014c00: 6e64 2062 656c 6f77 2074 6f20 636f 6e64  nd below to cond
+00014c10: 7563 7420 7468 6520 7261 7465 2063 6f72  uct the rate cor
+00014c20: 7265 6374 696f 6e20 616e 642f 6f72 206d  rection and/or m
+00014c30: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
+00014c40: 616e 616c 7973 6973 2e0a 0a60 6060 0a77  analysis...```.w
+00014c50: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
+00014c60: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
+00014c70: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
+00014c80: 7376 202d 6661 2041 7175 696c 6567 6961  sv -fa Aquilegia
+00014c90: 5f63 6f65 7275 6c65 6120 2d65 706b 2077  _coerulea -epk w
+00014ca0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
+00014cb0: 5f63 6f65 7275 6c65 612e 6b73 2e74 7376  _coerulea.ks.tsv
+00014cc0: 202d 6170 2077 6764 5f73 796e 2f69 6164   -ap wgd_syn/iad
+00014cd0: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
+00014ce0: 6f69 6e74 732e 7478 7420 2d73 7020 7370  oints.txt -sp sp
+00014cf0: 6563 6965 7374 7265 652e 6e77 202d 6f20  eciestree.nw -o 
+00014d00: 7767 645f 7669 7a5f 6d69 7865 645f 4b73  wgd_viz_mixed_Ks
+00014d10: 202d 2d70 6c6f 7465 6c6d 6d20 2d2d 706c   --plotelmm --pl
+00014d20: 6f74 6170 676d 6d20 2d2d 7265 7765 6967  otapgmm --reweig
+00014d30: 6874 0a60 6060 0a0a 6f72 2075 7369 6e67  ht.```..or using
+00014d40: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+00014d50: 6f77 2c20 7768 6963 6820 636f 6d62 696e  ow, which combin
+00014d60: 6573 2074 6865 2074 776f 2073 7465 7073  es the two steps
+00014d70: 2061 626f 7665 2069 6e20 6f6e 652e 204e   above in one. N
+00014d80: 6f74 6520 7468 6174 2077 6520 7375 6767  ote that we sugg
+00014d90: 6573 7420 6f66 2074 616b 696e 6720 7477  est of taking tw
+00014da0: 6f20 7365 7061 7261 7465 2073 7465 7073  o separate steps
+00014db0: 2069 6e20 7768 6963 6820 6077 6764 206b   in which `wgd k
+00014dc0: 7364 6020 756e 6465 7274 616b 6573 2074  sd` undertakes t
+00014dd0: 6865 2063 616c 6375 6c61 7469 6f6e 206f  he calculation o
+00014de0: 6620 6f72 7468 6f6c 6f67 6f75 7320 2a4b  f orthologous *K
+00014df0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+00014e00: 7374 7269 6275 7469 6f6e 2077 6869 6c65  stribution while
+00014e10: 2060 7767 6420 7669 7a60 2063 6172 7269   `wgd viz` carri
+00014e20: 6573 206f 7574 2074 6865 2072 6174 6520  es out the rate 
+00014e30: 636f 7272 6563 7469 6f6e 2061 6e64 2047  correction and G
+00014e40: 4d4d 2061 6e61 6c79 7369 7320 7375 6368  MM analysis such
+00014e50: 2074 6861 7420 6974 2773 2065 6173 6965   that it's easie
+00014e60: 7220 746f 2064 6562 7567 2e0a 6060 600a  r to debug..```.
+00014e70: 7767 6420 6b73 6420 7767 645f 676c 6f62  wgd ksd wgd_glob
+00014e80: 616c 6d72 6268 2f67 6c6f 6261 6c5f 4d52  almrbh/global_MR
+00014e90: 4248 2e74 7376 2041 7175 696c 6567 6961  BH.tsv Aquilegia
+00014ea0: 5f63 6f65 7275 6c65 6120 5072 6f74 6561  _coerulea Protea
+00014eb0: 5f63 796e 6172 6f69 6465 7320 4163 6f72  _cynaroides Acor
+00014ec0: 7573 5f61 6d65 7269 6361 6e75 7320 5669  us_americanus Vi
+00014ed0: 7469 735f 7669 6e69 6665 7261 202d 2d65  tis_vinifera --e
+00014ee0: 7874 7261 7061 7261 6e6f 6d65 6b73 2077  xtraparanomeks w
+00014ef0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
+00014f00: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00014f10: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
+00014f20: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
+00014f30: 6874 202d 6f20 7767 645f 676c 6f62 616c  ht -o wgd_global
+00014f40: 6d72 6268 5f6b 735f 7261 7465 5f63 6f72  mrbh_ks_rate_cor
+00014f50: 7265 6374 696f 6e20 2d66 6120 4171 7569  rection -fa Aqui
+00014f60: 6c65 6769 615f 636f 6572 756c 6561 202d  legia_coerulea -
+00014f70: 6170 2077 6764 5f73 796e 2f69 6164 686f  ap wgd_syn/iadho
+00014f80: 7265 2d6f 7574 2f61 6e63 686f 7270 6f69  re-out/anchorpoi
+00014f90: 6e74 732e 7478 7420 2d2d 706c 6f74 656c  nts.txt --plotel
+00014fa0: 6d6d 202d 2d70 6c6f 7461 7067 6d6d 0a60  mm --plotapgmm.`
+00014fb0: 6060 0a0a 5468 6520 6669 6c65 2060 7370  ``..The file `sp
+00014fc0: 6563 6965 7374 7265 652e 6e77 6020 6973  eciestree.nw` is
+00014fd0: 2074 6865 2074 6578 7420 6669 6c65 206f   the text file o
+00014fe0: 6620 7370 6563 6965 7320 7472 6565 2069  f species tree i
+00014ff0: 6e20 6e65 7769 636b 2074 6861 7420 7261  n newick that ra
+00015000: 7465 2063 6f72 7265 6374 696f 6e20 776f  te correction wo
+00015010: 756c 6420 6265 2063 6f6e 6475 6374 6564  uld be conducted
+00015020: 206f 6e2e 2049 7473 2063 6f6e 7465 6e74   on. Its content
+00015030: 2069 7320 6173 2062 656c 6f77 2e20 5573   is as below. Us
+00015040: 6572 7320 6361 6e20 6f70 7469 6f6e 616c  ers can optional
+00015050: 6c79 2070 726f 7669 6465 2074 6865 2073  ly provide the s
+00015060: 7065 6369 6573 2070 6169 7273 2074 6f20  pecies pairs to 
+00015070: 6265 2070 6c6f 7474 6564 2062 7574 2077  be plotted but w
+00015080: 6520 7375 6767 6573 7420 6f66 206a 7573  e suggest of jus
+00015090: 7420 7573 696e 6720 602d 6661 2041 7175  t using `-fa Aqu
+000150a0: 696c 6567 6961 5f63 6f65 7275 6c65 6160  ilegia_coerulea`
+000150b0: 2074 6f20 706c 6f74 2061 6c6c 2070 6f73   to plot all pos
+000150c0: 7369 626c 6520 666f 6361 6c2d 7369 7374  sible focal-sist
+000150d0: 6572 2073 7065 6369 6573 2070 6169 7273  er species pairs
+000150e0: 2e20 5765 2073 7567 6765 7374 2061 6464  . We suggest add
+000150f0: 696e 6720 7468 6520 6f70 7469 6f6e 2060  ing the option `
+00015100: 2d2d 7265 7765 6967 6874 6020 746f 2072  --reweight` to r
+00015110: 6563 616c 6375 6c61 7465 2074 6865 2077  ecalculate the w
+00015120: 6569 6768 7420 7065 7220 7370 6563 6965  eight per specie
+00015130: 7320 7061 6972 2073 7563 6820 7468 6174  s pair such that
+00015140: 2074 6865 2077 6569 6768 7420 6f66 206f   the weight of o
+00015150: 7274 686f 6c6f 676f 7573 2067 656e 6520  rthologous gene 
+00015160: 7061 6972 7320 7769 6c6c 2062 6563 6f6d  pairs will becom
+00015170: 6520 312e 2045 7874 7261 2063 6f6c 6c69  e 1. Extra colli
+00015180: 6e65 6172 2064 6174 6120 6361 6e20 6265  near data can be
+00015190: 2061 6464 6564 2062 7920 7468 6520 6f70   added by the op
+000151a0: 7469 6f6e 2060 2d61 7060 2061 6e64 2061  tion `-ap` and a
+000151b0: 6464 6974 696f 6e61 6c20 636c 7573 7465  dditional cluste
+000151c0: 7269 6e67 2061 6e61 6c79 7369 7320 6361  ring analysis ca
+000151d0: 6e20 6265 2069 6e69 7469 6174 6564 2062  n be initiated b
+000151e0: 7920 7365 7474 696e 6720 7468 6520 6f70  y setting the op
+000151f0: 7469 6f6e 2060 2d2d 706c 6f74 6170 676d  tion `--plotapgm
+00015200: 6d60 2061 6e64 2060 2d2d 706c 6f74 656c  m` and `--plotel
+00015210: 6d6d 602e 0a0a 6060 600a 2828 2856 6974  mm`...```.(((Vit
+00015220: 6973 5f76 696e 6966 6572 612c 5072 6f74  is_vinifera,Prot
+00015230: 6561 5f63 796e 6172 6f69 6465 7329 2c41  ea_cynaroides),A
+00015240: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00015250: 6129 2c41 636f 7275 735f 616d 6572 6963  a),Acorus_americ
+00015260: 616e 7573 293b 0a60 6060 0a0a 215b 5d28  anus);.```..![](
+00015270: 6461 7461 2f4d 6978 6564 2e6b 732e 4171  data/Mixed.ks.Aq
+00015280: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00015290: 2e6e 6f64 652e 7765 6967 6874 6564 2e70  .node.weighted.p
+000152a0: 6e67 290a 0a54 6865 206d 6978 6564 202a  ng)..The mixed *
+000152b0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
+000152c0: 6973 7472 6962 7574 696f 6e20 7368 6f77  istribution show
+000152d0: 6e20 6162 6f76 6520 6973 2061 2070 7562  n above is a pub
+000152e0: 6c69 6361 7469 6f6e 2d72 6561 6479 2066  lication-ready f
+000152f0: 6967 7572 6520 7468 6174 2061 7373 656d  igure that assem
+00015300: 626c 6573 2074 6865 2072 6573 756c 7473  bles the results
+00015310: 206f 6620 454c 4d4d 2c20 474d 4d20 616e   of ELMM, GMM an
+00015320: 6420 7261 7465 2063 6f72 7265 6374 696f  d rate correctio
+00015330: 6e2e 2054 6865 206f 6e65 2d76 732d 6f6e  n. The one-vs-on
+00015340: 6520 6f72 7468 6f6c 6f67 6f75 7320 2a4b  e orthologous *K
+00015350: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+00015360: 7374 7269 6275 7469 6f6e 7320 6973 2061  stributions is a
+00015370: 6c73 6f20 6175 746f 6d61 7469 6361 6c6c  lso automaticall
+00015380: 7920 7072 6f64 7563 6564 2077 6974 6820  y produced with 
+00015390: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
+000153a0: 7265 7375 6c74 7320 7375 7065 7269 6d70  results superimp
+000153b0: 6f73 6564 2077 6865 7265 2061 7661 696c  osed where avail
+000153c0: 6162 6c65 2061 7320 7368 6f77 6e20 6265  able as shown be
+000153d0: 6c6f 772e 0a0a 215b 5d28 6461 7461 2f46  low...![](data/F
+000153e0: 6f63 7573 5f73 6973 7465 725f 7061 6972  ocus_sister_pair
+000153f0: 732e 6b73 2e6e 6f64 652e 7765 6967 6874  s.ks.node.weight
+00015400: 6564 2e70 6e67 290a 215b 5d28 6461 7461  ed.png).![](data
+00015410: 2f41 6c6c 5f70 6169 7273 2e6b 732e 6e6f  /All_pairs.ks.no
+00015420: 6465 2e77 6569 6768 7465 642e 706e 6729  de.weighted.png)
+00015430: 0a0a 4265 7369 6465 7320 7468 6520 6162  ..Besides the ab
+00015440: 6f76 6520 6b73 7261 7465 732d 6c69 6b65  ove ksrates-like
+00015450: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00015460: 2070 6c6f 742c 2061 206d 6f72 6520 636c   plot, a more cl
+00015470: 6173 7369 6320 2a4b 2a3c 7375 623e 533c  assic *K*<sub>S<
+00015480: 2f73 7562 3e20 706c 6f74 2063 616e 2062  /sub> plot can b
+00015490: 6520 6d61 6465 2062 7920 6164 6469 6e67  e made by adding
+000154a0: 2074 6865 206f 7074 696f 6e20 602d 2d63   the option `--c
+000154b0: 6c61 7373 6963 6020 746f 2074 6170 206d  lassic` to tap m
+000154c0: 6f72 6520 6465 7461 696c 6564 6c79 2069  ore detailedly i
+000154d0: 6e74 6f20 7468 6520 7661 7269 6174 696f  nto the variatio
+000154e0: 6e20 6f66 2073 796e 6f6e 796d 6f75 7320  n of synonymous 
+000154f0: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
+00015500: 652e 0a0a 5573 696e 6720 7468 6520 636f  e...Using the co
+00015510: 6d6d 616e 6420 6265 6c6f 772c 2074 6865  mmand below, the
+00015520: 2064 6972 6563 7469 6f6e 206f 6620 7261   direction of ra
+00015530: 7465 2063 6f72 7265 6374 696f 6e20 616e  te correction an
+00015540: 6420 6465 6772 6565 206f 6620 7261 7465  d degree of rate
+00015550: 2076 6172 6961 7469 6f6e 2063 616e 2062   variation can b
+00015560: 6520 6f62 7365 7276 6564 206d 6f72 6520  e observed more 
+00015570: 6469 7265 6374 6c79 2e0a 0a60 6060 0a77  directly...```.w
+00015580: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
+00015590: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
+000155a0: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
+000155b0: 7376 202d 6661 2041 7175 696c 6567 6961  sv -fa Aquilegia
+000155c0: 5f63 6f65 7275 6c65 6120 2d65 706b 2077  _coerulea -epk w
+000155d0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
+000155e0: 5f63 6f65 7275 6c65 612e 6b73 2e74 7376  _coerulea.ks.tsv
+000155f0: 202d 6170 2077 6764 5f73 796e 2f69 6164   -ap wgd_syn/iad
+00015600: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
+00015610: 6f69 6e74 732e 7478 7420 2d73 7020 7370  oints.txt -sp sp
+00015620: 6563 6965 7374 7265 652e 6e77 202d 6f20  eciestree.nw -o 
+00015630: 7767 645f 7669 7a5f 6d69 7865 645f 4b73  wgd_viz_mixed_Ks
+00015640: 202d 2d70 6c6f 7465 6c6d 6d20 2d2d 706c   --plotelmm --pl
+00015650: 6f74 6170 676d 6d20 2d2d 7265 7765 6967  otapgmm --reweig
+00015660: 6874 202d 2d70 6c6f 746b 6465 202d 2d63  ht --plotkde --c
+00015670: 6c61 7373 6963 0a60 6060 0a0a 215b 5d28  lassic.```..![](
+00015680: 6461 7461 2f41 7175 696c 6567 6961 5f63  data/Aquilegia_c
+00015690: 6f65 7275 6c65 615f 476c 6f62 616c 6d72  oerulea_Globalmr
+000156a0: 6268 4b73 5f45 6c6d 6d5f 4170 676d 6d5f  bhKs_Elmm_Apgmm_
+000156b0: 436f 7272 6563 7465 642e 6b73 642e 7376  Corrected.ksd.sv
+000156c0: 6729 0a0a 4173 2073 686f 776e 2061 626f  g)..As shown abo
+000156d0: 7665 2c20 6265 6361 7573 6520 6f66 2074  ve, because of t
+000156e0: 6865 2068 6967 6865 7220 7375 6273 7469  he higher substi
+000156f0: 7475 7469 6f6e 2072 6174 6520 6f66 202a  tution rate of *
+00015700: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
+00015710: 6561 2a2c 2074 6865 206f 7269 6769 6e61  ea*, the origina
+00015720: 6c20 6f72 7468 6f6c 6f67 6f75 7320 2a4b  l orthologous *K
+00015730: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
+00015740: 6c75 6573 2077 6572 6520 6163 7475 616c  lues were actual
+00015750: 6c79 2075 6e64 6572 6573 7469 6d61 7465  ly underestimate
+00015760: 6420 696e 2074 6865 2074 696d 652d 6672  d in the time-fr
+00015770: 616d 6520 6f66 202a 4171 7569 6c65 6769  ame of *Aquilegi
+00015780: 6120 636f 6572 756c 6561 2a2e 2057 6865  a coerulea*. Whe
+00015790: 6e20 7765 2072 6563 6f76 6572 6564 2074  n we recovered t
+000157a0: 6865 2064 6976 6572 6765 6e63 6520 7375  he divergence su
+000157b0: 6273 7469 7475 7469 6f6e 2064 6973 7461  bstitution dista
+000157c0: 6e63 6520 696e 2074 6572 6d73 206f 6620  nce in terms of 
+000157d0: 7477 6f20 7469 6d65 7320 6f66 2074 6865  two times of the
+000157e0: 2062 7261 6e63 682d 7370 6563 6966 6963   branch-specific
+000157f0: 2063 6f6e 7472 6962 7574 696f 6e20 6f66   contribution of
+00015800: 202a 412e 2063 6f65 7275 6c65 612a 2073   *A. coerulea* s
+00015810: 696e 6365 2069 7473 2064 6976 6572 6765  ince its diverge
+00015820: 6e63 6520 7769 7468 2074 6865 2073 6973  nce with the sis
+00015830: 7465 7220 7370 6563 6965 7320 706c 7573  ter species plus
+00015840: 2074 6865 2073 6861 7265 6420 7375 6273   the shared subs
+00015850: 7469 7475 7469 6f6e 2064 6973 7461 6e63  titution distanc
+00015860: 6520 6265 666f 7265 2064 6976 6572 6765  e before diverge
+00015870: 6e63 6520 2869 6e20 7265 6c61 7469 7665  nce (in relative
+00015880: 2074 6f20 7468 6520 6f75 7467 726f 7570   to the outgroup
+00015890: 292c 2074 6865 2063 6f72 7265 6374 6564  ), the corrected
+000158a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000158b0: 206d 6f64 6520 6265 6361 6d65 206c 6172   mode became lar
+000158c0: 6765 722e 0a0a 4e6f 7465 2074 6861 7420  ger...Note that 
+000158d0: 7765 2063 616e 2065 6173 696c 7920 7368  we can easily sh
+000158e0: 6f77 2074 6861 7420 2a41 7175 696c 6567  ow that *Aquileg
+000158f0: 6961 2063 6f65 7275 6c65 612a 2068 6173  ia coerulea* has
+00015900: 2068 6967 6865 7220 7375 6273 7469 7475   higher substitu
+00015910: 7469 6f6e 2072 6174 6520 7468 616e 202a  tion rate than *
+00015920: 5072 6f74 6561 2063 796e 6172 6f69 6465  Protea cynaroide
+00015930: 732a 2061 6e64 202a 5669 7469 7320 7669  s* and *Vitis vi
+00015940: 6e69 6665 7261 2a20 6279 2063 6f6d 7061  nifera* by compa
+00015950: 7269 6e67 2074 6865 6972 2073 7562 7374  ring their subst
+00015960: 6974 7574 696f 6e20 6469 7374 616e 6365  itution distance
+00015970: 2069 6e20 7265 6761 7264 2074 6f20 7468   in regard to th
+00015980: 6520 7361 6d65 2064 6976 6572 6765 6e63  e same divergenc
+00015990: 6520 6576 656e 7420 7769 7468 206f 7574  e event with out
+000159a0: 6772 6f75 7020 7370 6563 6965 7320 2a41  group species *A
+000159b0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
+000159c0: 2a2c 2075 7369 6e67 2063 6f6d 6d61 6e64  *, using command
+000159d0: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
+000159e0: 2076 697a 202d 6420 7767 645f 676c 6f62   viz -d wgd_glob
+000159f0: 616c 6d72 6268 5f6b 732f 676c 6f62 616c  almrbh_ks/global
+00015a00: 5f4d 5242 482e 7473 762e 6b73 2e74 7376  _MRBH.tsv.ks.tsv
+00015a10: 202d 7370 2073 7065 6369 6573 7472 6565   -sp speciestree
+00015a20: 2e6e 7720 2d2d 7265 7765 6967 6874 202d  .nw --reweight -
+00015a30: 6f20 7767 645f 7669 7a5f 436f 6d70 6172  o wgd_viz_Compar
+00015a40: 655f 7261 7465 202d 2d73 7061 6972 2022  e_rate --spair "
+00015a50: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
+00015a60: 733b 5072 6f74 6561 5f63 796e 6172 6f69  s;Protea_cynaroi
+00015a70: 6465 7322 202d 2d73 7061 6972 2022 4171  des" --spair "Aq
+00015a80: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00015a90: 3b41 636f 7275 735f 616d 6572 6963 616e  ;Acorus_american
+00015aa0: 7573 2220 2d2d 7370 6169 7220 2256 6974  us" --spair "Vit
+00015ab0: 6973 5f76 696e 6966 6572 613b 4163 6f72  is_vinifera;Acor
+00015ac0: 7573 5f61 6d65 7269 6361 6e75 7322 202d  us_americanus" -
+00015ad0: 2d70 6c6f 746b 6465 202d 2d63 6c61 7373  -plotkde --class
+00015ae0: 6963 0a60 6060 0a0a 215b 5d28 6461 7461  ic.```..![](data
+00015af0: 2f52 6177 5f4f 7274 686f 6c6f 6775 6573  /Raw_Orthologues
+00015b00: 5f43 6f6d 7061 7265 5f72 6174 652e 6b73  _Compare_rate.ks
+00015b10: 642e 7376 6729 0a0a 4173 2064 6973 706c  d.svg)..As displ
+00015b20: 6179 6564 2061 626f 7665 2c20 7468 6520  ayed above, the 
+00015b30: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
+00015b40: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
+00015b50: 6573 2062 6577 7465 656e 202a 4171 7569  es bewteen *Aqui
+00015b60: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
+00015b70: 616e 6420 2a41 636f 7275 7320 616d 6572  and *Acorus amer
+00015b80: 6963 616e 7573 2a20 6861 7320 7468 6520  icanus* has the 
+00015b90: 6869 6768 6573 7420 6d6f 6465 2c20 696e  highest mode, in
+00015ba0: 6469 6361 7469 6e67 2074 6865 2066 6173  dicating the fas
+00015bb0: 7465 7220 7375 6273 7469 7475 7469 6f6e  ter substitution
+00015bc0: 2072 6174 6520 6f66 202a 412e 2063 6f65   rate of *A. coe
+00015bd0: 7275 6c65 612a 2063 6f6d 7061 7265 6420  rulea* compared 
+00015be0: 746f 202a 5072 6f74 6561 2063 796e 6172  to *Protea cynar
+00015bf0: 6f69 6465 732a 2061 6e64 202a 5669 7469  oides* and *Viti
+00015c00: 7320 7669 6e69 6665 7261 2a2e 0a0a 4265  s vinifera*...Be
+00015c10: 666f 7265 2076 322e 302e 3231 2c20 7468  fore v2.0.21, th
+00015c20: 6520 6765 6e65 2d73 7065 6369 6573 206d  e gene-species m
+00015c30: 6170 2066 696c 6520 6973 206e 6563 6365  ap file is necce
+00015c40: 7373 6172 696c 7920 6e65 6564 6564 2066  ssarily needed f
+00015c50: 6f72 2069 7473 2069 6d70 6c65 6d65 6e74  or its implement
+00015c60: 6174 696f 6e20 696e 2060 7767 6420 7669  ation in `wgd vi
+00015c70: 7a60 2c20 7768 6963 6820 7368 6f75 6c64  z`, which should
+00015c80: 2062 6520 6175 746f 6d61 7465 6c79 2070   be automately p
+00015c90: 726f 6475 6365 6420 6279 2074 6865 206c  roduced by the l
+00015ca0: 6173 7420 6077 6764 206b 7364 6020 7374  ast `wgd ksd` st
+00015cb0: 6570 2067 6976 656e 2074 6865 2060 7370  ep given the `sp
+00015cc0: 6169 7260 2061 6e64 2060 7370 6563 6965  air` and `specie
+00015cd0: 7374 7265 6560 2070 6172 616d 6574 6572  stree` parameter
+00015ce0: 732e 2054 6865 2060 6765 6e65 5f73 7065  s. The `gene_spe
+00015cf0: 6369 6573 2e6d 6170 6020 6861 7320 636f  cies.map` has co
+00015d00: 6e74 656e 7473 2061 7320 6265 6c6f 7720  ntents as below 
+00015d10: 696e 2077 6869 6368 2065 6163 6820 6c69  in which each li
+00015d20: 6e65 2069 7320 7468 6520 6a6f 696e 6564  ne is the joined
+00015d30: 2073 7472 696e 6720 6f66 2067 656e 6520   string of gene 
+00015d40: 6e61 6d65 2061 6e64 2073 7065 6369 6573  name and species
+00015d50: 206e 616d 6520 6279 2073 7061 6365 2e20   name by space. 
+00015d60: 4166 7465 7220 7632 2e30 2e32 3120 2869  After v2.0.21 (i
+00015d70: 6e63 6c75 6465 6429 2c20 7468 6520 6765  ncluded), the ge
+00015d80: 6e65 2d73 7065 6369 6573 206d 6170 2066  ne-species map f
+00015d90: 696c 6520 6973 206e 6f74 206e 6563 6365  ile is not necce
+00015da0: 7373 6172 696c 7920 6e65 6564 6564 2061  ssarily needed a
+00015db0: 6e79 6d6f 7265 2e0a 0a60 6060 0a41 7163  nymore...```.Aqc
+00015dc0: 6f65 3647 3035 3738 3030 2e31 2041 7175  oe6G057800.1 Aqu
+00015dd0: 696c 6567 6961 5f63 6f65 7275 6c65 610a  ilegia_coerulea.
+00015de0: 5676 695f 5649 545f 3230 3173 3030 3131  Vvi_VIT_201s0011
+00015df0: 6730 3135 3330 2e31 2056 6974 6973 5f76  g01530.1 Vitis_v
+00015e00: 696e 6966 6572 610a 5063 795f 5072 6f63  inifera.Pcy_Proc
+00015e10: 7930 3167 3038 3531 3020 5072 6f74 6561  y01g08510 Protea
+00015e20: 5f63 796e 6172 6f69 6465 730a 4161 6d5f  _cynaroides.Aam_
+00015e30: 4163 6f72 612e 3034 4731 3432 3930 302e  Acora.04G142900.
+00015e40: 3120 4163 6f72 7573 5f61 6d65 7269 6361  1 Acorus_america
+00015e50: 6e75 730a 6060 600a 0a41 6e20 616c 7465  nus.```..An alte
+00015e60: 726e 6174 6976 6520 7761 7920 746f 2063  rnative way to c
+00015e70: 616c 6375 6c61 7465 2074 6865 206f 7274  alculate the ort
+00015e80: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
+00015e90: 3e53 3c2f 7375 623e 2069 7320 746f 2064  >S</sub> is to d
+00015ea0: 6972 6563 746c 7920 7573 6520 7468 6520  irectly use the 
+00015eb0: 6f72 7468 6f67 726f 7570 7320 696e 7374  orthogroups inst
+00015ec0: 6561 6420 6f66 2067 6c6f 6261 6c20 4d52  ead of global MR
+00015ed0: 4248 2066 616d 696c 792e 2054 6861 7420  BH family. That 
+00015ee0: 7761 7920 7765 2064 6f6e 2774 2075 7365  way we don't use
+00015ef0: 2074 6865 2070 7265 2d69 6e66 6572 7265   the pre-inferre
+00015f00: 6420 7061 7261 6e6f 6d65 202a 4b2a 3c73  d paranome *K*<s
+00015f10: 7562 3e53 3c2f 7375 623e 2062 7574 2074  ub>S</sub> but t
+00015f20: 6865 2070 6172 616c 6f67 6f75 7320 6765  he paralogous ge
+00015f30: 6e65 2070 6169 7273 2069 6e73 6964 6520  ne pairs inside 
+00015f40: 6561 6368 206f 7274 686f 6772 6f75 7020  each orthogroup 
+00015f50: 696e 7374 6561 642e 2054 6f20 6163 6869  instead. To achi
+00015f60: 6576 6520 7468 6174 2c20 7765 2066 6972  eve that, we fir
+00015f70: 7374 206e 6565 6420 746f 2069 6e66 6572  st need to infer
+00015f80: 206f 7274 686f 6772 6f75 7073 2075 7369   orthogroups usi
+00015f90: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
+00015fa0: 656c 6f77 2e0a 0a60 6060 0a77 6764 2064  elow...```.wgd d
+00015fb0: 6d64 2041 7175 696c 6567 6961 5f63 6f65  md Aquilegia_coe
+00015fc0: 7275 6c65 6120 5072 6f74 6561 5f63 796e  rulea Protea_cyn
+00015fd0: 6172 6f69 6465 7320 4163 6f72 7573 5f61  aroides Acorus_a
+00015fe0: 6d65 7269 6361 6e75 7320 5669 7469 735f  mericanus Vitis_
+00015ff0: 7669 6e69 6665 7261 202d 2d6f 7274 686f  vinifera --ortho
+00016000: 696e 6665 7220 2d6f 2077 6764 5f6f 7274  infer -o wgd_ort
+00016010: 686f 2028 2d2d 6f6e 6c79 6f72 7468 6f29  ho (--onlyortho)
+00016020: 200a 6060 600a 0a55 7365 7273 2063 616e   .```..Users can
+00016030: 2064 6563 6964 6520 746f 206f 6e6c 7920   decide to only 
+00016040: 636f 6e64 7563 7420 7468 6520 6f72 7468  conduct the orth
+00016050: 6f67 726f 7570 2061 6e61 6c79 7369 7320  ogroup analysis 
+00016060: 7768 696c 6520 736b 6970 7069 6e67 206f  while skipping o
+00016070: 7468 6572 2061 6e61 6c79 7369 7320 6279  ther analysis by
+00016080: 2061 6464 696e 6720 7468 6520 666c 6167   adding the flag
+00016090: 2060 2d2d 6f6e 6c79 6f72 7468 6f60 2e20   `--onlyortho`. 
+000160a0: 4e65 7874 2073 7465 7020 6973 2074 6865  Next step is the
+000160b0: 2073 616d 6520 7769 7468 2067 6c6f 6261   same with globa
+000160c0: 6c20 4d52 4248 2066 616d 696c 792e 0a0a  l MRBH family...
+000160d0: 6060 600a 7767 6420 6b73 6420 7767 645f  ```.wgd ksd wgd_
+000160e0: 6f72 7468 6f2f 4f72 7468 6f67 726f 7570  ortho/Orthogroup
+000160f0: 732e 7370 2e74 7376 2041 7175 696c 6567  s.sp.tsv Aquileg
+00016100: 6961 5f63 6f65 7275 6c65 6120 5072 6f74  ia_coerulea Prot
+00016110: 6561 5f63 796e 6172 6f69 6465 7320 4163  ea_cynaroides Ac
+00016120: 6f72 7573 5f61 6d65 7269 6361 6e75 7320  orus_americanus 
+00016130: 5669 7469 735f 7669 6e69 6665 7261 202d  Vitis_vinifera -
+00016140: 6f20 7767 645f 6f72 7468 6f5f 6b73 0a77  o wgd_ortho_ks.w
+00016150: 6764 2076 697a 202d 6420 7767 645f 6f72  gd viz -d wgd_or
+00016160: 7468 6f5f 6b73 2f4f 7274 686f 6772 6f75  tho_ks/Orthogrou
+00016170: 7073 2e73 702e 7473 762e 6b73 2e74 7376  ps.sp.tsv.ks.tsv
+00016180: 202d 6570 6b20 7767 645f 6b73 642f 4171   -epk wgd_ksd/Aq
+00016190: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000161a0: 2e6b 732e 7473 7620 2d73 7020 7370 6563  .ks.tsv -sp spec
+000161b0: 6965 7374 7265 652e 6e77 202d 2d72 6577  iestree.nw --rew
+000161c0: 6569 6768 7420 2d61 7020 7767 645f 7379  eight -ap wgd_sy
+000161d0: 6e2f 6961 6468 6f72 652d 6f75 742f 616e  n/iadhore-out/an
+000161e0: 6368 6f72 706f 696e 7473 2e74 7874 202d  chorpoints.txt -
+000161f0: 706c 6f74 656c 6d6d 202d 2d70 6c6f 7461  plotelmm --plota
+00016200: 7067 6d6d 202d 6f20 7767 645f 6f72 7468  pgmm -o wgd_orth
+00016210: 6f5f 6b73 5f72 6174 655f 636f 7272 6563  o_ks_rate_correc
+00016220: 7469 6f6e 0a60 6060 0a0a 215b 5d28 6461  tion.```..![](da
+00016230: 7461 2f4d 6978 6564 2e6b 732e 4171 7569  ta/Mixed.ks.Aqui
+00016240: 6c65 6769 615f 636f 6572 756c 6561 2e6e  legia_coerulea.n
+00016250: 6f64 652e 7765 6967 6874 6564 5f4f 7274  ode.weighted_Ort
+00016260: 686f 5f4b 732e 706e 6729 0a21 5b5d 2864  ho_Ks.png).![](d
+00016270: 6174 612f 466f 6375 735f 7369 7374 6572  ata/Focus_sister
+00016280: 5f70 6169 7273 2e6b 732e 6e6f 6465 2e77  _pairs.ks.node.w
+00016290: 6569 6768 7465 645f 4f72 7468 6f5f 4b73  eighted_Ortho_Ks
+000162a0: 2e70 6e67 290a 215b 5d28 6461 7461 2f41  .png).![](data/A
+000162b0: 6c6c 5f70 6169 7273 2e6b 732e 6e6f 6465  ll_pairs.ks.node
+000162c0: 2e77 6569 6768 7465 645f 4f72 7468 6f5f  .weighted_Ortho_
+000162d0: 4b73 2e70 6e67 290a 0a41 7320 7368 6f77  Ks.png)..As show
+000162e0: 6e20 6162 6f76 652c 2074 6865 206e 756d  n above, the num
+000162f0: 6265 7220 6f66 206f 7274 686f 6c6f 676f  ber of orthologo
+00016300: 7573 2067 656e 6520 7061 6972 7320 6973  us gene pairs is
+00016310: 2064 6966 6665 7265 6e74 2074 6861 6e20   different than 
+00016320: 7468 6520 6f6e 6520 6672 6f6d 2067 6c6f  the one from glo
+00016330: 6261 6c20 4d52 4248 2066 616d 696c 6965  bal MRBH familie
+00016340: 7320 696e 2074 6861 7420 6865 7265 2077  s in that here w
+00016350: 6520 706c 6f74 7465 6420 616c 6c20 6f72  e plotted all or
+00016360: 7468 6f6c 6f67 6f75 7320 6765 6e65 2070  thologous gene p
+00016370: 6169 7273 2069 6e73 7465 6164 206f 6620  airs instead of 
+00016380: 6f6e 6c79 2067 6c6f 6261 6c20 4d52 4248  only global MRBH
+00016390: 2066 616d 696c 6965 732c 2074 6f67 6574   families, toget
+000163a0: 6865 7220 7769 7468 2064 6966 6665 7265  her with differe
+000163b0: 6e74 2072 6563 616c 6375 6c61 7465 6420  nt recalculated 
+000163c0: 7765 6967 6874 732e 0a0a 4166 7465 7220  weights...After 
+000163d0: 7468 6520 7068 796c 6f67 656e 6574 6963  the phylogenetic
+000163e0: 2074 696d 696e 6720 6f66 2074 6865 2052   timing of the R
+000163f0: 616e 756e 6375 6c61 6c65 7320 5747 442c  anunculales WGD,
+00016400: 2077 6520 6361 6e20 6675 7274 6865 7220   we can further 
+00016410: 696e 6665 7220 6974 7320 6162 736f 6c75  infer its absolu
+00016420: 7465 2061 6765 2e20 4669 7273 7420 7765  te age. First we
+00016430: 2069 6e66 6572 2074 6865 2063 7265 6469   infer the credi
+00016440: 626c 6520 7261 6e67 6520 6f66 2061 6e63  ble range of anc
+00016450: 686f 7220 7061 6972 7320 6279 202a 4b2a  hor pairs by *K*
+00016460: 3c73 7562 3e53 3c2f 7375 623e 2068 6575  <sub>S</sub> heu
+00016470: 7269 7374 6963 616c 6c79 2075 7369 6e67  ristically using
+00016480: 2074 6865 2070 726f 6772 616d 2060 7767   the program `wg
+00016490: 6420 7065 616b 602e 0a0a 6060 600a 7767  d peak`...```.wg
+000164a0: 6420 7065 616b 202d 2d68 6575 7269 7374  d peak --heurist
+000164b0: 6963 2077 6764 5f6b 7364 2f41 7175 696c  ic wgd_ksd/Aquil
+000164c0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+000164d0: 762e 6b73 2e74 7376 202d 6170 2077 6764  v.ks.tsv -ap wgd
+000164e0: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
+000164f0: 2f61 6e63 686f 7270 6f69 6e74 732e 7478  /anchorpoints.tx
+00016500: 7420 2d73 6d20 7767 645f 7379 6e2f 6961  t -sm wgd_syn/ia
+00016510: 6468 6f72 652d 6f75 742f 7365 676d 656e  dhore-out/segmen
+00016520: 7473 2e74 7874 202d 6c65 2077 6764 5f73  ts.txt -le wgd_s
+00016530: 796e 2f69 6164 686f 7265 2d6f 7574 2f6c  yn/iadhore-out/l
+00016540: 6973 745f 656c 656d 656e 7473 2e74 7874  ist_elements.txt
+00016550: 202d 6d70 2077 6764 5f73 796e 2f69 6164   -mp wgd_syn/iad
+00016560: 686f 7265 2d6f 7574 2f6d 756c 7469 706c  hore-out/multipl
+00016570: 6963 6f6e 5f70 6169 7273 2e74 7874 202d  icon_pairs.txt -
+00016580: 6f20 7767 645f 7065 616b 0a60 6060 0a0a  o wgd_peak.```..
+00016590: 215b 5d28 6461 7461 2f41 6e63 686f 724b  ![](data/AnchorK
+000165a0: 735f 5065 616b 4349 5f41 7175 696c 6567  s_PeakCI_Aquileg
+000165b0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+000165c0: 6b73 2e74 7376 5f6e 6f64 655f 7765 6967  ks.tsv_node_weig
+000165d0: 6874 6564 2e73 7667 290a 0a41 7320 7368  hted.svg)..As sh
+000165e0: 6f77 6e20 6162 6f76 652c 2077 6520 6173  own above, we as
+000165f0: 7375 6d65 6420 6120 6c6f 676e 6f72 6d61  sumed a lognorma
+00016600: 6c20 6469 7374 7269 6275 7469 6f6e 2061  l distribution a
+00016610: 7420 7468 6520 7065 616b 206c 6f63 6174  t the peak locat
+00016620: 696f 6e20 6465 7465 6374 6564 2062 7920  ion detected by 
+00016630: 7468 6520 6073 6967 6e61 6c60 206d 6f64  the `signal` mod
+00016640: 756c 6520 6f66 2060 7363 6970 7960 206c  ule of `scipy` l
+00016650: 6962 7261 7279 2e20 5468 6520 3935 2520  ibrary. The 95% 
+00016660: 636f 6e66 6964 656e 6365 206c 6576 656c  confidence level
+00016670: 206f 6620 7468 6520 6c6f 676e 6f72 6d61   of the lognorma
+00016680: 6c20 6469 7374 7269 6275 7469 6f6e 2077  l distribution w
+00016690: 6173 2061 7070 6c69 6564 2c20 692e 652e  as applied, i.e.
+000166a0: 2c20 302e 3638 2d32 2e37 342c 2069 6e20  , 0.68-2.74, in 
+000166b0: 6675 7274 6865 7220 6d6f 6c65 6375 6c61  further molecula
+000166c0: 7220 6461 7469 6e67 2e20 5468 6520 6669  r dating. The fi
+000166d0: 6c65 2060 4171 7569 6c65 6769 615f 636f  le `Aquilegia_co
+000166e0: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
+000166f0: 765f 3935 2543 495f 4150 5f66 6f72 5f64  v_95%CI_AP_for_d
+00016700: 6174 696e 675f 7765 6967 6874 6564 5f66  ating_weighted_f
+00016710: 6f72 6d61 742e 7473 7660 2069 7320 7768  ormat.tsv` is wh
+00016720: 6174 2077 6520 6e65 6564 2066 6f72 206e  at we need for n
+00016730: 6578 7420 7374 6570 2e20 546f 2062 7569  ext step. To bui
+00016740: 6c64 2074 6865 206f 7274 686f 6772 6f75  ld the orthogrou
+00016750: 7073 2075 7365 6420 696e 2070 6879 6c6f  ps used in phylo
+00016760: 6765 6e65 7469 6320 6461 7469 6e67 2c20  genetic dating, 
+00016770: 7765 206e 6565 6420 746f 2073 656c 6563  we need to selec
+00016780: 7420 736f 6d65 2073 7065 6369 6573 2061  t some species a
+00016790: 6e64 2066 6f72 6d20 6120 7374 6172 7469  nd form a starti
+000167a0: 6e67 2074 7265 6520 7769 7468 2070 726f  ng tree with pro
+000167b0: 7065 7220 666f 7373 696c 2063 616c 6962  per fossil calib
+000167c0: 7261 7469 6f6e 732e 2057 6520 7072 6f76  rations. We prov
+000167d0: 6964 6520 6f6e 6520 696e 206d 636d 6374  ide one in mcmct
+000167e0: 7265 6520 666f 726d 6174 2061 7320 6265  ree format as be
+000167f0: 6c6f 772e 0a0a 6060 600a 3137 2031 0a28  low...```.17 1.(
+00016800: 2828 2850 6f74 616d 6f67 6574 6f6e 5f61  (((Potamogeton_a
+00016810: 6375 7469 666f 6c69 7573 2c28 5370 6972  cutifolius,(Spir
+00016820: 6f64 656c 615f 696e 7465 726d 6564 6961  odela_intermedia
+00016830: 2c41 6d6f 7270 686f 7068 616c 6c75 735f  ,Amorphophallus_
+00016840: 6b6f 6e6a 6163 2929 2c28 4163 616e 7468  konjac)),(Acanth
+00016850: 6f63 686c 616d 7973 5f62 7261 6374 6561  ochlamys_bractea
+00016860: 7461 2c28 4469 6f73 636f 7265 615f 616c  ta,(Dioscorea_al
+00016870: 6174 612c 4469 6f73 636f 7265 615f 726f  ata,Dioscorea_ro
+00016880: 7475 6e64 6174 6129 2927 3e30 2e35 3630  tundata))'>0.560
+00016890: 303c 312e 3238 3633 2729 273e 302e 3833  0<1.2863')'>0.83
+000168a0: 3630 3c31 2e32 3836 3327 2c28 4163 6f72  60<1.2863',(Acor
+000168b0: 7573 5f61 6d65 7269 6361 6e75 732c 4163  us_americanus,Ac
+000168c0: 6f72 7573 5f74 6174 6172 696e 6f77 6969  orus_tatarinowii
+000168d0: 2929 273e 302e 3833 3630 3c31 2e32 3836  ))'>0.8360<1.286
+000168e0: 3327 2c28 2828 2854 6574 7261 6365 6e74  3',((((Tetracent
+000168f0: 726f 6e5f 7369 6e65 6e73 652c 5472 6f63  ron_sinense,Troc
+00016900: 686f 6465 6e64 726f 6e5f 6172 616c 696f  hodendron_aralio
+00016910: 6964 6573 292c 2842 7578 7573 5f61 7573  ides),(Buxus_aus
+00016920: 7472 6f79 756e 6e61 6e65 6e73 6973 2c42  troyunnanensis,B
+00016930: 7578 7573 5f73 696e 6963 6129 2927 3e31  uxus_sinica))'>1
+00016940: 2e31 3038 303c 312e 3238 3633 272c 284e  .1080<1.2863',(N
+00016950: 656c 756d 626f 5f6e 7563 6966 6572 612c  elumbo_nucifera,
+00016960: 2854 656c 6f70 6561 5f73 7065 6369 6f73  (Telopea_specios
+00016970: 6973 7369 6d61 2c50 726f 7465 615f 6379  issima,Protea_cy
+00016980: 6e61 726f 6964 6573 2929 2927 3e31 2e31  naroides)))'>1.1
+00016990: 3038 303c 312e 3238 3633 272c 2841 7175  080<1.2863',(Aqu
+000169a0: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
+000169b0: 6170 312c 4171 7569 6c65 6769 615f 636f  ap1,Aquilegia_co
+000169c0: 6572 756c 6561 5f61 7032 2929 273e 312e  erulea_ap2))'>1.
+000169d0: 3130 3830 3c31 2e32 3836 3327 2927 3e31  1080<1.2863')'>1
+000169e0: 2e32 3732 303c 322e 3437 3230 273b 0a60  .2720<2.4720';.`
+000169f0: 6060 0a0a 4173 2070 7265 7365 6e74 6564  ``..As presented
+00016a00: 2061 626f 7665 2c20 7468 6520 666f 6375   above, the focu
+00016a10: 7320 7370 6563 6965 7320 7468 6174 2069  s species that i
+00016a20: 7320 6162 6f75 7420 746f 2062 6520 6461  s about to be da
+00016a30: 7465 6420 6e65 6564 7320 746f 2062 6520  ted needs to be 
+00016a40: 7265 706c 6163 6564 2077 6974 6820 6028  replaced with `(
+00016a50: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00016a60: 6561 5f61 7031 2c41 7175 696c 6567 6961  ea_ap1,Aquilegia
+00016a70: 5f63 6f65 7275 6c65 615f 6170 3229 602e  _coerulea_ap2)`.
+00016a80: 2057 6974 6820 7468 6973 2073 7461 7274   With this start
+00016a90: 696e 6720 7472 6565 2061 6e64 2070 7265  ing tree and pre
+00016aa0: 646f 776e 6c6f 6164 6564 2063 6473 2066  downloaded cds f
+00016ab0: 696c 6573 206f 6620 616c 6c20 7468 6520  iles of all the 
+00016ac0: 7370 6563 6965 732c 2077 6520 6361 6e20  species, we can 
+00016ad0: 6275 696c 6420 7468 6520 6f72 7468 6f67  build the orthog
+00016ae0: 726f 7570 2075 7365 6420 696e 2074 6865  roup used in the
+00016af0: 2066 696e 616c 206d 6f6c 6563 756c 6172   final molecular
+00016b00: 2064 6174 696e 6720 7573 696e 6720 7468   dating using th
+00016b10: 6520 636f 6d6d 616e 6420 6173 2062 656c  e command as bel
+00016b20: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
+00016b30: 202d 6620 4171 7569 6c65 6769 615f 636f   -f Aquilegia_co
+00016b40: 6572 756c 6561 202d 6170 2077 6764 5f70  erulea -ap wgd_p
+00016b50: 6561 6b2f 4171 7569 6c65 6769 615f 636f  eak/Aquilegia_co
+00016b60: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
+00016b70: 765f 3935 2543 495f 4150 5f66 6f72 5f64  v_95%CI_AP_for_d
+00016b80: 6174 696e 675f 7765 6967 6874 6564 5f66  ating_weighted_f
+00016b90: 6f72 6d61 742e 7473 7620 2d6f 2077 6764  ormat.tsv -o wgd
+00016ba0: 5f64 6d64 5f6f 7274 686f 2050 6f74 616d  _dmd_ortho Potam
+00016bb0: 6f67 6574 6f6e 5f61 6375 7469 666f 6c69  ogeton_acutifoli
+00016bc0: 7573 2053 7069 726f 6465 6c61 5f69 6e74  us Spirodela_int
+00016bd0: 6572 6d65 6469 6120 416d 6f72 7068 6f70  ermedia Amorphop
+00016be0: 6861 6c6c 7573 5f6b 6f6e 6a61 6320 4163  hallus_konjac Ac
+00016bf0: 616e 7468 6f63 686c 616d 7973 5f62 7261  anthochlamys_bra
+00016c00: 6374 6561 7461 2044 696f 7363 6f72 6561  cteata Dioscorea
+00016c10: 5f61 6c61 7461 2044 696f 7363 6f72 6561  _alata Dioscorea
+00016c20: 5f72 6f74 756e 6461 7461 2041 636f 7275  _rotundata Acoru
+00016c30: 735f 616d 6572 6963 616e 7573 2041 636f  s_americanus Aco
+00016c40: 7275 735f 7461 7461 7269 6e6f 7769 6920  rus_tatarinowii 
+00016c50: 5465 7472 6163 656e 7472 6f6e 5f73 696e  Tetracentron_sin
+00016c60: 656e 7365 2054 726f 6368 6f64 656e 6472  ense Trochodendr
+00016c70: 6f6e 5f61 7261 6c69 6f69 6465 7320 4275  on_aralioides Bu
+00016c80: 7875 735f 6175 7374 726f 7975 6e6e 616e  xus_austroyunnan
+00016c90: 656e 7369 7320 4275 7875 735f 7369 6e69  ensis Buxus_sini
+00016ca0: 6361 204e 656c 756d 626f 5f6e 7563 6966  ca Nelumbo_nucif
+00016cb0: 6572 6120 5465 6c6f 7065 615f 7370 6563  era Telopea_spec
+00016cc0: 696f 7369 7373 696d 6120 5072 6f74 6561  iosissima Protea
+00016cd0: 5f63 796e 6172 6f69 6465 7320 4171 7569  _cynaroides Aqui
+00016ce0: 6c65 6769 615f 636f 6572 756c 6561 0a60  legia_coerulea.`
+00016cf0: 6060 0a0a 5468 6520 7265 7375 6c74 2066  ``..The result f
+00016d00: 696c 6520 606d 6572 6765 5f66 6f63 7573  ile `merge_focus
+00016d10: 5f61 702e 7473 7660 2069 7320 7768 6174  _ap.tsv` is what
+00016d20: 2077 6520 6e65 6564 2066 6f72 2074 6865   we need for the
+00016d30: 2066 696e 616c 2073 7465 7020 6f66 206d   final step of m
+00016d40: 6f6c 6563 756c 6172 2064 6174 696e 6720  olecular dating 
+00016d50: 696e 2070 726f 6772 616d 2060 7767 6420  in program `wgd 
+00016d60: 666f 6375 7360 2e0a 0a60 6060 0a77 6764  focus`...```.wgd
+00016d70: 2066 6f63 7573 202d 2d70 726f 7464 6174   focus --protdat
+00016d80: 696e 6720 2d2d 6161 6d6f 6465 6c20 6c67  ing --aamodel lg
+00016d90: 2077 6764 5f64 6d64 5f6f 7274 686f 2f6d   wgd_dmd_ortho/m
+00016da0: 6572 6765 5f66 6f63 7573 5f61 702e 7473  erge_focus_ap.ts
+00016db0: 7620 2d73 7020 6461 7469 6e67 5f74 7265  v -sp dating_tre
+00016dc0: 652e 6e77 202d 6f20 7767 645f 6461 7469  e.nw -o wgd_dati
+00016dd0: 6e67 202d 6420 6d63 6d63 7472 6565 202d  ng -d mcmctree -
+00016de0: 6473 2027 6275 726e 696e 203d 2032 3030  ds 'burnin = 200
+00016df0: 3027 202d 6473 2027 7361 6d70 6672 6571  0' -ds 'sampfreq
+00016e00: 203d 2031 3030 3027 202d 6473 2027 6e73   = 1000' -ds 'ns
+00016e10: 616d 706c 6520 3d20 3230 3030 3027 2050  ample = 20000' P
+00016e20: 6f74 616d 6f67 6574 6f6e 5f61 6375 7469  otamogeton_acuti
+00016e30: 666f 6c69 7573 2053 7069 726f 6465 6c61  folius Spirodela
+00016e40: 5f69 6e74 6572 6d65 6469 6120 416d 6f72  _intermedia Amor
+00016e50: 7068 6f70 6861 6c6c 7573 5f6b 6f6e 6a61  phophallus_konja
+00016e60: 6320 4163 616e 7468 6f63 686c 616d 7973  c Acanthochlamys
+00016e70: 5f62 7261 6374 6561 7461 2044 696f 7363  _bracteata Diosc
+00016e80: 6f72 6561 5f61 6c61 7461 2044 696f 7363  orea_alata Diosc
+00016e90: 6f72 6561 5f72 6f74 756e 6461 7461 2041  orea_rotundata A
+00016ea0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
+00016eb0: 2041 636f 7275 735f 7461 7461 7269 6e6f   Acorus_tatarino
+00016ec0: 7769 6920 5465 7472 6163 656e 7472 6f6e  wii Tetracentron
+00016ed0: 5f73 696e 656e 7365 2054 726f 6368 6f64  _sinense Trochod
+00016ee0: 656e 6472 6f6e 5f61 7261 6c69 6f69 6465  endron_aralioide
+00016ef0: 7320 4275 7875 735f 6175 7374 726f 7975  s Buxus_austroyu
+00016f00: 6e6e 616e 656e 7369 7320 4275 7875 735f  nnanensis Buxus_
+00016f10: 7369 6e69 6361 204e 656c 756d 626f 5f6e  sinica Nelumbo_n
+00016f20: 7563 6966 6572 6120 5465 6c6f 7065 615f  ucifera Telopea_
+00016f30: 7370 6563 696f 7369 7373 696d 6120 5072  speciosissima Pr
+00016f40: 6f74 6561 5f63 796e 6172 6f69 6465 7320  otea_cynaroides 
+00016f50: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00016f60: 6561 0a60 6060 0a0a 4865 7265 2077 6520  ea.```..Here we 
+00016f70: 6f6e 6c79 2069 6d70 6c65 6d65 6e74 6564  only implemented
+00016f80: 2074 6865 2063 6f6e 6361 7465 6e61 7469   the concatenati
+00016f90: 6f6e 2061 6e61 6c79 7369 7320 7573 696e  on analysis usin
+00016fa0: 6720 7072 6f74 6569 6e20 7365 7175 656e  g protein sequen
+00016fb0: 6365 2062 7920 6164 6469 6e67 2074 6865  ce by adding the
+00016fc0: 2066 6c61 6720 602d 2d70 726f 7464 6174   flag `--protdat
+00016fd0: 696e 6760 2061 6e64 2077 6520 7365 7420  ing` and we set 
+00016fe0: 7468 6520 7061 7261 6d65 7465 7220 666f  the parameter fo
+00016ff0: 7220 606d 636d 6374 7265 6560 2076 6961  r `mcmctree` via
+00017000: 2074 6865 206f 7074 696f 6e20 602d 6473   the option `-ds
+00017010: 602e 204e 6f74 6520 7468 6174 206f 7468  `. Note that oth
+00017020: 6572 2064 6174 696e 6720 7072 6f67 7261  er dating progra
+00017030: 6d20 7375 6368 2061 7320 6072 3873 6020  m such as `r8s` 
+00017040: 616e 6420 6062 6561 7374 6020 6172 6520  and `beast` are 
+00017050: 616c 736f 2061 7661 696c 6162 6c65 2067  also available g
+00017060: 6976 656e 2073 6f6d 6520 6d61 6e64 6174  iven some mandat
+00017070: 6f72 7920 7061 7261 6d65 7465 7273 2e20  ory parameters. 
+00017080: 5468 6520 6669 6e61 6c20 6c6f 6720 6f66  The final log of
+00017090: 2074 6865 2073 7563 6365 7373 6675 6c20   the successful 
+000170a0: 7275 6e20 6973 2061 7320 6265 6c6f 772e  run is as below.
+000170b0: 0a0a 6060 600a 3136 3a30 343a 3235 2049  ..```.16:04:25 I
+000170c0: 4e46 4f20 2020 2020 5275 6e6e 696e 6720  NFO     Running 
+000170d0: 6d63 6d63 7472 6565 2075 7369 6e67 2048  mcmctree using H
+000170e0: 6573 7369 616e 206d 6174 7269 7820 6f66  essian matrix of
+000170f0: 204c 472b 4761 6d6d 6120 2063 6f72 652e   LG+Gamma  core.
+00017100: 7079 3a39 3637 0a20 2020 2020 2020 2020  py:967.         
+00017110: 2020 2020 2020 2020 2066 6f72 2070 726f           for pro
+00017120: 7465 696e 206d 6f64 656c 0a32 333a 3439  tein model.23:49
+00017130: 3a33 3720 494e 464f 2020 2020 2050 6f73  :37 INFO     Pos
+00017140: 7465 7269 6f72 206d 6561 6e20 666f 7220  terior mean for 
+00017150: 7468 6520 6167 6573 206f 6620 7767 6420  the ages of wgd 
+00017160: 6973 2031 3132 2e38 3934 3520 6d63 6d63  is 112.8945 mcmc
+00017170: 7472 6565 2e70 793a 3239 360a 2020 2020  tree.py:296.    
+00017180: 2020 2020 2020 2020 2020 2020 2020 6d69                mi
+00017190: 6c6c 696f 6e20 7965 6172 7320 6672 6f6d  llion years from
+000171a0: 2043 6f6e 6361 7465 6e61 7465 6420 7065   Concatenated pe
+000171b0: 7074 6964 650a 2020 2020 2020 2020 2020  ptide.          
+000171c0: 2020 2020 2020 2020 616c 6967 6e6d 656e          alignmen
+000171d0: 7420 616e 6420 3935 2520 6372 6564 6962  t and 95% credib
+000171e0: 696c 6974 7920 696e 7465 7276 616c 7320  ility intervals 
+000171f0: 2843 4929 0a20 2020 2020 2020 2020 2020  (CI).           
+00017200: 2020 2020 2020 2069 7320 3130 312e 3232         is 101.22
+00017210: 342d 3132 332e 3132 3120 6d69 6c6c 696f  4-123.121 millio
+00017220: 6e20 7965 6172 730a 2020 2020 2020 2020  n years.        
+00017230: 2049 4e46 4f20 2020 2020 546f 7461 6c20   INFO     Total 
+00017240: 7275 6e20 7469 6d65 3a20 3239 3137 3573  run time: 29175s
+00017250: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00017260: 2020 2020 2020 2020 2020 2020 2020 636c                cl
+00017270: 692e 7079 3a32 3431 0a20 2020 2020 2020  i.py:241.       
+00017280: 2020 494e 464f 2020 2020 2044 6f6e 6520    INFO     Done 
+00017290: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000172a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+000172b0: 2020 2020 2020 2020 2020 2020 2020 2063                 c
+000172c0: 6c69 2e70 793a 3234 320a 6060 600a 0a54  li.py:242.```..T
+000172d0: 6f20 7669 7375 616c 697a 6520 7468 6520  o visualize the 
+000172e0: 6461 7465 2c20 7765 2061 6c73 6f20 7072  date, we also pr
+000172f0: 6f76 6964 6564 2061 2070 7974 686f 6e20  ovided a python 
+00017300: 7363 7269 7074 2074 6f20 706c 6f74 2074  script to plot t
+00017310: 6865 2057 4744 2064 6174 6573 2069 6e20  he WGD dates in 
+00017320: 7468 6520 6077 6764 6020 666f 6c64 6572  the `wgd` folder
+00017330: 2e20 5573 6572 7320 6e65 6564 2074 6f20  . Users need to 
+00017340: 6578 7472 6163 7420 7468 6520 7261 7720  extract the raw 
+00017350: 6461 7465 7320 6672 6f6d 2074 6865 2060  dates from the `
+00017360: 6d63 6d63 2e74 7874 6020 666f 7220 7468  mcmc.txt` for th
+00017370: 6520 5747 4420 6e6f 6465 2066 6972 7374  e WGD node first
+00017380: 2061 6e64 2073 6176 6520 6974 2061 7320   and save it as 
+00017390: 6669 6c65 2060 6461 7465 732e 7478 7460  file `dates.txt`
+000173a0: 2028 6f72 2077 6861 7465 7665 7220 7072   (or whatever pr
+000173b0: 6566 6572 7265 6420 6e61 6d65 292e 2041  eferred name). A
+000173c0: 6e20 6578 616d 706c 6520 636f 6d6d 616e  n example comman
+000173d0: 6420 6973 2061 7320 6265 6c6f 772e 0a0a  d is as below...
+000173e0: 6060 600a 7079 7468 6f6e 2024 5041 5448  ```.python $PATH
+000173f0: 2f70 6f73 7470 6c6f 742e 7079 2070 6f73  /postplot.py pos
+00017400: 7464 6973 2064 6174 6573 2e74 7874 202d  tdis dates.txt -
+00017410: 2d70 6572 6365 6e74 696c 6520 3930 202d  -percentile 90 -
+00017420: 2d74 6974 6c65 2022 5747 4420 6461 7465  -title "WGD date
+00017430: 2220 2d2d 6870 6420 2d6f 2022 5261 6e75  " --hpd -o "Ranu
+00017440: 6e63 756c 616c 6573 5f57 4744 5f64 6174  nculales_WGD_dat
+00017450: 652e 7376 6722 0a60 6060 0a0a 215b 5d28  e.svg".```..![](
+00017460: 6461 7461 2f52 616e 756e 6375 6c61 6c65  data/Ranunculale
+00017470: 735f 5747 445f 6461 7465 2e73 7667 290a  s_WGD_date.svg).
+00017480: 0a54 6865 2070 6f73 7465 7269 6f72 206d  .The posterior m
+00017490: 6561 6e2c 206d 6564 6961 6e20 616e 6420  ean, median and 
+000174a0: 6d6f 6465 206f 6620 7468 6520 5261 6e75  mode of the Ranu
+000174b0: 6e63 756c 616c 6573 2057 4744 2061 6765  nculales WGD age
+000174c0: 2069 7320 3131 322e 3932 2c20 3131 332e   is 112.92, 113.
+000174d0: 3434 2061 6e64 2031 3132 2e35 3420 6d79  44 and 112.54 my
+000174e0: 612c 2077 6974 6820 3930 2520 4850 4420  a, with 90% HPD 
+000174f0: 3130 352e 3037 202d 2031 3232 2e33 3220  105.07 - 122.32 
+00017500: 6d79 6120 6173 206d 616e 6966 6573 7465  mya as manifeste
+00017510: 6420 6162 6f76 652e 0a0a 2323 2320 4b73  d above...### Ks
+00017520: 7472 6565 0a0a 496e 2061 6464 6974 696f  tree..In additio
+00017530: 6e20 746f 2070 6169 7277 6973 6520 2a4b  n to pairwise *K
+00017540: 2a3c 7375 623e 533c 2f73 7562 3e20 6573  *<sub>S</sub> es
+00017550: 7469 6d61 7469 6f6e 2c20 6120 2a4b 2a3c  timation, a *K*<
+00017560: 7375 623e 533c 2f73 7562 3e20 7472 6565  sub>S</sub> tree
+00017570: 2077 6974 6820 6272 616e 6368 206c 656e   with branch len
+00017580: 6774 6820 696e 202a 4b2a 3c73 7562 3e53  gth in *K*<sub>S
+00017590: 3c2f 7375 623e 2075 6e69 7420 6361 6e20  </sub> unit can 
+000175a0: 616c 736f 2062 6520 6465 7269 7665 6420  also be derived 
+000175b0: 6672 6f6d 2074 6865 2070 726f 6772 616d  from the program
+000175c0: 2060 7767 6420 6b73 6460 2067 6976 656e   `wgd ksd` given
+000175d0: 2074 6865 206f 7074 696f 6e20 602d 2d6b   the option `--k
+000175e0: 7374 7265 6560 2061 6e64 2060 2d2d 7370  stree` and `--sp
+000175f0: 6563 6965 7374 7265 6560 2e20 4e6f 7465  eciestree`. Note
+00017600: 2074 6861 7420 7468 6520 6164 6469 7469   that the additi
+00017610: 6f6e 616c 206f 7074 696f 6e20 602d 2d6f  onal option `--o
+00017620: 6e6c 7963 6f6e 6361 746b 7374 7265 6560  nlyconcatkstree`
+00017630: 2077 696c 6c20 6f6e 6c79 2063 616c 6c20   will only call 
+00017640: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
+00017650: 7562 3e20 6573 7469 6d61 7469 6f6e 2066  ub> estimation f
+00017660: 6f72 2074 6865 2063 6f6e 6361 7465 6e61  or the concatena
+00017670: 7465 6420 616c 6967 6e6d 656e 7420 7261  ted alignment ra
+00017680: 7468 6572 2074 6861 6e20 616c 6c20 7468  ther than all th
+00017690: 6520 616c 6967 6e6d 656e 7473 2e20 5573  e alignments. Us
+000176a0: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
+000176b0: 6964 6520 6120 7072 6573 6574 2073 7065  ide a preset spe
+000176c0: 6369 6573 2074 7265 6520 666f 7220 7468  cies tree for th
+000176d0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+000176e0: 3e20 7472 6565 2069 6e66 6572 656e 6365  > tree inference
+000176f0: 206f 6620 7468 6520 636f 6e63 6174 656e   of the concaten
+00017700: 6174 6564 2061 6c69 676e 6d65 6e74 2077  ated alignment w
+00017710: 6869 6c65 2074 6865 2072 656d 6169 6e69  hile the remaini
+00017720: 6e67 2061 6c69 676e 6d65 6e74 7320 7769  ng alignments wi
+00017730: 6c6c 2062 6520 6167 6169 6e73 7420 616e  ll be against an
+00017740: 2061 7574 6f6d 6174 656c 7920 696e 6665   automately infe
+00017750: 7272 6564 2074 7265 6520 6672 6f6d 2060  rred tree from `
+00017760: 6661 7374 7472 6565 6020 6f72 2060 6971  fasttree` or `iq
+00017770: 7472 6565 602e 2049 6e20 7468 6520 656e  tree`. In the en
+00017780: 642c 2075 7365 7273 2077 696c 6c20 6765  d, users will ge
+00017790: 7420 6120 2a4b 2a3c 7375 623e 533c 2f73  t a *K*<sub>S</s
+000177a0: 7562 3e20 7472 6565 2c20 6120 2a4b 2a3c  ub> tree, a *K*<
+000177b0: 7375 623e 413c 2f73 7562 3e20 7472 6565  sub>A</sub> tree
+000177c0: 2061 6e64 2061 20cf 8920 7472 6565 2070   and a .. tree p
+000177d0: 6572 2066 616d 2061 6e64 2066 6f72 2074  er fam and for t
+000177e0: 6865 2063 6f6e 6361 7465 6e61 7465 6420  he concatenated 
+000177f0: 616c 6967 6e6d 656e 742e 0a0a 6060 600a  alignment...```.
+00017800: 7767 6420 6b73 6420 6461 7461 2f6b 7374  wgd ksd data/kst
+00017810: 7265 655f 6461 7461 2f66 616d 2e74 7376  ree_data/fam.tsv
+00017820: 2064 6174 612f 6b73 7472 6565 5f64 6174   data/kstree_dat
+00017830: 612f 4163 6f72 7573 5f74 6174 6172 696e  a/Acorus_tatarin
+00017840: 6f77 6969 2064 6174 612f 6b73 7472 6565  owii data/kstree
+00017850: 5f64 6174 612f 416d 626f 7265 6c6c 615f  _data/Amborella_
+00017860: 7472 6963 686f 706f 6461 2064 6174 612f  trichopoda data/
+00017870: 6b73 7472 6565 5f64 6174 612f 4171 7569  kstree_data/Aqui
+00017880: 6c65 6769 615f 636f 6572 756c 6561 2064  legia_coerulea d
+00017890: 6174 612f 6b73 7472 6565 5f64 6174 612f  ata/kstree_data/
+000178a0: 4172 6973 746f 6c6f 6368 6961 5f66 696d  Aristolochia_fim
+000178b0: 6272 6961 7461 2064 6174 612f 6b73 7472  briata data/kstr
+000178c0: 6565 5f64 6174 612f 4379 6361 735f 7061  ee_data/Cycas_pa
+000178d0: 6e7a 6869 6875 6165 6e73 6920 2d2d 6b73  nzhihuaensi --ks
+000178e0: 7472 6565 202d 2d73 7065 6369 6573 7472  tree --speciestr
+000178f0: 6565 2064 6174 612f 6b73 7472 6565 5f64  ee data/kstree_d
+00017900: 6174 612f 7370 6563 6965 735f 7472 6565  ata/species_tree
+00017910: 312e 6e77 202d 2d6f 6e6c 7963 6f6e 6361  1.nw --onlyconca
+00017920: 746b 7374 7265 6520 2d6f 2077 6764 5f6b  tkstree -o wgd_k
+00017930: 7374 7265 655f 746f 706f 6c6f 6779 310a  stree_topology1.
+00017940: 6060 600a 0a21 5b5d 2864 6174 612f 6b73  ```..![](data/ks
+00017950: 7472 6565 5f72 6573 756c 7473 2f6b 7374  tree_results/kst
+00017960: 7265 652e 7376 6729 0a0a 4162 6f76 6520  ree.svg)..Above 
+00017970: 7765 2075 7365 6420 7468 7265 6520 616c  we used three al
+00017980: 7465 726e 6174 6976 6520 746f 706f 6c6f  ternative topolo
+00017990: 6769 6573 2074 6f20 696e 6665 7220 7468  gies to infer th
+000179a0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+000179b0: 3e20 7472 6565 2077 6869 6368 206c 6564  > tree which led
+000179c0: 2074 6f20 6469 6666 6572 656e 7420 6272   to different br
+000179d0: 616e 6368 206c 656e 6774 6820 6573 7469  anch length esti
+000179e0: 6d61 7469 6f6e 2e20 4e6f 7465 2074 6861  mation. Note tha
+000179f0: 7420 7468 6520 6661 6d69 6c69 6573 2077  t the families w
+00017a00: 6520 7573 6564 2077 6572 6520 6f6e 6c79  e used were only
+00017a10: 2074 776f 2067 6c6f 6261 6c20 4d52 4248   two global MRBH
+00017a20: 2066 616d 696c 6965 7320 666f 7220 7468   families for th
+00017a30: 6520 7075 7270 6f73 6520 6f66 2069 6c6c  e purpose of ill
+00017a40: 7573 7472 6174 696f 6e2e 2054 6f20 6163  ustration. To ac
+00017a50: 7175 6972 6520 616e 2061 6363 7572 6174  quire an accurat
+00017a60: 6520 7072 6f66 696c 6520 6f66 2074 6865  e profile of the
+00017a70: 2073 7562 7374 6974 7574 696f 6e20 7261   substitution ra
+00017a80: 7465 2076 6172 6961 7469 6f6e 2c20 6f72  te variation, or
+00017a90: 7468 6f6c 6f67 7565 7320 6174 2074 6865  thologues at the
+00017aa0: 2077 686f 6c65 2067 656e 6f6d 6520 7363   whole genome sc
+00017ab0: 616c 6520 7368 6f75 6c64 2062 6520 7573  ale should be us
+00017ac0: 6564 2e0a 0a23 2320 4369 7461 7469 6f6e  ed...## Citation
+00017ad0: 0a20 0a50 6c65 6173 6520 6369 7465 2075  . .Please cite u
+00017ae0: 7320 6174 2068 7474 7073 3a2f 2f64 6f69  s at https://doi
+00017af0: 2e6f 7267 2f31 302e 3130 3037 2f39 3738  .org/10.1007/978
+00017b00: 2d31 2d30 3731 362d 3235 3631 2d33 5f31  -1-0716-2561-3_1
+00017b10: 2061 6e64 2068 7474 7073 3a2f 2f64 6f69   and https://doi
+00017b20: 2e6f 7267 2f31 302e 3130 3933 2f62 696f  .org/10.1093/bio
+00017b30: 696e 666f 726d 6174 6963 732f 6274 6165  informatics/btae
+00017b40: 3237 322e 0a0a 6060 600a 4865 6e67 6368  272...```.Hengch
+00017b50: 6920 4368 656e 2c20 4172 7468 7572 205a  i Chen, Arthur Z
+00017b60: 7761 656e 6570 6f65 6c20 2832 3032 3329  waenepoel (2023)
+00017b70: 2e20 496e 6665 7265 6e63 6520 6f66 2041  . Inference of A
+00017b80: 6e63 6965 6e74 2050 6f6c 7970 6c6f 6964  ncient Polyploid
+00017b90: 7920 6672 6f6d 2047 656e 6f6d 6963 2044  y from Genomic D
+00017ba0: 6174 612e 2049 6e3a 2056 616e 2064 6520  ata. In: Van de 
+00017bb0: 5065 6572 2c20 592e 2028 6564 7329 2050  Peer, Y. (eds) P
+00017bc0: 6f6c 7970 6c6f 6964 792e 204d 6574 686f  olyploidy. Metho
+00017bd0: 6473 2069 6e20 4d6f 6c65 6375 6c61 7220  ds in Molecular 
+00017be0: 4269 6f6c 6f67 792c 2076 6f6c 2032 3534  Biology, vol 254
+00017bf0: 352e 2048 756d 616e 612c 204e 6577 2059  5. Humana, New Y
+00017c00: 6f72 6b2c 204e 592e 2068 7474 7073 3a2f  ork, NY. https:/
+00017c10: 2f64 6f69 2e6f 7267 2f31 302e 3130 3037  /doi.org/10.1007
+00017c20: 2f39 3738 2d31 2d30 3731 362d 3235 3631  /978-1-0716-2561
+00017c30: 2d33 5f31 0a48 656e 6763 6869 2043 6865  -3_1.Hengchi Che
+00017c40: 6e2c 2041 7274 6875 7220 5a77 6165 6e65  n, Arthur Zwaene
+00017c50: 706f 656c 2c20 5976 6573 2056 616e 2064  poel, Yves Van d
+00017c60: 6520 5065 6572 2028 3230 3234 292e 2077  e Peer (2024). w
+00017c70: 6764 2076 323a 2061 2073 7569 7465 206f  gd v2: a suite o
+00017c80: 6620 746f 6f6c 7320 746f 2075 6e63 6f76  f tools to uncov
+00017c90: 6572 2061 6e64 2064 6174 6520 616e 6369  er and date anci
+00017ca0: 656e 7420 706f 6c79 706c 6f69 6479 2061  ent polyploidy a
+00017cb0: 6e64 2077 686f 6c65 2d67 656e 6f6d 6520  nd whole-genome 
+00017cc0: 6475 706c 6963 6174 696f 6e2e 2042 696f  duplication. Bio
+00017cd0: 696e 666f 726d 6174 6963 732c 2068 7474  informatics, htt
+00017ce0: 7073 3a2f 2f64 6f69 2e6f 7267 2f31 302e  ps://doi.org/10.
+00017cf0: 3130 3933 2f62 696f 696e 666f 726d 6174  1093/bioinformat
+00017d00: 6963 732f 6274 6165 3237 320a 6060 600a  ics/btae272.```.
+00017d10: 0a46 6f72 2063 6974 6174 696f 6e20 6f66  .For citation of
+00017d20: 2074 6865 2074 6f6f 6c73 2075 7365 6420   the tools used 
+00017d30: 696e 2077 6764 2c20 706c 6561 7365 2063  in wgd, please c
+00017d40: 6f6e 7375 6c74 2074 6865 2064 6f63 756d  onsult the docum
+00017d50: 656e 7461 7469 6f6e 2061 740a 6874 7470  entation at.http
+00017d60: 733a 2f2f 7767 6476 322e 7265 6164 7468  s://wgdv2.readth
+00017d70: 6564 6f63 732e 696f 2f65 6e2f 6c61 7465  edocs.io/en/late
+00017d80: 7374 2f63 6974 6174 696f 6e2e 6874 6d6c  st/citation.html
+00017d90: 2e0a 0a                                  ...
```

### Comparing `wgd-2.0.34/wgd.egg-info/requires.txt` & `wgd-2.0.35/wgd.egg-info/requires.txt`

 * *Files identical despite different names*

